version 1.0

task Anvigengenelevelstatsdatabases {
  input {
    String? contigs_db
    String? profile_db
    String? collection_name
    String? bin_id
    File? bin_ids_file
    Boolean? zeros_are_outliers
    Int? outliers_threshold
    Boolean? just_do_it
    Boolean? in_seq_stats
    String genes_db
  }
  command <<<
    anvi_gen_gene_level_stats_databases \
      ~{genes_db} \
      ~{if defined(contigs_db) then ("--contigs-db " +  '"' + contigs_db + '"') else ""} \
      ~{if defined(profile_db) then ("--profile-db " +  '"' + profile_db + '"') else ""} \
      ~{if defined(collection_name) then ("--collection-name " +  '"' + collection_name + '"') else ""} \
      ~{if defined(bin_id) then ("--bin-id " +  '"' + bin_id + '"') else ""} \
      ~{if defined(bin_ids_file) then ("--bin-ids-file " +  '"' + bin_ids_file + '"') else ""} \
      ~{if (zeros_are_outliers) then "--zeros-are-outliers" else ""} \
      ~{if defined(outliers_threshold) then ("--outliers-threshold " +  '"' + outliers_threshold + '"') else ""} \
      ~{if (just_do_it) then "--just-do-it" else ""} \
      ~{if (in_seq_stats) then "--inseq-stats" else ""}
  >>>
  runtime {
    docker: "quay.io/biocontainers/anvio-minimal:7--py_0"
  }
  parameter_meta {
    contigs_db: "Anvi'o contigs database generated by 'anvi-gen-\\ncontigs-database' (default: None)"
    profile_db: "Anvi'o profile database (default: None)"
    collection_name: "Collection name. (default: None)"
    bin_id: "Bin name you are interested in. (default: None)"
    bin_ids_file: "Text file for bins (each line should be a unique bin\\nid). (default: None)"
    zeros_are_outliers: "If you want all zero coverage positions to be treated\\nlike outliers then use this flag. The reason to treat\\nzero coverage as outliers is because when mapping\\nreads to a reference we could get many zero positions\\ndue to accessory genes. These positions then skew the\\naverage values that we compute. (default: False)"
    outliers_threshold: "Threshold to use for the outlier detection. The\\ndefault value is '1.5'. Absolute deviation around the\\nmedian is used. To read more about the method please\\nrefer to: 'How to Detect and Handle Outliers' by Boris\\nIglewicz and David Hoaglin\\n(doi:10.1016/j.jesp.2013.03.013)."
    just_do_it: "Don't bother me with questions or warnings, just do\\nit. (default: False)"
    in_seq_stats: "Provide if working with INSeq/Tn-Seq genomic data.\\nWith this, all gene level coverage stats will be\\ncalculated using INSeq/Tn-Seq statistical methods.\\n(default: False)"
    genes_db: "üç∫ More on `anvi-gen-gene-level-stats-databases`:"
  }
  output {
    File out_stdout = stdout()
  }
}