from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Int, Boolean, String

Gatk_To_Gvcf_V0_1_0 = CommandToolBuilder(tool="gatk_to_gvcf", base_command=["gatk_to_gvcf"], inputs=[ToolInput(tag="in_chrom_depth_file", input_type=File(optional=True), prefix="--chrom-depth-file", doc=InputDocumentation(doc="Read mean depth for each chromosome from file,\nand use these values for maximum site depth\nfilteration. File should contain one line per\nchromosome, where each line begins with:\n'chrom_name<TAB>depth' (default: no chrom depth\nfiltration)")), ToolInput(tag="in_max_depth_factor", input_type=Int(optional=True), prefix="--max-depth-factor", doc=InputDocumentation(doc="(=3.0) If a chrom depth file is supplied then loci\nwith depth exceeding the mean chrom depth times\nthis value are filtered")), ToolInput(tag="in_min_gq_x", input_type=Int(optional=True), prefix="--min-gqx", doc=InputDocumentation(doc="(=20.0)         Minimum locus GQX")), ToolInput(tag="in_min_mq", input_type=Int(optional=True), prefix="--min-mq", doc=InputDocumentation(doc="(=20.0)          Minimum site MQ")), ToolInput(tag="in_min_qd", input_type=Int(optional=True), prefix="--min-qd", doc=InputDocumentation(doc="(=3.73)          Minimum locus QD")), ToolInput(tag="in_max_site_fs", input_type=Int(optional=True), prefix="--max-site-fs", doc=InputDocumentation(doc="(=60.0)     Maximum site FS")), ToolInput(tag="in_max_hap_score", input_type=Int(optional=True), prefix="--max-hapscore", doc=InputDocumentation(doc="(=13.0)    Maximum site HaplotypeScore")), ToolInput(tag="in_min_m_qrs", input_type=Int(optional=True), prefix="--min-mqrs", doc=InputDocumentation(doc="(=-12.5)       Minimum site MQRankSum")), ToolInput(tag="in_min_site_r_prs", input_type=Int(optional=True), prefix="--min-site-rprs", doc=InputDocumentation(doc="(=-2.386) Minimum site ReadPosRankSum")), ToolInput(tag="in_max_in_del_fs", input_type=Int(optional=True), prefix="--max-indel-fs", doc=InputDocumentation(doc="(=200.0)   Maximum indel FS")), ToolInput(tag="in_min_in_del_r_prs", input_type=Int(optional=True), prefix="--min-indel-rprs", doc=InputDocumentation(doc="(=-20.0) Minimum indel ReadPosRankSum")), ToolInput(tag="in_no_default_filters", input_type=Boolean(optional=True), prefix="--no-default-filters", doc=InputDocumentation(doc="Clear all default filters. Any individual\nfilter threshold changes above will still be in\neffect")), ToolInput(tag="in_min_block_able_non_ref", input_type=Int(optional=True), prefix="--min-blockable-nonref", doc=InputDocumentation(doc="(=0.2) If AD present, only compress non-variant\nsite if 1-AD[0]/DP < value")), ToolInput(tag="in_skip_header", input_type=Boolean(optional=True), prefix="--skip-header", doc=InputDocumentation(doc="Write gVCF output without header")), ToolInput(tag="in_block_range_factor", input_type=Int(optional=True), prefix="--block-range-factor", doc=InputDocumentation(doc="(=0.3)       Non-variant blocks are restricted to\nrange [x,y], y <= max(x+3,x*(1+block-ra\nnge-factor))")), ToolInput(tag="in_block_label", input_type=Int(optional=True), prefix="--block-label", doc=InputDocumentation(doc="(=BLOCKAVG_min30p3a)\nVCF INFO key used to annotate\ncompressed non-variant blocks")), ToolInput(tag="in_block_stats", input_type=String(optional=True), prefix="--block-stats", doc=InputDocumentation(doc="Write non-variant block stats to the")), ToolInput(tag="in_file", input_type=File(), position=0, doc=InputDocumentation(doc="--no-block-compression                Turn off block compression"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Gatk_To_Gvcf_V0_1_0().translate("wdl", allow_empty_container=True)

