from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import String, File, Directory, Boolean

Metaxa2_V0_1_0 = CommandToolBuilder(tool="metaxa2", base_command=["metaxa2"], inputs=[ToolInput(tag="in_dna_fasta_fastq", input_type=String(optional=True), prefix="-i", doc=InputDocumentation(doc=": DNA FASTA or FASTQ input file to investigate")), ToolInput(tag="in_base_names_output", input_type=File(optional=True), prefix="-o", doc=InputDocumentation(doc=": Base for the names of output file(s)")), ToolInput(tag="in_dna_fastq_first", input_type=String(optional=True), prefix="-1", doc=InputDocumentation(doc=": DNA FASTQ input file containing the first reads in the read pairs to investigate")), ToolInput(tag="in_dna_fastq_second", input_type=String(optional=True), prefix="-2", doc=InputDocumentation(doc=": DNA FASTQ input file containing the second reads in the pairs to investigate")), ToolInput(tag="in_f", input_type=String(optional=True), prefix="-f", doc=InputDocumentation(doc=": Specifies the format of the input file, default = auto")), ToolInput(tag="in_specifies_format_f", input_type=String(optional=True), prefix="-z", doc=InputDocumentation(doc=": Specifies the format of the input file, default = f (off)")), ToolInput(tag="in_specifies_barcoding_gene", input_type=String(optional=True), prefix="-g", doc=InputDocumentation(doc=": Specifies the barcoding gene Metaxa should look for, default is ssu")), ToolInput(tag="in_pair_file", input_type=String(optional=True), prefix="--pairfile", doc=InputDocumentation(doc=": DNA FASTQ file containing the pairs to the sequences in the input file")), ToolInput(tag="in_format", input_type=String(optional=True), prefix="--format", doc=InputDocumentation(doc=": Specifies the format of the input file, default = auto")), ToolInput(tag="in_mode", input_type=String(optional=True), prefix="--mode", doc=InputDocumentation(doc=": Controls the Metaxa2 operating mode, default = metagenome")), ToolInput(tag="in_run_only_extraction_part", input_type=String(optional=True), prefix="-x", doc=InputDocumentation(doc=": Run only the extraction part of Metaxa2, without classification, off (F) by default")), ToolInput(tag="in_run_only_classification_part", input_type=String(optional=True), prefix="-c", doc=InputDocumentation(doc=": Run only the classification part of Metaxa2, without prior extraction, off (F) by default")), ToolInput(tag="in_path_directory_hmmprofile", input_type=String(optional=True), prefix="-p", doc=InputDocumentation(doc=": A path to a directory of HMM-profile collections representing rRNA conserved regions, default is in the same directory as metaxa itself")), ToolInput(tag="in_blast_databased_used", input_type=String(optional=True), prefix="-d", doc=InputDocumentation(doc=": The BLAST databased used for classification, default is in the same directory as metaxa itself")), ToolInput(tag="in_hmms_can", input_type=File(optional=True), prefix="--hmmscan", doc=InputDocumentation(doc=": If the hmmscan has already been performed, this option can be used as the base for the hmmscan output files, and the hmmscan step will be skipped. Overrides the -o option, while a DNA FASTA file must still be supplied!")), ToolInput(tag="in_date", input_type=Directory(optional=True), prefix="--date", doc=InputDocumentation(doc=": Adds a date and time stamp to the output directory, off (F) by default")), ToolInput(tag="in_plus", input_type=String(optional=True), prefix="--plus", doc=InputDocumentation(doc=": Runs blast search through blast+ instead of the legacy blastall engine, off (F) by default")), ToolInput(tag="in_u_search", input_type=String(optional=True), prefix="--usearch", doc=InputDocumentation(doc=": Runs usearch instead of blast, specify version, off (0) by default")), ToolInput(tag="in_u_search_bin", input_type=String(optional=True), prefix="--usearch_bin", doc=InputDocumentation(doc=": Specifies the location of the Usearch binary to be used, default is 'usearch' only")), ToolInput(tag="in_u_blast", input_type=String(optional=True), prefix="--ublast", doc=InputDocumentation(doc=": Runs the Ublast algorithm instead of Usearch algorithm, default is on (T)")), ToolInput(tag="in_reset", input_type=String(optional=True), prefix="--reset", doc=InputDocumentation(doc=": Rebuilds the HMM database. Useful if HMMER has been updated or searches seem to fail mysteriously, off (F) by default")), ToolInput(tag="in_minimum_quality_value", input_type=String(optional=True), prefix="-q", doc=InputDocumentation(doc=": Minimum quality value for basecalling, default = 20")), ToolInput(tag="in_quality_filter", input_type=String(optional=True), prefix="--quality_filter", doc=InputDocumentation(doc=": Filter out low-quality reads (below specified -q value), off (F) by default")), ToolInput(tag="in_quality_trim", input_type=String(optional=True), prefix="--quality_trim", doc=InputDocumentation(doc=": Trim away ends of low quality (below -q value), off (F) by default")), ToolInput(tag="in_quality_percent", input_type=String(optional=True), prefix="--quality_percent", doc=InputDocumentation(doc=": Percentage of low-quality (below -q value) accepted before filtering/trimming, default=10")), ToolInput(tag="in_ignore_paired_read", input_type=String(optional=True), prefix="--ignore_paired_read", doc=InputDocumentation(doc=": Do not discard the entire pair if only one of the reads is of bad quality, on (T) by default")), ToolInput(tag="in_distance", input_type=String(optional=True), prefix="--distance", doc=InputDocumentation(doc=": Specifies the distance between the sequence pairs, default = 150")), ToolInput(tag="in_profile_set_use", input_type=String(optional=True), prefix="-t", doc=InputDocumentation(doc=": Profile set to use for the search (comma-separated), default is all")), ToolInput(tag="in_domain_evalue_cutoff", input_type=String(optional=True), prefix="-E", doc=InputDocumentation(doc=": Domain E-value cutoff for a sequence to be included in the output, default = 1")), ToolInput(tag="in_domain_score_cutoff", input_type=String(optional=True), prefix="-S", doc=InputDocumentation(doc=": Domain score cutoff for a sequence to be included in the output, default = 12")), ToolInput(tag="in_minimal_number_match", input_type=String(optional=True), prefix="-N", doc=InputDocumentation(doc=": The minimal number of domains that must match a sequence before it is included, default = 2")), ToolInput(tag="in_number_sequence_matches", input_type=String(optional=True), prefix="-M", doc=InputDocumentation(doc=": Number of sequence matches to consider for classification, default = 5")), ToolInput(tag="in_reliability_cutoff_taxonomic", input_type=String(optional=True), prefix="-R", doc=InputDocumentation(doc=": Reliability cutoff for taxonomic classification, default = 75")), ToolInput(tag="in_sets_cutoff_classified", input_type=String(optional=True), prefix="-T", doc=InputDocumentation(doc=": Sets the percent identity cutoff to be classified at a certain taxonomic level\nBy default, these values are specified by the database used.\nOrder of the values is:         Kingdom/Domain,Phylum,Class,Order,Family,Genus,Species\nDefault values for the SSU are: 0,60,70,75,85,90,97")), ToolInput(tag="in_number_prediction_given", input_type=String(optional=True), prefix="-H", doc=InputDocumentation(doc=": The number of points that the Metaxa Extractor prediction is given, default is the same as the number of sequences (-M option)")), ToolInput(tag="in_selection_priority", input_type=String(optional=True), prefix="--selection_priority", doc=InputDocumentation(doc=": Selects what will be of highest priority when determining the origin of the sequence, default is score")), ToolInput(tag="in_scoring_model", input_type=String(optional=True), prefix="--scoring_model", doc=InputDocumentation(doc=": Selects the scoring model to be used for classification, select 'old' to use the pre 2.2 scoring model, default = new")), ToolInput(tag="in_search_eval", input_type=String(optional=True), prefix="--search_eval", doc=InputDocumentation(doc=": The E-value cutoff used in the HMMER search, high numbers may slow down the process, cannot be used with the --search_score option, default is to use score cutoff, not E-value")), ToolInput(tag="in_search_score", input_type=String(optional=True), prefix="--search_score", doc=InputDocumentation(doc=": The score cutoff used in the HMMER search, low numbers may slow down the process, cannot be used with the --search_eval option, default = 0")), ToolInput(tag="in_blast_eval", input_type=String(optional=True), prefix="--blast_eval", doc=InputDocumentation(doc=": The E-value cutoff used in the BLAST search, high numbers may slow down the process, cannot be used with the --blast_score option, default is 1e-5")), ToolInput(tag="in_blast_score", input_type=String(optional=True), prefix="--blast_score", doc=InputDocumentation(doc=": The score cutoff used in the BLAST search, low numbers may slow down the process, cannot be used with the --blast_eval option, default is to used E-value cutoff, not score")), ToolInput(tag="in_blast_word_size", input_type=String(optional=True), prefix="--blast_wordsize", doc=InputDocumentation(doc=": The word-size used for the BLAST-based classification, default is 14")), ToolInput(tag="in_allow_single_domain", input_type=String(optional=True), prefix="--allow_single_domain", doc=InputDocumentation(doc=": Allow inclusion of sequences that only find a single domain, given that they satisfy the given E-value and score thresholds, on with parameters 1e-10,0 by default")), ToolInput(tag="in_allow_reorder", input_type=String(optional=True), prefix="--allow_reorder", doc=InputDocumentation(doc=": Allows profiles to be in the wrong order on extracted sequences, on (T) by default")), ToolInput(tag="in_complement", input_type=String(optional=True), prefix="--complement", doc=InputDocumentation(doc=": Checks both DNA strands against the database, creating reverse complements, on (T) by default")), ToolInput(tag="in_cpu", input_type=String(optional=True), prefix="--cpu", doc=InputDocumentation(doc=": The number of CPU threads to use, default is 1")), ToolInput(tag="in_multi_thread", input_type=String(optional=True), prefix="--multi_thread", doc=InputDocumentation(doc=": Multi-thread the HMMER-search, on (T) if number of CPUs (--cpu option > 1), else off (F) by default")), ToolInput(tag="in_heuristics", input_type=String(optional=True), prefix="--heuristics", doc=InputDocumentation(doc=": Selects whether to use HMMER's heuristic filtering, on (T) by default")), ToolInput(tag="in_mega_blast", input_type=String(optional=True), prefix="--megablast", doc=InputDocumentation(doc=": Uses megablast for classification for better speed but less accuracy, off (F) by default")), ToolInput(tag="in_reference", input_type=String(optional=True), prefix="--reference", doc=InputDocumentation(doc=": A file in FASTA format containing reference sequences to be sent to a separate file in the analysis, default is blank (unused)")), ToolInput(tag="in_ref_identity", input_type=String(optional=True), prefix="--ref_identity", doc=InputDocumentation(doc=": The sequence identity cutoff to be considered a sequence to be derived from a reference entry, default = 99")), ToolInput(tag="in_summary", input_type=String(optional=True), prefix="--summary", doc=InputDocumentation(doc=": Summary of results output, on (T) by default")), ToolInput(tag="in_graphical", input_type=String(optional=True), prefix="--graphical", doc=InputDocumentation(doc=": 'Graphical' output, on (T) by default")), ToolInput(tag="in_fast_a", input_type=String(optional=True), prefix="--fasta", doc=InputDocumentation(doc=": FASTA-format output of extracted rRNA sequences, on (T) by default")), ToolInput(tag="in_split_pairs", input_type=String(optional=True), prefix="--split_pairs", doc=InputDocumentation(doc=": Outputs the two read pairs separately instead of as a joint rRNA sequence, off (F) by default")), ToolInput(tag="in_table", input_type=String(optional=True), prefix="--table", doc=InputDocumentation(doc=": Table format output of sequences containing probable rRNAs, off (F) by default")), ToolInput(tag="in_taxonomy", input_type=String(optional=True), prefix="--taxonomy", doc=InputDocumentation(doc=": Table format output of probable taxonomic origin for sequences, on (T) by default")), ToolInput(tag="in_rel_tax", input_type=String(optional=True), prefix="--reltax", doc=InputDocumentation(doc=": Output of probable taxonomic origin for sequences with reliability scores at each rank, off (F) by default")), ToolInput(tag="in_tax_level", input_type=String(optional=True), prefix="--taxlevel", doc=InputDocumentation(doc=": Force Metaxa to classify sequences at a certain taxonomy level, regardless of reliability score, off (0) by default")), ToolInput(tag="in_not_found", input_type=String(optional=True), prefix="--not_found", doc=InputDocumentation(doc=": Saves a list of non-found entries, off (F) by default")), ToolInput(tag="in_align", input_type=String(optional=True), prefix="--align", doc=InputDocumentation(doc=": Outputs alignments of BLAST matches to query in all, uncertain or no cases, requires MAFFT to be installed, default is 'none'")), ToolInput(tag="in_truncate", input_type=String(optional=True), prefix="--truncate", doc=InputDocumentation(doc=": Truncates the FASTA output to only contain the putative rRNA sequence found, on (T) by default")), ToolInput(tag="in_guess_species", input_type=String(optional=True), prefix="--guess_species", doc=InputDocumentation(doc=": Writes a species guess (which can be pretty far off) to the FASTA definition line, off (F) by default (depreciated option, use --taxonomy instead)")), ToolInput(tag="in_silent", input_type=String(optional=True), prefix="--silent", doc=InputDocumentation(doc=": Supresses printing progress info to stderr, off (F) by default")), ToolInput(tag="in_graph_scale", input_type=String(optional=True), prefix="--graph_scale", doc=InputDocumentation(doc=": Sets the scale of the graph output, if value is zero, a percentage view is shown, default = 0")), ToolInput(tag="in_save_raw", input_type=String(optional=True), prefix="--save_raw", doc=InputDocumentation(doc=": saves all raw data for searches etc. instead of removing it on finish, off (F) by default")), ToolInput(tag="in_help", input_type=Boolean(optional=True), prefix="--help", doc=InputDocumentation(doc=": displays this help message")), ToolInput(tag="in_bugs", input_type=Boolean(optional=True), prefix="--bugs", doc=InputDocumentation(doc=": displays the bug fixes and known bugs in this version of Metaxa")), ToolInput(tag="in_license", input_type=Boolean(optional=True), prefix="--license", doc=InputDocumentation(doc=": displays licensing information"))], outputs=[ToolOutput(tag="out_base_names_output", output_type=File(optional=True), selector=InputSelector(input_to_select="in_base_names_output", type_hint=File()), doc=OutputDocumentation(doc=": Base for the names of output file(s)")), ToolOutput(tag="out_hmms_can", output_type=File(optional=True), selector=InputSelector(input_to_select="in_hmms_can", type_hint=File()), doc=OutputDocumentation(doc=": If the hmmscan has already been performed, this option can be used as the base for the hmmscan output files, and the hmmscan step will be skipped. Overrides the -o option, while a DNA FASTA file must still be supplied!")), ToolOutput(tag="out_date", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_date", type_hint=File()), doc=OutputDocumentation(doc=": Adds a date and time stamp to the output directory, off (F) by default"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Metaxa2_V0_1_0().translate("wdl", allow_empty_container=True)

