from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, File

Bsmap_V0_1_0 = CommandToolBuilder(tool="bsmap", base_command=["bsmap"], inputs=[ToolInput(tag="in_str_query_file", input_type=Boolean(optional=True), prefix="-a", doc=InputDocumentation(doc="<str>   query a file, FASTA/FASTQ/BAM format")), ToolInput(tag="in_str_reference_sequences", input_type=Boolean(optional=True), prefix="-d", doc=InputDocumentation(doc="<str>   reference sequences file, FASTA format")), ToolInput(tag="in_str_file_format", input_type=File(optional=True), prefix="-o", doc=InputDocumentation(doc="<str>   output alignment file, BSP/SAM/BAM format, if omitted, the output will be written to STDOUT in SAM format.")), ToolInput(tag="in_int_seed_size", input_type=Boolean(optional=True), prefix="-s", doc=InputDocumentation(doc="<int>   seed size, default=16(WGBS mode), 12(RRBS mode). min=8, max=16.")), ToolInput(tag="in_float_value_is", input_type=Boolean(optional=True), prefix="-v", doc=InputDocumentation(doc="<float> if this value is between 0 and 1, it's interpreted as the mismatch rate w.r.t to the read length.\notherwise it's interpreted as the maximum number of mismatches allowed on a read, <=15.\nexample: -v 5 (max #mismatches = 5), -v 0.1 (max #mismatches = read_length * 10%)\ndefault=0.08.")), ToolInput(tag="in_int_gap_size", input_type=Boolean(optional=True), prefix="-g", doc=InputDocumentation(doc="<int>   gap size, BSMAP only allow 1 continuous gap (insert or deletion) with up to 3 nucleotides\ndefault=0")), ToolInput(tag="in_int_maximum_number", input_type=Boolean(optional=True), prefix="-w", doc=InputDocumentation(doc="<int>   maximum number of equal best hits to count, <=1000")), ToolInput(tag="in_using_mapping_approach", input_type=Boolean(optional=True), prefix="-3", doc=InputDocumentation(doc="using 3-nucleotide mapping approach")), ToolInput(tag="in_int_start_read", input_type=Boolean(optional=True), prefix="-B", doc=InputDocumentation(doc="<int>   start from the Nth read or read pair, default: 1")), ToolInput(tag="in_int_end_nth", input_type=Boolean(optional=True), prefix="-E", doc=InputDocumentation(doc="<int>   end at the Nth read or read pair, default: 4,294,967,295")), ToolInput(tag="in_int_index_interval", input_type=Boolean(optional=True), prefix="-I", doc=InputDocumentation(doc="<int>   index interval, default=4")), ToolInput(tag="in_float_set_ratio", input_type=Boolean(optional=True), prefix="-k", doc=InputDocumentation(doc="<float> set the cut-off ratio for over-represented kmers, default=5e-07\nexample: -k 1e-6 means the top 0.0001% over-represented kmer will be skipped in alignment")), ToolInput(tag="in_int_number_use", input_type=Boolean(optional=True), prefix="-p", doc=InputDocumentation(doc="<int>   number of processors to use, default=8")), ToolInput(tag="in_activating_rrbs_mode", input_type=Boolean(optional=True), prefix="-D", doc=InputDocumentation(doc="<str>   activating RRBS mapping mode and set restriction enzyme digestion sites.\ndigestion position marked by '-', example: -D C-CGG for MspI digestion.\ndefault: none (whole genome shotgun bisulfite mapping mode)")), ToolInput(tag="in_int_seed_random", input_type=Boolean(optional=True), prefix="-S", doc=InputDocumentation(doc="<int>   seed for random number generation used in selecting multiple hits\nother seed values generate pseudo random number based on read index number, to allow reproducible mapping results.\ndefault=0. (get seed from system clock, mapping results not resproducible.)")), ToolInput(tag="in_set_alignment_information", input_type=Boolean(optional=True), prefix="-M", doc=InputDocumentation(doc="<str>   set alignment information for the additional nucleotide transition.\n<str> is in the form of two different nucleotides N1N2,\nindicating N1 in the reads could be mapped to N2 in the reference sequences.\ndefault: -M TC, corresponds to C=>U(T) transition in bisulfite conversion.\nexample: -M GA could be used to detect A=>I(G) transition in RNA editing.")), ToolInput(tag="in_int_quality_threshold", input_type=Boolean(optional=True), prefix="-q", doc=InputDocumentation(doc="<int>   quality threshold in trimming, 0-40, default=0 (no trim)")), ToolInput(tag="in_int_base_quality", input_type=Boolean(optional=True), prefix="-z", doc=InputDocumentation(doc="<int>   base quality, default=33 [Illumina is using 64, Sanger Institute is using 33]")), ToolInput(tag="in_int_filter_lowquality", input_type=Boolean(optional=True), prefix="-f", doc=InputDocumentation(doc="<int>   filter low-quality reads containing >n Ns, default=5")), ToolInput(tag="in_end_sequence_default", input_type=Boolean(optional=True), prefix="-A", doc=InputDocumentation(doc="<str>   3-end adapter sequence, default: none (no trim)")), ToolInput(tag="in_int_map_nucleotides", input_type=Boolean(optional=True), prefix="-L", doc=InputDocumentation(doc="<int>   map the first N nucleotides of the read, default:160 (map the whole read).")), ToolInput(tag="in_how_report_repeat", input_type=Boolean(optional=True), prefix="-r", doc=InputDocumentation(doc="[0,1,2] how to report repeat hits, 0=none(unique hit/pair); 1=random one; 2=all(slow), default:1.")), ToolInput(tag="in_print_corresponding_reference", input_type=Boolean(optional=True), prefix="-R", doc=InputDocumentation(doc="print corresponding reference sequences in SAM output, default=off")), ToolInput(tag="in_report_unmapped_reads", input_type=Boolean(optional=True), prefix="-u", doc=InputDocumentation(doc="report unmapped reads, default=off")), ToolInput(tag="in_print_header_information", input_type=Boolean(optional=True), prefix="-H", doc=InputDocumentation(doc="do not print header information in SAM format output")), ToolInput(tag="in_verbose_level_message", input_type=Boolean(optional=True), prefix="-V", doc=InputDocumentation(doc="[0,1,2] verbose level: 0=no message displayed (quiet mode);\n1=major message (default); 2=detailed message.")), ToolInput(tag="in_str_query_b", input_type=Boolean(optional=True), prefix="-b", doc=InputDocumentation(doc="<str>   query b file")), ToolInput(tag="in_int_minimal_insert", input_type=Boolean(optional=True), prefix="-m", doc=InputDocumentation(doc="<int>   minimal insert size allowed, default=28")), ToolInput(tag="in_int_maximal_insert", input_type=Boolean(optional=True), prefix="-x", doc=InputDocumentation(doc="<int>   maximal insert size allowed, default=1000"))], outputs=[ToolOutput(tag="out_str_file_format", output_type=File(optional=True), selector=InputSelector(input_to_select="in_str_file_format", type_hint=File()), doc=OutputDocumentation(doc="<str>   output alignment file, BSP/SAM/BAM format, if omitted, the output will be written to STDOUT in SAM format."))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Bsmap_V0_1_0().translate("wdl", allow_empty_container=True)

