from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, String, Boolean, Int, Float

Hisatgenotype_Locus_Py_V0_1_0 = CommandToolBuilder(tool="hisatgenotype_locus.py", base_command=["hisatgenotype_locus.py"], inputs=[ToolInput(tag="in_base_fname", input_type=File(optional=True), prefix="--base-fname", doc=InputDocumentation(doc="base filename for backbone sequence, variants, and\nlinking info (default: hla)")), ToolInput(tag="in_locus_list", input_type=String(optional=True), prefix="--locus-list", doc=InputDocumentation(doc="A comma-separated list of genes (default: empty, all\ngenes)")), ToolInput(tag="in_genotype_genome", input_type=String(optional=True), prefix="--genotype-genome", doc=InputDocumentation(doc="Base name for genotype genome, which the program will\nuse instead of region-based small indexes (default:\nempty)")), ToolInput(tag="in_fast_a", input_type=Boolean(optional=True), prefix="--fasta", doc=InputDocumentation(doc="FASTA format")), ToolInput(tag="in_filename_singleend_reads", input_type=File(optional=True), prefix="-U", doc=InputDocumentation(doc="filename for single-end reads")), ToolInput(tag="in_one", input_type=Int(optional=True), prefix="-1", doc=InputDocumentation(doc="filename for paired-end reads")), ToolInput(tag="in_two", input_type=Int(optional=True), prefix="-2", doc=InputDocumentation(doc="filename for paired-end reads")), ToolInput(tag="in_alignment", input_type=File(optional=True), prefix="--alignment", doc=InputDocumentation(doc="BAM file name")), ToolInput(tag="in_threads", input_type=Int(optional=True), prefix="--threads", doc=InputDocumentation(doc="Number of threads")), ToolInput(tag="in_no_partial", input_type=Boolean(optional=True), prefix="--no-partial", doc=InputDocumentation(doc="Include partial alleles (e.g. A_nuc.fasta)")), ToolInput(tag="in_aligner_list", input_type=Int(optional=True), prefix="--aligner-list", doc=InputDocumentation(doc="A comma-separated list of aligners such as\nhisat2.graph,hisat2.linear,bowtie2.linear (default:\nhisat2.graph)")), ToolInput(tag="in_simulate_interval", input_type=Int(optional=True), prefix="--simulate-interval", doc=InputDocumentation(doc="Reads simulated at every these base pairs (default:\n10)")), ToolInput(tag="in_read_len", input_type=Int(optional=True), prefix="--read-len", doc=InputDocumentation(doc="Length of simulated reads (default: 100)")), ToolInput(tag="in_fragment_len", input_type=Int(optional=True), prefix="--fragment-len", doc=InputDocumentation(doc="Length of fragments (default: 350)")), ToolInput(tag="in_best_alleles", input_type=Int(optional=True), prefix="--best-alleles", doc=InputDocumentation(doc="A seeding number for randomness (default: 1)")), ToolInput(tag="in_num_edit_dist", input_type=Int(optional=True), prefix="--num-editdist", doc=InputDocumentation(doc="Maximum number of mismatches per read alignment to be\nconsidered (default: 2)")), ToolInput(tag="in_per_base_error_rate", input_type=Float(optional=True), prefix="--perbase-errorrate", doc=InputDocumentation(doc="Per basepair error rate in percentage when simulating\nreads (default: 0.0)")), ToolInput(tag="in_per_base_snp_rate", input_type=Float(optional=True), prefix="--perbase-snprate", doc=InputDocumentation(doc="Per basepair SNP rate in percentage when simulating\nreads (default: 0.0)")), ToolInput(tag="in_skip_fragment_regions", input_type=Int(optional=True), prefix="--skip-fragment-regions", doc=InputDocumentation(doc="A comma-separated list of regions from which no reads\noriginate, e.g., 500-600,1200-1400 (default: None).")), ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="--verbose", doc=InputDocumentation(doc="also print some statistics to stderr")), ToolInput(tag="in_verbose_level", input_type=Int(optional=True), prefix="--verbose-level", doc=InputDocumentation(doc="also print some statistics to stderr (default: 0)")), ToolInput(tag="in_debug", input_type=Int(optional=True), prefix="--debug", doc=InputDocumentation(doc="e.g., test_id:10,read_id:10000,basic_test")), ToolInput(tag="in_assembly_base", input_type=File(optional=True), prefix="--assembly-base", doc=InputDocumentation(doc="base file name (default: assembly_graph)")), ToolInput(tag="in_assembly", input_type=Boolean(optional=True), prefix="--assembly", doc=InputDocumentation(doc="Perform assembly")), ToolInput(tag="in_no_error_correction", input_type=Boolean(optional=True), prefix="--no-error-correction", doc=InputDocumentation(doc="Correct sequencing errors")), ToolInput(tag="in_keep_alignment", input_type=Boolean(optional=True), prefix="--keep-alignment", doc=InputDocumentation(doc="Keep alignment file")), ToolInput(tag="in_only_locus_list", input_type=String(optional=True), prefix="--only-locus-list", doc=InputDocumentation(doc="A comma-separated list of genes (default: empty, all\ngenes)")), ToolInput(tag="in_discordant", input_type=Boolean(optional=True), prefix="--discordant", doc=InputDocumentation(doc="Allow discordantly mapped pairs or singletons")), ToolInput(tag="in_type_primary_exons", input_type=Boolean(optional=True), prefix="--type-primary-exons", doc=InputDocumentation(doc="Look at primary exons first")), ToolInput(tag="in_keep_low_abundance_alleles", input_type=Boolean(optional=True), prefix="--keep-low-abundance-alleles", doc=InputDocumentation(doc="Do not remove alleles with low abundance while\nperforming typing")), ToolInput(tag="in_assembly_verbose", input_type=Boolean(optional=True), prefix="--assembly-verbose", doc=InputDocumentation(doc="Output intermediate assembly information")), ToolInput(tag="in_display_alleles", input_type=String(optional=True), prefix="--display-alleles", doc=InputDocumentation(doc="A comma-separated list of alleles to display in HTML\n(default: empty)\n"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Hisatgenotype_Locus_Py_V0_1_0().translate("wdl", allow_empty_container=True)

