from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Int, File, String, Boolean, Float

Iqtree2_V0_1_0 = CommandToolBuilder(tool="iqtree2", base_command=["iqtree2"], inputs=[ToolInput(tag="in_seqtype", input_type=Int(optional=True), prefix="--seqtype", doc=InputDocumentation(doc="BIN, DNA, AA, NT2AA, CODON, MORPH (default: auto-detect)")), ToolInput(tag="in_parsrand_starting_tree", input_type=File(optional=True), prefix="-t", doc=InputDocumentation(doc="|PARS|RAND    Starting tree (default: 99 parsimony and BIONJ)")), ToolInput(tag="in_outgroup_taxon_list", input_type=String(optional=True), prefix="-o", doc=InputDocumentation(doc="[,...,TAX]     Outgroup taxon (list) for writing .treefile")), ToolInput(tag="in_prefix", input_type=String(optional=True), prefix="--prefix", doc=InputDocumentation(doc="Prefix for all output files (default: aln/partition)")), ToolInput(tag="in_seed", input_type=Int(optional=True), prefix="--seed", doc=InputDocumentation(doc="Random seed number, normally used for debugging purpose")), ToolInput(tag="in_safe", input_type=Boolean(optional=True), prefix="--safe", doc=InputDocumentation(doc="Safe likelihood kernel to avoid numerical underflow")), ToolInput(tag="in_mem", input_type=Int(optional=True), prefix="--mem", doc=InputDocumentation(doc="[G|M|%]     Maximal RAM usage in GB | MB | %")), ToolInput(tag="in_runs", input_type=Int(optional=True), prefix="--runs", doc=InputDocumentation(doc="Number of indepedent runs (default: 1)")), ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="--verbose", doc=InputDocumentation(doc="Verbose mode, printing more messages to screen")), ToolInput(tag="in_quiet", input_type=Boolean(optional=True), prefix="--quiet", doc=InputDocumentation(doc="Quiet mode, suppress printing to screen (stdout)")), ToolInput(tag="in_f_const", input_type=Int(optional=True), prefix="-fconst", doc=InputDocumentation(doc=",...,fN    Add constant patterns into alignment (N=no. states)")), ToolInput(tag="in_epsilon", input_type=Int(optional=True), prefix="--epsilon", doc=InputDocumentation(doc="Likelihood epsilon for parameter estimate (default 0.01)")), ToolInput(tag="in_auto_coresthreads_default", input_type=Int(optional=True), prefix="-T", doc=InputDocumentation(doc="|AUTO          No. cores/threads or AUTO-detect (default: 1)")), ToolInput(tag="in_threads_max", input_type=Int(optional=True), prefix="--threads-max", doc=InputDocumentation(doc="Max number of threads for -T AUTO (default: all cores)")), ToolInput(tag="in_redo", input_type=Boolean(optional=True), prefix="--redo", doc=InputDocumentation(doc="Redo both ModelFinder and tree search")), ToolInput(tag="in_redo_tree", input_type=Boolean(optional=True), prefix="--redo-tree", doc=InputDocumentation(doc="Restore ModelFinder and only redo tree search")), ToolInput(tag="in_undo", input_type=Boolean(optional=True), prefix="--undo", doc=InputDocumentation(doc="Revoke finished run, used when changing some options")), ToolInput(tag="in_cp_time", input_type=Int(optional=True), prefix="--cptime", doc=InputDocumentation(doc="Minimum checkpoint interval (default: 60 sec and adapt)")), ToolInput(tag="in_dir_nexusraxml_partition", input_type=File(optional=True), prefix="-p", doc=InputDocumentation(doc="|DIR          NEXUS/RAxML partition file or directory with alignments\nEdge-linked proportional partition model")), ToolInput(tag="in_dir_edgelinked_model", input_type=File(optional=True), prefix="-q", doc=InputDocumentation(doc="|DIR          Like -p but edge-linked equal partition model")), ToolInput(tag="in_dir_edgeunlinked_model", input_type=File(optional=True), prefix="-Q", doc=InputDocumentation(doc="|DIR          Like -p but edge-unlinked partition model")), ToolInput(tag="in_dir_p_separate", input_type=File(optional=True), prefix="-S", doc=InputDocumentation(doc="|DIR          Like -p but separate tree inference")), ToolInput(tag="in_subsample", input_type=Int(optional=True), prefix="--subsample", doc=InputDocumentation(doc="Randomly sub-sample partitions (negative for complement)")), ToolInput(tag="in_subsample_seed", input_type=Int(optional=True), prefix="--subsample-seed", doc=InputDocumentation(doc="Random number seed for --subsample")), ToolInput(tag="in_l_map", input_type=Int(optional=True), prefix="--lmap", doc=InputDocumentation(doc="Number of quartets for likelihood mapping analysis")), ToolInput(tag="in_lmc_lust", input_type=File(optional=True), prefix="--lmclust", doc=InputDocumentation(doc="NEXUS file containing clusters for likelihood mapping")), ToolInput(tag="in_quartet_lh", input_type=Boolean(optional=True), prefix="--quartetlh", doc=InputDocumentation(doc="Print quartet log-likelihoods to .quartetlh file")), ToolInput(tag="in_nin_it", input_type=Int(optional=True), prefix="--ninit", doc=InputDocumentation(doc="Number of initial parsimony trees (default: 100)")), ToolInput(tag="in_ntop", input_type=Int(optional=True), prefix="--ntop", doc=InputDocumentation(doc="Number of top initial trees (default: 20)")), ToolInput(tag="in_n_best", input_type=Int(optional=True), prefix="--nbest", doc=InputDocumentation(doc="Number of best trees retained during search (defaut: 5)")), ToolInput(tag="in_fix_number_stop", input_type=Int(optional=True), prefix="-n", doc=InputDocumentation(doc="Fix number of iterations to stop (default: OFF)")), ToolInput(tag="in_n_stop", input_type=Int(optional=True), prefix="--nstop", doc=InputDocumentation(doc="Number of unsuccessful iterations to stop (default: 100)")), ToolInput(tag="in_perturb", input_type=Int(optional=True), prefix="--perturb", doc=InputDocumentation(doc="Perturbation strength for randomized NNI (default: 0.5)")), ToolInput(tag="in_radius", input_type=Int(optional=True), prefix="--radius", doc=InputDocumentation(doc="Radius for parsimony SPR search (default: 6)")), ToolInput(tag="in_all_nni", input_type=Boolean(optional=True), prefix="--allnni", doc=InputDocumentation(doc="Perform more thorough NNI search (default: OFF)")), ToolInput(tag="in_multifurcating_topological_constraint", input_type=File(optional=True), prefix="-g", doc=InputDocumentation(doc="(Multifurcating) topological constraint tree file")), ToolInput(tag="in_fast", input_type=Boolean(optional=True), prefix="--fast", doc=InputDocumentation(doc="Fast search to resemble FastTree")), ToolInput(tag="in_poly_to_my", input_type=Boolean(optional=True), prefix="--polytomy", doc=InputDocumentation(doc="Collapse near-zero branches into polytomy")), ToolInput(tag="in_tree_fix", input_type=Boolean(optional=True), prefix="--tree-fix", doc=InputDocumentation(doc="Fix -t tree (no tree search performed)")), ToolInput(tag="in_t_reels", input_type=Boolean(optional=True), prefix="--treels", doc=InputDocumentation(doc="Write locally optimal trees into .treels file")), ToolInput(tag="in_show_lh", input_type=Boolean(optional=True), prefix="--show-lh", doc=InputDocumentation(doc="Compute tree likelihood without optimisation")), ToolInput(tag="in_terrace", input_type=Boolean(optional=True), prefix="--terrace", doc=InputDocumentation(doc="Check if the tree lies on a phylogenetic terrace")), ToolInput(tag="in_uf_boot", input_type=Int(optional=True), prefix="--ufboot", doc=InputDocumentation(doc="Replicates for ultrafast bootstrap (>=1000)")), ToolInput(tag="in_uf_jack", input_type=Int(optional=True), prefix="--ufjack", doc=InputDocumentation(doc="Replicates for ultrafast jackknife (>=1000)")), ToolInput(tag="in_jack_prop", input_type=Int(optional=True), prefix="--jack-prop", doc=InputDocumentation(doc="Subsampling proportion for jackknife (default: 0.5)")), ToolInput(tag="in_sampling", input_type=String(optional=True), prefix="--sampling", doc=InputDocumentation(doc="GENE|GENESITE resampling for partitions (default: SITE)")), ToolInput(tag="in_boot_trees", input_type=Boolean(optional=True), prefix="--boot-trees", doc=InputDocumentation(doc="Write bootstrap trees to .ufboot file (default: none)")), ToolInput(tag="in_wbt_l", input_type=Boolean(optional=True), prefix="--wbtl", doc=InputDocumentation(doc="Like --boot-trees but also writing branch lengths")), ToolInput(tag="in_nmax", input_type=Int(optional=True), prefix="--nmax", doc=InputDocumentation(doc="Maximum number of iterations (default: 1000)")), ToolInput(tag="in_nstep", input_type=Int(optional=True), prefix="--nstep", doc=InputDocumentation(doc="Iterations for UFBoot stopping rule (default: 100)")), ToolInput(tag="in_bc_or", input_type=Int(optional=True), prefix="--bcor", doc=InputDocumentation(doc="Minimum correlation coefficient (default: 0.99)")), ToolInput(tag="in_be_ps", input_type=Int(optional=True), prefix="--beps", doc=InputDocumentation(doc="RELL epsilon to break tie (default: 0.5)")), ToolInput(tag="in_bnn_i", input_type=Boolean(optional=True), prefix="--bnni", doc=InputDocumentation(doc="Optimize UFBoot trees by NNI on bootstrap alignment")), ToolInput(tag="in_boot", input_type=Int(optional=True), prefix="--boot", doc=InputDocumentation(doc="Replicates for bootstrap + ML tree + consensus tree")), ToolInput(tag="in_jack", input_type=Int(optional=True), prefix="--jack", doc=InputDocumentation(doc="Replicates for jackknife + ML tree + consensus tree")), ToolInput(tag="in_bc_on", input_type=Int(optional=True), prefix="--bcon", doc=InputDocumentation(doc="Replicates for bootstrap + consensus tree")), ToolInput(tag="in_b_only", input_type=Int(optional=True), prefix="--bonly", doc=InputDocumentation(doc="Replicates for bootstrap only")), ToolInput(tag="in_t_be", input_type=Boolean(optional=True), prefix="--tbe", doc=InputDocumentation(doc="Transfer bootstrap expectation")), ToolInput(tag="in_a_bayes", input_type=Boolean(optional=True), prefix="--abayes", doc=InputDocumentation(doc="approximate Bayes test (Anisimova et al. 2011)")), ToolInput(tag="in_lbp", input_type=Int(optional=True), prefix="--lbp", doc=InputDocumentation(doc="Replicates for fast local bootstrap probabilities")), ToolInput(tag="in_msub", input_type=String(optional=True), prefix="--msub", doc=InputDocumentation(doc="Amino-acid model source\n(nuclear, mitochondrial, chloroplast or viral)")), ToolInput(tag="in_m_freq", input_type=String(optional=True), prefix="--mfreq", doc=InputDocumentation(doc=",...      List of state frequencies")), ToolInput(tag="in_m_rate", input_type=String(optional=True), prefix="--mrate", doc=InputDocumentation(doc=",...      List of rate heterogeneity among sites\n(e.g. -mrate E,I,G,I+G,R is used for -m MF)")), ToolInput(tag="in_cm_in", input_type=Int(optional=True), prefix="--cmin", doc=InputDocumentation(doc="Min categories for FreeRate model [+R] (default: 2)")), ToolInput(tag="in_cmax", input_type=Int(optional=True), prefix="--cmax", doc=InputDocumentation(doc="Max categories for FreeRate model [+R] (default: 10)")), ToolInput(tag="in_merit", input_type=String(optional=True), prefix="--merit", doc=InputDocumentation(doc="|AICc|BIC  Akaike|Bayesian information criterion (default: BIC)")), ToolInput(tag="in_mtree", input_type=Boolean(optional=True), prefix="--mtree", doc=InputDocumentation(doc="Perform full tree search for every model")), ToolInput(tag="in_madd", input_type=String(optional=True), prefix="--madd", doc=InputDocumentation(doc=",...       List of mixture models to consider")), ToolInput(tag="in_m_def", input_type=File(optional=True), prefix="--mdef", doc=InputDocumentation(doc="Model definition NEXUS file (see Manual)")), ToolInput(tag="in_modelo_matic", input_type=Boolean(optional=True), prefix="--modelomatic", doc=InputDocumentation(doc="Find best codon/protein/DNA models (Whelan et al. 2015)")), ToolInput(tag="in_r_cluster", input_type=Int(optional=True), prefix="--rcluster", doc=InputDocumentation(doc="Percentage of partition pairs for rcluster algorithm")), ToolInput(tag="in_r_cluster_f", input_type=Int(optional=True), prefix="--rclusterf", doc=InputDocumentation(doc="Percentage of partition pairs for rclusterf algorithm")), ToolInput(tag="in_r_cluster_max", input_type=Int(optional=True), prefix="--rcluster-max", doc=InputDocumentation(doc="Max number of partition pairs (default: 10*partitions)")), ToolInput(tag="in_alpha_min", input_type=Int(optional=True), prefix="--alpha-min", doc=InputDocumentation(doc="Min Gamma shape parameter for site rates (default: 0.02)")), ToolInput(tag="in_gamma_median", input_type=Boolean(optional=True), prefix="--gamma-median", doc=InputDocumentation(doc="Median approximation for +G site rates (default: mean)")), ToolInput(tag="in_rate", input_type=Boolean(optional=True), prefix="--rate", doc=InputDocumentation(doc="Write empirical Bayesian site rates to .rate file")), ToolInput(tag="in_ml_rate", input_type=Boolean(optional=True), prefix="--mlrate", doc=InputDocumentation(doc="Write maximum likelihood site rates to .mlrate file")), ToolInput(tag="in_input_counts_file", input_type=File(optional=True), prefix="-s", doc=InputDocumentation(doc="Input counts file (see manual)")), ToolInput(tag="in_mix_opt", input_type=Boolean(optional=True), prefix="--mix-opt", doc=InputDocumentation(doc="Optimize mixture weights (default: detect)")), ToolInput(tag="in_asc_ascertainment_bias", input_type=Boolean(optional=True), prefix="-m", doc=InputDocumentation(doc="...+ASC           Ascertainment bias correction")), ToolInput(tag="in_tree_freq", input_type=File(optional=True), prefix="--tree-freq", doc=InputDocumentation(doc="Input tree to infer site frequency model")), ToolInput(tag="in_site_freq", input_type=File(optional=True), prefix="--site-freq", doc=InputDocumentation(doc="Input site frequency model file")), ToolInput(tag="in_freq_max", input_type=Boolean(optional=True), prefix="--freq-max", doc=InputDocumentation(doc="Posterior maximum instead of mean approximation")), ToolInput(tag="in_trees", input_type=File(optional=True), prefix="--trees", doc=InputDocumentation(doc="Set of trees to evaluate log-likelihoods")), ToolInput(tag="in_test", input_type=Int(optional=True), prefix="--test", doc=InputDocumentation(doc="Replicates for topology test")), ToolInput(tag="in_test_weight", input_type=Boolean(optional=True), prefix="--test-weight", doc=InputDocumentation(doc="Perform weighted KH and SH tests")), ToolInput(tag="in_test_au", input_type=Boolean(optional=True), prefix="--test-au", doc=InputDocumentation(doc="Approximately unbiased (AU) test (Shimodaira 2002)")), ToolInput(tag="in_site_lh", input_type=Boolean(optional=True), prefix="--sitelh", doc=InputDocumentation(doc="Write site log-likelihoods to .sitelh file")), ToolInput(tag="in_ancestral", input_type=Boolean(optional=True), prefix="--ancestral", doc=InputDocumentation(doc="Ancestral state reconstruction by empirical Bayes")), ToolInput(tag="in_asr_min", input_type=Int(optional=True), prefix="--asr-min", doc=InputDocumentation(doc="Min probability of ancestral state (default: equil freq)")), ToolInput(tag="in_sym_test", input_type=Boolean(optional=True), prefix="--symtest", doc=InputDocumentation(doc="Perform three tests of symmetry")), ToolInput(tag="in_sym_test_only", input_type=Boolean(optional=True), prefix="--symtest-only", doc=InputDocumentation(doc="Do --symtest then exist")), ToolInput(tag="in_sym_test_remove_bad", input_type=Boolean(optional=True), prefix="--symtest-remove-bad", doc=InputDocumentation(doc="Do --symtest and remove bad partitions")), ToolInput(tag="in_sym_test_remove_good", input_type=Boolean(optional=True), prefix="--symtest-remove-good", doc=InputDocumentation(doc="Do --symtest and remove good partitions")), ToolInput(tag="in_sym_test_type", input_type=Int(optional=True), prefix="--symtest-type", doc=InputDocumentation(doc="|INT  Use MARginal/INTernal test when removing partitions")), ToolInput(tag="in_sym_test_pval", input_type=Float(optional=True), prefix="--symtest-pval", doc=InputDocumentation(doc="P-value cutoff (default: 0.05)")), ToolInput(tag="in_sym_test_keep_zero", input_type=Boolean(optional=True), prefix="--symtest-keep-zero", doc=InputDocumentation(doc="Keep NAs in the tests")), ToolInput(tag="in_gcf", input_type=File(optional=True), prefix="--gcf", doc=InputDocumentation(doc="Set of source trees for gene concordance factor (gCF)")), ToolInput(tag="in_df_tree", input_type=Boolean(optional=True), prefix="--df-tree", doc=InputDocumentation(doc="Write discordant trees associated with gDF1")), ToolInput(tag="in_scf", input_type=Int(optional=True), prefix="--scf", doc=InputDocumentation(doc="Number of quartets for site concordance factor (sCF)")), ToolInput(tag="in_cf_verbose", input_type=Boolean(optional=True), prefix="--cf-verbose", doc=InputDocumentation(doc="Write CF per tree/locus to cf.stat_tree/_loci")), ToolInput(tag="in_cf_quartet", input_type=Boolean(optional=True), prefix="--cf-quartet", doc=InputDocumentation(doc="Write sCF for all resampled quartets to .cf.quartet")), ToolInput(tag="in_date_tip", input_type=Int(optional=True), prefix="--date-tip", doc=InputDocumentation(doc="Tip dates as a real number or YYYY-MM-DD")), ToolInput(tag="in_date_root", input_type=Int(optional=True), prefix="--date-root", doc=InputDocumentation(doc="Root date as a real number or YYYY-MM-DD")), ToolInput(tag="in_date_ci", input_type=Int(optional=True), prefix="--date-ci", doc=InputDocumentation(doc="Number of replicates to compute confidence interval")), ToolInput(tag="in_clock_sd", input_type=Int(optional=True), prefix="--clock-sd", doc=InputDocumentation(doc="Std-dev for lognormal relaxed clock (default: 0.2)")), ToolInput(tag="in_date_no_out_group", input_type=Boolean(optional=True), prefix="--date-no-outgroup", doc=InputDocumentation(doc="Exclude outgroup from time tree")), ToolInput(tag="in_date_outlier", input_type=Int(optional=True), prefix="--date-outlier", doc=InputDocumentation(doc="Z-score cutoff to remove outlier tips/nodes (e.g. 3)")), ToolInput(tag="in_date_options", input_type=Boolean(optional=True), prefix="--date-options", doc=InputDocumentation(doc="'..'  Extra options passing directly to LSD2")), ToolInput(tag="in_dating", input_type=String(optional=True), prefix="--dating", doc=InputDocumentation(doc="Dating method: LSD for least square dating (default)")), ToolInput(tag="in_sup_min", input_type=Int(optional=True), prefix="--sup-min", doc=InputDocumentation(doc="Min split support, 0.5 for majority-rule consensus\n(default: 0, extended consensus)")), ToolInput(tag="in_burnin", input_type=Int(optional=True), prefix="--burnin", doc=InputDocumentation(doc="Burnin number of trees to ignore")), ToolInput(tag="in_con_tree", input_type=Boolean(optional=True), prefix="--con-tree", doc=InputDocumentation(doc="Compute consensus tree to .contree file")), ToolInput(tag="in_con_net", input_type=Boolean(optional=True), prefix="--con-net", doc=InputDocumentation(doc="Computing consensus network to .nex file")), ToolInput(tag="in_support", input_type=File(optional=True), prefix="--support", doc=InputDocumentation(doc="Assign support values into this tree from -t trees")), ToolInput(tag="in_sup_tag", input_type=String(optional=True), prefix="--suptag", doc=InputDocumentation(doc="Node name (or ALL) to assign tree IDs where node occurs")), ToolInput(tag="in_tree_dist_all", input_type=Boolean(optional=True), prefix="--tree-dist-all", doc=InputDocumentation(doc="Compute all-to-all RF distances for -t trees")), ToolInput(tag="in_tree_dist", input_type=File(optional=True), prefix="--tree-dist", doc=InputDocumentation(doc="Compute RF distances between -t trees and this set")), ToolInput(tag="in_tree_dist_two", input_type=File(optional=True), prefix="--tree-dist2", doc=InputDocumentation(doc="Like -rf but trees can have unequal taxon sets")), ToolInput(tag="in_taxa_yuleharding_tree", input_type=Int(optional=True), prefix="-r", doc=InputDocumentation(doc="No. taxa for Yule-Harding random tree")), ToolInput(tag="in_rand", input_type=String(optional=True), prefix="--rand", doc=InputDocumentation(doc="|CAT|BAL   UNIform | CATerpillar | BALanced random tree")), ToolInput(tag="in_rlen", input_type=Int(optional=True), prefix="--rlen", doc=InputDocumentation(doc="NUM NUM   min, mean, and max random branch lengths")), ToolInput(tag="in_keep_ident", input_type=Boolean(optional=True), prefix="--keep-ident", doc=InputDocumentation(doc="Keep identical sequences (default: remove & finally add)")), ToolInput(tag="in_bl_fix", input_type=Boolean(optional=True), prefix="-blfix", doc=InputDocumentation(doc="Fix branch lengths of user tree passed via -te")), ToolInput(tag="in_bl_scale", input_type=Boolean(optional=True), prefix="-blscale", doc=InputDocumentation(doc="Scale branch lengths of user tree passed via -t")), ToolInput(tag="in_blm_in", input_type=Boolean(optional=True), prefix="-blmin", doc=InputDocumentation(doc="Min branch length for optimization (default 0.000001)")), ToolInput(tag="in_bl_max", input_type=Boolean(optional=True), prefix="-blmax", doc=InputDocumentation(doc="Max branch length for optimization (default 100)")), ToolInput(tag="in_w_slr", input_type=Boolean(optional=True), prefix="-wslr", doc=InputDocumentation(doc="Write site log-likelihoods per rate category")), ToolInput(tag="in_w_slm", input_type=Boolean(optional=True), prefix="-wslm", doc=InputDocumentation(doc="Write site log-likelihoods per mixture class")), ToolInput(tag="in_wsl_mr", input_type=Boolean(optional=True), prefix="-wslmr", doc=InputDocumentation(doc="Write site log-likelihoods per mixture+rate class")), ToolInput(tag="in_w_spr", input_type=Boolean(optional=True), prefix="-wspr", doc=InputDocumentation(doc="Write site probabilities per rate category")), ToolInput(tag="in_ws_pm", input_type=Boolean(optional=True), prefix="-wspm", doc=InputDocumentation(doc="Write site probabilities per mixture class")), ToolInput(tag="in_wsp_mr", input_type=Boolean(optional=True), prefix="-wspmr", doc=InputDocumentation(doc="Write site probabilities per mixture+rate class")), ToolInput(tag="in_part_lh", input_type=Boolean(optional=True), prefix="--partlh", doc=InputDocumentation(doc="Write partition log-likelihoods to .partlh file")), ToolInput(tag="in_no_out_files", input_type=Boolean(optional=True), prefix="--no-outfiles", doc=InputDocumentation(doc="Suppress printing output files")), ToolInput(tag="in_eigen_lib", input_type=Boolean(optional=True), prefix="--eigenlib", doc=InputDocumentation(doc="Use Eigen3 library")), ToolInput(tag="in_aln_info", input_type=Boolean(optional=True), prefix="-alninfo", doc=InputDocumentation(doc="Print alignment sites statistics to .alninfo")), ToolInput(tag="in_iq_tree", input_type=String(), position=0, doc=InputDocumentation(doc=""))], outputs=[], container="quay.io/biocontainers/iqtree:2.1.2--h56fc30b_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Iqtree2_V0_1_0().translate("wdl")

