from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, Int, File, Float, String

Skesa_V0_1_0 = CommandToolBuilder(tool="skesa", base_command=["skesa"], inputs=[ToolInput(tag="in__print_version", input_type=Boolean(optional=True), prefix="-v", doc=InputDocumentation(doc="[ --version ]              Print version")), ToolInput(tag="in_cores", input_type=Int(optional=True), prefix="--cores", doc=InputDocumentation(doc="(=0)              Number of cores to use (default all) [integer]")), ToolInput(tag="in_memory", input_type=Int(optional=True), prefix="--memory", doc=InputDocumentation(doc="(=32)            Memory available (GB, only for sorted counter)\n[integer]")), ToolInput(tag="in_hash_count", input_type=Boolean(optional=True), prefix="--hash_count", doc=InputDocumentation(doc="Use hash counter [flag]")), ToolInput(tag="in_estimated_km_ers", input_type=Int(optional=True), prefix="--estimated_kmers", doc=InputDocumentation(doc="(=100)  Estimated number of unique kmers for bloom\nfilter (M, only for hash counter) [integer]")), ToolInput(tag="in_skip_bloom_filter", input_type=Boolean(optional=True), prefix="--skip_bloom_filter", doc=InputDocumentation(doc="Don't do bloom filter; use --estimated_kmers as\nthe hash table size (only for hash counter)\n[flag]")), ToolInput(tag="in_fast_a", input_type=File(optional=True), prefix="--fasta", doc=InputDocumentation(doc="Input fasta file(s) (could be used multiple\ntimes for different runs) [string]")), ToolInput(tag="in_fast_q", input_type=File(optional=True), prefix="--fastq", doc=InputDocumentation(doc="Input fastq file(s) (could be used multiple\ntimes for different runs) [string]")), ToolInput(tag="in_use_paired_ends", input_type=Boolean(optional=True), prefix="--use_paired_ends", doc=InputDocumentation(doc="Indicates that a single (not comma separated)\nfasta/fastq file contains paired reads [flag]")), ToolInput(tag="in_contigs_out", input_type=File(optional=True), prefix="--contigs_out", doc=InputDocumentation(doc="Output file for contigs (stdout if not\nspecified) [string]")), ToolInput(tag="in_km_er", input_type=Int(optional=True), prefix="--kmer", doc=InputDocumentation(doc="(=21)              Minimal kmer length for assembly [integer]")), ToolInput(tag="in_min_count", input_type=Int(optional=True), prefix="--min_count", doc=InputDocumentation(doc="Minimal count for kmers retained for comparing\nalternate choices [integer]")), ToolInput(tag="in_max_km_er_count", input_type=Int(optional=True), prefix="--max_kmer_count", doc=InputDocumentation(doc="Minimum acceptable average count for estimating\nthe maximal kmer length in reads [integer]")), ToolInput(tag="in_vector_percent", input_type=Float(optional=True), prefix="--vector_percent", doc=InputDocumentation(doc="(=0.05)  Count for  vectors as a fraction of the read\nnumber (1. disables) [float (0,1]]")), ToolInput(tag="in_insert_size", input_type=Int(optional=True), prefix="--insert_size", doc=InputDocumentation(doc="Expected insert size for paired reads (if not\nprovided, it will be estimated) [integer]")), ToolInput(tag="in_steps", input_type=Int(optional=True), prefix="--steps", doc=InputDocumentation(doc="(=11)             Number of assembly iterations from minimal to\nmaximal kmer length in reads [integer]")), ToolInput(tag="in_fraction", input_type=Float(optional=True), prefix="--fraction", doc=InputDocumentation(doc="(=0.1)         Maximum noise to signal ratio acceptable for\nextension [float [0,1)]")), ToolInput(tag="in_max_snp_len", input_type=Int(optional=True), prefix="--max_snp_len", doc=InputDocumentation(doc="(=150)      Maximal snp length [integer]")), ToolInput(tag="in_min_contig", input_type=Int(optional=True), prefix="--min_contig", doc=InputDocumentation(doc="(=200)       Minimal contig length reported in output\n[integer]")), ToolInput(tag="in_allow_snps", input_type=Boolean(optional=True), prefix="--allow_snps", doc=InputDocumentation(doc="Allow additional step for snp discovery [flag]")), ToolInput(tag="in_force_single_ends", input_type=Boolean(optional=True), prefix="--force_single_ends", doc=InputDocumentation(doc="Don't use paired-end information [flag]")), ToolInput(tag="in_seeds", input_type=File(optional=True), prefix="--seeds", doc=InputDocumentation(doc="Input file with seeds [string]")), ToolInput(tag="in_all", input_type=String(optional=True), prefix="--all", doc=InputDocumentation(doc="Output fasta for each iteration [string]")), ToolInput(tag="in_dbg_out", input_type=File(optional=True), prefix="--dbg_out", doc=InputDocumentation(doc="Output kmer file [string]")), ToolInput(tag="in_hist", input_type=File(optional=True), prefix="--hist", doc=InputDocumentation(doc="File for histogram [string]")), ToolInput(tag="in_connected_reads", input_type=File(optional=True), prefix="--connected_reads", doc=InputDocumentation(doc="File for connected paired reads [string]"))], outputs=[ToolOutput(tag="out_contigs_out", output_type=File(optional=True), selector=InputSelector(input_to_select="in_contigs_out", type_hint=File()), doc=OutputDocumentation(doc="Output file for contigs (stdout if not\nspecified) [string]")), ToolOutput(tag="out_dbg_out", output_type=File(optional=True), selector=InputSelector(input_to_select="in_dbg_out", type_hint=File()), doc=OutputDocumentation(doc="Output kmer file [string]"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Skesa_V0_1_0().translate("wdl", allow_empty_container=True)

