class: CommandLineTool
id: NanoPlot.cwl
inputs:
- id: threads
  doc: Set the allowed number of threads to be used by the script
  type: string
  inputBinding:
    prefix: --threads
- id: verbose
  doc: Write log messages also to terminal.
  type: boolean
  inputBinding:
    prefix: --verbose
- id: store
  doc: Store the extracted data in a pickle file for future plotting.
  type: boolean
  inputBinding:
    prefix: --store
- id: raw
  doc: Store the extracted data in tab separated file.
  type: boolean
  inputBinding:
    prefix: --raw
- id: huge
  doc: Input data is one very large file.
  type: boolean
  inputBinding:
    prefix: --huge
- id: outdir
  doc: Specify directory in which output has to be created.
  type: string
  inputBinding:
    prefix: --outdir
- id: prefix
  doc: Specify an optional prefix to be used for the output files.
  type: string
  inputBinding:
    prefix: --prefix
- id: maxlength
  doc: Hide reads longer than length specified.
  type: string
  inputBinding:
    prefix: --maxlength
- id: minlength
  doc: Hide reads shorter than length specified.
  type: string
  inputBinding:
    prefix: --minlength
- id: drop_outliers
  doc: Drop outlier reads with extreme long length.
  type: boolean
  inputBinding:
    prefix: --drop_outliers
- id: down_sample
  doc: Reduce dataset to N reads by random sampling.
  type: string
  inputBinding:
    prefix: --downsample
- id: log_length
  doc: Additionally show logarithmic scaling of lengths in plots.
  type: boolean
  inputBinding:
    prefix: --loglength
- id: percent_qual
  doc: Use qualities as theoretical percent identities.
  type: boolean
  inputBinding:
    prefix: --percentqual
- id: a_length
  doc: Use aligned read lengths rather than sequenced length (bam mode)
  type: boolean
  inputBinding:
    prefix: --alength
- id: min_qual
  doc: Drop reads with an average quality lower than specified.
  type: string
  inputBinding:
    prefix: --minqual
- id: runtime_until
  doc: Only take the N first hours of a run
  type: string
  inputBinding:
    prefix: --runtime_until
- id: read_type
  doc: '{1D,2D,1D2} Which read type to extract information about from summary. Options
    are 1D, 2D, 1D2'
  type: boolean
  inputBinding:
    prefix: --readtype
- id: barcoded
  doc: Use if you want to split the summary file by barcode
  type: boolean
  inputBinding:
    prefix: --barcoded
- id: color
  doc: Specify a valid matplotlib color for the plots
  type: string
  inputBinding:
    prefix: --color
- id: colormap
  doc: Specify a valid matplotlib colormap for the heatmap
  type: string
  inputBinding:
    prefix: --colormap
- id: format
  doc: Specify the output format of the plots.
  type: string
  inputBinding:
    prefix: --format
- id: plots
  doc: '[{kde,hex,dot,pauvre} [{kde,hex,dot,pauvre} ...]] Specify which bivariate
    plots have to be made.'
  type: boolean
  inputBinding:
    prefix: --plots
- id: list_colors
  doc: List the colors which are available for plotting and exit.
  type: boolean
  inputBinding:
    prefix: --listcolors
- id: list_colormaps
  doc: List the colors which are available for plotting and exit.
  type: boolean
  inputBinding:
    prefix: --listcolormaps
- id: no_n50
  doc: Hide the N50 mark in the read length histogram
  type: boolean
  inputBinding:
    prefix: --no-N50
- id: n50
  doc: Show the N50 mark in the read length histogram
  type: boolean
  inputBinding:
    prefix: --N50
- id: title
  doc: Add a title to all plots, requires quoting if using spaces
  type: string
  inputBinding:
    prefix: --title
- id: font_scale
  doc: Scale the font of the plots by a factor
  type: string
  inputBinding:
    prefix: --font_scale
- id: dpi
  doc: Set the dpi for saving images
  type: string
  inputBinding:
    prefix: --dpi
- id: hide_stats
  doc: Not adding Pearson R stats in some bivariate plots
  type: boolean
  inputBinding:
    prefix: --hide_stats
- id: fast_q
  doc: Data is in one or more default fastq file(s).
  type: File[]
  inputBinding:
    prefix: --fastq
- id: fast_a
  doc: Data is in one or more fasta file(s).
  type: File[]
  inputBinding:
    prefix: --fasta
- id: fast_q_rich
  doc: Data is in one or more fastq file(s) generated by albacore, MinKNOW or guppy
    with additional information concerning channel and time.
  type: File[]
  inputBinding:
    prefix: --fastq_rich
- id: fast_q_minimal
  doc: Data is in one or more fastq file(s) generated by albacore, MinKNOW or guppy
    with additional information concerning channel and time. Is extracted swiftly
    without elaborate checks.
  type: File[]
  inputBinding:
    prefix: --fastq_minimal
- id: summary
  doc: Data is in one or more summary file(s) generated by albacore or guppy.
  type: File[]
  inputBinding:
    prefix: --summary
- id: bam
  doc: Data is in one or more sorted bam file(s).
  type: File[]
  inputBinding:
    prefix: --bam
- id: ub_am
  doc: Data is in one or more unmapped bam file(s).
  type: File[]
  inputBinding:
    prefix: --ubam
- id: cram
  doc: Data is in one or more sorted cram file(s).
  type: File[]
  inputBinding:
    prefix: --cram
- id: pickle
  doc: Data is a pickle file stored earlier.
  type: string
  inputBinding:
    prefix: --pickle
outputs: []
cwlVersion: v1.1
baseCommand:
- NanoPlot
