from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, Directory, Int, String, File

Hal2Assemblyhub_Py_V0_1_0 = CommandToolBuilder(tool="hal2assemblyHub.py", base_command=["hal2assemblyHub.py"], inputs=[ToolInput(tag="in_cp_hal_file_to_out", input_type=Boolean(optional=True), prefix="--cpHalFileToOut", doc=InputDocumentation(doc="If specified, copy the input halfile to the output\ndirectory (instead of just make a softlink).\nDefault=False")), ToolInput(tag="in_ucsc_names", input_type=Boolean(optional=True), prefix="--ucscNames", doc=InputDocumentation(doc="Assume that sequence headers use the UCSC naming\nconvention, (i.e. 'genome.chr'), and  attempt to\nrename the sequences so that their names will end up\nas 'chr'")), ToolInput(tag="in_hub", input_type=Directory(optional=True), prefix="--hub", doc=InputDocumentation(doc="a single-word name of the directory containing the\ntrack hub files. Not displayed to hub users.\nDefault=myHub")), ToolInput(tag="in_short_label", input_type=Int(optional=True), prefix="--shortLabel", doc=InputDocumentation(doc="the short name for the track hub. Suggested maximum\nlength is 17 characters. Displayed as the hub name on\nthe Track Hubs page and the track group name on the\nbrowser tracks page. Default=my hub")), ToolInput(tag="in_long_label", input_type=Int(optional=True), prefix="--longLabel", doc=InputDocumentation(doc="a longer descriptive label for the track hub.\nSuggested maximum length is 80 characters. Displayed\nin the description field on the Track Hubs page.\nDefault=my hub")), ToolInput(tag="in_email", input_type=String(optional=True), prefix="--email", doc=InputDocumentation(doc="the contact to whom questions regarding the track hub\nshould be directed. Default=NoEmail")), ToolInput(tag="in_genomes", input_type=File(optional=True), prefix="--genomes", doc=InputDocumentation(doc="File specified list of genomes to make browser for. If\nspecified, only create browsers for these genomes in\nthe order provided by the list. Otherwise create\nbrowsers for all genomes in the input hal file")), ToolInput(tag="in_rename", input_type=File(optional=True), prefix="--rename", doc=InputDocumentation(doc="File that maps halfile genomeNames to names displayed\non the browser. Format:\n<halGenomeName>\t<genomeNameToDisplayOnBrowser>.\nDefault=none")), ToolInput(tag="in_tree", input_type=File(optional=True), prefix="--tree", doc=InputDocumentation(doc="Newick binary tree. The order of the tracks and the\ndefault track layout will be based on this tree if\noption 'genomes' is not specified. If not specified,\ntry to extract the newick tree from the input halfile.")), ToolInput(tag="in_url", input_type=String(optional=True), prefix="--url", doc=InputDocumentation(doc="Public url of the hub location")), ToolInput(tag="in_two_bit_dir", input_type=File(optional=True), prefix="--twobitdir", doc=InputDocumentation(doc="Optional. Directory containing the 2bit files of each\ngenomes. Default: extract from the input hal file.")), ToolInput(tag="in_lod", input_type=Boolean(optional=True), prefix="--lod", doc=InputDocumentation(doc="If specified, create 'level of detail' (lod) hal files\nand will put the lod.txt at the bigUrl instead of the\noriginal hal file. Default=False")), ToolInput(tag="in_lod_txt_file", input_type=File(optional=True), prefix="--lodTxtFile", doc=InputDocumentation(doc="'hal Level of detail' lod text file. If specified,\nwill put this at the bigUrl instead of the hal file.\nDefault=none")), ToolInput(tag="in_lod_dir", input_type=File(optional=True), prefix="--lodDir", doc=InputDocumentation(doc="'hal Level of detail' lod dir. If specified, will put\nthis at the bigUrl instead of the hal file.\nDefault=none")), ToolInput(tag="in_lod_max_block", input_type=Int(optional=True), prefix="--lodMaxBlock", doc=InputDocumentation(doc="Maximum number of blocks to display in a hal level of\ndetail (see halLodInterpolate.py --help for the\ndefault value).")), ToolInput(tag="in_lod_scale", input_type=String(optional=True), prefix="--lodScale", doc=InputDocumentation(doc="Scaling factor between two successive levels of detail\n(see halLodInterpolate.py --help for the default\nvalue).")), ToolInput(tag="in_lod_max_dna", input_type=Int(optional=True), prefix="--lodMaxDNA", doc=InputDocumentation(doc="Maximum query length such that its hal level of detail\nwill contain nucleotide information. Default=none (see\nhalLodInterpolate.py --help for the default value).")), ToolInput(tag="in_lod_in_memory", input_type=Boolean(optional=True), prefix="--lodInMemory", doc=InputDocumentation(doc="Load entire hal file into memory when generating\nlevels of detail instead of using hdf5 cache. Could\nresult in drastic speedup. Default=False.")), ToolInput(tag="in_lod_min_seq_frac", input_type=Int(optional=True), prefix="--lodMinSeqFrac", doc=InputDocumentation(doc="Minumum sequence length to sample as fraction of step\nsize for level of detail generation: ie sequences with\nlength <= floor(minSeqFrac * step) are ignored (see\nhalLodExtract --help for default value).")), ToolInput(tag="in_lod_min_cov_frac", input_type=Int(optional=True), prefix="--lodMinCovFrac", doc=InputDocumentation(doc="Minimum fraction of a genome that must be covered by\nsequences that exceed --minSeqFrac * step.  LODs that\nwould violate this threshold will not be generated (or\ndisplayed in  the browser).  This is seen a better\nthan the alternative, which is to produce unreasonably\nsparse LODs because half the sequences were not\nsampled (see halLodInterpolate.py --help for default\nvalue).")), ToolInput(tag="in_lod_chunk", input_type=Int(optional=True), prefix="--lodChunk", doc=InputDocumentation(doc="HDF5 chunk size for generated levels of detail (see\nhalLodExtract --help for default value).")), ToolInput(tag="in_bed_dirs", input_type=Directory(optional=True), prefix="--bedDirs", doc=InputDocumentation(doc="comma separated list of directories containing bed\nfiles of the input genomes. Each directory represents\na type of annotation. The annotations of each genome\nwill then be liftovered to all other genomes in the\nMSA. Example: 'genes,genomicIsland,tRNA'. Format of\neach directory: bedDir/ then genome1/ then chr1.bed,\nchr2.bed... Default=none")), ToolInput(tag="in_final_big_bed_dirs", input_type=Directory(optional=True), prefix="--finalBigBedDirs", doc=InputDocumentation(doc="comma separated list of directories containing final\nbig bed files to be displayed. No liftover will be\ndone for these files. Each directory represents a type\nof annotation. Example: 'genes,genomicIsland,tRNA'.\nFormat of each directory: bbDir/ then queryGenome/\nthen targetGenome1.bb, targetGenome2.bb ... (so\nannotation of queryGenome has been mapped to\ntargetGenomes and will be display on the targetGenome\nbrowsers). Default=none")), ToolInput(tag="in_bed_dirs_two", input_type=Int(optional=True), prefix="--bedDirs2", doc=InputDocumentation(doc="Similar to --bedDirs, except these tracks will be kept\nseparately and out of the composite track.\nDefault=none")), ToolInput(tag="in_final_big_bed_dirs_two", input_type=Int(optional=True), prefix="--finalBigBedDirs2", doc=InputDocumentation(doc="Similar to --finalBigBedDirs, except these tracks will\nbe kept separately and out of the composite track.\nDefault=none")), ToolInput(tag="in_no_bed_lift_over", input_type=Boolean(optional=True), prefix="--noBedLiftover", doc=InputDocumentation(doc="If specified, will not lift over the bed annotations.\nDefault=False")), ToolInput(tag="in_tabbed", input_type=Boolean(optional=True), prefix="--tabBed", doc=InputDocumentation(doc="If specified, treat tab as the delimiter of all the\nbed files. Default: any white space.")), ToolInput(tag="in_wig_dirs", input_type=Directory(optional=True), prefix="--wigDirs", doc=InputDocumentation(doc="comma separated list of directories containing wig\nfiles of the input genomes. Each directory represents\na type of annotation. The annotations of each genome\nwill then be liftovered to all other genomes in the\nMSA. Example: 'genes,genomicIsland,tRNA'. Format of\neach directory: wigDir/ then genome1/ then chr1.wig,\nchr2.wig... Default=none")), ToolInput(tag="in_final_bigwig_dirs", input_type=Directory(optional=True), prefix="--finalBigwigDirs", doc=InputDocumentation(doc="comma separated list of directories containing final\nbig wig files to be displayed. No liftover will be\ndone for these files. Each directory represents a type\nof annotation. Example: 'readCoverage,'. Format of\neach directory: bwDir/ then queryGenome/ then\ntargetGenome1.bw, targetGenome2.bw ... (so annotation\nof queryGenome has been mapped to targetGenomes and\nwill be display on the targetGenome browsers).\nDefault=none")), ToolInput(tag="in_no_wig_lift_over", input_type=Boolean(optional=True), prefix="--nowigLiftover", doc=InputDocumentation(doc="If specified, will not lift over the wig annotations.\nDefault=False")), ToolInput(tag="in_rms_k_dir", input_type=Directory(optional=True), prefix="--rmskDir", doc=InputDocumentation(doc="Directory containing repeatMasker's output files for\neach genome. Format: rmskDir/ then genome1/ then\ngenome.rmsk.SINE.bb, genome.rmsk.LINE.bb, ...\nDefault=none")), ToolInput(tag="in_gc_content", input_type=Boolean(optional=True), prefix="--gcContent", doc=InputDocumentation(doc="If specified, make GC-content tracks. Default=False")), ToolInput(tag="in_align_ability", input_type=Boolean(optional=True), prefix="--alignability", doc=InputDocumentation(doc="If specified, make Alignability (aka Alignment Depth)\ntracks. Default=False")), ToolInput(tag="in_conservation", input_type=File(optional=True), prefix="--conservation", doc=InputDocumentation(doc="Bed file providing regions to calculate the\nconservation tracks.")), ToolInput(tag="in_conservation_dir", input_type=Directory(optional=True), prefix="--conservationDir", doc=InputDocumentation(doc="Optional. Directory contains conservation bigwigs if\navailable. These bigwigs will be used. If this is not\nspecified, the program will compute the conservation\ntracks.")), ToolInput(tag="in_conservation_genome_name", input_type=File(optional=True), prefix="--conservationGenomeName", doc=InputDocumentation(doc="Name of the genome of the bed file provided in the '--\nconversation' option")), ToolInput(tag="in_conservation_tree", input_type=String(optional=True), prefix="--conservationTree", doc=InputDocumentation(doc="Optional. Newick tree for the conservation track")), ToolInput(tag="in_conservation_num_proc", input_type=Int(optional=True), prefix="--conservationNumProc", doc=InputDocumentation(doc="Optional. Number of processors to run conservation")), ToolInput(tag="in_clade_exclusive_regions", input_type=Boolean(optional=True), prefix="--cladeExclusiveRegions", doc=InputDocumentation(doc="If specified, will generate tracks of regions that are\nexclusive to each branch (including leaf 'branches',\nwhich will be genome-exclusive regions) on the tree.\nDefault=False")), ToolInput(tag="in_max_out_group_genomes", input_type=Int(optional=True), prefix="--maxOutgroupGenomes", doc=InputDocumentation(doc="Maximum number of outgroup genomes that a region is\nallowed to be in. Default=0")), ToolInput(tag="in_min_in_group_genomes", input_type=Int(optional=True), prefix="--minIngroupGenomes", doc=InputDocumentation(doc="Minimum number of ingroup genomes that a region must\nappear in. Default=all ingroup genomes (branch node\nand all its children).")), ToolInput(tag="in_self_alignment_snakes", input_type=Boolean(optional=True), prefix="--selfAlignmentSnakes", doc=InputDocumentation(doc="Produce a self-alignment snake track for every genome")), ToolInput(tag="in_logoff", input_type=Boolean(optional=True), prefix="--logOff", doc=InputDocumentation(doc="Turn off logging. (default is CRITICAL)")), ToolInput(tag="in_loginfo", input_type=Boolean(optional=True), prefix="--logInfo", doc=InputDocumentation(doc="Turn on logging at INFO level. (default is CRITICAL)")), ToolInput(tag="in_log_debug", input_type=Boolean(optional=True), prefix="--logDebug", doc=InputDocumentation(doc="Turn on logging at DEBUG level. (default is CRITICAL)")), ToolInput(tag="in_loglevel", input_type=String(optional=True), prefix="--logLevel", doc=InputDocumentation(doc="Log at level (may be either OFF/INFO/DEBUG/CRITICAL).\n(default is CRITICAL)")), ToolInput(tag="in_log_file", input_type=File(optional=True), prefix="--logFile", doc=InputDocumentation(doc="File to log in")), ToolInput(tag="in_rotating_logging", input_type=Boolean(optional=True), prefix="--rotatingLogging", doc=InputDocumentation(doc="Turn on rotating logging, which prevents log files\ngetting too big.")), ToolInput(tag="in_job_tree", input_type=File(optional=True), prefix="--jobTree", doc=InputDocumentation(doc="Directory in which to place job management files and\nthe global accessed temporary file directories(this\nneeds to be globally accessible by all machines\nrunning jobs). If you pass an existing directory it\nwill check if it's a valid existing job tree, then try\nand restart the jobs in it. The default=./jobTree")), ToolInput(tag="in_stats", input_type=Boolean(optional=True), prefix="--stats", doc=InputDocumentation(doc="Records statistics about the job-tree to be used by\njobTreeStats. default=False")), ToolInput(tag="in_batch_system", input_type=String(optional=True), prefix="--batchSystem", doc=InputDocumentation(doc="The type of batch system to run the job(s) with,\ncurrently can be 'singleMachine'/'parasol'/'acidTest'/\n'gridEngine'/'lsf'. default=singleMachine")), ToolInput(tag="in_max_threads", input_type=Int(optional=True), prefix="--maxThreads", doc=InputDocumentation(doc="The maximum number of threads (technically processes\nat this point) to use when running in single machine\nmode. Increasing this will allow more jobs to run\nconcurrently when running on a single machine.\ndefault=4")), ToolInput(tag="in_parasol_command", input_type=String(optional=True), prefix="--parasolCommand", doc=InputDocumentation(doc="The command to run the parasol program default=parasol")), ToolInput(tag="in_default_memory", input_type=Int(optional=True), prefix="--defaultMemory", doc=InputDocumentation(doc="The default amount of memory to request for a job (in\nbytes), by default is 2^31 = 2 gigabytes,\ndefault=2147483648")), ToolInput(tag="in_default_cpu", input_type=Int(optional=True), prefix="--defaultCpu", doc=InputDocumentation(doc="The default the number of cpus to dedicate a job.\ndefault=1")), ToolInput(tag="in_max_cpus", input_type=Int(optional=True), prefix="--maxCpus", doc=InputDocumentation(doc="The maximum number of cpus to request from the batch\nsystem at any one time. default=9223372036854775807")), ToolInput(tag="in_max_memory", input_type=Int(optional=True), prefix="--maxMemory", doc=InputDocumentation(doc="The maximum amount of memory to request from the batch\nsystem at any one time. default=9223372036854775807")), ToolInput(tag="in_retry_count", input_type=Int(optional=True), prefix="--retryCount", doc=InputDocumentation(doc="Number of times to retry a failing job before giving\nup and labeling job failed. default=0")), ToolInput(tag="in_max_job_duration", input_type=Int(optional=True), prefix="--maxJobDuration", doc=InputDocumentation(doc="Maximum runtime of a job (in seconds) before we kill\nit (this is a lower bound, and the actual time before\nkilling the job may be longer).\ndefault=9223372036854775807")), ToolInput(tag="in_rescue_jobs_frequency", input_type=String(optional=True), prefix="--rescueJobsFrequency", doc=InputDocumentation(doc="Period of time to wait (in seconds) between checking\nfor missing/overlong jobs, that is jobs which get lost\nby the batch system. Expert parameter. (default is set\nby the batch system)")), ToolInput(tag="in_big_batch_system", input_type=String(optional=True), prefix="--bigBatchSystem", doc=InputDocumentation(doc="The batch system to run for jobs with larger\nmemory/cpus requests, currently can be\n'singleMachine'/'parasol'/'acidTest'/'gridEngine'.\ndefault=none")), ToolInput(tag="in_big_memory_threshold", input_type=Int(optional=True), prefix="--bigMemoryThreshold", doc=InputDocumentation(doc="The memory threshold above which to submit to the big\nqueue. default=9223372036854775807")), ToolInput(tag="in_big_cpu_threshold", input_type=Int(optional=True), prefix="--bigCpuThreshold", doc=InputDocumentation(doc="The cpu threshold above which to submit to the big\nqueue. default=9223372036854775807")), ToolInput(tag="in_big_max_cpus", input_type=Int(optional=True), prefix="--bigMaxCpus", doc=InputDocumentation(doc="The maximum number of big batch system cpus to allow\nat one time on the big queue.\ndefault=9223372036854775807")), ToolInput(tag="in_big_max_memory", input_type=Int(optional=True), prefix="--bigMaxMemory", doc=InputDocumentation(doc="The maximum amount of memory to request from the big\nbatch system at any one time.\ndefault=9223372036854775807")), ToolInput(tag="in_job_time", input_type=Int(optional=True), prefix="--jobTime", doc=InputDocumentation(doc="The approximate time (in seconds) that you'd like a\nlist of child jobs to be run serially before being\nparallelized. This parameter allows one to avoid over\nparallelizing tiny jobs, and therefore paying\nsignificant scheduling overhead, by running tiny jobs\nin series on a single node/core of the cluster.\ndefault=30")), ToolInput(tag="in_max_log_file_size", input_type=Int(optional=True), prefix="--maxLogFileSize", doc=InputDocumentation(doc="The maximum size of a job log file to keep (in bytes),\nlog files larger than this will be truncated to the\nlast X bytes. Default is 50 kilobytes, default=50120")), ToolInput(tag="in_command", input_type=String(optional=True), prefix="--command", doc=InputDocumentation(doc="The command to run (which will generate subsequent\njobs). This is deprecated\n")), ToolInput(tag="in_hal_file", input_type=String(), position=0, doc=InputDocumentation(doc="")), ToolInput(tag="in_output_directory", input_type=String(), position=1, doc=InputDocumentation(doc=""))], outputs=[ToolOutput(tag="out_rms_k_dir", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_rms_k_dir", type_hint=File()), doc=OutputDocumentation(doc="Directory containing repeatMasker's output files for\neach genome. Format: rmskDir/ then genome1/ then\ngenome.rmsk.SINE.bb, genome.rmsk.LINE.bb, ...\nDefault=none"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Hal2Assemblyhub_Py_V0_1_0().translate("wdl", allow_empty_container=True)

