class: CommandLineTool
id: moss.cwl
inputs:
- id: in_bam
  doc: "original BAM files for one sample,\ncan be addressed multiple times"
  type: string?
  inputBinding:
    prefix: --bam
- id: in_realigned
  doc: "realigned BAM files for one sample,\ncan be addressed multiple times"
  type: string?
  inputBinding:
    prefix: --realigned
- id: in_ref
  doc: reference FASTA file
  type: File?
  inputBinding:
    prefix: --ref
- id: in_output
  doc: output VCF file
  type: File?
  inputBinding:
    prefix: --output
- id: in_normal
  doc: normal sample's germline VCF result
  type: string?
  inputBinding:
    prefix: --normal
- id: in_vcf
  doc: tumor samples' somatic VCF result
  type: string?
  inputBinding:
    prefix: --vcf
- id: in_loci
  doc: candidate loci VCF files generated by python script
  type: string?
  inputBinding:
    prefix: --loci
- id: in_tau
  doc: optional threshold for somatic score, default is 0
  type: long?
  inputBinding:
    prefix: --tau
- id: in_mu
  doc: set somatic mutation prior, 1-5x10^(-m), default is 1-5e-4
  type: double?
  inputBinding:
    prefix: --mu
- id: in_min_base_qual
  doc: minimum threshold for base quality, default is 13
  type: long?
  inputBinding:
    prefix: --min-base-qual
- id: in_min_mapping_qual
  doc: minimum threshold for mapping quality, default is 30
  type: long?
  inputBinding:
    prefix: --min-mapping-qual
- id: in_max_dep
  doc: max depth of the dataset, default is 500
  type: long?
  inputBinding:
    prefix: --max_dep
- id: in_grid_size
  doc: "number of grid points in [0, 1] for approximate integration,\ndefault is 20,\
    \ must be non-negative"
  type: long?
  inputBinding:
    prefix: --grid-size
- id: in_dry
  doc: set to dry run
  type: boolean?
  inputBinding:
    prefix: --dry
- id: in_filter_total
  doc: set to filter variants with total tumor depth < 150
  type: boolean?
  inputBinding:
    prefix: --filter-total
- id: in_filter_vaf
  doc: set to filter variants with VAF in any samples < 0.1
  type: boolean?
  inputBinding:
    prefix: --filter-vaf
- id: in_ignore_zero
  doc: ignore samples with all reads same as the normal allele
  type: boolean?
  inputBinding:
    prefix: --ignore0
- id: in_pile_up
  doc: output pileup, only effective in dry-run mode
  type: boolean?
  inputBinding:
    prefix: --pileup
outputs:
- id: out_stdout
  doc: Standard output stream
  type: stdout
- id: out_output
  doc: output VCF file
  type: File?
  outputBinding:
    glob: $(inputs.in_output)
hints:
- class: DockerRequirement
  dockerPull: quay.io/biocontainers/moss:0.1.1--h179b981_0
cwlVersion: v1.1
baseCommand:
- moss
