version 1.0

task OutputTranscriptInfo {
  input {
    File? coordinate_file
    File? gtf_file
    File? transcript_file
    String? output_longest_trans_cr_pts
    String? output_all_trans_cr_pts
  }
  command <<<
    OutputTranscriptInfo \
      ~{if defined(coordinate_file) then ("--coordinateFile " +  '"' + coordinate_file + '"') else ""} \
      ~{if defined(gtf_file) then ("--gtfFile " +  '"' + gtf_file + '"') else ""} \
      ~{if defined(transcript_file) then ("--transcriptFile " +  '"' + transcript_file + '"') else ""} \
      ~{if defined(output_longest_trans_cr_pts) then ("--output-longest-transcrpts " +  '"' + output_longest_trans_cr_pts + '"') else ""} \
      ~{if defined(output_all_trans_cr_pts) then ("--output-all-transcrpts " +  '"' + output_all_trans_cr_pts + '"') else ""}
  >>>
  runtime {
    docker: "quay.io/biocontainers/ribominer:0.2.3.1--pyh3252c3a_0"
  }
  parameter_meta {
    coordinate_file: "The file should contain the coordinate of start and\\nstop codon of each transcript. Generated by\\nRiboCode.[required]"
    gtf_file: "Reference gene annotation in gtf format.[required]"
    transcript_file: "transcript file with fasta format generated by\\nRiboCode.[required]"
    output_longest_trans_cr_pts: "Longest transcripts information."
    output_all_trans_cr_pts: "All transcripts information.\\n"
  }
  output {
    File out_stdout = stdout()
  }
}