from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Directory, Int, Boolean

Outrigger_Validate_V0_1_0 = CommandToolBuilder(tool="outrigger_validate", base_command=["outrigger", "validate"], inputs=[ToolInput(tag="in_fast_a", input_type=File(optional=True), prefix="--fasta", doc=InputDocumentation(doc="Location of the genome fasta file for which to get the\nsplice site sequences from")), ToolInput(tag="in_genome", input_type=File(optional=True), prefix="--genome", doc=InputDocumentation(doc="Either the genome name (e.g. 'mm10' or 'hg19') or\nlocation of the genome chromosome sizes file for\n'bedtools flank' to make sure we do not accidentally\nask for genome positions that are outside of the\ndefined range")), ToolInput(tag="in_index", input_type=Directory(optional=True), prefix="--index", doc=InputDocumentation(doc="Name of the folder where you saved the output from\n'outrigger index' (default is\n./outrigger_output/index, which is relative to the\ndirectory where you called this program, assuming you\nhave called 'outrigger psi' in the same folder as you\ncalled 'outrigger index').")), ToolInput(tag="in_output", input_type=Directory(optional=True), prefix="--output", doc=InputDocumentation(doc="Name of the folder where you saved the output from\n'outrigger index' (default is ./outrigger_output,\nwhich is relative to the directory where you called\nthe program).")), ToolInput(tag="in_valid_splice_sites", input_type=Int(optional=True), prefix="--valid-splice-sites", doc=InputDocumentation(doc="The intron-definition based splice sites that are\nallowed in the data, which is in the format 5'/3' of\nthe intron, and separated by commas for different\ntypes. Default is GT/AG,GC/AG,AT/AC, which are the\nmajor and minor spliceosome splice sites in mammalian\nsystems.")), ToolInput(tag="in_debug", input_type=Boolean(optional=True), prefix="--debug", doc=InputDocumentation(doc="If given, print debugging logging information to\nstandard out")), ToolInput(tag="in_low_memory", input_type=Boolean(optional=True), prefix="--low-memory", doc=InputDocumentation(doc="If set, then use a smaller memory footprint. By\ndefault, this is off.\n"))], outputs=[ToolOutput(tag="out_index", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_index", type_hint=File()), doc=OutputDocumentation(doc="Name of the folder where you saved the output from\n'outrigger index' (default is\n./outrigger_output/index, which is relative to the\ndirectory where you called this program, assuming you\nhave called 'outrigger psi' in the same folder as you\ncalled 'outrigger index').")), ToolOutput(tag="out_output", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_output", type_hint=File()), doc=OutputDocumentation(doc="Name of the folder where you saved the output from\n'outrigger index' (default is ./outrigger_output,\nwhich is relative to the directory where you called\nthe program)."))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Outrigger_Validate_V0_1_0().translate("wdl", allow_empty_container=True)

