from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, String

Showseq_V0_1_0 = CommandToolBuilder(tool="showseq", base_command=["showseq"], inputs=[ToolInput(tag="in_format", input_type=Boolean(optional=True), prefix="-format", doc=InputDocumentation(doc="menu       [2] Display format (Values: 0 (Enter your\nown list of things to display); 1 (Sequence\nonly); 2 (Default sequence with features); 3\n(Pretty sequence); 4 (One frame\ntranslation); 5 (Three frame translations);\n6 (Six frame translations); 7 (Restriction\nenzyme map); 8 (Baroque))")), ToolInput(tag="in_compframe_translation_t", input_type=Boolean(optional=True), prefix="-3", doc=InputDocumentation(doc="(CompFrame3 translation); T (Ticks line);")), ToolInput(tag="in_translate", input_type=Boolean(optional=True), prefix="-translate", doc=InputDocumentation(doc="range      [If this is left blank the complete sequence\nis translated.] Regions to translate (if\ntranslating).\nIf this is left blank the complete sequence\nis translated.\nA set of regions is specified by a set of\npairs of positions.\nThe positions are integers.\nThey are separated by any non-digit,\nnon-alpha character.\nExamples of region specifications are:\n24-45, 56-78\n1:45, 67=99;765..888")), ToolInput(tag="in_rev_translate", input_type=Boolean(optional=True), prefix="-revtranslate", doc=InputDocumentation(doc="range      [If this is left blank the complete reverse\nsequence is translated.] Regions to\ntranslate (if translating).\nIf this is left blank the complete sequence\nis translated.\nA set of regions is specified by a set of\npairs of positions.\nThe positions are integers.\nThey are separated by any non-digit,\nnon-alpha character.\nExamples of region specifications are:\n78-56, 45-24,\n888..765, 99=67; 45:1")), ToolInput(tag="in_uppercase", input_type=Boolean(optional=True), prefix="-uppercase", doc=InputDocumentation(doc="range      [If this is left blank, then the sequence\ncase is left alone.] Regions to put in\nuppercase.\nIf this is left blank, then the sequence\ncase is left alone.\nA set of regions is specified by a set of\npairs of positions.\nThe positions are integers.\nThey are separated by any non-digit,\nnon-alpha character.\nExamples of region specifications are:\n24-45, 56-78\n1:45, 67=99;765..888\n1,5,8,10,23,45,57,99")), ToolInput(tag="in_highlight", input_type=Boolean(optional=True), prefix="-highlight", doc=InputDocumentation(doc="range      [(full sequence)] Regions to colour if\nformatting for HTML.\nIf this is left blank, then the sequence is\nleft alone.\nA set of regions is specified by a set of\npairs of positions.\nThe positions are integers.\nThey are followed by any valid HTML font\ncolour.\nExamples of region specifications are:\n24-45 blue 56-78 orange\n1-100 green 120-156 red\nA file of ranges to colour (one range per\nline) can be specified as '@filename'.")), ToolInput(tag="in_annotation", input_type=Boolean(optional=True), prefix="-annotation", doc=InputDocumentation(doc="range      [If this is left blank, then no annotation\nis added.] Regions to annotate by marking.\nIf this is left blank, then no annotation is\nadded.\nA set of regions is specified by a set of\npairs of positions followed by optional\ntext.\nThe positions are integers.\nThey are followed by any text (but not\ndigits when on the command-line).\nExamples of region specifications are:\n24-45 new domain 56-78 match to Mouse\n1-100 First part 120-156 oligo\nA file of ranges to annotate (one range per\nline) can be specified as '@filename'.")), ToolInput(tag="in_enzymes", input_type=Boolean(optional=True), prefix="-enzymes", doc=InputDocumentation(doc="string     [all] The name 'all' reads in all enzyme\nnames from the REBASE database. You can\nspecify enzymes by giving their names with\ncommas between then, such as:\n'HincII,hinfI,ppiI,hindiii'.\nThe case of the names is not important. You\ncan specify a file of enzyme names to read\nin by giving the name of the file holding\nthe enzyme names with a '@' character in\nfront of it, for example, '@enz.list'.\nBlank lines and lines starting with a hash\ncharacter or '!' are ignored and all other\nlines are concatenated together with a comma\ncharacter ',' and then treated as the list\nof enzymes to search for.\nAn example of a file of enzyme names is:\n! my enzymes\nHincII, ppiII\n! other enzymes\nhindiii\nHinfI\nPpiI (Any string)")), ToolInput(tag="in_table", input_type=Boolean(optional=True), prefix="-table", doc=InputDocumentation(doc="menu       [0] Genetic code to use (Values: 0\n(Standard); 1 (Standard (with alternative\ninitiation codons)); 2 (Vertebrate\nMitochondrial); 3 (Yeast Mitochondrial); 4\n(Mold, Protozoan, Coelenterate Mitochondrial\nand Mycoplasma/Spiroplasma); 5\n(Invertebrate Mitochondrial); 6 (Ciliate\nMacronuclear and Dasycladacean); 9\n(Echinoderm Mitochondrial); 10 (Euplotid\nNuclear); 11 (Bacterial); 12 (Alternative\nYeast Nuclear); 13 (Ascidian Mitochondrial);\n14 (Flatworm Mitochondrial); 15\n(Blepharisma Macronuclear); 16\n(Chlorophycean Mitochondrial); 21 (Trematode\nMitochondrial); 22 (Scenedesmus obliquus);\n23 (Thraustochytrium Mitochondrial))")), ToolInput(tag="in_source_match", input_type=Boolean(optional=True), prefix="-sourcematch", doc=InputDocumentation(doc="string     [*] By default any feature source in the\nfeature table is shown. You can set this to\nmatch any feature source you wish to show.\nThe source name is usually either the name\nof the program that detected the feature or\nit is the feature table (eg: EMBL) that the\nfeature came from.\nThe source may be wildcarded by using '*'.\nIf you wish to show more than one source,\nseparate their names with the character '|',\neg:\ngene* | embl (Any string)")), ToolInput(tag="in_type_match", input_type=Boolean(optional=True), prefix="-typematch", doc=InputDocumentation(doc="string     [*] By default any feature type in the\nfeature table is shown. You can set this to\nmatch any feature type you wish to show.\nSee http://www.ebi.ac.uk/embl/WebFeat/ for a\nlist of the EMBL feature types and see\nAppendix A of the Swissprot user manual in\nhttp://www.expasy.org/sprot/userman.html for\na list of the Swissprot feature types.\nThe type may be wildcarded by using '*'.\nIf you wish to show more than one type,\nseparate their names with the character '|',\neg:\n*UTR | intron (Any string)")), ToolInput(tag="in_sense_match", input_type=Boolean(optional=True), prefix="-sensematch", doc=InputDocumentation(doc="integer    [0 - any sense, 1 - forward sense, -1 -\nreverse sense] By default any feature type\nin the feature table is shown. You can set\nthis to match any feature sense you wish to\nshow. 0 - any sense, 1 - forward sense, -1 -\nreverse sense (Integer from -1 to 1)")), ToolInput(tag="in_min_score", input_type=Boolean(optional=True), prefix="-minscore", doc=InputDocumentation(doc="float      [0.0] Minimum score of feature to display\n(see also maxscore) (Any numeric value)")), ToolInput(tag="in_max_score", input_type=Boolean(optional=True), prefix="-maxscore", doc=InputDocumentation(doc="float      [0.0] Maximum score of feature to display.\nIf both minscore and maxscore are zero (the\ndefault), then any score is ignored (Any\nnumeric value)")), ToolInput(tag="in_tag_match", input_type=Boolean(optional=True), prefix="-tagmatch", doc=InputDocumentation(doc="string     [*] Tags are the types of extra values that\na feature may have. For example in the EMBL\nfeature table, a 'CDS' type of feature may\nhave the tags '/codon', '/codon_start',\n'/db_xref', '/EC_number', '/evidence',\n'/exception', '/function', '/gene',\n'/label', '/map', '/note', '/number',\n'/partial', '/product', '/protein_id',\n'/pseudo', '/standard_name', '/translation',\n'/transl_except', '/transl_table', or\n'/usedin'. Some of these tags also have\nvalues, for example '/gene' can have the\nvalue of the gene name.\nBy default any feature tag in the feature\ntable is shown. You can set this to match\nany feature tag you wish to show.\nThe tag may be wildcarded by using '*'.\nIf you wish to show more than one tag,\nseparate their names with the character '|',\neg:\ngene | label (Any string)")), ToolInput(tag="in_value_match", input_type=Boolean(optional=True), prefix="-valuematch", doc=InputDocumentation(doc="string     [*] Tag values are the values associated\nwith a feature tag. Tags are the types of\nextra values that a feature may have. For\nexample in the EMBL feature table, a 'CDS'\ntype of feature may have the tags '/codon',\n'/codon_start', '/db_xref', '/EC_number',\n'/evidence', '/exception', '/function',\n'/gene', '/label', '/map', '/note',\n'/number', '/partial', '/product',\n'/protein_id', '/pseudo', '/standard_name',\n'/translation', '/transl_except',\n'/transl_table', or '/usedin'. Only some of\nthese tags can have values, for example\n'/gene' can have the value of the gene name.\nBy default any feature tag value in the\nfeature table is shown. You can set this to\nmatch any feature tag value you wish to\nshow.\nThe tag value may be wildcarded by using\n'*'.\nIf you wish to show more than one tag value,\nseparate their names with the character\n'|', eg:\npax* | 10 (Any string)")), ToolInput(tag="in_strict_tags", input_type=Boolean(optional=True), prefix="-stricttags", doc=InputDocumentation(doc="boolean    [N] By default if any tag/value pair in a\nfeature matches the specified tag and value,\nthen all the tags/value pairs of that\nfeature will be displayed. If this is set to\nbe true, then only those tag/value pairs in\na feature that match the specified tag and\nvalue will be displayed.")), ToolInput(tag="in_m_file", input_type=Boolean(optional=True), prefix="-mfile", doc=InputDocumentation(doc="datafile   [Emethylsites.dat] Restriction enzyme\nmethylation data file")), ToolInput(tag="in_flat_reformat", input_type=Boolean(optional=True), prefix="-flatreformat", doc=InputDocumentation(doc="boolean    [N] This changes the output format to one\nwhere the recognition site is indicated by a\nrow of '===' characters and the cut site is\npointed to by a '>' character in the\nforward sense, or a '<' in the reverse sense\nstrand.")), ToolInput(tag="in_min_cuts", input_type=Boolean(optional=True), prefix="-mincuts", doc=InputDocumentation(doc="integer    [1] This sets the minimum number of cuts for\nany restriction enzyme that will be\nconsidered. Any enzymes that cut fewer times\nthan this will be ignored. (Integer from 1\nto 1000)")), ToolInput(tag="in_max_cuts", input_type=Boolean(optional=True), prefix="-maxcuts", doc=InputDocumentation(doc="integer    [2000000000] This sets the maximum number of\ncuts for any restriction enzyme that will\nbe considered. Any enzymes that cut more\ntimes than this will be ignored. (Any\ninteger value)")), ToolInput(tag="in_site_len", input_type=Boolean(optional=True), prefix="-sitelen", doc=InputDocumentation(doc="integer    [4] This sets the minimum length of the\nrestriction enzyme recognition site. Any\nenzymes with sites shorter than this will be\nignored. (Integer from 2 to 20)")), ToolInput(tag="in_single", input_type=Boolean(optional=True), prefix="-single", doc=InputDocumentation(doc="boolean    [N] If this is set then this forces the\nvalues of the mincuts and maxcuts qualifiers\nto both be 1. Any other value you may have\nset them to will be ignored.")), ToolInput(tag="in_plasmid", input_type=Boolean(optional=True), prefix="-plasmid", doc=InputDocumentation(doc="boolean    [N] If this is set then this allows searches\nfor restriction enzyme recognition site and\ncut positions that span the end of the\nsequence to be considered.")), ToolInput(tag="in_methylation", input_type=Boolean(optional=True), prefix="-methylation", doc=InputDocumentation(doc="boolean    [N] If this is set then RE recognition sites\nwill not match methylated bases.")), ToolInput(tag="in_three_letter", input_type=Boolean(optional=True), prefix="-threeletter", doc=InputDocumentation(doc="boolean    [N] Display protein sequences in\nthree-letter code")), ToolInput(tag="in_number", input_type=Boolean(optional=True), prefix="-number", doc=InputDocumentation(doc="boolean    [N] Number the sequences")), ToolInput(tag="in_width", input_type=Boolean(optional=True), prefix="-width", doc=InputDocumentation(doc="integer    [60] Width of sequence to display (Integer 1\nor more)")), ToolInput(tag="in_length", input_type=Boolean(optional=True), prefix="-length", doc=InputDocumentation(doc="integer    [0] Line length of page (0 for indefinite)\n(Integer 0 or more)")), ToolInput(tag="in_margin", input_type=Boolean(optional=True), prefix="-margin", doc=InputDocumentation(doc="integer    [10] Margin around sequence for numbering\n(Integer 0 or more)")), ToolInput(tag="in_offset", input_type=Boolean(optional=True), prefix="-offset", doc=InputDocumentation(doc="integer    [1] Offset to start numbering the sequence\nfrom (Any integer value)")), ToolInput(tag="in_html", input_type=Boolean(optional=True), prefix="-html", doc=InputDocumentation(doc="boolean    [N] Use HTML formatting")), ToolInput(tag="in_considered_dot", input_type=String(), position=0, doc=InputDocumentation(doc="-[no]ambiguity      boolean    [Y] This allows those enzymes which have one")), ToolInput(tag="in_the_mdot", input_type=String(), position=0, doc=InputDocumentation(doc="-orfminsize         integer    [0] This sets the minimum size of Open"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Showseq_V0_1_0().translate("wdl", allow_empty_container=True)

