from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Directory, Boolean, Int, String

Coverage2Cytosine_V0_1_0 = CommandToolBuilder(tool="coverage2cytosine", base_command=["coverage2cytosine"], inputs=[ToolInput(tag="in_output", input_type=File(optional=True), prefix="--output", doc=InputDocumentation(doc="Name of the output file, mandatory.")), ToolInput(tag="in_dir", input_type=Directory(optional=True), prefix="--dir", doc=InputDocumentation(doc="Output directory. Output is written to the current directory if not specified explicitly.")), ToolInput(tag="in_genome_folder", input_type=File(optional=True), prefix="--genome_folder", doc=InputDocumentation(doc="Enter the genome folder you wish to use to extract sequences from (relative or full path). Accepted\nformats are FastA files ending with '.fa' or '.fasta'. Specifying a genome folder path is mandatory.")), ToolInput(tag="in_cx_context", input_type=File(optional=True), prefix="--CX_context", doc=InputDocumentation(doc="The output file contains information on every single cytosine in the genome irrespective of\nits context. This applies to both forward and reverse strands. Please be aware that this will\ngenerate output files with > 1.1 billion lines for a mammalian genome such as human or mouse.\nDefault: OFF (i.e. Default = CpG context only).")), ToolInput(tag="in_merge_cpg", input_type=Boolean(optional=True), prefix="--merge_CpG", doc=InputDocumentation(doc="Using this option will post-process the genome-wide report to write out an additional coverage\nfile (see above for the coverage file format) which has the top and bottom strand methylation\nevidence pooled into a single CpG dinucleotide entity. This may be the desirable input format\nfor some downstream processing tools such as the R-package bsseq (by K.D. Hansen). An example would be:\ngenome-wide CpG report (old)\ngi|9626372|ref|NC_001422.1|     157     +       313     156     CG\ngi|9626372|ref|NC_001422.1|     158     -       335     156     CG\nmerged CpG evidence coverage file (new)\ngi|9626372|ref|NC_001422.1|     157     158     67.500000       648     312\nThis option is currently experimental, and only works if CpG context only and a single genome-wide report\nwere specified (i.e. it doesn't work with the options --CX or --split_by_chromosome).\nIf --zero_based was specified as well the output coverage file is in zero-based half-open format like\nbedGraph files.")), ToolInput(tag="in_discordance", input_type=Int(optional=True), prefix="--discordance", doc=InputDocumentation(doc="When in '--merge_CpG' mode, apply a filter for the maximum allowed discordance between top and bottom\nstrand methylation values expressed as the absolute difference in percent methylation. Discordant CpGs\nare written to a file called 'discordant_CpG_evidence.cov' (not merged). As example consider:\ntop:     gi|170079663|ref|NC_010473.1|   573     +       5       6       CG      CGC\nbottom:  gi|170079663|ref|NC_010473.1|   574     -       13      4       CG      CGG\nwith '--discordance 20'\nThe methylation % difference here is 31%, so the read would go into the discordant.cov file. CpG positions\nfor which either the top or bottom strand was not measured at all will not be assessed for discordance, and\nhence appear in the regular 'merged_CpG_evidence.cov' file.")), ToolInput(tag="in_gc_context", input_type=File(optional=True), prefix="--gc_context", doc=InputDocumentation(doc="In addition to normal processing this option will reprocess the genome to find methylation in\nGpC context. This might be useful for specialist applications where GpC methylases had been\ndeployed. The output format is exactly the same as for the normal cytosine report, and only\npositions covered by at least one read are reported (output file ends in .GpC_report.txt). In addition\nthis will write out a Bismark coverage file (ending in GpC.cov).")), ToolInput(tag="in_nome_seq", input_type=Boolean(optional=True), prefix="--nome-seq", doc=InputDocumentation(doc="Sample is NOMe-Seq (nucleosome occupancy and methylome sequencing) where accessible DNA gets methylated\nin a GpC context (sets option '--gc' as well). The option '--nome-seq':\n(i) filters the genome-wide CpG-report to only output cytosines in ACG and TCG context\n(ii) filters the GC context output to only report cytosines in GCA, GCC and GCT context\nBoth of these measures aim to reduce unwanted biases, i.e. the influence of GCG and CCG on endogenous\nCpG methylation, and the inlfluence of CpG methylation on (the NOMe-Seq specific) GC context methylation.\nPLEASE NOTE that NOMe-Seq data requires a .cov.gz file as input which has been generated in\nnon-CG mode (--CX), else the GpC output file will be empty... Default: OFF.")), ToolInput(tag="in_coverage_threshold", input_type=Int(optional=True), prefix="--coverage_threshold", doc=InputDocumentation(doc="Positions have to be covered by at least INT calls (irrespective of their methylation state) before\nthey get reported. For NOMe-seq, the minimum threshold is automatically set to 1 unless specified explicitly.\nSetting a coverage threshold does not work in conjunction with --merge_CpGs. Default: 0 (i.e. all genomic positions get reported).")), ToolInput(tag="in_ff", input_type=Boolean(optional=True), prefix="--ff", doc=InputDocumentation(doc="In addition to the standard output selecting --ff will also extract a four and five (tetra/penta)\nnucleotide context for the cytosines in question. Too short sequences (e.g. at the edges of the\nchromosome) will be left blank; sequences containing Ns are ignored.ll")), ToolInput(tag="in_zero_based", input_type=Boolean(optional=True), prefix="--zero_based", doc=InputDocumentation(doc="Uses 0-based coordinates instead of 1-based coordinates throughout. Default: OFF.")), ToolInput(tag="in_split_by_chromosome", input_type=Boolean(optional=True), prefix="--split_by_chromosome", doc=InputDocumentation(doc="Writes the output into individual files for each chromosome instead of a single output file. Files\nwill be named to include the input filename and the chromosome number.")), ToolInput(tag="in_gzip", input_type=File(optional=True), prefix="--gzip", doc=InputDocumentation(doc="Output file will be GZIP compressed (ending in .gz). Only works for standard CpG- and CX-output.")), ToolInput(tag="in_input", input_type=String(optional=True), position=0, doc=InputDocumentation(doc=""))], outputs=[ToolOutput(tag="out_output", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output", type_hint=File()), doc=OutputDocumentation(doc="Name of the output file, mandatory.")), ToolOutput(tag="out_dir", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_dir", type_hint=File()), doc=OutputDocumentation(doc="Output directory. Output is written to the current directory if not specified explicitly.")), ToolOutput(tag="out_cx_context", output_type=File(optional=True), selector=InputSelector(input_to_select="in_cx_context", type_hint=File()), doc=OutputDocumentation(doc="The output file contains information on every single cytosine in the genome irrespective of\nits context. This applies to both forward and reverse strands. Please be aware that this will\ngenerate output files with > 1.1 billion lines for a mammalian genome such as human or mouse.\nDefault: OFF (i.e. Default = CpG context only).")), ToolOutput(tag="out_gc_context", output_type=File(optional=True), selector=InputSelector(input_to_select="in_gc_context", type_hint=File()), doc=OutputDocumentation(doc="In addition to normal processing this option will reprocess the genome to find methylation in\nGpC context. This might be useful for specialist applications where GpC methylases had been\ndeployed. The output format is exactly the same as for the normal cytosine report, and only\npositions covered by at least one read are reported (output file ends in .GpC_report.txt). In addition\nthis will write out a Bismark coverage file (ending in GpC.cov).")), ToolOutput(tag="out_gzip", output_type=File(optional=True), selector=InputSelector(input_to_select="in_gzip", type_hint=File()), doc=OutputDocumentation(doc="Output file will be GZIP compressed (ending in .gz). Only works for standard CpG- and CX-output."))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Coverage2Cytosine_V0_1_0().translate("wdl", allow_empty_container=True)

