class: CommandLineTool
id: ../../../LiBis.cwl
inputs:
- id: in_file
  doc: "Enter a number, 0 means using parameter to set up, 1\nmeans using text file\
    \ to set up"
  type: File
  inputBinding:
    prefix: --file
- id: in_setting_file
  doc: Setting txt file name. Ignore if -f is 0.
  type: File
  inputBinding:
    prefix: --settingfile
- id: in_required_fastq_file
  doc: "[NAME [NAME ...]], --name [NAME [NAME ...]]\nRequired. Fastq file name."
  type: boolean
  inputBinding:
    prefix: -n
- id: in_clip
  doc: 'Clip mode. 0 means close. 1 means open. default: 0'
  type: long
  inputBinding:
    prefix: --clip
- id: in_labels_for_samples
  doc: "[LABEL [LABEL ...]], --label [LABEL [LABEL ...]]\nLabels for samples"
  type: boolean
  inputBinding:
    prefix: -l
- id: in_genome
  doc: "Genome the reference belong to.(Use for plotting)\nhg18/hg19/mm10/mm9 and\
    \ so on. Plotting script will not\navaliable if leave it blank. default=hg38."
  type: long
  inputBinding:
    prefix: --genome
- id: in_window
  doc: Window length for clipping mode, default=30
  type: long
  inputBinding:
    prefix: --window
- id: in_step
  doc: Step size for clipping mode.
  type: long
  inputBinding:
    prefix: --step
- id: in_process
  doc: "Process using for one pipeline. Normally bsmap will\ncost 8 cpu number. So\
    \ total will be 8p."
  type: long
  inputBinding:
    prefix: --process
- id: in_ref
  doc: Required. Reference genome file name.
  type: File
  inputBinding:
    prefix: --ref
- id: in_quality_control
  doc: Do quality control
  type: boolean
  inputBinding:
    prefix: --QualityControl
- id: in_trim
  doc: Do trimming.
  type: boolean
  inputBinding:
    prefix: --trim
- id: in_binsize
  doc: "Plot setting. Set the bin size for averaging\nmethylation ratio among samples,\
    \ default=1000k"
  type: long
  inputBinding:
    prefix: --binsize
- id: in_filter
  doc: Minimal length for recombined reads
  type: long
  inputBinding:
    prefix: --filter
- id: in_bam
  doc: "Processed bam file for the first step. If bam files\nare offered here, the\
    \ first step of bsmap will be\nskipped. BAM files can only be generated by BSMAP.\n\
    Different files should be seperated by ','. If there's\nno bam file for part of\
    \ the samples, leave the space\nblank. For example: a.bam,b.bam,,,,f.bam"
  type: File
  inputBinding:
    prefix: --bam
- id: in_m_call
  doc: Run mcall for mapped bams or not
  type: boolean
  inputBinding:
    prefix: --mcall
- id: in_plot
  doc: Generate the final report or not
  type: boolean
  inputBinding:
    prefix: --plot
- id: in_no_check
  doc: "Skip the checking step for result folders. Using this\nparameter may rewrite\
    \ the previous results."
  type: boolean
  inputBinding:
    prefix: --nocheck
- id: in_moab_s_module
  doc: "Run LiBis as MOABS module. Please only apply this when\nintegrating LiBis\
    \ to MOABS."
  type: boolean
  inputBinding:
    prefix: --MOABSmodule
- id: in_multiple_mapping
  doc: Apply multiple mapping after first around bsmap
  type: boolean
  inputBinding:
    prefix: --multiplemapping
- id: in_pair_end_filter
  doc: "Apply pairend filter for all LiBis rescued reads. Only\nkeep reads have uniquely\
    \ mapped pair by BSMAP or\nLiBis."
  type: boolean
  inputBinding:
    prefix: --pairendfilter
- id: in_bam_u
  doc: "Given processed bam file contains unmapped reads. Only\nuse with -bam is open."
  type: boolean
  inputBinding:
    prefix: --bamu
- id: in_gzip
  doc: Temporary files are in gz format.
  type: boolean
  inputBinding:
    prefix: --gzip
- id: in_full_mode
  doc: Keep all temp files.
  type: boolean
  inputBinding:
    prefix: --fullmode
- id: in_name
  doc: ''
  type: string
  inputBinding:
    position: 0
outputs:
- id: out_stdout
  doc: Standard output stream
  type: stdout
cwlVersion: v1.1
baseCommand:
- LiBis
