#!/usr/bin/env cwl-runner

baseCommand:
- LiBis
class: CommandLineTool
cwlVersion: v1.0
id: libis
inputs:
- doc: Enter a number, 0 means using parameter to set up, 1 means using text file
    to set up
  id: file
  inputBinding:
    prefix: --file
  type: File
- doc: Setting txt file name. Ignore if -f is 0.
  id: setting_file
  inputBinding:
    prefix: --settingfile
  type: string
- doc: '[NAME [NAME ...]], --name [NAME [NAME ...]] Required. Fastq file name.'
  id: n
  inputBinding:
    prefix: -n
  type: boolean
- doc: 'Clip mode. 0 means close. 1 means open. default: 0'
  id: clip
  inputBinding:
    prefix: --clip
  type: string
- doc: '[LABEL [LABEL ...]], --label [LABEL [LABEL ...]] Labels for samples'
  id: l
  inputBinding:
    prefix: -l
  type: boolean
- doc: Genome the reference belong to.(Use for plotting) hg18/hg19/mm10/mm9 and so
    on. Plotting script will not avaliable if leave it blank. default=hg38.
  id: genome
  inputBinding:
    prefix: --genome
  type: string
- doc: Window length for clipping mode, default=30
  id: window
  inputBinding:
    prefix: --window
  type: string
- doc: Step size for clipping mode.
  id: step
  inputBinding:
    prefix: --step
  type: string
- doc: Process using for one pipeline. Normally bsmap will cost 8 cpu number. So total
    will be 8p.
  id: process
  inputBinding:
    prefix: --process
  type: string
- doc: Required. Reference genome file name.
  id: ref
  inputBinding:
    prefix: --ref
  type: string
- doc: Do quality control
  id: quality_control
  inputBinding:
    prefix: --QualityControl
  type: boolean
- doc: Do trimming.
  id: trim
  inputBinding:
    prefix: --trim
  type: boolean
- doc: Plot setting. Set the bin size for averaging methylation ratio among samples,
    default=1000k
  id: binsize
  inputBinding:
    prefix: --binsize
  type: string
- doc: Minimal length for recombined reads
  id: filter
  inputBinding:
    prefix: --filter
  type: string
- doc: "Processed bam file for the first step. If bam files are offered here, the\
    \ first step of bsmap will be skipped. BAM files can only be generated by BSMAP.\
    \ Different files should be seperated by ','. If there's no bam file for part\
    \ of the samples, leave the space blank. For example: a.bam,b.bam,,,,f.bam"
  id: bam
  inputBinding:
    prefix: --bam
  type: string
- doc: Run mcall for mapped bams or not
  id: m_call
  inputBinding:
    prefix: --mcall
  type: boolean
- doc: Generate the final report or not
  id: plot
  inputBinding:
    prefix: --plot
  type: boolean
- doc: Skip the checking step for result folders. Using this parameter may rewrite
    the previous results.
  id: no_check
  inputBinding:
    prefix: --nocheck
  type: boolean
- doc: Run LiBis as MOABS module. Please only apply this when integrating LiBis to
    MOABS.
  id: moab_s_module
  inputBinding:
    prefix: --MOABSmodule
  type: boolean
- doc: Apply multiple mapping after first around bsmap
  id: multiple_mapping
  inputBinding:
    prefix: --multiplemapping
  type: boolean
- doc: Apply pairend filter for all LiBis rescued reads. Only keep reads have uniquely
    mapped pair by BSMAP or LiBis.
  id: pair_end_filter
  inputBinding:
    prefix: --pairendfilter
  type: boolean
- doc: Given processed bam file contains unmapped reads. Only use with -bam is open.
  id: bam_u
  inputBinding:
    prefix: --bamu
  type: boolean
- doc: Temporary files are in gz format.
  id: gzip
  inputBinding:
    prefix: --gzip
  type: boolean
- doc: Keep all temp files.
  id: full_mode
  inputBinding:
    prefix: --fullmode
  type: boolean
