from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, Int, File, Float

Mason_Methylation_V0_1_0 = CommandToolBuilder(tool="mason_methylation", base_command=["mason_methylation"], inputs=[ToolInput(tag="in_version_check", input_type=Boolean(optional=True), prefix="--version-check", doc=InputDocumentation(doc="Turn this option off to disable version update notifications of the\napplication. One of 1, ON, TRUE, T, YES, 0, OFF, FALSE, F, and NO.\nDefault: 1.")), ToolInput(tag="in_quiet", input_type=Boolean(optional=True), prefix="--quiet", doc=InputDocumentation(doc="Low verbosity.")), ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="--verbose", doc=InputDocumentation(doc="Higher verbosity.")), ToolInput(tag="in_very_verbose", input_type=Boolean(optional=True), prefix="--very-verbose", doc=InputDocumentation(doc="Highest verbosity.")), ToolInput(tag="in_seed", input_type=Int(optional=True), prefix="--seed", doc=InputDocumentation(doc="Seed for RNG. Default: 0.")), ToolInput(tag="in_in", input_type=File(optional=True), prefix="--in", doc=InputDocumentation(doc="Input FASTA file with genome. Valid filetypes are: .sam[.*],\n.raw[.*], .gbk[.*], .frn[.*], .fq[.*], .fna[.*], .ffn[.*],\n.fastq[.*], .fasta[.*], .faa[.*], .fa[.*], .embl[.*], and .bam,\nwhere * is any of the following extensions: gz and bgzf for\ntransparent (de)compression.")), ToolInput(tag="in_out", input_type=File(optional=True), prefix="--out", doc=InputDocumentation(doc="Input FASTA file with genome. Valid filetypes are: .sam[.*],\n.raw[.*], .frn[.*], .fq[.*], .fna[.*], .ffn[.*], .fastq[.*],\n.fasta[.*], .faa[.*], .fa[.*], and .bam, where * is any of the\nfollowing extensions: gz and bgzf for transparent (de)compression.")), ToolInput(tag="in_methylation_levels", input_type=Boolean(optional=True), prefix="--methylation-levels", doc=InputDocumentation(doc="Enable methylation level simulation.")), ToolInput(tag="in_meth_cg_mu", input_type=Float(optional=True), prefix="--meth-cg-mu", doc=InputDocumentation(doc="Median of beta distribution for methylation level of CpG loci. In\nrange [0..1]. Default: 0.6.")), ToolInput(tag="in_meth_cg_sigma", input_type=Float(optional=True), prefix="--meth-cg-sigma", doc=InputDocumentation(doc="Standard deviation of beta distribution for methylation level of CpG\nloci. In range [0..1]. Default: 0.03.")), ToolInput(tag="in_meth_chg_mu", input_type=Float(optional=True), prefix="--meth-chg-mu", doc=InputDocumentation(doc="Median of beta distribution for methylation level of CHG loci. In\nrange [0..1]. Default: 0.08.")), ToolInput(tag="in_meth_chg_sigma", input_type=Float(optional=True), prefix="--meth-chg-sigma", doc=InputDocumentation(doc="Standard deviation of beta distribution for methylation level of CHG\nloci. In range [0..1]. Default: 0.008.")), ToolInput(tag="in_meth_chh_mu", input_type=Float(optional=True), prefix="--meth-chh-mu", doc=InputDocumentation(doc="Median of beta distribution for methylation level of CHH loci. In\nrange [0..1]. Default: 0.05.")), ToolInput(tag="in_meth_chh_sigma", input_type=Float(optional=True), prefix="--meth-chh-sigma", doc=InputDocumentation(doc="Standard deviation of beta distribution for methylation level of CHH\nloci. In range [0..1]. Default: 0.005."))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Mason_Methylation_V0_1_0().translate("wdl", allow_empty_container=True)

