from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import String, Boolean, Int, File, Float

Instrain_Profile_V0_1_0 = CommandToolBuilder(tool="inStrain_profile", base_command=["inStrain", "profile"], inputs=[ToolInput(tag="in_output", input_type=String(optional=True), prefix="--output", doc=InputDocumentation(doc="Output prefix (default: inStrain)")), ToolInput(tag="in_use_full_fast_a_header", input_type=Boolean(optional=True), prefix="--use_full_fasta_header", doc=InputDocumentation(doc="Instead of using the fasta ID (space in header before\nspace), use the full header. Needed for some mapping\ntools (including bbMap) (default: False)")), ToolInput(tag="in_force_compress", input_type=Boolean(optional=True), prefix="--force_compress", doc=InputDocumentation(doc="Force compression of all output files (default: False)")), ToolInput(tag="in_processes", input_type=Int(optional=True), prefix="--processes", doc=InputDocumentation(doc="Number of processes to use (default: 6)")), ToolInput(tag="in_debug", input_type=Boolean(optional=True), prefix="--debug", doc=InputDocumentation(doc="Make extra debugging output (default: False)")), ToolInput(tag="in_min_read_ani", input_type=Int(optional=True), prefix="--min_read_ani", doc=InputDocumentation(doc="Minimum percent identity of read pairs to consensus to\nuse the reads. Must be >, not >= (default: 0.95)")), ToolInput(tag="in_min_mapq", input_type=Int(optional=True), prefix="--min_mapq", doc=InputDocumentation(doc="Minimum mapq score of EITHER read in a pair to use\nthat pair. Must be >, not >= (default: -1)")), ToolInput(tag="in_max_insert_relative", input_type=Int(optional=True), prefix="--max_insert_relative", doc=InputDocumentation(doc="Multiplier to determine maximum insert size between\ntwo reads - default is to use 3x median insert size.\nMust be >, not >= (default: 3)")), ToolInput(tag="in_min_insert", input_type=Int(optional=True), prefix="--min_insert", doc=InputDocumentation(doc="Minimum insert size between two reads - default is 50\nbp. If two reads are 50bp each and overlap completely,\ntheir insert will be 50. Must be >, not >= (default:\n50)")), ToolInput(tag="in_pairing_filter", input_type=String(optional=True), prefix="--pairing_filter", doc=InputDocumentation(doc="How should paired reads be handled?\npaired_only = Only paired reads are retained\nnon_discordant = Keep all paired reads and singleton reads that map to a single scaffold\nall_reads = Keep all reads regardless of pairing status (NOT RECOMMENDED; See documentation for deatils)\n(default: paired_only)")), ToolInput(tag="in_priority_reads", input_type=File(optional=True), prefix="--priority_reads", doc=InputDocumentation(doc="The location of a list of reads that should be\nretained regardless of pairing status (for example\nlong reads or merged reads). This can be a .fastq file\nor text file with list of read names (will assume file\nis compressed if ends in .gz (default: None)")), ToolInput(tag="in_detailed_mapping_info", input_type=Boolean(optional=True), prefix="--detailed_mapping_info", doc=InputDocumentation(doc="Make a detailed read report indicating deatils about\neach individual mapped read (default: False)")), ToolInput(tag="in_min_cov", input_type=Int(optional=True), prefix="--min_cov", doc=InputDocumentation(doc="Minimum coverage to call an variant (default: 5)")), ToolInput(tag="in_min_freq", input_type=Int(optional=True), prefix="--min_freq", doc=InputDocumentation(doc="Minimum SNP frequency to confirm a SNV (both this AND\nthe FDR snp count cutoff must be true to call a SNP).\n(default: 0.05)")), ToolInput(tag="in_fdr", input_type=Float(optional=True), prefix="--fdr", doc=InputDocumentation(doc="SNP false discovery rate- based on simulation data\nwith a 0.1 percent error rate (Q30) (default: 1e-06)")), ToolInput(tag="in_gene_file", input_type=File(optional=True), prefix="--gene_file", doc=InputDocumentation(doc="Path to prodigal .fna genes file. If file ends in .gb\nor .gbk, will treat as a genbank file (EXPERIMENTAL;\nthe name of the gene must be in the gene qualifier)\n(default: None)")), ToolInput(tag="in_scaffold_bin_be", input_type=Boolean(optional=True), prefix="-s", doc=InputDocumentation(doc="[STB [STB ...]], --stb [STB [STB ...]]\nScaffold to bin. This can be a file with each line\nlisting a scaffold and a bin name, tab-seperated. This\ncan also be a space-seperated list of .fasta files,\nwith one genome per .fasta file. If nothing is\nprovided, all scaffolds will be treated as belonging\nto the same genome (default: [])")), ToolInput(tag="in_mm_level", input_type=Boolean(optional=True), prefix="--mm_level", doc=InputDocumentation(doc="Create output files on the mm level (see documentation\nfor info) (default: False)")), ToolInput(tag="in_skip_mm_profiling", input_type=Boolean(optional=True), prefix="--skip_mm_profiling", doc=InputDocumentation(doc="Dont perform analysis on an mm level; saves RAM and\ntime; impacts plots and raw_data (default: False)")), ToolInput(tag="in_database_mode", input_type=Boolean(optional=True), prefix="--database_mode", doc=InputDocumentation(doc="Set a number of parameters to values appropriate for\nmapping to a large fasta file. Will set:\n--min_read_ani 0.92 --skip_mm_profiling\n--min_genome_coverage 1 (default: False)")), ToolInput(tag="in_min_scaffold_reads", input_type=Int(optional=True), prefix="--min_scaffold_reads", doc=InputDocumentation(doc="Minimum number of reads mapping to a scaffold to\nproceed with profiling it (default: 1)")), ToolInput(tag="in_min_genome_coverage", input_type=Int(optional=True), prefix="--min_genome_coverage", doc=InputDocumentation(doc="Minimum number of reads mapping to a genome to proceed\nwith profiling it. MUST profile .stb if this is set\n(default: 0)")), ToolInput(tag="in_min_snp", input_type=Int(optional=True), prefix="--min_snp", doc=InputDocumentation(doc="Absolute minimum number of reads connecting two SNPs\nto calculate LD between them. (default: 20)")), ToolInput(tag="in_store_everything", input_type=Boolean(optional=True), prefix="--store_everything", doc=InputDocumentation(doc="Store intermediate dictionaries in the pickle file;\nwill result in significantly more RAM and disk usage\n(default: False)")), ToolInput(tag="in_scaffolds_to_profile", input_type=File(optional=True), prefix="--scaffolds_to_profile", doc=InputDocumentation(doc="Path to a file containing a list of scaffolds to\nprofile- if provided will ONLY profile those scaffolds\n(default: None)")), ToolInput(tag="in_rarefied_coverage", input_type=Int(optional=True), prefix="--rarefied_coverage", doc=InputDocumentation(doc="When calculating nucleotide diversity, also calculate\na rarefied version with this much coverage (default:\n50)")), ToolInput(tag="in_window_length", input_type=Int(optional=True), prefix="--window_length", doc=InputDocumentation(doc="Break scaffolds into windows of this length when\nprofiling (default: 10000)")), ToolInput(tag="in_skip_genome_wide", input_type=Boolean(optional=True), prefix="--skip_genome_wide", doc=InputDocumentation(doc="Do not generate tables that consider groups of\nscaffolds belonging to genomes (default: False)")), ToolInput(tag="in_skip_plot_generation", input_type=Boolean(optional=True), prefix="--skip_plot_generation", doc=InputDocumentation(doc="Do not make plots (default: False)\n")), ToolInput(tag="in_bam", input_type=String(), position=0, doc=InputDocumentation(doc="Sorted .bam file")), ToolInput(tag="in_fast_a", input_type=String(), position=1, doc=InputDocumentation(doc="Fasta file the bam is mapped to"))], outputs=[], container="quay.io/biocontainers/instrain:1.5.3--py_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Instrain_Profile_V0_1_0().translate("wdl")

