from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, String, Int, Float, File

Instrain_Compare_V0_1_0 = CommandToolBuilder(tool="inStrain_compare", base_command=["inStrain", "compare"], inputs=[ToolInput(tag="in_list_instrain_none", input_type=Boolean(optional=True), prefix="-i", doc=InputDocumentation(doc="[INPUT [INPUT ...]], --input [INPUT [INPUT ...]]\nA list of inStrain objects, all mapped to the same\n.fasta file (default: None)")), ToolInput(tag="in_output", input_type=String(optional=True), prefix="--output", doc=InputDocumentation(doc="Output prefix (default: instrainComparer)")), ToolInput(tag="in_processes", input_type=Int(optional=True), prefix="--processes", doc=InputDocumentation(doc="Number of processes to use (default: 6)")), ToolInput(tag="in_debug", input_type=Boolean(optional=True), prefix="--debug", doc=InputDocumentation(doc="Make extra debugging output (default: False)")), ToolInput(tag="in_min_cov", input_type=Int(optional=True), prefix="--min_cov", doc=InputDocumentation(doc="Minimum coverage to call an variant (default: 5)")), ToolInput(tag="in_min_freq", input_type=Int(optional=True), prefix="--min_freq", doc=InputDocumentation(doc="Minimum SNP frequency to confirm a SNV (both this AND\nthe FDR snp count cutoff must be true to call a SNP).\n(default: 0.05)")), ToolInput(tag="in_fdr", input_type=Float(optional=True), prefix="--fdr", doc=InputDocumentation(doc="SNP false discovery rate- based on simulation data\nwith a 0.1 percent error rate (Q30) (default: 1e-06)")), ToolInput(tag="in_scaffolds", input_type=File(optional=True), prefix="--scaffolds", doc=InputDocumentation(doc="Location to a list of scaffolds to compare. You can\nalso make this a .fasta file and it will load the\nscaffold names (default: None)")), ToolInput(tag="in_store_coverage_overlap", input_type=Boolean(optional=True), prefix="--store_coverage_overlap", doc=InputDocumentation(doc="Also store coverage overlap on an mm level (default:\nFalse)")), ToolInput(tag="in_store_mismatch_locations", input_type=Boolean(optional=True), prefix="--store_mismatch_locations", doc=InputDocumentation(doc="Store the locations of SNPs (default: False)")), ToolInput(tag="in_include_self_comparisons", input_type=Boolean(optional=True), prefix="--include_self_comparisons", doc=InputDocumentation(doc="Also compare IS profiles against themself (default:\nFalse)")), ToolInput(tag="in_greedy_clustering", input_type=Boolean(optional=True), prefix="--greedy_clustering", doc=InputDocumentation(doc="Dont do pair-wise comparisons, do greedy clustering to\nonly find the number of clsuters. If this is set, use\nthe parameters below as well (default: False)")), ToolInput(tag="in_g_ani", input_type=Float(optional=True), prefix="--g_ani", doc=InputDocumentation(doc="ANI threshold for greedy clustering- put the fraction\nnot the percentage (e.g. 0.99, not 99) (default: 0.99)")), ToolInput(tag="in_g_cov", input_type=Float(optional=True), prefix="--g_cov", doc=InputDocumentation(doc="Alignment coverage for greedy clustering- put the\nfraction not the percentage (e.g. 0.5, not 10)\n(default: 0.99)")), ToolInput(tag="in_g_mm", input_type=Int(optional=True), prefix="--g_mm", doc=InputDocumentation(doc="Maximum read mismatch level (default: 100)")), ToolInput(tag="in_input", input_type=String(optional=True), position=0, doc=InputDocumentation(doc=""))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Instrain_Compare_V0_1_0().translate("wdl", allow_empty_container=True)

