class: CommandLineTool
id: mamotif.cwl
inputs:
- id: p1
  doc: 'Path of peaks file of sample 1. BED and MACS format are currently supported.Please
    refer to documents for details. (default: None)'
  type: string
  inputBinding:
    prefix: --p1
- id: p2
  doc: 'Path of peaks file of sample 2. (default: None)'
  type: string
  inputBinding:
    prefix: --p2
- id: r1
  doc: 'Path of reads file of sample 1. BED format are currently supported. (default:
    None)'
  type: string
  inputBinding:
    prefix: --r1
- id: r2
  doc: 'Path of reads file of sample 2. (default: None)'
  type: string
  inputBinding:
    prefix: --r2
- id: s1
  doc: "Reads shiftsize of sample 1. This value is used to shift reads towards 3'\
    \ directionto account for the actual binding site. Set as half of the fragment\
    \ length. (default: 100)"
  type: string
  inputBinding:
    prefix: --s1
- id: s2
  doc: 'Reads shiftsize of sample 2. (default: 100)'
  type: string
  inputBinding:
    prefix: --s2
- id: motif
  doc: "Required pre-compiled motif file, which is generated by 'motifcompile'. (default:\
    \ None)"
  type: string
  inputBinding:
    prefix: --motif
- id: genome
  doc: "Required genome directory, which contains genome sequence and chromosome length.\
    \ You can run 'genomecompile' to generate the directory. (default: None)"
  type: string
  inputBinding:
    prefix: --genome
- id: annotation
  doc: 'Gene annotation file, which is used to generate random controls when performing
    enrichment analysis. Sequence features on promoters are very different from those
    on whole genome. We will pick up random control regions with similar genomic locations
    for each input region instead of the whole genome. (default: None)'
  type: string
  inputBinding:
    prefix: --annotation
- id: w
  doc: 'Half width of the window size when calculating read densities. Each window
    with length of 2*width is centered at peak summit or midpoint. This value should
    match the typical length of peaks, thus we recommend 1000 for sharp histone marks
    like H3K4me3 and H3K9/27ac, or 500 for transcription factors or DNase- Seq. (default:
    1000)'
  type: string
  inputBinding:
    prefix: -w
- id: d
  doc: 'Summit to summit distance cutoff for common peaks. Default=width/2. Only peaks
    with summit distance less than than this value are considered as real common peaks
    of two samples. (default: None)'
  type: string
  inputBinding:
    prefix: -d
- id: n
  doc: 'Times of permutation to test the enrichment of peak overlap between two samples.
    (default: 5)'
  type: string
  inputBinding:
    prefix: -n
- id: p
  doc: 'P-value cutoff to define biased (sample 1/2-specific) peaks. (default: 0.01)'
  type: string
  inputBinding:
    prefix: -p
- id: m_cut_off
  doc: 'M-value cutoff to distinguish biased peaks from unbiased peaks. Peaks with
    M-value>=M_cutoff and P-value<=P_cutoff as defined as sample1-biased(specific)
    peaks, while peaks with M-value<=-1*M_cutoff and P-value<=P_cutoff are defined
    as sample2-biased peaks. (default: 1.0)'
  type: string
  inputBinding:
    prefix: --m_cutoff
- id: motif_list
  doc: 'Motif list file. If specified, the program will only perform MotifScan on
    given motifs in this list. (default: None)'
  type: string
  inputBinding:
    prefix: --motif_list
- id: no_enrichment
  doc: 'If set, MotifScan will neither generate random control regions nor perform
    enrichment analysis. (default: True)'
  type: boolean
  inputBinding:
    prefix: --no_enrichment
- id: r
  doc: "Genomic location to apply MotifScan. 'all': perform MotifScan on the all input\
    \ regions. 'promoter': perform MotifScan on regions which located at promoter\
    \ regions. 'distal': perform MotifScan on regions which located at distal regions.\
    \ 'gene': perform MotifScan on regions which is the target(nearest one) of genes\
    \ within 10kbp distance cutoff. (default: all)"
  type: string
  inputBinding:
    prefix: -r
- id: upstream
  doc: "Upstream distance to TSS to define promoter regions. Valid when option -r\
    \ is set as 'promoter' or 'distal'. (default: 4000)"
  type: string
  inputBinding:
    prefix: --upstream
- id: downstream
  doc: "Upstream distance to TSS to define promoter regions. Valid when option -r\
    \ is set as 'promoter' or 'distal'. (default: 2000)"
  type: string
  inputBinding:
    prefix: --downstream
- id: random_times
  doc: 'The number of random control regions generated for each input region. For
    example, if you provide 1000 regions, 5000 random regions will be generated by
    default. (default: 5)'
  type: string
  inputBinding:
    prefix: --random_times
- id: peak_length
  doc: 'The length of input regions to perform MotifScan around peak summit/midpoint.
    If set to 0, the whole peak region is used to perform MotifScan. (default: 1000)'
  type: string
  inputBinding:
    prefix: --peak_length
- id: negative
  doc: 'Using negative test of this pk (default: False)'
  type: boolean
  inputBinding:
    prefix: --negative
- id: correction
  doc: 'correction type of pvalues, no correction or benjamin or bonferroni,default=benjamin
    (default: benjamin)'
  type: string
  inputBinding:
    prefix: --correction
- id: s
  doc: 'By default, MAnorm will output the results of unique and merged common peaks
    of two samples. With this option on, MAnorm will output two extra files containing
    the results of the original(unmerged) peaks in two samples. With this option on,
    MAmotif will also output the exact target sites of all motifs. (default: False)'
  type: boolean
  inputBinding:
    prefix: -s
- id: o
  doc: 'Comparison name, this is used as the folder name and prefix of output files.
    (default: None)'
  type: string
  inputBinding:
    prefix: -o
outputs: []
cwlVersion: v1.1
baseCommand:
- mamotif
