from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, String, Float, Int, Boolean

Methylpy_Test_Allc_V0_1_0 = CommandToolBuilder(tool="methylpy_test_allc", base_command=["methylpy", "test-allc"], inputs=[ToolInput(tag="in_all_c_file", input_type=File(optional=True), prefix="--allc-file", doc=InputDocumentation(doc="allc file to be tested. (default: None)")), ToolInput(tag="in_sample", input_type=String(optional=True), prefix="--sample", doc=InputDocumentation(doc="sample name (default: None)")), ToolInput(tag="in_unmethylated_control", input_type=Float(optional=True), prefix="--unmethylated-control", doc=InputDocumentation(doc="name of the chromosome/region that you want to use to\nestimate the non-conversion rate of your sample, or\nthe non-conversion rate you would like to use.\nConsequently, control is either a string, or a\ndecimal. If control is a string then it should be in\nthe following format: 'chrom:start-end'. If you would\nlike to specify an entire chromosome simply use\n'chrom:' (default: None)")), ToolInput(tag="in_path_to_output", input_type=File(optional=True), prefix="--path-to-output", doc=InputDocumentation(doc="Path to a directory where you would like the output to\nbe stored. The default is the same directory as the\ninput fastqs. (default: )")), ToolInput(tag="in_num_procs", input_type=Int(optional=True), prefix="--num-procs", doc=InputDocumentation(doc="Number of processors you wish to use to parallelize\nthis function (default: 1)")), ToolInput(tag="in_min_cov", input_type=Int(optional=True), prefix="--min-cov", doc=InputDocumentation(doc="Minimum number of reads that must cover a site for it\nto be tested. (default: 2)")), ToolInput(tag="in_compress_output", input_type=Boolean(optional=True), prefix="--compress-output", doc=InputDocumentation(doc="Boolean indicating whether to compress (by gzip) the\nfinal output (default: True)")), ToolInput(tag="in_sig_cut_off", input_type=Float(optional=True), prefix="--sig-cutoff", doc=InputDocumentation(doc="Float indicating at what FDR you want to consider a\nresult significant. (default: 0.01)")), ToolInput(tag="in_sort_mem", input_type=Int(optional=True), prefix="--sort-mem", doc=InputDocumentation(doc="Parameter to pass to unix sort with -S/--buffer-size\ncommand (default: 500M)")), ToolInput(tag="in_remove_chr_prefix", input_type=Boolean(optional=True), prefix="--remove-chr-prefix", doc=InputDocumentation(doc="Boolean indicates whether to remove in the final\noutput the 'chr' prefix in the chromosome name\n(default: True)\n"))], outputs=[ToolOutput(tag="out_path_to_output", output_type=File(optional=True), selector=InputSelector(input_to_select="in_path_to_output", type_hint=File()), doc=OutputDocumentation(doc="Path to a directory where you would like the output to\nbe stored. The default is the same directory as the\ninput fastqs. (default: )"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Methylpy_Test_Allc_V0_1_0().translate("wdl", allow_empty_container=True)

