from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import String, Boolean, Directory, Array, File

Vtools_Admin_V0_1_0 = CommandToolBuilder(tool="vtools_admin", base_command=["vtools", "admin"], inputs=[ToolInput(tag="in_verbosity", input_type=String(optional=True), prefix="--verbosity", doc=InputDocumentation(doc="Output error and warning (0), info (1), debug (2) and\ntrace (3) information to standard output (default to\n1).")), ToolInput(tag="in_update_resource", input_type=Boolean(optional=True), prefix="--update_resource", doc=InputDocumentation(doc="[TYPE], --update-resource [TYPE]\nDownload resources of specified type, which can be\n'current' (latest version of all resources), 'all'\n(all resources including obsolete databases),\n'existing' (only update resources that exist locally),\n'hg18' or 'hg19' (all resources for reference genome\nhg18 or hg19), 'annotation' (all current annotation\ndatabases), 'format' (all formats), and 'snapshot'\n(all online snapshots). Identical resources that are\navailable locally (under ~/.variant_tools or runtime\noption $local_resource) are ignored. Note that option\n'all' will download all versions of annotation\ndatabases which can be slow and take a lot of disk\nspaces.")), ToolInput(tag="in_mirror_repository", input_type=Directory(optional=True), prefix="--mirror_repository", doc=InputDocumentation(doc="Mirror the main variant tools repository to a local\ndirectory. This command will check files under dest,\ndownload all missing or outdated files. Existing files\nthat do not belong to the repository will not be\nremoved.")), ToolInput(tag="in_merge_samples", input_type=Boolean(optional=True), prefix="--merge_samples", doc=InputDocumentation(doc="Merge samples with the same sample names by combining\ngenotypes belonging to these samples. Phenotypes\nrelated to individual samples will be merged.")), ToolInput(tag="in_rename_samples", input_type=Array(t=String(), optional=True), prefix="--rename_samples", doc=InputDocumentation(doc="This argument takes a condition by which samples are\nselected, followed by either a new sample name (assign\na new name to selected samples) or an OLDNAME NEWNAME\npair of patterns for which the first instance of\nOLDNAME in sample names will be replaced by NEWNAME.")), ToolInput(tag="in_rename_table", input_type=String(optional=True), prefix="--rename_table", doc=InputDocumentation(doc="NEW_NAME, --rename-table NAME NEW_NAME\nChange table NAME to a NEW_NAME.")), ToolInput(tag="in_describe_table", input_type=String(optional=True), prefix="--describe_table", doc=InputDocumentation(doc="NEW_DESCRIPTION, --describe-table TABLE NEW_DESCRIPTION\nUpdate description for TABLE with a NEW_DESCRIPTION.")), ToolInput(tag="in_validate_build", input_type=Boolean(optional=True), prefix="--validate_build", doc=InputDocumentation(doc="Validate the build of project's reference genome by\nchecking if the reference alleles of variants agree\nwith the reference genome sequence. A reference genome\nwill be automatically downloaded if it does not exist\nin the local resource directory.")), ToolInput(tag="in_validate_sex", input_type=Boolean(optional=True), prefix="--validate_sex", doc=InputDocumentation(doc="[{report,force-heterozygote}], --validate-sex [{report,force-heterozygote}]\nValidate the sex of samples by checking the genotypes\nof samples on sex chromosomes (excluding pseudo-\nautosomal regions). Sex of samples are determined by a\nphenotype named sex or gender with values 1/2, M/F or\nMale/Female. Inconsistency will be reported if, for\nexample, a female sample has genotypes on chromosome\nY. This argument accepts an option argument, which is\nreport by default (report inconsistent genotype or\nsex), but can also be 'force-heterozygote' for male\nindividuals on chromosome X.")), ToolInput(tag="in_save_snapshot", input_type=File(optional=True), prefix="--save_snapshot", doc=InputDocumentation(doc="MESSAGE, --save-snapeshot NAME MESSAGE\nCreate a snapshot of the current project with NAME,\nwhich could be re-loaded using command 'vtools admin\n--load_snapshot'. A filename with extension .tar, .tgz\nor .tar.gz can be used to save the snapshot to a\nspecific directory with compression but such snapshots\nare not listed by command 'vtools show snapshots'.")), ToolInput(tag="in_extra_files", input_type=Boolean(optional=True), prefix="--extra_files", doc=InputDocumentation(doc="[FILE [FILE ...]], --extra-files [FILE [FILE ...]]\nAdditional files that will be saved along with the\nproject and genotype databases. This could include\ncustomized format files, project-specific annotations,\nand results. Files outside of the current project\ndirectory are not allowed due to security\nconsiderations.")), ToolInput(tag="in_load_snapshot", input_type=File(optional=True), prefix="--load_snapshot", doc=InputDocumentation(doc="Revert the current project to specified snapshot. All\nchanges since the that snapshot will be lost. The NAME\nshould be one of the project snapshots or online\nsnapshots listed by command 'vtools show snapshots',\nor name of a local snapshot file (with extension .tar,\n.tgz or .tar.gz).")), ToolInput(tag="in_set_runtime_option", input_type=Array(t=String(), optional=True), prefix="--set_runtime_option", doc=InputDocumentation(doc="Set value to internal options such as the batch size\nfor database options. The default values of these\noptions were chosen to fit most usage patterns but\ntweaking them might provide better performance under\ncertain circumstances. Please use command 'vtools show\nruntime_options' to list all currently supported\nruntime options.")), ToolInput(tag="in_reset_runtime_option", input_type=String(optional=True), prefix="--reset_runtime_option", doc=InputDocumentation(doc="Reset value to a runtime option to its default value.")), ToolInput(tag="in_global", input_type=Boolean(optional=True), prefix="--global", doc=InputDocumentation(doc="Save option to user_options.py so that it will be\nautomatically set for all user projects.")), ToolInput(tag="in_fast_a_two_crr", input_type=Array(t=String(), optional=True), prefix="--fasta2crr", doc=InputDocumentation(doc="Convert fasta files to a crr file (a binary format for\nfaster access) that can be used by variant tools. This\nis only needed if you are working with a reference\ngenome that is not supported by variant tools. This\nparameter accepts a list of fastq files (URLs and .gz\nformat are acceptable) followed by the name of the\n.crr file. The .crr file should be put under the\nproject directory or the local resource directory\n(under directory reference) to be usable by variant\ntools.\n")), ToolInput(tag="in_type", input_type=String(optional=True), position=0, doc=InputDocumentation(doc=""))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Vtools_Admin_V0_1_0().translate("wdl", allow_empty_container=True)

