from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, File

Hts_Seqscreener_V0_1_0 = CommandToolBuilder(tool="hts_SeqScreener", base_command=["hts_SeqScreener"], inputs=[ToolInput(tag="in__version_print", input_type=Boolean(optional=True), prefix="-v", doc=InputDocumentation(doc="[ --version ]                      Version print")), ToolInput(tag="in_arg_notes_json", input_type=Boolean(optional=True), prefix="-N", doc=InputDocumentation(doc="[ --notes ] arg                    Notes for the stats JSON")), ToolInput(tag="in_arg_statslog_write", input_type=Boolean(optional=True), prefix="-L", doc=InputDocumentation(doc="[ --stats-file ] arg (=stats.log)  Write to stats file name")), ToolInput(tag="in_arg_append_stats", input_type=Boolean(optional=True), prefix="-A", doc=InputDocumentation(doc="[ --append-stats-file ] arg        Append to stats file name")), ToolInput(tag="in_one", input_type=Boolean(optional=True), prefix="-1", doc=InputDocumentation(doc="[ --read1-input ] arg              Read 1 paired end fastq input <space\nseparated for multiple files>")), ToolInput(tag="in_two", input_type=Boolean(optional=True), prefix="-2", doc=InputDocumentation(doc="[ --read2-input ] arg              Read 2 paired end fastq input <space\nseparated for multiple files>")), ToolInput(tag="in_arg_single_end", input_type=Boolean(optional=True), prefix="-U", doc=InputDocumentation(doc="[ --singleend-input ] arg          Single end read fastq input <space\nseparated for multiple files>")), ToolInput(tag="in_arg_interleaved_fastq", input_type=Boolean(optional=True), prefix="-I", doc=InputDocumentation(doc="[ --interleaved-input ] arg        Interleaved fastq input <space\nseparated for multiple files>")), ToolInput(tag="in_arg_tabdelimited_input", input_type=Boolean(optional=True), prefix="-T", doc=InputDocumentation(doc="[ --tab-input ] arg                Tab-delimited (tab6) input <space\nseparated for multiple files>")), ToolInput(tag="in_forces_overwrite_files", input_type=Boolean(optional=True), prefix="-F", doc=InputDocumentation(doc="[ --force ]                        Forces overwrite of files")), ToolInput(tag="in_output_uncompressed_files", input_type=Boolean(optional=True), prefix="-u", doc=InputDocumentation(doc="[ --uncompressed ]                 Output uncompressed (not gzipped) files")), ToolInput(tag="in_arg_output_fastq", input_type=Boolean(optional=True), prefix="-f", doc=InputDocumentation(doc="[ --fastq-output ] arg             Output to Fastq files <PE AND/OR SE\nfiles>")), ToolInput(tag="in_arg_output_interleaved", input_type=Boolean(optional=True), prefix="-i", doc=InputDocumentation(doc="[ --interleaved-output ] arg       Output to interleaved fastq files\n<INTERLEAVED PE AND/OR SE files>")), ToolInput(tag="in_arg_output_tabdelimited", input_type=File(optional=True), prefix="-t", doc=InputDocumentation(doc="[ --tab-output ] arg               Output to tab-delimited (tab6) file")), ToolInput(tag="in_arg_output_unmapped", input_type=File(optional=True), prefix="-z", doc=InputDocumentation(doc="[ --unmapped-output ] arg          Output to unmapped sam file")), ToolInput(tag="in_arg_supply_file", input_type=Boolean(optional=True), prefix="-s", doc=InputDocumentation(doc="[ --seq ] arg                      Please supply a fasta file - default -\nPhix Sequence - default\nhttps://www.ncbi.nlm.nih.gov/nuccore/96\n26372")), ToolInput(tag="in_check_r_r", input_type=Boolean(optional=True), prefix="-C", doc=InputDocumentation(doc="[ --check-read-2 ]                 Check R2 as well as R1 (pe)")), ToolInput(tag="in_arg_kmer_size", input_type=Boolean(optional=True), prefix="-k", doc=InputDocumentation(doc="[ --kmer ] arg (=12)               Kmer size of the lookup table (min 5,\nmax 256)")), ToolInput(tag="in_arg_proportion_kmer", input_type=Boolean(optional=True), prefix="-x", doc=InputDocumentation(doc="[ --percentage-hits ] arg (=0.25)  Proportion of kmer percentage-hits to\nsequence need to happen to discard (min\n0.0, max 1.0)")), ToolInput(tag="in_output_reads_kmerhit", input_type=Boolean(optional=True), prefix="-n", doc=InputDocumentation(doc="[ --inverse ]                      Output reads that are ABOVE the kmer\nhit threshold")), ToolInput(tag="in_only_record_reads", input_type=Boolean(optional=True), prefix="-r", doc=InputDocumentation(doc="[ --record ]                       Only record the reads that pass the\nkmer hit threshold, output all reads"))], outputs=[ToolOutput(tag="out_arg_output_tabdelimited", output_type=File(optional=True), selector=InputSelector(input_to_select="in_arg_output_tabdelimited", type_hint=File()), doc=OutputDocumentation(doc="[ --tab-output ] arg               Output to tab-delimited (tab6) file")), ToolOutput(tag="out_arg_output_unmapped", output_type=File(optional=True), selector=InputSelector(input_to_select="in_arg_output_unmapped", type_hint=File()), doc=OutputDocumentation(doc="[ --unmapped-output ] arg          Output to unmapped sam file"))], container="quay.io/biocontainers/htstream:1.3.3--h5ca1c16_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Hts_Seqscreener_V0_1_0().translate("wdl")

