from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Array, String, Int, Boolean

Mageck_Count_V0_1_0 = CommandToolBuilder(tool="mageck_count", base_command=["mageck", "count"], inputs=[ToolInput(tag="in_list_seq", input_type=File(optional=True), prefix="--list-seq", doc=InputDocumentation(doc="A file containing the list of sgRNA names, their\nsequences and associated genes. Support file format:\ncsv and txt. Provide an empty file for collecting all\npossible sgRNA counts.")), ToolInput(tag="in_fast_q", input_type=Array(t=String(), optional=True), prefix="--fastq", doc=InputDocumentation(doc="Sample fastq files (or fastq.gz files, or SAM/BAM\nfiles after v0.5.5), separated by space; use comma (,)\nto indicate technical replicates of the same sample.\nFor example, '--fastq\nsample1_replicate1.fastq,sample1_replicate2.fastq\nsample2_replicate1.fastq,sample2_replicate2.fastq'\nindicates two samples with 2 technical replicates for\neach sample.")), ToolInput(tag="in_count_table", input_type=File(optional=True), prefix="--count-table", doc=InputDocumentation(doc="The read count table file. Only 1 file is accepted.")), ToolInput(tag="in_norm_method", input_type=String(optional=True), prefix="--norm-method", doc=InputDocumentation(doc="Method for normalization, including 'none' (no\nnormalization), 'median' (median normalization,\ndefault), 'total' (normalization by total read\ncounts), 'control' (normalization by control sgRNAs\nspecified by the --control-sgrna option).")), ToolInput(tag="in_control_sg_rna", input_type=String(optional=True), prefix="--control-sgrna", doc=InputDocumentation(doc="A list of control sgRNAs for normalization and for\ngenerating the null distribution of RRA.")), ToolInput(tag="in_control_gene", input_type=String(optional=True), prefix="--control-gene", doc=InputDocumentation(doc="A list of genes whose sgRNAs are used as control\nsgRNAs for normalization and for generating the null\ndistribution of RRA.")), ToolInput(tag="in_sample_label", input_type=Int(optional=True), prefix="--sample-label", doc=InputDocumentation(doc="Sample labels, separated by comma (,). Must be equal\nto the number of samples provided (in --fastq option).\nDefault 'sample1,sample2,...'.")), ToolInput(tag="in_output_prefix", input_type=File(optional=True), prefix="--output-prefix", doc=InputDocumentation(doc="The prefix of the output file(s). Default sample1.")), ToolInput(tag="in_unmapped_to_file", input_type=Boolean(optional=True), prefix="--unmapped-to-file", doc=InputDocumentation(doc="Save unmapped reads to file, with sgRNA lengths\nspecified by --sgrna-len option.")), ToolInput(tag="in_keep_tmp", input_type=Boolean(optional=True), prefix="--keep-tmp", doc=InputDocumentation(doc="Keep intermediate files.")), ToolInput(tag="in_test_run", input_type=Boolean(optional=True), prefix="--test-run", doc=InputDocumentation(doc="Test running. If this option is on, MAGeCK will only\nprocess the first 1M records for each file.")), ToolInput(tag="in_fast_q_two", input_type=Array(t=Int(), optional=True), prefix="--fastq-2", doc=InputDocumentation(doc="Paired sample fastq files (or fastq.gz files), the\norder of which should be consistent with that in fastq\noption.")), ToolInput(tag="in_count_pair", input_type=String(optional=True), prefix="--count-pair", doc=InputDocumentation(doc="Report all valid alignments per read or pair (default:\nFalse).")), ToolInput(tag="in_trim_five", input_type=Int(optional=True), prefix="--trim-5", doc=InputDocumentation(doc="Length of trimming the 5' of the reads. Users can\nspecify multiple trimming lengths, separated by comma\n(,); for example, '7,8'. Use 'AUTO' to allow MAGeCK to\nautomatically determine the trimming length. Default\nAUTO.")), ToolInput(tag="in_sg_rna_len", input_type=Int(optional=True), prefix="--sgrna-len", doc=InputDocumentation(doc="Length of the sgRNA. Default 20. ATTENTION: after\nv0.5.3, the program will automatically determine the\nsgRNA length from library file; so only use this if\nyou turn on the --unmapped-to-file option.")), ToolInput(tag="in_count_n", input_type=Boolean(optional=True), prefix="--count-n", doc=InputDocumentation(doc="Count sgRNAs with Ns. By default, sgRNAs containing N\nwill be discarded.")), ToolInput(tag="in_reverse_complement", input_type=Boolean(optional=True), prefix="--reverse-complement", doc=InputDocumentation(doc="Reverse complement the sequences in library for read")), ToolInput(tag="in_pdf_report", input_type=Boolean(optional=True), prefix="--pdf-report", doc=InputDocumentation(doc="Generate pdf report of the fastq files.")), ToolInput(tag="in_day_zero_label", input_type=Int(optional=True), prefix="--day0-label", doc=InputDocumentation(doc="Turn on the negative selection QC and specify the\nlabel for control sample (usually day 0 or plasmid).\nFor every other sample label, the negative selection\nQC will compare it with day0 sample, and estimate the\ndegree of negative selections in essential genes.")), ToolInput(tag="in_gmt_file", input_type=File(optional=True), prefix="--gmt-file", doc=InputDocumentation(doc="The pathway file used for QC, in GMT format. By\ndefault it will use the GMT file provided by MAGeCK.\n")), ToolInput(tag="in_mapping_dot", input_type=String(), position=0, doc=InputDocumentation(doc="Optional arguments for quality controls:"))], outputs=[ToolOutput(tag="out_output_prefix", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output_prefix", type_hint=File()), doc=OutputDocumentation(doc="The prefix of the output file(s). Default sample1."))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Mageck_Count_V0_1_0().translate("wdl", allow_empty_container=True)

