from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, String

Amptk_Pb_Dada2_V0_1_0 = CommandToolBuilder(tool="amptk_pb_dada2", base_command=["amptk", "pb-dada2"], inputs=[ToolInput(tag="in_fast_q", input_type=Boolean(optional=True), prefix="--fastq", doc=InputDocumentation(doc="Input FASTQ file (Required)")), ToolInput(tag="in_out", input_type=Boolean(optional=True), prefix="--out", doc=InputDocumentation(doc="Output base name.")), ToolInput(tag="in_min_reads", input_type=Boolean(optional=True), prefix="--min_reads", doc=InputDocumentation(doc="Minimum number of reads per sample. Default: 10")), ToolInput(tag="in_read_qual", input_type=Boolean(optional=True), prefix="--read_qual", doc=InputDocumentation(doc="Minimum read quality. Default: 0.95 [0-1]")), ToolInput(tag="in_barcode_qual", input_type=Boolean(optional=True), prefix="--barcode_qual", doc=InputDocumentation(doc="Barcode Quality score. Default: 80 [0-100]")), ToolInput(tag="in_pct_otu", input_type=Boolean(optional=True), prefix="--pct_otu", doc=InputDocumentation(doc="OTU Clustering Radius (percent). Default: 97")), ToolInput(tag="in_platform", input_type=Boolean(optional=True), prefix="--platform", doc=InputDocumentation(doc="Sequencing platform. Default: pacbio")), ToolInput(tag="in_chimera_method", input_type=Boolean(optional=True), prefix="--chimera_method", doc=InputDocumentation(doc="DADA2 de novo chimera method. Default: consensus [consensus,pooled,per-sample]")), ToolInput(tag="in_pool", input_type=Boolean(optional=True), prefix="--pool", doc=InputDocumentation(doc="Pool all samples together for DADA2. Default: off")), ToolInput(tag="in_pseudo_pool", input_type=Boolean(optional=True), prefix="--pseudopool", doc=InputDocumentation(doc="Pseudopooling samples for DADA2. Default: off")), ToolInput(tag="in_u_chime_ref", input_type=Boolean(optional=True), prefix="--uchime_ref", doc=InputDocumentation(doc="Run Ref Chimera filtering. Default: off [ITS, LSU, COI, 16S, custom path]")), ToolInput(tag="in_cpus", input_type=Boolean(optional=True), prefix="--cpus", doc=InputDocumentation(doc="Number of CPUs to use. Default: all")), ToolInput(tag="in_debug", input_type=Boolean(optional=True), prefix="--debug", doc=InputDocumentation(doc="Keep intermediate files.")), ToolInput(tag="in_arguments", input_type=String(), position=0, doc=InputDocumentation(doc=""))], outputs=[], container="quay.io/biocontainers/amptk:1.5.2--py_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Amptk_Pb_Dada2_V0_1_0().translate("wdl")

