from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Int, String, File

Kissreads_V0_1_0 = CommandToolBuilder(tool="kissreads", base_command=["kissreads"], inputs=[ToolInput(tag="in_use_seeds_length", input_type=Int(optional=True), prefix="-k", doc=InputDocumentation(doc=": will use seeds of length size_seed. Default: 25.")), ToolInput(tag="in_read_mapped_is", input_type=String(optional=True), prefix="-O", doc=InputDocumentation(doc=": a read can be mapped if its overlap is a least 'minimal_read_overlap'. Default: k")), ToolInput(tag="in_sequence_covered_at", input_type=Int(optional=True), prefix="-c", doc=InputDocumentation(doc=": a sequence is covered by at least min_coverage coherent reads. Default: 2")), ToolInput(tag="in_maximal_number_allowed", input_type=Int(optional=True), prefix="-d", doc=InputDocumentation(doc=": Maximal number of substitutions authorized between a read and a fragment. Note that no substitution is allowed on the central position while anaylizing the kissnp output. Default: 1.")), ToolInput(tag="in_write_readcoherent_outputs", input_type=File(optional=True), prefix="-o", doc=InputDocumentation(doc=": write read-coherent outputs. Default: standard output")), ToolInput(tag="in_write_unreadcoherent_outputs", input_type=File(optional=True), prefix="-u", doc=InputDocumentation(doc=": write unread-coherent outputs. Default: standard output")), ToolInput(tag="in_input_file_tocheckfasta_kissnp", input_type=File(optional=True), prefix="-n", doc=InputDocumentation(doc="input file (toCheck.fasta) is a kissnp output (incompatible with -I option)\nin this case: 1/ only the upper characters are considered (no mapping done on the extensions) and 2/ the central position (where the SNP occurs) is strictly mapped, no subsitution is authorized on this position.")), ToolInput(tag="in_input_file_tocheckfasta_intl", input_type=File(optional=True), prefix="-I", doc=InputDocumentation(doc="input file (toCheck.fasta) is an Intl output (incompatible with -n option)")), ToolInput(tag="in_int_value_heuristic", input_type=Int(optional=True), prefix="-i", doc=InputDocumentation(doc="(int value). This is a heuristic for limiting the memory footprint. Instead of indexing each kmer of the sequences contained into the toCheck.fasta, kissreads indexes kmers occurring each 'index_stride' position. Default = 1 (no heuristic)")), ToolInput(tag="in_number_threads_also", input_type=Int(optional=True), prefix="-t", doc=InputDocumentation(doc="number of threads (also limited by number of input files)")), ToolInput(tag="in_write_file_alignfile", input_type=File(optional=True), prefix="-m", doc=InputDocumentation(doc=", write a file of reads mapped to sequences in file align_file")), ToolInput(tag="in_mode", input_type=String(optional=True), prefix="-s", doc=InputDocumentation(doc="mode")), ToolInput(tag="in_coherent_events_standard", input_type=File(optional=True), prefix="-f", doc=InputDocumentation(doc="coherent events in a standard fasta file format")), ToolInput(tag="in_this_message_exit", input_type=String(optional=True), prefix="-h", doc=InputDocumentation(doc="this message and exit"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Kissreads_V0_1_0().translate("wdl", allow_empty_container=True)

