from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Boolean, String

Multigps_V0_1_0 = CommandToolBuilder(tool="multigps", base_command=["multigps"], inputs=[ToolInput(tag="in_threads", input_type=File(optional=True), prefix="--threads", doc=InputDocumentation(doc="<number of threads to use (default=1)>")), ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="--verbose", doc=InputDocumentation(doc="[flag to print intermediate files and extra output]")), ToolInput(tag="in_config", input_type=Boolean(optional=True), prefix="--config", doc=InputDocumentation(doc="<config file: all options here can be specified in a name<space>value text file, over-ridden by command-line args>")), ToolInput(tag="in_geninfo", input_type=File(optional=True), prefix="--geninfo", doc=InputDocumentation(doc="AND --seq <fasta seq directory reqd if using motif prior>")), ToolInput(tag="in_expt", input_type=File(optional=True), prefix="--expt", doc=InputDocumentation(doc="AND --format <SAM/BED/SCIDX>")), ToolInput(tag="in_ctrl", input_type=Boolean(optional=True), prefix="--ctrl", doc=InputDocumentation(doc="<file name (optional argument. must be same format as expt files)>")), ToolInput(tag="in_design", input_type=Boolean(optional=True), prefix="--design", doc=InputDocumentation(doc="<experiment design file name to use instead of --expt and --ctrl; see website for format>")), ToolInput(tag="in_fixed_pb", input_type=Boolean(optional=True), prefix="--fixedpb", doc=InputDocumentation(doc="<fixed per base limit (default: estimated from background model)>")), ToolInput(tag="in_poisson_gauss_pb", input_type=String(optional=True), prefix="--poissongausspb", doc=InputDocumentation(doc="[flag to use non-unique reads]")), ToolInput(tag="in_mapp_ability", input_type=Boolean(optional=True), prefix="--mappability", doc=InputDocumentation(doc="<fraction of the genome that is mappable for these experiments (default=0.8)>")), ToolInput(tag="in_nocache", input_type=Boolean(optional=True), prefix="--nocache", doc=InputDocumentation(doc="[flag to turn off caching of the entire set of experiments (i.e. run slower with less memory)]")), ToolInput(tag="in_no_scaling", input_type=Boolean(optional=True), prefix="--noscaling", doc=InputDocumentation(doc="[flag to turn off auto estimation of signal vs control scaling factor]")), ToolInput(tag="in_median_scale", input_type=Boolean(optional=True), prefix="--medianscale", doc=InputDocumentation(doc="[flag to use scaling by median ratio (default = scaling by NCIS)]")), ToolInput(tag="in_regression_scale", input_type=Boolean(optional=True), prefix="--regressionscale", doc=InputDocumentation(doc="[flag to use scaling by regression (default = scaling by NCIS)]")), ToolInput(tag="in_ses_scale", input_type=Boolean(optional=True), prefix="--sesscale", doc=InputDocumentation(doc="[flag to use scaling by SES (default = scaling by NCIS)]")), ToolInput(tag="in_fixed_scaling", input_type=Boolean(optional=True), prefix="--fixedscaling", doc=InputDocumentation(doc="<multiply control counts by total tag count ratio and then by this factor (default: NCIS)>")), ToolInput(tag="in_scale_win", input_type=Boolean(optional=True), prefix="--scalewin", doc=InputDocumentation(doc="<window size for scaling procedure (default=10000)>")), ToolInput(tag="in_plot_scaling", input_type=Boolean(optional=True), prefix="--plotscaling", doc=InputDocumentation(doc="[flag to plot diagnostic information for the chosen scaling method]")), ToolInput(tag="in_max_model_update", input_type=File(optional=True), prefix="--d", doc=InputDocumentation(doc="<max. model update rounds, default=3>")), ToolInput(tag="in_no_model_update", input_type=Boolean(optional=True), prefix="--nomodelupdate", doc=InputDocumentation(doc="[flag to turn off binding model updates]")), ToolInput(tag="in_min_model_update_events", input_type=Boolean(optional=True), prefix="--minmodelupdateevents", doc=InputDocumentation(doc="<minimum number of events to support an update (default=500)>")), ToolInput(tag="in_no_model_smoothing", input_type=Boolean(optional=True), prefix="--nomodelsmoothing", doc=InputDocumentation(doc="[flag to turn off binding model smoothing]")), ToolInput(tag="in_spline_smooth_param", input_type=Boolean(optional=True), prefix="--splinesmoothparam", doc=InputDocumentation(doc="<spline smoothing parameter (default=30)>")), ToolInput(tag="in_gauss_model_smoothing", input_type=Boolean(optional=True), prefix="--gaussmodelsmoothing", doc=InputDocumentation(doc="[flag to turn on Gaussian model smoothing (default = cubic spline)]")), ToolInput(tag="in_gauss_smooth_param", input_type=Boolean(optional=True), prefix="--gausssmoothparam", doc=InputDocumentation(doc="<Gaussian smoothing std dev (default=3)>")), ToolInput(tag="in_joint_in_model", input_type=Boolean(optional=True), prefix="--jointinmodel", doc=InputDocumentation(doc="[flag to allow joint events in model updates (default=do not)]")), ToolInput(tag="in_fixed_model_range", input_type=Boolean(optional=True), prefix="--fixedmodelrange", doc=InputDocumentation(doc="[flag to keep binding model range fixed to inital size (default: vary automatically)]")), ToolInput(tag="in_pr_log_conf", input_type=Boolean(optional=True), prefix="--prlogconf", doc=InputDocumentation(doc="<Poisson log threshold for potential region scanning(default=-6)>")), ToolInput(tag="in_alpha_scale", input_type=Boolean(optional=True), prefix="--alphascale", doc=InputDocumentation(doc="<alpha scaling factor(default=1.0>")), ToolInput(tag="in_fixed_alpha", input_type=Boolean(optional=True), prefix="--fixedalpha", doc=InputDocumentation(doc="<impose this alpha (default: set automatically)>")), ToolInput(tag="in_ml_config_not_shared", input_type=Boolean(optional=True), prefix="--mlconfignotshared", doc=InputDocumentation(doc="[flag to not share component configs in the ML step]")), ToolInput(tag="in_no_pos_prior", input_type=Boolean(optional=True), prefix="--noposprior", doc=InputDocumentation(doc="[flag to turn off inter-experiment positional prior (default=on)]")), ToolInput(tag="in_prob_shared", input_type=Boolean(optional=True), prefix="--probshared", doc=InputDocumentation(doc="<probability that events are shared across conditions (default=0.9)>")), ToolInput(tag="in_no_motifs", input_type=Boolean(optional=True), prefix="--nomotifs", doc=InputDocumentation(doc="[flag to turn off motif-finding & motif priors]")), ToolInput(tag="in_no_motif_prior", input_type=Boolean(optional=True), prefix="--nomotifprior", doc=InputDocumentation(doc="[flag to turn off motif priors only]")), ToolInput(tag="in_meme_path", input_type=Boolean(optional=True), prefix="--memepath", doc=InputDocumentation(doc="<path to the meme bin dir (default: meme is in $PATH)>")), ToolInput(tag="in_me_men_motifs", input_type=Boolean(optional=True), prefix="--memenmotifs", doc=InputDocumentation(doc="<number of motifs MEME should find for each condition (default=3)>")), ToolInput(tag="in_meme_min_w", input_type=Boolean(optional=True), prefix="--mememinw", doc=InputDocumentation(doc="<minw arg for MEME (default=6)>")), ToolInput(tag="in_meme_max_w", input_type=Boolean(optional=True), prefix="--mememaxw", doc=InputDocumentation(doc="<maxw arg for MEME (default=18)>")), ToolInput(tag="in_meme_args", input_type=Boolean(optional=True), prefix="--memeargs", doc=InputDocumentation(doc="<additional args for MEME (default=  -dna -mod zoops -revcomp -nostatus)>")), ToolInput(tag="in_meme_one_proc", input_type=Boolean(optional=True), prefix="--meme1proc", doc=InputDocumentation(doc="[flag to enforce non-parallel version of MEME]")), ToolInput(tag="in_qvalue_minimum_default", input_type=Boolean(optional=True), prefix="--q", doc=InputDocumentation(doc="<Q-value minimum (default=0.001)>")), ToolInput(tag="in_min_fold", input_type=Boolean(optional=True), prefix="--minfold", doc=InputDocumentation(doc="<minimum event fold-change vs scaled control (default=1.5)>")), ToolInput(tag="in_no_diff_tests", input_type=Boolean(optional=True), prefix="--nodifftests", doc=InputDocumentation(doc="[flag to turn off differential enrichment tests]")), ToolInput(tag="in_rpath", input_type=Boolean(optional=True), prefix="--rpath", doc=InputDocumentation(doc="<path to the R bin dir (default: R is in $PATH). Note that you need to install edgeR separately>")), ToolInput(tag="in_edge_rod", input_type=Boolean(optional=True), prefix="--edgerod", doc=InputDocumentation(doc="<EdgeR overdispersion parameter (default=0.15)>")), ToolInput(tag="in_diff_p", input_type=Boolean(optional=True), prefix="--diffp", doc=InputDocumentation(doc="<minimum p-value for reporting differential enrichment (default=0.01)>")), ToolInput(tag="in_events_are_txt", input_type=Boolean(optional=True), prefix="--eventsaretxt", doc=InputDocumentation(doc="[add .txt to events file extension]"))], outputs=[ToolOutput(tag="out_threads", output_type=File(optional=True), selector=InputSelector(input_to_select="in_threads", type_hint=File()), doc=OutputDocumentation(doc="<number of threads to use (default=1)>"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Multigps_V0_1_0().translate("wdl", allow_empty_container=True)

