from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, Int, String

Vtools_Report_Sequence_V0_1_0 = CommandToolBuilder(tool="vtools_report_sequence", base_command=["vtools_report", "sequence"], inputs=[ToolInput(tag="in_build", input_type=Boolean(optional=True), prefix="--build", doc=InputDocumentation(doc="[BUILD]       Output sequence at specified build of reference\ngenome. The primary reference genome of the project\nwill be used if by default.")), ToolInput(tag="in_numbered", input_type=Boolean(optional=True), prefix="--numbered", doc=InputDocumentation(doc="[{left,right}]\nIf specified, add position of the first or last\nbasepair of each line to the left or right of the\nline, and insert a space at every 10 basepair")), ToolInput(tag="in_char_per_line", input_type=Int(optional=True), prefix="--char_per_line", doc=InputDocumentation(doc="Number of characters (excluding space and leading\nnumbers) per line. Default to 70 in regular and 60 in\nnumbered format.")), ToolInput(tag="in_transcribe", input_type=Boolean(optional=True), prefix="--transcribe", doc=InputDocumentation(doc="[GENE [GENE ...]]\nTranscribe DNA sequence into RNA sequence. variant\ntools will look in the refGene database, find all\ngenes that overlap with the region, locate exons and\ncoding regions, transcribe the DNA sequence to RNA\nsequence (basically discard introns, and complement if\non reverse strand). The complete mRNA sequence will be\nprinted regardless of the bounaries of specified\nregions. If one or more names (refGene.name) are\nspecified, only specified genes will be translated.")), ToolInput(tag="in_t_tou", input_type=Boolean(optional=True), prefix="--TtoU", doc=InputDocumentation(doc="Print U for T for RNA sequence. Otherwise use T.")), ToolInput(tag="in_translate", input_type=Boolean(optional=True), prefix="--translate", doc=InputDocumentation(doc="[GENE [GENE ...]]\nTranslate DNA sequence into protein sequence. variant\ntools will look in the refGene database, find all\ngenes that overlap with the region, locate exons and\ncoding regions, transcribe and translate the DNA\nsequence to protein sequence. The complete protein\nsequence will be printed regardless of the boundaries\nof specified regions. If one or more names\n(refGene.name) are specified, only specified genes\nwill be translated.")), ToolInput(tag="in_mark", input_type=Boolean(optional=True), prefix="--mark", doc=InputDocumentation(doc="[MARK [MARK ...]]\nMark a location (--mark chr pos), a variant (--mark\nchr pos ref alt), a region (e.g.\nrefGene_exon.name:NM_000947), or a particular sequence\n(e.g. TCGGA) in red in the output. If a variant is\nspecified, the changed nucleotide or amino acid will\nbe printed. Currently only single nucleotide\npolymorphisms are supported.")), ToolInput(tag="in_mark_complement", input_type=Boolean(optional=True), prefix="--mark_complement", doc=InputDocumentation(doc="If set, also try to mark the complement of the\nspecified sequence")), ToolInput(tag="in_mark_reverse", input_type=Boolean(optional=True), prefix="--mark_reverse", doc=InputDocumentation(doc="If set, also try to mark the reverse sequence of the\nspecified sequence. If both mark_complemnt and\nmark_reverse are set, four different sequences will be\nsearched.")), ToolInput(tag="in_first_transcript", input_type=Boolean(optional=True), prefix="--first_transcript", doc=InputDocumentation(doc="If set, only display the first transcript of RNA or\nProtein sequence")), ToolInput(tag="in_show_transcript", input_type=Boolean(optional=True), prefix="--show_transcript", doc=InputDocumentation(doc="Put transcript name before transcript")), ToolInput(tag="in_marked_region", input_type=Boolean(optional=True), prefix="--marked_region", doc=InputDocumentation(doc="[MARKED_REGION [MARKED_REGION ...]], --marked-region [MARKED_REGION [MARKED_REGION ...]]\nIf set to a number or pair of number, print only n bp\nto the left and m (m=n if only one number is\nspecified) of the marked region. The sequence itself\nis no longer marked. This option is not yet supported\nin all combinations of options.")), ToolInput(tag="in_hide_unmatched", input_type=Boolean(optional=True), prefix="--hide_unmatched", doc=InputDocumentation(doc="If set, only display regions with marked variants or\nsequences")), ToolInput(tag="in_zero_based", input_type=Boolean(optional=True), prefix="--zero_based", doc=InputDocumentation(doc="If set, user input is zero based and will be\ntranslated to 1-based coordinates before query. The\noutput is always 1-based")), ToolInput(tag="in_verbosity", input_type=String(optional=True), prefix="--verbosity", doc=InputDocumentation(doc="Output error and warning (0), info (1), debug (2) and\ntrace (3) information to standard output (default to\n1).\n")), ToolInput(tag="in_regions", input_type=String(), position=0, doc=InputDocumentation(doc="One or more chromosome regions in the format of\nchr:start-end (e.g. chr21:33,031,597-33,041,570),\nField:Value from a region-based annotation database\n(e.g. refGene.name2:TRIM2 or\nrefGene_exon.name:NM_000947), or set options of\nseveral regions (&, |, -, and ^ for intersection,\nunion, difference, and symmetric difference). Several\nregions will be printed if the name matches more than\none regions. Chromosome positions are 1-based and are\ninclusive at both ends so the chromosome region has a\nlength of end-start+1 bp. A reversed complementary\nsequence will be outputted if starting position is\nafter the ending position."))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Vtools_Report_Sequence_V0_1_0().translate("wdl", allow_empty_container=True)

