from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Array, File, String, Int, Float, Directory, Boolean

Macs2_Filterdup_V0_1_0 = CommandToolBuilder(tool="macs2_filterdup", base_command=["macs2", "filterdup"], inputs=[ToolInput(tag="in_i_file", input_type=Array(t=File(), optional=True), prefix="--ifile", doc=InputDocumentation(doc="Alignment file. If multiple files are given as '-t A B\nC', then they will all be read and combined. Note that\npair-end data is not supposed to work with this\ncommand. REQUIRED.")), ToolInput(tag="in_format", input_type=String(optional=True), prefix="--format", doc=InputDocumentation(doc="Format of tag file, 'AUTO', 'BED' or 'ELAND' or\n'ELANDMULTI' or 'ELANDEXPORT' or 'SAM' or 'BAM' or\n'BOWTIE' or 'BAMPE' or 'BEDPE'. The default AUTO\noption will let 'macs2 filterdup' decide which format\nthe file is. Please check the definition in README\nfile if you choose\nELAND/ELANDMULTI/ELANDEXPORT/SAM/BAM/BOWTIE or\nBAMPE/BEDPE. DEFAULT: 'AUTO'")), ToolInput(tag="in_g_size", input_type=Int(optional=True), prefix="--gsize", doc=InputDocumentation(doc="Effective genome size. It can be 1.0e+9 or 1000000000,\nor shortcuts:'hs' for human (2.7e9), 'mm' for mouse\n(1.87e9), 'ce' for C. elegans (9e7) and 'dm' for\nfruitfly (1.2e8), DEFAULT:hs")), ToolInput(tag="in_t_size", input_type=Int(optional=True), prefix="--tsize", doc=InputDocumentation(doc="Tag size. This will override the auto detected tag\nsize. DEFAULT: Not set")), ToolInput(tag="in_p_value", input_type=Float(optional=True), prefix="--pvalue", doc=InputDocumentation(doc="Pvalue cutoff for binomial distribution test.\nDEFAULT:1e-5")), ToolInput(tag="in_keep_dup", input_type=Int(optional=True), prefix="--keep-dup", doc=InputDocumentation(doc="It controls the 'macs2 filterdup' behavior towards\nduplicate tags/pairs at the exact same location -- the\nsame coordination and the same strand. The 'auto'\noption makes 'macs2 filterdup' calculate the maximum\ntags at the exact same location based on binomal\ndistribution using given -p as pvalue cutoff; and the\n'all' option keeps every tags (useful if you only want\nto convert formats). If an integer is given, at most\nthis number of tags will be kept at the same location.\nNote, MACS2 callpeak function uses KEEPDUPLICATES=1 as\ndefault. Note, if you've used samtools or picard to\nflag reads as 'PCR/Optical duplicate' in bit 1024,\nMACS2 will still read them although the reads may be\ndecided by MACS2 as duplicate later. Default: auto")), ToolInput(tag="in_buffer_size", input_type=Int(optional=True), prefix="--buffer-size", doc=InputDocumentation(doc="Buffer size for incrementally increasing internal\narray size to store reads alignment information. In\nmost cases, you don't have to change this parameter.\nHowever, if there are large number of\nchromosomes/contigs/scaffolds in your alignment, it's\nrecommended to specify a smaller buffer size in order\nto decrease memory usage (but it will take longer time\nto read alignment files). Minimum memory requested for\nreading an alignment file is about # of CHROMOSOME *\nBUFFER_SIZE * 8 Bytes. DEFAULT: 100000")), ToolInput(tag="in_verbose", input_type=Int(optional=True), prefix="--verbose", doc=InputDocumentation(doc="Set verbose level. 0: only show critical message, 1:\nshow additional warning message, 2: show process\ninformation, 3: show debug messages. If you want to\nknow where are the duplicate reads, use 3. DEFAULT:2")), ToolInput(tag="in_outdir", input_type=Directory(optional=True), prefix="--outdir", doc=InputDocumentation(doc="If specified all output files will be written to that\ndirectory. Default: the current working directory")), ToolInput(tag="in_ofile", input_type=File(optional=True), prefix="--ofile", doc=InputDocumentation(doc="Output BED file name. If not specified, will write to\nstandard output. Note, if the input format is BAMPE or\nBEDPE, the output will be in BEDPE format. DEFAULT:\nstdout")), ToolInput(tag="in_dry_run", input_type=Boolean(optional=True), prefix="--dry-run", doc=InputDocumentation(doc="When set, filterdup will only output numbers instead\nof writing output files, including maximum allowable\nduplicates, total number of reads before filtering,\ntotal number of reads after filtering, and redundant\nrate. Default: not set\n"))], outputs=[ToolOutput(tag="out_outdir", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_outdir", type_hint=File()), doc=OutputDocumentation(doc="If specified all output files will be written to that\ndirectory. Default: the current working directory")), ToolOutput(tag="out_ofile", output_type=File(optional=True), selector=InputSelector(input_to_select="in_ofile", type_hint=File()), doc=OutputDocumentation(doc="Output BED file name. If not specified, will write to\nstandard output. Note, if the input format is BAMPE or\nBEDPE, the output will be in BEDPE format. DEFAULT:\nstdout"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Macs2_Filterdup_V0_1_0().translate("wdl", allow_empty_container=True)

