from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, String, File

Getdiffexpression_Pl_V0_1_0 = CommandToolBuilder(tool="getDiffExpression.pl", base_command=["getDiffExpression.pl"], inputs=[ToolInput(tag="in_rna", input_type=Boolean(optional=True), prefix="-rna", doc=InputDocumentation(doc="(for analyzeRNA.pl formatted input, default)")), ToolInput(tag="in_repeats", input_type=Boolean(optional=True), prefix="-repeats", doc=InputDocumentation(doc="(for analyzeRepeats.pl formatted input file)")), ToolInput(tag="in_peaks", input_type=Boolean(optional=True), prefix="-peaks", doc=InputDocumentation(doc="(for annotatePeaks.pl formatted input file)")), ToolInput(tag="in_loop", input_type=String(optional=True), prefix="-loop", doc=InputDocumentation(doc="(for output files from findTADsAndLoops.pl)")), ToolInput(tag="in_pc_one", input_type=Boolean(optional=True), prefix="-pc1", doc=InputDocumentation(doc="(for output files from annotatePeaks.pl looking at PC1 bedGraphs, try -pc1options)")), ToolInput(tag="in_basic", input_type=Boolean(optional=True), prefix="-basic", doc=InputDocumentation(doc="(for simple file with one column of gene identifiers and then the count data)")), ToolInput(tag="in_dispersion", input_type=Boolean(optional=True), prefix="-dispersion", doc=InputDocumentation(doc="<#> (edgeR common dispersion to use if no replicates, default: 0.05)")), ToolInput(tag="in_norm_two_total", input_type=Boolean(optional=True), prefix="-norm2total", doc=InputDocumentation(doc="(normalize using tag directory totals, default: normalize to gene totals(i.e.table)")), ToolInput(tag="in_avs_a", input_type=Boolean(optional=True), prefix="-AvsA", doc=InputDocumentation(doc="(compare each group vs. each group, default: compare 1st group vs. others)")), ToolInput(tag="in_show_r", input_type=Boolean(optional=True), prefix="-showR", doc=InputDocumentation(doc="(Show R status updates, command output)")), ToolInput(tag="in_de_seq_two", input_type=Boolean(optional=True), prefix="-DESeq2", doc=InputDocumentation(doc="(use DESeq2, now the default)")), ToolInput(tag="in_de_seq", input_type=Boolean(optional=True), prefix="-DESeq", doc=InputDocumentation(doc="(use DESeq instead of DESeq2 - doesn't support batch mode)")), ToolInput(tag="in_edger", input_type=Boolean(optional=True), prefix="-edgeR", doc=InputDocumentation(doc="(use edgeR, - doesn't support -norm2total for normalization to total mapped reads)")), ToolInput(tag="in_limma", input_type=Boolean(optional=True), prefix="-limma", doc=InputDocumentation(doc="(use limma, set to work on linear, non-integer values i.e. TAD IRs, PC1 values, etc.)")), ToolInput(tag="in_export", input_type=File(optional=True), prefix="-export", doc=InputDocumentation(doc="(output differential expression gene lists with filename prefix)")), ToolInput(tag="in_fdr", input_type=Boolean(optional=True), prefix="-fdr", doc=InputDocumentation(doc="<#> (FDR threshold for diff. expression reporting, default: 0.05)")), ToolInput(tag="in_p_value", input_type=Boolean(optional=True), prefix="-pvalue", doc=InputDocumentation(doc="<#> (p-value threshold for diff. expression reporting, default: uses -fdr)")), ToolInput(tag="in_log_twofold", input_type=Boolean(optional=True), prefix="-log2fold", doc=InputDocumentation(doc="<#> (Log2 Fold threshold for diff. expression reporting, default: 1.0)")), ToolInput(tag="in_r_log", input_type=Boolean(optional=True), prefix="-rlog", doc=InputDocumentation(doc="(Use DESeq2's rlog transform to normalize output matrix, default)")), ToolInput(tag="in_vst", input_type=Boolean(optional=True), prefix="-vst", doc=InputDocumentation(doc="(Use DESeq2's vst transform, quicker for large sample sets)")), ToolInput(tag="in_simple_norm", input_type=Boolean(optional=True), prefix="-simpleNorm", doc=InputDocumentation(doc="(Normalize to experiment totals, i.e. basic normalization)")), ToolInput(tag="in_raw", input_type=Boolean(optional=True), prefix="-raw", doc=InputDocumentation(doc="(do not normalize read counts in output table, keep raw values)")), ToolInput(tag="in_data_file", input_type=File(), position=0, doc=InputDocumentation(doc="")), ToolInput(tag="in_group_code_dot_dot_dot", input_type=String(), position=1, doc=InputDocumentation(doc=""))], outputs=[ToolOutput(tag="out_export", output_type=File(optional=True), selector=InputSelector(input_to_select="in_export", type_hint=File()), doc=OutputDocumentation(doc="(output differential expression gene lists with filename prefix)"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Getdiffexpression_Pl_V0_1_0().translate("wdl", allow_empty_container=True)

