from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import String, File, Directory, Boolean

Qualimap_Rnaseq_V0_1_0 = CommandToolBuilder(tool="qualimap_rnaseq", base_command=["qualimap", "rnaseq"], inputs=[ToolInput(tag="in_algorithm", input_type=String(optional=True), prefix="--algorithm", doc=InputDocumentation(doc="Counting algorithm:\nuniquely-mapped-reads(default) or\nproportional.")), ToolInput(tag="in_bam", input_type=File(optional=True), prefix="-bam", doc=InputDocumentation(doc="Input mapping file in BAM format.")), ToolInput(tag="in_gtf", input_type=File(optional=True), prefix="-gtf", doc=InputDocumentation(doc="Annotations file in Ensembl GTF format.")), ToolInput(tag="in_oc", input_type=File(optional=True), prefix="-oc", doc=InputDocumentation(doc="Output file for computed counts. If only name\nof the file is provided, then the file will be\nsaved in the output folder.")), ToolInput(tag="in_outdir", input_type=Directory(optional=True), prefix="-outdir", doc=InputDocumentation(doc="Output folder for HTML report and raw data.")), ToolInput(tag="in_outfile", input_type=File(optional=True), prefix="-outfile", doc=InputDocumentation(doc="Output file for PDF report (default value is\nreport.pdf).")), ToolInput(tag="in_out_format", input_type=String(optional=True), prefix="-outformat", doc=InputDocumentation(doc="Format of the output report (PDF, HTML or both\nPDF:HTML, default is HTML).")), ToolInput(tag="in_sequencing_protocol", input_type=String(optional=True), prefix="--sequencing-protocol", doc=InputDocumentation(doc="Sequencing library protocol:\nstrand-specific-forward,\nstrand-specific-reverse or non-strand-specific\n(default)")), ToolInput(tag="in_paired", input_type=Boolean(optional=True), prefix="--paired", doc=InputDocumentation(doc="Setting this flag for paired-end experiments\nwill result in counting fragments instead of\nreads")), ToolInput(tag="in_sorted", input_type=Boolean(optional=True), prefix="--sorted", doc=InputDocumentation(doc="This flag indicates that the input file is\nalready sorted by name. If not set, additional\nsorting by name will be performed. Only\nrequired for paired-end analysis.\n")), ToolInput(tag="in_arg", input_type=String(optional=True), position=0, doc=InputDocumentation(doc=""))], outputs=[ToolOutput(tag="out_oc", output_type=File(optional=True), selector=InputSelector(input_to_select="in_oc", type_hint=File()), doc=OutputDocumentation(doc="Output file for computed counts. If only name\nof the file is provided, then the file will be\nsaved in the output folder.")), ToolOutput(tag="out_outdir", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_outdir", type_hint=File()), doc=OutputDocumentation(doc="Output folder for HTML report and raw data.")), ToolOutput(tag="out_outfile", output_type=File(optional=True), selector=InputSelector(input_to_select="in_outfile", type_hint=File()), doc=OutputDocumentation(doc="Output file for PDF report (default value is\nreport.pdf)."))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Qualimap_Rnaseq_V0_1_0().translate("wdl", allow_empty_container=True)

