from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Directory, Boolean, File, Int, String

Kb_Count_V0_1_0 = CommandToolBuilder(tool="kb_count", base_command=["kb", "count"], inputs=[ToolInput(tag="in_tmp", input_type=Directory(optional=True), prefix="--tmp", doc=InputDocumentation(doc="Override default temporary directory")), ToolInput(tag="in_keep_tmp", input_type=Boolean(optional=True), prefix="--keep-tmp", doc=InputDocumentation(doc="Do not delete the tmp directory")), ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="--verbose", doc=InputDocumentation(doc="Print debugging information")), ToolInput(tag="in_path_output_directory", input_type=File(optional=True), prefix="-o", doc=InputDocumentation(doc="Path to output directory (default: current directory)")), ToolInput(tag="in_path_file_correct", input_type=File(optional=True), prefix="-w", doc=InputDocumentation(doc="Path to file of whitelisted barcodes to correct to. If\nnot provided and bustools supports the technology, a\npre-packaged whitelist is used. Otherwise, or if\n'None', is provided, the bustools whitelist command is\nused. (`kb --list` to view whitelists)")), ToolInput(tag="in_number_threads_use", input_type=Int(optional=True), prefix="-t", doc=InputDocumentation(doc="Number of threads to use (default: 8)")), ToolInput(tag="in_maximum_memory_used", input_type=Int(optional=True), prefix="-m", doc=InputDocumentation(doc="Maximum memory used (default: 4G)")), ToolInput(tag="in_workflow", input_type=Boolean(optional=True), prefix="--workflow", doc=InputDocumentation(doc="{standard,lamanno,nucleus,kite,kite:10xFB}\nType of workflow. Use `lamanno` for RNA velocity based\non La Manno et al. 2018 logic. Use `nucleus` for RNA\nvelocity on single-nucleus RNA-seq reads. Use `kite`\nfor feature barcoding. Use `kite:10xFB` for 10x\nGenomics Feature Barcoding technology. (default:\nstandard)")), ToolInput(tag="in_mm", input_type=Boolean(optional=True), prefix="--mm", doc=InputDocumentation(doc="Include reads that pseudoalign to multiple genes.")), ToolInput(tag="in_tcc", input_type=Boolean(optional=True), prefix="--tcc", doc=InputDocumentation(doc="Generate a TCC matrix instead of a gene count matrix.")), ToolInput(tag="in_filter", input_type=Boolean(optional=True), prefix="--filter", doc=InputDocumentation(doc="[{bustools}]\nProduce a filtered gene count matrix (default:\nbustools)")), ToolInput(tag="in_overwrite", input_type=File(optional=True), prefix="--overwrite", doc=InputDocumentation(doc="Overwrite existing output.bus file")), ToolInput(tag="in_dry_run", input_type=Boolean(optional=True), prefix="--dry-run", doc=InputDocumentation(doc="Dry run")), ToolInput(tag="in_laman_no", input_type=Boolean(optional=True), prefix="--lamanno", doc=InputDocumentation(doc="Deprecated. Use `--workflow lamanno` instead.")), ToolInput(tag="in_nucleus", input_type=Boolean(optional=True), prefix="--nucleus", doc=InputDocumentation(doc="Deprecated. Use `--workflow nucleus` instead.")), ToolInput(tag="in_loom", input_type=Boolean(optional=True), prefix="--loom", doc=InputDocumentation(doc="Generate loom file from count matrix")), ToolInput(tag="in_h_five_ad", input_type=Boolean(optional=True), prefix="--h5ad", doc=InputDocumentation(doc="Generate h5ad file from count matrix")), ToolInput(tag="in_cell_ranger", input_type=Boolean(optional=True), prefix="--cellranger", doc=InputDocumentation(doc="Convert count matrices to cellranger-compatible format")), ToolInput(tag="in_report", input_type=Boolean(optional=True), prefix="--report", doc=InputDocumentation(doc="Generate a HTML report containing run statistics and\nbasic plots. Using this option may cause kb to use\nmore memory than specified with the `-m` option. It\nmay also cause it to crash due to memory.")), ToolInput(tag="in_path_kallisto_commadelimited", input_type=File(optional=True), prefix="-i", doc=InputDocumentation(doc="Path to kallisto index/indices, comma-delimited")), ToolInput(tag="in_path_transcripttogene_mapping", input_type=Int(optional=True), prefix="-g", doc=InputDocumentation(doc="Path to transcript-to-gene mapping")), ToolInput(tag="in_singlecell_technology_used", input_type=String(optional=True), prefix="-x", doc=InputDocumentation(doc="Single-cell technology used (`kb --list` to view)")), ToolInput(tag="in_cone", input_type=Int(optional=True), prefix="-c1", doc=InputDocumentation(doc="Path to cDNA transcripts-to-capture")), ToolInput(tag="in_c_two", input_type=Int(optional=True), prefix="-c2", doc=InputDocumentation(doc="Path to intron transcripts-to-captured")), ToolInput(tag="in_fast_qs", input_type=String(), position=0, doc=InputDocumentation(doc="FASTQ files. For technology `SMARTSEQ`, all input\nFASTQs are alphabetically sorted by path and paired in\norder, and cell IDs are assigned as incrementing\nintegers starting from zero. A single batch TSV with\ncell ID, read 1, and read 2 as columns can be provided\nto override this behavior."))], outputs=[ToolOutput(tag="out_path_output_directory", output_type=File(optional=True), selector=InputSelector(input_to_select="in_path_output_directory", type_hint=File()), doc=OutputDocumentation(doc="Path to output directory (default: current directory)")), ToolOutput(tag="out_overwrite", output_type=File(optional=True), selector=InputSelector(input_to_select="in_overwrite", type_hint=File()), doc=OutputDocumentation(doc="Overwrite existing output.bus file"))], container="quay.io/biocontainers/kb-python:0.25.1--py_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Kb_Count_V0_1_0().translate("wdl")

