from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Array, String, File, Boolean

Makedb_Py_Ihmm_V0_1_0 = CommandToolBuilder(tool="MakeDb.py_ihmm", base_command=["MakeDb.py", "ihmm"], inputs=[ToolInput(tag="in_explicit_output_file", input_type=Array(t=String(), optional=True), prefix="-o", doc=InputDocumentation(doc="Explicit output file name. Note, this argument cannot\nbe used with the --failed, --outdir, or --outname\narguments. If unspecified, then the output filename\nwill be based on the input filename(s). (default:\nNone)")), ToolInput(tag="in_outdir", input_type=File(optional=True), prefix="--outdir", doc=InputDocumentation(doc="Specify to changes the output directory to the\nlocation specified. The input file directory is used\nif this is not specified. (default: None)")), ToolInput(tag="in_out_name", input_type=File(optional=True), prefix="--outname", doc=InputDocumentation(doc="Changes the prefix of the successfully processed\noutput file to the string specified. May not be\nspecified with multiple input files. (default: None)")), ToolInput(tag="in_log", input_type=File(optional=True), prefix="--log", doc=InputDocumentation(doc="Specify to write verbose logging to a file. May not be\nspecified with multiple input files. (default: None)")), ToolInput(tag="in_failed", input_type=Boolean(optional=True), prefix="--failed", doc=InputDocumentation(doc="If specified create files containing records that fail\nprocessing. (default: False)")), ToolInput(tag="in_format", input_type=String(optional=True), prefix="--format", doc=InputDocumentation(doc="Specify input and output format. (default: airr)")), ToolInput(tag="in_ihmmunealign_output_file", input_type=File(optional=True), prefix="-i", doc=InputDocumentation(doc="iHMMune-Align output file. (default: None)")), ToolInput(tag="in_list_folders_andor", input_type=Array(t=String(), optional=True), prefix="-r", doc=InputDocumentation(doc="List of folders and/or FASTA files containing the set\nof germline sequences used by iHMMune-Align. These\nreference sequences must contain IMGT-numbering\nspacers (gaps) in the V segment. (default: None)")), ToolInput(tag="in_list_containing_sequences", input_type=Array(t=String(), optional=True), prefix="-s", doc=InputDocumentation(doc="List of input FASTA files (with .fasta, .fna or .fa\nextension) containing sequences. (default: None)")), ToolInput(tag="in_one_zero_x", input_type=Array(t=File(), optional=True), prefix="--10x", doc=InputDocumentation(doc="Table file containing 10X annotations (with .csv or\n.tsv extension). (default: None)")), ToolInput(tag="in_as_is_id", input_type=Boolean(optional=True), prefix="--asis-id", doc=InputDocumentation(doc="Specify to prevent input sequence headers from being\nparsed to add new columns to database. Parsing of\nsequence headers requires headers to be in the pRESTO\nannotation format, so this should be specified when\nsequence headers are incompatible with the pRESTO\nannotation scheme. Note, unrecognized header formats\nwill default to this behavior. (default: False)")), ToolInput(tag="in_partial", input_type=Boolean(optional=True), prefix="--partial", doc=InputDocumentation(doc="If specified, include incomplete V(D)J alignments in\nthe pass file instead of the fail file. An incomplete\nalignment is defined as a record for which a valid\nIMGT-gapped sequence cannot be built or that is\nmissing a V gene assignment, J gene assignment,\njunction region, or productivity call. (default:\nFalse)")), ToolInput(tag="in_extended", input_type=File(optional=True), prefix="--extended", doc=InputDocumentation(doc="Specify to include additional aligner specific fields\nin the output. Adds the path score of the iHMMune-\nAlign hidden Markov model as vdj_score; adds fwr1,\nfwr2, fwr3, fwr4, cdr1, cdr2 and cdr3. (default:\nFalse)\n"))], outputs=[ToolOutput(tag="out_ihmmunealign_output_file", output_type=File(optional=True), selector=InputSelector(input_to_select="in_ihmmunealign_output_file", type_hint=File()), doc=OutputDocumentation(doc="iHMMune-Align output file. (default: None)")), ToolOutput(tag="out_extended", output_type=File(optional=True), selector=InputSelector(input_to_select="in_extended", type_hint=File()), doc=OutputDocumentation(doc="Specify to include additional aligner specific fields\nin the output. Adds the path score of the iHMMune-\nAlign hidden Markov model as vdj_score; adds fwr1,\nfwr2, fwr3, fwr4, cdr1, cdr2 and cdr3. (default:\nFalse)\n"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Makedb_Py_Ihmm_V0_1_0().translate("wdl", allow_empty_container=True)

