from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import String, Boolean, File, Int

Snpeff_V0_1_0 = CommandToolBuilder(tool="snpEff", base_command=["snpEff"], inputs=[ToolInput(tag="in_chr", input_type=String(optional=True), prefix="-chr", doc=InputDocumentation(doc=": Prepend 'string' to chromosome name (e.g. 'chr1' instead of '1'). Only on TXT output.")), ToolInput(tag="in_classic", input_type=Boolean(optional=True), prefix="-classic", doc=InputDocumentation(doc=": Use old style annotations instead of Sequence Ontology and Hgvs.")), ToolInput(tag="in_csv_stats", input_type=File(optional=True), prefix="-csvStats", doc=InputDocumentation(doc=": Create CSV summary file.")), ToolInput(tag="in_download", input_type=Boolean(optional=True), prefix="-download", doc=InputDocumentation(doc=": Download reference genome if not available. Default: true")), ToolInput(tag="in_input_format_default", input_type=String(optional=True), prefix="-i", doc=InputDocumentation(doc=": Input format [ vcf, bed ]. Default: VCF.")), ToolInput(tag="in_file_list", input_type=Boolean(optional=True), prefix="-fileList", doc=InputDocumentation(doc=": Input actually contains a list of files to process.")), ToolInput(tag="in_ouput_format_default", input_type=String(optional=True), prefix="-o", doc=InputDocumentation(doc=": Ouput format [ vcf, gatk, bed, bedAnn ]. Default: VCF.")), ToolInput(tag="in_html_stats", input_type=Boolean(optional=True), prefix="-htmlStats", doc=InputDocumentation(doc=": Create HTML summary file.  Default is 'snpEff_summary.html'")), ToolInput(tag="in_no_stats", input_type=Boolean(optional=True), prefix="-noStats", doc=InputDocumentation(doc=": Do not create stats (summary) file")), ToolInput(tag="in_filter_interval", input_type=Boolean(optional=True), prefix="-filterInterval", doc=InputDocumentation(doc="<file>   : Only analyze changes that intersect with the intervals specified in this file (you may use this option many times)")), ToolInput(tag="in_no_downstream", input_type=Boolean(optional=True), prefix="-no-downstream", doc=InputDocumentation(doc=": Do not show DOWNSTREAM changes")), ToolInput(tag="in_no_intergenic", input_type=Boolean(optional=True), prefix="-no-intergenic", doc=InputDocumentation(doc=": Do not show INTERGENIC changes")), ToolInput(tag="in_no_intron", input_type=Boolean(optional=True), prefix="-no-intron", doc=InputDocumentation(doc=": Do not show INTRON changes")), ToolInput(tag="in_no_upstream", input_type=Boolean(optional=True), prefix="-no-upstream", doc=InputDocumentation(doc=": Do not show UPSTREAM changes")), ToolInput(tag="in_no_utr", input_type=Boolean(optional=True), prefix="-no-utr", doc=InputDocumentation(doc=": Do not show 5_PRIME_UTR or 3_PRIME_UTR changes")), ToolInput(tag="in_no", input_type=String(optional=True), prefix="-no", doc=InputDocumentation(doc=": Do not show 'EffectType'. This option can be used several times.")), ToolInput(tag="in_cancer", input_type=Boolean(optional=True), prefix="-cancer", doc=InputDocumentation(doc=": Perform 'cancer' comparisons (Somatic vs Germline). Default: false")), ToolInput(tag="in_cancer_samples", input_type=File(optional=True), prefix="-cancerSamples", doc=InputDocumentation(doc=": Two column TXT file defining 'oringinal \t derived' samples.")), ToolInput(tag="in_format_eff", input_type=Boolean(optional=True), prefix="-formatEff", doc=InputDocumentation(doc=": Use 'EFF' field compatible with older versions (instead of 'ANN').")), ToolInput(tag="in_geneid", input_type=Boolean(optional=True), prefix="-geneId", doc=InputDocumentation(doc=": Use gene ID instead of gene name (VCF output). Default: false")), ToolInput(tag="in_hgvs", input_type=Boolean(optional=True), prefix="-hgvs", doc=InputDocumentation(doc=": Use HGVS annotations for amino acid sub-field. Default: true")), ToolInput(tag="in_hgv_sold", input_type=Boolean(optional=True), prefix="-hgvsOld", doc=InputDocumentation(doc=": Use old HGVS notation. Default: false")), ToolInput(tag="in_hgvs_one_letter_aa", input_type=Boolean(optional=True), prefix="-hgvs1LetterAa", doc=InputDocumentation(doc=": Use one letter Amino acid codes in HGVS notation. Default: false")), ToolInput(tag="in_hgv_strid", input_type=Boolean(optional=True), prefix="-hgvsTrId", doc=InputDocumentation(doc=": Use transcript ID in HGVS notation. Default: false")), ToolInput(tag="in_l_of", input_type=Boolean(optional=True), prefix="-lof", doc=InputDocumentation(doc=": Add loss of function (LOF) and Nonsense mediated decay (NMD) tags.")), ToolInput(tag="in_no_hgvs", input_type=Boolean(optional=True), prefix="-noHgvs", doc=InputDocumentation(doc=": Do not add HGVS annotations.")), ToolInput(tag="in_nol_of", input_type=Boolean(optional=True), prefix="-noLof", doc=InputDocumentation(doc=": Do not add LOF and NMD annotations.")), ToolInput(tag="in_no_shift_hgvs", input_type=Boolean(optional=True), prefix="-noShiftHgvs", doc=InputDocumentation(doc=": Do not shift variants according to HGVS notation (most 3prime end).")), ToolInput(tag="in_oic_r", input_type=Boolean(optional=True), prefix="-oicr", doc=InputDocumentation(doc=": Add OICR tag in VCF file. Default: false")), ToolInput(tag="in_sequence_ontology", input_type=Boolean(optional=True), prefix="-sequenceOntology", doc=InputDocumentation(doc=": Use Sequence Ontology terms. Default: true")), ToolInput(tag="in_config", input_type=Boolean(optional=True), prefix="-config", doc=InputDocumentation(doc=": Specify config file")), ToolInput(tag="in_config_option", input_type=File(optional=True), prefix="-configOption", doc=InputDocumentation(doc="=value     : Override a config file option")), ToolInput(tag="in_debug", input_type=Boolean(optional=True), prefix="-debug", doc=InputDocumentation(doc=": Debug mode (very verbose).")), ToolInput(tag="in_datadir", input_type=File(optional=True), prefix="-dataDir", doc=InputDocumentation(doc=": Override data_dir parameter from config file.")), ToolInput(tag="in_no_download", input_type=Boolean(optional=True), prefix="-nodownload", doc=InputDocumentation(doc=": Do not download a SnpEff database, if not available locally.")), ToolInput(tag="in_no_log", input_type=Boolean(optional=True), prefix="-noLog", doc=InputDocumentation(doc=": Do not report usage statistics to server")), ToolInput(tag="in_use_multiple_threads", input_type=Boolean(optional=True), prefix="-t", doc=InputDocumentation(doc=": Use multiple threads (implies '-noStats'). Default 'off'")), ToolInput(tag="in_quiet", input_type=Boolean(optional=True), prefix="-quiet", doc=InputDocumentation(doc=": Quiet mode (do not show any messages or errors)")), ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="-verbose", doc=InputDocumentation(doc=": Verbose mode")), ToolInput(tag="in_version", input_type=Boolean(optional=True), prefix="-version", doc=InputDocumentation(doc=": Show version number and exit")), ToolInput(tag="in_canon", input_type=Boolean(optional=True), prefix="-canon", doc=InputDocumentation(doc=": Only use canonical transcripts.")), ToolInput(tag="in_canon_list", input_type=File(optional=True), prefix="-canonList", doc=InputDocumentation(doc=": Only use canonical transcripts, replace some transcripts using the 'gene_id       transcript_id' entries in <file>.")), ToolInput(tag="in_interaction", input_type=Boolean(optional=True), prefix="-interaction", doc=InputDocumentation(doc=": Annotate using inteactions (requires interaciton database). Default: true")), ToolInput(tag="in_interval", input_type=File(optional=True), prefix="-interval", doc=InputDocumentation(doc=": Use a custom intervals in TXT/BED/BigBed/VCF/GFF file (you may use this option many times)")), ToolInput(tag="in_max_tsl", input_type=Int(optional=True), prefix="-maxTSL", doc=InputDocumentation(doc=": Only use transcripts having Transcript Support Level lower than <TSL_number>.")), ToolInput(tag="in_motif", input_type=Boolean(optional=True), prefix="-motif", doc=InputDocumentation(doc=": Annotate using motifs (requires Motif database). Default: true")), ToolInput(tag="in_next_prot", input_type=Boolean(optional=True), prefix="-nextProt", doc=InputDocumentation(doc=": Annotate using NextProt (requires NextProt database).")), ToolInput(tag="in_no_genome", input_type=Boolean(optional=True), prefix="-noGenome", doc=InputDocumentation(doc=": Do not load any genomic database (e.g. annotate using custom files).")), ToolInput(tag="in_noexpand_iub", input_type=Boolean(optional=True), prefix="-noExpandIUB", doc=InputDocumentation(doc=": Disable IUB code expansion in input variants")), ToolInput(tag="in_no_interaction", input_type=Boolean(optional=True), prefix="-noInteraction", doc=InputDocumentation(doc=": Disable inteaction annotations")), ToolInput(tag="in_no_motif", input_type=Boolean(optional=True), prefix="-noMotif", doc=InputDocumentation(doc=": Disable motif annotations.")), ToolInput(tag="in_no_next_prot", input_type=Boolean(optional=True), prefix="-noNextProt", doc=InputDocumentation(doc=": Disable NextProt annotations.")), ToolInput(tag="in_only_reg", input_type=Boolean(optional=True), prefix="-onlyReg", doc=InputDocumentation(doc=": Only use regulation tracks.")), ToolInput(tag="in_only_protein", input_type=Boolean(optional=True), prefix="-onlyProtein", doc=InputDocumentation(doc=": Only use protein coding transcripts. Default: false")), ToolInput(tag="in_only_tr", input_type=File(optional=True), prefix="-onlyTr", doc=InputDocumentation(doc=": Only use the transcripts in this file. Format: One transcript ID per line.")), ToolInput(tag="in_reg", input_type=String(optional=True), prefix="-reg", doc=InputDocumentation(doc=": Regulation track to use (this option can be used add several times).")), ToolInput(tag="in_splice_site_size", input_type=Int(optional=True), prefix="-spliceSiteSize", doc=InputDocumentation(doc=": Set size for splice sites (donor and acceptor) in bases. Default: 2")), ToolInput(tag="in_splice_region_exon_size", input_type=Int(optional=True), prefix="-spliceRegionExonSize", doc=InputDocumentation(doc=": Set size for splice site region within exons. Default: 3 bases")), ToolInput(tag="in_splice_region_intron_min", input_type=Int(optional=True), prefix="-spliceRegionIntronMin", doc=InputDocumentation(doc=": Set minimum number of bases for splice site region within intron. Default: 3 bases")), ToolInput(tag="in_splice_region_intron_max", input_type=Int(optional=True), prefix="-spliceRegionIntronMax", doc=InputDocumentation(doc=": Set maximum number of bases for splice site region within intron. Default: 8 bases")), ToolInput(tag="in_strict", input_type=Boolean(optional=True), prefix="-strict", doc=InputDocumentation(doc=": Only use 'validated' transcripts (i.e. sequence has been checked). Default: false")), ToolInput(tag="in_up_downstream_len", input_type=Int(optional=True), prefix="-upDownStreamLen", doc=InputDocumentation(doc=": Set upstream downstream interval length (in bases)")), ToolInput(tag="in_variants_file", input_type=String(), position=0, doc=InputDocumentation(doc=": Default is STDIN"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Snpeff_V0_1_0().translate("wdl", allow_empty_container=True)

