from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import String, File, Directory, Int, Float, Boolean

Targqc_V0_1_0 = CommandToolBuilder(tool="targqc", base_command=["targqc"], inputs=[ToolInput(tag="in_test", input_type=String(optional=True), prefix="--test", doc=InputDocumentation(doc="Quick test of correct installation.")), ToolInput(tag="in_amplicons", input_type=File(optional=True), prefix="--amplicons", doc=InputDocumentation(doc="BED file with regions to analyse. If not specified,\nCDS across full genome will be analysed")), ToolInput(tag="in_output_dir", input_type=Directory(optional=True), prefix="--output-dir", doc=InputDocumentation(doc="Output directory (or directory name in case of bcbio\nfinal dir)")), ToolInput(tag="in_genome", input_type=File(optional=True), prefix="--genome", doc=InputDocumentation(doc="Genome build (hg19 or hg38), used to pick genome\nannotation BED file if not specified, or path to .fa\nbwa prefix")), ToolInput(tag="in_bwa_bwa_prefix", input_type=File(optional=True), prefix="--bwa--bwa-prefix", doc=InputDocumentation(doc="Path to BWA index prefix to align if input is FastQ")), ToolInput(tag="in_fai", input_type=File(optional=True), prefix="--fai", doc=InputDocumentation(doc="Path to fai fasta index - sort BAM and BED files, and\nto get chromosome lengths for proper padding BED\nfiles; optional")), ToolInput(tag="in_padding", input_type=Int(optional=True), prefix="--padding", doc=InputDocumentation(doc="integer indicating the number of bases to extend each\ntarget region up and down-stream. Default is 200")), ToolInput(tag="in_down_sample_pairs_num", input_type=Int(optional=True), prefix="--downsample-pairs-num", doc=InputDocumentation(doc="If input is FastQ, select N random read pairs from\neach input set (instead of default fraction 0.05).")), ToolInput(tag="in_down_sample_fraction", input_type=Float(optional=True), prefix="--downsample-fraction", doc=InputDocumentation(doc="If input is FastQ, select K% random read pairs from\neach input set. Default is 0.05%. To turn off (align\nall reads), set --downsample 1")), ToolInput(tag="in_threads", input_type=Int(optional=True), prefix="--threads", doc=InputDocumentation(doc="Number of threads")), ToolInput(tag="in_reuse", input_type=Boolean(optional=True), prefix="--reuse", doc=InputDocumentation(doc="reuse intermediate non-empty files in the work dir\nfrom previous run")), ToolInput(tag="in_scheduler", input_type=String(optional=True), prefix="--scheduler", doc=InputDocumentation(doc="Scheduler to use for ipython parallel")), ToolInput(tag="in_queue", input_type=String(optional=True), prefix="--queue", doc=InputDocumentation(doc="Scheduler queue to run jobs on, for ipython parallel")), ToolInput(tag="in_resources", input_type=String(optional=True), prefix="--resources", doc=InputDocumentation(doc="Cluster specific resources specifications. Can be\nspecified multiple times. Supports SGE, Torque, LSF\nand SLURM parameters.")), ToolInput(tag="in_re_annotate", input_type=Boolean(optional=True), prefix="--reannotate", doc=InputDocumentation(doc="Re-annotate BED file with gene names, even if it's 4\ncolumns or more\n"))], outputs=[ToolOutput(tag="out_output_dir", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_output_dir", type_hint=File()), doc=OutputDocumentation(doc="Output directory (or directory name in case of bcbio\nfinal dir)"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Targqc_V0_1_0().translate("wdl", allow_empty_container=True)

