from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Float, Int, Boolean, Directory

Metagenome_Pipeline_Py_V0_1_0 = CommandToolBuilder(tool="metagenome_pipeline.py", base_command=["metagenome_pipeline.py"], inputs=[ToolInput(tag="in_input", input_type=File(optional=True), prefix="--input", doc=InputDocumentation(doc="Input table of sequence abundances (BIOM, TSV, or\nmothur shared file format).")), ToolInput(tag="in_function", input_type=File(optional=True), prefix="--function", doc=InputDocumentation(doc="Table of predicted gene family copy numbers (output of\nhsp.py).")), ToolInput(tag="in_marker", input_type=File(optional=True), prefix="--marker", doc=InputDocumentation(doc="Table of predicted marker gene copy numbers (output of\nhsp.py - typically for 16S).")), ToolInput(tag="in_max_nsti", input_type=Float(optional=True), prefix="--max_nsti", doc=InputDocumentation(doc="Sequences with NSTI values above this value will be\nexcluded (default: 2).")), ToolInput(tag="in_min_reads", input_type=Int(optional=True), prefix="--min_reads", doc=InputDocumentation(doc="Minimum number of reads across all samples for each\ninput ASV. ASVs below this cut-off will be counted as\npart of the 'RARE' category in the stratified output\n(default: 1).")), ToolInput(tag="in_min_samples", input_type=Int(optional=True), prefix="--min_samples", doc=InputDocumentation(doc="Minimum number of samples that an ASV needs to be\nidentfied within. ASVs below this cut-off will be\ncounted as part of the 'RARE' category in the\nstratified output (default: 1).")), ToolInput(tag="in_strat_out", input_type=Boolean(optional=True), prefix="--strat_out", doc=InputDocumentation(doc="Output table stratified by sequences as well. By\ndefault this will be in 'contributional' format (i.e.\nlong-format) unless the '--wide_table' option is set.\nThe startified outfile is named\n'metagenome_contrib.tsv.gz' when in long-format.")), ToolInput(tag="in_wide_table", input_type=Boolean(optional=True), prefix="--wide_table", doc=InputDocumentation(doc="Output wide-format stratified table of metagenome\npredictions when '--strat_out' is set. This is the\ndeprecated method of generating stratified tables\nsince it is extremely memory intensive. The startified\noutfile is named 'pred_metagenome_strat.tsv.gz' when\nthis option is set.")), ToolInput(tag="in_skip_norm", input_type=Boolean(optional=True), prefix="--skip_norm", doc=InputDocumentation(doc="Skip normalizing sequence abundances by predicted\nmarker gene copy numbers (typically 16S rRNA genes).\nThis step will be performed automatically unless this\noption is specified.")), ToolInput(tag="in_out_dir", input_type=Directory(optional=True), prefix="--out_dir", doc=InputDocumentation(doc="Output directory for metagenome predictions. (default:\nmetagenome_out)."))], outputs=[ToolOutput(tag="out_out_dir", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_out_dir", type_hint=File()), doc=OutputDocumentation(doc="Output directory for metagenome predictions. (default:\nmetagenome_out)."))], container="quay.io/biocontainers/picrust2:2.4.1--py_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Metagenome_Pipeline_Py_V0_1_0().translate("wdl")

