from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, Int, String, File

Last_Dotplot_V0_1_0 = CommandToolBuilder(tool="last_dotplot", base_command=["last-dotplot"], inputs=[ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="--verbose", doc=InputDocumentation(doc="show progress messages & data about the plot")), ToolInput(tag="in_width", input_type=Int(optional=True), prefix="--width", doc=InputDocumentation(doc="maximum width in pixels (default: 1000)")), ToolInput(tag="in_height", input_type=Int(optional=True), prefix="--height", doc=InputDocumentation(doc="maximum height in pixels (default: 1000)")), ToolInput(tag="in_max_seqs", input_type=Int(optional=True), prefix="--maxseqs", doc=InputDocumentation(doc="maximum number of horizontal or vertical sequences\n(default=100)")), ToolInput(tag="in_seq_one", input_type=Int(optional=True), prefix="--seq1", doc=InputDocumentation(doc="which sequences to show from the 1st genome")), ToolInput(tag="in_seq_two", input_type=Int(optional=True), prefix="--seq2", doc=InputDocumentation(doc="which sequences to show from the 2nd genome")), ToolInput(tag="in_forward_color", input_type=String(optional=True), prefix="--forwardcolor", doc=InputDocumentation(doc="color for forward alignments (default: red)")), ToolInput(tag="in_reverse_color", input_type=String(optional=True), prefix="--reversecolor", doc=InputDocumentation(doc="color for reverse alignments (default: blue)")), ToolInput(tag="in_alignments", input_type=File(optional=True), prefix="--alignments", doc=InputDocumentation(doc="secondary alignments")), ToolInput(tag="in_sort_one", input_type=Int(optional=True), prefix="--sort1", doc=InputDocumentation(doc="genome1 sequence order: 0=input order, 1=name order,\n2=length order, 3=alignment order (default=1)")), ToolInput(tag="in_sort_two", input_type=Int(optional=True), prefix="--sort2", doc=InputDocumentation(doc="genome2 sequence order: 0=input order, 1=name order,\n2=length order, 3=alignment order (default=1)")), ToolInput(tag="in_strands_one", input_type=Int(optional=True), prefix="--strands1", doc=InputDocumentation(doc="genome1 sequence orientation: 0=forward orientation,\n1=alignment orientation (default=0)")), ToolInput(tag="in_strands_two", input_type=Int(optional=True), prefix="--strands2", doc=InputDocumentation(doc="genome2 sequence orientation: 0=forward orientation,\n1=alignment orientation (default=0)")), ToolInput(tag="in_max_gap_one", input_type=Int(optional=True), prefix="--max-gap1", doc=InputDocumentation(doc="maximum unaligned (end,mid) gap in genome1: fraction\nof aligned length (default=0.5,2)")), ToolInput(tag="in_max_gap_two", input_type=Int(optional=True), prefix="--max-gap2", doc=InputDocumentation(doc="maximum unaligned (end,mid) gap in genome2: fraction\nof aligned length (default=0.5,2)")), ToolInput(tag="in_pad", input_type=Int(optional=True), prefix="--pad", doc=InputDocumentation(doc="pad length when cutting unaligned gaps: fraction of\naligned length (default=0.04)")), ToolInput(tag="in_join", input_type=Int(optional=True), prefix="--join", doc=InputDocumentation(doc="join: 0=nothing, 1=alignments adjacent in genome1,\n2=alignments adjacent in genome2 (default=0)")), ToolInput(tag="in_border_pixels", input_type=Int(optional=True), prefix="--border-pixels", doc=InputDocumentation(doc="number of pixels between sequences (default=1)")), ToolInput(tag="in_border_color", input_type=String(optional=True), prefix="--border-color", doc=InputDocumentation(doc="color for pixels between sequences (default=black)")), ToolInput(tag="in_margin_color", input_type=String(optional=True), prefix="--margin-color", doc=InputDocumentation(doc="margin color")), ToolInput(tag="in_font_file", input_type=File(optional=True), prefix="--fontfile", doc=InputDocumentation(doc="TrueType or OpenType font file")), ToolInput(tag="in_font_size", input_type=Int(optional=True), prefix="--fontsize", doc=InputDocumentation(doc="TrueType or OpenType font size (default: 14)")), ToolInput(tag="in_labels_one", input_type=Int(optional=True), prefix="--labels1", doc=InputDocumentation(doc="genome1 labels: 0=name, 1=name:length,\n2=name:start:length, 3=name:start-end (default=0)")), ToolInput(tag="in_labels_two", input_type=Int(optional=True), prefix="--labels2", doc=InputDocumentation(doc="genome2 labels: 0=name, 1=name:length,\n2=name:start:length, 3=name:start-end (default=0)")), ToolInput(tag="in_rot_one", input_type=Int(optional=True), prefix="--rot1", doc=InputDocumentation(doc="text rotation for the 1st genome (default=h)")), ToolInput(tag="in_rot_two", input_type=Int(optional=True), prefix="--rot2", doc=InputDocumentation(doc="text rotation for the 2nd genome (default=v)")), ToolInput(tag="in_be_done", input_type=File(optional=True), prefix="--bed1", doc=InputDocumentation(doc="read genome1 annotations from BED file")), ToolInput(tag="in_bed_two", input_type=File(optional=True), prefix="--bed2", doc=InputDocumentation(doc="read genome2 annotations from BED file")), ToolInput(tag="in_rms_kone", input_type=File(optional=True), prefix="--rmsk1", doc=InputDocumentation(doc="read genome1 repeats from RepeatMasker .out or\nrmsk.txt file")), ToolInput(tag="in_rms_k_two", input_type=File(optional=True), prefix="--rmsk2", doc=InputDocumentation(doc="read genome2 repeats from RepeatMasker .out or\nrmsk.txt file")), ToolInput(tag="in_gene_pre_done", input_type=File(optional=True), prefix="--genePred1", doc=InputDocumentation(doc="read genome1 genes from genePred file")), ToolInput(tag="in_gene_pred_two", input_type=File(optional=True), prefix="--genePred2", doc=InputDocumentation(doc="read genome2 genes from genePred file")), ToolInput(tag="in_exon_color", input_type=String(optional=True), prefix="--exon-color", doc=InputDocumentation(doc="color for exons (default=PaleGreen)")), ToolInput(tag="in_cds_color", input_type=String(optional=True), prefix="--cds-color", doc=InputDocumentation(doc="color for protein-coding regions (default=LimeGreen)")), ToolInput(tag="in_gap_one", input_type=File(optional=True), prefix="--gap1", doc=InputDocumentation(doc="read genome1 unsequenced gaps from agp or gap file")), ToolInput(tag="in_gap_two", input_type=File(optional=True), prefix="--gap2", doc=InputDocumentation(doc="read genome2 unsequenced gaps from agp or gap file")), ToolInput(tag="in_bridged_color", input_type=String(optional=True), prefix="--bridged-color", doc=InputDocumentation(doc="color for bridged gaps (default: yellow)")), ToolInput(tag="in_un_bridged_color", input_type=String(optional=True), prefix="--unbridged-color", doc=InputDocumentation(doc="color for unbridged gaps (default: orange)\n"))], outputs=[ToolOutput(tag="out_rms_kone", output_type=File(optional=True), selector=InputSelector(input_to_select="in_rms_kone", type_hint=File()), doc=OutputDocumentation(doc="read genome1 repeats from RepeatMasker .out or\nrmsk.txt file")), ToolOutput(tag="out_rms_k_two", output_type=File(optional=True), selector=InputSelector(input_to_select="in_rms_k_two", type_hint=File()), doc=OutputDocumentation(doc="read genome2 repeats from RepeatMasker .out or\nrmsk.txt file"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Last_Dotplot_V0_1_0().translate("wdl", allow_empty_container=True)

