from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Int, String, File, Directory, Boolean

Isaac_Align_V0_1_0 = CommandToolBuilder(tool="isaac_align", base_command=["isaac-align"], inputs=[ToolInput(tag="in_allow_empty_flow_cells", input_type=Int(optional=True), prefix="--allow-empty-flowcells", doc=InputDocumentation(doc="(=0)                Avoid failure when some of\nthe --base-calls contain no\ndata")), ToolInput(tag="in_anchor_mate", input_type=Int(optional=True), prefix="--anchor-mate", doc=InputDocumentation(doc="(=1)                          Allow entire pair to be\nanchored by only one read if\nit has not been realigned. If\nnot set, each read is\nanchored individually and\ndoes not affect anchoring of\nits mate.")), ToolInput(tag="in_anomalous_pair_handicap", input_type=Int(optional=True), prefix="--anomalous-pair-handicap", doc=InputDocumentation(doc="(=240)            When deciding between an\nanomalous pair and a rescued\npair, this is proportional to\nthe number of mismatches\nanomalous pair needs to have\nless in order to be accepted\ninstead of a rescued pair.")), ToolInput(tag="in_bam_exclude_tags", input_type=String(optional=True), prefix="--bam-exclude-tags", doc=InputDocumentation(doc="(=ZX,ZY)                 Comma-separated list of\nregular tags to exclude from\nthe output BAM files. Allowed\nvalues are: all,none,AS,BC,NM\n,OC,RG,SM,ZX,ZY")), ToolInput(tag="in_bam_gzip_level", input_type=Int(optional=True), prefix="--bam-gzip-level", doc=InputDocumentation(doc="(=1)                       Gzip level to use for BAM")), ToolInput(tag="in_bam_header_tag", input_type=File(optional=True), prefix="--bam-header-tag", doc=InputDocumentation(doc="Additional bam entries that\nare copied into the header of\neach produced bam file. Use\n'\t' to represent tab\nseparators.")), ToolInput(tag="in_bam_pessimistic_mapq", input_type=Int(optional=True), prefix="--bam-pessimistic-mapq", doc=InputDocumentation(doc="(=0)                 When set, the MAPQ is\ncomputed as MAPQ:=min(60,\nmin(SM, AS)), otherwise\nMAPQ:=min(60, max(SM, AS))")), ToolInput(tag="in_bam_produce_md_five", input_type=File(optional=True), prefix="--bam-produce-md5", doc=InputDocumentation(doc="(=1)                      Controls whether a separate\nfile containing md5 checksum\nis produced for each output\nbam.")), ToolInput(tag="in_bam_pu_format", input_type=Int(optional=True), prefix="--bam-pu-format", doc=InputDocumentation(doc="(=%F:%L:%B)                 Template string for bam\nheader RG tag PU field.\nOrdinary characters are\ndirectly copied. The\nfollowing placeholders are\nsupported:\n- %F             : Flowcell\nID\n- %L             : Lane\nnumber\n- %B             : Barcode")), ToolInput(tag="in_barcode_mismatches", input_type=Directory(optional=True), prefix="--barcode-mismatches", doc=InputDocumentation(doc="(=1)                   Multiple entries allowed.\nEach entry is applied to the\ncorresponding base-calls.\nLast entry applies to all the\nbases-calls-directory that do\nnot have barcode-mismatches\nspecified. Last component\nmismatch value applies to all\nsubsequent barcode components\nshould there be more than\none. Examples:\n- 1:0             : allow\none mismatch for the first\nbarcode component and no\nmismatches for the subsequent\ncomponents.\n- 1               : allow\none mismatch for every\nbarcode component.\n- 0               : no\nmismatches allowed in any\nbarcode component. This is\nthe default.")), ToolInput(tag="in_arg_full_path_base", input_type=Boolean(optional=True), prefix="-b", doc=InputDocumentation(doc="[ --base-calls ] arg                         full path to the base calls.\nMultiple entries allowed.\nPath should point either to a\ndirectory or a file depending\non --base-calls-format")), ToolInput(tag="in_arg_multiple_entries", input_type=Boolean(optional=True), prefix="-f", doc=InputDocumentation(doc="[ --base-calls-format ] arg                  Multiple entries allowed.\nEach entry is applied to the\ncorresponding base-calls.\nLast entry is applied to all\n--base-calls that don't have\n--base-calls-format\nspecified.\n- bam             :\n--base-calls points to a Bam\nfile. All data found in bam\nfile is assumed to come from\nlane 1 of a single flowcell.\n- bcl             :\n--base-calls points to\nRunInfo.xml file. Data is\nmade of uncompressed bcl\nfiles.\n- bcl-gz          :\n--base-calls points to\nRunInfo.xml file. Bcl cycle\ntile files are individually\ncompressed and named\ns_X_YYYY.bcl.gz\n- bcl-bgzf        :\n--base-calls points to\nRunInfo.xml file. Bcl data is\nstored in cycle files that\nare named CCCC.bcl.bgzf\n- fastq           :\n--base-calls points to a\ndirectory containing one\nfastq per lane/read named\nlane<X>_read<Y>.fastq. Use\nlane<X>_read1.fastq for\nsingle-ended data.\n- fastq-gz        :\n--base-calls points to a\ndirectory containing one\ncompressed fastq per\nlane/read named\nlane<X>_read<Y>.fastq.gz. Use\nlane<X>_read1.fastq.gz for\nsingle-ended data.")), ToolInput(tag="in_base_quality_cut_off", input_type=Int(optional=True), prefix="--base-quality-cutoff", doc=InputDocumentation(doc="(=15)                 3' end quality trimming\ncutoff. Value above 0 causes\nlow quality bases to be\nsoft-clipped. 0 turns the\ntrimming off.")), ToolInput(tag="in_bcl_tiles_per_chunk", input_type=Int(optional=True), prefix="--bcl-tiles-per-chunk", doc=InputDocumentation(doc="(=1)                  Increase this number when the\ntiles are too small for the\nprocessing to be efficient.\nIn particular, collecting the\ntemplate length statistics\nrequires several tens of\nthousands clusters to work.\nIf tiles are small and data\nis heavily multiplexed, there\nmight be not enough clusters\nin a single tile to collect\nthe tls for a sample")), ToolInput(tag="in_bin_regex", input_type=String(optional=True), prefix="--bin-regex", doc=InputDocumentation(doc="(=all)                          Define which bins appear in\nthe output bam files\nall                   :\nInclude all bins in the bam\nand all contig entries in the\nbam header.\nskip-empty             :\nInclude only the contigs that\nhave aligned data.\nREGEX                 : Is\ntreated as comma-separated\nlist of regular expressions.\nBam files will be filtered to\ncontain only the bins that\nmatch by the name.")), ToolInput(tag="in_candidate_matches_max", input_type=Int(optional=True), prefix="--candidate-matches-max", doc=InputDocumentation(doc="(=800)              Maximum number of candidate\nmatches to be considered for\nfinding the best alignment.\nIf seeds yield a greater\nnumber, the alignment\ngenerally is not performed.\nOther mechanisms such as\nshadow rescue may still place\nthe fragment.")), ToolInput(tag="in_cleanup_intermediary", input_type=Int(optional=True), prefix="--cleanup-intermediary", doc=InputDocumentation(doc="(=0)                 When set, Isaac will erase\nintermediate input files for\nthe stages that have been\ncompleted. Notice that this\nwill prevent resumption from\nthe stages that have their\ninput files removed.\n--start-from Last will still\nwork.")), ToolInput(tag="in_clip_overlapping", input_type=Int(optional=True), prefix="--clip-overlapping", doc=InputDocumentation(doc="(=1)                     When set, the pairs that have\nread ends overlapping each\nother will have the\nlower-quality end\nsoft-clipped.")), ToolInput(tag="in_clip_semi_aligned", input_type=Int(optional=True), prefix="--clip-semialigned", doc=InputDocumentation(doc="(=0)                     When set, reads have their\nbases soft-clipped on either\nsides until a stretch of 5\nmatches is found")), ToolInput(tag="in_arg_restrict_alignment", input_type=Boolean(optional=True), prefix="-c", doc=InputDocumentation(doc="[ --cluster ] arg                            Restrict the alignment to the\nspecified cluster Id\n(multiple entries allowed)")), ToolInput(tag="in_clusters_at_a_time", input_type=Int(optional=True), prefix="--clusters-at-a-time", doc=InputDocumentation(doc="(=8000000)             Bam and fastq only. When not\nset, number of clusters to\nprocess together when input\nis bam or fastq is computed\nautomatically based on the\namount of available RAM. Set\nto non-zero value to force\ndeterministic behavior.")), ToolInput(tag="in_decoy_regex", input_type=String(optional=True), prefix="--decoy-regex", doc=InputDocumentation(doc="(=decoy)                      Contigs that have matching\nnames are marked as decoys\nand enjoy reduced effort. In\nparticular:\n- Smith waterman is not\nused for alignments\n- Suspicious alignments are\nmarked dodgyFor example, to\nmark everything that does not\nbegin with chr as decoy use\nthe following regex:\n^(?!chr.*)")), ToolInput(tag="in_default_adapters", input_type=Int(optional=True), prefix="--default-adapters", doc=InputDocumentation(doc="Multiple entries allowed.\nEach entry is associated with\nthe corresponding base-calls.\nFlowcells that don't have\ndefault-adapters provided,\ndon't get adapters clipped in\nthe data.\nEach entry is a\ncomma-separated list of\nadapter sequences written in\nthe direction of the\nreference. Wildcard (*\ncharacter) is allowed only on\none side of the sequence.\nEntries with * apply only to\nthe alignments on the\nmatching strand. Entries\nwithout * apply to all strand\nalignments and are matched in\nthe order of appearance in\nthe list.\nExamples:\nACGT*,*TGCA       : Will\nclip ACGT and all subsequent\nbases in the forward-strand\nalignments and mirror the\nbehavior for the\nreverse-strand alignments.\nACGT,TGCA         : Will\nfind the following sequences\nin the reads: ACGT, TGCA,\nACGTTGCA  (but not TGCAACGT!)\nregardless of the alignment\nstrand. Then will attempt to\nclip off the side of the read\nthat is shorter. If both\nsides are roughly equal\nlength, will clip off the\nside that has less matches.\nStandard          :\nStandard protocol adapters.\nSame as AGATCGGAAGAGC*,*GCTCT\nTCCGATCT\nNextera           : Nextera\nstandard. Same as\nCTGTCTCTTATACACATCT*,*AGATGTG\nTATAAGAGACAG\nNexteraMp         : Nextera\nmate-pair. Same as\nCTGTCTCTTATACACATCT,AGATGTGTA\nTAAGAGACAG")), ToolInput(tag="in_description", input_type=String(optional=True), prefix="--description", doc=InputDocumentation(doc="Free form text to be stored\nin the Isaac @PG DS bam\nheader tag")), ToolInput(tag="in_detect_template_block_size", input_type=Int(optional=True), prefix="--detect-template-block-size", doc=InputDocumentation(doc="(=10000)       Number of pairs to use as a\nsingle block for template\nlength statistics detection")), ToolInput(tag="in_disable_resume", input_type=Int(optional=True), prefix="--disable-resume", doc=InputDocumentation(doc="(=0)                       If eanbled, Isaac does not\npersist the state of the\nanalysis on disk. This might\nsave noticeable amount of\nruntime at the expense of not\nbeing able to use\n--start-from option.")), ToolInput(tag="in_dodgy_alignment_score", input_type=Int(optional=True), prefix="--dodgy-alignment-score", doc=InputDocumentation(doc="(=0)                Controls the behavior for\ntemplates where alignment\nscore is impossible to\nassign:\n- Unaligned        : marks\ntemplate fragments as\nunaligned\n- 0-254            : exact\nMAPQ value to be set in bam\n- Unknown          : assigns\nvalue 255 for bam MAPQ.\nEnsures SM and AS are not\nspecified in the bam")), ToolInput(tag="in_enable_numa", input_type=Boolean(optional=True), prefix="--enable-numa", doc=InputDocumentation(doc="[=arg(=1)] (=0)                   Replicate static data across\nNUMA nodes, lock threads to\ntheir NUMA nodes, allocate\nthread private data on the\ncorresponding NUMA node")), ToolInput(tag="in_expected_bg_zf_ratio", input_type=Int(optional=True), prefix="--expected-bgzf-ratio", doc=InputDocumentation(doc="(=1)                  compressed = ratio *\nuncompressed. To avoid memory\noverallocation during the bam\ngeneration, Isaac has to\nassume certain compression\nratio. If Isaac estimates\nless memory than is actually\nrequired, it will fail at\nruntime. You can check how\nfar you are from the\ndangerous zone by looking at\nthe resident/swap memory\nnumbers for your process\nduring the bam generation. If\nyou see too much showing as\n'swap', it is safe to reduce\nthe --expected-bgzf-ratio.")), ToolInput(tag="in_expected_coverage", input_type=Int(optional=True), prefix="--expected-coverage", doc=InputDocumentation(doc="(=60)                   Expected coverage is required\nfor Isaac to estimate the\nefficient binning of the\naligned data.")), ToolInput(tag="in_fast_q_q_zero", input_type=Int(optional=True), prefix="--fastq-q0", doc=InputDocumentation(doc="(=!)                             Character to serve as base\nquality 0 in fastq input.")), ToolInput(tag="in_gap_scoring", input_type=Int(optional=True), prefix="--gap-scoring", doc=InputDocumentation(doc="(=bwa)                        Gapped alignment algorithm\nparameters:\n- eland            :\nequivalent of 2:-1:-15:-3:-25\n- bwa              :\nequivalent of 0:-3:-11:-4:-20\n- bwa-mem          :\nequivalent of 1:-4:-6:-1:-20\n- m:mm:go:ge:me:gl :\ncolon-delimited string of\nvalues where:\nm              : match\nscore\nmm             :\nmismatch score\ngo             : gap\nopen score\nge             : gap\nextend score\nme             : min\nextend score (all gaps\nreaching this score will be\ntreated as equal)")), ToolInput(tag="in_hash_table_buckets", input_type=Int(optional=True), prefix="--hash-table-buckets", doc=InputDocumentation(doc="(=0)                   Number of buckets to use for\nreference hash table. Larger\nnumber of buckets requires\nmore RAM but it tends to\nspeed up the execution and\nimprove sensitivity. Value of\n0 indicates default bucket\ncount: 2^({seed-length}*2)")), ToolInput(tag="in_help_defaults", input_type=Boolean(optional=True), prefix="--help-defaults", doc=InputDocumentation(doc="produce tab-delimited list of\ncommand line options and\ntheir default values")), ToolInput(tag="in_help_md", input_type=Boolean(optional=True), prefix="--help-md", doc=InputDocumentation(doc="produce help message\npre-formatted as a markdown\nfile section and exit")), ToolInput(tag="in_ignore_missing_bcls", input_type=File(optional=True), prefix="--ignore-missing-bcls", doc=InputDocumentation(doc="(=0)                  When set, missing bcl files\nare treated as all clusters\nhaving N bases for the\ncorresponding tile cycle.\nOtherwise, encountering a\nmissing bcl file causes the\nanalysis to fail.")), ToolInput(tag="in_ignore_missing_filters", input_type=File(optional=True), prefix="--ignore-missing-filters", doc=InputDocumentation(doc="(=0)               When set, missing filter\nfiles are treated as if all\nclusters pass filter for the\ncorresponding tile.\nOtherwise, encountering a\nmissing filter file causes\nthe analysis to fail.")), ToolInput(tag="in_input_concurrent_load", input_type=Int(optional=True), prefix="--input-concurrent-load", doc=InputDocumentation(doc="(=64)               Maximum number of concurrent\nfile read operations for\n--base-calls")), ToolInput(tag="in_arg_maximum_number", input_type=Boolean(optional=True), prefix="-j", doc=InputDocumentation(doc="[ --jobs ] arg (=8)                          Maximum number of compute\nthreads to run in parallel")), ToolInput(tag="in_keep_duplicates", input_type=File(optional=True), prefix="--keep-duplicates", doc=InputDocumentation(doc="(=1)                      Keep duplicate pairs in the\nbam file (with 0x400 flag set\nin all but the best one)")), ToolInput(tag="in_keep_unaligned", input_type=File(optional=True), prefix="--keep-unaligned", doc=InputDocumentation(doc="(=back)                    Available options:\n- discard          : discard\nclusters where both reads are\nnot aligned\n- front            : keep\nunaligned clusters in the\nfront of the BAM file\n- back             : keep\nunaligned clusters in the\nback of the BAM file")), ToolInput(tag="in_known_indels", input_type=File(optional=True), prefix="--known-indels", doc=InputDocumentation(doc="path to a VCF file containing\nknown indels fore\nrealignment.")), ToolInput(tag="in_lane_number_max", input_type=Int(optional=True), prefix="--lane-number-max", doc=InputDocumentation(doc="(=8)                      Maximum lane number to look\nfor in --base-calls-directory\n(fastq only).")), ToolInput(tag="in_mapq_threshold", input_type=Int(optional=True), prefix="--mapq-threshold", doc=InputDocumentation(doc="(=-1)                      If any fragment alignment in\ntemplate is below the\nthreshold, template is not\nstored in the BAM.")), ToolInput(tag="in_mark_duplicates", input_type=Int(optional=True), prefix="--mark-duplicates", doc=InputDocumentation(doc="(=1)                      If not set and --keep-duplica\ntes is set, the duplicates\nare not discarded and not\nflagged.")), ToolInput(tag="in_match_finder_shadow_split_repeats", input_type=Int(optional=True), prefix="--match-finder-shadow-split-repeats", doc=InputDocumentation(doc="(=100000)\nMaximum number of seed\ncandidate matches to be\nconsidered for finding a\npossible alignment split.")), ToolInput(tag="in_match_finder_too_many_repeats", input_type=Int(optional=True), prefix="--match-finder-too-many-repeats", doc=InputDocumentation(doc="(=4000)     Maximum number of seed\nmatches to be looked at for\neach attempted seed")), ToolInput(tag="in_match_finder_way_too_many_repeats", input_type=Int(optional=True), prefix="--match-finder-way-too-many-repeats", doc=InputDocumentation(doc="(=100000)\nMaximum number of seed\nmatches to be looked at if in\na pair one read has candidate\nalignments and the otherhas\ngone over match-finder-too-ma\nny-repeats on all seeds or\nover candidate-matches-max\nwhen seed position merge was\nattempted")), ToolInput(tag="in_memory_control", input_type=String(optional=True), prefix="--memory-control", doc=InputDocumentation(doc="(=off)                     Define the behavior in case\nunexpected memory allocations\nare detected:\n- warning         : Log\nWARNING about the allocation.\n- off             : Don't\nmonitor dynamic memory usage.\n- strict          : Fail\nmemory allocation. Intended\nfor development use.")), ToolInput(tag="in_arg_limits_operations", input_type=Boolean(optional=True), prefix="-m", doc=InputDocumentation(doc="[ --memory-limit ] arg (=0)                  Limits major memory\nconsumption operations to a\nset number of gigabytes. 0\nmeans no limit, however 0 is\nnot allowed as in such case\nIsaac will most likely\nconsume all the memory on the\nsystem and cause it to crash.\nDefault value is taken from\nulimit -v.")), ToolInput(tag="in_neighborhood_size_threshold", input_type=Int(optional=True), prefix="--neighborhood-size-threshold", doc=InputDocumentation(doc="(=0)          Threshold used to decide if\nthe number of reference\n32-mers sharing the same\nprefix (16 bases) is small\nenough to justify the\nneighborhood search. Use\nlarge enough value e.g. 10000\nto enable alignment to\npositions where seeds don't\nmatch exactly.")), ToolInput(tag="in_output_concurrent_save", input_type=Int(optional=True), prefix="--output-concurrent-save", doc=InputDocumentation(doc="(=120)             Maximum number of concurrent\nfile write operations for\n--output-directory")), ToolInput(tag="in_arg_directory_stored", input_type=Directory(optional=True), prefix="-o", doc=InputDocumentation(doc="[ --output-directory ] arg (=./Aligned)      Directory where the final\nalignment data be stored")), ToolInput(tag="in_per_tile_tls", input_type=Int(optional=True), prefix="--per-tile-tls", doc=InputDocumentation(doc="(=0)                         Forces template length\nstatistics(TLS) to be\nrecomputed for each tile.\nWhen not set, the first tile\nthat produces stable TLS will\ndetermine TLS for the rest of\nthe tiles of the lane. Notice\nthat as the tiles are not\nguaranteed to be processed in\nthe same order between\ndifferent runs, some pair\nalignments might vary between\ntwo runs on the same data\nunless --per-tile-tls is set.\nIt is not recommended to set\n--per-tile-tls when input\ndata is not randomly\ndistributed (such as bam) as\nin such cases, the shadow\nrescue range will be biased\nby the input data ordering.")), ToolInput(tag="in_pf_only", input_type=File(optional=True), prefix="--pf-only", doc=InputDocumentation(doc="(=1)                              When set, only the fragments\npassing filter (PF) are\ngenerated in the BAM file")), ToolInput(tag="in_pre_allocate_bins", input_type=Int(optional=True), prefix="--pre-allocate-bins", doc=InputDocumentation(doc="(=0)                    Use fallocate to reduce the\nbin file fragmentation. Since\nbin files are pre-allocated\nbased on the estimation of\ntheir size, it is recommended\nto turn bin pre-allocation\noff when using RAM disk as\ntemporary storage.")), ToolInput(tag="in_pre_sort_bins", input_type=Int(optional=True), prefix="--pre-sort-bins", doc=InputDocumentation(doc="(=1)                        Unset this value if you are\nworking with references that\nhave many contigs (1000+)")), ToolInput(tag="in_read_name_length", input_type=Int(optional=True), prefix="--read-name-length", doc=InputDocumentation(doc="(=0)                     Maximum read name length\n(fastq and bam only). Value\nof 0 causes the read name\nlength to be determined by\nreading the first records of\nthe input data. Shorter than\nneeded read names can cause\nduplicate names in the output\nbam files.")), ToolInput(tag="in_realign_dodgy", input_type=Int(optional=True), prefix="--realign-dodgy", doc=InputDocumentation(doc="(=0)                        If not set, the reads without\nalignment score are not\nrealigned against gaps found\nin other reads.")), ToolInput(tag="in_realign_gaps", input_type=Int(optional=True), prefix="--realign-gaps", doc=InputDocumentation(doc="(=sample)                    For reads overlapping the\ngaps occurring on other\nreads, check if applying\nthose gaps reduces mismatch\ncount. Significantly reduces\nnumber of false SNPs reported\naround short indels.\n- no              : no gap\nrealignment\n- sample          : realign\nagainst gaps found in the\nsame sample\n- project         : realign\nagainst gaps found in all\nsamples of the same project\n- all             : realign\nagainst gaps found in all\nsamples")), ToolInput(tag="in_realign_mapq_min", input_type=Int(optional=True), prefix="--realign-mapq-min", doc=InputDocumentation(doc="(=60)                    Gaps from alignments with\nlower MAPQ will not be used\nas candidates for gap\nrealignment")), ToolInput(tag="in_realign_vigorously", input_type=Int(optional=True), prefix="--realign-vigorously", doc=InputDocumentation(doc="(=0)                   If set, the realignment\nresult will be used to search\nfor more gaps and attempt\nanother realignment,\neffectively extending the\nrealignment over multiple\ndeletions not covered by the\noriginal alignment.")), ToolInput(tag="in_realigned_gaps_per_fragment", input_type=Int(optional=True), prefix="--realigned-gaps-per-fragment", doc=InputDocumentation(doc="(=4)          Maximum number of gaps the\nrealigner can introduce into\na fragment. For 100 bases\nlong DNA it is reasonable to\nkeep it no bigger than 2. RNA\nreads can overlap multiple\nintrons. Therefore a larger\nnumber is probably required\nfor RNA. Notice that bigger\nvalues can significantly slow\ndown the bam generation as\nthere is a n choose k\ncombination try with n being\nthe number of gaps detected\nby all other fragment\nalignments that overlap the\nfragment being realigned.")), ToolInput(tag="in_arg_full_path_referencegenome", input_type=Boolean(optional=True), prefix="-r", doc=InputDocumentation(doc="[ --reference-genome ] arg                   Full path to the reference\ngenome XML descriptor.")), ToolInput(tag="in_arg_default_name", input_type=Boolean(optional=True), prefix="-n", doc=InputDocumentation(doc="[ --reference-name ] arg (=default)          Unique symbolic name of the\nreference. Multiple entries\nallowed. Each entry is\nassociated with the\ncorresponding --reference-gen\nome and will be matched\nagainst the 'reference'\ncolumn in the sample sheet.\nSpecial names:\n- unknown         : default\nreference to use with data\nthat did not match any\nbarcode.\n- default         :\nreference to use for the data\nwith no matching value in\nsample sheet 'reference'\ncolumn.")), ToolInput(tag="in_remap_q_scores", input_type=Int(optional=True), prefix="--remap-qscores", doc=InputDocumentation(doc="Replace the base calls\nqscores according to the\nrules provided.\n- identity   : No remapping.\nOriginal qscores are\npreserved\n- bin:8      : Equivalent of\n0-1:0,2-9:7,10-19:11,20-24:22\n,25-29:27,30-34:32,35-39:37,4\n0-63:40")), ToolInput(tag="in_repeat_threshold", input_type=Int(optional=True), prefix="--repeat-threshold", doc=InputDocumentation(doc="(=100)                   Threshold used to decide if\nmatches must be discarded as\ntoo abundant (when the number\nof repeats is greater or\nequal to the threshold)")), ToolInput(tag="in_rescue_shadows", input_type=Int(optional=True), prefix="--rescue-shadows", doc=InputDocumentation(doc="(=1)                       Scan within dominant template\nrange off an orphan, for a\npossible shadow alignment")), ToolInput(tag="in_response_file", input_type=File(optional=True), prefix="--response-file", doc=InputDocumentation(doc="file with more command line")), ToolInput(tag="in_scatter_repeats", input_type=Int(optional=True), prefix="--scatter-repeats", doc=InputDocumentation(doc="(=1)                      When set, extra care will be\ntaken to scatter pairs\naligning to repeats across\nthe repeat locations")), ToolInput(tag="in_seed_base_quality_min", input_type=Int(optional=True), prefix="--seed-base-quality-min", doc=InputDocumentation(doc="(=3)                Minimum base quality for the\nseed to be used in alignment\ncandidate search.")), ToolInput(tag="in_seed_length", input_type=Int(optional=True), prefix="--seed-length", doc=InputDocumentation(doc="(=16)                         Length of the seed in bases.\nOnly 10 11 12 13 14 15 16 17\n18 19 20  are allowed. Longer\nseeds reduce sensitivity on\nnoisy data but improve repeat\nresolution and run time.")), ToolInput(tag="in_shadow_scan_range", input_type=Int(optional=True), prefix="--shadow-scan-range", doc=InputDocumentation(doc="(=-1)                   -1     - scan for possible\nmate alignments between\ntemplate min and max\n>=0    - scan for possible\nmate alignments in range of\ntemplate median +=\nshadow-scan-range")), ToolInput(tag="in_single_library_samples", input_type=Int(optional=True), prefix="--single-library-samples", doc=InputDocumentation(doc="(=1)               If set, the duplicate\ndetection will occur across\nall read pairs in the sample.\nIf not set, different lanes\nare assumed to originate from\ndifferent libraries and\nduplicate detection is not\nperformed across lanes.")), ToolInput(tag="in_smith_waterman_gap_size_max", input_type=Int(optional=True), prefix="--smith-waterman-gap-size-max", doc=InputDocumentation(doc="(=16)         Maximum length of gap\ndetectable by smith waterman\nalgorithm.")), ToolInput(tag="in_smith_waterman_gaps_max", input_type=Int(optional=True), prefix="--smith-waterman-gaps-max", doc=InputDocumentation(doc="(=4)              Maximum number of gaps that\ncan be introduced into an\nalignment by Smith-Waterman\nalgorithm. If the optimum\nalignment has more gaps, it\nis simply ignored as an\nalignment candidate.")), ToolInput(tag="in_split_alignments", input_type=Int(optional=True), prefix="--split-alignments", doc=InputDocumentation(doc="(=1)                     When set, alignments crossing\na structural variant are\nallowed to be split with SA\ntag.")), ToolInput(tag="in_split_gap_length", input_type=Int(optional=True), prefix="--split-gap-length", doc=InputDocumentation(doc="(=10000)                 Maximum length of insertion\nor deletion allowed to exist\nin a read. If a gap exceeds\nthis limit, the read gets\nbroken up around the gap with\nSA tag introduced")), ToolInput(tag="in_start_from", input_type=String(optional=True), prefix="--start-from", doc=InputDocumentation(doc="(=Start)                       Start processing at the\nspecified stage:\n- Start            : don't\nresume, start from beginning\n- Align            : same\nas Start\n- AlignmentReports :\nregenerate alignment reports\nand bam\n- Bam              : resume\nat bam generation\n- Finish           : Same\nas Bam.\n- Last             : resume\nfrom the last successful step\nNote that although Isaac\nattempts to perform some\nbasic validation, the only\nsafe option is 'Start' The\nprimary purpose of the\nfeature is to reduce the time\nrequired to diagnose the\nissues rather than be used on\na regular basis.")), ToolInput(tag="in_stats_image_format", input_type=String(optional=True), prefix="--stats-image-format", doc=InputDocumentation(doc="(=none)                Format to use for images\nduring stats generation\n- gif        : produce .gif\ntype plots\n- none       : no stat\ngeneration")), ToolInput(tag="in_stop_at", input_type=String(optional=True), prefix="--stop-at", doc=InputDocumentation(doc="(=Finish)                         Stop processing after the\nspecified stage is complete:\n- Start            :\nperform the first stage only\n- Align            : same\nas Start\n- AlignmentReports : don't\nperform bam generation\n- Bam              : finish\nwhen bam is done\n- Finish           : stop\nat the end.\n- Last             :\nperform up to the last\nsuccessful step only\nNote that although Isaac\nattempts to perform some\nbasic validation, the only\nsafe option is 'Finish' The\nprimary purpose of the\nfeature is to reduce the time\nrequired to diagnose the\nissues rather than be used on\na regular basis.")), ToolInput(tag="in_target_bin_size", input_type=Int(optional=True), prefix="--target-bin-size", doc=InputDocumentation(doc="(=0)                      Isaac will attempt to bin\ntemporary data so that each\nbin is close to targetBinSize\nin megabytes (1024 * 1024\nbytes). Value of 0 will cause\nIsaac to compute the target\nbin size automatically based\non the available memory.")), ToolInput(tag="in_temp_concurrent_load", input_type=Int(optional=True), prefix="--temp-concurrent-load", doc=InputDocumentation(doc="(=4)                 Maximum number of concurrent\nfile read operations for\n--temp-directory")), ToolInput(tag="in_temp_concurrent_save", input_type=Int(optional=True), prefix="--temp-concurrent-save", doc=InputDocumentation(doc="(=680)               Maximum number of concurrent\nfile write operations for\n--temp-directory")), ToolInput(tag="in_arg_directory_be", input_type=Boolean(optional=True), prefix="-t", doc=InputDocumentation(doc="[ --temp-directory ] arg (=./Temp)           Directory where the temporary\nfiles will be stored\n(matches, unsorted\nalignments, etc.)")), ToolInput(tag="in_tiles", input_type=Int(optional=True), prefix="--tiles", doc=InputDocumentation(doc="Comma-separated list of\nregular expressions to select\nonly a subset of the tiles\navailable in the flow-cell.\n- to select all the tiles\nending with '5' in all lanes:\n--tiles [0-9][0-9][0-9]5\n- to select tile 2 in lane 1\nand all the tiles in the\nother lanes: --tiles\ns_1_0002,s_[2-8]\nMultiple entries allowed,\neach applies to the\ncorresponding base-calls.")), ToolInput(tag="in_tls", input_type=Int(optional=True), prefix="--tls", doc=InputDocumentation(doc="Template-length statistics in\nthe format 'min:median:max:lo\nwStdDev:highStdDev:M0:M1',\nwhere M0 and M1 are the\nnumeric value of the models\n(0=FFp, 1=FRp, 2=RFp, 3=RRp,\n4=FFm, 5=FRm, 6=RFm, 7=RRm)")), ToolInput(tag="in_trim_pe", input_type=Int(optional=True), prefix="--trim-pe", doc=InputDocumentation(doc="(=1)                              Trim overhanging ends of PE")), ToolInput(tag="in_use_smith_waterman", input_type=String(optional=True), prefix="--use-smith-waterman", doc=InputDocumentation(doc="(=smart)               One of the following:\n- always           : Use\nsmith-waterman to reduce the\namount of mismatches in\naligned reads\n- smart            : apply\nheuristics to avoid executing\ncostly smith-waterman on\nsequences that are unlikely\nto produce gaps\n- never            : Don't\nuse smith-waterman")), ToolInput(tag="in_variable_read_length", input_type=Int(optional=True), prefix="--variable-read-length", doc=InputDocumentation(doc="Unless set, Isaac will fail\nif the length of the sequence\nchanges between the records\nof a fastq or a bam file.")), ToolInput(tag="in_verbosity", input_type=Int(optional=True), prefix="--verbosity", doc=InputDocumentation(doc="(=2)                            Verbosity: FATAL(0),\nERRORS(1), WARNINGS(2),\nINFO(3), DEBUG(4) (not\nsupported yet)")), ToolInput(tag="in_print_program_version", input_type=Boolean(optional=True), prefix="-v", doc=InputDocumentation(doc="[ --version ]                                print program version")), ToolInput(tag="in_arguments", input_type=String(), position=0, doc=InputDocumentation(doc="-s [ --sample-sheet ] arg                       Multiple entries allowed. ")), ToolInput(tag="in_alignments", input_type=String(), position=0, doc=InputDocumentation(doc="--use-bases-mask arg                            Conversion mask characters:\n- Y or y          : use\n- N or n          : discard\n- I or i          : use for")), ToolInput(tag="in_indexing", input_type=String(), position=1, doc=InputDocumentation(doc="If not given, the mask will ")), ToolInput(tag="in_information", input_type=String(), position=0, doc=InputDocumentation(doc="Failed to parse the options: /opt/conda/conda-bld/isaac4_1595676738617/work/src/c++/lib/options/AlignOptions.cpp(614): Throw in function void isaac::options::AlignOptions::verifyMandatoryPaths(boost::program_options::variables_map&)"))], outputs=[ToolOutput(tag="out_bam_produce_md_five", output_type=File(optional=True), selector=InputSelector(input_to_select="in_bam_produce_md_five", type_hint=File()), doc=OutputDocumentation(doc="(=1)                      Controls whether a separate\nfile containing md5 checksum\nis produced for each output\nbam.")), ToolOutput(tag="out_arg_directory_stored", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_arg_directory_stored", type_hint=File()), doc=OutputDocumentation(doc="[ --output-directory ] arg (=./Aligned)      Directory where the final\nalignment data be stored"))], container="quay.io/biocontainers/isaac4:04.18.11.09--h07bff40_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Isaac_Align_V0_1_0().translate("wdl")

