from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Int, Float, Boolean, String

Anvi_Script_Filter_Fasta_By_Blast_V0_1_0 = CommandToolBuilder(tool="anvi_script_filter_fasta_by_blast", base_command=["anvi-script-filter-fasta-by-blast"], inputs=[ToolInput(tag="in_fast_a_file", input_type=File(optional=True), prefix="--fasta-file", doc=InputDocumentation(doc="A FASTA-formatted input file (default: None)")), ToolInput(tag="in_output_file", input_type=File(optional=True), prefix="--output-file", doc=InputDocumentation(doc="File path to store results. (default: None)")), ToolInput(tag="in_delimited_file_", input_type=File(optional=True), prefix="-b", doc=InputDocumentation(doc="DELIMITED FILE, --blast-output TAB DELIMITED FILE\nBLAST table generated with blastp. `--outfmt 6` as the\noutput format is assumed. (default: None)")), ToolInput(tag="in_out_fmt", input_type=Int(optional=True), prefix="--outfmt", doc=InputDocumentation(doc="Specify the column ordering of your BLAST report. We\nadd the following paramter to our BLAST searches so\nthe output report contains the `qlen` field, which is\nnot included by default: `-outfmt '6 qseqid sseqid\npident length mismatch gapopen qstart qend sstart send\nevalue bitscore qlen slen'`. You may have used a\ndifferent `-outfmt` paramter, and you should use this\nparameter to explicitly define the column names in\nyour output file. For instance, if you had used the\nparameter mentioned above, then the correct version of\nthis parameter would be: 'qseqid sseqid pident length\nmismatch gapopen qstart qend sstart send evalue\nbitscore qlen slen'. Regardless of the BLAST output\nformat, your columns MUST contain the following\nparameters for this program to work properly:\n'qseqid', 'bitscore', 'length', 'qlen', and 'pident'.\n(default: None)")), ToolInput(tag="in_threshold", input_type=Float(optional=True), prefix="--threshold", doc=InputDocumentation(doc="What `proper_pident` threshold do you want to use for\nfiltering out sequences whose top bit-score matches\nhave `proper_pident`s less than this threshold? We\nhave defined `proper_pident` to be the percentage of\nthe query amino acids that both aligned to and were\nidentical to the corresponding matched amino acid.\nNote that the `pident` parameter output by BLAST does\nnot include regions of the query sequence unaligned to\nthe matched sequence, whereas `proper_pident` does.\nFor example, a sequence that's only half aligned by a\nmatch but with 100% identity at matched regions has a\n`pident` of 100 but a `proper_pident` of 50. The\ndefault is 30.0%.")), ToolInput(tag="in_just_do_it", input_type=Boolean(optional=True), prefix="--just-do-it", doc=InputDocumentation(doc="Don't bother me with questions or warnings, just do\nit. (default: False)")), ToolInput(tag="in_contigs_fast_a", input_type=String(), position=0, doc=InputDocumentation(doc="üç∫ More on `anvi-script-filter-fasta-by-blast`:"))], outputs=[ToolOutput(tag="out_output_file", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output_file", type_hint=File()), doc=OutputDocumentation(doc="File path to store results. (default: None)")), ToolOutput(tag="out_delimited_file_", output_type=File(optional=True), selector=InputSelector(input_to_select="in_delimited_file_", type_hint=File()), doc=OutputDocumentation(doc="DELIMITED FILE, --blast-output TAB DELIMITED FILE\nBLAST table generated with blastp. `--outfmt 6` as the\noutput format is assumed. (default: None)"))], container="quay.io/biocontainers/anvio-minimal:7--py_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Anvi_Script_Filter_Fasta_By_Blast_V0_1_0().translate("wdl")

