from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Boolean, String, Int, Float

Pbwt_V0_1_0 = CommandToolBuilder(tool="pbwt", base_command=["pbwt"], inputs=[ToolInput(tag="in_log", input_type=File(optional=True), prefix="-log", doc=InputDocumentation(doc="log file; '-' for stderr")), ToolInput(tag="in_check", input_type=Boolean(optional=True), prefix="-check", doc=InputDocumentation(doc="do various checks")), ToolInput(tag="in_stats", input_type=Boolean(optional=True), prefix="-stats", doc=InputDocumentation(doc="print stats depending on commands; writes to stdout")), ToolInput(tag="in_read", input_type=File(optional=True), prefix="-read", doc=InputDocumentation(doc="read pbwt file; '-' for stdin")), ToolInput(tag="in_read_sites", input_type=File(optional=True), prefix="-readSites", doc=InputDocumentation(doc="read sites file; '-' for stdin")), ToolInput(tag="in_read_samples", input_type=File(optional=True), prefix="-readSamples", doc=InputDocumentation(doc="read samples file; '-' for stdin")), ToolInput(tag="in_read_missing", input_type=File(optional=True), prefix="-readMissing", doc=InputDocumentation(doc="read missing file; '-' for stdin")), ToolInput(tag="in_read_dosage", input_type=File(optional=True), prefix="-readDosage", doc=InputDocumentation(doc="read dosage file; '-' for stdin")), ToolInput(tag="in_read_reverse", input_type=File(optional=True), prefix="-readReverse", doc=InputDocumentation(doc="read reverse file; '-' for stdin")), ToolInput(tag="in_read_all", input_type=String(optional=True), prefix="-readAll", doc=InputDocumentation(doc="read .pbwt and if present .sites, .samples, .missing - note not by default dosage")), ToolInput(tag="in_read_vcf_gt", input_type=File(optional=True), prefix="-readVcfGT", doc=InputDocumentation(doc="read GTs from vcf or bcf file; '-' for stdin vcf only ; biallelic sites only - require diploid!")), ToolInput(tag="in_read_vc_fpl", input_type=File(optional=True), prefix="-readVcfPL", doc=InputDocumentation(doc="read PLs from vcf or bcf file; '-' for stdin vcf only ; biallelic sites only - require diploid!")), ToolInput(tag="in_read_macs", input_type=File(optional=True), prefix="-readMacs", doc=InputDocumentation(doc="read MaCS output file; '-' for stdin")), ToolInput(tag="in_read_vcf_q", input_type=File(optional=True), prefix="-readVcfq", doc=InputDocumentation(doc="read VCFQ file; '-' for stdin")), ToolInput(tag="in_read_gen", input_type=File(optional=True), prefix="-readGen", doc=InputDocumentation(doc="<chrom>   read impute2 gen file - must set chrom")), ToolInput(tag="in_read_hap", input_type=File(optional=True), prefix="-readHap", doc=InputDocumentation(doc="<chrom>   read impute2 hap file - must set chrom")), ToolInput(tag="in_read_hap_legend", input_type=File(optional=True), prefix="-readHapLegend", doc=InputDocumentation(doc="<legend_file> <chrom>\nread impute2 hap and legend file - must set chrom")), ToolInput(tag="in_read_phase", input_type=File(optional=True), prefix="-readPhase", doc=InputDocumentation(doc="read Li and Stephens phase file")), ToolInput(tag="in_checkpoint", input_type=String(optional=True), prefix="-checkpoint", doc=InputDocumentation(doc="checkpoint every n sites while reading")), ToolInput(tag="in_merge", input_type=File(optional=True), prefix="-merge", doc=InputDocumentation(doc="...         merge two or more pbwt files")), ToolInput(tag="in_write", input_type=File(optional=True), prefix="-write", doc=InputDocumentation(doc="write pbwt file; '-' for stdout")), ToolInput(tag="in_write_sites", input_type=File(optional=True), prefix="-writeSites", doc=InputDocumentation(doc="write sites file; '-' for stdout")), ToolInput(tag="in_write_samples", input_type=File(optional=True), prefix="-writeSamples", doc=InputDocumentation(doc="write samples file; '-' for stdout")), ToolInput(tag="in_write_missing", input_type=File(optional=True), prefix="-writeMissing", doc=InputDocumentation(doc="write missing file; '-' for stdout")), ToolInput(tag="in_write_dosage", input_type=File(optional=True), prefix="-writeDosage", doc=InputDocumentation(doc="write missing file; '-' for stdout")), ToolInput(tag="in_write_reverse", input_type=File(optional=True), prefix="-writeReverse", doc=InputDocumentation(doc="write reverse file; '-' for stdout")), ToolInput(tag="in_write_all", input_type=String(optional=True), prefix="-writeAll", doc=InputDocumentation(doc="write .pbwt and if present .sites, .samples, .missing, .dosage")), ToolInput(tag="in_write_impute_ref", input_type=String(optional=True), prefix="-writeImputeRef", doc=InputDocumentation(doc="write .imputeHaps and .imputeLegend")), ToolInput(tag="in_write_impute_haps_g", input_type=File(optional=True), prefix="-writeImputeHapsG", doc=InputDocumentation(doc="write haplotype file for IMPUTE -known_haps_g")), ToolInput(tag="in_write_phase", input_type=File(optional=True), prefix="-writePhase", doc=InputDocumentation(doc="write FineSTRUCTURE/ChromoPainter input format (Impute/ShapeIT output format) phase file")), ToolInput(tag="in_write_transpose_haplotypes", input_type=File(optional=True), prefix="-writeTransposeHaplotypes", doc=InputDocumentation(doc="write transposed haplotype file (one hap per row); '-' for stdout")), ToolInput(tag="in_haps", input_type=File(optional=True), prefix="-haps", doc=InputDocumentation(doc="write haplotype file; '-' for stdout")), ToolInput(tag="in_write_gen", input_type=File(optional=True), prefix="-writeGen", doc=InputDocumentation(doc="write impute2 gen file; '-' for stdout")), ToolInput(tag="in_write_vcf_gz", input_type=File(optional=True), prefix="-writeVcfGz", doc=InputDocumentation(doc="write VCF or BCF; uncompressed or bgzip (Gz) compressed file; '-' for stdout")), ToolInput(tag="in_reference_fast_a", input_type=File(optional=True), prefix="-referenceFasta", doc=InputDocumentation(doc="reference fasta filename for VCF/BCF writing (optional)")), ToolInput(tag="in_subsites", input_type=String(optional=True), prefix="-subsites", doc=InputDocumentation(doc="<frac>   subsample <frac> sites with AF > <fmin>")), ToolInput(tag="in_subsample", input_type=String(optional=True), prefix="-subsample", doc=InputDocumentation(doc="<n>    subsample <n> samples from index <start>")), ToolInput(tag="in_subrange", input_type=String(optional=True), prefix="-subrange", doc=InputDocumentation(doc="<end>   cut down to sites in [start,end)")), ToolInput(tag="in_corrupt_sites", input_type=String(optional=True), prefix="-corruptSites", doc=InputDocumentation(doc="<q>     randomise fraction q of positions at fraction p of sites, according to site frequency")), ToolInput(tag="in_corrupt_samples", input_type=String(optional=True), prefix="-corruptSamples", doc=InputDocumentation(doc="<q>   randomise fraction q of positions for fraction p of samples, according to site frequency")), ToolInput(tag="in_copy_samples", input_type=Int(optional=True), prefix="-copySamples", doc=InputDocumentation(doc="<len>    make M new samples copied from current haplotypes with mean switch length len")), ToolInput(tag="in_select_sites", input_type=File(optional=True), prefix="-selectSites", doc=InputDocumentation(doc="select sites as in sites file")), ToolInput(tag="in_remove_sites", input_type=File(optional=True), prefix="-removeSites", doc=InputDocumentation(doc="remove sites as in sites file")), ToolInput(tag="in_select_samples", input_type=File(optional=True), prefix="-selectSamples", doc=InputDocumentation(doc="select samples as in samples file")), ToolInput(tag="in_long_within", input_type=String(optional=True), prefix="-longWithin", doc=InputDocumentation(doc="find matches within set longer than L")), ToolInput(tag="in_max_within", input_type=Boolean(optional=True), prefix="-maxWithin", doc=InputDocumentation(doc="find maximal matches within set")), ToolInput(tag="in_match_naive", input_type=File(optional=True), prefix="-matchNaive", doc=InputDocumentation(doc="maximal match seqs in pbwt file to reference")), ToolInput(tag="in_match_indexed", input_type=File(optional=True), prefix="-matchIndexed", doc=InputDocumentation(doc="maximal match seqs in pbwt file to reference")), ToolInput(tag="in_match_dynamic", input_type=File(optional=True), prefix="-matchDynamic", doc=InputDocumentation(doc="maximal match seqs in pbwt file to reference")), ToolInput(tag="in_impute_explore", input_type=String(optional=True), prefix="-imputeExplore", doc=InputDocumentation(doc="n'th impute test")), ToolInput(tag="in_phase", input_type=String(optional=True), prefix="-phase", doc=InputDocumentation(doc="phase with n sparse pbwts")), ToolInput(tag="in_reference_phase", input_type=String(optional=True), prefix="-referencePhase", doc=InputDocumentation(doc="phase current pbwt against reference whose root name is the argument - only keeps shared sites")), ToolInput(tag="in_reference_impute", input_type=Int(optional=True), prefix="-referenceImpute", doc=InputDocumentation(doc="[nSparse=1] [fSparse=1]  impute current pbwt into reference whose root name is the first argument;\ndoes not rephase either pbwt; optional nSparse > 1 also does sparse matching, fSparse is relative weight")), ToolInput(tag="in_genotype_compare", input_type=String(optional=True), prefix="-genotypeCompare", doc=InputDocumentation(doc="compare genotypes with those from reference whose root name is the argument - need compatible sites")), ToolInput(tag="in_impute_missing", input_type=Boolean(optional=True), prefix="-imputeMissing", doc=InputDocumentation(doc="impute data marked as missing")), ToolInput(tag="in_fit_alphabet_a", input_type=Float(optional=True), prefix="-fitAlphaBeta", doc=InputDocumentation(doc="fit probabilistic model 1..3")), ToolInput(tag="in_ll_copy_model", input_type=String(optional=True), prefix="-llCopyModel", doc=InputDocumentation(doc="<rho>  log likelihood of Li-Stephens model")), ToolInput(tag="in_paint", input_type=File(optional=True), prefix="-paint", doc=InputDocumentation(doc="[n] output painting co-ancestry matrix to fileroot, optionally specififying the number per region")), ToolInput(tag="in_paint_sparse", input_type=File(optional=True), prefix="-paintSparse", doc=InputDocumentation(doc="[n] output sparse painting to fileroot, optionally specififying the number per region")), ToolInput(tag="in_pretty", input_type=File(optional=True), prefix="-pretty", doc=InputDocumentation(doc="<k>        pretty plot at site k")), ToolInput(tag="in_sfs", input_type=Boolean(optional=True), prefix="-sfs", doc=InputDocumentation(doc="print site frequency spectrum (log scale) - also writes sites.freq file")), ToolInput(tag="in_ref_freq", input_type=File(optional=True), prefix="-refFreq", doc=InputDocumentation(doc="read site frequency information into the refFreq field of current sites")), ToolInput(tag="in_site_info", input_type=File(optional=True), prefix="-siteInfo", doc=InputDocumentation(doc="<kmin> <kmax> export PBWT information at sites with allele count kmin <= k < kmax")), ToolInput(tag="in_build_reverse", input_type=Boolean(optional=True), prefix="-buildReverse", doc=InputDocumentation(doc="build reverse pbwt")), ToolInput(tag="in_read_genetic_map", input_type=File(optional=True), prefix="-readGeneticMap", doc=InputDocumentation(doc="read Oxford format genetic map file")), ToolInput(tag="in_four_haps_stats", input_type=Boolean(optional=True), prefix="-4hapsStats", doc=InputDocumentation(doc="mu:rho 4 hap test stats"))], outputs=[ToolOutput(tag="out_read_macs", output_type=File(optional=True), selector=InputSelector(input_to_select="in_read_macs", type_hint=File()), doc=OutputDocumentation(doc="read MaCS output file; '-' for stdin"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Pbwt_V0_1_0().translate("wdl", allow_empty_container=True)

