from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, String, Boolean, Float, Int

Deepbgc_Pipeline_V0_1_0 = CommandToolBuilder(tool="deepbgc_pipeline", base_command=["deepbgc", "pipeline"], inputs=[ToolInput(tag="in_output", input_type=File(optional=True), prefix="--output", doc=InputDocumentation(doc="Custom output directory path")), ToolInput(tag="in_limit_to_record", input_type=String(optional=True), prefix="--limit-to-record", doc=InputDocumentation(doc="Process only specific record ID. Can be provided multiple times")), ToolInput(tag="in_minimal_output", input_type=File(optional=True), prefix="--minimal-output", doc=InputDocumentation(doc="Produce minimal output with just the GenBank sequence file")), ToolInput(tag="in_prodigal_meta_mode", input_type=Boolean(optional=True), prefix="--prodigal-meta-mode", doc=InputDocumentation(doc="Run Prodigal in '-p meta' mode to enable detecting genes in short contigs")), ToolInput(tag="in_protein", input_type=Boolean(optional=True), prefix="--protein", doc=InputDocumentation(doc="Accept amino-acid protein sequences as input (experimental). Will treat each file as a single record with multiple proteins.")), ToolInput(tag="in_detector", input_type=File(optional=True), prefix="--detector", doc=InputDocumentation(doc="Trained detection model name (run 'deepbgc download' to download models) or path to trained model pickle file. Can be provided multiple times (-d first -d second)")), ToolInput(tag="in_no_detector", input_type=Boolean(optional=True), prefix="--no-detector", doc=InputDocumentation(doc="Disable BGC detection")), ToolInput(tag="in_label", input_type=String(optional=True), prefix="--label", doc=InputDocumentation(doc="Label for detected clusters (equal to --detector by default). If multiple detectors are provided, a label should be provided for each one")), ToolInput(tag="in_score", input_type=Float(optional=True), prefix="--score", doc=InputDocumentation(doc="Average protein-wise DeepBGC score threshold for extracting BGC regions from Pfam sequences (default: 0.5)")), ToolInput(tag="in_merge_max_protein_gap", input_type=Int(optional=True), prefix="--merge-max-protein-gap", doc=InputDocumentation(doc="Merge detected BGCs within given number of proteins (default: 0)")), ToolInput(tag="in_merge_max_nucl_gap", input_type=Int(optional=True), prefix="--merge-max-nucl-gap", doc=InputDocumentation(doc="Merge detected BGCs within given number of nucleotides (default: 0)")), ToolInput(tag="in_min_nucl", input_type=Int(optional=True), prefix="--min-nucl", doc=InputDocumentation(doc="Minimum BGC nucleotide length (default: 1)")), ToolInput(tag="in_min_proteins", input_type=Int(optional=True), prefix="--min-proteins", doc=InputDocumentation(doc="Minimum number of proteins in a BGC (default: 1)")), ToolInput(tag="in_min_domains", input_type=Int(optional=True), prefix="--min-domains", doc=InputDocumentation(doc="Minimum number of protein domains in a BGC (default: 1)")), ToolInput(tag="in_min_bio_domains", input_type=Int(optional=True), prefix="--min-bio-domains", doc=InputDocumentation(doc="Minimum number of known biosynthetic (as defined by antiSMASH) protein domains in a BGC (default: 0)")), ToolInput(tag="in_classifier", input_type=File(optional=True), prefix="--classifier", doc=InputDocumentation(doc="Trained classification model name (run 'deepbgc download' to download models) or path to trained model pickle file. Can be provided multiple times (-c first -c second)")), ToolInput(tag="in_no_classifier", input_type=Boolean(optional=True), prefix="--no-classifier", doc=InputDocumentation(doc="Disable BGC classification")), ToolInput(tag="in_classifier_score", input_type=Float(optional=True), prefix="--classifier-score", doc=InputDocumentation(doc="DeepBGC classification score threshold for assigning classes to BGCs (default: 0.5)\n")), ToolInput(tag="in_inputs", input_type=String(), position=0, doc=InputDocumentation(doc="Input sequence file path (FASTA, GenBank, Pfam CSV)"))], outputs=[ToolOutput(tag="out_output", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output", type_hint=File()), doc=OutputDocumentation(doc="Custom output directory path")), ToolOutput(tag="out_minimal_output", output_type=File(optional=True), selector=InputSelector(input_to_select="in_minimal_output", type_hint=File()), doc=OutputDocumentation(doc="Produce minimal output with just the GenBank sequence file"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Deepbgc_Pipeline_V0_1_0().translate("wdl", allow_empty_container=True)

