from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Boolean, Int, String

Ngseqbasic_V0_1_0 = CommandToolBuilder(tool="NGseqBasic", base_command=["NGseqBasic"], inputs=[ToolInput(tag="in_outfile", input_type=File(optional=True), prefix="--outfile", doc=InputDocumentation(doc="(the STDOUT log file name in your RUN COMMAND - see above )")), ToolInput(tag="in_errfile", input_type=File(optional=True), prefix="--errfile", doc=InputDocumentation(doc="(the STDERR log file name in your RUN COMMAND - see above )")), ToolInput(tag="in_orange_blue", input_type=Boolean(optional=True), prefix="--orangeBlue", doc=InputDocumentation(doc="(use orange-blue color scheme instead of red-green)")), ToolInput(tag="in_gz", input_type=Boolean(optional=True), prefix="--gz", doc=InputDocumentation(doc="(input files are provided in file.fastq.gz compressed format )")), ToolInput(tag="in_lanes", input_type=Int(optional=True), prefix="--lanes", doc=InputDocumentation(doc="(set this to be the number of lanes, if there are more than 1 lanes in your fastq files)")), ToolInput(tag="in_single_end", input_type=Boolean(optional=True), prefix="--singleEnd", doc=InputDocumentation(doc="- to run single end sequencing files (default behavior is paired end files)")), ToolInput(tag="in_how_many_processors", input_type=Int(optional=True), prefix="-p", doc=InputDocumentation(doc=": to how many processors we will parallelise the bowtie part of the run")), ToolInput(tag="in_bowtie_one", input_type=Boolean(optional=True), prefix="--bowtie1", doc=InputDocumentation(doc="(default is bowtie1 - decide if bowtie1 or bowtie2 is to be used. bowtie2 is better to long reads - read lenght more than 70b, fragment lenght more than 350b)")), ToolInput(tag="in_chunk_mb", input_type=Int(optional=True), prefix="--chunkmb", doc=InputDocumentation(doc="- memory allocated to Bowtie, defaults to 256mb - only affects bowtie1 run")), ToolInput(tag="in_run_maps_more", input_type=Int(optional=True), prefix="-M", doc=InputDocumentation(doc="run with bowtie parameter M=2 (if maps more than M times, report one alignment in random) - only affects bowtie1 run")), ToolInput(tag="in_max_ins", input_type=Int(optional=True), prefix="--maxins", doc=InputDocumentation(doc=": sets the TRUE fragment (max) lenght to 350bp. Bowtie1 default 250, bowtie2 default 500, this script default 350. (Bowtie mappings resulting in fragments longer than --maxins are not reported)")), ToolInput(tag="in_no_bowtie", input_type=String(optional=True), prefix="--noBowtie", doc=InputDocumentation(doc="only pipe AFTER bowtie (assumes PIPE_mappedBamPaths.txt, see above)")), ToolInput(tag="in_flash", input_type=Boolean(optional=True), prefix="--flash", doc=InputDocumentation(doc="**/noFlash (run/do-not-run Flash - for unmapped reads 'try to merge overlapping short reads to longer single end read' to enhance mapping)")), ToolInput(tag="in_blacklist_filter", input_type=File(optional=True), prefix="--blacklistFilter", doc=InputDocumentation(doc="/noBlacklistFilter - supported genomes mm9 mm10 hg18 hg19 (filter/do-not-filter blacklisted regions out from the final bam file)")), ToolInput(tag="in_footprint", input_type=Boolean(optional=True), prefix="--footPrint", doc=InputDocumentation(doc=": generate also footprint tracks (see footprint(FP) parameters below)")), ToolInput(tag="in_peak_call", input_type=Boolean(optional=True), prefix="--peakCall", doc=InputDocumentation(doc=": generate also peak call (see peak call(P) parameters below)")), ToolInput(tag="in_next_era", input_type=Boolean(optional=True), prefix="--nextera", doc=InputDocumentation(doc=": use Nextera adaptors in trimming, instead of standard illumina PE adapters")), ToolInput(tag="in_trim", input_type=Boolean(optional=True), prefix="--trim", doc=InputDocumentation(doc="/noTrim** (run/do-not-run TrimGalore for the data - Illumina PE standard adapter filter, trims on 3' end)")), ToolInput(tag="in_trim_five", input_type=Boolean(optional=True), prefix="--trim5", doc=InputDocumentation(doc="(run trimming also for 5' end of the data - Illumina PE standard adapter filter, combination of TrimGalore and cutadapt)")), ToolInput(tag="in_ada_three_read_one", input_type=Int(optional=True), prefix="--ada3read1", doc=InputDocumentation(doc=": custom adapters 3' trimming, R1 and R2 (give both if PE custom trimming is needed, SE trimming needs only R1) - these adapters will be used instead of Illumina default / atac adapters. SEQUENCE has to be in CAPITAL letters ATCG")), ToolInput(tag="in_ada_five_read_one", input_type=Int(optional=True), prefix="--ada5read1", doc=InputDocumentation(doc=": custom adapters 5' trimming, R1 and R2 (give both if PE custom trimming is needed, SE trimming needs only R1) - these adapters will be used instead of Illumina default / atac adapters. SEQUENCE has to be in CAPITAL letters ATCG")), ToolInput(tag="in_no_window", input_type=Boolean(optional=True), prefix="--noWindow", doc=InputDocumentation(doc=": no windowing - instead plot 1b resolution raw read depths")), ToolInput(tag="in_window_size", input_type=Int(optional=True), prefix="--windowSize", doc=InputDocumentation(doc=": custom window size (instead of default 300b) - this value has to be even number (or is rounded into one).\nIf no custom INCREMENT is given, increment is set to 10% of windowSize , i.e. the windowSize value has to be divisible by 10 (or will be rounded to behave as such).")), ToolInput(tag="in_window_incr", input_type=Int(optional=True), prefix="--windowIncr", doc=InputDocumentation(doc=": custom window increment (instead of default 10%). The value of windowIncr should be even number, given in BASES (10 bases : windowIncr=10), not percentages.\nIf no custom windowSize is given, the windowSize will be set to 300bases, and increment rounded so that 300bases is divisible by the given increment (allows only values 10 and 30).")), ToolInput(tag="in_save_unmapped", input_type=Boolean(optional=True), prefix="--saveUnmapped", doc=InputDocumentation(doc="(save fastq files UNMAPPED_1.fastq UNMAPPED_2.fastq for not-paired-mapped reads : only for PE data)")), ToolInput(tag="in_save_unpaired", input_type=Boolean(optional=True), prefix="--saveUnpaired", doc=InputDocumentation(doc="(save bam files singleEnd_bowtie_READ1.bam singleEnd_bowtie_READ2.bam for unpaired mapped reads - not blacklisted/duplicate-filtered)")), ToolInput(tag="in_save_unpaired_filtered", input_type=Boolean(optional=True), prefix="--saveUnpairedFiltered", doc=InputDocumentation(doc="(save bam files singleEnd_bowtie_filtered_READ1.bam singleEnd_bowtie_filtered_READ1.bam for unpaired mapped reads - blacklisted/duplicate-filtered)")), ToolInput(tag="in_save_unfiltered", input_type=File(optional=True), prefix="--saveUnfiltered", doc=InputDocumentation(doc="(save original bam file bowtie_out.bam - straight from bowtie output)")), ToolInput(tag="in_save_unfiltered_mapped", input_type=File(optional=True), prefix="--saveUnfilteredMapped", doc=InputDocumentation(doc="(save original bam file mapped.bam - straight from bowtie output. Do not print out unmapped reads (single end) / non-proper pairs (paired end).)")), ToolInput(tag="in_save_untrimmed", input_type=File(optional=True), prefix="--saveUntrimmed", doc=InputDocumentation(doc="(run EXTRA bowtie-run before trimming, save the bam file UNTRIMMED_bowtie_out.bam - straight from before-trimming bowtie output - includes all reads)")), ToolInput(tag="in_save_untrimmed_mapped", input_type=File(optional=True), prefix="--saveUntrimmedMapped", doc=InputDocumentation(doc="(run EXTRA bowtie-run before trimming, save the bam file UNTRIMMED_onlyMapped_bowtie_out.bam - straight from before-trimming bowtie output. Do not print out unmapped reads (single end) / non-proper pairs (paired end).)")), ToolInput(tag="in_save_bdg", input_type=Boolean(optional=True), prefix="--saveBDG", doc=InputDocumentation(doc="(save unpacked BDG files - default : save only BIGWIG packed versions of these files")), ToolInput(tag="in_pyramid_rerun", input_type=Boolean(optional=True), prefix="--pyramidRerun", doc=InputDocumentation(doc=": if you are running peak calls / footprints via PYRAMID pipeline, set this on.")), ToolInput(tag="in_only_peak_call", input_type=Boolean(optional=True), prefix="--onlyPeakCall", doc=InputDocumentation(doc=": only run Peak Call (to finetune peak call part) - expects PIPE_previousRunPaths.txt (see above)")), ToolInput(tag="in_only_fp_and_pc", input_type=Boolean(optional=True), prefix="--onlyFPandPC", doc=InputDocumentation(doc=": only run Peak Call and FootPrinting (to finetune FootPrint windowing and peak call parameters) - expects PIPE_previousRunPaths.txt (see above)")), ToolInput(tag="in_only_hub", input_type=Boolean(optional=True), prefix="--onlyHub", doc=InputDocumentation(doc=": only generates data hub - NOTE !! this DOES NOT include data from any 'footprint / peak call rerun' folders - it hubs only the ORIGINAL data set - (expects PIPE_previousRunPaths.txt, see above)")), ToolInput(tag="in_window_fp", input_type=Int(optional=True), prefix="--windowFP", doc=InputDocumentation(doc=": (windowing the signal for visualisation) - sliding window, where overlap 2*WINDOW, and window size 4*WINDOW, resulting in 2*WINDOW size increments in graph. Value 0 means NO WINDOWING (only raw 1b resolution track is given)")), ToolInput(tag="in_depth_p", input_type=Int(optional=True), prefix="--depthP", doc=InputDocumentation(doc=": (first filter)  RANGE of depthP - Each region should have at least 40 reads to be included to the peak call.")), ToolInput(tag="in_merge_p", input_type=Int(optional=True), prefix="--mergeP", doc=InputDocumentation(doc=": (second filter) Merges regions separated by 0bp or less.")), ToolInput(tag="in_contig_p", input_type=Int(optional=True), prefix="--contigP", doc=InputDocumentation(doc=": (third filter)  Regions (merged, depth-filtered) should be at least 20bp wide to be considered."))], outputs=[ToolOutput(tag="out_blacklist_filter", output_type=File(optional=True), selector=InputSelector(input_to_select="in_blacklist_filter", type_hint=File()), doc=OutputDocumentation(doc="/noBlacklistFilter - supported genomes mm9 mm10 hg18 hg19 (filter/do-not-filter blacklisted regions out from the final bam file)")), ToolOutput(tag="out_save_unfiltered", output_type=File(optional=True), selector=InputSelector(input_to_select="in_save_unfiltered", type_hint=File()), doc=OutputDocumentation(doc="(save original bam file bowtie_out.bam - straight from bowtie output)")), ToolOutput(tag="out_save_unfiltered_mapped", output_type=File(optional=True), selector=InputSelector(input_to_select="in_save_unfiltered_mapped", type_hint=File()), doc=OutputDocumentation(doc="(save original bam file mapped.bam - straight from bowtie output. Do not print out unmapped reads (single end) / non-proper pairs (paired end).)")), ToolOutput(tag="out_save_untrimmed", output_type=File(optional=True), selector=InputSelector(input_to_select="in_save_untrimmed", type_hint=File()), doc=OutputDocumentation(doc="(run EXTRA bowtie-run before trimming, save the bam file UNTRIMMED_bowtie_out.bam - straight from before-trimming bowtie output - includes all reads)")), ToolOutput(tag="out_save_untrimmed_mapped", output_type=File(optional=True), selector=InputSelector(input_to_select="in_save_untrimmed_mapped", type_hint=File()), doc=OutputDocumentation(doc="(run EXTRA bowtie-run before trimming, save the bam file UNTRIMMED_onlyMapped_bowtie_out.bam - straight from before-trimming bowtie output. Do not print out unmapped reads (single end) / non-proper pairs (paired end).)"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Ngseqbasic_V0_1_0().translate("wdl", allow_empty_container=True)

