from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import String, File, Int, Boolean

Bcbio_Variation_V0_1_0 = CommandToolBuilder(tool="bcbio_variation", base_command=["bcbio-variation"], inputs=[ToolInput(tag="in_analysis_type", input_type=String(optional=True), prefix="--analysis_type", doc=InputDocumentation(doc="Name of the tool to run")), ToolInput(tag="in_arg_file", input_type=File(optional=True), prefix="--arg_file", doc=InputDocumentation(doc="Reads arguments from the\nspecified file")), ToolInput(tag="in_input_file", input_type=File(optional=True), prefix="--input_file", doc=InputDocumentation(doc="Input file containing sequence\ndata (SAM or BAM)")), ToolInput(tag="in_read_buffer_size", input_type=Int(optional=True), prefix="--read_buffer_size", doc=InputDocumentation(doc="Number of reads per SAM file to\nbuffer in memory")), ToolInput(tag="in_phone_home", input_type=String(optional=True), prefix="--phone_home", doc=InputDocumentation(doc="Run reporting mode (NO_ET|AWS|\nSTDOUT)")), ToolInput(tag="in_g_atk_key", input_type=File(optional=True), prefix="--gatk_key", doc=InputDocumentation(doc="GATK key file required to run\nwith -et NO_ET")), ToolInput(tag="in_tag", input_type=String(optional=True), prefix="--tag", doc=InputDocumentation(doc="Tag to identify this GATK run\nas part of a group of runs")), ToolInput(tag="in_read_filter", input_type=String(optional=True), prefix="--read_filter", doc=InputDocumentation(doc="Filters to apply to reads\nbefore analysis")), ToolInput(tag="in_intervals", input_type=String(optional=True), prefix="--intervals", doc=InputDocumentation(doc="One or more genomic intervals\nover which to operate")), ToolInput(tag="in_exclude_intervals", input_type=String(optional=True), prefix="--excludeIntervals", doc=InputDocumentation(doc="One or more genomic intervals\nto exclude from processing")), ToolInput(tag="in_interval_set_rule", input_type=String(optional=True), prefix="--interval_set_rule", doc=InputDocumentation(doc="Set merging approach to use for\ncombining interval inputs\n(UNION|INTERSECTION)")), ToolInput(tag="in_interval_merging", input_type=String(optional=True), prefix="--interval_merging", doc=InputDocumentation(doc="Interval merging rule for\nabutting intervals (ALL|\nOVERLAPPING_ONLY)")), ToolInput(tag="in_interval_padding", input_type=String(optional=True), prefix="--interval_padding", doc=InputDocumentation(doc="Amount of padding (in bp) to\nadd to each interval")), ToolInput(tag="in_reference_sequence", input_type=File(optional=True), prefix="--reference_sequence", doc=InputDocumentation(doc="Reference sequence file")), ToolInput(tag="in_nondeterministic_random_seed", input_type=Boolean(optional=True), prefix="--nonDeterministicRandomSeed", doc=InputDocumentation(doc="Use a non-deterministic random")), ToolInput(tag="in_max_runtime_units", input_type=Int(optional=True), prefix="--maxRuntimeUnits", doc=InputDocumentation(doc="Unit of time used by maxRuntime\n(NANOSECONDS|MICROSECONDS|\nMILLISECONDS|SECONDS|MINUTES|\nHOURS|DAYS)")), ToolInput(tag="in_downsampling_type", input_type=String(optional=True), prefix="--downsampling_type", doc=InputDocumentation(doc="Type of read downsampling to\nemploy at a given locus (NONE|\nALL_READS|BY_SAMPLE)")), ToolInput(tag="in_down_sample_to_fraction", input_type=String(optional=True), prefix="--downsample_to_fraction", doc=InputDocumentation(doc="Fraction of reads to downsample")), ToolInput(tag="in_baq", input_type=String(optional=True), prefix="--baq", doc=InputDocumentation(doc="Type of BAQ calculation to\napply in the engine (OFF|\nCALCULATE_AS_NECESSARY|\nRECALCULATE)")), ToolInput(tag="in_baq_gap_open_penalty", input_type=String(optional=True), prefix="--baqGapOpenPenalty", doc=InputDocumentation(doc="BAQ gap open penalty")), ToolInput(tag="in_refactor_ndn_cigar_string", input_type=Boolean(optional=True), prefix="--refactor_NDN_cigar_string", doc=InputDocumentation(doc="refactor cigar string with NDN\nelements to one element")), ToolInput(tag="in_fix_m_is_encoded_quality_scores", input_type=Boolean(optional=True), prefix="--fix_misencoded_quality_scores", doc=InputDocumentation(doc="Fix mis-encoded base quality")), ToolInput(tag="in_use_original_qualities", input_type=Boolean(optional=True), prefix="--useOriginalQualities", doc=InputDocumentation(doc="Use the base quality scores\nfrom the OQ tag")), ToolInput(tag="in_default_base_qualities", input_type=String(optional=True), prefix="--defaultBaseQualities", doc=InputDocumentation(doc="Assign a default base quality")), ToolInput(tag="in_performance_log", input_type=File(optional=True), prefix="--performanceLog", doc=InputDocumentation(doc="Write GATK runtime performance\nlog to this file")), ToolInput(tag="in_b_qsr", input_type=File(optional=True), prefix="--BQSR", doc=InputDocumentation(doc="Input covariates table file for\non-the-fly base quality score\nrecalibration")), ToolInput(tag="in_disable_in_del_quals", input_type=Boolean(optional=True), prefix="--disable_indel_quals", doc=InputDocumentation(doc="Disable printing of base\ninsertion and deletion tags\n(with -BQSR)")), ToolInput(tag="in_emit_original_quals", input_type=Boolean(optional=True), prefix="--emit_original_quals", doc=InputDocumentation(doc="Emit the OQ tag with the\noriginal base qualities (with\n-BQSR)")), ToolInput(tag="in_preserve_q_scores_less_than", input_type=String(optional=True), prefix="--preserve_qscores_less_than", doc=InputDocumentation(doc="Don't recalibrate bases with\nquality scores less than this\nthreshold (with -BQSR)")), ToolInput(tag="in_global_q_score_prior", input_type=String(optional=True), prefix="--globalQScorePrior", doc=InputDocumentation(doc="Global Qscore Bayesian prior to\nuse for BQSR")), ToolInput(tag="in_validation_strictness", input_type=String(optional=True), prefix="--validation_strictness", doc=InputDocumentation(doc="How strict should we be with\nvalidation (STRICT|LENIENT|\nSILENT)")), ToolInput(tag="in_remove_program_records", input_type=Boolean(optional=True), prefix="--remove_program_records", doc=InputDocumentation(doc="Remove program records from the\nSAM header")), ToolInput(tag="in_keep_program_records", input_type=Boolean(optional=True), prefix="--keep_program_records", doc=InputDocumentation(doc="Keep program records in the SAM")), ToolInput(tag="in_unsafe", input_type=String(optional=True), prefix="--unsafe", doc=InputDocumentation(doc="Enable unsafe operations:\nnothing will be checked at\nruntime (ALLOW_N_CIGAR_READS|\nALLOW_UNINDEXED_BAM|\nALLOW_UNSET_BAM_SORT_ORDER|\nNO_READ_ORDER_VERIFICATION|\nALLOW_SEQ_DICT_INCOMPATIBILITY|\nLENIENT_VCF_PROCESSING|ALL)")), ToolInput(tag="in_num_threads", input_type=Int(optional=True), prefix="--num_threads", doc=InputDocumentation(doc="Number of data threads to\nallocate to this analysis")), ToolInput(tag="in_num_cpu_threads_per_data_thread", input_type=Int(optional=True), prefix="--num_cpu_threads_per_data_thread", doc=InputDocumentation(doc="Number of CPU threads to\nallocate per data thread")), ToolInput(tag="in_monitor_thread_efficiency", input_type=Boolean(optional=True), prefix="--monitorThreadEfficiency", doc=InputDocumentation(doc="Enable threading efficiency")), ToolInput(tag="in_pedigree_string", input_type=String(optional=True), prefix="--pedigreeString", doc=InputDocumentation(doc="Pedigree string for samples")), ToolInput(tag="in_pedigree_validation_type", input_type=String(optional=True), prefix="--pedigreeValidationType", doc=InputDocumentation(doc="Validation strictness for\npedigree information (STRICT|\nSILENT)")), ToolInput(tag="in_variant_index_type", input_type=String(optional=True), prefix="--variant_index_type", doc=InputDocumentation(doc="Type of IndexCreator to use for\nVCF/BCF indices (DYNAMIC_SEEK|\nDYNAMIC_SIZE|LINEAR|INTERVAL)")), ToolInput(tag="in_variant_index_parameter", input_type=String(optional=True), prefix="--variant_index_parameter", doc=InputDocumentation(doc="Parameter to pass to the\nVCF/BCF IndexCreator")), ToolInput(tag="in_logging_level", input_type=String(optional=True), prefix="--logging_level", doc=InputDocumentation(doc="Set the minimum level of")), ToolInput(tag="in_max_runtime", input_type=Int(optional=True), prefix="-maxRuntime", doc=InputDocumentation(doc="")), ToolInput(tag="in_jar", input_type=String(optional=True), prefix="-jar", doc=InputDocumentation(doc="")), ToolInput(tag="in_seed", input_type=String(), position=0, doc=InputDocumentation(doc="-maxRuntime,--maxRuntime <maxRuntime>                                                   Stop execution cleanly as soon ")), ToolInput(tag="in_to", input_type=String(), position=0, doc=InputDocumentation(doc="-dcov,--downsample_to_coverage <downsample_to_coverage>                                 Target coverage threshold for ")), ToolInput(tag="in_scores", input_type=String(), position=0, doc=InputDocumentation(doc="-allowPotentiallyMisencodedQuals,--allow_potentially_misencoded_quality_scores          Ignore warnings about base ")), ToolInput(tag="in_header", input_type=String(), position=0, doc=InputDocumentation(doc="-sample_rename_mapping_file,--sample_rename_mapping_file <sample_rename_mapping_file>   Rename sample IDs on-the-fly at ")), ToolInput(tag="in_monitoring", input_type=String(), position=0, doc=InputDocumentation(doc="-bfh,--num_bam_file_handles <num_bam_file_handles>                                      Total number of BAM file ")), ToolInput(tag="in_simultaneously", input_type=String(), position=0, doc=InputDocumentation(doc="-rgbl,--read_group_black_list <read_group_black_list>                                   Exclude read groups based on ")), ToolInput(tag="in_tags", input_type=String(), position=1, doc=InputDocumentation(doc="-ped,--pedigree <pedigree>                                                              Pedigree files for samples")), ToolInput(tag="in_logging", input_type=String(), position=2, doc=InputDocumentation(doc="-log,--log_to_file <log_to_file>                                                        Set the logging location")), ToolInput(tag="in_annotator", input_type=String(), position=3, doc=InputDocumentation(doc="VariantAnnotator              Annotates variant calls with context information.")), ToolInput(tag="in_beagle", input_type=String(), position=4, doc=InputDocumentation(doc="BeagleOutputToVCF             Takes files produced by Beagle imputation engine and creates a vcf with modified ")), ToolInput(tag="in_annotations_dot", input_type=String(), position=5, doc=InputDocumentation(doc="ProduceBeagleInput            Converts the input VCF into a format accepted by the Beagle imputation/analysis ")), ToolInput(tag="in_program_dot", input_type=String(), position=6, doc=InputDocumentation(doc="VariantsToBeagleUnphased      Produces an input file to Beagle imputation engine, listing unphased, hard-called ")), ToolInput(tag="in_callableloci_emits_file", input_type=String(), position=0, doc=InputDocumentation(doc="CallableLoci                  Emits a data file containing information about callable, uncallable, poorly mapped, and \nother parts of the genome <p/>")), ToolInput(tag="in_compare_callable_loci", input_type=String(), position=1, doc=InputDocumentation(doc="Test routine for new VariantContext object")), ToolInput(tag="in_depth_of_coverage", input_type=String(), position=2, doc=InputDocumentation(doc="Assess sequence coverage by a wide array of metrics, partitioned by sample, read group, \nor library")), ToolInput(tag="in_gc_content_by_interval", input_type=String(), position=3, doc=InputDocumentation(doc="Walks along reference and calculates the GC content for each interval.")), ToolInput(tag="in_diagnostics", input_type=String(), position=4, doc=InputDocumentation(doc="CoveredByNSamplesSites        Print intervals file with all the variant sites for which most of the samples have good ")), ToolInput(tag="in_errorratepercycle_compute_rate", input_type=String(), position=5, doc=InputDocumentation(doc="ErrorRatePerCycle             Compute the read error rate per position")), ToolInput(tag="in_read_group_properties", input_type=String(), position=6, doc=InputDocumentation(doc="Emits a GATKReport containing read group, sample, library, platform, center, sequencing \ndata, paired end status, simple read type name (e.g.")), ToolInput(tag="in_read_length_distribution", input_type=String(), position=7, doc=InputDocumentation(doc="Outputs the read lengths of all the reads in a file.")), ToolInput(tag="in_diff_engine", input_type=String(), position=8, doc=InputDocumentation(doc="DiffObjects                   A generic engine for comparing tree-structured objects")), ToolInput(tag="in_examples", input_type=String(), position=9, doc=InputDocumentation(doc="GATKPaperGenotyper            A simple Bayesian genotyper, that outputs a text based call format.")), ToolInput(tag="in_fast_a", input_type=String(), position=10, doc=InputDocumentation(doc="FastaAlternateReferenceMaker  Generates an alternative reference sequence over the specified interval.")), ToolInput(tag="in_fast_a_reference_maker", input_type=String(), position=11, doc=InputDocumentation(doc="Renders a new reference in FASTA format consisting of only those loci provided in the \ninput data set.")), ToolInput(tag="in_fast_a_stats", input_type=String(), position=12, doc=InputDocumentation(doc="Calculate basic statistics about the reference sequence itself")), ToolInput(tag="in_filters", input_type=String(), position=13, doc=InputDocumentation(doc="VariantFiltration             Filters variant calls using a number of user-selectable, parameterizable criteria.")), ToolInput(tag="in_qc", input_type=String(), position=14, doc=InputDocumentation(doc="CheckPileup                   Compare GATK's internal pileup to a reference Samtools pileup")), ToolInput(tag="in_count_bases", input_type=String(), position=15, doc=InputDocumentation(doc="Walks over the input data set, calculating the number of bases seen for diagnostic ")), ToolInput(tag="in_countintervals_count_contiguous", input_type=String(), position=16, doc=InputDocumentation(doc="CountIntervals                Count contiguous regions in an interval list.")), ToolInput(tag="in_count_loci", input_type=String(), position=17, doc=InputDocumentation(doc="Walks over the input data set, calculating the total number of covered loci for \ndiagnostic purposes.")), ToolInput(tag="in_count_males", input_type=String(), position=18, doc=InputDocumentation(doc="Walks over the input data set, calculating the number of reads seen from male samples \nfor diagnostic purposes.")), ToolInput(tag="in_count_read_events", input_type=String(), position=19, doc=InputDocumentation(doc="Walks over the input data set, counting the number of read events (from the CIGAR \noperator)")), ToolInput(tag="in_count_reads", input_type=String(), position=20, doc=InputDocumentation(doc="Walks over the input data set, calculating the number of reads seen for diagnostic ")), ToolInput(tag="in_countrods_prints_encountered", input_type=String(), position=21, doc=InputDocumentation(doc="CountRODs                     Prints out counts of the number of reference ordered data objects encountered.")), ToolInput(tag="in_count_rods_by_ref", input_type=String(), position=22, doc=InputDocumentation(doc="Prints out counts of the number of reference ordered data objects encountered along the ")), ToolInput(tag="in_reference_dot", input_type=String(), position=23, doc=InputDocumentation(doc="CountTerminusEvent            Walks over the input data set, counting the number of reads ending in ")), ToolInput(tag="in_error_throwing", input_type=String(), position=0, doc=InputDocumentation(doc="A walker that simply throws errors.")), ToolInput(tag="in_flag_stat", input_type=String(), position=1, doc=InputDocumentation(doc="A reimplementation of the 'samtools flagstat' subcommand in the GATK")), ToolInput(tag="in_pile_up", input_type=String(), position=2, doc=InputDocumentation(doc="Emulates the samtools pileup command to print aligned reads")), ToolInput(tag="in_print_rods", input_type=String(), position=3, doc=InputDocumentation(doc="Prints out all of the RODs in the input data set.")), ToolInput(tag="in_qc_ref", input_type=String(), position=4, doc=InputDocumentation(doc="Quality control for the reference fasta")), ToolInput(tag="in_read_clipping_stats", input_type=String(), position=5, doc=InputDocumentation(doc="Read clipping statistics for all reads.")), ToolInput(tag="in_read_utils", input_type=String(), position=6, doc=InputDocumentation(doc="ClipReads                     Read clipping based on quality, position or sequence matching")), ToolInput(tag="in_print_reads", input_type=String(), position=7, doc=InputDocumentation(doc="Renders, in SAM/BAM format, all reads from the input data set in the order in which \nthey appear in the input file.")), ToolInput(tag="in_read_adaptor_trimmer", input_type=String(), position=8, doc=InputDocumentation(doc="Utility tool to blindly strip base adaptors.")), ToolInput(tag="in_splits_am_file", input_type=String(), position=9, doc=InputDocumentation(doc="Divides the input data set into separate BAM files, one for each sample in the input \ndata set.")), ToolInput(tag="in_variant_eval", input_type=String(), position=10, doc=InputDocumentation(doc="VariantEval                   General-purpose tool for variant evaluation (% in dbSNP, genotype concordance, Ti/Tv \nratios, and a lot more)")), ToolInput(tag="in_variant_utils", input_type=String(), position=11, doc=InputDocumentation(doc="CombineVariants               Combines VCF records from different sources.")), ToolInput(tag="in_filter_lifted_variants", input_type=String(), position=12, doc=InputDocumentation(doc="Filters a lifted-over VCF file for ref bases that have been changed.")), ToolInput(tag="in_genotype_concordance", input_type=String(), position=13, doc=InputDocumentation(doc="Genotype concordance (per-sample and aggregate counts and frequencies, NRD/NRS and site \nallele overlaps) between two callsets")), ToolInput(tag="in_left_align_and_trim_variants", input_type=String(), position=14, doc=InputDocumentation(doc="Left-aligns indels from a variants file.")), ToolInput(tag="in_lift_over_variants", input_type=String(), position=15, doc=InputDocumentation(doc="Lifts a VCF file over from one build to another.")), ToolInput(tag="in_randomly_split_variants", input_type=String(), position=16, doc=InputDocumentation(doc="Takes a VCF file, randomly splits variants into two different sets, and outputs 2 new \nVCFs with the results.")), ToolInput(tag="in_select_headers", input_type=String(), position=17, doc=InputDocumentation(doc="Selects headers from a VCF source.")), ToolInput(tag="in_select_variants", input_type=String(), position=18, doc=InputDocumentation(doc="Selects variants from a VCF source.")), ToolInput(tag="in_validate_variants", input_type=String(), position=19, doc=InputDocumentation(doc="Validates a VCF file with an extra strict set of criteria.")), ToolInput(tag="in_variants_to_allelic_primitives", input_type=String(), position=20, doc=InputDocumentation(doc="Takes alleles from a variants file and breaks them up (if possible) into more \nbasic/primitive alleles.")), ToolInput(tag="in_variants_to_binary_ped", input_type=String(), position=21, doc=InputDocumentation(doc="Converts a VCF file to a binary plink Ped file (.bed/.bim/.fam)")), ToolInput(tag="in_variants_to_table", input_type=String(), position=22, doc=InputDocumentation(doc="Emits specific fields from a VCF file to a tab-deliminated table")), ToolInput(tag="in_variants_to_vcf", input_type=String(), position=23, doc=InputDocumentation(doc="Converts variants from other file formats to VCF format.")), ToolInput(tag="in_variant_validation_assessor", input_type=String(), position=24, doc=InputDocumentation(doc="Annotates a validation (from Sequenom for example) VCF with QC metrics (HW-equilibrium, \n% failed probes)")), ToolInput(tag="in_vcf_walker", input_type=String(), position=25, doc=InputDocumentation(doc="VcfSimpleStatsWalker          ")), ToolInput(tag="in_walkers", input_type=String(), position=26, doc=InputDocumentation(doc="LeftAlignVariants             "))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Bcbio_Variation_V0_1_0().translate("wdl", allow_empty_container=True)

