from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, Directory, Int, String

Runreadstogenome_Pl_V0_1_0 = CommandToolBuilder(tool="runReadsToGenome.pl", base_command=["runReadsToGenome.pl"], inputs=[ToolInput(tag="in_spaceseparated_pairedend_reads", input_type=Boolean(optional=True), prefix="-p", doc=InputDocumentation(doc="'leftSequenceFile rightSequenceFile'\nSpace-separated paired-end reads in quote")), ToolInput(tag="in_sequencefileprovides_file_containing", input_type=Boolean(optional=True), prefix="-u", doc=InputDocumentation(doc="sequenceFile\nProvides a file containing single-end reads.")), ToolInput(tag="in_long", input_type=Boolean(optional=True), prefix="-long", doc=InputDocumentation(doc="long reads file in fasta format.")), ToolInput(tag="in_pac_bio", input_type=Boolean(optional=True), prefix="--pacbio", doc=InputDocumentation(doc="<bool> using this flag combined with -long for Pacbio long reads (bwa only)")), ToolInput(tag="in_ref", input_type=Boolean(optional=True), prefix="-ref", doc=InputDocumentation(doc="reference sequences file in fasta format")), ToolInput(tag="in_pre", input_type=Boolean(optional=True), prefix="-pre", doc=InputDocumentation(doc="output files' prefix (default 'ReadsMapping')")), ToolInput(tag="in_output_directory", input_type=Directory(optional=True), prefix="-d", doc=InputDocumentation(doc="output directory")), ToolInput(tag="in_consensus", input_type=Boolean(optional=True), prefix="-consensus", doc=InputDocumentation(doc="<bool> output consensus fasta file (default: on, set 0 to turn off)")), ToolInput(tag="in_aligner", input_type=Boolean(optional=True), prefix="-aligner", doc=InputDocumentation(doc="bwa or bowtie or snap or minimap2 (default: bwa)")), ToolInput(tag="in_bwa_options", input_type=Int(optional=True), prefix="-bwa_options", doc=InputDocumentation(doc="bwa options\ntype 'bwa mem' to see options\ndefault: '-t 4 '\n-t        <int> number of threads [4]\n-I        the input is in the Illumina 1.3+ FASTQ-like format")), ToolInput(tag="in_bowtie_options", input_type=Int(optional=True), prefix="-bowtie_options", doc=InputDocumentation(doc="bowtie options\ntype 'bowtie2 -h' to see options\ndefault: '-p 4 -a '\n-p           <int> number of alignment threads to launch [4]\n-a           report all alignments; very slow\n--phred64    qualities are Phred+64")), ToolInput(tag="in_snap_options", input_type=Boolean(optional=True), prefix="-snap_options", doc=InputDocumentation(doc="snap options\ntype 'snap paired' to see options")), ToolInput(tag="in_minimap_two_options", input_type=Boolean(optional=True), prefix="-minimap2_options", doc=InputDocumentation(doc="type 'minimap2' to see options\ndefault: '-t 4 '")), ToolInput(tag="in_skip_aln", input_type=Boolean(optional=True), prefix="-skip_aln", doc=InputDocumentation(doc="<bool> skip the alignment steps, assume bam files were generated\nand with proper prefix,outpurDir.  default: off")), ToolInput(tag="in_no_plot", input_type=Boolean(optional=True), prefix="-no_plot", doc=InputDocumentation(doc="<bool> default: off")), ToolInput(tag="in_no_snp", input_type=Boolean(optional=True), prefix="-no_snp", doc=InputDocumentation(doc="<bool> default: off")), ToolInput(tag="in_debug", input_type=Boolean(optional=True), prefix="-debug", doc=InputDocumentation(doc="<bool> default: off")), ToolInput(tag="in_cpu", input_type=Boolean(optional=True), prefix="-cpu", doc=InputDocumentation(doc="number of CPUs [4]. will overwrite aligner options.")), ToolInput(tag="in_min_in_del_candidate_depth", input_type=Int(optional=True), prefix="-min_indel_candidate_depth", doc=InputDocumentation(doc="number gapped reads for indel candidates [3]")), ToolInput(tag="in_min_alt_bases", input_type=Boolean(optional=True), prefix="-min_alt_bases", doc=InputDocumentation(doc="minimum number of alternate bases [3]")), ToolInput(tag="in_max_depth", input_type=Boolean(optional=True), prefix="-max_depth", doc=InputDocumentation(doc="maximum read depth [1000000]")), ToolInput(tag="in_min_depth", input_type=Boolean(optional=True), prefix="-min_depth", doc=InputDocumentation(doc="minimum read depth [7]")), ToolInput(tag="in_snp_gap_filter", input_type=Boolean(optional=True), prefix="-snp_gap_filter", doc=InputDocumentation(doc="SNP within INT bp around a gap to be filtered [3]")), ToolInput(tag="in_perl", input_type=String(), position=0, doc=InputDocumentation(doc=""))], outputs=[ToolOutput(tag="out_output_directory", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_output_directory", type_hint=File()), doc=OutputDocumentation(doc="output directory"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Runreadstogenome_Pl_V0_1_0().translate("wdl", allow_empty_container=True)

