from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, String, Int, Float

F5C_Call_Methylation_V0_1_0 = CommandToolBuilder(tool="f5c_call_methylation", base_command=["f5c", "call-methylation"], inputs=[ToolInput(tag="in_fastqfasta_read_file", input_type=File(optional=True), prefix="-r", doc=InputDocumentation(doc="fastq/fasta read file")), ToolInput(tag="in_sorted_bam_file", input_type=File(optional=True), prefix="-b", doc=InputDocumentation(doc="sorted bam file")), ToolInput(tag="in_reference_genome", input_type=File(optional=True), prefix="-g", doc=InputDocumentation(doc="reference genome")), ToolInput(tag="in_limit_processing_genomic", input_type=String(optional=True), prefix="-w", doc=InputDocumentation(doc="limit processing to genomic region string of format chr:start-end")), ToolInput(tag="in_number_processing_threads", input_type=Int(optional=True), prefix="-t", doc=InputDocumentation(doc="number of processing threads [8]")), ToolInput(tag="in_batch_size_max", input_type=Int(optional=True), prefix="-K", doc=InputDocumentation(doc="batch size (max number of reads loaded at once) [512]")), ToolInput(tag="in_var_6", input_type=Float(optional=True), prefix="-B", doc=InputDocumentation(doc="[K/M/G]            max number of bases loaded at once [5.0M]")), ToolInput(tag="in_output_to_file", input_type=File(optional=True), prefix="-o", doc=InputDocumentation(doc="output to file [stdout]")), ToolInput(tag="in_parameter_profile_used", input_type=Int(optional=True), prefix="-x", doc=InputDocumentation(doc="parameter profile to be used for better performance (always applied before other options)\ne.g., laptop, desktop, hpc; see https://f5c.page.link/profiles for the full list")), ToolInput(tag="in_iop", input_type=Int(optional=True), prefix="--iop", doc=InputDocumentation(doc="number of I/O processes to read fast5 files [1]")), ToolInput(tag="in_min_mapq", input_type=Int(optional=True), prefix="--min-mapq", doc=InputDocumentation(doc="minimum mapping quality [20]")), ToolInput(tag="in_secondary", input_type=String(optional=True), prefix="--secondary", doc=InputDocumentation(doc="|no         consider secondary mappings or not [no]")), ToolInput(tag="in_verbose", input_type=Int(optional=True), prefix="--verbose", doc=InputDocumentation(doc="verbosity level [0]")), ToolInput(tag="in_skip_ultra", input_type=File(optional=True), prefix="--skip-ultra", doc=InputDocumentation(doc="skip ultra long reads and write those entries to the bam file provided as the argument")), ToolInput(tag="in_ultra_thresh", input_type=Int(optional=True), prefix="--ultra-thresh", doc=InputDocumentation(doc="threshold to skip ultra long reads [100000]")), ToolInput(tag="in_skip_unreadable", input_type=Int(optional=True), prefix="--skip-unreadable", doc=InputDocumentation(doc="|no   skip any unreadable fast5 or terminate program [yes]")), ToolInput(tag="in_km_er_model", input_type=File(optional=True), prefix="--kmer-model", doc=InputDocumentation(doc="custom nucleotide k-mer model file (format similar to test/r9-models/r9.4_450bps.nucleotide.6mer.template.model)")), ToolInput(tag="in_meth_model", input_type=File(optional=True), prefix="--meth-model", doc=InputDocumentation(doc="custom methylation k-mer model file (format similar to test/r9-models/r9.4_450bps.cpg.6mer.template.model)")), ToolInput(tag="in_meth_out_version", input_type=Int(optional=True), prefix="--meth-out-version", doc=InputDocumentation(doc="methylation tsv output version (set 2 to print the strand column) [1]")), ToolInput(tag="in_print_events", input_type=String(optional=True), prefix="--print-events", doc=InputDocumentation(doc="|no      prints the event table")), ToolInput(tag="in_print_banded_aln", input_type=String(optional=True), prefix="--print-banded-aln", doc=InputDocumentation(doc="|no  prints the event alignment")), ToolInput(tag="in_print_scaling", input_type=String(optional=True), prefix="--print-scaling", doc=InputDocumentation(doc="|no     prints the estimated scalings")), ToolInput(tag="in_print_raw", input_type=String(optional=True), prefix="--print-raw", doc=InputDocumentation(doc="|no         prints the raw signal")), ToolInput(tag="in_debug_break", input_type=Int(optional=True), prefix="--debug-break", doc=InputDocumentation(doc="break after processing the specified no. of batches")), ToolInput(tag="in_profile_cpu", input_type=String(optional=True), prefix="--profile-cpu", doc=InputDocumentation(doc="|no       process section by section (used for profiling on CPU)")), ToolInput(tag="in_write_dump", input_type=File(optional=True), prefix="--write-dump", doc=InputDocumentation(doc="|no        write the fast5 dump to a file or not")), ToolInput(tag="in_read_dump", input_type=File(optional=True), prefix="--read-dump", doc=InputDocumentation(doc="|no         read from a fast5 dump file or not"))], outputs=[ToolOutput(tag="out_output_to_file", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output_to_file", type_hint=File()), doc=OutputDocumentation(doc="output to file [stdout]"))], container="quay.io/biocontainers/f5c:0.6--h8b6279f_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    F5C_Call_Methylation_V0_1_0().translate("wdl")

