version 1.0

task MethylpyDMRfind {
  input {
    Array[String] all_c_files
    String? output_prefix
    Array[String] samples
    Array[String] chrom_s
    Array[String] mc_type
    String? num_procs
    Int? min_cov
    String? dmr_max_dist
    String? sig_cut_off
    String? num_sims
    Int? min_tests
    Int? min_num_dms
    Array[String] sample_category
    String? mc_max_dist
    String? resid_cut_off
    String? keep_temp_files
    Int? min_cluster
    String? seed
  }
  command <<<
    methylpy DMRfind \
      ~{if defined(all_c_files) then ("--allc-files " +  '"' + all_c_files + '"') else ""} \
      ~{if defined(output_prefix) then ("--output-prefix " +  '"' + output_prefix + '"') else ""} \
      ~{if defined(samples) then ("--samples " +  '"' + samples + '"') else ""} \
      ~{if defined(chrom_s) then ("--chroms " +  '"' + chrom_s + '"') else ""} \
      ~{if defined(mc_type) then ("--mc-type " +  '"' + mc_type + '"') else ""} \
      ~{if defined(num_procs) then ("--num-procs " +  '"' + num_procs + '"') else ""} \
      ~{if defined(min_cov) then ("--min-cov " +  '"' + min_cov + '"') else ""} \
      ~{if defined(dmr_max_dist) then ("--dmr-max-dist " +  '"' + dmr_max_dist + '"') else ""} \
      ~{if defined(sig_cut_off) then ("--sig-cutoff " +  '"' + sig_cut_off + '"') else ""} \
      ~{if defined(num_sims) then ("--num-sims " +  '"' + num_sims + '"') else ""} \
      ~{if defined(min_tests) then ("--min-tests " +  '"' + min_tests + '"') else ""} \
      ~{if defined(min_num_dms) then ("--min-num-dms " +  '"' + min_num_dms + '"') else ""} \
      ~{if defined(sample_category) then ("--sample-category " +  '"' + sample_category + '"') else ""} \
      ~{if defined(mc_max_dist) then ("--mc-max-dist " +  '"' + mc_max_dist + '"') else ""} \
      ~{if defined(resid_cut_off) then ("--resid-cutoff " +  '"' + resid_cut_off + '"') else ""} \
      ~{if defined(keep_temp_files) then ("--keep-temp-files " +  '"' + keep_temp_files + '"') else ""} \
      ~{if defined(min_cluster) then ("--min-cluster " +  '"' + min_cluster + '"') else ""} \
      ~{if defined(seed) then ("--seed " +  '"' + seed + '"') else ""}
  >>>
  parameter_meta {
    all_c_files: "List of allc files. (default: None)"
    output_prefix: "String indicating the prefix for output files (default: None)"
    samples: "List of space separated samples matching allc files. By default sample names will be inferred from allc filenames (default: None)"
    chrom_s: "Space separated listing of chromosomes where DMRs will be called. If not specified, DMRs will be called across the chromosomes/contigs that contained any data in all allc files. (default: None)"
    mc_type: "List of space separated mc nucleotide contexts for which you want to look for DMRs. These classifications may use the wildcards H (indicating anything but a G) and N (indicating any nucleotide). (default: ['CGN'])"
    num_procs: "Number of processors you wish to use to parallelize this function (default: 1)"
    min_cov: "Minimum number of reads that must cover a site for it to be considered. (default: 0)"
    dmr_max_dist: "Maximum distance two significant sites can be to be included in the same block. (default: 250)"
    sig_cut_off: "Float indicating at what FDR you want to consider a result significant. (default: 0.01)"
    num_sims: "Number of permutation tests you would like to run to estimate the p-values of the differential methylation tests (default: 3000)"
    min_tests: "Minimum number of permuation tests you\ would d like to run for each mC (default: 100)"
    min_num_dms: "The minimum number of differentially methylated sites that a differentially methylated region needs to contain to be reported (default: 0)"
    sample_category: "A list of categories that each respective sample belongs to; the categories must begin at 0 and increase by 1 for each category added. ex: samples [A,B,C] categories [0,1,2] or categories [0, 1, 0] (default: False)"
    mc_max_dist: "Integer indicating the maximum distance two sites can be from one another for their methylation counts to be combined. This option helps with low coverage experiments where you may want to leverage the correlation of methylation between sites to get more statistical power. (default: 0)"
    resid_cut_off: "Results will have to show deviations in the contingency table in the same direction as the rest of the window (default: 0.01)"
    keep_temp_files: "Boolean indicating that you would like to keep the intermediate files generated by this function. This can be useful for debugging, but in general should be left False. (default: False)"
    min_cluster: "The minimum number of each sample category that must be present in every block that is output. (default: 2)"
    seed: "A seed to provide to the random number generator for permutation testing. Only change this if you are debugging and want to make sure the permutation output is consistent (default: -1)"
  }
}