class: CommandLineTool
id: ../../../RiboCode.cwl
inputs:
- id: an_not_dir
  doc: transcripts annotation directory, generated by prepare_transcripts.
  type: string
  inputBinding:
    prefix: --annot_dir
- id: config_file
  doc: list bam file and P-sites information in this file, please refer to the example
    file in data folder.
  type: string
  inputBinding:
    prefix: --config_file
- id: longest_orf
  doc: 'Default: yes, the region from most distal AUG to stop was defined as an ORF.
    If set to no , the position of start codon will be automatically determined by
    program.'
  type: string
  inputBinding:
    prefix: --longest-orf
- id: pval_cut_off
  doc: P-value cutoff for ORF filtering, default 0.05
  type: string
  inputBinding:
    prefix: --pval-cutoff
- id: start_codon
  doc: 'The canonical start codon. default: ATG'
  type: string
  inputBinding:
    prefix: --start_codon
- id: alt_start_codons
  doc: 'The alternative start codon, such as CTG,GTG, default: None. Multiple codons
    should be separated by comma.'
  type: string
  inputBinding:
    prefix: --alt_start_codons
- id: stop_codon
  doc: 'Stop codon, default: TAA,TAG,TGA'
  type: string
  inputBinding:
    prefix: --stop_codon
- id: transl_table
  doc: 'ORF translation table(Default: 1). Assign the correct genetic code based on
    your organism, [please refer: ht tps://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi
    ]'
  type: string
  inputBinding:
    prefix: --transl_table
- id: min_aa_length
  doc: The minimal length of predicted peptides,default 20
  type: long
  inputBinding:
    prefix: --min-AA-length
- id: output_gtf
  doc: output the gtf file of predicted ORFs
  type: boolean
  inputBinding:
    prefix: --output-gtf
- id: output_bed
  doc: output the bed file of predicted ORFs
  type: boolean
  inputBinding:
    prefix: --output-bed
- id: output_name
  doc: 'output file name, default: final_result'
  type: string
  inputBinding:
    prefix: --output-name
outputs: []
cwlVersion: v1.1
baseCommand:
- RiboCode
