class: CommandLineTool
id: anvi_summarize.cwl
inputs:
- id: in_pan_or_profile_db
  doc: "Anvi'o pan or profile database (and even genes\ndatabase in appropriate contexts).\
    \ (default: None)"
  type: string?
  inputBinding:
    prefix: --pan-or-profile-db
- id: in_contigs_db
  doc: "Anvi'o contigs database generated by 'anvi-gen-\ncontigs-database' (default:\
    \ None)"
  type: string?
  inputBinding:
    prefix: --contigs-db
- id: in_genomes_storage
  doc: "Anvi'o genomes storage file (default: None)"
  type: File?
  inputBinding:
    prefix: --genomes-storage
- id: in_in_it_gene_coverages
  doc: "Initialize gene coverage and detection data. This is a\nvery computationally\
    \ expensive step, but it is\nnecessary when you need gene level coverage data.\
    \ The\nreason this is very computationally expensive is\nbecause anvi'o computes\
    \ gene coverages by going back\nto actual coverage values of each gene to average\n\
    them, instead of using contig average coverage values,\nfor extreme accuracy.\
    \ (default: False)"
  type: boolean?
  inputBinding:
    prefix: --init-gene-coverages
- id: in_reformat_contig_names
  doc: "Reformat contig names while generating the summary\noutput so they look fancy.\
    \ With this flag, anvi'o will\nreplace the original names of contigs to those\
    \ that\ninclude the bin name as a prefix in resulting summary\noutput files per\
    \ bin. Use this flag carefully as it\nmay influence your downstream analyses due\
    \ to the fact\nthat your original contig names in your input FASTA\nfile for the\
    \ contigs database will not be in the\nsummary output. Although, anvi'o will report\
    \ a\nconversion map per bin so you can recover the original\ncontig name if you\
    \ have to. (default: False)"
  type: boolean?
  inputBinding:
    prefix: --reformat-contig-names
- id: in_report_aa_seqs_for_gene_calls
  doc: "You can use this flag if you would like amino acid AND\ndna sequences for\
    \ your gene calls in the genes output\nfile. By default, only dna sequences are\
    \ reported.\n(default: False)"
  type: File?
  inputBinding:
    prefix: --report-aa-seqs-for-gene-calls
- id: in_report_dna_sequences
  doc: "By default, this program reports amino acid sequences.\nUse this flag to report\
    \ DNA sequences instead. Also\nnote, since gene clusters are aligned via amino\
    \ acid\nsequences, using this flag removes alignment\ninformation manifesting\
    \ in the form of gap characters\n(`-` characters) that would be present if amino\
    \ acid\nsequences were reported. Read the warnings during\nruntime for more detailed\
    \ information. (default:\nFalse)"
  type: boolean?
  inputBinding:
    prefix: --report-DNA-sequences
- id: in_collection_name
  doc: 'Collection name. (default: None)'
  type: string?
  inputBinding:
    prefix: --collection-name
- id: in_output_dir
  doc: 'Directory path for output files (default: None)'
  type: File?
  inputBinding:
    prefix: --output-dir
- id: in_list_collections
  doc: 'Show available collections and exit. (default: False)'
  type: boolean?
  inputBinding:
    prefix: --list-collections
- id: in_cog_data_dir
  doc: "The directory path for your COG setup. Anvi'o will try\nto use the default\
    \ path if you do not specify\nanything. (default: None)"
  type: File?
  inputBinding:
    prefix: --cog-data-dir
- id: in_quick_summary
  doc: "When declared the summary output will be generated as\nquickly as possible,\
    \ with minimum amount of essential\ninformation about bins. (default: False)"
  type: boolean?
  inputBinding:
    prefix: --quick-summary
- id: in_just_do_it
  doc: "Don't bother me with questions or warnings, just do\nit. (default: False)"
  type: boolean?
  inputBinding:
    prefix: --just-do-it
- id: in_summary
  doc: 'üç∫ More on `anvi-summarize`:'
  type: string
  inputBinding:
    position: 0
outputs:
- id: out_stdout
  doc: Standard output stream
  type: stdout
- id: out_report_aa_seqs_for_gene_calls
  doc: "You can use this flag if you would like amino acid AND\ndna sequences for\
    \ your gene calls in the genes output\nfile. By default, only dna sequences are\
    \ reported.\n(default: False)"
  type: File?
  outputBinding:
    glob: $(inputs.in_report_aa_seqs_for_gene_calls)
- id: out_output_dir
  doc: 'Directory path for output files (default: None)'
  type: File?
  outputBinding:
    glob: $(inputs.in_output_dir)
hints:
- class: DockerRequirement
  dockerPull: quay.io/biocontainers/anvio-minimal:7--py_0
cwlVersion: v1.1
baseCommand:
- anvi-summarize
