from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Directory, Boolean, String, Int, Float

Krakenhll_Download_V0_1_0 = CommandToolBuilder(tool="krakenhll_download", base_command=["krakenhll-download"], inputs=[ToolInput(tag="in_folder_files_downloaded", input_type=Directory(optional=True), prefix="-o", doc=InputDocumentation(doc="Folder to which the files are downloaded. Default: '.'")), ToolInput(tag="in_db", input_type=Directory(optional=True), prefix="--db", doc=InputDocumentation(doc="Alternative to -o: Download to <directory>/{library,taxonomy}.")), ToolInput(tag="in_threads", input_type=Boolean(optional=True), prefix="--threads", doc=InputDocumentation(doc="<# of threads>  Number of processes when downloading (uses xargs). Default: '5'")), ToolInput(tag="in_rsync", input_type=Boolean(optional=True), prefix="--rsync", doc=InputDocumentation(doc="Download using rsync.")), ToolInput(tag="in_overwrite", input_type=Boolean(optional=True), prefix="--overwrite", doc=InputDocumentation(doc="Redownload and overwrite files with the same name.")), ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="--verbose", doc=InputDocumentation(doc="Be verbose.")), ToolInput(tag="in_term", input_type=String(optional=True), prefix="--term", doc=InputDocumentation(doc="Download all sequences returned from a NCBI nucleotide search.")), ToolInput(tag="in_mappings", input_type=Int(optional=True), prefix="--mappings", doc=InputDocumentation(doc="Try mapping accession IDs using the mapping files for the specified divisions (comma-separated).\nDefault: nucl_gb. Possible values: nucl_est, nucl_gb, nucl_gss, nucl_wgs.\nDownloaded from ftp://ftp.ncbi.nlm.nih.gov/pub/taxonomy/accession2taxid/.")), ToolInput(tag="in_ret_mode", input_type=Float(optional=True), prefix="--retmode", doc=InputDocumentation(doc="Specify return mode. Possible options: text (default), asn.1, xml.")), ToolInput(tag="in_ret_type", input_type=Int(optional=True), prefix="--rettype", doc=InputDocumentation(doc="Specify return type for download. Note that no mapping files are generated when\nrettype is specified. Possible options: fasta (default), gb, gbc, native, acc, seqid, ft, gbwithparts, fasta_cds_na, fasta_cds_aa. Note that only gb and fasta files are split, while the other formats stay in chunks.\nDownloaded from ftp://ftp.ncbi.nlm.nih.gov/pub/taxonomy/accession2taxid/.")), ToolInput(tag="in_fna", input_type=String(optional=True), prefix="--fna", doc=InputDocumentation(doc="Comma-separated list of sequence types, including genomic, rna, rna_from_genomic, cds_from_genomic. Default: genomic.\nSee the assembly project FTP site for available sequences")), ToolInput(tag="in_filter_unplaced_sequences", input_type=Boolean(optional=True), prefix="-u", doc=InputDocumentation(doc="Filter unplaced sequences.")), ToolInput(tag="in_modify_sequence_header", input_type=Boolean(optional=True), prefix="-l", doc=InputDocumentation(doc="Modify sequence header to include taxonomy ID for Kraken (i.e. add '>kraken:taxid|TAXID' to each sequence).")), ToolInput(tag="in_dust", input_type=Boolean(optional=True), prefix="--dust", doc=InputDocumentation(doc="Mask low-complexity regions using dustmasker.")), ToolInput(tag="in_include_viral_neighbors", input_type=Boolean(optional=True), prefix="--include-viral-neighbors", doc=InputDocumentation(doc="Include viral neighbors. Deprecated. Add 'viral-neighbors' to the arguments, instead."))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Krakenhll_Download_V0_1_0().translate("wdl", allow_empty_container=True)

