from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, File, String

Daa2Info_V0_1_0 = CommandToolBuilder(tool="daa2info", base_command=["daa2info"], inputs=[ToolInput(tag="in_in", input_type=Boolean(optional=True), prefix="--in", doc=InputDocumentation(doc="[string]                    Input DAA file. Mandatory option.")), ToolInput(tag="in_out", input_type=File(optional=True), prefix="--out", doc=InputDocumentation(doc="[string]                   Output file or '-' for stdout. Default value: -.")), ToolInput(tag="in_list_more", input_type=Boolean(optional=True), prefix="--listMore", doc=InputDocumentation(doc="List more info about file (if meganized). Default value: false.")), ToolInput(tag="in_class_two_count", input_type=Boolean(optional=True), prefix="--class2count", doc=InputDocumentation(doc="[string(s)]      List class to count for named classification(s) (Possible values: EGGNOG INTERPRO2GO KEGG SEED Taxonomy).")), ToolInput(tag="in_read_two_class", input_type=Boolean(optional=True), prefix="--read2class", doc=InputDocumentation(doc="[string(s)]       List read to class assignments for named classification(s) (Possible values: EGGNOG INTERPRO2GO KEGG SEED Taxonomy).")), ToolInput(tag="in_names", input_type=Boolean(optional=True), prefix="--names", doc=InputDocumentation(doc="Report class names rather than class Id numbers. Default value: false.")), ToolInput(tag="in_paths", input_type=Boolean(optional=True), prefix="--paths", doc=InputDocumentation(doc="Report class paths rather than class Id numbers. Default value: false.")), ToolInput(tag="in_prefix_rank", input_type=Boolean(optional=True), prefix="--prefixRank", doc=InputDocumentation(doc="When reporting class paths for taxonomy, prefix single letter to indicate taxonomic rank. Default value: false.")), ToolInput(tag="in_major_ranks_only", input_type=Boolean(optional=True), prefix="--majorRanksOnly", doc=InputDocumentation(doc="Only use major taxonomic ranks. Default value: false.")), ToolInput(tag="in_bacteria_only", input_type=Boolean(optional=True), prefix="--bacteriaOnly", doc=InputDocumentation(doc="Only report bacterial reads and counts in taxonomic report. Default value: false.")), ToolInput(tag="in_virus_only", input_type=Boolean(optional=True), prefix="--virusOnly", doc=InputDocumentation(doc="Only report viral reads and counts in taxonomic report. Default value: false.")), ToolInput(tag="in_ignore_unassigned", input_type=Boolean(optional=True), prefix="--ignoreUnassigned", doc=InputDocumentation(doc="Don't report on reads that are unassigned. Default value: true.")), ToolInput(tag="in_extract_summary_file", input_type=File(optional=True), prefix="--extractSummaryFile", doc=InputDocumentation(doc="[string]   Output a MEGAN summary file (contains all classifications, but no reads or alignments.")), ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="--verbose", doc=InputDocumentation(doc="Echo commandline options and be verbose. Default value: false.")), ToolInput(tag="in_commands", input_type=String(), position=0, doc=InputDocumentation(doc="-l, --list                           List general info about file. Default value: false."))], outputs=[ToolOutput(tag="out_out", output_type=File(optional=True), selector=InputSelector(input_to_select="in_out", type_hint=File()), doc=OutputDocumentation(doc="[string]                   Output file or '-' for stdout. Default value: -.")), ToolOutput(tag="out_extract_summary_file", output_type=File(optional=True), selector=InputSelector(input_to_select="in_extract_summary_file", type_hint=File()), doc=OutputDocumentation(doc="[string]   Output a MEGAN summary file (contains all classifications, but no reads or alignments."))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Daa2Info_V0_1_0().translate("wdl", allow_empty_container=True)

