#!/usr/bin/env cwl-runner

baseCommand:
- anvi-profile
class: CommandLineTool
cwlVersion: v1.0
id: anvi-profile
inputs:
- doc: Sorted and indexed BAM file to analyze. Takes a long time depending on the
    length of the file and parameters used for profiling.
  id: input_file
  inputBinding:
    prefix: --input-file
  type: string
- doc: Anvi'o contigs database generated by 'anvi-gen- contigs'
  id: contigs_db
  inputBinding:
    prefix: --contigs-db
  type: string
- doc: If you only have contig sequences, but no mapping data (i.e., you found a genome
    and would like to take a look from it), this flag will become very hand. After
    creating a contigs database for your contigs, you can create a blank anvi'o profile
    database to use anvi'o interactive interface with that contigs database without
    any mapping data.
  id: blank_profile
  inputBinding:
    prefix: --blank-profile
  type: boolean
- doc: Directory path for output files
  id: output_dir
  inputBinding:
    prefix: --output-dir
  type: string
- doc: Overwrite if the output files and/or directories exist.
  id: overwrite_output_destinations
  inputBinding:
    prefix: --overwrite-output-destinations
  type: boolean
- doc: It is important to set a sample name (using only ASCII letters and digits and
    without spaces) that is unique (considering all others). If you do not provide
    one, anvi'o will try to make up one for you based on other information, although,
    you should never let the software to decide these things).
  id: sample_name
  inputBinding:
    prefix: --sample-name
  type: string
- doc: One of the things anvi-profile does is to store information about variable
    nucleotide positions. Usually it does not report every variable position, since
    not every variable position is genuine variation. Say, if you have 1,000 coverage,
    and all nucleotides at that position are Ts and only one of them is a C, the confidence
    of that C being a real variation is quite low. anvi'o has a simple algorithm in
    place to reduce the impact of noise. However, using this flag you can disable
    it and ask profiler to report every single variation (which may result in very
    large output files and millions of reports, but you are the boss). Do not forget
    to take a look at '-- min-coverage-for-variability' parameter
  id: report_variability_full
  inputBinding:
    prefix: --report-variability-full
  type: boolean
- doc: By default, anvi'o characterizes single-nucleotide variation in each sample.
    The use of this flag will instruct profiler to skip that step. Please remember
    that parameters and flags must be identical between different profiles using the
    same contigs database for them to merge properly.
  id: skip_snv_profiling
  inputBinding:
    prefix: --skip-SNV-profiling
  type: boolean
- doc: Anvi'o can perform accurate characterization of codon frequencies in genes
    during profiling. While having codon frequencies opens doors to powerful evolutionary
    insights in downstream analyses, due to its computational complexity, this feature
    comes 'off' by default. Using this flag you can rise against the authority, as
    you always should, and make anvi'o profile codons.
  id: profile_s_cvs
  inputBinding:
    prefix: --profile-SCVs
  type: boolean
- doc: A plain text file that contains some description about the project. You can
    use Markdwon syntax. The description text will be rendered and shown in all relevant
    interfaces, including the anvi'o interactive interface, or anvi'o summary outputs.
  id: description
  inputBinding:
    prefix: --description
  type: string
- doc: Single profiles are rarely used for genome binning or visualization, and since
    clustering step increases the profiling runtime for no good reason, the default
    behavior is to not cluster contigs for individual runs. However, if you are planning
    to do binning on one sample, you must use this flag to tell anvi'o to run cluster
    configurations for single runs on your sample.
  id: cluster_contigs
  inputBinding:
    prefix: --cluster-contigs
  type: boolean
- doc: If you are not planning to use the interactive interface (or if you have other
    means to add a tree of contigs in the database) you may skip the step where hierarchical
    clustering of your items are preformed based on default clustering recipes matching
    to your database type.
  id: skip_hierarchical_clustering
  inputBinding:
    prefix: --skip-hierarchical-clustering
  type: boolean
- doc: The distance metric for the hierarchical clustering. Only relevant if you are
    using `--cluster-contigs` flag. The default is "euclidean".
  id: distance
  inputBinding:
    prefix: --distance
  type: string
- doc: The linkage method for the hierarchical clustering. Just like the distance
    metric this is only relevant if you are using it with `--cluster-contigs` flag.
    The default is "ward".
  id: linkage
  inputBinding:
    prefix: --linkage
  type: string
- doc: Minimum length of contigs in a BAM file to analyze. The minimum length should
    be long enough for tetra- nucleotide frequency analysis to be meaningful. There
    is no way to define a golden number of minimum length that would be applicable
    to genomes found in all environments, but we chose the default to be 1000, and
    have been happy with it. You are welcome to experiment, but we advise to never
    go below 1,000. You also should remember that the lower you go, the more time
    it will take to analyze all contigs. You can use --list-contigs parameter to have
    an idea how many contigs would be discarded for a given M.
  id: min_contig_length
  inputBinding:
    prefix: --min-contig-length
  type: long
- doc: Just like the minimum contig length parameter, but to set a maximum. Basically
    this will remove any contig longer than a certain value. Why would anyone need
    this? Who knows. But if you ever do, it is here.
  id: max_contig_length
  inputBinding:
    prefix: --max-contig-length
  type: long
- doc: Minimum mean coverage for contigs to be kept in the analysis. The default value
    is 0, which is for your best interest if you are going to profile multiple BAM
    files which are then going to be merged for a cross- sectional or time series
    analysis. Do not change it if you are not sure this is what you want to do.
  id: min_mean_coverage
  inputBinding:
    prefix: --min-mean-coverage
  type: long
- doc: Minimum coverage of a nucleotide position to be subjected to SNV profiling.
    By default, anvi'o will not attempt to make sense of variation in a given nucleotide
    position if it is covered less than 10X. You can change that minimum using this
    parameter.
  id: min_coverage_for_variability
  inputBinding:
    prefix: --min-coverage-for-variability
  type: long
- doc: When declared, the program will list contigs in the BAM file and exit gracefully
    without any further analysis.
  id: list_contigs
  inputBinding:
    prefix: --list-contigs
  type: boolean
- doc: It is possible to analyze only a group of contigs from a given BAM file. If
    you provide a text file, in which every contig of interest is listed line by line,
    the profiler would engine only on those contigs in the BAM file and ignore the
    rest. This can be used for debugging purposes, or to engine on a particular group
    of contigs that were identified as relevant during the interactive analysis.
  id: contigs_of_interest
  inputBinding:
    prefix: --contigs-of-interest
  type: File
- doc: Maximum number of threads to use for multithreading whenever possible. Very
    conservatively, the default is 1. It is a good idea to not exceed the number of
    CPUs / cores on your system. Plus, please be careful with this option if you are
    running your commands on a SGE --if you are clusterizing your runs, and asking
    for multiple threads to use, you may deplete your resources very fast.
  id: num_threads
  inputBinding:
    prefix: --num-threads
  type: string
- doc: The queue size for worker threads to store data to communicate to the main
    thread. The default is set by the class based on the number of threads. If you
    have *any* hesitation about whether you know what you are doing, you should not
    change this value.
  id: queue_size
  inputBinding:
    prefix: --queue-size
  type: long
- doc: How many items should be kept in memory before they are written do the disk.
    The default is 500 per thread. So a single-threaded job would have a write buffer
    size of 500, whereas a job with 4 threads would have a write buffer size of 4*500.
    The larger the buffer size, the less frequent the program will access to the disk,
    yet the more memory will be consumed since the processed items will be cleared
    off the memory only after they are written to the disk. The default buffer size
    will likely work for most cases. Please keep an eye on the memory usage output
    to make sure the memory use never exceeds the size of the physical memory.
  id: write_buffer_size_per_thread
  inputBinding:
    prefix: --write-buffer-size-per-thread
  type: long
- doc: This is not useful to non-developers. It forces the multi-process routine even
    when 1 thread is chosen.
  id: force_multi
  inputBinding:
    prefix: --force-multi
  type: boolean
