version 1.0

task AnviDisplayStructure {
  input {
    String? structure_db
    String? profile_db
    String? contigs_db
    String? variability_profile
    File? splits_of_interest
    String? collection_name
    String? bin_id
    File? samples_of_interest
    File? genes_of_interest
    String? gene_caller_ids
    Float? min_departure_from_consensus
    Boolean? sa_avs_only
    Boolean? s_cvs_only
    String? ip_address
    Int? port_number
    File? browser_path
    Boolean? server_only
    Boolean? password_protected
  }
  command <<<
    anvi-display-structure \
      ~{if defined(structure_db) then ("--structure-db " +  '"' + structure_db + '"') else ""} \
      ~{if defined(profile_db) then ("--profile-db " +  '"' + profile_db + '"') else ""} \
      ~{if defined(contigs_db) then ("--contigs-db " +  '"' + contigs_db + '"') else ""} \
      ~{if defined(variability_profile) then ("--variability-profile " +  '"' + variability_profile + '"') else ""} \
      ~{if defined(splits_of_interest) then ("--splits-of-interest " +  '"' + splits_of_interest + '"') else ""} \
      ~{if defined(collection_name) then ("--collection-name " +  '"' + collection_name + '"') else ""} \
      ~{if defined(bin_id) then ("--bin-id " +  '"' + bin_id + '"') else ""} \
      ~{if defined(samples_of_interest) then ("--samples-of-interest " +  '"' + samples_of_interest + '"') else ""} \
      ~{if defined(genes_of_interest) then ("--genes-of-interest " +  '"' + genes_of_interest + '"') else ""} \
      ~{if defined(gene_caller_ids) then ("--gene-caller-ids " +  '"' + gene_caller_ids + '"') else ""} \
      ~{if defined(min_departure_from_consensus) then ("--min-departure-from-consensus " +  '"' + min_departure_from_consensus + '"') else ""} \
      ~{true="--SAAVs-only" false="" sa_avs_only} \
      ~{true="--SCVs-only" false="" s_cvs_only} \
      ~{if defined(ip_address) then ("--ip-address " +  '"' + ip_address + '"') else ""} \
      ~{if defined(port_number) then ("--port-number " +  '"' + port_number + '"') else ""} \
      ~{if defined(browser_path) then ("--browser-path " +  '"' + browser_path + '"') else ""} \
      ~{true="--server-only" false="" server_only} \
      ~{true="--password-protected" false="" password_protected}
  >>>
  parameter_meta {
    structure_db: "Anvi'o structure database."
    profile_db: "Anvi'o profile database"
    contigs_db: "Anvi'o contigs database generated by 'anvi-gen- contigs'"
    variability_profile: "The output of anvi-gen-variability-profile, or a different variant-calling output that has been converted to the anvi'o format."
    splits_of_interest: "A file with split names. There should be only one column in the file, and each line should correspond to a unique split name."
    collection_name: "Collection name."
    bin_id: "If provided, any genes found in both your bin and your structure database will be available for display."
    samples_of_interest: "A file with samples names. There should be only one column in the file, and each line should correspond to a unique sample name (without a column header)."
    genes_of_interest: "A file with anvi'o gene caller IDs. There should be only one column in the file, and each line should correspond to a unique gene caller id (without a column header)."
    gene_caller_ids: "Gene caller ids. Multiple of them can be declared separated by a delimiter (the default is a comma). In anvi-gen-variability-profile, if you declare nothing you will get all genes matching your other filtering criteria. In other programs, you may get everything, nothing, or an error. It really depends on the situation. Fortunately, mistakes are cheap, so it's worth a try."
    min_departure_from_consensus: "Takes a value between 0 and 1, where 1 is maximum divergence from the consensus. it can be an expensive operation to display every variable position, and so the default is 0.05. To display every variable position, set this parameter to 0"
    sa_avs_only: "If provided, variability will be generated for single amino acid variants (SAAVs) and not for single codon variants (SCVs). This could save you some time if you're only interested in SAAVs."
    s_cvs_only: "If provided, variability will be generated for single codon variants (SCVs) and not for single amino acid variants (SAAVs). This could save you some time if you're only interested in SCVs."
    ip_address: "IP address for the HTTP server. The default ip address (0.0.0.0) should work just fine for most."
    port_number: "Port number to use for anvi'o services. If nothing is declared, anvi'o will try to find a suitable port number, starting from the default port number, 8080."
    browser_path: "By default, anvi'o will use your default browser to launch the interactive interface. If you would like to use something else than your system default, you can provide a full path for an alternative browser using this parameter, and hope for the best. For instance we are using this parameter to call Google's experimental browser, Canary, which performs better with demanding visualizations."
    server_only: "The default behavior is to start the local server, and fire up a browser that connects to the server. If you have other plans, and want to start the server without calling the browser, this is the flag you need."
    password_protected: "If this flag is set, command line tool will ask you to enter a password and interactive interface will be only accessible after entering same password. This option is recommended for shared machines like clusters or shared networks where computers are not isolated."
  }
}