#!/usr/bin/env cwl-runner

baseCommand:
- anvi-import-taxonomy-for-genes
class: CommandLineTool
cwlVersion: v1.0
id: anvi-import-taxonomy-for-genes
inputs:
- doc: Anvi'o contigs database generated by 'anvi-gen- contigs'
  id: contigs_db
  inputBinding:
    prefix: --contigs-db
  type: string
- doc: "Parser to make sense of the input files. There are 3 parsers readily available:\
    \ ['default_matrix', 'centrifuge', 'kaiju']. It is OK if you do not select a parser,\
    \ but in that case there will be no additional contigs available except the identification\
    \ of single- copy genes in your contigs for later use. Using a parser will not\
    \ prevent the analysis of single-copy genes, but make anvio more powerful to help\
    \ you make sense of your results. Please see the documentation, or get in touch\
    \ with the developers if you have any questions regarding parsers."
  id: parser
  inputBinding:
    prefix: --parser
  type: string
- doc: (S) [FILE(S) ...], --input-files FILE(S) [FILE(S) ...] Input file(s) for selected
    parser. Each parser (except "blank") requires input files to process that you
    generate before running anvio. Please see the documentation for details.
  id: i
  inputBinding:
    prefix: -i
  type: File
- doc: Don't bother me with questions or warnings, just do it.
  id: just_do_it
  inputBinding:
    prefix: --just-do-it
  type: boolean
