class: CommandLineTool
id: anvi_summarize.cwl
inputs:
- id: pan_or_profile_db
  doc: Anvi'o pan or profile database (and even genes database in appropriate contexts).
  type: string
  inputBinding:
    prefix: --pan-or-profile-db
- id: contigs_db
  doc: Anvi'o contigs database generated by 'anvi-gen- contigs'
  type: string
  inputBinding:
    prefix: --contigs-db
- id: genomes_storage
  doc: Anvi'o genomes storage file
  type: string
  inputBinding:
    prefix: --genomes-storage
- id: in_it_gene_coverages
  doc: Initialize gene coverage and detection data. This is a very computationally
    expensive step, but it is necessary when you need gene level coverage data. The
    reason this is very computationally expensive is because anvi'o computes gene
    coverages by going back to actual coverage values of each gene to average them,
    instead of using contig average coverage values, for extreme accuracy.
  type: boolean
  inputBinding:
    prefix: --init-gene-coverages
- id: reformat_contig_names
  doc: Reformat contig names while generating the summary output so they look fancy.
    With this flag, anvi'o will replace the original names of contigs to those that
    include the bin name as a prefix in resulting summary output files per bin. Use
    this flag carefully as it may influence your downstream analyses due to the fact
    that your original contig names in your input FASTA file for the contigs database
    will not be in the summary output. Although, anvi'o will report a conversion map
    per bin so you can recover the original contig name if you have to.
  type: boolean
  inputBinding:
    prefix: --reformat-contig-names
- id: report_aa_seqs_for_gene_calls
  doc: You can use this flag if you would like amino acid AND dna sequences for your
    gene calls in the genes output file. By default, only dna sequences are reported.
  type: boolean
  inputBinding:
    prefix: --report-aa-seqs-for-gene-calls
- id: report_dna_sequences
  doc: By default, this program reports amino acid sequences. Use this flag to report
    DNA sequences instead. Also note, since gene clusters are aligned via amino acid
    sequences, using this flag removes alignment information manifesting in the form
    of gap characters (`-` characters) that would be present if amino acid sequences
    were reported. Read the warnings during runtime for more detailed information.
  type: boolean
  inputBinding:
    prefix: --report-DNA-sequences
- id: collection_name
  doc: Collection name.
  type: string
  inputBinding:
    prefix: --collection-name
- id: output_dir
  doc: Directory path for output files
  type: string
  inputBinding:
    prefix: --output-dir
- id: list_collections
  doc: Show available collections and exit.
  type: boolean
  inputBinding:
    prefix: --list-collections
- id: taxonomic_level
  doc: The taxonomic level to use whenever relevant and/or available. The default
    taxonomic level is t_genus, but if you choose something specific, anvi'o will
    focus on that whenever possible.
  type: string
  inputBinding:
    prefix: --taxonomic-level
- id: cog_data_dir
  doc: The directory path for your COG setup. Anvi'o will try to use the default path
    if you do not specify anything.
  type: string
  inputBinding:
    prefix: --cog-data-dir
- id: quick_summary
  doc: When declared the summary output will be generated as quickly as possible,
    with minimum amount of essential information about bins.
  type: boolean
  inputBinding:
    prefix: --quick-summary
- id: just_do_it
  doc: Don't bother me with questions or warnings, just do it.
  type: boolean
  inputBinding:
    prefix: --just-do-it
outputs: []
cwlVersion: v1.1
baseCommand:
- anvi-summarize
