#!/usr/bin/env cwl-runner

baseCommand:
- anvi-gen-gene-level-stats-databases
class: CommandLineTool
cwlVersion: v1.0
id: anvi-gen-gene-level-stats-databases
inputs:
- doc: Anvi'o contigs database generated by 'anvi-gen- contigs'
  id: contigs_db
  inputBinding:
    prefix: --contigs-db
  type: string
- doc: Anvi'o profile database
  id: profile_db
  inputBinding:
    prefix: --profile-db
  type: string
- doc: Collection name.
  id: collection_name
  inputBinding:
    prefix: --collection-name
  type: string
- doc: Bin name you are interested in.
  id: bin_id
  inputBinding:
    prefix: --bin-id
  type: string
- doc: Text file for bins (each line should be a unique bin id).
  id: bin_ids_file
  inputBinding:
    prefix: --bin-ids-file
  type: File
- doc: If you want all zero coverage positions to be treated like outliers then use
    this flag. The reason to treat zero coverage as outliers is because when mapping
    reads to a reference we could get many zero positions due to accessory genes.
    These positions then skew the average values that we compute.
  id: zeros_are_outliers
  inputBinding:
    prefix: --zeros-are-outliers
  type: boolean
- doc: "Threshold to use for the outlier detection. The default value is '1.5'. Absolute\
    \ deviation around the median is used. To read more about the method please refer\
    \ to: 'How to Detect and Handle Outliers' by Boris Iglewicz and David Hoaglin\
    \ (doi:10.1016/j.jesp.2013.03.013)."
  id: outliers_threshold
  inputBinding:
    prefix: --outliers-threshold
  type: string
- doc: Don't bother me with questions or warnings, just do it.
  id: just_do_it
  inputBinding:
    prefix: --just-do-it
  type: boolean
- doc: Provide if working with INSeq/Tn-Seq genomic data. With this, all gene level
    coverage stats will be calculated using INSeq/Tn-Seq statistical methods.
  id: in_seq_stats
  inputBinding:
    prefix: --inseq-stats
  type: boolean
