version 1.0

task AnviscriptpredictCPRgenomes {
  input {
    String? contigs_db
    String? profile_db
    String? collection_name
    Boolean? list_collections
    Boolean? report_only_cpr
    Int? min_genome_size
    Int? min_percent_completion
    Int? max_percent_redundancy
    Int? min_class_probability
    File? output_file
    Boolean? just_do_it
    String classifier
  }
  command <<<
    anvi_script_predict_CPR_genomes \
      ~{classifier} \
      ~{if defined(contigs_db) then ("--contigs-db " +  '"' + contigs_db + '"') else ""} \
      ~{if defined(profile_db) then ("--profile-db " +  '"' + profile_db + '"') else ""} \
      ~{if defined(collection_name) then ("--collection-name " +  '"' + collection_name + '"') else ""} \
      ~{if (list_collections) then "--list-collections" else ""} \
      ~{if (report_only_cpr) then "--report-only-cpr" else ""} \
      ~{if defined(min_genome_size) then ("--min-genome-size " +  '"' + min_genome_size + '"') else ""} \
      ~{if defined(min_percent_completion) then ("--min-percent-completion " +  '"' + min_percent_completion + '"') else ""} \
      ~{if defined(max_percent_redundancy) then ("--max-percent-redundancy " +  '"' + max_percent_redundancy + '"') else ""} \
      ~{if defined(min_class_probability) then ("--min-class-probability " +  '"' + min_class_probability + '"') else ""} \
      ~{if defined(output_file) then ("--output-file " +  '"' + output_file + '"') else ""} \
      ~{if (just_do_it) then "--just-do-it" else ""}
  >>>
  parameter_meta {
    contigs_db: "Anvi'o contigs database generated by 'anvi-gen-\\ncontigs'"
    profile_db: "Anvi'o profile database"
    collection_name: "Collection name."
    list_collections: "Show available collections and exit."
    report_only_cpr: "Include only bins that look like CPR genomes."
    min_genome_size: "Minimum genome size to consider for CPR in Mbp.\\nDefault is 0.500000"
    min_percent_completion: "Minimum percent completion estimate based on anvi'o\\ndefault single-copy gene collections. Default is 50"
    max_percent_redundancy: "Maxumum percent redundancy or single-copy genes in an\\nanvi'o bin, or a genome to consider for\\nclassification. The default is 30"
    min_class_probability: "If the classification confidence is below this don't\\nbother. Default is 75."
    output_file: "File path to store results."
    just_do_it: "Don't bother me with questions or warnings, just do\\nit.\\n"
    classifier: "optional arguments:"
  }
  output {
    File out_stdout = stdout()
    File out_output_file = "${in_output_file}"
  }
}