#!/usr/bin/env cwl-runner

baseCommand:
- anvi-summarize
class: CommandLineTool
cwlVersion: v1.0
id: anvi-summarize
inputs:
- doc: Anvi'o pan or profile database (and even genes database in appropriate contexts).
  id: pan_or_profile_db
  inputBinding:
    prefix: --pan-or-profile-db
  type: string
- doc: Anvi'o contigs database generated by 'anvi-gen- contigs'
  id: contigs_db
  inputBinding:
    prefix: --contigs-db
  type: string
- doc: Anvi'o genomes storage file
  id: genomes_storage
  inputBinding:
    prefix: --genomes-storage
  type: string
- doc: Initialize gene coverage and detection data. This is a very computationally
    expensive step, but it is necessary when you need gene level coverage data. The
    reason this is very computationally expensive is because anvi'o computes gene
    coverages by going back to actual coverage values of each gene to average them,
    instead of using contig average coverage values, for extreme accuracy.
  id: in_it_gene_coverages
  inputBinding:
    prefix: --init-gene-coverages
  type: boolean
- doc: Reformat contig names while generating the summary output so they look fancy.
    With this flag, anvi'o will replace the original names of contigs to those that
    include the bin name as a prefix in resulting summary output files per bin. Use
    this flag carefully as it may influence your downstream analyses due to the fact
    that your original contig names in your input FASTA file for the contigs database
    will not be in the summary output. Although, anvi'o will report a conversion map
    per bin so you can recover the original contig name if you have to.
  id: reformat_contig_names
  inputBinding:
    prefix: --reformat-contig-names
  type: boolean
- doc: You can use this flag if you would like amino acid AND dna sequences for your
    gene calls in the genes output file. By default, only dna sequences are reported.
  id: report_aa_seqs_for_gene_calls
  inputBinding:
    prefix: --report-aa-seqs-for-gene-calls
  type: boolean
- doc: By default, this program reports amino acid sequences. Use this flag to report
    DNA sequences instead. Also note, since gene clusters are aligned via amino acid
    sequences, using this flag removes alignment information manifesting in the form
    of gap characters (`-` characters) that would be present if amino acid sequences
    were reported. Read the warnings during runtime for more detailed information.
  id: report_dna_sequences
  inputBinding:
    prefix: --report-DNA-sequences
  type: boolean
- doc: Collection name.
  id: collection_name
  inputBinding:
    prefix: --collection-name
  type: string
- doc: Directory path for output files
  id: output_dir
  inputBinding:
    prefix: --output-dir
  type: string
- doc: Show available collections and exit.
  id: list_collections
  inputBinding:
    prefix: --list-collections
  type: boolean
- doc: The taxonomic level to use whenever relevant and/or available. The default
    taxonomic level is t_genus, but if you choose something specific, anvi'o will
    focus on that whenever possible.
  id: taxonomic_level
  inputBinding:
    prefix: --taxonomic-level
  type: string
- doc: The directory path for your COG setup. Anvi'o will try to use the default path
    if you do not specify anything.
  id: cog_data_dir
  inputBinding:
    prefix: --cog-data-dir
  type: string
- doc: When declared the summary output will be generated as quickly as possible,
    with minimum amount of essential information about bins.
  id: quick_summary
  inputBinding:
    prefix: --quick-summary
  type: boolean
- doc: Don't bother me with questions or warnings, just do it.
  id: just_do_it
  inputBinding:
    prefix: --just-do-it
  type: boolean
