version 1.0

task AnviGenGeneLevelStatsDatabases {
  input {
    String? contigs_db
    String? profile_db
    String? collection_name
    String? bin_id
    File? bin_ids_file
    Boolean? zeros_are_outliers
    String? outliers_threshold
    Boolean? just_do_it
    Boolean? in_seq_stats
  }
  command <<<
    anvi-gen-gene-level-stats-databases \
      ~{if defined(contigs_db) then ("--contigs-db " +  '"' + contigs_db + '"') else ""} \
      ~{if defined(profile_db) then ("--profile-db " +  '"' + profile_db + '"') else ""} \
      ~{if defined(collection_name) then ("--collection-name " +  '"' + collection_name + '"') else ""} \
      ~{if defined(bin_id) then ("--bin-id " +  '"' + bin_id + '"') else ""} \
      ~{if defined(bin_ids_file) then ("--bin-ids-file " +  '"' + bin_ids_file + '"') else ""} \
      ~{true="--zeros-are-outliers" false="" zeros_are_outliers} \
      ~{if defined(outliers_threshold) then ("--outliers-threshold " +  '"' + outliers_threshold + '"') else ""} \
      ~{true="--just-do-it" false="" just_do_it} \
      ~{true="--inseq-stats" false="" in_seq_stats}
  >>>
  parameter_meta {
    contigs_db: "Anvi'o contigs database generated by 'anvi-gen- contigs'"
    profile_db: "Anvi'o profile database"
    collection_name: "Collection name."
    bin_id: "Bin name you are interested in."
    bin_ids_file: "Text file for bins (each line should be a unique bin id)."
    zeros_are_outliers: "If you want all zero coverage positions to be treated like outliers then use this flag. The reason to treat zero coverage as outliers is because when mapping reads to a reference we could get many zero positions due to accessory genes. These positions then skew the average values that we compute."
    outliers_threshold: "Threshold to use for the outlier detection. The default value is '1.5'. Absolute deviation around the median is used. To read more about the method please refer to: 'How to Detect and Handle Outliers' by Boris Iglewicz and David Hoaglin (doi:10.1016/j.jesp.2013.03.013)."
    just_do_it: "Don't bother me with questions or warnings, just do it."
    in_seq_stats: "Provide if working with INSeq/Tn-Seq genomic data. With this, all gene level coverage stats will be calculated using INSeq/Tn-Seq statistical methods."
  }
}