#!/usr/bin/env cwl-runner

baseCommand:
- anvi-refine
class: CommandLineTool
cwlVersion: v1.0
id: anvi-refine
inputs:
- doc: Anvi'o profile database
  id: profile_db
  inputBinding:
    prefix: --profile-db
  type: string
- doc: Anvi'o contigs database generated by 'anvi-gen- contigs'
  id: contigs_db
  inputBinding:
    prefix: --contigs-db
  type: string
- doc: Collection name.
  id: collection_name
  inputBinding:
    prefix: --collection-name
  type: string
- doc: Bin name you are interested in.
  id: bin_id
  inputBinding:
    prefix: --bin-id
  type: string
- doc: Text file for bins (each line should be a unique bin id).
  id: bin_ids_file
  inputBinding:
    prefix: --bin-ids-file
  type: File
- doc: If you don't know the bin name you want to work with but if you know the split
    name it contains you can use this parameter to tell anvi'o the split name, and
    so it can find the bin for you automatically. This is something extremely difficult
    for anvi'o to do, but it does it anyway because you.
  id: find_from_split_name
  inputBinding:
    prefix: --find-from-split-name
  type: string
- doc: NEWICK formatted tree structure
  id: tree
  inputBinding:
    prefix: --tree
  type: string
- doc: Skip hierarchical clustering for the splits in the refined bin, if you skip
    clustering you need to provide your own newick formatted tree using --tree parameter.
  id: skip_hierarchical_clustering
  inputBinding:
    prefix: --skip-hierarchical-clustering
  type: boolean
- doc: Often the minimum and maximum values defined for the an entire profile database
    that contains all contigs do not scale well when you wish to work with a single
    bin in the refine mode. For this reason, the default behavior of anvi-refine is
    to ignore min/max values set in the default state. This flag is your way of telling
    anvi'o to not do that, and load the state stored in the profile database as is.
  id: load_full_state
  inputBinding:
    prefix: --load-full-state
  type: boolean
- doc: A TAB-delimited file for an additional view to be used in the interface. This
    file should contain all split names, and values for each of them in all samples.
    Each column in this file must correspond to a sample name. Content of this file
    will be called 'user_view', which will be available as a new item in the 'views'
    combo box in the interface
  id: additional_view
  inputBinding:
    prefix: --additional-view
  type: string
- doc: A TAB-delimited file for additional layers for splits. The first column of
    this file must be split names, and the remaining columns should be unique attributes.
    The file does not need to contain all split names, or values for each split in
    every column. Anvi'o will try to deal with missing data nicely. Each column in
    this file will be visualized as a new layer in the tree.
  id: additional_layers
  inputBinding:
    prefix: --additional-layers
  type: string
- doc: When declared, this flag tells the interface to split every gene found in HMM
    searches that were performed against non-singlecopy gene HMM profiles into their
    own layer. Please see the documentation for details.
  id: split_hmm_layers
  inputBinding:
    prefix: --split-hmm-layers
  type: boolean
- doc: The taxonomic level to use whenever relevant and/or available. The default
    taxonomic level is t_genus, but if you choose something specific, anvi'o will
    focus on that whenever possible.
  id: taxonomic_level
  inputBinding:
    prefix: --taxonomic-level
  type: string
- doc: "During profiling, anvi'o marks positions of single- nucleotide variations\
    \ (SNVs) that originate from places in contigs where coverage values are a bit\
    \ 'sketchy'. If you would like to avoid SNVs in those positions of splits in applicable\
    \ projects you can use this flag, and the interface would hide SNVs that are marked\
    \ as 'outlier' (although it is clearly the best to see everything, no one will\
    \ judge you if you end up using this flag) (plus, there may or may not be some\
    \ historical data on this here: https://github.com/meren/anvio/issues/309)."
  id: hide_outlier_sn_vs
  inputBinding:
    prefix: --hide-outlier-SNVs
  type: boolean
- doc: Title for the interface. If you are working with a RUNINFO dict, the title
    will be determined based on information stored in that file. Regardless, you can
    override that value using this parameter.
  id: title
  inputBinding:
    prefix: --title
  type: string
- doc: The SVG output file path.
  id: export_svg
  inputBinding:
    prefix: --export-svg
  type: File
- doc: Don't do anything real. Test everything, and stop right before wherever the
    developer said 'well, this is enough testing', and decided to print out results.
  id: dry_run
  inputBinding:
    prefix: --dry-run
  type: boolean
- doc: When declared, function calls for genes will not be initialized (therefore
    will be missing from all relevant interfaces or output files). The use of this
    flag may reduce the memory fingerprint and processing time for large datasets.
  id: skip_in_it_functions
  inputBinding:
    prefix: --skip-init-functions
  type: boolean
- doc: When declared, the attempt to include automatically generated orders of items
    based on additional data is skipped. In case those buggers cause issues with your
    data, and you still want to see your stuff and deal with the other issue maybe
    later.
  id: skip_auto_ordering
  inputBinding:
    prefix: --skip-auto-ordering
  type: boolean
- doc: IP address for the HTTP server. The default ip address (0.0.0.0) should work
    just fine for most.
  id: ip_address
  inputBinding:
    prefix: --ip-address
  type: string
- doc: Port number to use for anvi'o services. If nothing is declared, anvi'o will
    try to find a suitable port number, starting from the default port number, 8080.
  id: port_number
  inputBinding:
    prefix: --port-number
  type: long
- doc: By default, anvi'o will use your default browser to launch the interactive
    interface. If you would like to use something else than your system default, you
    can provide a full path for an alternative browser using this parameter, and hope
    for the best. For instance we are using this parameter to call Google's experimental
    browser, Canary, which performs better with demanding visualizations.
  id: browser_path
  inputBinding:
    prefix: --browser-path
  type: File
- doc: When the interactive interface is started with this flag, all 'database write'
    operations will be disabled.
  id: read_only
  inputBinding:
    prefix: --read-only
  type: boolean
- doc: The default behavior is to start the local server, and fire up a browser that
    connects to the server. If you have other plans, and want to start the server
    without calling the browser, this is the flag you need.
  id: server_only
  inputBinding:
    prefix: --server-only
  type: boolean
- doc: If this flag is set, command line tool will ask you to enter a password and
    interactive interface will be only accessible after entering same password. This
    option is recommended for shared machines like clusters or shared networks where
    computers are not isolated.
  id: password_protected
  inputBinding:
    prefix: --password-protected
  type: boolean
