#!/usr/bin/env cwl-runner

baseCommand:
- anvi-get-short-reads-mapping-to-a-gene
class: CommandLineTool
cwlVersion: v1.0
id: anvi-get-short-reads-mapping-to-a-gene
inputs:
- doc: Anvi'o contigs database generated by 'anvi-gen- contigs'
  id: contigs_db
  inputBinding:
    prefix: --contigs-db
  type: string
- doc: (S) [INPUT_BAM(S) ...], --input-files INPUT_BAM(S) [INPUT_BAM(S) ...] Sorted
    and indexed BAM files to analyze. It is essential that all BAM files must be the
    result of mappings against the same contigs.
  id: i
  inputBinding:
    prefix: -i
  type: string
- doc: A single gene id.
  id: gene_caller_id
  inputBinding:
    prefix: --gene-caller-id
  type: string
- doc: A file with anvi'o gene caller IDs. There should be only one column in the
    file, and each line should correspond to a unique gene caller id (without a column
    header).
  id: genes_of_interest
  inputBinding:
    prefix: --genes-of-interest
  type: File
- doc: The minimum number of nucleotides for a given short read mapping into the gene
    context for it to be reported. You must consider the length of your short reads,
    as well as the length of the gene you are targeting. The default is 100 nts.
  id: leeway
  inputBinding:
    prefix: --leeway
  type: string
- doc: A prefix to be used while naming the output files (no file type extensions
    please; just a prefix).
  id: output_file_prefix
  inputBinding:
    prefix: --output-file-prefix
  type: File
