version 1.0

task AnviGetSequencesForGeneCalls {
  input {
    String? contigs_db
    String? gene_caller_ids
    String? delimiter
    Boolean? report_extended_def_lines
    String? wrap
    Boolean? export_gff_three
    Boolean? get_aa_sequences
    String? genomes_storage
    String? genome_names
    File? output_file
  }
  command <<<
    anvi-get-sequences-for-gene-calls \
      ~{if defined(contigs_db) then ("--contigs-db " +  '"' + contigs_db + '"') else ""} \
      ~{if defined(gene_caller_ids) then ("--gene-caller-ids " +  '"' + gene_caller_ids + '"') else ""} \
      ~{if defined(delimiter) then ("--delimiter " +  '"' + delimiter + '"') else ""} \
      ~{true="--report-extended-deflines" false="" report_extended_def_lines} \
      ~{if defined(wrap) then ("--wrap " +  '"' + wrap + '"') else ""} \
      ~{true="--export-gff3" false="" export_gff_three} \
      ~{true="--get-aa-sequences" false="" get_aa_sequences} \
      ~{if defined(genomes_storage) then ("--genomes-storage " +  '"' + genomes_storage + '"') else ""} \
      ~{if defined(genome_names) then ("--genome-names " +  '"' + genome_names + '"') else ""} \
      ~{if defined(output_file) then ("--output-file " +  '"' + output_file + '"') else ""}
  >>>
  parameter_meta {
    contigs_db: "Anvi'o contigs database generated by 'anvi-gen- contigs'"
    gene_caller_ids: "Gene caller ids. Multiple of them can be declared separated by a delimiter (the default is a comma). In anvi-gen-variability-profile, if you declare nothing you will get all genes matching your other filtering criteria. In other programs, you may get everything, nothing, or an error. It really depends on the situation. Fortunately, mistakes are cheap, so it's worth a try."
    delimiter: "The delimiter to parse multiple input terms. The default is ','."
    report_extended_def_lines: "When declared, the deflines in the resulting FASTA file will contain more information."
    wrap: "When to wrap sequences when storing them in a FASTA file. The default is '120'. A value of '0' would be equivalent to 'do not wrap'."
    export_gff_three: "If this is true, the output file will be in GFF3 format."
    get_aa_sequences: "Store amino acid sequences instead."
    genomes_storage: "Anvi'o genomes storage file"
    genome_names: "Genome names to 'focus'. You can use this parameter to limit the genomes included in your analysis. You can provide these names as a comma-separated list of names, or you can put them in a file, where you have a single genome name in each line, and provide the file path."
    output_file: "File path to store results."
  }
}