class: CommandLineTool
id: ../../../anvi_import_taxonomy_for_genes.cwl
inputs:
- id: contigs_db
  doc: Anvi'o contigs database generated by 'anvi-gen- contigs'
  type: string
  inputBinding:
    prefix: --contigs-db
- id: parser
  doc: "Parser to make sense of the input files. There are 3 parsers readily available:\
    \ ['default_matrix', 'centrifuge', 'kaiju']. It is OK if you do not select a parser,\
    \ but in that case there will be no additional contigs available except the identification\
    \ of single- copy genes in your contigs for later use. Using a parser will not\
    \ prevent the analysis of single-copy genes, but make anvio more powerful to help\
    \ you make sense of your results. Please see the documentation, or get in touch\
    \ with the developers if you have any questions regarding parsers."
  type: string
  inputBinding:
    prefix: --parser
- id: s_input_files
  doc: (S) [FILE(S) ...], --input-files FILE(S) [FILE(S) ...] Input file(s) for selected
    parser. Each parser (except "blank") requires input files to process that you
    generate before running anvio. Please see the documentation for details.
  type: File
  inputBinding:
    prefix: -i
- id: just_do_it
  doc: Don't bother me with questions or warnings, just do it.
  type: boolean
  inputBinding:
    prefix: --just-do-it
outputs: []
cwlVersion: v1.1
baseCommand:
- anvi-import-taxonomy-for-genes
