class: CommandLineTool
id: anvi_export_gene_calls.cwl
inputs:
- id: contigs_db
  doc: Anvi'o contigs database generated by 'anvi-gen- contigs'
  type: string
  inputBinding:
    prefix: --contigs-db
- id: output_file
  doc: File path to store results.
  type: File
  inputBinding:
    prefix: --output-file
- id: gene_caller
  doc: Which gene caller(s) would you like to export gene calls for? If providing
    multiple they should be comma- separated (no spaces). If you don't know, use --list-
    gene-callers
  type: string
  inputBinding:
    prefix: --gene-caller
- id: list_gene_callers
  doc: List available gene callers in the contigs database and quit.
  type: boolean
  inputBinding:
    prefix: --list-gene-callers
- id: skip_sequence_reporting
  doc: By default, exported gene calls have an amino acid sequences column in the
    output. Turn this behavior off with this flag
  type: boolean
  inputBinding:
    prefix: --skip-sequence-reporting
outputs: []
cwlVersion: v1.1
baseCommand:
- anvi-export-gene-calls
