version 1.0

task AnviEstimateScgTaxonomy {
  input {
    String? contigs_db
    Boolean? meta_genome_mode
    String? profile_db
    String? collection_name
    File? meta_genomes
    File? output_file
    File? output_file_prefix
    String? taxonomic_level
    Boolean? matrix_format
    Boolean? raw_output
    String? num_threads
    String? scg_name_for_meta_genome_mode
    File? report_scg_frequencies
    Boolean? just_do_it
    Boolean? simplify_taxonomy_information
    Boolean? compute_scg_coverages
    Boolean? update_profile_db_with_taxonomy
    File? taxonomy_database
  }
  command <<<
    anvi-estimate-scg-taxonomy \
      ~{if defined(contigs_db) then ("--contigs-db " +  '"' + contigs_db + '"') else ""} \
      ~{true="--metagenome-mode" false="" meta_genome_mode} \
      ~{if defined(profile_db) then ("--profile-db " +  '"' + profile_db + '"') else ""} \
      ~{if defined(collection_name) then ("--collection-name " +  '"' + collection_name + '"') else ""} \
      ~{if defined(meta_genomes) then ("--metagenomes " +  '"' + meta_genomes + '"') else ""} \
      ~{if defined(output_file) then ("--output-file " +  '"' + output_file + '"') else ""} \
      ~{if defined(output_file_prefix) then ("--output-file-prefix " +  '"' + output_file_prefix + '"') else ""} \
      ~{if defined(taxonomic_level) then ("--taxonomic-level " +  '"' + taxonomic_level + '"') else ""} \
      ~{true="--matrix-format" false="" matrix_format} \
      ~{true="--raw-output" false="" raw_output} \
      ~{if defined(num_threads) then ("--num-threads " +  '"' + num_threads + '"') else ""} \
      ~{if defined(scg_name_for_meta_genome_mode) then ("--scg-name-for-metagenome-mode " +  '"' + scg_name_for_meta_genome_mode + '"') else ""} \
      ~{if defined(report_scg_frequencies) then ("--report-scg-frequencies " +  '"' + report_scg_frequencies + '"') else ""} \
      ~{true="--just-do-it" false="" just_do_it} \
      ~{true="--simplify-taxonomy-information" false="" simplify_taxonomy_information} \
      ~{true="--compute-scg-coverages" false="" compute_scg_coverages} \
      ~{true="--update-profile-db-with-taxonomy" false="" update_profile_db_with_taxonomy} \
      ~{if defined(taxonomy_database) then ("--taxonomy-database " +  '"' + taxonomy_database + '"') else ""}
  >>>
  parameter_meta {
    contigs_db: "Anvi'o contigs database generated by 'anvi-gen- contigs'"
    meta_genome_mode: "Treat a given contigs database as a metagenome rather than treating it as a single genome."
    profile_db: "Anvi'o profile database"
    collection_name: "Collection name."
    meta_genomes: "A two-column TAB-delimited flat text file. The header line must contain these columns: 'name', 'contigs_db_path', and 'profile_db_path'. Each line should list a single entry, where 'name' can be any name to describe the metagenome stored in the anvi'o contigs database. In this context, the anvi'o profiles associated with contigs database must be SINGLE PROFILES, as in generated by the program `anvi- profile` and not `anvi-merge`."
    output_file: "File path to store results."
    output_file_prefix: "A prefix to be used while naming the output files (no file type extensions please; just a prefix)."
    taxonomic_level: "The taxonomic level to use whenever relevant and/or available. The default taxonomic level is None, but if you choose something specific, anvi'o will focus on that whenever possible."
    matrix_format: "If you want the reports to look like sparse matrices whenever possible, declare this flag. Matrices are especially good to use when you are working with internal/external genomes since they can show you quickly the distribution of each taxon across all metagenomes in programs like EXCEL. WELL TRY IT AND SEE."
    raw_output: "Just store the raw output without any processing of the primary data structure."
    num_threads: "Maximum number of threads to use for multithreading whenever possible. Very conservatively, the default is 1. It is a good idea to not exceed the number of CPUs / cores on your system. Plus, please be careful with this option if you are running your commands on a SGE --if you are clusterizing your runs, and asking for multiple threads to use, you may deplete your resources very fast."
    scg_name_for_meta_genome_mode: "When running in metagenome mode, anvi'o automatically chooses the most frequent single-copy core gene to estimate the taxonomic composition within a contigs database. If you have a different preference you can use this parameter to communicate that."
    report_scg_frequencies: "Report SCG frequencies in a TAB-delimited file and quit. This is a great way to decide which SCG name to use in metagenome mode (we often wish to use the most frequent SCG to increase the detection of taxa)."
    just_do_it: "Don't bother me with questions or warnings, just do it."
    simplify_taxonomy_information: "The taxonomy output may include a large number of names that contain clade-specific code for not-yet- characterized taxa. With this flag you can simplify taxon names. This will influence all output files and displays as the use of this flag will on-the-fly trim taxonomic levels with clade-specific code names."
    compute_scg_coverages: "When this flag is declared, anvi'o will go back to the profile database to learn coverage statistics of single-copy core genes for which we have taxonomy information."
    update_profile_db_with_taxonomy: "When anvi'o knows all both taxonomic affiliations and coverages across samples for single-copy core genes, it can, in theory add this information to the profile database. With this flag you can instruct anvi'o to do that and find information on taxonomy in the `layers` tab of your interactive interface."
    taxonomy_database: "Path to the directory that contains the BLAST databases for single-copy core genes. You will almost never need to use this parameter unless you are trying something very fancy. But when you do, you can tell anvi'o where to look for database files through this parameter."
  }
}