from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, String, Boolean, Int, Directory

Fastq_Dump_2_10_9_V0_1_0 = CommandToolBuilder(tool="fastq_dump.2.10.9", base_command=["fastq-dump.2.10.9"], inputs=[ToolInput(tag="in_accession", input_type=File(optional=True), prefix="--accession", doc=InputDocumentation(doc="Replaces accession derived from <path> in\nfilename(s) and deflines (only for\nsingle table dump)")), ToolInput(tag="in_table", input_type=String(optional=True), prefix="--table", doc=InputDocumentation(doc="Table name within cSRA object, default is\n'SEQUENCE'")), ToolInput(tag="in_split_spot", input_type=Boolean(optional=True), prefix="--split-spot", doc=InputDocumentation(doc="Split spots into individual reads")), ToolInput(tag="in_min_spot_id", input_type=String(optional=True), prefix="--minSpotId", doc=InputDocumentation(doc="Minimum spot id")), ToolInput(tag="in_max_spot_id", input_type=String(optional=True), prefix="--maxSpotId", doc=InputDocumentation(doc="Maximum spot id")), ToolInput(tag="in_spot_groups", input_type=Boolean(optional=True), prefix="--spot-groups", doc=InputDocumentation(doc="<[list]>[,...]  Filter by SPOT_GROUP (member): name[,...]")), ToolInput(tag="in_clip", input_type=Boolean(optional=True), prefix="--clip", doc=InputDocumentation(doc="Remove adapter sequences from reads")), ToolInput(tag="in_min_read_len", input_type=Int(optional=True), prefix="--minReadLen", doc=InputDocumentation(doc="Filter by sequence length >= <len>")), ToolInput(tag="in_read_filter", input_type=String(optional=True), prefix="--read-filter", doc=InputDocumentation(doc="Split into files by READ_FILTER value\n[split], optionally filter by value:\n[pass|reject|criteria|redacted]")), ToolInput(tag="in_qual_filter", input_type=Boolean(optional=True), prefix="--qual-filter", doc=InputDocumentation(doc="Filter used in early 1000 Genomes data: no\nsequences starting or ending with >= 10N")), ToolInput(tag="in_qual_filter_one", input_type=Boolean(optional=True), prefix="--qual-filter-1", doc=InputDocumentation(doc="Filter used in current 1000 Genomes data")), ToolInput(tag="in_aligned", input_type=Boolean(optional=True), prefix="--aligned", doc=InputDocumentation(doc="Dump only aligned sequences")), ToolInput(tag="in_unaligned", input_type=Boolean(optional=True), prefix="--unaligned", doc=InputDocumentation(doc="Dump only unaligned sequences")), ToolInput(tag="in_aligned_region", input_type=Boolean(optional=True), prefix="--aligned-region", doc=InputDocumentation(doc="<name[:from-to]>\nFilter by position on genome. Name can\neiter by accession.version (ex:\nNC_000001.10) or file specific name (ex:\n'chr1' or '1'. 'from' and 'to' are\n1-based coordinates")), ToolInput(tag="in_mate_pair_distance", input_type=String(optional=True), prefix="--matepair_distance", doc=InputDocumentation(doc="Filter by distance between matepairs. Use\n'unknown' to find matepairs split\nbetween the references. Use from-to to\nlimit matepair distance on the same\nreference")), ToolInput(tag="in_skip_technical", input_type=Boolean(optional=True), prefix="--skip-technical", doc=InputDocumentation(doc="Dump only biological reads")), ToolInput(tag="in_outdir", input_type=Directory(optional=True), prefix="--outdir", doc=InputDocumentation(doc="Output directory, default is working\ndirectory '.'")), ToolInput(tag="in_stdout", input_type=Boolean(optional=True), prefix="--stdout", doc=InputDocumentation(doc="Output to stdout, all split data become\njoined into single stream")), ToolInput(tag="in_gzip", input_type=Boolean(optional=True), prefix="--gzip", doc=InputDocumentation(doc="Compress output using gzip: deprecated,\nnot recommended")), ToolInput(tag="in_bzip_two", input_type=Boolean(optional=True), prefix="--bzip2", doc=InputDocumentation(doc="Compress output using bzip2: deprecated,\nnot recommended")), ToolInput(tag="in_split_files", input_type=Boolean(optional=True), prefix="--split-files", doc=InputDocumentation(doc="Write reads into separate files. Read\nnumber will be suffixed to the file\nname. NOTE! The `--split-3` option is\nrecommended. In cases where not all\nspots have the same number of reads,\nthis option will produce files that WILL\nCAUSE ERRORS in most programs which\nprocess split pair fastq files.")), ToolInput(tag="in_split_three", input_type=Boolean(optional=True), prefix="--split-3", doc=InputDocumentation(doc="3-way splitting for mate-pairs. For each\nspot, if there are two biological reads\nsatisfying filter conditions, the first\nis placed in the `*_1.fastq` file, and\nthe second is placed in the `*_2.fastq`\nfile. If there is only one biological\nread satisfying the filter conditions,\nit is placed in the `*.fastq` file.All\nother reads in the spot are ignored.")), ToolInput(tag="in_spot_group", input_type=Boolean(optional=True), prefix="--spot-group", doc=InputDocumentation(doc="Split into files by SPOT_GROUP (member\nname)")), ToolInput(tag="in_group_in_dirs", input_type=Boolean(optional=True), prefix="--group-in-dirs", doc=InputDocumentation(doc="Split into subdirectories instead of files")), ToolInput(tag="in_keep_empty_files", input_type=Boolean(optional=True), prefix="--keep-empty-files", doc=InputDocumentation(doc="Do not delete empty files")), ToolInput(tag="in_dump_cs", input_type=String(optional=True), prefix="--dumpcs", doc=InputDocumentation(doc="Formats sequence using color space\n(default for SOLiD), 'cskey' may be\nspecified for translation or else\nspecify 'dflt' to use the default value")), ToolInput(tag="in_dump_base", input_type=Boolean(optional=True), prefix="--dumpbase", doc=InputDocumentation(doc="Formats sequence using base space (default\nfor other than SOLiD).")), ToolInput(tag="in_offset", input_type=Boolean(optional=True), prefix="--offset", doc=InputDocumentation(doc="<integer             Offset to use for quality conversion,\ndefault is 33")), ToolInput(tag="in_fast_a", input_type=Int(optional=True), prefix="--fasta", doc=InputDocumentation(doc="FASTA only, no qualities, with can be\n'default' or '0' for no wrapping")), ToolInput(tag="in_suppress_qual_for_cs_key", input_type=Boolean(optional=True), prefix="--suppress-qual-for-cskey", doc=InputDocumentation(doc="suppress quality-value for cskey")), ToolInput(tag="in_orig_fmt", input_type=Boolean(optional=True), prefix="--origfmt", doc=InputDocumentation(doc="Defline contains only original sequence")), ToolInput(tag="in_h_elicos", input_type=Boolean(optional=True), prefix="--helicos", doc=InputDocumentation(doc="Helicos style defline")), ToolInput(tag="in_def_line_seq", input_type=String(optional=True), prefix="--defline-seq", doc=InputDocumentation(doc="Defline format specification for sequence.")), ToolInput(tag="in_def_line_qual", input_type=Int(optional=True), prefix="--defline-qual", doc=InputDocumentation(doc="Defline format specification for quality.\n<fmt> is string of characters and/or\nvariables. The variables can be one of:\n$ac - accession, $si spot id, $sn spot\nname, $sg spot group (barcode), $sl spot\nlength in bases, $ri read number, $rn\nread name, $rl read length in bases.\n'[]' could be used for an optional\noutput: if all vars in [] yield empty\nvalues whole group is not printed. Empty\nvalue is empty string or for numeric\nvariables. Ex: @$sn[_$rn]/$ri '_$rn' is\nomitted if name is empty")), ToolInput(tag="in_ngc", input_type=File(optional=True), prefix="--ngc", doc=InputDocumentation(doc="<path> to ngc file")), ToolInput(tag="in_perm", input_type=File(optional=True), prefix="--perm", doc=InputDocumentation(doc="<path> to permission file")), ToolInput(tag="in_location", input_type=String(optional=True), prefix="--location", doc=InputDocumentation(doc="location in cloud")), ToolInput(tag="in_cart", input_type=File(optional=True), prefix="--cart", doc=InputDocumentation(doc="<path> to cart file")), ToolInput(tag="in_disable_multithreading", input_type=Boolean(optional=True), prefix="--disable-multithreading", doc=InputDocumentation(doc="disable multithreading")), ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="--verbose", doc=InputDocumentation(doc="Increase the verbosity of the program\nstatus messages. Use multiple times for\nmore verbosity.")), ToolInput(tag="in_log_level", input_type=Int(optional=True), prefix="--log-level", doc=InputDocumentation(doc="Logging level as number or enum string.\nOne of\n(fatal|sys|int|err|warn|info|debug) or\n(0-6) Current/default is warn")), ToolInput(tag="in_option_file", input_type=File(optional=True), prefix="--option-file", doc=InputDocumentation(doc="Read more options and parameters from the")), ToolInput(tag="in_name", input_type=String(), position=0, doc=InputDocumentation(doc="-I|--readids                     Append read id after spot id as")), ToolInput(tag="in_file_dot", input_type=File(), position=0, doc=InputDocumentation(doc="-h|--help                        print this message"))], outputs=[ToolOutput(tag="out_outdir", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_outdir", type_hint=File()), doc=OutputDocumentation(doc="Output directory, default is working\ndirectory '.'"))], container="quay.io/biocontainers/sra-tools:2.10.9--pl526haddd2b5_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Fastq_Dump_2_10_9_V0_1_0().translate("wdl")

