version 1.0

task RsemRunEbseq {
  input {
    File? ng_vector
    Boolean? h_slash_help
    String data_matrix_file
    String conditions
    String output_file_dot_normalized_data_matrix
    String output_file_dot_pattern
    String output_file_dot_cond_means
  }
  command <<<
    rsem-run-ebseq \
      ~{data_matrix_file} \
      ~{conditions} \
      ~{output_file_dot_normalized_data_matrix} \
      ~{output_file_dot_pattern} \
      ~{output_file_dot_cond_means} \
      ~{if defined(ng_vector) then ("--ngvector " +  '"' + ng_vector + '"') else ""} \
      ~{true="-h/--help" false="" h_slash_help}
  >>>
  parameter_meta {
    ng_vector: "This option provides the grouping information required by EBSeq for isoform-level differential expression analysis. The file can be generated by 'rsem-generate-ngvector'. Turning this option on is highly recommended for isoform-level differential expression analysis. (Default: off)"
    h_slash_help: "Show help information."
    data_matrix_file: "This file is a m by n matrix. m is the number of genes/transcripts and n is the number of total samples. Each element in the matrix represents the expected count for a particular gene/transcript in a particular sample. Users can use 'rsem-generate-data-matrix' to generate this file from expression result files."
    conditions: "Comma-separated list of values representing the number of replicates for each condition. For example, \"3,3\" means the data set contains 2 conditions and each condition has 3 replicates. \"2,3,3\" means the data set contains 3 conditions, with 2, 3, and 3 replicates for each condition respectively."
    output_file_dot_normalized_data_matrix: "This file contains the median normalized version of the input data matrix."
    output_file_dot_pattern: "This file is only generated when there are more than 2 conditions. It defines all possible expression patterns over the conditions using a matrix with names. Each row of the matrix refers to a different expression pattern and each column gives the expression status of a different condition. Two conditions are equally expressed if and only if their statuses are the same."
    output_file_dot_cond_means: "This file is only generated when there are more than 2 conditions. It gives the normalized mean count value for each gene/transcript at each condition. It is formatted as a matrix with names. Each row represents a gene/transcript and each column represent a condition. The order of genes/transcripts is the same as 'output_file'. This file can be used to calculate fold changes between conditions which users are interested in."
  }
}