class: CommandLineTool
id: SNPsplit.cwl
inputs:
- id: in_snp_file
  doc: "Mandatory file specifying SNP positions to be considered; may be a plain text\
    \ file or\ngzip compressed. Currently, the SNP file is expected to be in the following\
    \ format:\nSNP-ID     Chromosome  Position    Strand   Ref/SNP\nexample:   33941939\
    \        9       68878541       1       T/G\nOnly the information contained in\
    \ fields 'Chromosome', 'Position' and 'Reference/SNP base'\nare being used for\
    \ analysis. The genome referred to as 'Ref' will be used as genome 1,\nthe genome\
    \ containing the 'SNP' base as genome 2."
  type: File
  inputBinding:
    prefix: --snp_file
- id: in_paired
  doc: 'Paired-end mode. (Default: OFF).'
  type: boolean
  inputBinding:
    prefix: --paired
- id: in_oslash_outdir
  doc: "Write all output files into this directory. By default the output files will\
    \ be written into\nthe same folder as the input file(s). If the specified folder\
    \ does not exist, SNPsplit will attempt\nto create it first. The path to the output\
    \ folder can be either relative or absolute."
  type: File
  inputBinding:
    prefix: -o/--outdir
- id: in_singletons
  doc: "If the allele-tagged paired-end file also contains singleton alignments (which\
    \ is the\ndefault for e.g. TopHat), these will be written out to extra files (ending\
    \ in _st.bam)\ninstead of writing everything to combined paired-end and singleton\
    \ files. Default: OFF."
  type: File
  inputBinding:
    prefix: --singletons
- id: in_no_sort
  doc: "This option skips the sorting step if BAM files are already sorted by read\
    \ name (e.g.\nHi-C files generated by HiCUP). Please note that setting --no_sort\
    \ for unsorted paired-end\nfiles will break the tagging process!"
  type: boolean
  inputBinding:
    prefix: --no_sort
- id: in_hic
  doc: "Assumes Hi-C data processed with HiCUP (www.bioinformatics.babraham.ac.uk/projects/hicup/)\n\
    as input, i.e. the input BAM file is paired-end and Reads 1 and 2 follow each\
    \ other. Thus,\nthis option also sets the flags --paired and --no_sort. Default:\
    \ OFF."
  type: boolean
  inputBinding:
    prefix: --hic
- id: in_bisulfite
  doc: "Assumes Bisulfite-Seq data processed with Bismark (www.bioinformatics.babraham.ac.uk/projects/bismark/)\n\
    as input. In paired-end mode (--paired), Read 1 and Read 2 of a pair are expected\
    \ to follow\neach other in consecutive lines. SNPsplit will run a quick check\
    \ at the start of a run to see if the\nfile provided appears to be a Bismark file,\
    \ and set the flags --bisulfite and/or --paired\nautomatically. In addition it\
    \ will perform a quick check to see if a paired-end file appears\nto have been\
    \ positionally sorted, and if not will set the --no_sort flag."
  type: boolean
  inputBinding:
    prefix: --bisulfite
- id: in_sam_tools_path
  doc: "The path to your Samtools installation, e.g. /home/user/samtools/. Does not\
    \ need to\nbe specified explicitly if Samtools is in the PATH already."
  type: boolean
  inputBinding:
    prefix: --samtools_path
- id: in_verbose
  doc: Verbose output (for debugging).
  type: boolean
  inputBinding:
    prefix: --verbose
- id: in_conflicting_slash_weird
  doc: "Reads or read pairs that were classified as 'Conflicting' (XX:Z:CF) will be\
    \ written to\nan extra file (ending in .conflicting.bam) instead of being simply\
    \ skipped. Reads may be\nclassified as 'Conflicting' if a single read contains\
    \ SNP information for both genomes at\nthe same time, or if the SNP position was\
    \ deleted from the read. Read-pairs are considered\n'Conflicting' if either read\
    \ is was tagged with the XX:Z:CF flag. Default: OFF."
  type: boolean
  inputBinding:
    prefix: --conflicting/--weird
- id: in_sam
  doc: "The output will be written out in SAM format instead of BAM (default). SNPsplit\
    \ will attempt to use\nthe path to Samtools that was specified with --samtools_path,\
    \ or, if it hasn't been\nspecified, attempt to find Samtools in the PATH environment.\
    \ If no installation of\nSamtools can be found, the SAM output will be compressed\
    \ with GZIP instead (yielding a\n.sam.gz output file)."
  type: File
  inputBinding:
    prefix: --sam
outputs:
- id: out_stdout
  doc: Standard output stream
  type: stdout
- id: out_singletons
  doc: "If the allele-tagged paired-end file also contains singleton alignments (which\
    \ is the\ndefault for e.g. TopHat), these will be written out to extra files (ending\
    \ in _st.bam)\ninstead of writing everything to combined paired-end and singleton\
    \ files. Default: OFF."
  type: File
  outputBinding:
    glob: $(inputs.in_singletons)
- id: out_sam
  doc: "The output will be written out in SAM format instead of BAM (default). SNPsplit\
    \ will attempt to use\nthe path to Samtools that was specified with --samtools_path,\
    \ or, if it hasn't been\nspecified, attempt to find Samtools in the PATH environment.\
    \ If no installation of\nSamtools can be found, the SAM output will be compressed\
    \ with GZIP instead (yielding a\n.sam.gz output file)."
  type: File
  outputBinding:
    glob: $(inputs.in_sam)
cwlVersion: v1.1
baseCommand:
- SNPsplit
