from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Int, String, Directory, Boolean, File

Basenji_Sat_Bed_Py_V0_1_0 = CommandToolBuilder(tool="basenji_sat_bed.py", base_command=["basenji_sat_bed.py"], inputs=[ToolInput(tag="in_nucleotides_downstream_mutate", input_type=Int(optional=True), prefix="-d", doc=InputDocumentation(doc="Nucleotides downstream of center sequence to mutate\n[Default: 0]")), ToolInput(tag="in_genome_fasta_sequences", input_type=String(optional=True), prefix="-f", doc=InputDocumentation(doc="Genome FASTA for sequences [Default: none]")), ToolInput(tag="in_length_center_sequence", input_type=Int(optional=True), prefix="-l", doc=InputDocumentation(doc="Length of center sequence to mutate [Default: 0]")), ToolInput(tag="in_output_directory", input_type=Directory(optional=True), prefix="-o", doc=InputDocumentation(doc="Output directory [Default: sat_mut]")), ToolInput(tag="in_plots", input_type=Boolean(optional=True), prefix="--plots", doc=InputDocumentation(doc="Make heatmap plots [Default: False]")), ToolInput(tag="in_number_processes_passed", input_type=Int(optional=True), prefix="-p", doc=InputDocumentation(doc="Number of processes, passed by multi script")), ToolInput(tag="in_rc", input_type=Boolean(optional=True), prefix="--rc", doc=InputDocumentation(doc="Ensemble forward and reverse complement predictions\n[Default: False]")), ToolInput(tag="in_shifts", input_type=Int(optional=True), prefix="--shifts", doc=InputDocumentation(doc="Ensemble prediction shifts [Default: 0]")), ToolInput(tag="in_stats", input_type=String(optional=True), prefix="--stats", doc=InputDocumentation(doc="Comma-separated list of stats to save. [Default: sum]")), ToolInput(tag="in_file_specifying_indexes", input_type=File(optional=True), prefix="-t", doc=InputDocumentation(doc="File specifying target indexes and labels in table format")), ToolInput(tag="in_nucleotides_upstream_center", input_type=Int(optional=True), prefix="-u", doc=InputDocumentation(doc="Nucleotides upstream of center sequence to mutate\n[Default: 0]\n")), ToolInput(tag="in_params_file", input_type=String(), position=0, doc=InputDocumentation(doc="")), ToolInput(tag="in_model_file", input_type=String(), position=1, doc=InputDocumentation(doc="")), ToolInput(tag="in_bed_file", input_type=String(), position=2, doc=InputDocumentation(doc=""))], outputs=[ToolOutput(tag="out_output_directory", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_output_directory", type_hint=File()), doc=OutputDocumentation(doc="Output directory [Default: sat_mut]"))], container="quay.io/biocontainers/basenji:0.5.1--pyhdfd78af_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Basenji_Sat_Bed_Py_V0_1_0().translate("wdl")

