from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Float, Int, File, Boolean, String, Directory

Comparem_Aai_Wf_V0_1_0 = CommandToolBuilder(tool="comparem_aai_wf", base_command=["comparem", "aai_wf"], inputs=[ToolInput(tag="in_evalue", input_type=Float(optional=True), prefix="--evalue", doc=InputDocumentation(doc="e-value cutoff for identifying initial blast hits\n(default: 0.001)")), ToolInput(tag="in_per_identity", input_type=Float(optional=True), prefix="--per_identity", doc=InputDocumentation(doc="percent identity for defining homology (default: 30.0)")), ToolInput(tag="in_per_aln_len", input_type=Int(optional=True), prefix="--per_aln_len", doc=InputDocumentation(doc="percent alignment length of query sequence for\ndefining homology (default: 70.0)")), ToolInput(tag="in_file_ext", input_type=File(optional=True), prefix="--file_ext", doc=InputDocumentation(doc="extension of files to process (default: fna)")), ToolInput(tag="in_proteins", input_type=Boolean(optional=True), prefix="--proteins", doc=InputDocumentation(doc="indicates the input files contain protein sequences")), ToolInput(tag="in_force_table", input_type=String(optional=True), prefix="--force_table", doc=InputDocumentation(doc="force use of specific translation table")), ToolInput(tag="in_blastp", input_type=Boolean(optional=True), prefix="--blastp", doc=InputDocumentation(doc="use blastp instead of diamond")), ToolInput(tag="in_sensitive", input_type=Boolean(optional=True), prefix="--sensitive", doc=InputDocumentation(doc="use sensitive mode of DIAMOND")), ToolInput(tag="in_keep_headers", input_type=Boolean(optional=True), prefix="--keep_headers", doc=InputDocumentation(doc="indicates FASTA headers already have the format\n<genome_id>~<gene_id>")), ToolInput(tag="in_keep_r_bhs", input_type=Boolean(optional=True), prefix="--keep_rbhs", doc=InputDocumentation(doc="create file with reciprocal best hits")), ToolInput(tag="in_tmp_dir", input_type=Directory(optional=True), prefix="--tmp_dir", doc=InputDocumentation(doc="specify alternative directory for temporary files\n(default: /tmp)")), ToolInput(tag="in_cpus", input_type=Int(optional=True), prefix="--cpus", doc=InputDocumentation(doc="number of CPUs to use (default: 1)")), ToolInput(tag="in_silent", input_type=Boolean(optional=True), prefix="--silent", doc=InputDocumentation(doc="suppress output")), ToolInput(tag="in_input_files", input_type=String(), position=0, doc=InputDocumentation(doc="genome files")), ToolInput(tag="in_output_dir", input_type=String(), position=1, doc=InputDocumentation(doc="output directory"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Comparem_Aai_Wf_V0_1_0().translate("wdl", allow_empty_container=True)

