from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, File, String, Int

Gtftk_Great_Reg_Domains_V0_1_0 = CommandToolBuilder(tool="gtftk_great_reg_domains", base_command=["gtftk", "great_reg_domains"], inputs=[ToolInput(tag="in_input_file", input_type=Boolean(optional=True), prefix="--inputfile", doc=InputDocumentation(doc="Path to the GTF file. Default to STDIN (default: <stdin>)")), ToolInput(tag="in_output_file", input_type=File(optional=True), prefix="--outputfile", doc=InputDocumentation(doc="Output file. (default: <stdout>)")), ToolInput(tag="in_go_id", input_type=Boolean(optional=True), prefix="--go-id", doc=InputDocumentation(doc="The GO ID (e.g GO:0003700). Default is to return all genes. (default: None)")), ToolInput(tag="in_species", input_type=Boolean(optional=True), prefix="--species", doc=InputDocumentation(doc="The dataset/species. Use select_by_go to get the list of species (default: hsapiens)")), ToolInput(tag="in_upstream", input_type=Boolean(optional=True), prefix="--upstream", doc=InputDocumentation(doc="Extend the TSS in 5' by a given value. (default: 5000)")), ToolInput(tag="in_downstream", input_type=Boolean(optional=True), prefix="--downstream", doc=InputDocumentation(doc="Extend the TSS 3' by a given value. (default: 1000)")), ToolInput(tag="in_distal", input_type=Boolean(optional=True), prefix="--distal", doc=InputDocumentation(doc="Maximum extension in one direction (default: 1000000)")), ToolInput(tag="in_chrom_info", input_type=Boolean(optional=True), prefix="--chrom-info", doc=InputDocumentation(doc="Tabulated two-columns file. Chromosomes as column 1, sizes as column 2 (default: None)")), ToolInput(tag="in_mode", input_type=Boolean(optional=True), prefix="--mode", doc=InputDocumentation(doc="The type of 'association rule'. (default: basal_plus_extension)")), ToolInput(tag="in_http_proxy", input_type=Boolean(optional=True), prefix="--http-proxy", doc=InputDocumentation(doc="Use this http proxy (not tested/experimental). (default: )")), ToolInput(tag="in_https_proxy", input_type=Boolean(optional=True), prefix="--https-proxy", doc=InputDocumentation(doc="Use this https proxy (not tested/experimental). (default: )")), ToolInput(tag="in_verbosity", input_type=Boolean(optional=True), prefix="--verbosity", doc=InputDocumentation(doc="Set output verbosity ([0-3]). (default: 0)")), ToolInput(tag="in_no_date", input_type=File(optional=True), prefix="--no-date", doc=InputDocumentation(doc="Do not add date to output file names. (default: False)")), ToolInput(tag="in_add_chr", input_type=Boolean(optional=True), prefix="--add-chr", doc=InputDocumentation(doc="Add 'chr' to chromosome names before printing output. (default: False)")), ToolInput(tag="in_tmp_dir", input_type=Boolean(optional=True), prefix="--tmp-dir", doc=InputDocumentation(doc="Keep all temporary files into this folder. (default: None)")), ToolInput(tag="in_keep_all", input_type=Boolean(optional=True), prefix="--keep-all", doc=InputDocumentation(doc="Try to keep all temporary files even if process does not terminate normally. (default: False)")), ToolInput(tag="in_logger_file", input_type=Boolean(optional=True), prefix="--logger-file", doc=InputDocumentation(doc="Stores the arguments passed to the command into a file. (default: None)")), ToolInput(tag="in_write_message_to_file", input_type=Boolean(optional=True), prefix="--write-message-to-file", doc=InputDocumentation(doc="Store all message into a file. (default: None)")), ToolInput(tag="in_produced", input_type=String(), position=0, doc=InputDocumentation(doc="a set of 'labeled' regions with the same rules as those described in GREAT (Genomic")), ToolInput(tag="in_argument", input_type=Int(), position=0, doc=InputDocumentation(doc="Argument:"))], outputs=[ToolOutput(tag="out_output_file", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output_file", type_hint=File()), doc=OutputDocumentation(doc="Output file. (default: <stdout>)")), ToolOutput(tag="out_no_date", output_type=File(optional=True), selector=InputSelector(input_to_select="in_no_date", type_hint=File()), doc=OutputDocumentation(doc="Do not add date to output file names. (default: False)"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Gtftk_Great_Reg_Domains_V0_1_0().translate("wdl", allow_empty_container=True)

