from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, Directory, File

Repeatmodeler_V0_1_0 = CommandToolBuilder(tool="RepeatModeler", base_command=["RepeatModeler"], inputs=[ToolInput(tag="in_database", input_type=Boolean(optional=True), prefix="-database", doc=InputDocumentation(doc="The name of the sequence database to run an analysis on. This is the\nname that was provided to the BuildDatabase script using the '-name'\noption.")), ToolInput(tag="in_pa", input_type=Boolean(optional=True), prefix="-pa", doc=InputDocumentation(doc="#\nSpecify the number of parallel search jobs to run. RMBlast jobs will\nuse 4 cores each and ABBlast jobs will use a single core each. i.e.\non a machine with 12 cores and running with RMBlast you would use\n-pa 3 to fully utilize the machine.")), ToolInput(tag="in_recover_dir", input_type=Directory(optional=True), prefix="-recoverDir", doc=InputDocumentation(doc="If a run fails in the middle of processing, it may be possible\nrecover some results and continue where the previous run left off.\nSimply supply the output directory where the results of the failed\nrun were saved and the program will attempt to recover and continue\nthe run.")), ToolInput(tag="in_sr_and", input_type=Boolean(optional=True), prefix="-srand", doc=InputDocumentation(doc="#\nOptionally set the seed of the random number generator to a known\nvalue before the batches are randomly selected ( using Fisher Yates\nShuffling ). This is only useful if you need to reproduce the sample\nchoice between runs. This should be an integer number.")), ToolInput(tag="in_ltr_struct", input_type=Boolean(optional=True), prefix="-LTRStruct", doc=InputDocumentation(doc="Run the LTR structural discovery pipeline ( LTR_Harvest and\nLTR_retreiver ) and combine results with the RepeatScout/RECON\npipeline. [optional]")), ToolInput(tag="in_genome_sample_size_max", input_type=Boolean(optional=True), prefix="-genomeSampleSizeMax", doc=InputDocumentation(doc="#\nOptionally change the maximum bp of the genome to sample in all\nrounds of RECON (default=243000000).")), ToolInput(tag="in_r_scout_dir", input_type=File(optional=True), prefix="-rscout_dir", doc=InputDocumentation(doc="The path to the installation of the RepeatScout ( 1.0.6 or higher )\nde-novo repeatfinding program.")), ToolInput(tag="in_cd_hit_dir", input_type=File(optional=True), prefix="-cdhit_dir", doc=InputDocumentation(doc="The path to the installation of the CD-Hit sequence clustering\npackage.")), ToolInput(tag="in_maff_t_dir", input_type=File(optional=True), prefix="-mafft_dir", doc=InputDocumentation(doc="The path to the installation of the MAFFT multiple alignment\nprogram.")), ToolInput(tag="in_rmb_last_dir", input_type=File(optional=True), prefix="-rmblast_dir", doc=InputDocumentation(doc="The path to the installation of the RMBLAST sequence alignment\nprogram.")), ToolInput(tag="in_trf_prgm", input_type=File(optional=True), prefix="-trf_prgm", doc=InputDocumentation(doc="The full path including the name for the TRF program ( 4.0.9 or\nhigher )")), ToolInput(tag="in_repeatmasker_dir", input_type=File(optional=True), prefix="-repeatmasker_dir", doc=InputDocumentation(doc="The path to the installation of RepeatMasker.")), ToolInput(tag="in_ltr_retriever_dir", input_type=File(optional=True), prefix="-ltr_retriever_dir", doc=InputDocumentation(doc="The path to the installation of the LTR_Retriever structural LTR\nanalysis package.")), ToolInput(tag="in_abb_last_dir", input_type=File(optional=True), prefix="-abblast_dir", doc=InputDocumentation(doc="The path to the installation of the ABBLAST sequence alignment\nprogram.")), ToolInput(tag="in_genome_tools_dir", input_type=File(optional=True), prefix="-genometools_dir", doc=InputDocumentation(doc="The path to the installation of the GenomeTools package.")), ToolInput(tag="in_ninja_dir", input_type=File(optional=True), prefix="-ninja_dir", doc=InputDocumentation(doc="The path to the installation of the Ninja phylogenetic analysis\npackage.")), ToolInput(tag="in_recon_dir", input_type=File(optional=True), prefix="-recon_dir", doc=InputDocumentation(doc="The path to the installation of the RECON de-novo repeatfinding\nprogram."))], outputs=[ToolOutput(tag="out_recover_dir", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_recover_dir", type_hint=File()), doc=OutputDocumentation(doc="If a run fails in the middle of processing, it may be possible\nrecover some results and continue where the previous run left off.\nSimply supply the output directory where the results of the failed\nrun were saved and the program will attempt to recover and continue\nthe run."))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Repeatmodeler_V0_1_0().translate("wdl", allow_empty_container=True)

