version 1.0

task Gottcha.pl {
  input {
    Boolean? i_string_input
    Boolean? database
    Boolean? threads
    Boolean? db_level
    Boolean? outdir
    Boolean? prefix
    Boolean? rel_abu
    Boolean? mode
    Boolean? no_plasmid_hit
    Boolean? bwa_opt
    Boolean? st_dir
    Boolean? dumps_am
    Boolean? min_q
    Boolean? fix_l
    Boolean? ascii
    Boolean? min_cov
    Boolean? min_ml_hl
    Boolean? c_cov
    Boolean? min_len
    Boolean? min_hits
    Boolean? help_slash_h_slash
  }
  command <<<
    gottcha.pl \
      ~{true="--input" false="" i_string_input} \
      ~{true="--database" false="" database} \
      ~{true="--threads" false="" threads} \
      ~{true="--dbLevel" false="" db_level} \
      ~{true="--outdir" false="" outdir} \
      ~{true="--prefix" false="" prefix} \
      ~{true="--relAbu" false="" rel_abu} \
      ~{true="--mode" false="" mode} \
      ~{true="--noPlasmidHit" false="" no_plasmid_hit} \
      ~{true="--bwaOpt" false="" bwa_opt} \
      ~{true="--stDir" false="" st_dir} \
      ~{true="--dumpSam" false="" dumps_am} \
      ~{true="--minQ" false="" min_q} \
      ~{true="--fixL" false="" fix_l} \
      ~{true="--ascii" false="" ascii} \
      ~{true="--minCov" false="" min_cov} \
      ~{true="--minMLHL" false="" min_ml_hl} \
      ~{true="--cCov" false="" c_cov} \
      ~{true="--minLen" false="" min_len} \
      ~{true="--minHits" false="" min_hits} \
      ~{true="--help/h/" false="" help_slash_h_slash}
  >>>
  parameter_meta {
    i_string_input: "|i    <STRING>  Input one or multiple FASTQ file(s). Use comma (,) to separate multiple input files."
    database: "|d <STRING>  The path of signature database. The database can be in FASTA format or BWA index (5 files)."
    threads: "|t  <INT>     Number of threads [default: auto-detect]"
    db_level: "|l  <STRING>  Specify the taxonomic level of the input database  (e.g. family, species, genus, strain, etc.). The value will be auto-detected if the input database ended with levels (e.g. GOTTCHA_db.species). [default: none]"
    outdir: "|o   <STRING>  Output directory [default: ./]"
    prefix: "|p   <STRING>  Filename prefix of the output. [default: <INPUT_FILENAME_PREFIX>]"
    rel_abu: "|r   <STRING>  The field will be used to calculate relative abundance. You can specify one of the following fields: \"LINEAR_LENGTH\", \"TOTAL_BP_MAPPED\", \"HIT_COUNT\", \"LINEAR_DOC\". [default: LINEAR_DOC]"
    mode: "|m     <STRING>  You can specify one of the output mode: \"summary\" : this mode will report a summary of profiling result to *.gottcha.tsv file. \"full\"    : other than a summary, this mode will report unfiltered result to *.gottcha_full.tsv with more detail. \"all\"     : other than two tables, this mode will  keep all output files that were  generated by each profiling step. [default: summary]"
    no_plasmid_hit: "|n       Ignore alignments that hit to plasmids [default: null]"
    bwa_opt: "|b   <STRING>  BWA-MEM in this script is used to map input reads to  GOTTCHA database. If you want to run it with your own parameters, use this option to specify. [default: \"-k 30 -T 0 -B 100 -O 100 -E 100\"]"
    st_dir: "|s    <STRING>  Specify a directory contains pre-splitrimmed input files. E.g. input file is \"test.fastq\", the script will looking for \"test_splitrim.fastq\" and \"test_splitrim.stats.txt\" in the specified directory."
    dumps_am: "Dump the mapping result in SAM format."
    min_q: "<INT>     Minimum quality for a read to be considered valid (0-41) [default: 20]"
    fix_l: "<INT>     Fixed length to which each trimmed read will be cut  down to [default: 30]"
    ascii: "<INT>     ASCII encoding of quality score (33 or 64) [default:  33] "
    min_cov: "<FLOAT>   Minimum linear coverage to be considered valid in  abundance calculation [default: 0.005]"
    min_ml_hl: "<INT>     Minimum Mean-Linear-Hit-Length to be considered valid in abundance calculation [default: 5]"
    c_cov: "<FLOAT>   Critical coverage below which --minMLHL will cause an organism to fail [default: 0.006]"
    min_len: "<INT>     Minimum unique length to be considered valid in abundance calculation [default: 100]"
    min_hits: "<INT>     Minimum number of hits to be considered valid in  abundance calculation [10]"
    help_slash_h_slash: "?             display this help                   "
  }
}