class: CommandLineTool
id: anvi_import_taxonomy_for_genes.cwl
inputs:
- id: in_contigs_db
  doc: "Anvi'o contigs database generated by 'anvi-gen-\ncontigs-database' (default:\
    \ None)"
  type: string?
  inputBinding:
    prefix: --contigs-db
- id: in_parser
  doc: "Parser to make sense of the input files. There are 3\nparsers readily available:\
    \ ['default_matrix',\n'centrifuge', 'kaiju']. It is OK if you do not select\n\
    a parser, but in that case there will be no additional\ncontigs available except\
    \ the identification of single-\ncopy genes in your contigs for later use. Using\
    \ a\nparser will not prevent the analysis of single-copy\ngenes, but make anvio\
    \ more powerful to help you make\nsense of your results. Please see the documentation,\n\
    or get in touch with the developers if you have any\nquestions regarding parsers.\
    \ (default: None)"
  type: long?
  inputBinding:
    prefix: --parser
- id: in_s_input_files
  doc: "(S) [FILE(S) ...], --input-files FILE(S) [FILE(S) ...]\nInput file(s) for\
    \ selected parser. Each parser (except\n\"blank\") requires input files to process\
    \ that you\ngenerate before running anvio. Please see the\ndocumentation for details.\
    \ (default: None)"
  type: File?
  inputBinding:
    prefix: -i
- id: in_just_do_it
  doc: "Don't bother me with questions or warnings, just do\nit. (default: False)"
  type: boolean?
  inputBinding:
    prefix: --just-do-it
- id: in_gene_taxonomy
  doc: 'üç∫ More on `anvi-import-taxonomy-for-genes`:'
  type: string
  inputBinding:
    position: 0
outputs:
- id: out_stdout
  doc: Standard output stream
  type: stdout
hints:
- class: DockerRequirement
  dockerPull: quay.io/biocontainers/anvio-minimal:7--py_0
cwlVersion: v1.1
baseCommand:
- anvi-import-taxonomy-for-genes
