from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Directory, Array, String, Int, Boolean, Float

Phenix_Py_Vcf2Distancematrix_V0_1_0 = CommandToolBuilder(tool="phenix.py_vcf2distancematrix", base_command=["phenix.py", "vcf2distancematrix"], inputs=[ToolInput(tag="in_out", input_type=File(optional=True), prefix="--out", doc=InputDocumentation(doc="Path to the maxtrix output file in given format.\n[REQUIRED. default format is tab separated. use\n--format to change format]")), ToolInput(tag="in_directory", input_type=Directory(optional=True), prefix="--directory", doc=InputDocumentation(doc="Path to the directory with .vcf files. Input option 1.")), ToolInput(tag="in_input", input_type=Array(t=String(), optional=True), prefix="--input", doc=InputDocumentation(doc="List of VCF files to process. Input option 2.")), ToolInput(tag="in_alignment_input", input_type=File(optional=True), prefix="--alignment-input", doc=InputDocumentation(doc="FASTA FILE, -a MULTI FASTA FILE\nMulti fasta file with whole genome input alignment.\nInput option 3.")), ToolInput(tag="in_deletion", input_type=String(optional=True), prefix="--deletion", doc=InputDocumentation(doc="Method of recombination filtering. Either 'pairwise'\nor 'complete' ['pairwise']")), ToolInput(tag="in_substitution", input_type=Int(optional=True), prefix="--substitution", doc=InputDocumentation(doc="Substituition model. Either 'number_of_differences',\n'jc69', 'k80', 'tn84' or 't93'\n['number_of_differences']")), ToolInput(tag="in_include", input_type=File(optional=True), prefix="--include", doc=InputDocumentation(doc="FILE    Only include positions in BED file in the FASTA")), ToolInput(tag="in_exclude", input_type=File(optional=True), prefix="--exclude", doc=InputDocumentation(doc="FILE    Exclude any positions specified in the BED file.")), ToolInput(tag="in_remove_recombination", input_type=Boolean(optional=True), prefix="--remove-recombination", doc=InputDocumentation(doc="Attempt to remove recombination from distance matrix.\n[don't]")), ToolInput(tag="in_ref_genome", input_type=File(optional=True), prefix="--refgenome", doc=InputDocumentation(doc="FILE, -g FASTA FILE\nReference genome used for SNP calling [Required for\nrecombination removal].")), ToolInput(tag="in_ref_genome_name", input_type=Int(optional=True), prefix="--refgenomename", doc=InputDocumentation(doc="Name of reference genome in input alignment [Required\nif input option 3 is used and reference is not named\n'reference'].")), ToolInput(tag="in_threshold", input_type=Float(optional=True), prefix="--threshold", doc=InputDocumentation(doc="Density tyhreshold above mean density for relevant\npair. [1.0].")), ToolInput(tag="in_threads", input_type=Int(optional=True), prefix="--threads", doc=InputDocumentation(doc="Number of threads to use. [1].")), ToolInput(tag="in_format", input_type=File(optional=True), prefix="--format", doc=InputDocumentation(doc="Change format for output file. Available options csv,\ntsv and mega. [tsv]")), ToolInput(tag="in_tree", input_type=File(optional=True), prefix="--tree", doc=InputDocumentation(doc="Make an NJ tree and write it to the given file in\nnewick format. [Default: Don't make tree, only matrix]")), ToolInput(tag="in_with_stats", input_type=Boolean(optional=True), prefix="--with-stats", doc=InputDocumentation(doc="Write additional files with information on removed\nrecombinant SNPs. [don't]\n"))], outputs=[ToolOutput(tag="out_out", output_type=File(optional=True), selector=InputSelector(input_to_select="in_out", type_hint=File()), doc=OutputDocumentation(doc="Path to the maxtrix output file in given format.\n[REQUIRED. default format is tab separated. use\n--format to change format]")), ToolOutput(tag="out_format", output_type=File(optional=True), selector=InputSelector(input_to_select="in_format", type_hint=File()), doc=OutputDocumentation(doc="Change format for output file. Available options csv,\ntsv and mega. [tsv]"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Phenix_Py_Vcf2Distancematrix_V0_1_0().translate("wdl", allow_empty_container=True)

