from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Boolean, Int, String

Rnaforester_V0_1_0 = CommandToolBuilder(tool="RNAforester", base_command=["RNAforester"], inputs=[ToolInput(tag="in_read_input_file", input_type=File(optional=True), prefix="-f", doc=InputDocumentation(doc="read input from file")), ToolInput(tag="in_score", input_type=Boolean(optional=True), prefix="--score", doc=InputDocumentation(doc="compute only scores, no alignment")), ToolInput(tag="in_no_scale", input_type=Boolean(optional=True), prefix="--noscale", doc=InputDocumentation(doc="suppress output of scale")), ToolInput(tag="in_tables", input_type=Boolean(optional=True), prefix="--tables", doc=InputDocumentation(doc="shows dynamic programming tables")), ToolInput(tag="in_backtrace", input_type=Boolean(optional=True), prefix="--backtrace", doc=InputDocumentation(doc="shows backtrace call table cells")), ToolInput(tag="in_calculate_alignment_top", input_type=Boolean(optional=True), prefix="-t", doc=InputDocumentation(doc="calculate alignment top down instead of bottom up")), ToolInput(tag="in_calculate_distance_similarity", input_type=Boolean(optional=True), prefix="-d", doc=InputDocumentation(doc="calculate distance instead of similarity")), ToolInput(tag="in_calculate_relative_score", input_type=Boolean(optional=True), prefix="-r", doc=InputDocumentation(doc="calculate relative score")), ToolInput(tag="in_local_similarity", input_type=Boolean(optional=True), prefix="-l", doc=InputDocumentation(doc="local similarity")), ToolInput(tag="in_so", input_type=Int(optional=True), prefix="-so", doc=InputDocumentation(doc="local suboptimal alignments within int%")), ToolInput(tag="in_smallinlarge_similarity", input_type=Boolean(optional=True), prefix="-s", doc=InputDocumentation(doc="small-in-large similarity")), ToolInput(tag="in_anchor", input_type=Boolean(optional=True), prefix="--anchor", doc=InputDocumentation(doc="use shape anchoring for speedup")), ToolInput(tag="in_affine_gap_scoring", input_type=Boolean(optional=True), prefix="-a", doc=InputDocumentation(doc="affine gap scoring")), ToolInput(tag="in_multiple_alignment_mode", input_type=Boolean(optional=True), prefix="-m", doc=InputDocumentation(doc="multiple alignment mode")), ToolInput(tag="in_ribo_sum", input_type=Boolean(optional=True), prefix="--RIBOSUM", doc=InputDocumentation(doc="RIBOSUM85-60 scoring matrix (base-pair substitutions)")), ToolInput(tag="in_cbm_in", input_type=String(optional=True), prefix="-cbmin", doc=InputDocumentation(doc="minimum base frequency for consensus structure")), ToolInput(tag="in_cm_in", input_type=String(optional=True), prefix="-cmin", doc=InputDocumentation(doc="minimum basepair frequency for consensus structure")), ToolInput(tag="in_mt", input_type=String(optional=True), prefix="-mt", doc=InputDocumentation(doc="clustering threshold")), ToolInput(tag="in_mc", input_type=String(optional=True), prefix="-mc", doc=InputDocumentation(doc="clustering cutoff")), ToolInput(tag="in_predict_structures_sequences", input_type=Boolean(optional=True), prefix="-p", doc=InputDocumentation(doc="predict structures from sequences")), ToolInput(tag="in_pm_in", input_type=String(optional=True), prefix="-pmin", doc=InputDocumentation(doc="minimum basepair frequency for prediction")), ToolInput(tag="in_sp", input_type=File(optional=True), prefix="-sp", doc=InputDocumentation(doc="save profile")), ToolInput(tag="in_ps", input_type=File(optional=True), prefix="-ps", doc=InputDocumentation(doc="profile search")), ToolInput(tag="in_pm", input_type=Int(optional=True), prefix="-pm", doc=InputDocumentation(doc="basepair(bond) match score")), ToolInput(tag="in_pd", input_type=Int(optional=True), prefix="-pd", doc=InputDocumentation(doc="basepair bond indel score")), ToolInput(tag="in_pdo", input_type=Int(optional=True), prefix="-pdo", doc=InputDocumentation(doc="basepair bond indel open score")), ToolInput(tag="in_bm", input_type=Int(optional=True), prefix="-bm", doc=InputDocumentation(doc="base match score")), ToolInput(tag="in_br", input_type=Int(optional=True), prefix="-br", doc=InputDocumentation(doc="base mismatch score")), ToolInput(tag="in_bd", input_type=Int(optional=True), prefix="-bd", doc=InputDocumentation(doc="base indel score")), ToolInput(tag="in_b_do", input_type=Int(optional=True), prefix="-bdo", doc=InputDocumentation(doc="base indel open score")), ToolInput(tag="in_two_d", input_type=Boolean(optional=True), prefix="-2d", doc=InputDocumentation(doc="generate alignment 2D plots in postscript format")), ToolInput(tag="in_two_d_hide_base_num", input_type=Boolean(optional=True), prefix="--2d_hidebasenum", doc=InputDocumentation(doc="hide base numbers in 2D plot")), ToolInput(tag="in_two_d_base_num_interval", input_type=Int(optional=True), prefix="--2d_basenuminterval", doc=InputDocumentation(doc="show every n-th base number")), ToolInput(tag="in_two_d_grey", input_type=Boolean(optional=True), prefix="--2d_grey", doc=InputDocumentation(doc="use only grey colors in 2D plots")), ToolInput(tag="in_two_d_scale", input_type=Int(optional=True), prefix="--2d_scale", doc=InputDocumentation(doc="scale factor for the 2d plots")), ToolInput(tag="in_two_d_fig", input_type=Boolean(optional=True), prefix="--2d_fig", doc=InputDocumentation(doc="generate additional fig file of 2d plot")), ToolInput(tag="in_fast_a", input_type=Boolean(optional=True), prefix="--fasta", doc=InputDocumentation(doc="generate fasta output of alignments"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Rnaforester_V0_1_0().translate("wdl", allow_empty_container=True)

