from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, File, String

Spectrast2Tsv_Py_V0_1_0 = CommandToolBuilder(tool="spectrast2tsv.py", base_command=["spectrast2tsv.py"], inputs=[ToolInput(tag="in_remove_duplicate_masses", input_type=Boolean(optional=True), prefix="-d", doc=InputDocumentation(doc="Remove duplicate masses from labeling")), ToolInput(tag="in_use_theoretical_mass", input_type=Boolean(optional=True), prefix="-e", doc=InputDocumentation(doc="Use theoretical mass")), ToolInput(tag="in_fastafile_fasta_file", input_type=Boolean(optional=True), prefix="-f", doc=InputDocumentation(doc="fasta_file    Fasta file to relate peptides to their proteins (this is optional).")), ToolInput(tag="in_massmodifs_list_g", input_type=Boolean(optional=True), prefix="-g", doc=InputDocumentation(doc="mass_modifs   List of allowed fragment mass modifications. Useful for phosphorylation and neutral losses. Example: -g -79.97,-97.98,-17.03,-18.01")), ToolInput(tag="in_labelingfile_file_containing", input_type=Boolean(optional=True), prefix="-i", doc=InputDocumentation(doc="labeling_file File containing the amino acid isotopic labeling mass shifts. If this option is used, heavy transitions will be generated.")), ToolInput(tag="in_outputkey_select_provided", input_type=Boolean(optional=True), prefix="-k", doc=InputDocumentation(doc="output_key    Select the output provided. Keys available: openswath, peakview. Default: peakview")), ToolInput(tag="in_masslimits_lower_limits", input_type=Boolean(optional=True), prefix="-l", doc=InputDocumentation(doc="mass_limits   Lower and upper mass limits of fragment ions. Example: -l 400,2000")), ToolInput(tag="in_modsfile_file_modifications", input_type=Boolean(optional=True), prefix="-m", doc=InputDocumentation(doc="mods_file     File with the modifications delta mass")), ToolInput(tag="in_int_max_number", input_type=Boolean(optional=True), prefix="-n", doc=InputDocumentation(doc="int           Max number of reported ions per peptide/z. Default: 20")), ToolInput(tag="in_int_min_number", input_type=Boolean(optional=True), prefix="-o", doc=InputDocumentation(doc="int           Min number of reported ions per peptide/z. Default: 3")), ToolInput(tag="in_float_error_allowed", input_type=Boolean(optional=True), prefix="-p", doc=InputDocumentation(doc="float         Maximum error allowed at the annotation of a fragment ion. Default: 0.05")), ToolInput(tag="in_int_number_use", input_type=Boolean(optional=True), prefix="-q", doc=InputDocumentation(doc="int            Number of processors to use (only for isoforms!). Default: 1")), ToolInput(tag="in_ionseries_list_used", input_type=Boolean(optional=True), prefix="-s", doc=InputDocumentation(doc="ion_series    List of ion series to be used. Example: -s y,b")), ToolInput(tag="in_timescale_options_minutes", input_type=Boolean(optional=True), prefix="-t", doc=InputDocumentation(doc="time-scale    Options: minutes, seconds. Default: seconds.")), ToolInput(tag="in_unimodcode_use_code", input_type=Boolean(optional=True), prefix="-u", doc=InputDocumentation(doc="unimod-code    Use this unimod code as a switching modification. Useful for phosphorylations. Example: -u 21")), ToolInput(tag="in_verbose_mode", input_type=Boolean(optional=True), prefix="-v", doc=InputDocumentation(doc="Verbose mode.")), ToolInput(tag="in_swathsfile_file_containing", input_type=Boolean(optional=True), prefix="-w", doc=InputDocumentation(doc="swaths_file   File containing the swath ranges. This is used to remove transitions with Q3 falling in the swath mass range. (line breaks in windows/unix format)")), ToolInput(tag="in_fragment_ion_charge", input_type=Boolean(optional=True), prefix="-x", doc=InputDocumentation(doc="allowed_frg_z Fragment ion charge states allowed. Default: 1,2")), ToolInput(tag="in_uisorder_when_using", input_type=Boolean(optional=True), prefix="-y", doc=InputDocumentation(doc="UIS-order     When using a switching modification, this determines the UIS order to be calculated. If -1 is set, all transitions for each isoform will be reported. Default : 2")), ToolInput(tag="in_outfile_output_file", input_type=File(optional=True), prefix="-a", doc=InputDocumentation(doc="outfile       Output file name (default: appends _peakview.txt)")), ToolInput(tag="in_spectra_st_file", input_type=String(), position=0, doc=InputDocumentation(doc=""))], outputs=[ToolOutput(tag="out_outfile_output_file", output_type=File(optional=True), selector=InputSelector(input_to_select="in_outfile_output_file", type_hint=File()), doc=OutputDocumentation(doc="outfile       Output file name (default: appends _peakview.txt)"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Spectrast2Tsv_Py_V0_1_0().translate("wdl", allow_empty_container=True)

