from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, File

Mummerplot_V0_1_0 = CommandToolBuilder(tool="mummerplot", base_command=["mummerplot"], inputs=[ToolInput(tag="in_breaklen_highlight_alignments", input_type=Boolean(optional=True), prefix="-b", doc=InputDocumentation(doc="|breaklen     Highlight alignments with breakpoints further than\nbreaklen nucleotides from the nearest sequence end")), ToolInput(tag="in_depend", input_type=Boolean(optional=True), prefix="--depend", doc=InputDocumentation(doc="Print the dependency information and exit")), ToolInput(tag="in_filter", input_type=Boolean(optional=True), prefix="--filter", doc=InputDocumentation(doc="Only display .delta alignments which represent the 'best'\nhit to any particular spot on either sequence, i.e. a\none-to-one mapping of reference and query subsequences")), ToolInput(tag="in_layout", input_type=Boolean(optional=True), prefix="--layout", doc=InputDocumentation(doc="Layout a .delta multiplot in an intelligible fashion,\nthis option requires the -R -Q options")), ToolInput(tag="in_fat", input_type=Boolean(optional=True), prefix="--fat", doc=InputDocumentation(doc="Layout sequences using fattest alignment only")), ToolInput(tag="in_prefix_set_prefix", input_type=Boolean(optional=True), prefix="-p", doc=InputDocumentation(doc="|prefix       Set the prefix of the output files (default 'out')")), ToolInput(tag="in_rv", input_type=Boolean(optional=True), prefix="-rv", doc=InputDocumentation(doc="Reverse video for x11 plots")), ToolInput(tag="in_idr_plot_id", input_type=Boolean(optional=True), prefix="-r", doc=InputDocumentation(doc="|IdR          Plot a particular reference sequence ID on the X-axis")), ToolInput(tag="in_idq_plot_particular", input_type=Boolean(optional=True), prefix="-q", doc=InputDocumentation(doc="|IdQ          Plot a particular query sequence ID on the Y-axis")), ToolInput(tag="in_rfile_plot_set", input_type=Boolean(optional=True), prefix="-R", doc=InputDocumentation(doc="|Rfile        Plot an ordered set of reference sequences from Rfile")), ToolInput(tag="in_qfile_plot_set", input_type=Boolean(optional=True), prefix="-Q", doc=InputDocumentation(doc="|Qfile        Plot an ordered set of query sequences from Qfile\nRfile/Qfile Can either be the original DNA multi-FastA\nfiles or lists of sequence IDs, lens and dirs [ /+/-]")), ToolInput(tag="in_size_set_size", input_type=Boolean(optional=True), prefix="-s", doc=InputDocumentation(doc="|size         Set the output size to small, medium or large")), ToolInput(tag="in_medium", input_type=Boolean(optional=True), prefix="--medium", doc=InputDocumentation(doc="(default 'small')")), ToolInput(tag="in_snp", input_type=Boolean(optional=True), prefix="--SNP", doc=InputDocumentation(doc="Highlight SNP locations in each alignment")), ToolInput(tag="in_terminal_set_terminal", input_type=Boolean(optional=True), prefix="-t", doc=InputDocumentation(doc="|terminal     Set the output terminal to x11, postscript or png")), ToolInput(tag="in_postscript", input_type=Boolean(optional=True), prefix="--postscript", doc=InputDocumentation(doc="(default 'x11')")), ToolInput(tag="in_xrange_set_xrange", input_type=Boolean(optional=True), prefix="-x", doc=InputDocumentation(doc="|xrange       Set the xrange for the plot '[min:max]'")), ToolInput(tag="in_yrange_set_yrange", input_type=Boolean(optional=True), prefix="-y", doc=InputDocumentation(doc="|yrange       Set the yrange for the plot '[min:max]'")), ToolInput(tag="in_match_file", input_type=File(), position=0, doc=InputDocumentation(doc=""))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Mummerplot_V0_1_0().translate("wdl", allow_empty_container=True)

