class: CommandLineTool
id: bismark2report.cwl
inputs:
- id: in_oslash_output
  doc: "Name of the output file (optional). If not specified explicitly, the output\
    \ filename will be derived\nfrom the Bismark alignment report file. Specifying\
    \ an output filename only works if the HTML report is\nto be generated for a single\
    \ Bismark alignment report (and potentially additional reports)."
  type: File
  inputBinding:
    prefix: -o/--output
- id: in_dir
  doc: Output directory. Output is written to the current directory if not specified
    explicitly.
  type: Directory
  inputBinding:
    prefix: --dir
- id: in_alignment_report
  doc: "If not specified explicitly, bismark2report attempts to find Bismark report\
    \ file(s) in the current\ndirectory and produces a separate HTML report for each\
    \ mapping report file. Based on the basename of\nthe Bismark mapping report, bismark2report\
    \ will also attempt to find the other Bismark reports (see below)\nfor inclusion\
    \ into the HTML report. Specifying a Bismark alignment report file is mandatory."
  type: File
  inputBinding:
    prefix: --alignment_report
- id: in_ded_up_report
  doc: "If not specified explicitly, bismark2report attempts to find a deduplication\
    \ report file with the same\nbasename as the Bismark mapping report (generated\
    \ by deduplicate_bismark) in the\ncurrent working directory. Including a deduplication\
    \ report is optional, and using the FILE 'none'\nwill skip this step entirely."
  type: File
  inputBinding:
    prefix: --dedup_report
- id: in_splitting_report
  doc: "If not specified explicitly, bismark2report attempts to find a splitting report\
    \ file with the same\nbasename as the Bismark mapping report (generated by the\
    \ Bismark methylation extractor) in the current\nworking directory. Including\
    \ a splitting report is optional, and using the FILE 'none' will skip this\nstep\
    \ entirely."
  type: File
  inputBinding:
    prefix: --splitting_report
- id: in_m_bias_report
  doc: "If not specified explicitly, bismark2report attempts to find a single M-bias\
    \ report file with the same\nbasename as the Bismark mapping report (generated\
    \ by the Bismark methylation extractor) in the current\nworking directory. Including\
    \ an M-Bias report is optional, and using the FILE 'none' will skip this step\n\
    entirely."
  type: File
  inputBinding:
    prefix: --mbias_report
- id: in_nucleotide_report
  doc: "If not specified explicitly, bismark2report attempts to find a single nucleotide\
    \ coverage report file\nwith the same basename as the Bismark mapping report (generated\
    \ by Bismark with the option\n'--nucleotide_coverage') in the current working\
    \ directory. Including a nucleotide coverage statistics\nreport is optional, and\
    \ using the FILE 'none' will skip this report entirely.\nScript last modified:\
    \ 03 May 2019\n"
  type: File
  inputBinding:
    prefix: --nucleotide_report
outputs:
- id: out_stdout
  doc: Standard output stream
  type: stdout
- id: out_oslash_output
  doc: "Name of the output file (optional). If not specified explicitly, the output\
    \ filename will be derived\nfrom the Bismark alignment report file. Specifying\
    \ an output filename only works if the HTML report is\nto be generated for a single\
    \ Bismark alignment report (and potentially additional reports)."
  type: File
  outputBinding:
    glob: $(inputs.in_oslash_output)
- id: out_dir
  doc: Output directory. Output is written to the current directory if not specified
    explicitly.
  type: Directory
  outputBinding:
    glob: $(inputs.in_dir)
cwlVersion: v1.1
baseCommand:
- bismark2report
