from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Directory, Int, Float, String

Wgfast_V0_1_0 = CommandToolBuilder(tool="wgfast", base_command=["wgfast"], inputs=[ToolInput(tag="in_reference_directory", input_type=File(optional=True), prefix="--reference_directory", doc=InputDocumentation(doc="path to reference file directory [REQUIRED]")), ToolInput(tag="in_read_directory", input_type=Directory(optional=True), prefix="--read_directory", doc=InputDocumentation(doc="path to directory of fastq files [REQUIRED]")), ToolInput(tag="in_processors", input_type=Int(optional=True), prefix="--processors", doc=InputDocumentation(doc="# of processors to use - defaults to 2")), ToolInput(tag="in_coverage", input_type=Int(optional=True), prefix="--coverage", doc=InputDocumentation(doc="minimum SNP coverage required to be called a SNP -\ndefaults to 3")), ToolInput(tag="in_proportion", input_type=Float(optional=True), prefix="--proportion", doc=InputDocumentation(doc="proportion of alleles to be called a SNP, defaults to\n0.9")), ToolInput(tag="in_keep", input_type=String(optional=True), prefix="--keep", doc=InputDocumentation(doc="keep temp files?  Defaults to F")), ToolInput(tag="in_subsample", input_type=String(optional=True), prefix="--subsample", doc=InputDocumentation(doc="Run subsample routine? Defaults to T")), ToolInput(tag="in_sub_nums", input_type=Int(optional=True), prefix="--subnums", doc=InputDocumentation(doc="number of subsamples to process, defaults to 100")), ToolInput(tag="in_doc", input_type=String(optional=True), prefix="--doc", doc=InputDocumentation(doc="run depth of coverage on all files?  Defaults to T")), ToolInput(tag="in_temp", input_type=Directory(optional=True), prefix="--temp", doc=InputDocumentation(doc="temporary directory for GATK analysis, defaults to\n/tmp")), ToolInput(tag="in_fudge_factor", input_type=Float(optional=True), prefix="--fudge_factor", doc=InputDocumentation(doc="How close does a subsample have to be from true\nplacement?  Defaults to 0.1")), ToolInput(tag="in_only_subs", input_type=String(optional=True), prefix="--only_subs", doc=InputDocumentation(doc="Only run sub-sample routine and exit? Defaults to F")), ToolInput(tag="in_model", input_type=String(optional=True), prefix="--model", doc=InputDocumentation(doc="which model to run with raxml, GTRGAMMA, ASC_GTRGAMMA,\nGTRCAT, ASC_GTRCAT")), ToolInput(tag="in_g_atk_method", input_type=String(optional=True), prefix="--gatk_method", doc=InputDocumentation(doc="How to call GATK? Defaults to\nEMIT_ALL_CONFIDENT_SITES, can be EMIT_ALL_SITES\n"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Wgfast_V0_1_0().translate("wdl", allow_empty_container=True)

