from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, File

Tag2Sort_V0_1_0 = CommandToolBuilder(tool="tag2sort", base_command=["tag2sort"], inputs=[ToolInput(tag="in_paired", input_type=Boolean(optional=True), prefix="--paired", doc=InputDocumentation(doc="Paired-end mode. (Default: OFF).")), ToolInput(tag="in_outdir", input_type=File(optional=True), prefix="--outdir", doc=InputDocumentation(doc="Write all output files into this directory. By default the output files will be written into\nthe same folder as the input file(s).")), ToolInput(tag="in_singletons", input_type=File(optional=True), prefix="--singletons", doc=InputDocumentation(doc="If the allele-tagged paired-end file also contains singleton alignments (which is the\ndefault for e.g. TopHat), these will be written out to extra files (ending in _st.bam)\ninstead of writing everything to combined paired-end and singleton files. Default: OFF.")), ToolInput(tag="in_hic", input_type=Boolean(optional=True), prefix="--hic", doc=InputDocumentation(doc="Assumes Hi-C data processed with HiCUP (www.bioinformatics.babraham.ac.uk/projects/hicup/)\nas input, i.e. the input BAM file is paired-end and Reads 1 and 2 follow each other. Thus,\nthis option also sets the flags --paired and --no_sort. Default: OFF.")), ToolInput(tag="in_sam_tools_path", input_type=Boolean(optional=True), prefix="--samtools_path", doc=InputDocumentation(doc="The path to your Samtools installation, e.g. /home/user/samtools/. Does not need to\nbe specified explicitly if Samtools is in the PATH already.")), ToolInput(tag="in_conflicting", input_type=Boolean(optional=True), prefix="--conflicting", doc=InputDocumentation(doc="Reads or read pairs that were classified as 'Conflicting' (XX:Z:CF) will be written to\nan extra file (ending in .conflicting.bam) instead of being simply skipped. Reads may be\nclassified as 'Conflicting' if a single read contains SNP information for both genomes at\nthe same time, or if the SNP position was deleted from the read. Read-pairs are considered\n'Conflicting' if either read is was tagged with the XX:Z:CF flag. Default: OFF."))], outputs=[ToolOutput(tag="out_singletons", output_type=File(optional=True), selector=InputSelector(input_to_select="in_singletons", type_hint=File()), doc=OutputDocumentation(doc="If the allele-tagged paired-end file also contains singleton alignments (which is the\ndefault for e.g. TopHat), these will be written out to extra files (ending in _st.bam)\ninstead of writing everything to combined paired-end and singleton files. Default: OFF."))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Tag2Sort_V0_1_0().translate("wdl", allow_empty_container=True)

