from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean

Var2Vcf_Paired_Pl_V0_1_0 = CommandToolBuilder(tool="var2vcf_paired.pl", base_command=["var2vcf_paired.pl"], inputs=[ToolInput(tag="in_print_this_usage", input_type=Boolean(optional=True), prefix="-H", doc=InputDocumentation(doc="Print this usage.")), ToolInput(tag="in_set_chrosomes_have", input_type=Boolean(optional=True), prefix="-C", doc=InputDocumentation(doc="If set, chrosomes will have names of 1,2,3,...,X,Y, instead of chr1, chr2, ..., chrX, chrY")), ToolInput(tag="in_set_variants_pass", input_type=Boolean(optional=True), prefix="-S", doc=InputDocumentation(doc="If set, variants that didn't pass filters will not be present in VCF file")), ToolInput(tag="in_set_output_only", input_type=Boolean(optional=True), prefix="-M", doc=InputDocumentation(doc="If set, output only candidate somatic")), ToolInput(tag="in_indicate_output_variants", input_type=Boolean(optional=True), prefix="-A", doc=InputDocumentation(doc="Indicate to output all variants at the same position.  By default, only the variant with the highest allele frequency is converted to VCF.")), ToolInput(tag="in_float_deprecatedthe_minimum", input_type=Boolean(optional=True), prefix="-D", doc=InputDocumentation(doc="float (0-1) # Deprecated\nThe minimum allele frequency difference between two samples required in addition to p-value.  Not compitable\nwith -M option.  It's for interest of identifying variants with different AF, not just somatic.")), ToolInput(tag="in_two_somatic_candidates", input_type=Boolean(optional=True), prefix="-c", doc=InputDocumentation(doc="int\nIf two somatic candidates are within {int} bp, they're both filtered.  Default: 0 or no filtering")), ToolInput(tag="in_intthe_maximum_nonmonomer", input_type=Boolean(optional=True), prefix="-I", doc=InputDocumentation(doc="int\nThe maximum non-monomer MSI allowed for a HT variant with AF < 0.6.  By default, 12, or any variants with AF < 0.6 in a region\nwith > 12 non-monomer MSI will be considered false positive.  For monomers, that number is 10.")), ToolInput(tag="in_intthe_maximum_mean", input_type=Boolean(optional=True), prefix="-m", doc=InputDocumentation(doc="int\nThe maximum mean mismatches allowed.  Default: 5.25, or if a variant is supported by reads with more than 5.25 mismathes, it'll be considered\nfalse positive.  Mismatches don't includes indels in the alignment.")), ToolInput(tag="in_namesthe_sample_names", input_type=Boolean(optional=True), prefix="-N", doc=InputDocumentation(doc="Name(s)\nThe sample name(s).  If only one name is given, the matched will be simply names as 'name-match'.  Two names\nare given separated by '|', such as 'tumor|blood'.")), ToolInput(tag="in_maximum_pvalue_default", input_type=Boolean(optional=True), prefix="-P", doc=InputDocumentation(doc="float\nThe maximum p-value.  Default to 0.05.")), ToolInput(tag="in_minimum_mean_position", input_type=Boolean(optional=True), prefix="-p", doc=InputDocumentation(doc="float\nThe minimum mean position of variants in the read.  Default: 5.")), ToolInput(tag="in_minimum_mean_base", input_type=Boolean(optional=True), prefix="-q", doc=InputDocumentation(doc="float\nThe minimum mean base quality.  Default to 22.5 for Illumina sequencing")), ToolInput(tag="in_minimum_mapping_quality", input_type=Boolean(optional=True), prefix="-Q", doc=InputDocumentation(doc="float\nThe minimum mapping quality.  Default to 0 for Illumina sequencing")), ToolInput(tag="in_minimum_total_depth", input_type=Boolean(optional=True), prefix="-d", doc=InputDocumentation(doc="integer\nThe minimum total depth.  Default to 5")), ToolInput(tag="in_minimum_variant_depth", input_type=Boolean(optional=True), prefix="-v", doc=InputDocumentation(doc="integer\nThe minimum variant depth.  Default to 3")), ToolInput(tag="in_minimum_allele_frequency_default", input_type=Boolean(optional=True), prefix="-f", doc=InputDocumentation(doc="float\nThe minimum allele frequency.  Default to 0.02")), ToolInput(tag="in_signalnoisethe_minimum_signal", input_type=Boolean(optional=True), prefix="-o", doc=InputDocumentation(doc="signal/noise\nThe minimum signal to noise, or the ratio of hi/(lo+0.5).  Default to 1.5.  Set it higher for deep sequencing.")), ToolInput(tag="in_minimum_allele_frequency_consider", input_type=Boolean(optional=True), prefix="-F", doc=InputDocumentation(doc="float\nThe minimum allele frequency to consider to be homozygous.  Default to 0.2.  Thus frequency > 0.8 (1-0.2) will\nbe considered homozygous '1/1', between 0.5 - (1-0.2) will be '1/0', between (-f) - 0.5 will be '0/1',\nbelow (-f) will be '0/0'."))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Var2Vcf_Paired_Pl_V0_1_0().translate("wdl", allow_empty_container=True)

