from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, Int, Float, File

Filterbam_V0_1_0 = CommandToolBuilder(tool="filterBam", base_command=["filterBam"], inputs=[ToolInput(tag="in_best", input_type=Boolean(optional=True), prefix="--best", doc=InputDocumentation(doc="output all best matches that satisfy minId and minCover (default 0)")), ToolInput(tag="in_no_introns", input_type=Boolean(optional=True), prefix="--noIntrons", doc=InputDocumentation(doc="do not allow longer gaps -for RNA-RNA alignments- (default 0)")), ToolInput(tag="in_paired", input_type=Boolean(optional=True), prefix="--paired", doc=InputDocumentation(doc="require that paired reads are on opposite strands of same target\n(default 0). NOTE: see prerequisite section above.")), ToolInput(tag="in_uniq", input_type=Boolean(optional=True), prefix="--uniq", doc=InputDocumentation(doc="take only best match, iff, second best is much worse  (default 0)")), ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="--verbose", doc=InputDocumentation(doc="output debugging info (default 0)")), ToolInput(tag="in_insert_limit", input_type=Int(optional=True), prefix="--insertLimit", doc=InputDocumentation(doc="maximum assumed size of inserts (default 10)")), ToolInput(tag="in_max_intron_len", input_type=Int(optional=True), prefix="--maxIntronLen", doc=InputDocumentation(doc="maximal separation of paired reads (default 500000)")), ToolInput(tag="in_max_sortes_test", input_type=Int(optional=True), prefix="--maxSortesTest", doc=InputDocumentation(doc="maximal sortedness (default 100000)")), ToolInput(tag="in_min_cover", input_type=Int(optional=True), prefix="--minCover", doc=InputDocumentation(doc="minimal percentage of coverage of the query read (default 80)")), ToolInput(tag="in_mini_d", input_type=Int(optional=True), prefix="--minId", doc=InputDocumentation(doc="minimal percentage of identity (default 92)")), ToolInput(tag="in_min_intron_len", input_type=Int(optional=True), prefix="--minIntronLen", doc=InputDocumentation(doc="minimal     ''     ''   ''    ''   (default 35)")), ToolInput(tag="in_uniq_thresh", input_type=Float(optional=True), prefix="--uniqThresh", doc=InputDocumentation(doc="threshold % for uniq, second best must be at most this\nfraction of best (default 0.96)")), ToolInput(tag="in_common_gene_file", input_type=File(optional=True), prefix="--commonGeneFile", doc=InputDocumentation(doc="file name in which to write cases where one read maps to\nseveral different genes")), ToolInput(tag="in_pair_bed_file", input_type=Int(optional=True), prefix="--pairBedFile", doc=InputDocumentation(doc="file name of pairedness coverage:\noptions:\na .bed format file in which for each position the number of\nfiltered read pairs is reported that contain the position in\nor between the reads")), ToolInput(tag="in_pairwise_alignments", input_type=Boolean(optional=True), prefix="--pairwiseAlignments", doc=InputDocumentation(doc="use in case alignments were done in pairwise fashion (default:  0)"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Filterbam_V0_1_0().translate("wdl", allow_empty_container=True)

