from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Boolean, Int, String

Outrigger_Index_V0_1_0 = CommandToolBuilder(tool="outrigger_index", base_command=["outrigger", "index"], inputs=[ToolInput(tag="in_output", input_type=File(optional=True), prefix="--output", doc=InputDocumentation(doc="Name of the folder where you saved the output from\n'outrigger index' (default is ./outrigger_output,\nwhich is relative to the directory where you called\nthe program)'. You will need this file for the next\nstep, 'outrigger psi'")), ToolInput(tag="in_sjouttab_files_star", input_type=Boolean(optional=True), prefix="-j", doc=InputDocumentation(doc="[SJ_OUT_TAB [SJ_OUT_TAB ...]], --sj-out-tab [SJ_OUT_TAB [SJ_OUT_TAB ...]]\nSJ.out.tab files from STAR aligner output. Not\nrequired if you specify '--compiled-junction-reads'")), ToolInput(tag="in_junction_reads_csv", input_type=File(optional=True), prefix="--junction-reads-csv", doc=InputDocumentation(doc="Name of the splice junction files to detect novel\nexons and build an index of alternative splicing\nevents from. Not required if you specify SJ.out.tab\nfile with '--sj-out-tab'")), ToolInput(tag="in_location_bam_use", input_type=Boolean(optional=True), prefix="-b", doc=InputDocumentation(doc="[BAM [BAM ...]], --bam [BAM [BAM ...]]\nLocation of bam files to use for finding events.")), ToolInput(tag="in_min_reads", input_type=Int(optional=True), prefix="--min-reads", doc=InputDocumentation(doc="Minimum number of reads per junction for that junction\nto count in creating the index of splicing events\n(default=10)")), ToolInput(tag="in_ignore_multi_mapping", input_type=Boolean(optional=True), prefix="--ignore-multimapping", doc=InputDocumentation(doc="Applies to STAR SJ.out.tab files only. If this flag is\nused, then do not include reads that mapped to\nmultiple locations in the genome, not uniquely to a\nlocus, in the read count for a junction. If inputting\n'bam' files, then this means that reads with a mapping\nquality (MAPQ) of less than 255 are considered\n'multimapped.' This is the same thing as what the STAR\naligner does. By default, this is off, and all reads\nare used.")), ToolInput(tag="in_max_de_novo_exon_length", input_type=Int(optional=True), prefix="--max-de-novo-exon-length", doc=InputDocumentation(doc="Maximum length of an exon detected *de novo* from the\ndataset. This is to prevent multiple kilobase long\nexons from being accidentally created. (default=100)")), ToolInput(tag="in_gtf_filename", input_type=File(optional=True), prefix="--gtf-filename", doc=InputDocumentation(doc="Name of the gtf file you want to use. If a gffutils\nfeature database doesn't already exist at this\nlocation plus '.db' (e.g. if your gtf is\ngencode.v19.annotation.gtf, then the database is\ninferred to be gencode.v19.annotation.gtf.db), then a\ndatabase will be auto-created. Not required if you\nprovide a pre-built database with '--gffutils-db'")), ToolInput(tag="in_gff_utils_db", input_type=File(optional=True), prefix="--gffutils-db", doc=InputDocumentation(doc="Name of the gffutils database file you want to use.\nThe exon IDs defined here will be used in the function\nwhen creating splicing event names. Not required if\nyou provide a gtf file with '--gtf-filename'")), ToolInput(tag="in_debug", input_type=Boolean(optional=True), prefix="--debug", doc=InputDocumentation(doc="If given, print debugging logging information to\nstandard out (Warning: LOTS of output. Not recommended\nunless you think something is going wrong)")), ToolInput(tag="in_n_jobs", input_type=Int(optional=True), prefix="--n-jobs", doc=InputDocumentation(doc="Number of threads to use when parallelizing exon\nfinding and file reading. Default is -1, which means\nto use as many threads as are available.")), ToolInput(tag="in_low_memory", input_type=Boolean(optional=True), prefix="--low-memory", doc=InputDocumentation(doc="If set, then use a smaller memory footprint. By\ndefault, this is off.")), ToolInput(tag="in_splice_types", input_type=String(optional=True), prefix="--splice-types", doc=InputDocumentation(doc="Which splice types to find. By default, 'all' are used\nwhich at this point is skipped exon (SE) and mutually\nexclusive exon (MXE) events. Can also specify only\none, e.g. 'se' or both 'se,mxe'")), ToolInput(tag="in_force", input_type=Boolean(optional=True), prefix="--force", doc=InputDocumentation(doc="If the 'outrigger index' command was interrupted,\nthere will be intermediate files remaining. If you\nwish to restart outrigger and overwrite them all, use\nthis flag. If you want to continue from where you left\noff, use the '--resume' flag. If neither is specified,\nthe program exits and complains to the user.")), ToolInput(tag="in_resume", input_type=Boolean(optional=True), prefix="--resume", doc=InputDocumentation(doc="If the 'outrigger index' command was interrupted,\nthere will be intermediate files remaining. If you\nwant to continue from where you left off, use this\nflag. The default action is to do nothing and ask the\nuser for input.\n"))], outputs=[ToolOutput(tag="out_output", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output", type_hint=File()), doc=OutputDocumentation(doc="Name of the folder where you saved the output from\n'outrigger index' (default is ./outrigger_output,\nwhich is relative to the directory where you called\nthe program)'. You will need this file for the next\nstep, 'outrigger psi'")), ToolOutput(tag="out_junction_reads_csv", output_type=File(optional=True), selector=InputSelector(input_to_select="in_junction_reads_csv", type_hint=File()), doc=OutputDocumentation(doc="Name of the splice junction files to detect novel\nexons and build an index of alternative splicing\nevents from. Not required if you specify SJ.out.tab\nfile with '--sj-out-tab'"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Outrigger_Index_V0_1_0().translate("wdl", allow_empty_container=True)

