from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, File

Translatorx_Vlocal_Pl_V0_1_0 = CommandToolBuilder(tool="translatorx_vLocal.pl", base_command=["translatorx_vLocal.pl"], inputs=[ToolInput(tag="in_file_containing_sequences", input_type=Boolean(optional=True), prefix="-i", doc=InputDocumentation(doc=": the file containing the nucleotide sequences in FASTA format (Required)")), ToolInput(tag="in_output_file_optional", input_type=File(optional=True), prefix="-o", doc=InputDocumentation(doc=": output file (Optional). Default: 'translatorx_res'.")), ToolInput(tag="in_file_containing_alignment", input_type=Boolean(optional=True), prefix="-a", doc=InputDocumentation(doc=": file containing the amino acid sequence alignment (Optional)")), ToolInput(tag="in_program_build_alignment", input_type=Boolean(optional=True), prefix="-p", doc=InputDocumentation(doc=": program to build the multiple alignment (Optional). Available options are:\nM/C/F/P, standing for Muscle, Clustalw, Prank, and maFft\nDefault: Muscle")), ToolInput(tag="in_genetic_code_translate", input_type=Boolean(optional=True), prefix="-c", doc=InputDocumentation(doc=": genetic code to translate the sequences (Optional). Available options are:\n1   Standard\n2   Vertebrate Mitochondrial\n3   Yeast Mitochondrial\n4   Mold Mitochondrial; Protozoan Mitochondrial; Coelenterate Mitochondrial; Mycoplasma; Spiroplasma\n5   Invertebrate Mitochondrial\n6   Ciliate Nuclear; Dasycladacean Nuclear; Hexamita Nuclear\n9   Echinoderm Mitochondrial; Flatworm Mitochondrial\n10  Euplotid Nuclear\n11  Bacterial and Plant Plastid\n12  Alternative Yeast Nuclear\n13  Ascidian Mitochondrial\n14  Alternative Flatworm Mitochondrial\n15  Blepharisma Macronuclear\n16  Chlorophycean Mitochondrial\n21  Trematode Mitochondrial\n22  Scenedesmus obliquus Mitochondrial\n23  Thraustochytrium Mitochondrial\n100 Ancestral Arthropod Mitochondrial Code (AGG=K)\n101 Hemichordate Mitochondrial\nDefault: 1 (Standard code)")), ToolInput(tag="in_file_containing_codes", input_type=Boolean(optional=True), prefix="-j", doc=InputDocumentation(doc=": file containing alternative genetic codes for each taxon. (Optional)\nFile format: Taxon        genetic_code[number]")), ToolInput(tag="in_parameters_gblocks_g", input_type=Boolean(optional=True), prefix="-g", doc=InputDocumentation(doc=": parameters for GBlocks (Optional)\nExample: -g '-b2 x -b3 x -b4 x...'")), ToolInput(tag="in_ft_guess_reading", input_type=Boolean(optional=True), prefix="-t", doc=InputDocumentation(doc=": F/T. Guess reading frame (Optional)[default F]"))], outputs=[ToolOutput(tag="out_output_file_optional", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output_file_optional", type_hint=File()), doc=OutputDocumentation(doc=": output file (Optional). Default: 'translatorx_res'."))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Translatorx_Vlocal_Pl_V0_1_0().translate("wdl", allow_empty_container=True)

