version 1.0

task Anvidisplaystructure {
  input {
    String? profile_db
    String? contigs_db
    String? variability_profile
    File? splits_of_interest
    String? collection_name
    String? bin_id
    File? samples_of_interest
    File? genes_of_interest
    String? gene_caller_ids
    Float? min_departure_from_consensus
    Boolean? sa_avs_only
    Boolean? s_cvs_only
    Float? ip_address
    Int? port_number
    File? browser_path
    Boolean? server_only
    Boolean? password_protected
    String? s
    String an_vi_gen_structure_database_dot
  }
  command <<<
    anvi_display_structure \
      ~{an_vi_gen_structure_database_dot} \
      ~{if defined(profile_db) then ("--profile-db " +  '"' + profile_db + '"') else ""} \
      ~{if defined(contigs_db) then ("--contigs-db " +  '"' + contigs_db + '"') else ""} \
      ~{if defined(variability_profile) then ("--variability-profile " +  '"' + variability_profile + '"') else ""} \
      ~{if defined(splits_of_interest) then ("--splits-of-interest " +  '"' + splits_of_interest + '"') else ""} \
      ~{if defined(collection_name) then ("--collection-name " +  '"' + collection_name + '"') else ""} \
      ~{if defined(bin_id) then ("--bin-id " +  '"' + bin_id + '"') else ""} \
      ~{if defined(samples_of_interest) then ("--samples-of-interest " +  '"' + samples_of_interest + '"') else ""} \
      ~{if defined(genes_of_interest) then ("--genes-of-interest " +  '"' + genes_of_interest + '"') else ""} \
      ~{if defined(gene_caller_ids) then ("--gene-caller-ids " +  '"' + gene_caller_ids + '"') else ""} \
      ~{if defined(min_departure_from_consensus) then ("--min-departure-from-consensus " +  '"' + min_departure_from_consensus + '"') else ""} \
      ~{if (sa_avs_only) then "--SAAVs-only" else ""} \
      ~{if (s_cvs_only) then "--SCVs-only" else ""} \
      ~{if defined(ip_address) then ("--ip-address " +  '"' + ip_address + '"') else ""} \
      ~{if defined(port_number) then ("--port-number " +  '"' + port_number + '"') else ""} \
      ~{if defined(browser_path) then ("--browser-path " +  '"' + browser_path + '"') else ""} \
      ~{if (server_only) then "--server-only" else ""} \
      ~{if (password_protected) then "--password-protected" else ""} \
      ~{if defined(s) then ("-s " +  '"' + s + '"') else ""}
  >>>
  runtime {
    docker: "None"
  }
  parameter_meta {
    profile_db: "Anvi'o profile database"
    contigs_db: "Anvi'o contigs database generated by 'anvi-gen-\\ncontigs'"
    variability_profile: "The output of anvi-gen-variability-profile, or a\\ndifferent variant-calling output that has been\\nconverted to the anvi'o format."
    splits_of_interest: "A file with split names. There should be only one\\ncolumn in the file, and each line should correspond to\\na unique split name."
    collection_name: "Collection name."
    bin_id: "If provided, any genes found in both your bin and your\\nstructure database will be available for display."
    samples_of_interest: "A file with samples names. There should be only one\\ncolumn in the file, and each line should correspond to\\na unique sample name (without a column header)."
    genes_of_interest: "A file with anvi'o gene caller IDs. There should be\\nonly one column in the file, and each line should\\ncorrespond to a unique gene caller id (without a\\ncolumn header)."
    gene_caller_ids: "Gene caller ids. Multiple of them can be declared\\nseparated by a delimiter (the default is a comma). In\\nanvi-gen-variability-profile, if you declare nothing\\nyou will get all genes matching your other filtering\\ncriteria. In other programs, you may get everything,\\nnothing, or an error. It really depends on the\\nsituation. Fortunately, mistakes are cheap, so it's\\nworth a try."
    min_departure_from_consensus: "Takes a value between 0 and 1, where 1 is maximum\\ndivergence from the consensus. it can be an expensive\\noperation to display every variable position, and so\\nthe default is 0.05. To display every variable\\nposition, set this parameter to 0"
    sa_avs_only: "If provided, variability will be generated for single\\namino acid variants (SAAVs) and not for single codon\\nvariants (SCVs). This could save you some time if\\nyou're only interested in SAAVs."
    s_cvs_only: "If provided, variability will be generated for single\\ncodon variants (SCVs) and not for single amino acid\\nvariants (SAAVs). This could save you some time if\\nyou're only interested in SCVs."
    ip_address: "IP address for the HTTP server. The default ip address\\n(0.0.0.0) should work just fine for most."
    port_number: "Port number to use for anvi'o services. If nothing is\\ndeclared, anvi'o will try to find a suitable port\\nnumber, starting from the default port number, 8080."
    browser_path: "By default, anvi'o will use your default browser to\\nlaunch the interactive interface. If you would like to\\nuse something else than your system default, you can\\nprovide a full path for an alternative browser using\\nthis parameter, and hope for the best. For instance we\\nare using this parameter to call Google's experimental\\nbrowser, Canary, which performs better with demanding\\nvisualizations."
    server_only: "The default behavior is to start the local server, and\\nfire up a browser that connects to the server. If you\\nhave other plans, and want to start the server without\\ncalling the browser, this is the flag you need."
    password_protected: "If this flag is set, command line tool will ask you to\\nenter a password and interactive interface will be\\nonly accessible after entering same password. This\\noption is recommended for shared machines like\\nclusters or shared networks where computers are not\\nisolated.\\n"
    s: ""
    an_vi_gen_structure_database_dot: "-s STRUCTURE_DB, --structure-db STRUCTURE_DB\\nAnvi'o structure database."
  }
  output {
    File out_stdout = stdout()
  }
}