from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Directory, File, String, Float, Int, Boolean

Somaticseq_Parallel_Py_V0_1_0 = CommandToolBuilder(tool="somaticseq_parallel.py", base_command=["somaticseq_parallel.py"], inputs=[ToolInput(tag="in_output_directory", input_type=Directory(optional=True), prefix="--output-directory", doc=InputDocumentation(doc="output directory (default: .)")), ToolInput(tag="in_genome_reference", input_type=File(optional=True), prefix="--genome-reference", doc=InputDocumentation(doc=".fasta.fai file to get the contigs (default: None)")), ToolInput(tag="in_truth_snv", input_type=String(optional=True), prefix="--truth-snv", doc=InputDocumentation(doc="VCF of true hits (default: None)")), ToolInput(tag="in_truth_in_del", input_type=String(optional=True), prefix="--truth-indel", doc=InputDocumentation(doc="VCF of true hits (default: None)")), ToolInput(tag="in_classifier_snv", input_type=String(optional=True), prefix="--classifier-snv", doc=InputDocumentation(doc="RData for SNV (default: None)")), ToolInput(tag="in_classifier_in_del", input_type=String(optional=True), prefix="--classifier-indel", doc=InputDocumentation(doc="RData for INDEL (default: None)")), ToolInput(tag="in_pass_threshold", input_type=Float(optional=True), prefix="--pass-threshold", doc=InputDocumentation(doc="SCORE for PASS (default: 0.5)")), ToolInput(tag="in_low_qual_threshold", input_type=Float(optional=True), prefix="--lowqual-threshold", doc=InputDocumentation(doc="SCORE for LowQual (default: 0.1)")), ToolInput(tag="in_algorithm", input_type=String(optional=True), prefix="--algorithm", doc=InputDocumentation(doc="ada or xgboost (default: xgboost)")), ToolInput(tag="in_homozygous_threshold", input_type=Float(optional=True), prefix="--homozygous-threshold", doc=InputDocumentation(doc="VAF for homozygous (default: 0.85)")), ToolInput(tag="in_heterozygous_threshold", input_type=Float(optional=True), prefix="--heterozygous-threshold", doc=InputDocumentation(doc="VAF for heterozygous (default: 0.01)")), ToolInput(tag="in_minimum_mapping_quality", input_type=Int(optional=True), prefix="--minimum-mapping-quality", doc=InputDocumentation(doc="Minimum mapping quality below which is considered poor\n(default: 1)")), ToolInput(tag="in_minimum_base_quality", input_type=Int(optional=True), prefix="--minimum-base-quality", doc=InputDocumentation(doc="Minimum base quality below which is considered poor\n(default: 5)")), ToolInput(tag="in_minimum_num_callers", input_type=Int(optional=True), prefix="--minimum-num-callers", doc=InputDocumentation(doc="Minimum number of tools to be considered (default:\n0.5)")), ToolInput(tag="in_dbsnp_vcf", input_type=String(optional=True), prefix="--dbsnp-vcf", doc=InputDocumentation(doc="dbSNP VCF (default: None)")), ToolInput(tag="in_cosmic_vcf", input_type=String(optional=True), prefix="--cosmic-vcf", doc=InputDocumentation(doc="COSMIC VCF (default: None)")), ToolInput(tag="in_inclusion_region", input_type=String(optional=True), prefix="--inclusion-region", doc=InputDocumentation(doc="inclusion bed (default: None)")), ToolInput(tag="in_exclusion_region", input_type=String(optional=True), prefix="--exclusion-region", doc=InputDocumentation(doc="exclusion bed (default: None)")), ToolInput(tag="in_threads", input_type=Int(optional=True), prefix="--threads", doc=InputDocumentation(doc="number of threads (default: 1)")), ToolInput(tag="in_somatic_seq_train", input_type=Boolean(optional=True), prefix="--somaticseq-train", doc=InputDocumentation(doc="Invoke training mode with ground truths (default:\nFalse)")), ToolInput(tag="in_seed", input_type=Int(optional=True), prefix="--seed", doc=InputDocumentation(doc="seed for xgboost training (default: 0)")), ToolInput(tag="in_tree_depth", input_type=Int(optional=True), prefix="--tree-depth", doc=InputDocumentation(doc="max tree depth for xgboost training (default: 12)")), ToolInput(tag="in_iterations", input_type=Int(optional=True), prefix="--iterations", doc=InputDocumentation(doc="num boosting rounds for xgboost: default is 500 for\ntraining and 100 for predicting, i.e., by default, 500\ntrees are built for classifier, but only the first 100\ntrees are used. (default: None)")), ToolInput(tag="in_features_excluded", input_type=Boolean(optional=True), prefix="--features-excluded", doc=InputDocumentation(doc="[FEATURES_EXCLUDED [FEATURES_EXCLUDED ...]]\nfeatures to exclude for xgboost training. Must be same\nfor train/predict. (default: [])")), ToolInput(tag="in_keep_intermediates", input_type=Boolean(optional=True), prefix="--keep-intermediates", doc=InputDocumentation(doc="Keep intermediate files (default: False)"))], outputs=[ToolOutput(tag="out_output_directory", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_output_directory", type_hint=File()), doc=OutputDocumentation(doc="output directory (default: .)"))], container="quay.io/biocontainers/somaticseq:3.6.2--py_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Somaticseq_Parallel_Py_V0_1_0().translate("wdl")

