from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, Directory, Int, String

Funannotate_Predict_V0_1_0 = CommandToolBuilder(tool="funannotate_predict", base_command=["funannotate", "predict"], inputs=[ToolInput(tag="in_input", input_type=Boolean(optional=True), prefix="--input", doc=InputDocumentation(doc="Genome multi-FASTA file (softmasked repeats)")), ToolInput(tag="in_out", input_type=Directory(optional=True), prefix="--out", doc=InputDocumentation(doc="Output folder name")), ToolInput(tag="in_species", input_type=Boolean(optional=True), prefix="--species", doc=InputDocumentation(doc="Species name, use quotes for binomial, e.g. 'Aspergillus fumigatus'")), ToolInput(tag="in_parameters", input_type=Boolean(optional=True), prefix="--parameters", doc=InputDocumentation(doc="Ab intio parameters JSON file to use for gene predictors")), ToolInput(tag="in_isolate", input_type=Boolean(optional=True), prefix="--isolate", doc=InputDocumentation(doc="Isolate name, e.g. Af293")), ToolInput(tag="in_strain", input_type=Boolean(optional=True), prefix="--strain", doc=InputDocumentation(doc="Strain name, e.g. FGSCA4")), ToolInput(tag="in_name", input_type=Boolean(optional=True), prefix="--name", doc=InputDocumentation(doc="Locus tag name (assigned by NCBI?). Default: FUN_")), ToolInput(tag="in_numbering", input_type=Boolean(optional=True), prefix="--numbering", doc=InputDocumentation(doc="Specify where gene numbering starts. Default: 1")), ToolInput(tag="in_maker_gff", input_type=Boolean(optional=True), prefix="--maker_gff", doc=InputDocumentation(doc="MAKER2 GFF file. Parse results directly to EVM.")), ToolInput(tag="in_pasa_gff", input_type=Boolean(optional=True), prefix="--pasa_gff", doc=InputDocumentation(doc="PASA generated gene models. filename:weight")), ToolInput(tag="in_other_gff", input_type=Boolean(optional=True), prefix="--other_gff", doc=InputDocumentation(doc="Annotation pass-through to EVM. filename:weight")), ToolInput(tag="in_rna_bam", input_type=Boolean(optional=True), prefix="--rna_bam", doc=InputDocumentation(doc="RNA-seq mapped to genome to train Augustus/GeneMark-ET")), ToolInput(tag="in_string_tie", input_type=Boolean(optional=True), prefix="--stringtie", doc=InputDocumentation(doc="StringTie GTF result")), ToolInput(tag="in_weights", input_type=Boolean(optional=True), prefix="--weights", doc=InputDocumentation(doc="Ab-initio predictor and EVM weight. Example: augustus:2 or pasa:10")), ToolInput(tag="in_augustus_species", input_type=Boolean(optional=True), prefix="--augustus_species", doc=InputDocumentation(doc="Augustus species config. Default: uses species name")), ToolInput(tag="in_min_training_models", input_type=Boolean(optional=True), prefix="--min_training_models", doc=InputDocumentation(doc="Minimum number of models to train Augustus. Default: 200")), ToolInput(tag="in_genemark_mode", input_type=Boolean(optional=True), prefix="--genemark_mode", doc=InputDocumentation(doc="GeneMark mode. Default: ES [ES,ET]")), ToolInput(tag="in_genemark_mod", input_type=Boolean(optional=True), prefix="--genemark_mod", doc=InputDocumentation(doc="GeneMark ini mod file")), ToolInput(tag="in_busco_seed_species", input_type=Boolean(optional=True), prefix="--busco_seed_species", doc=InputDocumentation(doc="Augustus pre-trained species to start BUSCO. Default: anidulans")), ToolInput(tag="in_optimize_augustus", input_type=Boolean(optional=True), prefix="--optimize_augustus", doc=InputDocumentation(doc="Run 'optimze_augustus.pl' to refine training (long runtime)")), ToolInput(tag="in_busco_db", input_type=Boolean(optional=True), prefix="--busco_db", doc=InputDocumentation(doc="BUSCO models. Default: dikarya. `funannotate outgroups --show_buscos`")), ToolInput(tag="in_organism", input_type=Boolean(optional=True), prefix="--organism", doc=InputDocumentation(doc="Fungal-specific options. Default: fungus. [fungus,other]")), ToolInput(tag="in_ploidy", input_type=Boolean(optional=True), prefix="--ploidy", doc=InputDocumentation(doc="Ploidy of assembly. Default: 1")), ToolInput(tag="in_tbl_two_asn", input_type=Boolean(optional=True), prefix="--tbl2asn", doc=InputDocumentation(doc="Assembly parameters for tbl2asn. Default: '-l paired-ends'")), ToolInput(tag="in_database", input_type=Boolean(optional=True), prefix="--database", doc=InputDocumentation(doc="Path to funannotate database. Default: $FUNANNOTATE_DB")), ToolInput(tag="in_protein_evidence", input_type=Boolean(optional=True), prefix="--protein_evidence", doc=InputDocumentation(doc="Proteins to map to genome (prot1.fa prot2.fa uniprot.fa). Default: uniprot.fa")), ToolInput(tag="in_protein_alignments", input_type=Boolean(optional=True), prefix="--protein_alignments", doc=InputDocumentation(doc="Pre-computed protein alignments in GFF3 format")), ToolInput(tag="in_p_two_g_p_ident", input_type=Boolean(optional=True), prefix="--p2g_pident", doc=InputDocumentation(doc="Exonerate percent identity. Default: 80")), ToolInput(tag="in_p_two_g_diamond_db", input_type=Boolean(optional=True), prefix="--p2g_diamond_db", doc=InputDocumentation(doc="Premade diamond genome database for protein2genome mapping")), ToolInput(tag="in_p_two_g_prefilter", input_type=Boolean(optional=True), prefix="--p2g_prefilter", doc=InputDocumentation(doc="Pre-filter hits software selection. Default: diamond [tblastn]")), ToolInput(tag="in_transcript_evidence", input_type=Boolean(optional=True), prefix="--transcript_evidence", doc=InputDocumentation(doc="mRNA/ESTs to align to genome (trans1.fa ests.fa trinity.fa). Default: none")), ToolInput(tag="in_transcript_alignments", input_type=Boolean(optional=True), prefix="--transcript_alignments", doc=InputDocumentation(doc="Pre-computed transcript alignments in GFF3 format")), ToolInput(tag="in_augustus_gff", input_type=Boolean(optional=True), prefix="--augustus_gff", doc=InputDocumentation(doc="Pre-computed AUGUSTUS GFF3 results (must use --stopCodonExcludedFromCDS=False)")), ToolInput(tag="in_genemark_gtf", input_type=Boolean(optional=True), prefix="--genemark_gtf", doc=InputDocumentation(doc="Pre-computed GeneMark GTF results")), ToolInput(tag="in_trnascan", input_type=Boolean(optional=True), prefix="--trnascan", doc=InputDocumentation(doc="Pre-computed tRNAscanSE results")), ToolInput(tag="in_min_intron_len", input_type=Boolean(optional=True), prefix="--min_intronlen", doc=InputDocumentation(doc="Minimum intron length. Default: 10")), ToolInput(tag="in_max_intron_len", input_type=Boolean(optional=True), prefix="--max_intronlen", doc=InputDocumentation(doc="Maximum intron length. Default: 3000")), ToolInput(tag="in_soft_mask", input_type=Boolean(optional=True), prefix="--soft_mask", doc=InputDocumentation(doc="Softmasked length threshold for GeneMark. Default: 2000")), ToolInput(tag="in_min_prot_len", input_type=Boolean(optional=True), prefix="--min_protlen", doc=InputDocumentation(doc="Minimum protein length. Default: 50")), ToolInput(tag="in_repeats_two_evm", input_type=Boolean(optional=True), prefix="--repeats2evm", doc=InputDocumentation(doc="Use repeats in EVM consensus model building")), ToolInput(tag="in_keep_evm", input_type=Boolean(optional=True), prefix="--keep_evm", doc=InputDocumentation(doc="Keep existing EVM results (for rerunning pipeline)")), ToolInput(tag="in_evm_partition_interval", input_type=Int(optional=True), prefix="--evm-partition-interval", doc=InputDocumentation(doc="length between genes to make a partition: Default: 1500")), ToolInput(tag="in_no_evm_partitions", input_type=Boolean(optional=True), prefix="--no-evm-partitions", doc=InputDocumentation(doc="Do not split contigs into partitions")), ToolInput(tag="in_repeat_filter", input_type=Boolean(optional=True), prefix="--repeat_filter", doc=InputDocumentation(doc="Repetitive gene model filtering. Default: overlap blast [overlap,blast,none]")), ToolInput(tag="in_keep_no_stops", input_type=Boolean(optional=True), prefix="--keep_no_stops", doc=InputDocumentation(doc="Keep gene models without valid stops")), ToolInput(tag="in_seq_center", input_type=Boolean(optional=True), prefix="--SeqCenter", doc=InputDocumentation(doc="Sequencing facilty for NCBI tbl file. Default: CFMR")), ToolInput(tag="in_seq_accession", input_type=Boolean(optional=True), prefix="--SeqAccession", doc=InputDocumentation(doc="Sequence accession number for NCBI tbl file. Default: 12345")), ToolInput(tag="in_force", input_type=Boolean(optional=True), prefix="--force", doc=InputDocumentation(doc="Annotated unmasked genome")), ToolInput(tag="in_cpus", input_type=Boolean(optional=True), prefix="--cpus", doc=InputDocumentation(doc="Number of CPUs to use. Default: 2")), ToolInput(tag="in_no_progress", input_type=Boolean(optional=True), prefix="--no-progress", doc=InputDocumentation(doc="Do not print progress to stdout for long sub jobs")), ToolInput(tag="in_arguments", input_type=String(), position=0, doc=InputDocumentation(doc=""))], outputs=[ToolOutput(tag="out_out", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_out", type_hint=File()), doc=OutputDocumentation(doc="Output folder name"))], container="quay.io/biocontainers/funannotate:1.8.5--pyh3252c3a_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Funannotate_Predict_V0_1_0().translate("wdl")

