from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import String, Int, Boolean

Traitar_Phenotype_V0_1_0 = CommandToolBuilder(tool="traitar_phenotype", base_command=["traitar", "phenotype"], inputs=[ToolInput(tag="in_gene_gff_type", input_type=String(optional=True), prefix="--gene_gff_type", doc=InputDocumentation(doc="if the input is amino acids the type of gene\nprediction GFF file can be specified for mapping the\nphenotype predictions to the genes")), ToolInput(tag="in_cpus", input_type=Int(optional=True), prefix="--cpus", doc=InputDocumentation(doc="number of cpus used for the individual steps; maximum\nis number of samples; needs parallel")), ToolInput(tag="in_primary_models", input_type=String(optional=True), prefix="--primary_models", doc=InputDocumentation(doc="primary phenotype models archive")), ToolInput(tag="in_secondary_models", input_type=String(optional=True), prefix="--secondary_models", doc=InputDocumentation(doc="secondary phenotype models archive")), ToolInput(tag="in_rearrange_heat_map", input_type=String(optional=True), prefix="--rearrange_heatmap", doc=InputDocumentation(doc="recompute the phenotype heatmaps based on a subset of\npreviously annotated and phenotyped samples")), ToolInput(tag="in_no_heat_map_sample_clustering", input_type=Boolean(optional=True), prefix="--no_heatmap_sample_clustering", doc=InputDocumentation(doc="if option is set, don't cluster the phenotype heatmaps\nby samples and keep input ordering")), ToolInput(tag="in_no_heat_map_phenotype_clustering", input_type=Boolean(optional=True), prefix="--no_heatmap_phenotype_clustering", doc=InputDocumentation(doc="if option is set, don't cluster the heatmaps by\nphenotype and keep input ordering")), ToolInput(tag="in_heat_map_format", input_type=String(optional=True), prefix="--heatmap_format", doc=InputDocumentation(doc="choose file format for the heatmap\n")), ToolInput(tag="in_positional_arguments", input_type=String(), position=0, doc=InputDocumentation(doc="positional arguments:")), ToolInput(tag="in_input_dir", input_type=String(), position=1, doc=InputDocumentation(doc="directory with the input data")), ToolInput(tag="in_sample_two_file", input_type=Int(), position=2, doc=InputDocumentation(doc="mapping from samples to fasta files (also see gitHub\nhelp): sample1_file_name{tab}sample1_name\nsample2_file_name{tab}sample2_name")), ToolInput(tag="in_directory_output_created", input_type=String(), position=0, doc=InputDocumentation(doc="directory for the output; will be created if it\ndoesn't exist yet"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Traitar_Phenotype_V0_1_0().translate("wdl", allow_empty_container=True)

