from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean

Purge_Haplotigs_Purge_V0_1_0 = CommandToolBuilder(tool="purge_haplotigs_purge", base_command=["purge_haplotigs", "purge"], inputs=[ToolInput(tag="in_genome", input_type=Boolean(optional=True), prefix="-genome", doc=InputDocumentation(doc="Genome assembly in fasta format. Needs to be indexed with samtools faidx.")), ToolInput(tag="in_coverage", input_type=Boolean(optional=True), prefix="-coverage", doc=InputDocumentation(doc="Contig by contig coverage stats csv file from the previous step.")), ToolInput(tag="in_threads", input_type=Boolean(optional=True), prefix="-threads", doc=InputDocumentation(doc="Number of worker threads to use. DEFAULT = 4")), ToolInput(tag="in_out_prefix", input_type=Boolean(optional=True), prefix="-outprefix", doc=InputDocumentation(doc="Prefix for the curated assembly. DEFAULT = 'curated'")), ToolInput(tag="in_repeats", input_type=Boolean(optional=True), prefix="-repeats", doc=InputDocumentation(doc="BED-format file of repeats to ignore during analysis.")), ToolInput(tag="in_dot_plots", input_type=Boolean(optional=True), prefix="-dotplots", doc=InputDocumentation(doc="Generate dotplots for manual inspection.")), ToolInput(tag="in_bam", input_type=Boolean(optional=True), prefix="-bam", doc=InputDocumentation(doc="Samtools-indexed bam file of aligned and sorted reads/subreads to the\nreference, required for generating dotplots.")), ToolInput(tag="in_falcon_naming", input_type=Boolean(optional=True), prefix="-falconNaming", doc=InputDocumentation(doc="Rename contigs in the style used by FALCON/FALCON-unzip")), ToolInput(tag="in_align_cov", input_type=Boolean(optional=True), prefix="-align_cov", doc=InputDocumentation(doc="Percent cutoff for identifying a contig as a haplotig. DEFAULT = 70")), ToolInput(tag="in_max_match", input_type=Boolean(optional=True), prefix="-max_match", doc=InputDocumentation(doc="Percent cutoff for identifying repetitive contigs. Ignored when\nusing repeat annotations (-repeats). DEFAULT = 250")), ToolInput(tag="in_minimap_indexing_drop", input_type=Boolean(optional=True), prefix="-I", doc=InputDocumentation(doc="Minimap2 indexing, drop minimisers every N bases, DEFAULT = 4G")), ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="-verbose", doc=InputDocumentation(doc="Print EVERYTHING.")), ToolInput(tag="in_limit_io", input_type=Boolean(optional=True), prefix="-limit_io", doc=InputDocumentation(doc="Limit for I/O intensive jobs. DEFAULT = -threads")), ToolInput(tag="in_wind_min", input_type=Boolean(optional=True), prefix="-wind_min", doc=InputDocumentation(doc="Min window size for BED coverage plots (for dotplots). DEFAULT = 5000")), ToolInput(tag="in_wind_nmax", input_type=Boolean(optional=True), prefix="-wind_nmax", doc=InputDocumentation(doc="Max windows per contig for BED coverage plots (for dotplots). DEFAULT = 200"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Purge_Haplotigs_Purge_V0_1_0().translate("wdl", allow_empty_container=True)

