class: CommandLineTool
id: anvi_export_gene_calls.cwl
inputs:
- id: in_contigs_db
  doc: "Anvi'o contigs database generated by 'anvi-gen-\ncontigs'"
  type: string?
  inputBinding:
    prefix: --contigs-db
- id: in_output_file
  doc: File path to store results.
  type: File?
  inputBinding:
    prefix: --output-file
- id: in_gene_caller
  doc: "Which gene caller(s) would you like to export gene\ncalls for? If providing\
    \ multiple they should be comma-\nseparated (no spaces). If you don't know, use\
    \ --list-\ngene-callers"
  type: string?
  inputBinding:
    prefix: --gene-caller
- id: in_list_gene_callers
  doc: "List available gene callers in the contigs database\nand quit."
  type: boolean?
  inputBinding:
    prefix: --list-gene-callers
- id: in_skip_sequence_reporting
  doc: "By default, exported gene calls have an amino acid\nsequences column in the\
    \ output. Turn this behavior off\nwith this flag\n"
  type: boolean?
  inputBinding:
    prefix: --skip-sequence-reporting
outputs:
- id: out_stdout
  doc: Standard output stream
  type: stdout
- id: out_output_file
  doc: File path to store results.
  type: File?
  outputBinding:
    glob: $(inputs.in_output_file)
hints: []
cwlVersion: v1.1
baseCommand:
- anvi-export-gene-calls
