version 1.0

task Anviexportgenecalls {
  input {
    String? contigs_db
    File? output_file
    String? gene_caller
    Boolean? list_gene_callers
    Boolean? skip_sequence_reporting
  }
  command <<<
    anvi_export_gene_calls \
      ~{if defined(contigs_db) then ("--contigs-db " +  '"' + contigs_db + '"') else ""} \
      ~{if defined(output_file) then ("--output-file " +  '"' + output_file + '"') else ""} \
      ~{if defined(gene_caller) then ("--gene-caller " +  '"' + gene_caller + '"') else ""} \
      ~{if (list_gene_callers) then "--list-gene-callers" else ""} \
      ~{if (skip_sequence_reporting) then "--skip-sequence-reporting" else ""}
  >>>
  parameter_meta {
    contigs_db: "Anvi'o contigs database generated by 'anvi-gen-\\ncontigs'"
    output_file: "File path to store results."
    gene_caller: "Which gene caller(s) would you like to export gene\\ncalls for? If providing multiple they should be comma-\\nseparated (no spaces). If you don't know, use --list-\\ngene-callers"
    list_gene_callers: "List available gene callers in the contigs database\\nand quit."
    skip_sequence_reporting: "By default, exported gene calls have an amino acid\\nsequences column in the output. Turn this behavior off\\nwith this flag\\n"
  }
  output {
    File out_stdout = stdout()
    File out_output_file = "${in_output_file}"
  }
}