class: CommandLineTool
id: NanoComp.cwl
inputs:
- id: in_threads
  doc: Set the allowed number of threads to be used by the script
  type: long?
  inputBinding:
    prefix: --threads
- id: in_outdir
  doc: Specify directory in which output has to be created.
  type: Directory?
  inputBinding:
    prefix: --outdir
- id: in_prefix
  doc: Specify an optional prefix to be used for the output files.
  type: string?
  inputBinding:
    prefix: --prefix
- id: in_verbose
  doc: Write log messages also to terminal.
  type: boolean?
  inputBinding:
    prefix: --verbose
- id: in_raw
  doc: Store the extracted data in tab separated file.
  type: boolean?
  inputBinding:
    prefix: --raw
- id: in_store
  doc: Store the extracted data in a pickle file for future plotting.
  type: boolean?
  inputBinding:
    prefix: --store
- id: in_tsv_stats
  doc: Output the stats file as a properly formatted TSV.
  type: File?
  inputBinding:
    prefix: --tsv_stats
- id: in_read_type
  doc: "Which read type to extract information about from summary. Options are 1D,\
    \ 2D,\n1D2"
  type: string?
  inputBinding:
    prefix: --readtype
- id: in_maxlength
  doc: Drop reads longer than length specified.
  type: long?
  inputBinding:
    prefix: --maxlength
- id: in_minlength
  doc: Drop reads shorter than length specified.
  type: long?
  inputBinding:
    prefix: --minlength
- id: in_barcoded
  doc: Barcoded experiment in summary format, splitting per barcode.
  type: boolean?
  inputBinding:
    prefix: --barcoded
- id: in_split_runs
  doc: "File: Split the summary on run IDs and use names in tsv file. Mandatory header\n\
    fields are 'NAME' and 'RUN_ID'."
  type: File?
  inputBinding:
    prefix: --split_runs
- id: in_format
  doc: Specify the output format of the plots.
  type: string?
  inputBinding:
    prefix: --format
- id: in_names
  doc: Specify the names to be used for the datasets
  type: string[]
  inputBinding:
    prefix: --names
- id: in_colors
  doc: Specify the colors to be used for the datasets
  type: string[]
  inputBinding:
    prefix: --colors
- id: in_plot
  doc: "Which plot type to use: 'box', 'violin' (default), 'ridge' (joyplot) or 'false'\n\
    (no plots)"
  type: string?
  inputBinding:
    prefix: --plot
- id: in_title
  doc: Add a title to all plots, requires quoting if using spaces
  type: string?
  inputBinding:
    prefix: --title
- id: in_dpi
  doc: Set the dpi for saving images
  type: string?
  inputBinding:
    prefix: --dpi
- id: in_fast_a
  doc: Data is in (compressed) fasta format.
  type: File[]
  inputBinding:
    prefix: --fasta
- id: in_fast_q
  doc: Data is in (compressed) fastq format.
  type: string[]
  inputBinding:
    prefix: --fastq
- id: in_summary
  doc: Data is in (compressed) summary files generated by albacore or guppy.
  type: string[]
  inputBinding:
    prefix: --summary
- id: in_bam
  doc: Data is in sorted bam files.
  type: string[]
  inputBinding:
    prefix: --bam
- id: in_ub_am
  doc: Data is in one or more unmapped bam file(s).
  type: File[]
  inputBinding:
    prefix: --ubam
- id: in_cram
  doc: Data is in one or more sorted cram file(s).
  type: File[]
  inputBinding:
    prefix: --cram
- id: in_pickle
  doc: Data is in one or more pickle file(s) from using NanoComp/NanoPlot.
  type: File[]
  inputBinding:
    prefix: --pickle
- id: in_feather
  doc: Data is in one or more feather file(s).
  type: File[]
  inputBinding:
    prefix: --feather
- id: in_v
  doc: ''
  type: boolean?
  inputBinding:
    prefix: -v
outputs:
- id: out_stdout
  doc: Standard output stream
  type: stdout
- id: out_outdir
  doc: Specify directory in which output has to be created.
  type: Directory?
  outputBinding:
    glob: $(inputs.in_outdir)
- id: out_tsv_stats
  doc: Output the stats file as a properly formatted TSV.
  type: File?
  outputBinding:
    glob: $(inputs.in_tsv_stats)
hints:
- class: DockerRequirement
  dockerPull: quay.io/biocontainers/nanocomp:1.13.1--py_0
cwlVersion: v1.1
baseCommand:
- NanoComp
