from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Int, Array, String, Boolean, Float, Directory

Riboloco_V0_1_0 = CommandToolBuilder(tool="riboloco", base_command=["riboloco"], inputs=[ToolInput(tag="in_samples", input_type=File(optional=True), prefix="--samples", doc=InputDocumentation(doc="In reference generation mode this may either be a .csv\nfile of samples (you MUST ensure that the file is of\n'.csv' otherwise it will not be recognised) or a\nsingle bed file. In conversion mode it must be a\nsingle bed file. Bed files should be transcriptome-\naligned; only reads in the + strand are used. Bed\nfiles should be 6 column, with transcript_id, start,\nend, and strand in columns 1, 2, 3 and 6 respectively\n(the default output from bedtools' 'bamtobed'). Bed\nfiles can be in .gzip format if desired.")), ToolInput(tag="in_info", input_type=File(optional=True), prefix="--info", doc=InputDocumentation(doc="Info file on transcripts. This should be a tab\nseparated file with details on the CDS within each\ntranscript. It should contain the columns\n'transcript_id', 'cds_start' and 'cds_stop'. The\ncoordinates MUST be 1-based!")), ToolInput(tag="in_transcript_fast_a", input_type=File(optional=True), prefix="--transcript_fasta", doc=InputDocumentation(doc="Fasta file of transcripts")), ToolInput(tag="in_output", input_type=File(optional=True), prefix="--output", doc=InputDocumentation(doc="Output file")), ToolInput(tag="in_orf_file", input_type=File(optional=True), prefix="--orf_file", doc=InputDocumentation(doc="An orf csv generated with riboloco_find_orfs.\nSupplying this will activate more intensive searching\nfor out of frame ribosomes.")), ToolInput(tag="in_min_abundance", input_type=Int(optional=True), prefix="--min_abundance", doc=InputDocumentation(doc="Minimum abundance of read length/frame to be included\nin final output. Default is 0.01. Set to zero to\ndisable. This can me lower than --min_abundance_ref,\nthe rationale being that you want to use abundant read\nlengths to build the reference, but any read length\nthat matches the reference well should be included in\nthe final file.")), ToolInput(tag="in_min_abundance_ref", input_type=Int(optional=True), prefix="--min_abundance_ref", doc=InputDocumentation(doc="Minimum fraction of total reads that a read type must\nrepresent for calculation of a reference offset (using\nstart and stop codon enrichment) to be attempted.\nDefault=0.1 (10pc). Warning - using low values may\npromote inclusion of reads which are primarily out of\nframe. Recommended to keep above 0.05. Read fractions\nare calculated for reads within the annotated CDS -\nUTRs are ignored.")), ToolInput(tag="in_read_type", input_type=Array(t=String(), optional=True), prefix="--read_type", doc=InputDocumentation(doc="Set the read type for which the reference is\ncalculated in reference generation mode.Additionally,\nyou can specify the offset with a colon, eg 28_0:-12.")), ToolInput(tag="in_plot_graphs", input_type=Boolean(optional=True), prefix="--plot_graphs", doc=InputDocumentation(doc="When selected, dislocate plots various graphs and\nheatmaps which may be useful for downstream analysis,\nor to verify accuracy of offset assignments.")), ToolInput(tag="in_min_score", input_type=Int(optional=True), prefix="--min_score", doc=InputDocumentation(doc="The minimum correlation between the reference and the\nRUST ratios for the assigned offset for the file to be\nwritten. Default = 0.7")), ToolInput(tag="in_generate_reference", input_type=Boolean(optional=True), prefix="--generate_reference", doc=InputDocumentation(doc="Activates reference generation mode - use this mode to\nmake a reference before converting bed files to single\nnucleotide resolution")), ToolInput(tag="in_reference", input_type=File(optional=True), prefix="--reference", doc=InputDocumentation(doc="Pre-computed reference file csv, generated by\ndislocate in 'generate reference' mode. Multiple\nreferences can be specified by adding a colon between\nfiles. Optional when running in conversion mode.")), ToolInput(tag="in_allow_out_of_frame", input_type=Boolean(optional=True), prefix="--allow_out_of_frame", doc=InputDocumentation(doc="Allow out of frame offsets to be assigned")), ToolInput(tag="in_conversion_types_list", input_type=Array(t=String(), optional=True), prefix="--conversion_types_list", doc=InputDocumentation(doc="Types to output. Can specify an offset with colon, eg\n28_0:-12. Using this command will cause other types to\nbe ignored, unless you also use --keep_all_valid")), ToolInput(tag="in_keep_all_valid", input_type=Boolean(optional=True), prefix="--keep_all_valid", doc=InputDocumentation(doc="This option (only applicable during conversion mode)\nkeeps all valid read types (i.e. all those that pass\nperiodicity and abundance filters) even when specific\nread lengths and offsets are set.")), ToolInput(tag="in_reference_use_kl", input_type=Boolean(optional=True), prefix="--reference_use_KL", doc=InputDocumentation(doc="Use KL divergence to find best offset during reference\ngeneration")), ToolInput(tag="in_use_kl", input_type=Boolean(optional=True), prefix="--use_KL", doc=InputDocumentation(doc="Use KL divergence to determine best A site offset\nduring assignment")), ToolInput(tag="in_periodicity", input_type=Int(optional=True), prefix="--periodicity", doc=InputDocumentation(doc="Periodicity filter - the minimum ratio of reads in the\nmajor frame to the minor frame for a given read length\nto pass filtering. Default is 2; higher numbers are\nmore stringent. Set to 1 to remove filtering")), ToolInput(tag="in_kl_length", input_type=Int(optional=True), prefix="--kl_length", doc=InputDocumentation(doc="The number of codons to use for KL-based determination\nof offsets. By default = 2 i.e. the P and A sites.")), ToolInput(tag="in_use_stop", input_type=Boolean(optional=True), prefix="--use_stop", doc=InputDocumentation(doc="If argument is used, riboloco will attempt to assign\noffsets based on stop codon as well as the start\ncodon. Riboloco will use this value during reference\ngeneration if either it is consistent with the start\ncodon determined offset, or if no start codon-based\noffset could be determined (e.g. with disomes)")), ToolInput(tag="in_min_a_offset", input_type=Int(optional=True), prefix="--min_A_offset", doc=InputDocumentation(doc="The miniumum offset length from the 3' end of the E\nsite. (Not the A site) Length is measured in nt.\nDefault = 3")), ToolInput(tag="in_max_a_offset", input_type=Int(optional=True), prefix="--max_A_offset", doc=InputDocumentation(doc="The maximum offset length from the 3' end of the E\nsite (not the A site). Length is measured in nt.\nDefault = -22")), ToolInput(tag="in_min_offset", input_type=Int(optional=True), prefix="--min_offset", doc=InputDocumentation(doc="The miniumum which is analysed when plotting Length is\nmeasured in nt. Default = 10")), ToolInput(tag="in_max_offset", input_type=Int(optional=True), prefix="--max_offset", doc=InputDocumentation(doc="The maximum offset which is analysed when plotting.\nLength is measured in nt. Default = -40")), ToolInput(tag="in_ignore", input_type=Int(optional=True), prefix="--ignore", doc=InputDocumentation(doc="Read types to ignore. Can add multiple with a colon\nseparator. Eg -ig 27_2:23_1.")), ToolInput(tag="in_write_individual_files", input_type=Boolean(optional=True), prefix="--write_individual_files", doc=InputDocumentation(doc="When selected, dislocate also writes individual\nbedgraph files for each read length/frame. This could\nbe useful for downstream analysis.")), ToolInput(tag="in_keep_read_types_distinct", input_type=Boolean(optional=True), prefix="--keep_read_types_distinct", doc=InputDocumentation(doc="Write out a single bedgraph, but keep the read types\ndistinct (useful for downstreamanalysis). Default =\nFalse")), ToolInput(tag="in_save_stats", input_type=Boolean(optional=True), prefix="--save_stats", doc=InputDocumentation(doc="Save a csv of the r values for each type, read and")), ToolInput(tag="in_min_ratio_start", input_type=Int(optional=True), prefix="--min_ratio_start", doc=InputDocumentation(doc="The minimum ratio of the start codon counts of a given\nread type versus the previous position for an offset\nbased on the start codon to be confidently assigned.\nDefault=4")), ToolInput(tag="in_min_counts_stop", input_type=Int(optional=True), prefix="--min_counts_stop", doc=InputDocumentation(doc="The minimum number of counts of a given read type at\nthe stop codon for a read's offset to be confidently\nassigned. Default=25. To block stop offsets being\nused, set to large number eg 100000")), ToolInput(tag="in_min_ratio_stop", input_type=Int(optional=True), prefix="--min_ratio_stop", doc=InputDocumentation(doc="The minimum ratio of the stop codon counts of a given\nread type versus the next position for an offset based\non the stop codon to be confidently assigned.\nDefault=4")), ToolInput(tag="in_max_distance", input_type=Int(optional=True), prefix="--max_distance", doc=InputDocumentation(doc="Maximum distance in nucleotides around start and stop\ncodons for which offsets are attempted to be\ncalculated. Default is 20. Twenty is plenty.")), ToolInput(tag="in_frame_ness_ratio", input_type=Float(optional=True), prefix="--frameness_ratio", doc=InputDocumentation(doc="The level of enrichment near the start codon versus\ndownstream for a read type to be reported as\npotentially having strong bias towards out of frame\nribosomes. Default = 1.2 i.e. 20percent")), ToolInput(tag="in_mismatches", input_type=Boolean(optional=True), prefix="--mismatches", doc=InputDocumentation(doc="Whether to consider first nt mismatches. Only\navailable when a bam file is provided")), ToolInput(tag="in_ambiguity", input_type=Float(optional=True), prefix="--ambiguity", doc=InputDocumentation(doc="Much much better the best r value must be compared to\nthe second best for an offset to be confidently\nassigned. Default = 0.8, i.e. the r value of second\nbest offset must be less than 0.8x the value of the\nbest offset. Lower values are more stringent. Should\nbe less than 1, and more than 0")), ToolInput(tag="in_o_of_plot_start", input_type=Int(optional=True), prefix="--oof_plot_start", doc=InputDocumentation(doc="How far downstream of the annotated start to look for\nout of frame reads. Default=0. Set to negative values\nto search for uORFs")), ToolInput(tag="in_o_of_plot_end", input_type=Int(optional=True), prefix="--oof_plot_end", doc=InputDocumentation(doc="How far downstream of the annotated start to look for\nout of frame reads. Default=2000.")), ToolInput(tag="in_o_of_plot_stride", input_type=Int(optional=True), prefix="--oof_plot_stride", doc=InputDocumentation(doc="How wide each of the windows in the oof heatmap should\nbe in nucleotides. Default=50nt")), ToolInput(tag="in_no_iterative_improvement", input_type=Boolean(optional=True), prefix="--no_iterative_improvement", doc=InputDocumentation(doc="By default, when using RUST ratio correlations to\ndetermine offsets, when new matches are found they are\nadded to the reference. This option stops this\nbehaviour")), ToolInput(tag="in_o_of_smooth_window", input_type=String(optional=True), prefix="--oof_smooth_window", doc=InputDocumentation(doc="Rolling average smoothing for oof heatmap")), ToolInput(tag="in_sample_dir", input_type=Directory(optional=True), prefix="--sample_dir", doc=InputDocumentation(doc="Directory of the input files. This optional argument\ncan be useful when passing a csv of filenames to -s in\nreference generation mode.")), ToolInput(tag="in_output_directory", input_type=Directory(optional=True), prefix="--output_directory", doc=InputDocumentation(doc="The directory to save outputs.")), ToolInput(tag="in_mono_some_priority", input_type=Boolean(optional=True), prefix="--monosome_priority", doc=InputDocumentation(doc="During reference generation mode, and if a .csv of\nsample files is passed to the function, this option\nensures that only monosome files, i.e. those with\n'Mon' in the sample name(!), are used for reference\ngeneration.")), ToolInput(tag="in_four_column_bed", input_type=Boolean(optional=True), prefix="--four_column_bed", doc=InputDocumentation(doc="select if using a four column bed with the strand in\nthe 4th column")), ToolInput(tag="in_write_full_data", input_type=File(optional=True), prefix="--write_full_data", doc=InputDocumentation(doc="Write out the raw bed/bam file reads with read type\nand info\n")), ToolInput(tag="in_offset_dot", input_type=String(), position=0, doc=InputDocumentation(doc="--min_counts_start MIN_COUNTS_START"))], outputs=[ToolOutput(tag="out_samples", output_type=File(optional=True), selector=InputSelector(input_to_select="in_samples", type_hint=File()), doc=OutputDocumentation(doc="In reference generation mode this may either be a .csv\nfile of samples (you MUST ensure that the file is of\n'.csv' otherwise it will not be recognised) or a\nsingle bed file. In conversion mode it must be a\nsingle bed file. Bed files should be transcriptome-\naligned; only reads in the + strand are used. Bed\nfiles should be 6 column, with transcript_id, start,\nend, and strand in columns 1, 2, 3 and 6 respectively\n(the default output from bedtools' 'bamtobed'). Bed\nfiles can be in .gzip format if desired.")), ToolOutput(tag="out_output", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output", type_hint=File()), doc=OutputDocumentation(doc="Output file")), ToolOutput(tag="out_output_directory", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_output_directory", type_hint=File()), doc=OutputDocumentation(doc="The directory to save outputs.")), ToolOutput(tag="out_write_full_data", output_type=File(optional=True), selector=InputSelector(input_to_select="in_write_full_data", type_hint=File()), doc=OutputDocumentation(doc="Write out the raw bed/bam file reads with read type\nand info\n"))], container="quay.io/biocontainers/riboloco:0.3.9--pyh5e36f6f_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Riboloco_V0_1_0().translate("wdl")

