from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import String, File, Int, Float, Boolean

Moss_V0_1_0 = CommandToolBuilder(tool="moss", base_command=["moss"], inputs=[ToolInput(tag="in_bam", input_type=String(optional=True), prefix="--bam", doc=InputDocumentation(doc="original BAM files for one sample,\ncan be addressed multiple times")), ToolInput(tag="in_realigned", input_type=String(optional=True), prefix="--realigned", doc=InputDocumentation(doc="realigned BAM files for one sample,\ncan be addressed multiple times")), ToolInput(tag="in_ref", input_type=File(optional=True), prefix="--ref", doc=InputDocumentation(doc="reference FASTA file")), ToolInput(tag="in_output", input_type=File(optional=True), prefix="--output", doc=InputDocumentation(doc="output VCF file")), ToolInput(tag="in_normal", input_type=String(optional=True), prefix="--normal", doc=InputDocumentation(doc="normal sample's germline VCF result")), ToolInput(tag="in_vcf", input_type=String(optional=True), prefix="--vcf", doc=InputDocumentation(doc="tumor samples' somatic VCF result")), ToolInput(tag="in_loci", input_type=String(optional=True), prefix="--loci", doc=InputDocumentation(doc="candidate loci VCF files generated by python script")), ToolInput(tag="in_tau", input_type=Int(optional=True), prefix="--tau", doc=InputDocumentation(doc="optional threshold for somatic score, default is 0")), ToolInput(tag="in_mu", input_type=Float(optional=True), prefix="--mu", doc=InputDocumentation(doc="set somatic mutation prior, 1-5x10^(-m), default is 1-5e-4")), ToolInput(tag="in_min_base_qual", input_type=Int(optional=True), prefix="--min-base-qual", doc=InputDocumentation(doc="minimum threshold for base quality, default is 13")), ToolInput(tag="in_min_mapping_qual", input_type=Int(optional=True), prefix="--min-mapping-qual", doc=InputDocumentation(doc="minimum threshold for mapping quality, default is 30")), ToolInput(tag="in_max_dep", input_type=Int(optional=True), prefix="--max_dep", doc=InputDocumentation(doc="max depth of the dataset, default is 500")), ToolInput(tag="in_grid_size", input_type=Int(optional=True), prefix="--grid-size", doc=InputDocumentation(doc="number of grid points in [0, 1] for approximate integration,\ndefault is 20, must be non-negative")), ToolInput(tag="in_dry", input_type=Boolean(optional=True), prefix="--dry", doc=InputDocumentation(doc="set to dry run")), ToolInput(tag="in_filter_total", input_type=Boolean(optional=True), prefix="--filter-total", doc=InputDocumentation(doc="set to filter variants with total tumor depth < 150")), ToolInput(tag="in_filter_vaf", input_type=Boolean(optional=True), prefix="--filter-vaf", doc=InputDocumentation(doc="set to filter variants with VAF in any samples < 0.1")), ToolInput(tag="in_ignore_zero", input_type=Boolean(optional=True), prefix="--ignore0", doc=InputDocumentation(doc="ignore samples with all reads same as the normal allele")), ToolInput(tag="in_pile_up", input_type=Boolean(optional=True), prefix="--pileup", doc=InputDocumentation(doc="output pileup, only effective in dry-run mode"))], outputs=[ToolOutput(tag="out_output", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output", type_hint=File()), doc=OutputDocumentation(doc="output VCF file"))], container="quay.io/biocontainers/moss:0.1.1--h179b981_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Moss_V0_1_0().translate("wdl")

