from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean

Coprarna2_Pl_V0_1_0 = CommandToolBuilder(tool="CopraRNA2.pl", base_command=["CopraRNA2.pl"], inputs=[ToolInput(tag="in_srna_seq", input_type=Boolean(optional=True), prefix="--srnaseq", doc=InputDocumentation(doc="FASTA file with small RNA sequences (def:input_sRNA.fa)")), ToolInput(tag="in_region", input_type=Boolean(optional=True), prefix="--region", doc=InputDocumentation(doc="region to scan in whole genome target prediction (def:5utr)\n'5utr' for start codon\n'3utr' for stop codon\n'cds' for entire transcript")), ToolInput(tag="in_nt_up", input_type=Boolean(optional=True), prefix="--ntup", doc=InputDocumentation(doc="amount of nucleotides upstream of '--region' to parse for targeting (def:200)")), ToolInput(tag="in_nt_down", input_type=Boolean(optional=True), prefix="--ntdown", doc=InputDocumentation(doc="amount of nucleotides downstream of '--region' to parse for targeting (def:100)")), ToolInput(tag="in_cores", input_type=Boolean(optional=True), prefix="--cores", doc=InputDocumentation(doc="amount of cores to use for parallel computation (def:1)")), ToolInput(tag="in_rc_size", input_type=Boolean(optional=True), prefix="--rcsize", doc=InputDocumentation(doc="minimum amount (%) of putative target homologs that need to be available\nfor a target cluster to be considered in the CopraRNA1 part (see --cop1) of the prediction (def:0.5)")), ToolInput(tag="in_winsize", input_type=Boolean(optional=True), prefix="--winsize", doc=InputDocumentation(doc="IntaRNA target (--tAccW) window size parameter (def:150)")), ToolInput(tag="in_max_bp_dist", input_type=Boolean(optional=True), prefix="--maxbpdist", doc=InputDocumentation(doc="IntaRNA target (--tAccL) maximum base pair distance parameter (def:100)")), ToolInput(tag="in_cop_one", input_type=Boolean(optional=True), prefix="--cop1", doc=InputDocumentation(doc="switch for CopraRNA1 prediction (def:off)")), ToolInput(tag="in_cons", input_type=Boolean(optional=True), prefix="--cons", doc=InputDocumentation(doc="controls consensus prediction (def:0)\n'0' for off\n'1' for organism of interest based consensus\n'2' for overall consensus based prediction")), ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="--verbose", doc=InputDocumentation(doc="switch to print verbose output to terminal during computation (def:off)")), ToolInput(tag="in_web_srv", input_type=Boolean(optional=True), prefix="--websrv", doc=InputDocumentation(doc="switch to provide webserver output files (def:off)")), ToolInput(tag="in_no_clean", input_type=Boolean(optional=True), prefix="--noclean", doc=InputDocumentation(doc="switch to prevent removal of temporary files (def:off)")), ToolInput(tag="in_enrich", input_type=Boolean(optional=True), prefix="--enrich", doc=InputDocumentation(doc="if entered then DAVID-WS functional enrichment is calculated with given amount of top predictions (def:off)")), ToolInput(tag="in_no_oi", input_type=Boolean(optional=True), prefix="--nooi", doc=InputDocumentation(doc="if set then the CopraRNA2 prediction mode is set not to focus on the organism of interest (def:off)")), ToolInput(tag="in_ooi_filt", input_type=Boolean(optional=True), prefix="--ooifilt", doc=InputDocumentation(doc="post processing filter for organism of interest p-value 0=off (def:0)")), ToolInput(tag="in_root", input_type=Boolean(optional=True), prefix="--root", doc=InputDocumentation(doc="specifies root function to apply to the weights (def:1)")), ToolInput(tag="in_top_count", input_type=Boolean(optional=True), prefix="--topcount", doc=InputDocumentation(doc="specifies the amount of top predictions to return and use for the extended regions plots (def:200)"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Coprarna2_Pl_V0_1_0().translate("wdl", allow_empty_container=True)

