from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Boolean, String

Varscan_Fpfilter_V0_1_0 = CommandToolBuilder(tool="varscan_fpfilter", base_command=["varscan", "fpfilter"], inputs=[ToolInput(tag="in_output_file", input_type=File(optional=True), prefix="--output-file", doc=InputDocumentation(doc="Optional output file for filter-pass variants")), ToolInput(tag="in_filtered_file", input_type=File(optional=True), prefix="--filtered-file", doc=InputDocumentation(doc="Optional output file for filter-fail variants")), ToolInput(tag="in_dream_three_settings", input_type=Boolean(optional=True), prefix="--dream3-settings", doc=InputDocumentation(doc="If set to 1, optimizes filter parameters based on TCGA-ICGC DREAM-3 SNV Challenge results")), ToolInput(tag="in_keep_failures", input_type=File(optional=True), prefix="--keep-failures", doc=InputDocumentation(doc="If set to 1, includes failures in the output file")), ToolInput(tag="in_min_var_count", input_type=Boolean(optional=True), prefix="--min-var-count", doc=InputDocumentation(doc="Minimum number of variant-supporting reads [4]")), ToolInput(tag="in_min_var_count_lc", input_type=Boolean(optional=True), prefix="--min-var-count-lc", doc=InputDocumentation(doc="Minimum number of variant-supporting reads when depth below somaticPdepth [2]")), ToolInput(tag="in_min_var_freq", input_type=Boolean(optional=True), prefix="--min-var-freq", doc=InputDocumentation(doc="Minimum variant allele frequency [0.05]")), ToolInput(tag="in_max_somatic_p", input_type=Boolean(optional=True), prefix="--max-somatic-p", doc=InputDocumentation(doc="Maximum somatic p-value [1.0]")), ToolInput(tag="in_max_somatic_p_depth", input_type=Boolean(optional=True), prefix="--max-somatic-p-depth", doc=InputDocumentation(doc="Depth required to test max somatic p-value [10]")), ToolInput(tag="in_min_ref_read_pos", input_type=Boolean(optional=True), prefix="--min-ref-readpos", doc=InputDocumentation(doc="Minimum average read position of ref-supporting reads [0.1]")), ToolInput(tag="in_min_var_read_pos", input_type=Boolean(optional=True), prefix="--min-var-readpos", doc=InputDocumentation(doc="Minimum average read position of var-supporting reads [0.1]")), ToolInput(tag="in_min_ref_dist_three", input_type=Boolean(optional=True), prefix="--min-ref-dist3", doc=InputDocumentation(doc="Minimum average distance to effective 3' end (ref) [0.1]")), ToolInput(tag="in_min_var_dist_three", input_type=Boolean(optional=True), prefix="--min-var-dist3", doc=InputDocumentation(doc="Minimum average distance to effective 3' end (var) [0.1]")), ToolInput(tag="in_min_stranded_ness", input_type=Boolean(optional=True), prefix="--min-strandedness", doc=InputDocumentation(doc="Minimum fraction of variant reads from each strand [0.01]")), ToolInput(tag="in_min_strand_reads", input_type=Boolean(optional=True), prefix="--min-strand-reads", doc=InputDocumentation(doc="Minimum allele depth required to perform the strand tests [5]")), ToolInput(tag="in_min_ref_base_qual", input_type=Boolean(optional=True), prefix="--min-ref-basequal", doc=InputDocumentation(doc="Minimum average base quality for ref allele [15]")), ToolInput(tag="in_min_var_base_qual", input_type=Boolean(optional=True), prefix="--min-var-basequal", doc=InputDocumentation(doc="Minimum average base quality for var allele [15]")), ToolInput(tag="in_max_base_qual_diff", input_type=Boolean(optional=True), prefix="--max-basequal-diff", doc=InputDocumentation(doc="Maximum average base quality diff (ref - var) [50]")), ToolInput(tag="in_min_ref_avg_rl", input_type=Boolean(optional=True), prefix="--min-ref-avgrl", doc=InputDocumentation(doc="Minimum average trimmed read length for ref allele [90]")), ToolInput(tag="in_min_var_avg_rl", input_type=Boolean(optional=True), prefix="--min-var-avgrl", doc=InputDocumentation(doc="Minimum average trimmed read length for var allele [90]")), ToolInput(tag="in_max_rl_diff", input_type=Boolean(optional=True), prefix="--max-rl-diff", doc=InputDocumentation(doc="Maximum average relative read length difference (ref - var) [0.25]")), ToolInput(tag="in_max_ref_mm_qs", input_type=Boolean(optional=True), prefix="--max-ref-mmqs", doc=InputDocumentation(doc="Maximum mismatch quality sum of reference-supporting reads [100]")), ToolInput(tag="in_max_var_mm_qs", input_type=Boolean(optional=True), prefix="--max-var-mmqs", doc=InputDocumentation(doc="Maximum mismatch quality sum of variant-supporting reads [100]")), ToolInput(tag="in_min_mm_qs_diff", input_type=Boolean(optional=True), prefix="--min-mmqs-diff", doc=InputDocumentation(doc="Minimum average mismatch quality sum (var - ref) [0]")), ToolInput(tag="in_max_mm_qs_diff", input_type=Boolean(optional=True), prefix="--max-mmqs-diff", doc=InputDocumentation(doc="Maximum average mismatch quality sum (var - ref) [50]")), ToolInput(tag="in_min_ref_map_qual", input_type=Boolean(optional=True), prefix="--min-ref-mapqual", doc=InputDocumentation(doc="Minimum average mapping quality for ref allele [15]")), ToolInput(tag="in_min_var_map_qual", input_type=Boolean(optional=True), prefix="--min-var-mapqual", doc=InputDocumentation(doc="Minimum average mapping quality for var allele [15]")), ToolInput(tag="in_max_map_qual_diff", input_type=Boolean(optional=True), prefix="--max-mapqual-diff", doc=InputDocumentation(doc="Maximum average mapping quality (ref - var) [50]")), ToolInput(tag="in_jar", input_type=String(optional=True), prefix="-jar", doc=InputDocumentation(doc="")), ToolInput(tag="in_java", input_type=String(), position=0, doc=InputDocumentation(doc="")), ToolInput(tag="in_fp_filter", input_type=String(), position=1, doc=InputDocumentation(doc="")), ToolInput(tag="in_variant", input_type=String(optional=True), position=2, doc=InputDocumentation(doc="")), ToolInput(tag="in_file", input_type=File(optional=True), position=3, doc=InputDocumentation(doc="")), ToolInput(tag="in_read_count", input_type=String(optional=True), position=4, doc=InputDocumentation(doc=""))], outputs=[ToolOutput(tag="out_output_file", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output_file", type_hint=File()), doc=OutputDocumentation(doc="Optional output file for filter-pass variants")), ToolOutput(tag="out_filtered_file", output_type=File(optional=True), selector=InputSelector(input_to_select="in_filtered_file", type_hint=File()), doc=OutputDocumentation(doc="Optional output file for filter-fail variants")), ToolOutput(tag="out_keep_failures", output_type=File(optional=True), selector=InputSelector(input_to_select="in_keep_failures", type_hint=File()), doc=OutputDocumentation(doc="If set to 1, includes failures in the output file"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Varscan_Fpfilter_V0_1_0().translate("wdl", allow_empty_container=True)

