from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, File, Int, String

Pycoqc_V0_1_0 = CommandToolBuilder(tool="pycoQC", base_command=["pycoQC"], inputs=[ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="--verbose", doc=InputDocumentation(doc="Increase verbosity")), ToolInput(tag="in_quiet", input_type=Boolean(optional=True), prefix="--quiet", doc=InputDocumentation(doc="Reduce verbosity")), ToolInput(tag="in_summary_file", input_type=Boolean(optional=True), prefix="--summary_file", doc=InputDocumentation(doc="[SUMMARY_FILE [SUMMARY_FILE ...]], -f [SUMMARY_FILE [SUMMARY_FILE ...]]\nPath to a sequencing_summary generated by Albacore\n1.0.0 + (read_fast5_basecaller.py) / Guppy 2.1.3+\n(guppy_basecaller). One can also pass multiple space\nseparated file paths or a UNIX style regex matching\nmultiple files (Required)")), ToolInput(tag="in_barcode_file", input_type=Boolean(optional=True), prefix="--barcode_file", doc=InputDocumentation(doc="[BARCODE_FILE [BARCODE_FILE ...]], -b [BARCODE_FILE [BARCODE_FILE ...]]\nPath to the barcode_file generated by Guppy 2.1.3+\n(guppy_barcoder) or Deepbinner 0.2.0+. This is not a\nrequired file. One can also pass multiple space\nseparated file paths or a UNIX style regex matching\nmultiple files (optional)")), ToolInput(tag="in_bam_file", input_type=Boolean(optional=True), prefix="--bam_file", doc=InputDocumentation(doc="[BAM_FILE [BAM_FILE ...]], -a [BAM_FILE [BAM_FILE ...]]\nPath to a Bam file corresponding to reads in the\nsummary_file. Preferably aligned with Minimap2 One can\nalso pass multiple space separated file paths or a\nUNIX style regex matching multiple files (optional)")), ToolInput(tag="in_html_outfile", input_type=File(optional=True), prefix="--html_outfile", doc=InputDocumentation(doc="Path to an output html file report (required if\njson_outfile not given)")), ToolInput(tag="in_json_outfile", input_type=File(optional=True), prefix="--json_outfile", doc=InputDocumentation(doc="Path to an output json file report (required if\nhtml_outfile not given)")), ToolInput(tag="in_min_pass_qual", input_type=Int(optional=True), prefix="--min_pass_qual", doc=InputDocumentation(doc="Minimum quality to consider a read as 'pass' (default:\n7)")), ToolInput(tag="in_min_pass_len", input_type=Int(optional=True), prefix="--min_pass_len", doc=InputDocumentation(doc="Minimum read length to consider a read as 'pass'\n(default: 0)")), ToolInput(tag="in_filter_calibration", input_type=Boolean(optional=True), prefix="--filter_calibration", doc=InputDocumentation(doc="If given, reads flagged as calibration strand by the\nbasecaller are removed (default: False)")), ToolInput(tag="in_filter_duplicated", input_type=Boolean(optional=True), prefix="--filter_duplicated", doc=InputDocumentation(doc="If given, duplicated read_ids are removed but the\nfirst occurence is kept (Guppy sometimes outputs the\nsame read multiple times) (default: False)")), ToolInput(tag="in_min_barcode_percent", input_type=Int(optional=True), prefix="--min_barcode_percent", doc=InputDocumentation(doc="Minimal percent of total reads to retain barcode\nlabel. If below, the barcode value is set as\n`unclassified` (default: 0.1)")), ToolInput(tag="in_report_title", input_type=String(optional=True), prefix="--report_title", doc=InputDocumentation(doc="Title to use in the html report (default: PycoQC\nreport)")), ToolInput(tag="in_template_file", input_type=File(optional=True), prefix="--template_file", doc=InputDocumentation(doc="Jinja2 html template for the html report (default: )")), ToolInput(tag="in_config_file", input_type=File(optional=True), prefix="--config_file", doc=InputDocumentation(doc="Path to a JSON configuration file for the html report.\nIf not provided, looks for it in ~/.pycoQC and\n~/.config/pycoQC/config. If it's still not found,\nfalls back to default parameters. The first level keys\nare the names of the plots to be included. The second\nlevel keys are the parameters to pass to each plotting\nfunction (default: )')")), ToolInput(tag="in_skip_coverage_plot", input_type=Boolean(optional=True), prefix="--skip_coverage_plot", doc=InputDocumentation(doc="Skip the coverage plot in HTML report. Useful when\nusing a reference file containing many sequences, i.e.\ntranscriptome (default: False)")), ToolInput(tag="in_sample", input_type=Int(optional=True), prefix="--sample", doc=InputDocumentation(doc="If not None a n number of reads will be randomly\nselected instead of the entire dataset for ploting\nfunction (deterministic sampling) (default: 100000)")), ToolInput(tag="in_default_config", input_type=Boolean(optional=True), prefix="--default_config", doc=InputDocumentation(doc="Print default configuration file. Can be used to\ngenerate a template JSON file (default: False)\n"))], outputs=[ToolOutput(tag="out_html_outfile", output_type=File(optional=True), selector=InputSelector(input_to_select="in_html_outfile", type_hint=File()), doc=OutputDocumentation(doc="Path to an output html file report (required if\njson_outfile not given)")), ToolOutput(tag="out_json_outfile", output_type=File(optional=True), selector=InputSelector(input_to_select="in_json_outfile", type_hint=File()), doc=OutputDocumentation(doc="Path to an output json file report (required if\nhtml_outfile not given)"))], container="quay.io/biocontainers/pycoqc:2.5.2--py_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Pycoqc_V0_1_0().translate("wdl")

