from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Boolean, String

Whatshap_Split_V0_1_0 = CommandToolBuilder(tool="whatshap_split", base_command=["whatshap", "split"], inputs=[ToolInput(tag="in_output_hone", input_type=File(optional=True), prefix="--output-h1", doc=InputDocumentation(doc="Output file to write reads from Haplotype 1 to. Use\nending .gz to create gzipped file.")), ToolInput(tag="in_output_h_two", input_type=File(optional=True), prefix="--output-h2", doc=InputDocumentation(doc="Output file to write reads from Haplotype 2 to. Use\nending .gz to create gzipped file.")), ToolInput(tag="in_output_untagged", input_type=File(optional=True), prefix="--output-untagged", doc=InputDocumentation(doc="Output file to write untagged reads to. Use ending .gz\nto create gzipped file.")), ToolInput(tag="in_add_untagged", input_type=Boolean(optional=True), prefix="--add-untagged", doc=InputDocumentation(doc="Add reads without tag to both H1 and H2 output")), ToolInput(tag="in_discard_unknown_reads", input_type=Boolean(optional=True), prefix="--discard-unknown-reads", doc=InputDocumentation(doc="Only check the haplotype of reads listed in the\nhaplotag list file. Reads (read names) not contained\nin this file will be discarded. In the default case (=\nkeep unknown reads), those reads would be considered\nuntagged and end up in the respective output file.\nPlease be sure that the read names match between the\ninput FASTQ/BAM and the haplotag list file.")), ToolInput(tag="in_read_lengths_histogram", input_type=File(optional=True), prefix="--read-lengths-histogram", doc=InputDocumentation(doc="Output file to write read lengths histogram to in tab\nseparated format.\n")), ToolInput(tag="in_reads", input_type=String(), position=0, doc=InputDocumentation(doc="Input FASTQ/BAM file with reads (FASTQ can be gzipped)")), ToolInput(tag="in_list", input_type=String(), position=1, doc=InputDocumentation(doc="Tab-separated list with (at least) two columns\n<readname> and <haplotype> (can be gzipped).\nCurrently, the two haplotypes have to be named H1 and\nH2 (or none). Alternatively, the output of the\n'haplotag' command can be used (4 columns), and this\nis required for using the '--only-largest-block'\noption (need phaseset and chromosome info).")), ToolInput(tag="in_streams_dot", input_type=String(), position=0, doc=InputDocumentation(doc="--only-largest-block  Only consider reads to be tagged if they belong to the"))], outputs=[ToolOutput(tag="out_output_hone", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output_hone", type_hint=File()), doc=OutputDocumentation(doc="Output file to write reads from Haplotype 1 to. Use\nending .gz to create gzipped file.")), ToolOutput(tag="out_output_h_two", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output_h_two", type_hint=File()), doc=OutputDocumentation(doc="Output file to write reads from Haplotype 2 to. Use\nending .gz to create gzipped file.")), ToolOutput(tag="out_output_untagged", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output_untagged", type_hint=File()), doc=OutputDocumentation(doc="Output file to write untagged reads to. Use ending .gz\nto create gzipped file.")), ToolOutput(tag="out_read_lengths_histogram", output_type=File(optional=True), selector=InputSelector(input_to_select="in_read_lengths_histogram", type_hint=File()), doc=OutputDocumentation(doc="Output file to write read lengths histogram to in tab\nseparated format.\n"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Whatshap_Split_V0_1_0().translate("wdl", allow_empty_container=True)

