from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, String

Rsat_Get_Orthologs_V0_1_0 = CommandToolBuilder(tool="rsat_get_orthologs", base_command=["rsat", "get-orthologs"], inputs=[ToolInput(tag="in_unique_species", input_type=Boolean(optional=True), prefix="-unique_species", doc=InputDocumentation(doc=", but at the level of the genus. At this level we")), ToolInput(tag="in_return", input_type=String(optional=True), prefix="-return", doc=InputDocumentation(doc=",e_value,rank")), ToolInput(tag="in_get_orthologs", input_type=String(), position=0, doc=InputDocumentation(doc="[1mDESCRIPTION[0m")), ToolInput(tag="in_ref_id", input_type=String(), position=0, doc=InputDocumentation(doc="ID of the reference (target) gene")), ToolInput(tag="in_ref_organism", input_type=String(), position=1, doc=InputDocumentation(doc="Name of the reference (target) organism")), ToolInput(tag="in_query_id", input_type=String(), position=2, doc=InputDocumentation(doc="ID of the query gene")), ToolInput(tag="in_query_name", input_type=String(), position=3, doc=InputDocumentation(doc="Common name of the query gene")), ToolInput(tag="in_ref_name", input_type=String(), position=4, doc=InputDocumentation(doc="Common name of the gene found in the reference genome.\n[1mBeware:[0m this output field requires to load synonym tables\nfor all the organisms of the reference taxon, which\nincreases the running time (a few seconds per taxon).")), ToolInput(tag="in_query_organism", input_type=String(), position=5, doc=InputDocumentation(doc="Name of the query organism")), ToolInput(tag="in_ident", input_type=String(), position=6, doc=InputDocumentation(doc="Percent of identity (a number between 0 and 100)")), ToolInput(tag="in_ali_len", input_type=String(), position=7, doc=InputDocumentation(doc="Alignment lengths (in residues)")), ToolInput(tag="in_mism_at", input_type=String(), position=8, doc=InputDocumentation(doc="Number of mismatches")), ToolInput(tag="in_gap_open", input_type=String(), position=9, doc=InputDocumentation(doc="Number of gap openings")), ToolInput(tag="in_e_value", input_type=String(), position=10, doc=InputDocumentation(doc="E-value (expected number of false positives)")), ToolInput(tag="in_bit_sc", input_type=String(), position=11, doc=InputDocumentation(doc="Bit score")), ToolInput(tag="in_rank", input_type=String(), position=12, doc=InputDocumentation(doc="Rank")), ToolInput(tag="in_s_rank", input_type=String(), position=13, doc=InputDocumentation(doc="Source rank (rank of the hit for the query organism)."))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Rsat_Get_Orthologs_V0_1_0().translate("wdl", allow_empty_container=True)

