from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, String, Boolean, Int, Float, Directory

Featurecounts_V0_1_0 = CommandToolBuilder(tool="featureCounts", base_command=["featureCounts"], inputs=[ToolInput(tag="in_name_annotation_file", input_type=File(optional=True), prefix="-a", doc=InputDocumentation(doc="Name of an annotation file. GTF/GFF format by default. See")), ToolInput(tag="in_more_format_information", input_type=String(optional=True), prefix="-F", doc=InputDocumentation(doc="for more format information. Inbuilt annotations")), ToolInput(tag="in_name_including_read", input_type=File(optional=True), prefix="-o", doc=InputDocumentation(doc="Name of output file including read counts. A separate file\nincluding summary statistics of counting results is also\nincluded in the output ('<string>.summary'). Both files\nare in tab delimited format.")), ToolInput(tag="in_specify_feature_types", input_type=String(optional=True), prefix="-t", doc=InputDocumentation(doc="Specify feature type(s) in a GTF annotation. If multiple\ntypes are provided, they should be separated by ',' with\nno space in between. 'exon' by default. Rows in the\nannotation with a matched feature will be extracted and\nused for read mapping.")), ToolInput(tag="in_specify_attribute_type", input_type=String(optional=True), prefix="-g", doc=InputDocumentation(doc="Specify attribute type in GTF annotation. 'gene_id' by\ndefault. Meta-features used for read counting will be\nextracted from annotation using the provided value.")), ToolInput(tag="in_extra_attributes", input_type=Boolean(optional=True), prefix="--extraAttributes", doc=InputDocumentation(doc="Extract extra attribute types from the provided GTF\nannotation and include them in the counting output. These\nattribute types will not be used to group features. If\nmore than one attribute type is provided they should be\nseparated by comma.")), ToolInput(tag="in_provide_alias_file", input_type=File(optional=True), prefix="-A", doc=InputDocumentation(doc="Provide a chromosome name alias file to match chr names in\nannotation with those in the reads. This should be a two-\ncolumn comma-delimited text file. Its first column should\ninclude chr names in the annotation and its second column\nshould include chr names in the reads. Chr names are case\nsensitive. No column header should be included in the\nfile.")), ToolInput(tag="in_perform_read_counting", input_type=Boolean(optional=True), prefix="-f", doc=InputDocumentation(doc="Perform read counting at feature level (eg. counting\nreads for exons rather than genes).")), ToolInput(tag="in_assign_reads_their", input_type=Boolean(optional=True), prefix="-O", doc=InputDocumentation(doc="Assign reads to all their overlapping meta-features (or\nfeatures if -f is specified).")), ToolInput(tag="in_min_overlap", input_type=Int(optional=True), prefix="--minOverlap", doc=InputDocumentation(doc="Minimum number of overlapping bases in a read that is\nrequired for read assignment. 1 by default. Number of\noverlapping bases is counted from both reads if paired\nend. If a negative value is provided, then a gap of up\nto specified size will be allowed between read and the\nfeature that the read is assigned to.")), ToolInput(tag="in_frac_overlap", input_type=Float(optional=True), prefix="--fracOverlap", doc=InputDocumentation(doc="Minimum fraction of overlapping bases in a read that is\nrequired for read assignment. Value should be within range\n[0,1]. 0 by default. Number of overlapping bases is\ncounted from both reads if paired end. Both this option\nand '--minOverlap' option need to be satisfied for read\nassignment.")), ToolInput(tag="in_frac_overlap_feature", input_type=Float(optional=True), prefix="--fracOverlapFeature", doc=InputDocumentation(doc="Minimum fraction of overlapping bases in a\nfeature that is required for read assignment. Value\nshould be within range [0,1]. 0 by default.")), ToolInput(tag="in_largest_overlap", input_type=Boolean(optional=True), prefix="--largestOverlap", doc=InputDocumentation(doc="Assign reads to a meta-feature/feature that has the\nlargest number of overlapping bases.")), ToolInput(tag="in_non_overlap", input_type=Int(optional=True), prefix="--nonOverlap", doc=InputDocumentation(doc="Maximum number of non-overlapping bases in a read (or a\nread pair) that is allowed when being assigned to a\nfeature. No limit is set by default.")), ToolInput(tag="in_non_overlap_feature", input_type=Int(optional=True), prefix="--nonOverlapFeature", doc=InputDocumentation(doc="Maximum number of non-overlapping bases in a feature\nthat is allowed in read assignment. No limit is set by\ndefault.")), ToolInput(tag="in_read_extension_five", input_type=Int(optional=True), prefix="--readExtension5", doc=InputDocumentation(doc="Reads are extended upstream by <int> bases from their\n5' end.")), ToolInput(tag="in_read_extension_three", input_type=Int(optional=True), prefix="--readExtension3", doc=InputDocumentation(doc="Reads are extended upstream by <int> bases from their\n3' end.")), ToolInput(tag="in_read_two_pos", input_type=Boolean(optional=True), prefix="--read2pos", doc=InputDocumentation(doc="<5:3>    Reduce reads to their 5' most base or 3' most base. Read\ncounting is then performed based on the single base the\nread is reduced to.")), ToolInput(tag="in_multimapping_reads_also", input_type=Boolean(optional=True), prefix="-M", doc=InputDocumentation(doc="Multi-mapping reads will also be counted. For a multi-\nmapping read, all its reported alignments will be\ncounted. The 'NH' tag in BAM/SAM input is used to detect\nmulti-mapping reads.")), ToolInput(tag="in_fraction", input_type=Boolean(optional=True), prefix="--fraction", doc=InputDocumentation(doc="Assign fractional counts to features. This option must\nbe used together with '-M' or '-O' or both. When '-M' is\nspecified, each reported alignment from a multi-mapping\nread (identified via 'NH' tag) will carry a fractional\ncount of 1/x, instead of 1 (one), where x is the total\nnumber of alignments reported for the same read. When '-O'\nis specified, each overlapping feature will receive a\nfractional count of 1/y, where y is the total number of\nfeatures overlapping with the read. When both '-M' and\n'-O' are specified, each alignment will carry a fractional\ncount of 1/(x*y).")), ToolInput(tag="in_minimum_mapping_quality", input_type=Int(optional=True), prefix="-Q", doc=InputDocumentation(doc="The minimum mapping quality score a read must satisfy in\norder to be counted. For paired-end reads, at least one\nend should satisfy this criteria. 0 by default.")), ToolInput(tag="in_split_only", input_type=Boolean(optional=True), prefix="--splitOnly", doc=InputDocumentation(doc="Count split alignments only (ie. alignments with CIGAR\nstring containing 'N'). An example of split alignments is\nexon-spanning reads in RNA-seq data.")), ToolInput(tag="in_non_split_only", input_type=Boolean(optional=True), prefix="--nonSplitOnly", doc=InputDocumentation(doc="If specified, only non-split alignments (CIGAR strings do\nnot contain letter 'N') will be counted. All the other\nalignments will be ignored.")), ToolInput(tag="in_primary", input_type=Boolean(optional=True), prefix="--primary", doc=InputDocumentation(doc="Count primary alignments only. Primary alignments are\nidentified using bit 0x100 in SAM/BAM FLAG field.")), ToolInput(tag="in_ignored_up", input_type=Boolean(optional=True), prefix="--ignoreDup", doc=InputDocumentation(doc="Ignore duplicate reads in read counting. Duplicate reads\nare identified using bit Ox400 in BAM/SAM FLAG field. The\nwhole read pair is ignored if one of the reads is a\nduplicate read for paired end data.")), ToolInput(tag="in_perform_strandspecific_read", input_type=Int(optional=True), prefix="-s", doc=InputDocumentation(doc="Perform strand-specific read counting. A single integer\nvalue (applied to all input files) or a string of comma-\nseparated values (applied to each corresponding input\nfile) should be provided. Possible values include:\n0 (unstranded), 1 (stranded) and 2 (reversely stranded).\nDefault value is 0 (ie. unstranded read counting carried\nout for all input files).")), ToolInput(tag="in_count_number_reads", input_type=Boolean(optional=True), prefix="-J", doc=InputDocumentation(doc="Count number of reads supporting each exon-exon junction.\nJunctions were identified from those exon-spanning reads\nin the input (containing 'N' in CIGAR string). Counting\nresults are saved to a file named '<output_file>.jcounts'")), ToolInput(tag="in_provide_name_fastaformat", input_type=File(optional=True), prefix="-G", doc=InputDocumentation(doc="Provide the name of a FASTA-format file that contains the\nreference sequences used in read mapping that produced the\nprovided SAM/BAM files. This optional argument can be used\nwith '-J' option to improve read counting for junctions.")), ToolInput(tag="in_specified_fragments_templates", input_type=Boolean(optional=True), prefix="-p", doc=InputDocumentation(doc="If specified, fragments (or templates) will be counted\ninstead of reads. This option is only applicable for\npaired-end reads; single-end reads are always counted as\nreads.")), ToolInput(tag="in_only_count_read", input_type=Boolean(optional=True), prefix="-B", doc=InputDocumentation(doc="Only count read pairs that have both ends aligned.")), ToolInput(tag="in_check_validity_counting", input_type=Boolean(optional=True), prefix="-P", doc=InputDocumentation(doc="Check validity of paired-end distance when counting read\npairs. Use -d and -D to set thresholds.")), ToolInput(tag="in_minimum_fragmenttemplate_length", input_type=Int(optional=True), prefix="-d", doc=InputDocumentation(doc="Minimum fragment/template length, 50 by default.")), ToolInput(tag="in_maximum_fragmenttemplate_length", input_type=Int(optional=True), prefix="-D", doc=InputDocumentation(doc="Maximum fragment/template length, 600 by default.")), ToolInput(tag="in_count_read_pairs", input_type=Boolean(optional=True), prefix="-C", doc=InputDocumentation(doc="Do not count read pairs that have their two ends mapping\nto different chromosomes or mapping to same chromosome\nbut on different strands.")), ToolInput(tag="in_do_not_sort", input_type=Boolean(optional=True), prefix="--donotsort", doc=InputDocumentation(doc="Do not sort reads in BAM/SAM input. Note that reads from\nthe same pair are required to be located next to each\nother in the input.")), ToolInput(tag="in_number_threads_default", input_type=Int(optional=True), prefix="-T", doc=InputDocumentation(doc="Number of the threads. 1 by default.")), ToolInput(tag="in_by_read_group", input_type=Boolean(optional=True), prefix="--byReadGroup", doc=InputDocumentation(doc="Assign reads by read group. 'RG' tag is required to be\npresent in the input BAM/SAM files.")), ToolInput(tag="in_count_long_reads", input_type=Boolean(optional=True), prefix="-L", doc=InputDocumentation(doc="Count long reads such as Nanopore and PacBio reads. Long\nread counting can only run in one thread and only reads\n(not read-pairs) can be counted. There is no limitation on\nthe number of 'M' operations allowed in a CIGAR string in\nlong read counting.")), ToolInput(tag="in_output_detailed_results", input_type=File(optional=True), prefix="-R", doc=InputDocumentation(doc="Output detailed assignment results for each read or read-\npair. Results are saved to a file that is in one of the\nfollowing formats: CORE, SAM and BAM. See Users Guide for\nmore info about these formats.")), ToolInput(tag="in_rpath", input_type=Directory(optional=True), prefix="--Rpath", doc=InputDocumentation(doc="Specify a directory to save the detailed assignment\nresults. If unspecified, the directory where counting\nresults are saved is used.")), ToolInput(tag="in_tmpdir", input_type=Directory(optional=True), prefix="--tmpDir", doc=InputDocumentation(doc="Directory under which intermediate files are saved (later\nremoved). By default, intermediate files will be saved to\nthe directory specified in '-o' argument.")), ToolInput(tag="in_max_mop", input_type=Int(optional=True), prefix="--maxMOp", doc=InputDocumentation(doc="Maximum number of 'M' operations allowed in a CIGAR\nstring. 10 by default. Both 'X' and '=' are treated as 'M'\nand adjacent 'M' operations are merged in the CIGAR\nstring.")), ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="--verbose", doc=InputDocumentation(doc="Output verbose information for debugging, such as un-\nmatched chromosome/contig names.")), ToolInput(tag="in_output_version_program", input_type=Boolean(optional=True), prefix="-v", doc=InputDocumentation(doc="Output version of the program.")), ToolInput(tag="in_input_file_one", input_type=Int(), position=0, doc=InputDocumentation(doc="")), ToolInput(tag="in_input_file_two", input_type=Int(optional=True), position=1, doc=InputDocumentation(doc=""))], outputs=[ToolOutput(tag="out_name_including_read", output_type=File(optional=True), selector=InputSelector(input_to_select="in_name_including_read", type_hint=File()), doc=OutputDocumentation(doc="Name of output file including read counts. A separate file\nincluding summary statistics of counting results is also\nincluded in the output ('<string>.summary'). Both files\nare in tab delimited format.")), ToolOutput(tag="out_output_detailed_results", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output_detailed_results", type_hint=File()), doc=OutputDocumentation(doc="Output detailed assignment results for each read or read-\npair. Results are saved to a file that is in one of the\nfollowing formats: CORE, SAM and BAM. See Users Guide for\nmore info about these formats."))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Featurecounts_V0_1_0().translate("wdl", allow_empty_container=True)

