from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, Int, Array, String, File

Pauvre_Marginplot_V0_1_0 = CommandToolBuilder(tool="pauvre_marginplot", base_command=["pauvre", "marginplot"], inputs=[ToolInput(tag="in_quiet", input_type=Boolean(optional=True), prefix="--quiet", doc=InputDocumentation(doc="Do not output warnings to stderr")), ToolInput(tag="in_dpi", input_type=Int(optional=True), prefix="--dpi", doc=InputDocumentation(doc="Change the dpi from the default 600 if you need it")), ToolInput(tag="in_file_form", input_type=Array(t=String(), optional=True), prefix="--fileform", doc=InputDocumentation(doc="Which output format would you like? Def.=png")), ToolInput(tag="in_filt_maxlen", input_type=Int(optional=True), prefix="--filt_maxlen", doc=InputDocumentation(doc="This sets the max read length filter reads.")), ToolInput(tag="in_filt_max_qual", input_type=Int(optional=True), prefix="--filt_maxqual", doc=InputDocumentation(doc="This sets the max mean read quality to filter reads.")), ToolInput(tag="in_filt_min_len", input_type=Int(optional=True), prefix="--filt_minlen", doc=InputDocumentation(doc="This sets the min read length to filter reads.")), ToolInput(tag="in_filt_min_qual", input_type=Int(optional=True), prefix="--filt_minqual", doc=InputDocumentation(doc="This sets the min mean read quality to filter reads.")), ToolInput(tag="in_km_erdf", input_type=Int(optional=True), prefix="--kmerdf", doc=InputDocumentation(doc="Pass the filename of a data matrix if you would like\nto plot read length versus number of kmers in that\nread. The data matrix is a tab-separated text file\nwith columns 'id length numks and kmers', where: <id>\n= read id <length> = the length of the read <numks> =\nthe number of canonical kmers in the read <kmers> = a\nlist representation of kmers ie ['GAT', 'GTA']")), ToolInput(tag="in_no_transparent", input_type=Boolean(optional=True), prefix="--no_transparent", doc=InputDocumentation(doc="Specify this option if you don't want a transparent\nbackground. Default is on.")), ToolInput(tag="in_no_timestamp", input_type=File(optional=True), prefix="--no_timestamp", doc=InputDocumentation(doc="Turn off time stamps in the filename output.")), ToolInput(tag="in_output_base_name", input_type=File(optional=True), prefix="--output_base_name", doc=InputDocumentation(doc="Specify a base name for the output file(s). The input\nfile base name is the default.")), ToolInput(tag="in_plot_maxlen", input_type=Int(optional=True), prefix="--plot_maxlen", doc=InputDocumentation(doc="Sets the maximum viewing area in the length dimension.")), ToolInput(tag="in_plot_max_qual", input_type=Int(optional=True), prefix="--plot_maxqual", doc=InputDocumentation(doc="Sets the maximum viewing area in the quality\ndimension.")), ToolInput(tag="in_plot_min_len", input_type=Int(optional=True), prefix="--plot_minlen", doc=InputDocumentation(doc="Sets the minimum viewing area in the length dimension.")), ToolInput(tag="in_plot_min_qual", input_type=Int(optional=True), prefix="--plot_minqual", doc=InputDocumentation(doc="Sets the minimum viewing area in the quality\ndimension.")), ToolInput(tag="in_length_bin", input_type=Int(optional=True), prefix="--lengthbin", doc=InputDocumentation(doc="This sets the bin size to use for length.")), ToolInput(tag="in_qual_bin", input_type=Int(optional=True), prefix="--qualbin", doc=InputDocumentation(doc="This sets the bin size to use for quality")), ToolInput(tag="in_title", input_type=String(optional=True), prefix="--title", doc=InputDocumentation(doc="This sets the title for the whole plot. Use --title\n'Crustacean's DNA read quality' if you need single\nquote or apostrophe inside title.")), ToolInput(tag="in_add_y_axes", input_type=Boolean(optional=True), prefix="--add-yaxes", doc=InputDocumentation(doc="Add Y-axes to both marginal histograms.")), ToolInput(tag="in_f", input_type=String(optional=True), prefix="-f", doc=InputDocumentation(doc="")), ToolInput(tag="in_higher", input_type=String(), position=0, doc=InputDocumentation(doc="-f FASTQ, --fastq FASTQ"))], outputs=[ToolOutput(tag="out_no_timestamp", output_type=File(optional=True), selector=InputSelector(input_to_select="in_no_timestamp", type_hint=File()), doc=OutputDocumentation(doc="Turn off time stamps in the filename output."))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Pauvre_Marginplot_V0_1_0().translate("wdl", allow_empty_container=True)

