from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import String, Directory, Int

Tepeaks_Narrow_V0_1_0 = CommandToolBuilder(tool="TEpeaks_narrow", base_command=["TEpeaks", "narrow"], inputs=[ToolInput(tag="in_treatment", input_type=String(optional=True), prefix="--treatment", doc=InputDocumentation(doc="IP sample ( BAM )")), ToolInput(tag="in_control", input_type=String(optional=True), prefix="--control", doc=InputDocumentation(doc="Control IP samples ( BAM )")), ToolInput(tag="in_output_dir", input_type=Directory(optional=True), prefix="--outputdir", doc=InputDocumentation(doc="Directory to write output to")), ToolInput(tag="in_fra_glen", input_type=Int(optional=True), prefix="--fraglen", doc=InputDocumentation(doc="Fragment size (default: 200)\n--keepDup=STRING          How to deal with duplicate reads. The valid values are 'auto', 'all', or 1 (default: auto)\n--shift=INT               Shift reads towards 3' end, if positive, or 5' end if negative. (default: 0)\n--lmfold=INT              Lower limit of fold ratio against background to build model (default: 10)\n--hmfold=INT              Higher limit of fold ratio against background to build model (default: 30)")), ToolInput(tag="in_prj_name", input_type=String(optional=True), prefix="--prjname", doc=InputDocumentation(doc="Name of the prject (default: NONAME)")), ToolInput(tag="in_pval", input_type=Int(optional=True), prefix="--pval", doc=InputDocumentation(doc="P-value cutoff (default: 1e-5)\n--fdr=DOUBLE              False discovery rate cutoff (default: 0.05)\n--toLarge                 Scale library size to large sample (default: off)")), ToolInput(tag="in_species", input_type=Int(optional=True), prefix="--species", doc=InputDocumentation(doc="Species e.g., hs (Human hg19),  mm (Mouse mm9). (default: hs)")), ToolInput(tag="in_g_size", input_type=Int(optional=True), prefix="--gsize", doc=InputDocumentation(doc="Effective genome size (default: human genome 2.7e9)\n--threads=INT             Number of threads to use (default: 1)\n--pileup=INT              The minuim pileup required for peaks with multi-reads (default: 20)\n--fe=DOUBLE               The minuim fold enrichment required for peaks with multi-reads (default: 3)")), ToolInput(tag="in_num_it_r", input_type=Int(optional=True), prefix="--numItr", doc=InputDocumentation(doc="Number of iterations (default: 50)")), ToolInput(tag="in_arguments", input_type=String(optional=True), position=0, doc=InputDocumentation(doc="")), ToolInput(tag="in_alignment_files", input_type=String(), position=1, doc=InputDocumentation(doc=""))], outputs=[ToolOutput(tag="out_output_dir", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_output_dir", type_hint=File()), doc=OutputDocumentation(doc="Directory to write output to"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Tepeaks_Narrow_V0_1_0().translate("wdl", allow_empty_container=True)

