from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean

Fastme_V0_1_0 = CommandToolBuilder(tool="fastme", base_command=["fastme"], inputs=[ToolInput(tag="in_minput_data_filem", input_type=Boolean(optional=True), prefix="-i", doc=InputDocumentation(doc="[00;04minput data file[00;01m, --input_data=[00;04minput data file[00;00m\nThe [00;04minput data file[00;00m contains sequence alignment(s)[00;00m\nor a distance matrix(ces).")), ToolInput(tag="in_minput_user_tree", input_type=Boolean(optional=True), prefix="-u", doc=InputDocumentation(doc="[00;04minput user tree file[00;01m, --user_tree=[00;04minput user tree file[00;00m\n[00;01mFastME [00;00mmay use an existing topology available in the [00;04minput user tree file[00;00m\nwhich corresponds to the input dataset.")), ToolInput(tag="in_moutput_tree_filem", input_type=Boolean(optional=True), prefix="-o", doc=InputDocumentation(doc="[00;04moutput tree file[00;01m, --output_tree=[00;04moutput tree file[00;00m\n[00;01mFastME [00;00mwill write the infered tree into the [00;04moutput tree file[00;00m.")), ToolInput(tag="in_moutput_matrix_filem", input_type=Boolean(optional=True), prefix="-O", doc=InputDocumentation(doc="[00;04moutput matrix file[00;01m, --output_matrix=[00;04moutput matrix file[00;00m\nUse this option if you want [00;01mFastME [00;00mto write the distances[00;00m\nmatrix computed from the input alignment in the [00;04moutput matrix file[00;00m.")), ToolInput(tag="in_moutput_information_filem", input_type=Boolean(optional=True), prefix="-I", doc=InputDocumentation(doc="[00;04moutput information file[00;01m, --output_info=[00;04moutput information file[00;00m\nUse this option if you want [00;01mFastME [00;00mto write information[00;00m\nabout its execution in the [00;04moutput information file[00;00m.")), ToolInput(tag="in_moutput_bootstrap_trees", input_type=Boolean(optional=True), prefix="-B", doc=InputDocumentation(doc="[00;04moutput bootstrap trees file[00;01m, --output_boot=[00;04moutput bootstrap trees file[00;00m\nUse this option if you want [00;01mFastME [00;00mto write bootstrap trees[00;00m\nin the [00;04mbootstrap trees file[00;00m.")), ToolInput(tag="in_append", input_type=Boolean(optional=True), prefix="--append", doc=InputDocumentation(doc="[00;00m\nUse this option to append results to existing output files (if any).[00;00m\nBy default output files will be overwritten.")), ToolInput(tag="in_mmethodm__methodmmethodmmfastme", input_type=Boolean(optional=True), prefix="-m", doc=InputDocumentation(doc="[00;04mmethod[00;01m, --method=[00;04mmethod[00;00m\n[00;01mFastME [00;00mcomputes a tree using a distance algorithm.[00;00m\nYou may choose this [00;04mmethod[00;00m from:[00;00m\n[00;01mTaxAdd_(B)alME[00;00m, [00;01mTaxAdd_(O)LSME[00;00m, [00;01mB(I)ONJ[00;00m (default),[00;00m\n[00;01m(N)J[00;00m or [00;01m(U)NJ[00;00m.")), ToolInput(tag="in_muse_option_dna", input_type=Boolean(optional=True), prefix="-d", doc=InputDocumentation(doc="[00;04m[model][00;01m, --dna=[00;04m[model][00;00m\nUse this option if your input data file contains DNA sequences alignment(s).[00;00m\nYou may also indicate the evolutionary [00;04m[model][00;00m which can be choosen from:[00;00m\n[00;01m(p)-distance[00;00m, [00;01mR(Y) symmetric[00;00m, [00;01m(R)Y[00;00m, [00;01m(J)C69[00;00m, [00;01m(K)2P[00;00m, [00;01mF8(1)[00;00m, [00;01mF8(4)[00;00m (default), [00;01m(T)N93[00;00m, [00;01m(L)ogDet[00;00m.")), ToolInput(tag="in_muse_option_protein", input_type=Boolean(optional=True), prefix="-p", doc=InputDocumentation(doc="[00;04m[model][00;01m, --protein=[00;04m[model][00;00m\nUse this option if your input data file contains protein sequences alignment(s).[00;00m\nYou may also indicate the evolutionary [00;04m[model][00;00m which can be choosen from:[00;00m\n[00;01m(p)-distance[00;00m, [00;01m(F)81 like[00;00m, [00;01m(L)G[00;00m (default), [00;01m(W)AG[00;00m, [00;01m(J)TT[00;00m, [00;01mDay(h)off[00;00m, [00;00m\n[00;01m(D)CMut[00;00m, [00;01m(C)pRev[00;00m, [00;01m(M)tREV[00;00m, [00;01m(R)tREV[00;00m, [00;01mHIV(b)[00;00m, [00;01mH(I)Vw[00;00m or [00;01mFL(U)[00;00m.")), ToolInput(tag="in_remove_gap", input_type=Boolean(optional=True), prefix="--remove_gap", doc=InputDocumentation(doc="[00;00m\nUse this option to completely remove any site which has a gap in[00;00m\nany sequence. By default, [00;01mFastME [00;00mis doing pairwise deletion of gaps.")), ToolInput(tag="in_equilibrium", input_type=Boolean(optional=True), prefix="--equilibrium", doc=InputDocumentation(doc="[00;00m\nThe equilibrium frequencies for DNA are always estimated by counting[00;00m\nthe occurence of the nucleotides in the input alignment.[00;00m\nFor amino-acid sequences, the equilibrium frequencies are estimated[00;00m\nusing the frequencies defined by the substitution model.[00;00m\nUse this option if you whish to estimate the amino-acid frequencies[00;00m\nby counting their occurence in the input alignment.")), ToolInput(tag="in_muse_option_you_wish", input_type=Boolean(optional=True), prefix="-g", doc=InputDocumentation(doc="[00;04m[alpha][00;01m, --gamma=[00;04m[alpha][00;00m\nUse this option if you wish to have gamma distributed rates across sites.[00;00m\nBy default, FastME runs with no gamma variation.[00;00m\nIf running FastME with gamma distributed rates across sites, the [00;04m[alpha][00;00m default value is 1.0.[00;00m\nOnly helpful when the input data file contains sequences alignment(s).")), ToolInput(tag="in_muse_option_do_m", input_type=Boolean(optional=True), prefix="-n", doc=InputDocumentation(doc="[00;04m[NNI][00;01m, --nni=[00;04m[NNI][00;00m\nUse this option to do [00;04m[NNI][00;00m tree topology improvement.[00;00m\nYou may choose the [00;04m[NNI][00;00m type from:[00;00m\n[00;01mNNI_(B)alME[00;00m (default) or [00;01mNNI_(O)LS[00;00m.")), ToolInput(tag="in_spr", input_type=Boolean(optional=True), prefix="--spr", doc=InputDocumentation(doc="[00;00m\nUse this option to do [00;04mSPR[00;00m tree topology improvement.")), ToolInput(tag="in_mbranchm__branchlengthmbranchmuse", input_type=Boolean(optional=True), prefix="-w", doc=InputDocumentation(doc="[00;04mbranch[00;01m, --branch_length=[00;04mbranch[00;00m\nUse this option to indicate the [00;04mbranch[00;00m length to assign to the tree.[00;00m\nYou may choose the [00;04mbranch[00;00m length from: [00;01m(B)alLS[00;00m (default), [00;01m(O)LS[00;00m\nor [00;01m(n)one[00;00m. [00;01m(n)one [00;00mis only available with BIONJ, NJ or UNJ.[00;00m\nOnly helpful when not improving the tree topology (no NNI nor SPR).")), ToolInput(tag="in_mdatasetsm__option", input_type=Boolean(optional=True), prefix="-D", doc=InputDocumentation(doc="[00;04mdatasets[00;01m, --datasets=[00;04mdatasets[00;00m\nUse this option to indicate the number of [00;04mdatasets[00;00m in your input[00;00m\ndata file. Default value is 1.")), ToolInput(tag="in_mreplicatesm__bootstrapmreplicatesmuse", input_type=Boolean(optional=True), prefix="-b", doc=InputDocumentation(doc="[00;04mreplicates[00;01m, --bootstrap=[00;04mreplicates[00;00m\nUse this option to indicate the number of [00;04mreplicates[00;01m FastME [00;00mwill[00;00m\ndo for bootstrapping. Default value is 0.[00;00m\nOnly helpful when the input data file contains sequences alignment(s).")), ToolInput(tag="in_mseedm_seedmseedmuse_option", input_type=Boolean(optional=True), prefix="-z", doc=InputDocumentation(doc="[00;04mseed[00;01m, --seed=[00;04mseed[00;00m\nUse this option to initialize randomization with [00;04mseed[00;00m value.[00;00m\nOnly helpful when bootstrapping.")), ToolInput(tag="in_muse_option_you_want", input_type=Boolean(optional=True), prefix="-c", doc=InputDocumentation(doc="[00;00m\nUse this option if you want FastME only to compute distance matrix.[00;00m\nOnly helpful when the input data file contains sequences alignment(s).")), ToolInput(tag="in_mnumber_digitsmuse_option", input_type=Boolean(optional=True), prefix="-f", doc=InputDocumentation(doc="[00;04mnumber of digits[00;00m\nUse this option to set the number of digits after the dot to use on output.[00;00m\nDefault precision is 12.")), ToolInput(tag="in_mnumber__nbthreadsmnumber", input_type=Boolean(optional=True), prefix="-T", doc=InputDocumentation(doc="[00;04mnumber of threads[00;01m, --nb_threads=[00;04mnumber of threads[00;00m\nUse this option to set the number of threads to use.[00;00m\nDefault [00;04mnumber of threads[00;00m is 8.")), ToolInput(tag="in_mvaluem__verbosemvaluemsets", input_type=Boolean(optional=True), prefix="-v", doc=InputDocumentation(doc="[00;04mvalue[00;01m, --verbose=[00;04mvalue[00;00m\nSets the verbose level to [00;04mvalue[00;00m [0-3].[00;00m\nDefault [00;04mvalue[00;00m is 0."))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Fastme_V0_1_0().translate("wdl", allow_empty_container=True)

