from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Int, File, Boolean, String

Pirs_Diploid_V0_1_0 = CommandToolBuilder(tool="pirs_diploid", base_command=["pirs", "diploid"], inputs=[ToolInput(tag="in_snp_rate", input_type=Int(optional=True), prefix="--snp-rate", doc=InputDocumentation(doc="A floating-point number in the interval [0, 1] that\nspecifies the heterozygous SNP rate.  Default: 0.001")), ToolInput(tag="in_in_del_rate", input_type=Int(optional=True), prefix="--indel-rate", doc=InputDocumentation(doc="A floating-point number in the interval [0, 1] that\nspecifies the heterozygous indel rate.\nDefault: 0.0001")), ToolInput(tag="in_sv_rate", input_type=Int(optional=True), prefix="--sv-rate", doc=InputDocumentation(doc="A floating-point number in the interval [0, 1] that\nspecifies the large-scale structural variation\n(insertion, deletion, inversion) rate in the diploid\ngenome. Default: 0.000001")), ToolInput(tag="in_transition_to_transversion_ratio", input_type=Int(optional=True), prefix="--transition-to-transversion-ratio", doc=InputDocumentation(doc="In a SNP, a transition is when a purine or pyrimidine\nis changed to one of the same (A <=> G, C <=> T)\nwhile a transversion is when a purine is changed\ninto a pyrimidine or vice-versa.  This option\nspecifies a floating-point number RATIO that gives\nthe ratio of the transition probability to the\ntransversion probability for simulated SNPs.\nDefault: 2.0")), ToolInput(tag="in_output_prefix", input_type=File(optional=True), prefix="--output-prefix", doc=InputDocumentation(doc="Use PREFIX as the prefix of the output file and logs.\nDefault: 'pirs_diploid'")), ToolInput(tag="in_output_file", input_type=File(optional=True), prefix="--output-file", doc=InputDocumentation(doc="Use FILE as the name of the output file. Use '-'\nfor standard output; this also moves the\ninformational messages from stdout to stderr.")), ToolInput(tag="in_output_file_type", input_type=File(optional=True), prefix="--output-file-type", doc=InputDocumentation(doc="The string 'text' or 'gzip' to specify the type of\nthe output FASTA file containing the diploid copy\nof the genome, as well as the log files.\nDefault: 'text'")), ToolInput(tag="in_no_logs", input_type=Boolean(optional=True), prefix="--no-logs", doc=InputDocumentation(doc="Do not write the log files.")), ToolInput(tag="in_random_seed", input_type=String(optional=True), prefix="--random-seed", doc=InputDocumentation(doc="Use SEED as the random seed. Default:\ntime(NULL) * getpid()")), ToolInput(tag="in_quiet", input_type=Boolean(optional=True), prefix="--quiet", doc=InputDocumentation(doc="Do not print informational messages.")), ToolInput(tag="in_one_bp", input_type=Int(), position=0, doc=InputDocumentation(doc="64.82%")), ToolInput(tag="in_two_bp", input_type=Int(), position=1, doc=InputDocumentation(doc="17.17%")), ToolInput(tag="in_three_bp", input_type=Int(), position=2, doc=InputDocumentation(doc="7.20%")), ToolInput(tag="in_four_bp", input_type=Int(), position=3, doc=InputDocumentation(doc="7.29%")), ToolInput(tag="in_five_bp", input_type=Int(), position=4, doc=InputDocumentation(doc="2.18%")), ToolInput(tag="in_six_bp", input_type=Int(), position=5, doc=InputDocumentation(doc="1.34%")), ToolInput(tag="in_one_zero_zero_bp", input_type=Int(), position=0, doc=InputDocumentation(doc="70%")), ToolInput(tag="in_two_zero_zero_bp", input_type=Int(), position=1, doc=InputDocumentation(doc="20%")), ToolInput(tag="in_five_zero_zero_bp", input_type=Int(), position=2, doc=InputDocumentation(doc="7%")), ToolInput(tag="in_one_zero_zero_zero_bp", input_type=Int(), position=3, doc=InputDocumentation(doc="2%")), ToolInput(tag="in_two_zero_zero_zero_bp", input_type=Int(), position=4, doc=InputDocumentation(doc="1%"))], outputs=[ToolOutput(tag="out_output_prefix", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output_prefix", type_hint=File()), doc=OutputDocumentation(doc="Use PREFIX as the prefix of the output file and logs.\nDefault: 'pirs_diploid'")), ToolOutput(tag="out_output_file", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output_file", type_hint=File()), doc=OutputDocumentation(doc="Use FILE as the name of the output file. Use '-'\nfor standard output; this also moves the\ninformational messages from stdout to stderr.")), ToolOutput(tag="out_output_file_type", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output_file_type", type_hint=File()), doc=OutputDocumentation(doc="The string 'text' or 'gzip' to specify the type of\nthe output FASTA file containing the diploid copy\nof the genome, as well as the log files.\nDefault: 'text'"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Pirs_Diploid_V0_1_0().translate("wdl", allow_empty_container=True)

