from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Directory, Int, Boolean, String

Slamdunk_Map_V0_1_0 = CommandToolBuilder(tool="slamdunk_map", base_command=["slamdunk", "map"], inputs=[ToolInput(tag="in_reference", input_type=File(optional=True), prefix="--reference", doc=InputDocumentation(doc="Reference fasta file")), ToolInput(tag="in_output_dir", input_type=Directory(optional=True), prefix="--outputDir", doc=InputDocumentation(doc="Output directory for mapped BAM files.")), ToolInput(tag="in_trim_five_p", input_type=Int(optional=True), prefix="--trim-5p", doc=InputDocumentation(doc="Number of bp removed from 5' end of all reads.\n(default: 12)")), ToolInput(tag="in_top_n", input_type=Int(optional=True), prefix="--topn", doc=InputDocumentation(doc="Max. number of alignments to report per read (default:\n1)")), ToolInput(tag="in_max_polya", input_type=Int(optional=True), prefix="--max-polya", doc=InputDocumentation(doc="Max number of As at the 3' end of a read. (default: 4)")), ToolInput(tag="in_threads", input_type=Int(optional=True), prefix="--threads", doc=InputDocumentation(doc="Thread number (default: 1)")), ToolInput(tag="in_quant_seq", input_type=Boolean(optional=True), prefix="--quantseq", doc=InputDocumentation(doc="Run plain Quantseq alignment without SLAM-seq scoring\n(default: False)")), ToolInput(tag="in_end_to_end", input_type=Boolean(optional=True), prefix="--endtoend", doc=InputDocumentation(doc="Use a end to end alignment algorithm for mapping.\n(default: False)")), ToolInput(tag="in_sample_name", input_type=String(optional=True), prefix="--sampleName", doc=InputDocumentation(doc="Use this sample name for all supplied samples\n(default: None)")), ToolInput(tag="in_sample_type", input_type=String(optional=True), prefix="--sampleType", doc=InputDocumentation(doc="Use this sample type for all supplied samples\n(default: pulse)")), ToolInput(tag="in_sample_time", input_type=Int(optional=True), prefix="--sampleTime", doc=InputDocumentation(doc="Use this sample time for all supplied samples\n(default: 0)")), ToolInput(tag="in_sample_index", input_type=Int(optional=True), prefix="--sample-index", doc=InputDocumentation(doc="Run analysis only for sample <i>. Use for distributing\nslamdunk analysis on a cluster (index is 1-based).\n(default: -1)")), ToolInput(tag="in_skip_sam", input_type=Boolean(optional=True), prefix="--skip-sam", doc=InputDocumentation(doc="Output BAM while mapping. Slower but, uses less hard\ndisk. (default: False)\n")), ToolInput(tag="in_files", input_type=String(), position=0, doc=InputDocumentation(doc="Single csv/tsv file (recommended) containing all\nsample files and sample info or a list of all sample\nBAM/FASTA(gz)/FASTQ(gz) files"))], outputs=[ToolOutput(tag="out_output_dir", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_output_dir", type_hint=File()), doc=OutputDocumentation(doc="Output directory for mapped BAM files."))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Slamdunk_Map_V0_1_0().translate("wdl", allow_empty_container=True)

