from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Int, File, Boolean, String

Kissplice_V0_1_0 = CommandToolBuilder(tool="kissplice", base_command=["kissplice"], inputs=[ToolInput(tag="in_input_fastaq_read", input_type=Int(optional=True), prefix="-r", doc=InputDocumentation(doc="input fasta/q read files or compressed (.gz) fasta/q\nfiles (mutiple, such as '-r file1 -r file2...')")), ToolInput(tag="in_kmer_size_default", input_type=Int(optional=True), prefix="-k", doc=InputDocumentation(doc="k-mer size (default=41)")), ToolInput(tag="in_maximum_number_branching", input_type=Int(optional=True), prefix="-b", doc=InputDocumentation(doc="maximum number of branching nodes (default 5)")), ToolInput(tag="in_maximal_length_shorter", input_type=Int(optional=True), prefix="-l", doc=InputDocumentation(doc="maximal length of the shorter path (default: 2k+1)")), ToolInput(tag="in_minimum_length_", input_type=Int(optional=True), prefix="-m", doc=InputDocumentation(doc="minimum length of the shorter path (default 2k-8)")), ToolInput(tag="in_maximum_length_exons", input_type=Int(optional=True), prefix="-M", doc=InputDocumentation(doc="maximum length of the longest path (default 1000000),\nskipped exons longer than UL_MAX are not reported")), ToolInput(tag="in_path_prefix_prebuilt", input_type=File(optional=True), prefix="-g", doc=InputDocumentation(doc="path and prefix to pre-built de Bruijn graph (suffixed\nby .edges/.nodes) if jointly used with -r, graph used\nto find bubbles and reads used for quantification")), ToolInput(tag="in_path_store_results", input_type=File(optional=True), prefix="-o", doc=InputDocumentation(doc="path to store the results and the summary log file\n(default = ./results)")), ToolInput(tag="in_specific_directory_absolute", input_type=File(optional=True), prefix="-d", doc=InputDocumentation(doc="specific directory (absolute path) where to build\ntemporary files (default temporary directory\notherwise)")), ToolInput(tag="in_number_cores_be", input_type=Int(optional=True), prefix="-t", doc=InputDocumentation(doc="number of cores (must be <= number of physical cores)")), ToolInput(tag="in_changes_types_be", input_type=Int(optional=True), prefix="-s", doc=InputDocumentation(doc="0, 1 or 2. Changes which types of SNPs will be output.\nIf 0 (default), will not output SNPs. If 1, will\noutput Type0a-SNPs. If 2, will output Type0a and\nType0b SNPs (warning: this option may increase a lot\nthe running time. You might also want to try the\nexperimental algorithm here)")), ToolInput(tag="in_verbose_mode", input_type=Boolean(optional=True), prefix="-v", doc=InputDocumentation(doc="Verbose mode")), ToolInput(tag="in_keep_nodesedges_file", input_type=Boolean(optional=True), prefix="-u", doc=InputDocumentation(doc="keep the nodes/edges file for unfinished bccs")), ToolInput(tag="in_integer_kmers_present", input_type=Int(optional=True), prefix="-c", doc=InputDocumentation(doc="an integer, k-mers present strictly less than this\nnumber of times in the dataset will be discarded\n(default 2)")), ToolInput(tag="in_percentage_edges_relative", input_type=Int(optional=True), prefix="-C", doc=InputDocumentation(doc="a percentage from [0,1), edges with relative coverage\nbelow this number are removed (default 0.05)")), ToolInput(tag="in_estimated_genometranscriptome_size", input_type=Int(optional=True), prefix="-z", doc=InputDocumentation(doc="estimated genome/transcriptome size (default = 1G)")), ToolInput(tag="in_edit_distance_threshold", input_type=Int(optional=True), prefix="-e", doc=InputDocumentation(doc="edit distance threshold, if the two sequences (paths)\nof a bubble have edit distance smaller than this\nthreshold, the bubble is classified as an inexact\nrepeat (default 3)")), ToolInput(tag="in_maximal_number_bubbles", input_type=Int(optional=True), prefix="-y", doc=InputDocumentation(doc="maximal number of bubbles enumeration in each bcc. If\nexceeded, no bubble is output for the bcc (default\n100M)")), ToolInput(tag="in_mismatches", input_type=Int(optional=True), prefix="--mismatches", doc=InputDocumentation(doc="Maximal number of substitutions authorized between a\nread and a fragment (for quantification only), default\n0. If you increase the mismatch and use --counts think\nof increasing min_overlap too.")), ToolInput(tag="in_counts", input_type=Int(optional=True), prefix="--counts", doc=InputDocumentation(doc="0,1 or 2 . Changes how the counts will be reported. If\n0 (default): total counts, if 1: counts on junctions,\nif 2: all counts. see User guide for more information")), ToolInput(tag="in_min_overlap", input_type=Int(optional=True), prefix="--min_overlap", doc=InputDocumentation(doc="Set how many nt must overlap a junction to be counted\nby --counts option. Default=3. see User guide for more\ninformation")), ToolInput(tag="in_timeout", input_type=Int(optional=True), prefix="--timeout", doc=InputDocumentation(doc="max amount of time (in seconds) spent for enumerating\nbubbles in each bcc. If exceeded, no bubble is output\nfor the bcc (default 100000)")), ToolInput(tag="in_output_context", input_type=Boolean(optional=True), prefix="--output-context", doc=InputDocumentation(doc="Will output the maximum non-ambiguous context of a")), ToolInput(tag="in_output_branch_count", input_type=Boolean(optional=True), prefix="--output-branch-count", doc=InputDocumentation(doc="Will output the number of branching nodes in each\npath.")), ToolInput(tag="in_keep_bccs", input_type=Boolean(optional=True), prefix="--keep-bccs", doc=InputDocumentation(doc="Keep the node/edges files for all bccs.")), ToolInput(tag="in_experimental", input_type=Boolean(optional=True), prefix="--experimental", doc=InputDocumentation(doc="Uses a new experimental algorithm that searches for\nbubbles by listing all paths.")), ToolInput(tag="in_max_memory", input_type=Int(optional=True), prefix="--max-memory", doc=InputDocumentation(doc="If you use the experimental algorithm, you must\nprovide the maximum size of the process's virtual\nmemory (address space) in megabytes (default\nunlimited). WARNING: this option does not work on Mac\noperating systems.")), ToolInput(tag="in_keep_counts", input_type=Boolean(optional=True), prefix="--keep-counts", doc=InputDocumentation(doc="Keep the .counts file after the sequencing-errors-\nremoval step.\n")), ToolInput(tag="in_bubble", input_type=String(), position=0, doc=InputDocumentation(doc="--output-path         Will output the id of the nodes composing the two"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Kissplice_V0_1_0().translate("wdl", allow_empty_container=True)

