from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Directory, String, Boolean, Int, Float

Dfast_V0_1_0 = CommandToolBuilder(tool="dfast", base_command=["dfast"], inputs=[ToolInput(tag="in_genome", input_type=File(optional=True), prefix="--genome", doc=InputDocumentation(doc="Genomic FASTA file")), ToolInput(tag="in_out", input_type=Directory(optional=True), prefix="--out", doc=InputDocumentation(doc="Output directory (default:OUT)")), ToolInput(tag="in_config", input_type=File(optional=True), prefix="--config", doc=InputDocumentation(doc="Configuration file (default config will be used if not\nspecified)")), ToolInput(tag="in_organism", input_type=String(optional=True), prefix="--organism", doc=InputDocumentation(doc="Organism name")), ToolInput(tag="in_strain", input_type=String(optional=True), prefix="--strain", doc=InputDocumentation(doc="Strain name")), ToolInput(tag="in_complete", input_type=Boolean(optional=True), prefix="--complete", doc=InputDocumentation(doc="Treat the query as a complete genome. Not required\nunless you need INSDC submission files.\n[t|f(=default)]")), ToolInput(tag="in_use_original_name", input_type=Boolean(optional=True), prefix="--use_original_name", doc=InputDocumentation(doc="Use original sequence names in a query FASTA file\n[t|f(=default)]")), ToolInput(tag="in_sort_sequence", input_type=Boolean(optional=True), prefix="--sort_sequence", doc=InputDocumentation(doc="Sort sequences by length [t(=default)|f]")), ToolInput(tag="in_minimum_length", input_type=Int(optional=True), prefix="--minimum_length", doc=InputDocumentation(doc="Minimum sequence length (default:200)")), ToolInput(tag="in_fix_origin", input_type=Boolean(optional=True), prefix="--fix_origin", doc=InputDocumentation(doc="Rotate/flip the chromosome so that the dnaA gene comes\nfirst. (ONLY FOR A FINISHED GENOME)")), ToolInput(tag="in_offset", input_type=Int(optional=True), prefix="--offset", doc=InputDocumentation(doc="Offset from the start codon of the dnaA gene. (for")), ToolInput(tag="in_locus_tag_prefix", input_type=String(optional=True), prefix="--locus_tag_prefix", doc=InputDocumentation(doc="Locus tag prefix (defaut:LOCUS)")), ToolInput(tag="in_step", input_type=Int(optional=True), prefix="--step", doc=InputDocumentation(doc="Increment step of locus tag (default:10)")), ToolInput(tag="in_use_separate_tags", input_type=Boolean(optional=True), prefix="--use_separate_tags", doc=InputDocumentation(doc="Use separate tags according to feature types\n[t(=default)|f]")), ToolInput(tag="in_threshold", input_type=Float(optional=True), prefix="--threshold", doc=InputDocumentation(doc="Thresholds for default database search (format:\n'pident,q_cov,s_cov,e_value', default: '0,75,75,1e-6')")), ToolInput(tag="in_database", input_type=File(optional=True), prefix="--database", doc=InputDocumentation(doc="Additional reference database to be searched against\nprior to the default database. (format:\ndb_path[,db_name[,pident,q_cov,s_cov,e_value]])")), ToolInput(tag="in_references", input_type=File(optional=True), prefix="--references", doc=InputDocumentation(doc="Reference file(s) for OrthoSearch. Use semicolons for\nmultiple files, e.g. 'genome1.faa;genome2.gbk'")), ToolInput(tag="in_aligner", input_type=String(optional=True), prefix="--aligner", doc=InputDocumentation(doc="Aligner to use [ghostx(=default)|blastp|diamond]")), ToolInput(tag="in_use_prodigal", input_type=Boolean(optional=True), prefix="--use_prodigal", doc=InputDocumentation(doc="Use Prodigal to predict CDS instead of MGA")), ToolInput(tag="in_use_gene_marks_two", input_type=Int(optional=True), prefix="--use_genemarks2", doc=InputDocumentation(doc="Use GeneMarkS2 to predict CDS instead of MGA.\n[auto|bact|arch]")), ToolInput(tag="in_use_trnascan", input_type=String(optional=True), prefix="--use_trnascan", doc=InputDocumentation(doc="Use tRNAscan-SE to predict tRNA instead of Aragorn.\n[bact|arch]")), ToolInput(tag="in_use_rn_ammer", input_type=String(optional=True), prefix="--use_rnammer", doc=InputDocumentation(doc="Use RNAmmer to predict rRNA instead of Barrnap.\n[bact|arch]")), ToolInput(tag="in_g_code", input_type=Int(optional=True), prefix="--gcode", doc=InputDocumentation(doc="Genetic code [11(=default),4(=Mycoplasma)]")), ToolInput(tag="in_no_hmm", input_type=Boolean(optional=True), prefix="--no_hmm", doc=InputDocumentation(doc="Disable HMMscan")), ToolInput(tag="in_no_cdd", input_type=Boolean(optional=True), prefix="--no_cdd", doc=InputDocumentation(doc="Disable CDDsearch")), ToolInput(tag="in_no_cds", input_type=Boolean(optional=True), prefix="--no_cds", doc=InputDocumentation(doc="Disable CDS prediction")), ToolInput(tag="in_no_rrna", input_type=Boolean(optional=True), prefix="--no_rrna", doc=InputDocumentation(doc="Disable rRNA prediction")), ToolInput(tag="in_no_trna", input_type=Boolean(optional=True), prefix="--no_trna", doc=InputDocumentation(doc="Disable tRNA prediction")), ToolInput(tag="in_no_crisp_r", input_type=Boolean(optional=True), prefix="--no_crispr", doc=InputDocumentation(doc="Disable CRISPR prediction")), ToolInput(tag="in_gff", input_type=String(optional=True), prefix="--gff", doc=InputDocumentation(doc="[Preliminary implementation] Read GFF to import\nstructural annotation. Ignores --use_original_name,\n--sort_sequence, --fix_origin.")), ToolInput(tag="in_seq_names", input_type=String(optional=True), prefix="--seq_names", doc=InputDocumentation(doc="Sequence names for each sequence (for complete genome)")), ToolInput(tag="in_seq_types", input_type=String(optional=True), prefix="--seq_types", doc=InputDocumentation(doc="Sequence types for each sequence (chromosome/plasmid,\nfor complete genome)")), ToolInput(tag="in_seq_topologies", input_type=String(optional=True), prefix="--seq_topologies", doc=InputDocumentation(doc="Sequence topologies for each sequence\n(linear/circular, for complete genome)")), ToolInput(tag="in_additional_modifiers", input_type=String(optional=True), prefix="--additional_modifiers", doc=InputDocumentation(doc="Additional modifiers for source features")), ToolInput(tag="in_metadata_file", input_type=File(optional=True), prefix="--metadata_file", doc=InputDocumentation(doc="Path to a metadata file (optional for DDBJ submission\nfile)")), ToolInput(tag="in_center_name", input_type=File(optional=True), prefix="--center_name", doc=InputDocumentation(doc="Genome center name (optional for GenBank submission\nfile)")), ToolInput(tag="in_cpu", input_type=Int(optional=True), prefix="--cpu", doc=InputDocumentation(doc="Number of CPUs to use")), ToolInput(tag="in_use_locus_tag_as_gene_id", input_type=Boolean(optional=True), prefix="--use_locustag_as_gene_id", doc=InputDocumentation(doc="Use locustag as gene ID for FASTA and GFF. (Useful\nwhen providing DFAST results to other tools such as\nRoary)")), ToolInput(tag="in_db_root", input_type=File(optional=True), prefix="--dbroot", doc=InputDocumentation(doc="DB root directory (default:APP_ROOT/db")), ToolInput(tag="in_force", input_type=Boolean(optional=True), prefix="--force", doc=InputDocumentation(doc="Force overwriting output")), ToolInput(tag="in_debug", input_type=Boolean(optional=True), prefix="--debug", doc=InputDocumentation(doc="Run in debug mode (Extra logging and retaining\ntemporary files)")), ToolInput(tag="in_show_config", input_type=Boolean(optional=True), prefix="--show_config", doc=InputDocumentation(doc="Show pipeline configuration and exit"))], outputs=[ToolOutput(tag="out_out", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_out", type_hint=File()), doc=OutputDocumentation(doc="Output directory (default:OUT)"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Dfast_V0_1_0().translate("wdl", allow_empty_container=True)

