class: CommandLineTool
id: motifscan.cwl
inputs:
- id: peak
  doc: 'Required input genomic regions file to scan motifs on. (default: None)'
  type: string
  inputBinding:
    prefix: --peak
- id: format
  doc: 'Format of input genomic regions. See documents for more information. (default:
    bed5col)'
  type: string
  inputBinding:
    prefix: --format
- id: motif
  doc: "Required pre-compiled motif file, which is generated by 'motifcompile'. (default:\
    \ None)"
  type: string
  inputBinding:
    prefix: --motif
- id: genome
  doc: "Required genome directory, which contains genome sequence and chromosome length.\
    \ You can run 'genomecompile' to generate the directory. (default: None)"
  type: string
  inputBinding:
    prefix: --genome
- id: annotation
  doc: 'Gene annotation file, which is used to generate random controls when performing
    enrichment analysis. Sequence features on promoters are very different from those
    on whole genome. We will pick up random control regions with similar genomic locations
    for each input region instead of the whole genome. (default: None)'
  type: string
  inputBinding:
    prefix: --annotation
- id: motif_list
  doc: 'Motif list file. If specified, the program will only perform MotifScan on
    given motifs in this list. (default: None)'
  type: string
  inputBinding:
    prefix: --motif_list
- id: no_enrichment
  doc: 'If set, MotifScan will neither generate random control regions nor perform
    enrichment analysis. (default: True)'
  type: boolean
  inputBinding:
    prefix: --no_enrichment
- id: control
  doc: 'User specified control regions. (default: None)'
  type: string
  inputBinding:
    prefix: --control
- id: c_format
  doc: 'Format of control regions (same as -f for input regions). (default: bed5col)'
  type: string
  inputBinding:
    prefix: --cformat
- id: r
  doc: "Genomic location to apply MotifScan. 'all': perform MotifScan on the all input\
    \ regions. 'promoter': perform MotifScan on regions which located at promoter\
    \ regions. 'distal': perform MotifScan on regions which located at distal regions.\
    \ 'gene': perform MotifScan on regions which is the target(nearest one) of genes\
    \ within 10kbp distance cutoff. (default: all)"
  type: string
  inputBinding:
    prefix: -r
- id: upstream
  doc: "Upstream distance to TSS to define promoter regions. Valid when option -r\
    \ is set as 'promoter' or 'distal'. (default: 4000)"
  type: string
  inputBinding:
    prefix: --upstream
- id: downstream
  doc: "Upstream distance to TSS to define promoter regions. Valid when option -r\
    \ is set as 'promoter' or 'distal'. (default: 2000)"
  type: string
  inputBinding:
    prefix: --downstream
- id: random_times
  doc: 'The number of random control regions generated for each input region. For
    example, if you provide 1000 regions, 5000 random regions will be generated by
    default. (default: 5)'
  type: string
  inputBinding:
    prefix: --random_times
- id: peak_length
  doc: 'The length of input regions to perform MotifScan around peak summit/midpoint.
    If set to 0, the whole peak region is used to perform MotifScan. (default: 1000)'
  type: string
  inputBinding:
    prefix: --peak_length
- id: target_site
  doc: 'If set, MotifScan will report the sequence and position of each target. (default:
    False)'
  type: boolean
  inputBinding:
    prefix: --target_site
- id: o
  doc: "Output directory to the write the results. If unspecified, the results will\
    \ be written to 'motifscan_output_PREFIX_OF_INPUT_REGION' under working directory.\
    \ (default: None)"
  type: string
  inputBinding:
    prefix: -o
outputs: []
cwlVersion: v1.1
baseCommand:
- motifscan
