from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, Int, Directory, Float

Ryuto_V0_1_0 = CommandToolBuilder(tool="ryuto", base_command=["ryuto"], inputs=[ToolInput(tag="in_arg_path_gtfgff", input_type=Boolean(optional=True), prefix="-g", doc=InputDocumentation(doc="[ --gtf-guide ] arg            Path to a gtf/gff file containing reference\ngenes. (default: none)")), ToolInput(tag="in_guide_trust_level", input_type=Int(optional=True), prefix="--guide-trustlevel", doc=InputDocumentation(doc="Level of trust in the given guides. 0 no\ntrust. 100 maximal trust. Abstract\nnon-linear variable! (default: 20)")), ToolInput(tag="in_arg_number_consecutive", input_type=Boolean(optional=True), prefix="-w", doc=InputDocumentation(doc="[ --pool ] arg                 Number of consecutive inputs to merge\ninternally in groups.")), ToolInput(tag="in_arg_number_use", input_type=Boolean(optional=True), prefix="-t", doc=InputDocumentation(doc="[ --threads ] arg              Number of threads to use in parallel.\n(default: 1)")), ToolInput(tag="in_arg_output_directory", input_type=Directory(optional=True), prefix="-o", doc=InputDocumentation(doc="[ --out-dir ] arg              The output directory for all data.\n(default: ./)")), ToolInput(tag="in_arg_frunstranded_frfirststrandfrsecondstrand", input_type=Boolean(optional=True), prefix="-l", doc=InputDocumentation(doc="[ --library-type ] arg         fr-unstranded, fr-firststrand,\nfr-secondstrand, ff-unstranded,\nff-firststrand, or ff-secondstrand\naccording to TopHat and Cufflinks\ndefinitions. (default: fr-unstranded)")), ToolInput(tag="in_add_option_ignore", input_type=Boolean(optional=True), prefix="-s", doc=InputDocumentation(doc="[ --unstranded ]               Add this option to ignore strand\ninformation and call everything on +\nstrand. (default: none)")), ToolInput(tag="in_arg_minimum_average", input_type=Boolean(optional=True), prefix="-c", doc=InputDocumentation(doc="[ --base-coverage-filter ] arg Minimum average coverage for features to be\nfiltered. Sets low-edge-mark, mean-filter,\nand score-filter (default: 4.0)")), ToolInput(tag="in_no_parse_heuristic", input_type=Boolean(optional=True), prefix="--no-parse-heuristic", doc=InputDocumentation(doc="If several input files are used, use this to disable\nthe heuristic to merge fragments. Increases needed RAM.\n(default: use heuristic)")), ToolInput(tag="in_compute_singles", input_type=Boolean(optional=True), prefix="--compute-singles", doc=InputDocumentation(doc="If several input files are used, use this to compute\ntranscripts for all individual transcripts. (default:\nonly joined output)")), ToolInput(tag="in_nc", input_type=Int(optional=True), prefix="--nc", doc=InputDocumentation(doc="Number of chromosomes processed in parallel. (# of\nthreads = nc * ng; default: 1)")), ToolInput(tag="in_ng", input_type=Int(optional=True), prefix="--ng", doc=InputDocumentation(doc="Number of graph per chromosome processed in parallel.\n(# of threads = nc * ng; default: 1)")), ToolInput(tag="in_pos_merge", input_type=Int(optional=True), prefix="--pos-merge", doc=InputDocumentation(doc="Positions within this range of bases will be\nclustered together. (default: 1)")), ToolInput(tag="in_min_intron_length", input_type=Int(optional=True), prefix="--min-intron-length", doc=InputDocumentation(doc="Minimal length of predicted introns. (default: 5)")), ToolInput(tag="in_max_intron_length", input_type=Int(optional=True), prefix="--max-intron-length", doc=InputDocumentation(doc="Maximal length of predicted introns. (default:\n1,000,000)")), ToolInput(tag="in_no_denoising", input_type=Boolean(optional=True), prefix="--no-denoising", doc=InputDocumentation(doc="Add this option to disable denoising of flow\ngraph capacities.")), ToolInput(tag="in_keep_introns", input_type=Boolean(optional=True), prefix="--keep-introns", doc=InputDocumentation(doc="Add this option to keep all introns with read\ncoverage intact.")), ToolInput(tag="in_intron_threshold", input_type=Int(optional=True), prefix="--intron-threshold", doc=InputDocumentation(doc="Percentage of coverage to existing splices\nneeded to keep introns intact. (default: 30%)")), ToolInput(tag="in_broken_intron_threshold", input_type=Int(optional=True), prefix="--broken-intron-threshold", doc=InputDocumentation(doc="Percentage of coverage to existing splices\nneeded to keep fragmented intron pieces intact.\n(default: 15%)")), ToolInput(tag="in_exon_join_distance", input_type=Int(optional=True), prefix="--exon-join-distance", doc=InputDocumentation(doc="Join exons no more distant than this amount of\nbp into one. (default: 5)")), ToolInput(tag="in_no_trimming", input_type=Boolean(optional=True), prefix="--no-trimming", doc=InputDocumentation(doc="Disable trimming of predicted transcripts based\non coverage.")), ToolInput(tag="in_trimming_rate", input_type=Int(optional=True), prefix="--trimming-rate", doc=InputDocumentation(doc="Don't join exons if one is smaller than this\npercentage of the bigger (default: 20%)")), ToolInput(tag="in_low_edge_mark", input_type=Float(optional=True), prefix="--low-edge-mark", doc=InputDocumentation(doc="Capacity of edges in the splitgraph considered\nunlikely. (default: 4.0)")), ToolInput(tag="in_min_junction_coverage", input_type=Int(optional=True), prefix="--min-junction-coverage", doc=InputDocumentation(doc="Minimal coverage to consider intron junctions in\nthe graph. (default: 1)")), ToolInput(tag="in_min_junction_anchor", input_type=Int(optional=True), prefix="--min-junction-anchor", doc=InputDocumentation(doc="Minimal length of bp evidence needed left and\nright of a junction to consider it. (default: 10)")), ToolInput(tag="in_group_vote_low", input_type=Int(optional=True), prefix="--group-vote-low", doc=InputDocumentation(doc="Multi assembly percentage required for simple\nconsensus voting (default: 30%)")), ToolInput(tag="in_group_vote_high", input_type=Int(optional=True), prefix="--group-vote-high", doc=InputDocumentation(doc="Multi assembly percentage required for\nsuper-majority consensus voting (default: 60%)")), ToolInput(tag="in_arg_minimum_capacity", input_type=Boolean(optional=True), prefix="-p", doc=InputDocumentation(doc="[ --percent-filter ] arg           Minimum capacity to report retrieved\ntranscripts compared to the largest\ntranscript found. (default: 0%)")), ToolInput(tag="in_mean_filter", input_type=Float(optional=True), prefix="--mean-filter", doc=InputDocumentation(doc="Minimum average capacity to report\nretrieved transcripts. (default: 4.0)")), ToolInput(tag="in_score_filter", input_type=Float(optional=True), prefix="--score-filter", doc=InputDocumentation(doc="Minimum score to report retrieved\ntranscripts. (default: 3.0)")), ToolInput(tag="in_min_single_coverage", input_type=Int(optional=True), prefix="--min-single-coverage", doc=InputDocumentation(doc="Minimal average coverage off single\nexon transcripts to report. (default:\n20)")), ToolInput(tag="in_min_single_length", input_type=Int(optional=True), prefix="--min-single-length", doc=InputDocumentation(doc="Minimal length off single exon\ntranscripts to report. (default: 300)")), ToolInput(tag="in_min_transcript_length_base", input_type=Int(optional=True), prefix="--min-transcript-length-base", doc=InputDocumentation(doc="Minimal length of a transcripts to\nreport in bp. (default: 200)")), ToolInput(tag="in_min_transcript_length_extension", input_type=Int(optional=True), prefix="--min-transcript-length-extension", doc=InputDocumentation(doc="Additional length in bp required per\nexon. (default: 50)")), ToolInput(tag="in_region_group_filter", input_type=Float(optional=True), prefix="--region-group-filter", doc=InputDocumentation(doc="Regions where all transcripts are below\nthis coverage are discarded. (default:\n5.5)")), ToolInput(tag="in_arg_reads_length", input_type=Boolean(optional=True), prefix="-r", doc=InputDocumentation(doc="[ --min-readlength ] arg      Reads below this length will not be used for\natom evidence. (default: 30)")), ToolInput(tag="in_arg_minimal_distance", input_type=Boolean(optional=True), prefix="-x", doc=InputDocumentation(doc="[ --min-paired-distance ] arg Minimal distance between two paired reads in\norder to be considered. (default: 0)")), ToolInput(tag="in_arg_maximal_distance", input_type=Boolean(optional=True), prefix="-y", doc=InputDocumentation(doc="[ --max-paired-distance ] arg Maximal distance between two paired reads in\norder to be considered. (default: 100000)")), ToolInput(tag="in_arc_granularity", input_type=Int(optional=True), prefix="--arc-granularity", doc=InputDocumentation(doc="Maximum number of arcs per original arc on\npseudo-polynomial convex cost flows.\n(default: 500)")), ToolInput(tag="in_min_raw_exon_length", input_type=Int(optional=True), prefix="--min-raw-exon-length", doc=InputDocumentation(doc="Minimal number of basepairs needed to\nconstitute a raw exon. (default: 3)")), ToolInput(tag="in_max_path_enumeration", input_type=Int(optional=True), prefix="--max-path-enumeration", doc=InputDocumentation(doc="Maximal number of distinct paths enumerated\non each side of an ambiguous node. (default:\n10,000)"))], outputs=[ToolOutput(tag="out_arg_output_directory", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_arg_output_directory", type_hint=File()), doc=OutputDocumentation(doc="[ --out-dir ] arg              The output directory for all data.\n(default: ./)"))], container="quay.io/biocontainers/ryuto:1.6.1--h986a166_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Ryuto_V0_1_0().translate("wdl")

