class: CommandLineTool
id: isONclust.cwl
inputs:
- id: fast_q
  doc: 'Path to consensus fastq file(s) (default: False)'
  type: string
  inputBinding:
    prefix: --fastq
- id: fl_nc
  doc: 'The flnc reads generated by the isoseq3 algorithm (BAM file) (default: False)'
  type: string
  inputBinding:
    prefix: --flnc
- id: ccs
  doc: 'Path to consensus BAM file(s) (default: False)'
  type: string
  inputBinding:
    prefix: --ccs
- id: t
  doc: 'Number of cores allocated for clustering (default: 8)'
  type: string
  inputBinding:
    prefix: --t
- id: d
  doc: 'For debugging, prints status of clustering and minimizer database every p
    reads processed. (default: 10000)'
  type: string
  inputBinding:
    prefix: --d
- id: q
  doc: 'Filters reads with average phred quality value under this number (default
    = 7.0). (default: 7.0)'
  type: string
  inputBinding:
    prefix: --q
- id: ont
  doc: 'Clustering of ONT transcript reads. (default: False)'
  type: boolean
  inputBinding:
    prefix: --ont
- id: iso_seq
  doc: 'Clustering of PacBio Iso-Seq reads. (default: False)'
  type: boolean
  inputBinding:
    prefix: --isoseq
- id: use_old_sorted_file
  doc: 'Using already existing sorted file if present in specified output directory.
    (default: False)'
  type: boolean
  inputBinding:
    prefix: --use_old_sorted_file
- id: consensus
  doc: 'After clustering, (1) run spoa on all clusters, (2) detect reverse complements,
    (3) run medaka. (default: False)'
  type: boolean
  inputBinding:
    prefix: --consensus
- id: abundance_ratio
  doc: 'Threshold for --consensus algorithm. Consider only clusters larger than a
    fraction of number of total reads (default 0.1) (default: 0.1)'
  type: string
  inputBinding:
    prefix: --abundance_ratio
- id: rc_identity_threshold
  doc: 'Threshold for --consensus algorithm. Define a reverse complement if identity
    is over this threshold (default 0.9) (default: 0.9)'
  type: string
  inputBinding:
    prefix: --rc_identity_threshold
- id: medaka
  doc: 'Run final medaka polishing algorithm. (default: False)'
  type: boolean
  inputBinding:
    prefix: --medaka
- id: k
  doc: 'Kmer size (default: 15)'
  type: string
  inputBinding:
    prefix: --k
- id: w
  doc: 'Window size (default: 50)'
  type: string
  inputBinding:
    prefix: --w
- id: min_shared
  doc: 'Minmum number of minimizers shared between read and cluster (default: 5)'
  type: long
  inputBinding:
    prefix: --min_shared
- id: mapped_threshold
  doc: 'Minmum mapped fraction of read to be included in cluster. The density of minimizers
    to classify a region as mapped depends on quality of the read. (default: 0.7)'
  type: string
  inputBinding:
    prefix: --mapped_threshold
- id: aligned_threshold
  doc: 'Minmum aligned fraction of read to be included in cluster. Aligned identity
    depends on the quality of the read. (default: 0.4)'
  type: string
  inputBinding:
    prefix: --aligned_threshold
- id: batch_type
  doc: 'In parrallel mode, how to split the reads into chunks "total_nt", "nr_reads",
    or "weighted" (default: total_nt) (default: total_nt)'
  type: string
  inputBinding:
    prefix: --batch_type
- id: min_fraction
  doc: 'Minmum fraction of minimizers shared compared to best hit, in order to continue
    mapping. (default: 0.8)'
  type: long
  inputBinding:
    prefix: --min_fraction
- id: min_prob_no_hits
  doc: 'Minimum probability for i consecutive minimizers to be different between read
    and representative and still considered as mapped region, under assumption that
    they come from the same transcript (depends on read quality). (default: 0.1)'
  type: long
  inputBinding:
    prefix: --min_prob_no_hits
- id: out_folder
  doc: 'A fasta file with transcripts that are shared between samples and have perfect
    illumina support. (default: None)'
  type: string
  inputBinding:
    prefix: --outfolder
outputs: []
cwlVersion: v1.1
baseCommand:
- isONclust
