from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, Int, File, Float, String

Mason_Simulator_V0_1_0 = CommandToolBuilder(tool="mason_simulator", base_command=["mason_simulator"], inputs=[ToolInput(tag="in_version_check", input_type=Boolean(optional=True), prefix="--version-check", doc=InputDocumentation(doc="Turn this option off to disable version update notifications of the\napplication. One of 1, ON, TRUE, T, YES, 0, OFF, FALSE, F, and NO.\nDefault: 1.")), ToolInput(tag="in_quiet", input_type=Boolean(optional=True), prefix="--quiet", doc=InputDocumentation(doc="Low verbosity.")), ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="--verbose", doc=InputDocumentation(doc="Higher verbosity.")), ToolInput(tag="in_very_verbose", input_type=Boolean(optional=True), prefix="--very-verbose", doc=InputDocumentation(doc="Highest verbosity.")), ToolInput(tag="in_seed", input_type=Int(optional=True), prefix="--seed", doc=InputDocumentation(doc="Seed to use for random number generator. Default: 0.")), ToolInput(tag="in_meth_seed", input_type=Int(optional=True), prefix="--meth-seed", doc=InputDocumentation(doc="Seed to use for methylation level random number generator. Default:\n0.")), ToolInput(tag="in_seed_spacing", input_type=Int(optional=True), prefix="--seed-spacing", doc=InputDocumentation(doc="Offset for seeds to use when multi-threading. Default: 2048.")), ToolInput(tag="in_num_threads", input_type=Int(optional=True), prefix="--num-threads", doc=InputDocumentation(doc="Number of threads to use. In range [1..inf]. Default: 1.")), ToolInput(tag="in_force_single_end", input_type=File(optional=True), prefix="--force-single-end", doc=InputDocumentation(doc="Force single-end simulation although --out-right file is given.")), ToolInput(tag="in_chunk_size", input_type=Int(optional=True), prefix="--chunk-size", doc=InputDocumentation(doc="Number of fragments to simulate in one batch. In range [65536..inf].\nDefault: 65536.")), ToolInput(tag="in_num_fragments", input_type=Int(optional=True), prefix="--num-fragments", doc=InputDocumentation(doc="Number of reads/pairs to simulate. In range [1..inf].")), ToolInput(tag="in_meth_fast_a_in", input_type=File(optional=True), prefix="--meth-fasta-in", doc=InputDocumentation(doc="FASTA file with methylation levels of the input file. Valid\nfiletypes are: .sam[.*], .raw[.*], .gbk[.*], .frn[.*], .fq[.*],\n.fna[.*], .ffn[.*], .fastq[.*], .fasta[.*], .faa[.*], .fa[.*],\n.embl[.*], and .bam, where * is any of the following extensions: gz\nand bgzf for transparent (de)compression.")), ToolInput(tag="in_out", input_type=File(optional=True), prefix="--out", doc=InputDocumentation(doc="Output of single-end/left end reads. Valid filetypes are: .sam[.*],\n.raw[.*], .frn[.*], .fq[.*], .fna[.*], .ffn[.*], .fastq[.*],\n.fasta[.*], .faa[.*], .fa[.*], and .bam, where * is any of the\nfollowing extensions: gz and bgzf for transparent (de)compression.")), ToolInput(tag="in_out_right", input_type=File(optional=True), prefix="--out-right", doc=InputDocumentation(doc="Output of right reads. Giving this options enables paired-end\nsimulation. Valid filetypes are: .sam[.*], .raw[.*], .frn[.*],\n.fq[.*], .fna[.*], .ffn[.*], .fastq[.*], .fasta[.*], .faa[.*],\n.fa[.*], and .bam, where * is any of the following extensions: gz\nand bgzf for transparent (de)compression.")), ToolInput(tag="in_out_alignment", input_type=File(optional=True), prefix="--out-alignment", doc=InputDocumentation(doc="SAM/BAM file with alignments. Valid filetypes are: .sam[.*] and\n.bam, where * is any of the following extensions: gz and bgzf for\ntransparent (de)compression.")), ToolInput(tag="in_input_reference", input_type=File(optional=True), prefix="--input-reference", doc=InputDocumentation(doc="Path to FASTA file to read the reference from. Valid filetypes are:\n.sam[.*], .raw[.*], .gbk[.*], .frn[.*], .fq[.*], .fna[.*], .ffn[.*],\n.fastq[.*], .fasta[.*], .faa[.*], .fa[.*], .embl[.*], and .bam,\nwhere * is any of the following extensions: gz and bgzf for\ntransparent (de)compression.")), ToolInput(tag="in_input_vcf", input_type=File(optional=True), prefix="--input-vcf", doc=InputDocumentation(doc="Path to the VCF file with variants to apply. Valid filetype is:\n.vcf[.*], where * is any of the following extensions: gz and bgzf\nfor transparent (de)compression.")), ToolInput(tag="in_methylation_levels", input_type=Boolean(optional=True), prefix="--methylation-levels", doc=InputDocumentation(doc="Enable methylation level simulation.")), ToolInput(tag="in_meth_cg_mu", input_type=Float(optional=True), prefix="--meth-cg-mu", doc=InputDocumentation(doc="Median of beta distribution for methylation level of CpG loci. In\nrange [0..1]. Default: 0.6.")), ToolInput(tag="in_meth_cg_sigma", input_type=Float(optional=True), prefix="--meth-cg-sigma", doc=InputDocumentation(doc="Standard deviation of beta distribution for methylation level of CpG\nloci. In range [0..1]. Default: 0.03.")), ToolInput(tag="in_meth_chg_mu", input_type=Float(optional=True), prefix="--meth-chg-mu", doc=InputDocumentation(doc="Median of beta distribution for methylation level of CHG loci. In\nrange [0..1]. Default: 0.08.")), ToolInput(tag="in_meth_chg_sigma", input_type=Float(optional=True), prefix="--meth-chg-sigma", doc=InputDocumentation(doc="Standard deviation of beta distribution for methylation level of CHG\nloci. In range [0..1]. Default: 0.008.")), ToolInput(tag="in_meth_chh_mu", input_type=Float(optional=True), prefix="--meth-chh-mu", doc=InputDocumentation(doc="Median of beta distribution for methylation level of CHH loci. In\nrange [0..1]. Default: 0.05.")), ToolInput(tag="in_meth_chh_sigma", input_type=Float(optional=True), prefix="--meth-chh-sigma", doc=InputDocumentation(doc="Standard deviation of beta distribution for methylation level of CHH\nloci. In range [0..1]. Default: 0.005.")), ToolInput(tag="in_fragment_size_model", input_type=Int(optional=True), prefix="--fragment-size-model", doc=InputDocumentation(doc="The model to use for the fragment size simulation. One of normal and\nuniform. Default: normal.")), ToolInput(tag="in_fragment_min_size", input_type=Int(optional=True), prefix="--fragment-min-size", doc=InputDocumentation(doc="Smallest fragment size to use when using uniform fragment size\nsimulation. In range [1..inf]. Default: 100.")), ToolInput(tag="in_fragment_max_size", input_type=Int(optional=True), prefix="--fragment-max-size", doc=InputDocumentation(doc="Largest fragment size to use when using uniform fragment size\nsimulation. In range [1..inf]. Default: 400.")), ToolInput(tag="in_fragment_mean_size", input_type=Int(optional=True), prefix="--fragment-mean-size", doc=InputDocumentation(doc="Mean fragment size for normally distributed fragment size\nsimulation. In range [1..inf]. Default: 300.")), ToolInput(tag="in_fragment_size_std_dev", input_type=Int(optional=True), prefix="--fragment-size-std-dev", doc=InputDocumentation(doc="Fragment size standard deviation when using normally distributed\nfragment size simulation. In range [1..inf]. Default: 30.")), ToolInput(tag="in_seq_technology", input_type=Int(optional=True), prefix="--seq-technology", doc=InputDocumentation(doc="Set sequencing technology to simulate. One of illumina, 454, and\nsanger. Default: illumina.")), ToolInput(tag="in_seq_mate_orientation", input_type=Int(optional=True), prefix="--seq-mate-orientation", doc=InputDocumentation(doc="Orientation for paired reads. See section Read Orientation below.\nOne of FR, RF, FF, and FF2. Default: FR.")), ToolInput(tag="in_seq_strands", input_type=String(optional=True), prefix="--seq-strands", doc=InputDocumentation(doc="Strands to simulate from, only applicable to paired sequencing\nsimulation. One of forward, reverse, and both. Default: both.")), ToolInput(tag="in_embed_read_info", input_type=Boolean(optional=True), prefix="--embed-read-info", doc=InputDocumentation(doc="Whether or not to embed read information.")), ToolInput(tag="in_read_name_prefix", input_type=String(optional=True), prefix="--read-name-prefix", doc=InputDocumentation(doc="Read names will have this prefix. Default: simulated..")), ToolInput(tag="in_enable_bs_seq", input_type=Boolean(optional=True), prefix="--enable-bs-seq", doc=InputDocumentation(doc="Enable BS-seq simulation.")), ToolInput(tag="in_bs_seq_protocol", input_type=String(optional=True), prefix="--bs-seq-protocol", doc=InputDocumentation(doc="Protocol to use for BS-Seq simulation. One of directional and\nundirectional. Default: directional.")), ToolInput(tag="in_bs_seq_conversion_rate", input_type=Float(optional=True), prefix="--bs-seq-conversion-rate", doc=InputDocumentation(doc="Conversion rate for unmethylated Cs to become Ts. In range [0..1].\nDefault: 0.99.")), ToolInput(tag="in_illumina_read_length", input_type=Int(optional=True), prefix="--illumina-read-length", doc=InputDocumentation(doc="Read length for Illumina simulation. In range [1..inf]. Default:\n100.")), ToolInput(tag="in_illumina_error_profile_file", input_type=File(optional=True), prefix="--illumina-error-profile-file", doc=InputDocumentation(doc="Path to file with Illumina error profile. The file must be a text\nfile with floating point numbers separated by space, each giving a\npositional error rate. Valid filetype is: .txt.")), ToolInput(tag="in_illumina_prob_insert", input_type=Float(optional=True), prefix="--illumina-prob-insert", doc=InputDocumentation(doc="Insert per-base probability for insertion in Illumina sequencing. In\nrange [0..1]. Default: 0.00005.")), ToolInput(tag="in_illumina_prob_deletion", input_type=Float(optional=True), prefix="--illumina-prob-deletion", doc=InputDocumentation(doc="Insert per-base probability for deletion in Illumina sequencing. In\nrange [0..1]. Default: 0.00005.")), ToolInput(tag="in_illumina_prob_mismatch_scale", input_type=Float(optional=True), prefix="--illumina-prob-mismatch-scale", doc=InputDocumentation(doc="Scaling factor for Illumina mismatch probability. In range [0..inf].\nDefault: 1.0.")), ToolInput(tag="in_illumina_prob_mismatch", input_type=Float(optional=True), prefix="--illumina-prob-mismatch", doc=InputDocumentation(doc="Average per-base mismatch probability in Illumina sequencing. In\nrange [0.0..1.0]. Default: 0.004.")), ToolInput(tag="in_illumina_prob_mismatch_begin", input_type=Float(optional=True), prefix="--illumina-prob-mismatch-begin", doc=InputDocumentation(doc="Per-base mismatch probability of first base in Illumina sequencing.\nIn range [0.0..1.0]. Default: 0.002.")), ToolInput(tag="in_illumina_prob_mismatch_end", input_type=Float(optional=True), prefix="--illumina-prob-mismatch-end", doc=InputDocumentation(doc="Per-base mismatch probability of last base in Illumina sequencing.\nIn range [0.0..1.0]. Default: 0.012.")), ToolInput(tag="in_illumina_position_raise", input_type=Int(optional=True), prefix="--illumina-position-raise", doc=InputDocumentation(doc="Point where the error curve raises in relation to read length. In\nrange [0.0..1.0]. Default: 0.66.")), ToolInput(tag="in_illumina_quality_mean_begin", input_type=Float(optional=True), prefix="--illumina-quality-mean-begin", doc=InputDocumentation(doc="Mean PHRED quality for non-mismatch bases of first base in Illumina\nsequencing. Default: 40.0.")), ToolInput(tag="in_illumina_quality_mean_end", input_type=Float(optional=True), prefix="--illumina-quality-mean-end", doc=InputDocumentation(doc="Mean PHRED quality for non-mismatch bases of last base in Illumina\nsequencing. Default: 39.5.")), ToolInput(tag="in_illumina_quality_stddev_begin", input_type=Float(optional=True), prefix="--illumina-quality-stddev-begin", doc=InputDocumentation(doc="Standard deviation of PHRED quality for non-mismatch bases of first\nbase in Illumina sequencing. Default: 0.05.")), ToolInput(tag="in_illumina_quality_stddev_end", input_type=Float(optional=True), prefix="--illumina-quality-stddev-end", doc=InputDocumentation(doc="Standard deviation of PHRED quality for non-mismatch bases of last\nbase in Illumina sequencing. Default: 10.0.")), ToolInput(tag="in_illumina_mismatch_quality_mean_begin", input_type=Float(optional=True), prefix="--illumina-mismatch-quality-mean-begin", doc=InputDocumentation(doc="Mean PHRED quality for mismatch bases of first base in Illumina\nsequencing. Default: 40.0.")), ToolInput(tag="in_illumina_mismatch_quality_mean_end", input_type=Float(optional=True), prefix="--illumina-mismatch-quality-mean-end", doc=InputDocumentation(doc="Mean PHRED quality for mismatch bases of last base in Illumina\nsequencing. Default: 30.0.")), ToolInput(tag="in_illumina_mismatch_quality_stddev_begin", input_type=Float(optional=True), prefix="--illumina-mismatch-quality-stddev-begin", doc=InputDocumentation(doc="Standard deviation of PHRED quality for mismatch bases of first base\nin Illumina sequencing. Default: 3.0.")), ToolInput(tag="in_illumina_mismatch_quality_stddev_end", input_type=Float(optional=True), prefix="--illumina-mismatch-quality-stddev-end", doc=InputDocumentation(doc="Standard deviation of PHRED quality for mismatch bases of last base\nin Illumina sequencing. Default: 15.0.")), ToolInput(tag="in_illumina_left_template_fast_q", input_type=File(optional=True), prefix="--illumina-left-template-fastq", doc=InputDocumentation(doc="FASTQ file to use for a template for left-end reads. Valid filetypes\nare: .sam[.*], .raw[.*], .gbk[.*], .frn[.*], .fq[.*], .fna[.*],\n.ffn[.*], .fastq[.*], .fasta[.*], .faa[.*], .fa[.*], .embl[.*], and\n.bam, where * is any of the following extensions: gz and bgzf for\ntransparent (de)compression.")), ToolInput(tag="in_illumina_right_template_fast_q", input_type=File(optional=True), prefix="--illumina-right-template-fastq", doc=InputDocumentation(doc="FASTQ file to use for a template for right-end reads. Valid\nfiletypes are: .sam[.*], .raw[.*], .gbk[.*], .frn[.*], .fq[.*],\n.fna[.*], .ffn[.*], .fastq[.*], .fasta[.*], .faa[.*], .fa[.*],\n.embl[.*], and .bam, where * is any of the following extensions: gz\nand bgzf for transparent (de)compression.")), ToolInput(tag="in_sanger_read_length_model", input_type=Int(optional=True), prefix="--sanger-read-length-model", doc=InputDocumentation(doc="The model to use for sampling the Sanger read length. One of normal\nand uniform. Default: normal.")), ToolInput(tag="in_sanger_read_length_min", input_type=Int(optional=True), prefix="--sanger-read-length-min", doc=InputDocumentation(doc="The minimal read length when the read length is sampled uniformly.\nIn range [0..inf]. Default: 400.")), ToolInput(tag="in_sanger_read_length_max", input_type=Int(optional=True), prefix="--sanger-read-length-max", doc=InputDocumentation(doc="The maximal read length when the read length is sampled uniformly.\nIn range [0..inf]. Default: 600.")), ToolInput(tag="in_sanger_read_length_mean", input_type=Int(optional=True), prefix="--sanger-read-length-mean", doc=InputDocumentation(doc="The mean read length when the read length is sampled with normal\ndistribution. In range [0..inf]. Default: 400.")), ToolInput(tag="in_sanger_read_length_error", input_type=Int(optional=True), prefix="--sanger-read-length-error", doc=InputDocumentation(doc="The read length standard deviation when the read length is sampled\nuniformly. In range [0..inf]. Default: 40.")), ToolInput(tag="in_sanger_prob_mismatch_scale", input_type=Float(optional=True), prefix="--sanger-prob-mismatch-scale", doc=InputDocumentation(doc="Scaling factor for Sanger mismatch probability. In range [0..inf].\nDefault: 1.0.")), ToolInput(tag="in_sanger_prob_mismatch_begin", input_type=Float(optional=True), prefix="--sanger-prob-mismatch-begin", doc=InputDocumentation(doc="Per-base mismatch probability of first base in Sanger sequencing. In\nrange [0.0..1.0]. Default: 0.005.")), ToolInput(tag="in_sanger_prob_mismatch_end", input_type=Float(optional=True), prefix="--sanger-prob-mismatch-end", doc=InputDocumentation(doc="Per-base mismatch probability of last base in Sanger sequencing. In\nrange [0.0..1.0]. Default: 0.001.")), ToolInput(tag="in_sanger_prob_insertion_begin", input_type=Float(optional=True), prefix="--sanger-prob-insertion-begin", doc=InputDocumentation(doc="Per-base insertion probability of first base in Sanger sequencing.\nIn range [0.0..1.0]. Default: 0.0025.")), ToolInput(tag="in_sanger_prob_insertion_end", input_type=Float(optional=True), prefix="--sanger-prob-insertion-end", doc=InputDocumentation(doc="Per-base insertion probability of last base in Sanger sequencing. In\nrange [0.0..1.0]. Default: 0.005.")), ToolInput(tag="in_sanger_prob_deletion_begin", input_type=Float(optional=True), prefix="--sanger-prob-deletion-begin", doc=InputDocumentation(doc="Per-base deletion probability of first base in Sanger sequencing. In\nrange [0.0..1.0]. Default: 0.0025.")), ToolInput(tag="in_sanger_prob_deletion_end", input_type=Float(optional=True), prefix="--sanger-prob-deletion-end", doc=InputDocumentation(doc="Per-base deletion probability of last base in Sanger sequencing. In\nrange [0.0..1.0]. Default: 0.005.")), ToolInput(tag="in_sanger_quality_match_start_mean", input_type=Float(optional=True), prefix="--sanger-quality-match-start-mean", doc=InputDocumentation(doc="Mean PHRED quality for non-mismatch bases of first base in Sanger\nsequencing. Default: 40.0.")), ToolInput(tag="in_sanger_quality_match_end_mean", input_type=Float(optional=True), prefix="--sanger-quality-match-end-mean", doc=InputDocumentation(doc="Mean PHRED quality for non-mismatch bases of last base in Sanger\nsequencing. Default: 39.5.")), ToolInput(tag="in_sanger_quality_match_start_stddev", input_type=Float(optional=True), prefix="--sanger-quality-match-start-stddev", doc=InputDocumentation(doc="Mean PHRED quality for non-mismatch bases of first base in Sanger\nsequencing. Default: 0.1.")), ToolInput(tag="in_sanger_quality_match_end_stddev", input_type=Int(optional=True), prefix="--sanger-quality-match-end-stddev", doc=InputDocumentation(doc="Mean PHRED quality for non-mismatch bases of last base in Sanger\nsequencing. Default: 2.")), ToolInput(tag="in_sanger_quality_error_start_mean", input_type=Int(optional=True), prefix="--sanger-quality-error-start-mean", doc=InputDocumentation(doc="Mean PHRED quality for erroneous bases of first base in Sanger\nsequencing. Default: 30.")), ToolInput(tag="in_sanger_quality_error_end_mean", input_type=Int(optional=True), prefix="--sanger-quality-error-end-mean", doc=InputDocumentation(doc="Mean PHRED quality for erroneous bases of last base in Sanger\nsequencing. Default: 20.")), ToolInput(tag="in_sanger_quality_error_start_stddev", input_type=Int(optional=True), prefix="--sanger-quality-error-start-stddev", doc=InputDocumentation(doc="Mean PHRED quality for erroneous bases of first base in Sanger\nsequencing. Default: 2.")), ToolInput(tag="in_sanger_quality_error_end_stddev", input_type=Int(optional=True), prefix="--sanger-quality-error-end-stddev", doc=InputDocumentation(doc="Mean PHRED quality for erroneous bases of last base in Sanger\nsequencing. Default: 5.")), ToolInput(tag="in_four_five_four_read_length_model", input_type=Int(optional=True), prefix="--454-read-length-model", doc=InputDocumentation(doc="The model to use for sampling the 454 read length. One of normal and\nuniform. Default: normal.")), ToolInput(tag="in_four_five_four_read_length_min", input_type=Int(optional=True), prefix="--454-read-length-min", doc=InputDocumentation(doc="The minimal read length when the read length is sampled uniformly.\nIn range [0..inf]. Default: 10.")), ToolInput(tag="in_four_five_four_read_length_max", input_type=Int(optional=True), prefix="--454-read-length-max", doc=InputDocumentation(doc="The maximal read length when the read length is sampled uniformly.\nIn range [0..inf]. Default: 600.")), ToolInput(tag="in_four_five_four_read_length_mean", input_type=Int(optional=True), prefix="--454-read-length-mean", doc=InputDocumentation(doc="The mean read length when the read length is sampled with normal\ndistribution. In range [0..inf]. Default: 400.")), ToolInput(tag="in_four_five_four_read_length_stddev", input_type=Int(optional=True), prefix="--454-read-length-stddev", doc=InputDocumentation(doc="The read length standard deviation when the read length is sampled\nwith normal distribution. In range [0..inf]. Default: 40.")), ToolInput(tag="in_four_five_four_no_sqrt_in_std_dev", input_type=Boolean(optional=True), prefix="--454-no-sqrt-in-std-dev", doc=InputDocumentation(doc="For error model, if set then (sigma = k * r)) is used, otherwise\n(sigma = k * sqrt(r)).")), ToolInput(tag="in_four_five_four_proportionality_factor", input_type=Int(optional=True), prefix="--454-proportionality-factor", doc=InputDocumentation(doc="Proportionality factor for calculating the standard deviation\nproportional to the read length. In range [0..inf]. Default: 0.15.")), ToolInput(tag="in_four_five_four_background_noise_mean", input_type=Int(optional=True), prefix="--454-background-noise-mean", doc=InputDocumentation(doc="Mean of lognormal distribution to use for the noise. In range\n[0..inf]. Default: 0.23.")), ToolInput(tag="in_four_five_four_background_noise_stddev", input_type=Int(optional=True), prefix="--454-background-noise-stddev", doc=InputDocumentation(doc="Standard deviation of lognormal distribution to use for the noise.\nIn range [0..inf]. Default: 0.15.")), ToolInput(tag="in_simulated_dot", input_type=String(), position=0, doc=InputDocumentation(doc="OPTIONS")), ToolInput(tag="in_file_dot", input_type=File(), position=0, doc=InputDocumentation(doc="After the generation of the haplotypes, fragments are sampled from the")), ToolInput(tag="in_nm", input_type=String(), position=0, doc=InputDocumentation(doc="Edit distance when aligned to the reference (i).")), ToolInput(tag="in_md", input_type=String(), position=1, doc=InputDocumentation(doc="String for mismatching positions (Z).")), ToolInput(tag="in_or", input_type=String(), position=2, doc=InputDocumentation(doc="Name of original reference, (Z).")), ToolInput(tag="in_oh", input_type=String(), position=3, doc=InputDocumentation(doc="Number of the original hhaplotype (1-based), (i).")), ToolInput(tag="in_op", input_type=String(), position=4, doc=InputDocumentation(doc="original position on the original reference (i).")), ToolInput(tag="in_os", input_type=String(), position=5, doc=InputDocumentation(doc="original strand, F/R for forward and reverse strand (A).")), ToolInput(tag="in_ur", input_type=String(), position=6, doc=InputDocumentation(doc="Reason for being unaligned, I/B for being in insertion or spanning\nover breakpoint.")), ToolInput(tag="in_xe", input_type=String(), position=7, doc=InputDocumentation(doc="Number of sequencing errors in the read (i).")), ToolInput(tag="in_xs", input_type=String(), position=8, doc=InputDocumentation(doc="Number of SNPs in the read alignment (i).")), ToolInput(tag="in_xi", input_type=String(), position=9, doc=InputDocumentation(doc="Number of small indels in the read alignment (i).")), ToolInput(tag="in_files_dot", input_type=String(), position=0, doc=InputDocumentation(doc="READ ORIENTATION")), ToolInput(tag="in_fr", input_type=String(), position=0, doc=InputDocumentation(doc="Reads are inward-facing, the same as Illumina paired-end reads: R1\n--> <-- R2.")), ToolInput(tag="in_rf", input_type=String(), position=1, doc=InputDocumentation(doc="Reads are outward-facing, the same as Illumina mate-pair reads: R1\n<-- --> R2.")), ToolInput(tag="in_ff", input_type=String(), position=2, doc=InputDocumentation(doc="Reads are on the same strand: R1 --> --> R2.")), ToolInput(tag="in_ff_two", input_type=Int(), position=3, doc=InputDocumentation(doc="Reads are on the same strand but the 'right' reads are sequenced to\nthe left of the 'left' reads, same as 454 paired: R2 --> --> R1."))], outputs=[ToolOutput(tag="out_force_single_end", output_type=File(optional=True), selector=InputSelector(input_to_select="in_force_single_end", type_hint=File()), doc=OutputDocumentation(doc="Force single-end simulation although --out-right file is given.")), ToolOutput(tag="out_out", output_type=File(optional=True), selector=InputSelector(input_to_select="in_out", type_hint=File()), doc=OutputDocumentation(doc="Output of single-end/left end reads. Valid filetypes are: .sam[.*],\n.raw[.*], .frn[.*], .fq[.*], .fna[.*], .ffn[.*], .fastq[.*],\n.fasta[.*], .faa[.*], .fa[.*], and .bam, where * is any of the\nfollowing extensions: gz and bgzf for transparent (de)compression.")), ToolOutput(tag="out_out_right", output_type=File(optional=True), selector=InputSelector(input_to_select="in_out_right", type_hint=File()), doc=OutputDocumentation(doc="Output of right reads. Giving this options enables paired-end\nsimulation. Valid filetypes are: .sam[.*], .raw[.*], .frn[.*],\n.fq[.*], .fna[.*], .ffn[.*], .fastq[.*], .fasta[.*], .faa[.*],\n.fa[.*], and .bam, where * is any of the following extensions: gz\nand bgzf for transparent (de)compression.")), ToolOutput(tag="out_out_alignment", output_type=File(optional=True), selector=InputSelector(input_to_select="in_out_alignment", type_hint=File()), doc=OutputDocumentation(doc="SAM/BAM file with alignments. Valid filetypes are: .sam[.*] and\n.bam, where * is any of the following extensions: gz and bgzf for\ntransparent (de)compression."))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Mason_Simulator_V0_1_0().translate("wdl", allow_empty_container=True)

