from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Boolean, String

Varscan_Somatic_V0_1_0 = CommandToolBuilder(tool="varscan_somatic", base_command=["varscan", "somatic"], inputs=[ToolInput(tag="in_output_snp", input_type=File(optional=True), prefix="--output-snp", doc=InputDocumentation(doc="- Output file for SNP calls [output.snp]")), ToolInput(tag="in_output_in_del", input_type=File(optional=True), prefix="--output-indel", doc=InputDocumentation(doc="- Output file for indel calls [output.indel]")), ToolInput(tag="in_min_coverage", input_type=Boolean(optional=True), prefix="--min-coverage", doc=InputDocumentation(doc="- Minimum coverage in normal and tumor to call variant [8]")), ToolInput(tag="in_min_coverage_normal", input_type=Boolean(optional=True), prefix="--min-coverage-normal", doc=InputDocumentation(doc="- Minimum coverage in normal to call somatic [8]")), ToolInput(tag="in_min_coverage_tumor", input_type=Boolean(optional=True), prefix="--min-coverage-tumor", doc=InputDocumentation(doc="- Minimum coverage in tumor to call somatic [6]")), ToolInput(tag="in_min_var_freq", input_type=Boolean(optional=True), prefix="--min-var-freq", doc=InputDocumentation(doc="- Minimum variant frequency to call a heterozygote [0.20]")), ToolInput(tag="in_min_freq_for_hom", input_type=Boolean(optional=True), prefix="--min-freq-for-hom", doc=InputDocumentation(doc="Minimum frequency to call homozygote [0.75]")), ToolInput(tag="in_normal_purity", input_type=Boolean(optional=True), prefix="--normal-purity", doc=InputDocumentation(doc="- Estimated purity (non-tumor content) of normal sample [1.00]")), ToolInput(tag="in_tumor_purity", input_type=Boolean(optional=True), prefix="--tumor-purity", doc=InputDocumentation(doc="- Estimated purity (tumor content) of tumor sample [1.00]")), ToolInput(tag="in_p_value", input_type=Boolean(optional=True), prefix="--p-value", doc=InputDocumentation(doc="- P-value threshold to call a heterozygote [0.99]")), ToolInput(tag="in_somatic_p_value", input_type=Boolean(optional=True), prefix="--somatic-p-value", doc=InputDocumentation(doc="- P-value threshold to call a somatic site [0.05]")), ToolInput(tag="in_strand_filter", input_type=Boolean(optional=True), prefix="--strand-filter", doc=InputDocumentation(doc="- If set to 1, removes variants with >90% strand bias [0]")), ToolInput(tag="in_validation", input_type=Boolean(optional=True), prefix="--validation", doc=InputDocumentation(doc="- If set to 1, outputs all compared positions even if non-variant")), ToolInput(tag="in_output_vcf", input_type=Boolean(optional=True), prefix="--output-vcf", doc=InputDocumentation(doc="- If set to 1, output VCF instead of VarScan native format")), ToolInput(tag="in_normal_pile_up", input_type=String(optional=True), position=0, doc=InputDocumentation(doc="")), ToolInput(tag="in_tumor_pile_up", input_type=String(optional=True), position=1, doc=InputDocumentation(doc=""))], outputs=[ToolOutput(tag="out_output_snp", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output_snp", type_hint=File()), doc=OutputDocumentation(doc="- Output file for SNP calls [output.snp]")), ToolOutput(tag="out_output_in_del", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output_in_del", type_hint=File()), doc=OutputDocumentation(doc="- Output file for indel calls [output.indel]"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Varscan_Somatic_V0_1_0().translate("wdl", allow_empty_container=True)

