version 1.0

task Anviimporttaxonomyforgenes {
  input {
    String? contigs_db
    Int? parser
    File? s_input_files
    Boolean? just_do_it
    String gene_taxonomy
  }
  command <<<
    anvi_import_taxonomy_for_genes \
      ~{gene_taxonomy} \
      ~{if defined(contigs_db) then ("--contigs-db " +  '"' + contigs_db + '"') else ""} \
      ~{if defined(parser) then ("--parser " +  '"' + parser + '"') else ""} \
      ~{if defined(s_input_files) then ("-i " +  '"' + s_input_files + '"') else ""} \
      ~{if (just_do_it) then "--just-do-it" else ""}
  >>>
  runtime {
    docker: "quay.io/biocontainers/anvio-minimal:7--py_0"
  }
  parameter_meta {
    contigs_db: "Anvi'o contigs database generated by 'anvi-gen-\\ncontigs-database' (default: None)"
    parser: "Parser to make sense of the input files. There are 3\\nparsers readily available: ['default_matrix',\\n'centrifuge', 'kaiju']. It is OK if you do not select\\na parser, but in that case there will be no additional\\ncontigs available except the identification of single-\\ncopy genes in your contigs for later use. Using a\\nparser will not prevent the analysis of single-copy\\ngenes, but make anvio more powerful to help you make\\nsense of your results. Please see the documentation,\\nor get in touch with the developers if you have any\\nquestions regarding parsers. (default: None)"
    s_input_files: "(S) [FILE(S) ...], --input-files FILE(S) [FILE(S) ...]\\nInput file(s) for selected parser. Each parser (except\\n\\\"blank\\\") requires input files to process that you\\ngenerate before running anvio. Please see the\\ndocumentation for details. (default: None)"
    just_do_it: "Don't bother me with questions or warnings, just do\\nit. (default: False)"
    gene_taxonomy: "üç∫ More on `anvi-import-taxonomy-for-genes`:"
  }
  output {
    File out_stdout = stdout()
  }
}