from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import String, Int, File, Boolean

Anvi_Get_Split_Coverages_V0_1_0 = CommandToolBuilder(tool="anvi_get_split_coverages", base_command=["anvi-get-split-coverages"], inputs=[ToolInput(tag="in_profile_db", input_type=String(optional=True), prefix="--profile-db", doc=InputDocumentation(doc="Anvi'o profile database (default: None)")), ToolInput(tag="in_contigs_db", input_type=String(optional=True), prefix="--contigs-db", doc=InputDocumentation(doc="Anvi'o contigs database generated by 'anvi-gen-\ncontigs-database' (default: None)")), ToolInput(tag="in_split_name", input_type=String(optional=True), prefix="--split-name", doc=InputDocumentation(doc="Split name. (default: None)")), ToolInput(tag="in_collection_name", input_type=String(optional=True), prefix="--collection-name", doc=InputDocumentation(doc="Collection name. (default: None)")), ToolInput(tag="in_bin_id", input_type=String(optional=True), prefix="--bin-id", doc=InputDocumentation(doc="Bin name you are interested in. (default: None)")), ToolInput(tag="in_gene_caller_id", input_type=String(optional=True), prefix="--gene-caller-id", doc=InputDocumentation(doc="A single gene id. (default: None)")), ToolInput(tag="in_flank_length", input_type=Int(optional=True), prefix="--flank-length", doc=InputDocumentation(doc="Extend sequences for gene calls with additional\nnucleotides from both ends. If the seqeunce for a\ntarget gene is between nucleotide positions START and\nSTOP, using a flank lenght of M will give you a\nsequence that starts at START - M and ends at STOP +\nM. (default: 0)")), ToolInput(tag="in_output_file", input_type=File(optional=True), prefix="--output-file", doc=InputDocumentation(doc="File path to store results. (default: None)")), ToolInput(tag="in_list_splits", input_type=Boolean(optional=True), prefix="--list-splits", doc=InputDocumentation(doc="When declared, the program will list split names in\nthe profile database and quite (default: False)")), ToolInput(tag="in_list_collections", input_type=Boolean(optional=True), prefix="--list-collections", doc=InputDocumentation(doc="Show available collections and exit. (default: False)")), ToolInput(tag="in_list_bins", input_type=Boolean(optional=True), prefix="--list-bins", doc=InputDocumentation(doc="List available bins in a collection and exit.\n(default: False)")), ToolInput(tag="in_coverages_txt", input_type=String(), position=0, doc=InputDocumentation(doc="üç∫ More on `anvi-get-split-coverages`:"))], outputs=[ToolOutput(tag="out_output_file", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output_file", type_hint=File()), doc=OutputDocumentation(doc="File path to store results. (default: None)"))], container="quay.io/biocontainers/anvio-minimal:7--py_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Anvi_Get_Split_Coverages_V0_1_0().translate("wdl")

