class: CommandLineTool
id: anvi_get_split_coverages.cwl
inputs:
- id: in_profile_db
  doc: "Anvi'o profile database (default: None)"
  type: string?
  inputBinding:
    prefix: --profile-db
- id: in_contigs_db
  doc: "Anvi'o contigs database generated by 'anvi-gen-\ncontigs-database' (default:\
    \ None)"
  type: string?
  inputBinding:
    prefix: --contigs-db
- id: in_split_name
  doc: 'Split name. (default: None)'
  type: string?
  inputBinding:
    prefix: --split-name
- id: in_collection_name
  doc: 'Collection name. (default: None)'
  type: string?
  inputBinding:
    prefix: --collection-name
- id: in_bin_id
  doc: 'Bin name you are interested in. (default: None)'
  type: string?
  inputBinding:
    prefix: --bin-id
- id: in_gene_caller_id
  doc: 'A single gene id. (default: None)'
  type: string?
  inputBinding:
    prefix: --gene-caller-id
- id: in_flank_length
  doc: "Extend sequences for gene calls with additional\nnucleotides from both ends.\
    \ If the seqeunce for a\ntarget gene is between nucleotide positions START and\n\
    STOP, using a flank lenght of M will give you a\nsequence that starts at START\
    \ - M and ends at STOP +\nM. (default: 0)"
  type: long?
  inputBinding:
    prefix: --flank-length
- id: in_output_file
  doc: 'File path to store results. (default: None)'
  type: File?
  inputBinding:
    prefix: --output-file
- id: in_list_splits
  doc: "When declared, the program will list split names in\nthe profile database\
    \ and quite (default: False)"
  type: boolean?
  inputBinding:
    prefix: --list-splits
- id: in_list_collections
  doc: 'Show available collections and exit. (default: False)'
  type: boolean?
  inputBinding:
    prefix: --list-collections
- id: in_list_bins
  doc: "List available bins in a collection and exit.\n(default: False)"
  type: boolean?
  inputBinding:
    prefix: --list-bins
- id: in_coverages_txt
  doc: 'üç∫ More on `anvi-get-split-coverages`:'
  type: string
  inputBinding:
    position: 0
outputs:
- id: out_stdout
  doc: Standard output stream
  type: stdout
- id: out_output_file
  doc: 'File path to store results. (default: None)'
  type: File?
  outputBinding:
    glob: $(inputs.in_output_file)
hints:
- class: DockerRequirement
  dockerPull: quay.io/biocontainers/anvio-minimal:7--py_0
cwlVersion: v1.1
baseCommand:
- anvi-get-split-coverages
