class: CommandLineTool
id: anvi_run_kegg_kofams.cwl
inputs:
- id: in_contigs_db
  doc: "Anvi'o contigs database generated by 'anvi-gen-\ncontigs-database' (default:\
    \ None)"
  type: string?
  inputBinding:
    prefix: --contigs-db
- id: in_profile_db
  doc: "If you only want to annotate gene calls in a subset of\nsequences, you can\
    \ provide a path to a profile\ndatabase with this flag. In this case, you must\
    \ also\nprovide a collection name so that we know which\nsequenes to focus on.\
    \ (default: None)"
  type: File?
  inputBinding:
    prefix: --profile-db
- id: in_collection_name
  doc: "Name of collection you want to focus on. Provide this\n(along with a profile\
    \ db) if you only want to annotate\ngene calls in a certain set of bins. (default:\
    \ None)"
  type: string?
  inputBinding:
    prefix: --collection-name
- id: in_kegg_data_dir
  doc: "The directory path for your KEGG setup, which will\ninclude things like KOfam\
    \ profiles and KEGG MODULE\ndata. Anvi'o will try to use the default path if you\n\
    do not specify anything. (default: None)"
  type: File?
  inputBinding:
    prefix: --kegg-data-dir
- id: in_num_threads
  doc: "Maximum number of threads to use for multithreading\nwhenever possible. Very\
    \ conservatively, the default is\n1. It is a good idea to not exceed the number\
    \ of CPUs\n/ cores on your system. Plus, please be careful with\nthis option if\
    \ you are running your commands on a SGE\n--if you are clusterizing your runs,\
    \ and asking for\nmultiple threads to use, you may deplete your\nresources very\
    \ fast. (default: 1)"
  type: long?
  inputBinding:
    prefix: --num-threads
- id: in_hmmer_program
  doc: "Which of the HMMER programs to use to run HMMs\n(hmmscan or hmmsearch). By\
    \ default anvi'o will use\nhmmscan for typical HMM operations like those in anvi-\n\
    run-hmms (as these tend to scan a very large number of\ngenes against a relatively\
    \ small number of HMMs), but\nif you are using this program to scan a very large\n\
    number of HMMs, hmmsearch might be a better choice for\nperformance. For this\
    \ reason, hmmsearch is the default\nin operations like anvi-run-pfams and anvi-run-kegg-\n\
    kofams. See this article for a discussion on the\nperformance of these two programs:\n\
    https://cryptogenomicon.org/2011/05/27/hmmscan-vs-\nhmmsearch-speed-the-numerology/\
    \ (default: None)"
  type: long?
  inputBinding:
    prefix: --hmmer-program
- id: in_keep_all_hits
  doc: "If you use this flag, anvi'o will not get rid of any\nraw HMM hits, even those\
    \ that are below the score\nthreshold. (default: False)"
  type: boolean?
  inputBinding:
    prefix: --keep-all-hits
- id: in_log_bit_scores
  doc: "Use this flag to generate a tab-delimited text file\ncontaining the bit scores\
    \ of every KOfam hit that is\nput in the contigs database. (default: False)"
  type: boolean?
  inputBinding:
    prefix: --log-bitscores
- id: in_just_do_it
  doc: "Don't bother me with questions or warnings, just do\nit. (default: False)"
  type: boolean?
  inputBinding:
    prefix: --just-do-it
outputs:
- id: out_stdout
  doc: Standard output stream
  type: stdout
hints:
- class: DockerRequirement
  dockerPull: quay.io/biocontainers/anvio-minimal:7--py_0
cwlVersion: v1.1
baseCommand:
- anvi-run-kegg-kofams
