from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import String, Boolean, Int, File

Anvi_Script_Predict_Cpr_Genomes_V0_1_0 = CommandToolBuilder(tool="anvi_script_predict_CPR_genomes", base_command=["anvi-script-predict-CPR-genomes"], inputs=[ToolInput(tag="in_contigs_db", input_type=String(optional=True), prefix="--contigs-db", doc=InputDocumentation(doc="Anvi'o contigs database generated by 'anvi-gen-\ncontigs-database' (default: None)")), ToolInput(tag="in_profile_db", input_type=String(optional=True), prefix="--profile-db", doc=InputDocumentation(doc="Anvi'o profile database (default: None)")), ToolInput(tag="in_collection_name", input_type=String(optional=True), prefix="--collection-name", doc=InputDocumentation(doc="Collection name. (default: None)")), ToolInput(tag="in_list_collections", input_type=Boolean(optional=True), prefix="--list-collections", doc=InputDocumentation(doc="Show available collections and exit. (default: False)")), ToolInput(tag="in_report_only_cpr", input_type=Boolean(optional=True), prefix="--report-only-cpr", doc=InputDocumentation(doc="Include only bins that look like CPR genomes.\n(default: False)")), ToolInput(tag="in_min_genome_size", input_type=Int(optional=True), prefix="--min-genome-size", doc=InputDocumentation(doc="Minimum genome size to consider for CPR in Mbp.\nDefault is 0.500000")), ToolInput(tag="in_min_percent_completion", input_type=Int(optional=True), prefix="--min-percent-completion", doc=InputDocumentation(doc="Minimum percent completion estimate based on anvi'o\ndefault single-copy gene collections. Default is 50")), ToolInput(tag="in_max_percent_redundancy", input_type=Int(optional=True), prefix="--max-percent-redundancy", doc=InputDocumentation(doc="Maxumum percent redundancy or single-copy genes in an\nanvi'o bin, or a genome to consider for\nclassification. The default is 30")), ToolInput(tag="in_min_class_probability", input_type=Int(optional=True), prefix="--min-class-probability", doc=InputDocumentation(doc="If the classification confidence is below this don't\nbother. Default is 75.")), ToolInput(tag="in_output_file", input_type=File(optional=True), prefix="--output-file", doc=InputDocumentation(doc="File path to store results. (default: None)")), ToolInput(tag="in_just_do_it", input_type=Boolean(optional=True), prefix="--just-do-it", doc=InputDocumentation(doc="Don't bother me with questions or warnings, just do\nit. (default: False)")), ToolInput(tag="in_classifier", input_type=String(), position=2, doc=InputDocumentation(doc="optional arguments:"))], outputs=[ToolOutput(tag="out_output_file", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output_file", type_hint=File()), doc=OutputDocumentation(doc="File path to store results. (default: None)"))], container="quay.io/biocontainers/anvio-minimal:7--py_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Anvi_Script_Predict_Cpr_Genomes_V0_1_0().translate("wdl")

