from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import String, File, Boolean, Int, Float

Anvi_Gen_Variability_Profile_V0_1_0 = CommandToolBuilder(tool="anvi_gen_variability_profile", base_command=["anvi-gen-variability-profile"], inputs=[ToolInput(tag="in_profile_db", input_type=String(optional=True), prefix="--profile-db", doc=InputDocumentation(doc="Anvi'o profile database (default: None)")), ToolInput(tag="in_contigs_db", input_type=String(optional=True), prefix="--contigs-db", doc=InputDocumentation(doc="Anvi'o contigs database generated by 'anvi-gen-\ncontigs-database' (default: None)")), ToolInput(tag="in_structure_db", input_type=String(optional=True), prefix="--structure-db", doc=InputDocumentation(doc="Anvi'o structure database. (default: None)")), ToolInput(tag="in_collection_name", input_type=String(optional=True), prefix="--collection-name", doc=InputDocumentation(doc="Collection name. (default: None)")), ToolInput(tag="in_bin_id", input_type=String(optional=True), prefix="--bin-id", doc=InputDocumentation(doc="Bin name you are interested in. (default: None)")), ToolInput(tag="in_splits_of_interest", input_type=File(optional=True), prefix="--splits-of-interest", doc=InputDocumentation(doc="A file with split names. There should be only one\ncolumn in the file, and each line should correspond to\na unique split name. (default: None)")), ToolInput(tag="in_genes_of_interest", input_type=File(optional=True), prefix="--genes-of-interest", doc=InputDocumentation(doc="A file with anvi'o gene caller IDs. There should be\nonly one column in the file, and each line should\ncorrespond to a unique gene caller id (without a\ncolumn header). (default: None)")), ToolInput(tag="in_gene_caller_ids", input_type=String(optional=True), prefix="--gene-caller-ids", doc=InputDocumentation(doc="Gene caller ids. Multiple of them can be declared\nseparated by a delimiter (the default is a comma). In\nanvi-gen-variability-profile, if you declare nothing\nyou will get all genes matching your other filtering\ncriteria. In other programs, you may get everything,\nnothing, or an error. It really depends on the\nsituation. Fortunately, mistakes are cheap, so it's\nworth a try. (default: None)")), ToolInput(tag="in_only_if_structure", input_type=Boolean(optional=True), prefix="--only-if-structure", doc=InputDocumentation(doc="If provided, your genes of interest will be further\nsubset to only include genes with structures in your\nstructure database, and therefore must be supplied in\nconjunction with a structure database, i.e. `-s\n<your_structure_database>`. If you did not specify\ngenes of interest, ALL genes will be subset to those\nthat have structures. (default: False)")), ToolInput(tag="in_samples_of_interest", input_type=File(optional=True), prefix="--samples-of-interest", doc=InputDocumentation(doc="A file with samples names. There should be only one\ncolumn in the file, and each line should correspond to\na unique sample name (without a column header).\n(default: None)")), ToolInput(tag="in_engine", input_type=String(optional=True), prefix="--engine", doc=InputDocumentation(doc="Variability engine. The default is 'NT'.")), ToolInput(tag="in_skip_synonym_ity", input_type=Boolean(optional=True), prefix="--skip-synonymity", doc=InputDocumentation(doc="Computing synonymity can be an expensive operation for\nlarge data sets. Provide this flag to skip computing\nsynonymity. It only makes sense to provide this flag\nwhen using --engine CDN. (default: False)")), ToolInput(tag="in_num_positions_from_each_split", input_type=Int(optional=True), prefix="--num-positions-from-each-split", doc=InputDocumentation(doc="Each split may have one or more variable positions. By\ndefault, anvi'o will report every SNV position found\nin a given split. This parameter will help you to\ndefine a cutoff for the maximum number of SNVs to be\nreported from a split (if the number of SNVs is more\nthan the number you declare using this parameter, the\npositions will be randomly subsampled). (default: 0)")), ToolInput(tag="in_min_departure_from_reference", input_type=Float(optional=True), prefix="--min-departure-from-reference", doc=InputDocumentation(doc="Takes a value between 0 and 1, where 1 is maximum\ndivergence from the reference. Default is 0.000000.\nThe reference here observation that corresponds to a\ngiven position in the mapped context.")), ToolInput(tag="in_max_departure_from_reference", input_type=Float(optional=True), prefix="--max-departure-from-reference", doc=InputDocumentation(doc="Similar to '--min-departure-from-reference', but\ndefines an upper limit for divergence. The default is\n1.000000.")), ToolInput(tag="in_min_departure_from_consensus", input_type=Float(optional=True), prefix="--min-departure-from-consensus", doc=InputDocumentation(doc="Takes a value between 0 and 1, where 1 is maximum\ndivergence from the consensus for a given position.\nThe default is 0.000000. The consensus is the most\nfrequent observation at a given position.")), ToolInput(tag="in_max_departure_from_consensus", input_type=Float(optional=True), prefix="--max-departure-from-consensus", doc=InputDocumentation(doc="Similar to '--min-departure-from-consensus', but\ndefines an upper limit for divergence. The default is\n1.000000.")), ToolInput(tag="in_min_occurrence", input_type=Int(optional=True), prefix="--min-occurrence", doc=InputDocumentation(doc="Minimum number of samples a nucleotide position should\nbe reported as variable. Default is 1. If you set it\nto 2, for instance, each eligible variable position\nwill be expected to appear in at least two samples,\nwhich will reduce the impact of stochastic, or\nunintelligible variable positions.")), ToolInput(tag="in_min_coverage_in_each_sample", input_type=Int(optional=True), prefix="--min-coverage-in-each-sample", doc=InputDocumentation(doc="Minimum coverage of a given variable nucleotide\nposition in all samples. If a nucleotide position is\ncovered less than this value even in one sample, it\nwill be removed from the analysis. Default is 0.")), ToolInput(tag="in_quince_mode", input_type=Boolean(optional=True), prefix="--quince-mode", doc=InputDocumentation(doc="The default behavior is to report base frequencies of\nnucleotide positions only if there is any variation\nreported during profiling (which by default uses some\nheuristics to minimize the impact of error-driven\nvariation). So, if there are 10 samples, and a given\nposition has been reported as a variable site during\nprofiling in only one of those samples, there will be\nno information will be stored in the database for the\nremaining 9. When this flag is used, we go back to\neach sample, and report base frequencies for each\nsample at this position even if they do not vary. It\nwill take considerably longer to report when this flag\nis on, and the use of it will increase the file size\ndramatically, however it is inevitable for some\nstatistical approaches (as well as for some beautiful\nvisualizations). (default: False)")), ToolInput(tag="in_output_file", input_type=File(optional=True), prefix="--output-file", doc=InputDocumentation(doc="File path to store results. (default: variability.txt)")), ToolInput(tag="in_include_contig_names", input_type=Boolean(optional=True), prefix="--include-contig-names", doc=InputDocumentation(doc="Use this flag if you would like contig names for each\nvariable position to be included in the output file as\na column. By default, we do not include contig names\nsince they can practically double the output file size\nwithout any actual benefit in most cases. (default:\nFalse)")), ToolInput(tag="in_include_split_names", input_type=File(optional=True), prefix="--include-split-names", doc=InputDocumentation(doc="Use this flag if you would like split names for each\nvariable position to be included in the output file as\na column. (default: False)")), ToolInput(tag="in_include_additional_data", input_type=Boolean(optional=True), prefix="--include-additional-data", doc=InputDocumentation(doc="Use this flag if you would like to append data stored\nin the `amino_acid_additional_data` table as\nadditional columns to your output. NOTE: This is not\nyet implemented for the `nucleotide_additional_data`\ntable. (default: False)")), ToolInput(tag="in_compute_gene_coverage_stats", input_type=Boolean(optional=True), prefix="--compute-gene-coverage-stats", doc=InputDocumentation(doc="If provided, gene coverage statistics will be appended\nfor each entry in variability report. This is very\nuseful information, but will not be included by\ndefault because it is an expensive operation, and may\ntake some additional time. (default: False)")), ToolInput(tag="in_splits_txt", input_type=String(), position=0, doc=InputDocumentation(doc="üçï Can provide: ")), ToolInput(tag="in_variability_profile_txt", input_type=String(), position=1, doc=InputDocumentation(doc="üç∫ More on `anvi-gen-variability-profile`:"))], outputs=[ToolOutput(tag="out_output_file", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output_file", type_hint=File()), doc=OutputDocumentation(doc="File path to store results. (default: variability.txt)")), ToolOutput(tag="out_include_split_names", output_type=File(optional=True), selector=InputSelector(input_to_select="in_include_split_names", type_hint=File()), doc=OutputDocumentation(doc="Use this flag if you would like split names for each\nvariable position to be included in the output file as\na column. (default: False)"))], container="quay.io/biocontainers/anvio-minimal:7--py_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Anvi_Gen_Variability_Profile_V0_1_0().translate("wdl")

