version 1.0

task Anvigetsplitcoverages {
  input {
    String? profile_db
    String? contigs_db
    String? split_name
    String? collection_name
    String? bin_id
    String? gene_caller_id
    Int? flank_length
    File? output_file
    Boolean? list_splits
    Boolean? list_collections
    Boolean? list_bins
    String coverages_txt
  }
  command <<<
    anvi_get_split_coverages \
      ~{coverages_txt} \
      ~{if defined(profile_db) then ("--profile-db " +  '"' + profile_db + '"') else ""} \
      ~{if defined(contigs_db) then ("--contigs-db " +  '"' + contigs_db + '"') else ""} \
      ~{if defined(split_name) then ("--split-name " +  '"' + split_name + '"') else ""} \
      ~{if defined(collection_name) then ("--collection-name " +  '"' + collection_name + '"') else ""} \
      ~{if defined(bin_id) then ("--bin-id " +  '"' + bin_id + '"') else ""} \
      ~{if defined(gene_caller_id) then ("--gene-caller-id " +  '"' + gene_caller_id + '"') else ""} \
      ~{if defined(flank_length) then ("--flank-length " +  '"' + flank_length + '"') else ""} \
      ~{if defined(output_file) then ("--output-file " +  '"' + output_file + '"') else ""} \
      ~{if (list_splits) then "--list-splits" else ""} \
      ~{if (list_collections) then "--list-collections" else ""} \
      ~{if (list_bins) then "--list-bins" else ""}
  >>>
  runtime {
    docker: "quay.io/biocontainers/anvio-minimal:7--py_0"
  }
  parameter_meta {
    profile_db: "Anvi'o profile database (default: None)"
    contigs_db: "Anvi'o contigs database generated by 'anvi-gen-\\ncontigs-database' (default: None)"
    split_name: "Split name. (default: None)"
    collection_name: "Collection name. (default: None)"
    bin_id: "Bin name you are interested in. (default: None)"
    gene_caller_id: "A single gene id. (default: None)"
    flank_length: "Extend sequences for gene calls with additional\\nnucleotides from both ends. If the seqeunce for a\\ntarget gene is between nucleotide positions START and\\nSTOP, using a flank lenght of M will give you a\\nsequence that starts at START - M and ends at STOP +\\nM. (default: 0)"
    output_file: "File path to store results. (default: None)"
    list_splits: "When declared, the program will list split names in\\nthe profile database and quite (default: False)"
    list_collections: "Show available collections and exit. (default: False)"
    list_bins: "List available bins in a collection and exit.\\n(default: False)"
    coverages_txt: "üç∫ More on `anvi-get-split-coverages`:"
  }
  output {
    File out_stdout = stdout()
    File out_output_file = "${in_output_file}"
  }
}