class: CommandLineTool
id: anvi_display_metabolism.cwl
inputs:
- id: in_contigs_db
  doc: "Anvi'o contigs database generated by 'anvi-gen-\ncontigs-database' (default:\
    \ None)"
  type: string?
  inputBinding:
    prefix: --contigs-db
- id: in_meta_genome_mode
  doc: "Treat a given contigs database as a metagenome rather\nthan treating it as\
    \ a single genome. (default: False)"
  type: boolean?
  inputBinding:
    prefix: --metagenome-mode
- id: in_kegg_data_dir
  doc: "The directory path for your KEGG setup, which will\ninclude things like KOfam\
    \ profiles and KEGG MODULE\ndata. Anvi'o will try to use the default path if you\n\
    do not specify anything. (default: None)"
  type: File?
  inputBinding:
    prefix: --kegg-data-dir
- id: in_collection_name
  doc: 'Collection name. (default: None)'
  type: string?
  inputBinding:
    prefix: --collection-name
- id: in_bin_id
  doc: 'Bin name you are interested in. (default: None)'
  type: string?
  inputBinding:
    prefix: --bin-id
- id: in_bin_ids_file
  doc: "Text file for bins (each line should be a unique bin\nid). (default: None)"
  type: File?
  inputBinding:
    prefix: --bin-ids-file
- id: in_module_completion_threshold
  doc: "This threshold defines the point at which we consider\na KEGG module to be\
    \ 'complete' or 'present' in a given\ngenome or bin. It is the fraction of steps\
    \ that must\nbe complete in in order for the entire module to be\nmarked complete.\
    \ The default is 0.75."
  type: long?
  inputBinding:
    prefix: --module-completion-threshold
- id: in_ip_address
  doc: "IP address for the HTTP server. The default ip address\n(0.0.0.0) should work\
    \ just fine for most."
  type: double?
  inputBinding:
    prefix: --ip-address
- id: in_port_number
  doc: "Port number to use for anvi'o services. If nothing is\ndeclared, anvi'o will\
    \ try to find a suitable port\nnumber, starting from the default port number,\
    \ 8080.\n(default: None)"
  type: long?
  inputBinding:
    prefix: --port-number
- id: in_browser_path
  doc: "By default, anvi'o will use your default browser to\nlaunch the interactive\
    \ interface. If you would like to\nuse something else than your system default,\
    \ you can\nprovide a full path for an alternative browser using\nthis parameter,\
    \ and hope for the best. For instance we\nare using this parameter to call Google's\
    \ experimental\nbrowser, Canary, which performs better with demanding\nvisualizations.\
    \ (default: None)"
  type: File?
  inputBinding:
    prefix: --browser-path
- id: in_server_only
  doc: "The default behavior is to start the local server, and\nfire up a browser\
    \ that connects to the server. If you\nhave other plans, and want to start the\
    \ server without\ncalling the browser, this is the flag you need.\n(default: False)"
  type: boolean?
  inputBinding:
    prefix: --server-only
- id: in_password_protected
  doc: "If this flag is set, command line tool will ask you to\nenter a password and\
    \ interactive interface will be\nonly accessible after entering same password.\
    \ This\noption is recommended for shared machines like\nclusters or shared networks\
    \ where computers are not\nisolated. (default: False)"
  type: boolean?
  inputBinding:
    prefix: --password-protected
- id: in_respectively_dot
  doc: "-p PROFILE_DB, --profile-db PROFILE_DB\nAnvi'o profile database (default:\
    \ None)"
  type: string
  inputBinding:
    position: 0
- id: in_interactive
  doc: 'üç∫ More on `anvi-display-metabolism`:'
  type: string
  inputBinding:
    position: 0
outputs:
- id: out_stdout
  doc: Standard output stream
  type: stdout
hints:
- class: DockerRequirement
  dockerPull: quay.io/biocontainers/anvio-minimal:7--py_0
cwlVersion: v1.1
baseCommand:
- anvi-display-metabolism
