version 1.0

task Anviexportgenecalls {
  input {
    String? anvianvigencontigsdatabase_default_none
    File? output_file
    String? gene_caller
    Boolean? list_gene_callers
    Boolean? skip_sequence_reporting
    String _can_provide
    String gene_calls_txt
  }
  command <<<
    anvi_export_gene_calls \
      ~{_can_provide} \
      ~{gene_calls_txt} \
      ~{if defined(anvianvigencontigsdatabase_default_none) then ("--contigs-db " +  '"' + anvianvigencontigsdatabase_default_none + '"') else ""} \
      ~{if defined(output_file) then ("--output-file " +  '"' + output_file + '"') else ""} \
      ~{if defined(gene_caller) then ("--gene-caller " +  '"' + gene_caller + '"') else ""} \
      ~{if (list_gene_callers) then "--list-gene-callers" else ""} \
      ~{if (skip_sequence_reporting) then "--skip-sequence-reporting" else ""}
  >>>
  runtime {
    docker: "quay.io/biocontainers/anvio-minimal:7--py_0"
  }
  parameter_meta {
    anvianvigencontigsdatabase_default_none: "Anvi'o contigs database generated by 'anvi-gen-\\ncontigs-database' (default: None)"
    output_file: "File path to store results. (default: None)"
    gene_caller: "Which gene caller(s) would you like to export gene\\ncalls for? If providing multiple they should be comma-\\nseparated (no spaces). If you don't know, use --list-\\ngene-callers (default: None)"
    list_gene_callers: "List available gene callers in the contigs database\\nand quit. (default: False)"
    skip_sequence_reporting: "By default, exported gene calls have an amino acid\\nsequences column in the output. Turn this behavior off\\nwith this flag (default: False)"
    _can_provide: "üçï Can provide: "
    gene_calls_txt: "üç∫ More on `anvi-export-gene-calls`:"
  }
  output {
    File out_stdout = stdout()
    File out_output_file = "${in_output_file}"
  }
}