version 1.0

task Anviestimatetrnataxonomy {
  input {
    String? contigs_db
    Boolean? meta_genome_mode
    String? profile_db
    String? collection_name
    File? meta_genomes
    Int? max_num_target_sequences
    File? output_file
    File? per_anticodon_output_file
    File? output_file_prefix
    String? taxonomic_level
    Boolean? matrix_format
    Boolean? raw_output
    Int? num_threads
    String? anticodon_for_meta_genome_mode
    File? report_anticodon_frequencies
    Boolean? just_do_it
    Boolean? simplify_taxonomy_information
    Boolean? compute_anticodon_coverages
    Boolean? update_profile_db_with_taxonomy
    File? taxonomy_database
    String consideration_dot
  }
  command <<<
    anvi_estimate_trna_taxonomy \
      ~{consideration_dot} \
      ~{if defined(contigs_db) then ("--contigs-db " +  '"' + contigs_db + '"') else ""} \
      ~{if (meta_genome_mode) then "--metagenome-mode" else ""} \
      ~{if defined(profile_db) then ("--profile-db " +  '"' + profile_db + '"') else ""} \
      ~{if defined(collection_name) then ("--collection-name " +  '"' + collection_name + '"') else ""} \
      ~{if defined(meta_genomes) then ("--metagenomes " +  '"' + meta_genomes + '"') else ""} \
      ~{if defined(max_num_target_sequences) then ("--max-num-target-sequences " +  '"' + max_num_target_sequences + '"') else ""} \
      ~{if defined(output_file) then ("--output-file " +  '"' + output_file + '"') else ""} \
      ~{if defined(per_anticodon_output_file) then ("--per-anticodon-output-file " +  '"' + per_anticodon_output_file + '"') else ""} \
      ~{if defined(output_file_prefix) then ("--output-file-prefix " +  '"' + output_file_prefix + '"') else ""} \
      ~{if defined(taxonomic_level) then ("--taxonomic-level " +  '"' + taxonomic_level + '"') else ""} \
      ~{if (matrix_format) then "--matrix-format" else ""} \
      ~{if (raw_output) then "--raw-output" else ""} \
      ~{if defined(num_threads) then ("--num-threads " +  '"' + num_threads + '"') else ""} \
      ~{if defined(anticodon_for_meta_genome_mode) then ("--anticodon-for-metagenome-mode " +  '"' + anticodon_for_meta_genome_mode + '"') else ""} \
      ~{if defined(report_anticodon_frequencies) then ("--report-anticodon-frequencies " +  '"' + report_anticodon_frequencies + '"') else ""} \
      ~{if (just_do_it) then "--just-do-it" else ""} \
      ~{if (simplify_taxonomy_information) then "--simplify-taxonomy-information" else ""} \
      ~{if (compute_anticodon_coverages) then "--compute-anticodon-coverages" else ""} \
      ~{if (update_profile_db_with_taxonomy) then "--update-profile-db-with-taxonomy" else ""} \
      ~{if defined(taxonomy_database) then ("--taxonomy-database " +  '"' + taxonomy_database + '"') else ""}
  >>>
  runtime {
    docker: "quay.io/biocontainers/anvio-minimal:7--py_0"
  }
  parameter_meta {
    contigs_db: "Anvi'o contigs database generated by 'anvi-gen-\\ncontigs-database' (default: None)"
    meta_genome_mode: "Treat a given contigs database as a metagenome rather\\nthan treating it as a single genome. (default: False)"
    profile_db: "Anvi'o profile database (default: None)"
    collection_name: "Collection name. (default: None)"
    meta_genomes: "A two-column TAB-delimited flat text file. The header\\nline must contain these columns: 'name',\\n'contigs_db_path', and 'profile_db_path'. Each line\\nshould list a single entry, where 'name' can be any\\nname to describe the metagenome stored in the anvi'o\\ncontigs database. In this context, the anvi'o profiles\\nassociated with contigs database must be SINGLE\\nPROFILES, as in generated by the program `anvi-\\nprofile` and not `anvi-merge`. (default: None)"
    max_num_target_sequences: "Maximum number of target sequences to request from\\nBLAST or DIAMOND searches. The default is 20%."
    output_file: "File path to store results. (default: None)"
    per_anticodon_output_file: "A more detailed output file that will describe\\ntaxonomy of each anticodon in a single bin. When\\nconsensus taxonomy is generated per bin or genome,\\ntaxonomy for each underlying item is not reported.\\nThis additional optional output file will elucidate\\nthings. (default: None)"
    output_file_prefix: "A prefix to be used while naming the output files (no\\nfile type extensions please; just a prefix). (default:\\nNone)"
    taxonomic_level: "The taxonomic level to use whenever relevant and/or\\navailable. The default taxonomic level is None, but if\\nyou choose something specific, anvi'o will focus on\\nthat whenever possible."
    matrix_format: "If you want the reports to look like sparse matrices\\nwhenever possible, declare this flag. Matrices are\\nespecially good to use when you are working with\\ninternal/external genomes since they can show you\\nquickly the distribution of each taxon across all\\nmetagenomes in programs like EXCEL. WELL TRY IT AND\\nSEE. (default: False)"
    raw_output: "Just store the raw output without any processing of\\nthe primary data structure. (default: False)"
    num_threads: "Maximum number of threads to use for multithreading\\nwhenever possible. Very conservatively, the default is\\n1. It is a good idea to not exceed the number of CPUs\\n/ cores on your system. Plus, please be careful with\\nthis option if you are running your commands on a SGE\\n--if you are clusterizing your runs, and asking for\\nmultiple threads to use, you may deplete your\\nresources very fast. (default: 1)"
    anticodon_for_meta_genome_mode: "When running in metagenome mode, anvi'o automatically\\nchooses the most frequent anticodon to estimate the\\ntaxonomic composition within a contigs database. If\\nyou have a different preference you can use this\\nparameter to communicate that. (default: None)"
    report_anticodon_frequencies: "Report anticodon frequencies in a TAB-delimited file\\nand quit. This is a great way to decide which\\nanticodon to use in metagenome mode (we often wish to\\nuse the most frequent anticodon to increase the\\ndetection of taxa). (default: None)"
    just_do_it: "Don't bother me with questions or warnings, just do\\nit. (default: False)"
    simplify_taxonomy_information: "The taxonomy output may include a large number of\\nnames that contain clade-specific code for not-yet-\\ncharacterized taxa. With this flag you can simplify\\ntaxon names. This will influence all output files and\\ndisplays as the use of this flag will on-the-fly trim\\ntaxonomic levels with clade-specific code names.\\n(default: False)"
    compute_anticodon_coverages: "When this flag is declared, anvi'o will go back to the\\nprofile database to learn coverage statistics of tRNA\\ngenes used for taxonomy. (default: False)"
    update_profile_db_with_taxonomy: "When anvi'o knows all both taxonomic affiliations and\\ncoverages across samples for single-copy core genes,\\nit can, in theory add this information to the profile\\ndatabase. With this flag you can instruct anvi'o to do\\nthat and find information on taxonomy in the `layers`\\ntab of your interactive interface. (default: False)"
    taxonomy_database: "Path to the directory that contains the BLAST\\ndatabases for single-copy core genes. You will almost\\nnever need to use this parameter unless you are trying\\nsomething very fancy. But when you do, you can tell\\nanvi'o where to look for database files through this\\nparameter. (default: None)"
    consideration_dot: "--dna-sequence FASTA  A FASTA-formatted input file (default: None)"
  }
  output {
    File out_stdout = stdout()
    File out_output_file = "${in_output_file}"
    File out_per_anticodon_output_file = "${in_per_anticodon_output_file}"
    File out_output_file_prefix = "${in_output_file_prefix}"
  }
}