version 1.0

task AnviExportGeneCalls {
  input {
    String? contigs_db
    File? output_file
    String? gene_caller
    Boolean? list_gene_callers
    Boolean? skip_sequence_reporting
  }
  command <<<
    anvi-export-gene-calls \
      ~{if defined(contigs_db) then ("--contigs-db " +  '"' + contigs_db + '"') else ""} \
      ~{if defined(output_file) then ("--output-file " +  '"' + output_file + '"') else ""} \
      ~{if defined(gene_caller) then ("--gene-caller " +  '"' + gene_caller + '"') else ""} \
      ~{true="--list-gene-callers" false="" list_gene_callers} \
      ~{true="--skip-sequence-reporting" false="" skip_sequence_reporting}
  >>>
  parameter_meta {
    contigs_db: "Anvi'o contigs database generated by 'anvi-gen- contigs'"
    output_file: "File path to store results."
    gene_caller: "Which gene caller(s) would you like to export gene calls for? If providing multiple they should be comma- separated (no spaces). If you don't know, use --list- gene-callers"
    list_gene_callers: "List available gene callers in the contigs database and quit."
    skip_sequence_reporting: "By default, exported gene calls have an amino acid sequences column in the output. Turn this behavior off with this flag"
  }
}