version 1.0

task OptiTypePipeline.py {
  input {
    Array[String] files_fished_raw
    Boolean? rna
    Boolean? dna
    String? beta
    String? enumerate
    String? outdir
    String? prefix
    Boolean? verbose
    String? config
    String opti_type
  }
  command <<<
    OptiTypePipeline.py \
      ~{opti_type} \
      ~{if defined(files_fished_raw) then ("--input " +  '"' + files_fished_raw + '"') else ""} \
      ~{true="--rna" false="" rna} \
      ~{true="--dna" false="" dna} \
      ~{if defined(beta) then ("--beta " +  '"' + beta + '"') else ""} \
      ~{if defined(enumerate) then ("--enumerate " +  '"' + enumerate + '"') else ""} \
      ~{if defined(outdir) then ("--outdir " +  '"' + outdir + '"') else ""} \
      ~{if defined(prefix) then ("--prefix " +  '"' + prefix + '"') else ""} \
      ~{true="--verbose" false="" verbose} \
      ~{if defined(config) then ("--config " +  '"' + config + '"') else ""}
  >>>
  parameter_meta {
    files_fished_raw: ".fastq file(s) (fished or raw) or .bam files stored for re-use, generated by an earlier OptiType run. One file: single-end mode, two files: paired-end mode."
    rna: "Use with RNA sequencing data."
    dna: "Use with DNA sequencing data."
    beta: "The beta value for for homozygosity detection (see paper). Default: 0.009. Handle with care."
    enumerate: "Number of enumerations. OptiType will output the optimal solution and the top N-1 suboptimal solutions in the results CSV. Default: 1"
    outdir: "Specifies the out directory to which all files should be written."
    prefix: "Specifies prefix of output files"
    verbose: "Set verbose mode on."
    config: "Path to config file. Default: config.ini in the same directory as this script"
    opti_type: ""
  }
}