from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, String, Directory, File, Int

Oncotator_V0_1_0 = CommandToolBuilder(tool="Oncotator", base_command=["Oncotator"], inputs=[ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="--verbose", doc=InputDocumentation(doc="set verbosity level [default: 5]")), ToolInput(tag="in_input_format", input_type=String(optional=True), prefix="--input_format", doc=InputDocumentation(doc="Input format. Note that MAFLITE will work for any tsv\nfile with appropriate headers, so long as all of the\nrequired headers (or an alias -- see maflite.config)\nare present. Note that '-i TCGAMAF' is the same as\nspecifying '-i MAFLITE --prune-tcga-maf-cols --allow-\noverwriting' [default: MAFLITE]")), ToolInput(tag="in_db_dir", input_type=Directory(optional=True), prefix="--db-dir", doc=InputDocumentation(doc="Main annotation database directory. [default: /xchip/c\nga/reference/annotation/db/oncotator_v1_ds_gencode_cur\nrent/]")), ToolInput(tag="in_output_format", input_type=String(optional=True), prefix="--output_format", doc=InputDocumentation(doc="Output format. [default: TCGAMAF]")), ToolInput(tag="in_override_config", input_type=File(optional=True), prefix="--override_config", doc=InputDocumentation(doc="File path to manual annotations in a config file\nformat (section is 'manual_annotations' and\nannotation:value pairs).")), ToolInput(tag="in_default_config", input_type=File(optional=True), prefix="--default_config", doc=InputDocumentation(doc="File path to default annotation values in a config\nfile format (section is 'manual_annotations' and\nannotation:value pairs).")), ToolInput(tag="in_no_multicore", input_type=Boolean(optional=True), prefix="--no-multicore", doc=InputDocumentation(doc="Disables all multicore functionality.")), ToolInput(tag="in_annotate_manual", input_type=Int(optional=True), prefix="--annotate-manual", doc=InputDocumentation(doc="Specify annotations to override. Can be specified\nmultiple times. E.g. -a 'name1:value1' -a\n'name2:value2'")), ToolInput(tag="in_annotate_default", input_type=Int(optional=True), prefix="--annotate-default", doc=InputDocumentation(doc="Specify default values for annotations. Can be\nspecified multiple times. E.g. -d 'name1:value1' -d\n'name2:value2'")), ToolInput(tag="in_cache_url", input_type=String(optional=True), prefix="--cache-url", doc=InputDocumentation(doc="URL to use for cache. See help for examples.")), ToolInput(tag="in_read_only_cache", input_type=Boolean(optional=True), prefix="--read_only_cache", doc=InputDocumentation(doc="Makes the cache read-only")), ToolInput(tag="in_tx_mode", input_type=String(optional=True), prefix="--tx-mode", doc=InputDocumentation(doc="Specify transcript mode for transcript providing\ndatasources that support multiple modes. [default:\nCANONICAL]")), ToolInput(tag="in_infer_genotypes", input_type=String(optional=True), prefix="--infer_genotypes", doc=InputDocumentation(doc="Forces the VCF output renderer to populate the output\ngenotypes as heterozygous. This option should only be\nused when converting a MAFLITE to a VCF; otherwise,\nthe option has no effect. [default: false]")), ToolInput(tag="in_skip_no_alt", input_type=Boolean(optional=True), prefix="--skip-no-alt", doc=InputDocumentation(doc="If specified, any mutation with annotation\nalt_allele_seen of 'False' will not be annotated or\nrendered. Do not use if output format is a VCF. If\nalt_allele_seen annotation is missing, render the\nmutation.")), ToolInput(tag="in_log_name", input_type=String(optional=True), prefix="--log_name", doc=InputDocumentation(doc="Specify log output location. Default: oncotator.log")), ToolInput(tag="in_prepend", input_type=Boolean(optional=True), prefix="--prepend", doc=InputDocumentation(doc="If specified for TCGAMAF output, will put a 'i_' in\nfront of fields that are not directly rendered in\nOncotator TCGA MAFs")), ToolInput(tag="in_infer_on_ps", input_type=Boolean(optional=True), prefix="--infer-onps", doc=InputDocumentation(doc="Will merge adjacent SNPs,DNPs,TNPs,etc if they are in\nthe same sample. This assumes that the input file is\nposition sorted. This may cause problems with VCF ->\nVCF conversion, and does not guarantee input order is\nmaintained.")), ToolInput(tag="in_canonical_tx_file", input_type=File(optional=True), prefix="--canonical-tx-file", doc=InputDocumentation(doc="Simple text file with list of transcript IDs (one per\nline) to always select where possible for variants.\nTranscript IDs must match the ones used by the\ntranscript provider in your datasource (e.g. gencode\nENST00000123456). If more than one transcript can be\nselected for a variant, uses the method defined by\n--tx-mode to break ties. Using this list means that a\ntranscript will be selected from this list first,\npossibly superseding a best-effect. Note that\ntranscript version number is not considered, whether\nincluded in the list or not.")), ToolInput(tag="in_collapse_filter_cols", input_type=Boolean(optional=True), prefix="--collapse-filter-cols", doc=InputDocumentation(doc="Render FILTER columns from VCF input as single\n'filter' column when using TCGAMAF output option.")), ToolInput(tag="in_re_annotate_tcga_maf_cols", input_type=Boolean(optional=True), prefix="--reannotate-tcga-maf-cols", doc=InputDocumentation(doc="Prefer new, annotated values to those specified by the\ninput file. Only useful when output is TCGA MAF and\nwhen --allow-overwriting is specified. Automatically\nturned on with -i TCGAMAF")), ToolInput(tag="in_allow_overwriting", input_type=Boolean(optional=True), prefix="--allow-overwriting", doc=InputDocumentation(doc="Allow annotations to be overwritten (no\nDuplicateAnnotationException errors). This should only\nbe used in rare cases and user should know when that\nis. Automatically turned on with -i TCGAMAF")), ToolInput(tag="in_collapse_number_annotations", input_type=Boolean(optional=True), prefix="--collapse-number-annotations", doc=InputDocumentation(doc="Advanced: For TCGA MAF output, collapse a set of known\nnumeric fields that may have been annotated with a\npipe. This can be useful for downstream tools that are\nexpecting a single value.")), ToolInput(tag="in_longer_other_tx", input_type=Boolean(optional=True), prefix="--longer-other-tx", doc=InputDocumentation(doc="Adds some select field(s) to the other_transcript")), ToolInput(tag="in_v", input_type=Boolean(optional=True), prefix="-V", doc=InputDocumentation(doc="")), ToolInput(tag="in_input_file", input_type=String(), position=0, doc=InputDocumentation(doc="Input file to be annotated. Type is specified through")), ToolInput(tag="in_options_dot", input_type=String(), position=1, doc=InputDocumentation(doc="output_file           Output file name of annotated file.")), ToolInput(tag="in_genome_build", input_type=String(), position=2, doc=InputDocumentation(doc="Genome build. For example: hg19")), ToolInput(tag="in_field", input_type=String(), position=0, doc=InputDocumentation(doc="--prune-filter-cols   Prune mutations from VCF input as based on 'filter'"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Oncotator_V0_1_0().translate("wdl", allow_empty_container=True)

