from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import String, File, Boolean, Int, Float

Anvi_Get_Sequences_For_Gene_Clusters_V0_1_0 = CommandToolBuilder(tool="anvi_get_sequences_for_gene_clusters", base_command=["anvi-get-sequences-for-gene-clusters"], inputs=[ToolInput(tag="in_pan_db", input_type=String(optional=True), prefix="--pan-db", doc=InputDocumentation(doc="Anvi'o pan database")), ToolInput(tag="in_genomes_storage", input_type=File(optional=True), prefix="--genomes-storage", doc=InputDocumentation(doc="Anvi'o genomes storage file")), ToolInput(tag="in_output_file", input_type=File(optional=True), prefix="--output-file", doc=InputDocumentation(doc="File path to store results.")), ToolInput(tag="in_report_dna_sequences", input_type=Boolean(optional=True), prefix="--report-DNA-sequences", doc=InputDocumentation(doc="By default, this program reports amino acid sequences.\nUse this flag to report DNA sequences instead.")), ToolInput(tag="in_gene_cluster_id", input_type=String(optional=True), prefix="--gene-cluster-id", doc=InputDocumentation(doc="Gene cluster ID you are interested in.")), ToolInput(tag="in_gene_cluster_ids_file", input_type=File(optional=True), prefix="--gene-cluster-ids-file", doc=InputDocumentation(doc="Text file for gene clusters (each line should contain\nbe a unique gene cluster id).")), ToolInput(tag="in_collection_name", input_type=String(optional=True), prefix="--collection-name", doc=InputDocumentation(doc="Collection name.")), ToolInput(tag="in_bin_id", input_type=String(optional=True), prefix="--bin-id", doc=InputDocumentation(doc="Bin name you are interested in.")), ToolInput(tag="in_min_num_genomes_gene_cluster_occurs", input_type=Int(optional=True), prefix="--min-num-genomes-gene-cluster-occurs", doc=InputDocumentation(doc="This filter will remove gene clusters from your\nreport. Let's assume you have 100 genomes in your pan\ngenome analysis. You can use this parameter if you\nwant to work only with gene clusters that occur in at\nleast X number of genomes. If you say '--min-num-\ngenomes-gene-cluster-occurs 90', each gene cluster in\nthe analysis will be required at least to appear in 90\ngenomes. If a gene occurs in less than that number of\ngenomes, it simply will not be reported. This is\nespecially useful for phylogenomic analyses, where you\nmay want to only focus on gene clusters that are\nprevalent across the set of genomes you wish to\nanalyze.")), ToolInput(tag="in_max_num_genomes_gene_cluster_occurs", input_type=Int(optional=True), prefix="--max-num-genomes-gene-cluster-occurs", doc=InputDocumentation(doc="This filter will remove gene clusters from your\nreport. Let's assume you have 100 genomes in your pan\ngenome analysis. You can use this parameter if you\nwant to work only with gene clusters that occur in at\nmost X number of genomes. If you say '--max-num-\ngenomes-gene-cluster-occurs 1', you will get gene\nclusters that are singletons. Combining this parameter\nwith --min-num-genomes-gene-cluster-occurs can give\nyou a very precise way to filter your gene clusters.")), ToolInput(tag="in_min_num_genes_from_each_genome", input_type=Int(optional=True), prefix="--min-num-genes-from-each-genome", doc=InputDocumentation(doc="This filter will remove gene clusters from your\nreport. If you say '--min-num-genes-from-each-genome\n2', this filter will remove every gene cluster, to\nwhich every genome in your analysis contributed less\nthan 2 genes. This can be useful to find out gene\nclusters with many genes from many genomes (such as\nconserved multi-copy genes within a clade).")), ToolInput(tag="in_max_num_genes_from_each_genome", input_type=Int(optional=True), prefix="--max-num-genes-from-each-genome", doc=InputDocumentation(doc="This filter will remove gene clusters from your\nreport. If you say '--max-num-genes-from-each-genome\n1', every gene cluster that has more than one gene\nfrom any genome that contributes to it will be removed\nfrom your analysis. This could be useful to remove\ngene clusters with paralogs from your report for\nappropriate phylogenomic analyses. For instance, using\n'--max-num-genes-from-each-genome 1' and 'min-num-\ngenomes-gene-cluster-occurs X' where X is the total\nnumber of your genomes, would give you the single-copy\ngene clusters in your pan genome.")), ToolInput(tag="in_max_num_gene_clusters_missing_from_genome", input_type=Int(optional=True), prefix="--max-num-gene-clusters-missing-from-genome", doc=InputDocumentation(doc="This filter will remove genomes from your report. If\nyou have a list of gene cluster names, you can use\nthis parameter to omit any genome from your report if\nit is missing more than a number of genes you desire.\nFor instance, if you have 100 genomes in your pan\ngenome, and you are interested in working only with\ngenomes that have all 5 specific gene clusters of your\nchoice, you can use '--max-num-gene-clusters-missing-\nfrom-genome 4' to remove remove the bins that are\nmissing more than 4 of those 5 genes. This is\nespecially useful for phylogenomic analyses. Parameter\n0 will remove any genome that is missing any of the\ngenes.")), ToolInput(tag="in_min_functional_homogeneity_index", input_type=Float(optional=True), prefix="--min-functional-homogeneity-index", doc=InputDocumentation(doc="This filter will remove gene clusters from your\nreport. If you say '--min-functional-homogeneity-index\n0.3', every gene cluster with a functional homogeneity\nindex less than 0.3 will be removed from your\nanalysis. This can be useful if you only want to look\nat gene clusters that are highly conserved in\nresulting function")), ToolInput(tag="in_max_functional_homogeneity_index", input_type=Float(optional=True), prefix="--max-functional-homogeneity-index", doc=InputDocumentation(doc="This filter will remove gene clusters from your\nreport. If you say '--max-functional-homogeneity-index\n0.5', every gene cluster with a functional homogeneity\nindex greater than 0.5 will be removed from your\nanalysis. This can be useful if you only want to look\nat gene clusters that don't seem to be functionally\nconserved")), ToolInput(tag="in_min_geometric_homogeneity_index", input_type=Float(optional=True), prefix="--min-geometric-homogeneity-index", doc=InputDocumentation(doc="This filter will remove gene clusters from your\nreport. If you say '--min-geometric-homogeneity-index\n0.3', every gene cluster with a geometric homogeneity\nindex less than 0.3 will be removed from your\nanalysis. This can be useful if you only want to look\nat gene clusters that are highly conserved in\ngeometric configuration")), ToolInput(tag="in_max_geometric_homogeneity_index", input_type=Float(optional=True), prefix="--max-geometric-homogeneity-index", doc=InputDocumentation(doc="This filter will remove gene clusters from your\nreport. If you say '--max-geometric-homogeneity-index\n0.5', every gene cluster with a geometric homogeneity\nindex greater than 0.5 will be removed from your\nanalysis. This can be useful if you only want to look\nat gene clusters that have many not be as conserved as\nothers")), ToolInput(tag="in_min_combined_homogeneity_index", input_type=Float(optional=True), prefix="--min-combined-homogeneity-index", doc=InputDocumentation(doc="This filter will remove gene clusters from your\nreport. If you say '--min-combined-homogeneity-index\n0.3', every gene cluster with a combined homogeneity\nindex less than 0.3 will be removed from your\nanalysis. This can be useful if you only want to look\nat gene clusters that are highly conserved overall")), ToolInput(tag="in_max_combined_homogeneity_index", input_type=Float(optional=True), prefix="--max-combined-homogeneity-index", doc=InputDocumentation(doc="This filter will remove gene clusters from your\nreport. If you say '--max-combined-homogeneity-index\n0.5', every gene cluster with a combined homogeneity\nindex greater than 0.5 will be removed from your\nanalysis. This can be useful if you only want to look\nat gene clusters that have many not be as conserved\noverall as others")), ToolInput(tag="in_add_into_items_additional_data_table", input_type=String(optional=True), prefix="--add-into-items-additional-data-table", doc=InputDocumentation(doc="If you use any of the filters, and would like to add\nthe resulting item names into the items additional\ndata table of your database, you can use this\nparameter. You will need to give a name for these\nresults to be saved. If the given name is already in\nthe items additional data table, its contents will be\nreplaced with the new one. Then you can run anvi-\ninteractive or anvi-display-pan to 'see' the results\nof your filters.")), ToolInput(tag="in_list_collections", input_type=Boolean(optional=True), prefix="--list-collections", doc=InputDocumentation(doc="Show available collections and exit.")), ToolInput(tag="in_list_bins", input_type=Boolean(optional=True), prefix="--list-bins", doc=InputDocumentation(doc="List available bins in a collection and exit.")), ToolInput(tag="in_concatenate_gene_clusters", input_type=Boolean(optional=True), prefix="--concatenate-gene-clusters", doc=InputDocumentation(doc="Concatenate output gene clusters in the same order to\ncreate a multi-gene alignment output that is suitable\nfor phylogenomic analyses.")), ToolInput(tag="in_partition_file", input_type=File(optional=True), prefix="--partition-file", doc=InputDocumentation(doc="Some commonly used software for phylogenetic analyses\n(e.g., IQ-TREE, RAxML, etc) allow users to\nspecify/test different substitution models for each\ngene of a concatenated multiple sequence alignments.\nFor this, they use a special file format called a\n'partition file', which indicates the site for each\ngene in the alignment. You can use this parameter to\ndeclare an output path for anvi'o to report a NEXUS\nformat partition file in addition to your FASTA output\n(requested by Massimiliano Molari in #1333).")), ToolInput(tag="in_separator", input_type=String(optional=True), prefix="--separator", doc=InputDocumentation(doc="Characters to separate things (the default is whatever\nis most suitable).")), ToolInput(tag="in_align_with", input_type=String(optional=True), prefix="--align-with", doc=InputDocumentation(doc="The multiple sequence alignment program to use when\nmultiple sequence alignment is necessary. To see all\navailable options, use the flag `--list-aligners`.")), ToolInput(tag="in_list_aligners", input_type=Boolean(optional=True), prefix="--list-aligners", doc=InputDocumentation(doc="Show available software for multiple sequence")), ToolInput(tag="in_just_do_it", input_type=Boolean(optional=True), prefix="--just-do-it", doc=InputDocumentation(doc="Don't bother me with questions or warnings, just do")), ToolInput(tag="in_alignment_dot", input_type=String(), position=0, doc=InputDocumentation(doc="LIFE SAVERS:")), ToolInput(tag="in_it_dot", input_type=String(), position=0, doc=InputDocumentation(doc="--dry-run             Don't do anything real. Test everything, and stop"))], outputs=[ToolOutput(tag="out_output_file", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output_file", type_hint=File()), doc=OutputDocumentation(doc="File path to store results.")), ToolOutput(tag="out_partition_file", output_type=File(optional=True), selector=InputSelector(input_to_select="in_partition_file", type_hint=File()), doc=OutputDocumentation(doc="Some commonly used software for phylogenetic analyses\n(e.g., IQ-TREE, RAxML, etc) allow users to\nspecify/test different substitution models for each\ngene of a concatenated multiple sequence alignments.\nFor this, they use a special file format called a\n'partition file', which indicates the site for each\ngene in the alignment. You can use this parameter to\ndeclare an output path for anvi'o to report a NEXUS\nformat partition file in addition to your FASTA output\n(requested by Massimiliano Molari in #1333)."))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Anvi_Get_Sequences_For_Gene_Clusters_V0_1_0().translate("wdl", allow_empty_container=True)

