from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, String, Int, File

Art_Illumina_V0_1_0 = CommandToolBuilder(tool="art_illumina", base_command=["art_illumina"], inputs=[ToolInput(tag="in_qpr_of_one", input_type=Boolean(optional=True), prefix="--qprof1", doc=InputDocumentation(doc="the first-read quality profile")), ToolInput(tag="in_qpr_of_two", input_type=Boolean(optional=True), prefix="--qprof2", doc=InputDocumentation(doc="the second-read quality profile")), ToolInput(tag="in_amplicon", input_type=String(optional=True), prefix="--amplicon", doc=InputDocumentation(doc="sequencing simulation")), ToolInput(tag="in_rcount", input_type=Boolean(optional=True), prefix="--rcount", doc=InputDocumentation(doc="number of reads/read pairs to be generated per sequence/amplicon (not be used together with -f/--fcov)")), ToolInput(tag="in_id", input_type=Boolean(optional=True), prefix="--id", doc=InputDocumentation(doc="the prefix identification tag for read ID")), ToolInput(tag="in_err_free", input_type=Boolean(optional=True), prefix="--errfree", doc=InputDocumentation(doc="indicate to generate the zero sequencing errors SAM file as well the regular one\nNOTE: the reads in the zero-error SAM file have the same alignment positions\nas those in the regular SAM file, but have no sequencing errors")), ToolInput(tag="in_f_cov", input_type=Boolean(optional=True), prefix="--fcov", doc=InputDocumentation(doc="the fold of read coverage to be simulated or number of reads/read pairs generated for each amplicon")), ToolInput(tag="in_in", input_type=Boolean(optional=True), prefix="--in", doc=InputDocumentation(doc="the filename of input DNA/RNA reference")), ToolInput(tag="in_ins_rate", input_type=Boolean(optional=True), prefix="--insRate", doc=InputDocumentation(doc="the first-read insertion rate (default: 0.00009)")), ToolInput(tag="in_ins_rate_two", input_type=Int(optional=True), prefix="--insRate2", doc=InputDocumentation(doc="second-read insertion rate (default: 0.00015)")), ToolInput(tag="in_del_rate", input_type=Boolean(optional=True), prefix="--delRate", doc=InputDocumentation(doc="the first-read deletion rate (default:  0.00011)")), ToolInput(tag="in_del_rate_two", input_type=Int(optional=True), prefix="--delRate2", doc=InputDocumentation(doc="second-read deletion rate (default: 0.00023)")), ToolInput(tag="in_max_in_del", input_type=Int(optional=True), prefix="--maxIndel", doc=InputDocumentation(doc="maximum total number of insertion and deletion per read (default: up to read length)")), ToolInput(tag="in_len", input_type=Boolean(optional=True), prefix="--len", doc=InputDocumentation(doc="the length of reads to be simulated")), ToolInput(tag="in_mf_len", input_type=Boolean(optional=True), prefix="--mflen", doc=InputDocumentation(doc="the mean size of DNA/RNA fragments for paired-end simulations")), ToolInput(tag="in_mate_pair", input_type=String(optional=True), prefix="--matepair", doc=InputDocumentation(doc="a mate-pair read simulation")), ToolInput(tag="in_cigar_m", input_type=Boolean(optional=True), prefix="--cigarM", doc=InputDocumentation(doc="indicate to use CIGAR 'M' instead of '=/X' for alignment match/mismatch")), ToolInput(tag="in_mask_n", input_type=Boolean(optional=True), prefix="--maskN", doc=InputDocumentation(doc="the cutoff frequency of 'N' in a window size of the read length for masking genomic regions\nNOTE: default: '-nf 1' to mask all regions with 'N'. Use '-nf 0' to turn off masking")), ToolInput(tag="in_no_aln", input_type=File(optional=True), prefix="--noALN", doc=InputDocumentation(doc="do not output ALN alignment file")), ToolInput(tag="in_out", input_type=File(optional=True), prefix="--out", doc=InputDocumentation(doc="the prefix of output filename")), ToolInput(tag="in_paired", input_type=Boolean(optional=True), prefix="--paired", doc=InputDocumentation(doc="indicate a paired-end read simulation or to generate reads from both ends of amplicons\nNOTE: art will automatically switch to a mate-pair simulation if the given mean fragment size >= 2000")), ToolInput(tag="in_quiet", input_type=Boolean(optional=True), prefix="--quiet", doc=InputDocumentation(doc="turn off end of run summary")), ToolInput(tag="in_min_q", input_type=Boolean(optional=True), prefix="--minQ", doc=InputDocumentation(doc="the minimum base quality score")), ToolInput(tag="in_maxq", input_type=Boolean(optional=True), prefix="--maxQ", doc=InputDocumentation(doc="the maxiumum base quality score")), ToolInput(tag="in_q_shift", input_type=Boolean(optional=True), prefix="--qShift", doc=InputDocumentation(doc="the amount to shift every first-read quality score by")), ToolInput(tag="in_q_shift_two", input_type=Boolean(optional=True), prefix="--qShift2", doc=InputDocumentation(doc="the amount to shift every second-read quality score by\nNOTE: For -qs/-qs2 option, a positive number will shift up quality scores (the max is 93)\nthat reduce substitution sequencing errors and a negative number will shift down\nquality scores that increase sequencing errors. If shifting scores by x, the error\nrate will be 1/(10^(x/10)) of the default profile.")), ToolInput(tag="in_rnd_seed", input_type=Boolean(optional=True), prefix="--rndSeed", doc=InputDocumentation(doc="the seed for random number generator (default: system time in second)\nNOTE: using a fixed seed to generate two identical datasets from different runs")), ToolInput(tag="in_sdev", input_type=Boolean(optional=True), prefix="--sdev", doc=InputDocumentation(doc="the standard deviation of DNA/RNA fragment size for paired-end simulations.")), ToolInput(tag="in_sam_out", input_type=Boolean(optional=True), prefix="--samout", doc=InputDocumentation(doc="indicate to generate SAM alignment file")), ToolInput(tag="in_sep_prof", input_type=Boolean(optional=True), prefix="--sepProf", doc=InputDocumentation(doc="indicate to use separate quality profiles for different bases (ATGC)")), ToolInput(tag="in_seq_sys", input_type=Boolean(optional=True), prefix="--seqSys", doc=InputDocumentation(doc="The name of Illumina sequencing system of the built-in profile used for simulation\nNOTE: sequencing system ID names are:\nGA1 - GenomeAnalyzer I (36bp,44bp), GA2 - GenomeAnalyzer II (50bp, 75bp)\nHS10 - HiSeq 1000 (100bp),          HS20 - HiSeq 2000 (100bp),      HS25 - HiSeq 2500 (125bp, 150bp)\nHSXn - HiSeqX PCR free (150bp),     HSXt - HiSeqX TruSeq (150bp),   MinS - MiniSeq TruSeq (50bp)\nMSv1 - MiSeq v1 (250bp),            MSv3 - MiSeq v3 (250bp),        NS50 - NextSeq500 v2 (75bp)"))], outputs=[ToolOutput(tag="out_no_aln", output_type=File(optional=True), selector=InputSelector(input_to_select="in_no_aln", type_hint=File()), doc=OutputDocumentation(doc="do not output ALN alignment file")), ToolOutput(tag="out_out", output_type=File(optional=True), selector=InputSelector(input_to_select="in_out", type_hint=File()), doc=OutputDocumentation(doc="the prefix of output filename"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Art_Illumina_V0_1_0().translate("wdl", allow_empty_container=True)

