class: CommandLineTool
id: tpp.cwl
inputs:
- id: in_protocol
  doc: '[Sassetti|Tn5|Mme1] # which sample prep protocol was used?; sassetti protocol
    is the default; this sets the default transposon and primer sequence'
  type: boolean?
  inputBinding:
    prefix: -protocol
- id: in_primer
  doc: '# prefix of reads corresponding to end of transposon at junction with genomic
    sequence; can override default seq'
  type: string?
  inputBinding:
    prefix: -primer
- id: in_mismatches
  doc: '# when searching for constant regions in reads 1 and 2; default is 1'
  type: long?
  inputBinding:
    prefix: -mismatches
- id: in_flags
  doc: '"<STRING>"  # args to pass to BWA'
  type: boolean?
  inputBinding:
    prefix: -flags
- id: in_bwa_alg
  doc: '[aln|mem]  # Default: mem. Algorithm to use for mapping reads with bwa'
  type: boolean?
  inputBinding:
    prefix: -bwa-alg
- id: in_primer_start_window
  doc: ',INT # position in read to search for start of primer; default is [0,20]'
  type: long?
  inputBinding:
    prefix: -primer-start-window
- id: in_window_size
  doc: '# automatic method to set window'
  type: long?
  inputBinding:
    prefix: -window-size
- id: in_bar_seq_catalog_out
  doc: "# if multiple replicons/genomes/contigs/sequences were provided in -ref, give\
    \ them names.\n# Enter 'auto' for autogenerated ids.\n"
  type: File?
  inputBinding:
    prefix: -barseq_catalog_out
- id: in_python
  doc: ''
  type: string
  inputBinding:
    position: 0
- id: in_path
  doc: ''
  type: File
  inputBinding:
    position: 1
outputs:
- id: out_stdout
  doc: Standard output stream
  type: stdout
hints: []
cwlVersion: v1.1
baseCommand:
- tpp
