from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Int, String, Boolean, Float

Rpkmforgenes_Py_V0_1_0 = CommandToolBuilder(tool="rpkmforgenes.py", base_command=["rpkmforgenes.py"], inputs=[ToolInput(tag="in_by_output_file", input_type=File(optional=True), prefix="-o", doc=InputDocumentation(doc="by output file")), ToolInput(tag="in_list_input_files", input_type=File(optional=True), prefix="-i", doc=InputDocumentation(doc="by list of input files (by default, guesses format from file extension)")), ToolInput(tag="in_by_annotation_file", input_type=File(optional=True), prefix="-a", doc=InputDocumentation(doc="by annotation file")), ToolInput(tag="in_bigwig_file_directory", input_type=Int(optional=True), prefix="-u", doc=InputDocumentation(doc="by a bigWig file, alternatively a directory for files for non-unique positions (lower case for nonunique k-mers (where k is the read length), upper case for unique; filenames are e.g. chr1.fa, can also be chr1_unique20-255.btxt etc")), ToolInput(tag="in_no_three_utr", input_type=Int(optional=True), prefix="-no3utr", doc=InputDocumentation(doc="remove 3'UTRs")), ToolInput(tag="in_full_transcript", input_type=Int(optional=True), prefix="-fulltranscript", doc=InputDocumentation(doc="not remove 3'UTRs (default)")), ToolInput(tag="in_maxlength", input_type=Int(optional=True), prefix="-maxlength", doc=InputDocumentation(doc="by a distance to cut each transcript from the 3' end, from 5' if negative (never seems to give better values)")), ToolInput(tag="in_max_genes", input_type=String(optional=True), prefix="-maxgenes", doc=InputDocumentation(doc="how many genes expression is calculated for (for testing purposes)")), ToolInput(tag="in_name_collapse", input_type=String(optional=True), prefix="-namecollapse", doc=InputDocumentation(doc="only consider overlap between isoform with the same gene identifier (shaky)")), ToolInput(tag="in_no_collapse", input_type=String(optional=True), prefix="-nocollapse", doc=InputDocumentation(doc="get isoform expressions (shaky)")), ToolInput(tag="in_no_overlap", input_type=String(optional=True), prefix="-nooverlap", doc=InputDocumentation(doc="ignore that transcripts can overlap (will count some reads several times)")), ToolInput(tag="in_rm_name_overlap", input_type=String(optional=True), prefix="-rmnameoverlap", doc=InputDocumentation(doc="ignore regions shared my multiple genes (seems to work well)")), ToolInput(tag="in_rm_regions", input_type=File(optional=True), prefix="-rmregions", doc=InputDocumentation(doc="by a bed file of regions which should be removed from genes")), ToolInput(tag="in_flat", input_type=String(optional=True), prefix="-flat", doc=InputDocumentation(doc="flatten all isoforms to one gene model (likely to give too low RPKM values)")), ToolInput(tag="in_tx_unique", input_type=String(optional=True), prefix="-txunique", doc=InputDocumentation(doc="ignore regions shared by multiple gene isoforms")), ToolInput(tag="in_only_coding", input_type=String(optional=True), prefix="-onlycoding", doc=InputDocumentation(doc="ignore noncoding transcripts")), ToolInput(tag="in_swap_strands", input_type=String(optional=True), prefix="-swapstrands", doc=InputDocumentation(doc="make reads on + strand map to genes on - and vice versa (and sets -strand)")), ToolInput(tag="in_introns", input_type=String(optional=True), prefix="-introns", doc=InputDocumentation(doc="gene expression from introns rather than exons (also removes exons of other isoforms)")), ToolInput(tag="in_keep_hap", input_type=String(optional=True), prefix="-keephap", doc=InputDocumentation(doc="not remove haplotype chromosome (_hap/_alt) annotation")), ToolInput(tag="in_no_random", input_type=String(optional=True), prefix="-norandom", doc=InputDocumentation(doc="remove genes on unplaced contigs")), ToolInput(tag="in_gene_pred", input_type=File(optional=True), prefix="-genePred", doc=InputDocumentation(doc="annotation file uses format of refGene.txt etc (default if cannot guess from file name suffix)")), ToolInput(tag="in_bed_ann", input_type=File(optional=True), prefix="-bedann", doc=InputDocumentation(doc="0-based bed file, chromosome start end and 9 optional fields")), ToolInput(tag="in_en_sgt_fann", input_type=String(optional=True), prefix="-ensgtfann", doc=InputDocumentation(doc="gtf format gene annotation (slow, generally better to convert to bed or genepred)")), ToolInput(tag="in_add_chr", input_type=Int(optional=True), prefix="-addchr", doc=InputDocumentation(doc="change a read aligning to 1 to chr1, 2 to chr2 etc")), ToolInput(tag="in_bed", input_type=File(optional=True), prefix="-bed", doc=InputDocumentation(doc="separated bed file (default if cannot guess from file name suffix)")), ToolInput(tag="in_bed_compacted", input_type=Int(optional=True), prefix="-bedcompacted", doc=InputDocumentation(doc="file with number of reads in the score column")), ToolInput(tag="in_bed_space", input_type=File(optional=True), prefix="-bedspace", doc=InputDocumentation(doc="separated bed file")), ToolInput(tag="in_bowtie", input_type=String(optional=True), prefix="-bowtie", doc=InputDocumentation(doc="default output format of bowtie")), ToolInput(tag="in_sam_se", input_type=String(optional=True), prefix="-samse", doc=InputDocumentation(doc="format, uniquely mapped reads (faster than -sam, , default for SAM))")), ToolInput(tag="in_bam_u", input_type=String(optional=True), prefix="-bamu", doc=InputDocumentation(doc="or SAM format, uniquely mapped reads (faster than -bam or -samse, default for BAM)")), ToolInput(tag="in_gff", input_type=File(optional=True), prefix="-gff", doc=InputDocumentation(doc="file with reads, no groups")), ToolInput(tag="in_mrna_norm", input_type=Int(optional=True), prefix="-mRNAnorm", doc=InputDocumentation(doc="normalize by the number of reads matching mRNA exons (default)")), ToolInput(tag="in_exon_norm", input_type=Int(optional=True), prefix="-exonnorm", doc=InputDocumentation(doc="normalize by the number of reads matching exons, including ncRNA")), ToolInput(tag="in_all_map_norm", input_type=Int(optional=True), prefix="-allmapnorm", doc=InputDocumentation(doc="normalize by the total number of mapped reads (default if annotation contains no mRNA)")), ToolInput(tag="in_forced_total", input_type=Int(optional=True), prefix="-forcedtotal", doc=InputDocumentation(doc="by a number of reads for each sample to set a constant to normalise by")), ToolInput(tag="in_read_count", input_type=Int(optional=True), prefix="-readcount", doc=InputDocumentation(doc="add the number of reads to the output")), ToolInput(tag="in_table", input_type=String(optional=True), prefix="-table", doc=InputDocumentation(doc="output format")), ToolInput(tag="in_sort_pos", input_type=String(optional=True), prefix="-sortpos", doc=InputDocumentation(doc="output sorted by genome position")), ToolInput(tag="in_export_ann", input_type=File(optional=True), prefix="-exportann", doc=InputDocumentation(doc="by a filename to write which exons have been used, also prints exon read counts for the last input file")), ToolInput(tag="in_read_present", input_type=Boolean(optional=True), prefix="-readpresent", doc=InputDocumentation(doc="- to suppress zero count entries - Ramu")), ToolInput(tag="in_strand", input_type=String(optional=True), prefix="-strand", doc=InputDocumentation(doc="use strand information of reads")), ToolInput(tag="in_both_ends", input_type=Float(optional=True), prefix="-bothends", doc=InputDocumentation(doc="also map the end positions to genes, each end counted as 0.5 (or 0.25 for paired-end reads)")), ToolInput(tag="in_both_ends_ceil", input_type=String(optional=True), prefix="-bothendsceil", doc=InputDocumentation(doc="set -bothends but round the read count upward")), ToolInput(tag="in_mid_read", input_type=String(optional=True), prefix="-midread", doc=InputDocumentation(doc="use middle of the read as read position")), ToolInput(tag="in_diff_reads", input_type=Int(optional=True), prefix="-diffreads", doc=InputDocumentation(doc="count only one read if several have the same position, strand and length (use with -bam or -sam if paired-end; samtools rmdup is generally better)")), ToolInput(tag="in_max_reads", input_type=Int(optional=True), prefix="-maxreads", doc=InputDocumentation(doc="by maximum number of reads to be used")), ToolInput(tag="in_random_reads", input_type=String(optional=True), prefix="-randomreads", doc=InputDocumentation(doc="make -maxreads pick reads at random")), ToolInput(tag="in_min_qual", input_type=Int(optional=True), prefix="-minqual", doc=InputDocumentation(doc="by an integer, to restrict reads to minimum this mapping quality (for sam, bam) or score (for bed, gff), default use all")), ToolInput(tag="in_max_nm", input_type=Int(optional=True), prefix="-maxNM", doc=InputDocumentation(doc="by an integer, to restrict reads to maximum this edit distance (nM or NM flag in sam, bam), default use all")), ToolInput(tag="in_list_sample_names", input_type=File(optional=True), prefix="-n", doc=InputDocumentation(doc="by list of sample names (input file names are otherwise used)")), ToolInput(tag="in_number_files_process", input_type=Int(optional=True), prefix="-p", doc=InputDocumentation(doc="by number of files to process in parallel")), ToolInput(tag="in_quite", input_type=String(optional=True), prefix="-quite", doc=InputDocumentation(doc="skip progress messages and warnings")), ToolInput(tag="in_print_message_quit", input_type=String(optional=True), prefix="-h", doc=InputDocumentation(doc="print this message and quit"))], outputs=[ToolOutput(tag="out_by_output_file", output_type=File(optional=True), selector=InputSelector(input_to_select="in_by_output_file", type_hint=File()), doc=OutputDocumentation(doc="by output file"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Rpkmforgenes_Py_V0_1_0().translate("wdl", allow_empty_container=True)

