class: CommandLineTool
id: ggcaller.cwl
inputs:
- id: in_graph
  doc: Bifrost GFA file generated by Bifrost build.
  type: File?
  inputBinding:
    prefix: --graph
- id: in_colours
  doc: Bifrost colours file generated by Bifrost build.
  type: File?
  inputBinding:
    prefix: --colours
- id: in_not_ref
  doc: "If using existing graph, was not graph built\nexclusively with assembled genomes.\
    \ [Default = False]"
  type: boolean?
  inputBinding:
    prefix: --not-ref
- id: in_refs
  doc: List of reference genomes (one file path per line).
  type: File?
  inputBinding:
    prefix: --refs
- id: in_reads
  doc: List of reference genomes (one file path per line).
  type: File?
  inputBinding:
    prefix: --reads
- id: in_codons
  doc: JSON file containing start and stop codon sequences.
  type: File?
  inputBinding:
    prefix: --codons
- id: in_km_er
  doc: K-mer size used in Bifrost build (bp). [Default = 31]
  type: long?
  inputBinding:
    prefix: --kmer
- id: in_path
  doc: "Maximum path length during traversal (bp). [Default =\n10000]"
  type: File?
  inputBinding:
    prefix: --path
- id: in_orf
  doc: Minimum ORF length to return (bp). [Default = 90]
  type: long?
  inputBinding:
    prefix: --orf
- id: in_max_overlap
  doc: "Maximum overlap allowed between overlapping ORFs.\n[Default = 60]"
  type: long?
  inputBinding:
    prefix: --maxoverlap
- id: in_min_path_score
  doc: "Minimum total score for a path of ORFs to be returned.\n[Default = 100]"
  type: long?
  inputBinding:
    prefix: --min-path-score
- id: in_min_orf_score
  doc: "Minimum individual score for an ORF to be returned.\n[Default = 100]"
  type: long?
  inputBinding:
    prefix: --min-orf-score
- id: in_no_filter
  doc: "Do not filter ORF calls using Balrog. Will return all\nORF calls. [Default\
    \ = False]"
  type: boolean?
  inputBinding:
    prefix: --no-filter
- id: in_no_write_idx
  doc: Do not write FMIndexes to file. [Default = False]
  type: boolean?
  inputBinding:
    prefix: --no-write-idx
- id: in_no_write_graph
  doc: "Do not write Bifrost GFA and colours to file. [Default\n= False]"
  type: boolean?
  inputBinding:
    prefix: --no-write-graph
- id: in_repeat
  doc: "Enable traversal of nodes mulitple times. [Default =\nFalse]"
  type: boolean?
  inputBinding:
    prefix: --repeat
- id: in_threads
  doc: Number of threads for FMIndexing [Default = 1]
  type: long?
  inputBinding:
    prefix: --threads
- id: in_out
  doc: Output FASTA file containing ORF sequences.
  type: File?
  inputBinding:
    prefix: --out
outputs:
- id: out_stdout
  doc: Standard output stream
  type: stdout
- id: out_out
  doc: Output FASTA file containing ORF sequences.
  type: File?
  outputBinding:
    glob: $(inputs.in_out)
hints:
- class: DockerRequirement
  dockerPull: quay.io/biocontainers/ggcaller:1.2.1--py39h2d02072_0
cwlVersion: v1.1
baseCommand:
- ggcaller
