from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import String, File, Boolean, Int, Directory

Pathoscope_Lib_V0_1_0 = CommandToolBuilder(tool="pathoscope_LIB", base_command=["pathoscope", "LIB"], inputs=[ToolInput(tag="in_genome_file", input_type=String(optional=True), prefix="-genomeFile", doc=InputDocumentation(doc="Specify reference genome(Download\nftp://ftp.ncbi.nih.gov/blast/db/FASTA/nt.gz)")), ToolInput(tag="in_taxonid_s", input_type=File(optional=True), prefix="-taxonIds", doc=InputDocumentation(doc="Specify taxon ids of your interest with comma\nseparated (if you have multiple taxon ids). If you do\nnot specify this option, it will work on all entries\nin the reference file. For taxonomy id lookup, refer\nto http://www.ncbi.nlm.nih.gov/taxonomy")), ToolInput(tag="in_exclude_taxonid_s", input_type=String(optional=True), prefix="-excludeTaxonIds", doc=InputDocumentation(doc="Specify taxon ids to exclude with comma separated (if\nyou have multiple taxon ids to exclude).")), ToolInput(tag="in_nodes_c", input_type=Boolean(optional=True), prefix="--noDesc", doc=InputDocumentation(doc="Do not keep an additional description in original\nfasta seq header.Depending on NGS aligner, a long\nsequence header may slow down its mapping process.")), ToolInput(tag="in_sub_tax", input_type=Boolean(optional=True), prefix="--subTax", doc=InputDocumentation(doc="To include all sub taxonomies under the query taxonomy\nid. e.g., if you set -t 4751 --subtax, it will cover\nall sub taxonomies under taxon id 4751 (fungi).")), ToolInput(tag="in_online", input_type=Boolean(optional=True), prefix="--online", doc=InputDocumentation(doc="To enable online searching in case you cannot find a\ncorrect taxonomy id for a given gi. When there are\nmany entries in nt whose gi is invalid, this option\nmay slow down the overall process.")), ToolInput(tag="in_dbhost", input_type=String(optional=True), prefix="-dbhost", doc=InputDocumentation(doc="specify hostname running mysql if you want to use\nmysql instead of hash method in mapping gi to taxonomy\nid")), ToolInput(tag="in_db_port", input_type=Int(optional=True), prefix="-dbport", doc=InputDocumentation(doc="provide mysql server port if different from default\n(3306)")), ToolInput(tag="in_dbuser", input_type=String(optional=True), prefix="-dbuser", doc=InputDocumentation(doc="user name to access mysql")), ToolInput(tag="in_db_passwd", input_type=String(optional=True), prefix="-dbpasswd", doc=InputDocumentation(doc="provide password associate with user")), ToolInput(tag="in_db", input_type=String(optional=True), prefix="-db", doc=InputDocumentation(doc="mysql pathoscope database name (default: pathodb)")), ToolInput(tag="in_outdir", input_type=Directory(optional=True), prefix="-outDir", doc=InputDocumentation(doc="Output Directory (Default=. (current directory))")), ToolInput(tag="in_out_prefix", input_type=String(optional=True), prefix="-outPrefix", doc=InputDocumentation(doc="specify an output prefix to name your target database\n"))], outputs=[ToolOutput(tag="out_outdir", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_outdir", type_hint=File()), doc=OutputDocumentation(doc="Output Directory (Default=. (current directory))"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Pathoscope_Lib_V0_1_0().translate("wdl", allow_empty_container=True)

