from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Directory, Boolean, String, File, Int

Hichipper_V0_1_0 = CommandToolBuilder(tool="hichipper", base_command=["hichipper"], inputs=[ToolInput(tag="in_out", input_type=Directory(optional=True), prefix="--out", doc=InputDocumentation(doc="Output directory name; must not be already\nexisting [Required]  [required]")), ToolInput(tag="in_keep_temp_files", input_type=Boolean(optional=True), prefix="--keep-temp-files", doc=InputDocumentation(doc="Keep temporary files?")), ToolInput(tag="in_input_vi", input_type=String(optional=True), prefix="--input-vi", doc=InputDocumentation(doc="Comma-separted list of interactions files for\nloop calling; option valid only in `call`\nmode")), ToolInput(tag="in_restriction_frags", input_type=File(optional=True), prefix="--restriction-frags", doc=InputDocumentation(doc="Filepath to restriction fragment files; will\noverwrite specification of this file when a\n.yaml is supplied for mode")), ToolInput(tag="in_peaks", input_type=File(optional=True), prefix="--peaks", doc=InputDocumentation(doc="Either 1 of 4 peak logic strings or a valid\nfilepath to a .bed (or similary formatted)\nfile; defers to what is in the .yaml")), ToolInput(tag="in_keep_samples", input_type=String(optional=True), prefix="--keep-samples", doc=InputDocumentation(doc="Comma separated list of sample names to keep;\nALL (special string) by default")), ToolInput(tag="in_ignore_samples", input_type=String(optional=True), prefix="--ignore-samples", doc=InputDocumentation(doc="Comma separated list of sample names to\nignore; NONE (special string) by default")), ToolInput(tag="in_read_length", input_type=Int(optional=True), prefix="--read-length", doc=InputDocumentation(doc="Length of reads from sequencing runs; default\n= 75")), ToolInput(tag="in_min_dist", input_type=Int(optional=True), prefix="--min-dist", doc=InputDocumentation(doc="Minimum distance for loop calls; default =")), ToolInput(tag="in_macs_two_genome", input_type=Int(optional=True), prefix="--macs2-genome", doc=InputDocumentation(doc="Argument to pass to the -g variable in MACS2\n(mm for mouse genome; hs for human genome);\ndefault = 'hs'")), ToolInput(tag="in_peak_pad", input_type=Int(optional=True), prefix="--peak-pad", doc=InputDocumentation(doc="Peak padding width (applied on both left and\nright); default = 500")), ToolInput(tag="in_merge_gap", input_type=Int(optional=True), prefix="--merge-gap", doc=InputDocumentation(doc="Merge nearby peaks (after all padding is\ncomplete; default = 500")), ToolInput(tag="in_no_merge", input_type=Boolean(optional=True), prefix="--no-merge", doc=InputDocumentation(doc="Completely skip anchor merging; will affect\nsummary statistics. Not recommended unless\nunderstood what is happening.")), ToolInput(tag="in_skip_res_frag_pad", input_type=Boolean(optional=True), prefix="--skip-resfrag-pad", doc=InputDocumentation(doc="Skip restriction fragment aware padding")), ToolInput(tag="in_skip_background_correction", input_type=Boolean(optional=True), prefix="--skip-background-correction", doc=InputDocumentation(doc="Skip restriction fragment aware background\ncorrection?")), ToolInput(tag="in_make_ucsc", input_type=Boolean(optional=True), prefix="--make-ucsc", doc=InputDocumentation(doc="Make additional output files that can support\nviewing in UCSC genome browser; requires\ntabix and bgzip; does the same thing as\n--make-washu.")), ToolInput(tag="in_make_washu", input_type=Boolean(optional=True), prefix="--make-washu", doc=InputDocumentation(doc="Make additional output files that can support\nviewing in WashU genome browser; requires\ntabix and bgzip; does the same thing as\n--make-ucsc.")), ToolInput(tag="in_basic_qc", input_type=Boolean(optional=True), prefix="--basic-qc", doc=InputDocumentation(doc="Create a simple QC report without Pandoc")), ToolInput(tag="in_skip_diff_loop", input_type=Boolean(optional=True), prefix="--skip-diffloop", doc=InputDocumentation(doc="Skip analyses in diffloop (e.g. Mango loop\ncalling; .rds generation)")), ToolInput(tag="in_bed_tools_path", input_type=File(optional=True), prefix="--bedtools-path", doc=InputDocumentation(doc="Path to bedtools; by default, assumes that\nbedtools is in PATH")), ToolInput(tag="in_macs_two_path", input_type=File(optional=True), prefix="--macs2-path", doc=InputDocumentation(doc="Path to macs2; by default, assumes that macs2\nis in PATH")), ToolInput(tag="in_tab_ix_path", input_type=File(optional=True), prefix="--tabix-path", doc=InputDocumentation(doc="Path to samtools; by default, assumes that\ntabix is in PATH")), ToolInput(tag="in_b_gzip_path", input_type=File(optional=True), prefix="--bgzip-path", doc=InputDocumentation(doc="Path to macs2; by default, assumes that bgzip\nis in PATH")), ToolInput(tag="in_r_path", input_type=File(optional=True), prefix="--r-path", doc=InputDocumentation(doc="Path to R; by default, assumes that R is in")), ToolInput(tag="in_ma__maxdist", input_type=Int(), position=0, doc=InputDocumentation(doc="-ma, --max-dist TEXT           Maximum distance for loop calls; default =")), ToolInput(tag="in__macsstring_text", input_type=Int(), position=1, doc=InputDocumentation(doc="--macs2-string TEXT            String of arguments to pass to MACS2; only is")), ToolInput(tag="in_path", input_type=File(), position=0, doc=InputDocumentation(doc="--help                         Show this message and exit."))], outputs=[ToolOutput(tag="out_out", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_out", type_hint=File()), doc=OutputDocumentation(doc="Output directory name; must not be already\nexisting [Required]  [required]"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Hichipper_V0_1_0().translate("wdl", allow_empty_container=True)

