from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Int, Directory, Boolean, String

Probabilistic2020_Oncogene_V0_1_0 = CommandToolBuilder(tool="probabilistic2020_oncogene", base_command=["probabilistic2020", "oncogene"], inputs=[ToolInput(tag="in_input", input_type=File(optional=True), prefix="--input", doc=InputDocumentation(doc="gene FASTA file from extract_gene_seq.py script")), ToolInput(tag="in_mutations", input_type=File(optional=True), prefix="--mutations", doc=InputDocumentation(doc="DNA mutations file (MAF file). Columns can be in any\norder, but should contain the correct column header\nnames.")), ToolInput(tag="in_bed", input_type=File(optional=True), prefix="--bed", doc=InputDocumentation(doc="BED file annotation of genes")), ToolInput(tag="in_processes", input_type=Int(optional=True), prefix="--processes", doc=InputDocumentation(doc="Number of processes to use for parallelization. 0\nindicates using a single process without using a\nmultiprocessing pool (more means Faster, default: 0).")), ToolInput(tag="in_num_iterations", input_type=Int(optional=True), prefix="--num-iterations", doc=InputDocumentation(doc="Number of iterations for null model. p-value precision\nincreases with more iterations, however this will also\nincrease the run time (Default: 100,000).")), ToolInput(tag="in_context", input_type=Int(optional=True), prefix="--context", doc=InputDocumentation(doc="Number of DNA bases to use as context. 0 indicates no\ncontext. 1 indicates only use the mutated base. 1.5\nindicates using the base context used in CHASM (http:/\n/wiki.chasmsoftware.org/index.php/CHASM_Overview). 2\nindicates using the mutated base and the upstream\nbase. 3 indicates using the mutated base and both the\nupstream and downstream bases. (Default: 1.5)")), ToolInput(tag="in_score_dir", input_type=Directory(optional=True), prefix="--score-dir", doc=InputDocumentation(doc="Directory containing VEST score information in pickle\nfiles (Default: None).")), ToolInput(tag="in_output_text_file", input_type=File(optional=True), prefix="--output", doc=InputDocumentation(doc="Output text file of probabilistic 20/20 results")), ToolInput(tag="in_stop_criteria", input_type=Int(optional=True), prefix="--stop-criteria", doc=InputDocumentation(doc="Number of iterations more significant then the\nobserved statistic to stop further computations. This\ndecreases compute time spent in resolving p-values for\nnon-significant genes. (Default: 1000).")), ToolInput(tag="in_recurrent", input_type=Int(optional=True), prefix="--recurrent", doc=InputDocumentation(doc="Minimum number of mutations at a position for it to be\nconsidered a recurrently mutated position (Default:\n3).")), ToolInput(tag="in_fraction", input_type=Int(optional=True), prefix="--fraction", doc=InputDocumentation(doc="Fraction of total mutations in a gene. This define the\nminimumm number of mutations for a position to be\ndefined as recurrently mutated (Defaul: .02).")), ToolInput(tag="in_unique", input_type=Boolean(optional=True), prefix="--unique", doc=InputDocumentation(doc="Only keep unique mutations for each tumor sample.\nMutations reported from heterogeneous sources may\ncontain duplicates, e.g. a tumor sample was sequenced\ntwice.")), ToolInput(tag="in_use_unmapped", input_type=Boolean(optional=True), prefix="--use-unmapped", doc=InputDocumentation(doc="Use mutations that are not mapped to the the single\nreference transcript for a gene specified in the bed\nfile indicated by the -b option.")), ToolInput(tag="in_genome", input_type=File(optional=True), prefix="--genome", doc=InputDocumentation(doc="Path to the genome fasta file. Required if --use-\nunmapped flag is used. (Default: None)")), ToolInput(tag="in_seed", input_type=Int(optional=True), prefix="--seed", doc=InputDocumentation(doc="Specify the seed for the pseudo random number\ngenerator. By default, the seed is randomly chosen.\nThe seed will be used for the monte carlo simulations\n(Default: 101).\n")), ToolInput(tag="in_find_evaluates_clustering", input_type=String(), position=0, doc=InputDocumentation(doc="Find statsitically significant oncogene-like genes. Evaluates clustering of"))], outputs=[ToolOutput(tag="out_output_text_file", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output_text_file", type_hint=File()), doc=OutputDocumentation(doc="Output text file of probabilistic 20/20 results"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Probabilistic2020_Oncogene_V0_1_0().translate("wdl", allow_empty_container=True)

