from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Boolean, String

Chexmix_V0_1_0 = CommandToolBuilder(tool="chexmix", base_command=["chexmix"], inputs=[ToolInput(tag="in_threads", input_type=File(optional=True), prefix="--threads", doc=InputDocumentation(doc="<number of threads to use (default=1)>")), ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="--verbose", doc=InputDocumentation(doc="[flag to print intermediate files and extra output]")), ToolInput(tag="in_config", input_type=Boolean(optional=True), prefix="--config", doc=InputDocumentation(doc="<config file: all options here can be specified in a name<space>value text file, over-ridden by command-line args>")), ToolInput(tag="in_geninfo", input_type=File(optional=True), prefix="--geninfo", doc=InputDocumentation(doc="AND --seq <fasta seq directory reqd if finding motif>")), ToolInput(tag="in_expt", input_type=File(optional=True), prefix="--expt", doc=InputDocumentation(doc="AND --format <SAM/BED/IDX>")), ToolInput(tag="in_ctrl", input_type=Boolean(optional=True), prefix="--ctrl", doc=InputDocumentation(doc="<file name (optional argument. must be same format as expt files)>")), ToolInput(tag="in_design", input_type=Boolean(optional=True), prefix="--design", doc=InputDocumentation(doc="<experiment design file name to use instead of --expt and --ctrl; see website for format>")), ToolInput(tag="in_fixed_pb", input_type=Boolean(optional=True), prefix="--fixedpb", doc=InputDocumentation(doc="<fixed per base limit (default: estimated from background model)>")), ToolInput(tag="in_poisson_gauss_pb", input_type=String(optional=True), prefix="--poissongausspb", doc=InputDocumentation(doc="[flag to use non-unique reads]")), ToolInput(tag="in_mapp_ability", input_type=Boolean(optional=True), prefix="--mappability", doc=InputDocumentation(doc="<fraction of the genome that is mappable for these experiments (default=0.8)>")), ToolInput(tag="in_nocache", input_type=Boolean(optional=True), prefix="--nocache", doc=InputDocumentation(doc="[flag to turn off caching of the entire set of experiments (i.e. run slower with less memory)]")), ToolInput(tag="in_no_scaling", input_type=Boolean(optional=True), prefix="--noscaling", doc=InputDocumentation(doc="[flag to turn off auto estimation of signal vs control scaling factor]")), ToolInput(tag="in_median_scale", input_type=Boolean(optional=True), prefix="--medianscale", doc=InputDocumentation(doc="[flag to use scaling by median ratio (default = scaling by NCIS)]")), ToolInput(tag="in_regression_scale", input_type=Boolean(optional=True), prefix="--regressionscale", doc=InputDocumentation(doc="[flag to use scaling by regression (default = scaling by NCIS)]")), ToolInput(tag="in_ses_scale", input_type=Boolean(optional=True), prefix="--sesscale", doc=InputDocumentation(doc="[flag to use scaling by SES (default = scaling by NCIS)]")), ToolInput(tag="in_fixed_scaling", input_type=Boolean(optional=True), prefix="--fixedscaling", doc=InputDocumentation(doc="<multiply control counts by total tag count ratio and then by this factor (default: NCIS)>")), ToolInput(tag="in_scale_win", input_type=Boolean(optional=True), prefix="--scalewin", doc=InputDocumentation(doc="<window size for scaling procedure (default=10000)>")), ToolInput(tag="in_plot_scaling", input_type=Boolean(optional=True), prefix="--plotscaling", doc=InputDocumentation(doc="[flag to plot diagnostic information for the chosen scaling method]")), ToolInput(tag="in_round", input_type=Boolean(optional=True), prefix="--round", doc=InputDocumentation(doc="<max. model update rounds (default=3)>")), ToolInput(tag="in_bm_window_max", input_type=Boolean(optional=True), prefix="--bmwindowmax", doc=InputDocumentation(doc="<max. window size for running a mixture model over binding events (default=2000)>")), ToolInput(tag="in_no_model_update", input_type=Boolean(optional=True), prefix="--nomodelupdate", doc=InputDocumentation(doc="[flag to turn off binding model updates]")), ToolInput(tag="in_min_model_update_events", input_type=Boolean(optional=True), prefix="--minmodelupdateevents", doc=InputDocumentation(doc="<minimum number of events to support an update using read distributions (default=100)>")), ToolInput(tag="in_pr_log_conf", input_type=Boolean(optional=True), prefix="--prlogconf", doc=InputDocumentation(doc="<Poisson log threshold for potential region scanning (default=-6)>")), ToolInput(tag="in_fixed_alpha", input_type=Boolean(optional=True), prefix="--fixedalpha", doc=InputDocumentation(doc="<binding events must have at least this number of reads (default: set automatically)>")), ToolInput(tag="in_alpha_scale", input_type=Boolean(optional=True), prefix="--alphascale", doc=InputDocumentation(doc="<alpha scaling factor; increase for stricter event calls (default=1.0)>")), ToolInput(tag="in_beta_scale", input_type=Boolean(optional=True), prefix="--betascale", doc=InputDocumentation(doc="<beta scaling factor; prior on subtype assignment (default=0.05)>")), ToolInput(tag="in_epsilon_scale", input_type=Boolean(optional=True), prefix="--epsilonscale", doc=InputDocumentation(doc="<epsilon scaling factor; increase for more weight on motif in subtype assignment (default=0.2)>")), ToolInput(tag="in_min_subtype_frac", input_type=Boolean(optional=True), prefix="--minsubtypefrac", doc=InputDocumentation(doc="<subtypes must have at least this percentage of associated binding events; increase for fewer subtypes (default=0.05)>")), ToolInput(tag="in_exclude", input_type=File(optional=True), prefix="--exclude", doc=InputDocumentation(doc="OR --excludebed <file of regions to ignore in bed format>")), ToolInput(tag="in_galaxy_html", input_type=Boolean(optional=True), prefix="--galaxyhtml", doc=InputDocumentation(doc="[flag to produce a html output appropreate for galaxy]")), ToolInput(tag="in_standard", input_type=Boolean(optional=True), prefix="--standard", doc=InputDocumentation(doc="[report events that pass significance threshold in condition as a whole (default mode)]")), ToolInput(tag="in_lenient", input_type=Boolean(optional=True), prefix="--lenient", doc=InputDocumentation(doc="[report events that pass significance in >=1 replicate *or* the condition as a whole.]")), ToolInput(tag="in_lenient_plus", input_type=Boolean(optional=True), prefix="--lenientplus", doc=InputDocumentation(doc="[report events that pass significance in condition OR (>=1 replicate AND no signif diff between replicates)]")), ToolInput(tag="in_meme_path", input_type=File(optional=True), prefix="--memepath", doc=InputDocumentation(doc="<path to the meme bin dir (default: meme is in $PATH)>")), ToolInput(tag="in_no_motifs", input_type=Boolean(optional=True), prefix="--nomotifs", doc=InputDocumentation(doc="[flag to turn off motif-finding & motif priors]")), ToolInput(tag="in_no_motif_prior", input_type=Boolean(optional=True), prefix="--nomotifprior", doc=InputDocumentation(doc="[flag to turn off motif priors only]")), ToolInput(tag="in_me_men_motifs", input_type=Boolean(optional=True), prefix="--memenmotifs", doc=InputDocumentation(doc="<number of motifs MEME should find for each condition (default=3)>")), ToolInput(tag="in_meme_min_w", input_type=Boolean(optional=True), prefix="--mememinw", doc=InputDocumentation(doc="<minw arg for MEME (default=6)>")), ToolInput(tag="in_meme_max_w", input_type=Boolean(optional=True), prefix="--mememaxw", doc=InputDocumentation(doc="<maxw arg for MEME (default=18)>")), ToolInput(tag="in_meme_args", input_type=Boolean(optional=True), prefix="--memeargs", doc=InputDocumentation(doc="<additional args for MEME (default=  -dna -mod zoops -revcomp -nostatus)>")), ToolInput(tag="in_min_roc", input_type=Boolean(optional=True), prefix="--minroc", doc=InputDocumentation(doc="<minimum motif ROC value (default=0.7)>")), ToolInput(tag="in_min_model_update_refs", input_type=Boolean(optional=True), prefix="--minmodelupdaterefs", doc=InputDocumentation(doc="<minimum number of motif reference to support an subtype distribution update (default=50)>")), ToolInput(tag="in_seqr_mth_res", input_type=Boolean(optional=True), prefix="--seqrmthres", doc=InputDocumentation(doc="<Filter out sequences with motifs below this threshold for recursively finding motifs (default=0.1)>")), ToolInput(tag="in_motif_pcc_th_res", input_type=Boolean(optional=True), prefix="--motifpccthres", doc=InputDocumentation(doc="<motif length adjusted similarity threshold for merging subtypes using motifs; decrease for fewer subtypes (default=0.95)>")), ToolInput(tag="in_no_clustering", input_type=Boolean(optional=True), prefix="--noclustering", doc=InputDocumentation(doc="[flag to turn off read distribution clustering]")), ToolInput(tag="in_pref", input_type=Boolean(optional=True), prefix="--pref", doc=InputDocumentation(doc="<preference value for read distribution clustering (default=-0.1)>")), ToolInput(tag="in_num_comps", input_type=Boolean(optional=True), prefix="--numcomps", doc=InputDocumentation(doc="<number of components to cluster (default=500)>")), ToolInput(tag="in_win", input_type=Boolean(optional=True), prefix="--win", doc=InputDocumentation(doc="<window size of read profiles (default=150)>")), ToolInput(tag="in_kl_divergence_th_res", input_type=Boolean(optional=True), prefix="--kldivergencethres", doc=InputDocumentation(doc="<KL divergence dissimilarity threshold for merging subtypes using read distributions; increase for fewer subtypes (default=-10)>")), ToolInput(tag="in_qvalue_minimum_default", input_type=Boolean(optional=True), prefix="--q", doc=InputDocumentation(doc="<Q-value minimum (default=0.01)>")), ToolInput(tag="in_min_fold", input_type=Boolean(optional=True), prefix="--minfold", doc=InputDocumentation(doc="<minimum event fold-change vs scaled control (default=1.5)>"))], outputs=[ToolOutput(tag="out_threads", output_type=File(optional=True), selector=InputSelector(input_to_select="in_threads", type_hint=File()), doc=OutputDocumentation(doc="<number of threads to use (default=1)>"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Chexmix_V0_1_0().translate("wdl", allow_empty_container=True)

