from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, Directory, String, File

Rgt_Tdf_Promotertest_V0_1_0 = CommandToolBuilder(tool="rgt_TDF_promotertest", base_command=["rgt-TDF", "promotertest"], inputs=[ToolInput(tag="in_input_file_name", input_type=Boolean(optional=True), prefix="-r", doc=InputDocumentation(doc="Input file name for RNA sequence (in fasta format)")), ToolInput(tag="in_rn", input_type=Boolean(optional=True), prefix="-rn", doc=InputDocumentation(doc="Define the RNA name")), ToolInput(tag="in_de", input_type=Boolean(optional=True), prefix="-de", doc=InputDocumentation(doc="Input file for target gene list (gene symbols or Ensembl\nID)")), ToolInput(tag="in_bed", input_type=Boolean(optional=True), prefix="-bed", doc=InputDocumentation(doc="Input BED file of the promoter regions of target genes")), ToolInput(tag="in_bg", input_type=Boolean(optional=True), prefix="-bg", doc=InputDocumentation(doc="Input BED file of the promoter regions of background")), ToolInput(tag="in_define_title_name", input_type=Boolean(optional=True), prefix="-t", doc=InputDocumentation(doc="Define the title name for the results under the Output\nname. (default is RNA name)")), ToolInput(tag="in_organism", input_type=Boolean(optional=True), prefix="-organism", doc=InputDocumentation(doc="Define the organism")), ToolInput(tag="in_gtf", input_type=Boolean(optional=True), prefix="-gtf", doc=InputDocumentation(doc="Define the GTF file for annotation (optional)")), ToolInput(tag="in_tss", input_type=Boolean(optional=True), prefix="-tss", doc=InputDocumentation(doc="Define the distance between the promoter regions and TSS\nalong gene body (default: 0)")), ToolInput(tag="in_pl", input_type=Boolean(optional=True), prefix="-pl", doc=InputDocumentation(doc="Define the promotor length (default: 1000)")), ToolInput(tag="in_show_dbs", input_type=Boolean(optional=True), prefix="-showdbs", doc=InputDocumentation(doc="Show the plots and statistics of DBS (DNA Binding sites)")), ToolInput(tag="in_score", input_type=Boolean(optional=True), prefix="-score", doc=InputDocumentation(doc="Load score column from input gene list or BED file for")), ToolInput(tag="in_ccf", input_type=Boolean(optional=True), prefix="-ccf", doc=InputDocumentation(doc="Define the cut off of triplex-forming promoters per DBD\nin percentage of total triplexes (0-1) or an integer.\n(default: 0.1)")), ToolInput(tag="in_rt", input_type=Boolean(optional=True), prefix="-rt", doc=InputDocumentation(doc="Remove temporary files (fa, txp...etc)")), ToolInput(tag="in_log", input_type=Boolean(optional=True), prefix="-log", doc=InputDocumentation(doc="Set the plots in log scale")), ToolInput(tag="in_obed", input_type=Boolean(optional=True), prefix="-obed", doc=InputDocumentation(doc="Output the BED files for DNA binding sites.")), ToolInput(tag="in_show_pa", input_type=Boolean(optional=True), prefix="-showpa", doc=InputDocumentation(doc="Show parallel and antiparallel bindings in the plot")), ToolInput(tag="in_protein_coding", input_type=Boolean(optional=True), prefix="-protein_coding", doc=InputDocumentation(doc="Apply filtering to get only protein coding genes.")), ToolInput(tag="in_known_only", input_type=Boolean(optional=True), prefix="-known_only", doc=InputDocumentation(doc="Apply filtering to get only known genes.")), ToolInput(tag="in_dump", input_type=Boolean(optional=True), prefix="-dump", doc=InputDocumentation(doc="Only dump the preprocessed file and leave the program.")), ToolInput(tag="in_rna_exp", input_type=Boolean(optional=True), prefix="-rnaexp", doc=InputDocumentation(doc="Given a file with RNA name and the expression value")), ToolInput(tag="in_no_file", input_type=Directory(optional=True), prefix="-nofile", doc=InputDocumentation(doc="Don't save any files in the output folder, except the")), ToolInput(tag="in_set_maximal_errorrate", input_type=Boolean(optional=True), prefix="-e", doc=InputDocumentation(doc="[Triplexes] Set the maximal error-rate in % tolerated\n(default: 20)")), ToolInput(tag="in_sets_tolerated_number", input_type=Boolean(optional=True), prefix="-c", doc=InputDocumentation(doc="[Triplexes] Sets the tolerated number of consecutive\nerrors with respect to the canonical triplex rules as\nsuch were found to greatly destabilize triplexes in\nvitro (default: 2)")), ToolInput(tag="in_fr", input_type=Boolean(optional=True), prefix="-fr", doc=InputDocumentation(doc="[Triplexes] Activates the filtering of low complexity\nregions and repeats in the sequence data (default: off)")), ToolInput(tag="in_fm", input_type=Boolean(optional=True), prefix="-fm", doc=InputDocumentation(doc="[Triplexes] Method to quickly discard non-hits (default:\n0).'0' = greedy approach; '1' = q-gram filtering.")), ToolInput(tag="in_of", input_type=Boolean(optional=True), prefix="-of", doc=InputDocumentation(doc="[Triplexes] Define output formats of Triplexes (default:\n1)")), ToolInput(tag="in_mf", input_type=Boolean(optional=True), prefix="-mf", doc=InputDocumentation(doc="[Triplexes] Merge overlapping features into a cluster\nand report the spanning region.")), ToolInput(tag="in_rm", input_type=Boolean(optional=True), prefix="-rm", doc=InputDocumentation(doc="[Triplexes] Set the multiprocessing")), ToolInput(tag="in_par", input_type=Boolean(optional=True), prefix="-par", doc=InputDocumentation(doc="[Triplexes] Define other parameters for Triplexes")), ToolInput(tag="in_genes", input_type=String(), position=0, doc=InputDocumentation(doc="-o                  Output directory name for all the results")), ToolInput(tag="in_analysis_dot", input_type=String(), position=1, doc=InputDocumentation(doc="-scoreh             Use the header of scores from the given gene list or BED")), ToolInput(tag="in_file_dot", input_type=File(), position=2, doc=InputDocumentation(doc="-a                  Define significance level for rejection null hypothesis")), ToolInput(tag="in_separately_dot", input_type=String(), position=0, doc=InputDocumentation(doc="-filter_havana      Apply filtering to remove HAVANA entries.")), ToolInput(tag="in_statistics_dot", input_type=String(), position=1, doc=InputDocumentation(doc="-l                  [Triplexes] Define the minimum length of triplex"))], outputs=[ToolOutput(tag="out_no_file", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_no_file", type_hint=File()), doc=OutputDocumentation(doc="Don't save any files in the output folder, except the"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Rgt_Tdf_Promotertest_V0_1_0().translate("wdl", allow_empty_container=True)

