class: CommandLineTool
id: anvi_script_gen_distribution_of_genes_in_a_bin.cwl
inputs:
- id: in_contigs_db
  doc: "Anvi'o contigs database generated by 'anvi-gen-\ncontigs'"
  type: string?
  inputBinding:
    prefix: --contigs-db
- id: in_profile_db
  doc: Anvi'o profile database
  type: string?
  inputBinding:
    prefix: --profile-db
- id: in_collection_name
  doc: Collection name.
  type: string?
  inputBinding:
    prefix: --collection-name
- id: in_bin_id
  doc: Bin name you are interested in.
  type: string?
  inputBinding:
    prefix: --bin-id
- id: in_min_detection
  doc: "For this entire thing to work, the genome you are\nfocusing on should be detected\
    \ in at least one\nmetagenome. If that is not the case, it would mean\nthat you\
    \ do not have any sample that represents the\nniche for this organism (or you\
    \ do not have enough\ndepth of coverage) to investigate the detection of\ngenes\
    \ in the environment. By default, this script\nrequires at least '0.5' of the\
    \ genome to be detected\nin at least one metagenome. This parameter allows you\n\
    to change that. 0 would mean no detection test\nrequired, 1 would mean the entire\
    \ genome must be\ndetected."
  type: double?
  inputBinding:
    prefix: --min-detection
- id: in_fraction_of_median_coverage
  doc: "The value set here will be used to remove a gene if\nits total coverage across\
    \ environments is less than\nthe median coverage of all genes multiplied by this\n\
    value. The default is 0.25, which means, if the median\ntotal coverage of all\
    \ genes across all samples is\n100X, then, a gene with a total coverage of less\
    \ than\n25X across all samples will be assumed not a part of\nthe 'environmental\
    \ core'.\n"
  type: double?
  inputBinding:
    prefix: --fraction-of-median-coverage
outputs:
- id: out_stdout
  doc: Standard output stream
  type: stdout
hints: []
cwlVersion: v1.1
baseCommand:
- anvi-script-gen-distribution-of-genes-in-a-bin
