from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean

_Showfeat_V0_1_0 = CommandToolBuilder(tool="_showfeat", base_command=["_showfeat"], inputs=[ToolInput(tag="in_source_match", input_type=Boolean(optional=True), prefix="-sourcematch", doc=InputDocumentation(doc="string     [*] By default any feature source in the\nfeature table is shown. You can set this to\nmatch any feature source you wish to show.\nThe source name is usually either the name\nof the program that detected the feature or\nit is the feature table (eg: EMBL) that the\nfeature came from.\nThe source may be wildcarded by using '*'.\nIf you wish to show more than one source,\nseparate their names with the character '|',\neg:\ngene* | embl (Any string)")), ToolInput(tag="in_type_match", input_type=Boolean(optional=True), prefix="-typematch", doc=InputDocumentation(doc="string     [*] By default any feature type in the\nfeature table is shown. You can set this to\nmatch any feature type you wish to show.\nSee http://www.ebi.ac.uk/embl/WebFeat/ for a\nlist of the EMBL feature types and see\nAppendix A of the Swissprot user manual in\nhttp://www.expasy.org/sprot/userman.html for\na list of the Swissprot feature types.\nThe type may be wildcarded by using '*'.\nIf you wish to show more than one type,\nseparate their names with the character '|',\neg:\n*UTR | intron (Any string)")), ToolInput(tag="in_tag_match", input_type=Boolean(optional=True), prefix="-tagmatch", doc=InputDocumentation(doc="string     [*] Tags are the types of extra values that\na feature may have. For example in the EMBL\nfeature table, a 'CDS' type of feature may\nhave the tags '/codon', '/codon_start',\n'/db_xref', '/EC_number', '/evidence',\n'/exception', '/function', '/gene',\n'/label', '/map', '/note', '/number',\n'/partial', '/product', '/protein_id',\n'/pseudo', '/standard_name', '/translation',\n'/transl_except', '/transl_table', or\n'/usedin'. Some of these tags also have\nvalues, for example '/gene' can have the\nvalue of the gene name.\nBy default any feature tag in the feature\ntable is shown. You can set this to match\nany feature tag you wish to show.\nThe tag may be wildcarded by using '*'.\nIf you wish to show more than one tag,\nseparate their names with the character '|',\neg:\ngene | label (Any string)")), ToolInput(tag="in_value_match", input_type=Boolean(optional=True), prefix="-valuematch", doc=InputDocumentation(doc="string     [*] Tag values are the values associated\nwith a feature tag. Tags are the types of\nextra values that a feature may have. For\nexample in the EMBL feature table, a 'CDS'\ntype of feature may have the tags '/codon',\n'/codon_start', '/db_xref', '/EC_number',\n'/evidence', '/exception', '/function',\n'/gene', '/label', '/map', '/note',\n'/number', '/partial', '/product',\n'/protein_id', '/pseudo', '/standard_name',\n'/translation', '/transl_except',\n'/transl_table', or '/usedin'. Only some of\nthese tags can have values, for example\n'/gene' can have the value of the gene name.\nBy default any feature tag value in the\nfeature table is shown. You can set this to\nmatch any feature tag value you wish to\nshow.\nThe tag value may be wildcarded by using\n'*'.\nIf you wish to show more than one tag value,\nseparate their names with the character\n'|', eg:\npax* | 10 (Any string)")), ToolInput(tag="in_sort", input_type=Boolean(optional=True), prefix="-sort", doc=InputDocumentation(doc="menu       [start] Sort features by Type, Start or\nSource, Nosort (don't sort - use input\norder) or join coding regions together and\nleave other features in the input order\n(Values: source (Sort by Source); start\n(Sort by Start position); type (Sort by\nType); nosort (No sorting done))")), ToolInput(tag="in_join_features", input_type=Boolean(optional=True), prefix="-joinfeatures", doc=InputDocumentation(doc="boolean    [N] Join coding regions together")), ToolInput(tag="in_annotation", input_type=Boolean(optional=True), prefix="-annotation", doc=InputDocumentation(doc="range      [If this is left blank, then no annotation\nis added.] Regions to annotate by marking.\nIf this is left blank, then no annotation is\nadded.\nA set of regions is specified by a set of\npairs of positions followed by optional\ntext.\nThe positions are integers.\nThey are followed by any text (but not\ndigits when on the command-line).\nExamples of region specifications are:\n24-45 new domain 56-78 match to Mouse\n1-100 First part 120-156 oligo\nA file of ranges to annotate (one range per\nline) can be specified as '@filename'.")), ToolInput(tag="in_html", input_type=Boolean(optional=True), prefix="-html", doc=InputDocumentation(doc="boolean    [N] Use HTML formatting")), ToolInput(tag="in_width", input_type=Boolean(optional=True), prefix="-width", doc=InputDocumentation(doc="integer    [60] You can expand (or contract) the width\nof the ASCII-character graphics display of\nthe positions of the features using this\nvalue.\nFor example, a width of 80 characters would\ncover a standard page width and a width a 10\ncharacters would be nearly unreadable.\nIf the width is set to less than 4, the\ngraphics lines and the scale line will not\nbe displayed. (Integer 0 or more)")), ToolInput(tag="in_collapse", input_type=Boolean(optional=True), prefix="-collapse", doc=InputDocumentation(doc="boolean    [N] If this is set, then features from the\nsame source and of the same type and sense\nare all printed on the same line. For\ninstance if there are several features from\nthe EMBL feature table (ie. the same source)\nwhich are all of type 'exon' in the same\nsense, then they will all be displayed on\nthe same line. This makes it hard to\ndistinguish overlapping features.\nIf this is set to false then each feature is\ndisplayed on a separate line making it\neasier to distinguish where features start\nand end.")), ToolInput(tag="in_strand", input_type=Boolean(optional=True), prefix="-strand", doc=InputDocumentation(doc="boolean    [N] Set this if you wish to display the\nstrand of the features. Protein features are\nalways directionless (indicated by '0'),\nforward is indicated by '+' and reverse is\n'-'.")), ToolInput(tag="in_source_show", input_type=Boolean(optional=True), prefix="-sourceshow", doc=InputDocumentation(doc="boolean    [N] Set this if you wish to display the\norigin of the features.\nThe source name is usually either the name\nof the program that detected the feature or\nit is the name of the feature table (eg:\nEMBL) that the feature came from.")), ToolInput(tag="in_position", input_type=Boolean(optional=True), prefix="-position", doc=InputDocumentation(doc="boolean    [N] Set this if you wish to display the\nstart and end position of the features. If\nseveral features are being displayed on the\nsame line, then the start and end positions\nwill be joined by a comma, for example:\n'189-189,225-225'.")), ToolInput(tag="in_tags_how", input_type=Boolean(optional=True), prefix="-tagshow", doc=InputDocumentation(doc="boolean    [N] Set this to be false if you do not wish\nto display the tags and values of the\nfeatures.")), ToolInput(tag="in_strict_tags", input_type=Boolean(optional=True), prefix="-stricttags", doc=InputDocumentation(doc="boolean    [N] By default if any tag/value pair in a\nfeature matches the specified tag and value,\nthen all the tags/value pairs of that\nfeature will be displayed. If this is set to\nbe true, then only those tag/value pairs in\na feature that match the specified tag and\nvalue will be displayed."))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    _Showfeat_V0_1_0().translate("wdl", allow_empty_container=True)

