version 1.0

task Mimseq {
  input {
    Int? species
    File? trnas_trnas_genomic
    File? file__trnaout
    File? trnas__mitotrnas
    Boolean? pre_trnas
    Boolean? cluster
    Boolean? cluster_id
    Boolean? de_conv_cov_ratio
    Int? threads
    Boolean? post_trans_mod_off
    String? control_condition
    Boolean? cca_analysis
    Boolean? double_cca
    Boolean? local_modo_mics
    Int? max_mismatches
    Int? remap_mismatches
    Boolean? snp_tolerance
    String? name__experiment
    Directory? out_dir
    Boolean? keep_temp
    Int? min_cov
    Int? max_multi
    Boolean? remap
    Boolean? mis_inc_thresh
    String sample_data
  }
  command <<<
    mimseq \
      ~{sample_data} \
      ~{if defined(species) then ("--species " +  '"' + species + '"') else ""} \
      ~{if defined(trnas_trnas_genomic) then ("-t " +  '"' + trnas_trnas_genomic + '"') else ""} \
      ~{if defined(file__trnaout) then ("-o " +  '"' + file__trnaout + '"') else ""} \
      ~{if defined(trnas__mitotrnas) then ("-m " +  '"' + trnas__mitotrnas + '"') else ""} \
      ~{if (pre_trnas) then "--pretRNAs" else ""} \
      ~{if (cluster) then "--cluster" else ""} \
      ~{if (cluster_id) then "--cluster-id" else ""} \
      ~{if (de_conv_cov_ratio) then "--deconv-cov-ratio" else ""} \
      ~{if defined(threads) then ("--threads " +  '"' + threads + '"') else ""} \
      ~{if (post_trans_mod_off) then "--posttrans-mod-off" else ""} \
      ~{if defined(control_condition) then ("--control-condition " +  '"' + control_condition + '"') else ""} \
      ~{if (cca_analysis) then "--cca-analysis" else ""} \
      ~{if (double_cca) then "--double-cca" else ""} \
      ~{if (local_modo_mics) then "--local-modomics" else ""} \
      ~{if defined(max_mismatches) then ("--max-mismatches " +  '"' + max_mismatches + '"') else ""} \
      ~{if defined(remap_mismatches) then ("--remap-mismatches " +  '"' + remap_mismatches + '"') else ""} \
      ~{if (snp_tolerance) then "--snp-tolerance" else ""} \
      ~{if defined(name__experiment) then ("-n " +  '"' + name__experiment + '"') else ""} \
      ~{if defined(out_dir) then ("--out-dir " +  '"' + out_dir + '"') else ""} \
      ~{if (keep_temp) then "--keep-temp" else ""} \
      ~{if defined(min_cov) then ("--min-cov " +  '"' + min_cov + '"') else ""} \
      ~{if defined(max_multi) then ("--max-multi " +  '"' + max_multi + '"') else ""} \
      ~{if (remap) then "--remap" else ""} \
      ~{if (mis_inc_thresh) then "--misinc-thresh" else ""}
  >>>
  runtime {
    docker: "quay.io/biocontainers/mimseq:0.3.3--py_0"
  }
  parameter_meta {
    species: "Species being analyzed for which to load pre-packaged\\ndata files (prioritized over -t, -o and -m). Options\\nare: Hsap, Hsap38, Mmus, Scer, Spom, Dmel, Drer, Ecol"
    trnas_trnas_genomic: "tRNAs, --trnas genomic tRNAs\\nGenomic tRNA fasta file, e.g. from gtRNAdb or\\ntRNAscan-SE. Already avalable in data folder for a few\\nmodel organisms."
    file__trnaout: "out file, --trnaout tRNA out file\\ntRNA.out file generated by tRNAscan-SE (also may be\\navailable on gtRNAdb). Contains information about tRNA\\nfeatures, including introns."
    trnas__mitotrnas: "tRNAs, --mito-trnas mitochondrial tRNAs\\nMitochondrial tRNA fasta file. Should be downloaded\\nfrom mitotRNAdb for species of interest. Already\\navailable in data folder for a few model organisms."
    pre_trnas: "Input reference sequences are pretRNAs. Enabling this\\noption will disable the removal of intron sequences\\nand addition of 3'-CCA to generate mature tRNA\\nsequences. Useful for mapping and discovering pretRNA\\nsequence reads."
    cluster: "Enable usearch sequence clustering of tRNAs by\\nisodecoder - drastically reduces rate of multi-mapping\\nreads."
    cluster_id: "[clustering identity threshold]\\nIdentity cutoff for usearch clustering between 0 and\\n1. Default is 0.97."
    de_conv_cov_ratio: "[deconvolution coverage threshold]\\nThreshold for ratio between coverage at 3' end and\\nmismatch used for deconvolution. Coverage reductions\\ngreater than the threshold will result in non-\\ndeconvoluted sequences. Default is 0.5 (i.e. less than\\n50% reduction required for deconvolution)."
    threads: "number\\nSet processor threads to use during read alignment and\\nread counting."
    post_trans_mod_off: "Disable post-transcriptional modification of tRNAs,\\ni.e. addition of 3'-CCA and 5'-G (His) to mature\\nsequences. Disable for certain prokaryotes (e.g. E.\\ncoli) where this is genomically encoded. Leave enabled\\n(default) for all eukaryotes."
    control_condition: "condition\\nName of control/wild-type condition as per user\\ndefined group specified in sample data input. This\\nmust exactly match the group name specified in sample\\ndata. This is used for differential expression\\nanalysis so that results are always in the form\\nmutant/treatment vs WT/control. REQUIRED"
    cca_analysis: "Enable analysis of 3'-CCA ends: Calculates proportions\\nof CC vs CCA ending reads per cluster and performs\\nDESeq2 analysis. Useful for comparing functional to\\nnon-functional mature tRNAs."
    double_cca: "Enable analysis of 3'-CCACCA tagging for tRNA\\ndegradation pathway. Note that this will alter the\\noutput of the CCA analysis pipeline."
    local_modo_mics: "Disable retrieval of Modomics data from online.\\nInstead use older locally stored data. Warning - this\\nleads to usage of older Modomics data!"
    max_mismatches: "mismatches\\nMaximum mismatches allowed. If specified between 0.0\\nand 1.0, then treated as a fraction of read length.\\nOtherwise, treated as integer number of mismatches.\\nDefault is an automatic ultrafast value calculated by\\nGSNAP; see GSNAP help for more info."
    remap_mismatches: "mismatches for remap\\nMaximum number of mismatches allowed during remapping\\nof all reads. Treated similarly to --max-mismatches.\\nThis is important to control misalignment of reads to\\nsimilar clusters/tRNAs Note that the SNP index will be\\nupdated with new SNPs from the first round of\\nalignment and so this should be relatively small to\\nprohibit misalignment."
    snp_tolerance: "Enable GSNAP SNP-tolerant read alignment, where known\\nmodifications from Modomics are mapped as SNPs."
    name__experiment: "name, --name experiment name\\nName of experiment. Note, output files and indices\\nwill have this as a prefix. REQUIRED"
    out_dir: "directory\\nOutput directory. Default is current directory. Cannot\\nbe an existing directory."
    keep_temp: "Keeps multi-mapping and unmapped bam files from GSNAP\\nalignments. Default is false."
    min_cov: "coverage per cluster\\nMinimum coverage per cluster required to include this\\ncluster in coverage plots, modification analysis, and\\n3'-CCA analysis. Can be a fraction of total mapped\\nreads between 0 and 1, or an integer of absolute\\ncoverage. Any cluster not meeting the threshold in 1\\nor more sample will be excluded. Note that all\\nclusters are included for differential expression\\nanalysis with DESeq2."
    max_multi: "coverage multithreading\\nMaximum number of bam files to run bedtools coverage\\non simultaneously. Increasing this number reduces\\nprocessing time by increasing number of files\\nprocessed simultaneously. However, depending on the\\nsize of the bam files to process and available memory,\\ntoo many files processed at once can cause termination\\nof mim-tRNAseq due to insufficient memory. If mim-\\ntRNAseq fails during coverage calculation, lower this\\nnumber. Increase at your own discretion. Default is 3."
    remap: "Enable detection of unannotated (potential)\\nmodifications from misincorporation data. These are\\ndefined as having a total misincorporation rate higher\\nthan the threshold set with --misinc-thresh. These\\nmodifications are then appended to already known ones,\\nand read alignment is reperformed. Very useful for\\npoorly annotated species in Modomics. Due to\\nrealignment and misincorporation parsing, enabling\\nthis option slows the analysis down considerably."
    mis_inc_thresh: "[threshold for unannotated mods]\\nThreshold of total misincorporation rate at a position\\nin a cluster used to call unannotated modifications.\\nValue between 0 and 1, default is 0.1 (10%\\nmisincorporation).\\n"
    sample_data: "Sample data sheet in text format, tab-separated.\\nColumn 1: full path to fastq (or fastq.gz). Column 2:\\ncondition/group."
  }
  output {
    File out_stdout = stdout()
    File out_file__trnaout = "${in_file__trnaout}"
    Directory out_out_dir = "${in_out_dir}"
  }
}