from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Int, Boolean, File, Directory

Snpmutator_V0_1_0 = CommandToolBuilder(tool="snpmutator", base_command=["snpmutator"], inputs=[ToolInput(tag="in_num_simulations", input_type=Int(optional=True), prefix="--num-simulations", doc=InputDocumentation(doc="Number of mutated sequences to generate. (default:\n100)")), ToolInput(tag="in_num_substitutions", input_type=Int(optional=True), prefix="--num-substitutions", doc=InputDocumentation(doc="Number of substitutions. (default: 500)")), ToolInput(tag="in_num_insertions", input_type=Int(optional=True), prefix="--num-insertions", doc=InputDocumentation(doc="Number of insertions. (default: 20)")), ToolInput(tag="in_num_deletions", input_type=Int(optional=True), prefix="--num-deletions", doc=InputDocumentation(doc="Number of deletions. (default: 20)")), ToolInput(tag="in_random_seed", input_type=Int(optional=True), prefix="--random-seed", doc=InputDocumentation(doc="Random number seed; if not set, the results are not\nreproducible. (default: None)")), ToolInput(tag="in_pool", input_type=Int(optional=True), prefix="--pool", doc=InputDocumentation(doc="Choose variants from a pool of eligible positions of\nthe specified size (default: 0)")), ToolInput(tag="in_group", input_type=Int(optional=True), prefix="--group", doc=InputDocumentation(doc="Group size. When greater than zero, this parameter\nchooses a new pool of positions for each group of\nreplicates. (default: None)")), ToolInput(tag="in_mono", input_type=Boolean(optional=True), prefix="--mono", doc=InputDocumentation(doc="Create monomorphic alleles (default: False)")), ToolInput(tag="in_seq_id", input_type=File(optional=True), prefix="--seqid", doc=InputDocumentation(doc="Output fasta description line sequence ID. Each\nmutated output file has only one sequence. If not\nspecified, the defline id will be the id of the first\nsequence in the input fasta file. The defline is\nalways suffixed with an annotation in this format:\n(mutated s=900 i=50 d=50). The seq id is also written\nto the CHROM column of the output VCF file. (default:\nNone)")), ToolInput(tag="in_ref", input_type=File(optional=True), prefix="--ref", doc=InputDocumentation(doc="Output concatenanted reference file with no mutations,\nbut all sequences concatenanted together. All the\nreplicates will be mutations of this file. (default:\nNone)")), ToolInput(tag="in_fast_a_dir", input_type=Directory(optional=True), prefix="--fasta-dir", doc=InputDocumentation(doc="Output directory for generated fasta files. (default:\n.)")), ToolInput(tag="in_summary", input_type=File(optional=True), prefix="--summary", doc=InputDocumentation(doc="Output positional summary file. (default: None)")), ToolInput(tag="in_vcf", input_type=File(optional=True), prefix="--vcf", doc=InputDocumentation(doc="Output VCF file. (default: None)")), ToolInput(tag="in_metrics", input_type=File(optional=True), prefix="--metrics", doc=InputDocumentation(doc="Output metrics file. (default: None)"))], outputs=[ToolOutput(tag="out_ref", output_type=File(optional=True), selector=InputSelector(input_to_select="in_ref", type_hint=File()), doc=OutputDocumentation(doc="Output concatenanted reference file with no mutations,\nbut all sequences concatenanted together. All the\nreplicates will be mutations of this file. (default:\nNone)")), ToolOutput(tag="out_fast_a_dir", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_fast_a_dir", type_hint=File()), doc=OutputDocumentation(doc="Output directory for generated fasta files. (default:\n.)")), ToolOutput(tag="out_summary", output_type=File(optional=True), selector=InputSelector(input_to_select="in_summary", type_hint=File()), doc=OutputDocumentation(doc="Output positional summary file. (default: None)")), ToolOutput(tag="out_vcf", output_type=File(optional=True), selector=InputSelector(input_to_select="in_vcf", type_hint=File()), doc=OutputDocumentation(doc="Output VCF file. (default: None)")), ToolOutput(tag="out_metrics", output_type=File(optional=True), selector=InputSelector(input_to_select="in_metrics", type_hint=File()), doc=OutputDocumentation(doc="Output metrics file. (default: None)"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Snpmutator_V0_1_0().translate("wdl", allow_empty_container=True)

