from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Int, Boolean, String, File

Mirdeep2_Pl_V0_1_0 = CommandToolBuilder(tool="miRDeep2.pl", base_command=["miRDeep2.pl"], inputs=[ToolInput(tag="in_minimum_read_stack", input_type=Int(optional=True), prefix="-a", doc=InputDocumentation(doc="minimum read stack height that triggers analysis. Using this option disables\nautomatic estimation of the optimal value and all detected precursors are analyzed")), ToolInput(tag="in_maximum_number_precursors", input_type=Int(optional=True), prefix="-g", doc=InputDocumentation(doc="maximum number of precursors to analyze when automatic excision gearing is used.\ndefault=50.000, if set to -1 all precursors will be analyzed")), ToolInput(tag="in_minimum_score_cutoff", input_type=Int(optional=True), prefix="-b", doc=InputDocumentation(doc="minimum score cut-off for predicted novel miRNAs to be displayed in the overview\ntable. This score cut-off is by default 0.")), ToolInput(tag="in_disable_randfold_analysis", input_type=Boolean(optional=True), prefix="-c", doc=InputDocumentation(doc="disable randfold analysis")), ToolInput(tag="in_disable_pdf_generation", input_type=Boolean(optional=True), prefix="-d", doc=InputDocumentation(doc="disable pdf generation")), ToolInput(tag="in_species_analyzed_used", input_type=String(optional=True), prefix="-t", doc=InputDocumentation(doc="species being analyzed - this is used to link to the appropriate UCSC browser entry")), ToolInput(tag="in_output_list_exit", input_type=Boolean(optional=True), prefix="-u", doc=InputDocumentation(doc="output list of UCSC browser species that are supported and exit")), ToolInput(tag="in_remove_directory_temporary", input_type=Boolean(optional=True), prefix="-v", doc=InputDocumentation(doc="remove directory with temporary files")), ToolInput(tag="in_sort_aligned_reads", input_type=Boolean(optional=True), prefix="-o", doc=InputDocumentation(doc="do not sort aligned reads in pdf files by sample, only used if multiple samples were used as input (see Readme for mor information)")), ToolInput(tag="in_file_known_mirbase", input_type=File(optional=True), prefix="-s", doc=InputDocumentation(doc="File with known miRBase star sequences")), ToolInput(tag="in_additional_tag_appended", input_type=String(optional=True), prefix="-z", doc=InputDocumentation(doc="Additional tag appended to current time stamp")), ToolInput(tag="in_use_switch_contain", input_type=Boolean(optional=True), prefix="-P", doc=InputDocumentation(doc="use this switch if mature_ref_miRNAs contain miRBase v18 identifiers (5p and 3p) instead of previous ids from v17")), ToolInput(tag="in_analyzed_dot", input_type=String(), position=0, doc=InputDocumentation(doc="precursors    miRBase miRNA precursor sequences in fasta format. These should be the known precursor"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Mirdeep2_Pl_V0_1_0().translate("wdl", allow_empty_container=True)

