from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, Int, File, String, Float

Mason_Variator_V0_1_0 = CommandToolBuilder(tool="mason_variator", base_command=["mason_variator"], inputs=[ToolInput(tag="in_version_check", input_type=Boolean(optional=True), prefix="--version-check", doc=InputDocumentation(doc="Turn this option off to disable version update notifications of the\napplication. One of 1, ON, TRUE, T, YES, 0, OFF, FALSE, F, and NO.\nDefault: 1.")), ToolInput(tag="in_quiet", input_type=Boolean(optional=True), prefix="--quiet", doc=InputDocumentation(doc="Set verbosity to a minimum.")), ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="--verbose", doc=InputDocumentation(doc="Enable verbose output.")), ToolInput(tag="in_very_verbose", input_type=Boolean(optional=True), prefix="--very-verbose", doc=InputDocumentation(doc="Enable very verbose output.")), ToolInput(tag="in_seed", input_type=Int(optional=True), prefix="--seed", doc=InputDocumentation(doc="The seed to use for the random number generator. Default: 0.")), ToolInput(tag="in_in_reference", input_type=File(optional=True), prefix="--in-reference", doc=InputDocumentation(doc="FASTA file with reference. Valid filetypes are: .fasta and .fa.")), ToolInput(tag="in_in_variant_tsv", input_type=File(optional=True), prefix="--in-variant-tsv", doc=InputDocumentation(doc="first two columns of this TSV file are")), ToolInput(tag="in_out_vcf", input_type=File(optional=True), prefix="--out-vcf", doc=InputDocumentation(doc="VCF file to write simulated variations to. Valid filetype is: .vcf.")), ToolInput(tag="in_out_fast_a", input_type=File(optional=True), prefix="--out-fasta", doc=InputDocumentation(doc="FASTA file to write simulated haplotypes to. Valid filetypes are:\n.fasta and .fa.")), ToolInput(tag="in_out_breakpoints", input_type=File(optional=True), prefix="--out-breakpoints", doc=InputDocumentation(doc="TSV file to write breakpoints in variants to. Valid filetypes are:\n.txt and .tsv.")), ToolInput(tag="in_haplotype_name_sep", input_type=String(optional=True), prefix="--haplotype-name-sep", doc=InputDocumentation(doc="Haplotype name separator in output FASTA. Default: /.")), ToolInput(tag="in_no_gen_var_ids", input_type=Boolean(optional=True), prefix="--no-gen-var-ids", doc=InputDocumentation(doc="Do not generate variant ids.")), ToolInput(tag="in_num_haplotypes", input_type=Int(optional=True), prefix="--num-haplotypes", doc=InputDocumentation(doc="The number of haplotypes to simulate. In range [1..inf]. Default: 1.")), ToolInput(tag="in_haplotype_sep", input_type=File(optional=True), prefix="--haplotype-sep", doc=InputDocumentation(doc="The separator between the chromosome and the haplotype name in the\noutput FASTA file. Default: /.")), ToolInput(tag="in_snp_rate", input_type=Float(optional=True), prefix="--snp-rate", doc=InputDocumentation(doc="Per-base SNP rate. In range [0.0..1.0]. Default: 0.0001.")), ToolInput(tag="in_small_in_del_rate", input_type=Float(optional=True), prefix="--small-indel-rate", doc=InputDocumentation(doc="Small indel rate. In range [0.0..1.0]. Default: 0.000001.")), ToolInput(tag="in_min_small_in_del_size", input_type=Int(optional=True), prefix="--min-small-indel-size", doc=InputDocumentation(doc="Minimal small indel size to simulate. In range [0..inf]. Default: 1.")), ToolInput(tag="in_max_small_in_del_size", input_type=Int(optional=True), prefix="--max-small-indel-size", doc=InputDocumentation(doc="Maximal small indel size to simulate. In range [0..inf]. Default: 6.")), ToolInput(tag="in_sv_in_del_rate", input_type=Float(optional=True), prefix="--sv-indel-rate", doc=InputDocumentation(doc="Per-base SNP rate. In range [0.0..1.0]. Default: 0.0000001.")), ToolInput(tag="in_sv_inversion_rate", input_type=Float(optional=True), prefix="--sv-inversion-rate", doc=InputDocumentation(doc="Per-base SNP rate. In range [0.0..1.0]. Default: 0.0000001.")), ToolInput(tag="in_sv_translocation_rate", input_type=Float(optional=True), prefix="--sv-translocation-rate", doc=InputDocumentation(doc="Per-base SNP rate. In range [0.0..1.0]. Default: 0.0000001.")), ToolInput(tag="in_sv_duplication_rate", input_type=Float(optional=True), prefix="--sv-duplication-rate", doc=InputDocumentation(doc="Per-base SNP rate. In range [0.0..1.0]. Default: 0.0000001.")), ToolInput(tag="in_min_sv_size", input_type=Int(optional=True), prefix="--min-sv-size", doc=InputDocumentation(doc="Minimal SV size to simulate. In range [0..inf]. Default: 50.")), ToolInput(tag="in_max_sv_size", input_type=Int(optional=True), prefix="--max-sv-size", doc=InputDocumentation(doc="Maximal SV size to simulate. In range [0..inf]. Default: 1000.")), ToolInput(tag="in_methylation_levels", input_type=Boolean(optional=True), prefix="--methylation-levels", doc=InputDocumentation(doc="Enable methylation level simulation.")), ToolInput(tag="in_meth_cg_mu", input_type=Float(optional=True), prefix="--meth-cg-mu", doc=InputDocumentation(doc="Median of beta distribution for methylation level of CpG loci. In\nrange [0..1]. Default: 0.6.")), ToolInput(tag="in_meth_cg_sigma", input_type=Float(optional=True), prefix="--meth-cg-sigma", doc=InputDocumentation(doc="Standard deviation of beta distribution for methylation level of CpG\nloci. In range [0..1]. Default: 0.03.")), ToolInput(tag="in_meth_chg_mu", input_type=Float(optional=True), prefix="--meth-chg-mu", doc=InputDocumentation(doc="Median of beta distribution for methylation level of CHG loci. In\nrange [0..1]. Default: 0.08.")), ToolInput(tag="in_meth_chg_sigma", input_type=Float(optional=True), prefix="--meth-chg-sigma", doc=InputDocumentation(doc="Standard deviation of beta distribution for methylation level of CHG\nloci. In range [0..1]. Default: 0.008.")), ToolInput(tag="in_meth_chh_mu", input_type=Float(optional=True), prefix="--meth-chh-mu", doc=InputDocumentation(doc="Median of beta distribution for methylation level of CHH loci. In\nrange [0..1]. Default: 0.05.")), ToolInput(tag="in_meth_chh_sigma", input_type=Float(optional=True), prefix="--meth-chh-sigma", doc=InputDocumentation(doc="Standard deviation of beta distribution for methylation level of CHH\nloci. In range [0..1]. Default: 0.005.")), ToolInput(tag="in_meth_fast_a_in", input_type=File(optional=True), prefix="--meth-fasta-in", doc=InputDocumentation(doc="Path to load original methylation levels from. Methylation levels\nare simulated if omitted. Valid filetypes are: .sam[.*], .raw[.*],\n.gbk[.*], .frn[.*], .fq[.*], .fna[.*], .ffn[.*], .fastq[.*],\n.fasta[.*], .faa[.*], .fa[.*], .embl[.*], and .bam, where * is any\nof the following extensions: gz and bgzf for transparent\n(de)compression.")), ToolInput(tag="in_meth_fast_a_out", input_type=File(optional=True), prefix="--meth-fasta-out", doc=InputDocumentation(doc="Path to write methylation levels to as FASTA. Only written if\n-of/--out-fasta is given. Valid filetypes are: .sam[.*], .raw[.*],\n.frn[.*], .fq[.*], .fna[.*], .ffn[.*], .fastq[.*], .fasta[.*],\n.faa[.*], .fa[.*], and .bam, where * is any of the following\nextensions: gz and bgzf for transparent (de)compression.")), ToolInput(tag="in_ins", input_type=String(), position=0, doc=InputDocumentation(doc="An insertion.")), ToolInput(tag="in_del", input_type=String(), position=1, doc=InputDocumentation(doc="A deletion.")), ToolInput(tag="in_in_v", input_type=String(), position=2, doc=InputDocumentation(doc="An inversion.")), ToolInput(tag="in_ctr", input_type=String(), position=3, doc=InputDocumentation(doc="An intra-chromosomal translocation.")), ToolInput(tag="in_dup", input_type=String(), position=4, doc=InputDocumentation(doc="A duplication"))], outputs=[ToolOutput(tag="out_out_breakpoints", output_type=File(optional=True), selector=InputSelector(input_to_select="in_out_breakpoints", type_hint=File()), doc=OutputDocumentation(doc="TSV file to write breakpoints in variants to. Valid filetypes are:\n.txt and .tsv.")), ToolOutput(tag="out_haplotype_sep", output_type=File(optional=True), selector=InputSelector(input_to_select="in_haplotype_sep", type_hint=File()), doc=OutputDocumentation(doc="The separator between the chromosome and the haplotype name in the\noutput FASTA file. Default: /.")), ToolOutput(tag="out_meth_fast_a_out", output_type=File(optional=True), selector=InputSelector(input_to_select="in_meth_fast_a_out", type_hint=File()), doc=OutputDocumentation(doc="Path to write methylation levels to as FASTA. Only written if\n-of/--out-fasta is given. Valid filetypes are: .sam[.*], .raw[.*],\n.frn[.*], .fq[.*], .fna[.*], .ffn[.*], .fastq[.*], .fasta[.*],\n.faa[.*], .fa[.*], and .bam, where * is any of the following\nextensions: gz and bgzf for transparent (de)compression."))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Mason_Variator_V0_1_0().translate("wdl", allow_empty_container=True)

