from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, String, Float

Gfacs_Pl_V0_1_0 = CommandToolBuilder(tool="gFACs.pl", base_command=["gFACs.pl"], inputs=[ToolInput(tag="in_specifying_format_step", input_type=Boolean(optional=True), prefix="-f", doc=InputDocumentation(doc="[format]     Specifying a format: A mandatory step to call upon the right script.")), ToolInput(tag="in_p", input_type=Boolean(optional=True), prefix="-p", doc=InputDocumentation(doc="[prefix]")), ToolInput(tag="in_rem_five_prime_three_prime_incompletes", input_type=Boolean(optional=True), prefix="--rem-5prime-3prime-incompletes", doc=InputDocumentation(doc="[nt number]")), ToolInput(tag="in_min_intron_size", input_type=Boolean(optional=True), prefix="--min-intron-size", doc=InputDocumentation(doc="[nt number]")), ToolInput(tag="in_min_cds_size", input_type=Boolean(optional=True), prefix="--min-CDS-size", doc=InputDocumentation(doc="[nt number]")), ToolInput(tag="in_en_tap_annotation", input_type=Boolean(optional=True), prefix="--entap-annotation", doc=InputDocumentation(doc="[/path/to/your/entap/annotation.tsv]")), ToolInput(tag="in_fast_a", input_type=Boolean(optional=True), prefix="--fasta", doc=InputDocumentation(doc="[/path/to/your/nucleotide/fasta.fasta]")), ToolInput(tag="in_rem_genes_without_start_and_stop_codon", input_type=Boolean(optional=True), prefix="--rem-genes-without-start-and-stop-codon", doc=InputDocumentation(doc="[number]")), ToolInput(tag="in_distributions", input_type=Boolean(optional=True), prefix="--distributions", doc=InputDocumentation(doc="[option] [option] etc...")), ToolInput(tag="in_g_facs_log_dot_txt", input_type=String(), position=0, doc=InputDocumentation(doc="gene_table.txt")), ToolInput(tag="in_braker_two_dot_zero_five_gtf", input_type=Float(), position=0, doc=InputDocumentation(doc="braker_2.05_gff")), ToolInput(tag="in_braker_two_dot_zero_five_gff_three", input_type=Float(), position=1, doc=InputDocumentation(doc="braker_2.0_gff3")), ToolInput(tag="in_braker_two_dot_zero_gff", input_type=Float(), position=2, doc=InputDocumentation(doc="braker_2.0_gtf")), ToolInput(tag="in_braker_two_dot_one_dot_two_gtf", input_type=Float(), position=3, doc=InputDocumentation(doc="maker_2.31.9_gff")), ToolInput(tag="in_pro_kk_a_one_dot_one_one_gff", input_type=Float(), position=4, doc=InputDocumentation(doc="gmap_2017_03_17_gff3")), ToolInput(tag="in_genome_threader_one_dot_six_dot_six_gff_three", input_type=Float(), position=5, doc=InputDocumentation(doc="stringtie_1.3.4_gtf")), ToolInput(tag="in_gff_read_zero_dot_nine_do_tone_two_gff_three", input_type=Float(), position=6, doc=InputDocumentation(doc="exonerate_2.4.0_gff")), ToolInput(tag="in_evm_one_dot_one_dot_one_gff_three", input_type=Float(), position=7, doc=InputDocumentation(doc="CoGe_1.0_gff")), ToolInput(tag="in_g_facs_gene_table", input_type=String(), position=8, doc=InputDocumentation(doc="gFACs_gtf")), ToolInput(tag="in_refseq_gff", input_type=String(), position=9, doc=InputDocumentation(doc="genbank_gbff - UNDER REPAIR")), ToolInput(tag="in_j_browse", input_type=String(), position=0, doc=InputDocumentation(doc="--compatibility [option] [option] etc... ")), ToolInput(tag="in_snp_eff", input_type=String(), position=1, doc=InputDocumentation(doc="EVM_1.1.1_gene_prediction       ")), ToolInput(tag="in_evm_one_dot_one_dot_one_alignment", input_type=Float(), position=2, doc=InputDocumentation(doc="Distribution table creation:")), ToolInput(tag="in_exon_lengths", input_type=String(), position=0, doc=InputDocumentation(doc="intron_lengths  ")), ToolInput(tag="in_cds_lengths", input_type=String(), position=1, doc=InputDocumentation(doc="gene_lengths    ")), ToolInput(tag="in_exon_position", input_type=String(), position=2, doc=InputDocumentation(doc="exon_position_data      "))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Gfacs_Pl_V0_1_0().translate("wdl", allow_empty_container=True)

