from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Boolean, Int, String, Float

Ngm_Log_V0_1_0 = CommandToolBuilder(tool="ngm_log", base_command=["ngm-log"], inputs=[ToolInput(tag="in_config", input_type=File(optional=True), prefix="--config", doc=InputDocumentation(doc="Path to the config file. The config file contains\nall advanced options. If this parameter is\nomitted, default values will be used.")), ToolInput(tag="in_reference", input_type=File(optional=True), prefix="--reference", doc=InputDocumentation(doc="Path to the reference genome\n(format: FASTA, can be gzipped).")), ToolInput(tag="in_qry", input_type=Boolean(optional=True), prefix="--qry", doc=InputDocumentation(doc="<path>              Path to the read file. If the file contains\ninterleaved mates use -p/--paired.")), ToolInput(tag="in_qry_one", input_type=File(optional=True), prefix="--qry1", doc=InputDocumentation(doc="Path to the read file containing mates 1.")), ToolInput(tag="in_qry_two", input_type=File(optional=True), prefix="--qry2", doc=InputDocumentation(doc="Path to the read file containing mates 2.\nValid input formats are: FASTA/Q (gzipped),\nSAM/BAM. If the query file(s) is/are omitted,\nNGM will only pre-process the reference.")), ToolInput(tag="in_paired", input_type=Boolean(optional=True), prefix="--paired", doc=InputDocumentation(doc="Input data is paired end.\nNOT required if -1/-2 are used. (default: off)")), ToolInput(tag="in_min_insert_size", input_type=Boolean(optional=True), prefix="--min-insert-size", doc=InputDocumentation(doc="The min insert size for paired end alignments\n(default: 0)")), ToolInput(tag="in_max_insert_size", input_type=Boolean(optional=True), prefix="--max-insert-size", doc=InputDocumentation(doc="The max insert size for paired end alignments\n(default: 1000)")), ToolInput(tag="in_max_read_length", input_type=Int(optional=True), prefix="--max-read-length", doc=InputDocumentation(doc="Length of longest read in input.\n(default: estimated from data)")), ToolInput(tag="in_force_r_length_check", input_type=Boolean(optional=True), prefix="--force-rlength-check", doc=InputDocumentation(doc="Forces NextgenMap to run through all reads to\nfind the max. read length. (default: off)")), ToolInput(tag="in_output", input_type=File(optional=True), prefix="--output", doc=InputDocumentation(doc="Path to output file.")), ToolInput(tag="in_top_n", input_type=Boolean(optional=True), prefix="--topn", doc=InputDocumentation(doc="Prints the <n> best alignments sorted by\nalignment score (default: 1)")), ToolInput(tag="in_strata", input_type=Boolean(optional=True), prefix="--strata", doc=InputDocumentation(doc="Only  output  the  highest  scoring  mappings\nfor any  given  read,  up  to <n> mappings per\nread. If a read has more than <n> mappings with\nthe same score, it is discarded and reported\nas unmapped.")), ToolInput(tag="in_bam", input_type=Boolean(optional=True), prefix="--bam", doc=InputDocumentation(doc="Output BAM instead of SAM.")), ToolInput(tag="in_keep_tags", input_type=Boolean(optional=True), prefix="--keep-tags", doc=InputDocumentation(doc="Copy BAM/SAM tags present in input file to\noutput file (default: off)")), ToolInput(tag="in_no_unal", input_type=File(optional=True), prefix="--no-unal", doc=InputDocumentation(doc="Don't print unaligned reads to output file.")), ToolInput(tag="in_hard_clip", input_type=Boolean(optional=True), prefix="--hard-clip", doc=InputDocumentation(doc="Hard instead of soft clipping in SAM output")), ToolInput(tag="in_silent_clip", input_type=Boolean(optional=True), prefix="--silent-clip", doc=InputDocumentation(doc="Hard clip reads but don't add clipping\ninformation to CIGAR string")), ToolInput(tag="in_rg_id", input_type=String(optional=True), prefix="--rg-id", doc=InputDocumentation(doc="Adds RG:Z:<string> to all alignments in SAM/BAM")), ToolInput(tag="in_rg_sm", input_type=String(optional=True), prefix="--rg-sm", doc=InputDocumentation(doc="RG header: Sample")), ToolInput(tag="in_rg_lb", input_type=String(optional=True), prefix="--rg-lb", doc=InputDocumentation(doc="RG header: Library")), ToolInput(tag="in_rg_pl", input_type=String(optional=True), prefix="--rg-pl", doc=InputDocumentation(doc="RG header: Platform")), ToolInput(tag="in_rg_ds", input_type=String(optional=True), prefix="--rg-ds", doc=InputDocumentation(doc="RG header: Description")), ToolInput(tag="in_rg_dt", input_type=String(optional=True), prefix="--rg-dt", doc=InputDocumentation(doc="RG header: Date (format: YYYY-MM-DD)")), ToolInput(tag="in_rg_pu", input_type=String(optional=True), prefix="--rg-pu", doc=InputDocumentation(doc="RG header: Platform unit")), ToolInput(tag="in_rg_pi", input_type=Int(optional=True), prefix="--rg-pi", doc=InputDocumentation(doc="RG header: Median insert size")), ToolInput(tag="in_rg_pg", input_type=String(optional=True), prefix="--rg-pg", doc=InputDocumentation(doc="RG header: Programs")), ToolInput(tag="in_rg_cn", input_type=String(optional=True), prefix="--rg-cn", doc=InputDocumentation(doc="RG header: sequencing center")), ToolInput(tag="in_rg_fo", input_type=String(optional=True), prefix="--rg-fo", doc=InputDocumentation(doc="RG header: Flow order")), ToolInput(tag="in_rg_ks", input_type=String(optional=True), prefix="--rg-ks", doc=InputDocumentation(doc="RG header: Key sequence")), ToolInput(tag="in_pe_delimiter", input_type=Int(optional=True), prefix="--pe-delimiter", doc=InputDocumentation(doc="Character used in suffix that identifies mate 1\nand 2. E.g. /1 and /2. This suffixes will be\nremoved to ensure proper SAM output\n(default: /)")), ToolInput(tag="in_threads", input_type=Int(optional=True), prefix="--threads", doc=InputDocumentation(doc="Number of candidate search threads")), ToolInput(tag="in_sensitivity", input_type=Float(optional=True), prefix="--sensitivity", doc=InputDocumentation(doc="A value between 0 and 1 that determines the\nnumber of candidate mapping regions that will\nbe evaluated with an sequence alignment.\n0: all CMR(s) will be evaluated\n1: only the best CMR(s) will be evaluated\nHigher values will reduce the runtime but also\nhave a negative effect on mapping sensitivity.\n(default: estimated from input data)")), ToolInput(tag="in_very_fast", input_type=Boolean(optional=True), prefix="--very-fast", doc=InputDocumentation(doc="Scale estimated sensitivity: Much faster, much less accurate")), ToolInput(tag="in_fast", input_type=Boolean(optional=True), prefix="--fast", doc=InputDocumentation(doc="Scale estimated sensitivity: Faster, less accurate")), ToolInput(tag="in_sensitive", input_type=Boolean(optional=True), prefix="--sensitive", doc=InputDocumentation(doc="Scale estimated sensitivity: More accurate, slower")), ToolInput(tag="in_very_sensitive", input_type=Boolean(optional=True), prefix="--very-sensitive", doc=InputDocumentation(doc="Scale estimated sensitivity: Much more accurate, much slower")), ToolInput(tag="in_min_identity", input_type=Int(optional=True), prefix="--min-identity", doc=InputDocumentation(doc="All reads mapped with an identity lower than\nthis threshold will be reported as unmapped\n(default: 0.65)")), ToolInput(tag="in_min_residues", input_type=Float(optional=True), prefix="--min-residues", doc=InputDocumentation(doc="All reads mapped with lower than\n<int> or <float> * read length residues\nwill be reported as unmapped. (default: 0.5)")), ToolInput(tag="in_min_mq", input_type=Int(optional=True), prefix="--min-mq", doc=InputDocumentation(doc="All reads mapped with lower than <int>\nmapping quality will be reported as unmapped.\n(default: 0)")), ToolInput(tag="in_gpu", input_type=Boolean(optional=True), prefix="--gpu", doc=InputDocumentation(doc="[int,...]            Use GPU(s) for alignment computation\nNOTE: GPU Ids are zero-based!\n-g or --gpu to use GPU\n-g 1 or --gpu 1 to use GPU 1\n-g 0,1 or --gpu 0,1 to use GPU 0 and 1\nIf -g/--gpu is omitted, alignments will be\ncomputed on the CPU (default)")), ToolInput(tag="in_bs_mapping", input_type=Boolean(optional=True), prefix="--bs-mapping", doc=InputDocumentation(doc="Enables bisulfite mapping.\nFor bs-mapping, kmer-skip will be applied to\nthe reads instead of the reference sequence.")), ToolInput(tag="in_bs_cut_off", input_type=Int(optional=True), prefix="--bs-cutoff", doc=InputDocumentation(doc="Max. number of Ts in a k-mer. All k-mers were\nthe number of Ts is higher than <int> are\nignored (default: 8)")), ToolInput(tag="in_km_er", input_type=Boolean(optional=True), prefix="--kmer", doc=InputDocumentation(doc="[10-14]             Kmer length in bases. (default: 13)")), ToolInput(tag="in_km_er_skip", input_type=Int(optional=True), prefix="--kmer-skip", doc=InputDocumentation(doc="Number of k-mers to skip when building the\nlookup table from the reference(default: 2)")), ToolInput(tag="in_km_er_min", input_type=Int(optional=True), prefix="--kmer-min", doc=InputDocumentation(doc="Minimal number of k-mer hits required to\nconsider a region a CMR. (default: 0)")), ToolInput(tag="in_max_cmrs", input_type=Int(optional=True), prefix="--max-cmrs", doc=InputDocumentation(doc="Reads that have more than <int> CMRs are\nignored. (default: infinite)")), ToolInput(tag="in_skip_save", input_type=Boolean(optional=True), prefix="--skip-save", doc=InputDocumentation(doc="Don't save index to disk. Saves disk space but\nincreases runtime for larger genomes.")), ToolInput(tag="in_local", input_type=Boolean(optional=True), prefix="--local", doc=InputDocumentation(doc="Perform local alignment. Ends might get clipped.\n(default: on)")), ToolInput(tag="in_end_to_end", input_type=Boolean(optional=True), prefix="--end-to-end", doc=InputDocumentation(doc="Perform end-to-end alignment. No clipping.\n(default: off)")), ToolInput(tag="in_affine", input_type=Boolean(optional=True), prefix="--affine", doc=InputDocumentation(doc="Use alignment algorithms that support affine gap\ncosts. This will give you better alignments for\nlonger indels but will also increase the runtime.\n(default: off)")), ToolInput(tag="in_max_consec_indels", input_type=Int(optional=True), prefix="--max-consec-indels", doc=InputDocumentation(doc="Maximum number of consecutive indels allowed.\n(default: computed based on avg. read length)")), ToolInput(tag="in_fast_pairing", input_type=Boolean(optional=True), prefix="--fast-pairing", doc=InputDocumentation(doc="Mates are mapped individually. If the best\nalignments for the mates give a proper pair\nthey are marked as paired in the output.\nIf not they are reported as broken pair.")), ToolInput(tag="in_pair_score_cut_off", input_type=Int(optional=True), prefix="--pair-score-cutoff", doc=InputDocumentation(doc="To find the best pairing all alignments of a\nread that have a score in the range of:\n[top score * pair-score-cutoff; top score]\nare taken into account.")), ToolInput(tag="in_skip_mate_check", input_type=Boolean(optional=True), prefix="--skip-mate-check", doc=InputDocumentation(doc="Don't check that both mates have the same name\n(default: off)")), ToolInput(tag="in_match_bonus", input_type=Int(optional=True), prefix="--match-bonus", doc=InputDocumentation(doc="Match score\n(default: 10, affine: 10, bs-mapping: 4)")), ToolInput(tag="in_mismatch", input_type=Int(optional=True), prefix="--mismatch", doc=InputDocumentation(doc="<int>      Mismatch penalty\n(default: 15, affine: 15, bs-mapping: 2)")), ToolInput(tag="in_gap_read_penalty", input_type=Int(optional=True), prefix="--gap-read-penalty", doc=InputDocumentation(doc="Penalty for a gap in the read\n(default: 20, affine: 33, bs-mapping: 10)")), ToolInput(tag="in_gap_ref_penalty", input_type=Int(optional=True), prefix="--gap-ref-penalty", doc=InputDocumentation(doc="Penalty for a gap in the reference\n(default: 20, affine: 33, bs-mapping: 10)")), ToolInput(tag="in_gap_extend_penalty", input_type=Int(optional=True), prefix="--gap-extend-penalty", doc=InputDocumentation(doc="Penalty for extending a gap\n(default: 20, affine: 3, bs-mapping: 10)")), ToolInput(tag="in_match_bonus_tt", input_type=Int(optional=True), prefix="--match-bonus-tt", doc=InputDocumentation(doc="Only for bs-mapping (default: 4)")), ToolInput(tag="in_match_bonus_tc", input_type=Int(optional=True), prefix="--match-bonus-tc", doc=InputDocumentation(doc="Only for bs-mapping (default: 4)")), ToolInput(tag="in_bin_size", input_type=Int(optional=True), prefix="--bin-size", doc=InputDocumentation(doc="Sets the size of the grid NextgenMap uses\nduring CMR search to: 2^n (default: 2)")), ToolInput(tag="in_update_check", input_type=Boolean(optional=True), prefix="--update-check", doc=InputDocumentation(doc="Perform an online check for a newer version of NGM")), ToolInput(tag="in_color", input_type=Boolean(optional=True), prefix="--color", doc=InputDocumentation(doc="Colored text output (default: off)")), ToolInput(tag="in_no_progress", input_type=Boolean(optional=True), prefix="--no-progress", doc=InputDocumentation(doc="Don't print progress info while mapping\n(default: off)\n")), ToolInput(tag="in_ngm", input_type=String(), position=0, doc=InputDocumentation(doc=""))], outputs=[ToolOutput(tag="out_output", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output", type_hint=File()), doc=OutputDocumentation(doc="Path to output file.")), ToolOutput(tag="out_no_unal", output_type=File(optional=True), selector=InputSelector(input_to_select="in_no_unal", type_hint=File()), doc=OutputDocumentation(doc="Don't print unaligned reads to output file."))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Ngm_Log_V0_1_0().translate("wdl", allow_empty_container=True)

