#!/usr/bin/env cwl-runner

baseCommand:
- fargene
class: CommandLineTool
cwlVersion: v1.0
id: fargene
inputs:
- doc: Input file(s) to be searched. Could either be in FASTA or FASTQ format.
  id: in_files
  inputBinding:
    prefix: --infiles
  type:
    items: string
    type: array
- doc: The Hidden Markov Model that should be used to analyse the data. Could either
    be one of the pre-defined models or the path to a custom HMM.
  id: hmm_model
  inputBinding:
    prefix: --hmm-model
  type: string
- doc: 'The threshold score for a sequence to be classified as a (almost) complete
    gene (default: None).'
  id: score
  inputBinding:
    prefix: --score
  type: string
- doc: 'If the input data is paired end metagenomic data (default: False).'
  id: meta
  inputBinding:
    prefix: --meta
  type: boolean
- doc: 'The threshold score for a fragment to be classified as a positive. Expressed
    as score per amino acid (default: None).'
  id: meta_score
  inputBinding:
    prefix: --meta-score
  type: string
- doc: 'The output directory for the whole run (default: ./fargene_output).'
  id: output
  inputBinding:
    prefix: --output
  type: string
- doc: 'Overwrite output directory if it exists (default: False).'
  id: force
  inputBinding:
    prefix: --force
  type: boolean
- doc: 'Directory for (sometimes large) intermediate files. (default: OUT_DIR/tmpdir)'
  id: tmp_dir
  inputBinding:
    prefix: --tmp-dir
  type: string
- doc: 'If the input sequence(s) is amino acids (default: False).'
  id: protein
  inputBinding:
    prefix: --protein
  type: boolean
- doc: 'Number of processes to be used when processing metagenomic data (default:
    1).'
  id: processes
  inputBinding:
    prefix: --processes
  type: string
- doc: 'The minimal length for a retrieved predicted ORF (nt). (default: 90% of the
    length of the chosen hmm.)'
  id: min_orf_length
  inputBinding:
    prefix: --min-orf-length
  type: long
- doc: 'Use this flag if the whole sequence where a hit is detected should be retrieved
    (default: False).'
  id: retrieve_whole
  inputBinding:
    prefix: --retrieve-whole
  type: boolean
- doc: Do not perform ORF prediction.
  id: no_orf_predict
  inputBinding:
    prefix: --no-orf-predict
  type: boolean
- doc: 'Use if no quality control should be performed on the metagenomic data (default:
    False).'
  id: no_quality_filtering
  inputBinding:
    prefix: --no-quality-filtering
  type: boolean
- doc: 'Use if you want to skip the assembly and retrieval of contigs for metagenomic
    data (default: False).'
  id: no_assembly
  inputBinding:
    prefix: --no-assembly
  type: boolean
- doc: 'Use NCBI ORFfinder instead of prodigal for ORF prediction of genomes/contigs
    (default: False).'
  id: orf_finder
  inputBinding:
    prefix: --orf-finder
  type: boolean
- doc: 'Store the translated sequences. Useful if you plan to redo the analysis using
    a different model and want to skip the preprocessing steps (default: False).'
  id: store_peptides
  inputBinding:
    prefix: --store-peptides
  type: boolean
- doc: 'Use of you want to redo the analysis or do the analysis using a different
    model and have kept either the nucletide or amino acid sequences. Please note
    that this only works if the input data is the same for both runs (default: False).'
  id: rerun
  inputBinding:
    prefix: --rerun
  type: boolean
- doc: Where the amino acid sequences generated by the method are located. Only to
    be used in combination with --rerun
  id: amino_dir
  inputBinding:
    prefix: --amino-dir
  type: string
- doc: Where the nucleotide sequences in FASTA generated by previous runs of the method
    are located. Only to be used in combination with --rerun
  id: fast_a_dir
  inputBinding:
    prefix: --fasta-dir
  type: string
- doc: 'The translation format that transeq should use. (default: pearson)'
  id: translation_format
  inputBinding:
    prefix: --translation-format
  type: string
- doc: 'Set logging level (default: INFO).'
  id: loglevel
  inputBinding:
    prefix: --loglevel
  type: string
- doc: 'Logfile (default: fargene_analysis.log).'
  id: log_file
  inputBinding:
    prefix: --logfile
  type: string
