from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, String, Int, File, Float, Directory

Mitofinder_V0_1_0 = CommandToolBuilder(tool="mitofinder", base_command=["mitofinder"], inputs=[ToolInput(tag="in_mega_hit", input_type=Boolean(optional=True), prefix="--megahit", doc=InputDocumentation(doc="Use Megahit for assembly. (Default)")), ToolInput(tag="in_i_dba", input_type=Boolean(optional=True), prefix="--idba", doc=InputDocumentation(doc="Use IDBA-UD for assembly.")), ToolInput(tag="in_meta_spades", input_type=Boolean(optional=True), prefix="--metaspades", doc=InputDocumentation(doc="Use MetaSPAdes for assembly.")), ToolInput(tag="in_trna_annotation", input_type=String(optional=True), prefix="--tRNA-annotation", doc=InputDocumentation(doc="'arwen'/'mitfi'/'trnascan' tRNA annotater to use.\nDefault = mitfi")), ToolInput(tag="in_seq_id", input_type=String(optional=True), prefix="--seqid", doc=InputDocumentation(doc="Sequence ID to be used throughout the process")), ToolInput(tag="in_paired_end_one", input_type=Int(optional=True), prefix="--Paired-end1", doc=InputDocumentation(doc="File with forward paired-end reads")), ToolInput(tag="in_paired_end_two", input_type=Int(optional=True), prefix="--Paired-end2", doc=InputDocumentation(doc="File with reverse paired-end reads")), ToolInput(tag="in_single_end", input_type=File(optional=True), prefix="--Single-end", doc=InputDocumentation(doc="File with single-end reads")), ToolInput(tag="in_assembly", input_type=File(optional=True), prefix="--assembly", doc=InputDocumentation(doc="File with your own assembly")), ToolInput(tag="in_max_memory", input_type=Int(optional=True), prefix="--max-memory", doc=InputDocumentation(doc="max memory to use in Go (MEGAHIT or MetaSPAdes)")), ToolInput(tag="in_length", input_type=Int(optional=True), prefix="--length", doc=InputDocumentation(doc="Shortest contig length to be used (MEGAHIT). Default =\n100")), ToolInput(tag="in_processors", input_type=Int(optional=True), prefix="--processors", doc=InputDocumentation(doc="Number of threads Mitofinder will use at most.")), ToolInput(tag="in_refseq", input_type=File(optional=True), prefix="--refseq", doc=InputDocumentation(doc="Reference mitochondrial genome in GenBank format\n(.gb).")), ToolInput(tag="in_blast_eval", input_type=Float(optional=True), prefix="--blast-eval", doc=InputDocumentation(doc="e-value of blast program used for contig\nidentification and annotation. Default = 0.00001")), ToolInput(tag="in_n_walk", input_type=Int(optional=True), prefix="--nwalk", doc=InputDocumentation(doc="Maximum number of codon steps to be tested on each\nsize of the gene to find the start and stop codon\nduring the annotation step. Default = 5 (30 bases)")), ToolInput(tag="in_override", input_type=Directory(optional=True), prefix="--override", doc=InputDocumentation(doc="This option forces MitoFinder to override the previous\noutput directory for the selected assembler.")), ToolInput(tag="in_adjust_direction", input_type=Boolean(optional=True), prefix="--adjust-direction", doc=InputDocumentation(doc="This option tells MitoFinder to adjust the direction\nof selected contig(s) (given the reference).")), ToolInput(tag="in_ignore", input_type=Boolean(optional=True), prefix="--ignore", doc=InputDocumentation(doc="This option tells MitoFinder to ignore the non-\nstandart mitochondrial genes.")), ToolInput(tag="in_new_genes", input_type=Boolean(optional=True), prefix="--new-genes", doc=InputDocumentation(doc="This option tells MitoFinder to try to annotate the\nnon-standard animal mitochondrial genes (e.g. rps3 in\nfungi). If several references are used, make sure the\nnon-standard genes have the same names in the several\nreferences")), ToolInput(tag="in_allow_intron", input_type=Boolean(optional=True), prefix="--allow-intron", doc=InputDocumentation(doc="This option tells MitoFinder to search for genes with\nintrons. Recommendation : Use it on mitochondrial\ncontigs previously found with MitoFinder without this\noption.")), ToolInput(tag="in_num_t", input_type=Boolean(optional=True), prefix="--numt", doc=InputDocumentation(doc="This option tells MitoFinder to search for both\nmitochondrial genes and NUMTs. Recommendation : Use it\non nuclear contigs previously found with MitoFinder\nwithout this option.")), ToolInput(tag="in_intron_size", input_type=Int(optional=True), prefix="--intron-size", doc=InputDocumentation(doc="Size of intron allowed. Default = 5000 bp")), ToolInput(tag="in_max_contig", input_type=Int(optional=True), prefix="--max-contig", doc=InputDocumentation(doc="Maximum number of contigs matching to the reference to\nkeep. Default = 0 (unlimited)")), ToolInput(tag="in_cds_merge", input_type=Boolean(optional=True), prefix="--cds-merge", doc=InputDocumentation(doc="This option tells MitoFinder to not merge the exons in\nthe NT and AA fasta files.")), ToolInput(tag="in_out_gb", input_type=File(optional=True), prefix="--out-gb", doc=InputDocumentation(doc="Do not create annotation output file in GenBank")), ToolInput(tag="in_blast_identity_nucl", input_type=Int(optional=True), prefix="--blast-identity-nucl", doc=InputDocumentation(doc="Nucleotide identity percentage for a hit to be\nretained. Default = 50")), ToolInput(tag="in_blast_identity_prot", input_type=Int(optional=True), prefix="--blast-identity-prot", doc=InputDocumentation(doc="Amino acid identity percentage for a hit to be\nretained. Default = 40")), ToolInput(tag="in_blast_size", input_type=Int(optional=True), prefix="--blast-size", doc=InputDocumentation(doc="Percentage of overlap in blast best hit to be\nretained. Default = 30")), ToolInput(tag="in_circular_size", input_type=Int(optional=True), prefix="--circular-size", doc=InputDocumentation(doc="Size to consider when checking for circularization.\nDefault = 45")), ToolInput(tag="in_circular_offset", input_type=Int(optional=True), prefix="--circular-offset", doc=InputDocumentation(doc="Offset from start and finish to consider when looking\nfor circularization. Default = 200")), ToolInput(tag="in_organism", input_type=Int(optional=True), prefix="--organism", doc=InputDocumentation(doc="Organism genetic code following NCBI table (integer):\n1. The Standard Code 2. The Vertebrate Mitochondrial\nCode 3. The Yeast Mitochondrial Code 4. The Mold,\nProtozoan, and Coelenterate Mitochondrial Code and the\nMycoplasma/Spiroplasma Code 5. The Invertebrate\nMitochondrial Code 6. The Ciliate, Dasycladacean and\nHexamita Nuclear Code 9. The Echinoderm and Flatworm\nMitochondrial Code 10. The Euplotid Nuclear Code 11.\nThe Bacterial, Archaeal and Plant Plastid Code 12. The\nAlternative Yeast Nuclear Code 13. The Ascidian\nMitochondrial Code 14. The Alternative Flatworm\nMitochondrial Code 16. Chlorophycean Mitochondrial\nCode 21. Trematode Mitochondrial Code 22. Scenedesmus\nobliquus Mitochondrial Code 23. Thraustochytrium\nMitochondrial Code 24. Pterobranchia Mitochondrial\nCode 25. Candidate Division SR1 and Gracilibacteria\nCode")), ToolInput(tag="in_example", input_type=Boolean(optional=True), prefix="--example", doc=InputDocumentation(doc="Print getting started examples")), ToolInput(tag="in_citation", input_type=Boolean(optional=True), prefix="--citation", doc=InputDocumentation(doc="How to cite MitoFinder")), ToolInput(tag="in_format_dot", input_type=String(), position=0, doc=InputDocumentation(doc="--min-contig-size MINCONTIGSIZE")), ToolInput(tag="in__maxcontigsize_maxcontigsize", input_type=Int(), position=0, doc=InputDocumentation(doc="--max-contig-size MAXCONTIGSIZE")), ToolInput(tag="in__renamecontig_rename", input_type=Int(), position=0, doc=InputDocumentation(doc="--rename-contig RENAME"))], outputs=[ToolOutput(tag="out_override", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_override", type_hint=File()), doc=OutputDocumentation(doc="This option forces MitoFinder to override the previous\noutput directory for the selected assembler.")), ToolOutput(tag="out_out_gb", output_type=File(optional=True), selector=InputSelector(input_to_select="in_out_gb", type_hint=File()), doc=OutputDocumentation(doc="Do not create annotation output file in GenBank"))], container="quay.io/biocontainers/mitofinder:1.4--py27h516909a_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Mitofinder_V0_1_0().translate("wdl")

