from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Int, String, Boolean, Float, Directory

Mimodd_Snap_V0_1_0 = CommandToolBuilder(tool="mimodd_snap", base_command=["mimodd", "snap"], inputs=[ToolInput(tag="in_ofile", input_type=File(optional=True), prefix="--ofile", doc=InputDocumentation(doc="name of the output file (required)")), ToolInput(tag="in_i_format", input_type=File(optional=True), prefix="--iformat", doc=InputDocumentation(doc="input file format; must be fastq, gz, sam or bam\n(default: bam)")), ToolInput(tag="in_o_format", input_type=File(optional=True), prefix="--oformat", doc=InputDocumentation(doc="output file format (sam or bam; default: bam)")), ToolInput(tag="in_header", input_type=File(optional=True), prefix="--header", doc=InputDocumentation(doc="a SAM file providing header information to be used for\nthe output (required for input in fastq format and\nwith unheadered SAM/BAM input, optional for headered\nSAM/BAM input; replaces header information found in\nthe input file")), ToolInput(tag="in_threads", input_type=Int(optional=True), prefix="--threads", doc=InputDocumentation(doc="number of threads to use (overrides config setting)")), ToolInput(tag="in_memory", input_type=String(optional=True), prefix="--memory", doc=InputDocumentation(doc="maximal amount of memory to be used in GB (overrides\nconfig setting); will be respected during sorting only")), ToolInput(tag="in_no_sort", input_type=Boolean(optional=True), prefix="--no-sort", doc=InputDocumentation(doc="output reads in their original order, i.e., do not\nsort by alignment location")), ToolInput(tag="in_quiet", input_type=Boolean(optional=True), prefix="--quiet", doc=InputDocumentation(doc="suppress original messages from SNAP")), ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="--verbose", doc=InputDocumentation(doc="verbose output (independent of SNAP)")), ToolInput(tag="in_cigar_strings_output", input_type=Boolean(optional=True), prefix="-X", doc=InputDocumentation(doc="CIGAR strings in the output should use = and X to\nindicate matches/mismatches rather than M (alignment\nmatch); USE OF THIS OPTION IS DISCOURAGED as =/X CIGAR\nstrings are still not fully supported by useful third-\nparty tools like IGV")), ToolInput(tag="in_m_match_notation", input_type=Boolean(optional=True), prefix="--mmatch-notation", doc=InputDocumentation(doc="legacy option retained for backwards compatibility;\nindicates that CIGAR strings in the output should use\nM (alignment match) rather than = and X (sequence\n(mis-)match); -M is implied by default, use -X to turn\noff")), ToolInput(tag="in_sort", input_type=File(optional=True), prefix="--sort", doc=InputDocumentation(doc="legacy option retained for backwards compatibility;\nsort output file by alignment location; implied by\ndefault, use --no-sort to turn off")), ToolInput(tag="in_consider_mate_pairs", input_type=Boolean(optional=True), prefix="-D", doc=InputDocumentation(doc="[RF|FR|FF|RR|ALL [RF|FR|FF|RR|ALL ...]], --discard-overlapping-mates [RF|FR|FF|RR|ALL [RF|FR|FF|RR|ALL ...]]\nconsider overlapping mate pairs of the given\norientation type(s) anomalous and discard them;\ndefault: keep all overlapping mate pairs")), ToolInput(tag="in_max__spacing", input_type=Int(optional=True), prefix="-s", doc=InputDocumentation(doc="MAX, --spacing MIN MAX\nmin and max spacing to allow between paired ends\n(default: 100 10000)")), ToolInput(tag="in_distance__maxdist", input_type=Int(optional=True), prefix="-d", doc=InputDocumentation(doc="DISTANCE, --maxdist EDIT DISTANCE\nmaximum edit distance allowed per read or pair\n(default: 8)")), ToolInput(tag="in_max_seeds", input_type=Int(optional=True), prefix="--maxseeds", doc=InputDocumentation(doc="number of seeds to use per read (default: 25)")), ToolInput(tag="in_max_hits", input_type=Int(optional=True), prefix="--maxhits", doc=InputDocumentation(doc="maximum hits to consider per seed (default: 250)")), ToolInput(tag="in_conf_diff", input_type=Int(optional=True), prefix="--confdiff", doc=InputDocumentation(doc="confidence threshold (default: 2)")), ToolInput(tag="in_conf_adapt", input_type=Int(optional=True), prefix="--confadapt", doc=InputDocumentation(doc="confidence adaptation threshold (default: 7)")), ToolInput(tag="in_error_rep", input_type=Boolean(optional=True), prefix="--error-rep", doc=InputDocumentation(doc="compute error rate assuming wgsim-generated reads")), ToolInput(tag="in_explore", input_type=Boolean(optional=True), prefix="--explore", doc=InputDocumentation(doc="explore some hits of overly popular seeds (useful for\nfiltering)")), ToolInput(tag="in_stop_on_first", input_type=Boolean(optional=True), prefix="--stop-on-first", doc=InputDocumentation(doc="stop on first match within edit distance limit\n(filtering mode)")), ToolInput(tag="in_filter_output", input_type=String(optional=True), prefix="--filter-output", doc=InputDocumentation(doc="retain only certain read classes in output (a=aligned\nonly, s=single hit only, u=unaligned only)")), ToolInput(tag="in_ignore", input_type=Boolean(optional=True), prefix="--ignore", doc=InputDocumentation(doc="ignore non-matching IDs in the paired-end aligner")), ToolInput(tag="in_selectivity", input_type=Int(optional=True), prefix="--selectivity", doc=InputDocumentation(doc="selectivity; randomly choose 1/selectivity of the\nreads to score")), ToolInput(tag="in_xxxx__clipping", input_type=Boolean(optional=True), prefix="-C", doc=InputDocumentation(doc="++|+x|x+|xx, --clipping ++|+x|x+|xx\nspecify a combination of two + or x symbols to\nindicate whether to clip low-quality bases from the\nfront and back of reads respectively; default: clip\nfrom front and back (-C ++)")), ToolInput(tag="in_gap_penalty", input_type=String(optional=True), prefix="--gap-penalty", doc=InputDocumentation(doc="specify a gap penalty to use when generating CIGAR\nstrings")), ToolInput(tag="in_idx_seed_size", input_type=Int(optional=True), prefix="--idx-seedsize", doc=InputDocumentation(doc="SIZE\nSeed size used in building the index (default: 20)")), ToolInput(tag="in_idx_slack", input_type=Float(optional=True), prefix="--idx-slack", doc=InputDocumentation(doc="Hash table slack for indexing (default: 0.3)")), ToolInput(tag="in_idx_overflow", input_type=Int(optional=True), prefix="--idx-overflow", doc=InputDocumentation(doc="factor (between 1 and 1000) to set the size of the\nindex build overflow space (default: 40)")), ToolInput(tag="in_idx_out", input_type=Directory(optional=True), prefix="--idx-out", doc=InputDocumentation(doc="DIR   name of the index directory to be created; if given,\nthe index directory will be permanent, otherwise a\ntemporary directory will be used")), ToolInput(tag="in_no_prefetch", input_type=Boolean(optional=True), prefix="--no-prefetch", doc=InputDocumentation(doc="disables cache prefetching in the genome; may be\nhelpful for machines with small caches or lots of\ncores/cache")), ToolInput(tag="in_bind_threads", input_type=Boolean(optional=True), prefix="--bind-threads", doc=InputDocumentation(doc="bind each thread to its processor (off by default)")), ToolInput(tag="in_snap", input_type=String(), position=0, doc=InputDocumentation(doc=""))], outputs=[ToolOutput(tag="out_ofile", output_type=File(optional=True), selector=InputSelector(input_to_select="in_ofile", type_hint=File()), doc=OutputDocumentation(doc="name of the output file (required)")), ToolOutput(tag="out_o_format", output_type=File(optional=True), selector=InputSelector(input_to_select="in_o_format", type_hint=File()), doc=OutputDocumentation(doc="output file format (sam or bam; default: bam)")), ToolOutput(tag="out_sort", output_type=File(optional=True), selector=InputSelector(input_to_select="in_sort", type_hint=File()), doc=OutputDocumentation(doc="legacy option retained for backwards compatibility;\nsort output file by alignment location; implied by\ndefault, use --no-sort to turn off"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Mimodd_Snap_V0_1_0().translate("wdl", allow_empty_container=True)

