from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import String, File, Boolean, Float, Int, Directory

Haystac_Database_V0_1_0 = CommandToolBuilder(tool="haystac_database", base_command=["haystac", "database"], inputs=[ToolInput(tag="in_mode", input_type=String(optional=True), prefix="--mode", doc=InputDocumentation(doc="Database creation mode for haystac [fetch, index,\nbuild]")), ToolInput(tag="in_output", input_type=File(optional=True), prefix="--output", doc=InputDocumentation(doc="Path to the database output directory")), ToolInput(tag="in_query", input_type=String(optional=True), prefix="--query", doc=InputDocumentation(doc="Database query in the NCBI query language. Please\nrefer to the documentation for assistance with\nconstructing a valid query.")), ToolInput(tag="in_query_file", input_type=File(optional=True), prefix="--query-file", doc=InputDocumentation(doc="File containing a database query in the NCBI query")), ToolInput(tag="in_sequences_file", input_type=File(optional=True), prefix="--sequences-file", doc=InputDocumentation(doc="Tab delimited file containing one record per row: the\nname of the taxon, a user defined accession code, and\nthe path to the fasta file (optionally compressed).")), ToolInput(tag="in_refseq_rep", input_type=String(optional=True), prefix="--refseq-rep", doc=InputDocumentation(doc="Use one of the RefSeq curated tables to construct a\nDB. Includes all prokaryotic species (excluding\nstrains) from the representative RefSeq DB, or all the\nspecies and strains from the viruses DB, or all the\nspecies and subspecies from the eukaryotes DB. If\nmultiple accessions exist for a given species/strain,\nthe first pair of species/accession is kept. Available\nRefSeq tables to use [prokaryote_rep, viruses,\neukaryotes].")), ToolInput(tag="in_force_accessions", input_type=Boolean(optional=True), prefix="--force-accessions", doc=InputDocumentation(doc="Disable validation checks for 'anomalous' assembly\nflags in NCBI (default: False)")), ToolInput(tag="in_exclude_accessions", input_type=String(optional=True), prefix="--exclude-accessions", doc=InputDocumentation(doc="[<accession> ...]\nList of NCBI accessions to exclude. (default: [])")), ToolInput(tag="in_resolve_accessions", input_type=Boolean(optional=True), prefix="--resolve-accessions", doc=InputDocumentation(doc="Pick the first accession when two accessions for a\ntaxon can be found in user provided input files\n(default: False)")), ToolInput(tag="in_bowtie_two_scaling", input_type=Float(optional=True), prefix="--bowtie2-scaling", doc=InputDocumentation(doc="Rescaling factor to keep the bowtie2 mutlifasta index\nbelow the maximum memory limit (default: 25.0)")), ToolInput(tag="in_rank", input_type=String(optional=True), prefix="--rank", doc=InputDocumentation(doc="Taxonomic rank to perform the identifications on\n[genus, species, subspecies, serotype] (default:\nspecies)")), ToolInput(tag="in_genera", input_type=String(optional=True), prefix="--genera", doc=InputDocumentation(doc="[<genus> ...]\nList of genera to restrict the abundance calculations.")), ToolInput(tag="in_mtdna", input_type=Boolean(optional=True), prefix="--mtDNA", doc=InputDocumentation(doc="For eukaryotes, download mitochondrial genomes only.\nNot to be used with --refseq-rep or queries containing\nprokaryotes (default: False)")), ToolInput(tag="in_seed", input_type=Int(optional=True), prefix="--seed", doc=InputDocumentation(doc="Random seed for database indexing")), ToolInput(tag="in_cores", input_type=Int(optional=True), prefix="--cores", doc=InputDocumentation(doc="Maximum number of CPU cores to use (default: 8)")), ToolInput(tag="in_mem", input_type=Int(optional=True), prefix="--mem", doc=InputDocumentation(doc="Maximum memory (MB) to use (default: 16012)")), ToolInput(tag="in_unlock", input_type=Directory(optional=True), prefix="--unlock", doc=InputDocumentation(doc="Unlock the output directory following a crash or hard\nrestart (default: False)")), ToolInput(tag="in_debug", input_type=Boolean(optional=True), prefix="--debug", doc=InputDocumentation(doc="Enable debugging mode (default: False)")), ToolInput(tag="in_snake_make", input_type=Boolean(optional=True), prefix="--snakemake", doc=InputDocumentation(doc="'<json>'  Pass additional flags to the `snakemake` scheduler.")), ToolInput(tag="in_language_dot", input_type=String(), position=0, doc=InputDocumentation(doc="--accessions-file <path>"))], outputs=[ToolOutput(tag="out_output", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output", type_hint=File()), doc=OutputDocumentation(doc="Path to the database output directory")), ToolOutput(tag="out_unlock", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_unlock", type_hint=File()), doc=OutputDocumentation(doc="Unlock the output directory following a crash or hard\nrestart (default: False)"))], container="quay.io/biocontainers/haystac:0.3.2--py36_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Haystac_Database_V0_1_0().translate("wdl")

