from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import String, File, Int

Anvi_Get_Short_Reads_Mapping_To_A_Gene_V0_1_0 = CommandToolBuilder(tool="anvi_get_short_reads_mapping_to_a_gene", base_command=["anvi-get-short-reads-mapping-to-a-gene"], inputs=[ToolInput(tag="in_contigs_db", input_type=String(optional=True), prefix="--contigs-db", doc=InputDocumentation(doc="Anvi'o contigs database generated by 'anvi-gen-\ncontigs'")), ToolInput(tag="in_s_sorted_files", input_type=String(optional=True), prefix="-i", doc=InputDocumentation(doc="(S) [INPUT_BAM(S) ...], --input-files INPUT_BAM(S) [INPUT_BAM(S) ...]\nSorted and indexed BAM files to analyze. It is\nessential that all BAM files must be the result of\nmappings against the same contigs.")), ToolInput(tag="in_gene_caller_id", input_type=String(optional=True), prefix="--gene-caller-id", doc=InputDocumentation(doc="A single gene id.")), ToolInput(tag="in_genes_of_interest", input_type=File(optional=True), prefix="--genes-of-interest", doc=InputDocumentation(doc="A file with anvi'o gene caller IDs. There should be\nonly one column in the file, and each line should\ncorrespond to a unique gene caller id (without a\ncolumn header).")), ToolInput(tag="in_leeway", input_type=Int(optional=True), prefix="--leeway", doc=InputDocumentation(doc="The minimum number of nucleotides for a given short\nread mapping into the gene context for it to be\nreported. You must consider the length of your short\nreads, as well as the length of the gene you are\ntargeting. The default is 100 nts.")), ToolInput(tag="in_output_file_prefix", input_type=File(optional=True), prefix="--output-file-prefix", doc=InputDocumentation(doc="A prefix to be used while naming the output files (no\nfile type extensions please; just a prefix).\n"))], outputs=[ToolOutput(tag="out_output_file_prefix", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output_file_prefix", type_hint=File()), doc=OutputDocumentation(doc="A prefix to be used while naming the output files (no\nfile type extensions please; just a prefix).\n"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Anvi_Get_Short_Reads_Mapping_To_A_Gene_V0_1_0().translate("wdl", allow_empty_container=True)

