version 1.0

task Rsemrunebseq {
  input {
    File? ng_vector
    Boolean? h_slash_help
    String analysis_dot
    String data_matrix_file
    String conditions
    String output_file_dot_normalized_data_matrix
    String outputfilepattern
    String output_file_dot_cond_means
    String interested_genelevel_analysis
  }
  command <<<
    rsem_run_ebseq \
      ~{analysis_dot} \
      ~{data_matrix_file} \
      ~{conditions} \
      ~{output_file_dot_normalized_data_matrix} \
      ~{outputfilepattern} \
      ~{output_file_dot_cond_means} \
      ~{interested_genelevel_analysis} \
      ~{if defined(ng_vector) then ("--ngvector " +  '"' + ng_vector + '"') else ""} \
      ~{if (h_slash_help) then "-h/--help" else ""}
  >>>
  parameter_meta {
    ng_vector: "This option provides the grouping information required by EBSeq for\\nisoform-level differential expression analysis. The file can be\\ngenerated by 'rsem-generate-ngvector'. Turning this option on is\\nhighly recommended for isoform-level differential expression\\nanalysis. (Default: off)"
    h_slash_help: "Show help information."
    analysis_dot: "SYNOPSIS"
    data_matrix_file: "This file is a m by n matrix. m is the number of genes/transcripts\\nand n is the number of total samples. Each element in the matrix\\nrepresents the expected count for a particular gene/transcript in a\\nparticular sample. Users can use 'rsem-generate-data-matrix' to\\ngenerate this file from expression result files."
    conditions: "Comma-separated list of values representing the number of replicates\\nfor each condition. For example, \\\"3,3\\\" means the data set contains 2\\nconditions and each condition has 3 replicates. \\\"2,3,3\\\" means the\\ndata set contains 3 conditions, with 2, 3, and 3 replicates for each\\ncondition respectively."
    output_file_dot_normalized_data_matrix: "This file contains the median normalized version of the input data"
    outputfilepattern: "output_file.pattern"
    output_file_dot_cond_means: "This file is only generated when there are more than 2 conditions.\\nIt gives the normalized mean count value for each gene/transcript at\\neach condition. It is formatted as a matrix with names. Each row\\nrepresents a gene/transcript and each column represent a condition.\\nThe order of genes/transcripts is the same as 'output_file'. This\\nfile can be used to calculate fold changes between conditions which\\nusers are interested in."
    interested_genelevel_analysis: "2) We're interested in gene-level analysis and there are 3 conditions."
  }
  output {
    File out_stdout = stdout()
  }
}