class: CommandLineTool
id: ../../../rsem_run_ebseq.cwl
inputs:
- id: ng_vector
  doc: "This option provides the grouping information required by EBSeq for isoform-level\
    \ differential expression analysis. The file can be generated by 'rsem-generate-ngvector'.\
    \ Turning this option on is highly recommended for isoform-level differential\
    \ expression analysis. (Default: off)"
  type: File
  inputBinding:
    prefix: --ngvector
- id: h_slash_help
  doc: Show help information.
  type: boolean
  inputBinding:
    prefix: -h/--help
- id: data_matrix_file
  doc: This file is a m by n matrix. m is the number of genes/transcripts and n is
    the number of total samples. Each element in the matrix represents the expected
    count for a particular gene/transcript in a particular sample. Users can use 'rsem-generate-data-matrix'
    to generate this file from expression result files.
  type: string
  inputBinding:
    position: 0
- id: conditions
  doc: Comma-separated list of values representing the number of replicates for each
    condition. For example, "3,3" means the data set contains 2 conditions and each
    condition has 3 replicates. "2,3,3" means the data set contains 3 conditions,
    with 2, 3, and 3 replicates for each condition respectively.
  type: string
  inputBinding:
    position: 1
- id: output_file_dot_normalized_data_matrix
  doc: This file contains the median normalized version of the input data matrix.
  type: string
  inputBinding:
    position: 1
- id: output_file_dot_pattern
  doc: This file is only generated when there are more than 2 conditions. It defines
    all possible expression patterns over the conditions using a matrix with names.
    Each row of the matrix refers to a different expression pattern and each column
    gives the expression status of a different condition. Two conditions are equally
    expressed if and only if their statuses are the same.
  type: string
  inputBinding:
    position: 2
- id: output_file_dot_cond_means
  doc: This file is only generated when there are more than 2 conditions. It gives
    the normalized mean count value for each gene/transcript at each condition. It
    is formatted as a matrix with names. Each row represents a gene/transcript and
    each column represent a condition. The order of genes/transcripts is the same
    as 'output_file'. This file can be used to calculate fold changes between conditions
    which users are interested in.
  type: string
  inputBinding:
    position: 3
outputs: []
cwlVersion: v1.1
baseCommand:
- rsem-run-ebseq
