from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Array, String, File, Boolean, Int

Convertdb_Py_Genbank_V0_1_0 = CommandToolBuilder(tool="ConvertDb.py_genbank", base_command=["ConvertDb.py", "genbank"], inputs=[ToolInput(tag="in_list_tab_defaultnone", input_type=Array(t=String(), optional=True), prefix="-d", doc=InputDocumentation(doc="A list of tab delimited database files. (default:\nNone)")), ToolInput(tag="in_explicit_output_file", input_type=Array(t=String(), optional=True), prefix="-o", doc=InputDocumentation(doc="Explicit output file name. Note, this argument cannot\nbe used with the --failed, --outdir, or --outname\narguments. If unspecified, then the output filename\nwill be based on the input filename(s). (default:\nNone)")), ToolInput(tag="in_outdir", input_type=File(optional=True), prefix="--outdir", doc=InputDocumentation(doc="Specify to changes the output directory to the\nlocation specified. The input file directory is used\nif this is not specified. (default: None)")), ToolInput(tag="in_out_name", input_type=File(optional=True), prefix="--outname", doc=InputDocumentation(doc="Changes the prefix of the successfully processed\noutput file to the string specified. May not be\nspecified with multiple input files. (default: None)")), ToolInput(tag="in_format", input_type=String(optional=True), prefix="--format", doc=InputDocumentation(doc="Specify input and output format. (default: airr)")), ToolInput(tag="in_mol", input_type=String(optional=True), prefix="--mol", doc=InputDocumentation(doc="The source molecule type. Usually one of 'mRNA' or\n'genomic DNA'. (default: mRNA)")), ToolInput(tag="in_product", input_type=String(optional=True), prefix="--product", doc=InputDocumentation(doc="The product name, such as 'immunoglobulin heavy\nchain'. (default: immunoglobulin heavy chain)")), ToolInput(tag="in_db", input_type=String(optional=True), prefix="--db", doc=InputDocumentation(doc="Name of the reference database used for alignment.\nUsually 'IMGT/GENE-DB'. (default: None)")), ToolInput(tag="in_inf", input_type=String(optional=True), prefix="--inf", doc=InputDocumentation(doc="Name and version of the inference tool used for\nreference alignment in the form tool:version.\n(default: None)")), ToolInput(tag="in_organism", input_type=String(optional=True), prefix="--organism", doc=InputDocumentation(doc="The scientific name of the organism. (default: None)")), ToolInput(tag="in_sex", input_type=String(optional=True), prefix="--sex", doc=InputDocumentation(doc="If specified, adds the given sex annotation to the\nfasta headers. (default: None)")), ToolInput(tag="in_isolate", input_type=String(optional=True), prefix="--isolate", doc=InputDocumentation(doc="If specified, adds the given isolate annotation\n(sample label) to the fasta headers. (default: None)")), ToolInput(tag="in_tissue", input_type=String(optional=True), prefix="--tissue", doc=InputDocumentation(doc="If specified, adds the given tissue-type annotation to\nthe fasta headers. (default: None)")), ToolInput(tag="in_cell_type", input_type=String(optional=True), prefix="--cell-type", doc=InputDocumentation(doc="If specified, adds the given cell-type annotation to\nthe fasta headers. (default: None)")), ToolInput(tag="in_yaml_file_specifying", input_type=File(optional=True), prefix="-y", doc=InputDocumentation(doc="A yaml file specifying sample features (BioSample\nattributes) in the form 'variable: value'. If\nspecified, any features provided in the yaml file will\noverride those provided at the commandline. Note, this\nconfig file applies to sample features only and cannot\nbe used for required source features such as the\n--product or --mol argument. (default: None)")), ToolInput(tag="in_label", input_type=String(optional=True), prefix="--label", doc=InputDocumentation(doc="If specified, add a field name to the sequence\nidentifier. Sequence identifiers will be output in the\nform <label>=<id>. (default: None)")), ToolInput(tag="in_cf", input_type=String(optional=True), prefix="--cf", doc=InputDocumentation(doc="Field containing the C region call. If unspecified,\nthe C region gene call will be excluded from the\nfeature table. (default: None)")), ToolInput(tag="in_nf", input_type=String(optional=True), prefix="--nf", doc=InputDocumentation(doc="If specified, use the provided column to add the\nAIRR_READ_COUNT note to the feature table. (default:\nNone)")), ToolInput(tag="in_if", input_type=String(optional=True), prefix="--if", doc=InputDocumentation(doc="If specified, use the provided column to add the\nAIRR_CELL_INDEX note to the feature table. (default:\nNone)")), ToolInput(tag="in_allow_stop", input_type=Boolean(optional=True), prefix="--allow-stop", doc=InputDocumentation(doc="If specified, retain records in the output with stop\ncodons in the junction region. In such records the CDS\nwill be removed and replaced with a similar\nmisc_feature in the feature table. (default: False)")), ToolInput(tag="in_as_is_id", input_type=Boolean(optional=True), prefix="--asis-id", doc=InputDocumentation(doc="If specified, use the existing sequence identifier for\nthe output identifier. By default, only the row number\nwill be used as the identifier to avoid the 50\ncharacter limit. (default: False)")), ToolInput(tag="in_as_is_calls", input_type=Boolean(optional=True), prefix="--asis-calls", doc=InputDocumentation(doc="Specify to prevent alleles from being parsed using the\nIMGT nomenclature. Note, this requires the gene\nassignments to be exact matches to valid records in\nthe references database specified by the --db\nargument. (default: False)")), ToolInput(tag="in_allele_delim", input_type=Int(optional=True), prefix="--allele-delim", doc=InputDocumentation(doc="The delimiter to use for splitting the gene name from\nthe allele number. Note, this only applies when\nspecifying --asis-calls. By default, this argument\nwill be ignored and allele numbers extracted under the\nexpectation of IMGT nomenclature consistency.\n(default: *)")), ToolInput(tag="in_asn", input_type=Boolean(optional=True), prefix="--asn", doc=InputDocumentation(doc="If specified, run tbl2asn to generate the .sqn\nsubmission file after making the .fsa and .tbl files.\n(default: False)")), ToolInput(tag="in_sbt", input_type=File(optional=True), prefix="--sbt", doc=InputDocumentation(doc="If provided along with --asn, use the specified file\nfor the template file argument to tbl2asn. (default:\nNone)")), ToolInput(tag="in_exec", input_type=Int(optional=True), prefix="--exec", doc=InputDocumentation(doc="The name or location of the tbl2asn executable.\n(default: tbl2asn)\n"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Convertdb_Py_Genbank_V0_1_0().translate("wdl", allow_empty_container=True)

