from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Boolean, String, Int

Snap_Aligner_Paired_V0_1_0 = CommandToolBuilder(tool="snap_aligner_paired", base_command=["snap-aligner", "paired"], inputs=[ToolInput(tag="in_filename_output_alignments", input_type=File(optional=True), prefix="-o", doc=InputDocumentation(doc="filename  output alignments to filename in SAM or BAM format, depending on the file extension or\nexplicit type specifier (see below).  Use a dash with an explicit type specifier to write to\nstdout, so for example -o -sam - would write SAM output to stdout")), ToolInput(tag="in_maximum_edit_distance", input_type=Boolean(optional=True), prefix="-d", doc=InputDocumentation(doc="maximum edit distance allowed per read or pair (default: 15)")), ToolInput(tag="in_number_seeds_use", input_type=Boolean(optional=True), prefix="-n", doc=InputDocumentation(doc="number of seeds to use per read")), ToolInput(tag="in_sc", input_type=Boolean(optional=True), prefix="-sc", doc=InputDocumentation(doc="Seed coverage (i.e., readSize/seedSize).  Floating point.  Exclusive with -n.  (default uses -n)")), ToolInput(tag="in_ms", input_type=Boolean(optional=True), prefix="-ms", doc=InputDocumentation(doc="minimum seed matches per location (default: 1)")), ToolInput(tag="in_number_threads_default", input_type=Boolean(optional=True), prefix="-t", doc=InputDocumentation(doc="number of threads (default is one per core)")), ToolInput(tag="in_bind_thread_its", input_type=Boolean(optional=True), prefix="-b", doc=InputDocumentation(doc="bind each thread to its processor (this is the default)")), ToolInput(tag="in_disables_cache_prefetching", input_type=Boolean(optional=True), prefix="-P", doc=InputDocumentation(doc="disables cache prefetching in the genome; may be helpful for machines\nwith small caches or lots of cores/cache")), ToolInput(tag="in_so", input_type=File(optional=True), prefix="-so", doc=InputDocumentation(doc="sort output file by alignment location")), ToolInput(tag="in_sm", input_type=Boolean(optional=True), prefix="-sm", doc=InputDocumentation(doc="memory to use for sorting in Gb")), ToolInput(tag="in_explore_hits_overly", input_type=Boolean(optional=True), prefix="-x", doc=InputDocumentation(doc="explore some hits of overly popular seeds (useful for filtering)")), ToolInput(tag="in_stop_first_match", input_type=Boolean(optional=True), prefix="-f", doc=InputDocumentation(doc="stop on first match within edit distance limit (filtering mode)")), ToolInput(tag="in_filter_output_aaligned", input_type=Boolean(optional=True), prefix="-F", doc=InputDocumentation(doc="filter output (a=aligned only, s=single hit only (MAPQ >= 10), u=unaligned only, l=long enough to align (see -mrl))")), ToolInput(tag="in_alternate_fully_general", input_type=Boolean(optional=True), prefix="-E", doc=InputDocumentation(doc="an alternate (and fully general) way to specify filter options.  Emit only these types s = single hit (MAPQ >= 10), m = multiple hit (MAPQ < 10),\nx = not long enough to align, u = unaligned, b = filter must apply to both ends of a paired-end read.  Combine the letters after\n-E, so for example -E smu will emit all reads that aren't too short/have too many Ns (because it leaves off l).  -E smx is the same\nas -F a, -E ux is the same as -F u, and so forth.\nWhen filtering in paired-end mode (either with -F or -E) unless you specify the b flag a read will be emitted if it's mate pair passes the filter\nEven if the read itself does not.  If you specify b mode, then a read will be emitted only if it and its partner both pass the filter.")), ToolInput(tag="in_suppress_output_onlyiindex", input_type=Boolean(optional=True), prefix="-S", doc=InputDocumentation(doc="suppress additional processing (sorted BAM output only)\ni=index, d=duplicate marking")), ToolInput(tag="in_ignore_ids_match", input_type=Boolean(optional=True), prefix="-I", doc=InputDocumentation(doc="ignore IDs that don't match in the paired-end aligner")), ToolInput(tag="in_cxx", input_type=String(optional=True), prefix="-Cxx", doc=InputDocumentation(doc="be followed by two + or - symbols saying whether to clip low-quality\nbases from front and back of read respectively; default: back only (-C-+)")), ToolInput(tag="in_indicates_strings_use", input_type=Boolean(optional=True), prefix="-M", doc=InputDocumentation(doc="indicates that CIGAR strings in the generated SAM file should use M (alignment\nmatch) rather than = and X (sequence (mis-)match).  This is the default")), ToolInput(tag="in_specify_gap_penalty", input_type=Boolean(optional=True), prefix="-G", doc=InputDocumentation(doc="specify a gap penalty to use when generating CIGAR strings")), ToolInput(tag="in_pf", input_type=Boolean(optional=True), prefix="-pf", doc=InputDocumentation(doc="specify the name of a file to contain the run speed")), ToolInput(tag="in_hp", input_type=String(optional=True), prefix="--hp", doc=InputDocumentation(doc="not to use huge pages (this may speed up index load and slow down alignment)  This is the default")), ToolInput(tag="in_specifies_extra_depth", input_type=Boolean(optional=True), prefix="-D", doc=InputDocumentation(doc="Specifies the extra search depth (the edit distance beyond the best hit that SNAP uses to compute MAPQ).  Default 2")), ToolInput(tag="in_rg", input_type=Boolean(optional=True), prefix="-rg", doc=InputDocumentation(doc="Specify the default read group if it is not specified in the input file")), ToolInput(tag="in_specify_entire_line", input_type=Boolean(optional=True), prefix="-R", doc=InputDocumentation(doc="Specify the entire read group line for the SAM/BAM output.  This must include an ID tag.  If it doesn't start with\n'@RG' SNAP will add that.  Specify tabs by \t.  Two backslashes will generate a single backslash.\nbackslash followed by anything else is illegal.  So, '-R @RG\tID:foo\tDS:my data' would generate reads\nwith defualt tag foo, and an @RG line that also included the DS:my data field.")), ToolInput(tag="in_sa", input_type=Boolean(optional=True), prefix="-sa", doc=InputDocumentation(doc="Include reads from SAM or BAM files with the secondary (0x100) or supplementary (0x800) flag set; default is to drop them.")), ToolInput(tag="in_om", input_type=Boolean(optional=True), prefix="-om", doc=InputDocumentation(doc="Output multiple alignments.  Takes as a parameter the maximum extra edit distance relative to the best alignment\nto allow for secondary alignments")), ToolInput(tag="in_omax", input_type=Int(optional=True), prefix="-omax", doc=InputDocumentation(doc="the number of alignments per read generated by -om.  This means that if -om would generate more\nthan -omax secondary alignments, SNAP will write out only the best -omax of them, where 'best' means\n'with the lowest edit distance'.  Ties are broken arbitrarily.")), ToolInput(tag="in_mpc", input_type=Int(optional=True), prefix="-mpc", doc=InputDocumentation(doc="the number of alignments generated by -om to this many per contig (chromosome/FASTA entry);\n'mpc' means 'max per contig; default unlimited.  This filter is applied prior to -omax.  The primary alignment\nis counted.")), ToolInput(tag="in_pc", input_type=Boolean(optional=True), prefix="-pc", doc=InputDocumentation(doc="Preserve the soft clipping for reads coming from SAM or BAM files")), ToolInput(tag="in_xf", input_type=Boolean(optional=True), prefix="-xf", doc=InputDocumentation(doc="Increase expansion factor for BAM and GZ files (default 1.0)")), ToolInput(tag="in_hdp", input_type=String(optional=True), prefix="-hdp", doc=InputDocumentation(doc="Hadoop-style prefixes (reporter:status:...) on error messages, and emit hadoop-style progress messages")), ToolInput(tag="in_mrl", input_type=Int(optional=True), prefix="-mrl", doc=InputDocumentation(doc="the minimum read length to align, reads shorter than this (after clipping) stay unaligned.  This should be\na good bit bigger than the seed length or you might get some questionable alignments.  Default 50")), ToolInput(tag="in_map", input_type=Int(optional=True), prefix="-map", doc=InputDocumentation(doc="file mapping to load the index rather than reading it.  This might speed up index loading in cases\nwhere SNAP is run repatedly on the same index, and the index is larger than half of the memory size\nof the machine.  On some operating systems, loading an index with -map is much slower than without if the\nindex is not in memory.  You might consider adding -pre to prefetch the index into system cache when loading\nwith -map when you don't expect the index to be in cache.")), ToolInput(tag="in_pre", input_type=String(optional=True), prefix="-pre", doc=InputDocumentation(doc="the index into system cache.  This is only meaningful with -map, and only helps if the index is not\nalready in memory and your operating system is slow at reading mapped files (i.e., some versions of Linux,\nbut not Windows).")), ToolInput(tag="in_lp", input_type=Boolean(optional=True), prefix="-lp", doc=InputDocumentation(doc="Run SNAP at low scheduling priority (Only implemented on Windows)")), ToolInput(tag="in_nu", input_type=Boolean(optional=True), prefix="-nu", doc=InputDocumentation(doc="No Ukkonen: don't reduce edit distance search based on prior candidates. This option is purely for\nevaluating the performance effect of using Ukkonen's algorithm rather than Smith-Waterman, and specifying\nit will slow down execution without improving the alignments.")), ToolInput(tag="in_no", input_type=Boolean(optional=True), prefix="-no", doc=InputDocumentation(doc="No Ordering: don't order the evalutation of reads so as to select more likely candidates first.  This option\nis purely for evaluating the performance effect of the read evaluation order, and specifying it will slow\ndown execution without improving alignments.")), ToolInput(tag="in_nt", input_type=Boolean(optional=True), prefix="-nt", doc=InputDocumentation(doc="Don't truncate searches based on missed seed hits.  This option is purely for evaluating the performance effect\nof candidate truncation, and specifying it will slow down execution without improving alignments.")), ToolInput(tag="in_wbs", input_type=Boolean(optional=True), prefix="-wbs", doc=InputDocumentation(doc="Write buffer size in megabytes.  Don't specify this unless you've gotten an error message saying to make it bigger.  Default 16.")), ToolInput(tag="in_di", input_type=Boolean(optional=True), prefix="-di", doc=InputDocumentation(doc="Drop the index after aligning and before sorting.  This frees up memory for the sort at the expense of not having the index loaded for your next run.")), ToolInput(tag="in_kts", input_type=Boolean(optional=True), prefix="-kts", doc=InputDocumentation(doc="Kill if too slow.  Monitor our progress and kill ourself if we're not moving fast enough.  This is intended for use on machines\nwith limited memory, where some alignment tasks will push SNAP into paging, and take disproportinaltely long.  This allows the script\nto move on to the next alignment.  Only works when generating output, and not during the sort phase.  If you're running out of memory\nsorting, try using -di.")), ToolInput(tag="in_sid", input_type=File(optional=True), prefix="-sid", doc=InputDocumentation(doc="Specifies the sort intermediate directory.  When SNAP is sorting, it aligns the reads in the order in which they come in, and writes\nthe aligned reads in batches to a temporary file.  When the aligning is done, it does a merge sort from the temporary file into the\nfinal output file.  By default, the intermediate file is in the same directory as the output file, but for performance or space\nreasons, you might want to put it elsewhere.  If so, use this option.")), ToolInput(tag="in_pro", input_type=Boolean(optional=True), prefix="-pro", doc=InputDocumentation(doc="Profile alignment to give you an idea of how much time is spent aligning and how much waiting for IO")), ToolInput(tag="in_min_max_spacing", input_type=Boolean(optional=True), prefix="-s", doc=InputDocumentation(doc="min and max spacing to allow between paired ends (default: 50 1000).")), ToolInput(tag="in_fs", input_type=Boolean(optional=True), prefix="-fs", doc=InputDocumentation(doc="force spacing to lie between min and max.")), ToolInput(tag="in_max_hits_intersecting", input_type=Boolean(optional=True), prefix="-H", doc=InputDocumentation(doc="max hits for intersecting aligner (default: 2000).")), ToolInput(tag="in_mcp", input_type=Int(optional=True), prefix="-mcp", doc=InputDocumentation(doc="the maximum candidate pool size (An internal data structure.\nOnly increase this if you get an error message saying to do so. If you're running\nout of memory, you may want to reduce it.  Default: 1000000)")), ToolInput(tag="in_ku", input_type=Boolean(optional=True), prefix="-ku", doc=InputDocumentation(doc="Keep unpaired-looking reads in SAM/BAM input.  Ordinarily, if a read doesn't specify\nmate information (RNEXT field is * and/or PNEXT is 0) then the code that matches reads will immdeiately\ndiscard it.  Specifying this flag may cause large memory usage for some input files,\nbut may be necessary for some strangely formatted input files.  You'll also need to specify this\nflag for SAM/BAM files that were aligned by a single-end aligner.\n")), ToolInput(tag="in_index_dir", input_type=String(), position=0, doc=InputDocumentation(doc=""))], outputs=[ToolOutput(tag="out_filename_output_alignments", output_type=File(optional=True), selector=InputSelector(input_to_select="in_filename_output_alignments", type_hint=File()), doc=OutputDocumentation(doc="filename  output alignments to filename in SAM or BAM format, depending on the file extension or\nexplicit type specifier (see below).  Use a dash with an explicit type specifier to write to\nstdout, so for example -o -sam - would write SAM output to stdout")), ToolOutput(tag="out_so", output_type=File(optional=True), selector=InputSelector(input_to_select="in_so", type_hint=File()), doc=OutputDocumentation(doc="sort output file by alignment location")), ToolOutput(tag="out_sid", output_type=File(optional=True), selector=InputSelector(input_to_select="in_sid", type_hint=File()), doc=OutputDocumentation(doc="Specifies the sort intermediate directory.  When SNAP is sorting, it aligns the reads in the order in which they come in, and writes\nthe aligned reads in batches to a temporary file.  When the aligning is done, it does a merge sort from the temporary file into the\nfinal output file.  By default, the intermediate file is in the same directory as the output file, but for performance or space\nreasons, you might want to put it elsewhere.  If so, use this option."))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Snap_Aligner_Paired_V0_1_0().translate("wdl", allow_empty_container=True)

