from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, Int, String, File

Clean_Genes_V0_1_0 = CommandToolBuilder(tool="clean_genes", base_command=["clean_genes"], inputs=[ToolInput(tag="in_start", input_type=Boolean(optional=True), prefix="--start", doc=InputDocumentation(doc="Require conserved start codons (all species)")), ToolInput(tag="in_stop", input_type=Boolean(optional=True), prefix="--stop", doc=InputDocumentation(doc="Require conserved stop codons (all species)")), ToolInput(tag="in_splice", input_type=Boolean(optional=True), prefix="--splice", doc=InputDocumentation(doc="Require conserved splice sites (all species).  By default,\nonly GT-AG, GC-AG, and AT-AC splice sites are allowed (see also\n--splice-strict)")), ToolInput(tag="in_f_shift", input_type=Boolean(optional=True), prefix="--fshift", doc=InputDocumentation(doc="Require that no frame-shift gap is present in any species.  Frame\nshifts are evaluated with respect to the reference sequence.  Gaps\nthat have non-multiple-of-three lengths are allowed if\ncompensatory gaps occur nearby (see source code for details).")), ToolInput(tag="in_nonsense", input_type=Boolean(optional=True), prefix="--nonsense", doc=InputDocumentation(doc="Require that no premature stop codon is present in any species.")), ToolInput(tag="in_conserved", input_type=Boolean(optional=True), prefix="--conserved", doc=InputDocumentation(doc="Implies --start, --stop, --splice, --fshift, and --nonsense.\nRecommended option for cross-species analysis.")), ToolInput(tag="in_n_limit", input_type=Int(optional=True), prefix="--N-limit", doc=InputDocumentation(doc="Maximum fraction of bases aligned to CDSs that are Ns in any\nspecies (<f> must be between 0 and 1).  Default is 0.05.  Set to 1\nto allow any number of Ns.")), ToolInput(tag="in_clean_gaps", input_type=Boolean(optional=True), prefix="--clean-gaps", doc=InputDocumentation(doc="Require all cds gaps to be multiples of three in length.  Can be\nused with --conserved.")), ToolInput(tag="in_in_del_strict", input_type=Boolean(optional=True), prefix="--indel-strict", doc=InputDocumentation(doc="Implies --clean_gaps, usually used with --conserved.  Prohibits\noverlapping cds gaps in different sequences, gaps near cds\nboundaries, and gaps in the reference sequence within and between\nflanking features (splice sites, etc.; see code for details).\nDesigned for use in training a phylo-HMM with an indel model.")), ToolInput(tag="in_splice_strict", input_type=Boolean(optional=True), prefix="--splice-strict", doc=InputDocumentation(doc="Implies --splice.  Allow only GT-AG canonical splice sites.  Useful\nwhen training a gene finder with a simple model for splice sites.")), ToolInput(tag="in_group_by", input_type=String(optional=True), prefix="--groupby", doc=InputDocumentation(doc="Group features according to specified tag (default\n'transcript_id').  If any feature within a group fails, the\nentire group will be discarded.  By choosing to group features\naccording to different criteria, you can make the program\n'clean' the data set at different levels.  For example, to\nclean at the level of individual exons, add a tag like\n'exon_id' to indicate exons (see the program 'refeature'),\nand then invoke clean_genes with '--groupby exon_id'.")), ToolInput(tag="in_msa_format", input_type=File(optional=True), prefix="--msa-format", doc=InputDocumentation(doc="|PHYLIP|MPM|MAF|SS\nAlignment file format.  Default is to guess format from file\ncontents.")), ToolInput(tag="in_refseq", input_type=File(optional=True), prefix="--refseq", doc=InputDocumentation(doc="(Required with --msa-format MAF)  Complete reference\nsequence for alignment (FASTA format).")), ToolInput(tag="in_offset_five", input_type=Int(optional=True), prefix="--offset5", doc=InputDocumentation(doc="(Default 0)  Offset of canonical 'GT' with respect to boundary\non *intron side* of annotated 5' splice sites.  Useful with\nannotations that describe a window around the canonical splice site.")), ToolInput(tag="in_offset_three", input_type=Int(optional=True), prefix="--offset3", doc=InputDocumentation(doc="(Default 0)  Offset of canonical 'AG' with respect to boundary\non intron side of annotated 3' splice sites.")), ToolInput(tag="in_log", input_type=File(optional=True), prefix="--log", doc=InputDocumentation(doc="Write human-readable log to specified file.")), ToolInput(tag="in_machine_log", input_type=String(optional=True), prefix="--machine-log", doc=InputDocumentation(doc="Like --log, but produces more concise, machine-readable log.")), ToolInput(tag="in_stats", input_type=File(optional=True), prefix="--stats", doc=InputDocumentation(doc="Write statistics on retained and discarded features to specified file.")), ToolInput(tag="in_discards", input_type=File(optional=True), prefix="--discards", doc=InputDocumentation(doc="Write discarded features to specified file.")), ToolInput(tag="in_no_output", input_type=File(optional=True), prefix="--no-output", doc=InputDocumentation(doc="Suppress output of 'cleaned' features to stdout.  Useful if only\nlog file and/or stats are of interest.")), ToolInput(tag="in_gff_fname", input_type=String(), position=0, doc=InputDocumentation(doc="")), ToolInput(tag="in_msa_fname", input_type=String(), position=1, doc=InputDocumentation(doc=""))], outputs=[ToolOutput(tag="out_no_output", output_type=File(optional=True), selector=InputSelector(input_to_select="in_no_output", type_hint=File()), doc=OutputDocumentation(doc="Suppress output of 'cleaned' features to stdout.  Useful if only\nlog file and/or stats are of interest."))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Clean_Genes_V0_1_0().translate("wdl", allow_empty_container=True)

