from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Boolean, Int, String

Maf_Parse_V0_1_0 = CommandToolBuilder(tool="maf_parse", base_command=["maf_parse"], inputs=[ToolInput(tag="in_out_format", input_type=File(optional=True), prefix="--out-format", doc=InputDocumentation(doc="|PHYLIP|FASTA|MPM|SS\n(Default MAF).  Output file format.  SS format is only\navailable un-ordered.  Note that some options, which involve\nreversing alignments based on strand, or stripping gaps,\ncannot be output in MAF format and use FASTA by default.\nAlso note that when output format is not MAF, the entire\noutput must be loaded into memory.")), ToolInput(tag="in_pretty", input_type=Boolean(optional=True), prefix="--pretty", doc=InputDocumentation(doc="Pretty-print alignment (use '.' when character matches\ncorresponding character in first sequence).  Ignored if\n--out-format SS is selected.")), ToolInput(tag="in_start", input_type=Int(optional=True), prefix="--start", doc=InputDocumentation(doc="Start index of sub-alignment (indexing starts with 1).\nCoordinates are in terms of the reference sequence unless\nthe --no-refseq option is used, in which case they are in\nterms of alignment columns.  Default is 1.")), ToolInput(tag="in_end", input_type=Int(optional=True), prefix="--end", doc=InputDocumentation(doc="End index of sub-alignment.  Default is length of alignment.\nCoordinates defined as in --start option, above.")), ToolInput(tag="in_seqs", input_type=Int(optional=True), prefix="--seqs", doc=InputDocumentation(doc="Comma-separated list of sequences to include (default)\nexclude (if --exclude).  Indicate by sequence number or name\n(numbering starts with 1 and is evaluated *after* --order is\napplied).")), ToolInput(tag="in_exclude", input_type=Boolean(optional=True), prefix="--exclude", doc=InputDocumentation(doc="Exclude rather than include specified sequences.")), ToolInput(tag="in_order", input_type=String(optional=True), prefix="--order", doc=InputDocumentation(doc="Change order of rows in alignment to match sequence names\nspecified in name_list.  The first name in the alignment becomes\nthe reference sequence.")), ToolInput(tag="in_no_refseq", input_type=Boolean(optional=True), prefix="--no-refseq", doc=InputDocumentation(doc="Do not assume first sequence in MAF is refseq.  Instead, use\ncoordinates  given by absolute position in alignment (starting\nfrom 1).")), ToolInput(tag="in_split", input_type=Int(optional=True), prefix="--split", doc=InputDocumentation(doc="Split MAF into pieces by length, and puts output in\noutRootX.maf, where X=1,2,...,numPieces.  outRoot can be\nmodified with --out-root, and the minimum number of digits in X\ncan be modified with --out-root-digits.\nSplits between blocks, so that each output file does not exceed\nspecified length.  By default, length is counted by distance\nspanned in alignment by refseq, unless --no-refseq is specified.")), ToolInput(tag="in_out_root", input_type=File(optional=True), prefix="--out-root", doc=InputDocumentation(doc="Filename root for output files produced by --split (default\n'maf_parse').")), ToolInput(tag="in_out_root_digits", input_type=Int(optional=True), prefix="--out-root-digits", doc=InputDocumentation(doc="(for use with --split).  The minimum number of digits used to\nindex each output file produced by split.")), ToolInput(tag="in_features", input_type=File(optional=True), prefix="--features", doc=InputDocumentation(doc="Annotations file.  May be GFF, BED, or genepred format.\nCoordinates assumed to be in frame of first sequence of\nalignment (reference sequence).  By default, outputs subset of\nMAF which are labeled in annotations file.  But can be used with\n--by-category, --by-group, and/or --do-cats to split MAF by\nannotation type.  Or if used with --mask-features, is only used\nto determine regions to mask.  Implies --strip-i-lines,\n--strip-e-lines")), ToolInput(tag="in_by_category", input_type=Boolean(optional=True), prefix="--by-category", doc=InputDocumentation(doc="(Requires --features).  Split by category, as defined by\nannotations file and (optionally) category map (see --catmap).")), ToolInput(tag="in_do_cats", input_type=String(optional=True), prefix="--do-cats", doc=InputDocumentation(doc="(For use with --by-category) Output sub-alignments for only the\nspecified categories.")), ToolInput(tag="in_cat_map", input_type=File(optional=True), prefix="--catmap", doc=InputDocumentation(doc="|<string>\n(Optionally use with --by-category) Mapping of feature types to\ncategory numbers.  Can either give a filename or an 'inline'\ndescription of a simple category map, e.g.,\n--catmap 'NCATS = 3 ; CDS 1-3' or\n--catmap 'NCATS = 1; UTR 1'.")), ToolInput(tag="in_by_group", input_type=File(optional=True), prefix="--by-group", doc=InputDocumentation(doc="(Requires --features).  Split by groups in annotation file, as\ndefined by specified tag.")), ToolInput(tag="in_mask_bases", input_type=Int(optional=True), prefix="--mask-bases", doc=InputDocumentation(doc="Mask all bases with quality score <= n.  Note that n is in the\nsame units as displayed in the MAF (ranging from 0-9), and\nrepresents min(9, floor(PHRED_score/5)).  Bases without any\nquality score will not be masked.")), ToolInput(tag="in_masked_file", input_type=File(optional=True), prefix="--masked-file", doc=InputDocumentation(doc="(For use with --mask-bases).  Write a file containing all the\nregions masked for low quality.  The file will be in 0-based\ncoordinates relative to the refseq, with an additional column\ngiving the name of the species masked.  Note that low-quality bases\nmasked at alignment columns with a gap in the reference sequence\nmay not be represented in the output file.")), ToolInput(tag="in_mask_features", input_type=String(optional=True), prefix="--mask-features", doc=InputDocumentation(doc="(Requires --features).  Mask all bases annotated in features in the\ngiven species (can be a comma-delimited list of species).  Note that\ncoordinates are always in terms of refseq, even if a different species\nis being masked.")), ToolInput(tag="in_strip_i_lines", input_type=Boolean(optional=True), prefix="--strip-i-lines", doc=InputDocumentation(doc="Remove lines in MAF starting with i.")), ToolInput(tag="in_strip_e_lines", input_type=Boolean(optional=True), prefix="--strip-e-lines", doc=InputDocumentation(doc="Remove lines in MAF starting with e.")), ToolInput(tag="in_in_file", input_type=String(), position=0, doc=InputDocumentation(doc=""))], outputs=[ToolOutput(tag="out_out_format", output_type=File(optional=True), selector=InputSelector(input_to_select="in_out_format", type_hint=File()), doc=OutputDocumentation(doc="|PHYLIP|FASTA|MPM|SS\n(Default MAF).  Output file format.  SS format is only\navailable un-ordered.  Note that some options, which involve\nreversing alignments based on strand, or stripping gaps,\ncannot be output in MAF format and use FASTA by default.\nAlso note that when output format is not MAF, the entire\noutput must be loaded into memory.")), ToolOutput(tag="out_out_root", output_type=File(optional=True), selector=InputSelector(input_to_select="in_out_root", type_hint=File()), doc=OutputDocumentation(doc="Filename root for output files produced by --split (default\n'maf_parse').")), ToolOutput(tag="out_masked_file", output_type=File(optional=True), selector=InputSelector(input_to_select="in_masked_file", type_hint=File()), doc=OutputDocumentation(doc="(For use with --mask-bases).  Write a file containing all the\nregions masked for low quality.  The file will be in 0-based\ncoordinates relative to the refseq, with an additional column\ngiving the name of the species masked.  Note that low-quality bases\nmasked at alignment columns with a gap in the reference sequence\nmay not be represented in the output file."))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Maf_Parse_V0_1_0().translate("wdl", allow_empty_container=True)

