from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, String

Vcf2Kinship_V0_1_0 = CommandToolBuilder(tool="vcf2kinship", base_command=["vcf2kinship"], inputs=[ToolInput(tag="in_in_vcf", input_type=Boolean(optional=True), prefix="--inVcf", doc=InputDocumentation(doc=": Input VCF File")), ToolInput(tag="in_out", input_type=Boolean(optional=True), prefix="--out", doc=InputDocumentation(doc=": Output prefix for autosomal kinship calculation")), ToolInput(tag="in_x_hemi", input_type=Boolean(optional=True), prefix="--xHemi", doc=InputDocumentation(doc=": Calculate kinship using non-PAR region X chromosome")), ToolInput(tag="in_x_region", input_type=Boolean(optional=True), prefix="--xRegion", doc=InputDocumentation(doc=": Specify PAR region (default: hg19), can be build number\ne.g. hg38, b37; or specify region, e.g.\n'60001-2699520,154931044-155260560'")), ToolInput(tag="in_ped", input_type=Boolean(optional=True), prefix="--ped", doc=InputDocumentation(doc=": Use pedigree method or specify ped file for X\nchromosome analysis.")), ToolInput(tag="in_ibs", input_type=Boolean(optional=True), prefix="--ibs", doc=InputDocumentation(doc=": Use IBS method.")), ToolInput(tag="in_bn", input_type=Boolean(optional=True), prefix="--bn", doc=InputDocumentation(doc=": Use Balding-Nicols method.")), ToolInput(tag="in_pc_a", input_type=Boolean(optional=True), prefix="--pca", doc=InputDocumentation(doc=": Decomoposite calculated kinship matrix.")), ToolInput(tag="in_store_genotype", input_type=Boolean(optional=True), prefix="--storeGenotype", doc=InputDocumentation(doc=": Store genotye matrix (sample by genotype).")), ToolInput(tag="in_dosage", input_type=Boolean(optional=True), prefix="--dosage", doc=InputDocumentation(doc=": Specify which dosage tag to use (e.g. EC/DS). Typical\ndosage are between 0.0 and 2.0.")), ToolInput(tag="in_people_include_id", input_type=Boolean(optional=True), prefix="--peopleIncludeID", doc=InputDocumentation(doc=": List IDs of people that will be included in study")), ToolInput(tag="in_people_include_file", input_type=Boolean(optional=True), prefix="--peopleIncludeFile", doc=InputDocumentation(doc=": From given file, set IDs of people that will be\nincluded in study")), ToolInput(tag="in_people_exclude_id", input_type=Boolean(optional=True), prefix="--peopleExcludeID", doc=InputDocumentation(doc=": List IDs of people that will be included in study")), ToolInput(tag="in_people_exclude_file", input_type=Boolean(optional=True), prefix="--peopleExcludeFile", doc=InputDocumentation(doc=": From given file, set IDs of people that will be\nincluded in study")), ToolInput(tag="in_range_list", input_type=Boolean(optional=True), prefix="--rangeList", doc=InputDocumentation(doc=": Specify some ranges to use, please use chr:begin-end")), ToolInput(tag="in_min_maf", input_type=Boolean(optional=True), prefix="--minMAF", doc=InputDocumentation(doc=": Specify the minimum MAF threshold to be included in\ncalculating kinship.")), ToolInput(tag="in_max_miss", input_type=Boolean(optional=True), prefix="--maxMiss", doc=InputDocumentation(doc=": Specify the maximum allows missing rate to be inclued\nin calculating kinship.")), ToolInput(tag="in_mins_it_equal", input_type=Boolean(optional=True), prefix="--minSiteQual", doc=InputDocumentation(doc=": Specify minimum site qual")), ToolInput(tag="in_an_no", input_type=Boolean(optional=True), prefix="--anno", doc=InputDocumentation(doc=": Specify the annotation type to be included in\ncalculating kinship.")), ToolInput(tag="in_ming_q", input_type=Boolean(optional=True), prefix="--minGQ", doc=InputDocumentation(doc=": Specify the minimum genotype quality, otherwise marked\nas missing genotype")), ToolInput(tag="in_ming_d", input_type=Boolean(optional=True), prefix="--minGD", doc=InputDocumentation(doc=": Specify the minimum genotype depth, otherwise marked as\nmissing genotype")), ToolInput(tag="in_update_id", input_type=Boolean(optional=True), prefix="--update-id", doc=InputDocumentation(doc=": Update VCF sample id using given file (column 1 and 2\nare old and new id).")), ToolInput(tag="in_thread", input_type=Boolean(optional=True), prefix="--thread", doc=InputDocumentation(doc=": Specify number of parallel threads to speed up")), ToolInput(tag="in_markers_dot", input_type=String(), position=0, doc=InputDocumentation(doc="--xLabel: Specify X chromosome label (default: 23,X")), ToolInput(tag="in_format_dot", input_type=String(), position=0, doc=InputDocumentation(doc="--rangeFile: Specify the file containing ranges, please use"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Vcf2Kinship_V0_1_0().translate("wdl", allow_empty_container=True)

