from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Int, File, String, Directory, Float, Boolean

Crispressopooled_V0_1_0 = CommandToolBuilder(tool="CRISPRessoPooled", base_command=["CRISPRessoPooled"], inputs=[ToolInput(tag="in_analysis", input_type=Int(optional=True), prefix="-Analysis", doc=InputDocumentation(doc="CRISPR/Cas9 outcomes from POOLED deep sequencing data-\n)                                            )")), ToolInput(tag="in_fast_q_r_one", input_type=Int(optional=True), prefix="--fastq_r1", doc=InputDocumentation(doc="First fastq file (default: Fastq filename)")), ToolInput(tag="in_fast_q_r_two", input_type=Int(optional=True), prefix="--fastq_r2", doc=InputDocumentation(doc="Second fastq file for paired end reads (default: )")), ToolInput(tag="in_amplicons_file", input_type=File(optional=True), prefix="--amplicons_file", doc=InputDocumentation(doc="Amplicons description file. In particular, this file,\nis a tab delimited text file with up to 5 columns (2\nrequired): AMPLICON_NAME: an identifier for the\namplicon (must be unique) AMPLICON_SEQUENCE: amplicon\nsequence used in the design of the experiment\nsgRNA_SEQUENCE (OPTIONAL): sgRNA sequence used for\nthis amplicon without the PAM sequence. If more than\none separate them by commas and not spaces. If not\navailable enter NA. EXPECTED_AMPLICON_AFTER_HDR\n(OPTIONAL): expected amplicon sequence in case of HDR.\nIf not available enter NA. CODING_SEQUENCE (OPTIONAL):\nSubsequence(s) of the amplicon corresponding to coding\nsequences. If more than one separate them by commas\nand not spaces. If not available enter NA. (default: )")), ToolInput(tag="in_bowtie_two_index", input_type=Int(optional=True), prefix="--bowtie2_index", doc=InputDocumentation(doc="Basename of Bowtie2 index for the reference genome\n(default: )")), ToolInput(tag="in_gene_annotations", input_type=File(optional=True), prefix="--gene_annotations", doc=InputDocumentation(doc="Gene Annotation Table from UCSC Genome Browser Tables\n(http://genome.ucsc.edu/cgi-\nbin/hgTables?command=start), please select as table\n'knowGene', as output format 'all fields from selected\ntable' and as file returned 'gzip compressed'\n(default: )")), ToolInput(tag="in_n_processes", input_type=Int(optional=True), prefix="--n_processes", doc=InputDocumentation(doc="Specify the number of processes to use for the\nquantification. Please use with caution since\nincreasing this parameter will increase significantly\nthe memory required to run CRISPResso. (default: 1)")), ToolInput(tag="in_bowtie_two_options_string", input_type=Int(optional=True), prefix="--bowtie2_options_string", doc=InputDocumentation(doc="Override options for the Bowtie2 alignment command\n(default: -k 1 --end-to-end -N 0 --np 0 )")), ToolInput(tag="in_min_reads_to_use_region", input_type=Int(optional=True), prefix="--min_reads_to_use_region", doc=InputDocumentation(doc="Minimum number of reads that align to a region to\nperform the CRISPResso analysis (default: 1000)")), ToolInput(tag="in_min_average_read_quality", input_type=Int(optional=True), prefix="--min_average_read_quality", doc=InputDocumentation(doc="Minimum average quality score (phred33) to keep a read\n(default: 0)")), ToolInput(tag="in_min_single_bp_quality", input_type=Int(optional=True), prefix="--min_single_bp_quality", doc=InputDocumentation(doc="Minimum single bp score (phred33) to keep a read\n(default: 0)")), ToolInput(tag="in_min_identity_score", input_type=Int(optional=True), prefix="--min_identity_score", doc=InputDocumentation(doc="Min identity score for the alignment (default: 60.0)")), ToolInput(tag="in_name", input_type=String(optional=True), prefix="--name", doc=InputDocumentation(doc="Output name (default: )")), ToolInput(tag="in_trim_sequences", input_type=Directory(optional=True), prefix="--trim_sequences", doc=InputDocumentation(doc="Enable the trimming of Illumina adapters with\nTrimmomatic (default: False)")), ToolInput(tag="in_trim_mo_matic_options_string", input_type=Float(optional=True), prefix="--trimmomatic_options_string", doc=InputDocumentation(doc="Override options for Trimmomatic (default:\nILLUMINACLIP:/usr/local/lib/python2.7/site-\npackages/CRISPResso/data/NexteraPE-\nPE.fa:0:90:10:0:true MINLEN:40)")), ToolInput(tag="in_min_paired_end_reads_overlap", input_type=Int(optional=True), prefix="--min_paired_end_reads_overlap", doc=InputDocumentation(doc="Minimum required overlap length between two reads to\nprovide a confident overlap. (default: 4)")), ToolInput(tag="in_max_paired_end_reads_overlap", input_type=Int(optional=True), prefix="--max_paired_end_reads_overlap", doc=InputDocumentation(doc="parameter for the flash merging step, this parameter\nis the maximum overlap length expected in\napproximately 90% of read pairs. Please see the flash\nmanual for more information. (default: 100)")), ToolInput(tag="in_hide_mutations_outside_window_nhej", input_type=Boolean(optional=True), prefix="--hide_mutations_outside_window_NHEJ", doc=InputDocumentation(doc="This parameter allows to visualize only the mutations\noverlapping the cleavage site and used to classify a\nread as NHEJ. This parameter has no effect on the\nquanitification of the NHEJ. It may be helpful to mask\na pre-existing and known mutations or sequencing\nerrors outside the window used for quantification of\nNHEJ events. (default: False)")), ToolInput(tag="in_window_around_sg_rna", input_type=Int(optional=True), prefix="--window_around_sgrna", doc=InputDocumentation(doc="Window(s) in bp around the cleavage position (half on\non each side) as determined by the provide guide RNA\nsequence to quantify the indels. Any indels outside\nthis window are excluded. A value of 0 disables this\nfilter. (default: 1)")), ToolInput(tag="in_cleavage_offset", input_type=Int(optional=True), prefix="--cleavage_offset", doc=InputDocumentation(doc="Cleavage offset to use within respect to the 3' end of\nthe provided sgRNA sequence. Remember that the sgRNA\nsequence must be entered without the PAM. The default\nis -3 and is suitable for the SpCas9 system. For\nalternate nucleases, other cleavage offsets may be\nappropriate, for example, if using Cpf1 this parameter\nwould be set to 1. (default: -3)")), ToolInput(tag="in_exclude_bp_from_left", input_type=Int(optional=True), prefix="--exclude_bp_from_left", doc=InputDocumentation(doc="Exclude bp from the left side of the amplicon sequence\nfor the quantification of the indels (default: 15)")), ToolInput(tag="in_exclude_bp_from_right", input_type=Int(optional=True), prefix="--exclude_bp_from_right", doc=InputDocumentation(doc="Exclude bp from the right side of the amplicon\nsequence for the quantification of the indels\n(default: 15)")), ToolInput(tag="in_hdr_perfect_alignment_threshold", input_type=Float(optional=True), prefix="--hdr_perfect_alignment_threshold", doc=InputDocumentation(doc="Sequence homology % for an HDR occurrence (default:\n98.0)")), ToolInput(tag="in_ignore_substitutions", input_type=Boolean(optional=True), prefix="--ignore_substitutions", doc=InputDocumentation(doc="Ignore substitutions events for the quantification and\nvisualization (default: False)")), ToolInput(tag="in_ignore_insertions", input_type=Boolean(optional=True), prefix="--ignore_insertions", doc=InputDocumentation(doc="Ignore insertions events for the quantification and\nvisualization (default: False)")), ToolInput(tag="in_ignore_deletions", input_type=Boolean(optional=True), prefix="--ignore_deletions", doc=InputDocumentation(doc="Ignore deletions events for the quantification and\nvisualization (default: False)")), ToolInput(tag="in_needle_options_string", input_type=Float(optional=True), prefix="--needle_options_string", doc=InputDocumentation(doc="Override options for the Needle aligner (default:\n-gapopen=10 -gapextend=0.5 -awidth3=5000)")), ToolInput(tag="in_keep_intermediate", input_type=Boolean(optional=True), prefix="--keep_intermediate", doc=InputDocumentation(doc="Keep all the intermediate files (default: False)")), ToolInput(tag="in_dump", input_type=Boolean(optional=True), prefix="--dump", doc=InputDocumentation(doc="Dump numpy arrays and pandas dataframes to file for\ndebugging purposes (default: False)")), ToolInput(tag="in_save_also_png", input_type=Boolean(optional=True), prefix="--save_also_png", doc=InputDocumentation(doc="Save also .png images additionally to .pdf files\n(default: False)\n"))], outputs=[ToolOutput(tag="out_gene_annotations", output_type=File(optional=True), selector=InputSelector(input_to_select="in_gene_annotations", type_hint=File()), doc=OutputDocumentation(doc="Gene Annotation Table from UCSC Genome Browser Tables\n(http://genome.ucsc.edu/cgi-\nbin/hgTables?command=start), please select as table\n'knowGene', as output format 'all fields from selected\ntable' and as file returned 'gzip compressed'\n(default: )")), ToolOutput(tag="out_trim_sequences", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_trim_sequences", type_hint=File()), doc=OutputDocumentation(doc="Enable the trimming of Illumina adapters with\nTrimmomatic (default: False)"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Crispressopooled_V0_1_0().translate("wdl", allow_empty_container=True)

