from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Int, Boolean, Directory, String

Savage_V0_1_0 = CommandToolBuilder(tool="savage", base_command=["savage"], inputs=[ToolInput(tag="in_path_input_fastq_containing_singleend", input_type=File(optional=True), prefix="-s", doc=InputDocumentation(doc="path to input fastq containing single-end reads")), ToolInput(tag="in_p_one", input_type=Int(optional=True), prefix="-p1", doc=InputDocumentation(doc="path to input fastq containing paired-end reads (/1)")), ToolInput(tag="in_p_two", input_type=Int(optional=True), prefix="-p2", doc=InputDocumentation(doc="path to input fastq containing paired-end reads (/2)")), ToolInput(tag="in_min_overlap_len", input_type=Int(optional=True), prefix="--min_overlap_len", doc=InputDocumentation(doc="minimum overlap length required between reads")), ToolInput(tag="in_num_threads", input_type=Int(optional=True), prefix="--num_threads", doc=InputDocumentation(doc="allowed number of cores")), ToolInput(tag="in_split", input_type=Int(optional=True), prefix="--split", doc=InputDocumentation(doc="split the data set into patches s.t. 500 < coverage/split_num < 1000")), ToolInput(tag="in_rev_comp", input_type=Boolean(optional=True), prefix="--revcomp", doc=InputDocumentation(doc="use this option when paired-end input reads are in forward-reverse orientation;\nthis option will take reverse complements of /2 reads (specified with -p2)\nplease see the SAVAGE manual for more information about input read orientations")), ToolInput(tag="in_outdir", input_type=Directory(optional=True), prefix="--outdir", doc=InputDocumentation(doc="specify output directory")), ToolInput(tag="in_ref", input_type=String(optional=True), prefix="--ref", doc=InputDocumentation(doc="reference genome in fasta format")), ToolInput(tag="in_no_stage_a", input_type=Boolean(optional=True), prefix="--no_stage_a", doc=InputDocumentation(doc="skip Stage a (initial contig formation)")), ToolInput(tag="in_no_stage_b", input_type=Boolean(optional=True), prefix="--no_stage_b", doc=InputDocumentation(doc="skip Stage b (extending initial contigs)")), ToolInput(tag="in_no_stage_c", input_type=Boolean(optional=True), prefix="--no_stage_c", doc=InputDocumentation(doc="skip Stage c (merging maximized contigs into master strains)")), ToolInput(tag="in_no_overlaps", input_type=Boolean(optional=True), prefix="--no_overlaps", doc=InputDocumentation(doc="skip overlap computations (use existing overlaps file instead)")), ToolInput(tag="in_no_preprocessing", input_type=Boolean(optional=True), prefix="--no_preprocessing", doc=InputDocumentation(doc="skip preprocessing procedure (i.e. creating data patches)")), ToolInput(tag="in_no_assembly", input_type=Boolean(optional=True), prefix="--no_assembly", doc=InputDocumentation(doc="skip all assembly steps; only use this option when using --count_strains separate from assembly (e.g. on a denovo assembly)")), ToolInput(tag="in_count_strains", input_type=Boolean(optional=True), prefix="--count_strains", doc=InputDocumentation(doc="compute a lower bound on the number of strains in this sample; note: this requires a reference genome.")), ToolInput(tag="in_ignore_sub_reads", input_type=Boolean(optional=True), prefix="--ignore_subreads", doc=InputDocumentation(doc="ignore subread info from previous stage")), ToolInput(tag="in_merge_contigs", input_type=String(optional=True), prefix="--merge_contigs", doc=InputDocumentation(doc="specify maximal distance between contigs for merging into master strains (stage c)")), ToolInput(tag="in_min_clique_size", input_type=Int(optional=True), prefix="--min_clique_size", doc=InputDocumentation(doc="minimum clique size used during error correction")), ToolInput(tag="in_overlap_len_stage_c", input_type=String(optional=True), prefix="--overlap_len_stage_c", doc=InputDocumentation(doc="min_overlap_len used in stage c")), ToolInput(tag="in_contig_len_stage_c", input_type=Int(optional=True), prefix="--contig_len_stage_c", doc=InputDocumentation(doc="minimum contig length required for stage c input contigs")), ToolInput(tag="in_keep_branches", input_type=Boolean(optional=True), prefix="--keep_branches", doc=InputDocumentation(doc="disable merging along branches by removing them from the graph (stage b & c)")), ToolInput(tag="in_sfo_mm", input_type=Int(optional=True), prefix="--sfo_mm", doc=InputDocumentation(doc="input parameter -e=SFO_MM for sfo: maximal mismatch rate 1/SFO_MM")), ToolInput(tag="in_diploid", input_type=Boolean(optional=True), prefix="--diploid", doc=InputDocumentation(doc="use this option for diploid genome assembly")), ToolInput(tag="in_diploid_contig_len", input_type=Int(optional=True), prefix="--diploid_contig_len", doc=InputDocumentation(doc="minimum contig length required for diploid step contigs")), ToolInput(tag="in_diploid_overlap_len", input_type=String(optional=True), prefix="--diploid_overlap_len", doc=InputDocumentation(doc="min_overlap_len used in diploid assembly step")), ToolInput(tag="in_average_read_len", input_type=Int(optional=True), prefix="--average_read_len", doc=InputDocumentation(doc="average length of the input reads; will be computed from the input if not specified")), ToolInput(tag="in_no_filtering", input_type=Boolean(optional=True), prefix="--no_filtering", doc=InputDocumentation(doc="disable kallisto-based filtering of contigs")), ToolInput(tag="in_max_tip_len", input_type=Int(optional=True), prefix="--max_tip_len", doc=InputDocumentation(doc="maximum extension length for a sequence to be called a tip\n"))], outputs=[ToolOutput(tag="out_outdir", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_outdir", type_hint=File()), doc=OutputDocumentation(doc="specify output directory"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Savage_V0_1_0().translate("wdl", allow_empty_container=True)

