from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Array, Int, File, Boolean, String

Tombo_Estimate_Kmer_Reference_V0_1_0 = CommandToolBuilder(tool="tombo_estimate_kmer_reference", base_command=["tombo", "estimate_kmer_reference"], inputs=[ToolInput(tag="in_fast_five_based_irs", input_type=Array(t=Int(), optional=True), prefix="--fast5-basedirs", doc=InputDocumentation(doc="Directories containing fast5 files.")), ToolInput(tag="in_tombo_model_filename", input_type=File(optional=True), prefix="--tombo-model-filename", doc=InputDocumentation(doc="Tombo model for event-less resquiggle and significance\ntesting. If no model is provided the default DNA or\nRNA tombo model will be used.")), ToolInput(tag="in_estimate_mean", input_type=Boolean(optional=True), prefix="--estimate-mean", doc=InputDocumentation(doc="Use the mean instead of median for model level\nestimation. Note:This can cause poor fits due to\noutliers")), ToolInput(tag="in_km_er_specific_sd", input_type=Boolean(optional=True), prefix="--kmer-specific-sd", doc=InputDocumentation(doc="Estimate standard deviation for each k-mers")), ToolInput(tag="in_downstream_bases", input_type=String(optional=True), prefix="--downstream-bases", doc=InputDocumentation(doc="Downstream bases in k-mer. Default: 2")), ToolInput(tag="in_minimum_test_reads", input_type=Int(optional=True), prefix="--minimum-test-reads", doc=InputDocumentation(doc="Number of reads required at a position to perform\nsignificance testing or contribute to model\nestimation. Default: 10")), ToolInput(tag="in_coverage_threshold", input_type=Int(optional=True), prefix="--coverage-threshold", doc=InputDocumentation(doc="Maximum mean coverage per region when estimating k-mer\nmodel (limits compute time for deep samples). Default:\n100")), ToolInput(tag="in_minimum_km_er_observations", input_type=Int(optional=True), prefix="--minimum-kmer-observations", doc=InputDocumentation(doc="Number of each k-mer observations required in order to\nproduce a reference (genomic locations for standard\nreference and per-read for alternative reference).\nDefault: 5")), ToolInput(tag="in_multiprocess_region_size", input_type=Int(optional=True), prefix="--multiprocess-region-size", doc=InputDocumentation(doc="Size of regions over which to multiprocesses statistic\ncomputation. For very deep samples a smaller value is\nrecommmended in order to control memory consumption.\nDefault: 10000")), ToolInput(tag="in_processes", input_type=Int(optional=True), prefix="--processes", doc=InputDocumentation(doc="Number of processes. Default: 1")), ToolInput(tag="in_corrected_group", input_type=Int(optional=True), prefix="--corrected-group", doc=InputDocumentation(doc="FAST5 group created by resquiggle command. Default:\nRawGenomeCorrected_000")), ToolInput(tag="in_base_call_subgroups", input_type=Array(t=String(), optional=True), prefix="--basecall-subgroups", doc=InputDocumentation(doc="FAST5 subgroup(s) (under Analyses/[corrected-group])\ncontaining basecalls. Default: ['BaseCalled_template']")), ToolInput(tag="in_quiet", input_type=Boolean(optional=True), prefix="--quiet", doc=InputDocumentation(doc="Don't print status information.")), ToolInput(tag="in_individually_dot", input_type=String(), position=0, doc=InputDocumentation(doc="--upstream-bases {0,1,2,3,4}"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Tombo_Estimate_Kmer_Reference_V0_1_0().translate("wdl", allow_empty_container=True)

