from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import String, File, Boolean

Anvi_Push_V0_1_0 = CommandToolBuilder(tool="anvi_push", base_command=["anvi-push"], inputs=[ToolInput(tag="in_user", input_type=String(optional=True), prefix="--user", doc=InputDocumentation(doc="The user for an anvi'server.")), ToolInput(tag="in_api_url", input_type=String(optional=True), prefix="--api-url", doc=InputDocumentation(doc="Anvi'server url")), ToolInput(tag="in_project_name", input_type=File(optional=True), prefix="--project-name", doc=InputDocumentation(doc="Name of the project. Please choose a short but\ndescriptive name (so anvi'o can use it whenever she\nneeds to name an output file, or add a new table in a\ndatabase, or name her first born).")), ToolInput(tag="in_tree", input_type=String(optional=True), prefix="--tree", doc=InputDocumentation(doc="NEWICK formatted tree structure")), ToolInput(tag="in_items_order", input_type=File(optional=True), prefix="--items-order", doc=InputDocumentation(doc="A flat file that contains the order of items you wish\nthe display using the interactive interface. You may\nwant to use this if you have a specific order of items\nin your mind, and do not want to display a tree in the\nmiddle (or simply you don't have one). The file format\nis simple: each line should have an item name, and\nthere should be no header.")), ToolInput(tag="in_fast_a_file", input_type=File(optional=True), prefix="--fasta-file", doc=InputDocumentation(doc="A FASTA-formatted input file")), ToolInput(tag="in_view_data", input_type=File(optional=True), prefix="--view-data", doc=InputDocumentation(doc="A TAB-delimited file for view data")), ToolInput(tag="in_additional_layers", input_type=File(optional=True), prefix="--additional-layers", doc=InputDocumentation(doc="A TAB-delimited file for additional layers for splits.\nThe first column of this file must be split names, and\nthe remaining columns should be unique attributes. The\nfile does not need to contain all split names, or\nvalues for each split in every column. Anvi'o will try\nto deal with missing data nicely. Each column in this\nfile will be visualized as a new layer in the tree.")), ToolInput(tag="in_state", input_type=File(optional=True), prefix="--state", doc=InputDocumentation(doc="State file, you can export states from database using\nanvi-export-state program")), ToolInput(tag="in_description", input_type=File(optional=True), prefix="--description", doc=InputDocumentation(doc="A plain text file that contains some description about\nthe project. You can use Markdwon syntax. The\ndescription text will be rendered and shown in all\nrelevant interfaces, including the anvi'o interactive\ninterface, or anvi'o summary outputs.")), ToolInput(tag="in_bins", input_type=File(optional=True), prefix="--bins", doc=InputDocumentation(doc="Tab-delimited file, first column contains tree leaves\n(gene clusters, splits, contigs etc.) and second\ncolumn contains which Bin they belong.")), ToolInput(tag="in_bins_info", input_type=File(optional=True), prefix="--bins-info", doc=InputDocumentation(doc="Additional information for bins. The file must contain\nthree TAB-delimited columns, where the first one must\nbe a unique bin name, the second should be a 'source',\nand the last one should be a 7 character HTML color\ncode (i.e., '#424242'). Source column must contain\ninformation about the origin of the bin. If these bins\nare automatically identified by a program like\nCONCOCT, this column could contain the program name\nand version. The source information will be associated\nwith the bin in various interfaces so in a sense it is\nnot *that* critical what it says there, but on the\nother hand it is, becuse we should also think about\npeople who may end up having to work with what we put\ntogether later.")), ToolInput(tag="in_delete_if_exists", input_type=Boolean(optional=True), prefix="--delete-if-exists", doc=InputDocumentation(doc="Be bold (at your own risk), and delete if exists."))], outputs=[ToolOutput(tag="out_project_name", output_type=File(optional=True), selector=InputSelector(input_to_select="in_project_name", type_hint=File()), doc=OutputDocumentation(doc="Name of the project. Please choose a short but\ndescriptive name (so anvi'o can use it whenever she\nneeds to name an output file, or add a new table in a\ndatabase, or name her first born)."))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Anvi_Push_V0_1_0().translate("wdl", allow_empty_container=True)

