from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, Directory, File, Int

Gtftk_Control_List_V0_1_0 = CommandToolBuilder(tool="gtftk_control_list", base_command=["gtftk", "control_list"], inputs=[ToolInput(tag="in_in_file", input_type=Boolean(optional=True), prefix="--in-file", doc=InputDocumentation(doc="A two columns tab-file. See notes. (default: None)")), ToolInput(tag="in_reference_gene_file", input_type=Boolean(optional=True), prefix="--reference-gene-file", doc=InputDocumentation(doc="The file containing the reference gene list (1 column, transcript ids). No header. (default: None)")), ToolInput(tag="in_out_dir", input_type=Directory(optional=True), prefix="--out-dir", doc=InputDocumentation(doc="Name of the output directory. (default: control_list)")), ToolInput(tag="in_log_two", input_type=Boolean(optional=True), prefix="--log2", doc=InputDocumentation(doc="If selected, data will be log transformed. (default: False)")), ToolInput(tag="in_pseudo_count", input_type=Boolean(optional=True), prefix="--pseudo-count", doc=InputDocumentation(doc="The value for a pseudo-count to be added. (default: 0)")), ToolInput(tag="in_page_width", input_type=File(optional=True), prefix="--page-width", doc=InputDocumentation(doc="Output pdf file width (e.g. 7 inches). (default: None)")), ToolInput(tag="in_page_height", input_type=File(optional=True), prefix="--page-height", doc=InputDocumentation(doc="Output file height (e.g. 5 inches). (default: None)")), ToolInput(tag="in_page_format", input_type=File(optional=True), prefix="--page-format", doc=InputDocumentation(doc="Output file format. (default: pdf)")), ToolInput(tag="in_dpi", input_type=Boolean(optional=True), prefix="--dpi", doc=InputDocumentation(doc="Dpi to use. (default: 300)")), ToolInput(tag="in_skip_first", input_type=Boolean(optional=True), prefix="--skip-first", doc=InputDocumentation(doc="Indicates that infile hase a header. (default: False)")), ToolInput(tag="in_rug", input_type=Boolean(optional=True), prefix="--rug", doc=InputDocumentation(doc="Add rugs to the diagram. (default: False)")), ToolInput(tag="in_jitter", input_type=Boolean(optional=True), prefix="--jitter", doc=InputDocumentation(doc="Add jittered points. (default: False)")), ToolInput(tag="in_user_img_file", input_type=Boolean(optional=True), prefix="--user-img-file", doc=InputDocumentation(doc="Provide an alternative path for the image. (default: None)")), ToolInput(tag="in_set_colors", input_type=Boolean(optional=True), prefix="--set-colors", doc=InputDocumentation(doc="Colors for the two sets (comma-separated). (default: #b2df8a,#6a3d9a)")), ToolInput(tag="in_verbosity", input_type=Boolean(optional=True), prefix="--verbosity", doc=InputDocumentation(doc="Set output verbosity ([0-3]). (default: 0)")), ToolInput(tag="in_no_date", input_type=File(optional=True), prefix="--no-date", doc=InputDocumentation(doc="Do not add date to output file names. (default: False)")), ToolInput(tag="in_add_chr", input_type=Boolean(optional=True), prefix="--add-chr", doc=InputDocumentation(doc="Add 'chr' to chromosome names before printing output. (default: False)")), ToolInput(tag="in_tmp_dir", input_type=Boolean(optional=True), prefix="--tmp-dir", doc=InputDocumentation(doc="Keep all temporary files into this folder. (default: None)")), ToolInput(tag="in_keep_all", input_type=Boolean(optional=True), prefix="--keep-all", doc=InputDocumentation(doc="Try to keep all temporary files even if process does not terminate normally. (default: False)")), ToolInput(tag="in_logger_file", input_type=Boolean(optional=True), prefix="--logger-file", doc=InputDocumentation(doc="Stores the arguments passed to the command into a file. (default: None)")), ToolInput(tag="in_write_message_to_file", input_type=Boolean(optional=True), prefix="--write-message-to-file", doc=InputDocumentation(doc="Store all message into a file. (default: None)")), ToolInput(tag="in_optional_arguments", input_type=Int(), position=0, doc=InputDocumentation(doc="optional arguments:"))], outputs=[ToolOutput(tag="out_out_dir", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_out_dir", type_hint=File()), doc=OutputDocumentation(doc="Name of the output directory. (default: control_list)")), ToolOutput(tag="out_page_width", output_type=File(optional=True), selector=InputSelector(input_to_select="in_page_width", type_hint=File()), doc=OutputDocumentation(doc="Output pdf file width (e.g. 7 inches). (default: None)")), ToolOutput(tag="out_page_height", output_type=File(optional=True), selector=InputSelector(input_to_select="in_page_height", type_hint=File()), doc=OutputDocumentation(doc="Output file height (e.g. 5 inches). (default: None)")), ToolOutput(tag="out_page_format", output_type=File(optional=True), selector=InputSelector(input_to_select="in_page_format", type_hint=File()), doc=OutputDocumentation(doc="Output file format. (default: pdf)")), ToolOutput(tag="out_no_date", output_type=File(optional=True), selector=InputSelector(input_to_select="in_no_date", type_hint=File()), doc=OutputDocumentation(doc="Do not add date to output file names. (default: False)"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Gtftk_Control_List_V0_1_0().translate("wdl", allow_empty_container=True)

