from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, String, Boolean, Int

Rsat_Position_Analysis_V0_1_0 = CommandToolBuilder(tool="rsat_position_analysis", base_command=["rsat", "position-analysis"], inputs=[ToolInput(tag="in_specified_standard_input", input_type=File(optional=True), prefix="-i", doc=InputDocumentation(doc="if not specified, the standard input is used.\nThis allows to place the command within a pipe.")), ToolInput(tag="in_seqtype", input_type=String(optional=True), prefix="-seqtype", doc=InputDocumentation(doc="|any\nSequence type")), ToolInput(tag="in_last", input_type=Boolean(optional=True), prefix="-last", doc=InputDocumentation(doc="#\nStop after # sequences (for quick testing)\nThe possibility to limit the analysis to a few sequences\n(e.g.50 of the input file can be useful for fine-tuning\nthe options and ensuring that the result will appear as\nrequested. Note that this option should not be used for\nthe final analysis, since the significance drastically\ndepends on the number of input sequences.")), ToolInput(tag="in_skip", input_type=Boolean(optional=True), prefix="-skip", doc=InputDocumentation(doc="#\nSkip the first # sequences.")), ToolInput(tag="in_first", input_type=Boolean(optional=True), prefix="-first", doc=InputDocumentation(doc="#\nFirst sequence to analyze.\nThe option '-first n' has the same effect as '-skip x',\nwhere x = n-1.  These two somewhat redundant options\nexist because in some cases it is easier to indicate the\nfirst sequences (e.g. for naming result files according\nto the first and last sequences analyzed).\nThe options -skip, -last, -first and -seqnb are\nconvenient for analyzing successive slices of at\ndifferent depths in the sequence file (e.g. from 1 to\n1000, then from 5001 to 6000, then from 10001 to 11000,\netc). This allows to test if the position bias of a\npattern is consistent over the whole sequence file, or if\nit is confined to the top sequences.\nAn example of useful application of this slice-based\nanalysis is to estimate the number of relevant sequences\nin peaks returned by applying peak-calling programs on\nChIP-seq reads.  Peak sequences should be sorted by\ndecreasing score values.")), ToolInput(tag="in_seq_nb", input_type=Boolean(optional=True), prefix="-seqnb", doc=InputDocumentation(doc="#\nMaximal number of sequences to analyze. The program stops\nreading the sequences after having reached this\nnumber. This option is convenient for analyzing\nsuccessive slices of at different depths in the sequence\nfile (e.g. for peak sequences sorted by decreasing\nscores.\nExample: analyze N sequences starting from the X_th one.\n-first X -seqnnb N\nWhen used without -skip or -first, the option -seqnb\ngives the same result as -last.")), ToolInput(tag="in_mask", input_type=String(optional=True), prefix="-mask", doc=InputDocumentation(doc="|lower\nMask lower or uppercases, respecively, i.e. replace\nselected case by N characters.")), ToolInput(tag="in_format", input_type=File(optional=True), prefix="-format", doc=InputDocumentation(doc="file format. Must be followed by one of the\nfollowing options:\nfasta (default)\nwconsensus\nIG\nfilelist\nraw\nSee below for the description of these formats.")), ToolInput(tag="in_outputfile_returns_list", input_type=File(optional=True), prefix="-o", doc=InputDocumentation(doc="outputfile. Returns a list of the oligonucleotides\nencountered in the sequences, with their frequencies.")), ToolInput(tag="in__verbose_level", input_type=Boolean(optional=True), prefix="-v", doc=InputDocumentation(doc="#    verbose level.")), ToolInput(tag="in_oligonucleotide_length", input_type=Boolean(optional=True), prefix="-l", doc=InputDocumentation(doc="oligonucleotide length.")), ToolInput(tag="in_ci", input_type=Boolean(optional=True), prefix="-ci", doc=InputDocumentation(doc="window interval (default 20 bases).\nThe width of the position windows (in number of bases)")), ToolInput(tag="in_origin", input_type=String(optional=True), prefix="-origin", doc=InputDocumentation(doc="| center | end\nReference for calculating positions.\nThe value should be chosen according to the sequence\ntype. For instance:\n-origin start for downstream sequences\n-origin end for promoter sequences\n-origin center can be useful for ChIP-seq peaks, which\ncan have variable lengths, but are supposed to\nbe more or less centred on the TF binding\nqsites.")), ToolInput(tag="in_offset", input_type=Boolean(optional=True), prefix="-offset", doc=InputDocumentation(doc="Add an offset to site positions. The offset value must\nbe an Integer number (positive, null or\nnegative). This option allows to select an arbitrary\nposition as origin.\nExample: the option I<-offset -100> can be used to\nspecify the transcription start site (TSS) as origin,\nin a collection of promoter sequences including 100\nresidues downstream of the TSS.\nNote: in previous versions, -origin was used to\nspecify both the reference point and the offset. Since\nMarch 2010, the offset is specified with the option\n-offset.")), ToolInput(tag="in_group_rc", input_type=Boolean(optional=True), prefix="-grouprc", doc=InputDocumentation(doc="group reverse complement pairs")), ToolInput(tag="in_nogroup_rc", input_type=Boolean(optional=True), prefix="-nogrouprc", doc=InputDocumentation(doc="do not group reverse complement pairs")), ToolInput(tag="in_sort", input_type=Boolean(optional=True), prefix="-sort", doc=InputDocumentation(doc="sort oligonucleotides according to the chi2\nstatistics, reflecting the level of bias in\ndistribution profile")), ToolInput(tag="in_one_str", input_type=Boolean(optional=True), prefix="-1str", doc=InputDocumentation(doc="inactivates the summation of occurrences on both\nstrands.")), ToolInput(tag="in_two_str", input_type=Boolean(optional=True), prefix="-2str", doc=InputDocumentation(doc="oligonucleotide occurrences found on both stands are\nsummed.")), ToolInput(tag="in_noov", input_type=Boolean(optional=True), prefix="-noov", doc=InputDocumentation(doc="no overlap\noverlapping occurrences of the same pattern are\nonly taken into account once")), ToolInput(tag="in_return", input_type=File(optional=True), prefix="-return", doc=InputDocumentation(doc="supported fields:\nhtml    HTML formatted results + index file\ndistrib k-mer occurrences per position window\nexp_occ expected occurrences for each window\nchi     chi-square value\nrank    rank of the pattern according to the sorting\ncriterion\ngraphs  one graph file per oligont profile\nclusters run k-mer clustering (regroup k-mers having\nsimilar occurrence profiles).\nSeveral return fields can be entered, separated by\ncommas.")), ToolInput(tag="in_task", input_type=Boolean(optional=True), prefix="-task", doc=InputDocumentation(doc="task1,task2,...\nSupported tasks:\npos: analyze oligonucleotide occurrences ('main'\nanalysis)\nclusters: cluster oligonucleotides according to\ntheir occurrence profiles.\nmatrices: build position-specific scoring matrices\nfrom clusters of oligonucleotides occupying\nsimilar positions.\nWhen different values would be entered for\noptions -min_clust_nb and -max_clust_nb,\nmatrices and logos are only generated for the\nmaximal number of cluster (max_clust_nb), for\nthe sake of time efficiency and readability of\nthe HTML reports.\ngraphs: generate a XY graph with the occurernce\nprofile of each oligonucleotide.\nindex: generate the HTML file with links to the\nresult files.\nall: run all the tasks above\nRun specific tasks rather than the whole analysis.\nThis option mainly serve to avoir re-running the whole\nsequence analysis for re-running cluster analysis with\nmodified parameters.")), ToolInput(tag="in_markov", input_type=Int(optional=True), prefix="-markov", doc=InputDocumentation(doc="Order for the Markov model use to compute\nposition-specific expected word frequencies.\nBy default, expected frequencies are estimated by\nassuming a homogeneous repartition of k-mer\noccurrences (as in the original publication by van\nHelden et al., 2000).\nAlternatively, the option -markov indicates that\nexpected frequency of each k-mer in each position\nwindow will be calculated on the basis of the observed\nfrequencies for smaller words.")), ToolInput(tag="in_lth_chi", input_type=Boolean(optional=True), prefix="-lth_chi", doc=InputDocumentation(doc="#      lower threshold on chi2\nreturn only words with a chi2 value > #")), ToolInput(tag="in_lth_sig", input_type=Boolean(optional=True), prefix="-lth_sig", doc=InputDocumentation(doc="#      lower threshold on significance\nreturn only words with a sig value > #")), ToolInput(tag="in_lth_occ", input_type=Boolean(optional=True), prefix="-lth_occ", doc=InputDocumentation(doc="#      lower threshold on occurrences\nreturn only words with a number of occurrences > #")), ToolInput(tag="in_uth_rank", input_type=Boolean(optional=True), prefix="-uth_rank", doc=InputDocumentation(doc="#     upper threshold on rank\nreturn maximum # words")), ToolInput(tag="in_max_graphs", input_type=Boolean(optional=True), prefix="-max_graphs", doc=InputDocumentation(doc="#   maximal number of graphs to export")), ToolInput(tag="in_pl", input_type=File(optional=True), prefix="-pl", doc=InputDocumentation(doc="A file containing a selection of patterns.  The\nanalysis is then restricted to these patterns.  The\nfirst word of each new line is considered as a new\npattern.  A score can be associated to each pattern\nwith the option -sc.")), ToolInput(tag="in_sc", input_type=Boolean(optional=True), prefix="-sc", doc=InputDocumentation(doc="#   score column\n(only valid whith the option -pl)\nThe column containing a score value for each pattern\nof the pattern file")), ToolInput(tag="in_min_pos", input_type=Boolean(optional=True), prefix="-minpos", doc=InputDocumentation(doc="#\nminimal position to take into account for the\nchi-square calculation This value must be a multiple\nof the window interval.")), ToolInput(tag="in_max_pos", input_type=Boolean(optional=True), prefix="-maxpos", doc=InputDocumentation(doc="#\nmaximal position to take into account for the\nchi-square calculation This value must be a multiple\nof the window interval.")), ToolInput(tag="in_no_check", input_type=Boolean(optional=True), prefix="-nocheck", doc=InputDocumentation(doc="do not check the applicability condition on the\nchi-square. By default, the program checks that each\nwindow has at least 5 observations. The chi-square is\nbracketed for words which do not fill this\nconditions. It is now recognized that this condition\nis too restrictive, and that the chi2 is still valid\nwith smaller clas effective. We allow to suppress the\nchecking, but the responsibility is left to the user,\nto decide whether the chi2 is or not significant.")), ToolInput(tag="in_no_filter", input_type=Boolean(optional=True), prefix="-nofilter", doc=InputDocumentation(doc="Do not discard oligos which do not fit the condition\nof applicability. Instead, mark them by including the\nchi2 value in curly brackets.")), ToolInput(tag="in_header", input_type=Int(optional=True), prefix="-header", doc=InputDocumentation(doc="| midfloor | min | max | interval\nInformation to display in column headers of the\ndistributions.\nmid: class middle points (centers)\nmidfloor: floor of class middle points (centers).\nthis avoids to display non-integer mid points\n(e.g. for interval 1-10 the midpoint is 5.5)\nmin: class min\nmax: class max\ninterval: class intervals: [min,max]")), ToolInput(tag="in_top_seq_for_matrices", input_type=Boolean(optional=True), prefix="-top_seq_for_matrices", doc=InputDocumentation(doc="#\nSelect the top # sequences for building position-specific\nscoring matrices (PSSM).  This argument is passed to the\nprogram matrix-from-pattern.\nThis option does not affect the number of sequences used\nfor motif discovery (detecting positionally biased\noligonucelotides), but it can be useful to test the\nsecond phase of motif discovery: extracion of matrices\nfrom the selected oligonucleotides.")), ToolInput(tag="in_img_format", input_type=Boolean(optional=True), prefix="-img_format", doc=InputDocumentation(doc="Image format (this parameter is passed to XYgraph).")), ToolInput(tag="in_title", input_type=Boolean(optional=True), prefix="-title", doc=InputDocumentation(doc="Title for the index table and position profile plots.")), ToolInput(tag="in_clust_method", input_type=Boolean(optional=True), prefix="-clust_method", doc=InputDocumentation(doc="Agglomeration rule for the hierarchical\nclustering. Supported: complete, average, single,\nward. Default: complete.")), ToolInput(tag="in_clust_nb", input_type=Boolean(optional=True), prefix="-clust_nb", doc=InputDocumentation(doc="#\nNumber of clusters (default: 8).")), ToolInput(tag="in_clust_suffix", input_type=Boolean(optional=True), prefix="-clust_suffix", doc=InputDocumentation(doc="Suffix to append to the cluster file and the directory\ncontianing cluster graphics. Default: 'clusters'.\n")), ToolInput(tag="in_position_analysis", input_type=String(), position=0, doc=InputDocumentation(doc="1998 by Jacques van Helden (Jacques.van-Helden@univ-amu.fr)")), ToolInput(tag="in_sequences", input_type=String(), position=0, doc=InputDocumentation(doc="pattern-discovery"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Rsat_Position_Analysis_V0_1_0().translate("wdl", allow_empty_container=True)

