class: CommandLineTool
id: anvi_script_predict_CPR_genomes.cwl
inputs:
- id: in_contigs_db
  doc: "Anvi'o contigs database generated by 'anvi-gen-\ncontigs-database' (default:\
    \ None)"
  type: string?
  inputBinding:
    prefix: --contigs-db
- id: in_profile_db
  doc: "Anvi'o profile database (default: None)"
  type: string?
  inputBinding:
    prefix: --profile-db
- id: in_collection_name
  doc: 'Collection name. (default: None)'
  type: string?
  inputBinding:
    prefix: --collection-name
- id: in_list_collections
  doc: 'Show available collections and exit. (default: False)'
  type: boolean?
  inputBinding:
    prefix: --list-collections
- id: in_report_only_cpr
  doc: "Include only bins that look like CPR genomes.\n(default: False)"
  type: boolean?
  inputBinding:
    prefix: --report-only-cpr
- id: in_min_genome_size
  doc: "Minimum genome size to consider for CPR in Mbp.\nDefault is 0.500000"
  type: long?
  inputBinding:
    prefix: --min-genome-size
- id: in_min_percent_completion
  doc: "Minimum percent completion estimate based on anvi'o\ndefault single-copy gene\
    \ collections. Default is 50"
  type: long?
  inputBinding:
    prefix: --min-percent-completion
- id: in_max_percent_redundancy
  doc: "Maxumum percent redundancy or single-copy genes in an\nanvi'o bin, or a genome\
    \ to consider for\nclassification. The default is 30"
  type: long?
  inputBinding:
    prefix: --max-percent-redundancy
- id: in_min_class_probability
  doc: "If the classification confidence is below this don't\nbother. Default is 75."
  type: long?
  inputBinding:
    prefix: --min-class-probability
- id: in_output_file
  doc: 'File path to store results. (default: None)'
  type: File?
  inputBinding:
    prefix: --output-file
- id: in_just_do_it
  doc: "Don't bother me with questions or warnings, just do\nit. (default: False)"
  type: boolean?
  inputBinding:
    prefix: --just-do-it
- id: in_classifier
  doc: 'optional arguments:'
  type: string
  inputBinding:
    position: 2
outputs:
- id: out_stdout
  doc: Standard output stream
  type: stdout
- id: out_output_file
  doc: 'File path to store results. (default: None)'
  type: File?
  outputBinding:
    glob: $(inputs.in_output_file)
hints:
- class: DockerRequirement
  dockerPull: quay.io/biocontainers/anvio-minimal:7--py_0
cwlVersion: v1.1
baseCommand:
- anvi-script-predict-CPR-genomes
