from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Int, String, Boolean, Array

Metagene_Count_V0_1_0 = CommandToolBuilder(tool="metagene_count", base_command=["metagene", "count"], inputs=[ToolInput(tag="in_min_counts", input_type=Int(optional=True), prefix="--min_counts", doc=InputDocumentation(doc="Minimum counts required in normalization region to be\nincluded in metagene average (Default: 10)")), ToolInput(tag="in_normalize_over", input_type=Int(optional=True), prefix="--normalize_over", doc=InputDocumentation(doc="N  Portion of each window against which its individual\nraw count profile will be normalized. Specify two\nintegers, in nucleotide distance from landmark\n(negative for upstream, positive for downstream.\nSurround negative numbers with quotes.). (Default: 20\n50)")), ToolInput(tag="in_norm_region", input_type=Int(optional=True), prefix="--norm_region", doc=InputDocumentation(doc="N     Deprecated. Use ``--normalize_over`` instead.\nFormerly, Portion of each window against which its\nindividual raw count profile will be normalized.\nSpecify two integers, in nucleotide distance, from 5'\nend of window. (Default: 70 100)")), ToolInput(tag="in_landmark", input_type=String(optional=True), prefix="--landmark", doc=InputDocumentation(doc="Name of landmark at zero point, optional.")), ToolInput(tag="in_use_mean", input_type=Boolean(optional=True), prefix="--use_mean", doc=InputDocumentation(doc="If supplied, use columnwise mean to generate profile\n(Default: use median)")), ToolInput(tag="in_keep", input_type=Boolean(optional=True), prefix="--keep", doc=InputDocumentation(doc="Save intermediate count files. Useful for additional\ncomputations (Default: False)")), ToolInput(tag="in_quiet", input_type=Boolean(optional=True), prefix="--quiet", doc=InputDocumentation(doc="Suppress all warning messages. Cannot use with '-v'.")), ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="--verbose", doc=InputDocumentation(doc="Increase verbosity. With '-v', show every warning.\nWith '-vv', turn warnings into exceptions. Cannot use\nwith '-q'. (Default: show each type of warning once)")), ToolInput(tag="in_count_files", input_type=Array(t=String(), optional=True), prefix="--count_files", doc=InputDocumentation(doc="One or more count or alignment file(s) from a single\nsample or set of samples to be pooled.")), ToolInput(tag="in_count_file_format", input_type=String(optional=True), prefix="--countfile_format", doc=InputDocumentation(doc="Format of file containing alignments or counts\n(Default: BAM)")), ToolInput(tag="in_sum", input_type=String(optional=True), prefix="--sum", doc=InputDocumentation(doc="Sum used in normalization of counts and RPKM/RPNT\ncalculations (Default: total mapped reads/counts in\ndataset)")), ToolInput(tag="in_min_length", input_type=Int(optional=True), prefix="--min_length", doc=InputDocumentation(doc="Minimum read length required to be included (BAM &\nbowtie files only. Default: 25)")), ToolInput(tag="in_max_length", input_type=Int(optional=True), prefix="--max_length", doc=InputDocumentation(doc="Maximum read length permitted to be included (BAM &\nbowtie files only. Default: 100)")), ToolInput(tag="in_maxmem", input_type=Int(optional=True), prefix="--maxmem", doc=InputDocumentation(doc="Maximum desired memory footprint in MB to devote to\nBigBed/BigWig files. May be exceeded by large queries.\n(Default: 0, No maximum)")), ToolInput(tag="in_big_genome", input_type=Boolean(optional=True), prefix="--big_genome", doc=InputDocumentation(doc="Use slower but memory-efficient implementation for big\ngenomes or for memory-limited computers. For wiggle &\nbowtie files only.")), ToolInput(tag="in_five_prime_variable", input_type=Boolean(optional=True), prefix="--fiveprime_variable", doc=InputDocumentation(doc="Map read alignment to a variable offset from 5'\nposition of read, with offset determined by read\nlength. Requires `--offset` below")), ToolInput(tag="in_five_prime", input_type=Boolean(optional=True), prefix="--fiveprime", doc=InputDocumentation(doc="Map read alignment to 5' position.")), ToolInput(tag="in_three_prime", input_type=Boolean(optional=True), prefix="--threeprime", doc=InputDocumentation(doc="Map read alignment to 3' position")), ToolInput(tag="in_center", input_type=Boolean(optional=True), prefix="--center", doc=InputDocumentation(doc="Subtract N positions from each end of read, and add\n1/(length-N), to each remaining position, where N is\nspecified by `--nibble`")), ToolInput(tag="in_offset", input_type=Int(optional=True), prefix="--offset", doc=InputDocumentation(doc="For `--fiveprime` or `--threeprime`, provide an\ninteger representing the offset into the read,\nstarting from either the 5' or 3' end, at which data\nshould be plotted. For `--fiveprime_variable`, provide\nthe filename of a two-column tab-delimited text file,\nin which first column represents read length or the\nspecial keyword `'default'`, and the second column\nrepresents the offset from the five prime end of that\nread length at which the read should be mapped.\n(Default: 0)")), ToolInput(tag="in_nibble", input_type=Int(optional=True), prefix="--nibble", doc=InputDocumentation(doc="For use with `--center` only. nt to remove from each\nend of read before mapping (Default: 0)")), ToolInput(tag="in_fig_format", input_type=String(optional=True), prefix="--figformat", doc=InputDocumentation(doc="File format for figure(s); Default: png)")), ToolInput(tag="in_fig_size", input_type=String(optional=True), prefix="--figsize", doc=InputDocumentation(doc="N         Figure width and height, in inches. (Default: use\nmatplotlibrc params)")), ToolInput(tag="in_title", input_type=String(optional=True), prefix="--title", doc=InputDocumentation(doc="Base title for plot(s).")), ToolInput(tag="in_cmap", input_type=Int(optional=True), prefix="--cmap", doc=InputDocumentation(doc="Matplotlib color map from which palette will be made\n(e.g. 'Blues','autumn','Set1'; default: use colors\nfrom ``--stylesheet`` if given, or color cycle in\nmatplotlibrc)")), ToolInput(tag="in_dpi", input_type=Int(optional=True), prefix="--dpi", doc=InputDocumentation(doc="Figure resolution (Default: 150)")), ToolInput(tag="in_stylesheet", input_type=Boolean(optional=True), prefix="--stylesheet", doc=InputDocumentation(doc="{Solarize_Light2,_classic_test_patch,bmh,classic,dark_background,fast,fivethirtyeight,ggplot,grayscale,seaborn,seaborn-bright,seaborn-colorblind,seaborn-dark,seaborn-dark-palette,seaborn-darkgrid,seaborn-deep,seaborn-muted,seaborn-notebook,seaborn-paper,seaborn-pastel,seaborn-poster,seaborn-talk,seaborn-ticks,seaborn-white,seaborn-whitegrid,tableau-colorblind10}\nUse this matplotlib stylesheet instead of matplotlibrc\nparams\n")), ToolInput(tag="in_roi_file", input_type=String(), position=0, doc=InputDocumentation(doc="Text file containing maximal spanning windows and\noffsets, generated by the ``generate`` subprogram.")), ToolInput(tag="in_out_base", input_type=String(), position=1, doc=InputDocumentation(doc="Basename for output files"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Metagene_Count_V0_1_0().translate("wdl", allow_empty_container=True)

