from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Array, String, Boolean, Int

Samtools_Split_V0_1_0 = CommandToolBuilder(tool="samtools_split", base_command=["samtools", "split"], inputs=[ToolInput(tag="in_output_filename_format", input_type=File(optional=True), prefix="-f", doc=InputDocumentation(doc="output filename format string ['%*_%#.%.']")), ToolInput(tag="in_put_reads_rg", input_type=File(optional=True), prefix="-u", doc=InputDocumentation(doc="put reads with no RG tag or an unrecognised RG tag in FILE1")), ToolInput(tag="in_override_header_only", input_type=Array(t=String(), optional=True), prefix="-h", doc=InputDocumentation(doc="override the header with FILE2 (-u file only)")), ToolInput(tag="in_verbose_output", input_type=Boolean(optional=True), prefix="-v", doc=InputDocumentation(doc="verbose output")), ToolInput(tag="in_no_pg", input_type=Boolean(optional=True), prefix="--no-PG", doc=InputDocumentation(doc="do not add a PG line")), ToolInput(tag="in_input_fmt_option", input_type=File(optional=True), prefix="--input-fmt-option", doc=InputDocumentation(doc="[=VAL]\nSpecify a single input file format option in the form\nof OPTION or OPTION=VALUE")), ToolInput(tag="in_output_fmt", input_type=String(optional=True), prefix="--output-fmt", doc=InputDocumentation(doc="[,OPT[=VAL]]...\nSpecify output format (SAM, BAM, CRAM)")), ToolInput(tag="in_output_fmt_option", input_type=File(optional=True), prefix="--output-fmt-option", doc=InputDocumentation(doc="[=VAL]\nSpecify a single output file format option in the form\nof OPTION or OPTION=VALUE")), ToolInput(tag="in_reference", input_type=File(optional=True), prefix="--reference", doc=InputDocumentation(doc="Reference sequence FASTA FILE [null]")), ToolInput(tag="in_threads", input_type=Int(optional=True), prefix="--threads", doc=InputDocumentation(doc="Number of additional threads to use [0]")), ToolInput(tag="in_write_index", input_type=Boolean(optional=True), prefix="--write-index", doc=InputDocumentation(doc="Automatically index the output files [off]")), ToolInput(tag="in_verbosity", input_type=Int(optional=True), prefix="--verbosity", doc=InputDocumentation(doc="Set level of verbosity"))], outputs=[ToolOutput(tag="out_output_filename_format", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output_filename_format", type_hint=File()), doc=OutputDocumentation(doc="output filename format string ['%*_%#.%.']")), ToolOutput(tag="out_output_fmt_option", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output_fmt_option", type_hint=File()), doc=OutputDocumentation(doc="[=VAL]\nSpecify a single output file format option in the form\nof OPTION or OPTION=VALUE"))], container="quay.io/biocontainers/samtools:1.12--hd5e65b6_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Samtools_Split_V0_1_0().translate("wdl")

