from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Directory, Boolean, File

Battenberg_Pl_V0_1_0 = CommandToolBuilder(tool="battenberg.pl", base_command=["battenberg.pl"], inputs=[ToolInput(tag="in_outdir", input_type=Directory(optional=True), prefix="-outdir", doc=InputDocumentation(doc="Folder to output result to.")), ToolInput(tag="in_reference", input_type=Boolean(optional=True), prefix="-reference", doc=InputDocumentation(doc="Path to reference genome index file *.fai")), ToolInput(tag="in_tumba_m", input_type=Boolean(optional=True), prefix="-tumbam", doc=InputDocumentation(doc="Path to tumour bam file\n- when '-a' defined sample name")), ToolInput(tag="in_normb_am", input_type=Boolean(optional=True), prefix="-normbam", doc=InputDocumentation(doc="Path to normal bam file\n- when '-a' defined sample name")), ToolInput(tag="in_gender", input_type=Boolean(optional=True), prefix="-gender", doc=InputDocumentation(doc="Gender, XX, XY or L (see -gl)")), ToolInput(tag="in_impute_info", input_type=Boolean(optional=True), prefix="-impute-info", doc=InputDocumentation(doc="Location of the impute info file")), ToolInput(tag="in_thousand_genomes_loc", input_type=Boolean(optional=True), prefix="-thousand-genomes-loc", doc=InputDocumentation(doc="Location of the directory containing 1k genomes data")), ToolInput(tag="in_ignore_contigs_file", input_type=Boolean(optional=True), prefix="-ignore-contigs-file", doc=InputDocumentation(doc="File containing contigs to ignore\n- specifically male sex chromosome, mitochondria and non primary contigs")), ToolInput(tag="in_allele_counts", input_type=Boolean(optional=True), prefix="-allele-counts", doc=InputDocumentation(doc="Provide a tar.gz containing the impute allele counts")), ToolInput(tag="in_prob_loci", input_type=Boolean(optional=True), prefix="-prob-loci", doc=InputDocumentation(doc="Location of prob_loci.txt file [included in release]")), ToolInput(tag="in_min_bq_all_count", input_type=Boolean(optional=True), prefix="-min-bq-allcount", doc=InputDocumentation(doc="Minimum base quality to permit allele counting [20]")), ToolInput(tag="in_segmentation_gamma", input_type=Boolean(optional=True), prefix="-segmentation-gamma", doc=InputDocumentation(doc="Segmentation gamma [10]")), ToolInput(tag="in_phasing_gamma", input_type=Boolean(optional=True), prefix="-phasing-gamma", doc=InputDocumentation(doc="Phasing gamma [1]")), ToolInput(tag="in_clonality_distance", input_type=Boolean(optional=True), prefix="-clonality-distance", doc=InputDocumentation(doc="Clonality distance [0]")), ToolInput(tag="in_as_cat_distance", input_type=Boolean(optional=True), prefix="-ascat-distance", doc=InputDocumentation(doc="ASCAT distance [1]")), ToolInput(tag="in_balanced_threshold", input_type=Boolean(optional=True), prefix="-balanced-threshold", doc=InputDocumentation(doc="Balanced threshold [0.51]")), ToolInput(tag="in_platform_gamma", input_type=Boolean(optional=True), prefix="-platform-gamma", doc=InputDocumentation(doc="Platform gamma [1]")), ToolInput(tag="in_min_ploidy", input_type=Boolean(optional=True), prefix="-min-ploidy", doc=InputDocumentation(doc="Min ploidy [1.6]")), ToolInput(tag="in_max_ploidy", input_type=Boolean(optional=True), prefix="-max-ploidy", doc=InputDocumentation(doc="Max ploidy [4.8]")), ToolInput(tag="in_min_rho", input_type=Boolean(optional=True), prefix="-min-rho", doc=InputDocumentation(doc="Min Rho [0.1]")), ToolInput(tag="in_min_goodness_of_fit", input_type=Boolean(optional=True), prefix="-min-goodness-of-fit", doc=InputDocumentation(doc="Min goodness of fit [0.63]")), ToolInput(tag="in_species", input_type=Boolean(optional=True), prefix="-species", doc=InputDocumentation(doc="Reference species []")), ToolInput(tag="in_assembly", input_type=Boolean(optional=True), prefix="-assembly", doc=InputDocumentation(doc="Reference assembly []")), ToolInput(tag="in_protocol", input_type=Boolean(optional=True), prefix="-protocol", doc=InputDocumentation(doc="Sequencing protocol [WGS]")), ToolInput(tag="in_platform", input_type=Boolean(optional=True), prefix="-platform", doc=InputDocumentation(doc="Sequencing platfrom [ILLUMINA]")), ToolInput(tag="in_gender_loci", input_type=Boolean(optional=True), prefix="-genderloci", doc=InputDocumentation(doc="List of gender loci, required when '-ge L' [share/gender/GRCh37d5_Y.loci]\n- these are loci that will not present at all in a female sample")), ToolInput(tag="in_threads", input_type=Boolean(optional=True), prefix="-threads", doc=InputDocumentation(doc="Number of threads allowed on this machine (default 1)")), ToolInput(tag="in_logs", input_type=Boolean(optional=True), prefix="-logs", doc=InputDocumentation(doc="Location to write logs (default is ./logs)")), ToolInput(tag="in_process", input_type=Boolean(optional=True), prefix="-process", doc=InputDocumentation(doc="Only process this step then exit, optionally set -index")), ToolInput(tag="in_index", input_type=Boolean(optional=True), prefix="-index", doc=InputDocumentation(doc="Optionally restrict '-p' to single job")), ToolInput(tag="in_jobs", input_type=Boolean(optional=True), prefix="-jobs", doc=InputDocumentation(doc="Declare with -p to determine how many jobs are needed for this step")), ToolInput(tag="in_man", input_type=Boolean(optional=True), prefix="-man", doc=InputDocumentation(doc="Full documentation.")), ToolInput(tag="in_is_male", input_type=Boolean(optional=True), prefix="-is-male", doc=InputDocumentation(doc="Flag the sample as male, otherwise female is assumed")), ToolInput(tag="in_file_dot", input_type=File(), position=0, doc=InputDocumentation(doc="-ignore-contigs-file"))], outputs=[ToolOutput(tag="out_outdir", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_outdir", type_hint=File()), doc=OutputDocumentation(doc="Folder to output result to."))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Battenberg_Pl_V0_1_0().translate("wdl", allow_empty_container=True)

