from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, String, Boolean, Int

Srnapipe_V0_1_0 = CommandToolBuilder(tool="srnapipe", base_command=["srnapipe"], inputs=[ToolInput(tag="in_fast_q", input_type=File(optional=True), prefix="--fastq", doc=InputDocumentation(doc="Fastq file to process")), ToolInput(tag="in_fast_q_n", input_type=String(optional=True), prefix="--fastq_n", doc=InputDocumentation(doc="Name of the content to process")), ToolInput(tag="in_ref", input_type=File(optional=True), prefix="--ref", doc=InputDocumentation(doc="Fasta file containing the reference genome")), ToolInput(tag="in_transcripts", input_type=File(optional=True), prefix="--transcripts", doc=InputDocumentation(doc="Fasta file containing the transcripts")), ToolInput(tag="in_te", input_type=File(optional=True), prefix="--TE", doc=InputDocumentation(doc="Fasta file containing the transposable elements")), ToolInput(tag="in_mirnas", input_type=File(optional=True), prefix="--miRNAs", doc=InputDocumentation(doc="Fasta file containing the miRNAs")), ToolInput(tag="in_snrnas", input_type=File(optional=True), prefix="--snRNAs", doc=InputDocumentation(doc="Fasta file containing the snRNAs")), ToolInput(tag="in_rrnas", input_type=File(optional=True), prefix="--rRNAs", doc=InputDocumentation(doc="Fasta file containing the rRNAs")), ToolInput(tag="in_trnas", input_type=File(optional=True), prefix="--tRNAs", doc=InputDocumentation(doc="Fasta file containing the tRNAs")), ToolInput(tag="in_html", input_type=Boolean(optional=True), prefix="--html", doc=InputDocumentation(doc="Main HTML file where results will be displayed")), ToolInput(tag="in_dir", input_type=Boolean(optional=True), prefix="--dir", doc=InputDocumentation(doc="Folder where results will be stored")), ToolInput(tag="in_min", input_type=Int(optional=True), prefix="--min", doc=InputDocumentation(doc="Minimum read size (default: 18)")), ToolInput(tag="in_max", input_type=Int(optional=True), prefix="--max", doc=InputDocumentation(doc="Maximum read size (default: 29)")), ToolInput(tag="in_si_min", input_type=Int(optional=True), prefix="--si_min", doc=InputDocumentation(doc="Lower bound of siRNA range (default: 21)")), ToolInput(tag="in_si_max", input_type=Int(optional=True), prefix="--si_max", doc=InputDocumentation(doc="Higher bound of siRNA range (default: 21)")), ToolInput(tag="in_pi_min", input_type=Int(optional=True), prefix="--pi_min", doc=InputDocumentation(doc="Lower bound of piRNA range (default: 23)")), ToolInput(tag="in_pi_max", input_type=Int(optional=True), prefix="--pi_max", doc=InputDocumentation(doc="Higher bound of piRNA range (default: 29)")), ToolInput(tag="in_mis", input_type=Int(optional=True), prefix="--mis", doc=InputDocumentation(doc="Maximal genome mismatches (default: 0)")), ToolInput(tag="in_mist_e", input_type=Int(optional=True), prefix="--misTE", doc=InputDocumentation(doc="Maximal TE mismatches (default: 3)")), ToolInput(tag="in_ppp_on", input_type=String(optional=True), prefix="--PPPon", doc=InputDocumentation(doc="Ping-pong partners (default: true)")), ToolInput(tag="in_threads", input_type=Int(optional=True), prefix="--threads", doc=InputDocumentation(doc="Number of threads used (default: 1)"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Srnapipe_V0_1_0().translate("wdl", allow_empty_container=True)

