class: CommandLineTool
id: GetProteinCodingSequence.cwl
inputs:
- id: in_input
  doc: Input file(s) in fasta format. generated by RiboCode
  type: File?
  inputBinding:
    prefix: --input
- id: in_coordinate_file
  doc: "The file should contain the coordinate of start and\nstop codon. Generated\
    \ by\nOutputTranscriptInfo.py.[required]"
  type: File?
  inputBinding:
    prefix: --coordinateFile
- id: in_select_trans_list
  doc: "Selected transcript list used for metagene\nanalysis.This files requires the\
    \ first column must be\nthe transcript ID  with a column name."
  type: string?
  inputBinding:
    prefix: --select_trans_list
- id: in_left_co_or
  doc: "The left position (codon level) of the sequence you\nwanna to extract.[required]"
  type: string?
  inputBinding:
    prefix: --left_coor
- id: in_right_co_or
  doc: "The right position (codon level) of the sequence you\nwanna to extract[required]"
  type: string?
  inputBinding:
    prefix: --right_coor
- id: in_ot_put_prefix
  doc: Prefix of output files.[required]
  type: string?
  inputBinding:
    prefix: --otput_prefix
- id: in_mode
  doc: Mode for sequence extraction.[whole or local]
  type: string?
  inputBinding:
    prefix: --mode
- id: in_table
  doc: "Which genetic code to use. default is the standard\ncode.table = 1, is the\
    \ standard table, ref: https://ww\nw.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi"
  type: long?
  inputBinding:
    prefix: --table
- id: in_id_type
  doc: "define the id type users input. the default is\ntranscript id, if not, will\
    \ be transformed into\ntranscript id. default=transcript_id\n"
  type: string?
  inputBinding:
    prefix: --id-type
outputs:
- id: out_stdout
  doc: Standard output stream
  type: stdout
hints:
- class: DockerRequirement
  dockerPull: quay.io/biocontainers/ribominer:0.2.3.1--pyh3252c3a_0
cwlVersion: v1.1
baseCommand:
- GetProteinCodingSequence
