version 1.0

task GetProteinCodingSequence {
  input {
    File? input_files_fasta
    File? coordinate_file
    String? select_trans_list
    String? left_co_or
    String? right_co_or
    String? ot_put_prefix
    String? mode
    Int? table
    String? id_type
  }
  command <<<
    GetProteinCodingSequence \
      ~{if defined(input_files_fasta) then ("--input " +  '"' + input_files_fasta + '"') else ""} \
      ~{if defined(coordinate_file) then ("--coordinateFile " +  '"' + coordinate_file + '"') else ""} \
      ~{if defined(select_trans_list) then ("--select_trans_list " +  '"' + select_trans_list + '"') else ""} \
      ~{if defined(left_co_or) then ("--left_coor " +  '"' + left_co_or + '"') else ""} \
      ~{if defined(right_co_or) then ("--right_coor " +  '"' + right_co_or + '"') else ""} \
      ~{if defined(ot_put_prefix) then ("--otput_prefix " +  '"' + ot_put_prefix + '"') else ""} \
      ~{if defined(mode) then ("--mode " +  '"' + mode + '"') else ""} \
      ~{if defined(table) then ("--table " +  '"' + table + '"') else ""} \
      ~{if defined(id_type) then ("--id-type " +  '"' + id_type + '"') else ""}
  >>>
  runtime {
    docker: "quay.io/biocontainers/ribominer:0.2.3.1--pyh3252c3a_0"
  }
  parameter_meta {
    input_files_fasta: "Input file(s) in fasta format. generated by RiboCode"
    coordinate_file: "The file should contain the coordinate of start and\\nstop codon. Generated by\\nOutputTranscriptInfo.py.[required]"
    select_trans_list: "Selected transcript list used for metagene\\nanalysis.This files requires the first column must be\\nthe transcript ID  with a column name."
    left_co_or: "The left position (codon level) of the sequence you\\nwanna to extract.[required]"
    right_co_or: "The right position (codon level) of the sequence you\\nwanna to extract[required]"
    ot_put_prefix: "Prefix of output files.[required]"
    mode: "Mode for sequence extraction.[whole or local]"
    table: "Which genetic code to use. default is the standard\\ncode.table = 1, is the standard table, ref: https://ww\\nw.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi"
    id_type: "define the id type users input. the default is\\ntranscript id, if not, will be transformed into\\ntranscript id. default=transcript_id\\n"
  }
  output {
    File out_stdout = stdout()
  }
}