from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Array, Int, String, File, Boolean, Float

Assemblepairs_Py_Sequential_V0_1_0 = CommandToolBuilder(tool="AssemblePairs.py_sequential", base_command=["AssemblePairs.py", "sequential"], inputs=[ToolInput(tag="in_ordered_list_containingheadprimary", input_type=Array(t=Int(), optional=True), prefix="-1", doc=InputDocumentation(doc="An ordered list of FASTA/FASTQ files containing\nhead/primary sequences. (default: None)")), ToolInput(tag="in_ordered_list_containingtailsecondary", input_type=Array(t=Int(), optional=True), prefix="-2", doc=InputDocumentation(doc="An ordered list of FASTA/FASTQ files containing\ntail/secondary sequences. (default: None)")), ToolInput(tag="in_explicit_output_file", input_type=Array(t=String(), optional=True), prefix="-o", doc=InputDocumentation(doc="Explicit output file name(s). Note, this argument\ncannot be used with the --failed, --outdir, or\n--outname arguments. If unspecified, then the output\nfilename will be based on the input filename(s).\n(default: None)")), ToolInput(tag="in_outdir", input_type=File(optional=True), prefix="--outdir", doc=InputDocumentation(doc="Specify to changes the output directory to the\nlocation specified. The input file directory is used\nif this is not specified. (default: None)")), ToolInput(tag="in_out_name", input_type=File(optional=True), prefix="--outname", doc=InputDocumentation(doc="Changes the prefix of the successfully processed\noutput file to the string specified. May not be\nspecified with multiple input files. (default: None)")), ToolInput(tag="in_log", input_type=File(optional=True), prefix="--log", doc=InputDocumentation(doc="Specify to write verbose logging to a file. May not be\nspecified with multiple input files. (default: None)")), ToolInput(tag="in_failed", input_type=Boolean(optional=True), prefix="--failed", doc=InputDocumentation(doc="If specified create files containing records that fail\nprocessing. (default: False)")), ToolInput(tag="in_fast_a", input_type=Boolean(optional=True), prefix="--fasta", doc=InputDocumentation(doc="Specify to force output as FASTA rather than FASTQ.\n(default: None)")), ToolInput(tag="in_delim", input_type=String(optional=True), prefix="--delim", doc=InputDocumentation(doc="DELIMITER DELIMITER\nA list of the three delimiters that separate\nannotation blocks, field names and values, and values\nwithin a field, respectively. (default: ('|', '=',\n','))")), ToolInput(tag="in_nproc", input_type=Int(optional=True), prefix="--nproc", doc=InputDocumentation(doc="The number of simultaneous computational processes to\nexecute (CPU cores to utilized). (default: 8)")), ToolInput(tag="in_coord", input_type=String(optional=True), prefix="--coord", doc=InputDocumentation(doc="The format of the sequence identifier which defines\nshared coordinate information across paired ends.\n(default: presto)")), ToolInput(tag="in_rc", input_type=String(optional=True), prefix="--rc", doc=InputDocumentation(doc="Specify which read to reverse complement before\nstitching. (default: tail)")), ToolInput(tag="in_one_f", input_type=Array(t=String(), optional=True), prefix="--1f", doc=InputDocumentation(doc="Specify annotation fields to copy from head records\ninto assembled record. (default: None)")), ToolInput(tag="in_two_f", input_type=Array(t=String(), optional=True), prefix="--2f", doc=InputDocumentation(doc="Specify annotation fields to copy from tail records\ninto assembled record. (default: None)")), ToolInput(tag="in_alpha", input_type=Float(optional=True), prefix="--alpha", doc=InputDocumentation(doc="Significance threshold for de novo paired-end\nassembly. (default: 1e-05)")), ToolInput(tag="in_max_error", input_type=Int(optional=True), prefix="--maxerror", doc=InputDocumentation(doc="Maximum allowable error rate for de novo assembly.\n(default: 0.3)")), ToolInput(tag="in_min_len", input_type=Int(optional=True), prefix="--minlen", doc=InputDocumentation(doc="Minimum sequence length to scan for overlap in de novo\nassembly. (default: 8)")), ToolInput(tag="in_maxlen", input_type=Int(optional=True), prefix="--maxlen", doc=InputDocumentation(doc="Maximum sequence length to scan for overlap in de novo\nassembly. (default: 1000)")), ToolInput(tag="in_scan_rev", input_type=Boolean(optional=True), prefix="--scanrev", doc=InputDocumentation(doc="If specified, scan past the end of the tail sequence\nin de novo assembly to allow the head sequence to\noverhang the end of the tail sequence. (default:\nFalse)")), ToolInput(tag="in_fasta_file_containing", input_type=File(optional=True), prefix="-r", doc=InputDocumentation(doc="A FASTA file containing the reference sequence\ndatabase. (default: None)")), ToolInput(tag="in_mini_dent", input_type=Float(optional=True), prefix="--minident", doc=InputDocumentation(doc="Minimum identity of the assembled sequence required to\ncall a valid reference guided assembly (between 0 and\n1). (default: 0.5)")), ToolInput(tag="in_evalue", input_type=Float(optional=True), prefix="--evalue", doc=InputDocumentation(doc="Minimum E-value for reference alignment for both the\nhead and tail sequence. (default: 1e-05)")), ToolInput(tag="in_max_hits", input_type=Int(optional=True), prefix="--maxhits", doc=InputDocumentation(doc="Maximum number of hits from the reference alignment to\ncheck for matching head and tail sequence assignments.\n(default: 100)")), ToolInput(tag="in_fill", input_type=Boolean(optional=True), prefix="--fill", doc=InputDocumentation(doc="Specify to change the behavior of inserted characters\nwhen the head and tail sequences do not overlap during\nreference guided assembly. If specified, this will\nresult in inserted of the V region reference sequence\ninstead of a sequence of Ns in the non-overlapping\nregion. Warning: you could end up making chimeric\nsequences by using this option. (default: False)")), ToolInput(tag="in_aligner", input_type=String(optional=True), prefix="--aligner", doc=InputDocumentation(doc="The local alignment tool to use. Must be one blastn\n(blast+ nucleotide) or usearch (ublast algorithm).\n(default: usearch)")), ToolInput(tag="in_exec", input_type=File(optional=True), prefix="--exec", doc=InputDocumentation(doc="The name or location of the aligner executable file\n(blastn or usearch). Defaults to the name specified by\nthe --aligner argument. (default: None)")), ToolInput(tag="in_db_exec", input_type=File(optional=True), prefix="--dbexec", doc=InputDocumentation(doc="The name or location of the executable file that\nbuilds the reference database. This defaults to\nmakeblastdb when blastn is specified to the --aligner\nargument, and usearch when usearch is specified.\n(default: None)\n"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Assemblepairs_Py_Sequential_V0_1_0().translate("wdl", allow_empty_container=True)

