from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Directory, Boolean, Int, Float, String

Matam_Assembly_Py_V0_1_0 = CommandToolBuilder(tool="matam_assembly.py", base_command=["matam_assembly.py"], inputs=[ToolInput(tag="in_input_fast_q", input_type=File(optional=True), prefix="--input_fastq", doc=InputDocumentation(doc="Input reads file (fasta or fastq format)")), ToolInput(tag="in_ref_db", input_type=File(optional=True), prefix="--ref_db", doc=InputDocumentation(doc="MATAM ref db. Default is\n$MATAM_DIR/db/SILVA_128_SSURef_NR95")), ToolInput(tag="in_out_dir", input_type=Directory(optional=True), prefix="--out_dir", doc=InputDocumentation(doc="Output directory.Default will be\n'matam_assembly'")), ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="--verbose", doc=InputDocumentation(doc="Increase verbosity")), ToolInput(tag="in_cpu", input_type=Int(optional=True), prefix="--cpu", doc=InputDocumentation(doc="Max number of CPU to use. Default is 1 cpu")), ToolInput(tag="in_max_memory", input_type=Int(optional=True), prefix="--max_memory", doc=InputDocumentation(doc="Maximum memory to use (in MBi). Default is\n10000 MBi")), ToolInput(tag="in_best", input_type=Int(optional=True), prefix="--best", doc=InputDocumentation(doc="Get up to --best good alignments per read.\nDefault is 10")), ToolInput(tag="in_evalue", input_type=Int(optional=True), prefix="--evalue", doc=InputDocumentation(doc="Max e-value to keep an alignment for. Default\nis 1e-05")), ToolInput(tag="in_score_threshold", input_type=Float(optional=True), prefix="--score_threshold", doc=InputDocumentation(doc="Score threshold (real between 0 and 1). Default\nis 0.9")), ToolInput(tag="in_straight_mode", input_type=Boolean(optional=True), prefix="--straight_mode", doc=InputDocumentation(doc="Use straight mode filtering. Default is\ngeometric mode")), ToolInput(tag="in_coverage_threshold", input_type=Int(optional=True), prefix="--coverage_threshold", doc=InputDocumentation(doc="Ref coverage threshold. By default set to 0 to\ndesactivate filtering")), ToolInput(tag="in_min_identity", input_type=Float(optional=True), prefix="--min_identity", doc=InputDocumentation(doc="Minimum identity of an overlap between 2 reads.\nDefault is 1.0")), ToolInput(tag="in_min_overlap_length", input_type=Int(optional=True), prefix="--min_overlap_length", doc=InputDocumentation(doc="Minimum length of an overlap. Default is 50")), ToolInput(tag="in_min_read_node", input_type=Int(optional=True), prefix="--min_read_node", doc=InputDocumentation(doc="Minimum number of read to keep a node. Default\nis 1")), ToolInput(tag="in_min_overlap_edge", input_type=Int(optional=True), prefix="--min_overlap_edge", doc=InputDocumentation(doc="Minimum number of overlap to keep an edge.\nDefault is 1")), ToolInput(tag="in_seed", input_type=Int(optional=True), prefix="--seed", doc=InputDocumentation(doc="Seed to initialize random generator. Default is\npicking seed from system time")), ToolInput(tag="in_quorum", input_type=Float(optional=True), prefix="--quorum", doc=InputDocumentation(doc="Quorum for LCA computing. Has to be between\n0.51 and 1. Default is 0.51")), ToolInput(tag="in_assembler", input_type=String(optional=True), prefix="--assembler", doc=InputDocumentation(doc="Select the assembler to be used. Default is SGA")), ToolInput(tag="in_read_correction", input_type=String(optional=True), prefix="--read_correction", doc=InputDocumentation(doc="Set the assembler read correction step. 'auto'\nmeans assemble the components with read\ncorrection enabled when the components coverage\nare sufficient (20X) and assemble the other\ncomponents without read correction. Default is\nauto")), ToolInput(tag="in_contigs_binning", input_type=Boolean(optional=True), prefix="--contigs_binning", doc=InputDocumentation(doc="Experimental. Perform contigs binning during")), ToolInput(tag="in_perform_taxonomic_assignment", input_type=Boolean(optional=True), prefix="--perform_taxonomic_assignment", doc=InputDocumentation(doc="Do the taxonomic assignment")), ToolInput(tag="in_training_model", input_type=String(optional=True), prefix="--training_model", doc=InputDocumentation(doc="The training model used for taxonomic\nassignment. Default is 16srrna")), ToolInput(tag="in_rdp_cut_off", input_type=Float(optional=True), prefix="--rdp_cutoff", doc=InputDocumentation(doc="Sequences assigned (by RDP) with a confidence\nscore < 0.8 (at genus level) will be tagged as\nan artificial 'unclassified' taxon")), ToolInput(tag="in_keep_tmp", input_type=Boolean(optional=True), prefix="--keep_tmp", doc=InputDocumentation(doc="Do not remove tmp files")), ToolInput(tag="in_debug", input_type=Boolean(optional=True), prefix="--debug", doc=InputDocumentation(doc="Output debug infos")), ToolInput(tag="in_resume_from", input_type=String(optional=True), prefix="--resume_from", doc=InputDocumentation(doc="Try to resume from given step. Steps are:\nreads_mapping, alignments_filtering,\noverlap_graph_building, graph_compaction,\ncontigs_assembly, scaffolding,\nabundance_calculation, taxonomic_assignment")), ToolInput(tag="in_filter_only", input_type=Boolean(optional=True), prefix="--filter_only", doc=InputDocumentation(doc="Perform only the first step of MATAM (i.e Reads\nmapping against ref db with sortmerna to filter\nthe reads). Relevant options for this step\ncorrespond to the 'Read mapping' section.\n")), ToolInput(tag="in_scaffolding_dot", input_type=String(), position=0, doc=InputDocumentation(doc="--min_scaffold_length MIN_SCAFFOLD_LENGTH                              Filter out small scaffolds"))], outputs=[ToolOutput(tag="out_out_dir", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_out_dir", type_hint=File()), doc=OutputDocumentation(doc="Output directory.Default will be\n'matam_assembly'"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Matam_Assembly_Py_V0_1_0().translate("wdl", allow_empty_container=True)

