from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, Int, File, String

Make_Midpoint_Matrix_V0_1_0 = CommandToolBuilder(tool="make_midpoint_matrix", base_command=["make_midpoint_matrix"], inputs=[ToolInput(tag="in_aggregate_output", input_type=Boolean(optional=True), prefix="--aggregate-output", doc=InputDocumentation(doc="Requests an aggregate matrix. See OUTPUT, below.")), ToolInput(tag="in_discrete_output", input_type=Boolean(optional=True), prefix="--discrete-output", doc=InputDocumentation(doc="Requests a discrete matrix. See OUTPUT, below.")), ToolInput(tag="in_bins", input_type=Int(optional=True), prefix="--bins", doc=InputDocumentation(doc="A list of fragment size bin groups and their\nresolutions. See BINNING, below.")), ToolInput(tag="in_exclude_flags", input_type=File(optional=True), prefix="--exclude-flags", doc=InputDocumentation(doc="A SAM flag used to exclude alignments from the BAM\nfile. More than one may be specified. Alignments\nmatching any exclude flag will not be counted. The\ndefault is to exclude all unmapped reads/mates by\nfiltering out any alignments with SAM flags 4 or 8\nset.")), ToolInput(tag="in_include_flags", input_type=File(optional=True), prefix="--include-flags", doc=InputDocumentation(doc="A SAM flag that determines which alignments from the\nBAM file will be included in the counts. More than one\nmay be specified. Any alignment matching any include\nflag will be counted. The default is to include\nproperly paired and mapped reads by filtering for SAM\nflags 83, 99, 147, or 163.")), ToolInput(tag="in_parallel", input_type=Int(optional=True), prefix="--parallel", doc=InputDocumentation(doc="The number of parallel scoring processes to use\n(default: 1).")), ToolInput(tag="in_quality", input_type=Int(optional=True), prefix="--quality", doc=InputDocumentation(doc="The minimum mapping quality required for a read to be\ncounted (default: 30).")), ToolInput(tag="in_region_extension", input_type=Int(optional=True), prefix="--region-extension", doc=InputDocumentation(doc="The number of bases to score on either side of the\nmotifs (default: 100).")), ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="--verbose", doc=InputDocumentation(doc="Requests more detailed output.")), ToolInput(tag="in_bam_file_of_aligned_reads", input_type=File(), position=0, doc=InputDocumentation(doc="The indexed BAM file containing the aligned ATAC-seq")), ToolInput(tag="in_reads_dot", input_type=String(), position=1, doc=InputDocumentation(doc="BED-file-of-motifs    The BED file containing the motifs. Use '-' to read")), ToolInput(tag="in_ed_otr_dot", input_type=String(), position=0, doc=InputDocumentation(doc="motif     e.r."))], outputs=[ToolOutput(tag="out_exclude_flags", output_type=File(optional=True), selector=InputSelector(input_to_select="in_exclude_flags", type_hint=File()), doc=OutputDocumentation(doc="A SAM flag used to exclude alignments from the BAM\nfile. More than one may be specified. Alignments\nmatching any exclude flag will not be counted. The\ndefault is to exclude all unmapped reads/mates by\nfiltering out any alignments with SAM flags 4 or 8\nset."))], container="quay.io/biocontainers/atactk:0.1.9--pyh3252c3a_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Make_Midpoint_Matrix_V0_1_0().translate("wdl")

