from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, Float, Int, File, String, Directory

Razers3_V0_1_0 = CommandToolBuilder(tool="razers3", base_command=["razers3"], inputs=[ToolInput(tag="in_version_check", input_type=Boolean(optional=True), prefix="--version-check", doc=InputDocumentation(doc="Turn this option off to disable version update notifications of the\napplication. One of 1, ON, TRUE, T, YES, 0, OFF, FALSE, F, and NO.\nDefault: 1.")), ToolInput(tag="in_percent_identity", input_type=Float(optional=True), prefix="--percent-identity", doc=InputDocumentation(doc="Percent identity threshold. In range [50..100]. Default: 95.")), ToolInput(tag="in_recognition_rate", input_type=Float(optional=True), prefix="--recognition-rate", doc=InputDocumentation(doc="Percent recognition rate. In range [80..100]. Default: 100.")), ToolInput(tag="in_no_gaps", input_type=Boolean(optional=True), prefix="--no-gaps", doc=InputDocumentation(doc="Allow only mismatches, no indels. Default: allow both.")), ToolInput(tag="in_forward", input_type=Boolean(optional=True), prefix="--forward", doc=InputDocumentation(doc="Map reads only to forward strands.")), ToolInput(tag="in_reverse", input_type=Boolean(optional=True), prefix="--reverse", doc=InputDocumentation(doc="Map reads only to reverse strands.")), ToolInput(tag="in_max_hits", input_type=Int(optional=True), prefix="--max-hits", doc=InputDocumentation(doc="Output only <NUM> of the best hits. In range [1..inf]. Default: 100.")), ToolInput(tag="in_unique", input_type=Boolean(optional=True), prefix="--unique", doc=InputDocumentation(doc="Output only unique best matches (-m 1 -dr 0 -pa).")), ToolInput(tag="in_trim_reads", input_type=Int(optional=True), prefix="--trim-reads", doc=InputDocumentation(doc="Trim reads to given length. Default: off. In range [14..inf].")), ToolInput(tag="in_output", input_type=File(optional=True), prefix="--output", doc=InputDocumentation(doc="Mapping result filename (use - to dump to stdout in razers format).\nDefault: <READS FILE>.razers. Valid filetypes are: .sam, .razers,\n.gff, .fasta, .fa, .eland, .bam, and .afg.")), ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="--verbose", doc=InputDocumentation(doc="Verbose mode.")), ToolInput(tag="in_v_verbose", input_type=Boolean(optional=True), prefix="--vverbose", doc=InputDocumentation(doc="Very verbose mode.")), ToolInput(tag="in_library_length", input_type=Int(optional=True), prefix="--library-length", doc=InputDocumentation(doc="Paired-end library length. In range [1..inf]. Default: 220.")), ToolInput(tag="in_library_error", input_type=Int(optional=True), prefix="--library-error", doc=InputDocumentation(doc="Paired-end library length tolerance. In range [0..inf]. Default: 50.")), ToolInput(tag="in_alignment", input_type=Boolean(optional=True), prefix="--alignment", doc=InputDocumentation(doc="Dump the alignment for each match (only razer or fasta format).")), ToolInput(tag="in_purge_ambiguous", input_type=Boolean(optional=True), prefix="--purge-ambiguous", doc=InputDocumentation(doc="Purge reads with more than <max-hits> best matches.")), ToolInput(tag="in_distance_range", input_type=Int(optional=True), prefix="--distance-range", doc=InputDocumentation(doc="Only consider matches with at most NUM more errors compared to the\nbest. Default: output all.")), ToolInput(tag="in_genome_naming", input_type=Int(optional=True), prefix="--genome-naming", doc=InputDocumentation(doc="Select how genomes are named (see Naming section below). In range\n[0..1]. Default: 0.")), ToolInput(tag="in_read_naming", input_type=Int(optional=True), prefix="--read-naming", doc=InputDocumentation(doc="Select how reads are named (see Naming section below). In range\n[0..3]. Default: 0.")), ToolInput(tag="in_full_read_id", input_type=Boolean(optional=True), prefix="--full-readid", doc=InputDocumentation(doc="Use the whole read id (don't clip after whitespace).")), ToolInput(tag="in_sort_order", input_type=Int(optional=True), prefix="--sort-order", doc=InputDocumentation(doc="Select how matches are sorted (see Sorting section below). In range\n[0..1]. Default: 0.")), ToolInput(tag="in_position_format", input_type=Int(optional=True), prefix="--position-format", doc=InputDocumentation(doc="Select begin/end position numbering (see Coordinate section below).\nIn range [0..1]. Default: 0.")), ToolInput(tag="in_dont_shrink_alignments", input_type=Boolean(optional=True), prefix="--dont-shrink-alignments", doc=InputDocumentation(doc="Disable alignment shrinking in SAM. This is required for generating\na gold mapping for Rabema.")), ToolInput(tag="in_filter", input_type=String(optional=True), prefix="--filter", doc=InputDocumentation(doc="Select k-mer filter. One of pigeonhole and swift. Default:\npigeonhole.")), ToolInput(tag="in_mutation_rate", input_type=Float(optional=True), prefix="--mutation-rate", doc=InputDocumentation(doc="Set the percent mutation rate (pigeonhole). In range [0..20].\nDefault: 5.")), ToolInput(tag="in_overlap_length", input_type=Int(optional=True), prefix="--overlap-length", doc=InputDocumentation(doc="Manually set the overlap length of adjacent k-mers (pigeonhole). In\nrange [0..inf].")), ToolInput(tag="in_param_dir", input_type=Directory(optional=True), prefix="--param-dir", doc=InputDocumentation(doc="Read user-computed parameter files in the directory <DIR> (swift).")), ToolInput(tag="in_threshold", input_type=Int(optional=True), prefix="--threshold", doc=InputDocumentation(doc="Manually set minimum k-mer count threshold (swift). In range\n[1..inf].")), ToolInput(tag="in_taboo_length", input_type=Int(optional=True), prefix="--taboo-length", doc=InputDocumentation(doc="Set taboo length (swift). In range [1..inf]. Default: 1.")), ToolInput(tag="in_shape", input_type=String(optional=True), prefix="--shape", doc=InputDocumentation(doc="Manually set k-mer shape.")), ToolInput(tag="in_overabundance_cut", input_type=Int(optional=True), prefix="--overabundance-cut", doc=InputDocumentation(doc="Set k-mer overabundance cut ratio. In range [0..1]. Default: 1.")), ToolInput(tag="in_repeat_length", input_type=Int(optional=True), prefix="--repeat-length", doc=InputDocumentation(doc="Skip simple-repeats of length <NUM>. In range [1..inf]. Default:\n1000.")), ToolInput(tag="in_load_factor", input_type=Float(optional=True), prefix="--load-factor", doc=InputDocumentation(doc="Set the load factor for the open addressing k-mer index. In range\n[1..inf]. Default: 1.6.")), ToolInput(tag="in_match_n", input_type=Boolean(optional=True), prefix="--match-N", doc=InputDocumentation(doc="N matches all other characters. Default: N matches nothing.")), ToolInput(tag="in_error_distr", input_type=File(optional=True), prefix="--error-distr", doc=InputDocumentation(doc="Write error distribution to FILE.")), ToolInput(tag="in_mismatch_file", input_type=File(optional=True), prefix="--mismatch-file", doc=InputDocumentation(doc="Write mismatch patterns to FILE.")), ToolInput(tag="in_compact_mult", input_type=Float(optional=True), prefix="--compact-mult", doc=InputDocumentation(doc="Multiply compaction threshold by this value after reaching and\ncompacting. In range [0..inf]. Default: 2.2.")), ToolInput(tag="in_no_compact_frac", input_type=Float(optional=True), prefix="--no-compact-frac", doc=InputDocumentation(doc="Don't compact if in this last fraction of genome. In range [0..1].\nDefault: 0.05.")), ToolInput(tag="in_thread_count", input_type=Int(optional=True), prefix="--thread-count", doc=InputDocumentation(doc="Set the number of threads to use (0 to force sequential mode). In\nrange [0..inf]. Default: 1.")), ToolInput(tag="in_parallel_window_size", input_type=Int(optional=True), prefix="--parallel-window-size", doc=InputDocumentation(doc="Collect candidates in windows of this length. In range [1..inf].\nDefault: 500000.")), ToolInput(tag="in_parallel_verification_size", input_type=Int(optional=True), prefix="--parallel-verification-size", doc=InputDocumentation(doc="Verify candidates in packages of this size. In range [1..inf].\nDefault: 100.")), ToolInput(tag="in_parallel_verification_max_package_count", input_type=Int(optional=True), prefix="--parallel-verification-max-package-count", doc=InputDocumentation(doc="Largest number of packages to create for verification per thread-1.\nIn range [1..inf]. Default: 100.")), ToolInput(tag="in_available_matches_memory_size", input_type=Int(optional=True), prefix="--available-matches-memory-size", doc=InputDocumentation(doc="Bytes of main memory available for storing matches. In range\n[-1..inf]. Default: 0.")), ToolInput(tag="in_match_histo_start_threshold", input_type=Int(optional=True), prefix="--match-histo-start-threshold", doc=InputDocumentation(doc="When to start histogram. In range [1..inf]. Default: 5.")), ToolInput(tag="in_two_zero_zero_three_six_zero_bp_dot", input_type=Int(), position=0, doc=InputDocumentation(doc="VERSION"))], outputs=[ToolOutput(tag="out_output", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output", type_hint=File()), doc=OutputDocumentation(doc="Mapping result filename (use - to dump to stdout in razers format).\nDefault: <READS FILE>.razers. Valid filetypes are: .sam, .razers,\n.gff, .fasta, .fa, .eland, .bam, and .afg."))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Razers3_V0_1_0().translate("wdl", allow_empty_container=True)

