from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Directory, Boolean, Int, String

Bismark2Bedgraph_V0_1_0 = CommandToolBuilder(tool="bismark2bedGraph", base_command=["bismark2bedGraph"], inputs=[ToolInput(tag="in_output", input_type=File(optional=True), prefix="--output", doc=InputDocumentation(doc="Name of the output file, mandatory.")), ToolInput(tag="in_dir", input_type=Directory(optional=True), prefix="--dir", doc=InputDocumentation(doc="Output directory. Output is written to the current directory if not specified explicitly.")), ToolInput(tag="in_cut_off", input_type=Boolean(optional=True), prefix="--cutoff", doc=InputDocumentation(doc="[threshold]       The minimum number of times a methylation state has to be seen for that nucleotide\nbefore its methylation percentage is reported. Default: 1.")), ToolInput(tag="in_remove_spaces", input_type=Boolean(optional=True), prefix="--remove_spaces", doc=InputDocumentation(doc="Replaces whitespaces in the sequence ID field with underscores to allow sorting.")), ToolInput(tag="in_cx_context", input_type=File(optional=True), prefix="--CX_context", doc=InputDocumentation(doc="The sorted bedGraph output file contains information on every single cytosine that was covered\nin the experiment irrespective of its sequence context. This applies to both forward and\nreverse strands. Please be aware that this option may generate large temporary and output files\nand may take a long time to sort (up to many hours). Default: OFF.\n(i.e. Default = CpG context only).")), ToolInput(tag="in_buffer_size", input_type=Int(optional=True), prefix="--buffer_size", doc=InputDocumentation(doc="This allows you to specify the main memory sort buffer when sorting the methylation information.\nEither specify a percentage of physical memory by appending % (e.g. --buffer_size 50%) or\na multiple of 1024 bytes, e.g. 'K' multiplies by 1024, 'M' by 1048576 and so on for 'T' etc.\n(e.g. --buffer_size 20G). For more information on sort type 'info sort' on a command line.\nDefaults to 2G.")), ToolInput(tag="in_scaffolds", input_type=Boolean(optional=True), prefix="--scaffolds", doc=InputDocumentation(doc="Users working with unfinished genomes sporting tens or even hundreds of thousands of\nscaffolds/contigs/chromosomes frequently encountered errors with pre-sorting reads to\nindividual chromosome files. These errors were caused by the operating system's limit\nof the number of filehandle that can be written to at any one time (typically 1024; to\nfind out this limit on Linux, type: ulimit -a).\nTo bypass the limitation of open filehandles, the option --scaffolds does not pre-sort\nmethylation calls into individual chromosome files. Instead, all input files are\ntemporarily merged into a single file (unless there is only a single file), and this\nfile will then be sorted by both chromosome AND position using the Unix sort command.\nPlease be aware that this option might take a looooong time to complete, depending on\nthe size of the input files, and the memory you allocate to this process (see --buffer_size).\nNevertheless, it seems to be working.")), ToolInput(tag="in_ample_memory", input_type=Boolean(optional=True), prefix="--ample_memory", doc=InputDocumentation(doc="Using this option will not sort chromosomal positions using the UNIX 'sort' command, but will\ninstead use two arrays to sort methylated and unmethylated calls, respectively. This may result\nin a faster sorting process for very large files, but this comes at the cost of a larger memory\nfootprint (as an estimate, two arrays of the length of (the largest) human chromosome 1 (nearly\n250 million bp) temporarily consume around 16GB of RAM). Note however that due to the overheads\nof creating and looping through arrays this option might in fact be *slower* for small-ish\nfiles (up to a few million alignments). Note also that this option is not currently compatible\nwith options '--scaffolds/--gazillion'.")), ToolInput(tag="in_zero_based", input_type=File(optional=True), prefix="--zero_based", doc=InputDocumentation(doc="Write out an additional coverage file (ending in .zero.cov) that uses 0-based genomic start\nand 1-based genomic end coordinates (zero-based, half-open), like used in the bedGraph file,\ninstead of using 1-based coordinates throughout. Default: OFF.")), ToolInput(tag="in_ucsc", input_type=File(optional=True), prefix="--ucsc", doc=InputDocumentation(doc="Writes out an additional bedGraph file (ending in '_UCSC.bedGraph.gz') that is compatible with the\nUCSC genome browser. If alignments were carried out against an Ensembl version of the genome, this\nstep will prefix chromosome names with 'chr', and rename the mitochondrial chromosome from 'MT' to\n'chrM'.")), ToolInput(tag="in_methylation", input_type=String(optional=True), position=0, doc=InputDocumentation(doc="")), ToolInput(tag="in_extractor", input_type=String(optional=True), position=1, doc=InputDocumentation(doc="")), ToolInput(tag="in_input", input_type=String(optional=True), position=2, doc=InputDocumentation(doc="")), ToolInput(tag="in_files", input_type=String(optional=True), position=3, doc=InputDocumentation(doc=""))], outputs=[ToolOutput(tag="out_output", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output", type_hint=File()), doc=OutputDocumentation(doc="Name of the output file, mandatory.")), ToolOutput(tag="out_dir", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_dir", type_hint=File()), doc=OutputDocumentation(doc="Output directory. Output is written to the current directory if not specified explicitly.")), ToolOutput(tag="out_cx_context", output_type=File(optional=True), selector=InputSelector(input_to_select="in_cx_context", type_hint=File()), doc=OutputDocumentation(doc="The sorted bedGraph output file contains information on every single cytosine that was covered\nin the experiment irrespective of its sequence context. This applies to both forward and\nreverse strands. Please be aware that this option may generate large temporary and output files\nand may take a long time to sort (up to many hours). Default: OFF.\n(i.e. Default = CpG context only).")), ToolOutput(tag="out_zero_based", output_type=File(optional=True), selector=InputSelector(input_to_select="in_zero_based", type_hint=File()), doc=OutputDocumentation(doc="Write out an additional coverage file (ending in .zero.cov) that uses 0-based genomic start\nand 1-based genomic end coordinates (zero-based, half-open), like used in the bedGraph file,\ninstead of using 1-based coordinates throughout. Default: OFF.")), ToolOutput(tag="out_ucsc", output_type=File(optional=True), selector=InputSelector(input_to_select="in_ucsc", type_hint=File()), doc=OutputDocumentation(doc="Writes out an additional bedGraph file (ending in '_UCSC.bedGraph.gz') that is compatible with the\nUCSC genome browser. If alignments were carried out against an Ensembl version of the genome, this\nstep will prefix chromosome names with 'chr', and rename the mitochondrial chromosome from 'MT' to\n'chrM'."))], container="quay.io/biocontainers/bismark:0.23.0--0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Bismark2Bedgraph_V0_1_0().translate("wdl")

