from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, String, Int, Boolean

Anvi_Compute_Genome_Similarity_V0_1_0 = CommandToolBuilder(tool="anvi_compute_genome_similarity", base_command=["anvi-compute-genome-similarity"], inputs=[ToolInput(tag="in_internal_genomes", input_type=File(optional=True), prefix="--internal-genomes", doc=InputDocumentation(doc="A five-column TAB-delimited flat text file. The header\nline must contain these columns: 'name', 'bin_id',\n'collection_id', 'profile_db_path', 'contigs_db_path'.\nEach line should list a single entry, where 'name' can\nbe any name to describe the anvi'o bin identified as\n'bin_id' that is stored in a collection. (default:\nNone)")), ToolInput(tag="in_external_genomes", input_type=File(optional=True), prefix="--external-genomes", doc=InputDocumentation(doc="A two-column TAB-delimited flat text file that lists\nanvi'o contigs databases. The first item in the header\nline should read 'name', and the second should read\n'contigs_db_path'. Each line in the file should\ndescribe a single entry, where the first column is the\nname of the genome (or MAG), and the second column is\nthe anvi'o contigs database generated for this genome.\n(default: None)")), ToolInput(tag="in_fast_a_text_file", input_type=File(optional=True), prefix="--fasta-text-file", doc=InputDocumentation(doc="A two-column TAB-delimited file that lists multiple\nFASTA files to import for analysis. If using for\n`anvi-dereplicate-genomes` or `anvi-compute-distance`,\neach FASTA is assumed to be a genome. The first item\nin the header line should read 'name', and the second\nitem should read 'path'. Each line in the field should\ndescribe a single entry, where the first column is the\nname of the FASTA file or corresponding sequence, and\nthe second column is the path to the FASTA file\nitself. (default: None)")), ToolInput(tag="in_output_dir", input_type=File(optional=True), prefix="--output-dir", doc=InputDocumentation(doc="Directory path for output files (default: None)")), ToolInput(tag="in_pan_db", input_type=String(optional=True), prefix="--pan-db", doc=InputDocumentation(doc="This is totally optional, but very useful when\napplicable. If you are running this for genomes for\nwhich you already have an anvi'o pangeome, then you\ncan show where the pan database is and anvi'o would\nautomatically add the results into the misc data\ntables of your pangenome. Those data can then be shown\nas heatmaps on the pan interactive interface through\nthe 'layers' tab. (default: None)")), ToolInput(tag="in_program", input_type=String(optional=True), prefix="--program", doc=InputDocumentation(doc="Tell anvi'o which program to run to process genome\nsimilarity. For ANI, you should either use pyANI or\nfastANI. If accuracy is paramount (for example,\ndistinguishing things less than 1 percent different),\nor for dealing with genomes < 80 percent similar,\npyANI is what we recommend. However, fastANI is much\nfaster. If you for some reason want to use mash\nsimilarity, you can use sourmash, but its really not\nintended for genome comparisons. If you don't choose\nanything here, anvi'o will reluctantly set the program\nto pyANI, but you really should be the one who is on\ntop of these things. (default: pyANI)")), ToolInput(tag="in_fast_ani_km_er_size", input_type=Int(optional=True), prefix="--fastani-kmer-size", doc=InputDocumentation(doc="Choose a kmer. The default is 16.")), ToolInput(tag="in_fragment_length", input_type=Int(optional=True), prefix="--fragment-length", doc=InputDocumentation(doc="Choose a fragment length. The default is 3000.")), ToolInput(tag="in_min_num_fragments", input_type=Int(optional=True), prefix="--min-num-fragments", doc=InputDocumentation(doc="Choose the minimum number of fragment lengths that can\nbe trusted. The default is 50.")), ToolInput(tag="in_method", input_type=String(optional=True), prefix="--method", doc=InputDocumentation(doc="Method for pyANI. The default is ANIb. You must have\nthe necessary binary in path for whichever method you\nchoose. According to the pyANI help for v0.2.7 at\nhttps://github.com/widdowquinn/pyani, the method\n'ANIm' uses MUMmer (NUCmer) to align the input\nsequences. 'ANIb' uses BLASTN+ to align 1020nt\nfragments of the input sequences. 'ANIblastall': uses\nthe legacy BLASTN to align 1020nt fragments Finally,\n'TETRA': calculates tetranucleotide frequencies of\neach input sequence")), ToolInput(tag="in_min_alignment_fraction", input_type=Int(optional=True), prefix="--min-alignment-fraction", doc=InputDocumentation(doc="In some cases you may get high raw ANI estimates\n(percent identity scores) between two genomes that\nhave little to do with each other simply because only\na small fraction of their content may be aligned. This\nfilter will set all ANI scores between two genomes to\n0 if the alignment fraction is less than you deem\ntrustable. When you set a value, anvi'o will go\nthrough the ANI results, and set percent identity\nscores between two genomes to 0 if the alignment\nfraction *between either of them* is less than the\nparameter described here. The default is 0.")), ToolInput(tag="in_significant_alignment_length", input_type=Int(optional=True), prefix="--significant-alignment-length", doc=InputDocumentation(doc="So --min-alignment-fraction discards any hit that is\ncoming from alignments that represent shorter\nfractions of genomes, but what if you still don't want\nto miss an alignment that is longer than an X number\nof nucleotides regardless of what fraction of the\ngenome it represents? Well, this parameter is to\nrecover things that may be lost due to --min-\nalignment-fraction parameter. Let's say, if you set\n--min-alignment-fraction to '0.05', and this parameter\nto '5000', anvi'o will keep hits from alignments that\nare longer than 5000 nts, EVEN IF THEY REPRESENT less\nthan 5 percent of a given genome pair. Basically if\n--min-alignment-fraction is your shield to protect\nyourself from incoming garbage, --significant-\nalignment-length is your chopstick to pick out those\nthat may be interesting, and you are a true warrior\nhere. (default: None)")), ToolInput(tag="in_min_full_percent_identity", input_type=Int(optional=True), prefix="--min-full-percent-identity", doc=InputDocumentation(doc="In some cases you may get high raw ANI estimates\n(percent identity scores) between two genomes that\nhave little to do with each other simply because only\na small fraction of their content may be aligned. This\ncan be partly alleviated by considering the *full*\npercent identity, which includes in its calculation\nregions that did not align. For example, if the\nalignment is a whopping 97 percent identity but only 8\npercent of the genome aligned, the *full* percent\nidentity is 0.970 * 0.080 = 0.078 OR 7.8 percent.\n*full* percent identity is always included in the\nreport, but you can also use it as a filter for other\nmetrics, such as percent identity. This filter will\nset all ANI measures between two genomes to 0 if the\n*full* percent identity is less than you deem\ntrustable. When you set a value, anvi'o will go\nthrough the ANI results, and set all ANI measures\nbetween two genomes to 0 if the *full* percent\nidentity *between either of them* is less than the\nparameter described here. The default is 0.")), ToolInput(tag="in_km_er_size", input_type=Int(optional=True), prefix="--kmer-size", doc=InputDocumentation(doc="Set the k-mer size for mash similarity checks. We\nfound 13 in almost all cases correlates best with\nalignment-based ANI. (default: None)")), ToolInput(tag="in_scale", input_type=Int(optional=True), prefix="--scale", doc=InputDocumentation(doc="Set the compression ratio for fasta signature file\ncomputations. The default is 1000. Smaller ratios\ndecrease sensitivity, while larger ratios will lead to\nlarge fasta signatures. (default: 1000)")), ToolInput(tag="in_distance", input_type=String(optional=True), prefix="--distance", doc=InputDocumentation(doc="The distance metric for the hierarchical clustering.\nThe default is 'euclidean'.")), ToolInput(tag="in_linkage", input_type=String(optional=True), prefix="--linkage", doc=InputDocumentation(doc="The linkage method for the hierarchical clustering.\nThe default is 'ward'.")), ToolInput(tag="in_num_threads", input_type=Int(optional=True), prefix="--num-threads", doc=InputDocumentation(doc="Maximum number of threads to use for multithreading\nwhenever possible. Very conservatively, the default is\n1. It is a good idea to not exceed the number of CPUs\n/ cores on your system. Plus, please be careful with\nthis option if you are running your commands on a SGE\n--if you are clusterizing your runs, and asking for\nmultiple threads to use, you may deplete your\nresources very fast. (default: 1)")), ToolInput(tag="in_just_do_it", input_type=Boolean(optional=True), prefix="--just-do-it", doc=InputDocumentation(doc="Don't bother me with questions or warnings, just do\nit. (default: False)")), ToolInput(tag="in_log_file", input_type=File(optional=True), prefix="--log-file", doc=InputDocumentation(doc="File path to store debug/output messages. (default:\nNone)")), ToolInput(tag="in_genome_similarity", input_type=String(), position=0, doc=InputDocumentation(doc="ðŸº More on `anvi-compute-genome-similarity`:"))], outputs=[ToolOutput(tag="out_output_dir", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output_dir", type_hint=File()), doc=OutputDocumentation(doc="Directory path for output files (default: None)")), ToolOutput(tag="out_log_file", output_type=File(optional=True), selector=InputSelector(input_to_select="in_log_file", type_hint=File()), doc=OutputDocumentation(doc="File path to store debug/output messages. (default:\nNone)"))], container="quay.io/biocontainers/anvio-minimal:7--py_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Anvi_Compute_Genome_Similarity_V0_1_0().translate("wdl")

