class: CommandLineTool
id: metaplots.cwl
inputs:
- id: in_an_not_dir
  doc: "transcripts annotation directory, generated by\nprepare_transcripts."
  type: Directory?
  inputBinding:
    prefix: --annot_dir
- id: in_rpf_mapping_file
  doc: ribo-seq BAM/SAM file aligned to the transcriptome.
  type: File?
  inputBinding:
    prefix: --rpf_mapping_file
- id: in_input_file
  doc: "the file list the ribo-seq BAM/SAM files aligned to\nthe transcriptome."
  type: File?
  inputBinding:
    prefix: --input_file
- id: in_stranded
  doc: "whether the data is strand-specific, reverse means\nreversed strand interpretation.(default:\
    \ yes)"
  type: string?
  inputBinding:
    prefix: --stranded
- id: in_minimum_length
  doc: minimum length of read to output, default 24
  type: long?
  inputBinding:
    prefix: --minimum-length
- id: in_maximum_length
  doc: maximum length of read to output, default 36
  type: long?
  inputBinding:
    prefix: --maximum-length
- id: in_p_value_one_cut_off
  doc: "pvalue cutoff of frame0 > frame2 for automatically\npredicting P-site location,\
    \ default 0.001"
  type: long?
  inputBinding:
    prefix: --pvalue1_cutoff
- id: in_p_value_two_cut_off
  doc: "pvalue cutoff of frame0 > frame2 for automatically\npredicting P-site location,\
    \ default 0.001"
  type: long?
  inputBinding:
    prefix: --pvalue2_cutoff
- id: in_frame_zero_percent
  doc: "proportion threshold of the number of reads in frame0,\ndefined by f0/(f0+f1+f2),\
    \ default 0.65"
  type: long?
  inputBinding:
    prefix: --frame0_percent
- id: in_out_name
  doc: 'name of output pdf file(default: metaplots)'
  type: File?
  inputBinding:
    prefix: --outname
outputs:
- id: out_stdout
  doc: Standard output stream
  type: stdout
- id: out_out_name
  doc: 'name of output pdf file(default: metaplots)'
  type: File?
  outputBinding:
    glob: $(inputs.in_out_name)
hints:
- class: DockerRequirement
  dockerPull: quay.io/biocontainers/ribocode:1.2.11--pyh145b6a8_1
cwlVersion: v1.1
baseCommand:
- metaplots
