from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, String, Int, Float

Ribocode_Onestep_V0_1_0 = CommandToolBuilder(tool="RiboCode_onestep", base_command=["RiboCode_onestep"], inputs=[ToolInput(tag="in_gtf", input_type=File(optional=True), prefix="--gtf", doc=InputDocumentation(doc="Default, suitable for GENCODE and ENSEMBL GTF file,\nplease refer: https://en.wikipedia.org/wiki/GENCODE")), ToolInput(tag="in_fast_a", input_type=File(optional=True), prefix="--fasta", doc=InputDocumentation(doc="The genome sequences file in fasta format.")), ToolInput(tag="in_rpf_mapping_file", input_type=File(optional=True), prefix="--rpf_mapping_file", doc=InputDocumentation(doc="ribo-seq BAM/SAM file aligned to the transcriptome.")), ToolInput(tag="in_input_file", input_type=File(optional=True), prefix="--input_file", doc=InputDocumentation(doc="the file list the ribo-seq BAM/SAM files aligned to\nthe transcriptome.")), ToolInput(tag="in_stranded", input_type=String(optional=True), prefix="--stranded", doc=InputDocumentation(doc="whether the data is strand-specific, reverse means\nreversed strand interpretation.(default: yes)")), ToolInput(tag="in_minimum_length", input_type=Int(optional=True), prefix="--minimum-length", doc=InputDocumentation(doc="minimum read length for metaplots analysis, default 24")), ToolInput(tag="in_maximum_length", input_type=Int(optional=True), prefix="--maximum-length", doc=InputDocumentation(doc="maximum read length for metaplots analysis, default 36")), ToolInput(tag="in_frame_zero_percent", input_type=Int(optional=True), prefix="--frame0_percent", doc=InputDocumentation(doc="proportion threshold of reads in frame0. For\nautomatically predicting P-site location, default 0.65")), ToolInput(tag="in_longest_orf", input_type=String(optional=True), prefix="--longest-orf", doc=InputDocumentation(doc="Default: yes, the region from most distal AUG to stop\nwas defined as an ORF. If set to no , the position of\nstart codon will be automatically determined by\nprogram.")), ToolInput(tag="in_pval_cut_off", input_type=Float(optional=True), prefix="--pval-cutoff", doc=InputDocumentation(doc="P-value cutoff for ORF filtering, default 0.05")), ToolInput(tag="in_start_codon", input_type=String(optional=True), prefix="--start_codon", doc=InputDocumentation(doc="The canonical start codon. default: ATG")), ToolInput(tag="in_alt_start_codons", input_type=String(optional=True), prefix="--alt_start_codons", doc=InputDocumentation(doc="The alternative start codon, such as CTG,GTG, default:\nNone. Multiple codons should be separated by comma.")), ToolInput(tag="in_stop_codon", input_type=String(optional=True), prefix="--stop_codon", doc=InputDocumentation(doc="Stop codon, default: TAA,TAG,TGA")), ToolInput(tag="in_transl_table", input_type=Int(optional=True), prefix="--transl_table", doc=InputDocumentation(doc="ORF translation table(Default: 1). Assign the correct\ngenetic code based on your organism, [please refer: ht\ntps://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi\n]")), ToolInput(tag="in_min_aa_length", input_type=Int(optional=True), prefix="--min-AA-length", doc=InputDocumentation(doc="The minimal length of predicted peptides,default 20")), ToolInput(tag="in_output_name", input_type=File(optional=True), prefix="--output-name", doc=InputDocumentation(doc="output file name, default: final_result")), ToolInput(tag="in_output_gtf", input_type=File(optional=True), prefix="--output-gtf", doc=InputDocumentation(doc="output the gtf file of predicted ORFs")), ToolInput(tag="in_output_bed", input_type=File(optional=True), prefix="--output-bed", doc=InputDocumentation(doc="output the bed file of predicted ORFs"))], outputs=[ToolOutput(tag="out_output_name", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output_name", type_hint=File()), doc=OutputDocumentation(doc="output file name, default: final_result")), ToolOutput(tag="out_output_gtf", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output_gtf", type_hint=File()), doc=OutputDocumentation(doc="output the gtf file of predicted ORFs")), ToolOutput(tag="out_output_bed", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output_bed", type_hint=File()), doc=OutputDocumentation(doc="output the bed file of predicted ORFs"))], container="quay.io/biocontainers/ribocode:1.2.11--pyh145b6a8_1", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Ribocode_Onestep_V0_1_0().translate("wdl")

