from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Boolean, Array, String, Int

Wgfastdb_V0_1_0 = CommandToolBuilder(tool="wgfastdb", base_command=["wgfastdb"], inputs=[ToolInput(tag="in_log", input_type=File(optional=True), prefix="--log", doc=InputDocumentation(doc="Set log file path (default: ./wgfastdb.log)")), ToolInput(tag="in_no_update", input_type=Boolean(optional=True), prefix="--no_update", doc=InputDocumentation(doc="Do not sync your collection with the latest assembly\nversions (default: False)")), ToolInput(tag="in_no_assembly_update", input_type=Boolean(optional=True), prefix="--no_assembly_update", doc=InputDocumentation(doc="Do not download the latest assembly summary and\ntaxonomy dump and use your local copies. (default:\nFalse)")), ToolInput(tag="in_download_only", input_type=Boolean(optional=True), prefix="--download_only", doc=InputDocumentation(doc="Run only the download step. (default: False)")), ToolInput(tag="in_species", input_type=Array(t=String(), optional=True), prefix="--species", doc=InputDocumentation(doc="List of species to build database. The species name\nmust match exactly a species directory at\nftp://ftp.ncbi.nlm.nih.gov/genomes/genbank/bacteria/\n(default: None)")), ToolInput(tag="in_config", input_type=File(optional=True), prefix="--config", doc=InputDocumentation(doc="Path to config table (.csv). Table MUST include a\n'species' column and may include a column 'reference'\nfor reference genomes. Curation parameters can also be\nset in the table using column headers: 'unknowns',\n'contigs', 'assembly_size', and 'distance'. If a\nparameter cell is left blank it will be replaced with\nthe default value or a value passed from the command\nline (default: None)")), ToolInput(tag="in_curate_only", input_type=Boolean(optional=True), prefix="--curate_only", doc=InputDocumentation(doc="Run only the curate step (default: False)")), ToolInput(tag="in_unknowns", input_type=Array(t=String(), optional=True), prefix="--unknowns", doc=InputDocumentation(doc="Maximum number of unknown bases (not A, T, C, G) for\ncuration. If more than one value is passed, the list\nmust be the same length as the number of species.\nOtherwise the same value is applied to all species.\n(default: [200])")), ToolInput(tag="in_contigs", input_type=Array(t=String(), optional=True), prefix="--contigs", doc=InputDocumentation(doc="Acceptable deviations from median number of contigs\nfor curation. If more than one value is passed, the\nlist must be the same length as the number of species.\nOtherwise the same value is applied to all species\n(default: [3.0])")), ToolInput(tag="in_assembly_size", input_type=Array(t=Int(), optional=True), prefix="--assembly_size", doc=InputDocumentation(doc="Acceptable devations from median assembly size for\ncuration. If more than one value is passed, the list\nmust be the same length as the number of species.\nOtherwise the same value is applied to all species\n(default: [3.0])")), ToolInput(tag="in_distance", input_type=Array(t=String(), optional=True), prefix="--distance", doc=InputDocumentation(doc="Acceptable deviations from median MASH distances for\ncuration. If more than one value is passed, the list\nmust be the same length as the number of species.\nOtherwise the same value is applied to all species\n(default: [3.0])")), ToolInput(tag="in_reference", input_type=Array(t=String(), optional=True), prefix="--reference", doc=InputDocumentation(doc="Define which genome to use as reference by providing\naccession number (GCA_XXXXXXXXX). This list should be\nthe same length as the number of species. This is\nREQUIRED from the command line or in the config file.\n(default: None)\n"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Wgfastdb_V0_1_0().translate("wdl", allow_empty_container=True)

