from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Directory, File, Int, String, Boolean, Float

Mob_Recon_V0_1_0 = CommandToolBuilder(tool="mob_recon", base_command=["mob_recon"], inputs=[ToolInput(tag="in_outdir", input_type=Directory(optional=True), prefix="--outdir", doc=InputDocumentation(doc="Output Directory to put results")), ToolInput(tag="in_in_file", input_type=File(optional=True), prefix="--infile", doc=InputDocumentation(doc="Input assembly fasta file to process")), ToolInput(tag="in_num_threads", input_type=Int(optional=True), prefix="--num_threads", doc=InputDocumentation(doc="Number of threads to be used")), ToolInput(tag="in_sample_id", input_type=String(optional=True), prefix="--sample_id", doc=InputDocumentation(doc="Sample Prefix for reports")), ToolInput(tag="in_force", input_type=Boolean(optional=True), prefix="--force", doc=InputDocumentation(doc="Overwrite existing directory")), ToolInput(tag="in_filter_db", input_type=File(optional=True), prefix="--filter_db", doc=InputDocumentation(doc="Path to fasta file to mask sequences")), ToolInput(tag="in_genome_filter_db_prefix", input_type=String(optional=True), prefix="--genome_filter_db_prefix", doc=InputDocumentation(doc="Prefix of mash sketch and blastdb of closed\nchromosomes to use for auto selection of close genomes\nfor filtering")), ToolInput(tag="in_mash_genome_neighbor_threshold", input_type=String(optional=True), prefix="--mash_genome_neighbor_threshold", doc=InputDocumentation(doc="Mash distance selecting valid closed genomes to filter")), ToolInput(tag="in_max_contig_size", input_type=Int(optional=True), prefix="--max_contig_size", doc=InputDocumentation(doc="Maximum size of a contig to be considered a plasmid")), ToolInput(tag="in_max_plasmid_size", input_type=Int(optional=True), prefix="--max_plasmid_size", doc=InputDocumentation(doc="Maximum size of a reconstructed plasmid")), ToolInput(tag="in_min_rep_evalue", input_type=Int(optional=True), prefix="--min_rep_evalue", doc=InputDocumentation(doc="Minimum evalue threshold for replicon blastn")), ToolInput(tag="in_min_mob_evalue", input_type=Int(optional=True), prefix="--min_mob_evalue", doc=InputDocumentation(doc="Minimum evalue threshold for relaxase tblastn")), ToolInput(tag="in_min_con_evalue", input_type=Int(optional=True), prefix="--min_con_evalue", doc=InputDocumentation(doc="Minimum evalue threshold for contig blastn")), ToolInput(tag="in_min_rpp_evalue", input_type=Int(optional=True), prefix="--min_rpp_evalue", doc=InputDocumentation(doc="Minimum evalue threshold for repetitve elements blastn")), ToolInput(tag="in_min_length", input_type=Int(optional=True), prefix="--min_length", doc=InputDocumentation(doc="Minimum length of contigs to classify")), ToolInput(tag="in_min_rep_ident", input_type=Int(optional=True), prefix="--min_rep_ident", doc=InputDocumentation(doc="Minimum sequence identity for replicons")), ToolInput(tag="in_min_mob_ident", input_type=Int(optional=True), prefix="--min_mob_ident", doc=InputDocumentation(doc="Minimum sequence identity for relaxases")), ToolInput(tag="in_min_con_ident", input_type=Int(optional=True), prefix="--min_con_ident", doc=InputDocumentation(doc="Minimum sequence identity for contigs")), ToolInput(tag="in_min_rpp_ident", input_type=Int(optional=True), prefix="--min_rpp_ident", doc=InputDocumentation(doc="Minimum sequence identity for repetitive elements")), ToolInput(tag="in_min_rep_cov", input_type=Int(optional=True), prefix="--min_rep_cov", doc=InputDocumentation(doc="Minimum percentage coverage of replicon query by input\nassembly")), ToolInput(tag="in_min_mob_cov", input_type=Int(optional=True), prefix="--min_mob_cov", doc=InputDocumentation(doc="Minimum percentage coverage of relaxase query by input\nassembly")), ToolInput(tag="in_min_con_cov", input_type=Int(optional=True), prefix="--min_con_cov", doc=InputDocumentation(doc="Minimum percentage coverage of assembly contig by the\nplasmid reference database to be considered")), ToolInput(tag="in_min_rpp_cov", input_type=Int(optional=True), prefix="--min_rpp_cov", doc=InputDocumentation(doc="Minimum percentage coverage of contigs by repetitive\nelements")), ToolInput(tag="in_min_overlap", input_type=Int(optional=True), prefix="--min_overlap", doc=InputDocumentation(doc="Minimum overlap of fragments")), ToolInput(tag="in_unicycle_r_contigs", input_type=Boolean(optional=True), prefix="--unicycler_contigs", doc=InputDocumentation(doc="Check for circularity flag generated by unicycler in\nfasta headers")), ToolInput(tag="in_run_overhang", input_type=Boolean(optional=True), prefix="--run_overhang", doc=InputDocumentation(doc="Detect circular contigs with assembly overhangs")), ToolInput(tag="in_keep_tmp", input_type=Boolean(optional=True), prefix="--keep_tmp", doc=InputDocumentation(doc="Do not delete temporary file directory")), ToolInput(tag="in_run_type_r", input_type=Boolean(optional=True), prefix="--run_typer", doc=InputDocumentation(doc="Automatically run Mob-typer on the identified plasmids")), ToolInput(tag="in_debug", input_type=Boolean(optional=True), prefix="--debug", doc=InputDocumentation(doc="Show debug information")), ToolInput(tag="in_plasmid_db", input_type=String(optional=True), prefix="--plasmid_db", doc=InputDocumentation(doc="Reference Database of complete plasmids")), ToolInput(tag="in_plasmid_mash_db", input_type=String(optional=True), prefix="--plasmid_mash_db", doc=InputDocumentation(doc="Companion Mash database of reference database")), ToolInput(tag="in_plasmid_meta", input_type=File(optional=True), prefix="--plasmid_meta", doc=InputDocumentation(doc="MOB-cluster plasmid cluster formatted file matched to\nthe reference plasmid db")), ToolInput(tag="in_plasmid_db_type", input_type=String(optional=True), prefix="--plasmid_db_type", doc=InputDocumentation(doc="Blast database type of reference database")), ToolInput(tag="in_plasmid_replicons", input_type=String(optional=True), prefix="--plasmid_replicons", doc=InputDocumentation(doc="Fasta of plasmid replicons")), ToolInput(tag="in_repetitive_mask", input_type=String(optional=True), prefix="--repetitive_mask", doc=InputDocumentation(doc="Fasta of known repetitive elements")), ToolInput(tag="in_plasmid_mob", input_type=String(optional=True), prefix="--plasmid_mob", doc=InputDocumentation(doc="Fasta of plasmid relaxases")), ToolInput(tag="in_plasmid_mpf", input_type=String(optional=True), prefix="--plasmid_mpf", doc=InputDocumentation(doc="Fasta of known plasmid mate-pair proteins")), ToolInput(tag="in_plasmid_or_it", input_type=String(optional=True), prefix="--plasmid_orit", doc=InputDocumentation(doc="Fasta of known plasmid oriT dna sequences")), ToolInput(tag="in_database_directory", input_type=Directory(optional=True), prefix="--database_directory", doc=InputDocumentation(doc="Directory you want to use for your databases. If the\ndatabases are not already downloaded, they will be\ndownloaded automatically. Defaults to\n/usr/local/lib/python3.8/site-\npackages/mob_suite/databases")), ToolInput(tag="in_primary_cluster_dist", input_type=Int(optional=True), prefix="--primary_cluster_dist", doc=InputDocumentation(doc="Mash distance for assigning primary cluster id 0 - 1")), ToolInput(tag="in_secondary_cluster_dist", input_type=Int(optional=True), prefix="--secondary_cluster_dist", doc=InputDocumentation(doc="Mash distance for assigning primary cluster id 0 - 1")), ToolInput(tag="in_three_dot_zero_dot_zero", input_type=Float(), position=0, doc=InputDocumentation(doc="optional arguments:"))], outputs=[ToolOutput(tag="out_outdir", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_outdir", type_hint=File()), doc=OutputDocumentation(doc="Output Directory to put results"))], container="quay.io/biocontainers/mob_suite:3.0.1--py_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Mob_Recon_V0_1_0().translate("wdl")

