from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, Int, File, String, Float, Directory

Stringtie_V0_1_0 = CommandToolBuilder(tool="stringtie", base_command=["stringtie"], inputs=[ToolInput(tag="in_conservative", input_type=Boolean(optional=True), prefix="--conservative", doc=InputDocumentation(doc=": conservative transcript assembly, same as -t -c 1.5 -f 0.05")), ToolInput(tag="in_rf", input_type=Boolean(optional=True), prefix="--rf", doc=InputDocumentation(doc=": assume stranded library fr-firststrand")), ToolInput(tag="in_fr", input_type=Boolean(optional=True), prefix="--fr", doc=InputDocumentation(doc=": assume stranded library fr-secondstrand")), ToolInput(tag="in_annotation_use_gtfgff", input_type=Int(optional=True), prefix="-G", doc=InputDocumentation(doc="annotation to use for guiding the assembly process (GTF/GFF3)")), ToolInput(tag="in_ptf", input_type=Boolean(optional=True), prefix="--ptf", doc=InputDocumentation(doc=": load point-features from a given 4 column feature file <f_tab>")), ToolInput(tag="in_pathfile_name_stdout", input_type=File(optional=True), prefix="-o", doc=InputDocumentation(doc="path/file name for the assembled transcripts GTF (default: stdout)")), ToolInput(tag="in_prefix_output_transcripts", input_type=String(optional=True), prefix="-l", doc=InputDocumentation(doc="prefix for output transcripts (default: STRG)")), ToolInput(tag="in_isoform_fraction_default", input_type=Float(optional=True), prefix="-f", doc=InputDocumentation(doc="isoform fraction (default: 0.01)")), ToolInput(tag="in_reads_processing_enforces", input_type=Float(optional=True), prefix="-L", doc=InputDocumentation(doc="reads processing; also enforces -s 1.5 -g 0 (default:false)")), ToolInput(tag="in_long_reads_provided", input_type=String(optional=True), prefix="-R", doc=InputDocumentation(doc="long reads are provided, just clean and collapse the reads but\ndo not assemble")), ToolInput(tag="in_assembled_transcript_length", input_type=Int(optional=True), prefix="-m", doc=InputDocumentation(doc="assembled transcript length (default: 200)")), ToolInput(tag="in_anchor_length_junctions", input_type=Int(optional=True), prefix="-a", doc=InputDocumentation(doc="anchor length for junctions (default: 10)")), ToolInput(tag="in_junction_coverage_default", input_type=Int(optional=True), prefix="-j", doc=InputDocumentation(doc="junction coverage (default: 1)")), ToolInput(tag="in_trimming_predicted_transcripts", input_type=String(optional=True), prefix="-t", doc=InputDocumentation(doc="trimming of predicted transcripts based on coverage\n(default: coverage trimming is enabled)")), ToolInput(tag="in_reads_consider_multiexon", input_type=Int(optional=True), prefix="-c", doc=InputDocumentation(doc="reads per bp coverage to consider for multi-exon transcript\n(default: 1)")), ToolInput(tag="in_reads_consider_singleexon", input_type=Float(optional=True), prefix="-s", doc=InputDocumentation(doc="reads per bp coverage to consider for single-exon transcript\n(default: 4.75)")), ToolInput(tag="in_log_bundle_processing", input_type=String(optional=True), prefix="-v", doc=InputDocumentation(doc="(log bundle processing details)")), ToolInput(tag="in_gap_allowed_read", input_type=Int(optional=True), prefix="-g", doc=InputDocumentation(doc="gap allowed between read mappings (default: 50)")), ToolInput(tag="in_bundle_allowed_covered", input_type=Int(optional=True), prefix="-M", doc=InputDocumentation(doc="of bundle allowed to be covered by multi-hit reads (default:1)")), ToolInput(tag="in_threads_cpus_use", input_type=Int(optional=True), prefix="-p", doc=InputDocumentation(doc="of threads (CPUs) to use (default: 1)")), ToolInput(tag="in_abundance_estimation_output", input_type=File(optional=True), prefix="-A", doc=InputDocumentation(doc="abundance estimation output file")), ToolInput(tag="in_window_possibly_erroneous", input_type=Int(optional=True), prefix="-E", doc=InputDocumentation(doc="window around possibly erroneous splice sites from long reads to\nlook out for correct splice sites (default: 25)")), ToolInput(tag="in_output_ballgown_o", input_type=Directory(optional=True), prefix="-B", doc=InputDocumentation(doc="output of Ballgown table files which will be created in the\nsame directory as the output GTF (requires -G, -o recommended)")), ToolInput(tag="in_output_ballgown_table", input_type=File(optional=True), prefix="-b", doc=InputDocumentation(doc="output of Ballgown table files but these files will be\ncreated under the directory path given as <dir_path>")), ToolInput(tag="in_estimate_abundance_requires", input_type=String(optional=True), prefix="-e", doc=InputDocumentation(doc="estimate the abundance of given reference transcripts (requires -G)")), ToolInput(tag="in_viral", input_type=Boolean(optional=True), prefix="--viral", doc=InputDocumentation(doc=": only relevant for long reads from viral data where splice sites\ndo not follow consensus (default:false)")), ToolInput(tag="in_assemble_transcripts_given", input_type=String(optional=True), prefix="-x", doc=InputDocumentation(doc="not assemble any transcripts on the given reference sequence(s)")), ToolInput(tag="in_multimapping_correction_default", input_type=String(optional=True), prefix="-u", doc=InputDocumentation(doc="multi-mapping correction (default: correction enabled)")), ToolInput(tag="in_usage_message_exit", input_type=String(optional=True), prefix="-h", doc=InputDocumentation(doc="this usage message and exit")), ToolInput(tag="in_minimum_input_transcript_fpkm", input_type=Int(optional=True), prefix="-F", doc=InputDocumentation(doc="minimum input transcript FPKM to include in the merge\n(default: 1.0)")), ToolInput(tag="in_minimum_input_transcript_include", input_type=Float(optional=True), prefix="-T", doc=InputDocumentation(doc="minimum input transcript TPM to include in the merge\n(default: 1.0)")), ToolInput(tag="in_keep_merged_transcripts", input_type=Boolean(optional=True), prefix="-i", doc=InputDocumentation(doc="keep merged transcripts with retained introns; by default\nthese are not kept unless there is strong evidence for them"))], outputs=[ToolOutput(tag="out_abundance_estimation_output", output_type=File(optional=True), selector=InputSelector(input_to_select="in_abundance_estimation_output", type_hint=File()), doc=OutputDocumentation(doc="abundance estimation output file")), ToolOutput(tag="out_output_ballgown_o", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_output_ballgown_o", type_hint=File()), doc=OutputDocumentation(doc="output of Ballgown table files which will be created in the\nsame directory as the output GTF (requires -G, -o recommended)")), ToolOutput(tag="out_output_ballgown_table", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output_ballgown_table", type_hint=File()), doc=OutputDocumentation(doc="output of Ballgown table files but these files will be\ncreated under the directory path given as <dir_path>"))], container="quay.io/biocontainers/stringtie:2.1.5--h7e0af3c_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Stringtie_V0_1_0().translate("wdl")

