from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Int, Float, Boolean, String

Curesim_V0_1_0 = CommandToolBuilder(tool="curesim", base_command=["curesim"], inputs=[ToolInput(tag="in_genome_fasta_file", input_type=File(optional=True), prefix="-f", doc=InputDocumentation(doc="[mandatory] genome fasta file or reads fastq file")), ToolInput(tag="in_o", input_type=File(optional=True), prefix="-o", doc=InputDocumentation(doc="[facultative] name of output fastq file [output.fastq]")), ToolInput(tag="in_var_2", input_type=Int(optional=True), prefix="-n", doc=InputDocumentation(doc="[facultative] number of reads to generate [50000]")), ToolInput(tag="in_m", input_type=Int(optional=True), prefix="-m", doc=InputDocumentation(doc="[facultative] read mean size [200]")), ToolInput(tag="in_sd", input_type=Int(optional=True), prefix="-sd", doc=InputDocumentation(doc="[facultative] standard deviation for read size [20.0]")), ToolInput(tag="in_r", input_type=Int(optional=True), prefix="-r", doc=InputDocumentation(doc="[facultative] number of random reads [0]")), ToolInput(tag="in_d", input_type=Float(optional=True), prefix="-d", doc=InputDocumentation(doc="[facultative] deletion rate [0.01]")), ToolInput(tag="in_i", input_type=Float(optional=True), prefix="-i", doc=InputDocumentation(doc="[facultative] insertion rate [0.005]")), ToolInput(tag="in_s", input_type=Float(optional=True), prefix="-s", doc=InputDocumentation(doc="[facultative] substitution rate [0.005]")), ToolInput(tag="in_ui", input_type=Boolean(optional=True), prefix="-ui", doc=InputDocumentation(doc="[facultative] uniform distribution for indels [homopolymers]")), ToolInput(tag="in_us", input_type=Boolean(optional=True), prefix="-us", doc=InputDocumentation(doc="[facultative] uniform distribution for substitutions [exponential]")), ToolInput(tag="in_q", input_type=Int(optional=True), prefix="-q", doc=InputDocumentation(doc="[facultative] quality encoding character ['5']")), ToolInput(tag="in_var_12", input_type=Int(optional=True), prefix="-N", doc=InputDocumentation(doc="[facultative] maximum number of Ns allowed per read [0]")), ToolInput(tag="in_v", input_type=Boolean(optional=True), prefix="-v", doc=InputDocumentation(doc="[facultative] verbose mode, you need R software in this mode [false]")), ToolInput(tag="in_skip", input_type=Boolean(optional=True), prefix="-skip", doc=InputDocumentation(doc="[facultative] skip the correction step [false]")), ToolInput(tag="in_jar", input_type=String(optional=True), prefix="-jar", doc=InputDocumentation(doc="")), ToolInput(tag="in_java", input_type=String(), position=0, doc=InputDocumentation(doc=""))], outputs=[ToolOutput(tag="out_o", output_type=File(optional=True), selector=InputSelector(input_to_select="in_o", type_hint=File()), doc=OutputDocumentation(doc="[facultative] name of output fastq file [output.fastq]"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Curesim_V0_1_0().translate("wdl", allow_empty_container=True)

