from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, String

Efetch_V0_1_0 = CommandToolBuilder(tool="efetch", base_command=["efetch"], inputs=[ToolInput(tag="in_format", input_type=Boolean(optional=True), prefix="-format", doc=InputDocumentation(doc="Format of record or report")), ToolInput(tag="in_mode", input_type=Boolean(optional=True), prefix="-mode", doc=InputDocumentation(doc="text, xml, asn.1, json")), ToolInput(tag="in_style", input_type=Boolean(optional=True), prefix="-style", doc=InputDocumentation(doc="withparts, conwithfeat")), ToolInput(tag="in_db", input_type=Boolean(optional=True), prefix="-db", doc=InputDocumentation(doc="Database name")), ToolInput(tag="in_id", input_type=Boolean(optional=True), prefix="-id", doc=InputDocumentation(doc="Unique identifier or accession number")), ToolInput(tag="in_seq_start", input_type=Boolean(optional=True), prefix="-seq_start", doc=InputDocumentation(doc="First sequence position to retrieve")), ToolInput(tag="in_seq_stop", input_type=Boolean(optional=True), prefix="-seq_stop", doc=InputDocumentation(doc="Last sequence position to retrieve")), ToolInput(tag="in_strand", input_type=Boolean(optional=True), prefix="-strand", doc=InputDocumentation(doc="1 = forward DNA strand, 2 = reverse complement")), ToolInput(tag="in_rev_comp", input_type=Boolean(optional=True), prefix="-revcomp", doc=InputDocumentation(doc="Shortcut for strand 2")), ToolInput(tag="in_chr_start", input_type=Boolean(optional=True), prefix="-chr_start", doc=InputDocumentation(doc="Sequence range from 0-based coordinates")), ToolInput(tag="in_chr_stop", input_type=Boolean(optional=True), prefix="-chr_stop", doc=InputDocumentation(doc="in gene docsum GenomicInfoType object")), ToolInput(tag="in_complexity", input_type=Boolean(optional=True), prefix="-complexity", doc=InputDocumentation(doc="0 = default, 1 = bioseq, 3 = nuc-prot set")), ToolInput(tag="in_extend", input_type=Boolean(optional=True), prefix="-extend", doc=InputDocumentation(doc="Extend sequence retrieval in both directions")), ToolInput(tag="in_extra_feat", input_type=Boolean(optional=True), prefix="-extrafeat", doc=InputDocumentation(doc="Bit flag specifying extra features")), ToolInput(tag="in_start", input_type=Boolean(optional=True), prefix="-start", doc=InputDocumentation(doc="First record to fetch")), ToolInput(tag="in_stop", input_type=Boolean(optional=True), prefix="-stop", doc=InputDocumentation(doc="Last record to fetch")), ToolInput(tag="in_raw", input_type=Boolean(optional=True), prefix="-raw", doc=InputDocumentation(doc="Skip database-specific XML modifications")), ToolInput(tag="in_json", input_type=Boolean(optional=True), prefix="-json", doc=InputDocumentation(doc="Convert adjusted XML output to JSON")), ToolInput(tag="in_full", input_type=String(), position=2, doc=InputDocumentation(doc="Same as native except for mesh")), ToolInput(tag="in_uid", input_type=String(), position=3, doc=InputDocumentation(doc="Unique Identifier List")), ToolInput(tag="in_url", input_type=String(), position=4, doc=InputDocumentation(doc="Entrez URL")), ToolInput(tag="in_xml", input_type=String(), position=5, doc=InputDocumentation(doc="Same as -format full -mode xml")), ToolInput(tag="in_bio_project", input_type=String(), position=6, doc=InputDocumentation(doc="native                      BioProject Report")), ToolInput(tag="in_xml_recordset_xml", input_type=String(), position=7, doc=InputDocumentation(doc="xml      RecordSet XML")), ToolInput(tag="in_bio_sample", input_type=String(), position=8, doc=InputDocumentation(doc="native                      BioSample Report")), ToolInput(tag="in_xml_biosampleset_xml", input_type=String(), position=9, doc=InputDocumentation(doc="xml      BioSampleSet XML")), ToolInput(tag="in_biosystems", input_type=String(), position=10, doc=InputDocumentation(doc="native             xml      Sys-set XML")), ToolInput(tag="in_clin_var", input_type=String(), position=11, doc=InputDocumentation(doc="variation                   Older Format")), ToolInput(tag="in_variation_id", input_type=String(), position=12, doc=InputDocumentation(doc="Transition Format")), ToolInput(tag="in_vcv", input_type=String(), position=13, doc=InputDocumentation(doc="VCV Report")), ToolInput(tag="in_clin_varset", input_type=String(), position=14, doc=InputDocumentation(doc="RCV Report")), ToolInput(tag="in_gds", input_type=String(), position=15, doc=InputDocumentation(doc="native             xml      RecordSet XML")), ToolInput(tag="in_summary", input_type=String(), position=16, doc=InputDocumentation(doc="Summary")), ToolInput(tag="in_gene", input_type=String(), position=17, doc=InputDocumentation(doc="full_report                 Detailed Report")), ToolInput(tag="in_gene_table", input_type=String(), position=18, doc=InputDocumentation(doc="Gene Table")), ToolInput(tag="in_tabular", input_type=String(), position=22, doc=InputDocumentation(doc="Tabular Report")), ToolInput(tag="in_alignmentscores_alignment_scores", input_type=String(), position=23, doc=InputDocumentation(doc="alignmentscores             Alignment Scores")), ToolInput(tag="in_fast_a", input_type=String(), position=24, doc=InputDocumentation(doc="FASTA")), ToolInput(tag="in_homologene_report", input_type=String(), position=25, doc=InputDocumentation(doc="Homologene Report")), ToolInput(tag="in_mesh", input_type=String(), position=29, doc=InputDocumentation(doc="full                        Full Record")), ToolInput(tag="in_nlm_catalog", input_type=String(), position=32, doc=InputDocumentation(doc="native                      Full Record")), ToolInput(tag="in_xml_nlmcatalogrecordset_xml", input_type=String(), position=33, doc=InputDocumentation(doc="xml      NLMCatalogRecordSet XML")), ToolInput(tag="in_pmc", input_type=String(), position=34, doc=InputDocumentation(doc="bioc                        PubTator Central BioC XML")), ToolInput(tag="in_var_43", input_type=String(), position=35, doc=InputDocumentation(doc="MEDLINE")), ToolInput(tag="in_xml_pmcarticleset_xml", input_type=String(), position=36, doc=InputDocumentation(doc="xml      pmc-articleset XML")), ToolInput(tag="in_pubmed", input_type=String(), position=37, doc=InputDocumentation(doc="abstract                    Abstract")), ToolInput(tag="in_bioc", input_type=String(), position=38, doc=InputDocumentation(doc="PubTator Central BioC XML")), ToolInput(tag="in_var_47", input_type=String(), position=39, doc=InputDocumentation(doc="MEDLINE")), ToolInput(tag="in_acc", input_type=String(), position=0, doc=InputDocumentation(doc="Accession Number")), ToolInput(tag="in_est", input_type=String(), position=1, doc=InputDocumentation(doc="EST Report")), ToolInput(tag="in_fast_a_cds_aa", input_type=String(), position=4, doc=InputDocumentation(doc="FASTA of CDS Products")), ToolInput(tag="in_fast_a_cds_na", input_type=String(), position=5, doc=InputDocumentation(doc="FASTA of Coding Regions")), ToolInput(tag="in_ft", input_type=String(), position=6, doc=InputDocumentation(doc="Feature Table")), ToolInput(tag="in_gbc", input_type=String(), position=9, doc=InputDocumentation(doc="xml      INSDSet XML")), ToolInput(tag="in_gene_fast_a", input_type=String(), position=10, doc=InputDocumentation(doc="FASTA of Gene")), ToolInput(tag="in_gpc", input_type=String(), position=13, doc=InputDocumentation(doc="xml      INSDSet XML")), ToolInput(tag="in_gss", input_type=String(), position=14, doc=InputDocumentation(doc="GSS Report")), ToolInput(tag="in_seq_id", input_type=String(), position=19, doc=InputDocumentation(doc="Seq-id ASN.1")), ToolInput(tag="in_snp", input_type=String(), position=20, doc=InputDocumentation(doc="json                        Reference SNP Report")), ToolInput(tag="in_sra", input_type=String(), position=21, doc=InputDocumentation(doc="native             xml      EXPERIMENT_PACKAGE_SET XML")), ToolInput(tag="in_run_info", input_type=String(), position=22, doc=InputDocumentation(doc="xml      SraRunInfo XML")), ToolInput(tag="in_structure", input_type=String(), position=23, doc=InputDocumentation(doc="mmdb                        Ncbi-mime-asn1 strucseq ASN.1")), ToolInput(tag="in_taxonomy", input_type=String(), position=26, doc=InputDocumentation(doc="native                      Taxonomy List")), ToolInput(tag="in_xml_taxaset_xml", input_type=String(), position=27, doc=InputDocumentation(doc="xml      TaxaSet XML"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Efetch_V0_1_0().translate("wdl", allow_empty_container=True)

