from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Int, Float, String, Boolean

Cnvhunter_V0_1_0 = CommandToolBuilder(tool="CnvHunter", base_command=["CnvHunter"], inputs=[ToolInput(tag="in_in", input_type=File(optional=True), prefix="-in", doc=InputDocumentation(doc="Input TSV files (one per sample) containing coverage data (chr, start, end, avg_depth). If only one file is given, each line in this file is interpreted as an input file path.")), ToolInput(tag="in_out", input_type=File(optional=True), prefix="-out", doc=InputDocumentation(doc="Output TSV file containing the detected CNVs.")), ToolInput(tag="in_number_most_use", input_type=Int(optional=True), prefix="-n", doc=InputDocumentation(doc="The number of most similar samples to use for reference construction.\nDefault value: '30'")), ToolInput(tag="in_min_z", input_type=Float(optional=True), prefix="-min_z", doc=InputDocumentation(doc="Minimum z-score for CNV seed detection.\nDefault value: '4'")), ToolInput(tag="in_ext_min_z", input_type=Float(optional=True), prefix="-ext_min_z", doc=InputDocumentation(doc="Minimum z-score for CNV extension around seeds.\nDefault value: '2'")), ToolInput(tag="in_ext_gap_span", input_type=Float(optional=True), prefix="-ext_gap_span", doc=InputDocumentation(doc="Percentage of orignal region size that can be spanned while merging nearby regions (0 disables it).\nDefault value: '20'")), ToolInput(tag="in_sam_min_depth", input_type=Float(optional=True), prefix="-sam_min_depth", doc=InputDocumentation(doc="QC: Minimum average depth of a sample.\nDefault value: '40'")), ToolInput(tag="in_sam_min_corr", input_type=Float(optional=True), prefix="-sam_min_corr", doc=InputDocumentation(doc="QC: Minimum correlation of sample to constructed reference sample.\nDefault value: '0.95'")), ToolInput(tag="in_sam_corr_regs", input_type=Int(optional=True), prefix="-sam_corr_regs", doc=InputDocumentation(doc="Maximum number of regions used for sample correlation calculation (to speed up comparisons for exoms etc.).\nDefault value: '20000'")), ToolInput(tag="in_reg_min_cov", input_type=Float(optional=True), prefix="-reg_min_cov", doc=InputDocumentation(doc="QC: Minimum (average) absolute depth of a target region.\nDefault value: '20'")), ToolInput(tag="in_reg_min_n_cov", input_type=Float(optional=True), prefix="-reg_min_ncov", doc=InputDocumentation(doc="QC: Minimum (average) normalized depth of a target region.\nDefault value: '0.01'")), ToolInput(tag="in_reg_max_cv", input_type=Float(optional=True), prefix="-reg_max_cv", doc=InputDocumentation(doc="QC: Maximum coefficient of variation (median/mad) of target region.\nDefault value: '0.4'")), ToolInput(tag="in_debug", input_type=String(optional=True), prefix="-debug", doc=InputDocumentation(doc="Writes debug information for the sample matching the given name (or for all samples if 'ALL' is given).\nDefault value: ''")), ToolInput(tag="in_seg", input_type=File(optional=True), prefix="-seg", doc=InputDocumentation(doc="Writes a SEG file for the sample matching the given name (used for visualization in IGV).\nDefault value: ''")), ToolInput(tag="in_par", input_type=Int(optional=True), prefix="-par", doc=InputDocumentation(doc="Comma-separated list of pseudo-autosomal regions on the X chromosome.\nDefault value: '1-2699520,154931044-155270560'")), ToolInput(tag="in_cnp_file", input_type=File(optional=True), prefix="-cnp_file", doc=InputDocumentation(doc="BED file containing copy-number-polymorphism (CNP) regions. They are excluded from the normalization/correlation calculation. E.g use the CNV map from http://dx.doi.org/10.1038/nrg3871.\nDefault value: ''")), ToolInput(tag="in_annotate", input_type=File(optional=True), prefix="-annotate", doc=InputDocumentation(doc="List of BED files used for annotation. Each file adds a column to the output file. The base filename is used as colum name and 4th column of the BED file is used as annotation value.\nDefault value: ''")), ToolInput(tag="in_gc_window", input_type=Int(optional=True), prefix="-gc_window", doc=InputDocumentation(doc="Moving median GC-content normalization window size (disabled by default).\nDefault value: '-1'")), ToolInput(tag="in_gc_extend", input_type=Int(optional=True), prefix="-gc_extend", doc=InputDocumentation(doc="Moving median GC-content normalization extension around target region.\nDefault value: '0'")), ToolInput(tag="in_ref", input_type=File(optional=True), prefix="-ref", doc=InputDocumentation(doc="Reference genome FASTA file. If unset, 'reference_genome' from the 'settings.ini' file is used.\nDefault value: ''")), ToolInput(tag="in_changelog", input_type=Boolean(optional=True), prefix="--changelog", doc=InputDocumentation(doc="Prints changeloge and exits.")), ToolInput(tag="in_tdx", input_type=Boolean(optional=True), prefix="--tdx", doc=InputDocumentation(doc="Writes a Tool Definition Xml file. The file name is the application name with the suffix '.tdx'."))], outputs=[ToolOutput(tag="out_out", output_type=File(optional=True), selector=InputSelector(input_to_select="in_out", type_hint=File()), doc=OutputDocumentation(doc="Output TSV file containing the detected CNVs.")), ToolOutput(tag="out_annotate", output_type=File(optional=True), selector=InputSelector(input_to_select="in_annotate", type_hint=File()), doc=OutputDocumentation(doc="List of BED files used for annotation. Each file adds a column to the output file. The base filename is used as colum name and 4th column of the BED file is used as annotation value.\nDefault value: ''"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Cnvhunter_V0_1_0().translate("wdl", allow_empty_container=True)

