from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Int, String

Gripss_V0_1_0 = CommandToolBuilder(tool="gripss", base_command=["gripss"], inputs=[ToolInput(tag="in_breakpoint_p_on", input_type=File(optional=True), prefix="-breakpoint_pon", doc=InputDocumentation(doc="Paired breakpoint pon\nbedpe file")), ToolInput(tag="in_hard_max_normal_absolute_support", input_type=Int(optional=True), prefix="-hard_max_normal_absolute_support", doc=InputDocumentation(doc="Hard max normal\nabsolute support [3]")), ToolInput(tag="in_hard_max_normal_relative_support", input_type=Int(optional=True), prefix="-hard_max_normal_relative_support", doc=InputDocumentation(doc="Hard max normal\nrelative support [0.06]")), ToolInput(tag="in_hard_min_tumor_qual", input_type=Int(optional=True), prefix="-hard_min_tumor_qual", doc=InputDocumentation(doc="Hard min tumor qual\n[100]")), ToolInput(tag="in_input_vcf", input_type=File(optional=True), prefix="-input_vcf", doc=InputDocumentation(doc="Path to GRIDSS VCF")), ToolInput(tag="in_max_in_exact_hom_length_short_del", input_type=Int(optional=True), prefix="-max_inexact_hom_length_short_del", doc=InputDocumentation(doc="Max inexact homology\nlength short del [5]")), ToolInput(tag="in_max_short_strand_bias", input_type=Int(optional=True), prefix="-max_short_strand_bias", doc=InputDocumentation(doc="Max short strand bias\n[0.95]")), ToolInput(tag="in_min_length", input_type=Int(optional=True), prefix="-min_length", doc=InputDocumentation(doc="Min length [32]")), ToolInput(tag="in_min_normal_coverage", input_type=Int(optional=True), prefix="-min_normal_coverage", doc=InputDocumentation(doc="Min normal coverage [8]")), ToolInput(tag="in_min_qual_break_end", input_type=Int(optional=True), prefix="-min_qual_break_end", doc=InputDocumentation(doc="Min qual break end\n[1000]")), ToolInput(tag="in_min_qual_break_point", input_type=Int(optional=True), prefix="-min_qual_break_point", doc=InputDocumentation(doc="Min qual break point\n[400]")), ToolInput(tag="in_min_qual_rescue_mobile_element_insertion", input_type=Int(optional=True), prefix="-min_qual_rescue_mobile_element_insertion", doc=InputDocumentation(doc="Min qual rescue mobile\nelement insertions\n[1000]")), ToolInput(tag="in_min_tumor_af", input_type=Int(optional=True), prefix="-min_tumor_af", doc=InputDocumentation(doc="Min tumor allelic\nfrequency [0.005]")), ToolInput(tag="in_output_vcf", input_type=File(optional=True), prefix="-output_vcf", doc=InputDocumentation(doc="Path to output VCF")), ToolInput(tag="in_ref_genome", input_type=String(optional=True), prefix="-ref_genome", doc=InputDocumentation(doc="Ref genome")), ToolInput(tag="in_reference", input_type=String(optional=True), prefix="-reference", doc=InputDocumentation(doc="Optional name of\nreference sample")), ToolInput(tag="in_soft_max_normal_relative_support", input_type=Int(optional=True), prefix="-soft_max_normal_relative_support", doc=InputDocumentation(doc="Soft max normal\nrelative support [0.03]")), ToolInput(tag="in_tumor", input_type=String(optional=True), prefix="-tumor", doc=InputDocumentation(doc="Name of tumor sample")), ToolInput(tag="in_grips_s", input_type=String(), position=0, doc=InputDocumentation(doc="-breakend_pon <arg>                               Single breakend pon bed")), ToolInput(tag="in_file", input_type=File(), position=1, doc=InputDocumentation(doc="-breakpoint_hotspot <arg>                         Paired breakpoint")), ToolInput(tag="in_input", input_type=String(), position=0, doc=InputDocumentation(doc="-max_hom_length_short_inv <arg>                   Max homology length"))], outputs=[ToolOutput(tag="out_output_vcf", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output_vcf", type_hint=File()), doc=OutputDocumentation(doc="Path to output VCF"))], container="quay.io/biocontainers/hmftools-gripss:1.9--0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Gripss_V0_1_0().translate("wdl")

