from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, Int, Directory, String, File

Clever_V0_1_0 = CommandToolBuilder(tool="clever", base_command=["clever"], inputs=[ToolInput(tag="in_sorted", input_type=Boolean(optional=True), prefix="--sorted", doc=InputDocumentation(doc="Input BAM file is sorted by position. Note that this\nrequires alternative alignments to be given as XA tags\n(like produced by BWA, stampy, etc.).")), ToolInput(tag="in_use_x_a", input_type=Boolean(optional=True), prefix="--use_xa", doc=InputDocumentation(doc="Interprete XA tags in input BAM file. This option\nSHOULD be given for mappers writing XA tags like BWA\nand stampy.")), ToolInput(tag="in_use_mapq", input_type=Boolean(optional=True), prefix="--use_mapq", doc=InputDocumentation(doc="Use MAPQ value instead re-computing posteriors.")), ToolInput(tag="in_number_threads_use", input_type=Int(optional=True), prefix="-T", doc=InputDocumentation(doc="Number of threads to use (default=1).")), ToolInput(tag="in_delete_old_result", input_type=Boolean(optional=True), prefix="-f", doc=InputDocumentation(doc="Delete old result and working directory first (if\npresent).")), ToolInput(tag="in_working_directory_default", input_type=Directory(optional=True), prefix="-w", doc=InputDocumentation(doc="Working directory (default: <result-directory>/work).")), ToolInput(tag="in_recompute_tags_ignored", input_type=Boolean(optional=True), prefix="-a", doc=InputDocumentation(doc="Do not (re-)compute AS tags. If given, the argument\n<ref.fasta(.gz)> is ignored.")), ToolInput(tag="in_keep_working_directory", input_type=Boolean(optional=True), prefix="-k", doc=InputDocumentation(doc="Keep working directory (default: delete directory when\nfinished).")), ToolInput(tag="in_read_groups_account", input_type=Boolean(optional=True), prefix="-r", doc=InputDocumentation(doc="Take read groups into account (multi sample mode).")), ToolInput(tag="in_additional_parameters_passed", input_type=String(optional=True), prefix="-C", doc=InputDocumentation(doc="Additional parameters to be passed to the CLEVER core\nalgorithm. Call 'clever-core' without parameters for a\nlist of options.")), ToolInput(tag="in_additional_parameters_postprocessing", input_type=String(optional=True), prefix="-P", doc=InputDocumentation(doc="Additional parameters for postprocessing results. Call\n'postprocess-predictions' without parameters for a\nlist of options.")), ToolInput(tag="in_create_plot_length", input_type=Boolean(optional=True), prefix="-I", doc=InputDocumentation(doc="Create a plot of internal segment size distribution\n(=fragment size - 2x read length). Also displays the\nestimated normal distribution (requires NumPy and\nmatplotlib).")), ToolInput(tag="in_chromosome", input_type=String(optional=True), prefix="--chromosome", doc=InputDocumentation(doc="Only process given chromosome (default: all).\n")), ToolInput(tag="in_bam_file", input_type=File(), position=0, doc=InputDocumentation(doc=""))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Clever_V0_1_0().translate("wdl", allow_empty_container=True)

