from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Array, String, Float, Int

Imfusion_Ctg_V0_1_0 = CommandToolBuilder(tool="imfusion_ctg", base_command=["imfusion-ctg"], inputs=[ToolInput(tag="in_insertions", input_type=File(optional=True), prefix="--insertions", doc=InputDocumentation(doc="Path to the merged insertions file from imfusion-\nmerge.")), ToolInput(tag="in_reference", input_type=File(optional=True), prefix="--reference", doc=InputDocumentation(doc="Path to the reference genome sequence (in fasta\nformat). Can either be the augmented reference genome\nor the original reference.")), ToolInput(tag="in_gene_ids", input_type=Array(t=String(), optional=True), prefix="--gene_ids", doc=InputDocumentation(doc="IDs of genes to test.")), ToolInput(tag="in_output", input_type=File(optional=True), prefix="--output", doc=InputDocumentation(doc="Path for the output CTG file.")), ToolInput(tag="in_threshold", input_type=Float(optional=True), prefix="--threshold", doc=InputDocumentation(doc="Minimum corrected p-value for CTGs. (Default = 0.01).")), ToolInput(tag="in_pattern", input_type=Int(optional=True), prefix="--pattern", doc=InputDocumentation(doc="Regular expression reflecting the nucleotide sequence\nat which the use transposon typically integrates (if\nany). Used to correct for sequence integration biases\nalong the genome. For example the pattern (AT|TA) is\nused for the T2onc2 transposon, which is biased\ntowards integrations as TA sites.")), ToolInput(tag="in_window", input_type=String(optional=True), prefix="--window", doc=InputDocumentation(doc="WINDOW\nWindow around the gene within which we test a given\ngene for enrichment in insertions.")), ToolInput(tag="in_chromosomes", input_type=Array(t=String(), optional=True), prefix="--chromosomes", doc=InputDocumentation(doc="Chromosomes to consider. Used to omit specific\nchromosomes from the CTG analysis.")), ToolInput(tag="in_min_depth", input_type=Int(optional=True), prefix="--min_depth", doc=InputDocumentation(doc="Minimum supporting number of reads for insertions to\nbe included in the CTG analysis. Can be used to omit\ninsertions with low support for more confidence in the\nanalysis.")), ToolInput(tag="in_expression", input_type=File(optional=True), prefix="--expression", doc=InputDocumentation(doc="Path to the merged expression file from imfusion-\nmerge.")), ToolInput(tag="in_de_threshold", input_type=String(optional=True), prefix="--de_threshold", doc=InputDocumentation(doc="Minimum p-value for a CTG to be considered as\ndifferentially expressed.\n"))], outputs=[ToolOutput(tag="out_output", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output", type_hint=File()), doc=OutputDocumentation(doc="Path for the output CTG file."))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Imfusion_Ctg_V0_1_0().translate("wdl", allow_empty_container=True)

