version 1.0

task ChorusNGSfilter {
  input {
    File? jellyfish
    File? genome
    Int? fastq_format_input
    File? jellyfish_file
    String? gzipped
    Int? threads
    Int? km_er
    File? probe
    File? output_bed_format
  }
  command <<<
    ChorusNGSfilter \
      ~{if defined(jellyfish) then ("--jellyfish " +  '"' + jellyfish + '"') else ""} \
      ~{if defined(genome) then ("--genome " +  '"' + genome + '"') else ""} \
      ~{if defined(fastq_format_input) then ("--input " +  '"' + fastq_format_input + '"') else ""} \
      ~{if defined(jellyfish_file) then ("--jellyfishfile " +  '"' + jellyfish_file + '"') else ""} \
      ~{if defined(gzipped) then ("--gzipped " +  '"' + gzipped + '"') else ""} \
      ~{if defined(threads) then ("--threads " +  '"' + threads + '"') else ""} \
      ~{if defined(km_er) then ("--kmer " +  '"' + km_er + '"') else ""} \
      ~{if defined(probe) then ("--probe " +  '"' + probe + '"') else ""} \
      ~{if defined(output_bed_format) then ("--output " +  '"' + output_bed_format + '"') else ""}
  >>>
  runtime {
    docker: "quay.io/biocontainers/chorus2:2.01--py37hf01694f_0"
  }
  parameter_meta {
    jellyfish: "The path where Jellyfish software installed"
    genome: "Fasta format genome file, should include all sequences\\nfrom genome"
    fastq_format_input: "Fastq format input files contain reads from whole\\ngenome shotgun sequencing, files can be gzipped.\\nMultiple files separate with \\\",\\\". For example:\\n1.fq.gz,2.fq.gz"
    jellyfish_file: "prebuild jellyfish index file, conflict with input\\nargument."
    gzipped: "Input fastq file is gzipped(gz) or uncompressed(text).\\n(Default: gz)"
    threads: "Number of threads or CPUs to use. (Default: 1)"
    km_er: "Length of k-mer used for counting k-mers in input\\nfastq files. (Default: 17)"
    probe: "The bed format probe file generated by Chorus"
    output_bed_format: "Output bed format probe file with k-mer score.\\n(Default: output.bed)"
  }
  output {
    File out_stdout = stdout()
    File out_output_bed_format = "${in_output_bed_format}"
  }
}