from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Float, Int, Boolean, File, String

Corset_V0_1_0 = CommandToolBuilder(tool="corset", base_command=["corset"], inputs=[ToolInput(tag="in_comma_separated_list", input_type=Float(optional=True), prefix="-d", doc=InputDocumentation(doc="A comma separated list of distance thresholds. The range must be\nbetween 0 and 1. e.g -d 0.4,0.5. If more than one distance threshold\nis supplied, the output filenames will be of the form:\ncounts-<threshold>.txt and clusters-<threshold>.txt\nDefault: 0.3")), ToolInput(tag="in_value_used_thresholding", input_type=Int(optional=True), prefix="-D", doc=InputDocumentation(doc="The value used for thresholding the log likelihood ratio. The default\nvalue will depend on the number of degrees of freedom (which is the\nnumber of groups -1). By default D = 17.5 + 2.5 * ndf, which corresponds\napproximately to a p-value threshold of 10^-5, when there are fewer than\n10 groups.")), ToolInput(tag="in_switches_log_be", input_type=Boolean(optional=True), prefix="-I", doc=InputDocumentation(doc="Switches off the log likelihood ratio test and should be used\nfor downstream differential transcript usage analysis. It will prevent\ndifferentially expressed transcript being split into different clusters.\nThis option is the equivalent of setting -D to a very large number.\nDefault: log likelihood ratio test on")), ToolInput(tag="in_filter_transcripts_fewer", input_type=Int(optional=True), prefix="-m", doc=InputDocumentation(doc="Filter out any transcripts with fewer than this many reads aligning.\nDefault: 10")), ToolInput(tag="in_specifies_grouping_belong", input_type=File(optional=True), prefix="-g", doc=InputDocumentation(doc="Specifies the grouping. i.e. which samples belong to which experimental\ngroups. The parameter must be a comma separated list (no spaces), with the\ngroupings given in the same order as the bam filename. For example:\n-g Group1,Group1,Group2,Group2 etc. If this option is not used, each sample\nis treated as an independent experimental group.")), ToolInput(tag="in_prefix_output_filenames", input_type=String(optional=True), prefix="-p", doc=InputDocumentation(doc="Prefix for the output filenames. The output files will be of the form\n<prefix>-counts.txt and <prefix>-clusters.txt. Default filenames are:\ncounts.txt and clusters.txt")), ToolInput(tag="in_specifies_files_overwritten", input_type=String(optional=True), prefix="-f", doc=InputDocumentation(doc="Specifies whether the output files should be overwritten if they already exist.\nDefault: false")), ToolInput(tag="in_specifies_sample_names", input_type=File(optional=True), prefix="-n", doc=InputDocumentation(doc="Specifies the sample names to be used in the header of the output count file.\nThis should be a comma separated list without spaces.\ne.g. -n Group1-ReplicateA,Group1-ReplicateB,Group2-ReplicateA etc.\nDefault: the input filenames will be used.")), ToolInput(tag="in_output_file_summarising", input_type=File(optional=True), prefix="-r", doc=InputDocumentation(doc="Output a file summarising the read alignments. This may be used if you\nwould like to read the bam files and run the clustering in seperate runs\nof corset. e.g. to read input bam files in parallel. The output will be the\nbam filename appended with .corset-reads.\nDefault: false")), ToolInput(tag="in_input_file_type", input_type=File(optional=True), prefix="-i", doc=InputDocumentation(doc="The input file type. Use -i corset, if you previously ran\ncorset with the -r option and would like to restart using those\nread summary files. Use salmon_eq_classes, if you aligned with salmon with\nthe flag --dumpEq and are passing corset the equivalent class files.\nRunning with either -i corset or salmon_eq_classes will switch off the -r option.\nDefault: bam")), ToolInput(tag="in_running_i_will", input_type=Int(optional=True), prefix="-l", doc=InputDocumentation(doc="If running with -i corset or salmon_eq_classes, this will filter out a link between contigs\nif the link is supported by less than this many reads (performed sample-wise). Reads will\nbe reassigned uniformly to the contigs in the equivalence class. This option will\nimprove runtime and memory usage, but will increase the number of clusters reported.\nDefault: 1 (no filtering)")), ToolInput(tag="in_running_i_option", input_type=Int(optional=True), prefix="-x", doc=InputDocumentation(doc="If running with -i corset or salmon_eq_classes, this option will filter out reads that\nalign to more than x contigs. Default: no filtering")), ToolInput(tag="in_input_bam_files", input_type=String(), position=0, doc=InputDocumentation(doc=""))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Corset_V0_1_0().translate("wdl", allow_empty_container=True)

