from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Array, String, Boolean, Int, File, Float

Srst2_V0_1_0 = CommandToolBuilder(tool="srst2", base_command=["srst2"], inputs=[ToolInput(tag="in_input_se", input_type=Array(t=String(), optional=True), prefix="--input_se", doc=InputDocumentation(doc="Single end read file(s) for analysing (may be gzipped)")), ToolInput(tag="in_input_pe", input_type=Array(t=String(), optional=True), prefix="--input_pe", doc=InputDocumentation(doc="Paired end read files for analysing (may be gzipped)")), ToolInput(tag="in_merge_paired", input_type=Boolean(optional=True), prefix="--merge_paired", doc=InputDocumentation(doc="Switch on if all the input read sets belong to a\nsingle sample, and you want to merge their data to get\na single result")), ToolInput(tag="in_forward", input_type=Int(optional=True), prefix="--forward", doc=InputDocumentation(doc="Designator for forward reads (only used if NOT in\nMiSeq format sample_S1_L001_R1_001.fastq.gz; otherwise\ndefault is _1, i.e. expect forward reads as\nsample_1.fastq.gz)")), ToolInput(tag="in_reverse", input_type=Int(optional=True), prefix="--reverse", doc=InputDocumentation(doc="Designator for reverse reads (only used if NOT in\nMiSeq format sample_S1_L001_R2_001.fastq.gz; otherwise\ndefault is _2, i.e. expect forward reads as\nsample_2.fastq.gz")), ToolInput(tag="in_read_type", input_type=String(optional=True), prefix="--read_type", doc=InputDocumentation(doc="Read file type (for bowtie2; default is q=fastq; other\noptions: qseq=solexa, f=fasta).")), ToolInput(tag="in_mlst_db", input_type=File(optional=True), prefix="--mlst_db", doc=InputDocumentation(doc="Fasta file of MLST alleles (optional)")), ToolInput(tag="in_mlst_delimiter", input_type=Int(optional=True), prefix="--mlst_delimiter", doc=InputDocumentation(doc="Character(s) separating gene name from allele number\nin MLST database (default '-', as in arcc-1)")), ToolInput(tag="in_mlst_definitions", input_type=String(optional=True), prefix="--mlst_definitions", doc=InputDocumentation(doc="ST definitions for MLST scheme (required if mlst_db\nsupplied and you want to calculate STs)")), ToolInput(tag="in_mlst_max_mismatch", input_type=Int(optional=True), prefix="--mlst_max_mismatch", doc=InputDocumentation(doc="Maximum number of mismatches per read for MLST allele\ncalling (default 10)")), ToolInput(tag="in_gene_db", input_type=Array(t=String(), optional=True), prefix="--gene_db", doc=InputDocumentation(doc="Fasta file/s for gene databases (optional)")), ToolInput(tag="in_no_gene_details", input_type=Boolean(optional=True), prefix="--no_gene_details", doc=InputDocumentation(doc="Switch OFF verbose reporting of gene typing")), ToolInput(tag="in_gene_max_mismatch", input_type=Int(optional=True), prefix="--gene_max_mismatch", doc=InputDocumentation(doc="Maximum number of mismatches per read for gene\ndetection and allele calling (default 10)")), ToolInput(tag="in_min_coverage", input_type=Int(optional=True), prefix="--min_coverage", doc=InputDocumentation(doc="Minimum %coverage cutoff for gene reporting (default\n90)")), ToolInput(tag="in_max_divergence", input_type=Int(optional=True), prefix="--max_divergence", doc=InputDocumentation(doc="Maximum %divergence cutoff for gene reporting (default\n10)")), ToolInput(tag="in_min_depth", input_type=Int(optional=True), prefix="--min_depth", doc=InputDocumentation(doc="Minimum mean depth to flag as dubious allele call\n(default 5)")), ToolInput(tag="in_min_edge_depth", input_type=Int(optional=True), prefix="--min_edge_depth", doc=InputDocumentation(doc="Minimum edge depth to flag as dubious allele call\n(default 2)")), ToolInput(tag="in_prob_err", input_type=Float(optional=True), prefix="--prob_err", doc=InputDocumentation(doc="Probability of sequencing error (default 0.01)")), ToolInput(tag="in_truncation_score_tolerance", input_type=String(optional=True), prefix="--truncation_score_tolerance", doc=InputDocumentation(doc="% increase in score allowed to choose non-truncated\nallele")), ToolInput(tag="in_stop_after", input_type=Int(optional=True), prefix="--stop_after", doc=InputDocumentation(doc="Stop mapping after this number of reads have been\nmapped (otherwise map all)")), ToolInput(tag="in_other", input_type=Int(optional=True), prefix="--other", doc=InputDocumentation(doc="Other arguments to pass to bowtie2 (must be escaped,\ne.g. '\--no-mixed'.")), ToolInput(tag="in_max_unaligned_overlap", input_type=Int(optional=True), prefix="--max_unaligned_overlap", doc=InputDocumentation(doc="Read discarded from alignment if either of its ends\nhas unaligned overlap with the reference that is\nlonger than this value (default 10)")), ToolInput(tag="in_mapq", input_type=Int(optional=True), prefix="--mapq", doc=InputDocumentation(doc="Samtools -q parameter (default 1)")), ToolInput(tag="in_base_q", input_type=Int(optional=True), prefix="--baseq", doc=InputDocumentation(doc="Samtools -Q parameter (default 20)")), ToolInput(tag="in_sam_tools_args", input_type=String(optional=True), prefix="--samtools_args", doc=InputDocumentation(doc="Other arguments to pass to samtools mpileup (must be\nescaped, e.g. '\-A').")), ToolInput(tag="in_output", input_type=Int(optional=True), prefix="--output", doc=InputDocumentation(doc="Prefix for srst2 output files")), ToolInput(tag="in_log", input_type=Boolean(optional=True), prefix="--log", doc=InputDocumentation(doc="Switch ON logging to file (otherwise log to stdout)")), ToolInput(tag="in_save_scores", input_type=Boolean(optional=True), prefix="--save_scores", doc=InputDocumentation(doc="Switch ON verbose reporting of all scores")), ToolInput(tag="in_report_new_consensus", input_type=Boolean(optional=True), prefix="--report_new_consensus", doc=InputDocumentation(doc="If a matching alleles is not found, report the\nconsensus allele. Note, only SNP differences are\nconsidered, not indels.")), ToolInput(tag="in_report_all_consensus", input_type=Boolean(optional=True), prefix="--report_all_consensus", doc=InputDocumentation(doc="Report the consensus allele for the most likely\nallele. Note, only SNP differences are considered, not\nindels.")), ToolInput(tag="in_use_existing_bowtie_two_sam", input_type=Boolean(optional=True), prefix="--use_existing_bowtie2_sam", doc=InputDocumentation(doc="Use existing SAM file generated by Bowtie2 if\navailable, otherwise they will be generated")), ToolInput(tag="in_use_existing_pile_up", input_type=Boolean(optional=True), prefix="--use_existing_pileup", doc=InputDocumentation(doc="Use existing pileups if available, otherwise they will\nbe generated")), ToolInput(tag="in_use_existing_scores", input_type=Boolean(optional=True), prefix="--use_existing_scores", doc=InputDocumentation(doc="Use existing scores files if available, otherwise they\nwill be generated")), ToolInput(tag="in_keep_interim_alignment", input_type=Boolean(optional=True), prefix="--keep_interim_alignment", doc=InputDocumentation(doc="Keep interim files (sam & unsorted bam), otherwise\nthey will be deleted after sorted bam is created")), ToolInput(tag="in_threads", input_type=String(optional=True), prefix="--threads", doc=InputDocumentation(doc="Use multiple threads in Bowtie and Samtools")), ToolInput(tag="in_prev_output", input_type=Array(t=String(), optional=True), prefix="--prev_output", doc=InputDocumentation(doc="SRST2 results files to compile (any new results from\nthis run will also be incorporated)\n"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Srst2_V0_1_0().translate("wdl", allow_empty_container=True)

