from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import String, File, Int, Boolean

Jannovar_Annotate_Vcf_V0_1_0 = CommandToolBuilder(tool="jannovar_annotate_vcf", base_command=["jannovar", "annotate-vcf"], inputs=[ToolInput(tag="in_d", input_type=String(optional=True), prefix="-d", doc=InputDocumentation(doc="[--interval INTERVAL]")), ToolInput(tag="in_input_vcf", input_type=File(optional=True), prefix="--input-vcf", doc=InputDocumentation(doc="Path to input VCF file")), ToolInput(tag="in_output_vcf", input_type=File(optional=True), prefix="--output-vcf", doc=InputDocumentation(doc="Path to output VCF file")), ToolInput(tag="in_database", input_type=File(optional=True), prefix="--database", doc=InputDocumentation(doc="Path to database .ser file")), ToolInput(tag="in_interval", input_type=String(optional=True), prefix="--interval", doc=InputDocumentation(doc="Interval with regions to annotate (optional)")), ToolInput(tag="in_pedigree_file", input_type=File(optional=True), prefix="--pedigree-file", doc=InputDocumentation(doc="Pedigree file  to  use  for  Mendelian inheritance\nannotation")), ToolInput(tag="in_ref_fast_a", input_type=File(optional=True), prefix="--ref-fasta", doc=InputDocumentation(doc="Path  to   FAI-indexed   reference   FASTA   file,\nrequired for dbSNP/ExAC/UK10K-based annotation")), ToolInput(tag="in_dbsnp_vcf", input_type=File(optional=True), prefix="--dbsnp-vcf", doc=InputDocumentation(doc="Path to dbSNP VCF file, activates dbSNP annotation")), ToolInput(tag="in_dbsnp_prefix", input_type=String(optional=True), prefix="--dbsnp-prefix", doc=InputDocumentation(doc="Prefix for dbSNP annotations")), ToolInput(tag="in_ex_ac_vcf", input_type=File(optional=True), prefix="--exac-vcf", doc=InputDocumentation(doc="Path to ExAC VCF file, activates ExAC annotation")), ToolInput(tag="in_ex_ac_prefix", input_type=String(optional=True), prefix="--exac-prefix", doc=InputDocumentation(doc="Prefix for ExAC annotations")), ToolInput(tag="in_g_nomad_exo_mes_vcf", input_type=File(optional=True), prefix="--gnomad-exomes-vcf", doc=InputDocumentation(doc="Path to gnomAD exomes  VCF  file, activates gnomAD\nexomes annotation")), ToolInput(tag="in_g_nomad_exo_mes_prefix", input_type=String(optional=True), prefix="--gnomad-exomes-prefix", doc=InputDocumentation(doc="Prefix for ExgnomAD exomes AC annotations")), ToolInput(tag="in_g_nomad_genomes_vcf", input_type=File(optional=True), prefix="--gnomad-genomes-vcf", doc=InputDocumentation(doc="Path to gnomAD genomes  VCF file, activates gnomAD\ngenomes annotation")), ToolInput(tag="in_g_nomad_genomes_prefix", input_type=String(optional=True), prefix="--gnomad-genomes-prefix", doc=InputDocumentation(doc="Prefix for ExgnomAD genomes AC annotations")), ToolInput(tag="in_uk_one_zero_k_vcf", input_type=Int(optional=True), prefix="--uk10k-vcf", doc=InputDocumentation(doc="Path to UK10K VCF file, activates UK10K annotation")), ToolInput(tag="in_uk_one_zero_k_prefix", input_type=Int(optional=True), prefix="--uk10k-prefix", doc=InputDocumentation(doc="Prefix for UK10K annotations")), ToolInput(tag="in_gone_k_vcf", input_type=Int(optional=True), prefix="--g1k-vcf", doc=InputDocumentation(doc="Path  to  thousand  genomes  VCF  file,  activates\nthousand genomes annotation")), ToolInput(tag="in_gone_k_prefix", input_type=Int(optional=True), prefix="--g1k-prefix", doc=InputDocumentation(doc="Prefix for thousand genomes annotations")), ToolInput(tag="in_clin_var_vcf", input_type=File(optional=True), prefix="--clinvar-vcf", doc=InputDocumentation(doc="Path to ClinVar file, activates ClinVar annotation")), ToolInput(tag="in_clin_var_prefix", input_type=String(optional=True), prefix="--clinvar-prefix", doc=InputDocumentation(doc="Prefix for ClinVar annotations")), ToolInput(tag="in_cosmic_vcf", input_type=File(optional=True), prefix="--cosmic-vcf", doc=InputDocumentation(doc="Path to COSMIC file, activates COSMIC annotation")), ToolInput(tag="in_cosmic_prefix", input_type=String(optional=True), prefix="--cosmic-prefix", doc=InputDocumentation(doc="Prefix for COSMIC annotations")), ToolInput(tag="in_inheritance_an_no_use_filters", input_type=Boolean(optional=True), prefix="--inheritance-anno-use-filters", doc=InputDocumentation(doc="Use filters in inheritance mode annotation")), ToolInput(tag="in_d_bnsf_p_tsv", input_type=File(optional=True), prefix="--dbnsfp-tsv", doc=InputDocumentation(doc="Patht to dbNSFP TSV file")), ToolInput(tag="in_d_bnsf_p_col_contig", input_type=String(optional=True), prefix="--dbnsfp-col-contig", doc=InputDocumentation(doc="Column index of contig in dbNSFP")), ToolInput(tag="in_d_bnsf_p_col_position", input_type=String(optional=True), prefix="--dbnsfp-col-position", doc=InputDocumentation(doc="Column index of position in dbNSFP")), ToolInput(tag="in_d_bnsf_p_prefix", input_type=String(optional=True), prefix="--dbnsfp-prefix", doc=InputDocumentation(doc="Prefix for dbNSFP annotations")), ToolInput(tag="in_d_bnsf_p_columns", input_type=File(optional=True), prefix="--dbnsfp-columns", doc=InputDocumentation(doc="Columns from dbDSFP file to use for annotation")), ToolInput(tag="in_bed_annotation", input_type=File(optional=True), prefix="--bed-annotation", doc=InputDocumentation(doc="Add BED file  to  use  for  annotating.  The value\nmust  be  of   the   format  'pathToBed:infoField:\ndescription[:colNo]'.")), ToolInput(tag="in_vcf_annotation", input_type=File(optional=True), prefix="--vcf-annotation", doc=InputDocumentation(doc="Add VCF file  to  use  for  annotating.  The value\nmust be of the format 'pathToVfFile:prefix:field1,\nfield2,field3'.")), ToolInput(tag="in_tsv_annotation", input_type=File(optional=True), prefix="--tsv-annotation", doc=InputDocumentation(doc="Add TSV file  to  use  for  annotating.  The value\nmust   be    of    the    format   'pathToTsvFile:\noneBasedOffset:colContig:colStart:colEnd:colRef\n(or=0):colAlt(or=0):isRefAnnotated(R=yes,A=no):\ncolValue:fieldType:fieldName:fieldDescription:\naccumulationStrategy'.")), ToolInput(tag="in_use_threshold_filters", input_type=Boolean(optional=True), prefix="--use-threshold-filters", doc=InputDocumentation(doc="Use threshold-based filters")), ToolInput(tag="in_gt_thresh_filt_min_cov_het", input_type=Int(optional=True), prefix="--gt-thresh-filt-min-cov-het", doc=InputDocumentation(doc="Minimal coverage for het. call")), ToolInput(tag="in_gt_thresh_filt_min_cov_hom_alt", input_type=Int(optional=True), prefix="--gt-thresh-filt-min-cov-hom-alt", doc=InputDocumentation(doc="Minimal coverage for hom. alt calls")), ToolInput(tag="in_gt_thresh_filt_max_cov", input_type=Int(optional=True), prefix="--gt-thresh-filt-max-cov", doc=InputDocumentation(doc="Maximal coverage for a sample")), ToolInput(tag="in_gt_thresh_filt_min_gq", input_type=Int(optional=True), prefix="--gt-thresh-filt-min-gq", doc=InputDocumentation(doc="Minimal genotype call quality")), ToolInput(tag="in_gt_thresh_filt_min_aaf_het", input_type=Int(optional=True), prefix="--gt-thresh-filt-min-aaf-het", doc=InputDocumentation(doc="Minimal het. call alternate allele fraction")), ToolInput(tag="in_gt_thresh_filt_max_aaf_het", input_type=Int(optional=True), prefix="--gt-thresh-filt-max-aaf-het", doc=InputDocumentation(doc="Maximal het. call alternate allele fraction")), ToolInput(tag="in_gt_thresh_filt_min_aaf_hom_alt", input_type=Int(optional=True), prefix="--gt-thresh-filt-min-aaf-hom-alt", doc=InputDocumentation(doc="Minimal hom. alt call alternate allele fraction")), ToolInput(tag="in_gt_thresh_filt_max_aaf_hom_ref", input_type=Int(optional=True), prefix="--gt-thresh-filt-max-aaf-hom-ref", doc=InputDocumentation(doc="Maximal hom. ref call alternate allele fraction")), ToolInput(tag="in_var_thresh_max_allele_freq_ad", input_type=Int(optional=True), prefix="--var-thresh-max-allele-freq-ad", doc=InputDocumentation(doc="Maximal allele  fraction  for  autosomal  dominant\ninheritance mode")), ToolInput(tag="in_var_thresh_max_allele_freq_ar", input_type=Int(optional=True), prefix="--var-thresh-max-allele-freq-ar", doc=InputDocumentation(doc="Maximal allele  fraction  for  autosomal recessive\ninheritance mode")), ToolInput(tag="in_var_thresh_max_hom_alt_ex_ac", input_type=Int(optional=True), prefix="--var-thresh-max-hom-alt-exac", doc=InputDocumentation(doc="Maximal count in homozygous  state  in ExAC before\nignoring")), ToolInput(tag="in_var_thresh_max_hom_alt_gone_k", input_type=Int(optional=True), prefix="--var-thresh-max-hom-alt-g1k", doc=InputDocumentation(doc="Maximal count in homozygous  state  in ExAC before\nignoring")), ToolInput(tag="in_de_novo_max_parent_ad_two", input_type=Int(optional=True), prefix="--de-novo-max-parent-ad2", doc=InputDocumentation(doc="Maximal support of  alternative  allele  in parent\nfor de novo variants.")), ToolInput(tag="in_enable_off_target_filter", input_type=Boolean(optional=True), prefix="--enable-off-target-filter", doc=InputDocumentation(doc="Enable filter for  on/off-target  based  on effect\nimpact")), ToolInput(tag="in_utr_is_off_target", input_type=Boolean(optional=True), prefix="--utr-is-off-target", doc=InputDocumentation(doc="Make UTR count as off-target  (default is to count\nUTR as on-target)")), ToolInput(tag="in_intronic_splice_is_off_target", input_type=Boolean(optional=True), prefix="--intronic-splice-is-off-target", doc=InputDocumentation(doc="Make intronic (non-consensus  site)  splice region\ncount as off-target (default  is  to  count as on-\ntarget)")), ToolInput(tag="in_no_escape_ann_field", input_type=Boolean(optional=True), prefix="--no-escape-ann-field", doc=InputDocumentation(doc="Disable escaping of INFO/ANN field in VCF output")), ToolInput(tag="in_show_all", input_type=Boolean(optional=True), prefix="--show-all", doc=InputDocumentation(doc="Show all effects")), ToolInput(tag="in_no_three_prime_shifting", input_type=Boolean(optional=True), prefix="--no-3-prime-shifting", doc=InputDocumentation(doc="Disable shifting towards 3' of transcript")), ToolInput(tag="in_three_letter_amino_acids", input_type=Boolean(optional=True), prefix="--3-letter-amino-acids", doc=InputDocumentation(doc="Enable usage of 3 letter amino acid codes")), ToolInput(tag="in_report_no_progress", input_type=Boolean(optional=True), prefix="--report-no-progress", doc=InputDocumentation(doc="Disable progress report, more quiet mode")), ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="--verbose", doc=InputDocumentation(doc="Enable verbose mode")), ToolInput(tag="in_very_verbose", input_type=Boolean(optional=True), prefix="--very-verbose", doc=InputDocumentation(doc="Enable very verbose mode")), ToolInput(tag="in_http_proxy", input_type=String(optional=True), prefix="--http-proxy", doc=InputDocumentation(doc="Set HTTP proxy to use, if any")), ToolInput(tag="in_https_proxy", input_type=String(optional=True), prefix="--https-proxy", doc=InputDocumentation(doc="Set HTTPS proxy to use, if any")), ToolInput(tag="in_ftp_proxy", input_type=String(optional=True), prefix="--ftp-proxy", doc=InputDocumentation(doc="Set FTP proxy to use, if any")), ToolInput(tag="in_annotate", input_type=String(), position=0, doc=InputDocumentation(doc="VCF  file  with   single  individual  as")), ToolInput(tag="in_singleton", input_type=String(), position=1, doc=InputDocumentation(doc="pedigree  (singleton   assumed   to  be")), ToolInput(tag="in_if", input_type=String(), position=0, doc=InputDocumentation(doc="one  parent's  genotype   is  affected,  apply")), ToolInput(tag="in_use", input_type=String(), position=0, doc=InputDocumentation(doc="advanced   pedigree-based   filters   (mainly"))], outputs=[ToolOutput(tag="out_output_vcf", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output_vcf", type_hint=File()), doc=OutputDocumentation(doc="Path to output VCF file"))], container="quay.io/biocontainers/jannovar-cli:0.35--0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Jannovar_Annotate_Vcf_V0_1_0().translate("wdl")

