from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, Int, File, String, Directory, Float

Anise_V0_1_0 = CommandToolBuilder(tool="anise", base_command=["anise"], inputs=[ToolInput(tag="in_quiet", input_type=Boolean(optional=True), prefix="--quiet", doc=InputDocumentation(doc="Set verbosity to a minimum.")), ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="--verbose", doc=InputDocumentation(doc="Enable verbose output.")), ToolInput(tag="in_very_verbose", input_type=Boolean(optional=True), prefix="--very-verbose", doc=InputDocumentation(doc="Enable very verbose output.")), ToolInput(tag="in_num_threads", input_type=Int(optional=True), prefix="--num-threads", doc=InputDocumentation(doc="Number of threads to use. In range [1..inf]. Default: 1.")), ToolInput(tag="in_debug_site_id", input_type=Int(optional=True), prefix="--debug-site-id", doc=InputDocumentation(doc="Debug site ID (-1 to disable). In range [-1..inf]. Default: -1.")), ToolInput(tag="in_debug_step_no", input_type=Int(optional=True), prefix="--debug-step-no", doc=InputDocumentation(doc="Debug step no (-1 to disable). In range [-1..inf]. Default: -1.")), ToolInput(tag="in_no_auto_tuning", input_type=Boolean(optional=True), prefix="--no-auto-tuning", doc=InputDocumentation(doc="Disable auto-tuning (see below).")), ToolInput(tag="in_input_reference", input_type=File(optional=True), prefix="--input-reference", doc=InputDocumentation(doc="Input FASTA file with reference. Valid filetypes are: fa and fasta.")), ToolInput(tag="in_input_vcf", input_type=File(optional=True), prefix="--input-vcf", doc=InputDocumentation(doc="Input VCF file with insert site candidates. Valid filetype is: vcf.")), ToolInput(tag="in_input_mapping", input_type=File(optional=True), prefix="--input-mapping", doc=InputDocumentation(doc="{sam,bam}\nInput SAM/BAM mapping file. Valid filetypes are: sam and bam.")), ToolInput(tag="in_output_fast_a", input_type=String(optional=True), prefix="--output-fasta", doc=InputDocumentation(doc="Output FASTA with contigs Valid filetypes are: fa and fasta.")), ToolInput(tag="in_output_mapping", input_type=File(optional=True), prefix="--output-mapping", doc=InputDocumentation(doc="Output SAM/BAM file with mapping fo reads to contigs in\n--output-fasta. Valid filetypes are: sam and bam.")), ToolInput(tag="in_output_debug_dir", input_type=Directory(optional=True), prefix="--output-debug-dir", doc=InputDocumentation(doc="Directory for debug output. Leave empty for no such output.")), ToolInput(tag="in_clean_up_tmp_files", input_type=Boolean(optional=True), prefix="--clean-up-tmp-files", doc=InputDocumentation(doc="Clean up temporary files.")), ToolInput(tag="in_recursion_max_steps", input_type=Int(optional=True), prefix="--recursion-max-steps", doc=InputDocumentation(doc="Maximal recursion depth. 0 for infinity. In range [0..inf]. Default:\n50.")), ToolInput(tag="in_no_realign_assembly", input_type=Boolean(optional=True), prefix="--no-realign-assembly", doc=InputDocumentation(doc="Do not realign the reads after assembly.")), ToolInput(tag="in_max_reads_factor", input_type=Float(optional=True), prefix="--max-reads-factor", doc=InputDocumentation(doc="Factor to use for the maximal read computation. ANISE stops for a\nsite if more than the number of reads expected from the expected\ncoverage times the max reads factor are assigned to the site. In\nrange [1..inf]. Default: 2.")), ToolInput(tag="in_stop_initial_read_count", input_type=Int(optional=True), prefix="--stop-initial-read-count", doc=InputDocumentation(doc="If there are more than this number of reads for a site in the\ninitial round then no assembly is performed. In range [0..inf].\nDefault: 4000.")), ToolInput(tag="in_stop_tex_read_count", input_type=Int(optional=True), prefix="--stop-tex-read-count", doc=InputDocumentation(doc="If there are more than this number of reads for a site in a later\nround then no triplet library extension is performed. In range\n[0..inf]. Default: 3000.")), ToolInput(tag="in_stop_read_count", input_type=Int(optional=True), prefix="--stop-read-count", doc=InputDocumentation(doc="If there are more than this number of reads for a site in a later\nround then no assembly is performed. In range [0..inf]. Default:\n30000.")), ToolInput(tag="in_stop_coverage", input_type=Int(optional=True), prefix="--stop-coverage", doc=InputDocumentation(doc="If the length sum of all reads for a site divided by the length sum\nof its contigs is higher than this value before assembly then the\nsite is deactivated. Set to 0 to deactivate check. In range\n[0..inf]. Default: 100.")), ToolInput(tag="in_realignment_bandwidth", input_type=Int(optional=True), prefix="--realignment-bandwidth", doc=InputDocumentation(doc="The bandwidth to use in the realignment step. In range [0..inf].\nDefault: 40.")), ToolInput(tag="in_realignment_border", input_type=Int(optional=True), prefix="--realignment-border", doc=InputDocumentation(doc="The border from the profile to extract around alignments. In range\n[0..inf]. Default: 30.")), ToolInput(tag="in_no_separate_repeats", input_type=Boolean(optional=True), prefix="--no-separate-repeats", doc=InputDocumentation(doc="Dont' repeat separation algorithm after realignment.")), ToolInput(tag="in_rep_sep_tammi_method", input_type=String(optional=True), prefix="--repsep-tammi-method", doc=InputDocumentation(doc="Variant of the Tammi method to use for repeat separation (simple or\nphred). One of phred and simple. Default: simple.")), ToolInput(tag="in_rep_sep_p_err", input_type=Float(optional=True), prefix="--repsep-p-err", doc=InputDocumentation(doc="Repeat separation per-base error for simple Tammi method. In range\n[0.0..1.0]. Default: 0.01.")), ToolInput(tag="in_rep_sep_max_random_correlation", input_type=Float(optional=True), prefix="--repsep-max-random-correlation", doc=InputDocumentation(doc="Repeat separation maximal random correlation. In range [0.0..1.0].\nDefault: 0.00001.")), ToolInput(tag="in_rep_sep_tau_min", input_type=Int(optional=True), prefix="--repsep-tau-min", doc=InputDocumentation(doc="Repeat separation tau_min value. In range [0..inf]. Default: 100000.")), ToolInput(tag="in_rep_sep_r_min", input_type=Int(optional=True), prefix="--repsep-r-min", doc=InputDocumentation(doc="Repeat separation r_min value. In range [0..inf]. Default: 100000.")), ToolInput(tag="in_rep_sep_min_overlap", input_type=Int(optional=True), prefix="--repsep-min-overlap", doc=InputDocumentation(doc="Repeat separation minimal overlap value. In range [0..inf]. Default:\n2.")), ToolInput(tag="in_rep_sep_start_compression_at", input_type=Int(optional=True), prefix="--repsep-start-compression-at", doc=InputDocumentation(doc="Repeat separation start compression. In range [2..inf]. Default:\n100.")), ToolInput(tag="in_rep_sep_split_d_min", input_type=Boolean(optional=True), prefix="--repsep-split-d-min", doc=InputDocumentation(doc="Repeat separation split at d_min deviations.")), ToolInput(tag="in_fragment_size_factor", input_type=Int(optional=True), prefix="--fragment-size-factor", doc=InputDocumentation(doc="Factor to multiple fragment size stddev with to get allowed error.\nIn range [0..inf]. Default: 8.")), ToolInput(tag="in_auto_library_num_records", input_type=Int(optional=True), prefix="--auto-library-num-records", doc=InputDocumentation(doc="Number of records to use for automatic library evaluation. Set to 0\nto evaluate all. In range [0..inf]. Default: 100000.")), ToolInput(tag="in_fragment_size_median", input_type=Int(optional=True), prefix="--fragment-size-median", doc=InputDocumentation(doc="Median fragment size. In range [0..inf]. Default: 250.")), ToolInput(tag="in_fragment_size_std_dev", input_type=Int(optional=True), prefix="--fragment-size-std-dev", doc=InputDocumentation(doc="Fragment size standard deviation. In range [0..inf]. Default: 30.")), ToolInput(tag="in_fragment_default_orientation", input_type=String(optional=True), prefix="--fragment-default-orientation", doc=InputDocumentation(doc="Default orientation. One of F+, F-, R+, and R-. Default: R+.")), ToolInput(tag="in_assembly_site_window_radius", input_type=Int(optional=True), prefix="--assembly-site-window-radius", doc=InputDocumentation(doc="Radius around insert site to cut for initial contigs. In range\n[100..inf]. Default: 1000.")), ToolInput(tag="in_assembly_site_fringe_radius", input_type=Int(optional=True), prefix="--assembly-site-fringe-radius", doc=InputDocumentation(doc="Radius around insert site to cut for collecting clippings. Set to -1\n(default) to consider all records with >= 15 clipped bases. In range\n[-1..inf]. Default: -1.")), ToolInput(tag="in_read_mapping_error_rate", input_type=Float(optional=True), prefix="--read-mapping-error-rate", doc=InputDocumentation(doc="Error rate of internal read mapping step in percent. In range\n[0..20]. Default: 5.")), ToolInput(tag="in_read_mapping_batch_size", input_type=Int(optional=True), prefix="--read-mapping-batch-size", doc=InputDocumentation(doc="Batch size for read mapping. In range [1..inf]. Default: 10000.")), ToolInput(tag="in_overlap_per_min_overlap_ratio", input_type=Int(optional=True), prefix="--overlapper-min-overlap-ratio", doc=InputDocumentation(doc="Overlapper min overlap rate in percent of the longer read. In range\n[0..inf]. Default: 40.")), ToolInput(tag="in_overlap_per_max_error_rate", input_type=Int(optional=True), prefix="--overlapper-max-error-rate", doc=InputDocumentation(doc="Overlapper maximum error rate in percent. In range [0..30]. Default:\n5.")), ToolInput(tag="in_no_read_correction", input_type=Boolean(optional=True), prefix="--no-read-correction", doc=InputDocumentation(doc="Whether or not to perform read correction")), ToolInput(tag="in_msa_score_match", input_type=Int(optional=True), prefix="--msa-score-match", doc=InputDocumentation(doc="PW match score in MSA. Default: 2.")), ToolInput(tag="in_msa_score_mismatch", input_type=Int(optional=True), prefix="--msa-score-mismatch", doc=InputDocumentation(doc="PW mismatch score in MSA. Default: -6.")), ToolInput(tag="in_msa_score_gap_open", input_type=Int(optional=True), prefix="--msa-score-gap-open", doc=InputDocumentation(doc="PW gap open score in MSA. Default: -4.")), ToolInput(tag="in_msa_score_gap_extend", input_type=Int(optional=True), prefix="--msa-score-gap-extend", doc=InputDocumentation(doc="PW gap extension score in MSA. Default: -9.")), ToolInput(tag="in_consensus_min_base_support", input_type=Int(optional=True), prefix="--consensus-min-base-support", doc=InputDocumentation(doc="Minimal base support for non-N call in consensus calling. Default:\n2.")), ToolInput(tag="in_consensus_min_contig_length_rate", input_type=Int(optional=True), prefix="--consensus-min-contig-length-rate", doc=InputDocumentation(doc="Minimal contig length in percent of average read length. Default:\n150.")), ToolInput(tag="in_repeat_dot", input_type=String(), position=0, doc=InputDocumentation(doc="AUTO TUNING"))], outputs=[ToolOutput(tag="out_output_mapping", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output_mapping", type_hint=File()), doc=OutputDocumentation(doc="Output SAM/BAM file with mapping fo reads to contigs in\n--output-fasta. Valid filetypes are: sam and bam.")), ToolOutput(tag="out_output_debug_dir", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_output_debug_dir", type_hint=File()), doc=OutputDocumentation(doc="Directory for debug output. Leave empty for no such output."))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Anise_V0_1_0().translate("wdl", allow_empty_container=True)

