class: CommandLineTool
id: autoAug.pl.cwl
inputs:
- id: in_genome
  doc: fasta file with DNA sequences for training
  type: File?
  inputBinding:
    prefix: --genome
- id: in_training_set
  doc: genesfile contains training genes in Genbank, GFF or protein FASTA format
  type: File?
  inputBinding:
    prefix: --trainingset
- id: in_species
  doc: species name as used by AUGUSTUS
  type: string?
  inputBinding:
    prefix: --species
- id: in_hints
  doc: hints for gene predictions with AUGUSTUS
  type: string?
  inputBinding:
    prefix: --hints
- id: in_est_ali
  doc: cDNA alignments in PSL format (as generated by BLAT and GMAP) are used to construct
    UTRs
  type: string?
  inputBinding:
    prefix: --estali
- id: in_pasa
  doc: use PASA to construct a training set
  type: boolean?
  inputBinding:
    prefix: --pasa
- id: in_cdna
  doc: a fasta file with cDNA sequences (ESTs, mRNA)
  type: File?
  inputBinding:
    prefix: --cdna
- id: in_pasa_polya_hints
  doc: use PASA polyA hints as hints for the prediction
  type: boolean?
  inputBinding:
    prefix: --pasapolyAhints
- id: in_use_existing
  doc: use and change the present config and parameter files if they exist for 'species'
  type: boolean?
  inputBinding:
    prefix: --useexisting
- id: in_verbose
  doc: print more status info. Cumulative option, e.g. use -v -v -v to make this script
    very verbose
  type: boolean?
  inputBinding:
    prefix: --verbose
- id: in_no_utr
  doc: do not train and predict UTRs.
  type: boolean?
  inputBinding:
    prefix: --noutr
- id: in_working_dir
  doc: "=/path/to/wd/           In the working directory results and temporary files\
    \ are stored.\nDefault: current working directory"
  type: boolean?
  inputBinding:
    prefix: --workingdir
- id: in_single_cpu
  doc: run the complete program sequentially instead of parallel execution of jobs
    on a cluster
  type: boolean?
  inputBinding:
    prefix: --singleCPU
- id: in_noninteractive
  doc: bypass all manual interaction when using a SGE cluster
  type: boolean?
  inputBinding:
    prefix: --noninteractive
- id: in_cname
  doc: cluster name, only use it when "noninteractive" default:fe
  type: string?
  inputBinding:
    prefix: --cname
- id: in_index
  doc: step index, default:0
  type: long?
  inputBinding:
    prefix: --index
- id: in_opt_rounds
  doc: optimization rounds - each meta parameter is optimized this often (default
    1)
  type: long?
  inputBinding:
    prefix: --optrounds
- id: in_max_intron_len
  doc: maximal length of an intron as used by PASA and BLAT, not by AUGUSTUS (default
    100000)
  type: long?
  inputBinding:
    prefix: --maxIntronLen
- id: in_use_gmap_for_pasa
  doc: use GMAP instead of BLAT in the PASA run
  type: boolean?
  inputBinding:
    prefix: --useGMAPforPASA
outputs:
- id: out_stdout
  doc: Standard output stream
  type: stdout
hints: []
cwlVersion: v1.1
baseCommand:
- autoAug.pl
