from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, String

Spaln_V0_1_0 = CommandToolBuilder(tool="spaln", base_command=["spaln"], inputs=[ToolInput(tag="in_generate_local_table", input_type=Boolean(optional=True), prefix="-E", doc=InputDocumentation(doc="Generate local lookup table for each block")), ToolInput(tag="in_xc", input_type=Boolean(optional=True), prefix="-XC", doc=InputDocumentation(doc="#    number of bit patterns < 6 (1)")), ToolInput(tag="in_xg", input_type=Boolean(optional=True), prefix="-XG", doc=InputDocumentation(doc="#    Maximum expected gene size (inferred from genome|db size)")), ToolInput(tag="in_xk", input_type=Boolean(optional=True), prefix="-Xk", doc=InputDocumentation(doc="#    Word size (inferred from genome|db size)")), ToolInput(tag="in_xb", input_type=Boolean(optional=True), prefix="-Xb", doc=InputDocumentation(doc="#    Block size (inferred from genome|db size)")), ToolInput(tag="in_x_a", input_type=Boolean(optional=True), prefix="-Xa", doc=InputDocumentation(doc="#    Abundance factor (10)")), ToolInput(tag="in_xr", input_type=Boolean(optional=True), prefix="-Xr", doc=InputDocumentation(doc="#    Minimum ORF length with -KP (30))")), ToolInput(tag="in_gzipped_output", input_type=Boolean(optional=True), prefix="-g", doc=InputDocumentation(doc="gzipped output")), ToolInput(tag="in_mutlithread_operation_threads", input_type=Boolean(optional=True), prefix="-t", doc=InputDocumentation(doc="#     Mutli-thread operation with # threads")), ToolInput(tag="in_minimum_score_report", input_type=Boolean(optional=True), prefix="-H", doc=InputDocumentation(doc="#     Minimum score for report (35)")), ToolInput(tag="in_ls", input_type=String(optional=True), prefix="-LS", doc=InputDocumentation(doc="#        semi-global or local alignment (-L)")), ToolInput(tag="in_number_outputs_omitted", input_type=Boolean(optional=True), prefix="-M", doc=InputDocumentation(doc="#[,#2]        Number of outputs per query (1) (4 if # is omitted)\n#2 (4) specifies the max number of candidate loci\nThis option is effective only for map-and-align modes")), ToolInput(tag="in_gvsac_gffgene_alignment", input_type=Boolean(optional=True), prefix="-O", doc=InputDocumentation(doc="#[,#2,..] (GvsA|C)    0:Gff3_gene; 1:alignment; 2:Gff3_match; 3:Bed; 4:exon-inf;\n5:intron-inf; 6:cDNA; 7:translated; 8:block-only;\n10:SAM; 12:binary; 15:query+GS (4)")), ToolInput(tag="in_dp_hspsearch_blocksearch", input_type=Boolean(optional=True), prefix="-Q", doc=InputDocumentation(doc="#     0:DP; 1-3:HSP-Search; 4-7; Block-Search (3)")), ToolInput(tag="in_read_block_file", input_type=Boolean(optional=True), prefix="-R", doc=InputDocumentation(doc="$     Read block information file *.bkn, *.bkp or *.bka")), ToolInput(tag="in_orientation_annotation_forward", input_type=Boolean(optional=True), prefix="-S", doc=InputDocumentation(doc="#     Orientation. 0:annotation; 1:forward; 2:reverse; 3:both (3)")), ToolInput(tag="in_subdirectory_where_reside", input_type=Boolean(optional=True), prefix="-T", doc=InputDocumentation(doc="$     Subdirectory where species-specific parameters reside")), ToolInput(tag="in_specify_aadb_run", input_type=Boolean(optional=True), prefix="-a", doc=InputDocumentation(doc="$     Specify AAdb. Must run `makeidx.pl -ia' breforehand")), ToolInput(tag="in_same_db_sequences", input_type=Boolean(optional=True), prefix="-A", doc=InputDocumentation(doc="$     Same as -a but db sequences are stored in memory")), ToolInput(tag="in_specify_genome_run", input_type=Boolean(optional=True), prefix="-d", doc=InputDocumentation(doc="$     Specify genome. Must run `makeidx.pl -i[n|p]' breforehand")), ToolInput(tag="in_same_d_db", input_type=Boolean(optional=True), prefix="-D", doc=InputDocumentation(doc="$     Same as -d but db sequences are stored in memory")), ToolInput(tag="in_number_characters_line", input_type=Boolean(optional=True), prefix="-l", doc=InputDocumentation(doc="#     Number of characters per line in alignment (60)")), ToolInput(tag="in_filedirectoryprefix_where_written", input_type=Boolean(optional=True), prefix="-o", doc=InputDocumentation(doc="$     File/directory/prefix where results are written (stdout)")), ToolInput(tag="in_pa", input_type=Boolean(optional=True), prefix="-pa", doc=InputDocumentation(doc="#    Remove 3' poly A >= # (0: don't remove)")), ToolInput(tag="in_pw", input_type=Boolean(optional=True), prefix="-pw", doc=InputDocumentation(doc="Report results even if the score is below the threshold")), ToolInput(tag="in_pq", input_type=Boolean(optional=True), prefix="-pq", doc=InputDocumentation(doc="Quiet mode")), ToolInput(tag="in_report_information_block", input_type=Boolean(optional=True), prefix="-r", doc=InputDocumentation(doc="$     Report information about block data file")), ToolInput(tag="in__gapextension_penalty", input_type=Boolean(optional=True), prefix="-u", doc=InputDocumentation(doc="#     Gap-extension penalty (3)")), ToolInput(tag="in__gapopen_penalty", input_type=Boolean(optional=True), prefix="-v", doc=InputDocumentation(doc="#     Gap-open penalty (8)")), ToolInput(tag="in_band_width_scan", input_type=Boolean(optional=True), prefix="-w", doc=InputDocumentation(doc="#     Band width for DP matrix scan (100)")), ToolInput(tag="in_ya", input_type=Boolean(optional=True), prefix="-ya", doc=InputDocumentation(doc="#    Stringency of splice site. 0->3:strong->weak")), ToolInput(tag="in_yl_three", input_type=Boolean(optional=True), prefix="-yl3", doc=InputDocumentation(doc="Ddouble affine gap penalty")), ToolInput(tag="in_ym", input_type=Boolean(optional=True), prefix="-ym", doc=InputDocumentation(doc="#    Nucleotide match score (2)")), ToolInput(tag="in_yn", input_type=Boolean(optional=True), prefix="-yn", doc=InputDocumentation(doc="#    Nucleotide mismatch score (-6)")), ToolInput(tag="in_yo", input_type=Boolean(optional=True), prefix="-yo", doc=InputDocumentation(doc="#    Penalty for a premature termination codon (100)")), ToolInput(tag="in_yx", input_type=Boolean(optional=True), prefix="-yx", doc=InputDocumentation(doc="#    Penalty for a frame shift error (100)")), ToolInput(tag="in_yy", input_type=Boolean(optional=True), prefix="-yy", doc=InputDocumentation(doc="#    Weight for splice site signal (8)")), ToolInput(tag="in_weight_coding_potential", input_type=Boolean(optional=True), prefix="-yz", doc=InputDocumentation(doc="#    Weight for coding potential (2)")), ToolInput(tag="in_yb", input_type=Boolean(optional=True), prefix="-yB", doc=InputDocumentation(doc="#    Weight for branch point signal (0)")), ToolInput(tag="in_yi", input_type=Boolean(optional=True), prefix="-yI", doc=InputDocumentation(doc="$    Intron length distribution")), ToolInput(tag="in_yl", input_type=Boolean(optional=True), prefix="-yL", doc=InputDocumentation(doc="#    Minimum expected length of intron (30)")), ToolInput(tag="in_ys", input_type=Boolean(optional=True), prefix="-yS", doc=InputDocumentation(doc="[#]  Use species-specific parameter set (0.0/0.5)")), ToolInput(tag="in_yx_zero", input_type=Boolean(optional=True), prefix="-yX0", doc=InputDocumentation(doc="Don't use parameter set for cross-species comparison")), ToolInput(tag="in_weight_intron_potential", input_type=Boolean(optional=True), prefix="-yZ", doc=InputDocumentation(doc="#    Weight for intron potential (0)")), ToolInput(tag="in_kd", input_type=Boolean(optional=True), prefix="-KD", doc=InputDocumentation(doc="")), ToolInput(tag="in_var_45", input_type=Boolean(optional=True), prefix="-W", doc=InputDocumentation(doc="")), ToolInput(tag="in_genome_dot_bkn", input_type=String(optional=True), position=0, doc=InputDocumentation(doc="")), ToolInput(tag="in_w_options", input_type=String(optional=True), position=1, doc=InputDocumentation(doc="")), ToolInput(tag="in_genome_dot_mfa", input_type=String(), position=2, doc=InputDocumentation(doc=""))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Spaln_V0_1_0().translate("wdl", allow_empty_container=True)

