from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, File

Mhap_V0_1_0 = CommandToolBuilder(tool="mhap", base_command=["mhap"], inputs=[ToolInput(tag="in_filter_threshold", input_type=Boolean(optional=True), prefix="--filter-threshold", doc=InputDocumentation(doc=", default = 1.0E-5\n[double], The cutoff at which the k-mer in the k-mer filter file is considered repetitive. This value for a specific k-mer is specified in the second column in the filter file. If no filter file is provided, this option is ignored.")), ToolInput(tag="in_max_shift", input_type=Boolean(optional=True), prefix="--max-shift", doc=InputDocumentation(doc=", default = 0.2\n[double], Region size to the left and right of the estimated overlap, as derived from the median shift and sequence length, where a k-mer matches are still considered valid. Second stage filter only.")), ToolInput(tag="in_min_olap_length", input_type=Boolean(optional=True), prefix="--min-olap-length", doc=InputDocumentation(doc=", default = 116\n[int], The minimum length of the read that used for overlapping. Used to filter out short reads from FASTA file.")), ToolInput(tag="in_min_store_length", input_type=Boolean(optional=True), prefix="--min-store-length", doc=InputDocumentation(doc=", default = 0\n[int], The minimum length of the read that is stored in the box. Used to filter out short reads from FASTA file.")), ToolInput(tag="in_no_rc", input_type=Boolean(optional=True), prefix="--no-rc", doc=InputDocumentation(doc=", default = false\nDo not store or do comparison of the reverse compliment strings.")), ToolInput(tag="in_no_self", input_type=Boolean(optional=True), prefix="--no-self", doc=InputDocumentation(doc=", default = false\nDo not compute the overlaps between sequences inside a box. Should be used when the to and from sequences are coming from different files.")), ToolInput(tag="in_no_tf", input_type=Boolean(optional=True), prefix="--no-tf", doc=InputDocumentation(doc=", default = false\nDo not perform the tf weighing, in the tf-idf weighing.")), ToolInput(tag="in_num_hashes", input_type=Boolean(optional=True), prefix="--num-hashes", doc=InputDocumentation(doc=", default = 512\n[int], Number of min-mers to be used in MinHashing.")), ToolInput(tag="in_num_min_matches", input_type=Boolean(optional=True), prefix="--num-min-matches", doc=InputDocumentation(doc=", default = 3\n[int], Minimum # min-mer that must be shared before computing second stage filter. Any sequences below that value are considered non-overlapping.")), ToolInput(tag="in_num_threads", input_type=Boolean(optional=True), prefix="--num-threads", doc=InputDocumentation(doc=", default = 8\n[int], nNumber of threads to use for computation. Typically set to #cores.")), ToolInput(tag="in_ordered_km_er_size", input_type=Boolean(optional=True), prefix="--ordered-kmer-size", doc=InputDocumentation(doc=", default = 12\n[int] The size of k-mers used in the ordered second stage filter.")), ToolInput(tag="in_ordered_sketch_size", input_type=Boolean(optional=True), prefix="--ordered-sketch-size", doc=InputDocumentation(doc=", default = 1536\n[int] The sketch size for second stage filter.")), ToolInput(tag="in_repeat_idf_scale", input_type=Boolean(optional=True), prefix="--repeat-idf-scale", doc=InputDocumentation(doc=", default = 3.0\n[double] The upper range of the idf (from tf-idf) scale. The full scale will be [1,X], where X is the parameter.")), ToolInput(tag="in_repeat_weight", input_type=Boolean(optional=True), prefix="--repeat-weight", doc=InputDocumentation(doc=", default = 0.9\n[double] Repeat suppression strength for tf-idf weighing. <0.0 do unweighted MinHash (version 1.0), >=1.0 do only the tf weighing. To perform no idf weighting, do no supply -f option.")), ToolInput(tag="in_settings", input_type=Boolean(optional=True), prefix="--settings", doc=InputDocumentation(doc=", default = 0\nSet all unset parameters for the default settings. Same defaults are applied to Nanopore and Pacbio reads. 0) None, 1) Default, 2) Fast, 3) Sensitive.")), ToolInput(tag="in_store_full_id", input_type=File(optional=True), prefix="--store-full-id", doc=InputDocumentation(doc=", default = false\nStore full IDs as seen in FASTA files, rather than storing just the sequence position in the file. Some FASTA files have long IDS, slowing output of results. This options is ignored when using compressed file format. Indexed file (-s) is indexed first, followed by -q files in alphabetical order.")), ToolInput(tag="in_supress_noise", input_type=Boolean(optional=True), prefix="--supress-noise", doc=InputDocumentation(doc=", default = 0\n[int] 0) Does nothing, 1) completely removes any k-mers not specified in the filter file, 2) supresses k-mers not specified in the filter file, similar to repeats.")), ToolInput(tag="in_threshold", input_type=Boolean(optional=True), prefix="--threshold", doc=InputDocumentation(doc=", default = 0.78\n[double], The threshold cutoff for the second stage sort-merge filter. This is based on the identity score computed from the Jaccard distance of k-mers (size given by ordered-kmer-size) in the overlapping regions.")), ToolInput(tag="in_default_kmer_file", input_type=File(optional=True), prefix="-f", doc=InputDocumentation(doc=", default = ''\nk-mer filter file used for filtering out highly repetative k-mers. Must be sorted in descending order of frequency (second column).")), ToolInput(tag="in_default_size_used", input_type=Boolean(optional=True), prefix="-k", doc=InputDocumentation(doc=", default = 16\n[int], k-mer size used for MinHashing. The k-mer size for second stage filter is seperate, and can also be modified.")), ToolInput(tag="in_only_directory_containing", input_type=Boolean(optional=True), prefix="-p", doc=InputDocumentation(doc=", default = ''\nUsage 2 only. The directory containing FASTA files that should be converted to binary format for storage.")), ToolInput(tag="in_default_usage_fasta", input_type=File(optional=True), prefix="-q", doc=InputDocumentation(doc=", default = ''\nUsage 1: The FASTA file of reads, or a directory of files, that will be compared to the set of reads in the box (see -s). Usage 2: The output directory for the binary formatted dat files.")), ToolInput(tag="in_only_fasta_binary", input_type=Boolean(optional=True), prefix="-s", doc=InputDocumentation(doc=", default = ''\nUsage 1 only. The FASTA or binary dat file (see Usage 2) of reads that will be stored in a box, and that all subsequent reads will be compared to.\n"))], outputs=[ToolOutput(tag="out_store_full_id", output_type=File(optional=True), selector=InputSelector(input_to_select="in_store_full_id", type_hint=File()), doc=OutputDocumentation(doc=", default = false\nStore full IDs as seen in FASTA files, rather than storing just the sequence position in the file. Some FASTA files have long IDS, slowing output of results. This options is ignored when using compressed file format. Indexed file (-s) is indexed first, followed by -q files in alphabetical order.")), ToolOutput(tag="out_default_kmer_file", output_type=File(optional=True), selector=InputSelector(input_to_select="in_default_kmer_file", type_hint=File()), doc=OutputDocumentation(doc=", default = ''\nk-mer filter file used for filtering out highly repetative k-mers. Must be sorted in descending order of frequency (second column).")), ToolOutput(tag="out_default_usage_fasta", output_type=File(optional=True), selector=InputSelector(input_to_select="in_default_usage_fasta", type_hint=File()), doc=OutputDocumentation(doc=", default = ''\nUsage 1: The FASTA file of reads, or a directory of files, that will be compared to the set of reads in the box (see -s). Usage 2: The output directory for the binary formatted dat files."))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Mhap_V0_1_0().translate("wdl", allow_empty_container=True)

