class: CommandLineTool
id: Periodicity.cwl
inputs:
- id: in_input
  doc: Input file(s) in bam format.
  type: File?
  inputBinding:
    prefix: --input
- id: in_an_not_dir
  doc: "transcripts annotation directory, generated by\nprepare_transcripts, a function\
    \ of RiboCode."
  type: Directory?
  inputBinding:
    prefix: --annot_dir
- id: in_coordinate_file
  doc: "The file should contain the coordinate of start and\nstop codon. Generated\
    \ by\nOutputTranscriptInfo.py.[required]"
  type: File?
  inputBinding:
    prefix: --coordinateFile
- id: in_ot_put_prefix
  doc: Prefix of output files.[required]
  type: string?
  inputBinding:
    prefix: --otput_prefix
- id: in_left_length
  doc: The left range of read length we will consider.
  type: long?
  inputBinding:
    prefix: --left_length
- id: in_right_length
  doc: The right range of read length we will consider.
  type: long?
  inputBinding:
    prefix: --right_length
- id: in_select_trans_list
  doc: "Selected transcript list used for metagene\nanalysis.This files requires the\
    \ first column must be\nthe transcript ID  with a column name."
  type: string?
  inputBinding:
    prefix: --select_trans_list
- id: in_id_type
  doc: "define the id type users input. the default is\ntranscript id, if not, will\
    \ be transformed into\ntranscript id. transcript_id=default\n"
  type: string?
  inputBinding:
    prefix: --id-type
outputs:
- id: out_stdout
  doc: Standard output stream
  type: stdout
hints:
- class: DockerRequirement
  dockerPull: quay.io/biocontainers/ribominer:0.2.3.1--pyh3252c3a_0
cwlVersion: v1.1
baseCommand:
- Periodicity
