from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import String, Boolean, Array, Int, File

Get_Count_Vectors_V0_1_0 = CommandToolBuilder(tool="get_count_vectors", base_command=["get_count_vectors"], inputs=[ToolInput(tag="in_out_prefix", input_type=String(optional=True), prefix="--out_prefix", doc=InputDocumentation(doc="Prefix to prepend to output files (default: no prefix)")), ToolInput(tag="in_format", input_type=String(optional=True), prefix="--format", doc=InputDocumentation(doc="printf-style format string for output (default:\n'%.8f')")), ToolInput(tag="in_quiet", input_type=Boolean(optional=True), prefix="--quiet", doc=InputDocumentation(doc="Suppress all warning messages. Cannot use with '-v'.")), ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="--verbose", doc=InputDocumentation(doc="Increase verbosity. With '-v', show every warning.\nWith '-vv', turn warnings into exceptions. Cannot use\nwith '-q'. (Default: show each type of warning once)")), ToolInput(tag="in_count_files", input_type=Array(t=String(), optional=True), prefix="--count_files", doc=InputDocumentation(doc="One or more count or alignment file(s) from a single\nsample or set of samples to be pooled.")), ToolInput(tag="in_count_file_format", input_type=String(optional=True), prefix="--countfile_format", doc=InputDocumentation(doc="Format of file containing alignments or counts\n(Default: BAM)")), ToolInput(tag="in_normalize", input_type=Boolean(optional=True), prefix="--normalize", doc=InputDocumentation(doc="Whether counts should be normalized to counts per\nmillion (usually not. default: False)")), ToolInput(tag="in_sum", input_type=String(optional=True), prefix="--sum", doc=InputDocumentation(doc="Sum used in normalization of counts and RPKM/RPNT\ncalculations (Default: total mapped reads/counts in\ndataset)")), ToolInput(tag="in_min_length", input_type=Int(optional=True), prefix="--min_length", doc=InputDocumentation(doc="Minimum read length required to be included (BAM &\nbowtie files only. Default: 25)")), ToolInput(tag="in_max_length", input_type=Int(optional=True), prefix="--max_length", doc=InputDocumentation(doc="Maximum read length permitted to be included (BAM &\nbowtie files only. Default: 100)")), ToolInput(tag="in_big_genome", input_type=Boolean(optional=True), prefix="--big_genome", doc=InputDocumentation(doc="Use slower but memory-efficient implementation for big\ngenomes or for memory-limited computers. For wiggle &\nbowtie files only.")), ToolInput(tag="in_five_prime_variable", input_type=Boolean(optional=True), prefix="--fiveprime_variable", doc=InputDocumentation(doc="Map read alignment to a variable offset from 5'\nposition of read, with offset determined by read\nlength. Requires `--offset` below")), ToolInput(tag="in_five_prime", input_type=Boolean(optional=True), prefix="--fiveprime", doc=InputDocumentation(doc="Map read alignment to 5' position.")), ToolInput(tag="in_three_prime", input_type=Boolean(optional=True), prefix="--threeprime", doc=InputDocumentation(doc="Map read alignment to 3' position")), ToolInput(tag="in_center", input_type=Boolean(optional=True), prefix="--center", doc=InputDocumentation(doc="Subtract N positions from each end of read, and add\n1/(length-N), to each remaining position, where N is\nspecified by `--nibble`")), ToolInput(tag="in_offset", input_type=Int(optional=True), prefix="--offset", doc=InputDocumentation(doc="For `--fiveprime` or `--threeprime`, provide an\ninteger representing the offset into the read,\nstarting from either the 5' or 3' end, at which data\nshould be plotted. For `--fiveprime_variable`, provide\nthe filename of a two-column tab-delimited text file,\nin which first column represents read length or the\nspecial keyword `'default'`, and the second column\nrepresents the offset from the five prime end of that\nread length at which the read should be mapped.\n(Default: 0)")), ToolInput(tag="in_nibble", input_type=Int(optional=True), prefix="--nibble", doc=InputDocumentation(doc="For use with `--center` only. nt to remove from each\nend of read before mapping (Default: 0)")), ToolInput(tag="in_annotation_files", input_type=File(optional=True), prefix="--annotation_files", doc=InputDocumentation(doc="[BED | BigBed | GTF2 | GFF3] [infile.[BED | BigBed | GTF2 | GFF3] ...]\nZero or more annotation files (max 1 file if BigBed)")), ToolInput(tag="in_annotation_format", input_type=String(optional=True), prefix="--annotation_format", doc=InputDocumentation(doc="Format of annotation_files (Default: GTF2). Note: GFF3\nassembly assumes SO v.2.5.2 feature ontologies, which\nmay or may not match your specific file.")), ToolInput(tag="in_add_three", input_type=Boolean(optional=True), prefix="--add_three", doc=InputDocumentation(doc="If supplied, coding regions will be extended by 3\nnucleotides at their 3' ends (except for GTF2 files\nthat explicitly include `stop_codon` features). Use if\nyour annotation file excludes stop codons from CDS.")), ToolInput(tag="in_tab_ix", input_type=Boolean(optional=True), prefix="--tabix", doc=InputDocumentation(doc="annotation_files are tabix-compressed and indexed\n(Default: False). Ignored for BigBed files.")), ToolInput(tag="in_sorted", input_type=Boolean(optional=True), prefix="--sorted", doc=InputDocumentation(doc="annotation_files are sorted by chromosomal position\n(Default: False)")), ToolInput(tag="in_bed_extra_columns", input_type=Array(t=String(), optional=True), prefix="--bed_extra_columns", doc=InputDocumentation(doc="Number of extra columns in BED file (e.g. in custom\nENCODE formats) or list of names for those columns.\n(Default: 0).")), ToolInput(tag="in_mask_bed_extra_columns", input_type=Array(t=String(), optional=True), prefix="--mask_bed_extra_columns", doc=InputDocumentation(doc="Number of extra columns in BED file (e.g. in custom\nENCODE formats) or list of names for those columns.\n(Default: 0).")), ToolInput(tag="in_maxmem", input_type=Int(optional=True), prefix="--maxmem", doc=InputDocumentation(doc="Maximum desired memory footprint in MB to devote to\nBigBed/BigWig files. May be exceeded by large queries.\n(Default: 0, No maximum)")), ToolInput(tag="in_mask_maxmem", input_type=Int(optional=True), prefix="--mask_maxmem", doc=InputDocumentation(doc="Maximum desired memory footprint in MB to devote to\nBigBed/BigWig files. May be exceeded by large queries.\n(Default: 0, No maximum)")), ToolInput(tag="in_gff_transcript_types", input_type=Array(t=String(), optional=True), prefix="--gff_transcript_types", doc=InputDocumentation(doc="GFF3 feature types to include as transcripts, even if\nno exons are present (for GFF3 only; default: use SO\nv2.5.3 specification)")), ToolInput(tag="in_gff_exon_types", input_type=Array(t=String(), optional=True), prefix="--gff_exon_types", doc=InputDocumentation(doc="GFF3 feature types to include as exons (for GFF3 only;\ndefault: use SO v2.5.3 specification)")), ToolInput(tag="in_gff_cds_types", input_type=Array(t=String(), optional=True), prefix="--gff_cds_types", doc=InputDocumentation(doc="GFF3 feature types to include as CDS (for GFF3 only;\ndefault: use SO v2.5.3 specification)")), ToolInput(tag="in_mask_gff_transcript_types", input_type=Array(t=String(), optional=True), prefix="--mask_gff_transcript_types", doc=InputDocumentation(doc="GFF3 feature types to include as transcripts, even if\nno exons are present (for GFF3 only; default: use SO\nv2.5.3 specification)")), ToolInput(tag="in_mask_gff_exon_types", input_type=Array(t=String(), optional=True), prefix="--mask_gff_exon_types", doc=InputDocumentation(doc="GFF3 feature types to include as exons (for GFF3 only;\ndefault: use SO v2.5.3 specification)")), ToolInput(tag="in_mask_gff_cds_types", input_type=Array(t=String(), optional=True), prefix="--mask_gff_cds_types", doc=InputDocumentation(doc="GFF3 feature types to include as CDS (for GFF3 only;\ndefault: use SO v2.5.3 specification)")), ToolInput(tag="in_mask_annotation_files", input_type=File(optional=True), prefix="--mask_annotation_files", doc=InputDocumentation(doc="[BED | BigBed | GTF2 | GFF3 | PSL] [infile.[BED | BigBed | GTF2 | GFF3 | PSL] ...]\nZero or more annotation files (max 1 file if BigBed)")), ToolInput(tag="in_mask_annotation_format", input_type=String(optional=True), prefix="--mask_annotation_format", doc=InputDocumentation(doc="Format of mask_annotation_files (Default: GTF2). Note:\nGFF3 assembly assumes SO v.2.5.2 feature ontologies,\nwhich may or may not match your specific file.")), ToolInput(tag="in_mask_add_three", input_type=Boolean(optional=True), prefix="--mask_add_three", doc=InputDocumentation(doc="If supplied, coding regions will be extended by 3\nnucleotides at their 3' ends (except for GTF2 files\nthat explicitly include `stop_codon` features). Use if\nyour annotation file excludes stop codons from CDS.")), ToolInput(tag="in_mask_tab_ix", input_type=Boolean(optional=True), prefix="--mask_tabix", doc=InputDocumentation(doc="mask_annotation_files are tabix-compressed and indexed\n(Default: False). Ignored for BigBed files.")), ToolInput(tag="in_mask_sorted", input_type=Boolean(optional=True), prefix="--mask_sorted", doc=InputDocumentation(doc="mask_annotation_files are sorted by chromosomal\nposition (Default: False)\n")), ToolInput(tag="in_out_folder", input_type=String(), position=0, doc=InputDocumentation(doc="Folder in which to save output vectors"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Get_Count_Vectors_V0_1_0().translate("wdl", allow_empty_container=True)

