from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Int, File, Boolean, String, Array

Slidejuncs_V0_1_0 = CommandToolBuilder(tool="slidejuncs", base_command=["slidejuncs"], inputs=[ToolInput(tag="in_max_slide", input_type=Int(optional=True), prefix="--maxslide", doc=InputDocumentation(doc="Maximum number of nt to search 5' and 3' of intron\nboundaries (Default: 10)")), ToolInput(tag="in_ref", input_type=File(optional=True), prefix="--ref", doc=InputDocumentation(doc="Reference file describing known splice junctions")), ToolInput(tag="in_slide_canonical", input_type=Boolean(optional=True), prefix="--slide_canonical", doc=InputDocumentation(doc="Slide junctions to canonical junctions if present\nwithin equal support region")), ToolInput(tag="in_quiet", input_type=Boolean(optional=True), prefix="--quiet", doc=InputDocumentation(doc="Suppress all warning messages. Cannot use with '-v'.")), ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="--verbose", doc=InputDocumentation(doc="Increase verbosity. With '-v', show every warning.\nWith '-vv', turn warnings into exceptions. Cannot use\nwith '-q'. (Default: show each type of warning once)")), ToolInput(tag="in_sequence_file", input_type=File(optional=True), prefix="--sequence_file", doc=InputDocumentation(doc="[fasta | fastq | twobit | genbank | embl]\nA file of DNA sequence")), ToolInput(tag="in_sequence_format", input_type=String(optional=True), prefix="--sequence_format", doc=InputDocumentation(doc="Format of sequence_file (Default: fasta).")), ToolInput(tag="in_mask_annotation_files", input_type=File(optional=True), prefix="--mask_annotation_files", doc=InputDocumentation(doc="[BED | BigBed | GTF2 | GFF3 | PSL] [infile.[BED | BigBed | GTF2 | GFF3 | PSL] ...]\nZero or more annotation files (max 1 file if BigBed)")), ToolInput(tag="in_mask_annotation_format", input_type=String(optional=True), prefix="--mask_annotation_format", doc=InputDocumentation(doc="Format of mask_annotation_files (Default: GTF2). Note:\nGFF3 assembly assumes SO v.2.5.2 feature ontologies,\nwhich may or may not match your specific file.")), ToolInput(tag="in_mask_add_three", input_type=Boolean(optional=True), prefix="--mask_add_three", doc=InputDocumentation(doc="If supplied, coding regions will be extended by 3\nnucleotides at their 3' ends (except for GTF2 files\nthat explicitly include `stop_codon` features). Use if\nyour annotation file excludes stop codons from CDS.")), ToolInput(tag="in_mask_tab_ix", input_type=Boolean(optional=True), prefix="--mask_tabix", doc=InputDocumentation(doc="mask_annotation_files are tabix-compressed and indexed\n(Default: False). Ignored for BigBed files.")), ToolInput(tag="in_mask_sorted", input_type=Boolean(optional=True), prefix="--mask_sorted", doc=InputDocumentation(doc="mask_annotation_files are sorted by chromosomal\nposition (Default: False)")), ToolInput(tag="in_mask_bed_extra_columns", input_type=Array(t=String(), optional=True), prefix="--mask_bed_extra_columns", doc=InputDocumentation(doc="Number of extra columns in BED file (e.g. in custom\nENCODE formats) or list of names for those columns.\n(Default: 0).")), ToolInput(tag="in_mask_maxmem", input_type=Int(optional=True), prefix="--mask_maxmem", doc=InputDocumentation(doc="Maximum desired memory footprint in MB to devote to\nBigBed/BigWig files. May be exceeded by large queries.\n(Default: 0, No maximum)")), ToolInput(tag="in_mask_gff_transcript_types", input_type=Array(t=String(), optional=True), prefix="--mask_gff_transcript_types", doc=InputDocumentation(doc="GFF3 feature types to include as transcripts, even if\nno exons are present (for GFF3 only; default: use SO\nv2.5.3 specification)")), ToolInput(tag="in_mask_gff_exon_types", input_type=Array(t=String(), optional=True), prefix="--mask_gff_exon_types", doc=InputDocumentation(doc="GFF3 feature types to include as exons (for GFF3 only;\ndefault: use SO v2.5.3 specification)")), ToolInput(tag="in_mask_gff_cds_types", input_type=Array(t=String(), optional=True), prefix="--mask_gff_cds_types", doc=InputDocumentation(doc="GFF3 feature types to include as CDS (for GFF3 only;\ndefault: use SO v2.5.3 specification)\n")), ToolInput(tag="in_sequence", input_type=String(), position=0, doc=InputDocumentation(doc="G   C   T   C   T   A   C   T   A   G   N   N   N   C   T   A   C   T   A   G   A   T   G   G")), ToolInput(tag="in_one_dot", input_type=Int(), position=0, doc=InputDocumentation(doc="If a mask file from crossmap\nis provided, junctions in which one or more of the 5' and 3' splice\nsites appear in a repetitive region of the genome are flagged as\nnon-informative and written to a separate file.")), ToolInput(tag="in_two_dot", input_type=Int(), position=1, doc=InputDocumentation(doc="For remaining splice junctions, the extent of locally repeated nucleotide\nsequence, if any, surrounding the query junction's splice donor and\nacceptor sites, are determined in both the 5' and 3' directions.\nThis is the maximum window (*equal-support region*) across which the\nactual splice junction could be moved without reducing sequence support.")), ToolInput(tag="in_three_dot", input_type=Int(), position=2, doc=InputDocumentation(doc="If there is one or more known splice junctions in this region, the\nquery junction is assumed to match these, and the known junctions are\nreported rather than the query.")), ToolInput(tag="in_four_dot", input_type=Int(), position=3, doc=InputDocumentation(doc="If (3) is not satisfied, and the query junction is a canonical splice\njunction, it is reported as is.")), ToolInput(tag="in_five_dot", input_type=Int(), position=4, doc=InputDocumentation(doc="If (3) is not satisfied, and the query junction represents a non-canonical\nsplice junction, the program determines if one or more canonical splice\njunctions is present in the equal-support region. If so, these canonical\nsplice junction are reported rather than the query junction.")), ToolInput(tag="in_six_dot", input_type=Int(), position=5, doc=InputDocumentation(doc="If (5) is not satisfied, the non-canonical query junction is reported as-is.")), ToolInput(tag="in_out_base_repetitive_dot_bed", input_type=String(), position=0, doc=InputDocumentation(doc="Splice junctions in which one or more of the splice sites lands\nin a repetitive/degenerate region of the genome, which gives rise to\nmapping ambiguities (step 1 above)")), ToolInput(tag="in_out_base_shifted_known_dot_bed", input_type=String(), position=1, doc=InputDocumentation(doc="The result of shifting query splice junctions to known splice junctions\nwith equal sequence support (step 3 above)")), ToolInput(tag="in_out_base_shifted_canonical_dot_bed", input_type=String(), position=2, doc=InputDocumentation(doc="The result of shifting non-canonical query splice junctions to canonical\nsplice junctions with equal sequence support (step 5 above)")), ToolInput(tag="in_out_base_untouched_dot_bed", input_type=String(), position=3, doc=InputDocumentation(doc="Query junctions reported without changes (steps 4 and 6 above)")), ToolInput(tag="in_input_dot_bed", input_type=String(), position=0, doc=InputDocumentation(doc="BED file describing discovered junctions")), ToolInput(tag="in_out_base", input_type=String(), position=1, doc=InputDocumentation(doc="Basename for output files"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Slidejuncs_V0_1_0().translate("wdl", allow_empty_container=True)

