from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, String, Boolean, Directory, Int

Beta_Basic_V0_1_0 = CommandToolBuilder(tool="BETA_basic", base_command=["BETA", "basic"], inputs=[ToolInput(tag="in_peak_file", input_type=File(optional=True), prefix="--peakfile", doc=InputDocumentation(doc="The bed format of peaks binding sites. (BETA support 3\nor 5 columns bed format, CHROM, START, END (NAME,\nSCORE))")), ToolInput(tag="in_diff_expr", input_type=File(optional=True), prefix="--diff_expr", doc=InputDocumentation(doc="IThe differential expression file get from limma for\nMicroArray ddata and cuffdiff for RNAseq data")), ToolInput(tag="in_kind", input_type=String(optional=True), prefix="--kind", doc=InputDocumentation(doc="the kind of your expression file,this is required,it\ncan be LIM, CUF, BSF, O. LIM for LIMMA standard\nformat. CUF for CUFDIFF standard format, BSF for BETA\nspecific format and O for other formats, if is 'O',\ncolumns infor required via --info")), ToolInput(tag="in_genome", input_type=String(optional=True), prefix="--genome", doc=InputDocumentation(doc="Specify your species, hg38,hg19,hg18,mm10,mm9. for\nother genome assembily versions of human and mouse or\nother species, ignore this paramter")), ToolInput(tag="in_g_name_two", input_type=Boolean(optional=True), prefix="--gname2", doc=InputDocumentation(doc="If this switch is on, gene or transcript IDs in files\ngiven through -e will be considered as official gene\nsymbols, DEFAULT=FALSE")), ToolInput(tag="in_info", input_type=File(optional=True), prefix="--info", doc=InputDocumentation(doc="specify the geneID, up/down status and statistcal\nvalues column of your expression data. NOTE: use a\ncomma as an connector. for example: 1,2,6 means geneID\nin the 1st column, logFC in 2nd column and FDR in 6th\ncolumn. DEFAULT:1,2,6 for LIMMA; 2,10,13 for Cuffdiff\nand 1,2,3 for BETA specific format. You'd better set\nit based on your exact expression file")), ToolInput(tag="in_reference", input_type=File(optional=True), prefix="--reference", doc=InputDocumentation(doc="the refgene info file downloaded from UCSC genome\nbrowser.input this file only if your genome is neither\nhg18, hg19,hg38 nor mm9,mm10")), ToolInput(tag="in_output", input_type=Directory(optional=True), prefix="--output", doc=InputDocumentation(doc="the directory to store all the output files, if you\ndon't set this, files will be output into the current\ndirectory")), ToolInput(tag="in_bl", input_type=Boolean(optional=True), prefix="--bl", doc=InputDocumentation(doc="whether or not use CTCF boundary to filter peaks\naround a gene, DEFAULT=FALSE")), ToolInput(tag="in_bf", input_type=File(optional=True), prefix="--bf", doc=InputDocumentation(doc="CTCF conserved peaks bed file, use this only when you\nset --bl and the genome is neither hg19 nor mm9")), ToolInput(tag="in_pn", input_type=Int(optional=True), prefix="--pn", doc=InputDocumentation(doc="the number of peaks you want to consider,\nDEFAULT=10000")), ToolInput(tag="in_method", input_type=String(optional=True), prefix="--method", doc=InputDocumentation(doc="Define the method to do the TF/CR function prediction,\nscore for regulatory potential, distance for the\ndistance to the proximal binding peak. DEFAULT:SCORE")), ToolInput(tag="in_name", input_type=String(optional=True), prefix="--name", doc=InputDocumentation(doc="this argument is used as the prefix for the result\nfiles. If not set, 'NA' will be used instead")), ToolInput(tag="in_distance", input_type=Int(optional=True), prefix="--distance", doc=InputDocumentation(doc="Set a number which unit is 'base'. It will get peaks\nwithin this distance from gene TSS. default:100000\n(100kb)")), ToolInput(tag="in_df", input_type=Int(optional=True), prefix="--df", doc=InputDocumentation(doc="Input a number 0~1 as a threshold to pick out the most\nsignificant differential expressed genes by FDR,\nDEFAULT = 1, that is select all the genes")), ToolInput(tag="in_da", input_type=Int(optional=True), prefix="--da", doc=InputDocumentation(doc="Get the most significant differential expressed genes\nby the percentage(0-1) or number(larger than 1)Input a\nnumber between 0-1, the rank based on fdr for example,\n2000, so that the script will only consider top 2000\ngenes as the differentially expressed genes. DEFAULT =\n0.5, that is select top 50 percent genes of up and\ndown seprately. NOTE: if you want to use diff_fdr,\nplease set this parameter to 1, otherwise it will get\nthe intersection of these two parameters")), ToolInput(tag="in_cut_off", input_type=Int(optional=True), prefix="--cutoff", doc=InputDocumentation(doc="Input a number between 0~1 as a threshold to select\nthe closer target gene list(up regulate or down\nregulate or both) with the p value was called by one\nside ks-test, DEFAULT = 0.001\n"))], outputs=[ToolOutput(tag="out_output", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_output", type_hint=File()), doc=OutputDocumentation(doc="the directory to store all the output files, if you\ndon't set this, files will be output into the current\ndirectory"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Beta_Basic_V0_1_0().translate("wdl", allow_empty_container=True)

