from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Int, String, Boolean, Float

Quasirecomb_V0_1_0 = CommandToolBuilder(tool="quasirecomb", base_command=["quasirecomb"], inputs=[ToolInput(tag="in_alignment_file_format", input_type=File(optional=True), prefix="-i", doc=InputDocumentation(doc=": Alignment file in BAM or SAM format.")), ToolInput(tag="in_path_output_directory", input_type=File(optional=True), prefix="-o", doc=InputDocumentation(doc=": Path to the output directory (default: current directory).")), ToolInput(tag="in_intint_interval_number", input_type=Int(optional=True), prefix="-K", doc=InputDocumentation(doc="or INT-INT     : The interval or fixed number of sequence generators, i.e. 1-4 or 2\nIn a grid enviroment the $SGE_TASK_ID.\nIn case of no input, K will be incremented as long as max BIC has not been reached, but will stop at K=5.")), ToolInput(tag="in_during_model_default", input_type=Int(optional=True), prefix="-m", doc=InputDocumentation(doc=": The number of EM restarts during model selection (default: 5).")), ToolInput(tag="in_for_best_k", input_type=Int(optional=True), prefix="-t", doc=InputDocumentation(doc=": The number of EM restarts for best K to find optimum (default: 50).")), ToolInput(tag="in_only_reconstruct_region", input_type=String(optional=True), prefix="-r", doc=InputDocumentation(doc=": Only reconstruct a specific region.")), ToolInput(tag="in_no_recomb", input_type=Boolean(optional=True), prefix="-noRecomb", doc=InputDocumentation(doc=": Do not allow recombination.")), ToolInput(tag="in_quality", input_type=Boolean(optional=True), prefix="-quality", doc=InputDocumentation(doc=": Account phred quality scores (slower runtime).")), ToolInput(tag="in_print_alignment", input_type=Boolean(optional=True), prefix="-printAlignment", doc=InputDocumentation(doc=": Save alignment.txt in a human readable format.")), ToolInput(tag="in_sample_proteins", input_type=Boolean(optional=True), prefix="-sampleProteins", doc=InputDocumentation(doc=": Sample full-length protein sequences in three reading frames.")), ToolInput(tag="in_coverage", input_type=Boolean(optional=True), prefix="-coverage", doc=InputDocumentation(doc=": If your dataset only contains a single region of interest,\nregions with a minimum coverage of 100x, 500x, 1,000x and 10,000x are reported.")), ToolInput(tag="in_refine", input_type=Boolean(optional=True), prefix="-refine", doc=InputDocumentation(doc=": Can only be used after QuasiRecomb has been executed once before on the same dataset in the same directory.\nThins the number of haplotypes.")), ToolInput(tag="in_nogaps", input_type=Boolean(optional=True), prefix="-noGaps", doc=InputDocumentation(doc=": Ignore gaps; useful if data is 454 and gaps are only technical errors.")), ToolInput(tag="in_conservative", input_type=Boolean(optional=True), prefix="-conservative", doc=InputDocumentation(doc=": Use this if the major haplotypes are only of interest.")), ToolInput(tag="in_max_del", input_type=Int(optional=True), prefix="-maxDel", doc=InputDocumentation(doc=": Remove reads with more than INT consecutive deletions.")), ToolInput(tag="in_max_perc_del", input_type=Float(optional=True), prefix="-maxPercDel", doc=InputDocumentation(doc=": Remove reads with more than DOUBLE ratio of deletions, between 0.0 - 1.0")), ToolInput(tag="in_unpaired", input_type=Boolean(optional=True), prefix="-unpaired", doc=InputDocumentation(doc=": If read names are not unique and reads are single-end, prevent pairing and merging.")), ToolInput(tag="in_xx", input_type=Boolean(optional=True), prefix="-XX", doc=InputDocumentation(doc=":NewRatio=9        : Reduces the memory consumption (RECOMMENDED to use).")), ToolInput(tag="in_x_mx_one_zero_g", input_type=Boolean(optional=True), prefix="-Xmx10G", doc=InputDocumentation(doc=": Increase heap space.")), ToolInput(tag="in_jar", input_type=String(optional=True), prefix="-jar", doc=InputDocumentation(doc="")), ToolInput(tag="in_java", input_type=String(), position=0, doc=InputDocumentation(doc="")), ToolInput(tag="in_options_dot_dot_dot", input_type=String(), position=1, doc=InputDocumentation(doc=""))], outputs=[ToolOutput(tag="out_path_output_directory", output_type=File(optional=True), selector=InputSelector(input_to_select="in_path_output_directory", type_hint=File()), doc=OutputDocumentation(doc=": Path to the output directory (default: current directory)."))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Quasirecomb_V0_1_0().translate("wdl", allow_empty_container=True)

