from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Int, String, Float, Boolean

Make_Consensi_Py_V0_1_0 = CommandToolBuilder(tool="make_consensi.py", base_command=["make-consensi.py"], inputs=[ToolInput(tag="in_dcs_one", input_type=File(optional=True), prefix="--dcs1", doc=InputDocumentation(doc="The file to output the first mates of the duplex consensus sequences into. Warning: This will be overwritten if it exists!")), ToolInput(tag="in_dcs_two", input_type=Int(optional=True), prefix="--dcs2", doc=InputDocumentation(doc="Same, but for mate 2.")), ToolInput(tag="in_sscs_one", input_type=Int(optional=True), prefix="--sscs1", doc=InputDocumentation(doc="Save the single-strand consensus sequences (mate 1) in this file (FASTA format). Warning: This will be overwritten if it exists!")), ToolInput(tag="in_sscs_two", input_type=Int(optional=True), prefix="--sscs2", doc=InputDocumentation(doc="Save the single-strand consensus sequences (mate 2) in this file (FASTA format). Warning: This will be overwritten if it exists!")), ToolInput(tag="in_qual_format", input_type=String(optional=True), prefix="--qual-format", doc=InputDocumentation(doc="FASTQ quality score format. Sanger scores are assumed to begin at '33' (!). Default: sanger.")), ToolInput(tag="in_fast_q_out", input_type=Int(optional=True), prefix="--fastq-out", doc=InputDocumentation(doc="Output in FASTQ instead of FASTA. You must specify the quality score to give to all bases. There is no meaningful quality score we can automatically give, so you will have to specify an artificial one. A good choice is 40, the maximum score normally output by sequencers.")), ToolInput(tag="in_min_reads", input_type=Int(optional=True), prefix="--min-reads", doc=InputDocumentation(doc="The minimum number of reads (from each strand) required to form a single-strand consensus. Strands with fewer reads will be skipped. Default: 3.")), ToolInput(tag="in_qual", input_type=Int(optional=True), prefix="--qual", doc=InputDocumentation(doc="Base quality threshold. Bases below this quality will not be counted. Default: 20.")), ToolInput(tag="in_cons_th_res", input_type=Float(optional=True), prefix="--cons-thres", doc=InputDocumentation(doc="The fractional threshold to use when making consensus sequences. The consensus base must be present in more than this fraction of the reads, or N will be used as the consensus base instead. Default: 0.5")), ToolInput(tag="in_min_cons_reads", input_type=Int(optional=True), prefix="--min-cons-reads", doc=InputDocumentation(doc="The absolute threshold to use when making consensus sequences. The consensus base must be present in more than this number of reads, or N will be used as the consensus base instead. Default: 0")), ToolInput(tag="in_phone_home", input_type=Boolean(optional=True), prefix="--phone-home", doc=InputDocumentation(doc="Report helpful usage data to the developer, to better understand the use cases and performance of the tool. The only data which will be recorded is the name and version of the tool, the size of the input data, the time and memory taken to process it, and the IP address of the machine running it. Also, if the script fails, it will report the name of the exception thrown and the line of code it occurred in. No filenames are sent, and the only parameters reported are the number of --processes and the --queue-size. All the reporting and recording code is available at https://github.com/NickSto/ET.")), ToolInput(tag="in_galaxy", input_type=Boolean(optional=True), prefix="--galaxy", doc=InputDocumentation(doc="Tell the script it's running on Galaxy. Currently this only affects data reported when phoning home.")), ToolInput(tag="in_test", input_type=Boolean(optional=True), prefix="--test", doc=InputDocumentation(doc="If reporting usage data, mark this as a test run.")), ToolInput(tag="in_log", input_type=File(optional=True), prefix="--log", doc=InputDocumentation(doc="Print log messages to this file instead of to stderr. Warning: Will overwrite the file.")), ToolInput(tag="in_processes", input_type=Int(optional=True), prefix="--processes", doc=InputDocumentation(doc="Number of worker subprocesses to use. If 0, no subprocesses will be started and everything will be done inside one process. Give 'auto' to use as many processes as there are CPU cores. Default: 0.")), ToolInput(tag="in_queue_size", input_type=Int(optional=True), prefix="--queue-size", doc=InputDocumentation(doc="How long to go accumulating responses from worker subprocesses before dealing with all of them. Default: 8 * the number of worker --processes.")), ToolInput(tag="in_families_dot_msa_dot_tsv", input_type=String(), position=0, doc=InputDocumentation(doc="The output of align_families.py. 6 columns:\n1. (canonical) barcode\n2. order ('ab' or 'ba')\n3. mate ('1' or '2')\n4. read name\n5. aligned sequence\n6. aligned quality scores."))], outputs=[ToolOutput(tag="out_dcs_one", output_type=File(optional=True), selector=InputSelector(input_to_select="in_dcs_one", type_hint=File()), doc=OutputDocumentation(doc="The file to output the first mates of the duplex consensus sequences into. Warning: This will be overwritten if it exists!"))], container="quay.io/biocontainers/dunovo:2.16--py39h38f01e4_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Make_Consensi_Py_V0_1_0().translate("wdl")

