from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Array, String, Int, Directory

Macs2_Refinepeak_V0_1_0 = CommandToolBuilder(tool="macs2_refinepeak", base_command=["macs2", "refinepeak"], inputs=[ToolInput(tag="in_candidate_peak_file", input_type=File(optional=True), prefix="-b", doc=InputDocumentation(doc="Candidate peak file in BED format. REQUIRED.")), ToolInput(tag="in_i_file", input_type=Array(t=File(), optional=True), prefix="--ifile", doc=InputDocumentation(doc="ChIP-seq alignment file. If multiple files are given\nas '-t A B C', then they will all be read and\ncombined. Note that pair-end data is not supposed to\nwork with this command. REQUIRED.")), ToolInput(tag="in_format", input_type=String(optional=True), prefix="--format", doc=InputDocumentation(doc="Format of tag file, 'AUTO', 'BED' or 'ELAND' or\n'ELANDMULTI' or 'ELANDEXPORT' or 'SAM' or 'BAM' or\n'BOWTIE'. The default AUTO option will let 'macs2\nrefinepeak' decide which format the file is. Please\ncheck the definition in README file if you choose\nELAND/ELANDMULTI/ELANDEXPORT/SAM/BAM/BOWTIE. DEFAULT:\n'AUTO'")), ToolInput(tag="in_cut_off", input_type=Int(optional=True), prefix="--cutoff", doc=InputDocumentation(doc="Cutoff DEFAULT: 5")), ToolInput(tag="in_window_size", input_type=Int(optional=True), prefix="--window-size", doc=InputDocumentation(doc="Scan window size on both side of the summit (default:\n100bp)")), ToolInput(tag="in_buffer_size", input_type=Int(optional=True), prefix="--buffer-size", doc=InputDocumentation(doc="Buffer size for incrementally increasing internal\narray size to store reads alignment information. In\nmost cases, you don't have to change this parameter.\nHowever, if there are large number of\nchromosomes/contigs/scaffolds in your alignment, it's\nrecommended to specify a smaller buffer size in order\nto decrease memory usage (but it will take longer time\nto read alignment files). Minimum memory requested for\nreading an alignment file is about # of CHROMOSOME *\nBUFFER_SIZE * 8 Bytes. DEFAULT: 100000")), ToolInput(tag="in_verbose", input_type=Int(optional=True), prefix="--verbose", doc=InputDocumentation(doc="Set verbose level. 0: only show critical message, 1:\nshow additional warning message, 2: show process\ninformation, 3: show debug messages. If you want to\nknow where are the duplicate reads, use 3. DEFAULT:2")), ToolInput(tag="in_outdir", input_type=Directory(optional=True), prefix="--outdir", doc=InputDocumentation(doc="If specified all output files will be written to that\ndirectory. Default: the current working directory")), ToolInput(tag="in_ofile", input_type=File(optional=True), prefix="--ofile", doc=InputDocumentation(doc="Output file name. Mutually exclusive with --o-prefix.")), ToolInput(tag="in_o_prefix", input_type=File(optional=True), prefix="--o-prefix", doc=InputDocumentation(doc="Output file prefix. Mutually exclusive with"))], outputs=[ToolOutput(tag="out_outdir", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_outdir", type_hint=File()), doc=OutputDocumentation(doc="If specified all output files will be written to that\ndirectory. Default: the current working directory")), ToolOutput(tag="out_ofile", output_type=File(optional=True), selector=InputSelector(input_to_select="in_ofile", type_hint=File()), doc=OutputDocumentation(doc="Output file name. Mutually exclusive with --o-prefix.")), ToolOutput(tag="out_o_prefix", output_type=File(optional=True), selector=InputSelector(input_to_select="in_o_prefix", type_hint=File()), doc=OutputDocumentation(doc="Output file prefix. Mutually exclusive with"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Macs2_Refinepeak_V0_1_0().translate("wdl", allow_empty_container=True)

