from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import String, File, Int, Boolean

Macaron_V0_1_0 = CommandToolBuilder(tool="MACARON", base_command=["MACARON"], inputs=[ToolInput(tag="in_script", input_type=String(optional=True), prefix="-Script", doc=InputDocumentation(doc="identify SnpClusters (SNPs within the same genetic codon)")), ToolInput(tag="in_in_file", input_type=File(optional=True), prefix="--infile", doc=InputDocumentation(doc="Full path of the input VCF file.")), ToolInput(tag="in_outfile", input_type=File(optional=True), prefix="--outfile", doc=InputDocumentation(doc="Path of the output txt file (Default Output file:\nMACARON_output.txt)")), ToolInput(tag="in_fields", input_type=File(optional=True), prefix="--fields", doc=InputDocumentation(doc="Single field name or comma-seperated ',' multiple\nfield names can be given. Field name should be given\naccording to the (INFO) field header of the input vcf\nfile. Example: -f Func.refGene,ExonicFunc.refGene,Gene\n.refGene,1000g2015aug_all,ExAC_ALL,ExAC_EAS,clinvar_20\n161128,gnomAD_exome_ALL,gnomAD_genome_ALL,EFF,CSQ")), ToolInput(tag="in_hg_ref", input_type=File(optional=True), prefix="--HG_REF", doc=InputDocumentation(doc="Indicate the full path to the reference genome fasta")), ToolInput(tag="in_java_options", input_type=Int(optional=True), prefix="--JAVA_OPTIONS", doc=InputDocumentation(doc="You can use this option to specify java arguments\nrequired by GATK (default: --JAVA-OPTIONS '-Xmx4G')")), ToolInput(tag="in_snp_eff", input_type=File(optional=True), prefix="--SNPEFF", doc=InputDocumentation(doc="You can use this option to directly indicate the full\npath to the snpEff jar or wrapper")), ToolInput(tag="in_snp_eff_hg", input_type=String(optional=True), prefix="--SNPEFF_HG", doc=InputDocumentation(doc="Indicate SnpEff human genome annotation database\nversion")), ToolInput(tag="in_verbosity", input_type=Boolean(optional=True), prefix="--verbosity", doc=InputDocumentation(doc="Use to print verbosity (Mostly GATK/SNPEFF output)")), ToolInput(tag="in_add_anim", input_type=Boolean(optional=True), prefix="--add_anim", doc=InputDocumentation(doc="Add animation while running (looks good but costs a\nthread)")), ToolInput(tag="in_keep_tmp", input_type=File(optional=True), prefix="--keep_tmp", doc=InputDocumentation(doc="Keep temporary files in the directory tmp_macaron, at\nthe same location than the output file.")), ToolInput(tag="in_g_atk_four_previous", input_type=Boolean(optional=True), prefix="--gatk4_previous", doc=InputDocumentation(doc="Use this option if you are using a version of gatk 4\nolder than gatk 4.1.4.1\n")), ToolInput(tag="in_file", input_type=File(), position=0, doc=InputDocumentation(doc="--GATK GATK_PATH      You can use this option to directly indicate the full"))], outputs=[ToolOutput(tag="out_outfile", output_type=File(optional=True), selector=InputSelector(input_to_select="in_outfile", type_hint=File()), doc=OutputDocumentation(doc="Path of the output txt file (Default Output file:\nMACARON_output.txt)")), ToolOutput(tag="out_keep_tmp", output_type=File(optional=True), selector=InputSelector(input_to_select="in_keep_tmp", type_hint=File()), doc=OutputDocumentation(doc="Keep temporary files in the directory tmp_macaron, at\nthe same location than the output file."))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Macaron_V0_1_0().translate("wdl", allow_empty_container=True)

