from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, String, Array, File, Int

Emblmygff3_V0_1_0 = CommandToolBuilder(tool="EMBLmyGFF3", base_command=["EMBLmyGFF3"], inputs=[ToolInput(tag="in_accession", input_type=Boolean(optional=True), prefix="--accession", doc=InputDocumentation(doc="Bolean. Accession number(s) for the entry. Default\nvalue: XXX. The proper value is automatically filled\nup by ENA during the submission by a unique accession\nnumber they will assign. The accession number is used\nto set up the AC line and the first token of the ID\nline as well. Please visit [this\npage](https://www.ebi.ac.uk/ena/submit/accession-\nnumber-formats) and [this\none](https://www.ebi.ac.uk/ena/submit/sequence-\nsubmission) to learn more about it. Activating the\noption will set the Accession number with the fasta\nsequence identifier.")), ToolInput(tag="in_created", input_type=String(optional=True), prefix="--created", doc=InputDocumentation(doc="Creation time of the original entry. The default value\nis the date of the day.")), ToolInput(tag="in_data_class", input_type=String(optional=True), prefix="--data_class", doc=InputDocumentation(doc="Data class of the sample. Default value 'XXX'. This\noption is used to set up the 5th token of the ID line.")), ToolInput(tag="in_organelle", input_type=String(optional=True), prefix="--organelle", doc=InputDocumentation(doc="Sample organelle. No default value.")), ToolInput(tag="in_locus_tag", input_type=String(optional=True), prefix="--locus_tag", doc=InputDocumentation(doc="Locus tag prefix used to set up the prefix of the\nlocus_tag qualifier. The locus tag has to be\nregistered at ENA prior any submission. More\ninformation\n[here](https://www.ebi.ac.uk/ena/submit/locus-tags).")), ToolInput(tag="in_keyword", input_type=Array(t=String(), optional=True), prefix="--keyword", doc=InputDocumentation(doc="Keywords for the entry. No default value.")), ToolInput(tag="in_classification", input_type=String(optional=True), prefix="--classification", doc=InputDocumentation(doc="Organism classification e.g 'Eukaryota; Opisthokonta;\nMetazoa'. The default value is the classification\nfound in the NCBI taxonomy DB from the species/taxid\ngiven as --species parameter. If none is found, 'Life'\nwill be the default value.")), ToolInput(tag="in_molecule_type", input_type=String(optional=True), prefix="--molecule_type", doc=InputDocumentation(doc="Molecule type of the sample. No default value.")), ToolInput(tag="in_output", input_type=File(optional=True), prefix="--output", doc=InputDocumentation(doc="Output filename.")), ToolInput(tag="in_project_id", input_type=String(optional=True), prefix="--project_id", doc=InputDocumentation(doc="Project ID. Default is 'XXX' (This is used to set up\nthe PR line).")), ToolInput(tag="in_quiet", input_type=Boolean(optional=True), prefix="--quiet", doc=InputDocumentation(doc="Decrease verbosity.")), ToolInput(tag="in_transl_table", input_type=String(optional=True), prefix="--transl_table", doc=InputDocumentation(doc="Translation table. No default. (This is used to set up\nthe translation table qualifier transl_table of the\nCDS features.) Please visit [NCBI genetic code](https:\n//www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi)\nfor more information.")), ToolInput(tag="in_species", input_type=String(optional=True), prefix="--species", doc=InputDocumentation(doc="Sample species, formatted as 'Genus species' or taxid.\nNo default. (This is used to set up the OS line.)")), ToolInput(tag="in_topology", input_type=String(optional=True), prefix="--topology", doc=InputDocumentation(doc="Sequence topology. No default. (This is used to set up\nthe Topology that is the 3rd token of the ID line.)")), ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="--verbose", doc=InputDocumentation(doc="Increase verbosity.")), ToolInput(tag="in_taxonomy", input_type=String(optional=True), prefix="--taxonomy", doc=InputDocumentation(doc="Source taxonomy. Default value 'XXX'. This option is\nused to set the taxonomic division within ID line (6th\ntoken).")), ToolInput(tag="in_gzip", input_type=File(optional=True), prefix="--gzip", doc=InputDocumentation(doc="Gzip output file.")), ToolInput(tag="in_advanced_help", input_type=String(optional=True), prefix="--advanced_help", doc=InputDocumentation(doc="Display advanced information of the parameter\nspecified or of all parameters if none specified.")), ToolInput(tag="in_de", input_type=String(optional=True), prefix="--de", doc=InputDocumentation(doc="Description. Default value 'XXX'.")), ToolInput(tag="in_author", input_type=Array(t=String(), optional=True), prefix="--author", doc=InputDocumentation(doc="Author for the reference. No default value.")), ToolInput(tag="in_rc", input_type=String(optional=True), prefix="--rc", doc=InputDocumentation(doc="Reference Comment. No default value.")), ToolInput(tag="in_rg", input_type=String(optional=True), prefix="--rg", doc=InputDocumentation(doc="Reference Group, the working groups/consortia that\nproduced the record. Default value 'XXX'.")), ToolInput(tag="in_rl", input_type=String(optional=True), prefix="--rl", doc=InputDocumentation(doc="Reference publishing location. No default value.")), ToolInput(tag="in_rt", input_type=String(optional=True), prefix="--rt", doc=InputDocumentation(doc="Reference Title. No default value.")), ToolInput(tag="in_rx", input_type=String(optional=True), prefix="--rx", doc=InputDocumentation(doc="Reference cross-reference. No default value")), ToolInput(tag="in_email", input_type=Int(optional=True), prefix="--email", doc=InputDocumentation(doc="Email used to fetch information from NCBI taxonomy\ndatabase. Default value 'EMBLmyGFF3@tool.org'.")), ToolInput(tag="in_expose_translations", input_type=Boolean(optional=True), prefix="--expose_translations", doc=InputDocumentation(doc="Copy feature and attribute mapping files to the\nworking directory. They will be used as mapping files\ninstead of the default internal JSON files. You may\nmodify them as it suits you.")), ToolInput(tag="in_force_unknown_features", input_type=Boolean(optional=True), prefix="--force_unknown_features", doc=InputDocumentation(doc="Force to keep feature types not accepted by EMBL. /!\\nOption not suitable for submission purpose.")), ToolInput(tag="in_force_un_complete_features", input_type=Boolean(optional=True), prefix="--force_uncomplete_features", doc=InputDocumentation(doc="Force to keep features whithout all the mandatory\nqualifiers. /!\ Option not suitable for submission\npurpose.")), ToolInput(tag="in_interleave_genes", input_type=Boolean(optional=True), prefix="--interleave_genes", doc=InputDocumentation(doc="Print gene features with interleaved mRNA and CDS")), ToolInput(tag="in_locus_numbering_start", input_type=String(optional=True), prefix="--locus_numbering_start", doc=InputDocumentation(doc="Start locus numbering with the provided value.")), ToolInput(tag="in_no_progress", input_type=Boolean(optional=True), prefix="--no_progress", doc=InputDocumentation(doc="Hide conversion progress counter.")), ToolInput(tag="in_no_wrap_qualifier", input_type=Boolean(optional=True), prefix="--no_wrap_qualifier", doc=InputDocumentation(doc="By default there is a line wrapping at 80 characters.\nThe cut is at the world level. Activating this option\nwill avoid the line-wrapping for the qualifiers.")), ToolInput(tag="in_shame", input_type=Boolean(optional=True), prefix="--shame", doc=InputDocumentation(doc="Suppress the shameless plug.")), ToolInput(tag="in_translate", input_type=Boolean(optional=True), prefix="--translate", doc=InputDocumentation(doc="Include translation in CDS features.")), ToolInput(tag="in_use_attribute_value_as_locus_tag", input_type=String(optional=True), prefix="--use_attribute_value_as_locus_tag", doc=InputDocumentation(doc="Use the value of the defined attribute as locus_tag.")), ToolInput(tag="in_uncompressed_log", input_type=Boolean(optional=True), prefix="--uncompressed_log", doc=InputDocumentation(doc="Some logs can be compressed for better lisibility,\nthey won't.")), ToolInput(tag="in_strain", input_type=String(optional=True), prefix="--strain", doc=InputDocumentation(doc="Strain from which sequence was obtained. May be needed\nwhen organism belongs to Bacteria.")), ToolInput(tag="in_environmental_sample", input_type=Boolean(optional=True), prefix="--environmental_sample", doc=InputDocumentation(doc="Bolean. Identifies sequences derived by direct\nmolecular isolation from a bulk environmental DNA\nsample with no reliable identification of the source\norganism. May be needed when organism belongs to\nBacteria.")), ToolInput(tag="in_isolation_source", input_type=String(optional=True), prefix="--isolation_source", doc=InputDocumentation(doc="Describes the physical, environmental and/or local\ngeographical source of the biological sample from\nwhich the sequence was derived. Mandatory when\nenvironmental_sample option used.")), ToolInput(tag="in_isolate", input_type=String(optional=True), prefix="--isolate", doc=InputDocumentation(doc="Individual isolate from which the sequence was\nobtained. May be needed when organism belongs to\nBacteria.\n")), ToolInput(tag="in_gff_file", input_type=String(), position=0, doc=InputDocumentation(doc="Input gff-file.")), ToolInput(tag="in_fast_a", input_type=String(), position=1, doc=InputDocumentation(doc="Input fasta sequence.")), ToolInput(tag="in_features_dot", input_type=String(), position=0, doc=InputDocumentation(doc="--keep_duplicates     Do not remove duplicate features during the process."))], outputs=[ToolOutput(tag="out_output", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output", type_hint=File()), doc=OutputDocumentation(doc="Output filename.")), ToolOutput(tag="out_gzip", output_type=File(optional=True), selector=InputSelector(input_to_select="in_gzip", type_hint=File()), doc=OutputDocumentation(doc="Gzip output file."))], container="quay.io/biocontainers/emblmygff3:2.1--py_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Emblmygff3_V0_1_0().translate("wdl")

