from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, File

Faqcs_V0_1_0 = CommandToolBuilder(tool="FaQCs", base_command=["FaQCs"], inputs=[ToolInput(tag="in_file_unpaired_reads", input_type=Boolean(optional=True), prefix="-u", doc=InputDocumentation(doc="<File> Unpaired reads")), ToolInput(tag="in_file_first_paired", input_type=Boolean(optional=True), prefix="-1", doc=InputDocumentation(doc="<File> First paired read file")), ToolInput(tag="in_file_second_paired", input_type=Boolean(optional=True), prefix="-2", doc=InputDocumentation(doc="<File> Second paired read file")), ToolInput(tag="in_mode", input_type=Boolean(optional=True), prefix="--mode", doc=InputDocumentation(doc="'HARD' or 'BWA' or 'BWA_plus' (default BWA_plus)\nBWA trim is NOT A HARD cutoff! (see bwa's bwa_trim_read() function in bwaseqio.c)")), ToolInput(tag="in_int_targets_quality", input_type=Boolean(optional=True), prefix="-q", doc=InputDocumentation(doc="<INT> Targets # as quality level (default 5) for trimming")), ToolInput(tag="in_five_end", input_type=Boolean(optional=True), prefix="--5end", doc=InputDocumentation(doc="<INT> Cut # bp from 5 end before quality trimming/filtering")), ToolInput(tag="in_three_end", input_type=Boolean(optional=True), prefix="--3end", doc=InputDocumentation(doc="<INT> Cut # bp from 3 end before quality trimming/filtering")), ToolInput(tag="in_adapter", input_type=Boolean(optional=True), prefix="--adapter", doc=InputDocumentation(doc="<bool> Trim reads with illumina adapter/primers (default: no)")), ToolInput(tag="in_rate", input_type=Boolean(optional=True), prefix="--rate", doc=InputDocumentation(doc="<FLOAT> Mismatch ratio of adapters' length (default: 0.2, allow 20% mismatches)")), ToolInput(tag="in_polya", input_type=Boolean(optional=True), prefix="--polyA", doc=InputDocumentation(doc="<bool>  Trim poly A ( > 15 )")), ToolInput(tag="in_artifact_file", input_type=Boolean(optional=True), prefix="--artifactFile", doc=InputDocumentation(doc="<File> additional artifact (adapters/primers/contaminations) reference file in fasta format")), ToolInput(tag="in_min_l", input_type=Boolean(optional=True), prefix="--min_L", doc=InputDocumentation(doc="<INT> Trimmed read should have to be at least this minimum length (default:50)")), ToolInput(tag="in_avg_q", input_type=Boolean(optional=True), prefix="--avg_q", doc=InputDocumentation(doc="<NUM> Average quality cutoff (default:0, no filtering)")), ToolInput(tag="in_int_trimmed_read_has", input_type=Boolean(optional=True), prefix="-n", doc=InputDocumentation(doc="<INT> Trimmed read has greater than or equal to this number of continuous base 'N' will be discarded.\n(default: 2, 'NN')")), ToolInput(tag="in_lc", input_type=Boolean(optional=True), prefix="--lc", doc=InputDocumentation(doc="<FLOAT> Low complexity filter ratio, Maximum fraction of mono-/di-nucleotide sequence  (default: 0.85)")), ToolInput(tag="in_phi_x", input_type=Boolean(optional=True), prefix="--phiX", doc=InputDocumentation(doc="<bool> Filter phiX reads (slow)")), ToolInput(tag="in_ascii", input_type=Boolean(optional=True), prefix="--ascii", doc=InputDocumentation(doc="Encoding type: 33 or 64 or autoCheck (default)\nType of ASCII encoding: 33 (standard) or 64 (illumina 1.3+)")), ToolInput(tag="in_out_ascii", input_type=Boolean(optional=True), prefix="--out_ascii", doc=InputDocumentation(doc="Output encoding. (default: 33)")), ToolInput(tag="in_prefix", input_type=File(optional=True), prefix="--prefix", doc=InputDocumentation(doc="<TEXT> Output file prefix. (default: QC)")), ToolInput(tag="in_stats", input_type=File(optional=True), prefix="--stats", doc=InputDocumentation(doc="<File> Statistical numbers output file (default: prefix.stats.txt)")), ToolInput(tag="in_path_output_directory", input_type=File(optional=True), prefix="-d", doc=InputDocumentation(doc="<PATH> Output directory.")), ToolInput(tag="in_int_cpus_run", input_type=Boolean(optional=True), prefix="-t", doc=InputDocumentation(doc="<INT > # of CPUs to run the script (default:2 )")), ToolInput(tag="in_split_size", input_type=Boolean(optional=True), prefix="--split_size", doc=InputDocumentation(doc="<INT> Split the input file into several sub files by sequence number (default: 1000000)")), ToolInput(tag="in_qc_only", input_type=Boolean(optional=True), prefix="--qc_only", doc=InputDocumentation(doc="<bool> no Filters, no Trimming, report numbers.")), ToolInput(tag="in_km_er_rarefaction", input_type=Boolean(optional=True), prefix="--kmer_rarefaction", doc=InputDocumentation(doc="<bool>\nTurn on the kmer calculation. Turn on will slow down ~10 times. (default:Calculation is off.)\n(meaningless if -subset is too small)")), ToolInput(tag="in_int_kmer_rarefaction", input_type=Boolean(optional=True), prefix="-m", doc=InputDocumentation(doc="<INT>     kmer for rarefaction curve (range:[2,31], default 31)")), ToolInput(tag="in_subset", input_type=Boolean(optional=True), prefix="--subset", doc=InputDocumentation(doc="<INT>   Use this nubmer x split_size for qc_only and kmer_rarefaction\n(default: 10,  10x1000000 SE reads, 20x1000000 PE reads)")), ToolInput(tag="in_discard", input_type=Boolean(optional=True), prefix="--discard", doc=InputDocumentation(doc="<bool> Output discarded reads to prefix.discard.fastq (default: 0, not output)")), ToolInput(tag="in_substitute", input_type=Boolean(optional=True), prefix="--substitute", doc=InputDocumentation(doc="<bool> Replace 'N' in the trimmed reads with random base A,T,C ,or G (default: 0, off)")), ToolInput(tag="in_trim_only", input_type=Boolean(optional=True), prefix="--trim_only", doc=InputDocumentation(doc="<bool> No quality report. Output trimmed reads only.")), ToolInput(tag="in_replace_to_n_q", input_type=Boolean(optional=True), prefix="--replace_to_N_q", doc=InputDocumentation(doc="<INT> Replace base G to N when below this quality score (default:0, off)")), ToolInput(tag="in_five_trim_off", input_type=Boolean(optional=True), prefix="--5trim_off", doc=InputDocumentation(doc="<bool> Turn off trimming from 5'end.")), ToolInput(tag="in_debug", input_type=Boolean(optional=True), prefix="--debug", doc=InputDocumentation(doc="<bool> Keep intermediate files"))], outputs=[ToolOutput(tag="out_prefix", output_type=File(optional=True), selector=InputSelector(input_to_select="in_prefix", type_hint=File()), doc=OutputDocumentation(doc="<TEXT> Output file prefix. (default: QC)")), ToolOutput(tag="out_stats", output_type=File(optional=True), selector=InputSelector(input_to_select="in_stats", type_hint=File()), doc=OutputDocumentation(doc="<File> Statistical numbers output file (default: prefix.stats.txt)")), ToolOutput(tag="out_path_output_directory", output_type=File(optional=True), selector=InputSelector(input_to_select="in_path_output_directory", type_hint=File()), doc=OutputDocumentation(doc="<PATH> Output directory."))], container="quay.io/biocontainers/faqcs:2.10--r40he513fc3_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Faqcs_V0_1_0().translate("wdl")

