from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, File, String

Bp_Genbank_Ref_Extractor_V0_1_0 = CommandToolBuilder(tool="bp_genbank_ref_extractor", base_command=["bp_genbank_ref_extractor"], inputs=[ToolInput(tag="in_assembly", input_type=Boolean(optional=True), prefix="--assembly", doc=InputDocumentation(doc="When retrieving the sequence, a specific assemly can be defined. The\nvalue expected is a regex that will be case-insensitive. If it\nmatches more than one assembly, it will use the first match. It\ndefauls to '(primary|reference) assembly'.")), ToolInput(tag="in_debug", input_type=File(optional=True), prefix="--debug", doc=InputDocumentation(doc="If set, even more output will be printed that may help on debugging.\nUnlike the messages from --verbose and --very-verbose, these will\nnot appear on the log file unless this option is selected. This\noption also sets --very-verbose.")), ToolInput(tag="in_downstream", input_type=Boolean(optional=True), prefix="--downstream", doc=InputDocumentation(doc="Specifies the number of extra base pairs to be retrieved downstream\nof the gene. This extra base pairs will only affect the gene\nsequence, not the transcript or proteins.")), ToolInput(tag="in_email", input_type=Boolean(optional=True), prefix="--email", doc=InputDocumentation(doc="A valid email used to connect to the NCBI servers. This may be used\nby NCBI to contact users in case of problems and before blocking\naccess in case of heavy usage.")), ToolInput(tag="in_format", input_type=Boolean(optional=True), prefix="--format", doc=InputDocumentation(doc="Specifies the format that the sequences will be saved. Defaults to\n*genbank* format. Valid formats are 'genbank' or 'fasta'.")), ToolInput(tag="in_genes", input_type=Boolean(optional=True), prefix="--genes", doc=InputDocumentation(doc="Specifies the name for gene file. By default, they are not saved. If\nno value is given defaults to its UID. Possible values are 'uid',\n'name', 'symbol' (the official symbol or nomenclature).")), ToolInput(tag="in_limit", input_type=Boolean(optional=True), prefix="--limit", doc=InputDocumentation(doc="When making a query, limit the result to these first specific\nresults. This is to prevent the use of specially unspecific queries\nand a warning will be given if a query returns more results than the\nlimit. The default value is 200. Note that this limit is for *each*\nsearch.")), ToolInput(tag="in_non_on_coding", input_type=Boolean(optional=True), prefix="--nonon-coding", doc=InputDocumentation(doc="Some protein coding genes have transcripts that are non-coding. By\ndefault, these sequences are saved as well. --nonon-coding can be\nused to ignore those transcripts.")), ToolInput(tag="in_proteins", input_type=Boolean(optional=True), prefix="--proteins", doc=InputDocumentation(doc="Specifies the name for proteins file. By default, they are not\nsaved. If no value is given defaults to its accession. Possible\nvalues are 'accession', 'description', 'gene' (the corresponding\ngene ID) and 'transcript' (the corresponding transcript accesion).\nNote that if not using 'accession' is possible for files to be\noverwritten. It is possible for the same gene to encode more than\none protein or different proteins to have the same description.")), ToolInput(tag="in_no_pseudo", input_type=Boolean(optional=True), prefix="--nopseudo", doc=InputDocumentation(doc="By default, sequences of pseudo genes will be saved. --nopseudo can\nbe used to ignore those genes.")), ToolInput(tag="in_save", input_type=Boolean(optional=True), prefix="--save", doc=InputDocumentation(doc="Specifies the path for the directory where the sequence and log\nfiles will be saved. If the directory does not exist it will be\ncreated altough the path to it must exist. Files on the directory\nmay be rewritten if necessary. If unspecified, a directory named\nextracted sequences on the current directory will be used.")), ToolInput(tag="in_save_data", input_type=Boolean(optional=True), prefix="--save-data", doc=InputDocumentation(doc="This options saves the data (gene UIDs, description, product\naccessions, etc) to a file. As an optional value, the file format\ncan be specified. Defaults to CSV.\nCurrently only CSV is supported.\nSaving the data structure as a CSV file, requires the installation\nof the Text::CSV module.")), ToolInput(tag="in_transcripts", input_type=Boolean(optional=True), prefix="--transcripts", doc=InputDocumentation(doc="Specifies the name for transcripts file. By default, they are not\nsaved. If no value is given defaults to its accession. Possible\nvalues are 'accession', 'description', 'gene' (the corresponding\ngene ID) and 'protein' (the protein the transcript encodes).\nNote that if not using 'accession' is possible for files to be\noverwritten. It is possible for the same gene to have more than one\ntranscript or different transcripts to have the same description.\nAlso, non-coding transcripts will create problems if using\n'protein'.")), ToolInput(tag="in_upstream", input_type=Boolean(optional=True), prefix="--upstream", doc=InputDocumentation(doc="Specifies the number of extra base pairs to be extracted upstream of\nthe gene. This extra base pairs will only affect the gene sequence,\nnot the transcript or proteins.")), ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="--verbose", doc=InputDocumentation(doc="If set, program becomes verbose. For an extremely verbose program,\nuse --very-verbose instead.")), ToolInput(tag="in_very_verbose", input_type=Boolean(optional=True), prefix="--very-verbose", doc=InputDocumentation(doc="If set, program becomes extremely verbose. Setting this option,\nautomatically sets --verbose as well. For help in debugging,\nconsider using --debug")), ToolInput(tag="in_bioperl_lat_bioperl_dot_org", input_type=String(), position=0, doc=InputDocumentation(doc="- General discussion")), ToolInput(tag="in_support", input_type=String(), position=0, doc=InputDocumentation(doc="Please direct usage questions or support issues to the mailing list:\n*bioperl-l@bioperl.org*\nrather than to the module maintainer directly. Many experienced and\nreponsive experts will be able look at the problem and quickly address\nit. Please include a thorough description of the problem with code and\ndata examples if at all possible."))], outputs=[ToolOutput(tag="out_debug", output_type=File(optional=True), selector=InputSelector(input_to_select="in_debug", type_hint=File()), doc=OutputDocumentation(doc="If set, even more output will be printed that may help on debugging.\nUnlike the messages from --verbose and --very-verbose, these will\nnot appear on the log file unless this option is selected. This\noption also sets --very-verbose."))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Bp_Genbank_Ref_Extractor_V0_1_0().translate("wdl", allow_empty_container=True)

