from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Int, String, Float, Boolean

Crispector_V0_1_0 = CommandToolBuilder(tool="crispector", base_command=["crispector"], inputs=[ToolInput(tag="in_tx_in_one", input_type=File(optional=True), prefix="--tx_in1", doc=InputDocumentation(doc="Tx read 1 input path or Tx merged FASTQ file")), ToolInput(tag="in_tx_in_two", input_type=File(optional=True), prefix="--tx_in2", doc=InputDocumentation(doc="Tx read 2 input path, if FASTQ files aren't\nmerged [OPTIONAL]")), ToolInput(tag="in_mock_in_one", input_type=File(optional=True), prefix="--mock_in1", doc=InputDocumentation(doc="Mock read 1 input path or mock merged FASTQ")), ToolInput(tag="in_experiment_config", input_type=File(optional=True), prefix="--experiment_config", doc=InputDocumentation(doc="A CSV (Comma Separated Values‚Äè) file with\nthe experiment data. Table has 11 columns:\nSiteName, AmpliconReference, gRNA, OnTarget,\nForwardPrimer, ReversePrimer,TxInput1Path\nTxInput2Path, MockInput1Path,\nMockInput2Path, DonorReference. The first 4\ncolumns are required, the rest are optional.\nHeader should be specified by the above\norder. Please check the README on GitHub\nfurther details and examples.  [required]")), ToolInput(tag="in_report_output", input_type=File(optional=True), prefix="--report_output", doc=InputDocumentation(doc="Path to output folder")), ToolInput(tag="in_cut_site_position", input_type=Int(optional=True), prefix="--cut_site_position", doc=InputDocumentation(doc="Expected cut-site position with respect to\nthe 3' end of the provided sgRNA sequence.\nNote, the sgRNA sequence must be entered\nwithout the PAM.  [default: -3]")), ToolInput(tag="in_crisp_ector_config", input_type=File(optional=True), prefix="--crispector_config", doc=InputDocumentation(doc="Path to crispector configuration in YAML\nformat.See 'Advanced usage' section in\nREADME on GitHub for further.")), ToolInput(tag="in_fast_p_options_string", input_type=String(optional=True), prefix="--fastp_options_string", doc=InputDocumentation(doc="Try 'fastp --help' for more details")), ToolInput(tag="in_min_num_of_reads", input_type=Int(optional=True), prefix="--min_num_of_reads", doc=InputDocumentation(doc="Minimum number of reads (per locus site) to\nevaluate edit events  [default: 500]")), ToolInput(tag="in_min_read_length_without_primers", input_type=Int(optional=True), prefix="--min_read_length_without_primers", doc=InputDocumentation(doc="Filter out any read shorter than\nmin_read_length_without_primers + length of\nforward and reverse primers. This threshold\nfilters primer-dimmer effect reads.\n[default: 10]")), ToolInput(tag="in_max_edit_distance_on_primers", input_type=Int(optional=True), prefix="--max_edit_distance_on_primers", doc=InputDocumentation(doc="Maximum edit distance to consider a read\nprefix (or suffix) as a match for a primer.\n[default: 8]")), ToolInput(tag="in_amplicon_min_score", input_type=Float(optional=True), prefix="--amplicon_min_score", doc=InputDocumentation(doc="RANGE\nMinimum normalized alignment score to\nconsider a read alignment as valid.\nNormalized alignment score is defined as the\nNeedleman-Wunch alignment score divided by\nthe maximum possible score. Below this\nalignment threshold, reads are discarded.\n[default: 30]")), ToolInput(tag="in_translocation_amplicon_min_score", input_type=Float(optional=True), prefix="--translocation_amplicon_min_score", doc=InputDocumentation(doc="RANGE\nMinimum alignment score to consider a read\nwith primer inconsistency as a possible\ntranslocation. Should be higher than\n--amplicon_min_score, because translocations\nreads are noisier.Score is normalized\nbetween 0 (not even one bp match) to 100\n(read is identical to  [default: 80]")), ToolInput(tag="in_min_editing_activity", input_type=Float(optional=True), prefix="--min_editing_activity", doc=InputDocumentation(doc="RANGE\nMinimum editing activity (%). Sites with\nediting activity lower than the minimum,\nwill be discarded from the translocation\ndetection.  [default: 0.1]")), ToolInput(tag="in_translocation_p_value", input_type=Float(optional=True), prefix="--translocation_p_value", doc=InputDocumentation(doc="RANGE\nTranslocations statistical significance\nlevel. This threshold is applied on the\ncorrected p_value,FDR (false discovery\nrate).  [default: 0.05]")), ToolInput(tag="in_disable_translocations", input_type=Boolean(optional=True), prefix="--disable_translocations", doc=InputDocumentation(doc="Disable translocations detection  [default:\nFalse]")), ToolInput(tag="in_override_noise_estimation", input_type=Boolean(optional=True), prefix="--override_noise_estimation", doc=InputDocumentation(doc="Override noise estimation with default q\nparameter from crispector_config file. It's\nadvisable to set this flag for experiment\nwith a low number of off-target sites (<5).\nq is defined as the probability of an indel\nto occur through an edit event. Check\nCRISPECTOR paper for more details.\n[default: False]")), ToolInput(tag="in_confidence_interval", input_type=Float(optional=True), prefix="--confidence_interval", doc=InputDocumentation(doc="RANGE\nConfidence interval for the evaluated\nediting activity  [default: 0.95]")), ToolInput(tag="in_enable_substitutions", input_type=Boolean(optional=True), prefix="--enable_substitutions", doc=InputDocumentation(doc="Enable substitutions events for the\nquantification of edit events  [default:\nFalse]")), ToolInput(tag="in_suppress_site_output", input_type=Boolean(optional=True), prefix="--suppress_site_output", doc=InputDocumentation(doc="Do not create plots for sites (save memory\nand runtime)  [default: False]")), ToolInput(tag="in_keep_intermediate_files", input_type=Boolean(optional=True), prefix="--keep_intermediate_files", doc=InputDocumentation(doc="Keep intermediate files for debug purposes\n[default: False; required]")), ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="--verbose", doc=InputDocumentation(doc="Higher verbosity  [default: False]")), ToolInput(tag="in_file", input_type=File(), position=0, doc=InputDocumentation(doc="-m_r2, --mock_in2 PATH          Mock read read 2 input path, if FASTQ files"))], outputs=[ToolOutput(tag="out_report_output", output_type=File(optional=True), selector=InputSelector(input_to_select="in_report_output", type_hint=File()), doc=OutputDocumentation(doc="Path to output folder"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Crispector_V0_1_0().translate("wdl", allow_empty_container=True)

