from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Boolean, String, Int, Float

Whatshap_Phase_V0_1_0 = CommandToolBuilder(tool="whatshap_phase", base_command=["whatshap", "phase"], inputs=[ToolInput(tag="in_output", input_type=File(optional=True), prefix="--output", doc=InputDocumentation(doc="Output VCF file. Add .gz to the file name to get\ncompressed output. If omitted, use standard output.")), ToolInput(tag="in_reference", input_type=File(optional=True), prefix="--reference", doc=InputDocumentation(doc="Reference file. Provide this to detect alleles through\nre-alignment. If no index (.fai) exists, it will be\ncreated")), ToolInput(tag="in_no_reference", input_type=Boolean(optional=True), prefix="--no-reference", doc=InputDocumentation(doc="Detect alleles without requiring a reference, at the\nexpense of phasing quality (in particular for long\nreads)")), ToolInput(tag="in_tag", input_type=String(optional=True), prefix="--tag", doc=InputDocumentation(doc="Store phasing information with PS tag (standardized)\nor HP tag (used by GATK ReadBackedPhasing) (default:\nPS)")), ToolInput(tag="in_output_read_list", input_type=File(optional=True), prefix="--output-read-list", doc=InputDocumentation(doc="Write reads that have been used for phasing to FILE.")), ToolInput(tag="in_algorithm", input_type=String(optional=True), prefix="--algorithm", doc=InputDocumentation(doc="Phasing algorithm to use (default: whatshap)")), ToolInput(tag="in_merge_reads", input_type=Boolean(optional=True), prefix="--merge-reads", doc=InputDocumentation(doc="Merge reads which are likely to come from the same\nhaplotype (default: do not merge reads)")), ToolInput(tag="in_internal_downsampling", input_type=Int(optional=True), prefix="--internal-downsampling", doc=InputDocumentation(doc="Coverage reduction parameter in the internal core\nphasing algorithm. Higher values increase runtime\n*exponentially* while possibly improving phasing\nquality marginally. Avoid using this in the normal\ncase! (default: 15)")), ToolInput(tag="in_mapping_quality", input_type=Int(optional=True), prefix="--mapping-quality", doc=InputDocumentation(doc="Minimum mapping quality (default: 20)")), ToolInput(tag="in_indels", input_type=Boolean(optional=True), prefix="--indels", doc=InputDocumentation(doc="Also phase indels (default: do not phase indels)")), ToolInput(tag="in_ignore_read_groups", input_type=Boolean(optional=True), prefix="--ignore-read-groups", doc=InputDocumentation(doc="Ignore read groups in BAM/CRAM header and assume all\nreads come from the same sample.")), ToolInput(tag="in_sample", input_type=String(optional=True), prefix="--sample", doc=InputDocumentation(doc="Name of a sample to phase. If not given, all samples\nin the input VCF are phased. Can be used multiple\ntimes.")), ToolInput(tag="in_chromosome", input_type=String(optional=True), prefix="--chromosome", doc=InputDocumentation(doc="Name of chromosome to phase. If not given, all\nchromosomes in the input VCF are phased. Can be used\nmultiple times.")), ToolInput(tag="in_error_rate", input_type=Float(optional=True), prefix="--error-rate", doc=InputDocumentation(doc="The probability that a nucleotide is wrong in read\nmerging model (default: 0.15).")), ToolInput(tag="in_maximum_error_rate", input_type=Int(optional=True), prefix="--maximum-error-rate", doc=InputDocumentation(doc="The maximum error rate of any edge of the read merging\ngraph before discarding it (default: 0.25).")), ToolInput(tag="in_threshold", input_type=Int(optional=True), prefix="--threshold", doc=InputDocumentation(doc="The threshold of the ratio between the probabilities\nthat a pair of reads come from the same haplotype and\ndifferent haplotypes in the read merging model\n(default: 1000000).")), ToolInput(tag="in_negative_threshold", input_type=Int(optional=True), prefix="--negative-threshold", doc=InputDocumentation(doc="The threshold of the ratio between the probabilities\nthat a pair of reads come from different haplotypes\nand the same haplotype in the read merging model\n(default: 1000).")), ToolInput(tag="in_full_genotyping", input_type=Boolean(optional=True), prefix="--full-genotyping", doc=InputDocumentation(doc="Completely re-genotype all variants based on read\ndata, ignores all genotype data that might be present\nin the VCF (EXPERIMENTAL FEATURE).")), ToolInput(tag="in_distrust_genotypes", input_type=Boolean(optional=True), prefix="--distrust-genotypes", doc=InputDocumentation(doc="Allow switching variants from hetero- to homozygous in\nan optimal solution (see documentation).")), ToolInput(tag="in_include_homozygous", input_type=Boolean(optional=True), prefix="--include-homozygous", doc=InputDocumentation(doc="Also work on homozygous variants, which might be\nturned to heterozygous")), ToolInput(tag="in_default_gq", input_type=Int(optional=True), prefix="--default-gq", doc=InputDocumentation(doc="Default genotype quality used as cost of changing a\ngenotype when no genotype likelihoods are available\n(default 30)")), ToolInput(tag="in_gl_regularize_r", input_type=Float(optional=True), prefix="--gl-regularizer", doc=InputDocumentation(doc="Constant (float) to be used to regularize genotype\nlikelihoods read from input VCF (default None).")), ToolInput(tag="in_changed_genotype_list", input_type=File(optional=True), prefix="--changed-genotype-list", doc=InputDocumentation(doc="Write list of changed genotypes to FILE.")), ToolInput(tag="in_ped", input_type=File(optional=True), prefix="--ped", doc=InputDocumentation(doc="/FAM         Use pedigree information in PED file to improve\nphasing (switches to PedMEC algorithm). Columns 2, 3,\n4 must refer to child, mother, and father sample names\nas used in the VCF and BAM/CRAM. Other columns are\nignored.")), ToolInput(tag="in_recombination_list", input_type=File(optional=True), prefix="--recombination-list", doc=InputDocumentation(doc="Write putative recombination events to FILE.")), ToolInput(tag="in_recomb_rate", input_type=Float(optional=True), prefix="--recombrate", doc=InputDocumentation(doc="Recombination rate in cM/Mb (used with --ped). If\ngiven, a constant recombination rate is assumed\n(default: 1.26cM/Mb).")), ToolInput(tag="in_gen_map", input_type=File(optional=True), prefix="--genmap", doc=InputDocumentation(doc="File with genetic map (used with --ped) to be used\ninstead of constant recombination rate, i.e. overrides\noption --recombrate.")), ToolInput(tag="in_no_genetic_haplo_typing", input_type=Boolean(optional=True), prefix="--no-genetic-haplotyping", doc=InputDocumentation(doc="Do not merge blocks that are not connected by reads\n(i.e. solely based on genotype status). Default: when\nin --ped mode, merge all blocks that contain at least\none homozygous genotype in at least one individual\ninto one block.")), ToolInput(tag="in_use_ped_samples", input_type=Boolean(optional=True), prefix="--use-ped-samples", doc=InputDocumentation(doc="Only work on samples mentioned in the provided PED\nfile.\n")), ToolInput(tag="in_vcf", input_type=String(), position=0, doc=InputDocumentation(doc="VCF or BCF file with variants to be phased (can be\ngzip-compressed)")), ToolInput(tag="in_phase_input", input_type=String(), position=1, doc=InputDocumentation(doc="BAM, CRAM, VCF or BCF file(s) with phase information,\neither through sequencing reads (BAM, CRAM) or through\nphased blocks (VCF, BCF)"))], outputs=[ToolOutput(tag="out_output", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output", type_hint=File()), doc=OutputDocumentation(doc="Output VCF file. Add .gz to the file name to get\ncompressed output. If omitted, use standard output."))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Whatshap_Phase_V0_1_0().translate("wdl", allow_empty_container=True)

