from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import String, Int, File

Sat_Bsa_V0_1_0 = CommandToolBuilder(tool="Sat_BSA", base_command=["Sat-BSA"], inputs=[ToolInput(tag="in_chromosome_name_selecting", input_type=String(optional=True), prefix="-c", doc=InputDocumentation(doc=": Chromosome name for selecting the aligned reads.")), ToolInput(tag="in_start_position_selecting", input_type=Int(optional=True), prefix="-s", doc=InputDocumentation(doc=": Start position for selecting the aligned reads.")), ToolInput(tag="in_end_position_selecting", input_type=Int(optional=True), prefix="-e", doc=InputDocumentation(doc=": End position for selecting the aligned reads.")), ToolInput(tag="in_full_path_listing_bam_files", input_type=File(optional=True), prefix="-b", doc=InputDocumentation(doc=": Full path of bam_list.txt listing bam files (Input format 1).")), ToolInput(tag="in_full_path_listing_fastagz_files", input_type=File(optional=True), prefix="-f", doc=InputDocumentation(doc=": Full path of fa_list.txt listing fasta.gz files (Input format 2).")), ToolInput(tag="in_thread_number_default", input_type=Int(optional=True), prefix="-t", doc=InputDocumentation(doc=": Thread number. Default=[1]")), ToolInput(tag="in_full_path_canu", input_type=File(optional=True), prefix="-d", doc=InputDocumentation(doc=": Full path of Canu.")), ToolInput(tag="in_genome_size_mb", input_type=Int(optional=True), prefix="-g", doc=InputDocumentation(doc=": Genome size in Mb set in Canu.")), ToolInput(tag="in_read_status_set", input_type=String(optional=True), prefix="-r", doc=InputDocumentation(doc=": Read status set in Canu. Default=[-nanopore-raw].")), ToolInput(tag="in_mapping_quality_value", input_type=Int(optional=True), prefix="-q", doc=InputDocumentation(doc=": The mapping quality value excluded from analysis. Default=[0]")), ToolInput(tag="in_pb_used_sequence", input_type=String(optional=True), prefix="-i", doc=InputDocumentation(doc="or pb  : The used sequence reads type (Oxford Nanopore Technologies[ont] or PacBio[pb]).")), ToolInput(tag="in_defining_promoter_size", input_type=Int(optional=True), prefix="-p", doc=InputDocumentation(doc=": Defining promoter size applied for identifying SVs. Default=[0]")), ToolInput(tag="in_threshold_pvalue_fishers", input_type=Int(optional=True), prefix="-v", doc=InputDocumentation(doc=": Threshold for P-value from Fishers exact test. Default=[0.05]")), ToolInput(tag="in_w", input_type=String(optional=True), prefix="-w", doc=InputDocumentation(doc="")), ToolInput(tag="in_required", input_type=String(optional=True), position=0, doc=InputDocumentation(doc="")), ToolInput(tag="in_arguments", input_type=String(optional=True), position=1, doc=InputDocumentation(doc=""))], outputs=[], container="quay.io/biocontainers/sat-bsa:1.12--0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Sat_Bsa_V0_1_0().translate("wdl")

