from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, File, Int

Gtftk_Tabulate_V0_1_0 = CommandToolBuilder(tool="gtftk_tabulate", base_command=["gtftk", "tabulate"], inputs=[ToolInput(tag="in_select_transcript_ids", input_type=Boolean(optional=True), prefix="--select-transcript-ids", doc=InputDocumentation(doc="A shortcuts for '-k transcript_id'. (default: False)")), ToolInput(tag="in_select_gene_ids", input_type=Boolean(optional=True), prefix="--select-gene_ids", doc=InputDocumentation(doc="A shortcuts for '-k gene_id'. (default: False)")), ToolInput(tag="in_select_gene_names", input_type=Boolean(optional=True), prefix="--select-gene-names", doc=InputDocumentation(doc="A shortcuts for '-k gene_name'. (default: False)")), ToolInput(tag="in_select_exon_ids", input_type=Boolean(optional=True), prefix="--select-exon-ids", doc=InputDocumentation(doc="A shortcuts for '-k exon_ids'. (default: False)")), ToolInput(tag="in_input_file", input_type=Boolean(optional=True), prefix="--inputfile", doc=InputDocumentation(doc="Path to the GTF file. Default to STDIN (default: <stdin>)")), ToolInput(tag="in_output_file", input_type=File(optional=True), prefix="--outputfile", doc=InputDocumentation(doc="Output file. (default: <stdout>)")), ToolInput(tag="in_separator", input_type=Boolean(optional=True), prefix="--separator", doc=InputDocumentation(doc="The output field separator. (default: )")), ToolInput(tag="in_key", input_type=Boolean(optional=True), prefix="--key", doc=InputDocumentation(doc="A comma-separated list of key names. (default: *)")), ToolInput(tag="in_unique", input_type=Boolean(optional=True), prefix="--unique", doc=InputDocumentation(doc="Print a non redondant list of lines. (default: False)")), ToolInput(tag="in_no_header", input_type=Boolean(optional=True), prefix="--no-header", doc=InputDocumentation(doc="Don't print the header line. (default: False)")), ToolInput(tag="in_no_unset", input_type=Boolean(optional=True), prefix="--no-unset", doc=InputDocumentation(doc="Don't print lines containing '.' (unset values) (default: False)")), ToolInput(tag="in_accept_undef", input_type=Boolean(optional=True), prefix="--accept-undef", doc=InputDocumentation(doc="Print line for which the key is undefined (i.e, '?', does not exists). (default: False)")), ToolInput(tag="in_no_basic", input_type=Boolean(optional=True), prefix="--no-basic", doc=InputDocumentation(doc="In case key is set to 'all' or '*', don't write basic attributes. (default: False)")), ToolInput(tag="in_verbosity", input_type=Boolean(optional=True), prefix="--verbosity", doc=InputDocumentation(doc="Set output verbosity ([0-3]). (default: 0)")), ToolInput(tag="in_no_date", input_type=File(optional=True), prefix="--no-date", doc=InputDocumentation(doc="Do not add date to output file names. (default: False)")), ToolInput(tag="in_add_chr", input_type=Boolean(optional=True), prefix="--add-chr", doc=InputDocumentation(doc="Add 'chr' to chromosome names before printing output. (default: False)")), ToolInput(tag="in_tmp_dir", input_type=Boolean(optional=True), prefix="--tmp-dir", doc=InputDocumentation(doc="Keep all temporary files into this folder. (default: None)")), ToolInput(tag="in_keep_all", input_type=Boolean(optional=True), prefix="--keep-all", doc=InputDocumentation(doc="Try to keep all temporary files even if process does not terminate normally. (default: False)")), ToolInput(tag="in_logger_file", input_type=Boolean(optional=True), prefix="--logger-file", doc=InputDocumentation(doc="Stores the arguments passed to the command into a file. (default: None)")), ToolInput(tag="in_write_message_to_file", input_type=Boolean(optional=True), prefix="--write-message-to-file", doc=InputDocumentation(doc="Store all message into a file. (default: None)")), ToolInput(tag="in_optional_arguments", input_type=Int(), position=0, doc=InputDocumentation(doc="optional arguments:"))], outputs=[ToolOutput(tag="out_output_file", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output_file", type_hint=File()), doc=OutputDocumentation(doc="Output file. (default: <stdout>)")), ToolOutput(tag="out_no_date", output_type=File(optional=True), selector=InputSelector(input_to_select="in_no_date", type_hint=File()), doc=OutputDocumentation(doc="Do not add date to output file names. (default: False)"))], container="quay.io/biocontainers/pygtftk:1.2.7--py36h4edd4cd_1", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Gtftk_Tabulate_V0_1_0().translate("wdl")

