from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Int, File, String, Directory, Boolean, Float

Tb_Profiler_Profile_V0_1_0 = CommandToolBuilder(tool="tb_profiler_profile", base_command=["tb-profiler", "profile"], inputs=[ToolInput(tag="in_read_one", input_type=Int(optional=True), prefix="--read1", doc=InputDocumentation(doc="First read file (default: None)")), ToolInput(tag="in_read_two", input_type=Int(optional=True), prefix="--read2", doc=InputDocumentation(doc="Second read file (default: None)")), ToolInput(tag="in_bam", input_type=File(optional=True), prefix="--bam", doc=InputDocumentation(doc="BAM file. Make sure it has been generated using the\nH37Rv genome (GCA_000195955.2) (default: None)")), ToolInput(tag="in_platform", input_type=String(optional=True), prefix="--platform", doc=InputDocumentation(doc="NGS Platform used to generate data (default: illumina)")), ToolInput(tag="in_db", input_type=String(optional=True), prefix="--db", doc=InputDocumentation(doc="Mutation panel name (default: tbdb)")), ToolInput(tag="in_external_db", input_type=File(optional=True), prefix="--external_db", doc=InputDocumentation(doc="Path to db files prefix (overrides '--db' parameter)\n(default: None)")), ToolInput(tag="in_prefix", input_type=String(optional=True), prefix="--prefix", doc=InputDocumentation(doc="Sample prefix for all results generated (default:\ntbprofiler)")), ToolInput(tag="in_dir", input_type=Directory(optional=True), prefix="--dir", doc=InputDocumentation(doc="Storage directory (default: .)")), ToolInput(tag="in_csv", input_type=Boolean(optional=True), prefix="--csv", doc=InputDocumentation(doc="Add CSV output (default: False)")), ToolInput(tag="in_txt", input_type=Boolean(optional=True), prefix="--txt", doc=InputDocumentation(doc="Add text output (default: False)")), ToolInput(tag="in_pdf", input_type=Boolean(optional=True), prefix="--pdf", doc=InputDocumentation(doc="Add PDF output. This requires pdflatex to be installed\n(default: False)")), ToolInput(tag="in_add_columns", input_type=String(optional=True), prefix="--add_columns", doc=InputDocumentation(doc="Add additional columns found in the mutation database\nto the text and csv results (default: None)")), ToolInput(tag="in_add_mutation_metadata", input_type=Boolean(optional=True), prefix="--add_mutation_metadata", doc=InputDocumentation(doc="Add mutation data from the tbdr database (not\ncurrently functional) (default: False)")), ToolInput(tag="in_mapper", input_type=String(optional=True), prefix="--mapper", doc=InputDocumentation(doc="Mapping tool to use. If you are using nanopore data it\nwill default to minimap2 (default: bwa)")), ToolInput(tag="in_caller", input_type=String(optional=True), prefix="--caller", doc=InputDocumentation(doc="Variant calling tool to use. (default: bcftools)")), ToolInput(tag="in_calling_params", input_type=String(optional=True), prefix="--calling_params", doc=InputDocumentation(doc="Override default parameters for variant calling\n(default: None)")), ToolInput(tag="in_min_depth", input_type=Int(optional=True), prefix="--min_depth", doc=InputDocumentation(doc="Minimum depth required to call variant. Bases with\ndepth below this cutoff will be marked as missing\n(default: 10)")), ToolInput(tag="in_af", input_type=Float(optional=True), prefix="--af", doc=InputDocumentation(doc="Minimum allele frequency to call variants (default:\n0.1)")), ToolInput(tag="in_reporting_af", input_type=Float(optional=True), prefix="--reporting_af", doc=InputDocumentation(doc="Minimum allele frequency to use variants for\nprediction (default: 0.1)")), ToolInput(tag="in_coverage_fraction_threshold", input_type=Int(optional=True), prefix="--coverage_fraction_threshold", doc=InputDocumentation(doc="Cutoff used to calculate fraction of region covered by\n<= this value (default: 0)")), ToolInput(tag="in_missing_cov_threshold", input_type=String(optional=True), prefix="--missing_cov_threshold", doc=InputDocumentation(doc="Cutoff used to positions/codons in genes which are\nmissing (this argument has now been merged with\n--min_depth argument and will be deprecated in future\nreleases) (default: None)")), ToolInput(tag="in_no_trim", input_type=Boolean(optional=True), prefix="--no_trim", doc=InputDocumentation(doc="Don't trim files using trimmomatic (default: False)")), ToolInput(tag="in_no_flag_stat", input_type=Boolean(optional=True), prefix="--no_flagstat", doc=InputDocumentation(doc="Don't collect flagstats (default: False)")), ToolInput(tag="in_no_delly", input_type=Boolean(optional=True), prefix="--no_delly", doc=InputDocumentation(doc="Don't run delly (default: False)")), ToolInput(tag="in_threads", input_type=Int(optional=True), prefix="--threads", doc=InputDocumentation(doc="Threads to use (default: 1)")), ToolInput(tag="in_verbose", input_type=String(optional=True), prefix="--verbose", doc=InputDocumentation(doc="Verbosity increases from 0 to 2 (default: 0)")), ToolInput(tag="in_no_cleanup", input_type=Boolean(optional=True), prefix="--no_cleanup", doc=InputDocumentation(doc="Don't remove temporary files on error (default: False)"))], outputs=[], container="quay.io/biocontainers/tb-profiler:3.0.1--pypyh3252c3a_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Tb_Profiler_Profile_V0_1_0().translate("wdl")

