from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Int, String, Boolean, File

Distanceest_Ssq_V0_1_0 = CommandToolBuilder(tool="DistanceEst_ssq", base_command=["DistanceEst-ssq"], inputs=[ToolInput(tag="in_mind", input_type=Int(optional=True), prefix="--mind", doc=InputDocumentation(doc="minimum distance between contigs [-(k-1)]")), ToolInput(tag="in_maxd", input_type=String(optional=True), prefix="--maxd", doc=InputDocumentation(doc="maximum distance between contigs")), ToolInput(tag="in_fr", input_type=Boolean(optional=True), prefix="--fr", doc=InputDocumentation(doc="force the orientation to forward-reverse")), ToolInput(tag="in_rf", input_type=Boolean(optional=True), prefix="--rf", doc=InputDocumentation(doc="force the orientation to reverse-forward")), ToolInput(tag="in_km_er", input_type=Int(optional=True), prefix="--kmer", doc=InputDocumentation(doc="set --mind to -(k-1) bp")), ToolInput(tag="in_min_align", input_type=Int(optional=True), prefix="--min-align", doc=InputDocumentation(doc="the minimal alignment size [1]")), ToolInput(tag="in_n_pairs", input_type=Int(optional=True), prefix="--npairs", doc=InputDocumentation(doc="minimum number of pairs")), ToolInput(tag="in_seed_length", input_type=Int(optional=True), prefix="--seed-length", doc=InputDocumentation(doc="minimum length of the seed contigs")), ToolInput(tag="in_min_mapq", input_type=Int(optional=True), prefix="--min-mapq", doc=InputDocumentation(doc="ignore alignments with mapping quality\nless than this threshold [10]")), ToolInput(tag="in_out", input_type=File(optional=True), prefix="--out", doc=InputDocumentation(doc="write result to FILE")), ToolInput(tag="in_mle", input_type=Boolean(optional=True), prefix="--mle", doc=InputDocumentation(doc="use the MLE [default]\n(maximum likelihood estimator)")), ToolInput(tag="in_mean", input_type=Boolean(optional=True), prefix="--mean", doc=InputDocumentation(doc="use the difference of the population mean\nand the sample mean")), ToolInput(tag="in_dist", input_type=Boolean(optional=True), prefix="--dist", doc=InputDocumentation(doc="output graph in dist format [default]")), ToolInput(tag="in_dot", input_type=Boolean(optional=True), prefix="--dot", doc=InputDocumentation(doc="output graph in dot format")), ToolInput(tag="in_threads", input_type=Int(optional=True), prefix="--threads", doc=InputDocumentation(doc="use N parallel threads [1]")), ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="--verbose", doc=InputDocumentation(doc="display verbose output")), ToolInput(tag="in_db", input_type=File(optional=True), prefix="--db", doc=InputDocumentation(doc="specify path of database repository in FILE")), ToolInput(tag="in_library", input_type=String(optional=True), prefix="--library", doc=InputDocumentation(doc="specify library NAME for sqlite")), ToolInput(tag="in_strain", input_type=String(optional=True), prefix="--strain", doc=InputDocumentation(doc="specify strain NAME for sqlite")), ToolInput(tag="in_species", input_type=String(optional=True), prefix="--species", doc=InputDocumentation(doc="specify species NAME for sqlite")), ToolInput(tag="in_hist", input_type=String(), position=0, doc=InputDocumentation(doc="distribution of fragments size")), ToolInput(tag="in_pair", input_type=String(), position=1, doc=InputDocumentation(doc="alignments between contigs"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Distanceest_Ssq_V0_1_0().translate("wdl", allow_empty_container=True)

