from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, Int, File, String, Directory

Minnow_V0_1_0 = CommandToolBuilder(tool="minnow", base_command=["minnow"], inputs=[ToolInput(tag="in_a_levin_mode", input_type=Boolean(optional=True), prefix="--alevin-mode", doc=InputDocumentation(doc="The program would assume that the input matrix is obtained from Alevin")), ToolInput(tag="in_splatter_mode", input_type=Boolean(optional=True), prefix="--splatter-mode", doc=InputDocumentation(doc="matrix file is obtained from running splatter")), ToolInput(tag="in_normal_mode", input_type=Boolean(optional=True), prefix="--normal-mode", doc=InputDocumentation(doc="user provided matrix")), ToolInput(tag="in_test_uniq_ness", input_type=Boolean(optional=True), prefix="--testUniqness", doc=InputDocumentation(doc="matrix file is obtained from running splatter")), ToolInput(tag="in_reverse_uniq_ness", input_type=Boolean(optional=True), prefix="--reverseUniqness", doc=InputDocumentation(doc="matrix file is obtained from running splatter")), ToolInput(tag="in_use_weibull", input_type=Boolean(optional=True), prefix="--useWeibull", doc=InputDocumentation(doc="matrix file is obtained from running splatter")), ToolInput(tag="in_num_of_doublets", input_type=Int(optional=True), prefix="--numOfDoublets", doc=InputDocumentation(doc="Number of doublets to be generated")), ToolInput(tag="in_mat_dir", input_type=File(optional=True), prefix="--matdir", doc=InputDocumentation(doc="directory with matrix file/ if this is a file instead of a dir")), ToolInput(tag="in_outdir", input_type=File(optional=True), prefix="--outdir", doc=InputDocumentation(doc="the simulated reads will be written here")), ToolInput(tag="in_num_molfile", input_type=Int(optional=True), prefix="--numMolFile", doc=InputDocumentation(doc="Number of molecules generated from each cell")), ToolInput(tag="in_gen_code", input_type=Boolean(optional=True), prefix="--gencode", doc=InputDocumentation(doc="gencode reference has | separator")), ToolInput(tag="in_ref_file", input_type=File(optional=True), prefix="--reffile", doc=InputDocumentation(doc="transcriptome reference file (assumed from fasta file)")), ToolInput(tag="in_g_two_t", input_type=Int(optional=True), prefix="--g2t", doc=InputDocumentation(doc="tab separated list of Gene to Transcirpt mapping")), ToolInput(tag="in_r_spd", input_type=String(optional=True), prefix="--rspd", doc=InputDocumentation(doc="tab separated read start position distribution")), ToolInput(tag="in_bfh", input_type=File(optional=True), prefix="--bfh", doc=InputDocumentation(doc="BFH file")), ToolInput(tag="in_gene_prob", input_type=File(optional=True), prefix="--geneProb", doc=InputDocumentation(doc="Gene level probability file (TSV)")), ToolInput(tag="in_count_prob", input_type=File(optional=True), prefix="--countProb", doc=InputDocumentation(doc="global scale count probability file")), ToolInput(tag="in_velocity", input_type=Boolean(optional=True), prefix="--velocity", doc=InputDocumentation(doc="In velocity mode we generate reads from exon-exon junction")), ToolInput(tag="in_binary", input_type=Boolean(optional=True), prefix="--binary", doc=InputDocumentation(doc="If the matrix file is written in binary")), ToolInput(tag="in_dbg", input_type=Boolean(optional=True), prefix="--dbg", doc=InputDocumentation(doc="Use the provided GFA file and BFH")), ToolInput(tag="in_no_dump", input_type=Boolean(optional=True), prefix="--noDump", doc=InputDocumentation(doc="will use the model file made")), ToolInput(tag="in_gfa", input_type=File(optional=True), prefix="--gfa", doc=InputDocumentation(doc="gfa file for contigs")), ToolInput(tag="in_dup_counts", input_type=Directory(optional=True), prefix="--dupCounts", doc=InputDocumentation(doc="for making minnow read the dup counts TSV filtered_cb_frequency.txt in the same folder")), ToolInput(tag="in_use_whitelist", input_type=Boolean(optional=True), prefix="--useWhiteList", doc=InputDocumentation(doc="Flag for making minnow read the dup counts TSV filtered_cb_frequency.txt in the same folder")), ToolInput(tag="in_generate_noisy_cells", input_type=Boolean(optional=True), prefix="--generateNoisyCells", doc=InputDocumentation(doc="Flag for making minnow read the dup counts TSV filtered_cb_frequency.txt in the same folder")), ToolInput(tag="in_polya", input_type=Boolean(optional=True), prefix="--polyA", doc=InputDocumentation(doc="Flag to sample with polyA sites this should accompany --polyAsite and --polyAfraction")), ToolInput(tag="in_polya_site", input_type=File(optional=True), prefix="--polyAsite", doc=InputDocumentation(doc="Fasta file with polyA sites")), ToolInput(tag="in_poly_a_fraction", input_type=File(optional=True), prefix="--polyAfraction", doc=InputDocumentation(doc="File with polyA site fraction")), ToolInput(tag="in_sample_cells", input_type=String(optional=True), prefix="--sampleCells", doc=InputDocumentation(doc="sample this many cells from the set of all cells")), ToolInput(tag="in_intron_file", input_type=File(optional=True), prefix="--intronfile", doc=InputDocumentation(doc="Intron bed file which contains the intron boundaries per transcript")), ToolInput(tag="in_genome", input_type=File(optional=True), prefix="--genome", doc=InputDocumentation(doc="genome FASTA file")), ToolInput(tag="in_number_of_cells", input_type=Int(optional=True), prefix="--numberOfCells", doc=InputDocumentation(doc="Number of cells required for simulation (default = 10)")), ToolInput(tag="in_number_of_transcripts", input_type=Int(optional=True), prefix="--numberOfTranscripts", doc=InputDocumentation(doc="Number of transcripts for simulation (default = 100)")), ToolInput(tag="in_clusters", input_type=Int(optional=True), prefix="--clusters", doc=InputDocumentation(doc="Gene cluster file (should be ported with --dbg)")), ToolInput(tag="in_pcr", input_type=Int(optional=True), prefix="--PCR", doc=InputDocumentation(doc="Maximum cycles to repeat PCR (default = 5)")), ToolInput(tag="in_pcr_model_six", input_type=Int(optional=True), prefix="--PCRmodel6", doc=InputDocumentation(doc="from Best, Katharine et al. (2015)")), ToolInput(tag="in_error_rate", input_type=String(optional=True), prefix="--error-rate", doc=InputDocumentation(doc="error rate to be introduced while producing sequences")), ToolInput(tag="in_num_threads", input_type=Int(optional=True), prefix="--num-threads", doc=InputDocumentation(doc="number of threads to parallelize the process")), ToolInput(tag="in_eq_class_dir", input_type=Directory(optional=True), prefix="--eqClassDir", doc=InputDocumentation(doc="directory containing relevent files produced by the python script")), ToolInput(tag="in_cluster", input_type=File(optional=True), prefix="--cluster", doc=InputDocumentation(doc="Optional cluster file to model cluster based histogram"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Minnow_V0_1_0().translate("wdl", allow_empty_container=True)

