from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Float, Directory, Int, String, File, Boolean

Constax_V0_1_0 = CommandToolBuilder(tool="constax", base_command=["constax"], inputs=[ToolInput(tag="in_conf", input_type=Float(optional=True), prefix="--conf", doc=InputDocumentation(doc="Classification confidence threshold (default: 0.8)")), ToolInput(tag="in_num_threads", input_type=Directory(optional=True), prefix="--num_threads", doc=InputDocumentation(doc="directory to for output files (default: 1)")), ToolInput(tag="in_m_hits", input_type=Int(optional=True), prefix="--mhits", doc=InputDocumentation(doc="Maximum number of BLAST hits to use, for use with -b\noption (default: 10)")), ToolInput(tag="in_evalue", input_type=Float(optional=True), prefix="--evalue", doc=InputDocumentation(doc="Maximum expect value of BLAST hits to use, for use\nwith -b option (default: 1.0)")), ToolInput(tag="in_p_iden", input_type=Float(optional=True), prefix="--p_iden", doc=InputDocumentation(doc="Minimum proportion identity of BLAST hits to use, for\nuse with -b option (default: 0.0)")), ToolInput(tag="in_db", input_type=String(optional=True), prefix="--db", doc=InputDocumentation(doc="Database to train classifiers, in FASTA format\n(default: )")), ToolInput(tag="in_train_file", input_type=File(optional=True), prefix="--trainfile", doc=InputDocumentation(doc="Path to which training files will be written (default:\n./training_files)")), ToolInput(tag="in_input", input_type=File(optional=True), prefix="--input", doc=InputDocumentation(doc="Input file in FASTA format containing sequence records\nto classify (default: otus.fasta)")), ToolInput(tag="in_output", input_type=Directory(optional=True), prefix="--output", doc=InputDocumentation(doc="Output directory for classifications (default:\n./outputs)")), ToolInput(tag="in_tax", input_type=Directory(optional=True), prefix="--tax", doc=InputDocumentation(doc="Directory for taxonomy assignments (default:\n./taxonomy_assignments)")), ToolInput(tag="in_train", input_type=Boolean(optional=True), prefix="--train", doc=InputDocumentation(doc="Complete training if specified (default: False)")), ToolInput(tag="in_blast", input_type=Boolean(optional=True), prefix="--blast", doc=InputDocumentation(doc="Use BLAST instead of UTAX if specified (default:\nFalse)")), ToolInput(tag="in_select_by_keyword", input_type=File(optional=True), prefix="--select_by_keyword", doc=InputDocumentation(doc="Takes a keyword argument and --input FASTA file to\nproduce a filtered database with headers containing\nthe keyword with name --output (default: False)")), ToolInput(tag="in_msu_hpcc", input_type=Boolean(optional=True), prefix="--msu_hpcc", doc=InputDocumentation(doc="If specified, use executable paths on Michigan State\nUniversity HPCC. Overrides other path arguments\n(default: False)")), ToolInput(tag="in_conservative", input_type=Boolean(optional=True), prefix="--conservative", doc=InputDocumentation(doc="If specified, use conservative consensus rule (2 False\n= False winner) (default: False)")), ToolInput(tag="in_make_plot", input_type=Boolean(optional=True), prefix="--make_plot", doc=InputDocumentation(doc="If specified, run R script to make plot of classified\ntaxa (default: False)")), ToolInput(tag="in_check", input_type=Boolean(optional=True), prefix="--check", doc=InputDocumentation(doc="If specified, runs checks but stops before training or\nclassifying (default: False)")), ToolInput(tag="in_mem", input_type=Int(optional=True), prefix="--mem", doc=InputDocumentation(doc="Memory available to use for RDP, in MB. 32000MB\nrecommended for UNITE, 128000MB for SILVA (default:\n32000)")), ToolInput(tag="in_sin_tax_path", input_type=File(optional=True), prefix="--sintax_path", doc=InputDocumentation(doc="Path to USEARCH/VSEARCH executable for SINTAX\nclassification (default: False)")), ToolInput(tag="in_u_tax_path", input_type=File(optional=True), prefix="--utax_path", doc=InputDocumentation(doc="Path to USEARCH executable for UTAX classification\n(default: False)")), ToolInput(tag="in_rdp_path", input_type=File(optional=True), prefix="--rdp_path", doc=InputDocumentation(doc="Path to RDP classifier.jar file (default: False)")), ToolInput(tag="in_cons_tax_path", input_type=File(optional=True), prefix="--constax_path", doc=InputDocumentation(doc="Path to CONSTAX scripts (default: False)")), ToolInput(tag="in_path_file", input_type=File(optional=True), prefix="--pathfile", doc=InputDocumentation(doc="File with paths to SINTAX, UTAX, RDP, and CONSTAX\nexecutables (default: pathfile.txt)")), ToolInput(tag="in_isolates", input_type=File(optional=True), prefix="--isolates", doc=InputDocumentation(doc="FASTA formatted file of isolates to use BLAST against\n(default: False)")), ToolInput(tag="in_isolates_query_coverage", input_type=Int(optional=True), prefix="--isolates_query_coverage", doc=InputDocumentation(doc="Threshold of sequence query coverage to report isolate\nmatches (default: 75)")), ToolInput(tag="in_isolates_percent_identity", input_type=Int(optional=True), prefix="--isolates_percent_identity", doc=InputDocumentation(doc="Threshold of aligned sequence percent identity to\nreport isolate matches (default: 1)")), ToolInput(tag="in_high_level_db", input_type=File(optional=True), prefix="--high_level_db", doc=InputDocumentation(doc="FASTA database file of representative sequences for\nassignment of high level taxonomy (default: False)")), ToolInput(tag="in_high_level_query_coverage", input_type=Int(optional=True), prefix="--high_level_query_coverage", doc=InputDocumentation(doc="Threshold of sequence query coverage to report high-\nlevel taxonomy matches (default: 75)")), ToolInput(tag="in_high_level_percent_identity", input_type=Int(optional=True), prefix="--high_level_percent_identity", doc=InputDocumentation(doc="Threshold of aligned sequence percent identity to\nreport high-level taxonomy matches (default: 1)"))], outputs=[ToolOutput(tag="out_num_threads", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_num_threads", type_hint=File()), doc=OutputDocumentation(doc="directory to for output files (default: 1)")), ToolOutput(tag="out_output", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_output", type_hint=File()), doc=OutputDocumentation(doc="Output directory for classifications (default:\n./outputs)"))], container="quay.io/biocontainers/constax:2.0.9--hdfd78af_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Constax_V0_1_0().translate("wdl")

