from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, Int, File, String

Pileupcaller_V0_1_0 = CommandToolBuilder(tool="pileupCaller", base_command=["pileupCaller"], inputs=[ToolInput(tag="in_random_haploid", input_type=Boolean(optional=True), prefix="--randomHaploid", doc=InputDocumentation(doc="This method samples one read at random at each site,\nand uses the allele on that read as the one for the\nactual genotype. This results in a haploid call")), ToolInput(tag="in_majority_call", input_type=Boolean(optional=True), prefix="--majorityCall", doc=InputDocumentation(doc="Pick the allele supported by the most reads at a\nsite. If an equal numbers of alleles fulfil this,\npick one at random. This results in a haploid call.\nSee --downSampling for best practices for calling\nrare variants")), ToolInput(tag="in_downsampling", input_type=Boolean(optional=True), prefix="--downSampling", doc=InputDocumentation(doc="When this switch is given, the MajorityCalling mode\nwill downsample from the total number of reads a\nnumber of reads (without replacement) equal to the\n--minDepth given. This mitigates reference bias in\nthe MajorityCalling model, which increases with\nhigher coverage. The recommendation for rare-allele\ncalling is --majorityCall --downsampling --minDepth 3")), ToolInput(tag="in_random_diploid", input_type=Boolean(optional=True), prefix="--randomDiploid", doc=InputDocumentation(doc="Sample two reads at random (without replacement) at\neach site and represent the individual by a diploid\ngenotype constructed from those two random picks.\nThis will always assign missing data to positions\nwhere only one read is present, even if minDepth=1.\nThe main use case for this option is for estimating\nmean heterozygosity across sites.")), ToolInput(tag="in_keep_incongruent_reads", input_type=Boolean(optional=True), prefix="--keepIncongruentReads", doc=InputDocumentation(doc="By default, pileupCaller now removes reads with\ntri-allelic alleles that are neither of the two\nalleles specified in the SNP file. To keep those\nreads for sampling, set this flag. With this option\ngiven, if the sampled read has a tri-allelic allele\nthat is neither of the two given alleles in the SNP\nfile, a missing genotype is generated. IMPORTANT\nNOTE: The default behaviour has changed in\npileupCaller version 1.4.0. If you want to emulate\nthe previous behaviour, use this flag. I recommend\nnow to NOT set this flag and use the new behaviour.")), ToolInput(tag="in_seed", input_type=Int(optional=True), prefix="--seed", doc=InputDocumentation(doc="random seed used for the random number generator. If\nnot given, use system clock to seed the random number\ngenerator.")), ToolInput(tag="in_min_depth", input_type=Int(optional=True), prefix="--minDepth", doc=InputDocumentation(doc="specify the minimum depth for a call. For sites with\nfewer reads than this number, declare\nMissing (default: 1)")), ToolInput(tag="in_skip_transitions", input_type=Boolean(optional=True), prefix="--skipTransitions", doc=InputDocumentation(doc="skip transition SNPs entirely in the output,\nresulting in a dataset with fewer sites.")), ToolInput(tag="in_transitions_missing", input_type=Boolean(optional=True), prefix="--transitionsMissing", doc=InputDocumentation(doc="mark transitions as missing in the output, but do\noutput the sites.")), ToolInput(tag="in_single_strand_mode", input_type=Boolean(optional=True), prefix="--singleStrandMode", doc=InputDocumentation(doc="[THIS IS CURRENTLY AN EXPERIMENTAL FEATURE]. At C/T\npolymorphisms, ignore reads aligning to the forward\nstrand. At G/A polymorphisms, ignore reads aligning\nto the reverse strand. This should remove post-mortem\ndamage in ancient DNA libraries prepared with the\nnon-UDG single-stranded protocol.")), ToolInput(tag="in_snp_file", input_type=File(optional=True), prefix="--snpFile", doc=InputDocumentation(doc="an Eigenstrat-formatted SNP list file for the\npositions and alleles to call. All positions in the\nSNP file will be output, adding missing data where\nthere is no data. Note that pileupCaller\nautomatically checks whether alleles in the SNP file\nare flipped with respect to the human reference, and\nin those cases flips the genotypes accordingly. But\nit assumes that the strand-orientation of the SNPs\ngiven in the SNP list is the one in the reference\ngenome used in the BAM file underlying the pileup\ninput. Note that both the SNP file and the incoming\npileup data have to be ordered by chromosome and\nposition, and this is checked. The chromosome order\nin humans is 1-22,X,Y,MT. Chromosome can generally\nbegin with 'chr'. In case of non-human data with\ndifferent chromosome names, you should convert all\nnames to numbers. They will always considered to be\nnumerically ordered, even beyond 22. Finally, I note\nthat for internally, X is converted to 23, Y to 24\nand MT to 90. This is the most widely used encoding\nin Eigenstrat databases for human data, so using a\nSNP file with that encoding will automatically be\ncorrectly aligned to pileup data with actual\nchromosome names X, Y and MT (or chrX, chrY and\nchrMT, respectively).")), ToolInput(tag="in_eigen_strat_out", input_type=File(optional=True), prefix="--eigenstratOut", doc=InputDocumentation(doc="Set Eigenstrat as output format. Specify the\nfilenames for the EigenStrat SNP and IND file\noutputs: <FILE_PREFIX>.snp.txt and\n<FILE_PREFIX>.ind.txt If not set, output will be\nFreqSum (Default). Note that freqSum format,\ndescribed at\nhttps://rarecoal-docs.readthedocs.io/en/latest/rarecoal-tools.html#vcf2freqsum,\nis useful for testing your pipeline, since it's\noutput to standard out")), ToolInput(tag="in_sample_pop_name", input_type=File(optional=True), prefix="--samplePopName", doc=InputDocumentation(doc="specify the population name of the samples, which is\nincluded in the output *.ind.txt file in Eigenstrat\noutput. This will be ignored if the output format is\nnot Eigenstrat (default: 'Unknown')")), ToolInput(tag="in_sample_names", input_type=Int(optional=True), prefix="--sampleNames", doc=InputDocumentation(doc=",NAME2,...\ngive the names of the samples as comma-separated list\n(no spaces)")), ToolInput(tag="in_sample_name_file", input_type=File(optional=True), prefix="--sampleNameFile", doc=InputDocumentation(doc="give the names of the samples in a file with one name\nper line\n")), ToolInput(tag="in_data_dot", input_type=String(), position=0, doc=InputDocumentation(doc="This tool is part of sequenceTools version 1.4.0.5"))], outputs=[ToolOutput(tag="out_eigen_strat_out", output_type=File(optional=True), selector=InputSelector(input_to_select="in_eigen_strat_out", type_hint=File()), doc=OutputDocumentation(doc="Set Eigenstrat as output format. Specify the\nfilenames for the EigenStrat SNP and IND file\noutputs: <FILE_PREFIX>.snp.txt and\n<FILE_PREFIX>.ind.txt If not set, output will be\nFreqSum (Default). Note that freqSum format,\ndescribed at\nhttps://rarecoal-docs.readthedocs.io/en/latest/rarecoal-tools.html#vcf2freqsum,\nis useful for testing your pipeline, since it's\noutput to standard out")), ToolOutput(tag="out_sample_pop_name", output_type=File(optional=True), selector=InputSelector(input_to_select="in_sample_pop_name", type_hint=File()), doc=OutputDocumentation(doc="specify the population name of the samples, which is\nincluded in the output *.ind.txt file in Eigenstrat\noutput. This will be ignored if the output format is\nnot Eigenstrat (default: 'Unknown')"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Pileupcaller_V0_1_0().translate("wdl", allow_empty_container=True)

