from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, File, Array, Int, Directory, String

Grafimo_V0_1_0 = CommandToolBuilder(tool="grafimo", base_command=["grafimo"], inputs=[ToolInput(tag="in_cores", input_type=Boolean(optional=True), prefix="--cores", doc=InputDocumentation(doc="[NCORES]      Number of cores to use. The default value is 0 (auto-\ndetection). If you chose to query the whole genome\nvariation graph note that the default option is to use\nonly one core to avoid memory issues")), ToolInput(tag="in_path_linear_required", input_type=Boolean(optional=True), prefix="-l", doc=InputDocumentation(doc="[LINEAR-GENOME], --linear-genome [LINEAR-GENOME]\nPath to the linear genome (FASTA format required)")), ToolInput(tag="in_path_vcf_file", input_type=Boolean(optional=True), prefix="-v", doc=InputDocumentation(doc="[VCF], --vcf [VCF]\nPath to the VCF file. The VCF must be compressed (e.g.\nmyvcf.vcf.gz)")), ToolInput(tag="in_path_vg_format", input_type=Boolean(optional=True), prefix="-g", doc=InputDocumentation(doc="[GRAPH-GENOME], --graph-genome [GRAPH-GENOME]\nPath to the VG genome graph file (VG or XG format)")), ToolInput(tag="in_path_directory_containing", input_type=Boolean(optional=True), prefix="-d", doc=InputDocumentation(doc="[GRAPH-GENOMES-DIR], --graph-genome-dir [GRAPH-GENOMES-DIR]\nPath to a directory containing a variable number of\nVGs graph genomes (VG or XG format)")), ToolInput(tag="in_bed_file", input_type=File(optional=True), prefix="--bedfile", doc=InputDocumentation(doc="Path to the BED file containing the regions to scan")), ToolInput(tag="in_motif", input_type=Array(t=Int(), optional=True), prefix="--motif", doc=InputDocumentation(doc="Path to the motif file (JASPAR or MEME format\nrequired)")), ToolInput(tag="in_chromosomes_vg_built", input_type=Boolean(optional=True), prefix="-c", doc=InputDocumentation(doc="[1 [X ...]], --chroms [1 [X ...]]\nChromosomes for which the VG will be built or in which\nGRAFIMO will search the given motif. To consider all\nthe chromosomes, just skip this argument. This\nargument can be used during the building of VGs or\nduring their scan for the occurrences of the given\nmotif")), ToolInput(tag="in_path_background_file", input_type=Boolean(optional=True), prefix="-k", doc=InputDocumentation(doc="[BACKGROUND], --bgfile [BACKGROUND]\nPath to the background file [optional]")), ToolInput(tag="in_pseudocount_add_motif", input_type=Boolean(optional=True), prefix="-p", doc=InputDocumentation(doc="[PSEUDOCOUNT], --pseudo [PSEUDOCOUNT]\nPseudocount to add to motif counts [optional]")), ToolInput(tag="in_threshold_applied_ispossible", input_type=Boolean(optional=True), prefix="-t", doc=InputDocumentation(doc="[THRESHOLD], --threshold [THRESHOLD]\nThreshold that will be applied on the P-values (by\ndefault) of each found motif occurrence. It is\npossible to apply the threshold on the q-values using\nthe -q (--qvalueT) option. Default is 1e-4 [optional]")), ToolInput(tag="in_no_q_value", input_type=Boolean(optional=True), prefix="--no-qvalue", doc=InputDocumentation(doc="With this parameter the q-values will not be computed")), ToolInput(tag="in_no_reverse", input_type=Boolean(optional=True), prefix="--no-reverse", doc=InputDocumentation(doc="With this parameter GRAFIMO will scan only the forward")), ToolInput(tag="in_recomb", input_type=Boolean(optional=True), prefix="--recomb", doc=InputDocumentation(doc="While scanning the genome variation graph for the\ngiven motif occurrences will be considered all\npossible recombinants that can be obtained with the\ngenomic variants used to build the VG")), ToolInput(tag="in_re_index", input_type=Boolean(optional=True), prefix="--reindex", doc=InputDocumentation(doc="When building the genome variation graph the VCF will\nbe indexed with tabix, even if its TBI file it is\nalready available. This option can only be used with\nthe 'buildvg' pipeline")), ToolInput(tag="in_q_value_t", input_type=Boolean(optional=True), prefix="--qvalueT", doc=InputDocumentation(doc="The threshold will be applied on the q-values, rather\nthan the P-values")), ToolInput(tag="in_top_graphs", input_type=Directory(optional=True), prefix="--top-graphs", doc=InputDocumentation(doc="[GRAPHS_NUM]\nThe PNG image of the regions containing the top\nGRAPHS_NUM sequences (sorted by P-value) will be\nstored in the output directory")), ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="--verbose", doc=InputDocumentation(doc="Output a lot of additional information about the\nexecution\n")), ToolInput(tag="in_workflow", input_type=String(), position=0, doc=InputDocumentation(doc="This option accpets only two values: 'buildvg' or\n'findmotif'. This option is mandatory. When used\n'buildvg', GRAFIMO will compute the genome variation\ngraph for each chromosome or a user defined subset,\nfrom the given reference genome and VCF file. When\nused 'findmotif' GRAFIMO will scan a given VG or all\nthe VGs contained in a given directory, for the given\nmotif occurrences")), ToolInput(tag="in_strand", input_type=String(), position=1, doc=InputDocumentation(doc="-f, --text-only       Print the results in TSV directly to the standard")), ToolInput(tag="in_output", input_type=String(), position=2, doc=InputDocumentation(doc="-o [OUTDIR], --out [OUTDIR]"))], outputs=[ToolOutput(tag="out_top_graphs", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_top_graphs", type_hint=File()), doc=OutputDocumentation(doc="[GRAPHS_NUM]\nThe PNG image of the regions containing the top\nGRAPHS_NUM sequences (sorted by P-value) will be\nstored in the output directory"))], container="quay.io/biocontainers/grafimo:1.1.3--py38h0731a9c_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Grafimo_V0_1_0().translate("wdl")

