from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import String, File, Boolean, Int, Float

Pureclip_V0_1_0 = CommandToolBuilder(tool="pureclip", base_command=["pureclip"], inputs=[ToolInput(tag="in_in", input_type=String(optional=True), prefix="--in", doc=InputDocumentation(doc="Target bam files. Valid filetype is: .bam.")), ToolInput(tag="in_bai", input_type=String(optional=True), prefix="--bai", doc=InputDocumentation(doc="Target bam index files. Valid filetype is: .bai.")), ToolInput(tag="in_genome", input_type=File(optional=True), prefix="--genome", doc=InputDocumentation(doc="Genome reference file. Valid filetypes are: .fa, .fasta, .fa.gz, and\n.fasta.gz.")), ToolInput(tag="in_out", input_type=File(optional=True), prefix="--out", doc=InputDocumentation(doc="Output file to write crosslink sites. Valid filetype is: .bed.")), ToolInput(tag="in_or", input_type=File(optional=True), prefix="--or", doc=InputDocumentation(doc="Output file to write binding regions. Valid filetype is: .bed.")), ToolInput(tag="in_par", input_type=File(optional=True), prefix="--par", doc=InputDocumentation(doc="Output file to write learned parameters.")), ToolInput(tag="in_ctr", input_type=Boolean(optional=True), prefix="--ctr", doc=InputDocumentation(doc="Assign crosslink sites to read start positions. Note: depends on RT\nenzyme, buffer conditions and likely on protein. Default: assign\ncrosslink sites to positions upstream of read starts.")), ToolInput(tag="in_st", input_type=Int(optional=True), prefix="--st", doc=InputDocumentation(doc="Scoring scheme. Default: 0 -> score_UC (log posterior probability\nratio of most likely and second most likely state). In range [0..3].")), ToolInput(tag="in_inter", input_type=File(optional=True), prefix="--inter", doc=InputDocumentation(doc="Genomic chromosomes to learn HMM parameters, e.g. 'chr1;chr2;chr3'.\nContigs have to be in the same order as in BAM file. Useful to\nreduce runtime and memory consumption. Default: all contigs from\nreference file are used (useful when applying to transcript-wise\nalignments or poor data).")), ToolInput(tag="in_chr", input_type=File(optional=True), prefix="--chr", doc=InputDocumentation(doc="Contigs to apply HMM, e.g. 'chr1;chr2;chr3;'. Contigs have to be in\nthe same order as in BAM file.")), ToolInput(tag="in_bc", input_type=Int(optional=True), prefix="--bc", doc=InputDocumentation(doc="Flag to set parameters according to binding characteristics of\nprotein: see description in section below. In range [0..1].")), ToolInput(tag="in_bdw", input_type=Int(optional=True), prefix="--bdw", doc=InputDocumentation(doc="Bandwidth for kernel density estimation used to access enrichment.\nNOTE: Increasing the bandwidth increases runtime and memory\nconsumption. Default: 50. In range [1..500].")), ToolInput(tag="in_bd_wn", input_type=Int(optional=True), prefix="--bdwn", doc=InputDocumentation(doc="Bandwidth for kernel density estimation used to estimate n for\nbinomial distributions. For proteins that are rather sliding along\nthe RNA or showing long crosslink clusters this should be increased,\ne.g. to 100 (should be <= 4*bdw). Default: same as bdw. In range\n[1..500].")), ToolInput(tag="in_dm", input_type=Int(optional=True), prefix="--dm", doc=InputDocumentation(doc="Distance used to merge individual crosslink sites to binding\nregions. Default: 8")), ToolInput(tag="in_ld", input_type=Boolean(optional=True), prefix="--ld", doc=InputDocumentation(doc="Use higher precision to store emission probabilities, state poster\nposterior probabilities etc. (i.e. long double). Should not be\nnecessary anymore, due to computations in log-space. Note: increases\nmemory consumption. Default: double.")), ToolInput(tag="in_ts", input_type=Int(optional=True), prefix="--ts", doc=InputDocumentation(doc="Size of look-up table for log-sum-exp values. Default: 600000")), ToolInput(tag="in_tmv", input_type=Int(optional=True), prefix="--tmv", doc=InputDocumentation(doc="Minimum value in look-up table for log-sum-exp values. Default:\n-2000")), ToolInput(tag="in_ur", input_type=Int(optional=True), prefix="--ur", doc=InputDocumentation(doc="Flag to define which read should be selected for the analysis:\n1->R1, 2->R2. Note: PureCLIP uses read starts corresponding to 3'\ncDNA ends. Thus if providing paired-end data, only the corresponding\nread should be selected (e.g. eCLIP->R2, iCLIP->R1). If applicable,\nused for input BAM file as well. Default: uses read starts of all\nprovided reads assuming single-end or pre-filtered data. In range\n[1..2].")), ToolInput(tag="in_is", input_type=File(optional=True), prefix="--is", doc=InputDocumentation(doc="Covariates file: position-wise values, e.g. smoothed reads start\ncounts (KDEs) from input data. Valid filetype is: .bed.")), ToolInput(tag="in_i_bam", input_type=File(optional=True), prefix="--ibam", doc=InputDocumentation(doc="File containing mapped reads from control experiment, e.g. eCLIP\ninput. Valid filetype is: .bam.")), ToolInput(tag="in_i_bai", input_type=File(optional=True), prefix="--ibai", doc=InputDocumentation(doc="File containing BAM index corresponding to mapped reads from control\nexperiment Valid filetype is: .bai.")), ToolInput(tag="in_fis", input_type=File(optional=True), prefix="--fis", doc=InputDocumentation(doc="Fimo input motif score covariates file. Valid filetype is: .bed.")), ToolInput(tag="in_nim", input_type=Int(optional=True), prefix="--nim", doc=InputDocumentation(doc="Max. motif ID to use. Default: Only covariates with motif ID 1 are\nused.")), ToolInput(tag="in_up_e", input_type=Boolean(optional=True), prefix="--upe", doc=InputDocumentation(doc="Use (n dependent) pseudo emission probabilities for crosslink state.")), ToolInput(tag="in_mib_r", input_type=Int(optional=True), prefix="--mibr", doc=InputDocumentation(doc="Maximum number of iterations within BRENT algorithm. In range\n[1..1000].")), ToolInput(tag="in_mib_w", input_type=Int(optional=True), prefix="--mibw", doc=InputDocumentation(doc="Maximum number of iterations within Baum-Welch algorithm. In range\n[0..500].")), ToolInput(tag="in_gone_km_in", input_type=Int(optional=True), prefix="--g1kmin", doc=InputDocumentation(doc="Minimum shape k of 'non-enriched' gamma distribution (g1.k). In\nrange [1.5..inf].")), ToolInput(tag="in_gone_kmax", input_type=Int(optional=True), prefix="--g1kmax", doc=InputDocumentation(doc="Maximum shape k of 'non-enriched' gamma distribution (g1.k).")), ToolInput(tag="in_g_two_km_in", input_type=Int(optional=True), prefix="--g2kmin", doc=InputDocumentation(doc="Minimum shape k of 'enriched' gamma distribution (g2.k).")), ToolInput(tag="in_g_two_kmax", input_type=Int(optional=True), prefix="--g2kmax", doc=InputDocumentation(doc="Maximum shape k of 'enriched' gamma distribution (g2.k).")), ToolInput(tag="in_fk", input_type=Boolean(optional=True), prefix="--fk", doc=InputDocumentation(doc="When incorporating input signal, do not constrain 'non-enriched'\nshape parameter k <= 'enriched' gamma parameter k.")), ToolInput(tag="in_mkn", input_type=Int(optional=True), prefix="--mkn", doc=InputDocumentation(doc="Max. k/N ratio (read start sites/N) used to learn truncation\nprobabilities for 'non-crosslink' and 'crosslink' emission\nprobabilities (high ratios might originate from mapping artifacts\nthat can disturb parameter learning). Default: 1.0 In range\n[0.5..1.5].")), ToolInput(tag="in_bone_p", input_type=Int(optional=True), prefix="--b1p", doc=InputDocumentation(doc="Initial value for binomial probability parameter of 'non-crosslink'\nstate. Default: 0.01.")), ToolInput(tag="in_btw_op", input_type=Int(optional=True), prefix="--b2p", doc=InputDocumentation(doc="Initial value for binomial probability parameter of 'crosslink'\nstate. Default: 0.15.")), ToolInput(tag="in_mtp", input_type=Int(optional=True), prefix="--mtp", doc=InputDocumentation(doc="Min. transition probability from state '2' to '3' (helpful for poor\ndata, where no clear distinction between 'enriched' and\n'non-enriched' is possible). Default: 0.0001.")), ToolInput(tag="in_m_kde", input_type=Int(optional=True), prefix="--mkde", doc=InputDocumentation(doc="Minimum KDE value used for fitting left-truncated gamma\ndistributions. Default: corresponding to singleton read start.")), ToolInput(tag="in_ntp", input_type=Int(optional=True), prefix="--ntp", doc=InputDocumentation(doc="Only sites with n >= ntp are used to learn binomial probability\nparameters (bin1.p, bin2.p). Default: 10")), ToolInput(tag="in_ntp_two", input_type=Int(optional=True), prefix="--ntp2", doc=InputDocumentation(doc="Only sites with n >= ntp2 are used to learn probability of\ntransition from state '2' to '2' or '3'. Useful for data with low\ntruncation rate at crosslink sites or in general high fraction of\nnon-coinciding read starts. Default: 0")), ToolInput(tag="in_a_ntp", input_type=Boolean(optional=True), prefix="--antp", doc=InputDocumentation(doc="Automatically choose n threshold (-ntp, -ntp2) to estimate\nparameters linked to crosslink states based on expected read start\ncount at crosslink sites.")), ToolInput(tag="in_pat", input_type=Int(optional=True), prefix="--pat", doc=InputDocumentation(doc="Length threshold for internal poly-X stretches to get excluded.")), ToolInput(tag="in_epal", input_type=Boolean(optional=True), prefix="--epal", doc=InputDocumentation(doc="Exclude intervals containing poly-A stretches from learning.")), ToolInput(tag="in_epa_a", input_type=Boolean(optional=True), prefix="--epaa", doc=InputDocumentation(doc="Exclude intervals containing poly-A stretches from analysis.")), ToolInput(tag="in_e_ptl", input_type=Boolean(optional=True), prefix="--eptl", doc=InputDocumentation(doc="Exclude intervals containing poly-U stretches from learning.")), ToolInput(tag="in_epta", input_type=Boolean(optional=True), prefix="--epta", doc=InputDocumentation(doc="Exclude intervals containing poly-U stretches from analysis.")), ToolInput(tag="in_m_rtf", input_type=Int(optional=True), prefix="--mrtf", doc=InputDocumentation(doc="Fit gamma shape k only for positions with min. covariate value.")), ToolInput(tag="in_mtc", input_type=Int(optional=True), prefix="--mtc", doc=InputDocumentation(doc="Maximum number of read starts at one position used for learning. For\nsites with counts above threshold the whole covered regions will be\nignored for learning! Default: 500. In range [50..50000].")), ToolInput(tag="in_mtc_two", input_type=Int(optional=True), prefix="--mtc2", doc=InputDocumentation(doc="Maximum number of read starts at one position stored. For sites with\ncounts above threshold the count will be truncated. Influences k and\nn. Default: 65000. In range [5000..65000].")), ToolInput(tag="in_pet", input_type=Int(optional=True), prefix="--pet", doc=InputDocumentation(doc="Prior enrichment threshold: a KDE threshold corresponding to 7 read\nstart counts at one position will be used for initial classification\nof 'non-enriched' and 'enriched' site. Default: 7 In range [2..50].")), ToolInput(tag="in_nt", input_type=Int(optional=True), prefix="--nt", doc=InputDocumentation(doc="Number of threads used for learning.")), ToolInput(tag="in_nta", input_type=Int(optional=True), prefix="--nta", doc=InputDocumentation(doc="Number of threads used for applying learned parameters. Increases\nmemory usage, if greater than number of chromosomes used for\nlearning, since HMM will be build for multiple chromosomes in\nparallel. Default: min(nt, no. of chromosomes/transcripts used for\nlearning).")), ToolInput(tag="in_oa", input_type=Boolean(optional=True), prefix="--oa", doc=InputDocumentation(doc="Outputs all sites with at least one read start in extended output\nformat.")), ToolInput(tag="in_quiet", input_type=Boolean(optional=True), prefix="--quiet", doc=InputDocumentation(doc="Set verbosity to a minimum.")), ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="--verbose", doc=InputDocumentation(doc="Enable verbose output.")), ToolInput(tag="in_very_verbose", input_type=Boolean(optional=True), prefix="--very-verbose", doc=InputDocumentation(doc="Enable very verbose output.")), ToolInput(tag="in_zero_dot_one_dot", input_type=Float(), position=0, doc=InputDocumentation(doc="In case of different binding characteristics adjust parameters -bdw,")), ToolInput(tag="in_clusters", input_type=String(), position=0, doc=InputDocumentation(doc="pureclip -i target.bam -bai target.bai -g ref.fasta -o called_crosslinksites.bed -nt 10  -iv '1;2;3;' -bc 1 -b2p 0.03 "))], outputs=[ToolOutput(tag="out_out", output_type=File(optional=True), selector=InputSelector(input_to_select="in_out", type_hint=File()), doc=OutputDocumentation(doc="Output file to write crosslink sites. Valid filetype is: .bed.")), ToolOutput(tag="out_or", output_type=File(optional=True), selector=InputSelector(input_to_select="in_or", type_hint=File()), doc=OutputDocumentation(doc="Output file to write binding regions. Valid filetype is: .bed.")), ToolOutput(tag="out_par", output_type=File(optional=True), selector=InputSelector(input_to_select="in_par", type_hint=File()), doc=OutputDocumentation(doc="Output file to write learned parameters."))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Pureclip_V0_1_0().translate("wdl", allow_empty_container=True)

