class: CommandLineTool
id: ../../../PhiSpy.py.cwl
inputs:
- id: in_output_dir
  doc: The output directory to write the results
  type: Directory
  inputBinding:
    prefix: --output_dir
- id: in_make_training_data
  doc: "Create training data from a set of annotated genome\nfiles. Requires is_phage=1\
    \ qualifier in prophage's\nCDSs"
  type: long
  inputBinding:
    prefix: --make_training_data
- id: in_training_set
  doc: "Choose the most closely related set to your genome.\n[Default data/trainSet_genericAll.txt]"
  type: string
  inputBinding:
    prefix: --training_set
- id: in_list
  doc: List the available training sets and exit
  type: boolean
  inputBinding:
    prefix: --list
- id: in_file_prefix
  doc: "An optional prefix to prepend to all of the output\nfiles"
  type: File
  inputBinding:
    prefix: --file_prefix
- id: in_run_evaluation_mode
  doc: "[EVALUATE], --evaluate [EVALUATE]\nRun in evaluation mode -- does not generate\
    \ new data,\nbut reruns the evaluation"
  type: boolean
  inputBinding:
    prefix: -e
- id: in_number
  doc: "Number of consecutive genes in a region of window size\nthat must be prophage\
    \ genes to be called. [Default: 5]"
  type: long
  inputBinding:
    prefix: --number
- id: in_min_contig_size
  doc: "Minimum contig size (in bp) to be included in the\nanalysis. Smaller contigs\
    \ will be dropped. [Default:\n30]"
  type: long
  inputBinding:
    prefix: --min_contig_size
- id: in_window_size
  doc: "Window size of consecutive genes to look through to\nfind phages. [Default:\
    \ 30]"
  type: long
  inputBinding:
    prefix: --window_size
- id: in_non_prophage_gene_gaps
  doc: "The number of non phage genes betweeen prophages.\n[Default: 10]"
  type: long
  inputBinding:
    prefix: --nonprophage_genegaps
- id: in_phage_genes
  doc: "The minimum number of genes that must be identified as\nbelonging to a phage\
    \ for the region to be included.\nThe default is 2 or more genes."
  type: long
  inputBinding:
    prefix: --phage_genes
- id: in_metrics
  doc: "The set of metrics to consider during classification.\nIf not set, all metrics\
    \ (orf_length_med,\nshannon_slope, at_skew, gc_skew, max_direction) will\nbe considered."
  type: string[]
  inputBinding:
    prefix: --metrics
- id: in_random_forest_trees
  doc: "Number of trees generated by Random Forest classifier.\n[Default: 500]"
  type: long
  inputBinding:
    prefix: --randomforest_trees
- id: in_expand_slope
  doc: "Use the product of the slope of the Shannon scores in\nmaking test sets"
  type: boolean
  inputBinding:
    prefix: --expand_slope
- id: in_km_ers_type
  doc: "Type of kmers used for calculating Shannon scores.\n[Default: all]"
  type: string
  inputBinding:
    prefix: --kmers_type
- id: in_ph_mms
  doc: "Phage HMM profile database (like pVOGs) will be mapped\nagainst the genome\
    \ of interest and used as additional\nfeature to identify prophages. Note that\
    \ this is\nexperimental at the moment"
  type: string
  inputBinding:
    prefix: --phmms
- id: in_color
  doc: "If set, within the output GenBank file CDSs with phmms\nhits will be colored\
    \ (for viewing in Artemis)."
  type: File
  inputBinding:
    prefix: --color
- id: in_threads
  doc: "Number of threads to use while searching with phmms\nand the random forest.\
    \ [default: 2]"
  type: long
  inputBinding:
    prefix: --threads
- id: in_output_choice
  doc: "Sum of codes for files to output. For more details see\nthe README.md file\
    \ at\nhttps://github.com/linsalrob/PhiSpy#choosing-which-\noutput-files-are-created.\
    \ [default: 3]"
  type: File
  inputBinding:
    prefix: --output_choice
- id: in_include_all_repeats
  doc: include all repeats in the genbank output
  type: boolean
  inputBinding:
    prefix: --include_all_repeats
- id: in_extra_dna
  doc: "additional DNA flanking the predicted prophage to test\nfor repeats [Default\
    \ 2000]"
  type: long
  inputBinding:
    prefix: --extra_dna
- id: in_min_repeat_len
  doc: 'minimum repeat length to search for [Default: 10]'
  type: long
  inputBinding:
    prefix: --min_repeat_len
- id: in_log
  doc: "Name of the log file to write details to [Default:\nphispy.log]"
  type: File
  inputBinding:
    prefix: --log
- id: in_quiet
  doc: Run in quiet mode
  type: boolean
  inputBinding:
    prefix: --quiet
- id: in_delete_temp_files
  doc: "[KEEP], --keep [KEEP]\nDo not delete temp files"
  type: boolean
  inputBinding:
    prefix: -k
outputs:
- id: out_stdout
  doc: Standard output stream
  type: stdout
- id: out_output_dir
  doc: The output directory to write the results
  type: Directory
  outputBinding:
    glob: $(inputs.in_output_dir)
- id: out_color
  doc: "If set, within the output GenBank file CDSs with phmms\nhits will be colored\
    \ (for viewing in Artemis)."
  type: File
  outputBinding:
    glob: $(inputs.in_color)
- id: out_output_choice
  doc: "Sum of codes for files to output. For more details see\nthe README.md file\
    \ at\nhttps://github.com/linsalrob/PhiSpy#choosing-which-\noutput-files-are-created.\
    \ [default: 3]"
  type: File
  outputBinding:
    glob: $(inputs.in_output_choice)
cwlVersion: v1.1
baseCommand:
- PhiSpy.py
