from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Boolean, String, Int, Float

Anvi_Pan_Genome_V0_1_0 = CommandToolBuilder(tool="anvi_pan_genome", base_command=["anvi-pan-genome"], inputs=[ToolInput(tag="in_genomes_storage", input_type=File(optional=True), prefix="--genomes-storage", doc=InputDocumentation(doc="Anvi'o genomes storage file")), ToolInput(tag="in_genome_names", input_type=File(optional=True), prefix="--genome-names", doc=InputDocumentation(doc="Genome names to 'focus'. You can use this parameter to\nlimit the genomes included in your analysis. You can\nprovide these names as a comma-separated list of\nnames, or you can put them in a file, where you have a\nsingle genome name in each line, and provide the file\npath.")), ToolInput(tag="in_skip_alignments", input_type=Boolean(optional=True), prefix="--skip-alignments", doc=InputDocumentation(doc="By default, anvi'o attempts to align amino acid\nsequences in each gene cluster using multiple sequnce\nalignment via muscle. You can use this flag to skip\nthat step and be upset later.")), ToolInput(tag="in_skip_homogeneity", input_type=Boolean(optional=True), prefix="--skip-homogeneity", doc=InputDocumentation(doc="By default, anvi'o attempts to calculate homogeneity\nvalues for every gene cluster, given that they are\naligned. You can use this flag to have anvi'o skip\nhomogeneity calculations. Anvi'o will ignore this flag\nif you decide to skip alignments")), ToolInput(tag="in_quick_homogeneity", input_type=Boolean(optional=True), prefix="--quick-homogeneity", doc=InputDocumentation(doc="By default, anvi'o will use a homogeneity algorithm\nthat checks for horizontal and vertical geometric\nhomogeneity (along with functional). With this flag,\nyou can tell anvi'o to skip horizontal geometric\nhomogeneity calculations. It will be less accurate but\nquicker. Anvi'o will ignore this flag if you skip\nhomogeneity calculations or alignments all together.")), ToolInput(tag="in_align_with", input_type=String(optional=True), prefix="--align-with", doc=InputDocumentation(doc="The multiple sequence alignment program to use when\nmultiple sequence alignment is necessary. To see all\navailable options, use the flag `--list-aligners`.")), ToolInput(tag="in_exclude_partial_gene_calls", input_type=Boolean(optional=True), prefix="--exclude-partial-gene-calls", doc=InputDocumentation(doc="By default, anvi'o includes all partial gene calls\nfrom the analysis, which, in some cases, may inflate\nthe number of gene clusters identified and introduce\nextra heterogeneity within those gene clusters. Using\nthis flag, you can request anvi'o to exclude partial\ngene calls from the analysis (whether a gene call is\npartial or not is an information that comes directly\nfrom the gene caller used to identify genes during the\ngeneration of the contigs database).")), ToolInput(tag="in_use_ncbi_blast", input_type=Boolean(optional=True), prefix="--use-ncbi-blast", doc=InputDocumentation(doc="This program uses DIAMOND by default, however, if you\nlike, you can use good ol' blastp from NCBI instead.")), ToolInput(tag="in_min_bit", input_type=Int(optional=True), prefix="--minbit", doc=InputDocumentation(doc="The minimum minbit value. The minbit heuristic\nprovides a mean to set a to eliminate weak matches\nbetween two amino acid sequences. We learned it from\nITEP (Benedict MN et al, doi:10.1186/1471-2164-15-8),\nwhich is a comprehensive analysis workflow for\npangenomes, and decided to use it in the anvi'o\npangenomic workflow, as well. Briefly, If you have two\namino acid sequences, 'A' and 'B', the minbit is\ndefined as 'BITSCORE(A, B) / MIN(BITSCORE(A, A),\nBITSCORE(B, B))'. So the minbit score between two\nsequences goes to 1 if they are very similar over the\nentire length of the 'shorter' amino acid sequence,\nand goes to 0 if (1) they match over a very short\nstretch compared even to the length of the shorter\namino acid sequence or (2) the match betwen sequence\nidentity is low. The default is 0.5.")), ToolInput(tag="in_mcl_inflation", input_type=Float(optional=True), prefix="--mcl-inflation", doc=InputDocumentation(doc="MCL inflation parameter, that defines the sensitivity\nof the algorithm during the identification of the gene\nclusters. More information on this parameter and it's\neffect on cluster granularity is here:\n(http://micans.org/mcl/man/mclfaq.html#faq7.2). The\ndefault is 2.")), ToolInput(tag="in_min_occurrence", input_type=Int(optional=True), prefix="--min-occurrence", doc=InputDocumentation(doc="Do you not want singletons?\ You don't? Well, this\nparameter will help you get rid of them (along with\ndoubletons, if you want). Anvi'o will remove gene\nclusters that occur less than the number you set using\nthis parameter from the analysis. The default is 1,\nwhich means everything will be kept. If you want to\nremove singletons, set it to 2, if you want to remove\ndoubletons as well, set it to 3, and so on.")), ToolInput(tag="in_min_percent_identity", input_type=Int(optional=True), prefix="--min-percent-identity", doc=InputDocumentation(doc="Minimum percent identity between the two amino acid\nsequences for them to have an edge for MCL analysis.\nThis value will be used to filter hits from Diamond\nsearch results. Because percent identity is not a\npredictor of a good match (since it does not\ncommunicate many other important factors such as the\nalignment length between the two sequences and its\nproportion to the entire length of those involved), we\nsuggest you rely on 'minbit' parameter. But you know\nwhat? Maybe you shouldn't listen to anyone, and\nexperiment on your own! The default is 0 percent.")), ToolInput(tag="in_sensitive", input_type=Boolean(optional=True), prefix="--sensitive", doc=InputDocumentation(doc="DIAMOND sensitivity. With this flag you can instruct\nDIAMOND to be 'sensitive', rather than 'fast' during\nthe search. It is likely the search will take\nremarkably longer. But, hey, if you are doing it for\nyour final analysis, maybe it should take longer and\nbe more accurate. This flag is only relevant if you\nare running DIAMOND.")), ToolInput(tag="in_project_name", input_type=File(optional=True), prefix="--project-name", doc=InputDocumentation(doc="Name of the project. Please choose a short but\ndescriptive name (so anvi'o can use it whenever she\nneeds to name an output file, or add a new table in a\ndatabase, or name her first born).")), ToolInput(tag="in_description", input_type=File(optional=True), prefix="--description", doc=InputDocumentation(doc="A plain text file that contains some description about\nthe project. You can use Markdwon syntax. The\ndescription text will be rendered and shown in all\nrelevant interfaces, including the anvi'o interactive\ninterface, or anvi'o summary outputs.")), ToolInput(tag="in_output_dir", input_type=File(optional=True), prefix="--output-dir", doc=InputDocumentation(doc="Directory path for output files")), ToolInput(tag="in_overwrite_output_destinations", input_type=Boolean(optional=True), prefix="--overwrite-output-destinations", doc=InputDocumentation(doc="Overwrite if the output files and/or directories\nexist.")), ToolInput(tag="in_num_threads", input_type=Int(optional=True), prefix="--num-threads", doc=InputDocumentation(doc="Maximum number of threads to use for multithreading\nwhenever possible. Very conservatively, the default is\n1. It is a good idea to not exceed the number of CPUs\n/ cores on your system. Plus, please be careful with\nthis option if you are running your commands on a SGE\n--if you are clusterizing your runs, and asking for\nmultiple threads to use, you may deplete your\nresources very fast.")), ToolInput(tag="in_enforce_hierarchical_clustering", input_type=Boolean(optional=True), prefix="--enforce-hierarchical-clustering", doc=InputDocumentation(doc="If you want anvi'o to try to generate a hierarchical\nclustering of your gene clusters even if the number of\ngene clusters exceeds its suggested limit for\nhierarchical clustering, you can use this flag to\nenforce it. Are you are a rebel of some sorts? Or did\ncomputers made you upset? Express your anger towards\nmachine using this flag.")), ToolInput(tag="in_distance", input_type=String(optional=True), prefix="--distance", doc=InputDocumentation(doc="The distance metric for the clustering of gene\nclusters. If you do not use this flag, the default\ndistance metric will be used for each clustering\nconfiguration which is 'euclidean'.")), ToolInput(tag="in_linkage", input_type=String(optional=True), prefix="--linkage", doc=InputDocumentation(doc="The same story with the `--distance`, except, the\nsystem default for this one is ward.\n")), ToolInput(tag="in_clusters_dot", input_type=String(), position=0, doc=InputDocumentation(doc="--skip-hierarchical-clustering\nAnvi'o attempts to generate a hierarchical clustering\nof your gene clusters once it identifies them so you\ncan use `anvi-display-pan` to play with it. But if you\nwant to skip this step, this is your flag."))], outputs=[ToolOutput(tag="out_project_name", output_type=File(optional=True), selector=InputSelector(input_to_select="in_project_name", type_hint=File()), doc=OutputDocumentation(doc="Name of the project. Please choose a short but\ndescriptive name (so anvi'o can use it whenever she\nneeds to name an output file, or add a new table in a\ndatabase, or name her first born).")), ToolOutput(tag="out_output_dir", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output_dir", type_hint=File()), doc=OutputDocumentation(doc="Directory path for output files"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Anvi_Pan_Genome_V0_1_0().translate("wdl", allow_empty_container=True)

