from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Int, String, Array, Directory, Float, File

Qcumber_2_V0_1_0 = CommandToolBuilder(tool="QCumber_2", base_command=["QCumber-2"], inputs=[ToolInput(tag="in_threads", input_type=Int(optional=True), prefix="--threads", doc=InputDocumentation(doc="Number of threads. Default: 4")), ToolInput(tag="in_config", input_type=String(optional=True), prefix="--config", doc=InputDocumentation(doc="Configfile to run pipeline. Additional parameters in\nthe commandline will override arguments in configfile.")), ToolInput(tag="in_input", input_type=Array(t=String(), optional=True), prefix="--input", doc=InputDocumentation(doc="input sample folder. Illumina filenames should end\nwith _<lane>_<R1|R2>_number, e.g.\nSample_12_345_R1_001.fastq, to find the right paired\nset.")), ToolInput(tag="in_read_one", input_type=Int(optional=True), prefix="--read1", doc=InputDocumentation(doc="Read 1 file")), ToolInput(tag="in_read_two", input_type=Int(optional=True), prefix="--read2", doc=InputDocumentation(doc="Read 2 file")), ToolInput(tag="in_technology", input_type=String(optional=True), prefix="--technology", doc=InputDocumentation(doc="If not set, automatically determine technology and\nsearch for fastq and bam files. Set technology to\nIonTorrent if all files are bam-files, else set\ntechnology to Illumina.")), ToolInput(tag="in_adapter", input_type=String(optional=True), prefix="--adapter", doc=InputDocumentation(doc="Adapter name for trimming. Default: all")), ToolInput(tag="in_reference", input_type=String(optional=True), prefix="--reference", doc=InputDocumentation(doc="Map reads against reference. Reference needs to be in\nfasta-format.")), ToolInput(tag="in_index", input_type=Int(optional=True), prefix="--index", doc=InputDocumentation(doc="Bowtie2 index if available.")), ToolInput(tag="in_kraken_db", input_type=String(optional=True), prefix="--kraken_db", doc=InputDocumentation(doc="Custom Kraken database. Default value is taken from\nenvironment variable KRAKEN_DB_PATH. Default: .")), ToolInput(tag="in_sav", input_type=Directory(optional=True), prefix="--sav", doc=InputDocumentation(doc="Illumina folder for SAV. Requires RunInfo.xml,\nRunParameter.xml and Interop folder.")), ToolInput(tag="in_trim_better", input_type=String(optional=True), prefix="--trimBetter", doc=InputDocumentation(doc="Optimize trimming parameter using 'Per sequence base\ncontent' from fastqc. Not recommended for amplicons.")), ToolInput(tag="in_illumina_clip", input_type=Int(optional=True), prefix="--illuminaclip", doc=InputDocumentation(doc="Illuminaclip option: <leading quality>:<trailing\nquality>:<sliding window>. Default: 2:30:10")), ToolInput(tag="in_min_len", input_type=Int(optional=True), prefix="--minlen", doc=InputDocumentation(doc="Minlen parameter for Trimmomatic. Drops read short\nthan minlen. Default: 50")), ToolInput(tag="in_trim_option", input_type=Int(optional=True), prefix="--trimOption", doc=InputDocumentation(doc="Additional Trimmomatic input. Default (if trimBetter\nis not set): SLIDINGWINDOW:4:20")), ToolInput(tag="in_trim_better_threshold", input_type=Float(optional=True), prefix="--trimBetter_threshold", doc=InputDocumentation(doc="Set -trimBetter to use this option.Default setting for\nIllumina: 0.15 and for IonTorrent: 0.25.")), ToolInput(tag="in_output", input_type=File(optional=True), prefix="--output", doc=InputDocumentation(doc="TSV File with two columns: <old sample name> <new\nsample name>"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Qcumber_2_V0_1_0().translate("wdl", allow_empty_container=True)

