from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Array, String, File, Int, Boolean

Analyze_Probe_Coverage_Py_V0_1_0 = CommandToolBuilder(tool="analyze_probe_coverage.py", base_command=["analyze_probe_coverage.py"], inputs=[ToolInput(tag="in_dataset", input_type=Array(t=String(), optional=True), prefix="--dataset", doc=InputDocumentation(doc="Labels for one or more target datasets; each can be a\nFASTA file, taxonomy ID to download, or dataset. The\nformat is as specified for --dataset in design.py.")), ToolInput(tag="in_probes_fast_a", input_type=File(optional=True), prefix="--probes-fasta", doc=InputDocumentation(doc="Path to a FASTA file that provides the probes (one per\nsequence) whose coverage should be analyzed against\nthe genomes in the given datasets")), ToolInput(tag="in_mismatches", input_type=Int(optional=True), prefix="--mismatches", doc=InputDocumentation(doc="Allow for this number of mismatches when determining\nwhether a probe covers a sequence")), ToolInput(tag="in_lcf_th_res", input_type=Int(optional=True), prefix="--lcf-thres", doc=InputDocumentation(doc="Say that a portion of a probe covers a portion of a\nsequence if the two share a substring with at most\nMISMATCHES mismatches that has length >= LCF_THRES bp")), ToolInput(tag="in_island_of_exact_match", input_type=Int(optional=True), prefix="--island-of-exact-match", doc=InputDocumentation(doc="(Optional) When determining whether a probe covers a\nsequence, require that there be an exact match (i.e.,\nno mismatches) of length at least\nISLAND_OF_EXACT_MATCH bp between a portion of the\nprobe and a portion of the sequence")), ToolInput(tag="in_cover_extension", input_type=Int(optional=True), prefix="--cover-extension", doc=InputDocumentation(doc="Extend the coverage of each side of a probe by this\nnumber of bp. That is, a probe covers a region that\nconsists of the portion of a sequence it hybridizes\nto, as well as this number of bp on each side of that\nportion. This is useful in modeling hybrid selection,\nwhere a probe hybridizes toa fragment that includes\nthe region targeted by the probe, along with\nsurrounding portions of the sequence. Increasing its\nvalue should reduce the number of probes required to\nachieve the desired coverage.")), ToolInput(tag="in_limit_target_genomes", input_type=String(optional=True), prefix="--limit-target-genomes", doc=InputDocumentation(doc="(Optional) Use only the first N target genomes in the\ndataset")), ToolInput(tag="in_print_analysis", input_type=Boolean(optional=True), prefix="--print-analysis", doc=InputDocumentation(doc="Print analysis of the probe set's coverage")), ToolInput(tag="in_write_analysis_to_tsv", input_type=File(optional=True), prefix="--write-analysis-to-tsv", doc=InputDocumentation(doc="The file to which to write a TSV-formatted matrix of\nthe probe set's coverage analysis")), ToolInput(tag="in_write_sliding_window_coverage", input_type=File(optional=True), prefix="--write-sliding-window-coverage", doc=InputDocumentation(doc="The file to which to write the average coverage\nachieved by the probe set within sliding windows of\neach target genome")), ToolInput(tag="in_write_probe_map_counts_to_tsv", input_type=Int(optional=True), prefix="--write-probe-map-counts-to-tsv", doc=InputDocumentation(doc="The file to which to write a TSV-formatted list of the\nnumber of sequences each probe maps to. This\nexplicitly does not count reverse complements.")), ToolInput(tag="in_max_num_processes", input_type=Int(optional=True), prefix="--max-num-processes", doc=InputDocumentation(doc="(Optional) An int >= 1 that gives the maximum number\nof processes to use in multiprocessing pools; uses\nmin(number of CPUs in the system, MAX_NUM_PROCESSES)\nprocesses")), ToolInput(tag="in_km_er_probe_map_k", input_type=Int(optional=True), prefix="--kmer-probe-map-k", doc=InputDocumentation(doc="(Optional) Use this value (KMER_PROBE_LENGTH_K) as the\nk-mer length when constructing a map of k-mers to the\nprobes that contain these k-mers. This map is used\nwhen mapping the given probes to target sequences and\nthe k-mers serve as seeds for calculating whether a\nprobe 'covers' a subsequence. The value should be\nsufficiently less than the probe length (PROBE_LENGTH)\nso that it can find mappings even when the candidate\nprobe and target sequence are divergent. In\nparticular, CATCH will try to find a value k >=\nKMER_PROBE_LENGTH_K (by default, >=10) such that k\ndivides PROBE_LENGTH and k < PROBE_LENGTH / MISMATCHES\n(if MISMATCHES=0, then k=PROBE_LENGTH). It will then\nuse this k as the k-mer length in mappings; if no such\nk exists, it will use a randomized approach with\nKMER_PROBE_LENGTH_K as the k-mer length.")), ToolInput(tag="in_debug", input_type=Boolean(optional=True), prefix="--debug", doc=InputDocumentation(doc="Debug output")), ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="--verbose", doc=InputDocumentation(doc="Verbose output"))], outputs=[], container="quay.io/biocontainers/catch:1.4.0--py_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Analyze_Probe_Coverage_Py_V0_1_0().translate("wdl")

