from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, Int, File, Float, String

Vsearch_V0_1_0 = CommandToolBuilder(tool="vsearch", base_command=["vsearch"], inputs=[ToolInput(tag="in_bzip_two_decompress", input_type=Boolean(optional=True), prefix="--bzip2_decompress", doc=InputDocumentation(doc="decompress input with bzip2 (required if pipe)")), ToolInput(tag="in_fast_a_width", input_type=Int(optional=True), prefix="--fasta_width", doc=InputDocumentation(doc="width of FASTA seq lines, 0 for no wrap (80)")), ToolInput(tag="in_gzip_decompress", input_type=Boolean(optional=True), prefix="--gzip_decompress", doc=InputDocumentation(doc="decompress input with gzip (required if pipe)")), ToolInput(tag="in_log", input_type=File(optional=True), prefix="--log", doc=InputDocumentation(doc="write messages, timing and memory info to file")), ToolInput(tag="in_max_seq_length", input_type=Int(optional=True), prefix="--maxseqlength", doc=InputDocumentation(doc="maximum sequence length (50000)")), ToolInput(tag="in_min_seq_length", input_type=Int(optional=True), prefix="--minseqlength", doc=InputDocumentation(doc="min seq length (clust/derep/search: 32, other:1)")), ToolInput(tag="in_no_progress", input_type=Boolean(optional=True), prefix="--no_progress", doc=InputDocumentation(doc="do not show progress indicator")), ToolInput(tag="in_not_run_c_labels", input_type=Boolean(optional=True), prefix="--notrunclabels", doc=InputDocumentation(doc="do not truncate labels at first space")), ToolInput(tag="in_quiet", input_type=Boolean(optional=True), prefix="--quiet", doc=InputDocumentation(doc="output just warnings and fatal errors to stderr")), ToolInput(tag="in_threads", input_type=Int(optional=True), prefix="--threads", doc=InputDocumentation(doc="number of threads to use, zero for all cores (0)")), ToolInput(tag="in_u_chime_de_novo", input_type=File(optional=True), prefix="--uchime_denovo", doc=InputDocumentation(doc="detect chimeras de novo")), ToolInput(tag="in_u_chime_two_de_novo", input_type=File(optional=True), prefix="--uchime2_denovo", doc=InputDocumentation(doc="detect chimeras de novo in denoised amplicons")), ToolInput(tag="in_u_chime_three_de_novo", input_type=File(optional=True), prefix="--uchime3_denovo", doc=InputDocumentation(doc="detect chimeras de novo in denoised amplicons")), ToolInput(tag="in_u_chime_ref", input_type=File(optional=True), prefix="--uchime_ref", doc=InputDocumentation(doc="detect chimeras using a reference database")), ToolInput(tag="in_dn", input_type=Float(optional=True), prefix="--dn", doc=InputDocumentation(doc="'no' vote pseudo-count (1.4)")), ToolInput(tag="in_min_diffs", input_type=Int(optional=True), prefix="--mindiffs", doc=InputDocumentation(doc="minimum number of differences in segment (3) *")), ToolInput(tag="in_mind_iv", input_type=Float(optional=True), prefix="--mindiv", doc=InputDocumentation(doc="minimum divergence from closest parent (0.8) *")), ToolInput(tag="in_minh", input_type=Float(optional=True), prefix="--minh", doc=InputDocumentation(doc="minimum score (0.28) * ignored in uchime2/3")), ToolInput(tag="in_size_in", input_type=Boolean(optional=True), prefix="--sizein", doc=InputDocumentation(doc="propagate abundance annotation from input")), ToolInput(tag="in_self", input_type=Boolean(optional=True), prefix="--self", doc=InputDocumentation(doc="exclude identical labels for --uchime_ref")), ToolInput(tag="in_self_id", input_type=Boolean(optional=True), prefix="--selfid", doc=InputDocumentation(doc="exclude identical sequences for --uchime_ref")), ToolInput(tag="in_xn", input_type=Float(optional=True), prefix="--xn", doc=InputDocumentation(doc="'no' vote weight (8.0)")), ToolInput(tag="in_borderline", input_type=File(optional=True), prefix="--borderline", doc=InputDocumentation(doc="output borderline chimeric sequences to file")), ToolInput(tag="in_chimeras", input_type=File(optional=True), prefix="--chimeras", doc=InputDocumentation(doc="output chimeric sequences to file")), ToolInput(tag="in_fast_a_score", input_type=Boolean(optional=True), prefix="--fasta_score", doc=InputDocumentation(doc="include chimera score in fasta output")), ToolInput(tag="in_non_chimeras", input_type=File(optional=True), prefix="--nonchimeras", doc=InputDocumentation(doc="output non-chimeric sequences to file")), ToolInput(tag="in_relabel", input_type=String(optional=True), prefix="--relabel", doc=InputDocumentation(doc="relabel nonchimeras with this prefix string")), ToolInput(tag="in_relabel_keep", input_type=Boolean(optional=True), prefix="--relabel_keep", doc=InputDocumentation(doc="keep the old label after the new when relabelling")), ToolInput(tag="in_relabel_md_five", input_type=Boolean(optional=True), prefix="--relabel_md5", doc=InputDocumentation(doc="relabel with md5 digest of normalized sequence")), ToolInput(tag="in_relabel_self", input_type=Boolean(optional=True), prefix="--relabel_self", doc=InputDocumentation(doc="relabel with the sequence itself as label")), ToolInput(tag="in_relabel_sha_one", input_type=Boolean(optional=True), prefix="--relabel_sha1", doc=InputDocumentation(doc="relabel with sha1 digest of normalized sequence")), ToolInput(tag="in_size_out", input_type=Boolean(optional=True), prefix="--sizeout", doc=InputDocumentation(doc="include abundance information when relabelling")), ToolInput(tag="in_uchi_meal_ns", input_type=File(optional=True), prefix="--uchimealns", doc=InputDocumentation(doc="output chimera alignments to file")), ToolInput(tag="in_uchi_me_out", input_type=File(optional=True), prefix="--uchimeout", doc=InputDocumentation(doc="output to chimera info to tab-separated file")), ToolInput(tag="in_uchi_me_out_five", input_type=Boolean(optional=True), prefix="--uchimeout5", doc=InputDocumentation(doc="make output compatible with uchime version 5")), ToolInput(tag="in_xsize", input_type=Boolean(optional=True), prefix="--xsize", doc=InputDocumentation(doc="strip abundance information in output")), ToolInput(tag="in_cluster_fast", input_type=File(optional=True), prefix="--cluster_fast", doc=InputDocumentation(doc="cluster sequences after sorting by length")), ToolInput(tag="in_cluster_size", input_type=File(optional=True), prefix="--cluster_size", doc=InputDocumentation(doc="cluster sequences after sorting by abundance")), ToolInput(tag="in_cluster_small_mem", input_type=File(optional=True), prefix="--cluster_smallmem", doc=InputDocumentation(doc="cluster already sorted sequences (see -usersort)")), ToolInput(tag="in_cluster_u_noise", input_type=File(optional=True), prefix="--cluster_unoise", doc=InputDocumentation(doc="denoise Illumina amplicon reads")), ToolInput(tag="in_cons_truncate", input_type=Boolean(optional=True), prefix="--cons_truncate", doc=InputDocumentation(doc="do not ignore terminal gaps in MSA for consensus")), ToolInput(tag="in_id", input_type=Float(optional=True), prefix="--id", doc=InputDocumentation(doc="reject if identity lower, accepted values: 0-1.0")), ToolInput(tag="in_id_def", input_type=Int(optional=True), prefix="--iddef", doc=InputDocumentation(doc="id definition, 0-4=CD-HIT,all,int,MBL,BLAST (2)")), ToolInput(tag="in_q_mask", input_type=String(optional=True), prefix="--qmask", doc=InputDocumentation(doc="|dust|soft      mask seqs with dust, soft or no method (dust)")), ToolInput(tag="in_strand", input_type=String(optional=True), prefix="--strand", doc=InputDocumentation(doc="|both          cluster using plus or both strands (plus)")), ToolInput(tag="in_user_sort", input_type=Boolean(optional=True), prefix="--usersort", doc=InputDocumentation(doc="indicate sequences not pre-sorted by length")), ToolInput(tag="in_minsize", input_type=Int(optional=True), prefix="--minsize", doc=InputDocumentation(doc="minimum abundance (unoise only) (8)")), ToolInput(tag="in_u_noise_alpha", input_type=Float(optional=True), prefix="--unoise_alpha", doc=InputDocumentation(doc="alpha parameter (unoise only) (2.0)")), ToolInput(tag="in_centroids", input_type=File(optional=True), prefix="--centroids", doc=InputDocumentation(doc="output centroid sequences to FASTA file")), ToolInput(tag="in_cluster_out_id", input_type=Boolean(optional=True), prefix="--clusterout_id", doc=InputDocumentation(doc="add cluster id info to consout and profile files")), ToolInput(tag="in_cluster_out_sort", input_type=Boolean(optional=True), prefix="--clusterout_sort", doc=InputDocumentation(doc="order msaout, consout, profile by decr abundance")), ToolInput(tag="in_clusters", input_type=File(optional=True), prefix="--clusters", doc=InputDocumentation(doc="output each cluster to a separate FASTA file")), ToolInput(tag="in_cons_out", input_type=File(optional=True), prefix="--consout", doc=InputDocumentation(doc="output cluster consensus sequences to FASTA file")), ToolInput(tag="in_mo_thur_shared_out", input_type=File(optional=True), prefix="--mothur_shared_out", doc=InputDocumentation(doc="filename for OTU table output in mothur format")), ToolInput(tag="in_msa_out", input_type=File(optional=True), prefix="--msaout", doc=InputDocumentation(doc="output multiple seq. alignments to FASTA file")), ToolInput(tag="in_o_tut_about", input_type=File(optional=True), prefix="--otutabout", doc=InputDocumentation(doc="filename for OTU table output in classic format")), ToolInput(tag="in_profile", input_type=File(optional=True), prefix="--profile", doc=InputDocumentation(doc="output sequence profile of each cluster to file")), ToolInput(tag="in_size_order", input_type=Boolean(optional=True), prefix="--sizeorder", doc=InputDocumentation(doc="sort accepted centroids by abundance, AGC")), ToolInput(tag="in_uc", input_type=File(optional=True), prefix="--uc", doc=InputDocumentation(doc="specify filename for UCLUST-like output")), ToolInput(tag="in_sff_convert", input_type=File(optional=True), prefix="--sff_convert", doc=InputDocumentation(doc="convert given SFF file to FASTQ format")), ToolInput(tag="in_fast_q_ascii_out", input_type=Int(optional=True), prefix="--fastq_asciiout", doc=InputDocumentation(doc="FASTQ output quality score ASCII base char (33)")), ToolInput(tag="in_fast_q_qmax_out", input_type=Int(optional=True), prefix="--fastq_qmaxout", doc=InputDocumentation(doc="maximum base quality value for FASTQ output (41)")), ToolInput(tag="in_fast_q_q_min_out", input_type=Int(optional=True), prefix="--fastq_qminout", doc=InputDocumentation(doc="minimum base quality value for FASTQ output (0)")), ToolInput(tag="in_de_rep_full_length", input_type=File(optional=True), prefix="--derep_fulllength", doc=InputDocumentation(doc="dereplicate sequences in the given FASTA file")), ToolInput(tag="in_de_rep_id", input_type=File(optional=True), prefix="--derep_id", doc=InputDocumentation(doc="dereplicate using both identifiers and sequences")), ToolInput(tag="in_de_rep_prefix", input_type=File(optional=True), prefix="--derep_prefix", doc=InputDocumentation(doc="dereplicate sequences in file based on prefixes")), ToolInput(tag="in_re_replicate", input_type=File(optional=True), prefix="--rereplicate", doc=InputDocumentation(doc="rereplicate sequences in the given FASTA file")), ToolInput(tag="in_min_unique_size", input_type=Int(optional=True), prefix="--minuniquesize", doc=InputDocumentation(doc="minimum abundance for output from dereplication")), ToolInput(tag="in_top_n", input_type=Int(optional=True), prefix="--topn", doc=InputDocumentation(doc="output only n most abundant sequences after derep")), ToolInput(tag="in_fast_q_convert", input_type=File(optional=True), prefix="--fastq_convert", doc=InputDocumentation(doc="convert between FASTQ file formats")), ToolInput(tag="in_fast_q_qmax", input_type=Int(optional=True), prefix="--fastq_qmax", doc=InputDocumentation(doc="maximum base quality value for FASTQ input (41)")), ToolInput(tag="in_fast_q_qm_in", input_type=Int(optional=True), prefix="--fastq_qmin", doc=InputDocumentation(doc="minimum base quality value for FASTQ input (0)")), ToolInput(tag="in_fast_q_chars", input_type=File(optional=True), prefix="--fastq_chars", doc=InputDocumentation(doc="analyse FASTQ file for version and quality range")), ToolInput(tag="in_fast_q_stats", input_type=File(optional=True), prefix="--fastq_stats", doc=InputDocumentation(doc="report statistics on FASTQ file")), ToolInput(tag="in_fast_q_ee_stats", input_type=File(optional=True), prefix="--fastq_eestats", doc=InputDocumentation(doc="quality score and expected error statistics")), ToolInput(tag="in_fast_q_ee_stats_two", input_type=File(optional=True), prefix="--fastq_eestats2", doc=InputDocumentation(doc="expected error and length cutoff statistics")), ToolInput(tag="in_fast_q_ascii", input_type=Int(optional=True), prefix="--fastq_ascii", doc=InputDocumentation(doc="FASTQ input quality score ASCII base char (33)")), ToolInput(tag="in_length_cutoffs", input_type=Int(optional=True), prefix="--length_cutoffs", doc=InputDocumentation(doc=",INT,INT fastq_eestats2 length (min,max,incr) (50,*,50)")), ToolInput(tag="in_output_file_statistics", input_type=File(optional=True), prefix="--output", doc=InputDocumentation(doc="output file for fastq_eestats(2) statistics")), ToolInput(tag="in_fast_x_mask", input_type=File(optional=True), prefix="--fastx_mask", doc=InputDocumentation(doc="mask sequences in the given FASTA or FASTQ file")), ToolInput(tag="in_hard_mask", input_type=Boolean(optional=True), prefix="--hardmask", doc=InputDocumentation(doc="mask by replacing with N instead of lower case")), ToolInput(tag="in_max_unmasked_pct", input_type=Boolean(optional=True), prefix="--max_unmasked_pct", doc=InputDocumentation(doc="max unmasked % of sequences to keep (100.0)")), ToolInput(tag="in_min_unmasked_pct", input_type=Boolean(optional=True), prefix="--min_unmasked_pct", doc=InputDocumentation(doc="min unmasked % of sequences to keep (0.0)")), ToolInput(tag="in_fast_q_out", input_type=File(optional=True), prefix="--fastqout", doc=InputDocumentation(doc="output to specified FASTQ file")), ToolInput(tag="in_mask_fast_a", input_type=File(optional=True), prefix="--maskfasta", doc=InputDocumentation(doc="mask sequences in the given FASTA file")), ToolInput(tag="in_fast_q_join", input_type=File(optional=True), prefix="--fastq_join", doc=InputDocumentation(doc="join paired-end reads into one sequence with gap")), ToolInput(tag="in_join_pad_gap", input_type=String(optional=True), prefix="--join_padgap", doc=InputDocumentation(doc="sequence string used for padding (NNNNNNNN)")), ToolInput(tag="in_join_pad_gap_q", input_type=String(optional=True), prefix="--join_padgapq", doc=InputDocumentation(doc="quality string used for padding (IIIIIIII)")), ToolInput(tag="in_fast_q_merge_pairs", input_type=File(optional=True), prefix="--fastq_mergepairs", doc=InputDocumentation(doc="merge paired-end reads into one sequence")), ToolInput(tag="in_fast_q_max_diff_pct", input_type=Float(optional=True), prefix="--fastq_maxdiffpct", doc=InputDocumentation(doc="maximum percentage diff. bases in overlap (100.0)")), ToolInput(tag="in_fast_q_max_diffs", input_type=Int(optional=True), prefix="--fastq_maxdiffs", doc=InputDocumentation(doc="maximum number of different bases in overlap (10)")), ToolInput(tag="in_fast_q_max_ee", input_type=String(optional=True), prefix="--fastq_maxee", doc=InputDocumentation(doc="maximum expected error value for merged sequence")), ToolInput(tag="in_fast_q_max_merge_len", input_type=Boolean(optional=True), prefix="--fastq_maxmergelen", doc=InputDocumentation(doc="maximum length of entire merged sequence")), ToolInput(tag="in_fast_q_max_ns", input_type=Int(optional=True), prefix="--fastq_maxns", doc=InputDocumentation(doc="maximum number of N's")), ToolInput(tag="in_fast_q_min_len", input_type=Int(optional=True), prefix="--fastq_minlen", doc=InputDocumentation(doc="minimum input read length after truncation (1)")), ToolInput(tag="in_fast_q_min_merge_len", input_type=Boolean(optional=True), prefix="--fastq_minmergelen", doc=InputDocumentation(doc="minimum length of entire merged sequence")), ToolInput(tag="in_fast_q_mi_nov_len", input_type=Boolean(optional=True), prefix="--fastq_minovlen", doc=InputDocumentation(doc="minimum length of overlap between reads (10)")), ToolInput(tag="in_fast_q_no_stagger", input_type=Boolean(optional=True), prefix="--fastq_nostagger", doc=InputDocumentation(doc="disallow merging of staggered reads (default)")), ToolInput(tag="in_fast_q_trunc_qual", input_type=Int(optional=True), prefix="--fastq_truncqual", doc=InputDocumentation(doc="base quality value for truncation")), ToolInput(tag="in_fast_a_out", input_type=File(optional=True), prefix="--fastaout", doc=InputDocumentation(doc="FASTA output filename for merged sequences")), ToolInput(tag="in_fast_a_out_not_merged_fwd", input_type=File(optional=True), prefix="--fastaout_notmerged_fwd", doc=InputDocumentation(doc="FASTA filename for non-merged forward sequences")), ToolInput(tag="in_fast_a_out_not_merged_rev", input_type=File(optional=True), prefix="--fastaout_notmerged_rev", doc=InputDocumentation(doc="FASTA filename for non-merged reverse sequences")), ToolInput(tag="in_fast_q_ee_out", input_type=Boolean(optional=True), prefix="--fastq_eeout", doc=InputDocumentation(doc="include expected errors (ee) in FASTQ output")), ToolInput(tag="in_fast_q_out_not_merged_fwd", input_type=File(optional=True), prefix="--fastqout_notmerged_fwd", doc=InputDocumentation(doc="FASTQ filename for non-merged forward sequences")), ToolInput(tag="in_fast_q_out_not_merged_rev", input_type=File(optional=True), prefix="--fastqout_notmerged_rev", doc=InputDocumentation(doc="FASTQ filename for non-merged reverse sequences")), ToolInput(tag="in_label_suffix", input_type=String(optional=True), prefix="--label_suffix", doc=InputDocumentation(doc="suffix to append to label of merged sequences")), ToolInput(tag="in_xee", input_type=Boolean(optional=True), prefix="--xee", doc=InputDocumentation(doc="remove expected errors (ee) info from output")), ToolInput(tag="in_all_pairs_global", input_type=File(optional=True), prefix="--allpairs_global", doc=InputDocumentation(doc="perform global alignment of all sequence pairs")), ToolInput(tag="in_aln_out", input_type=File(optional=True), prefix="--alnout", doc=InputDocumentation(doc="filename for human-readable alignment output")), ToolInput(tag="in_accept_all", input_type=Boolean(optional=True), prefix="--acceptall", doc=InputDocumentation(doc="output all pairwise alignments")), ToolInput(tag="in_cut", input_type=File(optional=True), prefix="--cut", doc=InputDocumentation(doc="filename of FASTA formatted input sequences")), ToolInput(tag="in_fast_a_out_rev", input_type=File(optional=True), prefix="--fastaout_rev", doc=InputDocumentation(doc="FASTA filename for fragments on reverse strand")), ToolInput(tag="in_fast_a_out_discarded", input_type=File(optional=True), prefix="--fastaout_discarded", doc=InputDocumentation(doc="FASTA filename for non-matching sequences")), ToolInput(tag="in_fast_a_out_discarded_rev", input_type=File(optional=True), prefix="--fastaout_discarded_rev", doc=InputDocumentation(doc="FASTA filename for non-matching, reverse compl.")), ToolInput(tag="in_fast_x_rev_comp", input_type=File(optional=True), prefix="--fastx_revcomp", doc=InputDocumentation(doc="reverse-complement seqs in FASTA or FASTQ file")), ToolInput(tag="in_search_exact", input_type=File(optional=True), prefix="--search_exact", doc=InputDocumentation(doc="filename of queries for exact match search")), ToolInput(tag="in_u_search_global", input_type=File(optional=True), prefix="--usearch_global", doc=InputDocumentation(doc="filename of queries for global alignment search")), ToolInput(tag="in_full_dp", input_type=Boolean(optional=True), prefix="--fulldp", doc=InputDocumentation(doc="full dynamic programming alignment (always on)")), ToolInput(tag="in_gap_ext", input_type=Int(optional=True), prefix="--gapext", doc=InputDocumentation(doc="penalties for gap extension (2I/1E)")), ToolInput(tag="in_gap_open", input_type=Int(optional=True), prefix="--gapopen", doc=InputDocumentation(doc="penalties for gap opening (20I/2E)")), ToolInput(tag="in_id_prefix", input_type=Int(optional=True), prefix="--idprefix", doc=InputDocumentation(doc="reject if first n nucleotides do not match")), ToolInput(tag="in_id_suffix", input_type=Int(optional=True), prefix="--idsuffix", doc=InputDocumentation(doc="reject if last n nucleotides do not match")), ToolInput(tag="in_left_just", input_type=Boolean(optional=True), prefix="--leftjust", doc=InputDocumentation(doc="reject if terminal gaps at alignment left end")), ToolInput(tag="in_match", input_type=Int(optional=True), prefix="--match", doc=InputDocumentation(doc="score for match (2)")), ToolInput(tag="in_max_accepts", input_type=Int(optional=True), prefix="--maxaccepts", doc=InputDocumentation(doc="number of hits to accept and show per strand (1)")), ToolInput(tag="in_max_diffs", input_type=Int(optional=True), prefix="--maxdiffs", doc=InputDocumentation(doc="reject if more substitutions or indels")), ToolInput(tag="in_max_gaps", input_type=Int(optional=True), prefix="--maxgaps", doc=InputDocumentation(doc="reject if more indels")), ToolInput(tag="in_max_hits", input_type=Int(optional=True), prefix="--maxhits", doc=InputDocumentation(doc="maximum number of hits to show (unlimited)")), ToolInput(tag="in_max_id", input_type=String(optional=True), prefix="--maxid", doc=InputDocumentation(doc="reject if identity higher")), ToolInput(tag="in_maxq_size", input_type=Int(optional=True), prefix="--maxqsize", doc=InputDocumentation(doc="reject if query abundance larger")), ToolInput(tag="in_max_qt", input_type=Int(optional=True), prefix="--maxqt", doc=InputDocumentation(doc="reject if query/target length ratio higher")), ToolInput(tag="in_max_rejects", input_type=Int(optional=True), prefix="--maxrejects", doc=InputDocumentation(doc="number of non-matching hits to consider (32)")), ToolInput(tag="in_maxsize_ratio", input_type=String(optional=True), prefix="--maxsizeratio", doc=InputDocumentation(doc="reject if query/target abundance ratio higher")), ToolInput(tag="in_max_sl", input_type=Int(optional=True), prefix="--maxsl", doc=InputDocumentation(doc="reject if shorter/longer length ratio higher")), ToolInput(tag="in_max_subs", input_type=Int(optional=True), prefix="--maxsubs", doc=InputDocumentation(doc="reject if more substitutions")), ToolInput(tag="in_mid", input_type=String(optional=True), prefix="--mid", doc=InputDocumentation(doc="reject if percent identity lower, ignoring gaps")), ToolInput(tag="in_min_cols", input_type=Int(optional=True), prefix="--mincols", doc=InputDocumentation(doc="reject if alignment length shorter")), ToolInput(tag="in_min_qt", input_type=Int(optional=True), prefix="--minqt", doc=InputDocumentation(doc="reject if query/target length ratio lower")), ToolInput(tag="in_minsize_ratio", input_type=String(optional=True), prefix="--minsizeratio", doc=InputDocumentation(doc="reject if query/target abundance ratio lower")), ToolInput(tag="in_mins_l", input_type=Int(optional=True), prefix="--minsl", doc=InputDocumentation(doc="reject if shorter/longer length ratio lower")), ToolInput(tag="in_mint_size", input_type=Int(optional=True), prefix="--mintsize", doc=InputDocumentation(doc="reject if target abundance lower")), ToolInput(tag="in_min_word_matches", input_type=Int(optional=True), prefix="--minwordmatches", doc=InputDocumentation(doc="minimum number of word matches required (12)")), ToolInput(tag="in_mismatch", input_type=Int(optional=True), prefix="--mismatch", doc=InputDocumentation(doc="score for mismatch (-4)")), ToolInput(tag="in_pattern", input_type=String(optional=True), prefix="--pattern", doc=InputDocumentation(doc="option is ignored")), ToolInput(tag="in_query_cov", input_type=String(optional=True), prefix="--query_cov", doc=InputDocumentation(doc="reject if fraction of query seq. aligned lower")), ToolInput(tag="in_right_just", input_type=Boolean(optional=True), prefix="--rightjust", doc=InputDocumentation(doc="reject if terminal gaps at alignment right end")), ToolInput(tag="in_slots", input_type=Int(optional=True), prefix="--slots", doc=InputDocumentation(doc="option is ignored")), ToolInput(tag="in_target_cov", input_type=String(optional=True), prefix="--target_cov", doc=InputDocumentation(doc="reject if fraction of target seq. aligned lower")), ToolInput(tag="in_weak_id", input_type=String(optional=True), prefix="--weak_id", doc=InputDocumentation(doc="include aligned hits with >= id; continue search")), ToolInput(tag="in_word_length", input_type=Int(optional=True), prefix="--wordlength", doc=InputDocumentation(doc="length of words for database index 3-15 (8)")), ToolInput(tag="in_biom_out", input_type=File(optional=True), prefix="--biomout", doc=InputDocumentation(doc="filename for OTU table output in biom 1.0 format")), ToolInput(tag="in_blast_six_out", input_type=File(optional=True), prefix="--blast6out", doc=InputDocumentation(doc="filename for blast-like tab-separated output")), ToolInput(tag="in_db_matched", input_type=File(optional=True), prefix="--dbmatched", doc=InputDocumentation(doc="FASTA file for matching database sequences")), ToolInput(tag="in_db_not_matched", input_type=File(optional=True), prefix="--dbnotmatched", doc=InputDocumentation(doc="FASTA file for non-matching database sequences")), ToolInput(tag="in_fast_a_pairs", input_type=File(optional=True), prefix="--fastapairs", doc=InputDocumentation(doc="FASTA file with pairs of query and target")), ToolInput(tag="in_matched", input_type=File(optional=True), prefix="--matched", doc=InputDocumentation(doc="FASTA file for matching query sequences")), ToolInput(tag="in_not_matched", input_type=File(optional=True), prefix="--notmatched", doc=InputDocumentation(doc="FASTA file for non-matching query sequences")), ToolInput(tag="in_output_no_hits", input_type=Boolean(optional=True), prefix="--output_no_hits", doc=InputDocumentation(doc="output non-matching queries to output files")), ToolInput(tag="in_row_len", input_type=Int(optional=True), prefix="--rowlen", doc=InputDocumentation(doc="width of alignment lines in alnout output (64)")), ToolInput(tag="in_sam_header", input_type=File(optional=True), prefix="--samheader", doc=InputDocumentation(doc="include a header in the SAM output file")), ToolInput(tag="in_sam_out", input_type=File(optional=True), prefix="--samout", doc=InputDocumentation(doc="filename for SAM format output")), ToolInput(tag="in_top_hits_only", input_type=Boolean(optional=True), prefix="--top_hits_only", doc=InputDocumentation(doc="output only hits with identity equal to the best")), ToolInput(tag="in_uc_all_hits", input_type=Boolean(optional=True), prefix="--uc_allhits", doc=InputDocumentation(doc="show all, not just top hit with uc output")), ToolInput(tag="in_user_fields", input_type=File(optional=True), prefix="--userfields", doc=InputDocumentation(doc="fields to output in userout file")), ToolInput(tag="in_user_out", input_type=File(optional=True), prefix="--userout", doc=InputDocumentation(doc="filename for user-defined tab-separated output")), ToolInput(tag="in_shuffle", input_type=File(optional=True), prefix="--shuffle", doc=InputDocumentation(doc="shuffle order of sequences in FASTA file randomly")), ToolInput(tag="in_sort_by_length", input_type=File(optional=True), prefix="--sortbylength", doc=InputDocumentation(doc="sort sequences by length in given FASTA file")), ToolInput(tag="in_sort_by_size", input_type=File(optional=True), prefix="--sortbysize", doc=InputDocumentation(doc="abundance sort sequences in given FASTA file")), ToolInput(tag="in_rand_seed", input_type=Int(optional=True), prefix="--randseed", doc=InputDocumentation(doc="seed for PRNG, zero to use random data source (0)")), ToolInput(tag="in_fast_x_subsample", input_type=File(optional=True), prefix="--fastx_subsample", doc=InputDocumentation(doc="subsample sequences from given FASTA/FASTQ file")), ToolInput(tag="in_sample_pct", input_type=Float(optional=True), prefix="--sample_pct", doc=InputDocumentation(doc="sampling percentage between 0.0 and 100.0")), ToolInput(tag="in_sample_size", input_type=Int(optional=True), prefix="--sample_size", doc=InputDocumentation(doc="sampling size")), ToolInput(tag="in_fast_q_out_discarded", input_type=File(optional=True), prefix="--fastqout_discarded", doc=InputDocumentation(doc="output non-subsampled sequences to FASTQ file")), ToolInput(tag="in_sin_tax", input_type=File(optional=True), prefix="--sintax", doc=InputDocumentation(doc="classify sequences in given FASTA/FASTQ file")), ToolInput(tag="in_sin_tax_cut_off", input_type=Float(optional=True), prefix="--sintax_cutoff", doc=InputDocumentation(doc="confidence value cutoff level (0.0)")), ToolInput(tag="in_fast_x_filter", input_type=File(optional=True), prefix="--fastx_filter", doc=InputDocumentation(doc="trim and filter sequences in FASTA/FASTQ file")), ToolInput(tag="in_fast_q_filter", input_type=File(optional=True), prefix="--fastq_filter", doc=InputDocumentation(doc="trim and filter sequences in FASTQ file")), ToolInput(tag="in_reverse", input_type=File(optional=True), prefix="--reverse", doc=InputDocumentation(doc="FASTQ file with other end of paired-end reads")), ToolInput(tag="in_fast_q_max_ee_rate", input_type=String(optional=True), prefix="--fastq_maxee_rate", doc=InputDocumentation(doc="discard if expected error rate is higher")), ToolInput(tag="in_fast_q_maxlen", input_type=Int(optional=True), prefix="--fastq_maxlen", doc=InputDocumentation(doc="discard if length of sequence is longer")), ToolInput(tag="in_fast_q_strip_left", input_type=Int(optional=True), prefix="--fastq_stripleft", doc=InputDocumentation(doc="delete given number of bases from the 5' end")), ToolInput(tag="in_fast_q_strip_right", input_type=Int(optional=True), prefix="--fastq_stripright", doc=InputDocumentation(doc="delete given number of bases from the 3' end")), ToolInput(tag="in_fast_q_trunc_ee", input_type=String(optional=True), prefix="--fastq_truncee", doc=InputDocumentation(doc="truncate to given maximum expected error")), ToolInput(tag="in_fast_q_trunc_len", input_type=Int(optional=True), prefix="--fastq_trunclen", doc=InputDocumentation(doc="truncate to given length (discard if shorter)")), ToolInput(tag="in_fast_q_trunc_len_keep", input_type=Int(optional=True), prefix="--fastq_trunclen_keep", doc=InputDocumentation(doc="truncate to given length (keep if shorter)")), ToolInput(tag="in_maxsize", input_type=Int(optional=True), prefix="--maxsize", doc=InputDocumentation(doc="discard if abundance of sequence is above")), ToolInput(tag="in_fast_q_out_discarded_rev", input_type=File(optional=True), prefix="--fastqout_discarded_rev", doc=InputDocumentation(doc="FASTQ filename for discarded reverse sequences")), ToolInput(tag="in_fast_q_out_rev", input_type=File(optional=True), prefix="--fastqout_rev", doc=InputDocumentation(doc="FASTQ filename for passed reverse sequences")), ToolInput(tag="in_make_udb_u_search", input_type=File(optional=True), prefix="--makeudb_usearch", doc=InputDocumentation(doc="make UDB file from given FASTA file")), ToolInput(tag="in_udb_two_fast_a", input_type=File(optional=True), prefix="--udb2fasta", doc=InputDocumentation(doc="output FASTA file from given UDB file")), ToolInput(tag="in_udb_info", input_type=File(optional=True), prefix="--udbinfo", doc=InputDocumentation(doc="show information about UDB file")), ToolInput(tag="in_udb_stats", input_type=File(optional=True), prefix="--udbstats", doc=InputDocumentation(doc="report statistics about indexed words in UDB file")), ToolInput(tag="in__db_database", input_type=String(), position=0, doc=InputDocumentation(doc="--db FILENAME               reference database for --uchime_ref")), ToolInput(tag="in_abskew_real_minimum", input_type=String(), position=1, doc=InputDocumentation(doc="--abskew REAL               minimum abundance ratio (2.0, 16.0 for uchime3)")), ToolInput(tag="in__sffclip_clip", input_type=String(), position=0, doc=InputDocumentation(doc="--sff_clip                  clip ends of sequences as indicated in file (no)")), ToolInput(tag="in__fastqout_filename_output", input_type=String(), position=1, doc=InputDocumentation(doc="--fastqout FILENAME         output converted sequences to given FASTQ file")), ToolInput(tag="in__maxuniquesize_abundance", input_type=String(), position=0, doc=InputDocumentation(doc="--maxuniquesize INT         maximum abundance for output from dereplication")), ToolInput(tag="in_output_filename_output_fasta", input_type=String(), position=1, doc=InputDocumentation(doc="--output FILENAME           output FASTA file")), ToolInput(tag="in_var_198", input_type=String(), position=0, doc=InputDocumentation(doc="--fastq_ascii INT           FASTQ input quality score ASCII base char (33)")), ToolInput(tag="in__fastqout_filename_fastq", input_type=String(), position=1, doc=InputDocumentation(doc="--fastqout FILENAME         FASTQ output filename for converted sequences")), ToolInput(tag="in_log_filename_file", input_type=String(), position=1, doc=InputDocumentation(doc="--log FILENAME              output file for fastq_stats statistics")), ToolInput(tag="in_var_201", input_type=String(), position=0, doc=InputDocumentation(doc="--fastq_ascii INT           FASTQ input quality score ASCII base char (33)")), ToolInput(tag="in__fastaout_output", input_type=String(), position=1, doc=InputDocumentation(doc="--fastaout FILENAME         output to specified FASTA file")), ToolInput(tag="in__hardmask_mask", input_type=String(), position=0, doc=InputDocumentation(doc="--hardmask                  mask by replacing with N instead of lower case")), ToolInput(tag="in_var_204", input_type=String(), position=1, doc=InputDocumentation(doc="--output FILENAME           output to specified FASTA file")), ToolInput(tag="in_var_205", input_type=String(), position=0, doc=InputDocumentation(doc="--reverse FILENAME          specify FASTQ file with reverse reads")), ToolInput(tag="in_for_joined_sequences", input_type=String(), position=1, doc=InputDocumentation(doc="--fastaout FILENAME         FASTA output filename for joined sequences")), ToolInput(tag="in_var_207", input_type=String(), position=0, doc=InputDocumentation(doc="--reverse FILENAME          specify FASTQ file with reverse reads")), ToolInput(tag="in__fastqallowmergestagger_allow", input_type=String(), position=1, doc=InputDocumentation(doc="--fastq_allowmergestagger   allow merging of staggered reads")), ToolInput(tag="in__eetabbedout_filename", input_type=String(), position=2, doc=InputDocumentation(doc="--eetabbedout FILENAME      output error statistics to specified file")), ToolInput(tag="in_cutpattern_string_pattern", input_type=String(), position=0, doc=InputDocumentation(doc="--cut_pattern STRING        pattern to match with ^ and _ at cut sites")), ToolInput(tag="in__fastaout_filename_filename", input_type=String(), position=1, doc=InputDocumentation(doc="--fastaout FILENAME         FASTA filename for fragments on forward strand")), ToolInput(tag="in_var_212", input_type=String(), position=0, doc=InputDocumentation(doc="--fastq_ascii INT           FASTQ input quality score ASCII base char (33)")), ToolInput(tag="in__fastaout_output_filename", input_type=String(), position=1, doc=InputDocumentation(doc="--fastaout FILENAME         FASTA output filename")), ToolInput(tag="in__db_filename", input_type=String(), position=0, doc=InputDocumentation(doc="--db FILENAME               name of UDB or FASTA database for search")), ToolInput(tag="in_var_215", input_type=String(), position=1, doc=InputDocumentation(doc="--dbmask none|dust|soft     mask db with dust, soft or no method (dust)")), ToolInput(tag="in__filename_filename", input_type=String(), position=2, doc=InputDocumentation(doc="--alnout FILENAME           filename for human-readable alignment output")), ToolInput(tag="in__maxsize_abundance", input_type=String(), position=0, doc=InputDocumentation(doc="--maxsize INT               maximum abundance for sortbysize")), ToolInput(tag="in_var_218", input_type=String(), position=1, doc=InputDocumentation(doc="--output FILENAME           output to specified FASTA file")), ToolInput(tag="in_var_219", input_type=String(), position=0, doc=InputDocumentation(doc="--fastq_ascii INT           FASTQ input quality score ASCII base char (33)")), ToolInput(tag="in__fastaout_subsampled", input_type=String(), position=1, doc=InputDocumentation(doc="--fastaout FILENAME         output subsampled sequences to FASTA file")), ToolInput(tag="in__db_reference", input_type=String(), position=0, doc=InputDocumentation(doc="--db FILENAME               taxonomic reference db in given FASTA or UDB file")), ToolInput(tag="in__tabbedout_filename", input_type=String(), position=1, doc=InputDocumentation(doc="--tabbedout FILENAME        write results to given tab-delimited file")), ToolInput(tag="in_var_223", input_type=String(), position=0, doc=InputDocumentation(doc="--fastq_ascii INT           FASTQ input quality score ASCII base char (33)")), ToolInput(tag="in__eeout_include", input_type=String(), position=1, doc=InputDocumentation(doc="--eeout                     include expected errors in output")), ToolInput(tag="in_var_225", input_type=String(), position=0, doc=InputDocumentation(doc="--dbmask none|dust|soft     mask db with dust, soft or no method (dust)")), ToolInput(tag="in_output_filename_udb", input_type=String(), position=1, doc=InputDocumentation(doc="--output FILENAME           UDB or FASTA output file"))], outputs=[ToolOutput(tag="out_borderline", output_type=File(optional=True), selector=InputSelector(input_to_select="in_borderline", type_hint=File()), doc=OutputDocumentation(doc="output borderline chimeric sequences to file")), ToolOutput(tag="out_chimeras", output_type=File(optional=True), selector=InputSelector(input_to_select="in_chimeras", type_hint=File()), doc=OutputDocumentation(doc="output chimeric sequences to file")), ToolOutput(tag="out_non_chimeras", output_type=File(optional=True), selector=InputSelector(input_to_select="in_non_chimeras", type_hint=File()), doc=OutputDocumentation(doc="output non-chimeric sequences to file")), ToolOutput(tag="out_uchi_meal_ns", output_type=File(optional=True), selector=InputSelector(input_to_select="in_uchi_meal_ns", type_hint=File()), doc=OutputDocumentation(doc="output chimera alignments to file")), ToolOutput(tag="out_uchi_me_out", output_type=File(optional=True), selector=InputSelector(input_to_select="in_uchi_me_out", type_hint=File()), doc=OutputDocumentation(doc="output to chimera info to tab-separated file")), ToolOutput(tag="out_centroids", output_type=File(optional=True), selector=InputSelector(input_to_select="in_centroids", type_hint=File()), doc=OutputDocumentation(doc="output centroid sequences to FASTA file")), ToolOutput(tag="out_clusters", output_type=File(optional=True), selector=InputSelector(input_to_select="in_clusters", type_hint=File()), doc=OutputDocumentation(doc="output each cluster to a separate FASTA file")), ToolOutput(tag="out_cons_out", output_type=File(optional=True), selector=InputSelector(input_to_select="in_cons_out", type_hint=File()), doc=OutputDocumentation(doc="output cluster consensus sequences to FASTA file")), ToolOutput(tag="out_mo_thur_shared_out", output_type=File(optional=True), selector=InputSelector(input_to_select="in_mo_thur_shared_out", type_hint=File()), doc=OutputDocumentation(doc="filename for OTU table output in mothur format")), ToolOutput(tag="out_msa_out", output_type=File(optional=True), selector=InputSelector(input_to_select="in_msa_out", type_hint=File()), doc=OutputDocumentation(doc="output multiple seq. alignments to FASTA file")), ToolOutput(tag="out_o_tut_about", output_type=File(optional=True), selector=InputSelector(input_to_select="in_o_tut_about", type_hint=File()), doc=OutputDocumentation(doc="filename for OTU table output in classic format")), ToolOutput(tag="out_profile", output_type=File(optional=True), selector=InputSelector(input_to_select="in_profile", type_hint=File()), doc=OutputDocumentation(doc="output sequence profile of each cluster to file")), ToolOutput(tag="out_uc", output_type=File(optional=True), selector=InputSelector(input_to_select="in_uc", type_hint=File()), doc=OutputDocumentation(doc="specify filename for UCLUST-like output")), ToolOutput(tag="out_output_file_statistics", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output_file_statistics", type_hint=File()), doc=OutputDocumentation(doc="output file for fastq_eestats(2) statistics")), ToolOutput(tag="out_fast_q_out", output_type=File(optional=True), selector=InputSelector(input_to_select="in_fast_q_out", type_hint=File()), doc=OutputDocumentation(doc="output to specified FASTQ file")), ToolOutput(tag="out_fast_a_out", output_type=File(optional=True), selector=InputSelector(input_to_select="in_fast_a_out", type_hint=File()), doc=OutputDocumentation(doc="FASTA output filename for merged sequences")), ToolOutput(tag="out_aln_out", output_type=File(optional=True), selector=InputSelector(input_to_select="in_aln_out", type_hint=File()), doc=OutputDocumentation(doc="filename for human-readable alignment output")), ToolOutput(tag="out_biom_out", output_type=File(optional=True), selector=InputSelector(input_to_select="in_biom_out", type_hint=File()), doc=OutputDocumentation(doc="filename for OTU table output in biom 1.0 format")), ToolOutput(tag="out_blast_six_out", output_type=File(optional=True), selector=InputSelector(input_to_select="in_blast_six_out", type_hint=File()), doc=OutputDocumentation(doc="filename for blast-like tab-separated output")), ToolOutput(tag="out_sam_header", output_type=File(optional=True), selector=InputSelector(input_to_select="in_sam_header", type_hint=File()), doc=OutputDocumentation(doc="include a header in the SAM output file")), ToolOutput(tag="out_sam_out", output_type=File(optional=True), selector=InputSelector(input_to_select="in_sam_out", type_hint=File()), doc=OutputDocumentation(doc="filename for SAM format output")), ToolOutput(tag="out_user_fields", output_type=File(optional=True), selector=InputSelector(input_to_select="in_user_fields", type_hint=File()), doc=OutputDocumentation(doc="fields to output in userout file")), ToolOutput(tag="out_user_out", output_type=File(optional=True), selector=InputSelector(input_to_select="in_user_out", type_hint=File()), doc=OutputDocumentation(doc="filename for user-defined tab-separated output")), ToolOutput(tag="out_fast_q_out_discarded", output_type=File(optional=True), selector=InputSelector(input_to_select="in_fast_q_out_discarded", type_hint=File()), doc=OutputDocumentation(doc="output non-subsampled sequences to FASTQ file")), ToolOutput(tag="out_udb_two_fast_a", output_type=File(optional=True), selector=InputSelector(input_to_select="in_udb_two_fast_a", type_hint=File()), doc=OutputDocumentation(doc="output FASTA file from given UDB file"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Vsearch_V0_1_0().translate("wdl", allow_empty_container=True)

