from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Int, Boolean, File, String

Display_Rate_Matrix_V0_1_0 = CommandToolBuilder(tool="display_rate_matrix", base_command=["display_rate_matrix"], inputs=[ToolInput(tag="in_output_pt_expqt", input_type=Int(optional=True), prefix="-t", doc=InputDocumentation(doc=": Output P(t) = exp(Qt) instead of Q.  Requires t >= 0.\nUse '-t A' to output a matrix for each branch of the tree.")), ToolInput(tag="in_show_equilibrium_frequencies", input_type=Boolean(optional=True), prefix="-f", doc=InputDocumentation(doc=":     Show equilibrium frequencies as an additional table row.\nIn list node they are shown with first tuple being -.")), ToolInput(tag="in_show_raw_matrix", input_type=Boolean(optional=True), prefix="-e", doc=InputDocumentation(doc=":     Show 'exchangeabilities' instead of raw matrix elements\n(that is, divide each element by the equilibrium frequency\nof its column).  Not available with -t.")), ToolInput(tag="in_suppress_printing_elements", input_type=Boolean(optional=True), prefix="-d", doc=InputDocumentation(doc=":     Suppress printing of elements on main diagonal.")), ToolInput(tag="in_format_table_typesetting", input_type=Boolean(optional=True), prefix="-L", doc=InputDocumentation(doc=":     Format table for typesetting with LATEX.  Incompatible with -l.")), ToolInput(tag="in_show_matrix_elements", input_type=Boolean(optional=True), prefix="-l", doc=InputDocumentation(doc=":     Show matrix elements as a list rather than as a table.\nWhen -t is not specified (rate matrix case), only off-diagonal\nelements will be printed.")), ToolInput(tag="in_use_l_only", input_type=Boolean(optional=True), prefix="-i", doc=InputDocumentation(doc=":     (For use with -l only) Report whether each substitution is\na transition or a transversion.")), ToolInput(tag="in_use_l_report", input_type=Boolean(optional=True), prefix="-z", doc=InputDocumentation(doc=":     (For use with -l) Report elements equal to zero (omitted by\ndefault, except with -t).  Implied by -a.")), ToolInput(tag="in_use_l_assume", input_type=Boolean(optional=True), prefix="-S", doc=InputDocumentation(doc=":     (For use with -l)  Assume a symmetric matrix and report half\nas many lines.  Useful with -e.")), ToolInput(tag="in_use_l_print", input_type=Boolean(optional=True), prefix="-E", doc=InputDocumentation(doc=":     (for use with -l) Print rates and probabilities\nin scientific notation (format %e instead of %f).")), ToolInput(tag="in_requires_model_order", input_type=Boolean(optional=True), prefix="-a", doc=InputDocumentation(doc=":     (Requires a model of order 3).  Replace a matrix of codon\nsubstitution rates with the induced matrix of amino acid\nsubstitution rates, according to the universal genetic\ncode.  See Yang, Nielsen, and Hasegawa, 1998.")), ToolInput(tag="in_use_include_stop", input_type=Boolean(optional=True), prefix="-s", doc=InputDocumentation(doc=":     (For use with -a)  Include stop codons (by default suppressed).")), ToolInput(tag="in_only_implies_implies", input_type=File(optional=True), prefix="-M", doc=InputDocumentation(doc=": (For use with -l only; implies -a) Read an amino-acid\nsubstitution matrix from file <f> and report values from\nthis matrix with the induced amino acid substitution rates.\nMatrix should be in the format used by BLAST (as\nproduced by the NCBI 'pam' program)")), ToolInput(tag="in_m_matrices_format", input_type=String(optional=True), prefix="-N", doc=InputDocumentation(doc="Like -M but for matrices in the format used by the PAML\npackage for amino acid substitution and rate matrices.")), ToolInput(tag="in_use_l_and", input_type=File(optional=True), prefix="-A", doc=InputDocumentation(doc=": (For use with -l only and not with -M/-N)  Read alternative\nsubstitution scores from file <f> and report values in\noutput.  File <f> should have three columns: a 'from'\ntuple, a 'to' tuple, and a real-valued score.\nSubstitutions not listed will be given null scores and\nreported as 'NA'.")), ToolInput(tag="in_a_compares_rates", input_type=File(optional=True), prefix="-B", doc=InputDocumentation(doc="Like -A but compares to rates of a single-nucleotide model\n(order 1).  File <f> should be a standard tree model (.mod) file.")), ToolInput(tag="in_report_contextdependent_rates", input_type=Boolean(optional=True), prefix="-C", doc=InputDocumentation(doc="Report context-dependent transition/transversion rates, as\nshown in Tables 2 and 3 of Morton et al., JME 45:227-231, 1997.\nRequires a model of order 3 with a DNA alphabet.")), ToolInput(tag="in_model_fname", input_type=String(), position=0, doc=InputDocumentation(doc=""))], outputs=[ToolOutput(tag="out_use_l_and", output_type=File(optional=True), selector=InputSelector(input_to_select="in_use_l_and", type_hint=File()), doc=OutputDocumentation(doc=": (For use with -l only and not with -M/-N)  Read alternative\nsubstitution scores from file <f> and report values in\noutput.  File <f> should have three columns: a 'from'\ntuple, a 'to' tuple, and a real-valued score.\nSubstitutions not listed will be given null scores and\nreported as 'NA'."))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Display_Rate_Matrix_V0_1_0().translate("wdl", allow_empty_container=True)

