from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import String, File, Directory, Array, Int, Boolean

Hicbuildmatrix_V0_1_0 = CommandToolBuilder(tool="hicBuildMatrix", base_command=["hicBuildMatrix"], inputs=[ToolInput(tag="in_sam_files", input_type=String(optional=True), prefix="--samFiles", doc=InputDocumentation(doc="sam files two sam files, -s two sam files two sam files\nThe two PE alignment sam files to process (default:\nNone)")), ToolInput(tag="in_out_filename", input_type=File(optional=True), prefix="--outFileName", doc=InputDocumentation(doc="Output file name for the Hi-C matrix. (default: None)")), ToolInput(tag="in_qc_folder", input_type=Directory(optional=True), prefix="--QCfolder", doc=InputDocumentation(doc="Path of folder to save the quality control data for\nthe matrix. The log files produced this way can be\nloaded into `hicQC` in order to compare the quality of\nmultiple Hi-C libraries. (default: None)")), ToolInput(tag="in_restriction_cut_file", input_type=File(optional=True), prefix="--restrictionCutFile", doc=InputDocumentation(doc="file [BED file ...], -rs BED file [BED file ...]\nBED file(s) with all restriction cut sites (output of\n'hicFindRestSite' command). Should only contain the\nrestriction sites of the same genome which has been\nused to generate the input sam files. Using regions of\na different genome version can generate false results!\nTo use more than one restriction enzyme, generate a\nrestrictionCutFile for each enzyne and list them space\nseperated. (default: None)")), ToolInput(tag="in_restriction_sequence", input_type=Array(t=String(), optional=True), prefix="--restrictionSequence", doc=InputDocumentation(doc="Sequence of the restriction site, if multiple are\nused, please list them space seperated. If a dangling\nsequence is listed at the same time, please preserve\nthe same order. (default: None)")), ToolInput(tag="in_dangling_sequence", input_type=Array(t=String(), optional=True), prefix="--danglingSequence", doc=InputDocumentation(doc="Sequence left by the restriction enzyme after cutting,\nif multiple are used, please list them space seperated\nand preserve the order. Each restriction enzyme\nrecognizes a different DNA sequence and, after\ncutting, they leave behind a specific 'sticky' end or\ndangling end sequence. For example, for HindIII the\nrestriction site is AAGCTT and the dangling end is\nAGCT. For DpnII, the restriction site and dangling end\nsequence are the same: GATC. This information is\neasily found on the description of the restriction\nenzyme. The dangling sequence is used to classify and\nreport reads whose 5' end starts with such sequence as\ndangling-end reads. A significant portion of dangling-\nend reads in a sample are indicative of a problem with\nthe re-ligation step of the protocol. (default: None)")), ToolInput(tag="in_out_bam", input_type=File(optional=True), prefix="--outBam", doc=InputDocumentation(doc="file, -b bam file\nOutput bam file to process. Optional parameter. A bam\nfile containing all valid Hi-C reads can be created\nusing this option. This bam file could be useful to\ninspect the distribution of valid Hi-C reads pairs or\nfor other downstream analyses, but is not used by any\nHiCExplorer tool. Computation will be significantly\nlonger if this option is set. (default: None)")), ToolInput(tag="in_binsize", input_type=Array(t=String(), optional=True), prefix="--binSize", doc=InputDocumentation(doc="Size in bp for the bins. The bin size depends on the\ndepth of sequencing. Use a larger bin size for\nlibraries sequenced with lower depth. If not given,\nmatrices of restriction site resolution will be built.\nOptionally for mcool file format: Define multiple\nresolutions which are all a multiple of the first\nvalue. Example: --binSize 10000 20000 50000 will\ncreate a mcool file formate containing the three\ndefined resolutions. (default: None)")), ToolInput(tag="in_min_distance", input_type=Int(optional=True), prefix="--minDistance", doc=InputDocumentation(doc="Minimum distance between restriction sites.\nRestriction sites that are closer than this distance\nare merged into one. This option only applies if\n--restrictionCutFile is given (Default: 300).")), ToolInput(tag="in_max_distance", input_type=Int(optional=True), prefix="--maxDistance", doc=InputDocumentation(doc="This parameter is now obsolete. Use\n--maxLibraryInsertSize instead (default: None)")), ToolInput(tag="in_max_library_insert_size", input_type=Int(optional=True), prefix="--maxLibraryInsertSize", doc=InputDocumentation(doc="The maximum library insert size defines different cut\noffs based on the maximum expected library size. *This\nis not the average fragment size* but the higher end\nof the the fragment size distribution (obtained using\nfor example a Fragment Analyzer or a Bioanalyzer)\nwhich usually is between 800 to 1500 bp. If this value\nis not known use the default value. The insert value\nis used to decide if two mates belong to the same\nfragment (by checking if they are within this max\ninsert size) and to decide if a mate is too far away\nfrom the nearest restriction site (Default: 1000).")), ToolInput(tag="in_genome_assembly", input_type=File(optional=True), prefix="--genomeAssembly", doc=InputDocumentation(doc="The genome the reads were mapped to. Used for metadata\nof cool file. (default: None)")), ToolInput(tag="in_region", input_type=Int(optional=True), prefix="--region", doc=InputDocumentation(doc=":START-END, -r CHR:START-END\nRegion of the genome to limit the operation to. The\nformat is chr:start-end. It is also possible to just\nspecify a chromosome, for example --region chr10\n(default: None)")), ToolInput(tag="in_remove_self_ligation", input_type=Int(optional=True), prefix="--removeSelfLigation", doc=InputDocumentation(doc="If set, inward facing reads less than 1000 bp apart\nand having a restrictionsite in between are removed.\nAlthough this reads do not contribute to any distant\ncontact, they are useful to account for bias in the\ndata (for the moment is always True). (default: True)")), ToolInput(tag="in_keep_self_circles", input_type=Boolean(optional=True), prefix="--keepSelfCircles", doc=InputDocumentation(doc="If set, outward facing reads without any restriction\nfragment (self circles) are kept. They will be counted\nand shown in the QC plots. (default: False)")), ToolInput(tag="in_min_mapping_quality", input_type=Int(optional=True), prefix="--minMappingQuality", doc=InputDocumentation(doc="minimum mapping quality for reads to be accepted.\nBecause the restriction enzyme site could be located\non top of the read, this may reduce the reported\nquality of the read. Thus, this parameter may be\nadjusted if too many low quality (but otherwise\nperfectly valid Hi-C reads) are found. A good strategy\nis to make a test run (using the --doTestRun), then\nchecking the results to see if too many low quality\nreads are present and then using the bam file\ngenerated to check if those low quality reads are\ncaused by the read not being mapped entirely (Default:\n15).")), ToolInput(tag="in_threads", input_type=Int(optional=True), prefix="--threads", doc=InputDocumentation(doc="Number of threads. Using the python multiprocessing\nmodule. One master process which is used to read the\ninput file into the buffer and one process which is\nmerging the output bam files of the processes into one\noutput bam file. All other threads do the actual\ncomputation. Minimum value for the '--thread'\nparameter is 2. The usage of 8 threads is optimal if\nyou have an HDD. A higher number of threads is only\nuseful if you have a fast SSD. Have in mind that the\nperformance of hicBuildMatrix is influenced by the\nnumber of threads, the speed of your hard drive and\nthe inputBufferSize. To clarify: the performance with\na higher thread number is not negative influenced but\nnot positive too. With a slow HDD and a high number of\nthreads many threads will do nothing most of the time\n(Default: 4).")), ToolInput(tag="in_input_buffer_size", input_type=Int(optional=True), prefix="--inputBufferSize", doc=InputDocumentation(doc="Size of the input buffer of each thread. 400,000 read\npairs per input file per thread is the default value.\nReduce this value to decrease memory usage. (default:\n400000)")), ToolInput(tag="in_do_test_run", input_type=Boolean(optional=True), prefix="--doTestRun", doc=InputDocumentation(doc="A test run is useful to test the quality of a Hi-C\nexperiment quickly. It works by testing only 1,000,000\nreads. This option is useful to get an idea of quality\ncontrol values like inter-chromosomal interactions,\nduplication rates etc. (default: False)")), ToolInput(tag="in_do_test_run_lines", input_type=Int(optional=True), prefix="--doTestRunLines", doc=InputDocumentation(doc="Number of lines to consider for the qc test run\n(Default: 1000000).")), ToolInput(tag="in_skip_duplication_check", input_type=Boolean(optional=True), prefix="--skipDuplicationCheck", doc=InputDocumentation(doc="Identification of duplicated read pairs is memory\nconsuming. Thus, in case of memory errors this check\ncan be skipped. However, consider running a\n`--doTestRun` first to get an estimation of the\nduplicated reads. (default: False)")), ToolInput(tag="in_chromosome_sizes", input_type=Int(optional=True), prefix="--chromosomeSizes", doc=InputDocumentation(doc="file, -cs txt file\nFile with the chromosome sizes for your genome. A tab-\ndelimited two column layout 'chr_name size' is\nexpectedUsually the sizes can be determined from the\nSAM/BAM input files, however, for cHi-C or scHi-C it\ncan be that at the start or end no data is present.\nPlease consider that this option causes that only\nreads are considered which are on the listed\nchromosomes.Use this option to guarantee fixed sizes.\nAn example file is available via UCSC: http://hgdownlo\nad.soe.ucsc.edu/goldenPath/dm3/bigZips/dm3.chrom.sizes\n(default: None)")), ToolInput(tag="in_sam", input_type=String(), position=0, doc=InputDocumentation(doc="")), ToolInput(tag="in_files", input_type=String(), position=1, doc=InputDocumentation(doc="")), ToolInput(tag="in_two", input_type=String(), position=2, doc=InputDocumentation(doc=""))], outputs=[ToolOutput(tag="out_out_filename", output_type=File(optional=True), selector=InputSelector(input_to_select="in_out_filename", type_hint=File()), doc=OutputDocumentation(doc="Output file name for the Hi-C matrix. (default: None)")), ToolOutput(tag="out_out_bam", output_type=File(optional=True), selector=InputSelector(input_to_select="in_out_bam", type_hint=File()), doc=OutputDocumentation(doc="file, -b bam file\nOutput bam file to process. Optional parameter. A bam\nfile containing all valid Hi-C reads can be created\nusing this option. This bam file could be useful to\ninspect the distribution of valid Hi-C reads pairs or\nfor other downstream analyses, but is not used by any\nHiCExplorer tool. Computation will be significantly\nlonger if this option is set. (default: None)"))], container="quay.io/biocontainers/hicexplorer:3.6--py_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Hicbuildmatrix_V0_1_0().translate("wdl")

