from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Int, Directory, Float, Boolean

Deblur_Workflow_V0_1_0 = CommandToolBuilder(tool="deblur_workflow", base_command=["deblur", "workflow"], inputs=[ToolInput(tag="in_seqs_fp", input_type=File(optional=True), prefix="--seqs-fp", doc=InputDocumentation(doc="Either a Demultiplexed FASTA or FASTQ file\nincluding all samples, or a directory of\nper-sample FASTA or FASTQ files. Gzip'd\nfiles are acceptable (e.g., .fastq.gz).\n[required]")), ToolInput(tag="in_output_dir", input_type=File(optional=True), prefix="--output-dir", doc=InputDocumentation(doc="Directory path to store output including\nBIOM table  [required]")), ToolInput(tag="in_trim_length", input_type=Int(optional=True), prefix="--trim-length", doc=InputDocumentation(doc="Sequence trim length. All reads shorter than\nthe trim-length will be discarded. A value\nof -1 can be specified to skip trimming;\nthis assumes all sequences have an identical\nlength.  [required]")), ToolInput(tag="in_left_trim_length", input_type=Int(optional=True), prefix="--left-trim-length", doc=InputDocumentation(doc="Trim the first N bases from every sequence.\nA value of 0 disables this trim.  [default:\n0]")), ToolInput(tag="in_pos_ref_fp", input_type=File(optional=True), prefix="--pos-ref-fp", doc=InputDocumentation(doc="Positive reference filtering database. Keep\nall sequences permissively aligning to any\nsequence in this FASTA file; these results\nare stored in the reference-hit.biom output\nfile. This defaults to the Greengenes 13_8\nOTUs at 88% identity. An e-value threshold\nof 10 is used with SortMeRNA against the\nreference. For multiple databases, specify\nthe argument multiple times, e.g., --pos-\nref-fp db1.fa --pos-ref-fp db2.fa")), ToolInput(tag="in_pos_ref_db_fp", input_type=File(optional=True), prefix="--pos-ref-db-fp", doc=InputDocumentation(doc="An indexed version of the positive filtering\ndatabase. This can be useful to avoid\nincurring the expense of reindexing on every\nrun. If not supplied, deblur will index the\ndatabase. For multiple databases, the order\nmust follow that of --pos-ref-fp, for\nexample, --pos-ref-db-fp db1.idx --pos-ref-\ndb-fp db2.idx ..")), ToolInput(tag="in_neg_ref_fp", input_type=File(optional=True), prefix="--neg-ref-fp", doc=InputDocumentation(doc="Negative (artifacts) filtering database.\nDrop all sequences which align to any record\nin this FASTA file. This defaults to a\ndatabase composed of multiple PhiX genomes\nand known Illumina adapters. For multiple\ndatabases, specify the argument mutiple\ntimes, e.g., --neg-ref-fp db1.fa --neg-ref-\nfp db2.fa")), ToolInput(tag="in_neg_ref_db_fp", input_type=File(optional=True), prefix="--neg-ref-db-fp", doc=InputDocumentation(doc="An indexed version of the negative filtering\ndatabase. If not supplied, deblur will index\nthe database.For multiple databases, the\norder must follow that of --neg-ref-fp, for\nexample, --neg-ref-db-fp db1.idx --neg-ref-\ndb-fp db2.idx ..")), ToolInput(tag="in_overwrite", input_type=Directory(optional=True), prefix="--overwrite", doc=InputDocumentation(doc="Overwrite output directory if exists.\n[default: False]")), ToolInput(tag="in_mean_error", input_type=Float(optional=True), prefix="--mean-error", doc=InputDocumentation(doc="The mean per nucleotide error rate, used for\noriginal sequence estimate. If not passed\ntypical illumina error rate is used.\n[default: 0.005]")), ToolInput(tag="in_error_dist", input_type=Int(optional=True), prefix="--error-dist", doc=InputDocumentation(doc="A comma separated list of error\nprobabilities for each Hamming distance. The\nlength of the list determines the number of\nHamming distances taken into account.\n[default: 1, 0.06, 0.02, 0.02, 0.01, 0.005,\n0.005, 0.005, 0.001, 0.001, 0.001, 0.0005]")), ToolInput(tag="in_in_del_prob", input_type=Float(optional=True), prefix="--indel-prob", doc=InputDocumentation(doc="Insertion/deletion (indel) probability. This\nprobability consistent for multiple indels;\nthere is not an added elongation penalty.\n[default: 0.01]")), ToolInput(tag="in_in_del_max", input_type=Int(optional=True), prefix="--indel-max", doc=InputDocumentation(doc="Maximum number of allowed indels.  [default:\n3]")), ToolInput(tag="in_min_reads", input_type=Int(optional=True), prefix="--min-reads", doc=InputDocumentation(doc="Keep only the sequences which appear at\nleast min-reads study wide (as opposed to\nper-sample).  [default: 10]")), ToolInput(tag="in_min_size", input_type=Int(optional=True), prefix="--min-size", doc=InputDocumentation(doc="Keep only sequences which appear at least\nmin-size times per-sample (as opposed to\nstudy wide).  [default: 2]")), ToolInput(tag="in_threads_per_sample", input_type=Int(optional=True), prefix="--threads-per-sample", doc=InputDocumentation(doc="Number of threads to use per sample (0 to\nuse all)  [default: 1]")), ToolInput(tag="in_keep_tmp_files", input_type=Boolean(optional=True), prefix="--keep-tmp-files", doc=InputDocumentation(doc="Keep temporary files (useful for debugging)\n[default: False]")), ToolInput(tag="in_log_level", input_type=Int(optional=True), prefix="--log-level", doc=InputDocumentation(doc="RANGE       Level of messages for log file(range 1-debug\nto 5-critical  [default: 2]")), ToolInput(tag="in_log_file", input_type=File(optional=True), prefix="--log-file", doc=InputDocumentation(doc="log file name  [default: deblur.log]")), ToolInput(tag="in_jobs_to_start", input_type=Int(optional=True), prefix="--jobs-to-start", doc=InputDocumentation(doc="Number of jobs to start (if to run in\nparallel)  [default: 1]")), ToolInput(tag="in_is_worker_thread", input_type=Boolean(optional=True), prefix="--is-worker-thread", doc=InputDocumentation(doc="indicates this is not the main deblur\nprocess (used by the parallel deblur - do\nnot use manually)  [default: False]"))], outputs=[ToolOutput(tag="out_output_dir", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output_dir", type_hint=File()), doc=OutputDocumentation(doc="Directory path to store output including\nBIOM table  [required]")), ToolOutput(tag="out_pos_ref_fp", output_type=File(optional=True), selector=InputSelector(input_to_select="in_pos_ref_fp", type_hint=File()), doc=OutputDocumentation(doc="Positive reference filtering database. Keep\nall sequences permissively aligning to any\nsequence in this FASTA file; these results\nare stored in the reference-hit.biom output\nfile. This defaults to the Greengenes 13_8\nOTUs at 88% identity. An e-value threshold\nof 10 is used with SortMeRNA against the\nreference. For multiple databases, specify\nthe argument multiple times, e.g., --pos-\nref-fp db1.fa --pos-ref-fp db2.fa")), ToolOutput(tag="out_overwrite", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_overwrite", type_hint=File()), doc=OutputDocumentation(doc="Overwrite output directory if exists.\n[default: False]"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Deblur_Workflow_V0_1_0().translate("wdl", allow_empty_container=True)

