from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Int, Boolean, Float, String

Indelfixer_V0_1_0 = CommandToolBuilder(tool="InDelFixer", base_command=["InDelFixer"], inputs=[ToolInput(tag="in_path_output_directory", input_type=File(optional=True), prefix="-o", doc=InputDocumentation(doc=": Path to the output directory (default: current directory).")), ToolInput(tag="in_path_ngs_input", input_type=File(optional=True), prefix="-i", doc=InputDocumentation(doc=": Path to the NGS input file (FASTA, FASTQ or SFF format) [REQUIRED].")), ToolInput(tag="in_ir", input_type=File(optional=True), prefix="-ir", doc=InputDocumentation(doc=": Path to the second paired end file (FASTQ) [ONLY REQUIRED if first file is also fastq].")), ToolInput(tag="in_path_reference_genomes", input_type=File(optional=True), prefix="-g", doc=InputDocumentation(doc=": Path to the reference genomes file (FASTA format) [REQUIRED].")), ToolInput(tag="in_region_reference_genome", input_type=Int(optional=True), prefix="-r", doc=InputDocumentation(doc=": Region on the reference genome (i.e. 342-944).")), ToolInput(tag="in_kmer_size_default", input_type=Int(optional=True), prefix="-k", doc=InputDocumentation(doc=": Kmer size (default 10).")), ToolInput(tag="in_kmer_offset_default", input_type=Int(optional=True), prefix="-v", doc=InputDocumentation(doc=": Kmer offset (default 2).")), ToolInput(tag="in_cut", input_type=Int(optional=True), prefix="-cut", doc=InputDocumentation(doc=": Cut given number of bases (primer) at 5' and 3' (default 0).")), ToolInput(tag="in_refine", input_type=Int(optional=True), prefix="-refine", doc=InputDocumentation(doc=": Computes a consensus sequence from alignment and re-aligns against that.\nRefinement is repeated as many times as specified.")), ToolInput(tag="in_mcc", input_type=Int(optional=True), prefix="-mcc", doc=InputDocumentation(doc=": Minimal coverage to replace a reference base in the consensus (default 1).")), ToolInput(tag="in_rm_del", input_type=Boolean(optional=True), prefix="-rmDel", doc=InputDocumentation(doc=": Removes conserved gaps from consensus sequence during refinement.")), ToolInput(tag="in_sensitive", input_type=Boolean(optional=True), prefix="-sensitive", doc=InputDocumentation(doc=": More sensitive but slower alignment.")), ToolInput(tag="in_fix", input_type=Boolean(optional=True), prefix="-fix", doc=InputDocumentation(doc=": Fill frame-shift causing deletions with consensus sequence.")), ToolInput(tag="in_no_hashing", input_type=Boolean(optional=True), prefix="-noHashing", doc=InputDocumentation(doc=": No fast kmer-matching to find approximate mapping region. Please use with PacBio data!")), ToolInput(tag="in_realign", input_type=Float(optional=True), prefix="-realign", doc=InputDocumentation(doc=": Reads are aligned to the whole reference sequence,\nif the relative mismatch rate is above the given threshold (default 0.1).")), ToolInput(tag="in_minimal_readlength_prior", input_type=Int(optional=True), prefix="-l", doc=InputDocumentation(doc=": Minimal read-length prior alignment (default 0).")), ToolInput(tag="in_la", input_type=Int(optional=True), prefix="-la", doc=InputDocumentation(doc=": Minimal read-length after alignment (default 0).")), ToolInput(tag="in_ins", input_type=Float(optional=True), prefix="-ins", doc=InputDocumentation(doc=": The maximum percentage of insertions allowed [range 0.0 - 1.0] (default 1.0).")), ToolInput(tag="in_del", input_type=Float(optional=True), prefix="-del", doc=InputDocumentation(doc=": The maximum percentage of deletions allowed [range 0.0 - 1.0] (default 1.0).")), ToolInput(tag="in_sub", input_type=Float(optional=True), prefix="-sub", doc=InputDocumentation(doc=": The maximum percentage of substitutions allowed [range 0.0 - 1.0] (default 0.5).")), ToolInput(tag="in_max_del", input_type=Int(optional=True), prefix="-maxDel", doc=InputDocumentation(doc=": The maximum number of consecutive deletions allowed (default no filtering).")), ToolInput(tag="in_minimal_average_phred", input_type=Int(optional=True), prefix="-q", doc=InputDocumentation(doc=": Minimal average Phred score of the aligned read (default 20).")), ToolInput(tag="in_gop", input_type=Boolean(optional=True), prefix="-gop", doc=InputDocumentation(doc=": Gap opening costs for Smith-Waterman (default 30).")), ToolInput(tag="in_gex", input_type=Boolean(optional=True), prefix="-gex", doc=InputDocumentation(doc=": Gap extension costs for Smith-Waterman (default 3).")), ToolInput(tag="in_four_five_four", input_type=Boolean(optional=True), prefix="-454", doc=InputDocumentation(doc=": 10 open / 1 extend")), ToolInput(tag="in_var_25", input_type=Boolean(optional=True), prefix="-illumina", doc=InputDocumentation(doc=": 30 open / 3 extend")), ToolInput(tag="in_var_26", input_type=Boolean(optional=True), prefix="-pacbio", doc=InputDocumentation(doc=": 5 open / 3 extend")), ToolInput(tag="in_jar", input_type=String(optional=True), prefix="-jar", doc=InputDocumentation(doc="")), ToolInput(tag="in_java_jar_indelfixerjar_referencegenomesfasta", input_type=String(), position=0, doc=InputDocumentation(doc=": java -jar InDelFixer.jar -i libCase102.ccs.fastq -g referenceGenomes.fasta -noHashing -pacbio")), ToolInput(tag="in_java_jar_indelfixerjar_g", input_type=String(), position=1, doc=InputDocumentation(doc=": java -jar InDelFixer.jar -i libCase102_R1.fastq -ir libCase102_R2.fastq -g referenceGenomes.fasta -illumina"))], outputs=[ToolOutput(tag="out_path_output_directory", output_type=File(optional=True), selector=InputSelector(input_to_select="in_path_output_directory", type_hint=File()), doc=OutputDocumentation(doc=": Path to the output directory (default: current directory)."))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Indelfixer_V0_1_0().translate("wdl", allow_empty_container=True)

