from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Boolean, Array, Int, String, Float

Mimodd_Map_V0_1_0 = CommandToolBuilder(tool="mimodd_map", base_command=["mimodd", "map"], inputs=[ToolInput(tag="in_ofile", input_type=File(optional=True), prefix="--ofile", doc=InputDocumentation(doc="redirect the binned variant counts to this file\n(default: stdout).")), ToolInput(tag="in_quiet", input_type=Boolean(optional=True), prefix="--quiet", doc=InputDocumentation(doc="suppress warning messages about plotting problems.")), ToolInput(tag="in_bin_sizes", input_type=Array(t=Int(), optional=True), prefix="--bin-sizes", doc=InputDocumentation(doc="list of bin sizes to be used for histogram plots and\nlinkage reports (default: 1Mb and 500Kb)")), ToolInput(tag="in_mapping_sample", input_type=File(optional=True), prefix="--mapping-sample", doc=InputDocumentation(doc="name of the sample (as appearing in the input vcf\nfile) for which variants should be mapped (cannot be\nused in SVD mode)")), ToolInput(tag="in_related_parent", input_type=String(optional=True), prefix="--related-parent", doc=InputDocumentation(doc="name of the sample to provide related parent strain\n(mutagenesis strain) variants for the analysis in\nVariant Allele Frequency (VAF) mode.")), ToolInput(tag="in_unrelated_parent", input_type=String(optional=True), prefix="--unrelated-parent", doc=InputDocumentation(doc="name of the sample to provide unrelated parent strain\n(mapping strain) variants for the analysis in Variant\nAllele Frequency (VAF) mode.")), ToolInput(tag="in_infer_missing", input_type=Boolean(optional=True), prefix="--infer-missing", doc=InputDocumentation(doc="if variant data for either the related or the\nunrelated parent strain is not provided, the tool can\ntry to infer the alleles present in that parent from\nthe allele spectrum found in the mapping sample. Use\nwith caution on carefully filtered variant lists only!")), ToolInput(tag="in_contrast_sample", input_type=File(optional=True), prefix="--contrast-sample", doc=InputDocumentation(doc="name of the sample (as appearing in the input vcf\nfile) that provides the contrast for the mapping\nsample")), ToolInput(tag="in_text_file", input_type=File(optional=True), prefix="--text-file", doc=InputDocumentation(doc="generate text-based output for every variant position\nand save it to the specified file. This file can be\nused as input during later runs of the tool, which\nwill speed up replotting.")), ToolInput(tag="in_seq_info", input_type=File(optional=True), prefix="--seqinfo", doc=InputDocumentation(doc="if an input file does not specify required contig\ninformation (chromosome names and sizes), extract this\ninformation from SEQINFO_FILE. This file can be a\nreference genome file in fasta format or a CloudMap-\nstyle sequence dictionary file. This option is never\nnecessary with MiModD-generated input, but can be\nuseful with input files generated by third-party tools\nthat do not record contig information in their output.")), ToolInput(tag="in_cloud_map", input_type=Boolean(optional=True), prefix="--cloudmap", doc=InputDocumentation(doc="generate valid input for the original CloudMap Mapping\ntools and save it to the text output file specified by\nthe '-t' option. This option can only be used in 'SVD'\nor 'VAF' mode, which have an equivalent in CloudMap.")), ToolInput(tag="in_plot_file", input_type=File(optional=True), prefix="--plot-file", doc=InputDocumentation(doc="generate graphical output and store it in the given\nfile (default: no graphical output)")), ToolInput(tag="in_fit_width", input_type=Boolean(optional=True), prefix="--fit-width", doc=InputDocumentation(doc="do not autoscale x-axes to size of largest contig")), ToolInput(tag="in_no_scatter", input_type=Boolean(optional=True), prefix="--no-scatter", doc=InputDocumentation(doc="do not produce scatter plots of observed segregation\nrates; just plot histograms")), ToolInput(tag="in_loess_span", input_type=Float(optional=True), prefix="--loess-span", doc=InputDocumentation(doc="span parameter for the Loess regression line through\nthe linkage data (default: 0.1, specify 0 to skip\ncalculation)")), ToolInput(tag="in_ylim_scatter", input_type=Float(optional=True), prefix="--ylim-scatter", doc=InputDocumentation(doc="upper limit for scatter plot y-axis (default: 1)")), ToolInput(tag="in_points_colors", input_type=Array(t=String(), optional=True), prefix="--points-colors", doc=InputDocumentation(doc="color(s) for scatter plot data points")), ToolInput(tag="in_loess_colors", input_type=Array(t=String(), optional=True), prefix="--loess-colors", doc=InputDocumentation(doc="color(s) for regression line(s) through scatter plot\ndata")), ToolInput(tag="in_no_hist", input_type=Boolean(optional=True), prefix="--no-hist", doc=InputDocumentation(doc="do not produce linkage histogram plots; only generate\nscatter plots")), ToolInput(tag="in_no_kde", input_type=Boolean(optional=True), prefix="--no-kde", doc=InputDocumentation(doc="do not add kernel density estimate lines to histogram")), ToolInput(tag="in_hist_colors", input_type=Array(t=String(), optional=True), prefix="--hist-colors", doc=InputDocumentation(doc="list of colors to be used for plotting histogram bars\nof different width (default: darkgrey and red)\n")), ToolInput(tag="in_analysis_mode", input_type=String(), position=0, doc=InputDocumentation(doc="specify 'SVD' for Simple Variant Density analysis,\n'VAF' for Variant Allele Frequency analysis or 'VAC'\nfor Variant Allele Contrast analysis.")), ToolInput(tag="in_input_file", input_type=String(), position=1, doc=InputDocumentation(doc="valid input files are VCF files or per-variant report\nfiles (as generated by this tool with the '-t' option\nor by the CloudMap Hawaiian Variant Density Mapping\ntool).")), ToolInput(tag="in_plots", input_type=String(), position=0, doc=InputDocumentation(doc="--ylim-hist FLOAT     upper limit for histogram plot y-axis (default: auto)"))], outputs=[ToolOutput(tag="out_plot_file", output_type=File(optional=True), selector=InputSelector(input_to_select="in_plot_file", type_hint=File()), doc=OutputDocumentation(doc="generate graphical output and store it in the given\nfile (default: no graphical output)"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Mimodd_Map_V0_1_0().translate("wdl", allow_empty_container=True)

