class: CommandLineTool
id: rsem_control_fdr.cwl
inputs:
- id: in_hard_threshold
  doc: "Use hard threshold method to control FDR. If this option is set,\nonly those\
    \ genes/transcripts with their PPDE >= 1 - fdr_rate are\ncalled as DE. (Default:\
    \ on)"
  type: boolean?
  inputBinding:
    prefix: --hard-threshold
- id: in_soft_threshold
  doc: "Use soft threshold method to control FDR. If this option is set,\nthis program\
    \ will try to report as many genes/transcripts as\npossible, as long as their\
    \ average PPDE >= 1 - fdr_rate. This option\nis equivalent to use EBSeq's 'crit_fun'\
    \ for FDR control. (Default:\noff)"
  type: boolean?
  inputBinding:
    prefix: --soft-threshold
- id: in_input_file
  doc: "This should be the main result file generated by 'rsem-run-ebseq',\nwhich\
    \ contains all genes/transcripts and their associated\nstatistics."
  type: string
  inputBinding:
    position: 0
- id: in_fdr_rate
  doc: The desire false discovery rate (FDR).
  type: string
  inputBinding:
    position: 1
- id: in_output_file
  doc: "This file is a subset of the 'input_file'. It only contains the\ngenes/transcripts\
    \ called as differentially expressed (DE). When more\nthan 2 conditions exist,\
    \ DE is defined as not all conditions are\nequally expressed. Because statistical\
    \ significance does not\nnecessarily mean biological significance, users should\
    \ also refer to\nthe fold changes to decide which genes/transcripts are biologically\n\
    significant. When more than two conditions exist, this file will not\ncontain\
    \ fold change information and users need to calculate it from\n'input_file.condmeans'\
    \ by themselves."
  type: string
  inputBinding:
    position: 2
outputs:
- id: out_stdout
  doc: Standard output stream
  type: stdout
hints: []
cwlVersion: v1.1
baseCommand:
- rsem-control-fdr
