from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, Int, Float, File

Plass_Assemble_V0_1_0 = CommandToolBuilder(tool="plass_assemble", base_command=["plass", "assemble"], inputs=[ToolInput(tag="in_add_self_matches", input_type=Boolean(optional=True), prefix="--add-self-matches", doc=InputDocumentation(doc="artificially add entries of queries with themselves (for clustering)")), ToolInput(tag="in_alph_size", input_type=Int(optional=True), prefix="--alph-size", doc=InputDocumentation(doc="alphabet size (range 2-21) [13]")), ToolInput(tag="in_mask", input_type=Int(optional=True), prefix="--mask", doc=InputDocumentation(doc="mask sequences in k-mer stage 0: w/o low complexity masking, 1: with low complexity masking [0]")), ToolInput(tag="in_mask_lower_case", input_type=Int(optional=True), prefix="--mask-lower-case", doc=InputDocumentation(doc="lowercase letters will be excluded from k-mer search 0: include region, 1: exclude region [0]")), ToolInput(tag="in_kmer_size_set", input_type=Int(optional=True), prefix="-k", doc=InputDocumentation(doc="k-mer size in the range (0: set automatically to optimum) [14]")), ToolInput(tag="in_split_memory_limit", input_type=Int(optional=True), prefix="--split-memory-limit", doc=InputDocumentation(doc="Set max memory per split. E.g. 800B, 5K, 10M, 1G. Defaults (0) to all available system memory. [0]")), ToolInput(tag="in_extend_sequences_is", input_type=Float(optional=True), prefix="-e", doc=InputDocumentation(doc="Extend sequences if the E-value is below [0.0, inf] [0.000]")), ToolInput(tag="in_list_matches_fraction", input_type=Float(optional=True), prefix="-c", doc=InputDocumentation(doc="list matches above this fraction of aligned (covered) residues (see --cov-mode) [0.000]")), ToolInput(tag="in_add_string_convert", input_type=Boolean(optional=True), prefix="-a", doc=InputDocumentation(doc="add backtrace string (convert to alignments with mmseqs convertalis utility)")), ToolInput(tag="in_cov_mode", input_type=Int(optional=True), prefix="--cov-mode", doc=InputDocumentation(doc="0: coverage of query and target, 1: coverage of target, 2: coverage of query 3: target seq. length needs to be at least x% of query length, 4: query seq. length needs to be at least x% of target length 5: short seq. needs to be at least x% of the other seq. length [0]")), ToolInput(tag="in_min_seq_id", input_type=Float(optional=True), prefix="--min-seq-id", doc=InputDocumentation(doc="Overlap sequence identity threshold [0.0, 1.0] [0.900]")), ToolInput(tag="in_min_aln_len", input_type=Int(optional=True), prefix="--min-aln-len", doc=InputDocumentation(doc="minimum alignment length (range 0-INT_MAX) [0]")), ToolInput(tag="in_seq_id_mode", input_type=Int(optional=True), prefix="--seq-id-mode", doc=InputDocumentation(doc="0: alignment length 1: shorter, 2: longer sequence [0]")), ToolInput(tag="in_km_er_per_seq", input_type=Int(optional=True), prefix="--kmer-per-seq", doc=InputDocumentation(doc="kmer per sequence [60]")), ToolInput(tag="in_adjust_km_er_len", input_type=Boolean(optional=True), prefix="--adjust-kmer-len", doc=InputDocumentation(doc="adjust k-mer length based on specificity (only for nucleotides)")), ToolInput(tag="in_hash_shift", input_type=Int(optional=True), prefix="--hash-shift", doc=InputDocumentation(doc="Shift k-mer hash [5]")), ToolInput(tag="in_include_only_extendable", input_type=Int(optional=True), prefix="--include-only-extendable", doc=InputDocumentation(doc="Include only extendable [1, set to 0 to disable]")), ToolInput(tag="in_ignore_multi_km_er", input_type=Int(optional=True), prefix="--ignore-multi-kmer", doc=InputDocumentation(doc="Skip kmers occuring multiple times (>=2) [1, set to 0 to disable]")), ToolInput(tag="in_num_iterations", input_type=Int(optional=True), prefix="--num-iterations", doc=InputDocumentation(doc="Number of assembly iterations [1, inf] [12]")), ToolInput(tag="in_re_score_mode", input_type=Int(optional=True), prefix="--rescore-mode", doc=InputDocumentation(doc="Rescore diagonal with: 0: Hamming distance, 1: local alignment (score only), 2: local alignment, 3: global alignment or 4: longest alignment fullfilling window quality criterion [3]")), ToolInput(tag="in_wrapped_scoring", input_type=Boolean(optional=True), prefix="--wrapped-scoring", doc=InputDocumentation(doc="Double the (nucleotide) query sequence during the scoring process to allow wrapped diagonal scoring around end and start")), ToolInput(tag="in_min_length", input_type=Int(optional=True), prefix="--min-length", doc=InputDocumentation(doc="minimum codon number in open reading frames [45]")), ToolInput(tag="in_max_length", input_type=Int(optional=True), prefix="--max-length", doc=InputDocumentation(doc="maximum codon number in open reading frames [32734]")), ToolInput(tag="in_max_gaps", input_type=Int(optional=True), prefix="--max-gaps", doc=InputDocumentation(doc="maximum number of codons with gaps or unknown residues before an open reading frame is rejected [2147483647]")), ToolInput(tag="in_contig_start_mode", input_type=Int(optional=True), prefix="--contig-start-mode", doc=InputDocumentation(doc="Contig start can be 0: incomplete, 1: complete, 2: both [2]")), ToolInput(tag="in_contig_end_mode", input_type=Int(optional=True), prefix="--contig-end-mode", doc=InputDocumentation(doc="Contig end can be 0: incomplete, 1: complete, 2: both  [2]")), ToolInput(tag="in_orf_start_mode", input_type=Int(optional=True), prefix="--orf-start-mode", doc=InputDocumentation(doc="Orf fragment can be 0: from start to stop, 1: from any to stop, 2: from last encountered start to stop (no start in the middle) [1]")), ToolInput(tag="in_forward_frames", input_type=Int(optional=True), prefix="--forward-frames", doc=InputDocumentation(doc="comma-seperated list of ORF frames on the forward strand to be extracted [1,2,3]")), ToolInput(tag="in_reverse_frames", input_type=Int(optional=True), prefix="--reverse-frames", doc=InputDocumentation(doc="comma-seperated list of ORF frames on the reverse strand to be extracted [1,2,3]")), ToolInput(tag="in_translation_table", input_type=Int(optional=True), prefix="--translation-table", doc=InputDocumentation(doc="1) CANONICAL, 2) VERT_MITOCHONDRIAL, 3) YEAST_MITOCHONDRIAL, 4) MOLD_MITOCHONDRIAL, 5) INVERT_MITOCHONDRIAL, 6) CILIATE, 9) FLATWORM_MITOCHONDRIAL, 10) EUPLOTID, 11) PROKARYOTE, 12) ALT_YEAST, 13) ASCIDIAN_MITOCHONDRIAL, 14) ALT_FLATWORM_MITOCHONDRIAL, 15) BLEPHARISMA, 16) CHLOROPHYCEAN_MITOCHONDRIAL, 21) TREMATODE_MITOCHONDRIAL, 22) SCENEDESMUS_MITOCHONDRIAL, 23) THRAUSTOCHYTRIUM_MITOCHONDRIAL, 24) PTEROBRANCHIA_MITOCHONDRIAL, 25) GRACILIBACTERIA, 26) PACHYSOLEN, 27) KARYORELICT, 28) CONDYLOSTOMA, 29) MESODINIUM, 30) PERTRICH, 31) BLASTOCRITHIDIA [1]")), ToolInput(tag="in_translate", input_type=Int(optional=True), prefix="--translate", doc=InputDocumentation(doc="translate ORF to amino acid [0]")), ToolInput(tag="in_use_all_table_starts", input_type=Boolean(optional=True), prefix="--use-all-table-starts", doc=InputDocumentation(doc="use all alteratives for a start codon in the genetic table, if false - only ATG (AUG)")), ToolInput(tag="in_id_offset", input_type=Int(optional=True), prefix="--id-offset", doc=InputDocumentation(doc="numeric ids in index file are offset by this value  [0]")), ToolInput(tag="in_protein_filter_threshold", input_type=Float(optional=True), prefix="--protein-filter-threshold", doc=InputDocumentation(doc="filter proteins lower than threshold [0.0,1.0] [0.200]")), ToolInput(tag="in_filter_proteins", input_type=Int(optional=True), prefix="--filter-proteins", doc=InputDocumentation(doc="filter proteins by a neural network [0,1] [1]")), ToolInput(tag="in_delete_tmp_inc", input_type=Int(optional=True), prefix="--delete-tmp-inc", doc=InputDocumentation(doc="delete temporary files incremental [0,1] [1]")), ToolInput(tag="in_remove_tmp_files", input_type=Boolean(optional=True), prefix="--remove-tmp-files", doc=InputDocumentation(doc="Delete temporary files")), ToolInput(tag="in_sub_mat", input_type=File(optional=True), prefix="--sub-mat", doc=InputDocumentation(doc="amino acid substitution matrix file [nucl:nucleotide.out,aa:blosum62.out]")), ToolInput(tag="in_db_load_mode", input_type=Int(optional=True), prefix="--db-load-mode", doc=InputDocumentation(doc="Database preload mode 0: auto, 1: fread, 2: mmap, 3: mmap+touch [0]")), ToolInput(tag="in_threads", input_type=Int(optional=True), prefix="--threads", doc=InputDocumentation(doc="number of cores used for the computation (uses all cores by default) [8]")), ToolInput(tag="in_compressed", input_type=Int(optional=True), prefix="--compressed", doc=InputDocumentation(doc="write results in compressed format [0]")), ToolInput(tag="in_verbosity_level_nothing", input_type=Int(optional=True), prefix="-v", doc=InputDocumentation(doc="verbosity level: 0=nothing, 1: +errors, 2: +warnings, 3: +info [3]")), ToolInput(tag="in_max_seq_len", input_type=Int(optional=True), prefix="--max-seq-len", doc=InputDocumentation(doc="maximum sequence length (range 1-32768]) [65535]")), ToolInput(tag="in_mpi_runner", input_type=Int(optional=True), prefix="--mpi-runner", doc=InputDocumentation(doc="Use MPI on compute grid with this MPI command (e.g. 'mpirun -np 42') []")), ToolInput(tag="in_filter_hits", input_type=Boolean(optional=True), prefix="--filter-hits", doc=InputDocumentation(doc="filter hits by seq.id. and coverage")), ToolInput(tag="in_sort_results", input_type=Int(optional=True), prefix="--sort-results", doc=InputDocumentation(doc="Sort results: 0: no sorting, 1: sort by evalue (Alignment) or seq.id. (Hamming) [0]")), ToolInput(tag="in_km_er_per_seq_scale", input_type=Float(optional=True), prefix="--kmer-per-seq-scale", doc=InputDocumentation(doc="scale kmer per sequence based on sequence length as kmer-per-seq val + scale x seqlen [0.000]")), ToolInput(tag="in_create_lookup", input_type=Int(optional=True), prefix="--create-lookup", doc=InputDocumentation(doc="Create database lookup file (can be very large) [0]"))], outputs=[ToolOutput(tag="out_sub_mat", output_type=File(optional=True), selector=InputSelector(input_to_select="in_sub_mat", type_hint=File()), doc=OutputDocumentation(doc="amino acid substitution matrix file [nucl:nucleotide.out,aa:blosum62.out]"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Plass_Assemble_V0_1_0().translate("wdl", allow_empty_container=True)

