from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Directory, Int, Boolean, Float

Scapp_V0_1_0 = CommandToolBuilder(tool="scapp", base_command=["scapp"], inputs=[ToolInput(tag="in_graph", input_type=File(optional=True), prefix="--graph", doc=InputDocumentation(doc="Assembly graph FASTG file to process")), ToolInput(tag="in_output_dir", input_type=Directory(optional=True), prefix="--output_dir", doc=InputDocumentation(doc="Output directory")), ToolInput(tag="in_max_k", input_type=Int(optional=True), prefix="--max_k", doc=InputDocumentation(doc="Integer reflecting maximum k value used by the\nassembler")), ToolInput(tag="in_min_length", input_type=Int(optional=True), prefix="--min_length", doc=InputDocumentation(doc="Minimum length required for reporting [default: 1000]")), ToolInput(tag="in_max_cv", input_type=Int(optional=True), prefix="--max_CV", doc=InputDocumentation(doc="Coefficient of variation used for pre-selection\n[default: 0.5, higher--> less restrictive]")), ToolInput(tag="in_num_processes", input_type=Int(optional=True), prefix="--num_processes", doc=InputDocumentation(doc="Number of processes to use")), ToolInput(tag="in_use_scores", input_type=Boolean(optional=True), prefix="--use_scores", doc=InputDocumentation(doc="Boolean flag of whether to use sequence classification\nscores in plasmid assembly")), ToolInput(tag="in_use_gene_hits", input_type=Boolean(optional=True), prefix="--use_gene_hits", doc=InputDocumentation(doc="Boolean flag of whether to use plasmid-specific gene\nhits in plasmid assembly")), ToolInput(tag="in_bam", input_type=File(optional=True), prefix="--bam", doc=InputDocumentation(doc="BAM file resulting from aligning reads to contigs\nfile, filtering for best matches")), ToolInput(tag="in_reads_one", input_type=Int(optional=True), prefix="--reads1", doc=InputDocumentation(doc="1st paired-end read file path")), ToolInput(tag="in_reads_two", input_type=Int(optional=True), prefix="--reads2", doc=InputDocumentation(doc="1st paired-end read file path")), ToolInput(tag="in_plas_class", input_type=File(optional=True), prefix="--plasclass", doc=InputDocumentation(doc="PlasClass score file with scores of the assembly graph\nnodes")), ToolInput(tag="in_plas_flow", input_type=File(optional=True), prefix="--plasflow", doc=InputDocumentation(doc="PlasFlow score file with scores of the assembly graph\nnodes")), ToolInput(tag="in_classification_thresh", input_type=Float(optional=True), prefix="--classification_thresh", doc=InputDocumentation(doc="threshold for classifying potential plasmid [0.5]")), ToolInput(tag="in_gene_match_thresh", input_type=Int(optional=True), prefix="--gene_match_thresh", doc=InputDocumentation(doc="threshold for % identity and fraction of length to\nmatch plasmid genes [0.75]")), ToolInput(tag="in_self_loop_score_thresh", input_type=Float(optional=True), prefix="--selfloop_score_thresh", doc=InputDocumentation(doc="threshold for self-loop plasmid score [0.9]")), ToolInput(tag="in_self_loop_mate_thresh", input_type=Float(optional=True), prefix="--selfloop_mate_thresh", doc=InputDocumentation(doc="threshold for self-loop off loop mates [0.1]")), ToolInput(tag="in_chromosome_score_thresh", input_type=Float(optional=True), prefix="--chromosome_score_thresh", doc=InputDocumentation(doc="threshold for high confidence chromosome node score\n[0.2]")), ToolInput(tag="in_chromosome_len_thresh", input_type=Int(optional=True), prefix="--chromosome_len_thresh", doc=InputDocumentation(doc="threshold for high confidence chromosome node length\n[10000]")), ToolInput(tag="in_plasmid_score_thresh", input_type=Float(optional=True), prefix="--plasmid_score_thresh", doc=InputDocumentation(doc="threshold for high confidence plasmid node score [0.9]")), ToolInput(tag="in_plasmid_len_thresh", input_type=Int(optional=True), prefix="--plasmid_len_thresh", doc=InputDocumentation(doc="threshold for high confidence plasmid node length\n[10000]")), ToolInput(tag="in_good_cyc_dominated_thresh", input_type=Float(optional=True), prefix="--good_cyc_dominated_thresh", doc=InputDocumentation(doc="threshold for # of mate-pairs off the cycle in\ndominated node [0.5]\n"))], outputs=[ToolOutput(tag="out_output_dir", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_output_dir", type_hint=File()), doc=OutputDocumentation(doc="Output directory"))], container="quay.io/biocontainers/scapp:0.1.4--py_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Scapp_V0_1_0().translate("wdl")

