from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Int, Float, Boolean, String

Bwa_Mem2_Avx512Bw_Mem_V0_1_0 = CommandToolBuilder(tool="bwa_mem2.avx512bw_mem", base_command=["bwa-mem2.avx512bw", "mem"], inputs=[ToolInput(tag="in_output_sam_file", input_type=File(optional=True), prefix="-o", doc=InputDocumentation(doc="Output SAM file name")), ToolInput(tag="in_number_of_threads", input_type=Int(optional=True), prefix="-t", doc=InputDocumentation(doc="number of threads [1]")), ToolInput(tag="in_minimum_seed_length", input_type=Int(optional=True), prefix="-k", doc=InputDocumentation(doc="minimum seed length [19]")), ToolInput(tag="in_band_width_banded", input_type=Int(optional=True), prefix="-w", doc=InputDocumentation(doc="band width for banded alignment [100]")), ToolInput(tag="in_offdiagonal_xdropoff", input_type=Int(optional=True), prefix="-d", doc=InputDocumentation(doc="off-diagonal X-dropoff [100]")), ToolInput(tag="in_look_internal_float", input_type=Float(optional=True), prefix="-r", doc=InputDocumentation(doc="look for internal seeds inside a seed longer than {-k} * FLOAT [1.5]")), ToolInput(tag="in_seed_occurrence_round", input_type=Int(optional=True), prefix="-y", doc=InputDocumentation(doc="seed occurrence for the 3rd round seeding [20]")), ToolInput(tag="in_skip_seeds_more", input_type=Int(optional=True), prefix="-c", doc=InputDocumentation(doc="skip seeds with more than INT occurrences [500]")), ToolInput(tag="in_drop_chains_shorter", input_type=Float(optional=True), prefix="-D", doc=InputDocumentation(doc="drop chains shorter than FLOAT fraction of the longest overlapping chain [0.50]")), ToolInput(tag="in_discard_chain_seeded", input_type=Int(optional=True), prefix="-W", doc=InputDocumentation(doc="discard a chain if seeded bases shorter than INT [0]")), ToolInput(tag="in_perform_most_int", input_type=Int(optional=True), prefix="-m", doc=InputDocumentation(doc="perform at most INT rounds of mate rescues for each read [50]")), ToolInput(tag="in_skip_mate_rescue", input_type=Boolean(optional=True), prefix="-S", doc=InputDocumentation(doc="skip mate rescue")), ToolInput(tag="in_skip_pairing_rescue", input_type=Boolean(optional=True), prefix="-P", doc=InputDocumentation(doc="skip pairing; mate rescue performed unless -S also in use")), ToolInput(tag="in_score_sequence_tdboelu", input_type=Int(optional=True), prefix="-A", doc=InputDocumentation(doc="score for a sequence match, which scales options -TdBOELU unless overridden [1]")), ToolInput(tag="in_penalty_for_mismatch", input_type=Int(optional=True), prefix="-B", doc=InputDocumentation(doc="penalty for a mismatch [4]")), ToolInput(tag="in_gap_open_penalties", input_type=Int(), prefix="-O", doc=InputDocumentation(doc="gap open penalties for deletions and insertions [6,6]")), ToolInput(tag="in_gap_extension_penalty", input_type=Int(), prefix="-E", doc=InputDocumentation(doc="gap extension penalty; a gap of size k cost '{-O} + {-E}*k' [1,1]")), ToolInput(tag="in_penalty_end_clipping", input_type=Int(), prefix="-L", doc=InputDocumentation(doc="penalty for 5'- and 3'-end clipping [5,5]")), ToolInput(tag="in_penalty_unpaired_read", input_type=Int(optional=True), prefix="-U", doc=InputDocumentation(doc="penalty for an unpaired read pair [17]")), ToolInput(tag="in_smart_pairing_ignoring", input_type=Boolean(optional=True), prefix="-p", doc=InputDocumentation(doc="smart pairing (ignoring in2.fq)")), ToolInput(tag="in_read_group_line", input_type=String(optional=True), prefix="-R", doc=InputDocumentation(doc="read group header line such as '@RG\tID:foo\tSM:bar' [null]")), ToolInput(tag="in_file_insert_str", input_type=File(optional=True), prefix="-H", doc=InputDocumentation(doc="/FILE   insert STR to header if it starts with @; or insert lines in FILE [null]")), ToolInput(tag="in_treat_alt_contigs", input_type=Boolean(optional=True), prefix="-j", doc=InputDocumentation(doc="treat ALT contigs as part of the primary assembly (i.e. ignore <idxbase>.alt file)")), ToolInput(tag="in_split_alignment_take", input_type=Boolean(optional=True), prefix="-5", doc=InputDocumentation(doc="for split alignment, take the alignment with the smallest coordinate as primary")), ToolInput(tag="in_modify_mapq_supplementary", input_type=Boolean(optional=True), prefix="-q", doc=InputDocumentation(doc="don't modify mapQ of supplementary alignments")), ToolInput(tag="in_process_int_input", input_type=Int(optional=True), prefix="-K", doc=InputDocumentation(doc="process INT input bases in each batch regardless of nThreads (for reproducibility) []")), ToolInput(tag="in_verbose_level_error", input_type=Int(optional=True), prefix="-v", doc=InputDocumentation(doc="verbose level: 1=error, 2=warning, 3=message, 4+=debugging [3]")), ToolInput(tag="in_minimum_score_output", input_type=Int(optional=True), prefix="-T", doc=InputDocumentation(doc="minimum score to output [30]")), ToolInput(tag="in_there_int_hits", input_type=Int(), prefix="-h", doc=InputDocumentation(doc="if there are <INT hits with score >80% of the max score, output all in XA [5,200]")), ToolInput(tag="in_output_alignments_pe", input_type=Boolean(optional=True), prefix="-a", doc=InputDocumentation(doc="output all alignments for SE or unpaired PE")), ToolInput(tag="in_append_fastafastq_comment", input_type=Boolean(optional=True), prefix="-C", doc=InputDocumentation(doc="append FASTA/FASTQ comment to SAM output")), ToolInput(tag="in_output_reference_header", input_type=Boolean(optional=True), prefix="-V", doc=InputDocumentation(doc="output the reference FASTA header in the XR tag")), ToolInput(tag="in_use_soft_clipping", input_type=Boolean(optional=True), prefix="-Y", doc=InputDocumentation(doc="use soft clipping for supplementary alignments")), ToolInput(tag="in_mark_shorter_hits", input_type=Boolean(optional=True), prefix="-M", doc=InputDocumentation(doc="mark shorter split hits as secondary")), ToolInput(tag="in_specify_mean_deviation", input_type=Float(), prefix="-I", doc=InputDocumentation(doc="specify the mean, standard deviation (10% of the mean if absent), max\n(4 sigma from the mean if absent) and min of the insert size distribution.\nFR orientation only. [inferred]")), ToolInput(tag="in_idx_base", input_type=String(), position=0, doc=InputDocumentation(doc="")), ToolInput(tag="in_in_one_dot_fq", input_type=Int(), position=1, doc=InputDocumentation(doc="")), ToolInput(tag="in_in_two_dot_fq", input_type=Int(optional=True), position=2, doc=InputDocumentation(doc=""))], outputs=[ToolOutput(tag="out_output_sam_file", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output_sam_file", type_hint=File()), doc=OutputDocumentation(doc="Output SAM file name"))], container="quay.io/biocontainers/bwa-mem2:2.2.1--he513fc3_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Bwa_Mem2_Avx512Bw_Mem_V0_1_0().translate("wdl")

