from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Int, Directory, Boolean, String, Array, Float, File

Mykrobe_Predict_V0_1_0 = CommandToolBuilder(tool="mykrobe_predict", base_command=["mykrobe", "predict"], inputs=[ToolInput(tag="in_km_er", input_type=Int(optional=True), prefix="--kmer", doc=InputDocumentation(doc="kmer length (default:21)")), ToolInput(tag="in_tmp", input_type=Directory(optional=True), prefix="--tmp", doc=InputDocumentation(doc="tmp directory (default: tmp/)")), ToolInput(tag="in_keep_tmp", input_type=Boolean(optional=True), prefix="--keep_tmp", doc=InputDocumentation(doc="Dont remove tmp files")), ToolInput(tag="in_skeleton_dir", input_type=Directory(optional=True), prefix="--skeleton_dir", doc=InputDocumentation(doc="directory for skeleton binaries")), ToolInput(tag="in_expected_depth", input_type=String(optional=True), prefix="--expected_depth", doc=InputDocumentation(doc="expected depth")), ToolInput(tag="in_seq", input_type=Array(t=String(), optional=True), prefix="--seq", doc=InputDocumentation(doc="sequence files (fasta,fastq,bam)")), ToolInput(tag="in_ctx", input_type=String(optional=True), prefix="--ctx", doc=InputDocumentation(doc="cortex graph binary")), ToolInput(tag="in_force", input_type=Boolean(optional=True), prefix="--force", doc=InputDocumentation(doc="force")), ToolInput(tag="in_ont", input_type=Boolean(optional=True), prefix="--ont", doc=InputDocumentation(doc="Set default for ONT data. Sets expected_error_rate to\n0.15 and to haploid")), ToolInput(tag="in_guess_sequence_method", input_type=Boolean(optional=True), prefix="--guess_sequence_method", doc=InputDocumentation(doc="Guess if ONT or Illumia based on error rate. If error\nrate is > 10%, ploidy is set to haploid and a\nconfidence threshold is used")), ToolInput(tag="in_ignore_minor_calls", input_type=Boolean(optional=True), prefix="--ignore_minor_calls", doc=InputDocumentation(doc="Ignore minor calls when running resistance prediction")), ToolInput(tag="in_ignore_filtered", input_type=String(optional=True), prefix="--ignore_filtered", doc=InputDocumentation(doc="don't include filtered genotypes")), ToolInput(tag="in_model", input_type=String(optional=True), prefix="--model", doc=InputDocumentation(doc="Genotype model used, default kmer_count. Options\nkmer_count, median_depth")), ToolInput(tag="in_ploidy", input_type=Int(optional=True), prefix="--ploidy", doc=InputDocumentation(doc="Use a diploid (includes 0/1 calls) or haploid\ngenotyping model")), ToolInput(tag="in_filters", input_type=Array(t=String(), optional=True), prefix="--filters", doc=InputDocumentation(doc="don't include filtered genotypes")), ToolInput(tag="in_report_all_calls", input_type=Boolean(optional=True), prefix="--report_all_calls", doc=InputDocumentation(doc="report all calls")), ToolInput(tag="in_expected_error_rate", input_type=Float(optional=True), prefix="--expected_error_rate", doc=InputDocumentation(doc="Expected sequencing error rate. Set to 0.15 for ONT\ngenotyping.")), ToolInput(tag="in_min_variant_conf", input_type=String(optional=True), prefix="--min_variant_conf", doc=InputDocumentation(doc="minimum genotype confidence for variant genotyping")), ToolInput(tag="in_min_gene_conf", input_type=Int(optional=True), prefix="--min_gene_conf", doc=InputDocumentation(doc="minimum genotype confidence for gene genotyping")), ToolInput(tag="in_min_proportion_expected_depth", input_type=Int(optional=True), prefix="--min_proportion_expected_depth", doc=InputDocumentation(doc="minimum depth required on the sum of both alleles.\nDefault 0.3 (30%)")), ToolInput(tag="in_min_gene_percent_co_vg_threshold", input_type=Int(optional=True), prefix="--min_gene_percent_covg_threshold", doc=InputDocumentation(doc="all genes alleles found above this percent coverage\nwill be reported (default 100 (only best alleles\nreported))")), ToolInput(tag="in_output", input_type=File(optional=True), prefix="--output", doc=InputDocumentation(doc="File path to save output json file as. Default is to")), ToolInput(tag="in_quiet", input_type=Boolean(optional=True), prefix="--quiet", doc=InputDocumentation(doc="Only output warnings/errors to stderr")), ToolInput(tag="in_debug", input_type=Boolean(optional=True), prefix="--debug", doc=InputDocumentation(doc="Output debugging information to stderr")), ToolInput(tag="in_panel", input_type=String(optional=True), prefix="--panel", doc=InputDocumentation(doc="Name of panel to use. Ignored if species is 'custom'")), ToolInput(tag="in_custom_probe_set_path", input_type=File(optional=True), prefix="--custom_probe_set_path", doc=InputDocumentation(doc="Required if species is 'custom'. Ignored otherwise.\nFile path to fasta file from `mykrobe make-probes`.")), ToolInput(tag="in_custom_variant_to_resistance_json", input_type=File(optional=True), prefix="--custom_variant_to_resistance_json", doc=InputDocumentation(doc="For use with `--panel custom`. Ignored otherwise. File\npath to JSON with key,value pairs of variant names and\ninduced drug resistance.")), ToolInput(tag="in_custom_lineage_json", input_type=File(optional=True), prefix="--custom_lineage_json", doc=InputDocumentation(doc="For use with `--panel custom`. Ignored otherwise. File\npath to JSON made by --lineage option of make-probes")), ToolInput(tag="in_format", input_type=String(optional=True), prefix="--format", doc=InputDocumentation(doc="Choose output format. Default: csv.\n")), ToolInput(tag="in_sample", input_type=String(), position=0, doc=InputDocumentation(doc="sample id")), ToolInput(tag="in_species", input_type=String(), position=1, doc=InputDocumentation(doc="species name, or 'custom' to use custom data, in which\ncase --custom_probe_set_path is required")), ToolInput(tag="in_threads", input_type=String(), position=0, doc=InputDocumentation(doc="-m MEMORY, --memory MEMORY")), ToolInput(tag="in_stdout_dot", input_type=String(), position=0, doc=InputDocumentation(doc="--panels_dir DIRNAME  Name of directory that contains panel data. Default:"))], outputs=[ToolOutput(tag="out_output", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output", type_hint=File()), doc=OutputDocumentation(doc="File path to save output json file as. Default is to"))], container="quay.io/biocontainers/mykrobe:0.9.0--py36hd181a71_2", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Mykrobe_Predict_V0_1_0().translate("wdl")

