from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean

Verifybamid2_V0_1_0 = CommandToolBuilder(tool="verifybamid2", base_command=["verifybamid2"], inputs=[ToolInput(tag="in_bam_file", input_type=Boolean(optional=True), prefix="--BamFile", doc=InputDocumentation(doc="[STR: Empty]        : [String] Bam or Cram file for the sample[Required if --PileupFile not specified]")), ToolInput(tag="in_pile_up_file", input_type=Boolean(optional=True), prefix="--PileupFile", doc=InputDocumentation(doc="[STR: Empty]        : [String] Pileup file for the sample[Required if --BamFile not specified]")), ToolInput(tag="in_reference", input_type=Boolean(optional=True), prefix="--Reference", doc=InputDocumentation(doc="[STR: Empty]        : [String] Reference file[Required]")), ToolInput(tag="in_svd_prefix", input_type=Boolean(optional=True), prefix="--SVDPrefix", doc=InputDocumentation(doc="[STR: Empty]        : [String] SVD related files prefix(normally shared by .UD, .mu and .bed files)[Required]")), ToolInput(tag="in_output", input_type=Boolean(optional=True), prefix="--Output", doc=InputDocumentation(doc="[STR: result]       : [String] Prefix of output files[optional]")), ToolInput(tag="in_within_ancestry", input_type=Boolean(optional=True), prefix="--WithinAncestry", doc=InputDocumentation(doc="[FLG: OFF]          : [Bool] Enabling withinAncestry assume target sample and contamination source are from the same populations,[default:BetweenAncestry] otherwise")), ToolInput(tag="in_disable_sanity_check", input_type=Boolean(optional=True), prefix="--DisableSanityCheck", doc=InputDocumentation(doc="[FLG: OFF]          : [Bool] Disable marker quality sanity check(no marker filtering)[default:false]")), ToolInput(tag="in_num_pc", input_type=Boolean(optional=True), prefix="--NumPC", doc=InputDocumentation(doc="[INT: 2]            : [Int] Set number of PCs to infer Allele Frequency[optional]")), ToolInput(tag="in_fix_pc", input_type=Boolean(optional=True), prefix="--FixPC", doc=InputDocumentation(doc="[STR: Empty]        : [String] Input fixed PCs to estimate Alpha[format PC1:PC2:PC3...]")), ToolInput(tag="in_fix_alpha", input_type=Boolean(optional=True), prefix="--FixAlpha", doc=InputDocumentation(doc="[FLT: -1.0e+00]     : [Double] Input fixed Alpha to estimate PC coordinates")), ToolInput(tag="in_known_af", input_type=Boolean(optional=True), prefix="--KnownAF", doc=InputDocumentation(doc="[STR: Empty]        : [String] known allele frequency file (chr  pos     freq)[Optional]")), ToolInput(tag="in_num_thread", input_type=Boolean(optional=True), prefix="--NumThread", doc=InputDocumentation(doc="[INT: 4]            : [Int] Set number of threads in likelihood calculation[default:4]")), ToolInput(tag="in_seed", input_type=Boolean(optional=True), prefix="--Seed", doc=InputDocumentation(doc="[INT: 12345]        : [Int] Random number seed[default:12345]")), ToolInput(tag="in_epsilon", input_type=Boolean(optional=True), prefix="--Epsilon", doc=InputDocumentation(doc="[FLT: 1.0e-08]      : [Double] Minimization procedure convergence threshold, usually a trade-off bettween accuracy and running time[default:1e-10]")), ToolInput(tag="in_output_pile_up", input_type=Boolean(optional=True), prefix="--OutputPileup", doc=InputDocumentation(doc="[FLG: OFF]          : [Bool] If output temp pileup file")), ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="--Verbose", doc=InputDocumentation(doc="[FLG: OFF]          : [Bool] If print the progress of the method on the screen")), ToolInput(tag="in_ref_vcf", input_type=Boolean(optional=True), prefix="--RefVCF", doc=InputDocumentation(doc="[STR: Empty]        : [String] VCF file from which to extract reference panel's genotype matrix[Required if no SVD files available]")), ToolInput(tag="in_min_bq", input_type=Boolean(optional=True), prefix="--min-BQ", doc=InputDocumentation(doc="[INT: 13]           : [Int] skip bases with baseQ/BAQ smaller than min-BQ")), ToolInput(tag="in_min_mq", input_type=Boolean(optional=True), prefix="--min-MQ", doc=InputDocumentation(doc="[INT: 2]            : [Int] skip alignments with mapQ smaller than min-MQ")), ToolInput(tag="in_adjust_mq", input_type=Boolean(optional=True), prefix="--adjust-MQ", doc=InputDocumentation(doc="[INT: 40]           : [Int] adjust mapping quality; recommended:50, disable:0")), ToolInput(tag="in_max_depth", input_type=Boolean(optional=True), prefix="--max-depth", doc=InputDocumentation(doc="[INT: 8000]         : [Int] max per-file depth")), ToolInput(tag="in_no_orphans", input_type=Boolean(optional=True), prefix="--no-orphans", doc=InputDocumentation(doc="[FLG: OFF]          : [Bool] do not use anomalous read pairs")), ToolInput(tag="in_incl_flags", input_type=Boolean(optional=True), prefix="--incl-flags", doc=InputDocumentation(doc="[INT: 1040]         : [Int] required flags: skip reads with mask bits unset")), ToolInput(tag="in_excl_flags", input_type=Boolean(optional=True), prefix="--excl-flags", doc=InputDocumentation(doc="[INT: 1796]         : [Int] filter flags: skip reads with mask bits set")), ToolInput(tag="in_ud_path", input_type=Boolean(optional=True), prefix="--UDPath", doc=InputDocumentation(doc="[STR: Empty]        : [String] UD matrix file from SVD result of genotype matrix")), ToolInput(tag="in_mean_path", input_type=Boolean(optional=True), prefix="--MeanPath", doc=InputDocumentation(doc="[STR: Empty]        : [String] Mean matrix file of genotype matrix")), ToolInput(tag="in_bed_path", input_type=Boolean(optional=True), prefix="--BedPath", doc=InputDocumentation(doc="[STR: Empty]        : [String] Bed file for markers used in this analysis,1 based pos(chr        pos-1   pos     refAllele       altAllele)[Required]"))], outputs=[], container="quay.io/biocontainers/verifybamid2:2.0.1--h32f71e1_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Verifybamid2_V0_1_0().translate("wdl")

