from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, File, Int, String, Float

Rtg_Vcffilter_V0_1_0 = CommandToolBuilder(tool="rtg_vcffilter", base_command=["rtg", "vcffilter"], inputs=[ToolInput(tag="in_clear_failed_samples", input_type=Boolean(optional=True), prefix="--clear-failed-samples", doc=InputDocumentation(doc=", and --fail-samples options for alternatives.")), ToolInput(tag="in_all_samples", input_type=Boolean(optional=True), prefix="--all-samples", doc=InputDocumentation(doc="apply sample-specific criteria to all\nsamples contained in the input VCF")), ToolInput(tag="in_bed_regions", input_type=File(optional=True), prefix="--bed-regions", doc=InputDocumentation(doc="if set, only read VCF records that overlap\nthe ranges contained in the specified BED\nfile")), ToolInput(tag="in_input", input_type=File(optional=True), prefix="--input", doc=InputDocumentation(doc="VCF file containing variants to be\nfiltered. Use '-' to read from standard\ninput")), ToolInput(tag="in_output", input_type=File(optional=True), prefix="--output", doc=InputDocumentation(doc="output VCF file. Use '-' to write to\nstandard output. This option is required,\nunless '--javascript' is being used")), ToolInput(tag="in_region", input_type=Int(optional=True), prefix="--region", doc=InputDocumentation(doc="if set, only read VCF records within the\nspecified range. The format is one of\n<sequence_name>,\n<sequence_name>:<start>-<end>,\n<sequence_name>:<pos>+<length> or\n<sequence_name>:<pos>~<padding>")), ToolInput(tag="in_sample", input_type=Int(optional=True), prefix="--sample", doc=InputDocumentation(doc="apply sample-specific criteria to the named\nsample contained in the input VCF. May be\nspecified 0 or more times")), ToolInput(tag="in_density_window", input_type=Int(optional=True), prefix="--density-window", doc=InputDocumentation(doc="window within which multiple variants are")), ToolInput(tag="in_exclude_vcf", input_type=File(optional=True), prefix="--exclude-vcf", doc=InputDocumentation(doc="discard all variants that overlap with the\nones in this file")), ToolInput(tag="in_include_bed", input_type=File(optional=True), prefix="--include-bed", doc=InputDocumentation(doc="only keep variants within the regions in\nthis BED file")), ToolInput(tag="in_include_vcf", input_type=File(optional=True), prefix="--include-vcf", doc=InputDocumentation(doc="only keep variants that overlap with the\nones in this file")), ToolInput(tag="in_javascript", input_type=File(optional=True), prefix="--javascript", doc=InputDocumentation(doc="javascript filtering functions for\ndetermining whether to keep record. May be\neither an expression or a file name. May be\nspecified 0 or more times")), ToolInput(tag="in_keep_expr", input_type=String(optional=True), prefix="--keep-expr", doc=InputDocumentation(doc="records for which this expression evaluates\nto true will be retained")), ToolInput(tag="in_keep_filter", input_type=Int(optional=True), prefix="--keep-filter", doc=InputDocumentation(doc="only keep variants with this FILTER tag.\nMay be specified 0 or more times, or as a\ncomma separated list")), ToolInput(tag="in_keep_info", input_type=Int(optional=True), prefix="--keep-info", doc=InputDocumentation(doc="only keep variants with this INFO tag. May\nbe specified 0 or more times, or as a comma\nseparated list")), ToolInput(tag="in_max_alleles", input_type=Int(optional=True), prefix="--max-alleles", doc=InputDocumentation(doc="maximum number of alleles (REF included)")), ToolInput(tag="in_max_combined_read_depth", input_type=Int(optional=True), prefix="--max-combined-read-depth", doc=InputDocumentation(doc="maximum allowed combined read depth")), ToolInput(tag="in_max_quality", input_type=Float(optional=True), prefix="--max-quality", doc=InputDocumentation(doc="maximum allowed quality")), ToolInput(tag="in_min_alleles", input_type=Int(optional=True), prefix="--min-alleles", doc=InputDocumentation(doc="minimum number of alleles (REF included)")), ToolInput(tag="in_min_combined_read_depth", input_type=Int(optional=True), prefix="--min-combined-read-depth", doc=InputDocumentation(doc="minimum allowed combined read depth")), ToolInput(tag="in_min_quality", input_type=Float(optional=True), prefix="--min-quality", doc=InputDocumentation(doc="minimum allowed quality")), ToolInput(tag="in_remove_filter", input_type=Int(optional=True), prefix="--remove-filter", doc=InputDocumentation(doc="remove variants with this FILTER tag. May\nbe specified 0 or more times, or as a comma\nseparated list")), ToolInput(tag="in_remove_info", input_type=Int(optional=True), prefix="--remove-info", doc=InputDocumentation(doc="remove variants with this INFO tag. May be\nspecified 0 or more times, or as a comma\nseparated list")), ToolInput(tag="in_remove_overlapping", input_type=Boolean(optional=True), prefix="--remove-overlapping", doc=InputDocumentation(doc="remove records that overlap with previous")), ToolInput(tag="in_max_ambiguity_ratio", input_type=Float(optional=True), prefix="--max-ambiguity-ratio", doc=InputDocumentation(doc="maximum allowed ambiguity ratio")), ToolInput(tag="in_max_avr_score", input_type=Float(optional=True), prefix="--max-avr-score", doc=InputDocumentation(doc="maximum allowed AVR score")), ToolInput(tag="in_max_de_novo_score", input_type=Float(optional=True), prefix="--max-denovo-score", doc=InputDocumentation(doc="maximum de novo score threshold")), ToolInput(tag="in_max_genotype_quality", input_type=Float(optional=True), prefix="--max-genotype-quality", doc=InputDocumentation(doc="maximum allowed genotype quality")), ToolInput(tag="in_max_read_depth", input_type=Int(optional=True), prefix="--max-read-depth", doc=InputDocumentation(doc="maximum allowed sample read depth")), ToolInput(tag="in_min_avr_score", input_type=Float(optional=True), prefix="--min-avr-score", doc=InputDocumentation(doc="minimum allowed AVR score")), ToolInput(tag="in_min_de_novo_score", input_type=Float(optional=True), prefix="--min-denovo-score", doc=InputDocumentation(doc="minimum de novo score threshold")), ToolInput(tag="in_min_genotype_quality", input_type=Float(optional=True), prefix="--min-genotype-quality", doc=InputDocumentation(doc="minimum allowed genotype quality")), ToolInput(tag="in_min_read_depth", input_type=Int(optional=True), prefix="--min-read-depth", doc=InputDocumentation(doc="minimum allowed sample read depth")), ToolInput(tag="in_non_snps_only", input_type=Boolean(optional=True), prefix="--non-snps-only", doc=InputDocumentation(doc="only keep where sample variant is MNP or")), ToolInput(tag="in_remove_same_as_ref", input_type=Boolean(optional=True), prefix="--remove-same-as-ref", doc=InputDocumentation(doc="remove where sample is same as reference")), ToolInput(tag="in_snps_only", input_type=Boolean(optional=True), prefix="--snps-only", doc=InputDocumentation(doc="only keep where sample variant is a simple")), ToolInput(tag="in_fail", input_type=String(optional=True), prefix="--fail", doc=InputDocumentation(doc="retain failed records, add the provided\nlabel to the FILTER field")), ToolInput(tag="in_fail_samples", input_type=String(optional=True), prefix="--fail-samples", doc=InputDocumentation(doc="retain failed records, add the provided\nlabel to the sample FT field")), ToolInput(tag="in_add_header", input_type=File(optional=True), prefix="--add-header", doc=InputDocumentation(doc="|FILE      file containing VCF header lines to add, or\na literal header line. May be specified 0\nor more times")), ToolInput(tag="in_no_gzip", input_type=Boolean(optional=True), prefix="--no-gzip", doc=InputDocumentation(doc="do not gzip the output")), ToolInput(tag="in_no_header", input_type=Boolean(optional=True), prefix="--no-header", doc=InputDocumentation(doc="prevent VCF header from being written")), ToolInput(tag="in_discarded", input_type=String(), position=0, doc=InputDocumentation(doc="--exclude-bed=FILE            discard all variants within the regions in")), ToolInput(tag="in_records", input_type=String(), position=0, doc=InputDocumentation(doc="Filtering (Sample based)")), ToolInput(tag="in_in_del", input_type=String(), position=1, doc=InputDocumentation(doc="--remove-all-same-as-ref      remove where all samples are same as")), ToolInput(tag="in_reference", input_type=String(), position=2, doc=InputDocumentation(doc="--remove-hom                  remove where sample is homozygous")), ToolInput(tag="in_snp", input_type=String(), position=3, doc=InputDocumentation(doc="Reporting"))], outputs=[ToolOutput(tag="out_output", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output", type_hint=File()), doc=OutputDocumentation(doc="output VCF file. Use '-' to write to\nstandard output. This option is required,\nunless '--javascript' is being used"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Rtg_Vcffilter_V0_1_0().translate("wdl", allow_empty_container=True)

