from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Boolean, Int, String

Slncky_V1_0_V0_1_0 = CommandToolBuilder(tool="slncky.v1.0", base_command=["slncky.v1.0"], inputs=[ToolInput(tag="in_config", input_type=File(optional=True), prefix="--config", doc=InputDocumentation(doc="path to assembly.config file. default uses config file\nin same directory as slncky")), ToolInput(tag="in_no_orth_search", input_type=Boolean(optional=True), prefix="--no_orth_search", doc=InputDocumentation(doc="flag if you only want to filter lncs but don't want to\nsearch for orthologs")), ToolInput(tag="in_no_filter", input_type=Boolean(optional=True), prefix="--no_filter", doc=InputDocumentation(doc="flag if you don't want lncs to be filtered before\nsearching for ortholog")), ToolInput(tag="in_overwrite", input_type=Boolean(optional=True), prefix="--overwrite", doc=InputDocumentation(doc="forces overwrite of out_prefix.bed")), ToolInput(tag="in_threads", input_type=Int(optional=True), prefix="--threads", doc=InputDocumentation(doc="number of threads. default = 5")), ToolInput(tag="in_min_overlap", input_type=Int(optional=True), prefix="--min_overlap", doc=InputDocumentation(doc="remove any transcript that overlap annotated coding\ngene > min_overlap%. default = 0%")), ToolInput(tag="in_min_cluster", input_type=Int(optional=True), prefix="--min_cluster", doc=InputDocumentation(doc="min size of duplication clusters to remove. default=2")), ToolInput(tag="in_min_coding", input_type=Int(optional=True), prefix="--min_coding", doc=InputDocumentation(doc="min exonic identity to filter out transcript that\naligns to orthologous coding gene. default is set by\nlearning coding alignment distribution from data")), ToolInput(tag="in_no_overlap", input_type=Boolean(optional=True), prefix="--no_overlap", doc=InputDocumentation(doc="flag if you don't want to overlap with coding")), ToolInput(tag="in_no_collapse", input_type=Boolean(optional=True), prefix="--no_collapse", doc=InputDocumentation(doc="flag if you don't want to collapse isoforms")), ToolInput(tag="in_no_dup", input_type=Boolean(optional=True), prefix="--no_dup", doc=InputDocumentation(doc="flag if don't want to align to duplicates")), ToolInput(tag="in_no_self", input_type=Boolean(optional=True), prefix="--no_self", doc=InputDocumentation(doc="flag if you don't want to self-align for duplicates")), ToolInput(tag="in_no_coding", input_type=Boolean(optional=True), prefix="--no_coding", doc=InputDocumentation(doc="flag if you don't want to align to orthologous coding")), ToolInput(tag="in_min_noncoding", input_type=Int(optional=True), prefix="--min_noncoding", doc=InputDocumentation(doc="min exonic identity to filter out transcript that\naligns to orthologous noncoding gene. default=0")), ToolInput(tag="in_no_bg", input_type=Boolean(optional=True), prefix="--no_bg", doc=InputDocumentation(doc="flag if you don't want to compare lnc-to-ortholog\nalignments to a background. This flag may be useful if\nyou want to do a 'quick-and-dirty' run of the ortholog\nsearch.")), ToolInput(tag="in_no_orf", input_type=Boolean(optional=True), prefix="--no_orf", doc=InputDocumentation(doc="flag if you don't want to search for orfs")), ToolInput(tag="in_bed_tools", input_type=File(optional=True), prefix="--bedtools", doc=InputDocumentation(doc="path to bedtools")), ToolInput(tag="in_lift_over", input_type=File(optional=True), prefix="--liftover", doc=InputDocumentation(doc="path to liftOver")), ToolInput(tag="in_min_match", input_type=Int(optional=True), prefix="--minMatch", doc=InputDocumentation(doc="minMatch parameter for liftover. default=0.1")), ToolInput(tag="in_pad", input_type=String(optional=True), prefix="--pad", doc=InputDocumentation(doc="# of basepairs to search up- and down-stream when\nlifting over lnc to ortholog")), ToolInput(tag="in_last_z", input_type=File(optional=True), prefix="--lastz", doc=InputDocumentation(doc="path to lastz")), ToolInput(tag="in_gap_extend", input_type=String(optional=True), prefix="--gap_extend", doc=InputDocumentation(doc="flag for if you want slncky to create a website\nvisualizing results\n")), ToolInput(tag="in_bed_file", input_type=String(), position=0, doc=InputDocumentation(doc="bed12 file of transcripts")), ToolInput(tag="in_assembly", input_type=String(), position=1, doc=InputDocumentation(doc="assembly")), ToolInput(tag="in_out_prefix", input_type=String(), position=2, doc=InputDocumentation(doc="out_prefix"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Slncky_V1_0_V0_1_0().translate("wdl", allow_empty_container=True)

