from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, String, Boolean, Directory, Int

Flatfile_To_Json_Pl_V0_1_0 = CommandToolBuilder(tool="flatfile_to_json.pl", base_command=["flatfile-to-json.pl"], inputs=[ToolInput(tag="in_bed", input_type=File(optional=True), prefix="--bed", doc=InputDocumentation(doc="Process a GFF3, BED, or GenBank file containing annotation data.\nThis script does not support GFF version 2 or GTF (GFF 2.5) input.\nGenBank input is limited to handling records for single genes.")), ToolInput(tag="in_track_label", input_type=String(optional=True), prefix="--trackLabel", doc=InputDocumentation(doc="Unique identifier for this track. Required.")), ToolInput(tag="in_key", input_type=Boolean(optional=True), prefix="--key", doc=InputDocumentation(doc="'<text>'\nHuman-readable track name.")), ToolInput(tag="in_out", input_type=Directory(optional=True), prefix="--out", doc=InputDocumentation(doc="Output directory to write to. Defaults to 'data/'.")), ToolInput(tag="in_track_type", input_type=String(optional=True), prefix="--trackType", doc=InputDocumentation(doc="/View/Track/HTMLFeatures\nOptional JavaScript class to use to display this track. For backward\ncompatibility, this defaults to 'HTMLFeatures'.\nUnless you have some reason to use HTMLFeatures tracks, though, it's\nrecommended to set this to 'CanvasFeatures' to use the newer\ncanvas-based feature track type.")), ToolInput(tag="in_classname", input_type=String(optional=True), prefix="--className", doc=InputDocumentation(doc="CSS class for features. Defaults to 'feature'. Only used by\nHTMLFeatures tracks.")), ToolInput(tag="in_url_template", input_type=Boolean(optional=True), prefix="--urltemplate", doc=InputDocumentation(doc="'http://example.com/idlookup?id={id}'\nTemplate for a URL to be visited when features are clicked on.")), ToolInput(tag="in_no_sub_features", input_type=Boolean(optional=True), prefix="--noSubfeatures", doc=InputDocumentation(doc="Do not format subfeature data. Only include top-level features.")), ToolInput(tag="in_arrowhead_class", input_type=String(optional=True), prefix="--arrowheadClass", doc=InputDocumentation(doc="CSS class for arrowheads. Only used by HTMLFeatures tracks.")), ToolInput(tag="in_sub_feature_classes", input_type=Boolean(optional=True), prefix="--subfeatureClasses", doc=InputDocumentation(doc="'{ JSON-format subfeature class map }'\nCSS classes for each subfeature type, in JSON syntax. Example:\n--subfeatureClasses '{'CDS': 'transcript-CDS', 'exon': 'transcript-exon'}'\nOnly used by HTMLFeatures tracks.")), ToolInput(tag="in_client_config", input_type=Boolean(optional=True), prefix="--clientConfig", doc=InputDocumentation(doc="'{ JSON-format style configuration for this track }'\nExtra configuration for the client, in JSON syntax. Example:\n--clientConfig '{'featureCss': 'background-color: #668; height: 8px;', 'histScale': 2}'\nFor historical reasons, this is only merged into the 'style' section\nof the new track's configuration.")), ToolInput(tag="in_config", input_type=Boolean(optional=True), prefix="--config", doc=InputDocumentation(doc="'{ JSON-format extra configuration for this track }'\nAdditional top-level configuration for the client, in JSON syntax.\nExample:\n--config '{ 'glyph': 'ProcessedTranscript' }'\nUnlike '--clientConfig', this is merged into the top level of the\nnew track configuration.")), ToolInput(tag="in_metadata", input_type=Boolean(optional=True), prefix="--metadata", doc=InputDocumentation(doc="'{ JSON metadata }'\nMetadata about this track. Example:\n--metadata '{'description': 'Genes from XYZ pipeline.', 'category': 'Transcripts' }'")), ToolInput(tag="in_type", input_type=File(optional=True), prefix="--type", doc=InputDocumentation(doc="Only process features of the given type. Can take either single type\nnames, e.g. 'mRNA', or type names qualified by 'source' name, for\nwhatever definition of 'source' your data file might have. For\nexample, 'mRNA:exonerate' will filter for only mRNA features that\nhave a source of 'exonerate'.\nMultiple type names can be specified by separating the type names\nwith commas, e.g. '--type mRNA:exonerate,ncRNA'.")), ToolInput(tag="in_name_attributes", input_type=Boolean(optional=True), prefix="--nameAttributes", doc=InputDocumentation(doc="'name,alias,id'\nComma-separated list of feature attributes (a.k.a. tags) that should\nbe treated as names for this features. Case insensitive. Defaults to\n'name,alias,id'.")), ToolInput(tag="in_ncl_chunk", input_type=Int(optional=True), prefix="--nclChunk", doc=InputDocumentation(doc="NCList chunk size; if you get 'json text or perl structure exceeds\nmaximum nesting level' errors, try setting this lower (default:\n50,000).")), ToolInput(tag="in_compress", input_type=Boolean(optional=True), prefix="--compress", doc=InputDocumentation(doc="Compress the output, making .jsonz (gzipped) JSON files. This can\nsave a lot of disk space, but note that web servers require some\nadditional configuration to serve these correctly.")), ToolInput(tag="in_sort_mem", input_type=Int(optional=True), prefix="--sortMem", doc=InputDocumentation(doc="Bytes of RAM to use for sorting features. Default 512MB.\nThe JSON NCList generation has to sort the features by reference\nsequence, start coordinate, and end coordinate. This is how much RAM\nin bytes the sorting process is allowed to use.")), ToolInput(tag="in_thick_type", input_type=String(optional=True), prefix="--thickType", doc=InputDocumentation(doc="Correspond to '<-thin_type'> and '<-thick_type'> in\nBio::FeatureIO::bed. Do '<perldoc Bio::FeatureIO::bed'> for details.")), ToolInput(tag="in_flat_file_to_json_do_tpl", input_type=String(), position=0, doc=InputDocumentation(doc="\\n( --gff <GFF3 file> | --bed <BED file> | --gbk <GenBank file> )         \\n--trackLabel <track identifier>                                         \\n[ --trackType <JS Class> ]                                              \\n[ --out <output directory> ]                                            \\n[ --key <human-readable track name> ]                                   \\n[ --className <CSS class name for displaying features> ]                \\n[ --urltemplate 'http://example.com/idlookup?id={id}' ]                 \\n[ --arrowheadClass <CSS class> ]                                        \\n[ --noSubfeatures ]                                                     \\n[ --subfeatureClasses '{ JSON-format subfeature class map }' ]          \\n[ --clientConfig '{ JSON-format style configuration for this track }' ] \\n[ --config '{ JSON-format extra configuration for this track }' ]       \\n[ --thinType <BAM -thin_type> ]                                         \\n[ --thicktype <BAM -thick_type>]                                        \\n[ --type <feature types to process> ]                                   \\n[ --nclChunk <chunk size for generated NCLs> ]                          \\n[ --compress ]                                                          \\n[ --sortMem <memory in bytes to use for sorting> ]                      \\n[ --maxLookback <maximum number of features to buffer in gff3 files> ]  \\n[ --nameAttributes 'name,alias,id' ]                                    \")), ToolInput(tag="in_required", input_type=String(), position=0, doc=InputDocumentation(doc="--gff <GFF3 file>")), ToolInput(tag="in_optional", input_type=String(), position=1, doc=InputDocumentation(doc="--help | -h | -?\nDisplay an extended help screen.")), ToolInput(tag="in_gff_three_specific", input_type=Int(), position=2, doc=InputDocumentation(doc="--maxLookback <integer>\nMaximum number of features to keep in memory when parsing GFF3\nfiles. Defaults to 10000.\nIf you receive 'orphan features' errors when parsing a GFF3 file\nthat doesn't contain enough '###' directives (which are important\nfor parsing), you can try setting this higher if your machine has\nenough memory.")), ToolInput(tag="in_bed_specific", input_type=String(), position=3, doc=InputDocumentation(doc="--thinType <type>"))], outputs=[ToolOutput(tag="out_out", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_out", type_hint=File()), doc=OutputDocumentation(doc="Output directory to write to. Defaults to 'data/'."))], container="quay.io/biocontainers/jbrowse:1.16.11--pl526hc9558a2_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Flatfile_To_Json_Pl_V0_1_0().translate("wdl")

