from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Directory, String, Boolean, Float, Int

Disentangle_Organelle_Assembly_Py_V0_1_0 = CommandToolBuilder(tool="disentangle_organelle_assembly.py", base_command=["disentangle_organelle_assembly.py"], inputs=[ToolInput(tag="in_input_fastg_format", input_type=File(optional=True), prefix="-g", doc=InputDocumentation(doc="input fastg format file.")), ToolInput(tag="in_input_tab_format", input_type=File(optional=True), prefix="-t", doc=InputDocumentation(doc="input tab format file (*.csv; the postfix 'csv' was in\nconformity with Bandage) produced by slim_graph.py.")), ToolInput(tag="in_output_directory", input_type=Directory(optional=True), prefix="-o", doc=InputDocumentation(doc="output directory.")), ToolInput(tag="in_organelle_type_embplantptotherptembplantmtembplantnranimalmtfungusmtfungusnranonym", input_type=String(optional=True), prefix="-F", doc=InputDocumentation(doc="organelle type: embplant_pt/other_pt/embplant_mt/embpl\nant_nr/animal_mt/fungus_mt/fungus_nr/anonym.")), ToolInput(tag="in_linear", input_type=Boolean(optional=True), prefix="--linear", doc=InputDocumentation(doc="By default, this script would only disentangle the\ncircular graph (the complete circular organelle\ngenome), and would directly give up linear/broken\ngraphs. Choose this option to try for linear/broken\ncases.")), ToolInput(tag="in_weight_f", input_type=Float(optional=True), prefix="--weight-f", doc=InputDocumentation(doc="weight factor for excluding non-target contigs.\nDefault:100.0")), ToolInput(tag="in_depth_f", input_type=Float(optional=True), prefix="--depth-f", doc=InputDocumentation(doc="Depth factor for excluding non-target contigs.\nDefault:10.0")), ToolInput(tag="in_type_f", input_type=Float(optional=True), prefix="--type-f", doc=InputDocumentation(doc="Type factor for identifying genome type tag.\nDefault:3.0")), ToolInput(tag="in_contamination_depth", input_type=Float(optional=True), prefix="--contamination-depth", doc=InputDocumentation(doc="Depth factor for confirming contaminating contigs.\nDefault:3.0")), ToolInput(tag="in_contamination_similarity", input_type=Float(optional=True), prefix="--contamination-similarity", doc=InputDocumentation(doc="Similarity threshold for confirming contaminating\ncontigs. Default:0.9")), ToolInput(tag="in_no_degenerate", input_type=Boolean(optional=True), prefix="--no-degenerate", doc=InputDocumentation(doc="Disable making consensus from parallel contig based on\nnucleotide degenerate table.")), ToolInput(tag="in_degenerate_depth", input_type=Float(optional=True), prefix="--degenerate-depth", doc=InputDocumentation(doc="Depth factor for confirming parallel contigs.\nDefault:1.5")), ToolInput(tag="in_degenerate_similarity", input_type=Float(optional=True), prefix="--degenerate-similarity", doc=InputDocumentation(doc="Similarity threshold for confirming parallel contigs.\nDefault:0.98")), ToolInput(tag="in_expected_max_size", input_type=Int(optional=True), prefix="--expected-max-size", doc=InputDocumentation(doc="Expected maximum target genome size. Default: 200000\n(-F embplant_pt/fungus_mt), 25000 (-F\nembplant_nr/animal_mt/fungus_nr), 600000 (-F\nembplant_mt/other_pt)")), ToolInput(tag="in_expected_min_size", input_type=Int(optional=True), prefix="--expected-min-size", doc=InputDocumentation(doc="Expected mininum target genome size. Default: 10000")), ToolInput(tag="in_reverse_lsc", input_type=Boolean(optional=True), prefix="--reverse-lsc", doc=InputDocumentation(doc="For '-F embplant_pt' with complete circular result, by\ndefault, the direction of the starting contig (usually\nthe LSC contig) is determined as the direction with\nless ORFs. Choose this option to reverse the direction\nof the starting contig when result is circular.\nActually, both directions are biologically equivalent\nto each other. The reordering of the direction is only\nfor easier downstream analysis.")), ToolInput(tag="in_max_paths_num", input_type=Int(optional=True), prefix="--max-paths-num", doc=InputDocumentation(doc="Repeats would dramatically increase the number of\npotential isomers (paths). This option was used to\nexport a certain amount of paths out of all possible\npaths per assembly graph. Default: 1000")), ToolInput(tag="in_keep_all_polymorphic", input_type=Boolean(optional=True), prefix="--keep-all-polymorphic", doc=InputDocumentation(doc="By default, this script would pick the contig with\nhighest coverage among all parallel (polymorphic)\ncontigs when degenerating was not applicable. Choose\nthis flag to export all combinations.")), ToolInput(tag="in_min_sigma", input_type=Int(optional=True), prefix="--min-sigma", doc=InputDocumentation(doc="Minimum deviation factor for excluding non-target\ncontigs. Default:0.1")), ToolInput(tag="in_min_depth", input_type=Int(optional=True), prefix="--min-depth", doc=InputDocumentation(doc="Minimum coverage for a contig to be included in\ndisentangling. Default:0.0")), ToolInput(tag="in_max_depth", input_type=Int(optional=True), prefix="--max-depth", doc=InputDocumentation(doc="Minimum coverage for a contig to be included in\ndisentangling. Default:inf")), ToolInput(tag="in_max_multiplicity", input_type=Int(optional=True), prefix="--max-multiplicity", doc=InputDocumentation(doc="Maximum multiplicity of contigs for disentangling\ngenome paths. Should be 1~12. Default:8")), ToolInput(tag="in_prefix", input_type=Directory(optional=True), prefix="--prefix", doc=InputDocumentation(doc="Prefix of output files inside output directory.\nDefault:target")), ToolInput(tag="in_keep_temp", input_type=Boolean(optional=True), prefix="--keep-temp", doc=InputDocumentation(doc="export intermediate graph file.")), ToolInput(tag="in_time_limit", input_type=Int(optional=True), prefix="--time-limit", doc=InputDocumentation(doc="time limit for the disentangling process. Default:3600")), ToolInput(tag="in_random_seed", input_type=Int(optional=True), prefix="--random-seed", doc=InputDocumentation(doc="Random seed (only for disentangling at this moment).\nDefault: 12345")), ToolInput(tag="in_continue", input_type=Boolean(optional=True), prefix="--continue", doc=InputDocumentation(doc="continue mode.")), ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="--verbose", doc=InputDocumentation(doc="verbose logging.")), ToolInput(tag="in_debug", input_type=Boolean(optional=True), prefix="--debug", doc=InputDocumentation(doc="for debug."))], outputs=[ToolOutput(tag="out_output_directory", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_output_directory", type_hint=File()), doc=OutputDocumentation(doc="output directory.")), ToolOutput(tag="out_prefix", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_prefix", type_hint=File()), doc=OutputDocumentation(doc="Prefix of output files inside output directory.\nDefault:target"))], container="quay.io/biocontainers/getorganelle:1.7.4.1--pyh5e36f6f_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Disentangle_Organelle_Assembly_Py_V0_1_0().translate("wdl")

