from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Int, Directory, String

Pathoscope_Map_V0_1_0 = CommandToolBuilder(tool="pathoscope_MAP", base_command=["pathoscope", "MAP"], inputs=[ToolInput(tag="in_input_read_fastq_file_unpairedsingleend", input_type=File(optional=True), prefix="-U", doc=InputDocumentation(doc="Input Read Fastq File (Unpaired/Single-end)")), ToolInput(tag="in_one", input_type=Int(optional=True), prefix="-1", doc=InputDocumentation(doc="Input Read Fastq File (Pair 1)")), ToolInput(tag="in_two", input_type=Int(optional=True), prefix="-2", doc=InputDocumentation(doc="Input Read Fastq File (Pair 2)")), ToolInput(tag="in_target_ref_files", input_type=File(optional=True), prefix="-targetRefFiles", doc=InputDocumentation(doc="Target Reference Genome Fasta Files Full Path (Comma\nSeparated)")), ToolInput(tag="in_filter_ref_files", input_type=File(optional=True), prefix="-filterRefFiles", doc=InputDocumentation(doc="Filter Reference Genome Fasta Files Full Path (Comma\nSeparated)")), ToolInput(tag="in_target_align_params", input_type=Int(optional=True), prefix="-targetAlignParams", doc=InputDocumentation(doc="Target Mapping Bowtie2 Parameters (Default: Pathoscope\nchosen best parameters)")), ToolInput(tag="in_filter_align_params", input_type=Int(optional=True), prefix="-filterAlignParams", doc=InputDocumentation(doc="Filter Mapping Bowtie2 Parameters (Default: Use the\nsame Target Mapping Bowtie2 parameters)")), ToolInput(tag="in_outdir", input_type=Directory(optional=True), prefix="-outDir", doc=InputDocumentation(doc="Output Directory (Default=. (current directory))")), ToolInput(tag="in_out_align", input_type=File(optional=True), prefix="-outAlign", doc=InputDocumentation(doc="Output Alignment File Name (Default=outalign.sam)")), ToolInput(tag="in_index_dir", input_type=Directory(optional=True), prefix="-indexDir", doc=InputDocumentation(doc="Index Directory (Default=. (current directory))")), ToolInput(tag="in_target_index_prefixes", input_type=String(optional=True), prefix="-targetIndexPrefixes", doc=InputDocumentation(doc="Target Index Prefixes (Comma Separated)")), ToolInput(tag="in_filter_index_prefixes", input_type=String(optional=True), prefix="-filterIndexPrefixes", doc=InputDocumentation(doc="Filter Index Prefixes (Comma Separated)")), ToolInput(tag="in_target_align_files", input_type=File(optional=True), prefix="-targetAlignFiles", doc=InputDocumentation(doc="Target Alignment Files Full Path (Comma Separated)")), ToolInput(tag="in_filter_align_files", input_type=File(optional=True), prefix="-filterAlignFiles", doc=InputDocumentation(doc="Filter Alignment Files Full Path (Comma Separated)")), ToolInput(tag="in_bt_home", input_type=File(optional=True), prefix="-btHome", doc=InputDocumentation(doc="Full Path to Bowtie2 binary directory (Default: Uses\nbowtie2 in system path)")), ToolInput(tag="in_num_threads", input_type=Int(optional=True), prefix="-numThreads", doc=InputDocumentation(doc="Number of threads to use by aligner (bowtie2) if\ndifferent from default (8)")), ToolInput(tag="in_exp_tag", input_type=String(optional=True), prefix="-expTag", doc=InputDocumentation(doc="Experiment Tag added to files generated for\nidentification (Default: pathomap)\n"))], outputs=[ToolOutput(tag="out_outdir", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_outdir", type_hint=File()), doc=OutputDocumentation(doc="Output Directory (Default=. (current directory))")), ToolOutput(tag="out_out_align", output_type=File(optional=True), selector=InputSelector(input_to_select="in_out_align", type_hint=File()), doc=OutputDocumentation(doc="Output Alignment File Name (Default=outalign.sam)"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Pathoscope_Map_V0_1_0().translate("wdl", allow_empty_container=True)

