class: CommandLineTool
id: run_snp_pipeline.sh.cwl
inputs:
- id: reference_file
  doc: Relative or absolute path to the reference fasta file
  type: string
  inputBinding:
    position: 0
- id: force
  doc: 'Force processing even when result files already exist and are newer than inputs
    (default: False)'
  type: boolean
  inputBinding:
    prefix: --force
- id: mirror
  doc: 'Create a mirror copy of the reference directory and all the sample directories.  Use
    this option to avoid polluting the reference directory and sample directories
    with the intermediate files generated by the snp pipeline.  A "reference" subdirectory
    and a "samples" subdirectory are created under the output directory (see the -o
    option).  One directory per sample is created under the "samples" directory. Three
    suboptions allow a choice of how the reference and samples are mirrored. -m soft
    : creates soft links to the fasta and fastq files instead of copying -m hard :
    creates hard links to the fasta and fastq files instead of copying -m copy : copies
    the fasta and fastq files (default: None)'
  type: string
  inputBinding:
    prefix: --mirror
- id: conf
  doc: 'Relative or absolute path to a configuration file for overriding defaults
    and defining extra parameters for the tools and scripts within the pipeline. Note:
    A default parameter configuration file named "snppipeline.conf" is used whenever
    the pipeline is run without the -c option.  The configuration file used for each
    run is copied into the log directory, capturing the parameters used during the
    run. (default: None)'
  type: File
  inputBinding:
    prefix: --conf
- id: q
  doc: '|torque, --queue_mgr grid|torque Job queue manager for remote parallel job
    execution in an HPC environment. Currently "grid" and "torque" are supported.
    If not specified, the pipeline will execute locally. (default: None)'
  type: string
  inputBinding:
    prefix: -Q
- id: out_dir
  doc: 'Output directory for the result files. Additional subdirectories are automatically
    created under the output directory for logs files and the mirrored samples and
    reference files (see the -m option). The output directory will be created if it
    does not already exist. If not specified, the output files are written to the
    current working directory. If you re- run the pipeline on previously processed
    samples, and specify a different output directory, the pipeline will not rebuild
    everything unless you either force a rebuild (see the -f option) or you request
    mirrored inputs (see the -m option). (default: .)'
  type: string
  inputBinding:
    prefix: --out_dir
- id: samples_dir
  doc: 'Relative or absolute path to the parent directory of all the sample directories.  The
    -s option should be used when all the sample directories are in subdirectories
    immediately below a parent directory. Note: You must specify either the -s or
    -S option, but not both. Note: The specified directory should contain only a collection
    of sample directories, nothing else. Note: Unless you request mirrored inputs,
    see the -m option, additional files will be written to each of the sample directories
    during the execution of the SNP Pipeline (default: None)'
  type: string
  inputBinding:
    prefix: --samples_dir
- id: samples_file
  doc: 'Relative or absolute path to a file listing all of the sample directories.  The
    -S option should be used when the samples are not under a common parent directory.
    Note: If you are not mirroring the samples (see the -m option), you can improve
    parallel processing performance by sorting the the list of directories descending
    by size, largest first. The -m option automatically generates a sorted directory
    list. Note: You must specify either the -s or -S option, but not both. Note: Unless
    you request mirrored inputs, see the -m option, additional files will be written
    to each of the sample directories during the execution of the SNP Pipeline (default:
    None)'
  type: File
  inputBinding:
    prefix: --samples_file
- id: purge
  doc: 'Purge the intermediate output files (the entire "samples" directory) when
    the pipeline completes successfully. (default: False)'
  type: boolean
  inputBinding:
    prefix: --purge
- id: v
  doc: '0..5, --verbose 0..5 Verbose message level (0=no info, 5=lots) (default: 1)'
  type: boolean
  inputBinding:
    prefix: -v
outputs: []
cwlVersion: v1.1
baseCommand:
- run_snp_pipeline.sh
