from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Directory, Boolean, Int, String, Float

Unfazed_V0_1_0 = CommandToolBuilder(tool="unfazed", base_command=["unfazed"], inputs=[ToolInput(tag="in_dn_ms", input_type=File(optional=True), prefix="--dnms", doc=InputDocumentation(doc="valid VCF OR BED file of the DNMs of interest> If BED,\nmust contain chrom, start, end, kid_id, var_type\ncolumns (default: None)")), ToolInput(tag="in_sites", input_type=File(optional=True), prefix="--sites", doc=InputDocumentation(doc="sorted/bgzipped/indexed VCF/BCF file of SNVs to\nidentify informative sites. Must contain each kid and\nboth parents (default: None)")), ToolInput(tag="in_ped", input_type=File(optional=True), prefix="--ped", doc=InputDocumentation(doc="ped file including the kid and both parent IDs\n(default: None)")), ToolInput(tag="in_bam_dir", input_type=Directory(optional=True), prefix="--bam-dir", doc=InputDocumentation(doc="directory where bam/cram files (named {sample_id}.bam\nor {sample_id}.cram) are stored for offspring. If not\nincluded, --bam-pairs must be set (default: None)")), ToolInput(tag="in_bam_pairs", input_type=Boolean(optional=True), prefix="--bam-pairs", doc=InputDocumentation(doc="[BAM_PAIRS [BAM_PAIRS ...]]\nspace-delimited list of pairs in the format\n{sample_id}:{bam_path} where {sample_id} matches an\noffspring id from the dnm file. Can be used with\n--bam-dir arg, must be used in its absence (default:\nNone)")), ToolInput(tag="in_threads", input_type=Int(optional=True), prefix="--threads", doc=InputDocumentation(doc="number of threads to use (default: 2)")), ToolInput(tag="in_output_type", input_type=String(optional=True), prefix="--output-type", doc=InputDocumentation(doc="choose output type. If --dnms is not a VCF/BCF, output\nmust be to BED format. Defaults to match --dnms input\nfile (default: None)")), ToolInput(tag="in_include_ambiguous", input_type=Boolean(optional=True), prefix="--include-ambiguous", doc=InputDocumentation(doc="include ambiguous phasing results (default: False)")), ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="--verbose", doc=InputDocumentation(doc="print verbose output including sites and reads used\nfor phasing. Only applies to BED output (default:\nFalse)")), ToolInput(tag="in_outfile", input_type=File(optional=True), prefix="--outfile", doc=InputDocumentation(doc="name for output file. Defaults to stdout (default:\n/dev/stdout)")), ToolInput(tag="in_reference", input_type=File(optional=True), prefix="--reference", doc=InputDocumentation(doc="reference fasta file (required for crams) (default:\nNone)")), ToolInput(tag="in_build", input_type=String(optional=True), prefix="--build", doc=InputDocumentation(doc="human genome build, used to determine sex chromosome\npseudoautosomal regions. If `na` option is chosen, sex\nchromosomes will not be auto-phased. HG19/GRCh37\ninterchangeable (default: None)")), ToolInput(tag="in_no_extended", input_type=Boolean(optional=True), prefix="--no-extended", doc=InputDocumentation(doc="do not perform extended read-based phasing (default\nTrue) (default: False)")), ToolInput(tag="in_multi_read_proc_min", input_type=Int(optional=True), prefix="--multiread-proc-min", doc=InputDocumentation(doc="min number of variants required to perform multiple\nparallel reads of the sites file (default: 1000)")), ToolInput(tag="in_quiet", input_type=Boolean(optional=True), prefix="--quiet", doc=InputDocumentation(doc="no logging of variant processing data (default: False)")), ToolInput(tag="in_min_gt_qual", input_type=Int(optional=True), prefix="--min-gt-qual", doc=InputDocumentation(doc="min genotype and base quality for informative sites\n(default: 20)")), ToolInput(tag="in_min_depth", input_type=Int(optional=True), prefix="--min-depth", doc=InputDocumentation(doc="min coverage for informative sites (default: 10)")), ToolInput(tag="in_ab_hom_ref", input_type=Float(optional=True), prefix="--ab-homref", doc=InputDocumentation(doc="allele balance range for homozygous reference\ninformative sites (default: 0.0:0.2)")), ToolInput(tag="in_ab_ho_malt", input_type=Float(optional=True), prefix="--ab-homalt", doc=InputDocumentation(doc="allele balance range for homozygous alternate\ninformative sites (default: 0.8:1.0)")), ToolInput(tag="in_ab_het", input_type=Float(optional=True), prefix="--ab-het", doc=InputDocumentation(doc="allele balance range for heterozygous informative\nsites (default: 0.2:0.8)")), ToolInput(tag="in_evidence_min_ratio", input_type=Int(optional=True), prefix="--evidence-min-ratio", doc=InputDocumentation(doc="minimum ratio of evidence for a parent to provide an\nunambiguous call. Default 10:1 (default: 10)")), ToolInput(tag="in_search_dist", input_type=Int(optional=True), prefix="--search-dist", doc=InputDocumentation(doc="maximum search distance from variant for informative\nsites (in bases) (default: 5000)")), ToolInput(tag="in_insert_size_max_sample", input_type=Int(optional=True), prefix="--insert-size-max-sample", doc=InputDocumentation(doc="maximum number of read inserts to sample in order to\nestimate concordant read insert size (default:\n1000000)")), ToolInput(tag="in_min_map_qual", input_type=Int(optional=True), prefix="--min-map-qual", doc=InputDocumentation(doc="minimum map quality for reads (default: 1)")), ToolInput(tag="in_stdev_s", input_type=Int(optional=True), prefix="--stdevs", doc=InputDocumentation(doc="number of standard deviations from the mean insert\nlength to define a discordant read (default: 3)")), ToolInput(tag="in_read_len", input_type=Int(optional=True), prefix="--readlen", doc=InputDocumentation(doc="expected length of input reads (default: 151)")), ToolInput(tag="in_split_error_margin", input_type=Int(optional=True), prefix="--split-error-margin", doc=InputDocumentation(doc="margin of error for the location of split read\nclipping in bases (default: 5)")), ToolInput(tag="in_max_reads", input_type=Int(optional=True), prefix="--max-reads", doc=InputDocumentation(doc="maximum number of reads to collect for phasing a\nsingle variant (default: 100)\n")), ToolInput(tag="in_v", input_type=Boolean(optional=True), prefix="-v", doc=InputDocumentation(doc=""))], outputs=[ToolOutput(tag="out_outfile", output_type=File(optional=True), selector=InputSelector(input_to_select="in_outfile", type_hint=File()), doc=OutputDocumentation(doc="name for output file. Defaults to stdout (default:\n/dev/stdout)"))], container="quay.io/biocontainers/unfazed:1.0.2--pyh3252c3a_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Unfazed_V0_1_0().translate("wdl")

