from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, File, String

Mapbed_V0_1_0 = CommandToolBuilder(tool="mapBed", base_command=["mapBed"], inputs=[ToolInput(tag="in_specify_columns_map", input_type=Boolean(optional=True), prefix="-c", doc=InputDocumentation(doc="Specify columns from the B file to map onto intervals in A.\nDefault: 5.\nMultiple columns can be specified in a comma-delimited list.")), ToolInput(tag="in_specify_operation_applied", input_type=Boolean(optional=True), prefix="-o", doc=InputDocumentation(doc="Specify the operation that should be applied to -c.\nValid operations:\nsum, min, max, absmin, absmax,\nmean, median, mode, antimode\nstdev, sstdev\ncollapse (i.e., print a delimited list (duplicates allowed)),\ndistinct (i.e., print a delimited list (NO duplicates allowed)),\ndistinct_sort_num (as distinct, sorted numerically, ascending),\ndistinct_sort_num_desc (as distinct, sorted numerically, desscending),\ndistinct_only (delimited list of only unique values),\ncount\ncount_distinct (i.e., a count of the unique values in the column),\nfirst (i.e., just the first value in the column),\nlast (i.e., just the last value in the column),\nDefault: sum\nMultiple operations can be specified in a comma-delimited list.\nIf there is only column, but multiple operations, all operations will be\napplied on that column. Likewise, if there is only one operation, but\nmultiple columns, that operation will be applied to all columns.\nOtherwise, the number of columns must match the the number of operations,\nand will be applied in respective order.\nE.g., '-c 5,4,6 -o sum,mean,count' will give the sum of column 5,\nthe mean of column 4, and the count of column 6.\nThe order of output columns will match the ordering given in the command.")), ToolInput(tag="in_delim", input_type=Boolean(optional=True), prefix="-delim", doc=InputDocumentation(doc="Specify a custom delimiter for the collapse operations.\n- Example: -delim '|'\n- Default: ','.")), ToolInput(tag="in_prec", input_type=Boolean(optional=True), prefix="-prec", doc=InputDocumentation(doc="Sets the decimal precision for output (Default: 5)")), ToolInput(tag="in_require_same_strandedness", input_type=Boolean(optional=True), prefix="-s", doc=InputDocumentation(doc="Require same strandedness.  That is, only report hits in B\nthat overlap A on the _same_ strand.\n- By default, overlaps are reported without respect to strand.")), ToolInput(tag="in_require_different_strandedness", input_type=Boolean(optional=True), prefix="-S", doc=InputDocumentation(doc="Require different strandedness.  That is, only report hits in B\nthat overlap A on the _opposite_ strand.\n- By default, overlaps are reported without respect to strand.")), ToolInput(tag="in_minimum_overlap_a", input_type=Boolean(optional=True), prefix="-f", doc=InputDocumentation(doc="Minimum overlap required as a fraction of A.\n- Default is 1E-9 (i.e., 1bp).\n- FLOAT (e.g. 0.50)")), ToolInput(tag="in_minimum_overlap_b", input_type=Boolean(optional=True), prefix="-F", doc=InputDocumentation(doc="Minimum overlap required as a fraction of B.\n- Default is 1E-9 (i.e., 1bp).\n- FLOAT (e.g. 0.50)")), ToolInput(tag="in_require_overlap_be", input_type=Boolean(optional=True), prefix="-r", doc=InputDocumentation(doc="Require that the fraction overlap be reciprocal for A AND B.\n- In other words, if -f is 0.90 and -r is used, this requires\nthat B overlap 90% of A and A _also_ overlaps 90% of B.")), ToolInput(tag="in_require_fraction_be", input_type=Boolean(optional=True), prefix="-e", doc=InputDocumentation(doc="Require that the minimum fraction be satisfied for A OR B.\n- In other words, if -e is used with -f 0.90 and -F 0.10 this requires\nthat either 90% of A is covered OR 10% of  B is covered.\nWithout -e, both fractions would have to be satisfied.")), ToolInput(tag="in_split", input_type=Boolean(optional=True), prefix="-split", doc=InputDocumentation(doc="Treat 'split' BAM or BED12 entries as distinct BED intervals.")), ToolInput(tag="in_provide_genome_file", input_type=Boolean(optional=True), prefix="-g", doc=InputDocumentation(doc="Provide a genome file to enforce consistent chromosome sort order\nacross input files. Only applies when used with -sorted option.")), ToolInput(tag="in_no_name_check", input_type=Boolean(optional=True), prefix="-nonamecheck", doc=InputDocumentation(doc="For sorted data, don't throw an error if the file has different naming conventions\nfor the same chromosome. ex. 'chr1' vs 'chr01'.")), ToolInput(tag="in_bed", input_type=Boolean(optional=True), prefix="-bed", doc=InputDocumentation(doc="If using BAM input, write output as BED.")), ToolInput(tag="in_header", input_type=File(optional=True), prefix="-header", doc=InputDocumentation(doc="the header from the A file prior to results.")), ToolInput(tag="in_no_buf", input_type=Boolean(optional=True), prefix="-nobuf", doc=InputDocumentation(doc="Disable buffered output. Using this option will cause each line\nof output to be printed as it is generated, rather than saved\nin a buffer. This will make printing large output files\nnoticeably slower, but can be useful in conjunction with\nother software tools and scripts that need to process one\nline of bedtools output at a time.")), ToolInput(tag="in_i_obuf", input_type=Boolean(optional=True), prefix="-iobuf", doc=InputDocumentation(doc="Specify amount of memory to use for input buffer.\nTakes an integer argument. Optional suffixes K/M/G supported.\nNote: currently has no effect with compressed files.")), ToolInput(tag="in_b", input_type=String(optional=True), prefix="-b", doc=InputDocumentation(doc="")), ToolInput(tag="in_a", input_type=String(optional=True), prefix="-a", doc=InputDocumentation(doc="")), ToolInput(tag="in_bed_tools", input_type=String(), position=0, doc=InputDocumentation(doc="")), ToolInput(tag="in_map", input_type=String(), position=1, doc=InputDocumentation(doc=""))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Mapbed_V0_1_0().translate("wdl", allow_empty_container=True)

