from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, String, Int, File, Array

Pauvre_Redwood_V0_1_0 = CommandToolBuilder(tool="pauvre_redwood", base_command=["pauvre", "redwood"], inputs=[ToolInput(tag="in_quiet", input_type=Boolean(optional=True), prefix="--quiet", doc=InputDocumentation(doc="Do not output warnings to stderr")), ToolInput(tag="in_your_fasta_file", input_type=String(optional=True), prefix="-d", doc=InputDocumentation(doc="[{main,rnaseq} ...], --doubled {main,rnaseq} [{main,rnaseq} ...]\nIf your fasta file was doubled to map circularly, use\nthis flag or you may encounter plotting errors.\nAccepts multiple arguments. 'main' is for the sam file\npassed with --sam, 'rnaseq' is for the sam file passed\nwith --rnaseq")), ToolInput(tag="in_dpi", input_type=Int(optional=True), prefix="--dpi", doc=InputDocumentation(doc="Change the dpi from the default 600 if you need it")), ToolInput(tag="in_gff", input_type=File(optional=True), prefix="--gff", doc=InputDocumentation(doc="The input filepath for the gff annotation to plot")), ToolInput(tag="in_interlace", input_type=Boolean(optional=True), prefix="--interlace", doc=InputDocumentation(doc="Interlace the reads so the pileup plot looks better.\nDoesn't work currently")), ToolInput(tag="in_invert", input_type=Boolean(optional=True), prefix="--invert", doc=InputDocumentation(doc="invert the image so that it looks better on a dark\nbackground. DOESN'T DO ANYTHING.")), ToolInput(tag="in_log", input_type=Boolean(optional=True), prefix="--log", doc=InputDocumentation(doc="Plot the RNAseq track with a log scale")), ToolInput(tag="in_main_bam", input_type=File(optional=True), prefix="--main_bam", doc=InputDocumentation(doc="The input filepath for the bam file to plot. Ideally\nwas plotted with a fasta file that is two copies of\nthe mitochondrial genome concatenated. This should be\nlong reads (ONT, PB) and will be displayed in the\ninterior of the redwood plot.")), ToolInput(tag="in_no_timestamp", input_type=File(optional=True), prefix="--no_timestamp", doc=InputDocumentation(doc="Turn off time stamps in the filename output.")), ToolInput(tag="in_output_base_name", input_type=File(optional=True), prefix="--output-base-name", doc=InputDocumentation(doc="Specify a base name for the output file(s). The input\nfile base name is the default.")), ToolInput(tag="in_query", input_type=Array(t=String(), optional=True), prefix="--query", doc=InputDocumentation(doc="Query your reads to change plotting options")), ToolInput(tag="in_rnase_q_bam", input_type=File(optional=True), prefix="--rnaseq_bam", doc=InputDocumentation(doc="The input filepath for the rnaseq bam file to plot")), ToolInput(tag="in_small_start", input_type=String(optional=True), prefix="--small_start", doc=InputDocumentation(doc="This determines where the shortest of the filtered\nreads will appear on the redwood plot: on the outside\nor on the inside? The default option puts the longest\nreads on the outside and the shortest reads on the\ninside.")), ToolInput(tag="in_sort", input_type=String(optional=True), prefix="--sort", doc=InputDocumentation(doc="What value to use to sort the order in which the reads\nare plotted?")), ToolInput(tag="in_ticks", input_type=Array(t=String(), optional=True), prefix="--ticks", doc=InputDocumentation(doc="Specify control for the number of ticks.")), ToolInput(tag="in_transparent", input_type=Boolean(optional=True), prefix="--transparent", doc=InputDocumentation(doc="Specify this option if you DON'T want a transparent\nbackground. Default is on.\n")), ToolInput(tag="in_higher", input_type=String(), position=0, doc=InputDocumentation(doc="--fileform STRING [STRING ...]"))], outputs=[ToolOutput(tag="out_no_timestamp", output_type=File(optional=True), selector=InputSelector(input_to_select="in_no_timestamp", type_hint=File()), doc=OutputDocumentation(doc="Turn off time stamps in the filename output."))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Pauvre_Redwood_V0_1_0().translate("wdl", allow_empty_container=True)

