from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, Directory, Int, Array, String, File

Pauvre_Synplot_V0_1_0 = CommandToolBuilder(tool="pauvre_synplot", base_command=["pauvre", "synplot"], inputs=[ToolInput(tag="in_quiet", input_type=Boolean(optional=True), prefix="--quiet", doc=InputDocumentation(doc="Do not output warnings to stderr")), ToolInput(tag="in_aln_dir", input_type=Directory(optional=True), prefix="--aln_dir", doc=InputDocumentation(doc="The directory where all the fasta alignments are")), ToolInput(tag="in_center_on", input_type=Int(optional=True), prefix="--center_on", doc=InputDocumentation(doc="Change the dpi from the default 600 if you need it")), ToolInput(tag="in_gff_paths", input_type=Array(t=String(), optional=True), prefix="--gff_paths", doc=InputDocumentation(doc="The input filepath. for the gff annotation to plot.\nIndividual filepaths separated by spaces. For example,\n--gff_paths sp1.gff sp2.gff sp3.gff")), ToolInput(tag="in_gff_labels", input_type=Array(t=String(), optional=True), prefix="--gff_labels", doc=InputDocumentation(doc="In case the gff names and sequence names don't match,\nchange the labels that will appear over the text.")), ToolInput(tag="in_no_timestamp", input_type=File(optional=True), prefix="--no_timestamp", doc=InputDocumentation(doc="Turn off time stamps in the filename output.")), ToolInput(tag="in_optimum_order", input_type=Boolean(optional=True), prefix="--optimum_order", doc=InputDocumentation(doc="If selected, this doesn't plot the optimum arrangement\nof things as they are input into gff_paths. Instead,\nit uses the first gff file as the top-most sequence in\nthe plot, and reorganizes the remaining gff files to\nminimize the number of intersections.")), ToolInput(tag="in_output_basename", input_type=File(optional=True), prefix="--output-basename", doc=InputDocumentation(doc="Specify a base name for the output file(s). The input\nfile base name is the default.")), ToolInput(tag="in_ratio", input_type=Array(t=String(), optional=True), prefix="--ratio", doc=InputDocumentation(doc="Enter the dimensions (arbitrary units) to plot the\nfigure. For example a figure that is seven times wider\nthan tall is: --ratio 7 1")), ToolInput(tag="in_sandwich", input_type=Boolean(optional=True), prefix="--sandwich", doc=InputDocumentation(doc="Put an additional copy of the first gff file on the\nbottom of the plot for comparison.")), ToolInput(tag="in_start_with_aligned_genes", input_type=Boolean(optional=True), prefix="--start_with_aligned_genes", doc=InputDocumentation(doc="Minimizes the number of intersections but only selects\ncombos where the first gene in each sequence is\naligned.")), ToolInput(tag="in_stop_codons", input_type=Boolean(optional=True), prefix="--stop_codons", doc=InputDocumentation(doc="Performs some internal corrections if the gff\nannotation includes the stop codons in the coding\nsequences.")), ToolInput(tag="in_transparent", input_type=Boolean(optional=True), prefix="--transparent", doc=InputDocumentation(doc="Specify this option if you DON'T want a transparent\nbackground. Default is on.\n")), ToolInput(tag="in_contained_dot", input_type=String(), position=0, doc=InputDocumentation(doc="--center_on CENTER_ON")), ToolInput(tag="in_higher", input_type=String(), position=0, doc=InputDocumentation(doc="--fileform {png,pdf,eps,jpeg,jpg,pdf,pgf,ps,raw,rgba,svg,svgz,tif,tiff} [{png,pdf,eps,jpeg,jpg,pdf,pgf,ps,raw,rgba,svg,svgz,tif,tiff} ...]"))], outputs=[ToolOutput(tag="out_no_timestamp", output_type=File(optional=True), selector=InputSelector(input_to_select="in_no_timestamp", type_hint=File()), doc=OutputDocumentation(doc="Turn off time stamps in the filename output."))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Pauvre_Synplot_V0_1_0().translate("wdl", allow_empty_container=True)

