from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, File, String, Int

Jgi_Summarize_Bam_Contig_Depths_V0_1_0 = CommandToolBuilder(tool="jgi_summarize_bam_contig_depths", base_command=["jgi_summarize_bam_contig_depths"], inputs=[ToolInput(tag="in_output_depth", input_type=Boolean(optional=True), prefix="--outputDepth", doc=InputDocumentation(doc="arg  The file to put the contig by bam depth matrix (default: STDOUT)")), ToolInput(tag="in_percent_identity", input_type=Boolean(optional=True), prefix="--percentIdentity", doc=InputDocumentation(doc="arg  The minimum end-to-end % identity of qualifying reads (default: 97)")), ToolInput(tag="in_paired_contigs", input_type=File(optional=True), prefix="--pairedContigs", doc=InputDocumentation(doc="arg  The file to output the sparse matrix of contigs which paired reads span (default: none)")), ToolInput(tag="in_unmapped_fast_q", input_type=File(optional=True), prefix="--unmappedFastq", doc=InputDocumentation(doc="arg  The prefix to output unmapped reads from each bam file suffixed by 'bamfile.bam.fastq.gz'")), ToolInput(tag="in_no_intra_depth_variance", input_type=Boolean(optional=True), prefix="--noIntraDepthVariance", doc=InputDocumentation(doc="Do not include variance from mean depth along the contig")), ToolInput(tag="in_show_depth", input_type=File(optional=True), prefix="--showDepth", doc=InputDocumentation(doc="Output a .depth file per bam for each contig base")), ToolInput(tag="in_min_map_qual", input_type=Boolean(optional=True), prefix="--minMapQual", doc=InputDocumentation(doc="arg  The minimum mapping quality necessary to count the read as mapped (default: 0)")), ToolInput(tag="in_weight_map_qual", input_type=Boolean(optional=True), prefix="--weightMapQual", doc=InputDocumentation(doc="arg  Weight per-base depth based on the MQ of the read (i.e uniqueness) (default: 0.0 (disabled))")), ToolInput(tag="in_include_edge_bases", input_type=Boolean(optional=True), prefix="--includeEdgeBases", doc=InputDocumentation(doc="When calculating depth & variance, include the 1-readlength edges (off by default)")), ToolInput(tag="in_max_edge_bases", input_type=Boolean(optional=True), prefix="--maxEdgeBases", doc=InputDocumentation(doc="When calculating depth & variance, and not --includeEdgeBases, the maximum length (default:75)")), ToolInput(tag="in_reference_fast_a", input_type=Boolean(optional=True), prefix="--referenceFasta", doc=InputDocumentation(doc="arg  The reference file.  (It must be the same fasta that bams used)")), ToolInput(tag="in_output_gc", input_type=Boolean(optional=True), prefix="--outputGC", doc=InputDocumentation(doc="arg  The file to print the gc coverage histogram")), ToolInput(tag="in_gc_window", input_type=Boolean(optional=True), prefix="--gcWindow", doc=InputDocumentation(doc="arg  The sliding window size for GC calculations")), ToolInput(tag="in_output_read_stats", input_type=Boolean(optional=True), prefix="--outputReadStats", doc=InputDocumentation(doc="arg  The file to print the per read statistics")), ToolInput(tag="in_output_km_ers", input_type=Boolean(optional=True), prefix="--outputKmers", doc=InputDocumentation(doc="arg  The file to print the perfect kmer counts")), ToolInput(tag="in_shred_length", input_type=Boolean(optional=True), prefix="--shredLength", doc=InputDocumentation(doc="arg  The maximum length of the shreds")), ToolInput(tag="in_shred_depth", input_type=Boolean(optional=True), prefix="--shredDepth", doc=InputDocumentation(doc="arg  The depth to generate overlapping shreds")), ToolInput(tag="in_min_contig_length", input_type=Boolean(optional=True), prefix="--minContigLength", doc=InputDocumentation(doc="arg  The mimimum length of contig to include for mapping and shredding")), ToolInput(tag="in_min_contig_depth", input_type=Boolean(optional=True), prefix="--minContigDepth", doc=InputDocumentation(doc="arg  The minimum depth along contig at which to break the contig")), ToolInput(tag="in_options", input_type=String(), position=0, doc=InputDocumentation(doc="")), ToolInput(tag="in_sorted_bam_one", input_type=Int(), position=1, doc=InputDocumentation(doc="")), ToolInput(tag="in_sorted_bam_two", input_type=Int(optional=True), position=2, doc=InputDocumentation(doc=""))], outputs=[ToolOutput(tag="out_paired_contigs", output_type=File(optional=True), selector=InputSelector(input_to_select="in_paired_contigs", type_hint=File()), doc=OutputDocumentation(doc="arg  The file to output the sparse matrix of contigs which paired reads span (default: none)")), ToolOutput(tag="out_unmapped_fast_q", output_type=File(optional=True), selector=InputSelector(input_to_select="in_unmapped_fast_q", type_hint=File()), doc=OutputDocumentation(doc="arg  The prefix to output unmapped reads from each bam file suffixed by 'bamfile.bam.fastq.gz'")), ToolOutput(tag="out_show_depth", output_type=File(optional=True), selector=InputSelector(input_to_select="in_show_depth", type_hint=File()), doc=OutputDocumentation(doc="Output a .depth file per bam for each contig base"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Jgi_Summarize_Bam_Contig_Depths_V0_1_0().translate("wdl", allow_empty_container=True)

