from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Directory, Boolean, String, Int

Msstitch_Filterperco_V0_1_0 = CommandToolBuilder(tool="msstitch_filterperco", base_command=["msstitch", "filterperco"], inputs=[ToolInput(tag="in_input_file_format", input_type=File(optional=True), prefix="-i", doc=InputDocumentation(doc="Input file of {} format")), ToolInput(tag="in_directory_to_output", input_type=Directory(optional=True), prefix="-d", doc=InputDocumentation(doc="Directory to output in")), ToolInput(tag="in_output_file", input_type=File(optional=True), prefix="-o", doc=InputDocumentation(doc="Output file")), ToolInput(tag="in_full_protein", input_type=Boolean(optional=True), prefix="--fullprotein", doc=InputDocumentation(doc="Store full protein sequences (at a minimum-match\nlength) in the SQLite file rather than tryptic\nsequences")), ToolInput(tag="in_de_ami_date", input_type=Boolean(optional=True), prefix="--deamidate", doc=InputDocumentation(doc="Filter against both normal peptides and deamidated\npeptides where a D->N transition has occurred.")), ToolInput(tag="in_fast_a", input_type=String(optional=True), prefix="--fasta", doc=InputDocumentation(doc="FASTA sequence database")), ToolInput(tag="in_min_len", input_type=Int(optional=True), prefix="--minlen", doc=InputDocumentation(doc="Minimum length of peptide to be included")), ToolInput(tag="in_dbfile", input_type=File(optional=True), prefix="--dbfile", doc=InputDocumentation(doc="Database lookup file")), ToolInput(tag="in_enforce_tryptic", input_type=Boolean(optional=True), prefix="--enforce-tryptic", doc=InputDocumentation(doc="When filtering peptides against whole proteins, filter\nout peptides that match a whole protein but only if\nthey are fully tryptic, i.e. the protein needs K,R 1\nposition upstream of the peptide, and the peptide\nC-terminal should also be K,R. Useful when discerning\nfrom pseudogenes")), ToolInput(tag="in_in_source_frag", input_type=Int(optional=True), prefix="--insourcefrag", doc=InputDocumentation(doc="Apply filter against both intact peptides and those\nthat match to the C-terminal part of a tryptic peptide\nfrom the database, resulting from in-source\nfragmentation, where some amino acids will be missing\nfrom the N-terminus. Specify the max number of amino\nacids that may be missing. Database should be built\nwith this flag in order for the lookup to work, since\nsequences will be stored and looked up reversed\n"))], outputs=[ToolOutput(tag="out_directory_to_output", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_directory_to_output", type_hint=File()), doc=OutputDocumentation(doc="Directory to output in")), ToolOutput(tag="out_output_file", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output_file", type_hint=File()), doc=OutputDocumentation(doc="Output file"))], container="quay.io/biocontainers/msstitch:3.5--py_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Msstitch_Filterperco_V0_1_0().translate("wdl")

