from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import String, File, Float, Boolean, Int

Purple_V0_1_0 = CommandToolBuilder(tool="PURPLE", base_command=["PURPLE"], inputs=[ToolInput(tag="in_circo_s", input_type=String(optional=True), prefix="-circos", doc=InputDocumentation(doc="Location of\ncircos binary")), ToolInput(tag="in_cobalt", input_type=File(optional=True), prefix="-cobalt", doc=InputDocumentation(doc="Path to COBALT")), ToolInput(tag="in_db_user", input_type=String(optional=True), prefix="-db_user", doc=InputDocumentation(doc="Database user")), ToolInput(tag="in_gc_profile", input_type=File(optional=True), prefix="-gc_profile", doc=InputDocumentation(doc="Path to GC")), ToolInput(tag="in_germline_vcf", input_type=String(optional=True), prefix="-germline_vcf", doc=InputDocumentation(doc="Optional\nlocation of\ngermline\nvariants to\nenrich and\nprocess in\ndriver catalog.")), ToolInput(tag="in_highly_diploid_percentage", input_type=Float(optional=True), prefix="-highly_diploid_percentage", doc=InputDocumentation(doc="Proportion of\ngenome that\nmust be diploid\nbefore using\nsomatic fit.\nDefault 0.97")), ToolInput(tag="in_max_ploidy", input_type=Float(optional=True), prefix="-max_ploidy", doc=InputDocumentation(doc="Maximum ploidy\n(default 8.0)")), ToolInput(tag="in_max_purity", input_type=Float(optional=True), prefix="-max_purity", doc=InputDocumentation(doc="Maximum purity\n(default 1.0)")), ToolInput(tag="in_min_diploid_tumor_ratio_count", input_type=String(optional=True), prefix="-min_diploid_tumor_ratio_count", doc=InputDocumentation(doc="Minimum ratio\ncount while\nsmoothing\nbefore diploid\nregions become\nsuspect.")), ToolInput(tag="in_min_diploid_tumor_ratio_count_centromere", input_type=String(optional=True), prefix="-min_diploid_tumor_ratio_count_centromere", doc=InputDocumentation(doc="Minimum ratio\ncount while\nsmoothing\nbefore diploid\nregions become\nsuspect while\napproaching\ncentromere.")), ToolInput(tag="in_min_ploidy", input_type=Float(optional=True), prefix="-min_ploidy", doc=InputDocumentation(doc="Minimum ploidy\n(default 1.0)")), ToolInput(tag="in_min_purity", input_type=Float(optional=True), prefix="-min_purity", doc=InputDocumentation(doc="Minimum purity\n(default 0.08)")), ToolInput(tag="in_no_charts", input_type=Boolean(optional=True), prefix="-no_charts", doc=InputDocumentation(doc="Disable charts")), ToolInput(tag="in_output_dir", input_type=File(optional=True), prefix="-output_dir", doc=InputDocumentation(doc="Path to the")), ToolInput(tag="in_ploidy_penalty_factor", input_type=Int(optional=True), prefix="-ploidy_penalty_factor", doc=InputDocumentation(doc="Penalty factor\nto apply to the\nnumber of copy\nnumber events")), ToolInput(tag="in_ploidy_penalty_min", input_type=String(optional=True), prefix="-ploidy_penalty_min", doc=InputDocumentation(doc="Minimum ploidy")), ToolInput(tag="in_ploidy_penalty_sub_min_additional", input_type=Int(optional=True), prefix="-ploidy_penalty_sub_min_additional", doc=InputDocumentation(doc="Additional\npenalty to\napply to major\nallele < 1 or\nminor allele <\n0")), ToolInput(tag="in_ploidy_penalty_sub_one_major_allele_multiplier", input_type=String(optional=True), prefix="-ploidy_penalty_sub_one_major_allele_multiplier", doc=InputDocumentation(doc="Penalty")), ToolInput(tag="in_purity_increment", input_type=String(optional=True), prefix="-purity_increment", doc=InputDocumentation(doc="Purity")), ToolInput(tag="in_ref_genome", input_type=File(optional=True), prefix="-ref_genome", doc=InputDocumentation(doc="Path to the ref\ngenome fasta\nfile.")), ToolInput(tag="in_name_of_the", input_type=String(optional=True), prefix="-reference", doc=InputDocumentation(doc="Name of the")), ToolInput(tag="in_somatic_min_peak", input_type=Int(optional=True), prefix="-somatic_min_peak", doc=InputDocumentation(doc="Minimum number\nof somatic\nvariants to\nconsider a\npeak. Default\n10")), ToolInput(tag="in_somatic_min_purity", input_type=Float(optional=True), prefix="-somatic_min_purity", doc=InputDocumentation(doc="Somatic fit\nwill not be\nused if both\nsomatic and\nfitted purities\nare less than\nthis value.\nDefault 0.17")), ToolInput(tag="in_somatic_min_purity_spread", input_type=String(optional=True), prefix="-somatic_min_purity_spread", doc=InputDocumentation(doc="Minimum spread")), ToolInput(tag="in_somatic_penalty_weight", input_type=String(optional=True), prefix="-somatic_penalty_weight", doc=InputDocumentation(doc="Proportion of")), ToolInput(tag="in_structural_vcf", input_type=String(optional=True), prefix="-structural_vcf", doc=InputDocumentation(doc="Optional\nlocation of\nstructural\nvariant vcf for\nmore accurate\nsegmentation.")), ToolInput(tag="in_sv_recovery_vcf", input_type=String(optional=True), prefix="-sv_recovery_vcf", doc=InputDocumentation(doc="Optional\nlocation of\nfailing\nstructural\nvariants that\nmay be\nrecovered.")), ToolInput(tag="in_number_of", input_type=Int(optional=True), prefix="-threads", doc=InputDocumentation(doc="Number of")), ToolInput(tag="in_tumor", input_type=String(optional=True), prefix="-tumor", doc=InputDocumentation(doc="Name of the\ntumor sample.\nThis should\ncorrespond to\nthe value used\nin AMBER and\nCOBALT.")), ToolInput(tag="in_tumor_only", input_type=Boolean(optional=True), prefix="-tumor_only", doc=InputDocumentation(doc="Tumor only\nmode. Disables\nsomatic\nfitting.")), ToolInput(tag="in_version", input_type=Boolean(optional=True), prefix="-version", doc=InputDocumentation(doc="Exit after")), ToolInput(tag="in_purity_ploidy_estimate_application", input_type=String(), position=0, doc=InputDocumentation(doc="-amber <arg>                                              Path to AMBER")), ToolInput(tag="in_tdot", input_type=String(), position=0, doc=InputDocumentation(doc="-db_enabled                                               Persist data to")), ToolInput(tag="in_dbpass_arg_database", input_type=String(), position=1, doc=InputDocumentation(doc="-db_pass <arg>                                            Database")), ToolInput(tag="in_password_dot", input_type=String(), position=2, doc=InputDocumentation(doc="-db_url <arg>                                             Database url in")), ToolInput(tag="in_po", input_type=String(), position=0, doc=InputDocumentation(doc="rt/database")), ToolInput(tag="in_name_dot", input_type=String(), position=1, doc=InputDocumentation(doc="-driver_catalog                                           Persist data to")), ToolInput(tag="in_drivergenepanel_arg_path", input_type=String(), position=2, doc=InputDocumentation(doc="-driver_gene_panel <arg>                                  Path to driver")), ToolInput(tag="in_profile_dot", input_type=String(), position=0, doc=InputDocumentation(doc="-germline_hotspots <arg>                                  Path to")), ToolInput(tag="in_germline", input_type=String(), position=1, doc=InputDocumentation(doc="hotspot VCF")), ToolInput(tag="in_output", input_type=String(), position=2, doc=InputDocumentation(doc="directory.")), ToolInput(tag="in_applied", input_type=String(), position=0, doc=InputDocumentation(doc="-ploidy_penalty_standard_deviation <arg>                  Standard")), ToolInput(tag="in_normal", input_type=String(), position=0, doc=InputDocumentation(doc="distribution")), ToolInput(tag="in_modelling", input_type=String(), position=1, doc=InputDocumentation(doc="ploidy")), ToolInput(tag="in_multiplier", input_type=String(), position=0, doc=InputDocumentation(doc="applied to")), ToolInput(tag="in_increment", input_type=String(), position=0, doc=InputDocumentation(doc="(default 0.01)")), ToolInput(tag="in_sample_this", input_type=String(), position=1, doc=InputDocumentation(doc="sample. This")), ToolInput(tag="in_should", input_type=String(), position=2, doc=InputDocumentation(doc="correspond to")), ToolInput(tag="in_cobalt_dot", input_type=String(), position=0, doc=InputDocumentation(doc="-somatic_hotspots <arg>                                   Path to somatic")), ToolInput(tag="in_within", input_type=String(), position=0, doc=InputDocumentation(doc="candidate")), ToolInput(tag="in_zero_dot_one_five", input_type=Float(), position=0, doc=InputDocumentation(doc="-somatic_min_variants <arg>                               Minimum number")), ToolInput(tag="in_variants", input_type=String(), position=0, doc=InputDocumentation(doc="required to")), ToolInput(tag="in_somatic", input_type=String(), position=0, doc=InputDocumentation(doc="deviation to")), ToolInput(tag="in_one_dot_zero", input_type=Float(), position=0, doc=InputDocumentation(doc="-somatic_vcf <arg>                                        Optional")), ToolInput(tag="in_highly_diploid", input_type=String(), position=0, doc=InputDocumentation(doc="samples.")), ToolInput(tag="in_default", input_type=String(), position=1, doc=InputDocumentation(doc="(default 2)")), ToolInput(tag="in_displaying", input_type=String(), position=2, doc=InputDocumentation(doc="version info."))], outputs=[], container="quay.io/biocontainers/hmftools-purple:2.53--0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Purple_V0_1_0().translate("wdl")

