from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean

Gstacks_V0_1_0 = CommandToolBuilder(tool="gstacks", base_command=["gstacks"], inputs=[ToolInput(tag="in_input_directory_containg", input_type=Boolean(optional=True), prefix="-P", doc=InputDocumentation(doc=": input directory containg '*.matches.bam' files created by the\nde novo Stacks pipeline, ustacks-cstacks-sstacks-tsv2bam")), ToolInput(tag="in_input_directory_containing", input_type=Boolean(optional=True), prefix="-I", doc=InputDocumentation(doc=": input directory containing BAM files")), ToolInput(tag="in_im_suffix_use", input_type=Boolean(optional=True), prefix="-S", doc=InputDocumentation(doc=": with -I/-M, suffix to use to build BAM file names: by default this\nis just '.bam', i.e. the program expects 'SAMPLE_NAME.bam'")), ToolInput(tag="in_input_bam_files", input_type=Boolean(optional=True), prefix="-B", doc=InputDocumentation(doc=": input BAM file(s)")), ToolInput(tag="in_path_population_map", input_type=Boolean(optional=True), prefix="-M", doc=InputDocumentation(doc=": path to a population map giving the list of samples")), ToolInput(tag="in_output_directory_default", input_type=Boolean(optional=True), prefix="-O", doc=InputDocumentation(doc=": output directory (default: none with -B; with -P same as the input\ndirectory)")), ToolInput(tag="in_threads", input_type=Boolean(optional=True), prefix="--threads", doc=InputDocumentation(doc=": number of threads to use (default: 1)")), ToolInput(tag="in_model", input_type=Boolean(optional=True), prefix="--model", doc=InputDocumentation(doc=": model to use to call variants and genotypes; one of\nmarukilow (default), marukihigh, or snp")), ToolInput(tag="in_var_alpha", input_type=Boolean(optional=True), prefix="--var-alpha", doc=InputDocumentation(doc=": alpha threshold for discovering SNPs (default: 0.01 for marukilow)")), ToolInput(tag="in_gt_alpha", input_type=Boolean(optional=True), prefix="--gt-alpha", doc=InputDocumentation(doc=": alpha threshold for calling genotypes (default: 0.05)")), ToolInput(tag="in_rm_pcr_duplicates", input_type=Boolean(optional=True), prefix="--rm-pcr-duplicates", doc=InputDocumentation(doc=": remove all but one set ofread pairs of the same sample that\nhave the same insert length (implies --rm-unpaired-reads)")), ToolInput(tag="in_rm_unpaired_reads", input_type=Boolean(optional=True), prefix="--rm-unpaired-reads", doc=InputDocumentation(doc=": discard unpaired reads")), ToolInput(tag="in_ignore_pe_reads", input_type=Boolean(optional=True), prefix="--ignore-pe-reads", doc=InputDocumentation(doc=": ignore paired-end reads even if present in the input")), ToolInput(tag="in_unpaired", input_type=Boolean(optional=True), prefix="--unpaired", doc=InputDocumentation(doc=": ignore read pairing (only for paired-end GBS; treat READ2's as if they were READ1's)")), ToolInput(tag="in_km_er_length", input_type=Boolean(optional=True), prefix="--kmer-length", doc=InputDocumentation(doc=": kmer length for the de Bruijn graph (default: 31, max. 31)")), ToolInput(tag="in_max_de_bruijn_reads", input_type=Boolean(optional=True), prefix="--max-debruijn-reads", doc=InputDocumentation(doc=": maximum number of reads to use in the de Bruijn graph (default: 1000)")), ToolInput(tag="in_min_km_er_cov", input_type=Boolean(optional=True), prefix="--min-kmer-cov", doc=InputDocumentation(doc=": minimum coverage to consider a kmer (default: 2)")), ToolInput(tag="in_write_alignments", input_type=Boolean(optional=True), prefix="--write-alignments", doc=InputDocumentation(doc=": save read alignments (heavy BAM files)")), ToolInput(tag="in_min_mapq", input_type=Boolean(optional=True), prefix="--min-mapq", doc=InputDocumentation(doc=": minimum PHRED-scaled mapping quality to consider a read (default: 10)")), ToolInput(tag="in_max_clipped", input_type=Boolean(optional=True), prefix="--max-clipped", doc=InputDocumentation(doc=": maximum soft-clipping level, in fraction of read length (default: 0.20)")), ToolInput(tag="in_max_insert_len", input_type=Boolean(optional=True), prefix="--max-insert-len", doc=InputDocumentation(doc=": maximum allowed sequencing insert length (default: 1000)")), ToolInput(tag="in_details", input_type=Boolean(optional=True), prefix="--details", doc=InputDocumentation(doc=": write a heavier output")), ToolInput(tag="in_phasing_co_occurrences_thr_range", input_type=Boolean(optional=True), prefix="--phasing-cooccurrences-thr-range", doc=InputDocumentation(doc=": range of edge coverage thresholds to\niterate over when building the graph of allele cooccurrences for\nSNP phasing (default: 1,2)")), ToolInput(tag="in_phasing_dont_prune_he_ts", input_type=Boolean(optional=True), prefix="--phasing-dont-prune-hets", doc=InputDocumentation(doc=": don't try to ignore dubious heterozygote\ngenotypes during phasing\n"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Gstacks_V0_1_0().translate("wdl", allow_empty_container=True)

