from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Array, String, File, Boolean, Int

Buildconsensus_Py_V0_1_0 = CommandToolBuilder(tool="BuildConsensus.py", base_command=["BuildConsensus.py"], inputs=[ToolInput(tag="in_list_fastafastq_files", input_type=Array(t=String(), optional=True), prefix="-s", doc=InputDocumentation(doc="A list of FASTA/FASTQ files containing sequences to\nprocess. (default: None)")), ToolInput(tag="in_explicit_output_file", input_type=Array(t=String(), optional=True), prefix="-o", doc=InputDocumentation(doc="Explicit output file name(s). Note, this argument\ncannot be used with the --failed, --outdir, or\n--outname arguments. If unspecified, then the output\nfilename will be based on the input filename(s).\n(default: None)")), ToolInput(tag="in_outdir", input_type=File(optional=True), prefix="--outdir", doc=InputDocumentation(doc="Specify to changes the output directory to the\nlocation specified. The input file directory is used\nif this is not specified. (default: None)")), ToolInput(tag="in_out_name", input_type=File(optional=True), prefix="--outname", doc=InputDocumentation(doc="Changes the prefix of the successfully processed\noutput file to the string specified. May not be\nspecified with multiple input files. (default: None)")), ToolInput(tag="in_log", input_type=File(optional=True), prefix="--log", doc=InputDocumentation(doc="Specify to write verbose logging to a file. May not be\nspecified with multiple input files. (default: None)")), ToolInput(tag="in_failed", input_type=Boolean(optional=True), prefix="--failed", doc=InputDocumentation(doc="If specified create files containing records that fail\nprocessing. (default: False)")), ToolInput(tag="in_fast_a", input_type=Boolean(optional=True), prefix="--fasta", doc=InputDocumentation(doc="Specify to force output as FASTA rather than FASTQ.\n(default: None)")), ToolInput(tag="in_delim", input_type=String(optional=True), prefix="--delim", doc=InputDocumentation(doc="DELIMITER DELIMITER\nA list of the three delimiters that separate\nannotation blocks, field names and values, and values\nwithin a field, respectively. (default: ('|', '=',\n','))")), ToolInput(tag="in_nproc", input_type=Int(optional=True), prefix="--nproc", doc=InputDocumentation(doc="The number of simultaneous computational processes to\nexecute (CPU cores to utilized). (default: 8)")), ToolInput(tag="in_minimum_number_sequences", input_type=Int(optional=True), prefix="-n", doc=InputDocumentation(doc="The minimum number of sequences needed to define a\nvalid consensus. (default: 1)")), ToolInput(tag="in_bf", input_type=String(optional=True), prefix="--bf", doc=InputDocumentation(doc="Position of description barcode field to group\nsequences by. (default: BARCODE)")), ToolInput(tag="in_consensus_quality_score", input_type=Int(optional=True), prefix="-q", doc=InputDocumentation(doc="Consensus quality score cut-off under which an\nambiguous character is assigned; does not apply when\nquality scores are unavailable. (default: 0)")), ToolInput(tag="in_freq", input_type=Int(optional=True), prefix="--freq", doc=InputDocumentation(doc="Fraction of character occurrences under which an\nambiguous character is assigned. (default: 0.6)")), ToolInput(tag="in_max_gap", input_type=Int(optional=True), prefix="--maxgap", doc=InputDocumentation(doc="If specified, this defines a cut-off for the frequency\nof allowed gap values for each position. Positions\nexceeding the threshold are deleted from the\nconsensus. If not defined, positions are always\nretained. (default: None)")), ToolInput(tag="in_pf", input_type=String(optional=True), prefix="--pf", doc=InputDocumentation(doc="Specifies the field name of the primer annotations\n(default: None)")), ToolInput(tag="in_pr_cons", input_type=String(optional=True), prefix="--prcons", doc=InputDocumentation(doc="Specify to define a minimum primer frequency required\nto assign a consensus primer, and filter out sequences\nwith minority primers from the consensus building\nstep. (default: None)")), ToolInput(tag="in_cf", input_type=Array(t=String(), optional=True), prefix="--cf", doc=InputDocumentation(doc="Specifies a set of additional annotation fields to\ncopy into the consensus sequence annotations.\n(default: None)")), ToolInput(tag="in_act", input_type=String(optional=True), prefix="--act", doc=InputDocumentation(doc="[{min,max,sum,set,majority} ...]\nList of actions to take for each copy field which\ndefines how each annotation will be combined into a\nsingle value. The actions 'min', 'max', 'sum' perform\nthe corresponding mathematical operation on numeric\nannotations. The action 'set' combines annotations\ninto a comma delimited list of unique values and adds\nan annotation named <FIELD>_COUNT specifying the count\nof each item in the set. The action 'majority' assigns\nthe most frequent annotation to the consensus\nannotation and adds an annotation named <FIELD>_FREQ\nspecifying the frequency of the majority value.\n(default: None)")), ToolInput(tag="in_dep", input_type=Boolean(optional=True), prefix="--dep", doc=InputDocumentation(doc="Specify to calculate consensus quality with a non-\nindependence assumption (default: False)")), ToolInput(tag="in_max_div", input_type=Int(optional=True), prefix="--maxdiv", doc=InputDocumentation(doc="Specify to calculate the nucleotide diversity of each\nread group (average pairwise error rate) and remove\ngroups exceeding the given diversity threshold.\nDiversity is calculate for all positions within the\nread group, ignoring any character filtering imposed\nby the -q, --freq and --maxgap arguments. Mutually\nexclusive with --maxerror. (default: None)")), ToolInput(tag="in_max_error", input_type=Int(optional=True), prefix="--maxerror", doc=InputDocumentation(doc="Specify to calculate the error rate of each read group\n(rate of mismatches from consensus) and remove groups\nexceeding the given error threshold. The error rate is\ncalculated against the final consensus sequence, which\nmay include masked positions due to the -q and --freq\narguments and may have deleted positions due to the\n--maxgap argument. Mutually exclusive with --maxdiv.\n(default: None)")), ToolInput(tag="in_consensus_pass", input_type=String(), position=0, doc=InputDocumentation(doc="consensus reads.")), ToolInput(tag="in_consensus_fail", input_type=String(), position=1, doc=InputDocumentation(doc="raw reads failing consensus filtering criteria.")), ToolInput(tag="in_primer", input_type=String(), position=0, doc=InputDocumentation(doc="a comma delimited list of unique primer annotations found within the\nbarcode read group.")), ToolInput(tag="in_pr_count", input_type=String(), position=1, doc=InputDocumentation(doc="a comma delimited list of the corresponding counts of unique primer")), ToolInput(tag="in_annotations_dot", input_type=String(), position=2, doc=InputDocumentation(doc="PRCONS")), ToolInput(tag="in_pr_freq", input_type=String(), position=0, doc=InputDocumentation(doc="the frequency of the majority primer.")), ToolInput(tag="in_cons_count", input_type=String(), position=1, doc=InputDocumentation(doc="the count of reads within the barcode read group which contributed to\nthe consensus sequence. This is the total size of the read group,\nminus sequence excluded due to user defined filtering criteria.\n"))], outputs=[], container="quay.io/biocontainers/presto:0.6.2--py_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Buildconsensus_Py_V0_1_0().translate("wdl")

