from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import String, File, Int, Directory, Boolean

Rgi_V0_1_0 = CommandToolBuilder(tool="rgi", base_command=["rgi"], inputs=[ToolInput(tag="in_input_type", input_type=String(optional=True), prefix="--input_type", doc=InputDocumentation(doc="must be one of contig, orf, protein, read (default:\ncontig)")), ToolInput(tag="in_input_sequence", input_type=File(optional=True), prefix="--input_sequence", doc=InputDocumentation(doc="input file must be in either FASTA (contig and\nprotein), FASTQ(read) or gzip format! e.g\nmyFile.fasta, myFasta.fasta.gz")), ToolInput(tag="in_num_threads", input_type=Int(optional=True), prefix="--num_threads", doc=InputDocumentation(doc="Number of threads (CPUs) to use in the BLAST search\n(default=32)")), ToolInput(tag="in_output_file", input_type=File(optional=True), prefix="--output_file", doc=InputDocumentation(doc="Output JSON file (default=Report)")), ToolInput(tag="in_loose_criteria", input_type=String(optional=True), prefix="--loose_criteria", doc=InputDocumentation(doc="The options are YES to include loose hits and NO to\nexclude loose hits. (default=NO to exclude loose hits)")), ToolInput(tag="in_clean", input_type=Directory(optional=True), prefix="--clean", doc=InputDocumentation(doc="This removes temporary files in the results directory\nafter run. Options are NO or YES (default=YES for\nremove)")), ToolInput(tag="in_data", input_type=String(optional=True), prefix="--data", doc=InputDocumentation(doc="Specify a data-type, i.e. wgs, chromosome, plasmid,\netc. (default = NA)")), ToolInput(tag="in_verbose", input_type=File(optional=True), prefix="--verbose", doc=InputDocumentation(doc="log progress to file. Options are OFF or ON (default =\nOFF for no logging)")), ToolInput(tag="in_alignment_tool", input_type=String(optional=True), prefix="--alignment_tool", doc=InputDocumentation(doc="choose between BLAST and DIAMOND. Options are BLAST or\nDIAMOND (default = BLAST)")), ToolInput(tag="in_db", input_type=File(optional=True), prefix="--db", doc=InputDocumentation(doc="specify path to CARD blast databases (default: None)")), ToolInput(tag="in_software_version", input_type=Boolean(optional=True), prefix="--software_version", doc=InputDocumentation(doc="Prints software number")), ToolInput(tag="in_data_version", input_type=Boolean(optional=True), prefix="--data_version", doc=InputDocumentation(doc="Prints data version number"))], outputs=[ToolOutput(tag="out_output_file", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output_file", type_hint=File()), doc=OutputDocumentation(doc="Output JSON file (default=Report)"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Rgi_V0_1_0().translate("wdl", allow_empty_container=True)

