from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, Float

Gthgetseq_V0_1_0 = CommandToolBuilder(tool="gthgetseq", base_command=["gthgetseq"], inputs=[ToolInput(tag="in_get_cdna", input_type=Boolean(optional=True), prefix="-getcdna", doc=InputDocumentation(doc="get cDNA/EST sequences\ndefault: no")), ToolInput(tag="in_get_cdna_comp", input_type=Boolean(optional=True), prefix="-getcdnacomp", doc=InputDocumentation(doc="get complement of cDNA/EST sequences\ndefault: no")), ToolInput(tag="in_get_protein", input_type=Boolean(optional=True), prefix="-getprotein", doc=InputDocumentation(doc="get protein sequences\ndefault: no")), ToolInput(tag="in_get_protein_comp", input_type=Boolean(optional=True), prefix="-getproteincomp", doc=InputDocumentation(doc="get complement of protein sequences\ndefault: no")), ToolInput(tag="in_get_genomic", input_type=Boolean(optional=True), prefix="-getgenomic", doc=InputDocumentation(doc="get genomic sequences\ndefault: no")), ToolInput(tag="in_get_genomic_comp", input_type=Boolean(optional=True), prefix="-getgenomiccomp", doc=InputDocumentation(doc="get complement of genomic sequences\ndefault: no")), ToolInput(tag="in_min_alignment_score", input_type=Float(optional=True), prefix="-minalignmentscore", doc=InputDocumentation(doc="the minimum alignment score for spliced alignments to be\nincluded into the set of spliced alignments\ndefault: 0.00")), ToolInput(tag="in_max_alignment_score", input_type=Float(optional=True), prefix="-maxalignmentscore", doc=InputDocumentation(doc="the maximum alignment score for spliced alignments to be\nincluded into the set of spliced alignments\ndefault: 1.00")), ToolInput(tag="in_min_coverage", input_type=Boolean(optional=True), prefix="-mincoverage", doc=InputDocumentation(doc="set the minimum coverage for spliced alignments to be\nincluded into the set of spliced alignments\ndefault: 0.00")), ToolInput(tag="in_max_coverage", input_type=Boolean(optional=True), prefix="-maxcoverage", doc=InputDocumentation(doc="set the maximum coverage for spliced alignments to be\nincluded into the set of spliced alignments\ndefault: 9999.99")), ToolInput(tag="in_gzip", input_type=Boolean(optional=True), prefix="-gzip", doc=InputDocumentation(doc="gzip compressed input file(s)\ndefault: no")), ToolInput(tag="in_bzip_two", input_type=Boolean(optional=True), prefix="-bzip2", doc=InputDocumentation(doc="bzip2 compressed input file(s)\ndefault: no")), ToolInput(tag="in_version", input_type=Boolean(optional=True), prefix="-version", doc=InputDocumentation(doc="display version information and exit"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Gthgetseq_V0_1_0().translate("wdl", allow_empty_container=True)

