from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Array, String, File, Boolean

Creategermlines_Py_V0_1_0 = CommandToolBuilder(tool="CreateGermlines.py", base_command=["CreateGermlines.py"], inputs=[ToolInput(tag="in_list_tab_defaultnone", input_type=Array(t=String(), optional=True), prefix="-d", doc=InputDocumentation(doc="A list of tab delimited database files. (default:\nNone)")), ToolInput(tag="in_explicit_output_file", input_type=Array(t=String(), optional=True), prefix="-o", doc=InputDocumentation(doc="Explicit output file name. Note, this argument cannot\nbe used with the --failed, --outdir, or --outname\narguments. If unspecified, then the output filename\nwill be based on the input filename(s). (default:\nNone)")), ToolInput(tag="in_outdir", input_type=File(optional=True), prefix="--outdir", doc=InputDocumentation(doc="Specify to changes the output directory to the\nlocation specified. The input file directory is used\nif this is not specified. (default: None)")), ToolInput(tag="in_out_name", input_type=File(optional=True), prefix="--outname", doc=InputDocumentation(doc="Changes the prefix of the successfully processed\noutput file to the string specified. May not be\nspecified with multiple input files. (default: None)")), ToolInput(tag="in_log", input_type=File(optional=True), prefix="--log", doc=InputDocumentation(doc="Specify to write verbose logging to a file. May not be\nspecified with multiple input files. (default: None)")), ToolInput(tag="in_failed", input_type=Boolean(optional=True), prefix="--failed", doc=InputDocumentation(doc="If specified create files containing records that fail\nprocessing. (default: False)")), ToolInput(tag="in_format", input_type=String(optional=True), prefix="--format", doc=InputDocumentation(doc="Specify input and output format. (default: airr)")), ToolInput(tag="in_list_folders_andor", input_type=Array(t=String(), optional=True), prefix="-r", doc=InputDocumentation(doc="List of folders and/or fasta files (with .fasta, .fna\nor .fa extension) with germline sequences. When using\nthe default Change-O sequence and coordinate fields,\nthese reference sequences must contain IMGT-numbering\nspacers (gaps) in the V segment. Alternative numbering\nschemes, or no numbering, may work for alternative\nsequence and coordinate definitions that define a\nvalid alignment, but a warning will be issued.\n(default: None)")), ToolInput(tag="in_specify_types_include", input_type=String(optional=True), prefix="-g", doc=InputDocumentation(doc="[{full,dmask,vonly,regions} ...]\nSpecify type(s) of germlines to include full germline,\ngermline with D segment masked, or germline for V\nsegment only. (default: ['dmask'])")), ToolInput(tag="in_cloned", input_type=Boolean(optional=True), prefix="--cloned", doc=InputDocumentation(doc="Specify to create only one germline per clone. Note,\nif allele calls are ambiguous within a clonal group,\nthis will place the germline call used for the entire\nclone within the germline_v_call, germline_d_call and\ngermline_j_call fields. (default: False)")), ToolInput(tag="in_sf", input_type=String(optional=True), prefix="--sf", doc=InputDocumentation(doc="Field containing the aligned sequence. Defaults to\nsequence_alignment (airr) or SEQUENCE_IMGT (changeo).\n(default: None)")), ToolInput(tag="in_vf", input_type=String(optional=True), prefix="--vf", doc=InputDocumentation(doc="Field containing the germline V segment call. Defaults\nto v_call (airr) or V_CALL (changeo). (default: None)")), ToolInput(tag="in_df", input_type=String(optional=True), prefix="--df", doc=InputDocumentation(doc="Field containing the germline D segment call. Defaults\nto d_call (airr) or D_CALL (changeo). (default: None)")), ToolInput(tag="in_jf", input_type=String(optional=True), prefix="--jf", doc=InputDocumentation(doc="Field containing the germline J segment call. Defaults\nto j_call (airr) or J_CALL (changeo). (default: None)")), ToolInput(tag="in_cf", input_type=String(optional=True), prefix="--cf", doc=InputDocumentation(doc="Field containing clone identifiers. Ignored if")), ToolInput(tag="in_germ_pass", input_type=String(), position=0, doc=InputDocumentation(doc="database with assigned germline sequences.")), ToolInput(tag="in_germ_fail", input_type=String(), position=1, doc=InputDocumentation(doc="database with records failing germline assignment.")), ToolInput(tag="in_clone_id", input_type=String(), position=0, doc=InputDocumentation(doc="output fields:"))], outputs=[], container="quay.io/biocontainers/changeo:1.0.2--py_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Creategermlines_Py_V0_1_0().translate("wdl")

