from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, Directory, String

Mutmap_V0_1_0 = CommandToolBuilder(tool="mutmap", base_command=["mutmap"], inputs=[ToolInput(tag="in_ref", input_type=Boolean(optional=True), prefix="--ref", doc=InputDocumentation(doc="Reference fasta.")), ToolInput(tag="in_cultivar", input_type=Boolean(optional=True), prefix="--cultivar", doc=InputDocumentation(doc="fastq or bam of cultivar. If you specify\nfastq, please separate pairs by comma,\ne.g. -c fastq1,fastq2. You can use this\noptiion multiple times")), ToolInput(tag="in_bulk", input_type=Boolean(optional=True), prefix="--bulk", doc=InputDocumentation(doc="fastq or bam of mutnat bulk. If you specify\nfastq, please separate pairs by comma,\ne.g. -b fastq1,fastq2. You can use this\noptiion multiple times")), ToolInput(tag="in_n_bulk", input_type=Boolean(optional=True), prefix="--N-bulk", doc=InputDocumentation(doc="Number of individuals in mutant bulk.")), ToolInput(tag="in_out", input_type=Directory(optional=True), prefix="--out", doc=InputDocumentation(doc="Output directory. Specified name must not")), ToolInput(tag="in_window", input_type=Boolean(optional=True), prefix="--window", doc=InputDocumentation(doc="Window size (kb). [2000]")), ToolInput(tag="in_step", input_type=Boolean(optional=True), prefix="--step", doc=InputDocumentation(doc="Step size (kb). [100]")), ToolInput(tag="in_max_depth", input_type=Boolean(optional=True), prefix="--max-depth", doc=InputDocumentation(doc="Maximum depth of variants which will be used.\nThis cutoff will be applied in both of cultivar\nand bulk. [250]")), ToolInput(tag="in_min_depth", input_type=Boolean(optional=True), prefix="--min-depth", doc=InputDocumentation(doc="Minimum depth of variants which will be used.\nThis cutoff will be applied in both of cultivar\nand bulk. [8]")), ToolInput(tag="in_n_rep", input_type=Boolean(optional=True), prefix="--N-rep", doc=InputDocumentation(doc="Number of replicates for simulation to make\nnull distribution. [5000]")), ToolInput(tag="in_trim", input_type=Boolean(optional=True), prefix="--trim", doc=InputDocumentation(doc="Trim fastq using trimmomatic.")), ToolInput(tag="in_adapter", input_type=Boolean(optional=True), prefix="--adapter", doc=InputDocumentation(doc="FASTA of adapter sequences. This will be used\nwhen you specify '-T' for trimming.")), ToolInput(tag="in_trim_params", input_type=Boolean(optional=True), prefix="--trim-params", doc=InputDocumentation(doc="Parameters for trimmomatic. Input parameters\nmust be separated by comma with following\norder: phred, ILLUMINACLIP, LEADING, TRAILING,\nSLIDINGWINDOW, MINLEN. If you want to remove\nadapters of illumina, please specify FASTA of\nthe adapter sequences with '--adapter'. Specified\nname will be inserted into <ADAPTER_FASTA>. If you\ndon't specify it, adapter trimming will be skipped.\n[33,<ADAPTER_FASTA>:2:30:10,20,20,4:15,75]")), ToolInput(tag="in_snp_eff", input_type=Boolean(optional=True), prefix="--snpEff", doc=InputDocumentation(doc="Predict causal variant using SnpEff. Please\ncheck available databases in SnpEff.")), ToolInput(tag="in_mem", input_type=Boolean(optional=True), prefix="--mem", doc=InputDocumentation(doc="Maximum memory per thread when bam sorted;\nsuffix K/M/G recognized. [1G]")), ToolInput(tag="in_min_mq", input_type=Boolean(optional=True), prefix="--min-MQ", doc=InputDocumentation(doc="Minimum mapping quality in mpileup. [40]")), ToolInput(tag="in_min_bq", input_type=Boolean(optional=True), prefix="--min-BQ", doc=InputDocumentation(doc="Minimum base quality in mpileup. [18]")), ToolInput(tag="in_adjust_mq", input_type=Boolean(optional=True), prefix="--adjust-MQ", doc=InputDocumentation(doc="'adjust-MQ' in mpileup. Default parameter\nis suited for BWA. [50]")), ToolInput(tag="in_species", input_type=Boolean(optional=True), prefix="--species", doc=InputDocumentation(doc="Consider multiple test correction derived by\nHuang et al. (2019). Please spesify a species name.\nWith this option. QTL-seq produces a theoretical threshold.\nCurrently, Arabidopsis, Cucumber, Maize, Rapeseed,\nRice, Tobacco, Tomato, Wheat, and Yeast are supported.")), ToolInput(tag="in_exist_dot", input_type=String(), position=0, doc=InputDocumentation(doc="-t , --threads     Number of threads. If you specify the number"))], outputs=[ToolOutput(tag="out_out", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_out", type_hint=File()), doc=OutputDocumentation(doc="Output directory. Specified name must not"))], container="quay.io/biocontainers/mutmap:2.3.0--py_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Mutmap_V0_1_0().translate("wdl")

