from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import String, Directory, File, Boolean

Findcsrnatss_Pl_Peak_File___Tss_V0_1_0 = CommandToolBuilder(tool="findcsRNATSS.pl_peak file | tss", base_command=["findcsRNATSS.pl", "peak file | tss"], inputs=[ToolInput(tag="in_o", input_type=String(optional=True), prefix="-o", doc=InputDocumentation(doc="(*see note below)")), ToolInput(tag="in_rna", input_type=Directory(optional=True), prefix="-rna", doc=InputDocumentation(doc="(*see note below)")), ToolInput(tag="in_genome", input_type=File(optional=True), prefix="-genome", doc=InputDocumentation(doc="(HOMER genome identifier or FASTA file)")), ToolInput(tag="in_gtf", input_type=File(optional=True), prefix="-gtf", doc=InputDocumentation(doc="(gene annotation, by default will use genome annotation if available)")), ToolInput(tag="in_n_tag_threshold", input_type=Boolean(optional=True), prefix="-ntagThreshold", doc=InputDocumentation(doc="<#> (minimum number of norm. tags per TSS cluster, default: 7)")), ToolInput(tag="in_size", input_type=Boolean(optional=True), prefix="-size", doc=InputDocumentation(doc="<#> (size of regions to check for significant TSS clusters, default: 150)")), ToolInput(tag="in_include_single_exons", input_type=Boolean(optional=True), prefix="-includeSingleExons", doc=InputDocumentation(doc="(include single exon genes for TP/FP calculations, default: only use multiexon genes)")), ToolInput(tag="in_meta_gene", input_type=Boolean(optional=True), prefix="-metagene", doc=InputDocumentation(doc="(Include a metagene plot of read distributions)")), ToolInput(tag="in_cpu", input_type=Boolean(optional=True), prefix="-cpu", doc=InputDocumentation(doc="<#> (max CPUs to use)")), ToolInput(tag="in_max_input_log_twofold", input_type=Boolean(optional=True), prefix="-maxInputLog2Fold", doc=InputDocumentation(doc="<#> (maximum log2 fold enrichment vs. input or RNA, default: 30000000000)")), ToolInput(tag="in_max_rna_log_twofold", input_type=Boolean(optional=True), prefix="-maxRNALog2Fold", doc=InputDocumentation(doc="<#> (maximum log2 fold enrichment vs. input or RNA, default: 30000000000)")), ToolInput(tag="in_mind_ist_diff", input_type=Boolean(optional=True), prefix="-minDistDiff", doc=InputDocumentation(doc="<#> (minimum CDF difference to consider for TP/FP calculation, default: 0.15)")), ToolInput(tag="in_default_log_twofold", input_type=Boolean(optional=True), prefix="-defaultLog2Fold", doc=InputDocumentation(doc="<#> (if annotation-based TP/FP threshold calculation failes, default: 1)")), ToolInput(tag="in_pseudo_count", input_type=Boolean(optional=True), prefix="-pseudoCount", doc=InputDocumentation(doc="<#> (number of normalizaed reads added to input/target read counts, default: 1.0)")), ToolInput(tag="in_local_fold_enrichment", input_type=Boolean(optional=True), prefix="-L", doc=InputDocumentation(doc="<#> (local fold enrichment used for initial TSS identification, default: 2)")), ToolInput(tag="in_no_filter_rna", input_type=Boolean(optional=True), prefix="-noFilterRNA", doc=InputDocumentation(doc="(only use RNAseq for stability calculations, not for filtering putative TSS)")), ToolInput(tag="in_sense_region", input_type=Boolean(optional=True), prefix="-senseRegion", doc=InputDocumentation(doc="<#>,<#> (Size of region to calculate RNA/input read density,default: -100,500)")), ToolInput(tag="in_antisense_region", input_type=Boolean(optional=True), prefix="-antisenseRegion", doc=InputDocumentation(doc="<#>,<#> (Size of region to calculate RNA/input read density,default: -500,100)")), ToolInput(tag="in_rna_region", input_type=Boolean(optional=True), prefix="-rnaRegion", doc=InputDocumentation(doc="<#>,<#> (Size of region to calculate RNA coverage for FP identification,default: -150,-10)")), ToolInput(tag="in_stable_threshold", input_type=Boolean(optional=True), prefix="-stableThreshold", doc=InputDocumentation(doc="<#> (minimum RNAseq read density for stable transcript, def: 2)")), ToolInput(tag="in_bi_threshold", input_type=Boolean(optional=True), prefix="-biThreshold", doc=InputDocumentation(doc="<#> (minimum divergent csRNAseq signal for bi/divergent TSS, def: 2)")), ToolInput(tag="in_dist", input_type=Boolean(optional=True), prefix="-dist", doc=InputDocumentation(doc="<#> (minimum distance from annotated TSS to assign 'distal' annotation, def: 500)"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Findcsrnatss_Pl_Peak_File___Tss_V0_1_0().translate("wdl", allow_empty_container=True)

