from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, Int, File

Runhicpca_Pl_V0_1_0 = CommandToolBuilder(tool="runHiCpca.pl", base_command=["runHiCpca.pl"], inputs=[ToolInput(tag="in_res", input_type=Boolean(optional=True), prefix="-res", doc=InputDocumentation(doc="<#> (resolution in bp, default: 50000)")), ToolInput(tag="in_window", input_type=Boolean(optional=True), prefix="-window", doc=InputDocumentation(doc="<#> (overlapping window resolution in bp, i.e. superRes, default: 100000)")), ToolInput(tag="in_active", input_type=Boolean(optional=True), prefix="-active", doc=InputDocumentation(doc="<K4me+ regions> (Regions to use to help decide sign on principal component [active=+])")), ToolInput(tag="in_inactive", input_type=Int(optional=True), prefix="-inactive", doc=InputDocumentation(doc="(Regions to use to help decide sign on principal component [inactive=-])")), ToolInput(tag="in_genome", input_type=File(optional=True), prefix="-genome", doc=InputDocumentation(doc="(If you don't have seed regions, this will use the TSS file as active seeds)")), ToolInput(tag="in_corr_depth", input_type=Boolean(optional=True), prefix="-corrDepth", doc=InputDocumentation(doc="<#> (number of expected reads needed per data point when calculating correlation, default: 3)")), ToolInput(tag="in_std", input_type=Boolean(optional=True), prefix="-std", doc=InputDocumentation(doc="<#> (exclude regions with sequencing depth exceeding # std deviations, default: 8)")), ToolInput(tag="in_min", input_type=Boolean(optional=True), prefix="-min", doc=InputDocumentation(doc="<#> (exclude regions with sequencing depth less than this fraction of mean, default: 0.05)")), ToolInput(tag="in_rpath", input_type=File(optional=True), prefix="-rpath", doc=InputDocumentation(doc="(If R is not accessible via the $PATH variable)")), ToolInput(tag="in_pc", input_type=Boolean(optional=True), prefix="-pc", doc=InputDocumentation(doc="<#> (number of principal components to report, default: 1)")), ToolInput(tag="in_cpu", input_type=Boolean(optional=True), prefix="-cpu", doc=InputDocumentation(doc="<#> (number of CPUs to use, default: 1)")), ToolInput(tag="in_custom_regions", input_type=File(optional=True), prefix="-customRegions", doc=InputDocumentation(doc="(instead of analysis by chr, separate by these regions, i.e. arms)")), ToolInput(tag="in_cluster", input_type=Boolean(optional=True), prefix="-cluster", doc=InputDocumentation(doc="<#> (cluster correlation matrix, define clusters using correlation threshold: i.e. 0.25)")), ToolInput(tag="in_min_p", input_type=Boolean(optional=True), prefix="-minp", doc=InputDocumentation(doc="<#> (minimum cluster size, as fraction of chromosome, default: 0.02)"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Runhicpca_Pl_V0_1_0().translate("wdl", allow_empty_container=True)

