from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, Int, File, Float, String

Prosolo_Single_Cell_Bulk_V0_1_0 = CommandToolBuilder(tool="prosolo_single_cell_bulk", base_command=["prosolo", "single-cell-bulk"], inputs=[ToolInput(tag="in_exclusive_end", input_type=Boolean(optional=True), prefix="--exclusive-end", doc=InputDocumentation(doc="Assume that the END tag is exclusive (i.e. it points to the position after the\nvariant). This is needed, e.g., for DELLY.")), ToolInput(tag="in_omit_fragment_evidence", input_type=Boolean(optional=True), prefix="--omit-fragment-evidence", doc=InputDocumentation(doc="Omit evidence consisting of read pairs with unexpected insert size (insert size\nparameters will be ignored).")), ToolInput(tag="in_omit_indels", input_type=Boolean(optional=True), prefix="--omit-indels", doc=InputDocumentation(doc="Don't call indels.")), ToolInput(tag="in_omit_sn_vs", input_type=Boolean(optional=True), prefix="--omit-snvs", doc=InputDocumentation(doc="Don't call SNVs.")), ToolInput(tag="in_bulk_max_n", input_type=Int(optional=True), prefix="--bulk-max-n", doc=InputDocumentation(doc="Maximum number of (theoretical) reads to work with in the bulk background, in\ncase the actual read count for a variant is higher (all read information will\nbe used, but probabilities will only be computed for all discrete allele\nfrequencies allowed by the maximum read count provided here). The code will\nwork with any number above bulk-min-n, but we use the cap of the currently\nused Lodato amplification bias model for the single cell sample as the\ndefault. [default: 100]")), ToolInput(tag="in_bulk_min_n", input_type=Int(optional=True), prefix="--bulk-min-n", doc=InputDocumentation(doc="Minimum number of (theoretical) reads to work with in the bulk background, in\ncase the actual read count for a variant site is lower (in this case,\nprobabilities will be computed for all discrete allele frequencies allowed by\nthe minimum read count provided here). The code will work with a minimum of 2,\nbut for a more even sampling of Event spaces, the default is at 8. [default:\n8]")), ToolInput(tag="in_candidates", input_type=File(optional=True), prefix="--candidates", doc=InputDocumentation(doc="VCF/BCF file to process (if omitted, read from STDIN).")), ToolInput(tag="in_in_del_window", input_type=Int(optional=True), prefix="--indel-window", doc=InputDocumentation(doc="Number of bases to consider left and right of indel breakpoint when\ncalculating read support. This number should not be too large in order to\navoid biases caused by other close variants. [default: 100]")), ToolInput(tag="in_max_in_del_len", input_type=Int(optional=True), prefix="--max-indel-len", doc=InputDocumentation(doc="Omit longer indels when calling [1000].")), ToolInput(tag="in_obs", input_type=File(optional=True), prefix="--obs", doc=InputDocumentation(doc="Optional path where read observations shall be written to. The resulting file\ncontains a line for each observation with tab-separated values.")), ToolInput(tag="in_output", input_type=File(optional=True), prefix="--output", doc=InputDocumentation(doc="BCF file that shall contain the results (if omitted, write to STDOUT).")), ToolInput(tag="in_pile_up_window", input_type=Int(optional=True), prefix="--pileup-window", doc=InputDocumentation(doc="Window to investigate for evidence left and right of each variant. [default:\n2500]")), ToolInput(tag="in_ploidy", input_type=Int(optional=True), prefix="--ploidy", doc=InputDocumentation(doc="General ploidy of sampled individual. [default: 2]")), ToolInput(tag="in_spurious_del_ext_rate", input_type=Float(optional=True), prefix="--spurious-delext-rate", doc=InputDocumentation(doc="Extension rate of spurious insertions by the sequencer (Illumina: 0.0, see\nSchirmer et al. BMC Bioinformatics 2016). [default: 0.0]")), ToolInput(tag="in_spurious_in_sext_rate", input_type=Float(optional=True), prefix="--spurious-insext-rate", doc=InputDocumentation(doc="Extension rate of spurious insertions by the sequencer (Illumina: 0.0, see\nSchirmer et al. BMC Bioinformatics 2016) [0.0]. [default: 0.0]")), ToolInput(tag="in_spurious_del_rate", input_type=Float(optional=True), prefix="--spurious-del-rate", doc=InputDocumentation(doc="Rate of spuriosly deleted bases by the sequencer (Illumina: 5.1e-6, see\nSchirmer et al. BMC Bioinformatics 2016). [default: 5.1e-6]")), ToolInput(tag="in_spurious_ins_rate", input_type=Float(optional=True), prefix="--spurious-ins-rate", doc=InputDocumentation(doc="Rate of spuriously inserted bases by the sequencer (Illumina: 2.8e-6, see\nSchirmer et al. BMC Bioinformatics 2016). [default: 2.8e-6]")), ToolInput(tag="in_flags", input_type=String(optional=True), position=0, doc=InputDocumentation(doc="")), ToolInput(tag="in_single_cell", input_type=String(), position=1, doc=InputDocumentation(doc="")), ToolInput(tag="in_bulk", input_type=String(), position=2, doc=InputDocumentation(doc=""))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Prosolo_Single_Cell_Bulk_V0_1_0().translate("wdl", allow_empty_container=True)

