from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Int, Array, String, Directory

Snaptools_Align_Single_End_V0_1_0 = CommandToolBuilder(tool="snaptools_align_single_end", base_command=["snaptools", "align-single-end"], inputs=[ToolInput(tag="in_input_reference", input_type=File(optional=True), prefix="--input-reference", doc=InputDocumentation(doc="reference genome file contains the reference genome\nthat reads are mapped against, the genome index must\nbe under the same folder (default: None)")), ToolInput(tag="in_input_fast_q_one", input_type=Int(optional=True), prefix="--input-fastq1", doc=InputDocumentation(doc="fastq file contains R1 reads, currently supports\nfastq, gz, bz2 file (default: None)")), ToolInput(tag="in_output_bam", input_type=File(optional=True), prefix="--output-bam", doc=InputDocumentation(doc="output bam file contains unfiltered alignments\n(default: None)")), ToolInput(tag="in_aligner", input_type=Int(optional=True), prefix="--aligner", doc=InputDocumentation(doc="aligner to use. Currently, snaptools supports bwa,\nbowtie, bowtie2 and minimap2. (default: bwa)")), ToolInput(tag="in_path_to_aligner", input_type=File(optional=True), prefix="--path-to-aligner", doc=InputDocumentation(doc="path to fold that contains bwa (default: None)")), ToolInput(tag="in_aligner_options", input_type=Array(t=String(), optional=True), prefix="--aligner-options", doc=InputDocumentation(doc="list of strings indicating options you would like\npassed to alignerstrongly do not recommand to change\nunless you know what you are doing. (default: None)")), ToolInput(tag="in_read_fast_q_command", input_type=File(optional=True), prefix="--read-fastq-command", doc=InputDocumentation(doc="command line to execute for each of the input file.\nThis commandshould generate FASTA or FASTQ text and\nsend it to stdout (default: None)")), ToolInput(tag="in_num_threads", input_type=Int(optional=True), prefix="--num-threads", doc=InputDocumentation(doc="number of alignment threads (default: 1)")), ToolInput(tag="in_min_cov", input_type=Int(optional=True), prefix="--min-cov", doc=InputDocumentation(doc="min number of fragments per barcode. barcodes of total\nfragments less than --min-cov will be filreted before\nalingment. Note: though this feature is included, we\nfound it barely speed up the process, so recommand to\nset it to be 0. (default: 0)")), ToolInput(tag="in_if_sort", input_type=File(optional=True), prefix="--if-sort", doc=InputDocumentation(doc="weather to sort the bam file based on the read name\n(default: True)")), ToolInput(tag="in_tmp_folder", input_type=Directory(optional=True), prefix="--tmp-folder", doc=InputDocumentation(doc="directory to store temporary files. If not given,\nsnaptools will automaticallygenerate a temporary\nlocation to store temporary files (default: None)")), ToolInput(tag="in_overwrite", input_type=File(optional=True), prefix="--overwrite", doc=InputDocumentation(doc="whether to overwrite the output file if it already\nexists (default: False)\n"))], outputs=[ToolOutput(tag="out_output_bam", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output_bam", type_hint=File()), doc=OutputDocumentation(doc="output bam file contains unfiltered alignments\n(default: None)")), ToolOutput(tag="out_overwrite", output_type=File(optional=True), selector=InputSelector(input_to_select="in_overwrite", type_hint=File()), doc=OutputDocumentation(doc="whether to overwrite the output file if it already\nexists (default: False)\n"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Snaptools_Align_Single_End_V0_1_0().translate("wdl", allow_empty_container=True)

