from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import String, Int, Float

Lr2Gene_Pl_V0_1_0 = CommandToolBuilder(tool="lr2gene.pl", base_command=["lr2gene.pl"], inputs=[ToolInput(tag="in_control_sample_used", input_type=String(optional=True), prefix="-c", doc=InputDocumentation(doc="that control sample is used for normalization")), ToolInput(tag="in_mode", input_type=String(optional=True), prefix="-y", doc=InputDocumentation(doc="mode")), ToolInput(tag="in_minimum_consecutive_amplicons", input_type=Int(optional=True), prefix="-s", doc=InputDocumentation(doc="The minimum consecutive amplicons to look for deletions and amplifications.  Default: 1.  Use with caution\nwhen it is less than 3.  There might be more false positives.  Though it has been successfully applied with\noption '-s 1' and identified one-exon deletion of PTEN and TP53 that were confirmed by RNA-seq.")), ToolInput(tag="in_minimum_amplified_default", input_type=Float(optional=True), prefix="-A", doc=InputDocumentation(doc="Minimum log2 ratio for a whole gene to be considered amplified.  Default: 1.50")), ToolInput(tag="in_minimum_ratio_deleted", input_type=Float(optional=True), prefix="-D", doc=InputDocumentation(doc="Minimum log2 ratio for a whole gene to be considered deleted.  Default: -2.00")), ToolInput(tag="in_minimum_mean_log", input_type=Float(optional=True), prefix="-E", doc=InputDocumentation(doc="Minimum mean log2 ratio difference for <3 exon deletion/amplification to be called.  Default: 1.25")), ToolInput(tag="in_when_considering_deletions", input_type=Float(optional=True), prefix="-M", doc=InputDocumentation(doc="When considering partial deletions less than 3 exons/amplicons, the minimum MAD value, in addition to -E,\nbefore considering it to be amplified or deleted.  Default: 10")), ToolInput(tag="in_when_considering_deletionamplification", input_type=Float(optional=True), prefix="-d", doc=InputDocumentation(doc="When considering >=3 exons deletion/amplification within a gene, the minimum differences between the log2 of two segments.\nDefault: 0.5")), ToolInput(tag="in_pvalue_when_are", input_type=Float(optional=True), prefix="-p", doc=InputDocumentation(doc="(0-1)\nThe p-value for t-test when consecutive exons/amplicons are >= 3.  Default: 0.00001")), ToolInput(tag="in_when_minimum_number", input_type=Float(optional=True), prefix="-e", doc=InputDocumentation(doc="When considering breakpoint in the middle of a gene, the minimum number of exons.  Default: 5")), ToolInput(tag="in_when_minimum_differences", input_type=Float(optional=True), prefix="-t", doc=InputDocumentation(doc="When considering breakpoint in the middle of a gene, the minimum differences between the log2 of two segments.\nDefault: 0.4")), ToolInput(tag="in_pvalue_when_is", input_type=Float(optional=True), prefix="-P", doc=InputDocumentation(doc="(0-1)\nThe p-value for t-test when the breakpoint is in the middle with min exons/amplicons >= [-e].  Default: 0.000001")), ToolInput(tag="in_breakpoint_detected_fraction", input_type=Float(optional=True), prefix="-R", doc=InputDocumentation(doc="(0-1)\nIf a breakpoint has been detected more than 'float' fraction of samples, it is considered false positive and removed.\nDefault: 0.03, or 3%.  Use in combination with -N")), ToolInput(tag="in_breakpoint_detected_samples", input_type=Int(optional=True), prefix="-N", doc=InputDocumentation(doc="If a breakpoint has been detected more than 'int' samples, it is considered false positives and removed.\nDefault: 5.  Use in combination with -R."))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Lr2Gene_Pl_V0_1_0().translate("wdl", allow_empty_container=True)

