from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import String, File, Int, Boolean

Fastani_V0_1_0 = CommandToolBuilder(tool="fastANI", base_command=["fastANI"], inputs=[ToolInput(tag="in_ref", input_type=String(optional=True), prefix="--ref", doc=InputDocumentation(doc="reference genome (fasta/fastq)[.gz]")), ToolInput(tag="in_ref_list", input_type=File(optional=True), prefix="--refList", doc=InputDocumentation(doc="a file containing list of reference genome files, one genome per line")), ToolInput(tag="in_query", input_type=String(optional=True), prefix="--query", doc=InputDocumentation(doc="query genome (fasta/fastq)[.gz]")), ToolInput(tag="in_query_list", input_type=File(optional=True), prefix="--queryList", doc=InputDocumentation(doc="a file containing list of query genome files, one genome per line")), ToolInput(tag="in_km_er", input_type=Int(optional=True), prefix="--kmer", doc=InputDocumentation(doc="kmer size <= 16 [default : 16]")), ToolInput(tag="in_threads", input_type=Int(optional=True), prefix="--threads", doc=InputDocumentation(doc="thread count for parallel execution [default : 1]")), ToolInput(tag="in_fra_glen", input_type=Int(optional=True), prefix="--fragLen", doc=InputDocumentation(doc="fragment length [default : 3,000]")), ToolInput(tag="in_min_fraction", input_type=Int(optional=True), prefix="--minFraction", doc=InputDocumentation(doc="minimum fraction of genome that must be shared for trusting ANI. If\nreference and query genome size differ, smaller one among the two is\nconsidered. [default : 0.2]")), ToolInput(tag="in_visualize", input_type=Boolean(optional=True), prefix="--visualize", doc=InputDocumentation(doc="output mappings for visualization, can be enabled for single genome to\nsingle genome comparison only [disabled by default]")), ToolInput(tag="in_matrix", input_type=File(optional=True), prefix="--matrix", doc=InputDocumentation(doc="also output ANI values as lower triangular matrix (format inspired from\nphylip). If enabled, you should expect an output file with .matrix\nextension [disabled by default]")), ToolInput(tag="in_output", input_type=File(optional=True), prefix="--output", doc=InputDocumentation(doc="[required]\noutput file name"))], outputs=[ToolOutput(tag="out_matrix", output_type=File(optional=True), selector=InputSelector(input_to_select="in_matrix", type_hint=File()), doc=OutputDocumentation(doc="also output ANI values as lower triangular matrix (format inspired from\nphylip). If enabled, you should expect an output file with .matrix\nextension [disabled by default]")), ToolOutput(tag="out_output", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output", type_hint=File()), doc=OutputDocumentation(doc="[required]\noutput file name"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Fastani_V0_1_0().translate("wdl", allow_empty_container=True)

