from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Array, String, File, Boolean, Int

Buildtrees_Py_V0_1_0 = CommandToolBuilder(tool="BuildTrees.py", base_command=["BuildTrees.py"], inputs=[ToolInput(tag="in_list_tab_defaultnone", input_type=Array(t=String(), optional=True), prefix="-d", doc=InputDocumentation(doc="A list of tab delimited database files. (default:\nNone)")), ToolInput(tag="in_outdir", input_type=File(optional=True), prefix="--outdir", doc=InputDocumentation(doc="Specify to changes the output directory to the\nlocation specified. The input file directory is used\nif this is not specified. (default: None)")), ToolInput(tag="in_out_name", input_type=File(optional=True), prefix="--outname", doc=InputDocumentation(doc="Changes the prefix of the successfully processed\noutput file to the string specified. May not be\nspecified with multiple input files. (default: None)")), ToolInput(tag="in_log", input_type=File(optional=True), prefix="--log", doc=InputDocumentation(doc="Specify to write verbose logging to a file. May not be\nspecified with multiple input files. (default: None)")), ToolInput(tag="in_failed", input_type=Boolean(optional=True), prefix="--failed", doc=InputDocumentation(doc="If specified create files containing records that fail\nprocessing. (default: False)")), ToolInput(tag="in_format", input_type=String(optional=True), prefix="--format", doc=InputDocumentation(doc="Specify input and output format. (default: airr)")), ToolInput(tag="in_collapse", input_type=Boolean(optional=True), prefix="--collapse", doc=InputDocumentation(doc="If specified, collapse identical sequences before\nexporting to fasta. (default: False)")), ToolInput(tag="in_nc_dr_three", input_type=Boolean(optional=True), prefix="--ncdr3", doc=InputDocumentation(doc="If specified, remove CDR3 from all sequences.\n(default: False)")), ToolInput(tag="in_n_mask", input_type=Boolean(optional=True), prefix="--nmask", doc=InputDocumentation(doc="If specified, do not attempt to mask split codons.\n(default: False)")), ToolInput(tag="in_md", input_type=File(optional=True), prefix="--md", doc=InputDocumentation(doc="List of fields to containing metadata to include in\noutput fasta file sequence headers. (default: None)")), ToolInput(tag="in_clones", input_type=Array(t=String(), optional=True), prefix="--clones", doc=InputDocumentation(doc="List of clone IDs to output, if specified. (default:\nNone)")), ToolInput(tag="in_min_seq", input_type=Int(optional=True), prefix="--minseq", doc=InputDocumentation(doc="Minimum number of data sequences. Any clones with\nfewer than the specified number of sequences will be\nexcluded. (default: 1)")), ToolInput(tag="in_sample", input_type=Int(optional=True), prefix="--sample", doc=InputDocumentation(doc="Depth of reads to be subsampled (before\ndeduplication). (default: -1)")), ToolInput(tag="in_append", input_type=Array(t=String(), optional=True), prefix="--append", doc=InputDocumentation(doc="List of columns to append to sequence ID to ensure\nuniqueness. (default: None)")), ToolInput(tag="in_ig_phy_ml", input_type=Boolean(optional=True), prefix="--igphyml", doc=InputDocumentation(doc="Run IgPhyML on output? (default: False)")), ToolInput(tag="in_nproc", input_type=Int(optional=True), prefix="--nproc", doc=InputDocumentation(doc="Number of threads to parallelize IgPhyML across.\n(default: 1)")), ToolInput(tag="in_clean", input_type=String(optional=True), prefix="--clean", doc=InputDocumentation(doc="Delete intermediate files? none: leave all\nintermediate files; all: delete all intermediate\nfiles. (default: none)")), ToolInput(tag="in_optimize", input_type=String(optional=True), prefix="--optimize", doc=InputDocumentation(doc="Optimize combination of topology (t) branch lengths\n(l) and parameters (r), or nothing (n), for IgPhyML.\n(default: lr)")), ToolInput(tag="in_omega", input_type=String(optional=True), prefix="--omega", doc=InputDocumentation(doc="Omega parameters to estimate for FWR,CDR respectively:\ne = estimate, ce = estimate + confidence interval\n(default: e,e)")), ToolInput(tag="in_kappa_parameters_estimate", input_type=String(optional=True), prefix="-t", doc=InputDocumentation(doc="Kappa parameters to estimate: e = estimate, ce =\nestimate + confidence interval (default: e)")), ToolInput(tag="in_motifs", input_type=Int(optional=True), prefix="--motifs", doc=InputDocumentation(doc="Which motifs to estimate mutability. (default:\nWRC_2:0,GYW_0:1,WA_1:2,TW_0:3,SYC_2:4,GRS_0:5)")), ToolInput(tag="in_hotness", input_type=String(optional=True), prefix="--hotness", doc=InputDocumentation(doc="Mutability parameters to estimate: e = estimate, ce =\nestimate + confidence interval (default: e,e,e,e,e,e)")), ToolInput(tag="in_o_format", input_type=String(optional=True), prefix="--oformat", doc=InputDocumentation(doc="IgPhyML output format. (default: tab)")), ToolInput(tag="in_no_hlp", input_type=Boolean(optional=True), prefix="--nohlp", doc=InputDocumentation(doc="Don't run HLP model? (default: False)")), ToolInput(tag="in_asr", input_type=Int(optional=True), prefix="--asr", doc=InputDocumentation(doc="Ancestral sequence reconstruction interval (0-1).\n(default: -1)")), ToolInput(tag="in_lineages", input_type=String(), position=0, doc=InputDocumentation(doc="successfully processed records.")), ToolInput(tag="in_lineages_fail", input_type=String(), position=1, doc=InputDocumentation(doc="database records failed processing.")), ToolInput(tag="in_ig_phy_ml_pass", input_type=String(), position=2, doc=InputDocumentation(doc="parameter estimates and lineage trees from running IgPhyML, if specified"))], outputs=[ToolOutput(tag="out_md", output_type=File(optional=True), selector=InputSelector(input_to_select="in_md", type_hint=File()), doc=OutputDocumentation(doc="List of fields to containing metadata to include in\noutput fasta file sequence headers. (default: None)"))], container="quay.io/biocontainers/changeo:1.0.2--py_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Buildtrees_Py_V0_1_0().translate("wdl")

