from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Int, File, Boolean, String

Agat_Sp_Extract_Sequences_Pl_V0_1_0 = CommandToolBuilder(tool="agat_sp_extract_sequences.pl", base_command=["agat_sp_extract_sequences.pl"], inputs=[ToolInput(tag="in_split", input_type=Int(optional=True), prefix="--split", doc=InputDocumentation(doc="extract 5'UTR with introns: agat_sp_extract_sequences.pl -g")), ToolInput(tag="in_gff", input_type=File(optional=True), prefix="--gff", doc=InputDocumentation(doc="String - Input GTF/GFF file.")), ToolInput(tag="in_fast_a", input_type=File(optional=True), prefix="--fasta", doc=InputDocumentation(doc="String - Input fasta file.")), ToolInput(tag="in_dnrc", input_type=Boolean(optional=True), prefix="--dnrc", doc=InputDocumentation(doc="Boolean - dnrc means `do not reverse complement`, by default the\nextrated sequence of a feature on the minus strand is reverse\ncomplemented. You can deactivate the behavior by using this\noption.")), ToolInput(tag="in_string_define_feature", input_type=Boolean(optional=True), prefix="-t", doc=InputDocumentation(doc="String - Define the feature you want to extract the sequence\nfrom. Default 'cds'. Most common choice are:\ngene,mrna,exon,cds,trna,three_prime_utr,five_prime_utr. When you\nchoose exon (or cds,utr,etc.), all the exons of a same parent\nfeature are attached together before to extract the sequence. If\nyou wish to extract each exon of an mRNA independently, see\noption --split. /!\ `-t mRNA` will extract the features labeled\nas 'mRNA' and corresponds to the cdna* because it contains the\nintrons if any. It does not actually extract the mRNAs as it is\ndefined biologicaly. To extract the mRNA as defined biologicaly\nyou must use `-t exon`. *Not a real cdna because it is not\nreversed")), ToolInput(tag="in_protein", input_type=Boolean(optional=True), prefix="--protein", doc=InputDocumentation(doc="Boolean - Will translate the extracted sequence in Amino acid.\nBy default the codon table used is the 1 (Standard). See --table\nparameter for more options.")), ToolInput(tag="in_codon", input_type=Int(optional=True), prefix="--codon", doc=InputDocumentation(doc="Integer - Allow to choose the codon table for the translation.\n[default 1]")), ToolInput(tag="in_alternative_start_codon", input_type=Int(optional=True), prefix="--alternative_start_codon", doc=InputDocumentation(doc="Bolean - When activated it can affect the translation of the\nstart codon. Indeed alternative start codons exist, and are\ntranslated by the cells'machinery by a Methionine (M). By\ndefault AGAT translates the first codon as other codons by the\ncorresponding AA. If you wish to translate the first codon by a\nM when it is a valid alternative start codon, activate this\nparameter. If the sequence you try to translate is a CDS (or\nstarting by a CDS), the phase is checked and the alternative\nstart codon is accepted only if the phase is 0.")), ToolInput(tag="in_eo", input_type=Boolean(optional=True), prefix="--eo", doc=InputDocumentation(doc="Boolean - Called 'extremity only', this option will extract only\nthe adjacent parts of a feature. This option has to be activated\nwith -u and/or -p option. /!\ using -u and -p together builds a\nchimeric sequence which will be the concatenation of the left\nand right extremities of a feature.")), ToolInput(tag="in_full", input_type=Boolean(optional=True), prefix="--full", doc=InputDocumentation(doc="Boolean - This option allows dealing with multifeature like cds\nor exon, to extract the full sequence from start extremity to\nthe end extremity, i.e with introns. The use of that option with\nexon feature will give the same result as extracting the mrna\nsequence (-t mRNA) and corresponds to the cdna*. (To actually\nextract an mRNA as it is defined biologicaly you need to use the\n`-t exon` option wihtout the --full option and wihtout the\n--split option) Use of that option on cds will give the cdna*\nwithout the untraslated regions (UTRs). *Not a real cdna because\nit is not reversed")), ToolInput(tag="in_upstream", input_type=Int(optional=True), prefix="-upstream", doc=InputDocumentation(doc="Integer - It will take that number of nucleotide in more at the\n5' extremity. /!\ You must activate the option '--full' if you\nwish to extract only the most upstream part of certain features\n(exon,cds,utr) otherwise you will extract each upstream parts of\nthe subfeatures (e.g many cds parts may be needed to shape a cds\nin its whole).")), ToolInput(tag="in_downstream", input_type=Int(optional=True), prefix="-downstream", doc=InputDocumentation(doc="Integer - It will take that number of nucleotide in more at the\n3' extremity. /!\ You must activate the option '--full' if you\nwith to extract only the most downstream part of certain feature\n(exon,cds,utr) otherwise you will extract each downstream parts\nof the subfeatures (e.g many cds parts may be needed to shape a\ncds in its whole).")), ToolInput(tag="in_cdna", input_type=Boolean(optional=True), prefix="--cdna", doc=InputDocumentation(doc="Boolean - This extract the cdna* sequence (i.e transcribed\nsequence (devoid of introns, but containing untranslated\nexons)). It corresponds to extract the exons sequences. *Not a\nreal cdna because it is not reversed")), ToolInput(tag="in_of_s", input_type=Boolean(optional=True), prefix="--ofs", doc=InputDocumentation(doc="String - Output Fields Separator for the description field of\nthe header of the created fasta sequences. By default it's a\nspace < > but can be modified by any String or character using\nthis option.")), ToolInput(tag="in_clean_internal_stop", input_type=Boolean(optional=True), prefix="--clean_internal_stop", doc=InputDocumentation(doc="Boolean - The Clean Internal Stop option allows replacing the\ntranslation of the stop codons present among the sequence that\nis represented by the <*> character by <X>. Indeed the <*>\ncharacter can be disturbing for many programs (e.g interproscan)")), ToolInput(tag="in_clean_final_stop", input_type=Boolean(optional=True), prefix="--clean_final_stop", doc=InputDocumentation(doc="Boolean - The Clean Final Stop option allows removing the\ntranslation of the final stop codons that is represented by the\n<*> character. This character can be disturbing for many\nprograms (e.g interproscan)")), ToolInput(tag="in_remove_orf_offset", input_type=Boolean(optional=True), prefix="--remove_orf_offset", doc=InputDocumentation(doc="Boolean - CDS can start with a phase different from 0 when a\ngene model is fragmented. When asking for protein translation\nthis (start) offset is trimmed out automatically. But when you\nextract CDS dna sequences, this (start) offset is not removed by\ndefault. To remove it activate this option. If --up or --do\noption are used too, the (start) offset is trimmed first, then\nis added the piece of sequence asked for.")), ToolInput(tag="in_output", input_type=File(optional=True), prefix="--output", doc=InputDocumentation(doc="String - Output fasta file. If no output file is specified, the\noutput will be written to STDOUT.")), ToolInput(tag="in_agat_sp_extract_sequences_do_tpl", input_type=String(), position=0, doc=InputDocumentation(doc="Description:"))], outputs=[ToolOutput(tag="out_output", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output", type_hint=File()), doc=OutputDocumentation(doc="String - Output fasta file. If no output file is specified, the\noutput will be written to STDOUT."))], container="quay.io/biocontainers/agat:0.5.1--pl526r35_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Agat_Sp_Extract_Sequences_Pl_V0_1_0().translate("wdl")

