from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, String

Novoindex_V0_1_0 = CommandToolBuilder(tool="novoindex", base_command=["novoindex"], inputs=[ToolInput(tag="in_kmer_length_used", input_type=Boolean(optional=True), prefix="-k", doc=InputDocumentation(doc="99        is the k-mer length to be used for the index. Typically 14.")), ToolInput(tag="in_step_size_index", input_type=Boolean(optional=True), prefix="-s", doc=InputDocumentation(doc="9         is the step size for the index. Typical values are from 1 to 3.")), ToolInput(tag="in_sets_number_threads", input_type=Boolean(optional=True), prefix="-t", doc=InputDocumentation(doc="9         sets number of threads to use for indexing.")), ToolInput(tag="in_sets_lower_case", input_type=Boolean(optional=True), prefix="-m", doc=InputDocumentation(doc="sets lower case masking on. Lower case sequence will not be indexed.")), ToolInput(tag="in_sets_bisulphite_indexing", input_type=Boolean(optional=True), prefix="-b", doc=InputDocumentation(doc="sets bisulphite indexing and alignment mode for methylation experiments.")), ToolInput(tag="in_add_m_tag", input_type=Boolean(optional=True), prefix="-5", doc=InputDocumentation(doc="add an M5 tag to sequence headers.")), ToolInput(tag="in_name_sets_name", input_type=Boolean(optional=True), prefix="-n", doc=InputDocumentation(doc="name      sets the an internal name for the reference sequence index. This is\nused in report headers and as the AS: field in SAM SQ record.\nDefaults to the indexfile name.")), ToolInput(tag="in_novo_index", input_type=String(), position=0, doc=InputDocumentation(doc="-k 99 -s 9 -m indexfile sequencefiles....")), ToolInput(tag="in_index_file", input_type=String(), position=0, doc=InputDocumentation(doc="is the filename for the indexed reference sequence generated by novoindex.")), ToolInput(tag="in_sequence_files", input_type=String(), position=1, doc=InputDocumentation(doc="a list of sequence files in fasta format to be included in the index."))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Novoindex_V0_1_0().translate("wdl", allow_empty_container=True)

