class: CommandLineTool
id: phyluce_assembly_get_fastas_from_match_counts.cwl
inputs:
- id: in_locus_db
  doc: "--match-count-output\nMATCH_COUNT_OUTPUT\n[--incomplete-matrix INCOMPLETE_MATRIX]\n\
    --output OUTPUT\n[--verbosity {INFO,WARN,CRITICAL}]\n[--log-path LOG_PATH]\n[--extend-locus-db\
    \ EXTEND_LOCUS_DB]\n[--extend-locus-contigs EXTEND_LOCUS_CONTIGS]"
  type: File
  inputBinding:
    prefix: --locus-db
- id: in_contigs
  doc: "The directory containing the assembled contigs in\nwhich you searched for\
    \ UCE loci. (default: None)"
  type: Directory
  inputBinding:
    prefix: --contigs
- id: in_match_count_output
  doc: "The output file containing taxa and loci in\ncomplete/incomplete matrices\
    \ generated by\nget_match_counts.py. (default: None)"
  type: File
  inputBinding:
    prefix: --match-count-output
- id: in_incomplete_matrix
  doc: "The path to the outfile for incomplete-matrix records.\nRequired when processing\
    \ an incomplete data matrix.\n(default: False)"
  type: File
  inputBinding:
    prefix: --incomplete-matrix
- id: in_output
  doc: "The path to the output FASTA file you want to create.\n(default: None)"
  type: File
  inputBinding:
    prefix: --output
- id: in_verbosity
  doc: 'The logging level to use. (default: INFO)'
  type: string
  inputBinding:
    prefix: --verbosity
- id: in_log_path
  doc: 'The path to a directory to hold logs. (default: None)'
  type: File
  inputBinding:
    prefix: --log-path
- id: in_extend_locus_db
  doc: "An SQLlite database file holding probe matches to\nother targeted loci. (default:\
    \ None)"
  type: File
  inputBinding:
    prefix: --extend-locus-db
- id: in_extend_locus_contigs
  doc: "A directory holding the assembled contigs (from\ngenomes or another study)\
    \ referenced by --extend-\nlocus-db. (default: None)\n"
  type: Directory
  inputBinding:
    prefix: --extend-locus-contigs
outputs:
- id: out_stdout
  doc: Standard output stream
  type: stdout
- id: out_locus_db
  doc: "--match-count-output\nMATCH_COUNT_OUTPUT\n[--incomplete-matrix INCOMPLETE_MATRIX]\n\
    --output OUTPUT\n[--verbosity {INFO,WARN,CRITICAL}]\n[--log-path LOG_PATH]\n[--extend-locus-db\
    \ EXTEND_LOCUS_DB]\n[--extend-locus-contigs EXTEND_LOCUS_CONTIGS]"
  type: File
  outputBinding:
    glob: $(inputs.in_locus_db)
- id: out_match_count_output
  doc: "The output file containing taxa and loci in\ncomplete/incomplete matrices\
    \ generated by\nget_match_counts.py. (default: None)"
  type: File
  outputBinding:
    glob: $(inputs.in_match_count_output)
- id: out_output
  doc: "The path to the output FASTA file you want to create.\n(default: None)"
  type: File
  outputBinding:
    glob: $(inputs.in_output)
cwlVersion: v1.1
baseCommand:
- phyluce_assembly_get_fastas_from_match_counts
