from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Boolean, Int, String, Float

Ngshmmalign_V0_1_0 = CommandToolBuilder(tool="ngshmmalign", base_command=["ngshmmalign"], inputs=[ToolInput(tag="in_file_containing_profilemsa", input_type=File(optional=True), prefix="-r", doc=InputDocumentation(doc="File containing the profile/MSA of the")), ToolInput(tag="in_alnsam_filename_written", input_type=File(optional=True), prefix="-o", doc=InputDocumentation(doc="(=aln.sam)                Filename where alignment will be written to")), ToolInput(tag="in_arg_devnull_filename", input_type=Boolean(optional=True), prefix="-w", doc=InputDocumentation(doc="[ --wrong ] arg (=/dev/null)  Filename where alignment will be written\nthat are filtered (too short, unpaired)")), ToolInput(tag="in_number_threads_use", input_type=Int(optional=True), prefix="-t", doc=InputDocumentation(doc="(=8)                      Number of threads to use for alignment.\nDefaults to number of logical cores found")), ToolInput(tag="in_clean_mafft_files", input_type=Boolean(optional=True), prefix="-l", doc=InputDocumentation(doc="Do not clean up MAFFT temporary MSA files")), ToolInput(tag="in_use_search_avoiding", input_type=Boolean(optional=True), prefix="-E", doc=InputDocumentation(doc="Use full-exhaustive search, avoiding indexed")), ToolInput(tag="in_consensus_name_willbe", input_type=String(optional=True), prefix="-N", doc=InputDocumentation(doc="(=CONSENSUS)              Name of consensus reference contig that will\nbe created")), ToolInput(tag="in_loci_ambiguous_bases", input_type=Boolean(optional=True), prefix="-U", doc=InputDocumentation(doc="Loci with ambiguous bases get their emission\nprobabilities according to their allele\nfrequencies. In practice this is\nundesirable, as it leads to systematic\naccumulation of gaps in homopolymeric\nregions with SNVs")), ToolInput(tag="in_arg_value_seed", input_type=Boolean(optional=True), prefix="-s", doc=InputDocumentation(doc="[ --seed ] arg (=42)          Value of seed for deterministic run. A value\nof 0 will pick a random seed from some\nnon-deterministic entropy source")), ToolInput(tag="in_hardclip_reads_bases", input_type=Boolean(optional=True), prefix="--hard", doc=InputDocumentation(doc="Hard-clip reads. Clipped bases will NOT be\nin the sequence in the alignment")), ToolInput(tag="in_extreme_hardclip_reads", input_type=Boolean(optional=True), prefix="--HARD", doc=InputDocumentation(doc="Extreme Hard-clip reads. Do not write\nhard-clip in CIGAR, as if the hard-clipped\nbases never existed. Mutually exclusive with\nprevious option")), ToolInput(tag="in_show_progress_indicator", input_type=Boolean(optional=True), prefix="-v", doc=InputDocumentation(doc="Show progress indicator while aligning")), ToolInput(tag="in_l_minimum_mapped", input_type=Int(optional=True), prefix="-M", doc=InputDocumentation(doc="(=L * 0.8)                Minimum mapped length of read")), ToolInput(tag="in_minimum_frequency_calling", input_type=Float(optional=True), prefix="-a", doc=InputDocumentation(doc="(=0.05)                   Minimum frequency for calling ambiguous base")), ToolInput(tag="in_error", input_type=Float(optional=True), prefix="--error", doc=InputDocumentation(doc="(=0.005)             Global substitution probability")), ToolInput(tag="in_go", input_type=Float(optional=True), prefix="--go", doc=InputDocumentation(doc="(=1e-4)                 Gap open probability")), ToolInput(tag="in_ge", input_type=Float(optional=True), prefix="--ge", doc=InputDocumentation(doc="(=0.30)                 Gap extend probability")), ToolInput(tag="in_io", input_type=Float(optional=True), prefix="--io", doc=InputDocumentation(doc="(=5e-5)                 Insert open probability")), ToolInput(tag="in_ie", input_type=Float(optional=True), prefix="--ie", doc=InputDocumentation(doc="(=0.50)                 Insert extend probability")), ToolInput(tag="in_ep", input_type=Int(optional=True), prefix="--ep", doc=InputDocumentation(doc="(=1/L)                  Jump to end probability; usually 1/L, where\nL is the average length of the reads")), ToolInput(tag="in_lco", input_type=Float(optional=True), prefix="--lco", doc=InputDocumentation(doc="(=0.10)                Left clip open probability")), ToolInput(tag="in_lce", input_type=Float(optional=True), prefix="--lce", doc=InputDocumentation(doc="(=0.90)                Left clip extend probability")), ToolInput(tag="in_rco", input_type=String(optional=True), prefix="--rco", doc=InputDocumentation(doc="(=lco/L)               Right clip open probability")), ToolInput(tag="in_rce", input_type=Float(optional=True), prefix="--rce", doc=InputDocumentation(doc="(=0.90)                Right clip extend probability")), ToolInput(tag="in_reference", input_type=String(), position=0, doc=InputDocumentation(doc="-R arg                           File containing the profile/MSA of the ")), ToolInput(tag="in_lookup", input_type=String(), position=0, doc=InputDocumentation(doc="-X                               Replace general aligned state 'M' with '=' "))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Ngshmmalign_V0_1_0().translate("wdl", allow_empty_container=True)

