#!/usr/bin/env cwl-runner

baseCommand:
- gottcha.pl
class: CommandLineTool
cwlVersion: v1.0
id: gottcha.pl
inputs:
- doc: '|i    <STRING>  Input one or multiple FASTQ file(s). Use comma (,) to separate
    multiple input files.'
  id: input
  inputBinding:
    prefix: --input
  type: boolean
- doc: '|d <STRING>  The path of signature database. The database can be in FASTA
    format or BWA index (5 files).'
  id: database
  inputBinding:
    prefix: --database
  type: boolean
- doc: '|t  <INT>     Number of threads [default: auto-detect]'
  id: threads
  inputBinding:
    prefix: --threads
  type: boolean
- doc: '|l  <STRING>  Specify the taxonomic level of the input database  (e.g. family,
    species, genus, strain, etc.). The value will be auto-detected if the input database
    ended with levels (e.g. GOTTCHA_db.species). [default: none]'
  id: db_level
  inputBinding:
    prefix: --dbLevel
  type: boolean
- doc: '|o   <STRING>  Output directory [default: ./]'
  id: outdir
  inputBinding:
    prefix: --outdir
  type: boolean
- doc: '|p   <STRING>  Filename prefix of the output. [default: <INPUT_FILENAME_PREFIX>]'
  id: prefix
  inputBinding:
    prefix: --prefix
  type: boolean
- doc: '|r   <STRING>  The field will be used to calculate relative abundance. You
    can specify one of the following fields: "LINEAR_LENGTH", "TOTAL_BP_MAPPED", "HIT_COUNT",
    "LINEAR_DOC". [default: LINEAR_DOC]'
  id: rel_abu
  inputBinding:
    prefix: --relAbu
  type: boolean
- doc: '|m     <STRING>  You can specify one of the output mode: "summary" : this
    mode will report a summary of profiling result to *.gottcha.tsv file. "full"    :
    other than a summary, this mode will report unfiltered result to *.gottcha_full.tsv
    with more detail. "all"     : other than two tables, this mode will  keep all
    output files that were  generated by each profiling step. [default: summary]'
  id: mode
  inputBinding:
    prefix: --mode
  type: boolean
- doc: '|n       Ignore alignments that hit to plasmids [default: null]'
  id: no_plasmid_hit
  inputBinding:
    prefix: --noPlasmidHit
  type: boolean
- doc: '|b   <STRING>  BWA-MEM in this script is used to map input reads to  GOTTCHA
    database. If you want to run it with your own parameters, use this option to specify.
    [default: "-k 30 -T 0 -B 100 -O 100 -E 100"]'
  id: bwa_opt
  inputBinding:
    prefix: --bwaOpt
  type: boolean
- doc: '|s    <STRING>  Specify a directory contains pre-splitrimmed input files.
    E.g. input file is "test.fastq", the script will looking for "test_splitrim.fastq"
    and "test_splitrim.stats.txt" in the specified directory.'
  id: st_dir
  inputBinding:
    prefix: --stDir
  type: boolean
- doc: Dump the mapping result in SAM format.
  id: dumps_am
  inputBinding:
    prefix: --dumpSam
  type: boolean
- doc: '<INT>     Minimum quality for a read to be considered valid (0-41) [default:
    20]'
  id: min_q
  inputBinding:
    prefix: --minQ
  type: boolean
- doc: '<INT>     Fixed length to which each trimmed read will be cut  down to [default:
    30]'
  id: fix_l
  inputBinding:
    prefix: --fixL
  type: boolean
- doc: '<INT>     ASCII encoding of quality score (33 or 64) [default:  33] '
  id: ascii
  inputBinding:
    prefix: --ascii
  type: boolean
- doc: '<FLOAT>   Minimum linear coverage to be considered valid in  abundance calculation
    [default: 0.005]'
  id: min_cov
  inputBinding:
    prefix: --minCov
  type: boolean
- doc: '<INT>     Minimum Mean-Linear-Hit-Length to be considered valid in abundance
    calculation [default: 5]'
  id: min_ml_hl
  inputBinding:
    prefix: --minMLHL
  type: boolean
- doc: '<FLOAT>   Critical coverage below which --minMLHL will cause an organism to
    fail [default: 0.006]'
  id: c_cov
  inputBinding:
    prefix: --cCov
  type: boolean
- doc: '<INT>     Minimum unique length to be considered valid in abundance calculation
    [default: 100]'
  id: min_len
  inputBinding:
    prefix: --minLen
  type: boolean
- doc: <INT>     Minimum number of hits to be considered valid in  abundance calculation
    [10]
  id: min_hits
  inputBinding:
    prefix: --minHits
  type: boolean
