class: CommandLineTool
id: gottcha.pl.cwl
inputs:
- id: input
  doc: '|i    <STRING>  Input one or multiple FASTQ file(s). Use comma (,) to separate
    multiple input files.'
  type: boolean
  inputBinding:
    prefix: --input
- id: database
  doc: '|d <STRING>  The path of signature database. The database can be in FASTA
    format or BWA index (5 files).'
  type: boolean
  inputBinding:
    prefix: --database
- id: threads
  doc: '|t  <INT>     Number of threads [default: auto-detect]'
  type: boolean
  inputBinding:
    prefix: --threads
- id: db_level
  doc: '|l  <STRING>  Specify the taxonomic level of the input database  (e.g. family,
    species, genus, strain, etc.). The value will be auto-detected if the input database
    ended with levels (e.g. GOTTCHA_db.species). [default: none]'
  type: boolean
  inputBinding:
    prefix: --dbLevel
- id: outdir
  doc: '|o   <STRING>  Output directory [default: ./]'
  type: boolean
  inputBinding:
    prefix: --outdir
- id: prefix
  doc: '|p   <STRING>  Filename prefix of the output. [default: <INPUT_FILENAME_PREFIX>]'
  type: boolean
  inputBinding:
    prefix: --prefix
- id: rel_abu
  doc: '|r   <STRING>  The field will be used to calculate relative abundance. You
    can specify one of the following fields: "LINEAR_LENGTH", "TOTAL_BP_MAPPED", "HIT_COUNT",
    "LINEAR_DOC". [default: LINEAR_DOC]'
  type: boolean
  inputBinding:
    prefix: --relAbu
- id: mode
  doc: '|m     <STRING>  You can specify one of the output mode: "summary" : this
    mode will report a summary of profiling result to *.gottcha.tsv file. "full"    :
    other than a summary, this mode will report unfiltered result to *.gottcha_full.tsv
    with more detail. "all"     : other than two tables, this mode will  keep all
    output files that were  generated by each profiling step. [default: summary]'
  type: boolean
  inputBinding:
    prefix: --mode
- id: no_plasmid_hit
  doc: '|n       Ignore alignments that hit to plasmids [default: null]'
  type: boolean
  inputBinding:
    prefix: --noPlasmidHit
- id: bwa_opt
  doc: '|b   <STRING>  BWA-MEM in this script is used to map input reads to  GOTTCHA
    database. If you want to run it with your own parameters, use this option to specify.
    [default: "-k 30 -T 0 -B 100 -O 100 -E 100"]'
  type: boolean
  inputBinding:
    prefix: --bwaOpt
- id: st_dir
  doc: '|s    <STRING>  Specify a directory contains pre-splitrimmed input files.
    E.g. input file is "test.fastq", the script will looking for "test_splitrim.fastq"
    and "test_splitrim.stats.txt" in the specified directory.'
  type: boolean
  inputBinding:
    prefix: --stDir
- id: dumps_am
  doc: Dump the mapping result in SAM format.
  type: boolean
  inputBinding:
    prefix: --dumpSam
- id: min_q
  doc: '<INT>     Minimum quality for a read to be considered valid (0-41) [default:
    20]'
  type: boolean
  inputBinding:
    prefix: --minQ
- id: fix_l
  doc: '<INT>     Fixed length to which each trimmed read will be cut  down to [default:
    30]'
  type: boolean
  inputBinding:
    prefix: --fixL
- id: ascii
  doc: '<INT>     ASCII encoding of quality score (33 or 64) [default:  33] '
  type: boolean
  inputBinding:
    prefix: --ascii
- id: min_cov
  doc: '<FLOAT>   Minimum linear coverage to be considered valid in  abundance calculation
    [default: 0.005]'
  type: boolean
  inputBinding:
    prefix: --minCov
- id: min_ml_hl
  doc: '<INT>     Minimum Mean-Linear-Hit-Length to be considered valid in abundance
    calculation [default: 5]'
  type: boolean
  inputBinding:
    prefix: --minMLHL
- id: c_cov
  doc: '<FLOAT>   Critical coverage below which --minMLHL will cause an organism to
    fail [default: 0.006]'
  type: boolean
  inputBinding:
    prefix: --cCov
- id: min_len
  doc: '<INT>     Minimum unique length to be considered valid in abundance calculation
    [default: 100]'
  type: boolean
  inputBinding:
    prefix: --minLen
- id: min_hits
  doc: <INT>     Minimum number of hits to be considered valid in  abundance calculation
    [10]
  type: boolean
  inputBinding:
    prefix: --minHits
outputs: []
cwlVersion: v1.1
baseCommand:
- gottcha.pl
