from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import String, File, Boolean, Int

Fasttreemp_V0_1_0 = CommandToolBuilder(tool="FastTreeMP", base_command=["FastTreeMP"], inputs=[ToolInput(tag="in_quiet", input_type=String(optional=True), prefix="-quiet", doc=InputDocumentation(doc="suppress reporting information")), ToolInput(tag="in_nopr", input_type=String(optional=True), prefix="-nopr", doc=InputDocumentation(doc="suppress progress indicator")), ToolInput(tag="in_log", input_type=File(optional=True), prefix="-log", doc=InputDocumentation(doc="-- save intermediate trees, settings, and model details")), ToolInput(tag="in_fastest", input_type=Boolean(optional=True), prefix="-fastest", doc=InputDocumentation(doc="-- speed up the neighbor joining phase & reduce memory usage\n(recommended for >50,000 sequences)")), ToolInput(tag="in_analyze_alignments_onlyuse", input_type=Int(optional=True), prefix="-n", doc=InputDocumentation(doc="to analyze multiple alignments (phylip format only)\n(use for global bootstrap, with seqboot and CompareToBootstrap.pl)")), ToolInput(tag="in_no_support", input_type=String(optional=True), prefix="-nosupport", doc=InputDocumentation(doc="not compute support values")), ToolInput(tag="in_in_tree", input_type=String(optional=True), prefix="-intree", doc=InputDocumentation(doc="optimize branch lengths for a fixed topology")), ToolInput(tag="in_in_tree_one", input_type=File(optional=True), prefix="-intree1", doc=InputDocumentation(doc="to use this starting tree for all the alignments\n(for faster global bootstrap on huge alignments)")), ToolInput(tag="in_pseudo", input_type=String(optional=True), prefix="-pseudo", doc=InputDocumentation(doc="use pseudocounts (recommended for highly gapped sequences)")), ToolInput(tag="in_gtr", input_type=Boolean(optional=True), prefix="-gtr", doc=InputDocumentation(doc="-- generalized time-reversible model (nucleotide alignments only)")), ToolInput(tag="in_lg", input_type=Boolean(optional=True), prefix="-lg", doc=InputDocumentation(doc="-- Le-Gascuel 2008 model (amino acid alignments only)")), ToolInput(tag="in_wag", input_type=Boolean(optional=True), prefix="-wag", doc=InputDocumentation(doc="-- Whelan-And-Goldman 2001 model (amino acid alignments only)")), ToolInput(tag="in_quote", input_type=Boolean(optional=True), prefix="-quote", doc=InputDocumentation(doc="-- allow spaces and other restricted characters (but not ' ) in\nsequence names and quote names in the output tree (fasta input only;\nFastTree will not be able to read these trees back in)")), ToolInput(tag="in_no_ml", input_type=String(optional=True), prefix="-noml", doc=InputDocumentation(doc="turn off maximum-likelihood")), ToolInput(tag="in_nome", input_type=String(optional=True), prefix="-nome", doc=InputDocumentation(doc="turn off minimum-evolution NNIs and SPRs\n(recommended if running additional ML NNIs with -intree)")), ToolInput(tag="in_cat", input_type=Boolean(optional=True), prefix="-cat", doc=InputDocumentation(doc="# to specify the number of rate categories of sites (default 20)\nor -nocat to use constant rates")), ToolInput(tag="in_gamma", input_type=Boolean(optional=True), prefix="-gamma", doc=InputDocumentation(doc="-- after optimizing the tree under the CAT approximation,\nrescale the lengths to optimize the Gamma20 likelihood")), ToolInput(tag="in_constraints", input_type=Int(optional=True), prefix="-constraints", doc=InputDocumentation(doc="to constrain the topology search\nconstraintAlignment should have 1s or 0s to indicates splits")), ToolInput(tag="in_expert", input_type=Boolean(optional=True), prefix="-expert", doc=InputDocumentation(doc="-- see more options"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Fasttreemp_V0_1_0().translate("wdl", allow_empty_container=True)

