class: CommandLineTool
id: anvi_display_structure.cwl
inputs:
- id: structure_db
  doc: Anvi'o structure database.
  type: string
  inputBinding:
    prefix: --structure-db
- id: profile_db
  doc: Anvi'o profile database
  type: string
  inputBinding:
    prefix: --profile-db
- id: contigs_db
  doc: Anvi'o contigs database generated by 'anvi-gen- contigs'
  type: string
  inputBinding:
    prefix: --contigs-db
- id: variability_profile
  doc: The output of anvi-gen-variability-profile, or a different variant-calling
    output that has been converted to the anvi'o format.
  type: string
  inputBinding:
    prefix: --variability-profile
- id: splits_of_interest
  doc: A file with split names. There should be only one column in the file, and each
    line should correspond to a unique split name.
  type: File
  inputBinding:
    prefix: --splits-of-interest
- id: collection_name
  doc: Collection name.
  type: string
  inputBinding:
    prefix: --collection-name
- id: bin_id
  doc: If provided, any genes found in both your bin and your structure database will
    be available for display.
  type: string
  inputBinding:
    prefix: --bin-id
- id: samples_of_interest
  doc: A file with samples names. There should be only one column in the file, and
    each line should correspond to a unique sample name (without a column header).
  type: File
  inputBinding:
    prefix: --samples-of-interest
- id: genes_of_interest
  doc: A file with anvi'o gene caller IDs. There should be only one column in the
    file, and each line should correspond to a unique gene caller id (without a column
    header).
  type: File
  inputBinding:
    prefix: --genes-of-interest
- id: gene_caller_ids
  doc: Gene caller ids. Multiple of them can be declared separated by a delimiter
    (the default is a comma). In anvi-gen-variability-profile, if you declare nothing
    you will get all genes matching your other filtering criteria. In other programs,
    you may get everything, nothing, or an error. It really depends on the situation.
    Fortunately, mistakes are cheap, so it's worth a try.
  type: string
  inputBinding:
    prefix: --gene-caller-ids
- id: min_departure_from_consensus
  doc: Takes a value between 0 and 1, where 1 is maximum divergence from the consensus.
    it can be an expensive operation to display every variable position, and so the
    default is 0.05. To display every variable position, set this parameter to 0
  type: double
  inputBinding:
    prefix: --min-departure-from-consensus
- id: sa_avs_only
  doc: If provided, variability will be generated for single amino acid variants (SAAVs)
    and not for single codon variants (SCVs). This could save you some time if you're
    only interested in SAAVs.
  type: boolean
  inputBinding:
    prefix: --SAAVs-only
- id: s_cvs_only
  doc: If provided, variability will be generated for single codon variants (SCVs)
    and not for single amino acid variants (SAAVs). This could save you some time
    if you're only interested in SCVs.
  type: boolean
  inputBinding:
    prefix: --SCVs-only
- id: ip_address
  doc: IP address for the HTTP server. The default ip address (0.0.0.0) should work
    just fine for most.
  type: string
  inputBinding:
    prefix: --ip-address
- id: port_number
  doc: Port number to use for anvi'o services. If nothing is declared, anvi'o will
    try to find a suitable port number, starting from the default port number, 8080.
  type: long
  inputBinding:
    prefix: --port-number
- id: browser_path
  doc: By default, anvi'o will use your default browser to launch the interactive
    interface. If you would like to use something else than your system default, you
    can provide a full path for an alternative browser using this parameter, and hope
    for the best. For instance we are using this parameter to call Google's experimental
    browser, Canary, which performs better with demanding visualizations.
  type: File
  inputBinding:
    prefix: --browser-path
- id: server_only
  doc: The default behavior is to start the local server, and fire up a browser that
    connects to the server. If you have other plans, and want to start the server
    without calling the browser, this is the flag you need.
  type: boolean
  inputBinding:
    prefix: --server-only
- id: password_protected
  doc: If this flag is set, command line tool will ask you to enter a password and
    interactive interface will be only accessible after entering same password. This
    option is recommended for shared machines like clusters or shared networks where
    computers are not isolated.
  type: boolean
  inputBinding:
    prefix: --password-protected
outputs: []
cwlVersion: v1.1
baseCommand:
- anvi-display-structure
