from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, File, Int, Directory, String, Float

Bamsnap_V0_1_0 = CommandToolBuilder(tool="bamsnap", base_command=["bamsnap"], inputs=[ToolInput(tag="in_bam", input_type=Boolean(optional=True), prefix="-bam", doc=InputDocumentation(doc="[BAM [BAM ...]]  bam or cram file(s)")), ToolInput(tag="in_bam_list", input_type=File(optional=True), prefix="-bamlist", doc=InputDocumentation(doc="list file with bam (or cram) file paths")), ToolInput(tag="in_title", input_type=Boolean(optional=True), prefix="-title", doc=InputDocumentation(doc="[TITLE [TITLE ...]]\ntitle (name) of bam (or cram) file(s)")), ToolInput(tag="in_no_title", input_type=Boolean(optional=True), prefix="-no_title", doc=InputDocumentation(doc="do not draw label.")), ToolInput(tag="in_title_font_size", input_type=Int(optional=True), prefix="-title_fontsize", doc=InputDocumentation(doc="font size of title")), ToolInput(tag="in_pos", input_type=Boolean(optional=True), prefix="-pos", doc=InputDocumentation(doc="[POS [POS ...]]  genomic position (ex. 1:816687-818057, 12:7462545)")), ToolInput(tag="in_vcf", input_type=File(optional=True), prefix="-vcf", doc=InputDocumentation(doc="list file with genomic positions with VCF format")), ToolInput(tag="in_bed", input_type=File(optional=True), prefix="-bed", doc=InputDocumentation(doc="list file with genomic positions with BED format")), ToolInput(tag="in_out", input_type=File(optional=True), prefix="-out", doc=InputDocumentation(doc="output file or title of output file")), ToolInput(tag="in_image_type", input_type=File(optional=True), prefix="-imagetype", doc=InputDocumentation(doc="output file type")), ToolInput(tag="in_conf", input_type=File(optional=True), prefix="-conf", doc=InputDocumentation(doc="configuration file")), ToolInput(tag="in_ref", input_type=File(optional=True), prefix="-ref", doc=InputDocumentation(doc="Reference sequence fasta file (ex. hg19.fa)")), ToolInput(tag="in_ref_index_rebuild", input_type=Boolean(optional=True), prefix="-ref_index_rebuild", doc=InputDocumentation(doc="if you want to rebuild fasta index file (.fai), when\nit is older than the fasta file. (Default: false)")), ToolInput(tag="in_ref_version", input_type=Int(optional=True), prefix="-refversion", doc=InputDocumentation(doc="Reference version (default: hg38)")), ToolInput(tag="in_margin", input_type=Int(optional=True), prefix="-margin", doc=InputDocumentation(doc="genomic margin size")), ToolInput(tag="in_save_image_only", input_type=Boolean(optional=True), prefix="-save_image_only", doc=InputDocumentation(doc="save image only")), ToolInput(tag="in_image_dir_name", input_type=Directory(optional=True), prefix="-image_dir_name", doc=InputDocumentation(doc="image directory name")), ToolInput(tag="in_zip_out", input_type=Boolean(optional=True), prefix="-zipout", doc=InputDocumentation(doc="make a single zip file")), ToolInput(tag="in_separator_height", input_type=String(optional=True), prefix="-separator_height", doc=InputDocumentation(doc="separator's height")), ToolInput(tag="in_process", input_type=Int(optional=True), prefix="-process", doc=InputDocumentation(doc="number of process for multi-processing (default: 1)")), ToolInput(tag="in_draw", input_type=Boolean(optional=True), prefix="-draw", doc=InputDocumentation(doc="[DRAW [DRAW ...]]\ntrack composition (default: -draw coordinates bamplot\nbase gene)")), ToolInput(tag="in_bam_plot", input_type=Boolean(optional=True), prefix="-bamplot", doc=InputDocumentation(doc="[BAMPLOT [BAMPLOT ...]]\ntrack composition of bamplot (default: -bamplot\ncoverage base read)")), ToolInput(tag="in_width", input_type=Int(optional=True), prefix="-width", doc=InputDocumentation(doc="image width (unit:px, default: 1000)")), ToolInput(tag="in_height", input_type=String(optional=True), prefix="-height", doc=InputDocumentation(doc="image height (unit:px)")), ToolInput(tag="in_bgcolor", input_type=String(optional=True), prefix="-bgcolor", doc=InputDocumentation(doc="background color (default: FFFFFF)")), ToolInput(tag="in_plot_margin_top", input_type=Int(optional=True), prefix="-plot_margin_top", doc=InputDocumentation(doc="top margin size of plot (default: 20)")), ToolInput(tag="in_plot_margin_bottom", input_type=Int(optional=True), prefix="-plot_margin_bottom", doc=InputDocumentation(doc="bottom margin size of plot (default: 20)")), ToolInput(tag="in_plot_margin_left", input_type=Int(optional=True), prefix="-plot_margin_left", doc=InputDocumentation(doc="left margin size of plot (default: 0)")), ToolInput(tag="in_plot_margin_right", input_type=Int(optional=True), prefix="-plot_margin_right", doc=InputDocumentation(doc="right margin size of plot (default: 0)")), ToolInput(tag="in_border", input_type=Boolean(optional=True), prefix="-border", doc=InputDocumentation(doc="draw border in plot (default: false)")), ToolInput(tag="in_grid", input_type=Int(optional=True), prefix="-grid", doc=InputDocumentation(doc="draw grid in plot (default: 0 (grid size))")), ToolInput(tag="in_read_thickness", input_type=String(optional=True), prefix="-read_thickness", doc=InputDocumentation(doc="read thickness (unit:px)")), ToolInput(tag="in_read_gap_height", input_type=String(optional=True), prefix="-read_gap_height", doc=InputDocumentation(doc="read gap height (unit:px)")), ToolInput(tag="in_read_gap_width", input_type=Int(optional=True), prefix="-read_gap_width", doc=InputDocumentation(doc="min size of read gap width (unit:px)")), ToolInput(tag="in_read_bgcolor", input_type=String(optional=True), prefix="-read_bgcolor", doc=InputDocumentation(doc="read background color")), ToolInput(tag="in_read_color", input_type=String(optional=True), prefix="-read_color", doc=InputDocumentation(doc="read color")), ToolInput(tag="in_read_pos_color", input_type=String(optional=True), prefix="-read_pos_color", doc=InputDocumentation(doc="positive strand read color")), ToolInput(tag="in_read_neg_color", input_type=String(optional=True), prefix="-read_neg_color", doc=InputDocumentation(doc="negative strand read color")), ToolInput(tag="in_read_color_deletion", input_type=String(optional=True), prefix="-read_color_deletion", doc=InputDocumentation(doc="read color of deletion")), ToolInput(tag="in_read_color_inversion", input_type=String(optional=True), prefix="-read_color_inversion", doc=InputDocumentation(doc="read color of inversion")), ToolInput(tag="in_insert_size_del_threshold", input_type=Int(optional=True), prefix="-insert_size_del_threshold", doc=InputDocumentation(doc="insert size threshold for deletion (default: 1000)")), ToolInput(tag="in_insert_size_ins_threshold", input_type=Int(optional=True), prefix="-insert_size_ins_threshold", doc=InputDocumentation(doc="insert size threshold for insertion (default: 50)")), ToolInput(tag="in_read_group", input_type=String(optional=True), prefix="-read_group", doc=InputDocumentation(doc="read group")), ToolInput(tag="in_read_color_by", input_type=String(optional=True), prefix="-read_color_by", doc=InputDocumentation(doc="read color by")), ToolInput(tag="in_read_color_inter_chrom_chr_one", input_type=Int(optional=True), prefix="-read_color_interchrom_chr1", doc=InputDocumentation(doc="paired read color located in chromosome 1")), ToolInput(tag="in_read_color_inter_chrom_chr_two", input_type=Int(optional=True), prefix="-read_color_interchrom_chr2", doc=InputDocumentation(doc="paired read color located in chromosome 2")), ToolInput(tag="in_read_color_inter_chrom_chr_three", input_type=Int(optional=True), prefix="-read_color_interchrom_chr3", doc=InputDocumentation(doc="paired read color located in chromosome 3")), ToolInput(tag="in_read_color_inter_chrom_chr_four", input_type=Int(optional=True), prefix="-read_color_interchrom_chr4", doc=InputDocumentation(doc="paired read color located in chromosome 4")), ToolInput(tag="in_read_color_inter_chrom_chr_five", input_type=Int(optional=True), prefix="-read_color_interchrom_chr5", doc=InputDocumentation(doc="paired read color located in chromosome 5")), ToolInput(tag="in_read_color_inter_chrom_chr_six", input_type=Int(optional=True), prefix="-read_color_interchrom_chr6", doc=InputDocumentation(doc="paired read color located in chromosome 6")), ToolInput(tag="in_read_color_inter_chrom_chr_seven", input_type=Int(optional=True), prefix="-read_color_interchrom_chr7", doc=InputDocumentation(doc="paired read color located in chromosome 7")), ToolInput(tag="in_read_color_inter_chrom_chr_eight", input_type=Int(optional=True), prefix="-read_color_interchrom_chr8", doc=InputDocumentation(doc="paired read color located in chromosome 8")), ToolInput(tag="in_read_color_inter_chrom_chr_nine", input_type=Int(optional=True), prefix="-read_color_interchrom_chr9", doc=InputDocumentation(doc="paired read color located in chromosome 9")), ToolInput(tag="in_read_color_inter_chrom_chr_one_zero", input_type=Int(optional=True), prefix="-read_color_interchrom_chr10", doc=InputDocumentation(doc="paired read color located in chromosome 10")), ToolInput(tag="in_read_color_inter_chrom_chr_one_one", input_type=Int(optional=True), prefix="-read_color_interchrom_chr11", doc=InputDocumentation(doc="paired read color located in chromosome 11")), ToolInput(tag="in_read_color_inter_chrom_chr_one_two", input_type=Int(optional=True), prefix="-read_color_interchrom_chr12", doc=InputDocumentation(doc="paired read color located in chromosome 12")), ToolInput(tag="in_read_color_inter_chrom_chr_one_three", input_type=Int(optional=True), prefix="-read_color_interchrom_chr13", doc=InputDocumentation(doc="paired read color located in chromosome 13")), ToolInput(tag="in_read_color_inter_chrom_chr_one_four", input_type=Int(optional=True), prefix="-read_color_interchrom_chr14", doc=InputDocumentation(doc="paired read color located in chromosome 14")), ToolInput(tag="in_read_color_inter_chrom_chr_one_five", input_type=Int(optional=True), prefix="-read_color_interchrom_chr15", doc=InputDocumentation(doc="paired read color located in chromosome 15")), ToolInput(tag="in_read_color_inter_chrom_chr_one_six", input_type=Int(optional=True), prefix="-read_color_interchrom_chr16", doc=InputDocumentation(doc="paired read color located in chromosome 16")), ToolInput(tag="in_read_color_inter_chrom_chr_one_seven", input_type=Int(optional=True), prefix="-read_color_interchrom_chr17", doc=InputDocumentation(doc="paired read color located in chromosome 17")), ToolInput(tag="in_read_color_inter_chrom_chr_one_eight", input_type=Int(optional=True), prefix="-read_color_interchrom_chr18", doc=InputDocumentation(doc="paired read color located in chromosome 18")), ToolInput(tag="in_read_color_inter_chrom_chr_one_nine", input_type=Int(optional=True), prefix="-read_color_interchrom_chr19", doc=InputDocumentation(doc="paired read color located in chromosome 19")), ToolInput(tag="in_read_color_inter_chrom_chr_two_zero", input_type=Int(optional=True), prefix="-read_color_interchrom_chr20", doc=InputDocumentation(doc="paired read color located in chromosome 20")), ToolInput(tag="in_read_color_inter_chrom_chr_two_one", input_type=Int(optional=True), prefix="-read_color_interchrom_chr21", doc=InputDocumentation(doc="paired read color located in chromosome 21")), ToolInput(tag="in_read_color_inter_chrom_chr_two_two", input_type=Int(optional=True), prefix="-read_color_interchrom_chr22", doc=InputDocumentation(doc="paired read color located in chromosome 22")), ToolInput(tag="in_read_color_inter_chrom_chrx", input_type=String(optional=True), prefix="-read_color_interchrom_chrX", doc=InputDocumentation(doc="paired read color located in chromosome X")), ToolInput(tag="in_read_color_inter_chrom_chry", input_type=String(optional=True), prefix="-read_color_interchrom_chrY", doc=InputDocumentation(doc="paired read color located in chromosome Y")), ToolInput(tag="in_read_color_inter_chrom_other", input_type=String(optional=True), prefix="-read_color_interchrom_other", doc=InputDocumentation(doc="paired read color located in other chromosome")), ToolInput(tag="in_show_soft_clipped", input_type=Boolean(optional=True), prefix="-show_soft_clipped", doc=InputDocumentation(doc="show soft clipped part (default: false)")), ToolInput(tag="in_show_hard_clipped", input_type=Boolean(optional=True), prefix="-show_hard_clipped", doc=InputDocumentation(doc="show hard clipped part (default: false)")), ToolInput(tag="in_show_clipped", input_type=Boolean(optional=True), prefix="-show_clipped", doc=InputDocumentation(doc="show soft and hard clipped part (default: false)")), ToolInput(tag="in_center_line", input_type=Boolean(optional=True), prefix="-center_line", doc=InputDocumentation(doc="draw center line")), ToolInput(tag="in_no_target_line", input_type=Boolean(optional=True), prefix="-no_target_line", doc=InputDocumentation(doc="do not draw target line")), ToolInput(tag="in_base_font_size", input_type=Int(optional=True), prefix="-base_fontsize", doc=InputDocumentation(doc="font size of base track")), ToolInput(tag="in_base_height", input_type=String(optional=True), prefix="-base_height", doc=InputDocumentation(doc="base track height")), ToolInput(tag="in_base_margin_top", input_type=Int(optional=True), prefix="-base_margin_top", doc=InputDocumentation(doc="top margin size of base track")), ToolInput(tag="in_base_margin_bottom", input_type=Int(optional=True), prefix="-base_margin_bottom", doc=InputDocumentation(doc="bottom margin size of base track")), ToolInput(tag="in_coverage_height", input_type=String(optional=True), prefix="-coverage_height", doc=InputDocumentation(doc="coverage track height")), ToolInput(tag="in_coverage_font_size", input_type=Int(optional=True), prefix="-coverage_fontsize", doc=InputDocumentation(doc="coverage font size")), ToolInput(tag="in_coverage_vaf", input_type=Float(optional=True), prefix="-coverage_vaf", doc=InputDocumentation(doc="coverage variant allele fraction threshold (default:\n0.2)")), ToolInput(tag="in_coverage_color", input_type=String(optional=True), prefix="-coverage_color", doc=InputDocumentation(doc="coverage color")), ToolInput(tag="in_coverage_bgcolor", input_type=String(optional=True), prefix="-coverage_bgcolor", doc=InputDocumentation(doc="coverage track background color")), ToolInput(tag="in_heat_map_height", input_type=String(optional=True), prefix="-heatmap_height", doc=InputDocumentation(doc="coverage heatmap height")), ToolInput(tag="in_heat_map_bgcolor", input_type=String(optional=True), prefix="-heatmap_bgcolor", doc=InputDocumentation(doc="coverage heatmap background color")), ToolInput(tag="in_gene_height", input_type=String(optional=True), prefix="-gene_height", doc=InputDocumentation(doc="gene track height")), ToolInput(tag="in_gene_font_size", input_type=Int(optional=True), prefix="-gene_fontsize", doc=InputDocumentation(doc="font size of gene track")), ToolInput(tag="in_gene_pos_color", input_type=String(optional=True), prefix="-gene_pos_color", doc=InputDocumentation(doc="positive strand color")), ToolInput(tag="in_gene_neg_color", input_type=String(optional=True), prefix="-gene_neg_color", doc=InputDocumentation(doc="negative strand color")), ToolInput(tag="in_coordinates_height", input_type=String(optional=True), prefix="-coordinates_height", doc=InputDocumentation(doc="coordinate height")), ToolInput(tag="in_coordinates_font_size", input_type=Int(optional=True), prefix="-coordinates_fontsize", doc=InputDocumentation(doc="coordinate font size")), ToolInput(tag="in_coordinates_axis_loc", input_type=String(optional=True), prefix="-coordinates_axisloc", doc=InputDocumentation(doc="coordinate axis location")), ToolInput(tag="in_coordinates_bgcolor", input_type=String(optional=True), prefix="-coordinates_bgcolor", doc=InputDocumentation(doc="coordinate background color")), ToolInput(tag="in_coordinates_label_color", input_type=String(optional=True), prefix="-coordinates_labelcolor", doc=InputDocumentation(doc="coordinate label color")), ToolInput(tag="in_separated_bam", input_type=Boolean(optional=True), prefix="-separated_bam", doc=InputDocumentation(doc="draw a plot for each bam")), ToolInput(tag="in_debug", input_type=Boolean(optional=True), prefix="-debug", doc=InputDocumentation(doc="turn on the debugging mode")), ToolInput(tag="in_silence", input_type=Boolean(optional=True), prefix="-silence", doc=InputDocumentation(doc="don't print any log.")), ToolInput(tag="in_sub_command", input_type=String(), position=0, doc=InputDocumentation(doc=""))], outputs=[ToolOutput(tag="out_out", output_type=File(optional=True), selector=InputSelector(input_to_select="in_out", type_hint=File()), doc=OutputDocumentation(doc="output file or title of output file")), ToolOutput(tag="out_image_type", output_type=File(optional=True), selector=InputSelector(input_to_select="in_image_type", type_hint=File()), doc=OutputDocumentation(doc="output file type"))], container="quay.io/biocontainers/bamsnap:0.2.18--py_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Bamsnap_V0_1_0().translate("wdl")

