from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, File, String

Promer_V0_1_0 = CommandToolBuilder(tool="promer", base_command=["promer"], inputs=[ToolInput(tag="in_mum", input_type=Boolean(optional=True), prefix="--mum", doc=InputDocumentation(doc="Use anchor matches that are unique in both the reference\nand query")), ToolInput(tag="in_m_umc_and", input_type=Boolean(optional=True), prefix="--mumcand", doc=InputDocumentation(doc="Same as --mumreference")), ToolInput(tag="in_mum_reference", input_type=Boolean(optional=True), prefix="--mumreference", doc=InputDocumentation(doc="Use anchor matches that are unique in in the reference\nbut not necessarily unique in the query (default behavior)")), ToolInput(tag="in_max_match", input_type=Boolean(optional=True), prefix="--maxmatch", doc=InputDocumentation(doc="Use all anchor matches regardless of their uniqueness")), ToolInput(tag="in_breaklen_set_distance", input_type=Boolean(optional=True), prefix="-b", doc=InputDocumentation(doc="|breaklen     Set the distance an alignment extension will attempt to\nextend poor scoring regions before giving up, measured in\namino acids (default 60)")), ToolInput(tag="in_mincluster_sets_length", input_type=Boolean(optional=True), prefix="-c", doc=InputDocumentation(doc="|mincluster   Sets the minimum length of a cluster of matches, measured in\namino acids (default 20)")), ToolInput(tag="in_depend", input_type=Boolean(optional=True), prefix="--depend", doc=InputDocumentation(doc="Print the dependency information and exit")), ToolInput(tag="in_diagfactor_set_factordefault", input_type=Boolean(optional=True), prefix="-d", doc=InputDocumentation(doc="|diagfactor   Set the clustering diagonal difference separation factor\n(default .11)")), ToolInput(tag="in_maxgap_set_gap", input_type=Boolean(optional=True), prefix="-g", doc=InputDocumentation(doc="|maxgap       Set the maximum gap between two adjacent matches in a\ncluster, measured in amino acids (default 30)")), ToolInput(tag="in_minmatch_set_length", input_type=Boolean(optional=True), prefix="-l", doc=InputDocumentation(doc="|minmatch     Set the minimum length of a single match, measured in amino\nacids (default 6)")), ToolInput(tag="in_masklen_set_maximum", input_type=Boolean(optional=True), prefix="-m", doc=InputDocumentation(doc="|masklen      Set the maximum bookend masking lenth, measured in amino\nacids (default 8)")), ToolInput(tag="in_coords", input_type=File(optional=True), prefix="--coords", doc=InputDocumentation(doc="Automatically generate the original PROmer1.1 '.coords'\noutput file using the 'show-coords' program")), ToolInput(tag="in_prefix_set_prefix", input_type=Boolean(optional=True), prefix="-p", doc=InputDocumentation(doc="|prefix       Set the prefix of the output files (default 'out')")), ToolInput(tag="in_matrix_set_number", input_type=Boolean(optional=True), prefix="-x", doc=InputDocumentation(doc="|matrix       Set the alignment matrix number to 1 [BLOSUM 45], 2 [BLOSUM\n62] or 3 [BLOSUM 80] (default 2)\n")), ToolInput(tag="in_reference", input_type=String(), position=0, doc=InputDocumentation(doc="Set the input reference multi-FASTA DNA file")), ToolInput(tag="in_query", input_type=String(), position=1, doc=InputDocumentation(doc="Set the input query multi-FASTA DNA file"))], outputs=[ToolOutput(tag="out_coords", output_type=File(optional=True), selector=InputSelector(input_to_select="in_coords", type_hint=File()), doc=OutputDocumentation(doc="Automatically generate the original PROmer1.1 '.coords'\noutput file using the 'show-coords' program"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Promer_V0_1_0().translate("wdl", allow_empty_container=True)

