version 1.0

task Anvirunkeggkofams {
  input {
    String? contigs_db
    File? profile_db
    String? collection_name
    File? kegg_data_dir
    Int? num_threads
    Int? hmmer_program
    Boolean? keep_all_hits
    Boolean? log_bit_scores
    Boolean? just_do_it
  }
  command <<<
    anvi_run_kegg_kofams \
      ~{if defined(contigs_db) then ("--contigs-db " +  '"' + contigs_db + '"') else ""} \
      ~{if defined(profile_db) then ("--profile-db " +  '"' + profile_db + '"') else ""} \
      ~{if defined(collection_name) then ("--collection-name " +  '"' + collection_name + '"') else ""} \
      ~{if defined(kegg_data_dir) then ("--kegg-data-dir " +  '"' + kegg_data_dir + '"') else ""} \
      ~{if defined(num_threads) then ("--num-threads " +  '"' + num_threads + '"') else ""} \
      ~{if defined(hmmer_program) then ("--hmmer-program " +  '"' + hmmer_program + '"') else ""} \
      ~{if (keep_all_hits) then "--keep-all-hits" else ""} \
      ~{if (log_bit_scores) then "--log-bitscores" else ""} \
      ~{if (just_do_it) then "--just-do-it" else ""}
  >>>
  runtime {
    docker: "quay.io/biocontainers/anvio-minimal:7--py_0"
  }
  parameter_meta {
    contigs_db: "Anvi'o contigs database generated by 'anvi-gen-\\ncontigs-database' (default: None)"
    profile_db: "If you only want to annotate gene calls in a subset of\\nsequences, you can provide a path to a profile\\ndatabase with this flag. In this case, you must also\\nprovide a collection name so that we know which\\nsequenes to focus on. (default: None)"
    collection_name: "Name of collection you want to focus on. Provide this\\n(along with a profile db) if you only want to annotate\\ngene calls in a certain set of bins. (default: None)"
    kegg_data_dir: "The directory path for your KEGG setup, which will\\ninclude things like KOfam profiles and KEGG MODULE\\ndata. Anvi'o will try to use the default path if you\\ndo not specify anything. (default: None)"
    num_threads: "Maximum number of threads to use for multithreading\\nwhenever possible. Very conservatively, the default is\\n1. It is a good idea to not exceed the number of CPUs\\n/ cores on your system. Plus, please be careful with\\nthis option if you are running your commands on a SGE\\n--if you are clusterizing your runs, and asking for\\nmultiple threads to use, you may deplete your\\nresources very fast. (default: 1)"
    hmmer_program: "Which of the HMMER programs to use to run HMMs\\n(hmmscan or hmmsearch). By default anvi'o will use\\nhmmscan for typical HMM operations like those in anvi-\\nrun-hmms (as these tend to scan a very large number of\\ngenes against a relatively small number of HMMs), but\\nif you are using this program to scan a very large\\nnumber of HMMs, hmmsearch might be a better choice for\\nperformance. For this reason, hmmsearch is the default\\nin operations like anvi-run-pfams and anvi-run-kegg-\\nkofams. See this article for a discussion on the\\nperformance of these two programs:\\nhttps://cryptogenomicon.org/2011/05/27/hmmscan-vs-\\nhmmsearch-speed-the-numerology/ (default: None)"
    keep_all_hits: "If you use this flag, anvi'o will not get rid of any\\nraw HMM hits, even those that are below the score\\nthreshold. (default: False)"
    log_bit_scores: "Use this flag to generate a tab-delimited text file\\ncontaining the bit scores of every KOfam hit that is\\nput in the contigs database. (default: False)"
    just_do_it: "Don't bother me with questions or warnings, just do\\nit. (default: False)"
  }
  output {
    File out_stdout = stdout()
  }
}