from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Boolean, Directory

Configmanta_Py_V0_1_0 = CommandToolBuilder(tool="configManta.py", base_command=["configManta.py"], inputs=[ToolInput(tag="in_config", input_type=File(optional=True), prefix="--config", doc=InputDocumentation(doc="provide a configuration file to override defaults in\nglobal config file (/usr/local/share/manta-1.6.0-0/bin\n/configManta.py.ini)")), ToolInput(tag="in_all_help", input_type=Boolean(optional=True), prefix="--allHelp", doc=InputDocumentation(doc="show all extended/hidden options")), ToolInput(tag="in_normal_bam", input_type=File(optional=True), prefix="--normalBam", doc=InputDocumentation(doc="Normal sample BAM or CRAM file. May be specified more\nthan once, multiple inputs will be treated as each BAM\nfile representing a different sample. [optional] (no\ndefault)")), ToolInput(tag="in_tumour_bam", input_type=File(optional=True), prefix="--tumourBam", doc=InputDocumentation(doc="Tumor sample BAM or CRAM file. Only up to one tumor\nbam file accepted. [optional] (no default)")), ToolInput(tag="in_exo_me", input_type=Boolean(optional=True), prefix="--exome", doc=InputDocumentation(doc="Set options for WES input: turn off depth filters")), ToolInput(tag="in_rna", input_type=Boolean(optional=True), prefix="--rna", doc=InputDocumentation(doc="Set options for RNA-Seq input. Must specify exactly\none bam input file")), ToolInput(tag="in_un_stranded_rna", input_type=Boolean(optional=True), prefix="--unstrandedRNA", doc=InputDocumentation(doc="Set if RNA-Seq input is unstranded: Allows splice-\njunctions on either strand")), ToolInput(tag="in_reference_fast_a", input_type=File(optional=True), prefix="--referenceFasta", doc=InputDocumentation(doc="samtools-indexed reference fasta file [required]")), ToolInput(tag="in_run_dir", input_type=Directory(optional=True), prefix="--runDir", doc=InputDocumentation(doc="Name of directory to be created where all workflow\nscripts and output will be written. Each analysis\nrequires a separate directory. (default:\nMantaWorkflow)")), ToolInput(tag="in_call_regions", input_type=File(optional=True), prefix="--callRegions", doc=InputDocumentation(doc="Optionally provide a bgzip-compressed/tabix-indexed\nBED file containing the set of regions to call. No VCF\noutput will be provided outside of these regions. The\nfull genome will still be used to estimate statistics\nfrom the input (such as expected fragment size\ndistribution). Only one BED file may be specified.\n(default: call the entire genome)"))], outputs=[ToolOutput(tag="out_run_dir", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_run_dir", type_hint=File()), doc=OutputDocumentation(doc="Name of directory to be created where all workflow\nscripts and output will be written. Each analysis\nrequires a separate directory. (default:\nMantaWorkflow)"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Configmanta_Py_V0_1_0().translate("wdl", allow_empty_container=True)

