from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, File, Int, String, Float

Samtools_Mpileup_V0_1_0 = CommandToolBuilder(tool="samtools_mpileup", base_command=["samtools", "mpileup"], inputs=[ToolInput(tag="in_assume_quality_illumina", input_type=Boolean(optional=True), prefix="-6", doc=InputDocumentation(doc="assume the quality is in the Illumina-1.3+ encoding")), ToolInput(tag="in_count_anomalous_pairs", input_type=Boolean(optional=True), prefix="-A", doc=InputDocumentation(doc="count anomalous read pairs")), ToolInput(tag="in_disable_baq_computation", input_type=Boolean(optional=True), prefix="-B", doc=InputDocumentation(doc="disable BAQ computation")), ToolInput(tag="in_list_input_bam", input_type=File(optional=True), prefix="-b", doc=InputDocumentation(doc="list of input BAM filenames, one per line [null]")), ToolInput(tag="in_parameter_adjusting_disable", input_type=Int(optional=True), prefix="-C", doc=InputDocumentation(doc="parameter for adjusting mapQ; 0 to disable [0]")), ToolInput(tag="in_max_perbam_depth", input_type=Int(optional=True), prefix="-d", doc=InputDocumentation(doc="max per-BAM depth to avoid excessive memory usage [250]")), ToolInput(tag="in_recalculate_extended_baq", input_type=Boolean(optional=True), prefix="-E", doc=InputDocumentation(doc="recalculate extended BAQ on the fly thus ignoring existing BQs")), ToolInput(tag="in_faidx_indexed_reference", input_type=File(optional=True), prefix="-f", doc=InputDocumentation(doc="faidx indexed reference sequence file [null]")), ToolInput(tag="in_exclude_read_groups", input_type=File(optional=True), prefix="-G", doc=InputDocumentation(doc="exclude read groups listed in FILE [null]")), ToolInput(tag="in_list_positions_chr", input_type=File(optional=True), prefix="-l", doc=InputDocumentation(doc="list of positions (chr pos) or regions (BED) [null]")), ToolInput(tag="in_cap_mapping_quality", input_type=Int(optional=True), prefix="-M", doc=InputDocumentation(doc="cap mapping quality at INT [60]")), ToolInput(tag="in_region_pileup_generated", input_type=String(optional=True), prefix="-r", doc=InputDocumentation(doc="region in which pileup is generated [null]")), ToolInput(tag="in_ignore_rg_tags", input_type=Boolean(optional=True), prefix="-R", doc=InputDocumentation(doc="ignore RG tags")), ToolInput(tag="in_skip_alignments_mapq", input_type=Int(optional=True), prefix="-q", doc=InputDocumentation(doc="skip alignments with mapQ smaller than INT [0]")), ToolInput(tag="in_skip_bases_smaller", input_type=Int(optional=True), prefix="-Q", doc=InputDocumentation(doc="skip bases with baseQ/BAQ smaller than INT [13]")), ToolInput(tag="in_rf", input_type=Int(optional=True), prefix="--rf", doc=InputDocumentation(doc="required flags: skip reads with mask bits unset []")), ToolInput(tag="in_ff", input_type=Int(optional=True), prefix="--ff", doc=InputDocumentation(doc="filter flags: skip reads with mask bits set []")), ToolInput(tag="in_output_persample_dp", input_type=Boolean(optional=True), prefix="-D", doc=InputDocumentation(doc="output per-sample DP in BCF (require -g/-u)")), ToolInput(tag="in_generate_bcf_likelihoods", input_type=Boolean(optional=True), prefix="-g", doc=InputDocumentation(doc="generate BCF output (genotype likelihoods)")), ToolInput(tag="in_output_base_positions", input_type=Boolean(optional=True), prefix="-O", doc=InputDocumentation(doc="output base positions on reads (disabled by -g/-u)")), ToolInput(tag="in_output_mapping_quality", input_type=Boolean(optional=True), prefix="-s", doc=InputDocumentation(doc="output mapping quality (disabled by -g/-u)")), ToolInput(tag="in_output_persample_strand", input_type=Boolean(optional=True), prefix="-S", doc=InputDocumentation(doc="output per-sample strand bias P-value in BCF (require -g/-u)")), ToolInput(tag="in_generate_uncompress_output", input_type=Boolean(optional=True), prefix="-u", doc=InputDocumentation(doc="generate uncompress BCF output")), ToolInput(tag="in_phredscaled_extension_seq", input_type=Int(optional=True), prefix="-e", doc=InputDocumentation(doc="Phred-scaled gap extension seq error probability [20]")), ToolInput(tag="in_minimum_fraction_gapped", input_type=Float(optional=True), prefix="-F", doc=InputDocumentation(doc="minimum fraction of gapped reads for candidates [0.002]")), ToolInput(tag="in_coefficient_homopolymer_errors", input_type=Int(optional=True), prefix="-h", doc=InputDocumentation(doc="coefficient for homopolymer errors [100]")), ToolInput(tag="in_perform_indel_calling", input_type=Boolean(optional=True), prefix="-I", doc=InputDocumentation(doc="do not perform indel calling")), ToolInput(tag="in_max_persample_depth", input_type=Int(optional=True), prefix="-L", doc=InputDocumentation(doc="max per-sample depth for INDEL calling [250]")), ToolInput(tag="in_minimum_gapped_reads", input_type=Int(optional=True), prefix="-m", doc=InputDocumentation(doc="minimum gapped reads for indel candidates [1]")), ToolInput(tag="in_phredscaled_gap_open", input_type=Int(optional=True), prefix="-o", doc=InputDocumentation(doc="Phred-scaled gap open sequencing error probability [40]")), ToolInput(tag="in_apply_m_persample", input_type=Boolean(optional=True), prefix="-p", doc=InputDocumentation(doc="apply -m and -F per-sample to increase sensitivity")), ToolInput(tag="in_comma_separated_list", input_type=String(optional=True), prefix="-P", doc=InputDocumentation(doc="comma separated list of platforms for indels [all]")), ToolInput(tag="in_in_one_dot_bam", input_type=Int(), position=0, doc=InputDocumentation(doc=""))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Samtools_Mpileup_V0_1_0().translate("wdl", allow_empty_container=True)

