from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Boolean, Int, Float

Exactsnp_V0_1_0 = CommandToolBuilder(tool="exactSNP", base_command=["exactSNP"], inputs=[ToolInput(tag="in_specify_name_input", input_type=File(optional=True), prefix="-i", doc=InputDocumentation(doc="Specify name of an input file including read mapping results. The")), ToolInput(tag="in_specify_name_sequences", input_type=File(optional=True), prefix="-g", doc=InputDocumentation(doc="Specify name of the file including all reference sequences. Only\none single FASTA format file should be provided.")), ToolInput(tag="in_specify_name_outputs", input_type=File(optional=True), prefix="-o", doc=InputDocumentation(doc="Specify name of the output file. This program outputs a VCF format\nfile that includes discovered SNPs.")), ToolInput(tag="in_provide_set_included", input_type=File(optional=True), prefix="-a", doc=InputDocumentation(doc="Provide a set of annotated SNPs (e.g. SNPs included in the dbSNP\ndatabase). The supplied file should be in VCF format (gzipped file\nis accepted). Providing known SNPs to the program should improve\nits SNP calling performance. Note that the provided SNPs may or\nmay not be called.")), ToolInput(tag="in_indicate_input_file", input_type=Boolean(optional=True), prefix="-b", doc=InputDocumentation(doc="Indicate the input file provided via -i is in BAM format.")), ToolInput(tag="in_specify_qvalue_cutoff", input_type=Int(optional=True), prefix="-Q", doc=InputDocumentation(doc="Specify the q-value cutoff for SNP calling at sequencing depth of\n50X. 12 by default. The corresponding p-value cutoff is 10^(-1*Q).\nNote that this program automatically adjusts the q-value cutoff\naccording to the sequencing depth at each chromosomal location.")), ToolInput(tag="in_specify_minimum_fraction", input_type=Float(optional=True), prefix="-f", doc=InputDocumentation(doc="Specify the minimum fraction of mis-matched bases a SNP-containing\nlocation must have. Its value must between 0 and 1. 0 by default.")), ToolInput(tag="in_specify_minimum_number_have", input_type=Int(optional=True), prefix="-n", doc=InputDocumentation(doc="Specify the minimum number of mis-matched bases a SNP-containing\nlocation must have. 1 by default.")), ToolInput(tag="in_specify_minimum_number_reads", input_type=Int(optional=True), prefix="-r", doc=InputDocumentation(doc="Specify the minimum number of mapped reads a SNP-containing\nlocation must have (ie. the minimum coverage). 1 by default.")), ToolInput(tag="in_specify_maximum_depth", input_type=Int(optional=True), prefix="-x", doc=InputDocumentation(doc="Specify the maximum depth a SNP location is allowed to have.\n1,000,000 reads by default. This option is useful for removing PCR\nartefacts.")), ToolInput(tag="in_specify_base_scores", input_type=Int(optional=True), prefix="-s", doc=InputDocumentation(doc="Specify the minimum base quality scores (Phred scores) read bases\nmust satisfy to be used for SNP calling. 13 by default. Read bases\nwith quality scores less than 13 will be excluded from the\nanalysis.")), ToolInput(tag="in_specify_number_bases", input_type=Int(optional=True), prefix="-t", doc=InputDocumentation(doc="Specify the number of bases trimmed off from each end of the read.\n3 by default.")), ToolInput(tag="in_specify_number_threads", input_type=Int(optional=True), prefix="-T", doc=InputDocumentation(doc="Specify the number of threads. 1 by default.")), ToolInput(tag="in_output_version_program", input_type=Boolean(optional=True), prefix="-v", doc=InputDocumentation(doc="output version of the program.")), ToolInput(tag="in_specify_path_save", input_type=File(optional=True), prefix="-C", doc=InputDocumentation(doc="Specify the path to save the temporary files."))], outputs=[ToolOutput(tag="out_specify_name_outputs", output_type=File(optional=True), selector=InputSelector(input_to_select="in_specify_name_outputs", type_hint=File()), doc=OutputDocumentation(doc="Specify name of the output file. This program outputs a VCF format\nfile that includes discovered SNPs."))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Exactsnp_V0_1_0().translate("wdl", allow_empty_container=True)

