from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, Int, File, String

Centrifuge_Build_Bin_V0_1_0 = CommandToolBuilder(tool="centrifuge_build_bin", base_command=["centrifuge-build-bin"], inputs=[ToolInput(tag="in_reference_sequences_given", input_type=Boolean(optional=True), prefix="-c", doc=InputDocumentation(doc="reference sequences given on cmd line (as\n<reference_in>)")), ToolInput(tag="in_no_auto", input_type=Boolean(optional=True), prefix="--noauto", doc=InputDocumentation(doc="disable automatic -p/--bmax/--dcv memory-fitting")), ToolInput(tag="in_bmax", input_type=Int(optional=True), prefix="--bmax", doc=InputDocumentation(doc="max bucket sz for blockwise suffix-array builder")), ToolInput(tag="in_bmax_divn", input_type=Int(optional=True), prefix="--bmaxdivn", doc=InputDocumentation(doc="max bucket sz as divisor of ref len (default: 4)")), ToolInput(tag="in_dcv", input_type=Int(optional=True), prefix="--dcv", doc=InputDocumentation(doc="diff-cover period for blockwise (default: 1024)")), ToolInput(tag="in_no_dc", input_type=Boolean(optional=True), prefix="--nodc", doc=InputDocumentation(doc="disable diff-cover (algorithm becomes quadratic)")), ToolInput(tag="in_no_ref", input_type=Boolean(optional=True), prefix="--noref", doc=InputDocumentation(doc="don't build .3/.4.bt2 (packed reference) portion")), ToolInput(tag="in_just_ref", input_type=Boolean(optional=True), prefix="--justref", doc=InputDocumentation(doc="just build .3/.4.bt2 (packed reference) portion")), ToolInput(tag="in_off_rate", input_type=Int(optional=True), prefix="--offrate", doc=InputDocumentation(doc="SA is sampled every 2^offRate BWT chars (default: 5)")), ToolInput(tag="in_f_tab_chars", input_type=Int(optional=True), prefix="--ftabchars", doc=InputDocumentation(doc="# of chars consumed in initial lookup (default: 10)")), ToolInput(tag="in_conversion_table", input_type=File(optional=True), prefix="--conversion-table", doc=InputDocumentation(doc="a table that converts any id to a taxonomy id")), ToolInput(tag="in_taxonomy_tree", input_type=Boolean(optional=True), prefix="--taxonomy-tree", doc=InputDocumentation(doc="<file name>  taxonomy tree")), ToolInput(tag="in_name_table", input_type=Boolean(optional=True), prefix="--name-table", doc=InputDocumentation(doc="<file name>  names corresponding to taxonomic IDs")), ToolInput(tag="in_size_table", input_type=Boolean(optional=True), prefix="--size-table", doc=InputDocumentation(doc="<file name>  table of contig (or genome) sizes")), ToolInput(tag="in_seed", input_type=Int(optional=True), prefix="--seed", doc=InputDocumentation(doc="seed for random number generator")), ToolInput(tag="in_quiet", input_type=Boolean(optional=True), prefix="--quiet", doc=InputDocumentation(doc="verbose output (for debugging)")), ToolInput(tag="in_threads", input_type=Int(optional=True), prefix="--threads", doc=InputDocumentation(doc="number of alignment threads to launch (1)")), ToolInput(tag="in_km_er_count", input_type=Int(optional=True), prefix="--kmer-count", doc=InputDocumentation(doc="k size for counting the number of distinct k-mer")), ToolInput(tag="in_reference_in", input_type=String(), position=0, doc=InputDocumentation(doc="comma-separated list of files with ref sequences")), ToolInput(tag="in_centrifuge_index_base", input_type=String(), position=1, doc=InputDocumentation(doc="write cf data to files with this dir/basename"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Centrifuge_Build_Bin_V0_1_0().translate("wdl", allow_empty_container=True)

