from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Int, Float, Boolean

Persim_V0_1_0 = CommandToolBuilder(tool="PERsim", base_command=["PERsim"], inputs=[ToolInput(tag="in_roi", input_type=File(optional=True), prefix="-roi", doc=InputDocumentation(doc="Target region BED file.")), ToolInput(tag="in_count", input_type=Int(optional=True), prefix="-count", doc=InputDocumentation(doc="Number of read pairs to generate.")), ToolInput(tag="in_out_one", input_type=File(optional=True), prefix="-out1", doc=InputDocumentation(doc="Forward reads output file in .FASTQ.GZ format.")), ToolInput(tag="in_out_two", input_type=File(optional=True), prefix="-out2", doc=InputDocumentation(doc="Reverse reads output file in .FASTQ.GZ format.")), ToolInput(tag="in_length", input_type=Int(optional=True), prefix="-length", doc=InputDocumentation(doc="Read length for forward/reverse reads.\nDefault value: '100'")), ToolInput(tag="in_ins_mean", input_type=Int(optional=True), prefix="-ins_mean", doc=InputDocumentation(doc="Library insert size mean value.\nDefault value: '200'")), ToolInput(tag="in_ins_stdev", input_type=Int(optional=True), prefix="-ins_stdev", doc=InputDocumentation(doc="Library insert size mean standard deviation.\nDefault value: '70'")), ToolInput(tag="in_error", input_type=Float(optional=True), prefix="-error", doc=InputDocumentation(doc="Base error probability (uniform distribution).\nDefault value: '0.01'")), ToolInput(tag="in_max_n", input_type=Int(optional=True), prefix="-max_n", doc=InputDocumentation(doc="Maximum number of N bases (from reference genome).\nDefault value: '5'")), ToolInput(tag="in_a_one", input_type=Int(optional=True), prefix="-a1", doc=InputDocumentation(doc="Forward read sequencing adapter sequence (for read-through).\nDefault value: 'AGATCGGAAGAGCACACGTCTGAACTCCAGTCACGAGTTA'")), ToolInput(tag="in_a_two", input_type=Int(optional=True), prefix="-a2", doc=InputDocumentation(doc="Reverse read sequencing adapter sequence (for read-through).\nDefault value: 'AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTC'")), ToolInput(tag="in_ref", input_type=File(optional=True), prefix="-ref", doc=InputDocumentation(doc="Reference genome FASTA file. If unset 'reference_genome' from the 'settings.ini' file is used.\nDefault value: ''")), ToolInput(tag="in_enable_verbose_debug", input_type=Boolean(optional=True), prefix="-v", doc=InputDocumentation(doc="Enable verbose debug output.\nDefault value: 'false'")), ToolInput(tag="in_changelog", input_type=Boolean(optional=True), prefix="--changelog", doc=InputDocumentation(doc="Prints changeloge and exits.")), ToolInput(tag="in_tdx", input_type=Boolean(optional=True), prefix="--tdx", doc=InputDocumentation(doc="Writes a Tool Definition Xml file. The file name is the application name with the suffix '.tdx'."))], outputs=[ToolOutput(tag="out_out_one", output_type=File(optional=True), selector=InputSelector(input_to_select="in_out_one", type_hint=File()), doc=OutputDocumentation(doc="Forward reads output file in .FASTQ.GZ format.")), ToolOutput(tag="out_out_two", output_type=File(optional=True), selector=InputSelector(input_to_select="in_out_two", type_hint=File()), doc=OutputDocumentation(doc="Reverse reads output file in .FASTQ.GZ format."))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Persim_V0_1_0().translate("wdl", allow_empty_container=True)

