from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean

Mirge3_0_V0_1_0 = CommandToolBuilder(tool="miRge3.0", base_command=["miRge3.0"], inputs=[ToolInput(tag="in_samples", input_type=Boolean(optional=True), prefix="--samples", doc=InputDocumentation(doc="list of one or more samples separated by comma or a file with list of samples separated by new line (accepts *.fastq, *.fastq.gz)")), ToolInput(tag="in_mir_db", input_type=Boolean(optional=True), prefix="--mir-DB", doc=InputDocumentation(doc="the reference database of miRNA. Options: miRBase and miRGeneDB (Default: miRBase)")), ToolInput(tag="in_libraries_path", input_type=Boolean(optional=True), prefix="--libraries-path", doc=InputDocumentation(doc="the path to miRge libraries")), ToolInput(tag="in_organism_name", input_type=Boolean(optional=True), prefix="--organism-name", doc=InputDocumentation(doc="the organism name can be human, mouse, fruitfly, nematode, rat or zebrafish")), ToolInput(tag="in_cr_threshold", input_type=Boolean(optional=True), prefix="--crThreshold", doc=InputDocumentation(doc="the threshold of the proportion of canonical reads for the miRNAs to retain. Range for ex (0 - 0.5), (Default: 0.1)")), ToolInput(tag="in_phred_six_four", input_type=Boolean(optional=True), prefix="--phred64", doc=InputDocumentation(doc="phred64 format (Default: 33)")), ToolInput(tag="in_spike_in", input_type=Boolean(optional=True), prefix="--spikeIn", doc=InputDocumentation(doc="switch to annotate spike-ins if spike-in bowtie index files are located at the path of bowtie's index files (Default: off)")), ToolInput(tag="in_isoform_entropy", input_type=Boolean(optional=True), prefix="--isoform-entropy", doc=InputDocumentation(doc="switch to calculate isomir entropy (default: off)")), ToolInput(tag="in_threads", input_type=Boolean(optional=True), prefix="--threads", doc=InputDocumentation(doc="the number of processors to use for trimming, qc, and alignment (Default: 1)")), ToolInput(tag="in_atoi", input_type=Boolean(optional=True), prefix="--AtoI", doc=InputDocumentation(doc="switch to calculate A to I editing (Default: off)")), ToolInput(tag="in_tcf_out", input_type=Boolean(optional=True), prefix="--tcf-out", doc=InputDocumentation(doc="switch to write trimmed and collapsed fasta file (Default: off)")), ToolInput(tag="in_gff_out", input_type=Boolean(optional=True), prefix="--gff-out", doc=InputDocumentation(doc="switch to output isomiR results in gff format (Default: off)")), ToolInput(tag="in_bam_out", input_type=Boolean(optional=True), prefix="--bam-out", doc=InputDocumentation(doc="switch to output results in bam format (Default: off)")), ToolInput(tag="in_trna_frag", input_type=Boolean(optional=True), prefix="--tRNA-frag", doc=InputDocumentation(doc="switch to analyze tRNA fragment and halves (Default: off)")), ToolInput(tag="in_outdir", input_type=Boolean(optional=True), prefix="--outDir", doc=InputDocumentation(doc="the directory of the outputs (Default: current directory)")), ToolInput(tag="in_diff_ex", input_type=Boolean(optional=True), prefix="--diffex", doc=InputDocumentation(doc="perform differential expression with DESeq2 (Default: off)")), ToolInput(tag="in_metadata", input_type=Boolean(optional=True), prefix="--metadata", doc=InputDocumentation(doc="the path to metadata file (Default: off, require '.csv' file format if -dex is opted)")), ToolInput(tag="in_quiet", input_type=Boolean(optional=True), prefix="--quiet", doc=InputDocumentation(doc="enable quiet/silent mode, only show warnings and errors (Default: off)")), ToolInput(tag="in_adapter", input_type=Boolean(optional=True), prefix="--adapter", doc=InputDocumentation(doc="Sequence of a 3' adapter. The adapter and subsequent bases are trimmed")), ToolInput(tag="in_front", input_type=Boolean(optional=True), prefix="--front", doc=InputDocumentation(doc="Sequence of a 5' adapter. The adapter and any preceding bases are trimmed")), ToolInput(tag="in_cut", input_type=Boolean(optional=True), prefix="--cut", doc=InputDocumentation(doc="Remove bases from each read. If LENGTH is positive, remove bases from the beginning. If LENGTH is negative, remove bases from the end")), ToolInput(tag="in_next_seq_trim", input_type=Boolean(optional=True), prefix="--nextseq-trim", doc=InputDocumentation(doc="NextSeq-specific quality trimming (each read). Trims also dark cycles appearing as high-quality G bases")), ToolInput(tag="in_quality_cut_off", input_type=Boolean(optional=True), prefix="--quality-cutoff", doc=InputDocumentation(doc="Trim low-quality bases from 5' and/or 3' ends of each read before adapter removal. If one value is given, only the 3' end is trimmed\nIf two comma-separated cutoffs are given, the 5' end is trimmed with the first cutoff, the 3' end with the second")), ToolInput(tag="in_length", input_type=Boolean(optional=True), prefix="--length", doc=InputDocumentation(doc="Shorten reads to LENGTH. Positive values remove bases at the end while negative ones remove bases at the beginning. This and the following\nmodifications are applied after adapter trimming")), ToolInput(tag="in_trim_n", input_type=Boolean(optional=True), prefix="--trim-n", doc=InputDocumentation(doc="Trim N's on ends of reads")), ToolInput(tag="in_minimum_length", input_type=Boolean(optional=True), prefix="--minimum-length", doc=InputDocumentation(doc="Discard reads shorter than LEN. (Default: 16)")), ToolInput(tag="in_uniq_mol_ids", input_type=Boolean(optional=True), prefix="--uniq-mol-ids", doc=InputDocumentation(doc="Removes PCR duplicates and trim UMI of length by specifying two comma-separated cutoffs as 5’ cutoff,3’ bp from both ends of the read. eg: 4,4 or 0,4")), ToolInput(tag="in_u_mid_ed_up", input_type=Boolean(optional=True), prefix="--umiDedup", doc=InputDocumentation(doc="Specifies argument to removes PCR duplicates (Default: False); if TRUE it will remove UMI and remove PCR duplicates otherwise it only remove UMI and keep the raw counts")), ToolInput(tag="in_qiagen_umi", input_type=Boolean(optional=True), prefix="--qiagenumi", doc=InputDocumentation(doc="Removes PCR duplicates of reads obtained from Qiagen platform (Default: Illumina; '-umi x,y ' Required)")), ToolInput(tag="in_novel_mirna", input_type=Boolean(optional=True), prefix="--novel-miRNA", doc=InputDocumentation(doc="include prediction of novel miRNAs")), ToolInput(tag="in_minlength", input_type=Boolean(optional=True), prefix="--minLength", doc=InputDocumentation(doc="the minimum length of the retained reads for novel miRNA detection (default: 16)")), ToolInput(tag="in_maxlength", input_type=Boolean(optional=True), prefix="--maxLength", doc=InputDocumentation(doc="the maximum length of the retained reads for novel miRNA detection (default: 25)")), ToolInput(tag="in_min_read_counts", input_type=Boolean(optional=True), prefix="--minReadCounts", doc=InputDocumentation(doc="the minimum read counts supporting novel miRNA detection (default: 2)")), ToolInput(tag="in_max_mapping_loci", input_type=Boolean(optional=True), prefix="--maxMappingLoci", doc=InputDocumentation(doc="the maximum number of mapping loci for the retained reads for novel miRNA detection (default: 3)")), ToolInput(tag="in_seed_length", input_type=Boolean(optional=True), prefix="--seedLength", doc=InputDocumentation(doc="the seed length when invoking Bowtie for novel miRNA detection (default: 25)")), ToolInput(tag="in_overlap_len_cut_off", input_type=Boolean(optional=True), prefix="--overlapLenCutoff", doc=InputDocumentation(doc="the length of overlapped seqence when joining reads into longer sequences based on the coordinate\non the genome for novel miRNA detection (default: 14)")), ToolInput(tag="in_cluster_length", input_type=Boolean(optional=True), prefix="--clusterLength", doc=InputDocumentation(doc="the maximum length of the clustered sequences for novel miRNA detection (default: 30)")), ToolInput(tag="in_bowtie_path", input_type=Boolean(optional=True), prefix="--bowtie-path", doc=InputDocumentation(doc="the path to system's directory containing bowtie binary")), ToolInput(tag="in_sam_tools_path", input_type=Boolean(optional=True), prefix="--samtools-path", doc=InputDocumentation(doc="the path to system's directory containing samtools binary")), ToolInput(tag="in_rna_fold_path", input_type=Boolean(optional=True), prefix="--RNAfold-path", doc=InputDocumentation(doc="the path to system's directory containing RNAfold binary"))], outputs=[], container="quay.io/biocontainers/mirge3:0.0.7--pyh3252c3a_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Mirge3_0_V0_1_0().translate("wdl")

