from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Directory, String, File, Boolean, Int

Gmap_Build_V0_1_0 = CommandToolBuilder(tool="gmap_build", base_command=["gmap_build"], inputs=[ToolInput(tag="in_dir", input_type=Directory(optional=True), prefix="--dir", doc=InputDocumentation(doc="Destination directory for installation (defaults to gmapdb\ndirectory specified at configure time)")), ToolInput(tag="in_genome_db", input_type=String(optional=True), prefix="--genomedb", doc=InputDocumentation(doc="Genome name (required)")), ToolInput(tag="in_names", input_type=File(optional=True), prefix="--names", doc=InputDocumentation(doc="Substitute names for contigs, provided in a file.\nThe file can have two formats:\n1.  A file with one column per line, with each line\ncorresponding to a FASTA file, in the order given to\ngmap_build.  The chromosome name for each FASTA file will\nbe replaced with the desired chromosome name in the file.\nEvery chromosome in the FASTA must have a corresponding line\nin the file.  This is useful if you want to rename chromosomes\nwith a systematic numbering pattern.\n2.  A file with two columns per line, separated by white\nspace.  In each line, the original FASTA chromosome name\nshould be in column 1 and the desired chromosome name\nwill be in column 2.\nThe meaning of file format 2 depends on whether\n--limit-to-names is specified.  If so, the genome build will\nbe limited to those chromosomes in this file.  Otherwise,\nall chromosomes in the FASTA file will be included,\nbut only those chromosomes in this file will be re-named, which\nprovides an easy way to change just a few chromosome names.\nThis file can be combined with the --sort=names option, in\nwhich the order of chromosomes is that given in the file.  In\nthis case, every chromosome must be listed in the file, and\nfor chromosome names that should not be changed, column 2 can\nbe blank (or the same as column 1).  The option of a blank\ncolumn 2 is allowed only when specifying --sort=names,\nbecause otherwise, the program cannot distinguish between a\n1-column and 2-column names file.")), ToolInput(tag="in_limit_to_names", input_type=Boolean(optional=True), prefix="--limit-to-names", doc=InputDocumentation(doc="Determines whether to limit the genome build to the lines listed\nin the --names file.  You can limit a genome build to certain\nchromosomes with this option, plus a --names file that either\nrenames chromosomes, or lists the same names in both columns for\nthe desired chromosomes.")), ToolInput(tag="in_km_er", input_type=Int(optional=True), prefix="--kmer", doc=InputDocumentation(doc="k-mer value for genomic index (allowed: 15 or less, default is 15)")), ToolInput(tag="in_sampling_interval_genomoe", input_type=Int(optional=True), prefix="-q", doc=InputDocumentation(doc="sampling interval for genomoe (allowed: 1-3, default 3)")), ToolInput(tag="in_sort", input_type=File(optional=True), prefix="--sort", doc=InputDocumentation(doc="Sort chromosomes using given method:\nnone - use chromosomes as found in FASTA file(s) (default)\nalpha - sort chromosomes alphabetically (chr10 before chr 1)\nnumeric-alpha - chr1, chr1U, chr2, chrM, chrU, chrX, chrY\nchrom - chr1, chr2, chrM, chrX, chrY, chr1U, chrU\nnames - sort chromosomes based on file provided to --names flag")), ToolInput(tag="in_gunzip", input_type=Boolean(optional=True), prefix="--gunzip", doc=InputDocumentation(doc="Files are gzipped, so need to gunzip each file first")), ToolInput(tag="in_fast_a_pipe", input_type=String(optional=True), prefix="--fasta-pipe", doc=InputDocumentation(doc="Interpret argument as a command, instead of a list of FASTA files")), ToolInput(tag="in_fast_q", input_type=Boolean(optional=True), prefix="--fastq", doc=InputDocumentation(doc="Files are in FASTQ format")), ToolInput(tag="in_rev_comp", input_type=Boolean(optional=True), prefix="--revcomp", doc=InputDocumentation(doc="Reverse complement all contigs")), ToolInput(tag="in_sleep_many_seconds", input_type=Int(optional=True), prefix="-w", doc=InputDocumentation(doc="Wait (sleep) this many seconds after each step (default 2)")), ToolInput(tag="in_circular", input_type=File(optional=True), prefix="--circular", doc=InputDocumentation(doc="Circular chromosomes (either a list of chromosomes separated\nby a comma, or a filename containing circular chromosomes,\none per line).  If you use the --names feature, then you\nshould use the substitute name of the chromosome, not the\noriginal name, for this option.  (NOTE: This behavior is different\nfrom previous versions, and starts with version 2020-10-20.)")), ToolInput(tag="in_alt_scaffold", input_type=File(optional=True), prefix="--altscaffold", doc=InputDocumentation(doc="File with alt scaffold info, listing alternate scaffolds,\none per line, tab-delimited, with the following fields:\n(1) alt_scaf_acc, (2) parent_name, (3) orientation,\n(4) alt_scaf_start, (5) alt_scaf_stop, (6) parent_start, (7) parent_end.")), ToolInput(tag="in_n_messages", input_type=Int(optional=True), prefix="--nmessages", doc=InputDocumentation(doc="Maximum number of messages (warnings, contig reports) to report (default 50)")), ToolInput(tag="in_md_flag", input_type=File(optional=True), prefix="--mdflag", doc=InputDocumentation(doc="Use MD file from NCBI for mapping contigs to\nchromosomal coordinates")), ToolInput(tag="in_contigs_are_mapped", input_type=Boolean(optional=True), prefix="--contigs-are-mapped", doc=InputDocumentation(doc="Find a chromosomal region in each FASTA header line.\nUseful for contigs that have been mapped\nto chromosomal coordinates.  Ignored if the --mdflag is provided.")), ToolInput(tag="in_transcriptome_db", input_type=String(optional=True), prefix="--transcriptomedb", doc=InputDocumentation(doc="Transcriptome name")), ToolInput(tag="in_transcripts", input_type=File(optional=True), prefix="--transcripts", doc=InputDocumentation(doc="FASTA file containing transcripts (required if specifying")), ToolInput(tag="in_n_threads", input_type=Int(optional=True), prefix="--nthreads", doc=InputDocumentation(doc="Number of threads for GMAP alignment of transcripts to genome\n(default 8)\n")), ToolInput(tag="in_options_dot_dot_dot", input_type=String(optional=True), position=0, doc=InputDocumentation(doc="")), ToolInput(tag="in_genome_fast_a_files", input_type=String(), position=1, doc=InputDocumentation(doc=""))], outputs=[], container="quay.io/biocontainers/gmap:2021.02.22--pl526h2f06484_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Gmap_Build_V0_1_0().translate("wdl")

