from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, File, Directory, String

Mirtrace_V0_1_0 = CommandToolBuilder(tool="mirtrace", base_command=["mirtrace"], inputs=[ToolInput(tag="in_species", input_type=Boolean(optional=True), prefix="--species", doc=InputDocumentation(doc="Species (miRBase encoding). EXAMPLE: 'hsa' for Homo sapiens.\nTo list all codes, run 'miRTrace --list-species'.")), ToolInput(tag="in_config", input_type=Boolean(optional=True), prefix="--config", doc=InputDocumentation(doc="List of FASTQ files to process. This is a CSV (comma separated\nvalue) file, i.e. with one entry per row.\nEach row consists of the following columns\n(only the first is required):\nfilename,name-displayed-in-report,adapter,PHRED-ASCII-offset\nNOTE: the PHRED ASCII offset can typically be reliably\nauto-detected and is not necessary to specify.\nEXAMPLE CONFIG FILE:\npath/sample1.fastq,sample 1 (control),TGGAATTC\npath/sample2.fastq,sample 2 (+drug X),TGGAATTC")), ToolInput(tag="in_adapter", input_type=Boolean(optional=True), prefix="--adapter", doc=InputDocumentation(doc="<DNA sequence>. [DEFAULT: none].")), ToolInput(tag="in_protocol", input_type=Boolean(optional=True), prefix="--protocol", doc=InputDocumentation(doc="One of the following (read structure schematic in parens):\nillumina (miRNA--3'-adapter--index) [DEFAULT]\nqiaseq (miRNA--3'-adapter--UMI--3'-adapter--index)\nNOTE: Only the first (leftmost) 3' adapter should be specified.\ncats (NNN--miRNA--poly-A--3'-adapter--index)\nNOTE: It's not possible to specify an adapter for -p cats.\nnextflex (NNNN--miRNA--NNNN--3'-adapter--index)")), ToolInput(tag="in_output_dir", input_type=File(optional=True), prefix="--output-dir", doc=InputDocumentation(doc="Directory for output files. [DEFAULT: <file listing>.output]")), ToolInput(tag="in_force", input_type=Directory(optional=True), prefix="--force", doc=InputDocumentation(doc="Overwrite output directory if it exists.")), ToolInput(tag="in_enable_pipes", input_type=Boolean(optional=True), prefix="--enable-pipes", doc=InputDocumentation(doc="Enable support for named pipes (fifos) as input.\nNOTE: Requires a config file with PHRED and adapter\ngiven for each entry. Input must not be compressed.")), ToolInput(tag="in_write_fast_a", input_type=Directory(optional=True), prefix="--write-fasta", doc=InputDocumentation(doc="Write QC-passed reads and unknown reads (as defined\nin the RNA type plot) to the output folder.\nIdentical reads are collapsed. Entries are sorted by abundance.")), ToolInput(tag="in_un_collapse_fast_a", input_type=Boolean(optional=True), prefix="--uncollapse-fasta", doc=InputDocumentation(doc="Write one FASTA entry per original FASTQ entry.")), ToolInput(tag="in_title", input_type=Boolean(optional=True), prefix="--title", doc=InputDocumentation(doc="Set the report title.")), ToolInput(tag="in_comment", input_type=Boolean(optional=True), prefix="--comment", doc=InputDocumentation(doc="Add a comment to the generated report. Multiple\narguments can be given.")), ToolInput(tag="in_num_threads", input_type=Boolean(optional=True), prefix="--num-threads", doc=InputDocumentation(doc="Maximum number of processing threads to use.\n[DEFAULT: <number of virtual cores>]")), ToolInput(tag="in_verbosity_level", input_type=Boolean(optional=True), prefix="--verbosity-level", doc=InputDocumentation(doc="Level of detail for debug messages. [default: 1]")), ToolInput(tag="in_global_mem_reserve", input_type=Boolean(optional=True), prefix="--global-mem-reserve", doc=InputDocumentation(doc="Amount of Java memory reserved for\n'housekeeping' tasks (in MB).\nIncrease only if OutOfMemoryErrors are occurring.\nDecrease only if available system memory is very low.\n[Current value: 400 MB]")), ToolInput(tag="in_custom_db_folder", input_type=Boolean(optional=True), prefix="--custom-db-folder", doc=InputDocumentation(doc="Folder containing user-generated reference databases.")), ToolInput(tag="in_list_species", input_type=Boolean(optional=True), prefix="--list-species", doc=InputDocumentation(doc="List all species that miRTrace has reference databases for.")), ToolInput(tag="in_cite", input_type=Boolean(optional=True), prefix="--cite", doc=InputDocumentation(doc="Show information about how to cite our paper.")), ToolInput(tag="in_trace", input_type=String(), position=0, doc=InputDocumentation(doc="miRNA trace mode. Produces a clade report. --species is ignored.")), ToolInput(tag="in_qc", input_type=String(), position=1, doc=InputDocumentation(doc="Quality control mode (full set of reports). --species must be given."))], outputs=[ToolOutput(tag="out_output_dir", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output_dir", type_hint=File()), doc=OutputDocumentation(doc="Directory for output files. [DEFAULT: <file listing>.output]")), ToolOutput(tag="out_force", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_force", type_hint=File()), doc=OutputDocumentation(doc="Overwrite output directory if it exists.")), ToolOutput(tag="out_write_fast_a", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_write_fast_a", type_hint=File()), doc=OutputDocumentation(doc="Write QC-passed reads and unknown reads (as defined\nin the RNA type plot) to the output folder.\nIdentical reads are collapsed. Entries are sorted by abundance."))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Mirtrace_V0_1_0().translate("wdl", allow_empty_container=True)

