from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, String, File, Int, Float, Directory

Poppunk_V0_1_0 = CommandToolBuilder(tool="poppunk", base_command=["poppunk"], inputs=[ToolInput(tag="in_create_db", input_type=Boolean(optional=True), prefix="--create-db", doc=InputDocumentation(doc="Create pairwise distances database between reference")), ToolInput(tag="in_use_model", input_type=Boolean(optional=True), prefix="--use-model", doc=InputDocumentation(doc="Apply a fitted model to a reference database to\nrestore database files")), ToolInput(tag="in_ref_db", input_type=String(optional=True), prefix="--ref-db", doc=InputDocumentation(doc="Location of built reference database")), ToolInput(tag="in_r_files", input_type=File(optional=True), prefix="--r-files", doc=InputDocumentation(doc="File listing reference input assemblies")), ToolInput(tag="in_distances", input_type=String(optional=True), prefix="--distances", doc=InputDocumentation(doc="Prefix of input pickle of pre-calculated distances")), ToolInput(tag="in_external_clustering", input_type=File(optional=True), prefix="--external-clustering", doc=InputDocumentation(doc="File with cluster definitions or other labels\ngenerated with any other method.")), ToolInput(tag="in_output", input_type=String(optional=True), prefix="--output", doc=InputDocumentation(doc="Prefix for output files")), ToolInput(tag="in_plot_fit", input_type=Int(optional=True), prefix="--plot-fit", doc=InputDocumentation(doc="Create this many plots of some fits relating k-mer to\ncore/accessory distances [default = 0]")), ToolInput(tag="in_overwrite", input_type=Boolean(optional=True), prefix="--overwrite", doc=InputDocumentation(doc="Overwrite any existing database files")), ToolInput(tag="in_graph_weights", input_type=Boolean(optional=True), prefix="--graph-weights", doc=InputDocumentation(doc="Save within-strain Euclidean distances into the graph")), ToolInput(tag="in_min_k", input_type=Int(optional=True), prefix="--min-k", doc=InputDocumentation(doc="Minimum kmer length [default = 13]")), ToolInput(tag="in_max_k", input_type=Int(optional=True), prefix="--max-k", doc=InputDocumentation(doc="Maximum kmer length [default = 29]")), ToolInput(tag="in_k_step", input_type=Int(optional=True), prefix="--k-step", doc=InputDocumentation(doc="K-mer step size [default = 4]")), ToolInput(tag="in_sketch_size", input_type=Int(optional=True), prefix="--sketch-size", doc=InputDocumentation(doc="Kmer sketch size [default = 10000]")), ToolInput(tag="in_codon_phased", input_type=Boolean(optional=True), prefix="--codon-phased", doc=InputDocumentation(doc="Used codon phased seeds X--X--X [default = False]")), ToolInput(tag="in_min_km_er_count", input_type=Int(optional=True), prefix="--min-kmer-count", doc=InputDocumentation(doc="Minimum k-mer count when using reads as input [default\n= 0]")), ToolInput(tag="in_exact_count", input_type=Boolean(optional=True), prefix="--exact-count", doc=InputDocumentation(doc="Use the exact k-mer counter with reads [default = use\ncountmin counter]")), ToolInput(tag="in_strand_preserved", input_type=Boolean(optional=True), prefix="--strand-preserved", doc=InputDocumentation(doc="Treat input as being on the same strand, and ignore\nreverse complement k-mers [default = use canonical\nk-mers]")), ToolInput(tag="in_qc_filter", input_type=String(optional=True), prefix="--qc-filter", doc=InputDocumentation(doc="Behaviour following sequence QC step: 'stop'\n[default], 'prune' (analyse data passing QC), or\n'continue' (analyse all data)")), ToolInput(tag="in_retain_failures", input_type=Boolean(optional=True), prefix="--retain-failures", doc=InputDocumentation(doc="Retain sketches of genomes that do not pass QC filters\nin separate database [default = False]")), ToolInput(tag="in_max_a_dist", input_type=Int(optional=True), prefix="--max-a-dist", doc=InputDocumentation(doc="Maximum accessory distance to permit [default = 0.5]")), ToolInput(tag="in_length_sigma", input_type=Int(optional=True), prefix="--length-sigma", doc=InputDocumentation(doc="Number of standard deviations of length distribution\nbeyond which sequences will be excluded [default = 5]")), ToolInput(tag="in_length_range", input_type=Int(optional=True), prefix="--length-range", doc=InputDocumentation(doc="LENGTH_RANGE\nAllowed length range, outside of which sequences will\nbe excluded [two values needed - lower and upper\nbounds]")), ToolInput(tag="in_prop_n", input_type=Float(optional=True), prefix="--prop-n", doc=InputDocumentation(doc="Threshold ambiguous base proportion above which\nsequences will be excluded [default = 0.1]")), ToolInput(tag="in_upper_n", input_type=String(optional=True), prefix="--upper-n", doc=InputDocumentation(doc="Threshold ambiguous base count above which sequences\nwill be excluded")), ToolInput(tag="in_maximum_number_mixture", input_type=Int(optional=True), prefix="--K", doc=InputDocumentation(doc="Maximum number of mixture components [default = 2]")), ToolInput(tag="in_maximum_number_default", input_type=Int(optional=True), prefix="--D", doc=InputDocumentation(doc="Maximum number of clusters in DBSCAN fitting [default\n= 100]")), ToolInput(tag="in_min_cluster_prop", input_type=Int(optional=True), prefix="--min-cluster-prop", doc=InputDocumentation(doc="Minimum proportion of points in a cluster in DBSCAN\nfitting [default = 0.0001]")), ToolInput(tag="in_threshold", input_type=String(optional=True), prefix="--threshold", doc=InputDocumentation(doc="Cutoff if using --fit-model threshold")), ToolInput(tag="in_pos_shift", input_type=String(optional=True), prefix="--pos-shift", doc=InputDocumentation(doc="Maximum amount to move the boundary away from origin\n[default = to between-strain mean]")), ToolInput(tag="in_neg_shift", input_type=String(optional=True), prefix="--neg-shift", doc=InputDocumentation(doc="Maximum amount to move the boundary towards the origin\n[default = to within-strain mean]")), ToolInput(tag="in_manual_start", input_type=File(optional=True), prefix="--manual-start", doc=InputDocumentation(doc="A file containing information for a start point. See\ndocumentation for help.")), ToolInput(tag="in_indiv_refine", input_type=String(optional=True), prefix="--indiv-refine", doc=InputDocumentation(doc="Also run refinement for core and accessory\nindividually")), ToolInput(tag="in_no_local", input_type=Boolean(optional=True), prefix="--no-local", doc=InputDocumentation(doc="Do not perform the local optimization step (speed up\non very large datasets)")), ToolInput(tag="in_model_dir", input_type=Directory(optional=True), prefix="--model-dir", doc=InputDocumentation(doc="Directory containing model to use for assigning\nqueries to clusters [default = reference database\ndirectory]")), ToolInput(tag="in_core_only", input_type=String(optional=True), prefix="--core-only", doc=InputDocumentation(doc="Save the core distance fit (with")), ToolInput(tag="in_ranks", input_type=Int(optional=True), prefix="--ranks", doc=InputDocumentation(doc="Comma separated list of ranks used in lineage\nclustering [default = 1,2,3]")), ToolInput(tag="in_use_accessory", input_type=Boolean(optional=True), prefix="--use-accessory", doc=InputDocumentation(doc="Use accessory distances for lineage definitions\n[default = use core distances]")), ToolInput(tag="in_threads", input_type=Int(optional=True), prefix="--threads", doc=InputDocumentation(doc="Number of threads to use [default = 1]")), ToolInput(tag="in_gpu_sketch", input_type=Boolean(optional=True), prefix="--gpu-sketch", doc=InputDocumentation(doc="Use a GPU when calculating sketches (read data only)\n[default = False]")), ToolInput(tag="in_gpu_dist", input_type=Boolean(optional=True), prefix="--gpu-dist", doc=InputDocumentation(doc="Use a GPU when calculating distances [default = False]")), ToolInput(tag="in_deviceid", input_type=Int(optional=True), prefix="--deviceid", doc=InputDocumentation(doc="CUDA device ID, if using GPU [default = 0]")), ToolInput(tag="in_sequences", input_type=String(), position=0, doc=InputDocumentation(doc="--fit-model {bgmm,dbscan,refine,lineage,threshold}"))], outputs=[], container="quay.io/biocontainers/poppunk:2.3.0--py_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Poppunk_V0_1_0().translate("wdl")

