from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, Int, File, String

Mccortex63_V0_1_0 = CommandToolBuilder(tool="mccortex63", base_command=["mccortex63"], inputs=[ToolInput(tag="in_quiet", input_type=Boolean(optional=True), prefix="--quiet", doc=InputDocumentation(doc="Silence status output normally printed to STDERR")), ToolInput(tag="in_force", input_type=Boolean(optional=True), prefix="--force", doc=InputDocumentation(doc="Overwrite output files if they already exist")), ToolInput(tag="in_memory", input_type=Int(optional=True), prefix="--memory", doc=InputDocumentation(doc="Memory e.g. 1GB [default: 1GB]")), ToolInput(tag="in_nk_mers", input_type=Int(optional=True), prefix="--nkmers", doc=InputDocumentation(doc="Hash entries [default: 4M, ~4 million]")), ToolInput(tag="in_threads", input_type=Int(optional=True), prefix="--threads", doc=InputDocumentation(doc="Limit on proccessing threads [default: 2]")), ToolInput(tag="in_out", input_type=File(optional=True), prefix="--out", doc=InputDocumentation(doc="Output file")), ToolInput(tag="in_paths", input_type=File(optional=True), prefix="--paths", doc=InputDocumentation(doc="Links file to load (can specify multiple times)")), ToolInput(tag="in_breakpoints", input_type=String(), position=0, doc=InputDocumentation(doc="use a trusted assembled genome to call large events")), ToolInput(tag="in_bubbles", input_type=String(), position=1, doc=InputDocumentation(doc="find bubbles in graph which are potential variants")), ToolInput(tag="in_build", input_type=String(), position=2, doc=InputDocumentation(doc="construct cortex graph from FASTA/FASTQ/BAM")), ToolInput(tag="in_calls_two_vcf", input_type=Int(), position=3, doc=InputDocumentation(doc="convert bubble/breakpoint calls to VCF")), ToolInput(tag="in_check", input_type=String(), position=4, doc=InputDocumentation(doc="load and check graph (.ctx) and path (.ctp) files")), ToolInput(tag="in_clean", input_type=String(), position=5, doc=InputDocumentation(doc="clean errors from a graph")), ToolInput(tag="in_contigs", input_type=String(), position=6, doc=InputDocumentation(doc="assemble contigs for a sample")), ToolInput(tag="in_correct", input_type=String(), position=7, doc=InputDocumentation(doc="error correct reads")), ToolInput(tag="in_coverage", input_type=String(), position=8, doc=InputDocumentation(doc="print contig coverage")), ToolInput(tag="in_dist", input_type=String(), position=9, doc=InputDocumentation(doc="make colour kmer distance matrix")), ToolInput(tag="in_index", input_type=String(), position=10, doc=InputDocumentation(doc="index a sorted cortex graph file")), ToolInput(tag="in_infer_edges", input_type=String(), position=11, doc=InputDocumentation(doc="infer graph edges between kmers before calling `thread`")), ToolInput(tag="in_join", input_type=String(), position=12, doc=InputDocumentation(doc="combine graphs, filter graph intersections")), ToolInput(tag="in_links", input_type=String(), position=13, doc=InputDocumentation(doc="clean and plot link files (.ctp)")), ToolInput(tag="in_p_join", input_type=String(), position=14, doc=InputDocumentation(doc="merge link files (.ctp)")), ToolInput(tag="in_pop_bubbles", input_type=String(), position=15, doc=InputDocumentation(doc="pop bubbles in the population graph")), ToolInput(tag="in_p_view", input_type=String(), position=16, doc=InputDocumentation(doc="text view of a cortex link file (.ctp)")), ToolInput(tag="in_reads", input_type=String(), position=17, doc=InputDocumentation(doc="filter reads against a graph")), ToolInput(tag="in_rm_substr", input_type=String(), position=18, doc=InputDocumentation(doc="reduce set of strings to remove substrings")), ToolInput(tag="in_server", input_type=String(), position=19, doc=InputDocumentation(doc="interactively query the graph")), ToolInput(tag="in_sort", input_type=String(), position=20, doc=InputDocumentation(doc="sort the kmers in a graph file")), ToolInput(tag="in_subgraph", input_type=String(), position=21, doc=InputDocumentation(doc="filter a subgraph using seed kmers")), ToolInput(tag="in_thread", input_type=String(), position=22, doc=InputDocumentation(doc="thread reads through cleaned graph to make links")), ToolInput(tag="in_uniq_km_ers", input_type=String(), position=23, doc=InputDocumentation(doc="generate random unique kmers")), ToolInput(tag="in_unit_igs", input_type=String(), position=24, doc=InputDocumentation(doc="pull out unitigs in FASTA, DOT or GFA format")), ToolInput(tag="in_vcf_cov", input_type=String(), position=25, doc=InputDocumentation(doc="coverage of a VCF against cortex graphs")), ToolInput(tag="in_vcf_geno", input_type=String(), position=26, doc=InputDocumentation(doc="genotype a VCF after running vcfcov")), ToolInput(tag="in_view", input_type=String(), position=27, doc=InputDocumentation(doc="text view of a cortex graph file (.ctx)"))], outputs=[ToolOutput(tag="out_out", output_type=File(optional=True), selector=InputSelector(input_to_select="in_out", type_hint=File()), doc=OutputDocumentation(doc="Output file"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Mccortex63_V0_1_0().translate("wdl", allow_empty_container=True)

