from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, String, Int, Boolean

Dumpstr_V0_1_0 = CommandToolBuilder(tool="dumpSTR", base_command=["dumpSTR"], inputs=[ToolInput(tag="in_vcf", input_type=File(optional=True), prefix="--vcf", doc=InputDocumentation(doc="Input STR VCF file")), ToolInput(tag="in_out", input_type=String(optional=True), prefix="--out", doc=InputDocumentation(doc="Prefix for output files")), ToolInput(tag="in_vcf_type", input_type=String(optional=True), prefix="--vcftype", doc=InputDocumentation(doc="Options=['gangstr', 'advntr', 'hipstr', 'eh',\n'popstr'] (default: auto)")), ToolInput(tag="in_min_locus_call_rate", input_type=Int(optional=True), prefix="--min-locus-callrate", doc=InputDocumentation(doc="Minimum locus call rate")), ToolInput(tag="in_min_locus_h_wep", input_type=Int(optional=True), prefix="--min-locus-hwep", doc=InputDocumentation(doc="Filter loci failing HWE at this p-value threshold")), ToolInput(tag="in_min_locus_het", input_type=Int(optional=True), prefix="--min-locus-het", doc=InputDocumentation(doc="Minimum locus heterozygosity")), ToolInput(tag="in_max_locus_het", input_type=Int(optional=True), prefix="--max-locus-het", doc=InputDocumentation(doc="Maximum locus heterozygosity")), ToolInput(tag="in_use_length", input_type=Boolean(optional=True), prefix="--use-length", doc=InputDocumentation(doc="Calculate per-locus stats (het, HWE) collapsing\nalleles by length (default: False)")), ToolInput(tag="in_filter_regions", input_type=String(optional=True), prefix="--filter-regions", doc=InputDocumentation(doc="Comma-separated list of BED files of regions to\nfilter. Must be bgzipped and tabix indexed")), ToolInput(tag="in_filter_regions_names", input_type=File(optional=True), prefix="--filter-regions-names", doc=InputDocumentation(doc="Comma-separated list of filter names for each BED\nfilter file")), ToolInput(tag="in_filter_h_run", input_type=Boolean(optional=True), prefix="--filter-hrun", doc=InputDocumentation(doc="Filter STRs with long homopolymer runs. (default:\nFalse)")), ToolInput(tag="in_drop_filtered", input_type=Boolean(optional=True), prefix="--drop-filtered", doc=InputDocumentation(doc="Drop filtered records from output (default: False)")), ToolInput(tag="in_hip_str_max_call_flank_in_del", input_type=Int(optional=True), prefix="--hipstr-max-call-flank-indel", doc=InputDocumentation(doc="Maximum call flank indel rate")), ToolInput(tag="in_hip_str_max_call_stutter", input_type=Int(optional=True), prefix="--hipstr-max-call-stutter", doc=InputDocumentation(doc="Maximum call stutter rate")), ToolInput(tag="in_hip_str_min_supp_reads", input_type=Int(optional=True), prefix="--hipstr-min-supp-reads", doc=InputDocumentation(doc="Minimum supporting reads for each allele")), ToolInput(tag="in_hip_str_min_call_dp", input_type=Int(optional=True), prefix="--hipstr-min-call-DP", doc=InputDocumentation(doc="Minimum call coverage")), ToolInput(tag="in_hip_str_max_call_dp", input_type=Int(optional=True), prefix="--hipstr-max-call-DP", doc=InputDocumentation(doc="Maximum call coverage")), ToolInput(tag="in_hip_str_min_call_q", input_type=Int(optional=True), prefix="--hipstr-min-call-Q", doc=InputDocumentation(doc="Minimum call quality score")), ToolInput(tag="in_gang_str_min_call_dp", input_type=Int(optional=True), prefix="--gangstr-min-call-DP", doc=InputDocumentation(doc="Minimum call coverage")), ToolInput(tag="in_gang_str_max_call_dp", input_type=Int(optional=True), prefix="--gangstr-max-call-DP", doc=InputDocumentation(doc="Maximum call coverage")), ToolInput(tag="in_gang_str_min_call_q", input_type=Int(optional=True), prefix="--gangstr-min-call-Q", doc=InputDocumentation(doc="Minimum call quality score")), ToolInput(tag="in_gang_str_expansion_prob_het", input_type=String(optional=True), prefix="--gangstr-expansion-prob-het", doc=InputDocumentation(doc="Expansion prob-value threshold. Filters calls with\nprobability of heterozygous expansion less than this")), ToolInput(tag="in_gang_str_expansion_prob_hom", input_type=String(optional=True), prefix="--gangstr-expansion-prob-hom", doc=InputDocumentation(doc="Expansion prob-value threshold. Filters calls with\nprobability of homozygous expansion less than this")), ToolInput(tag="in_gang_str_expansion_prob_total", input_type=String(optional=True), prefix="--gangstr-expansion-prob-total", doc=InputDocumentation(doc="Expansion prob-value threshold. Filters calls with\nprobability of total expansion less than this")), ToolInput(tag="in_gang_str_filter_span_only", input_type=Boolean(optional=True), prefix="--gangstr-filter-span-only", doc=InputDocumentation(doc="Filter out all calls that only have spanning read\nsupport (default: False)")), ToolInput(tag="in_gang_str_filter_span_bound_only", input_type=Boolean(optional=True), prefix="--gangstr-filter-spanbound-only", doc=InputDocumentation(doc="Filter out all reads except spanning and bounding\n(default: False)")), ToolInput(tag="in_gang_str_filter_bad_ci", input_type=Boolean(optional=True), prefix="--gangstr-filter-badCI", doc=InputDocumentation(doc="Filter regions where the ML estimate is not in the CI\n(default: False)")), ToolInput(tag="in_gang_str_require_support", input_type=String(optional=True), prefix="--gangstr-require-support", doc=InputDocumentation(doc="Require each allele call to have at least n supporting\nreads")), ToolInput(tag="in_gang_str_read_len", input_type=Int(optional=True), prefix="--gangstr-readlen", doc=InputDocumentation(doc="Read length used (bp). Required if using --require-\nsupport")), ToolInput(tag="in_ad_vntr_min_call_dp", input_type=Int(optional=True), prefix="--advntr-min-call-DP", doc=InputDocumentation(doc="Minimum call coverage")), ToolInput(tag="in_ad_vntr_max_call_dp", input_type=Int(optional=True), prefix="--advntr-max-call-DP", doc=InputDocumentation(doc="Maximum call coverage")), ToolInput(tag="in_ad_vntr_min_spanning", input_type=Int(optional=True), prefix="--advntr-min-spanning", doc=InputDocumentation(doc="Minimum spanning read count (SR field)")), ToolInput(tag="in_ad_vntr_min_flanking", input_type=Int(optional=True), prefix="--advntr-min-flanking", doc=InputDocumentation(doc="Minimum flanking read count (FR field)")), ToolInput(tag="in_ad_vntr_min_ml", input_type=Int(optional=True), prefix="--advntr-min-ML", doc=InputDocumentation(doc="Minimum value of maximum likelihood (ML field)")), ToolInput(tag="in_eh_min_adfl", input_type=Int(optional=True), prefix="--eh-min-ADFL", doc=InputDocumentation(doc="Minimum number of flanking reads consistent with the\nallele")), ToolInput(tag="in_eh_min_a_dir", input_type=Int(optional=True), prefix="--eh-min-ADIR", doc=InputDocumentation(doc="Minimum number of in-repeat reads consistent with the\nallele")), ToolInput(tag="in_eh_min_adsp", input_type=Int(optional=True), prefix="--eh-min-ADSP", doc=InputDocumentation(doc="Minimum number of spanning reads consistent with the\nallele")), ToolInput(tag="in_eh_min_call_lc", input_type=Int(optional=True), prefix="--eh-min-call-LC", doc=InputDocumentation(doc="Minimum call coverage")), ToolInput(tag="in_eh_max_call_lc", input_type=Int(optional=True), prefix="--eh-max-call-LC", doc=InputDocumentation(doc="Maximum call coverage")), ToolInput(tag="in_pop_str_min_call_dp", input_type=Int(optional=True), prefix="--popstr-min-call-DP", doc=InputDocumentation(doc="Minimum call coverage")), ToolInput(tag="in_pop_str_max_call_dp", input_type=Int(optional=True), prefix="--popstr-max-call-DP", doc=InputDocumentation(doc="Maximum call coverage")), ToolInput(tag="in_pop_str_require_support", input_type=String(optional=True), prefix="--popstr-require-support", doc=InputDocumentation(doc="Require each allele call to have at least n supporting\nreads")), ToolInput(tag="in_num_records", input_type=Int(optional=True), prefix="--num-records", doc=InputDocumentation(doc="Only process this many records")), ToolInput(tag="in_die_on_warning", input_type=Boolean(optional=True), prefix="--die-on-warning", doc=InputDocumentation(doc="Quit if a record can't be parsed (default: False)")), ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="--verbose", doc=InputDocumentation(doc="Print out extra info (default: False)")), ToolInput(tag="in_tool", input_type=String(), position=0, doc=InputDocumentation(doc="")), ToolInput(tag="in_for", input_type=String(), position=1, doc=InputDocumentation(doc="")), ToolInput(tag="in_filtering", input_type=String(), position=2, doc=InputDocumentation(doc="")), ToolInput(tag="in_and", input_type=String(), position=3, doc=InputDocumentation(doc="")), ToolInput(tag="in_qc", input_type=String(), position=4, doc=InputDocumentation(doc="")), ToolInput(tag="in_of", input_type=String(), position=5, doc=InputDocumentation(doc="")), ToolInput(tag="in_tr", input_type=String(), position=6, doc=InputDocumentation(doc="")), ToolInput(tag="in_genotypes", input_type=String(), position=7, doc=InputDocumentation(doc=""))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Dumpstr_V0_1_0().translate("wdl", allow_empty_container=True)

