from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean

Tutorial_Eqtlbma_R_V0_1_0 = CommandToolBuilder(tool="tutorial_eqtlbma.R", base_command=["tutorial_eqtlbma.R"], inputs=[ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="--verbose", doc=InputDocumentation(doc="verbosity level (0/default=1/2)")), ToolInput(tag="in_pkg", input_type=Boolean(optional=True), prefix="--pkg", doc=InputDocumentation(doc="absolute path to the package")), ToolInput(tag="in_nsb_grps", input_type=Boolean(optional=True), prefix="--nsbgrps", doc=InputDocumentation(doc="nb of tissues (default=3)")), ToolInput(tag="in_ninds", input_type=Boolean(optional=True), prefix="--ninds", doc=InputDocumentation(doc="nb of individuals per subgroup (default=200)\nindividuals are diploid\na single number, say 200, means same individuals in all subgroups\nseveral numbers, say 200-150-200, means different individuals between subgroups (must agree with --nsbgrps)")), ToolInput(tag="in_n_genes", input_type=Boolean(optional=True), prefix="--ngenes", doc=InputDocumentation(doc="nb of genes (default=1000)")), ToolInput(tag="in_nc_hrs", input_type=Boolean(optional=True), prefix="--nchrs", doc=InputDocumentation(doc="nb of chromosome(s) (default=1)")), ToolInput(tag="in_agl", input_type=Boolean(optional=True), prefix="--agl", doc=InputDocumentation(doc="average gene length (default=10000)")), ToolInput(tag="in_a_il", input_type=Boolean(optional=True), prefix="--ail", doc=InputDocumentation(doc="average intergenic length (default=50000)")), ToolInput(tag="in_anchor", input_type=Boolean(optional=True), prefix="--anchor", doc=InputDocumentation(doc="anchor for cis region (default=TSS/TSS+TES)")), ToolInput(tag="in_cr_five", input_type=Boolean(optional=True), prefix="--cr5", doc=InputDocumentation(doc="radius of cis region in 5' (default=1000)")), ToolInput(tag="in_cr_three", input_type=Boolean(optional=True), prefix="--cr3", doc=InputDocumentation(doc="radius of cis region in 3' (default=1000)")), ToolInput(tag="in_fsg", input_type=Boolean(optional=True), prefix="--fsg", doc=InputDocumentation(doc="fixed nb of cis SNPs per gene (or use --asg)")), ToolInput(tag="in_as_g", input_type=Boolean(optional=True), prefix="--asg", doc=InputDocumentation(doc="average nb of cis SNPs per gene (default=50)")), ToolInput(tag="in_maf", input_type=Boolean(optional=True), prefix="--maf", doc=InputDocumentation(doc="minor allele frequency (default=0.3)")), ToolInput(tag="in_rare", input_type=Boolean(optional=True), prefix="--rare", doc=InputDocumentation(doc="proportion of SNPs with rare alleles (with MAF=0.02, default=0.0)")), ToolInput(tag="in_pi_zero", input_type=Boolean(optional=True), prefix="--pi0", doc=InputDocumentation(doc="prior proba for a gene to have no eQTL in any subgroup (default=0.3)")), ToolInput(tag="in_cover_r", input_type=Boolean(optional=True), prefix="--coverr", doc=InputDocumentation(doc="error covariance between subgroups (default=1)\n0: the SxS covariance matrix is diagonal (usually the case if different individuals between subgroups), same for all genes\n1: the SxS covariance matrix is unconstrained (usually the case if same individuals in all subgroups), same for all genes")), ToolInput(tag="in_seed", input_type=Boolean(optional=True), prefix="--seed", doc=InputDocumentation(doc="seed for the RNG (default=1859)")), ToolInput(tag="in_dir", input_type=Boolean(optional=True), prefix="--dir", doc=InputDocumentation(doc="directory in which files are written (current by default)")), ToolInput(tag="in_n_cores", input_type=Boolean(optional=True), prefix="--ncores", doc=InputDocumentation(doc="nb of cores to run in parallel (default=1)"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Tutorial_Eqtlbma_R_V0_1_0().translate("wdl", allow_empty_container=True)

