from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, String

Bp_Load_Gff_Pl_V0_1_0 = CommandToolBuilder(tool="bp_load_gff.pl", base_command=["bp_load_gff.pl"], inputs=[ToolInput(tag="in_dsn", input_type=Boolean(optional=True), prefix="--dsn", doc=InputDocumentation(doc="'dbi:mysql:database=dmel_r5_1;host=myhost;port=myport'\ndna1.fa dna2.fa features1.gff features2.gff ...")), ToolInput(tag="in_of__database", input_type=String(optional=True), prefix="-d", doc=InputDocumentation(doc="of --database.")), ToolInput(tag="in_adaptor", input_type=String(optional=True), prefix="--adaptor", doc=InputDocumentation(doc="Schema adaptor (default dbi::mysqlopt)")), ToolInput(tag="in_user", input_type=Boolean(optional=True), prefix="--user", doc=InputDocumentation(doc="<user>      Username for mysql authentication")), ToolInput(tag="in_pass", input_type=Boolean(optional=True), prefix="--pass", doc=InputDocumentation(doc="<password>  Password for mysql authentication")), ToolInput(tag="in_fast_a", input_type=Boolean(optional=True), prefix="--fasta", doc=InputDocumentation(doc="<path>      Fasta file or directory containing fasta files for the DNA")), ToolInput(tag="in_create", input_type=Boolean(optional=True), prefix="--create", doc=InputDocumentation(doc="Force creation and initialization of database")), ToolInput(tag="in_max_feature", input_type=Boolean(optional=True), prefix="--maxfeature", doc=InputDocumentation(doc="Set the value of the maximum feature size (default 100 Mb; must be a power of 10)")), ToolInput(tag="in_group", input_type=Boolean(optional=True), prefix="--group", doc=InputDocumentation(doc="A list of one or more tag names (comma or space separated)\nto be used for grouping in the 9th column.")), ToolInput(tag="in_upgrade", input_type=Boolean(optional=True), prefix="--upgrade", doc=InputDocumentation(doc="Upgrade existing database to current schema")), ToolInput(tag="in_gff_three_munge", input_type=Boolean(optional=True), prefix="--gff3_munge", doc=InputDocumentation(doc="Activate GFF3 name munging (see Bio::DB::GFF)")), ToolInput(tag="in_quiet", input_type=Boolean(optional=True), prefix="--quiet", doc=InputDocumentation(doc="No progress reports")), ToolInput(tag="in_summary", input_type=Boolean(optional=True), prefix="--summary", doc=InputDocumentation(doc="Generate summary statistics for drawing coverage histograms.\nThis can be run on a previously loaded database or during\nthe load.")), ToolInput(tag="in_notes", input_type=String(), position=0, doc=InputDocumentation(doc="If the filename is given as '-' then the input is taken from standard\ninput. Compressed files (.gz, .Z, .bz2) are automatically uncompressed.\nFASTA format files are distinguished from GFF files by their filename\nextensions. Files ending in .fa, .fasta, .fast, .seq, .dna and their\nuppercase variants are treated as FASTA files. Everything else is\ntreated as a GFF file. If you wish to load -fasta files from STDIN, then\nuse the -f command-line swith with an argument of '-', as in\ngunzip my_data.fa.gz | bp_fast_load_gff.pl -d test -f -\nOn the first load of a database, you will see a number of 'unknown\ntable' errors. This is normal.\nAbout maxfeature: the default value is 100,000,000 bases. If you have\nfeatures that are close to or greater that 100Mb in length, then the\nvalue of maxfeature should be increased to 1,000,000,000, or another\npower of 10."))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Bp_Load_Gff_Pl_V0_1_0().translate("wdl", allow_empty_container=True)

