from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, String, Directory, Boolean, Array, Int

Agfusion_Batch_V0_1_0 = CommandToolBuilder(tool="agfusion_batch", base_command=["agfusion", "batch"], inputs=[ToolInput(tag="in_file", input_type=File(optional=True), prefix="--file", doc=InputDocumentation(doc="Output file from fusion-finding algorithm.")), ToolInput(tag="in_algorithm", input_type=String(optional=True), prefix="--algorithm", doc=InputDocumentation(doc="The fusion-finding algorithm. Can be one of the\nfollowing: bellerophontes, breakfusion, chimerascan,\nchimerscope, defuse, ericscript, fusioncatcher,\nfusionhunter, fusionmap, fusioninspector, infusion,\njaffa, mapsplice, starfusion, tophatfusion.")), ToolInput(tag="in_database", input_type=File(optional=True), prefix="--database", doc=InputDocumentation(doc="Path to the AGFusion database (e.g. --db\n/path/to/agfusion.homo_sapiens.87.db)")), ToolInput(tag="in_out", input_type=Directory(optional=True), prefix="--out", doc=InputDocumentation(doc="Directory to save results")), ToolInput(tag="in_noncanonical", input_type=Boolean(optional=True), prefix="--noncanonical", doc=InputDocumentation(doc="(Optional) Include non-canonical gene transcripts in\nthe analysis (default False).")), ToolInput(tag="in_protein_databases", input_type=Array(t=String(), optional=True), prefix="--protein_databases", doc=InputDocumentation(doc="(Optional) Space-delimited list of one or more protein\nfeature databases to include when visualizing\nproteins. Options are: pfam, smart, superfamily,\ntigrfam, pfscan (Prosite_profiles), tmhmm (i.e.\ntransmembrane), seg (low_complexity regions), ncoils\n(coiled coil regions), prints, pirsf, and signalp\n(signal peptide regions) (default: --protein_databases\npfam and tmhmm).")), ToolInput(tag="in_re_color", input_type=Int(optional=True), prefix="--recolor", doc=InputDocumentation(doc="(Optional) Re-color a domain. Provide the original\nname of the domain then your color (semi-colon\ndelimited, all in quotes). Can specify --recolor\nmultiples for each domain. (e.g. --color\n'Pkinase_Tyr;blue' --color 'I-set;#006600').")), ToolInput(tag="in_rename", input_type=String(optional=True), prefix="--rename", doc=InputDocumentation(doc="(Optional) Rename a domain. Provide the original name\nof the domain then your new name (semi-colon\ndelimited, all in quotes). Can specify --rename\nmultiples for each domain. (e.g. --rename\n'Pkinase_Tyr;Kinase').")), ToolInput(tag="in_exclude_domain", input_type=Array(t=String(), optional=True), prefix="--exclude_domain", doc=InputDocumentation(doc="(Optional) Exclude a certain domain(s) from plotting\nby providing a space-separated list of domain names.")), ToolInput(tag="in_type", input_type=File(optional=True), prefix="--type", doc=InputDocumentation(doc="(Optional) Image file type (png, jpeg, pdf). Default:")), ToolInput(tag="in_height", input_type=File(optional=True), prefix="--height", doc=InputDocumentation(doc="(Optional) Image file height in inches (default 3).")), ToolInput(tag="in_dpi", input_type=String(optional=True), prefix="--dpi", doc=InputDocumentation(doc="(Optional) Dots per inch.")), ToolInput(tag="in_font_size", input_type=Int(optional=True), prefix="--fontsize", doc=InputDocumentation(doc="(Optional) Fontsize (default 12).")), ToolInput(tag="in_wt", input_type=Boolean(optional=True), prefix="--WT", doc=InputDocumentation(doc="(Optional) Include this to plot wild-type\narchitechtures of the 5' and 3' genes")), ToolInput(tag="in_middle_star", input_type=Boolean(optional=True), prefix="--middlestar", doc=InputDocumentation(doc="(Optional) Insert a * at the junction position for the\ncdna, cds, and protein sequences (default False).")), ToolInput(tag="in_no_domain_labels", input_type=Boolean(optional=True), prefix="--no_domain_labels", doc=InputDocumentation(doc="(Optional) Do not label domains.")), ToolInput(tag="in_debug", input_type=Boolean(optional=True), prefix="--debug", doc=InputDocumentation(doc="(Optional) Enable debugging logging.")), ToolInput(tag="in_png", input_type=String(), position=0, doc=InputDocumentation(doc="-w WIDTH, --width WIDTH"))], outputs=[ToolOutput(tag="out_file", output_type=File(optional=True), selector=InputSelector(input_to_select="in_file", type_hint=File()), doc=OutputDocumentation(doc="Output file from fusion-finding algorithm."))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Agfusion_Batch_V0_1_0().translate("wdl", allow_empty_container=True)

