from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, Directory, File, String

Funannotate_Annotate_V0_1_0 = CommandToolBuilder(tool="funannotate_annotate", base_command=["funannotate", "annotate"], inputs=[ToolInput(tag="in_input", input_type=Boolean(optional=True), prefix="--input", doc=InputDocumentation(doc="Folder from funannotate predict")), ToolInput(tag="in_out", input_type=Directory(optional=True), prefix="--out", doc=InputDocumentation(doc="Output folder for results")), ToolInput(tag="in_fast_a", input_type=Boolean(optional=True), prefix="--fasta", doc=InputDocumentation(doc="Genome in multi-fasta format")), ToolInput(tag="in_species", input_type=Boolean(optional=True), prefix="--species", doc=InputDocumentation(doc="Species name, use quotes for binomial, e.g. 'Aspergillus fumigatus'")), ToolInput(tag="in_sbt", input_type=Boolean(optional=True), prefix="--sbt", doc=InputDocumentation(doc="NCBI submission template file. (Recommended)")), ToolInput(tag="in_annotations", input_type=Boolean(optional=True), prefix="--annotations", doc=InputDocumentation(doc="Custom annotations (3 column tsv file)")), ToolInput(tag="in_eggnog", input_type=Boolean(optional=True), prefix="--eggnog", doc=InputDocumentation(doc="Eggnog-mapper annotations file (if NOT installed)")), ToolInput(tag="in_anti_smash", input_type=File(optional=True), prefix="--antismash", doc=InputDocumentation(doc="antiSMASH secondary metabolism results (GBK file from output)")), ToolInput(tag="in_iprs_can", input_type=Boolean(optional=True), prefix="--iprscan", doc=InputDocumentation(doc="InterProScan5 XML file")), ToolInput(tag="in_ph_obius", input_type=Boolean(optional=True), prefix="--phobius", doc=InputDocumentation(doc="Phobius pre-computed results (if phobius NOT installed)")), ToolInput(tag="in_isolate", input_type=Boolean(optional=True), prefix="--isolate", doc=InputDocumentation(doc="Isolate name")), ToolInput(tag="in_strain", input_type=Boolean(optional=True), prefix="--strain", doc=InputDocumentation(doc="Strain name")), ToolInput(tag="in_rename", input_type=Boolean(optional=True), prefix="--rename", doc=InputDocumentation(doc="Rename GFF gene models with locus_tag from NCBI.")), ToolInput(tag="in_fix", input_type=Boolean(optional=True), prefix="--fix", doc=InputDocumentation(doc="Gene/Product names fixed (TSV: GeneID      Name    Product)")), ToolInput(tag="in_remove", input_type=Boolean(optional=True), prefix="--remove", doc=InputDocumentation(doc="Gene/Product names to remove (TSV: Gene    Product)")), ToolInput(tag="in_busco_db", input_type=Boolean(optional=True), prefix="--busco_db", doc=InputDocumentation(doc="BUSCO models. Default: dikarya")), ToolInput(tag="in_tbl_two_asn", input_type=Boolean(optional=True), prefix="--tbl2asn", doc=InputDocumentation(doc="Additional parameters for tbl2asn. Default: '-l paired-ends'")), ToolInput(tag="in_database", input_type=Boolean(optional=True), prefix="--database", doc=InputDocumentation(doc="Path to funannotate database. Default: $FUNANNOTATE_DB")), ToolInput(tag="in_force", input_type=Directory(optional=True), prefix="--force", doc=InputDocumentation(doc="Force over-write of output folder")), ToolInput(tag="in_cpus", input_type=Boolean(optional=True), prefix="--cpus", doc=InputDocumentation(doc="Number of CPUs to use. Default: 2")), ToolInput(tag="in_no_progress", input_type=Boolean(optional=True), prefix="--no-progress", doc=InputDocumentation(doc="Do not print progress to stdout for long sub jobs")), ToolInput(tag="in_arguments", input_type=String(), position=0, doc=InputDocumentation(doc=""))], outputs=[ToolOutput(tag="out_out", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_out", type_hint=File()), doc=OutputDocumentation(doc="Output folder for results")), ToolOutput(tag="out_anti_smash", output_type=File(optional=True), selector=InputSelector(input_to_select="in_anti_smash", type_hint=File()), doc=OutputDocumentation(doc="antiSMASH secondary metabolism results (GBK file from output)")), ToolOutput(tag="out_force", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_force", type_hint=File()), doc=OutputDocumentation(doc="Force over-write of output folder"))], container="quay.io/biocontainers/funannotate:1.8.7--pyh5e36f6f_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Funannotate_Annotate_V0_1_0().translate("wdl")

