from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import String, Directory, Int

Raiss_V0_1_0 = CommandToolBuilder(tool="raiss", base_command=["raiss"], inputs=[ToolInput(tag="in_chrom", input_type=String(optional=True), prefix="--chrom", doc=InputDocumentation(doc="chromosome to impute to the chr\d+ format")), ToolInput(tag="in_g_was", input_type=String(optional=True), prefix="--gwas", doc=InputDocumentation(doc="GWAS to impute to the consortia_trait format")), ToolInput(tag="in_ref_folder", input_type=Directory(optional=True), prefix="--ref-folder", doc=InputDocumentation(doc="reference panel location (used to determine which snp\nto impute)")), ToolInput(tag="in_ld_folder", input_type=Directory(optional=True), prefix="--ld-folder", doc=InputDocumentation(doc="Location LD correlation matrices")), ToolInput(tag="in_zscore_folder", input_type=Directory(optional=True), prefix="--zscore-folder", doc=InputDocumentation(doc="Location of the zscore files of the gwases to impute")), ToolInput(tag="in_output_folder", input_type=Directory(optional=True), prefix="--output-folder", doc=InputDocumentation(doc="Location of the impute zscore files")), ToolInput(tag="in_window_size", input_type=Int(optional=True), prefix="--window-size", doc=InputDocumentation(doc="Size of the non overlapping window")), ToolInput(tag="in_buffer_size", input_type=Int(optional=True), prefix="--buffer-size", doc=InputDocumentation(doc="Size of the buffer around the imputation window")), ToolInput(tag="in_l_two_regularization", input_type=Int(optional=True), prefix="--l2-regularization", doc=InputDocumentation(doc="Size of the small value added to the diagonal of the\ncovariance matrix before inversion")), ToolInput(tag="in_eigen_threshold", input_type=String(optional=True), prefix="--eigen-threshold", doc=InputDocumentation(doc="threshold under which eigen vectors are removed for\nthe computation of the pseudo inverse")), ToolInput(tag="in_r_two_threshold", input_type=Int(optional=True), prefix="--R2-threshold", doc=InputDocumentation(doc="R square (imputation quality) threshold bellow which\nSNPs are filtered from the output")), ToolInput(tag="in_ref_panel_suffix", input_type=String(optional=True), prefix="--ref-panel-suffix", doc=InputDocumentation(doc="end of the suffix for the reference panel files\n"))], outputs=[ToolOutput(tag="out_output_folder", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_output_folder", type_hint=File()), doc=OutputDocumentation(doc="Location of the impute zscore files"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Raiss_V0_1_0().translate("wdl", allow_empty_container=True)

