version 1.0

task HaphpipeBuildTreeNG {
  input {
    String? seqs
    String? in_type
    String? output_name
    String? outdir
    String? model
    Boolean? all
    String? branch_length
    String? con_sense
    String? rand_tree
    String? pars_tree
    String? user_tree
    Boolean? search
    Boolean? search_one_random
    String? constraint_tree
    String? out_group
    Boolean? bs_converge
    Boolean? bs_msa
    String? bs_trees
    String? bs_tree_cut_off
    String? bs_metric
    Boolean? bootstrap
    Boolean? check
    String? log
    Boolean? log_lh
    Boolean? terrace
    String? seed
    Boolean? redo
    Boolean? keep_tmp
    Boolean? quiet
    String? log_file
    String? n_cpu
    Boolean? debug
  }
  command <<<
    haphpipe build_tree_NG \
      ~{if defined(seqs) then ("--seqs " +  '"' + seqs + '"') else ""} \
      ~{if defined(in_type) then ("--in_type " +  '"' + in_type + '"') else ""} \
      ~{if defined(output_name) then ("--output_name " +  '"' + output_name + '"') else ""} \
      ~{if defined(outdir) then ("--outdir " +  '"' + outdir + '"') else ""} \
      ~{if defined(model) then ("--model " +  '"' + model + '"') else ""} \
      ~{true="--all" false="" all} \
      ~{if defined(branch_length) then ("--branch_length " +  '"' + branch_length + '"') else ""} \
      ~{if defined(con_sense) then ("--consense " +  '"' + con_sense + '"') else ""} \
      ~{if defined(rand_tree) then ("--rand_tree " +  '"' + rand_tree + '"') else ""} \
      ~{if defined(pars_tree) then ("--pars_tree " +  '"' + pars_tree + '"') else ""} \
      ~{if defined(user_tree) then ("--user_tree " +  '"' + user_tree + '"') else ""} \
      ~{true="--search" false="" search} \
      ~{true="--search_1random" false="" search_one_random} \
      ~{if defined(constraint_tree) then ("--constraint_tree " +  '"' + constraint_tree + '"') else ""} \
      ~{if defined(out_group) then ("--outgroup " +  '"' + out_group + '"') else ""} \
      ~{true="--bsconverge" false="" bs_converge} \
      ~{true="--bs_msa" false="" bs_msa} \
      ~{if defined(bs_trees) then ("--bs_trees " +  '"' + bs_trees + '"') else ""} \
      ~{if defined(bs_tree_cut_off) then ("--bs_tree_cutoff " +  '"' + bs_tree_cut_off + '"') else ""} \
      ~{if defined(bs_metric) then ("--bs_metric " +  '"' + bs_metric + '"') else ""} \
      ~{true="--bootstrap" false="" bootstrap} \
      ~{true="--check" false="" check} \
      ~{if defined(log) then ("--log " +  '"' + log + '"') else ""} \
      ~{true="--loglh" false="" log_lh} \
      ~{true="--terrace" false="" terrace} \
      ~{if defined(seed) then ("--seed " +  '"' + seed + '"') else ""} \
      ~{true="--redo" false="" redo} \
      ~{true="--keep_tmp" false="" keep_tmp} \
      ~{true="--quiet" false="" quiet} \
      ~{if defined(log_file) then ("--logfile " +  '"' + log_file + '"') else ""} \
      ~{if defined(n_cpu) then ("--ncpu " +  '"' + n_cpu + '"') else ""} \
      ~{true="--debug" false="" debug}
  >>>
  parameter_meta {
    seqs: "Input alignment in FASTA or PHYLIP format"
    in_type: "File format: FASTA or PHYLIP (Default is FASTA)"
    output_name: "Run name for trees"
    outdir: "Output directory (default: .)"
    model: "Substitution model OR path to partition file"
    all: "Run bootstrap search and find best ML tree (default: False)"
    branch_length: "Branch length estimation mode: linked, scaled, unlinked (partitioned analysis only)"
    con_sense: "Consensus tree building options: STRICT, MR, or MRE"
    rand_tree: "Start from a random topology"
    pars_tree: "Start from a tree generated by the parsimony-based randomized stepwise addition algorithm"
    user_tree: "Load a custom starting tree from the NEWICK file"
    search: "Find best scoring ML tree (default) (default: False)"
    search_one_random: "Find best scoring ML tree with 1 random tree (default: False)"
    constraint_tree: "Specify a constraint tree to e.g. enforce monophyly of certain groups"
    out_group: "Outgroup(s) for tree"
    bs_converge: "A posteriori bootstrap convergence test (default: False)"
    bs_msa: "Generate bootstrap replicate alignments (default: False)"
    bs_trees: "Number of bootstrap trees OR autoMRE"
    bs_tree_cut_off: "Change the bootstopping cutoff value to make the test more or less stringent"
    bs_metric: "Options: tbe or fbp,tbe"
    bootstrap: "Run non-parametric bootstrap analysis (default: False)"
    check: "Check alignment file and remove any columns consisting entirely of gaps (default: False)"
    log: "Options for output verbosity: ERROR, WARNING, RESULT, INFO, PROGRESS, VERBOSE, or DEBUG"
    log_lh: "Compute log-likelihood of a given tree without any optimization (default: False)"
    terrace: "Check whether a tree lies on a phylogenetic terrace (default: False)"
    seed: "Seed for random numbers (default: 12345)"
    redo: "Run even if there are existing files with the same name (use with caution!) (default: False)"
    keep_tmp: "Keep temporary directory (default: False)"
    quiet: "Do not write output to console (silence stdout and stderr) (default: False)"
    log_file: "Append console output to this file"
    n_cpu: "Number of CPU to use (default: 1)"
    debug: "Print commands but do not run (default: False)"
  }
}