class: CommandLineTool
id: gottcha.pl.cwl
inputs:
- id: in_input
  doc: "|i    <STRING>  Input one or multiple FASTQ file(s). Use comma (,)\nto separate\
    \ multiple input files."
  type: boolean
  inputBinding:
    prefix: --input
- id: in_database
  doc: "|d <STRING>  The path of signature database. The database can be\nin FASTA\
    \ format or BWA index (5 files)."
  type: boolean
  inputBinding:
    prefix: --database
- id: in_threads
  doc: '|t  <INT>     Number of threads [default: auto-detect]'
  type: boolean
  inputBinding:
    prefix: --threads
- id: in_db_level
  doc: "|l  <STRING>  Specify the taxonomic level of the input database\n(e.g. family,\
    \ species, genus, strain, etc.). The\nvalue will be auto-detected if the input\
    \ database\nended with levels (e.g. GOTTCHA_db.species).\n[default: none]"
  type: boolean
  inputBinding:
    prefix: --dbLevel
- id: in_outdir
  doc: '|o   <STRING>  Output directory [default: ./]'
  type: Directory
  inputBinding:
    prefix: --outdir
- id: in_prefix
  doc: "|p   <STRING>  Filename prefix of the output.\n[default: <INPUT_FILENAME_PREFIX>]"
  type: boolean
  inputBinding:
    prefix: --prefix
- id: in_rel_abu
  doc: "|r   <STRING>  The field will be used to calculate relative\nabundance. You\
    \ can specify one of the following\nfields: \"LINEAR_LENGTH\", \"TOTAL_BP_MAPPED\"\
    ,\n\"HIT_COUNT\", \"LINEAR_DOC\".\n[default: LINEAR_DOC]"
  type: boolean
  inputBinding:
    prefix: --relAbu
- id: in_mode
  doc: "|m     <STRING>  You can specify one of the output mode:\n\"summary\" : this\
    \ mode will report a summary of\nprofiling result to *.gottcha.tsv file.\n\"full\"\
    \    : other than a summary, this mode will\nreport unfiltered result to\n*.gottcha_full.tsv\
    \ with more detail.\n\"all\"     : other than two tables, this mode will\nkeep\
    \ all output files that were\ngenerated by each profiling step.\n[default: summary]"
  type: File
  inputBinding:
    prefix: --mode
- id: in_no_plasmid_hit
  doc: "|n       Ignore alignments that hit to plasmids\n[default: null]"
  type: boolean
  inputBinding:
    prefix: --noPlasmidHit
- id: in_bwa_opt
  doc: "|b   <STRING>  BWA-MEM in this script is used to map input reads to\nGOTTCHA\
    \ database. If you want to run it with your own\nparameters, use this option to\
    \ specify.\n[default: \"-k 30 -T 0 -B 100 -O 100 -E 100\"]"
  type: boolean
  inputBinding:
    prefix: --bwaOpt
- id: in_st_dir
  doc: "|s    <STRING>  Specify a directory contains pre-splitrimmed input\nfiles.\
    \ E.g. input file is \"test.fastq\", the script\nwill looking for \"test_splitrim.fastq\"\
    \ and\n\"test_splitrim.stats.txt\" in the specified directory."
  type: boolean
  inputBinding:
    prefix: --stDir
- id: in_dumps_am
  doc: Dump the mapping result in SAM format.
  type: boolean
  inputBinding:
    prefix: --dumpSam
- id: in_min_q
  doc: "<INT>     Minimum quality for a read to be considered valid\n(0-41) [default:\
    \ 20]"
  type: boolean
  inputBinding:
    prefix: --minQ
- id: in_fix_l
  doc: "<INT>     Fixed length to which each trimmed read will be cut\ndown to [default:\
    \ 30]"
  type: boolean
  inputBinding:
    prefix: --fixL
- id: in_ascii
  doc: "<INT>     ASCII encoding of quality score (33 or 64) [default:\n33]"
  type: boolean
  inputBinding:
    prefix: --ascii
- id: in_min_cov
  doc: "<FLOAT>   Minimum linear coverage to be considered valid in\nabundance calculation\
    \ [default: 0.005]"
  type: boolean
  inputBinding:
    prefix: --minCov
- id: in_min_ml_hl
  doc: "<INT>     Minimum Mean-Linear-Hit-Length to be considered valid\nin abundance\
    \ calculation [default: 5]"
  type: boolean
  inputBinding:
    prefix: --minMLHL
- id: in_c_cov
  doc: "<FLOAT>   Critical coverage below which --minMLHL will cause an\norganism\
    \ to fail [default: 0.006]"
  type: boolean
  inputBinding:
    prefix: --cCov
- id: in_min_len
  doc: "<INT>     Minimum unique length to be considered valid in\nabundance calculation\
    \ [default: 100]"
  type: boolean
  inputBinding:
    prefix: --minLen
- id: in_min_hits
  doc: "<INT>     Minimum number of hits to be considered valid in\nabundance calculation\
    \ [10]"
  type: boolean
  inputBinding:
    prefix: --minHits
- id: in_help_slash_h_slash
  doc: '?             display this help'
  type: boolean
  inputBinding:
    prefix: --help/h/
outputs:
- id: out_stdout
  doc: Standard output stream
  type: stdout
- id: out_outdir
  doc: '|o   <STRING>  Output directory [default: ./]'
  type: Directory
  outputBinding:
    glob: $(inputs.in_outdir)
- id: out_mode
  doc: "|m     <STRING>  You can specify one of the output mode:\n\"summary\" : this\
    \ mode will report a summary of\nprofiling result to *.gottcha.tsv file.\n\"full\"\
    \    : other than a summary, this mode will\nreport unfiltered result to\n*.gottcha_full.tsv\
    \ with more detail.\n\"all\"     : other than two tables, this mode will\nkeep\
    \ all output files that were\ngenerated by each profiling step.\n[default: summary]"
  type: File
  outputBinding:
    glob: $(inputs.in_mode)
cwlVersion: v1.1
baseCommand:
- gottcha.pl
