from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, String

Pbsv_Call_V0_1_0 = CommandToolBuilder(tool="pbsv_call", base_command=["pbsv", "call"], inputs=[ToolInput(tag="in_chunk_length", input_type=Boolean(optional=True), prefix="--chunk-length", doc=InputDocumentation(doc="STR   Process in chunks of N reference bp. [1M]")), ToolInput(tag="in_region", input_type=Boolean(optional=True), prefix="--region", doc=InputDocumentation(doc="STR   Limit discovery to this reference region: CHR|CHR:START-END.")), ToolInput(tag="in_types", input_type=Boolean(optional=True), prefix="--types", doc=InputDocumentation(doc="STR   Call these SV types: 'DEL', 'INS', 'INV', 'DUP', 'BND', 'CNV'.\n[DEL,INS,INV,DUP,BND,CNV]")), ToolInput(tag="in_min_sv_length", input_type=Boolean(optional=True), prefix="--min-sv-length", doc=InputDocumentation(doc="STR   Ignore variants with length < N bp. [20]")), ToolInput(tag="in_min_cnv_length", input_type=Boolean(optional=True), prefix="--min-cnv-length", doc=InputDocumentation(doc="STR   Ignore CNVs with length < N bp. [1K]")), ToolInput(tag="in_max_ins_length", input_type=Boolean(optional=True), prefix="--max-ins-length", doc=InputDocumentation(doc="STR   Ignore insertions with length > N bp. [10K]")), ToolInput(tag="in_max_dup_length", input_type=Boolean(optional=True), prefix="--max-dup-length", doc=InputDocumentation(doc="STR   Ignore duplications with length > N bp. [100K]")), ToolInput(tag="in_cluster_max_length_perc_diff", input_type=Boolean(optional=True), prefix="--cluster-max-length-perc-diff", doc=InputDocumentation(doc="INT   Do not cluster signatures with difference in length > P%. [25]")), ToolInput(tag="in_cluster_max_ref_pos_diff", input_type=Boolean(optional=True), prefix="--cluster-max-ref-pos-diff", doc=InputDocumentation(doc="STR   Do not cluster signatures > N bp apart in reference. [200]")), ToolInput(tag="in_cluster_min_base_pair_perc_id", input_type=Boolean(optional=True), prefix="--cluster-min-basepair-perc-id", doc=InputDocumentation(doc="INT   Do not cluster signatures with basepair identity < P%. [10]")), ToolInput(tag="in_max_consensus_coverage", input_type=Boolean(optional=True), prefix="--max-consensus-coverage", doc=InputDocumentation(doc="INT   Limit to N reads for variant consensus. [20]")), ToolInput(tag="in_poa_scores", input_type=Boolean(optional=True), prefix="--poa-scores", doc=InputDocumentation(doc="STR   Score POA alignment with triplet match,mismatch,gap. [1,-2,-2]")), ToolInput(tag="in_min_realign_length", input_type=Boolean(optional=True), prefix="--min-realign-length", doc=InputDocumentation(doc="STR   Consider segments with > N length for re-alignment. [100]")), ToolInput(tag="in_call_min_reads_all_samples", input_type=Boolean(optional=True), prefix="--call-min-reads-all-samples", doc=InputDocumentation(doc="INT   Ignore calls supported by < N reads total across samples. [2]")), ToolInput(tag="in_call_min_reads_one_sample", input_type=Boolean(optional=True), prefix="--call-min-reads-one-sample", doc=InputDocumentation(doc="INT   Ignore calls supported by < N reads in every sample. [2]")), ToolInput(tag="in_call_min_reads_per_strand_all_samples", input_type=Boolean(optional=True), prefix="--call-min-reads-per-strand-all-samples", doc=InputDocumentation(doc="INT   Ignore calls supported by < N reads per strand total across samples\n[1]")), ToolInput(tag="in_call_min_bnd_reads_all_samples", input_type=Boolean(optional=True), prefix="--call-min-bnd-reads-all-samples", doc=InputDocumentation(doc="INT   Ignore BND calls supported by < N reads total across samples [2]")), ToolInput(tag="in_call_min_read_perc_one_sample", input_type=Boolean(optional=True), prefix="--call-min-read-perc-one-sample", doc=InputDocumentation(doc="INT   Ignore calls supported by < P% of reads in every sample. [20]")), ToolInput(tag="in_ccs", input_type=Boolean(optional=True), prefix="--ccs", doc=InputDocumentation(doc="CCS optimized parameters: -A 1 -O 1 -S 0 -P 10.")), ToolInput(tag="in_gt_min_reads", input_type=Boolean(optional=True), prefix="--gt-min-reads", doc=InputDocumentation(doc="INT   Minimum supporting reads to assign a sample a non-reference\ngenotype. [1]")), ToolInput(tag="in_annotations", input_type=Boolean(optional=True), prefix="--annotations", doc=InputDocumentation(doc="FILE  Annotate variants by comparing with sequences in fasta.Default\nannotations are ALU, L1, SVA.")), ToolInput(tag="in_annotation_min_perc_sim", input_type=Boolean(optional=True), prefix="--annotation-min-perc-sim", doc=InputDocumentation(doc="INT   Annotate variant if sequence similarity > P%. [60]")), ToolInput(tag="in_min_n_in_gap", input_type=Boolean(optional=True), prefix="--min-N-in-gap", doc=InputDocumentation(doc="STR   Consider >= N consecutive 'N' bp as a reference gap. [50]")), ToolInput(tag="in_filter_near_reference_gap", input_type=Boolean(optional=True), prefix="--filter-near-reference-gap", doc=InputDocumentation(doc="STR   Flag variants < N bp from a gap as 'NearReferenceGap'. [1K]")), ToolInput(tag="in_filter_near_contig_end", input_type=Boolean(optional=True), prefix="--filter-near-contig-end", doc=InputDocumentation(doc="STR   Flag variants < N bp from a contig end as 'NearContigEnd'. [1K]")), ToolInput(tag="in_num_threads", input_type=Boolean(optional=True), prefix="--num-threads", doc=InputDocumentation(doc="INT   Number of threads to use, 0 means autodetection. [0]")), ToolInput(tag="in_log_level", input_type=Boolean(optional=True), prefix="--log-level", doc=InputDocumentation(doc="STR   Set log level. Valid choices: (TRACE, DEBUG, INFO, WARN, FATAL).\n[WARN]")), ToolInput(tag="in_log_file", input_type=Boolean(optional=True), prefix="--log-file", doc=InputDocumentation(doc="FILE  Log to a file, instead of stderr.")), ToolInput(tag="in_ref_dot_out_dot_vcf", input_type=String(), position=0, doc=InputDocumentation(doc="STR   Variant call format (VCF) output."))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Pbsv_Call_V0_1_0().translate("wdl", allow_empty_container=True)

