from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, String, File, Int, Float, Directory

Poppunk_V0_1_0 = CommandToolBuilder(tool="poppunk", base_command=["poppunk"], inputs=[ToolInput(tag="in_easy_run", input_type=Boolean(optional=True), prefix="--easy-run", doc=InputDocumentation(doc="Create clusters from assemblies with default settings")), ToolInput(tag="in_create_db", input_type=Boolean(optional=True), prefix="--create-db", doc=InputDocumentation(doc="Create pairwise distances database between reference")), ToolInput(tag="in_refine_model", input_type=Boolean(optional=True), prefix="--refine-model", doc=InputDocumentation(doc="Refine the accuracy of a fitted model")), ToolInput(tag="in_assign_query", input_type=Boolean(optional=True), prefix="--assign-query", doc=InputDocumentation(doc="Assign the cluster of query sequences without re-\nrunning the whole mixture model")), ToolInput(tag="in_generate_viz", input_type=Boolean(optional=True), prefix="--generate-viz", doc=InputDocumentation(doc="Generate files for a visualisation from an existing")), ToolInput(tag="in_use_model", input_type=Boolean(optional=True), prefix="--use-model", doc=InputDocumentation(doc="Apply a fitted model to a reference database to\nrestore database files")), ToolInput(tag="in_ref_db", input_type=String(optional=True), prefix="--ref-db", doc=InputDocumentation(doc="Location of built reference database")), ToolInput(tag="in_r_files", input_type=File(optional=True), prefix="--r-files", doc=InputDocumentation(doc="File listing reference input assemblies")), ToolInput(tag="in_q_files", input_type=File(optional=True), prefix="--q-files", doc=InputDocumentation(doc="File listing query input assemblies")), ToolInput(tag="in_distances", input_type=String(optional=True), prefix="--distances", doc=InputDocumentation(doc="Prefix of input pickle of pre-calculated distances")), ToolInput(tag="in_external_clustering", input_type=File(optional=True), prefix="--external-clustering", doc=InputDocumentation(doc="File with cluster definitions or other labels\ngenerated with any other method.")), ToolInput(tag="in_output", input_type=String(optional=True), prefix="--output", doc=InputDocumentation(doc="Prefix for output files (required)")), ToolInput(tag="in_plot_fit", input_type=Int(optional=True), prefix="--plot-fit", doc=InputDocumentation(doc="Create this many plots of some fits relating k-mer to\ncore/accessory distances [default = 0]")), ToolInput(tag="in_full_db", input_type=Boolean(optional=True), prefix="--full-db", doc=InputDocumentation(doc="Keep full reference database, not just representatives")), ToolInput(tag="in_update_db", input_type=Boolean(optional=True), prefix="--update-db", doc=InputDocumentation(doc="Update reference database with query sequences")), ToolInput(tag="in_overwrite", input_type=Boolean(optional=True), prefix="--overwrite", doc=InputDocumentation(doc="Overwrite any existing database files")), ToolInput(tag="in_min_k", input_type=Int(optional=True), prefix="--min-k", doc=InputDocumentation(doc="Minimum kmer length [default = 9]")), ToolInput(tag="in_max_k", input_type=Int(optional=True), prefix="--max-k", doc=InputDocumentation(doc="Maximum kmer length [default = 29]")), ToolInput(tag="in_k_step", input_type=Int(optional=True), prefix="--k-step", doc=InputDocumentation(doc="K-mer step size [default = 4]")), ToolInput(tag="in_sketch_size", input_type=Int(optional=True), prefix="--sketch-size", doc=InputDocumentation(doc="Kmer sketch size [default = 10000]")), ToolInput(tag="in_min_km_er_count", input_type=Int(optional=True), prefix="--min-kmer-count", doc=InputDocumentation(doc="Minimum k-mer count when using reads as input [default\n= 0]")), ToolInput(tag="in_exact_count", input_type=Boolean(optional=True), prefix="--exact-count", doc=InputDocumentation(doc="Use the exact k-mer counter with reads [default = use\ncountmin counter]")), ToolInput(tag="in_strand_preserved", input_type=Boolean(optional=True), prefix="--strand-preserved", doc=InputDocumentation(doc="Treat input as being on the same strand, and ignore\nreverse complement k-mers [default = use canonical\nk-mers]")), ToolInput(tag="in_max_a_dist", input_type=Int(optional=True), prefix="--max-a-dist", doc=InputDocumentation(doc="Maximum accessory distance to permit [default = 0.5]")), ToolInput(tag="in_ignore_length", input_type=Boolean(optional=True), prefix="--ignore-length", doc=InputDocumentation(doc="Ignore outliers in terms of assembly length [default =\nFalse]")), ToolInput(tag="in_estimated_length", input_type=Int(optional=True), prefix="--estimated-length", doc=InputDocumentation(doc="Provide an integer estimated genome length when using\n'--ignore-length' [default = 2000000]")), ToolInput(tag="in_maximum_number_mixture", input_type=Int(optional=True), prefix="--K", doc=InputDocumentation(doc="Maximum number of mixture components [default = 2]")), ToolInput(tag="in_dbs_can", input_type=Boolean(optional=True), prefix="--dbscan", doc=InputDocumentation(doc="Use DBSCAN rather than mixture model")), ToolInput(tag="in_maximum_number_default", input_type=Int(optional=True), prefix="--D", doc=InputDocumentation(doc="Maximum number of clusters in DBSCAN fitting [default\n= 100]")), ToolInput(tag="in_min_cluster_prop", input_type=Int(optional=True), prefix="--min-cluster-prop", doc=InputDocumentation(doc="Minimum proportion of points in a cluster in DBSCAN\nfitting [default = 0.0001]")), ToolInput(tag="in_pos_shift", input_type=Float(optional=True), prefix="--pos-shift", doc=InputDocumentation(doc="Maximum amount to move the boundary away from origin\n[default = 0.2]")), ToolInput(tag="in_neg_shift", input_type=Float(optional=True), prefix="--neg-shift", doc=InputDocumentation(doc="Maximum amount to move the boundary towards the origin\n[default = 0.4]")), ToolInput(tag="in_manual_start", input_type=File(optional=True), prefix="--manual-start", doc=InputDocumentation(doc="A file containing information for a start point. See\ndocumentation for help.")), ToolInput(tag="in_indiv_refine", input_type=Boolean(optional=True), prefix="--indiv-refine", doc=InputDocumentation(doc="Also run refinement for core and accessory")), ToolInput(tag="in_model_dir", input_type=Directory(optional=True), prefix="--model-dir", doc=InputDocumentation(doc="Directory containing model to use for assigning\nqueries to clusters [default = reference database\ndirectory]")), ToolInput(tag="in_previous_clustering", input_type=Directory(optional=True), prefix="--previous-clustering", doc=InputDocumentation(doc="Directory containing previous cluster definitions and\nnetwork [default = use that in the directory\ncontaining the model]")), ToolInput(tag="in_core_only", input_type=Boolean(optional=True), prefix="--core-only", doc=InputDocumentation(doc="Use a core-distance only model for assigning queries\n[default = False]")), ToolInput(tag="in_accessory_only", input_type=Boolean(optional=True), prefix="--accessory-only", doc=InputDocumentation(doc="Use an accessory-distance only model for assigning\nqueries [default = False]")), ToolInput(tag="in_subset", input_type=File(optional=True), prefix="--subset", doc=InputDocumentation(doc="File with list of sequences to include in\nvisualisation (with --generate-viz only)")), ToolInput(tag="in_micro_react", input_type=Boolean(optional=True), prefix="--microreact", doc=InputDocumentation(doc="Generate output files for microreact visualisation")), ToolInput(tag="in_cytoscape", input_type=Boolean(optional=True), prefix="--cytoscape", doc=InputDocumentation(doc="Generate network output files for Cytoscape")), ToolInput(tag="in_p_handango", input_type=Boolean(optional=True), prefix="--phandango", doc=InputDocumentation(doc="Generate phylogeny and TSV for Phandango visualisation")), ToolInput(tag="in_grape_tree", input_type=Boolean(optional=True), prefix="--grapetree", doc=InputDocumentation(doc="Generate phylogeny and CSV for grapetree visualisation")), ToolInput(tag="in_rapid_nj", input_type=File(optional=True), prefix="--rapidnj", doc=InputDocumentation(doc="Path to rapidNJ binary to build NJ tree for Microreact")), ToolInput(tag="in_perplexity", input_type=Float(optional=True), prefix="--perplexity", doc=InputDocumentation(doc="Perplexity used to calculate t-SNE projection (with\n--microreact) [default=20.0]")), ToolInput(tag="in_info_csv", input_type=String(optional=True), prefix="--info-csv", doc=InputDocumentation(doc="Epidemiological information CSV formatted for\nmicroreact (can be used with other outputs)")), ToolInput(tag="in_use_mash", input_type=Boolean(optional=True), prefix="--use-mash", doc=InputDocumentation(doc="Use the old mash sketch backend [default = False]")), ToolInput(tag="in_mash", input_type=String(optional=True), prefix="--mash", doc=InputDocumentation(doc="Location of mash executable")), ToolInput(tag="in_threads", input_type=Int(optional=True), prefix="--threads", doc=InputDocumentation(doc="Number of threads to use [default = 1]")), ToolInput(tag="in_use_gpu", input_type=Boolean(optional=True), prefix="--use-gpu", doc=InputDocumentation(doc="Use a GPU when calculating distances [default = False]")), ToolInput(tag="in_deviceid", input_type=Int(optional=True), prefix="--deviceid", doc=InputDocumentation(doc="CUDA device ID, if using GPU [default = 0]")), ToolInput(tag="in_no_stream", input_type=Boolean(optional=True), prefix="--no-stream", doc=InputDocumentation(doc="Use temporary files for mash dist interfacing. Reduce\nmemory use/increase disk use for large datasets")), ToolInput(tag="in_sequences", input_type=String(), position=0, doc=InputDocumentation(doc="--fit-model           Fit a mixture model to a reference database")), ToolInput(tag="in_database", input_type=String(), position=1, doc=InputDocumentation(doc="--threshold THRESHOLD")), ToolInput(tag="in_individually", input_type=String(), position=0, doc=InputDocumentation(doc="--no-local            Do not perform the local optimization step (speed up"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Poppunk_V0_1_0().translate("wdl", allow_empty_container=True)

