from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Int, File, Boolean, String, Float

Ribotish_Tisdiff_V0_1_0 = CommandToolBuilder(tool="ribotish_tisdiff", base_command=["ribotish", "tisdiff"], inputs=[ToolInput(tag="in_one", input_type=Int(optional=True), prefix="-1", doc=InputDocumentation(doc="Prediction results of group 1 TIS data")), ToolInput(tag="in_two", input_type=Int(optional=True), prefix="-2", doc=InputDocumentation(doc="Prediction results of group 2 TIS data")), ToolInput(tag="in_group_tis_files", input_type=Int(optional=True), prefix="-a", doc=InputDocumentation(doc="Group 1 TIS enriched riboseq bam files, comma")), ToolInput(tag="in_output_result_file", input_type=File(optional=True), prefix="-o", doc=InputDocumentation(doc="Output result file")), ToolInput(tag="in_t_is_one_para", input_type=Int(optional=True), prefix="--tis1para", doc=InputDocumentation(doc="Input offset parameter files for group 1 bam files")), ToolInput(tag="in_tis_two_para", input_type=Int(optional=True), prefix="--tis2para", doc=InputDocumentation(doc="Input offset parameter files for group 2 bam files")), ToolInput(tag="in_gene_format", input_type=File(optional=True), prefix="--geneformat", doc=InputDocumentation(doc="Gene annotation file format (gtf, bed, gpd, gff,\ndefault: auto)")), ToolInput(tag="in_max_nh", input_type=Int(optional=True), prefix="--maxNH", doc=InputDocumentation(doc="Max NH value allowed for bam alignments (default: 5)")), ToolInput(tag="in_min_mapq", input_type=Int(optional=True), prefix="--minMapQ", doc=InputDocumentation(doc="Min MapQ value required for bam alignments (default:\n1)")), ToolInput(tag="in_secondary", input_type=Boolean(optional=True), prefix="--secondary", doc=InputDocumentation(doc="Use bam secondary alignments")), ToolInput(tag="in_paired", input_type=Boolean(optional=True), prefix="--paired", doc=InputDocumentation(doc="Reads are paired end")), ToolInput(tag="in_lone", input_type=Int(optional=True), prefix="--l1", doc=InputDocumentation(doc="Labels for group 1 TIS data")), ToolInput(tag="in_l_two", input_type=Int(optional=True), prefix="--l2", doc=InputDocumentation(doc="Labels for group 2 TIS data")), ToolInput(tag="in_no_compatible", input_type=Boolean(optional=True), prefix="--nocompatible", doc=InputDocumentation(doc="Do not require reads compatible with transcript splice")), ToolInput(tag="in_chr_map", input_type=File(optional=True), prefix="--chrmap", doc=InputDocumentation(doc="Input chromosome id mapping table file if annotation\nchr ids are not same as chr ids in bam/fasta files")), ToolInput(tag="in_norm_comm", input_type=Boolean(optional=True), prefix="--normcomm", doc=InputDocumentation(doc="Use common TISs instead of union TISs for")), ToolInput(tag="in_rnase_q", input_type=String(optional=True), prefix="--rnaseq", doc=InputDocumentation(doc="RNASeq count input")), ToolInput(tag="in_scale_factor", input_type=Int(optional=True), prefix="--scalefactor", doc=InputDocumentation(doc="Input TIS scale factor of 2/1 (default: auto)")), ToolInput(tag="in_rna_scale", input_type=Int(optional=True), prefix="--rnascale", doc=InputDocumentation(doc="Input RNASeq scale factor of 2/1 (default: auto)")), ToolInput(tag="in_chi_two", input_type=Boolean(optional=True), prefix="--chi2", doc=InputDocumentation(doc="Use chisquare test instead of Fisher's exact test for\nmRNA referenced test")), ToolInput(tag="in_beta_bin_om", input_type=Boolean(optional=True), prefix="--betabinom", doc=InputDocumentation(doc="Use beta-binom test instead of Fisher's exact test for\nmRNA referenced test")), ToolInput(tag="in_export", input_type=String(optional=True), prefix="--export", doc=InputDocumentation(doc="Export count table for differential analysis with\nother tools")), ToolInput(tag="in_plot_out", input_type=File(optional=True), prefix="--plotout", doc=InputDocumentation(doc="Scatter plot output pdf file")), ToolInput(tag="in_fig_size", input_type=Int(optional=True), prefix="--figsize", doc=InputDocumentation(doc="Scatter plot figure size (default: 8,8)")), ToolInput(tag="in_plot_ma", input_type=File(optional=True), prefix="--plotma", doc=InputDocumentation(doc="TIS normalization MA plot output pdf file")), ToolInput(tag="in_qi", input_type=Int(optional=True), prefix="--qi", doc=InputDocumentation(doc="Index of TIS q value, 0 based (default: 15)")), ToolInput(tag="in_minimum_fold_change", input_type=Float(optional=True), prefix="-f", doc=InputDocumentation(doc="Minimum fold change threshold (default: 1.5)")), ToolInput(tag="in_ip_th", input_type=Float(optional=True), prefix="--ipth", doc=InputDocumentation(doc="Input TIS p value threshold (default: 0.05)")), ToolInput(tag="in_i_qth", input_type=Float(optional=True), prefix="--iqth", doc=InputDocumentation(doc="Input TIS q value threshold (default: 0.05)")), ToolInput(tag="in_opt_h", input_type=Float(optional=True), prefix="--opth", doc=InputDocumentation(doc="Output TIS diff p value threshold (default: 0.05)")), ToolInput(tag="in_o_qth", input_type=Float(optional=True), prefix="--oqth", doc=InputDocumentation(doc="Output TIS diff q value threshold (default: 0.05)")), ToolInput(tag="in_number_of_processes", input_type=Int(optional=True), prefix="-p", doc=InputDocumentation(doc="Number of processes")), ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="--verbose", doc=InputDocumentation(doc="Increase output verbosity")), ToolInput(tag="in_b", input_type=Boolean(optional=True), prefix="-b", doc=InputDocumentation(doc="")), ToolInput(tag="in_junctions", input_type=String(), position=2, doc=InputDocumentation(doc="--compatiblemis COMPATIBLEMIS")), ToolInput(tag="in_normalization", input_type=String(), position=0, doc=InputDocumentation(doc="--normanno            Use only annotated TISs for normalization"))], outputs=[ToolOutput(tag="out_output_result_file", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output_result_file", type_hint=File()), doc=OutputDocumentation(doc="Output result file")), ToolOutput(tag="out_plot_out", output_type=File(optional=True), selector=InputSelector(input_to_select="in_plot_out", type_hint=File()), doc=OutputDocumentation(doc="Scatter plot output pdf file")), ToolOutput(tag="out_plot_ma", output_type=File(optional=True), selector=InputSelector(input_to_select="in_plot_ma", type_hint=File()), doc=OutputDocumentation(doc="TIS normalization MA plot output pdf file"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Ribotish_Tisdiff_V0_1_0().translate("wdl", allow_empty_container=True)

