from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, Directory, Int, String, File, Float

Snippy_V0_1_0 = CommandToolBuilder(tool="snippy", base_command=["snippy"], inputs=[ToolInput(tag="in_citation", input_type=Boolean(optional=True), prefix="--citation", doc=InputDocumentation(doc="Print citation for referencing snippy")), ToolInput(tag="in_check", input_type=Boolean(optional=True), prefix="--check", doc=InputDocumentation(doc="Check dependences are installed then exit (default OFF)")), ToolInput(tag="in_force", input_type=Directory(optional=True), prefix="--force", doc=InputDocumentation(doc="Force overwrite of existing output folder (default OFF)")), ToolInput(tag="in_quiet", input_type=Boolean(optional=True), prefix="--quiet", doc=InputDocumentation(doc="No screen output (default OFF)")), ToolInput(tag="in_cpus", input_type=Int(optional=True), prefix="--cpus", doc=InputDocumentation(doc="Maximum number of CPU cores to use (default '8')")), ToolInput(tag="in_ram", input_type=Int(optional=True), prefix="--ram", doc=InputDocumentation(doc="Try and keep RAM under this many GB (default '8')")), ToolInput(tag="in_tmpdir", input_type=String(optional=True), prefix="--tmpdir", doc=InputDocumentation(doc="Fast temporary storage eg. local SSD (default '/tmp')")), ToolInput(tag="in_reference", input_type=String(optional=True), prefix="--reference", doc=InputDocumentation(doc="Reference genome. Supports FASTA, GenBank, EMBL (not GFF) (default '')")), ToolInput(tag="in_r_one", input_type=Int(optional=True), prefix="--R1", doc=InputDocumentation(doc="Reads, paired-end R1 (left) (default '')")), ToolInput(tag="in_r_two", input_type=Int(optional=True), prefix="--R2", doc=InputDocumentation(doc="Reads, paired-end R2 (right) (default '')")), ToolInput(tag="in_se", input_type=String(optional=True), prefix="--se", doc=InputDocumentation(doc="Single-end reads (default '')")), ToolInput(tag="in_ctg_s", input_type=String(optional=True), prefix="--ctgs", doc=InputDocumentation(doc="Don't have reads use these contigs (default '')")), ToolInput(tag="in_peil", input_type=Int(optional=True), prefix="--peil", doc=InputDocumentation(doc="Reads, paired-end R1/R2 interleaved (default '')")), ToolInput(tag="in_bam", input_type=File(optional=True), prefix="--bam", doc=InputDocumentation(doc="Use this BAM file instead of aligning reads (default '')")), ToolInput(tag="in_targets", input_type=File(optional=True), prefix="--targets", doc=InputDocumentation(doc="Only call SNPs from this BED file (default '')")), ToolInput(tag="in_subsample", input_type=Int(optional=True), prefix="--subsample", doc=InputDocumentation(doc="Subsample FASTQ to this proportion (default '1')")), ToolInput(tag="in_outdir", input_type=Directory(optional=True), prefix="--outdir", doc=InputDocumentation(doc="Output folder (default '')")), ToolInput(tag="in_prefix", input_type=String(optional=True), prefix="--prefix", doc=InputDocumentation(doc="Prefix for output files (default 'snps')")), ToolInput(tag="in_report", input_type=Boolean(optional=True), prefix="--report", doc=InputDocumentation(doc="Produce report with visual alignment per variant (default OFF)")), ToolInput(tag="in_cleanup", input_type=Boolean(optional=True), prefix="--cleanup", doc=InputDocumentation(doc="Remove most files not needed for snippy-core (inc. BAMs!) (default OFF)")), ToolInput(tag="in_rg_id", input_type=String(optional=True), prefix="--rgid", doc=InputDocumentation(doc="Use this @RG ID: in the BAM header (default '')")), ToolInput(tag="in_unmapped", input_type=Boolean(optional=True), prefix="--unmapped", doc=InputDocumentation(doc="Keep unmapped reads in BAM and write FASTQ (default OFF)")), ToolInput(tag="in_map_qual", input_type=Int(optional=True), prefix="--mapqual", doc=InputDocumentation(doc="Minimum read mapping quality to consider (default '60')")), ToolInput(tag="in_base_qual", input_type=Int(optional=True), prefix="--basequal", doc=InputDocumentation(doc="Minimum base quality to consider (default '13')")), ToolInput(tag="in_min_cov", input_type=Int(optional=True), prefix="--mincov", doc=InputDocumentation(doc="Minimum site depth to for calling alleles (default '10')")), ToolInput(tag="in_min_frac", input_type=Int(optional=True), prefix="--minfrac", doc=InputDocumentation(doc="Minumum proportion for variant evidence (0=AUTO) (default '0')")), ToolInput(tag="in_min_qual", input_type=Int(optional=True), prefix="--minqual", doc=InputDocumentation(doc="Minumum QUALITY in VCF column 6 (default '100')")), ToolInput(tag="in_max_soft", input_type=Int(optional=True), prefix="--maxsoft", doc=InputDocumentation(doc="Maximum soft clipping to allow (default '10')")), ToolInput(tag="in_bwa_opt", input_type=String(optional=True), prefix="--bwaopt", doc=InputDocumentation(doc="Extra BWA MEM options, eg. -x pacbio (default '')")), ToolInput(tag="in_fb_opt", input_type=Float(optional=True), prefix="--fbopt", doc=InputDocumentation(doc="Extra Freebayes options, eg. --theta 1E-6 --read-snp-limit 2 (default '')"))], outputs=[ToolOutput(tag="out_force", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_force", type_hint=File()), doc=OutputDocumentation(doc="Force overwrite of existing output folder (default OFF)")), ToolOutput(tag="out_outdir", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_outdir", type_hint=File()), doc=OutputDocumentation(doc="Output folder (default '')"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Snippy_V0_1_0().translate("wdl", allow_empty_container=True)

