from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Boolean, Int, String, Float

Bam_Somaticsniper_V0_1_0 = CommandToolBuilder(tool="bam_somaticsniper", base_command=["bam-somaticsniper"], inputs=[ToolInput(tag="in_required_reference_sequence", input_type=File(optional=True), prefix="-f", doc=InputDocumentation(doc="REQUIRED reference sequence in the FASTA format")), ToolInput(tag="in_display_version_information", input_type=Boolean(optional=True), prefix="-v", doc=InputDocumentation(doc="Display version information")), ToolInput(tag="in_filtering_reads_mapping", input_type=Int(optional=True), prefix="-q", doc=InputDocumentation(doc="filtering reads with mapping quality less than INT [0]")), ToolInput(tag="in_filtering_somatic_snv", input_type=Int(optional=True), prefix="-Q", doc=InputDocumentation(doc="filtering somatic snv output with somatic quality less than  INT [15]")), ToolInput(tag="in_report_loh_variants", input_type=String(optional=True), prefix="-L", doc=InputDocumentation(doc="do not report LOH variants as determined by genotypes")), ToolInput(tag="in_report_gain_determined", input_type=String(optional=True), prefix="-G", doc=InputDocumentation(doc="do not report Gain of Reference variants as determined by genotypes")), ToolInput(tag="in_disable_priors_somatic", input_type=String(optional=True), prefix="-p", doc=InputDocumentation(doc="disable priors in the somatic calculation. Increases sensitivity for solid tumors")), ToolInput(tag="in_use_prior_probabilities", input_type=String(optional=True), prefix="-J", doc=InputDocumentation(doc="Use prior probabilities accounting for the somatic mutation rate")), ToolInput(tag="in_prior_probability_somatic", input_type=Float(optional=True), prefix="-s", doc=InputDocumentation(doc="prior probability of a somatic mutation (implies -J) [0.010000]")), ToolInput(tag="in_theta_maq_consensus", input_type=Float(optional=True), prefix="-T", doc=InputDocumentation(doc="theta in maq consensus calling model (for -c/-g) [0.850000]")), ToolInput(tag="in_number_haplotypes_sample", input_type=Int(optional=True), prefix="-N", doc=InputDocumentation(doc="number of haplotypes in the sample (for -c/-g) [2]")), ToolInput(tag="in_prior_difference_two", input_type=Float(optional=True), prefix="-r", doc=InputDocumentation(doc="prior of a difference between two haplotypes (for -c/-g) [0.001000]")), ToolInput(tag="in_normal_sample_d", input_type=String(optional=True), prefix="-n", doc=InputDocumentation(doc="normal sample id (for VCF header) [NORMAL]")), ToolInput(tag="in_tumor_sample_d", input_type=String(optional=True), prefix="-t", doc=InputDocumentation(doc="tumor sample id (for VCF header) [TUMOR]")), ToolInput(tag="in_select_output_format", input_type=String(optional=True), prefix="-F", doc=InputDocumentation(doc="select output format [classic]\nAvailable formats:\nclassic\nvcf\nbed\n"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Bam_Somaticsniper_V0_1_0().translate("wdl", allow_empty_container=True)

