from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, String, Boolean, Int

Erne_Meth_V0_1_0 = CommandToolBuilder(tool="erne_meth", base_command=["erne-meth"], inputs=[ToolInput(tag="in_fast_a", input_type=File(optional=True), prefix="--fasta", doc=InputDocumentation(doc="reference fasta file (can be repeated several\ntime). [REQUIRED]")), ToolInput(tag="in_input", input_type=File(optional=True), prefix="--input", doc=InputDocumentation(doc="BAM/SAM file generated with erne-bs5 [REQUIRED]")), ToolInput(tag="in_output_prefix", input_type=String(optional=True), prefix="--output-prefix", doc=InputDocumentation(doc="prefix for output files (methylation analysis,\nreports). [REQUIRED]")), ToolInput(tag="in_annotations_erne", input_type=Boolean(optional=True), prefix="--annotations-erne", doc=InputDocumentation(doc="Generate methylation annotations in erne\nformat. Format is: <chr> <position> <strand>\n<context> <#C+#T> <methyl_level> <mult_reads>,\nwhere methyl_level = #C/(#C+#T) (or NA if\n#C+#T=0) and mult_reads is the number of\nmultiple reads disambiguated that cover the\nposition (only if --disambiguation-mode is\nspecified, NA otherwise). Coordinates are\n1-based. Default: disabled.")), ToolInput(tag="in_annotations_bismark", input_type=Boolean(optional=True), prefix="--annotations-bismark", doc=InputDocumentation(doc="Generate methylation annotations in bismark\ncytosine format. Format is: <chr> <position>\n<strand> <#C> <#T> <context> <detailed\ncontext>, where <position> for - strand (Gs on\n+ strand) is computed as position of the\ncorresponding G on + strand minus context\nlength plus one (context length = 2 for CG and\n3 for CHG/CHH). Context is one of CG/CHG/CHH,\nwhile detailed context is the exact 3 bases of\nthe context. Coordinates are 1-based. Default:\ndisabled.")), ToolInput(tag="in_annotations_epp", input_type=Boolean(optional=True), prefix="--annotations-epp", doc=InputDocumentation(doc="Generate methylation annotations in the format\nas output files from the Epigenome Processing\nPipeline (EPP) applied at the Broad Institute.\nFormat is (tab separated): <chr name> <start\ncontext(included)> <end context(excluded)>\n<methylation value and coverage as a string (\n'#C/(#T+#C)', '0/0' if not covered )>\n<methylation in range [0,1000]> <strand>\n<context(CG/CHG/CHH)>. Coordinates are 0-based.\nDefault: disabled.")), ToolInput(tag="in_compress", input_type=String(optional=True), prefix="--compress", doc=InputDocumentation(doc="Compress methylation annotations in the\nspecified format (example: --compress bz).\nDefault: do not compress.")), ToolInput(tag="in_write_bam", input_type=File(optional=True), prefix="--write-bam", doc=InputDocumentation(doc="write used alignments in a bam file. The bam\nfile is called 'output_prefix_alignments_used.b\nam', where output_prefix is the prefix\nspecified with --output-prefix.")), ToolInput(tag="in_target", input_type=Int(optional=True), prefix="--target", doc=InputDocumentation(doc="input bed file with targeted regions in the\nformat chr start_pos(included)\nend_pos(excluded). If specified, ERNE-METH will\nproduce also target statistics and three output\nfiles with extension _on_target.txt,\n_off_target.txt and _out_target_cov_distributio\nn.txt containing informations about % of\ncovered bases on target having a given minimum\ncoverage, number of covered bases off target\nhaving a given minimum coverage, and a list of\ntriples <distance_from_target, coverage,\nnumber_of_positions> respectively. Here,\ndistance_from_target is the distance from the\nnearest target region, coverage is the coverage\nof that position and number_of_positions is the\nnumber of positions having that coverage and\ndistance_from_target.")), ToolInput(tag="in_extend_target", input_type=Int(optional=True), prefix="--extend-target", doc=InputDocumentation(doc="Extend boundaries of the targeted regions by\narg positions (on both sides). Useful to take\ninto account tails of aligned reads at the\nextremities of targeted regions while computing\nstatistics. Default: 0.")), ToolInput(tag="in_on_target_annotations", input_type=Boolean(optional=True), prefix="--on-target-annotations", doc=InputDocumentation(doc="In the annotations file print only on-target\npositions (included extended positions, see\n--extend-target). Default: false (print all\npositions).")), ToolInput(tag="in_print_only_covered", input_type=Boolean(optional=True), prefix="--print-only-covered", doc=InputDocumentation(doc="In the annotations file print only covered\npositions. Default: false (print all\ncytosines).")), ToolInput(tag="in_single_mode", input_type=Boolean(optional=True), prefix="--single-mode", doc=InputDocumentation(doc="Modality 1: Use only single-mapping reads. This\nmodality is incompatible with --multiple-mode.\nDefault: enabled.")), ToolInput(tag="in_multiple_mode", input_type=Boolean(optional=True), prefix="--multiple-mode", doc=InputDocumentation(doc="Modality 2: With paired end, if one mate is\nsingle and the other multiple, use the primary\nalignment for the multiple-mapping read.\nDefault: disabled")), ToolInput(tag="in_disambiguation_mode", input_type=Boolean(optional=True), prefix="--disambiguation-mode", doc=InputDocumentation(doc="Modality 3: Use methylation information to\ndisambiguate multiple-mapping reads. WARNING:\ncan be used only on bam files produced with the\noption --print-all and single-end reads.\nDefault: disabled.")), ToolInput(tag="in_use_first", input_type=File(optional=True), prefix="--use-first", doc=InputDocumentation(doc="Use only the first arg alignments (or pairs if\npaired-end) in the input file. If set to 0, use\nall alignments. Default: 0.")), ToolInput(tag="in_de_duplicate", input_type=Boolean(optional=True), prefix="--deduplicate", doc=InputDocumentation(doc="Automatically remove PCR duplicates while\ncalling methylation. An alignment is considered\nduplicate if another alignment with same\nchromosome, starting position and orientation\nhas already been processed. Given a set of\nduplicates, only the first N appearing in the\ninput file will be used for the methylation\ncall, where N is specified with the option\n--allowed-duplicates. For paired-end reads, a\npair is considered duplicate if and only if\nanother pair with same chromosome, starting\nposition and orientation (for both reads) has\nbeen aligned before. Default: false.")), ToolInput(tag="in_allowed_duplicates", input_type=File(optional=True), prefix="--allowed-duplicates", doc=InputDocumentation(doc="If --deduplicate is specified, keep the first\narg duplicate alignments for a read/pair (in\nthe order seen in the bam file). Default: 1.")), ToolInput(tag="in_disable_un_proper_pair", input_type=Boolean(optional=True), prefix="--disable-unproper-pair", doc=InputDocumentation(doc="With paired end reads, discard unproper pairs\n(alignments on different strand/chromosome).\nUnproper pairs could reflect structural\nvariations. Default: disabled (accept unproper\npairs).")), ToolInput(tag="in_max_coverage", input_type=Int(optional=True), prefix="--max-coverage", doc=InputDocumentation(doc="Do not increment Cytosine coverage counters if\ncoverage of a Cytosine is > arg. Can be used to\nnormalize read depth across multiple samples.\nDefault: 2^16-1.")), ToolInput(tag="in_coverage_threshold", input_type=Int(optional=True), prefix="--coverage-threshold", doc=InputDocumentation(doc="If the coverage of a cytosine in the reference\nis < arg then it is considered as not covered.\nDefault: 1.")), ToolInput(tag="in_error_threshold", input_type=Int(optional=True), prefix="--error-threshold", doc=InputDocumentation(doc="(only with --disambiguation-mode) use an\nalignment for the methylation pattern extension\nonly if its average methylation distance per\ncytosine is less or equal this value in %.\nDefault: 100.")), ToolInput(tag="in_delta_m", input_type=Int(optional=True), prefix="--delta-m", doc=InputDocumentation(doc="(only with --disambiguation-mode) During\ndisambiguation phase if there are multiple\nalignments for a read then extend the\nmethylation profile with the best one ONLY if\nits methylation distance is at least by arg\nsmaller than the second best one. High values\nguarantee higher reliability. Default: 0.")), ToolInput(tag="in_min_c_cov", input_type=Int(optional=True), prefix="--min-C-cov", doc=InputDocumentation(doc="(only with --disambiguation-mode) Lower bound\nfor the minimum number of cytosines in an\nalignment that must be covered by the\nmethylation pattern to process the alignment.\nWhen C_cov reaches this value, the algorithm is\nterminated. Default: 10.")), ToolInput(tag="in_max_c_cov", input_type=Int(optional=True), prefix="--max-C-cov", doc=InputDocumentation(doc="(only with --disambiguation-mode) Upper bound\nfor the minimum number of cytosines in an\nalignment that must be covered by the\nmethylation pattern to process the alignment.\nThis is the starting value for C_cov. Default:\n20.")), ToolInput(tag="in_aligned_reads_thr", input_type=Int(optional=True), prefix="--aligned-reads-thr", doc=InputDocumentation(doc="(only with --disambiguation-mode) If during a\ncycle less or equal than arg reads have been\ndisambiguated then decrement C_cov. Default: 10")), ToolInput(tag="in_contamination_reference", input_type=File(optional=True), prefix="--contamination-reference", doc=InputDocumentation(doc="reference file to use for contamination check\n(in ERNE format)\n"))], outputs=[ToolOutput(tag="out_write_bam", output_type=File(optional=True), selector=InputSelector(input_to_select="in_write_bam", type_hint=File()), doc=OutputDocumentation(doc="write used alignments in a bam file. The bam\nfile is called 'output_prefix_alignments_used.b\nam', where output_prefix is the prefix\nspecified with --output-prefix."))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Erne_Meth_V0_1_0().translate("wdl", allow_empty_container=True)

