from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Boolean, String, Directory

Configurestrelkagermlineworkflow_Py_V0_1_0 = CommandToolBuilder(tool="configureStrelkaGermlineWorkflow.py", base_command=["configureStrelkaGermlineWorkflow.py"], inputs=[ToolInput(tag="in_config", input_type=File(optional=True), prefix="--config", doc=InputDocumentation(doc="provide a configuration file to override defaults in\nglobal config file (/usr/local/share/strelka-2.9.10-0/\nbin/configureStrelkaGermlineWorkflow.py.ini)")), ToolInput(tag="in_all_help", input_type=Boolean(optional=True), prefix="--allHelp", doc=InputDocumentation(doc="show all extended/hidden options")), ToolInput(tag="in_bam", input_type=File(optional=True), prefix="--bam", doc=InputDocumentation(doc="Sample BAM or CRAM file. May be specified more than\nonce, multiple inputs will be treated as each BAM file\nrepresenting a different sample. [required] (no\ndefault)")), ToolInput(tag="in_ploidy", input_type=File(optional=True), prefix="--ploidy", doc=InputDocumentation(doc="Provide ploidy file in VCF. The VCF should include one\nsample column per input sample labeled with the same\nsample names found in the input BAM/CRAM RG header\nsections. Ploidy should be provided in records using\nthe FORMAT/CN field, which are interpreted to span the\nrange [POS+1, INFO/END]. Any CN value besides 1 or 0\nwill be treated as 2. File must be tabix indexed. (no\ndefault)")), ToolInput(tag="in_no_compress", input_type=File(optional=True), prefix="--noCompress", doc=InputDocumentation(doc="Provide BED file of regions where gVCF block\ncompression is not allowed. File must be bgzip-\ncompressed/tabix-indexed. (no default)")), ToolInput(tag="in_call_continuous_vf", input_type=String(optional=True), prefix="--callContinuousVf", doc=InputDocumentation(doc="Call variants on CHROM without a ploidy prior\nassumption, issuing calls with continuous variant\nfrequencies (no default)")), ToolInput(tag="in_rna", input_type=Boolean(optional=True), prefix="--rna", doc=InputDocumentation(doc="Set options for RNA-Seq input.")), ToolInput(tag="in_reference_fast_a", input_type=File(optional=True), prefix="--referenceFasta", doc=InputDocumentation(doc="samtools-indexed reference fasta file [required]")), ToolInput(tag="in_in_del_candidates", input_type=File(optional=True), prefix="--indelCandidates", doc=InputDocumentation(doc="Specify a VCF of candidate indel alleles. These\nalleles are always evaluated but only reported in the\noutput when they are inferred to exist in the sample.\nThe VCF must be tabix indexed. All indel alleles must\nbe left-shifted/normalized, any unnormalized alleles\nwill be ignored. This option may be specified more\nthan once, multiple input VCFs will be merged.\n(default: None)")), ToolInput(tag="in_forced_gt", input_type=File(optional=True), prefix="--forcedGT", doc=InputDocumentation(doc="Specify a VCF of candidate alleles. These alleles are\nalways evaluated and reported even if they are\nunlikely to exist in the sample. The VCF must be tabix\nindexed. All indel alleles must be left-\nshifted/normalized, any unnormalized allele will\ntrigger a runtime error. This option may be specified\nmore than once, multiple input VCFs will be merged.\nNote that for any SNVs provided in the VCF, the SNV\nsite will be reported (and for gVCF, excluded from\nblock compression), but the specific SNV alleles are\nignored. (default: None)")), ToolInput(tag="in_targeted", input_type=Boolean(optional=True), prefix="--targeted", doc=InputDocumentation(doc="Set options for exome or other targeted input: note in\nparticular that this flag turns off high-depth filters")), ToolInput(tag="in_call_regions", input_type=File(optional=True), prefix="--callRegions", doc=InputDocumentation(doc="Optionally provide a bgzip-compressed/tabix-indexed\nBED file containing the set of regions to call. No VCF\noutput will be provided outside of these regions. The\nfull genome will still be used to estimate statistics\nfrom the input (such as expected depth per\nchromosome). Only one BED file may be specified.\n(default: call the entire genome)")), ToolInput(tag="in_run_dir", input_type=Directory(optional=True), prefix="--runDir", doc=InputDocumentation(doc="Name of directory to be created where all workflow\nscripts and output will be written. Each analysis\nrequires a separate directory. (default:\nStrelkaGermlineWorkflow)"))], outputs=[ToolOutput(tag="out_run_dir", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_run_dir", type_hint=File()), doc=OutputDocumentation(doc="Name of directory to be created where all workflow\nscripts and output will be written. Each analysis\nrequires a separate directory. (default:\nStrelkaGermlineWorkflow)"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Configurestrelkagermlineworkflow_Py_V0_1_0().translate("wdl", allow_empty_container=True)

