from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, Int, File, String

Bowtie_Align_L_V0_1_0 = CommandToolBuilder(tool="bowtie_align_l", base_command=["bowtie-align-l"], inputs=[ToolInput(tag="in_query_input_files_fastq", input_type=Boolean(optional=True), prefix="-q", doc=InputDocumentation(doc="query input files are FASTQ .fq/.fastq (default)")), ToolInput(tag="in_query_input_files_multifasta", input_type=Boolean(optional=True), prefix="-f", doc=InputDocumentation(doc="query input files are (multi-)FASTA .fa/.mfa")), ToolInput(tag="in_intiint_query_input", input_type=Int(optional=True), prefix="-F", doc=InputDocumentation(doc=":<int>,i:<int> query input files are continuous FASTA where reads\nare substrings (k-mers) extracted from a FASTA file <s>\nand aligned at offsets 1, 1+i, 1+2i ... end of reference")), ToolInput(tag="in_query_input_files_raw", input_type=Boolean(optional=True), prefix="-r", doc=InputDocumentation(doc="query input files are raw one-sequence-per-line")), ToolInput(tag="in_query_sequences_given", input_type=Boolean(optional=True), prefix="-c", doc=InputDocumentation(doc="query sequences given on cmd line (as <mates>, <singles>)")), ToolInput(tag="in_reads_index_colorspace", input_type=Boolean(optional=True), prefix="-C", doc=InputDocumentation(doc="reads and index are in colorspace")), ToolInput(tag="in_quals", input_type=File(optional=True), prefix="--quals", doc=InputDocumentation(doc="QV file(s) corresponding to CSFASTA inputs; use with -f -C")), ToolInput(tag="in_q_one", input_type=File(optional=True), prefix="--Q1", doc=InputDocumentation(doc="same as -Q, but for mate files 1 and 2 respectively")), ToolInput(tag="in_skip", input_type=Int(optional=True), prefix="--skip", doc=InputDocumentation(doc="skip the first <int> reads/pairs in the input")), ToolInput(tag="in_q_up_to", input_type=Int(optional=True), prefix="--qupto", doc=InputDocumentation(doc="stop after first <int> reads/pairs (excl. skipped reads)")), ToolInput(tag="in_trim_five", input_type=Int(optional=True), prefix="--trim5", doc=InputDocumentation(doc="trim <int> bases from 5' (left) end of reads")), ToolInput(tag="in_trim_three", input_type=Int(optional=True), prefix="--trim3", doc=InputDocumentation(doc="trim <int> bases from 3' (right) end of reads")), ToolInput(tag="in_phred_three_three_quals", input_type=Boolean(optional=True), prefix="--phred33-quals", doc=InputDocumentation(doc="input quals are Phred+33 (default)")), ToolInput(tag="in_phred_six_four_quals", input_type=Boolean(optional=True), prefix="--phred64-quals", doc=InputDocumentation(doc="input quals are Phred+64 (same as --solexa1.3-quals)")), ToolInput(tag="in_solexa_quals", input_type=Boolean(optional=True), prefix="--solexa-quals", doc=InputDocumentation(doc="input quals are from GA Pipeline ver. < 1.3")), ToolInput(tag="in_solexa_one_dot_three_quals", input_type=Boolean(optional=True), prefix="--solexa1.3-quals", doc=InputDocumentation(doc="input quals are from GA Pipeline ver. >= 1.3")), ToolInput(tag="in_integer_quals", input_type=Boolean(optional=True), prefix="--integer-quals", doc=InputDocumentation(doc="qualities are given as space-separated integers (not ASCII)")), ToolInput(tag="in_report_endtoend_hits", input_type=Int(optional=True), prefix="-v", doc=InputDocumentation(doc="report end-to-end hits w/ <=v mismatches; ignore qualities")), ToolInput(tag="in_maq_err", input_type=Int(optional=True), prefix="--maqerr", doc=InputDocumentation(doc="max sum of mismatch quals across alignment for -n (def: 70)")), ToolInput(tag="in_seed_len", input_type=Int(optional=True), prefix="--seedlen", doc=InputDocumentation(doc="seed length for -n (default: 28)")), ToolInput(tag="in_no_maq_round", input_type=Boolean(optional=True), prefix="--nomaqround", doc=InputDocumentation(doc="disable Maq-like quality rounding for -n (nearest 10 <= 30)")), ToolInput(tag="in_mini_ns", input_type=Int(optional=True), prefix="--minins", doc=InputDocumentation(doc="minimum insert size for paired-end alignment (default: 0)")), ToolInput(tag="in_max_ins", input_type=Int(optional=True), prefix="--maxins", doc=InputDocumentation(doc="maximum insert size for paired-end alignment (default: 250)")), ToolInput(tag="in_fr", input_type=String(optional=True), prefix="--fr", doc=InputDocumentation(doc="align fw/rev, rev/fw, fw/fw (default: --fr)")), ToolInput(tag="in_no_fw", input_type=Boolean(optional=True), prefix="--nofw", doc=InputDocumentation(doc="do not align to forward/reverse-complement reference strand")), ToolInput(tag="in_max_bts", input_type=Int(optional=True), prefix="--maxbts", doc=InputDocumentation(doc="max # backtracks for -n 2/3 (default: 125, 800 for --best)")), ToolInput(tag="in_pair_tries", input_type=Int(optional=True), prefix="--pairtries", doc=InputDocumentation(doc="max # attempts to find mate for anchor hit (default: 100)")), ToolInput(tag="in_try_hard", input_type=Boolean(optional=True), prefix="--tryhard", doc=InputDocumentation(doc="try hard to find valid alignments, at the expense of speed")), ToolInput(tag="in_chunk_mbs", input_type=Int(optional=True), prefix="--chunkmbs", doc=InputDocumentation(doc="max megabytes of RAM for best-first search frames (def: 64)")), ToolInput(tag="in_reads_per_batch", input_type=Boolean(optional=True), prefix="--reads-per-batch", doc=InputDocumentation(doc="# of reads to read from input file at once (default: 16)")), ToolInput(tag="in_report_int_alignments", input_type=Int(optional=True), prefix="-k", doc=InputDocumentation(doc="report up to <int> good alignments per read (default: 1)")), ToolInput(tag="in_all", input_type=Boolean(optional=True), prefix="--all", doc=InputDocumentation(doc="report all alignments per read (much slower than low -k)")), ToolInput(tag="in_suppress_alignments_exist", input_type=Int(optional=True), prefix="-m", doc=InputDocumentation(doc="suppress all alignments if > <int> exist (def: no limit)")), ToolInput(tag="in_m_reports_random", input_type=Int(optional=True), prefix="-M", doc=InputDocumentation(doc="like -m, but reports 1 random hit (MAPQ=0); requires --best")), ToolInput(tag="in_best", input_type=Boolean(optional=True), prefix="--best", doc=InputDocumentation(doc="hits guaranteed best stratum; ties broken by quality")), ToolInput(tag="in_strata", input_type=Boolean(optional=True), prefix="--strata", doc=InputDocumentation(doc="hits in sub-optimal strata aren't reported (requires --best)")), ToolInput(tag="in_time", input_type=Boolean(optional=True), prefix="--time", doc=InputDocumentation(doc="print wall-clock time taken by search phases")), ToolInput(tag="in_off_base", input_type=Int(optional=True), prefix="--offbase", doc=InputDocumentation(doc="leftmost ref offset = <int> in bowtie output (default: 0)")), ToolInput(tag="in_quiet", input_type=Boolean(optional=True), prefix="--quiet", doc=InputDocumentation(doc="print nothing but the alignments")), ToolInput(tag="in_ref_idx", input_type=Boolean(optional=True), prefix="--refidx", doc=InputDocumentation(doc="refer to ref. seqs by 0-based index rather than name")), ToolInput(tag="in_al", input_type=File(optional=True), prefix="--al", doc=InputDocumentation(doc="write aligned reads/pairs to file(s) <fname>")), ToolInput(tag="in_un", input_type=File(optional=True), prefix="--un", doc=InputDocumentation(doc="write unaligned reads/pairs to file(s) <fname>")), ToolInput(tag="in_no_unal", input_type=Boolean(optional=True), prefix="--no-unal", doc=InputDocumentation(doc="suppress SAM records for unaligned reads")), ToolInput(tag="in_max", input_type=Int(optional=True), prefix="--max", doc=InputDocumentation(doc="write reads/pairs over -m limit to file(s) <fname>")), ToolInput(tag="in_suppress", input_type=String(optional=True), prefix="--suppress", doc=InputDocumentation(doc="suppresses given columns (comma-delim'ed) in default output")), ToolInput(tag="in_full_ref", input_type=Boolean(optional=True), prefix="--fullref", doc=InputDocumentation(doc="write entire ref name (default: only up to 1st space)")), ToolInput(tag="in_snp_phred", input_type=Int(optional=True), prefix="--snpphred", doc=InputDocumentation(doc="Phred penalty for SNP when decoding colorspace (def: 30)")), ToolInput(tag="in_col_cseq", input_type=Boolean(optional=True), prefix="--col-cseq", doc=InputDocumentation(doc="print aligned colorspace seqs as colors, not decoded bases")), ToolInput(tag="in_col_c_qual", input_type=Boolean(optional=True), prefix="--col-cqual", doc=InputDocumentation(doc="print original colorspace quals, not decoded quals")), ToolInput(tag="in_col_keep_ends", input_type=Boolean(optional=True), prefix="--col-keepends", doc=InputDocumentation(doc="keep nucleotides at extreme ends of decoded alignment")), ToolInput(tag="in_sam", input_type=Boolean(optional=True), prefix="--sam", doc=InputDocumentation(doc="write hits in SAM format")), ToolInput(tag="in_mapq", input_type=Int(optional=True), prefix="--mapq", doc=InputDocumentation(doc="default mapping quality (MAPQ) to print for SAM alignments")), ToolInput(tag="in_sam_no_head", input_type=Boolean(optional=True), prefix="--sam-nohead", doc=InputDocumentation(doc="supppress header lines (starting with @) for SAM output")), ToolInput(tag="in_sam_no_sq", input_type=Boolean(optional=True), prefix="--sam-nosq", doc=InputDocumentation(doc="supppress @SQ header lines for SAM output")), ToolInput(tag="in_sam_rg", input_type=String(optional=True), prefix="--sam-RG", doc=InputDocumentation(doc="add <text> (usually 'lab=value') to @RG line of SAM header")), ToolInput(tag="in_off_rate", input_type=Int(optional=True), prefix="--offrate", doc=InputDocumentation(doc="override offrate of index; must be >= index's offrate")), ToolInput(tag="in_threads", input_type=Int(optional=True), prefix="--threads", doc=InputDocumentation(doc="number of alignment threads to launch (default: 1)")), ToolInput(tag="in_mm", input_type=Boolean(optional=True), prefix="--mm", doc=InputDocumentation(doc="use memory-mapped I/O for index; many 'bowtie's can share")), ToolInput(tag="in_shmem", input_type=Boolean(optional=True), prefix="--shmem", doc=InputDocumentation(doc="use shared mem for index; many 'bowtie's can share")), ToolInput(tag="in_seed", input_type=Int(optional=True), prefix="--seed", doc=InputDocumentation(doc="seed for random number generator")), ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="--verbose", doc=InputDocumentation(doc="verbose output (for debugging)"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Bowtie_Align_L_V0_1_0().translate("wdl", allow_empty_container=True)

