from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, String, File, Int, Array

Emapper_Py_V0_1_0 = CommandToolBuilder(tool="emapper.py", base_command=["emapper.py"], inputs=[ToolInput(tag="in_guess_db", input_type=Boolean(optional=True), prefix="--guessdb", doc=InputDocumentation(doc="guess eggnog db based on the provided taxid")), ToolInput(tag="in_database", input_type=Boolean(optional=True), prefix="--database", doc=InputDocumentation(doc="specify the target database for sequence searches.\nChoose among: euk,bact,arch, host:port, or a local\nhmmpressed database")), ToolInput(tag="in_data_dir", input_type=String(optional=True), prefix="--data_dir", doc=InputDocumentation(doc="Directory to use for DATA_PATH.")), ToolInput(tag="in_tax_scope", input_type=Boolean(optional=True), prefix="--tax_scope", doc=InputDocumentation(doc="Fix the taxonomic scope used for annotation, so only\northologs from a particular clade are used for\nfunctional transfer. By default, this is automatically\nadjusted for every query sequence.")), ToolInput(tag="in_target_orthologs", input_type=String(optional=True), prefix="--target_orthologs", doc=InputDocumentation(doc="defines what type of orthologs should be used for\nfunctional transfer")), ToolInput(tag="in_excluded_tax_a", input_type=Boolean(optional=True), prefix="--excluded_taxa", doc=InputDocumentation(doc="(for debugging and benchmark purposes)")), ToolInput(tag="in_go_evidence", input_type=String(optional=True), prefix="--go_evidence", doc=InputDocumentation(doc="Defines what type of GO terms should be used for\nannotation:experimental = Use only terms inferred from\nexperimental evidencenon-electronic = Use only non-\nelectronically curated terms")), ToolInput(tag="in_hmm_max_hits", input_type=Boolean(optional=True), prefix="--hmm_maxhits", doc=InputDocumentation(doc="Max number of hits to report. Default=1")), ToolInput(tag="in_hmm_evalue", input_type=Boolean(optional=True), prefix="--hmm_evalue", doc=InputDocumentation(doc="E-value threshold. Default=0.001")), ToolInput(tag="in_hmm_score", input_type=Boolean(optional=True), prefix="--hmm_score", doc=InputDocumentation(doc="Bit score threshold. Default=20")), ToolInput(tag="in_hmm_max_seq_len", input_type=Boolean(optional=True), prefix="--hmm_maxseqlen", doc=InputDocumentation(doc="Ignore query sequences larger than `maxseqlen`.\nDefault=5000")), ToolInput(tag="in_hmm_q_cov", input_type=Boolean(optional=True), prefix="--hmm_qcov", doc=InputDocumentation(doc="min query coverage (from 0 to 1). Default=(disabled)")), ToolInput(tag="in_fixed_database_size", input_type=Boolean(optional=True), prefix="--Z", doc=InputDocumentation(doc="Fixed database size used in phmmer/hmmscan (allows\ncomparing e-values among databases).\nDefault=40,000,000")), ToolInput(tag="in_dm_nd_db", input_type=File(optional=True), prefix="--dmnd_db", doc=InputDocumentation(doc="Path to DIAMOND-compatible database")), ToolInput(tag="in_matrix", input_type=String(optional=True), prefix="--matrix", doc=InputDocumentation(doc="Scoring matrix")), ToolInput(tag="in_gap_open", input_type=String(optional=True), prefix="--gapopen", doc=InputDocumentation(doc="Gap open penalty")), ToolInput(tag="in_gap_extend", input_type=String(optional=True), prefix="--gapextend", doc=InputDocumentation(doc="Gap extend penalty")), ToolInput(tag="in_query_cover", input_type=Int(optional=True), prefix="--query-cover", doc=InputDocumentation(doc="Report only alignments above the given percentage of\nquery cover. Default=0")), ToolInput(tag="in_subject_cover", input_type=Int(optional=True), prefix="--subject-cover", doc=InputDocumentation(doc="Report only alignments above the given percentage of\nsubject cover. Default=0")), ToolInput(tag="in_seed_ortholog_evalue", input_type=Boolean(optional=True), prefix="--seed_ortholog_evalue", doc=InputDocumentation(doc="Min E-value expected when searching for seed eggNOG\northolog. Applies to phmmer/diamond searches. Queries\nnot having a significant seed orthologs will not be\nannotated. Default=0.001")), ToolInput(tag="in_seed_ortholog_score", input_type=Boolean(optional=True), prefix="--seed_ortholog_score", doc=InputDocumentation(doc="Min bit score expected when searching for seed eggNOG\northolog. Applies to phmmer/diamond searches. Queries\nnot having a significant seed orthologs will not be\nannotated. Default=60")), ToolInput(tag="in_output", input_type=Boolean(optional=True), prefix="--output", doc=InputDocumentation(doc="base name for output files")), ToolInput(tag="in_resume", input_type=File(optional=True), prefix="--resume", doc=InputDocumentation(doc="Resumes a previous execution skipping reported hits in\nthe output file.")), ToolInput(tag="in_override", input_type=Boolean(optional=True), prefix="--override", doc=InputDocumentation(doc="Overwrites output files if they exist.")), ToolInput(tag="in_no_refine", input_type=Boolean(optional=True), prefix="--no_refine", doc=InputDocumentation(doc="Skip hit refinement, reporting only HMM hits.")), ToolInput(tag="in_no_an_not", input_type=Boolean(optional=True), prefix="--no_annot", doc=InputDocumentation(doc="Skip functional annotation, reporting only hits")), ToolInput(tag="in_no_search", input_type=Boolean(optional=True), prefix="--no_search", doc=InputDocumentation(doc="Skip HMM search mapping. Use existing hits file")), ToolInput(tag="in_predict_ortho", input_type=Boolean(optional=True), prefix="--predict_ortho", doc=InputDocumentation(doc="The list of predicted orthologs")), ToolInput(tag="in_report_orthologs", input_type=Boolean(optional=True), prefix="--report_orthologs", doc=InputDocumentation(doc="The list of orthologs used for functional transferred\nare dumped into a separate file")), ToolInput(tag="in_scratch_dir", input_type=File(optional=True), prefix="--scratch_dir", doc=InputDocumentation(doc="Write output files in a temporary scratch dir, move\nthem to final the final output dir when finished.\nSpeed up large computations using network file\nsystems.")), ToolInput(tag="in_output_dir", input_type=Boolean(optional=True), prefix="--output_dir", doc=InputDocumentation(doc="Where output files should be written")), ToolInput(tag="in_temp_dir", input_type=Boolean(optional=True), prefix="--temp_dir", doc=InputDocumentation(doc="Where temporary files are created. Better if this is a\nlocal disk.")), ToolInput(tag="in_no_file_comments", input_type=Boolean(optional=True), prefix="--no_file_comments", doc=InputDocumentation(doc="No header lines nor stats are included in the output")), ToolInput(tag="in_target_tax_a", input_type=Array(t=String(), optional=True), prefix="--target_taxa", doc=InputDocumentation(doc="taxa that will be searched for orthologs")), ToolInput(tag="in_predict_output_format", input_type=String(optional=True), prefix="--predict_output_format", doc=InputDocumentation(doc="Choose the output format among: per_query, per_species\n.Default = per_species")), ToolInput(tag="in_defaulthmmer", input_type=String(optional=True), prefix="-m", doc=InputDocumentation(doc="Default:hmmer")), ToolInput(tag="in_input_fasta_file", input_type=Boolean(optional=True), prefix="-i", doc=InputDocumentation(doc="Input FASTA file containing query sequences")), ToolInput(tag="in_translate", input_type=Boolean(optional=True), prefix="--translate", doc=InputDocumentation(doc="Assume sequences are genes instead of proteins")), ToolInput(tag="in_server_mode", input_type=Boolean(optional=True), prefix="--servermode", doc=InputDocumentation(doc="Loads target database in memory and keeps running in\nserver mode, so another instance of eggnog-mapper can\nconnect to this sever. Auto turns on the --usemem flag")), ToolInput(tag="in_use_mem", input_type=Boolean(optional=True), prefix="--usemem", doc=InputDocumentation(doc="If a local hmmpressed database is provided as target\nusing --db, this flag will allocate the whole database\nin memory using hmmpgmd. Database will be unloaded\nafter execution.")), ToolInput(tag="in_annotate_hits_table", input_type=Boolean(optional=True), prefix="--annotate_hits_table", doc=InputDocumentation(doc="Annotatate TSV formatted table of query->hits. 4\nfields required: query, hit, evalue, score. Implies\n--no_search and --no_refine.\n")), ToolInput(tag="in_files", input_type=String(), position=0, doc=InputDocumentation(doc="--keep_mapping_files  Do not delete temporary mapping files used for"))], outputs=[ToolOutput(tag="out_resume", output_type=File(optional=True), selector=InputSelector(input_to_select="in_resume", type_hint=File()), doc=OutputDocumentation(doc="Resumes a previous execution skipping reported hits in\nthe output file.")), ToolOutput(tag="out_scratch_dir", output_type=File(optional=True), selector=InputSelector(input_to_select="in_scratch_dir", type_hint=File()), doc=OutputDocumentation(doc="Write output files in a temporary scratch dir, move\nthem to final the final output dir when finished.\nSpeed up large computations using network file\nsystems."))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Emapper_Py_V0_1_0().translate("wdl", allow_empty_container=True)

