from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Int, Boolean, String

Bacgwasim_V0_1_0 = CommandToolBuilder(tool="BacGWASim", base_command=["BacGWASim"], inputs=[ToolInput(tag="in_config", input_type=File(optional=True), prefix="--config", doc=InputDocumentation(doc="Path to a config file")), ToolInput(tag="in_output_dir", input_type=File(optional=True), prefix="--output-dir", doc=InputDocumentation(doc="Path to the output directory")), ToolInput(tag="in_num_species", input_type=Int(optional=True), prefix="--num-species", doc=InputDocumentation(doc="Number of samples in the simulated population")), ToolInput(tag="in_genome_length", input_type=Int(optional=True), prefix="--genome-length", doc=InputDocumentation(doc="Length of the genome (bp)")), ToolInput(tag="in_mutation_rate", input_type=Boolean(optional=True), prefix="--mutation-rate", doc=InputDocumentation(doc="]0-1[       Mutation rate")), ToolInput(tag="in_recomb_rate", input_type=Boolean(optional=True), prefix="--recomb-rate", doc=InputDocumentation(doc="]0-1[         Recombination rate")), ToolInput(tag="in_maf", input_type=Boolean(optional=True), prefix="--maf", doc=InputDocumentation(doc="]0-1[                 Minor allele frequency threshold of rare alleles\nto be discarded")), ToolInput(tag="in_num_var", input_type=Int(optional=True), prefix="--num-var", doc=InputDocumentation(doc="Number of simulated variants. If '-1', variant\nnumber will be solely a function of mutation\nrate")), ToolInput(tag="in_random_seed", input_type=Int(optional=True), prefix="--random-seed", doc=InputDocumentation(doc="Random seed for reproducibility of results")), ToolInput(tag="in_phen_type", input_type=String(optional=True), prefix="--phen-type", doc=InputDocumentation(doc="Type of simulated phenotype. 'cc':binary case-\ncontrol, 'quant': quantitative")), ToolInput(tag="in_num_causal_var", input_type=Int(optional=True), prefix="--num-causal-var", doc=InputDocumentation(doc="Number of causal markers")), ToolInput(tag="in_causal_maf_min", input_type=Boolean(optional=True), prefix="--causal-maf-min", doc=InputDocumentation(doc="]0-1[      Minimum Minor Allele Frequency (MAF) of causal")), ToolInput(tag="in_effect_size_odr", input_type=Int(optional=True), prefix="--effect-size-odr", doc=InputDocumentation(doc="Effect sizes of causal markers (.odds ratios)")), ToolInput(tag="in_phen_replication", input_type=Int(optional=True), prefix="--phen-replication", doc=InputDocumentation(doc="Number of phenotype replication sets")), ToolInput(tag="in_heritability", input_type=Boolean(optional=True), prefix="--heritability", doc=InputDocumentation(doc="[0-1]        Heritability of phenotype")), ToolInput(tag="in_disease_prevalence", input_type=Boolean(optional=True), prefix="--disease-prevalence", doc=InputDocumentation(doc="]0-1[  Prevalence of phenotype")), ToolInput(tag="in_case", input_type=Int(optional=True), prefix="--case", doc=InputDocumentation(doc="In case of case-control binary phenotype\nsimulation, number of case and control samples\nmust be defined by 'case' and 'control'\nparameters")), ToolInput(tag="in_control", input_type=Int(optional=True), prefix="--control", doc=InputDocumentation(doc="In case of case-control binary phenotype\nsimulation, number of case and control samples\nmust be defined by 'case' and 'control'\nparameters")), ToolInput(tag="in_plot_ld", input_type=Boolean(optional=True), prefix="--plot-ld", doc=InputDocumentation(doc="Generate the LD plot")), ToolInput(tag="in_snp_limit", input_type=Int(optional=True), prefix="--snp-limit", doc=InputDocumentation(doc="Number of SNPs randomly selected for plotting\nlinkage map (Increasing this number will\nsignificantly increase computation time and\nrequire increasing the java heap size)")), ToolInput(tag="in_heap_size", input_type=Int(optional=True), prefix="--heap-size", doc=InputDocumentation(doc="Java heap_size for ld plot visualization (.mb)")), ToolInput(tag="in_ld_maf", input_type=Boolean(optional=True), prefix="--ld-maf", doc=InputDocumentation(doc="]0-1[              Minimum Minor Allele Frequency of markers for LD\nplotting (Lower this values, it is more\ndifficult to estimate accurate r2 values between\npairs of markers leading to more noisy plot)")), ToolInput(tag="in_cores", input_type=Int(optional=True), prefix="--cores", doc=InputDocumentation(doc="Number of cores available for computations")), ToolInput(tag="in_latency_wait", input_type=Int(optional=True), prefix="--latency-wait", doc=InputDocumentation(doc="Time to wait (in sec) after a job to ensure that\nall files are present\n"))], outputs=[ToolOutput(tag="out_output_dir", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output_dir", type_hint=File()), doc=OutputDocumentation(doc="Path to the output directory"))], container="quay.io/biocontainers/bacgwasim:2.0.0--py_1", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Bacgwasim_V0_1_0().translate("wdl")

