from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, File, String

Blast2Lca_V0_1_0 = CommandToolBuilder(tool="blast2lca", base_command=["blast2lca"], inputs=[ToolInput(tag="in_format", input_type=Boolean(optional=True), prefix="--format", doc=InputDocumentation(doc="[string]                BLAST format. Default value: Unknown Legal values: Unknown, DAA, BlastText, BlastXML, BlastTab, LastMAF, RapSearch2Aln, IlluminaReporter, RDPAssignmentDetails, RDPStandalone, Mothur, SAM, References_as_FastA")), ToolInput(tag="in_mode", input_type=Boolean(optional=True), prefix="--mode", doc=InputDocumentation(doc="[string]                  BLAST mode. Default value: Unknown Legal values: Unknown, BlastN, BlastP, BlastX, Classifier")), ToolInput(tag="in_kegg_output", input_type=File(optional=True), prefix="--keggOutput", doc=InputDocumentation(doc="[string]           KEGG output file. Default value: foo-kegg.txt")), ToolInput(tag="in_kegg", input_type=Boolean(optional=True), prefix="--kegg", doc=InputDocumentation(doc="Map reads to KEGG KOs?. Default value: false.")), ToolInput(tag="in_show_ranks", input_type=Boolean(optional=True), prefix="--showRanks", doc=InputDocumentation(doc="Show taxonomic ranks. Default value: true.")), ToolInput(tag="in_official_ranks_only", input_type=Boolean(optional=True), prefix="--officialRanksOnly", doc=InputDocumentation(doc="Report only taxa that have an official rank. Default value: true.")), ToolInput(tag="in_show_tax_ids", input_type=Boolean(optional=True), prefix="--showTaxIds", doc=InputDocumentation(doc="Report taxon ids rather than taxon names. Default value: false.")), ToolInput(tag="in_max_expected", input_type=Boolean(optional=True), prefix="--maxExpected", doc=InputDocumentation(doc="[number]          Max expected. Default value: 0.01.")), ToolInput(tag="in_top_percent", input_type=Boolean(optional=True), prefix="--topPercent", doc=InputDocumentation(doc="[number]          Top percent. Default value: 10.0.")), ToolInput(tag="in_min_percent_identity", input_type=Boolean(optional=True), prefix="--minPercentIdentity", doc=InputDocumentation(doc="[number]   Min percent identity. Default value: 0.0.")), ToolInput(tag="in_max_kegg_per_read", input_type=Boolean(optional=True), prefix="--maxKeggPerRead", doc=InputDocumentation(doc="[number]       Maximum number of KEGG assignments to report for a read. Default value: 4.")), ToolInput(tag="in_parse_tax_on_names", input_type=Boolean(optional=True), prefix="--parseTaxonNames", doc=InputDocumentation(doc="Parse taxon names. Default value: true.")), ToolInput(tag="in_map_db", input_type=Boolean(optional=True), prefix="--mapDB", doc=InputDocumentation(doc="[string]               MEGAN mapping db (file megan-map.db).")), ToolInput(tag="in_acc_two_tax_a", input_type=Boolean(optional=True), prefix="--acc2taxa", doc=InputDocumentation(doc="[string]            Accession-to-Taxonomy mapping file.")), ToolInput(tag="in_syn_two_tax_a", input_type=Boolean(optional=True), prefix="--syn2taxa", doc=InputDocumentation(doc="[string]            Synonyms-to-Taxonomy mapping file.")), ToolInput(tag="in_acc_two_kegg", input_type=Boolean(optional=True), prefix="--acc2kegg", doc=InputDocumentation(doc="[string]         Accession-to-KEGG mapping file.")), ToolInput(tag="in_syn_two_kegg", input_type=Boolean(optional=True), prefix="--syn2kegg", doc=InputDocumentation(doc="[string]         Synonyms-to-KEGG mapping file.")), ToolInput(tag="in_first_word_is_accession", input_type=Boolean(optional=True), prefix="--firstWordIsAccession", doc=InputDocumentation(doc="First word in reference header is accession number (set to 'true' for NCBI-nr downloaded Sep 2016 or later). Default value: true.")), ToolInput(tag="in_accession_tags", input_type=Boolean(optional=True), prefix="--accessionTags", doc=InputDocumentation(doc="[string(s)]   List of accession tags. Default value(s): 'gb|' 'ref|'.")), ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="--verbose", doc=InputDocumentation(doc="Echo commandline options and be verbose. Default value: false.")), ToolInput(tag="in_input", input_type=String(), position=0, doc=InputDocumentation(doc="-i, --input [string]                 Input BLAST file. Mandatory option.")), ToolInput(tag="in_output", input_type=String(), position=1, doc=InputDocumentation(doc="-o, --output [string]                Taxonomy output file. Default value: foo-taxonomy.txt")), ToolInput(tag="in_parameters", input_type=String(), position=0, doc=InputDocumentation(doc="-ms, --minScore [number]             Min score. Default value: 50.0.")), ToolInput(tag="in_true_dot", input_type=String(), position=0, doc=InputDocumentation(doc="Classification support:"))], outputs=[ToolOutput(tag="out_kegg_output", output_type=File(optional=True), selector=InputSelector(input_to_select="in_kegg_output", type_hint=File()), doc=OutputDocumentation(doc="[string]           KEGG output file. Default value: foo-kegg.txt"))], container="quay.io/biocontainers/megan:6.21.2--h9ee0642_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Blast2Lca_V0_1_0().translate("wdl")

