from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import String, Boolean, File, Int

Pre_Py_V0_1_0 = CommandToolBuilder(tool="pre.py", base_command=["pre.py"], inputs=[ToolInput(tag="in_location", input_type=String(optional=True), prefix="--location", doc=InputDocumentation(doc="Comma-separated list of locations [use naming after\npreprocessing], when not specified will use whole VCF.")), ToolInput(tag="in_pass_only", input_type=Boolean(optional=True), prefix="--pass-only", doc=InputDocumentation(doc="Keep only PASS variants.")), ToolInput(tag="in_filters_only", input_type=String(optional=True), prefix="--filters-only", doc=InputDocumentation(doc="Specify a comma-separated list of filters to apply (by\ndefault all filters are ignored / passed on.")), ToolInput(tag="in_restrict_regions", input_type=File(optional=True), prefix="--restrict-regions", doc=InputDocumentation(doc="Restrict analysis to given (sparse) regions (using -R\nin bcftools).")), ToolInput(tag="in_target_regions", input_type=File(optional=True), prefix="--target-regions", doc=InputDocumentation(doc="Restrict analysis to given (dense) regions (using -T\nin bcftools).")), ToolInput(tag="in_left_shift", input_type=Boolean(optional=True), prefix="--leftshift", doc=InputDocumentation(doc="Left-shift variants safely.")), ToolInput(tag="in_no_left_shift", input_type=Boolean(optional=True), prefix="--no-leftshift", doc=InputDocumentation(doc="Do not left-shift variants safely.")), ToolInput(tag="in_decompose", input_type=Boolean(optional=True), prefix="--decompose", doc=InputDocumentation(doc="Decompose variants into primitives. This results in\nmore granular counts.")), ToolInput(tag="in_no_decompose", input_type=Boolean(optional=True), prefix="--no-decompose", doc=InputDocumentation(doc="Do not decompose variants into primitives.")), ToolInput(tag="in_bcf_tools_norm", input_type=Boolean(optional=True), prefix="--bcftools-norm", doc=InputDocumentation(doc="Enable preprocessing through bcftools norm -c x -D\n(requires external preprocessing to be switched on).")), ToolInput(tag="in_fix_chr", input_type=Boolean(optional=True), prefix="--fixchr", doc=InputDocumentation(doc="Add chr prefix to VCF records where necessary\n(default: auto, attempt to match reference).")), ToolInput(tag="in_no_fix_chr", input_type=Boolean(optional=True), prefix="--no-fixchr", doc=InputDocumentation(doc="Do not add chr prefix to VCF records (default: auto,\nattempt to match reference).")), ToolInput(tag="in_bcf", input_type=Boolean(optional=True), prefix="--bcf", doc=InputDocumentation(doc="Use BCF internally. This is the default when the input\nfile is in BCF format already. Using BCF can speed up\ntemp file access, but may fail for VCF files that have\nbroken headers or records that don't comply with the\nheader.")), ToolInput(tag="in_somatic", input_type=Boolean(optional=True), prefix="--somatic", doc=InputDocumentation(doc="Assume the input file is a somatic call file and\nsquash all columns into one, putting all FORMATs into\nINFO + use half genotypes (see also --set-gt). This\nwill replace all sample columns and replace them with\na single one.")), ToolInput(tag="in_set_gt", input_type=String(optional=True), prefix="--set-gt", doc=InputDocumentation(doc="This is used to treat Strelka somatic files Possible\nvalues for this parameter: half / hemi / het / hom /\nhalf to assign one of the following genotypes to the\nresulting sample: 1 | 0/1 | 1/1 | ./1. This will\nreplace all sample columns and replace them with a\nsingle one.")), ToolInput(tag="in_filter_non_ref", input_type=Boolean(optional=True), prefix="--filter-nonref", doc=InputDocumentation(doc="Remove any variants genotyped as <NON_REF>.")), ToolInput(tag="in_convert_gvc_f_to_vcf", input_type=Boolean(optional=True), prefix="--convert-gvcf-to-vcf", doc=InputDocumentation(doc="Convert the input set from genome VCF format to a VCF\nbefore processing.")), ToolInput(tag="in_gender", input_type=String(optional=True), prefix="--gender", doc=InputDocumentation(doc="Specify sex. This determines how haploid calls on chrX\nget treated: for male samples, all non-ref calls (in\nthe truthset only when running through hap.py) are\ngiven a 1/1 genotype.")), ToolInput(tag="in_reference", input_type=File(optional=True), prefix="--reference", doc=InputDocumentation(doc="Specify a reference file.")), ToolInput(tag="in_window_size", input_type=Int(optional=True), prefix="--window-size", doc=InputDocumentation(doc="Preprocessing window size (variants further apart than\nthat size are not expected to interfere).")), ToolInput(tag="in_threads", input_type=Int(optional=True), prefix="--threads", doc=InputDocumentation(doc="Number of threads to use.")), ToolInput(tag="in_log_file", input_type=File(optional=True), prefix="--logfile", doc=InputDocumentation(doc="Write logging information into file rather than to")), ToolInput(tag="in_quiet", input_type=Boolean(optional=True), prefix="--quiet", doc=InputDocumentation(doc="Set logging level to output errors only.")), ToolInput(tag="in_input", input_type=String(), position=0, doc=InputDocumentation(doc="VCF file to process.")), ToolInput(tag="in_output", input_type=String(), position=1, doc=InputDocumentation(doc="Output filename.")), ToolInput(tag="in_stderr", input_type=String(), position=0, doc=InputDocumentation(doc="--verbose             Raise logging level from warning to info."))], outputs=[], container="quay.io/biocontainers/hap.py:0.3.14--py27h5c5a3ab_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Pre_Py_V0_1_0().translate("wdl")

