class: CommandLineTool
id: ../../../isONclust.cwl
inputs:
- id: in_fast_q
  doc: 'Path to consensus fastq file(s) (default: False)'
  type: File
  inputBinding:
    prefix: --fastq
- id: in_fl_nc
  doc: "The flnc reads generated by the isoseq3 algorithm (BAM\nfile) (default: False)"
  type: File
  inputBinding:
    prefix: --flnc
- id: in_ccs
  doc: 'Path to consensus BAM file(s) (default: False)'
  type: File
  inputBinding:
    prefix: --ccs
- id: in_number_cores_allocated
  doc: 'Number of cores allocated for clustering (default: 8)'
  type: long
  inputBinding:
    prefix: --t
- id: in_debugging_prints_status
  doc: "For debugging, prints status of clustering and\nminimizer database every p\
    \ reads processed. (default:\n10000)"
  type: long
  inputBinding:
    prefix: --d
- id: in_filters_reads_default
  doc: "Filters reads with average phred quality value under\nthis number (default\
    \ = 7.0). (default: 7.0)"
  type: long
  inputBinding:
    prefix: --q
- id: in_ont
  doc: 'Clustering of ONT transcript reads. (default: False)'
  type: boolean
  inputBinding:
    prefix: --ont
- id: in_iso_seq
  doc: 'Clustering of PacBio Iso-Seq reads. (default: False)'
  type: boolean
  inputBinding:
    prefix: --isoseq
- id: in_use_old_sorted_file
  doc: "Using already existing sorted file if present in\nspecified output directory.\
    \ (default: False)"
  type: File
  inputBinding:
    prefix: --use_old_sorted_file
- id: in_consensus
  doc: "After clustering, (1) run spoa on all clusters, (2)\ndetect reverse complements,\
    \ (3) run medaka. (default:\nFalse)"
  type: boolean
  inputBinding:
    prefix: --consensus
- id: in_abundance_ratio
  doc: "Threshold for --consensus algorithm. Consider only\nclusters larger than a\
    \ fraction of number of total\nreads (default 0.1) (default: 0.1)"
  type: long
  inputBinding:
    prefix: --abundance_ratio
- id: in_rc_identity_threshold
  doc: "Threshold for --consensus algorithm. Define a reverse\ncomplement if identity\
    \ is over this threshold (default\n0.9) (default: 0.9)"
  type: double
  inputBinding:
    prefix: --rc_identity_threshold
- id: in_medaka
  doc: 'Run final medaka polishing algorithm. (default: False)'
  type: boolean
  inputBinding:
    prefix: --medaka
- id: in_kmer_size_default
  doc: 'Kmer size (default: 15)'
  type: long
  inputBinding:
    prefix: --k
- id: in_window_size_default
  doc: 'Window size (default: 50)'
  type: long
  inputBinding:
    prefix: --w
- id: in_min_shared
  doc: "Minmum number of minimizers shared between read and\ncluster (default: 5)"
  type: long
  inputBinding:
    prefix: --min_shared
- id: in_mapped_threshold
  doc: "Minmum mapped fraction of read to be included in\ncluster. The density of\
    \ minimizers to classify a\nregion as mapped depends on quality of the read.\n\
    (default: 0.7)"
  type: double
  inputBinding:
    prefix: --mapped_threshold
- id: in_aligned_threshold
  doc: "Minmum aligned fraction of read to be included in\ncluster. Aligned identity\
    \ depends on the quality of\nthe read. (default: 0.4)"
  type: double
  inputBinding:
    prefix: --aligned_threshold
- id: in_batch_type
  doc: "In parrallel mode, how to split the reads into chunks\n\"total_nt\", \"nr_reads\"\
    , or \"weighted\" (default:\ntotal_nt) (default: total_nt)"
  type: string
  inputBinding:
    prefix: --batch_type
- id: in_min_fraction
  doc: "Minmum fraction of minimizers shared compared to best\nhit, in order to continue\
    \ mapping. (default: 0.8)"
  type: long
  inputBinding:
    prefix: --min_fraction
- id: in_min_prob_no_hits
  doc: "Minimum probability for i consecutive minimizers to be\ndifferent between\
    \ read and representative and still\nconsidered as mapped region, under assumption\
    \ that\nthey come from the same transcript (depends on read\nquality). (default:\
    \ 0.1)"
  type: long
  inputBinding:
    prefix: --min_prob_no_hits
- id: in_out_folder
  doc: "A fasta file with transcripts that are shared between\nsamples and have perfect\
    \ illumina support. (default:\nNone)\n"
  type: Directory
  inputBinding:
    prefix: --outfolder
outputs:
- id: out_stdout
  doc: Standard output stream
  type: stdout
- id: out_use_old_sorted_file
  doc: "Using already existing sorted file if present in\nspecified output directory.\
    \ (default: False)"
  type: File
  outputBinding:
    glob: $(inputs.in_use_old_sorted_file)
- id: out_out_folder
  doc: "A fasta file with transcripts that are shared between\nsamples and have perfect\
    \ illumina support. (default:\nNone)\n"
  type: Directory
  outputBinding:
    glob: $(inputs.in_out_folder)
cwlVersion: v1.1
baseCommand:
- isONclust
