from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, File, Int, String

Phyloflash_Makedb_Pl_V0_1_0 = CommandToolBuilder(tool="phyloFlash_makedb.pl", base_command=["phyloFlash_makedb.pl"], inputs=[ToolInput(tag="in_remote", input_type=Boolean(optional=True), prefix="--remote", doc=InputDocumentation(doc="Download databases via FTP")), ToolInput(tag="in_silva_file", input_type=File(optional=True), prefix="--silva_file", doc=InputDocumentation(doc="/to/silva_db\nPath to local copy of SILVA database file. Ignored if --remote\nflag is used.\nThis should be the Fasta-formatted SILVA SSURef file, clustered\nat 99% identity, with SILVA taxonomy strings in file header, and\nsequences truncated to SSU gene boundaries. The file name should\nbe in the form\nSILVA_[Release]_SSURef_Nr99_tax_silva_trunc.fasta.gz (release\n132 and before) or\nSILVA_[Release]_SSURef_NR99_tax_silva_trunc.fasta.gz (from\nrelease 138 onwards)")), ToolInput(tag="in_univ_ec_file", input_type=File(optional=True), prefix="--univec_file", doc=InputDocumentation(doc="/to/univec_db\nPath to local copy of Univec database file. Ignored if --remote\nflag is used.")), ToolInput(tag="in_emir_ge", input_type=Boolean(optional=True), prefix="--emirge", doc=InputDocumentation(doc="Index database with Bowtie v1 for Emirge. Requires bowtie-build\nto be in path.\nDefault: Yes (turn off with --noemirge)")), ToolInput(tag="in_sort_merna", input_type=Boolean(optional=True), prefix="--sortmerna", doc=InputDocumentation(doc="Index database for Sortmerna, if you wish to use it as an\nalternative to BBmap for extracting rRNA reads from the read\nfile. Requires indexdb_rna to be in path.\nDefault: No (--nosortmerna).")), ToolInput(tag="in_keep", input_type=Boolean(optional=True), prefix="--keep", doc=InputDocumentation(doc="Do not delete intermediary files")), ToolInput(tag="in_overwrite", input_type=Boolean(optional=True), prefix="--overwrite", doc=InputDocumentation(doc="Overwrite files if already present. Files are not overwritten by\ndefault, allowing you to restart the DB build process if it was\ninterrupted (but you will have to do find and delete corrupted\nfiles manually).\nDefault: No ('--nooverwrite')")), ToolInput(tag="in_ref_minlength", input_type=Boolean(optional=True), prefix="--ref_minlength", doc=InputDocumentation(doc="Minimum length of database sequences to keep after trimming of\nvectors and contaminant sequences. Default of 800 bp is about\n50% for SSU rRNA gene length, but should be changed if a custom\ndatabase is used.\nDefault: 800")), ToolInput(tag="in_cpus", input_type=Int(optional=True), prefix="--CPUs", doc=InputDocumentation(doc="Number of processors to use\nDefault: All available")), ToolInput(tag="in_mem", input_type=Int(optional=True), prefix="--mem", doc=InputDocumentation(doc="Memory limit (in Gb) for indexing tools. At least 10 is")), ToolInput(tag="in_log", input_type=File(optional=True), prefix="--log", doc=InputDocumentation(doc="Write phyloFlash_makedb.pl log to a file.\nDefault: None")), ToolInput(tag="in_check_env", input_type=Boolean(optional=True), prefix="--check_env", doc=InputDocumentation(doc="Check that required dependencies are available in your path. If\nspecifying optional tools --sortmerna and --emirge, put the\n--check_env argument at the end of the command.")), ToolInput(tag="in_man", input_type=Boolean(optional=True), prefix="--man", doc=InputDocumentation(doc="Full manual page")), ToolInput(tag="in_recommended_dot", input_type=String(), position=0, doc=InputDocumentation(doc="Default: 10"))], outputs=[], container="quay.io/biocontainers/phyloflash:3.4--0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Phyloflash_Makedb_Pl_V0_1_0().translate("wdl")

