from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, File, Float, Int, String

Mendelscan_Score_V0_1_0 = CommandToolBuilder(tool="mendelscan_score", base_command=["mendelscan", "score"], inputs=[ToolInput(tag="in_vep_file", input_type=Boolean(optional=True), prefix="--vep-file", doc=InputDocumentation(doc="Variant annotation in VEP format")), ToolInput(tag="in_ped_file", input_type=Boolean(optional=True), prefix="--ped-file", doc=InputDocumentation(doc="Pedigree file in 6-column tab-delimited format")), ToolInput(tag="in_gene_file", input_type=Boolean(optional=True), prefix="--gene-file", doc=InputDocumentation(doc="A list of gene expression values for tissue of interest")), ToolInput(tag="in_output_file", input_type=File(optional=True), prefix="--output-file", doc=InputDocumentation(doc="Output file to contain human-friendly scored variants")), ToolInput(tag="in_output_vcf", input_type=File(optional=True), prefix="--output-vcf", doc=InputDocumentation(doc="Output file to contain scored variants in VCF format")), ToolInput(tag="in_inheritance", input_type=Boolean(optional=True), prefix="--inheritance", doc=InputDocumentation(doc="Presumed model of inheritance: dominant, recessive, x-linked [dominant]")), ToolInput(tag="in_seg_score_case_ref", input_type=Boolean(optional=True), prefix="--seg-score-case-ref", doc=InputDocumentation(doc="A case sample was called reference/wild-type (0.50/0.10)")), ToolInput(tag="in_seg_score_case_het", input_type=Boolean(optional=True), prefix="--seg-score-case-het", doc=InputDocumentation(doc="A case sample was called heterozygous (NA/0.50)")), ToolInput(tag="in_seg_score_case_hom", input_type=Boolean(optional=True), prefix="--seg-score-case-hom", doc=InputDocumentation(doc="A case sample was called homozygous variant (0.80/NA)")), ToolInput(tag="in_seg_score_control_het", input_type=Float(optional=True), prefix="--seg-score-control-het", doc=InputDocumentation(doc="case sample was called heterozygous (0.10/NA)")), ToolInput(tag="in_seg_score_control_hom", input_type=Float(optional=True), prefix="--seg-score-control-hom", doc=InputDocumentation(doc="case sample was called homozygous variant (0.01/0.10)")), ToolInput(tag="in_max_vaf_for_ref", input_type=Boolean(optional=True), prefix="--max-vaf-for-ref", doc=InputDocumentation(doc="Maximum non-ref (variant) allele frequency at ref site to count as ref [0.05]")), ToolInput(tag="in_min_vaf_to_recall", input_type=Boolean(optional=True), prefix="--min-vaf-to-recall", doc=InputDocumentation(doc="Minimum VAF at which a reference genotype will be considered het. To disable recall, set to 1.01 [0.20]")), ToolInput(tag="in_pop_score_novel", input_type=Boolean(optional=True), prefix="--pop-score-novel", doc=InputDocumentation(doc="Variant is not present in dbSNP according to VCF (1.00)")), ToolInput(tag="in_pop_score_mutation", input_type=Boolean(optional=True), prefix="--pop-score-mutation", doc=InputDocumentation(doc="Variant from mutation (OMIM) or locus-specific databases (0.95)")), ToolInput(tag="in_pop_score_known", input_type=Boolean(optional=True), prefix="--pop-score-known", doc=InputDocumentation(doc="Variant known to dbSNP but no mutation or GMAF info (0.60)")), ToolInput(tag="in_pop_score_rare", input_type=Boolean(optional=True), prefix="--pop-score-rare", doc=InputDocumentation(doc="Variant in dbSNP with GMAF < 0.01 (0.20)")), ToolInput(tag="in_pop_score_uncommon", input_type=Boolean(optional=True), prefix="--pop-score-uncommon", doc=InputDocumentation(doc="Variant in dbSNP with GMAF 0.01-0.05 (0.02)")), ToolInput(tag="in_pop_score_common", input_type=Boolean(optional=True), prefix="--pop-score-common", doc=InputDocumentation(doc="Variant in dbSNP with GMAF >= 0.05 (0.01)")), ToolInput(tag="in_an_no_score_one", input_type=Boolean(optional=True), prefix="--anno-score-1", doc=InputDocumentation(doc="Score for intergenic mutations [0.01]")), ToolInput(tag="in_an_no_score_two", input_type=Boolean(optional=True), prefix="--anno-score-2", doc=InputDocumentation(doc="Score for intronic mutations [0.01]")), ToolInput(tag="in_an_no_score_three", input_type=Boolean(optional=True), prefix="--anno-score-3", doc=InputDocumentation(doc="Score for downstream mutations [0.01]")), ToolInput(tag="in_an_no_score_four", input_type=Boolean(optional=True), prefix="--anno-score-4", doc=InputDocumentation(doc="Score for upstream mutations [0.01]")), ToolInput(tag="in_an_no_score_five", input_type=Boolean(optional=True), prefix="--anno-score-5", doc=InputDocumentation(doc="Score for UTR mutations [0.01]")), ToolInput(tag="in_an_no_score_six", input_type=Boolean(optional=True), prefix="--anno-score-6", doc=InputDocumentation(doc="Score for ncRNA mutations [0.01]")), ToolInput(tag="in_an_no_score_seven", input_type=Boolean(optional=True), prefix="--anno-score-7", doc=InputDocumentation(doc="Score for miRNA mutations [0.01]")), ToolInput(tag="in_an_no_score_eight", input_type=Boolean(optional=True), prefix="--anno-score-8", doc=InputDocumentation(doc="Score for synonymous coding mutations [0.05]")), ToolInput(tag="in_an_no_score_nine", input_type=Boolean(optional=True), prefix="--anno-score-9", doc=InputDocumentation(doc="Score for splice region mutations [0.20]")), ToolInput(tag="in_an_no_score_one_zero", input_type=Int(optional=True), prefix="--anno-score-10", doc=InputDocumentation(doc="for nonstop mutations [1.00]")), ToolInput(tag="in_an_no_score_one_one", input_type=Int(optional=True), prefix="--anno-score-11", doc=InputDocumentation(doc="for missense mutations not predicted damaging [0.80]")), ToolInput(tag="in_an_no_score_one_two", input_type=Int(optional=True), prefix="--anno-score-12", doc=InputDocumentation(doc="for missense mutations damaging by 1/3 algorithms [0.95]")), ToolInput(tag="in_an_no_score_one_three", input_type=Int(optional=True), prefix="--anno-score-13", doc=InputDocumentation(doc="for missense mutations damaging by 2/3 algorithms [0.95]")), ToolInput(tag="in_an_no_score_one_four", input_type=Int(optional=True), prefix="--anno-score-14", doc=InputDocumentation(doc="for missense mutations damaging by 3/3 algorithms [0.95]")), ToolInput(tag="in_an_no_score_one_five", input_type=Int(optional=True), prefix="--anno-score-15", doc=InputDocumentation(doc="for essential splice site mutations [1.00]")), ToolInput(tag="in_an_no_score_one_six", input_type=Int(optional=True), prefix="--anno-score-16", doc=InputDocumentation(doc="for frameshift mutations [1.00]")), ToolInput(tag="in_an_no_score_one_seven", input_type=Int(optional=True), prefix="--anno-score-17", doc=InputDocumentation(doc="for nonsense mutations [1.00]")), ToolInput(tag="in_jar", input_type=String(optional=True), prefix="-jar", doc=InputDocumentation(doc="")), ToolInput(tag="in_java", input_type=String(), position=0, doc=InputDocumentation(doc="")), ToolInput(tag="in_score", input_type=String(), position=1, doc=InputDocumentation(doc="")), ToolInput(tag="in_vcf", input_type=String(optional=True), position=2, doc=InputDocumentation(doc=""))], outputs=[ToolOutput(tag="out_output_file", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output_file", type_hint=File()), doc=OutputDocumentation(doc="Output file to contain human-friendly scored variants")), ToolOutput(tag="out_output_vcf", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output_vcf", type_hint=File()), doc=OutputDocumentation(doc="Output file to contain scored variants in VCF format"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Mendelscan_Score_V0_1_0().translate("wdl", allow_empty_container=True)

