from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Int, String, Directory, Boolean, Float

Crispresso_V0_1_0 = CommandToolBuilder(tool="CRISPResso", base_command=["CRISPResso"], inputs=[ToolInput(tag="in_analysis", input_type=Int(optional=True), prefix="-Analysis", doc=InputDocumentation(doc="CRISPR/Cas9 outcomes from deep sequencing data-\n)")), ToolInput(tag="in_fast_q_r_one", input_type=Int(optional=True), prefix="--fastq_r1", doc=InputDocumentation(doc="First fastq file (default: Fastq filename)")), ToolInput(tag="in_fast_q_r_two", input_type=Int(optional=True), prefix="--fastq_r2", doc=InputDocumentation(doc="Second fastq file for paired end reads (default: )")), ToolInput(tag="in_amplicon_seq", input_type=String(optional=True), prefix="--amplicon_seq", doc=InputDocumentation(doc="Amplicon Sequence (default: None)")), ToolInput(tag="in_guide_seq", input_type=Int(optional=True), prefix="--guide_seq", doc=InputDocumentation(doc="sgRNA sequence, if more than one, please separate by\ncomma/s. Note that the sgRNA needs to be input as the\nguide RNA sequence (usually 20 nt) immediately\nadjacent to but not including the PAM sequence (5' of\nNGG for SpCas9). If the PAM is found on the opposite\nstrand with respect to the Amplicon Sequence, ensure\nthe sgRNA sequence is also found on the opposite\nstrand. The CRISPResso convention is to depict the\nexpected cleavage position using the value of the\nparameter cleavage_offset nt 3' from the end of the\nguide. In addition, the use of alternate nucleases to\nSpCas9 is supported. For example, if using the Cpf1\nsystem, enter the sequence (usually 20 nt) immediately\n3' of the PAM sequence and explicitly set the\ncleavage_offset parameter to 1, since the default\nsetting of -3 is suitable only for SpCas9. (default: )")), ToolInput(tag="in_expected_hdr_amplicon_seq", input_type=String(optional=True), prefix="--expected_hdr_amplicon_seq", doc=InputDocumentation(doc="Amplicon sequence expected after HDR (default: )")), ToolInput(tag="in_donor_seq", input_type=String(optional=True), prefix="--donor_seq", doc=InputDocumentation(doc="Donor Sequence. This optional input comprises a\nsubsequence of the expected HDR amplicon to be\nhighlighted in plots. (default: )")), ToolInput(tag="in_coding_seq", input_type=String(optional=True), prefix="--coding_seq", doc=InputDocumentation(doc="Subsequence/s of the amplicon sequence covering one or\nmore coding sequences for the frameshift analysis.If\nmore than one (for example, split by intron/s), please\nseparate by comma. (default: )")), ToolInput(tag="in_min_average_read_quality", input_type=Int(optional=True), prefix="--min_average_read_quality", doc=InputDocumentation(doc="Minimum average quality score (phred33) to keep a read\n(default: 0)")), ToolInput(tag="in_min_single_bp_quality", input_type=Int(optional=True), prefix="--min_single_bp_quality", doc=InputDocumentation(doc="Minimum single bp score (phred33) to keep a read\n(default: 0)")), ToolInput(tag="in_min_identity_score", input_type=Int(optional=True), prefix="--min_identity_score", doc=InputDocumentation(doc="Minimum identity score for the alignment (default:\n60.0)")), ToolInput(tag="in_name", input_type=String(optional=True), prefix="--name", doc=InputDocumentation(doc="Output name (default: )")), ToolInput(tag="in_split_paired_end", input_type=Directory(optional=True), prefix="--split_paired_end", doc=InputDocumentation(doc="Splits a single fastq file contating paired end reads\nin two files before running CRISPResso (default:\nFalse)")), ToolInput(tag="in_trim_sequences", input_type=Boolean(optional=True), prefix="--trim_sequences", doc=InputDocumentation(doc="Enable the trimming of Illumina adapters with\nTrimmomatic (default: False)")), ToolInput(tag="in_trim_mo_matic_options_string", input_type=Float(optional=True), prefix="--trimmomatic_options_string", doc=InputDocumentation(doc="Override options for Trimmomatic (default:\nILLUMINACLIP:/usr/local/lib/python2.7/site-\npackages/CRISPResso/data/NexteraPE-\nPE.fa:0:90:10:0:true MINLEN:40)")), ToolInput(tag="in_min_paired_end_reads_overlap", input_type=Int(optional=True), prefix="--min_paired_end_reads_overlap", doc=InputDocumentation(doc="Parameter for the FLASH read merging step. Minimum\nrequired overlap length between two reads to provide a\nconfident overlap. (default: 4)")), ToolInput(tag="in_max_paired_end_reads_overlap", input_type=Int(optional=True), prefix="--max_paired_end_reads_overlap", doc=InputDocumentation(doc="Parameter for the FLASH merging step. Maximum overlap\nlength expected in approximately 90% of read pairs.\nPlease see the FLASH manual for more information.\n(default: 100)")), ToolInput(tag="in_hide_mutations_outside_window_nhej", input_type=Boolean(optional=True), prefix="--hide_mutations_outside_window_NHEJ", doc=InputDocumentation(doc="This parameter allows to visualize only the mutations\noverlapping the cleavage site and used to classify a\nread as NHEJ. This parameter has no effect on the\nquanitification of the NHEJ. It may be helpful to mask\na pre-existing and known mutations or sequencing\nerrors outside the window used for quantification of\nNHEJ events. (default: False)")), ToolInput(tag="in_window_around_sg_rna", input_type=Int(optional=True), prefix="--window_around_sgrna", doc=InputDocumentation(doc="Window(s) in bp around the cleavage position (half on\non each side) as determined by the provide guide RNA\nsequence to quantify the indels. Any indels outside\nthis window are excluded. A value of 0 disables this\nfilter. (default: 1)")), ToolInput(tag="in_cleavage_offset", input_type=Int(optional=True), prefix="--cleavage_offset", doc=InputDocumentation(doc="Cleavage offset to use within respect to the 3' end of\nthe provided sgRNA sequence. Remember that the sgRNA\nsequence must be entered without the PAM. The default\nis -3 and is suitable for the SpCas9 system. For\nalternate nucleases, other cleavage offsets may be\nappropriate, for example, if using Cpf1 this parameter\nwould be set to 1. (default: -3)")), ToolInput(tag="in_exclude_bp_from_left", input_type=Int(optional=True), prefix="--exclude_bp_from_left", doc=InputDocumentation(doc="Exclude bp from the left side of the amplicon sequence\nfor the quantification of the indels (default: 15)")), ToolInput(tag="in_exclude_bp_from_right", input_type=Int(optional=True), prefix="--exclude_bp_from_right", doc=InputDocumentation(doc="Exclude bp from the right side of the amplicon\nsequence for the quantification of the indels\n(default: 15)")), ToolInput(tag="in_hdr_perfect_alignment_threshold", input_type=Float(optional=True), prefix="--hdr_perfect_alignment_threshold", doc=InputDocumentation(doc="Sequence homology % for an HDR occurrence (default:\n98.0)")), ToolInput(tag="in_ignore_substitutions", input_type=Boolean(optional=True), prefix="--ignore_substitutions", doc=InputDocumentation(doc="Ignore substitutions events for the quantification and\nvisualization (default: False)")), ToolInput(tag="in_ignore_insertions", input_type=Boolean(optional=True), prefix="--ignore_insertions", doc=InputDocumentation(doc="Ignore insertions events for the quantification and\nvisualization (default: False)")), ToolInput(tag="in_ignore_deletions", input_type=Boolean(optional=True), prefix="--ignore_deletions", doc=InputDocumentation(doc="Ignore deletions events for the quantification and\nvisualization (default: False)")), ToolInput(tag="in_needle_options_string", input_type=Float(optional=True), prefix="--needle_options_string", doc=InputDocumentation(doc="Override options for the Needle aligner (default:\n-gapopen=10 -gapextend=0.5 -awidth3=5000)")), ToolInput(tag="in_keep_intermediate", input_type=Boolean(optional=True), prefix="--keep_intermediate", doc=InputDocumentation(doc="Keep all the intermediate files (default: False)")), ToolInput(tag="in_dump", input_type=Boolean(optional=True), prefix="--dump", doc=InputDocumentation(doc="Dump numpy arrays and pandas dataframes to file for\ndebugging purposes (default: False)")), ToolInput(tag="in_save_also_png", input_type=Boolean(optional=True), prefix="--save_also_png", doc=InputDocumentation(doc="Save also .png images additionally to .pdf files\n(default: False)")), ToolInput(tag="in_n_processes", input_type=Int(optional=True), prefix="--n_processes", doc=InputDocumentation(doc="Specify the number of processes to use for the\nquantification. Please use with caution since\nincreasing this parameter will increase significantly\nthe memory required to run CRISPResso. (default: 1)")), ToolInput(tag="in_offset_around_cut_to_plot", input_type=Int(optional=True), prefix="--offset_around_cut_to_plot", doc=InputDocumentation(doc="Offset to use to summarize alleles around the cut site\nin the alleles table plot. (default: 20)")), ToolInput(tag="in_min_frequency_alleles_around_cut_to_plot", input_type=Int(optional=True), prefix="--min_frequency_alleles_around_cut_to_plot", doc=InputDocumentation(doc="Minimum % reads required to report an allele in the\nalleles table plot. (default: 0.2)")), ToolInput(tag="in_max_rows_alleles_around_cut_to_plot", input_type=Int(optional=True), prefix="--max_rows_alleles_around_cut_to_plot", doc=InputDocumentation(doc="Maximum number of rows to report in the alleles table\nplot. (default: 50)")), ToolInput(tag="in_debug", input_type=Boolean(optional=True), prefix="--debug", doc=InputDocumentation(doc="Print stack trace on error. (default: False)"))], outputs=[ToolOutput(tag="out_split_paired_end", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_split_paired_end", type_hint=File()), doc=OutputDocumentation(doc="Splits a single fastq file contating paired end reads\nin two files before running CRISPResso (default:\nFalse)"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Crispresso_V0_1_0().translate("wdl", allow_empty_container=True)

