class: CommandLineTool
id: anvi_get_sequences_for_hmm_hits.cwl
inputs:
- id: in_contigs_db
  doc: "Anvi'o contigs database generated by 'anvi-gen-\ncontigs'"
  type: string
  inputBinding:
    prefix: --contigs-db
- id: in_profile_db
  doc: Anvi'o profile database
  type: string
  inputBinding:
    prefix: --profile-db
- id: in_collection_name
  doc: Collection name.
  type: string
  inputBinding:
    prefix: --collection-name
- id: in_bin_id
  doc: Bin name you are interested in.
  type: string
  inputBinding:
    prefix: --bin-id
- id: in_bin_ids_file
  doc: "Text file for bins (each line should be a unique bin\nid)."
  type: File
  inputBinding:
    prefix: --bin-ids-file
- id: in_external_genomes
  doc: "A two-column TAB-delimited flat text file that lists\nanvi'o contigs databases.\
    \ The first item in the header\nline should read 'name', and the second should\
    \ read\n'contigs_db_path'. Each line in the file should\ndescribe a single entry,\
    \ where the first column is the\nname of the genome (or MAG), and the second column\
    \ is\nthe anvi'o contigs database generated for this genome."
  type: File
  inputBinding:
    prefix: --external-genomes
- id: in_internal_genomes
  doc: "A five-column TAB-delimited flat text file. The header\nline must contain\
    \ these columns: 'name', 'bin_id',\n'collection_id', 'profile_db_path', 'contigs_db_path'.\n\
    Each line should list a single entry, where 'name' can\nbe any name to describe\
    \ the anvi'o bin identified as\n'bin_id' that is stored in a collection."
  type: File
  inputBinding:
    prefix: --internal-genomes
- id: in_hmm_sources
  doc: "NAME\nGet sequences for a specific list of HMM sources. You\ncan list one\
    \ or more sources by separating them from\neach other with a comma character (i.e.,\
    \ '--hmm-\nsources source_1,source_2,source_3'). If you would\nlike to see a list\
    \ of available sources in the contigs\ndatabase, run this program with '--list-hmm-sources'\n\
    flag."
  type: long
  inputBinding:
    prefix: --hmm-sources
- id: in_gene_names
  doc: "HIT NAME\nGet sequences only for a specific gene name. Each name\nshould be\
    \ separated from each other by a comma\ncharacter. For instance, if you want to\
    \ get back only\nRecA and Ribosomal_L27, you can type '--gene-names\nRecA,Ribosomal_L27',\
    \ and you will get any and every\nhit that matches these names in any source.\
    \ If you\nwould like to see a list of available gene names, you\ncan use '--list-available-gene-names'\
    \ flag."
  type: long
  inputBinding:
    prefix: --gene-names
- id: in_list_hmm_sources
  doc: "List available HMM sources in the contigs database and\nquit."
  type: boolean
  inputBinding:
    prefix: --list-hmm-sources
- id: in_list_available_gene_names
  doc: "List available gene names in HMM sources selection and\nquit."
  type: boolean
  inputBinding:
    prefix: --list-available-gene-names
- id: in_output_file
  doc: File path to store results.
  type: File
  inputBinding:
    prefix: --output-file
- id: in_no_wrap
  doc: Do not be wrap sequences nicely in the output file.
  type: File
  inputBinding:
    prefix: --no-wrap
- id: in_get_aa_sequences
  doc: Store amino acid sequences instead.
  type: boolean
  inputBinding:
    prefix: --get-aa-sequences
- id: in_concatenate_genes
  doc: "Concatenate output genes in the same order to create a\nmulti-gene alignment\
    \ output that is suitable for\nphylogenomic analyses."
  type: boolean
  inputBinding:
    prefix: --concatenate-genes
- id: in_partition_file
  doc: "Some commonly used software for phylogenetic analyses\n(e.g., IQ-TREE, RAxML,\
    \ etc) allow users to\nspecify/test different substitution models for each\ngene\
    \ of a concatenated multiple sequence alignments.\nFor this, they use a special\
    \ file format called a\n'partition file', which indicates the site for each\n\
    gene in the alignment. You can use this parameter to\ndeclare an output path for\
    \ anvi'o to report a NEXUS\nformat partition file in addition to your FASTA output\n\
    (requested by Massimiliano Molari in #1333)."
  type: File
  inputBinding:
    prefix: --partition-file
- id: in_max_num_genes_missing_from_bin
  doc: "This filter removes bins (or genomes) from your\nanalysis. If you have a list\
    \ of gene names, you can\nuse this parameter to omit any bin (or external\ngenome)\
    \ that is missing more than a number of genes\nyou desire. For instance, if you\
    \ have 100 genome bins,\nand you are interested in working with 5 ribosomal\n\
    proteins, you can use '--max-num-genes-missing-from-\nbin 4' to remove the bins\
    \ that are missing more than 4\nof those 5 genes. This is especially useful for\n\
    phylogenomic analyses. Parameter 0 will remove any bin\nthat is missing any of\
    \ the genes."
  type: long
  inputBinding:
    prefix: --max-num-genes-missing-from-bin
- id: in_min_num_bins_gene_occurs
  doc: "This filter removes genes from your analysis. Let's\nassume you have 100 bins\
    \ to get sequences for HMM\nhits. If you want to work only with genes among all\n\
    the hits that occur in at least X number of bins, and\ndiscard the rest of them,\
    \ you can use this flag. If\nyou say '--min-num-bins-gene-occurs 90', each gene\
    \ in\nthe analysis will be required at least to appear in 90\ngenomes. If a gene\
    \ occurs in less than that number of\ngenomes, it simply will not be reported.\
    \ This is\nespecially useful for phylogenomic analyses, where you\nmay want to\
    \ only focus on genes that are prevalent\nacross the set of genomes you wish to\
    \ analyze."
  type: long
  inputBinding:
    prefix: --min-num-bins-gene-occurs
- id: in_align_with
  doc: "The multiple sequence alignment program to use when\nmultiple sequence alignment\
    \ is necessary. To see all\navailable options, use the flag `--list-aligners`."
  type: string
  inputBinding:
    prefix: --align-with
- id: in_separator
  doc: "A word that will be used to sepaate concatenated gene\nsequences from each\
    \ other (IF you are using this\nprogram with `--concatenate-genes` flag). The\
    \ default\nis \"XXX\" for amino acid sequences, and \"NNN\" for DNA\nsequences"
  type: string
  inputBinding:
    prefix: --separator
- id: in_return_best_hit
  doc: "A bin may contain more than one hit for a gene name in\na given HMM source.\
    \ For instance, there may be\nmultiple RecA hits in a genome bin from Campbell\
    \ et\nal.. Using this flag, will go through all of the gene\nnames that appear\
    \ multiple times, and remove all but\nthe one with the lowest e-value. Good for\
    \ whenever you\nreally need to get only a single copy of single-copy\ncore genes\
    \ from a genome bin."
  type: boolean
  inputBinding:
    prefix: --return-best-hit
- id: in_just_do_it
  doc: "Don't bother me with questions or warnings, just do\nit.\n"
  type: boolean
  inputBinding:
    prefix: --just-do-it
outputs:
- id: out_stdout
  doc: Standard output stream
  type: stdout
- id: out_output_file
  doc: File path to store results.
  type: File
  outputBinding:
    glob: $(inputs.in_output_file)
- id: out_no_wrap
  doc: Do not be wrap sequences nicely in the output file.
  type: File
  outputBinding:
    glob: $(inputs.in_no_wrap)
- id: out_partition_file
  doc: "Some commonly used software for phylogenetic analyses\n(e.g., IQ-TREE, RAxML,\
    \ etc) allow users to\nspecify/test different substitution models for each\ngene\
    \ of a concatenated multiple sequence alignments.\nFor this, they use a special\
    \ file format called a\n'partition file', which indicates the site for each\n\
    gene in the alignment. You can use this parameter to\ndeclare an output path for\
    \ anvi'o to report a NEXUS\nformat partition file in addition to your FASTA output\n\
    (requested by Massimiliano Molari in #1333)."
  type: File
  outputBinding:
    glob: $(inputs.in_partition_file)
cwlVersion: v1.1
baseCommand:
- anvi-get-sequences-for-hmm-hits
