from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Int, Boolean, String, Directory

Run_Panacota_Py_Annotate_V0_1_0 = CommandToolBuilder(tool="run_panacota.py_annotate", base_command=["run_panacota.py", "annotate"], inputs=[ToolInput(tag="in_path_folder_containing", input_type=File(optional=True), prefix="-d", doc=InputDocumentation(doc="Path to folder containing all multifasta genome files")), ToolInput(tag="in_path_folder_annotated", input_type=File(optional=True), prefix="-r", doc=InputDocumentation(doc="Path to folder where output annotated genomes must be")), ToolInput(tag="in_file_containing_list", input_type=File(optional=True), prefix="-l", doc=InputDocumentation(doc="File containing the list of genome filenames to\nannotate (1 genome per line). Each genome must be in\nmulti-fasta format. You can specify the species name\n(4 characters) you want to give to some genome, as\nwell as the download (or any other reason) date of\nyour choice. Format 'genome_name :: name.date'. name\nand date are optional. See doc for more information on\nthis file format. If you want to run this module from\n'prepare_module' results, use '--info' instead.")), ToolInput(tag="in_choose_name_genomes", input_type=Int(optional=True), prefix="-n", doc=InputDocumentation(doc="Choose a name for your annotated genomes. This name\nshould contain 4 alphanumeric characters. Generally,\nthey correspond to the 2 first letters of genus, and 2\nfirst letters of species, e.g. ESCO for Escherichia\nColi.")), ToolInput(tag="in_add_option_you_want_only", input_type=Boolean(optional=True), prefix="-Q", doc=InputDocumentation(doc="Add this option if you want only to do quality control\non your genomes (cut at 5N if asked, calculate L90 and\nnumber of contigs and plot their distributions). This\nallows you to check which genomes would be annotated\nwith the given parameters, and to modify those\nparameters if you want, before you launch the\nannotation and formatting steps.")), ToolInput(tag="in_info", input_type=Int(optional=True), prefix="--info", doc=InputDocumentation(doc="If you already ran the 'prepare' data module, or\nalready calculated yourself the L90 and number of\ncontigs for each genome, you can give this\ninformation, to go directly to annotation and\nformatting steps. This file contains at least 4\ncolumns, tab separated, with the following headers:\n'to_annotate', 'gsize', 'nb_conts', 'L90'. Any other\ncolumn will be ignored.")), ToolInput(tag="in_prodigal", input_type=Boolean(optional=True), prefix="--prodigal", doc=InputDocumentation(doc="Add this option if you only want syntactical\nannotation, given by prodigal, and not functional\nannotation requiring prokka and is slower.")), ToolInput(tag="in_small", input_type=Boolean(optional=True), prefix="--small", doc=InputDocumentation(doc="If you use Prodigal to annotate genomes, if you\nsequences are too small (less than 20000 characters),\nit cannot annotate them with the default options. Add\nthis option to use 'meta' procedure.")), ToolInput(tag="in_l_nine_zero", input_type=Int(optional=True), prefix="--l90", doc=InputDocumentation(doc="Maximum value of L90 allowed to keep a genome. Default\nis 100.")), ToolInput(tag="in_nb_cont", input_type=Int(optional=True), prefix="--nbcont", doc=InputDocumentation(doc="Maximum number of contigs allowed to keep a genome.\nDefault is 999.")), ToolInput(tag="in_cut_n", input_type=Int(optional=True), prefix="--cutn", doc=InputDocumentation(doc="By default, each genome will be cut into new contigs\nwhen at least 5 'N' in a row are found in its\nsequence. If you don't want to cut genomes into new\ncontigs when there are rows of 'N', put 0 to this\noption. If you want to cut from a different number of\n'N' occurrences, put this value to this option.")), ToolInput(tag="in_date", input_type=Int(optional=True), prefix="--date", doc=InputDocumentation(doc="Specify the date (MMYY) to give to your annotated\ngenomes. By default, will give today's date. The only\nrequirement on the given date is that it is 4\ncharacters long. You can use letters if you want. But\nthe common way is to use 4 digits, corresponding to\nMMYY.")), ToolInput(tag="in_tmp", input_type=String(optional=True), prefix="--tmp", doc=InputDocumentation(doc="Specify where the temporary files (sequence split by\nrows of 'N', sequence with new contig names etc.) must\nbe saved. By default, it will be saved in your\nresult_directory/tmp_files.")), ToolInput(tag="in_an_not_dir", input_type=Directory(optional=True), prefix="--annot_dir", doc=InputDocumentation(doc="Specify in which directory the prokka/prodigal output\nfiles (1 folder per genome, called\n<genome_name>-[prokka, Prodigal]Res) must be saved. By\ndefault, they are saved in the same directory as your\ntemporary files (see --tmp option to change it).")), ToolInput(tag="in_force", input_type=Boolean(optional=True), prefix="--force", doc=InputDocumentation(doc="Force run: Add this option if you want to (re)run\nannotation and formatting steps for all genomes even\nif their result folder (for annotation step) or files\n(for format step) already exist: override existing\nresults. Without this option, if there already are\nresults in the given result folder, the program stops.\nIf there are no results, but prokka/prodigal folder\nalready exists, prokka/prodigal won't run again, and\nthe formating step will use the already existing\nfolder if correct, or skip the genome if there are\nproblems in prokka/prodigal folder.")), ToolInput(tag="in_threads", input_type=Int(optional=True), prefix="--threads", doc=InputDocumentation(doc="Specify how many threads can be used (default=1)")), ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="--verbose", doc=InputDocumentation(doc="Increase verbosity in stdout/stderr.")), ToolInput(tag="in_quiet", input_type=Boolean(optional=True), prefix="--quiet", doc=InputDocumentation(doc="Do not display anything to stdout/stderr. log files\nwill still be created.")), ToolInput(tag="in_saved", input_type=String(), position=0, doc=InputDocumentation(doc="Optional arguments:"))], outputs=[ToolOutput(tag="out_path_folder_annotated", output_type=File(optional=True), selector=InputSelector(input_to_select="in_path_folder_annotated", type_hint=File()), doc=OutputDocumentation(doc="Path to folder where output annotated genomes must be")), ToolOutput(tag="out_an_not_dir", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_an_not_dir", type_hint=File()), doc=OutputDocumentation(doc="Specify in which directory the prokka/prodigal output\nfiles (1 folder per genome, called\n<genome_name>-[prokka, Prodigal]Res) must be saved. By\ndefault, they are saved in the same directory as your\ntemporary files (see --tmp option to change it)."))], container="quay.io/biocontainers/panacota:1.2.0--py_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Run_Panacota_Py_Annotate_V0_1_0().translate("wdl")

