from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Boolean, String

Fastqc_V0_1_0 = CommandToolBuilder(tool="fastqc", base_command=["fastqc"], inputs=[ToolInput(tag="in_outdir", input_type=File(optional=True), prefix="--outdir", doc=InputDocumentation(doc="Create all output files in the specified output directory.\nPlease note that this directory must exist as the program\nwill not create it.  If this option is not set then the\noutput file for each sequence file is created in the same\ndirectory as the sequence file which was processed.")), ToolInput(tag="in_casa_va", input_type=Boolean(optional=True), prefix="--casava", doc=InputDocumentation(doc="Files come from raw casava output. Files in the same sample\ngroup (differing only by the group number) will be analysed\nas a set rather than individually. Sequences with the filter\nflag set in the header will be excluded from the analysis.\nFiles must have the same names given to them by casava\n(including being gzipped and ending with .gz) otherwise they\nwon't be grouped together correctly.")), ToolInput(tag="in_nano", input_type=File(optional=True), prefix="--nano", doc=InputDocumentation(doc="Files come from nanopore sequences and are in fast5 format. In\nthis mode you can pass in directories to process and the program\nwill take in all fast5 files within those directories and produce\na single output file from the sequences found in all files.")), ToolInput(tag="in_no_filter", input_type=Boolean(optional=True), prefix="--nofilter", doc=InputDocumentation(doc="If running with --casava then don't remove read flagged by\ncasava as poor quality when performing the QC analysis.")), ToolInput(tag="in_extract", input_type=File(optional=True), prefix="--extract", doc=InputDocumentation(doc="If set then the zipped output file will be uncompressed in\nthe same directory after it has been created.  By default\nthis option will be set if fastqc is run in non-interactive\nmode.")), ToolInput(tag="in_java", input_type=Boolean(optional=True), prefix="--java", doc=InputDocumentation(doc="Provides the full path to the java binary you want to use to\nlaunch fastqc. If not supplied then java is assumed to be in\nyour path.")), ToolInput(tag="in_no_extract", input_type=File(optional=True), prefix="--noextract", doc=InputDocumentation(doc="Do not uncompress the output file after creating it.  You\nshould set this option if you do not wish to uncompress\nthe output when running in non-interactive mode.")), ToolInput(tag="in_nogroup", input_type=Boolean(optional=True), prefix="--nogroup", doc=InputDocumentation(doc="Disable grouping of bases for reads >50bp. All reports will\nshow data for every base in the read.  WARNING: Using this\noption will cause fastqc to crash and burn if you use it on\nreally long reads, and your plots may end up a ridiculous size.\nYou have been warned!")), ToolInput(tag="in_min_length", input_type=Boolean(optional=True), prefix="--min_length", doc=InputDocumentation(doc="Sets an artificial lower limit on the length of the sequence\nto be shown in the report.  As long as you set this to a value\ngreater or equal to your longest read length then this will be\nthe sequence length used to create your read groups.  This can\nbe useful for making directly comaparable statistics from\ndatasets with somewhat variable read lengths.")), ToolInput(tag="in_format", input_type=Boolean(optional=True), prefix="--format", doc=InputDocumentation(doc="Bypasses the normal sequence file format detection and\nforces the program to use the specified format.  Valid\nformats are bam,sam,bam_mapped,sam_mapped and fastq")), ToolInput(tag="in_threads", input_type=Boolean(optional=True), prefix="--threads", doc=InputDocumentation(doc="Specifies the number of files which can be processed")), ToolInput(tag="in_c", input_type=Boolean(optional=True), prefix="-c", doc=InputDocumentation(doc="Specifies a non-default file which contains the list of")), ToolInput(tag="in_contaminants", input_type=Boolean(optional=True), prefix="--contaminants", doc=InputDocumentation(doc="contaminants to screen overrepresented sequences against.\nThe file must contain sets of named contaminants in the\nform name[tab]sequence.  Lines prefixed with a hash will\nbe ignored.")), ToolInput(tag="in_a", input_type=Boolean(optional=True), prefix="-a", doc=InputDocumentation(doc="Specifies a non-default file which contains the list of")), ToolInput(tag="in_adapters", input_type=Boolean(optional=True), prefix="--adapters", doc=InputDocumentation(doc="adapter sequences which will be explicity searched against\nthe library. The file must contain sets of named adapters\nin the form name[tab]sequence.  Lines prefixed with a hash\nwill be ignored.")), ToolInput(tag="in_specifies_file_a", input_type=Boolean(optional=True), prefix="-l", doc=InputDocumentation(doc="Specifies a non-default file which contains a set of criteria")), ToolInput(tag="in_limits", input_type=File(optional=True), prefix="--limits", doc=InputDocumentation(doc="which will be used to determine the warn/error limits for the\nvarious modules.  This file can also be used to selectively\nremove some modules from the output all together.  The format\nneeds to mirror the default limits.txt file found in the\nConfiguration folder.")), ToolInput(tag="in_km_ers", input_type=Boolean(optional=True), prefix="--kmers", doc=InputDocumentation(doc="Specifies the length of Kmer to look for in the Kmer content\nmodule. Specified Kmer length must be between 2 and 10. Default\nlength is 7 if not specified.")), ToolInput(tag="in_quiet", input_type=Boolean(optional=True), prefix="--quiet", doc=InputDocumentation(doc="Supress all progress messages on stdout and only report errors.")), ToolInput(tag="in_dir", input_type=Boolean(optional=True), prefix="--dir", doc=InputDocumentation(doc="Selects a directory to be used for temporary files written when\ngenerating report images. Defaults to system temp directory if\nnot specified.")), ToolInput(tag="in_data_dot", input_type=String(), position=0, doc=InputDocumentation(doc="If no files to process are specified on the command line then the program")), ToolInput(tag="in_required_dot", input_type=String(), position=0, doc=InputDocumentation(doc="In this mode it is suitable for inclusion into a standardised")), ToolInput(tag="in_simultaneously_dot", input_type=String(), position=0, doc=InputDocumentation(doc="Each thread will be allocated 250MB of"))], outputs=[ToolOutput(tag="out_outdir", output_type=File(optional=True), selector=InputSelector(input_to_select="in_outdir", type_hint=File()), doc=OutputDocumentation(doc="Create all output files in the specified output directory.\nPlease note that this directory must exist as the program\nwill not create it.  If this option is not set then the\noutput file for each sequence file is created in the same\ndirectory as the sequence file which was processed.")), ToolOutput(tag="out_nano", output_type=File(optional=True), selector=InputSelector(input_to_select="in_nano", type_hint=File()), doc=OutputDocumentation(doc="Files come from nanopore sequences and are in fast5 format. In\nthis mode you can pass in directories to process and the program\nwill take in all fast5 files within those directories and produce\na single output file from the sequences found in all files.")), ToolOutput(tag="out_extract", output_type=File(optional=True), selector=InputSelector(input_to_select="in_extract", type_hint=File()), doc=OutputDocumentation(doc="If set then the zipped output file will be uncompressed in\nthe same directory after it has been created.  By default\nthis option will be set if fastqc is run in non-interactive\nmode.")), ToolOutput(tag="out_no_extract", output_type=File(optional=True), selector=InputSelector(input_to_select="in_no_extract", type_hint=File()), doc=OutputDocumentation(doc="Do not uncompress the output file after creating it.  You\nshould set this option if you do not wish to uncompress\nthe output when running in non-interactive mode.")), ToolOutput(tag="out_limits", output_type=File(optional=True), selector=InputSelector(input_to_select="in_limits", type_hint=File()), doc=OutputDocumentation(doc="which will be used to determine the warn/error limits for the\nvarious modules.  This file can also be used to selectively\nremove some modules from the output all together.  The format\nneeds to mirror the default limits.txt file found in the\nConfiguration folder."))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Fastqc_V0_1_0().translate("wdl", allow_empty_container=True)

