from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Array, File, Boolean, Int, Float, String, Directory

Recognicer_Df_V0_1_0 = CommandToolBuilder(tool="recognicer_df", base_command=["recognicer_df"], inputs=[ToolInput(tag="in_treatment_file", input_type=Array(t=File(), optional=True), prefix="--treatment_file", doc=InputDocumentation(doc="Names of the sample file you wish to run RECOGNICER\non. This can either be the relative or the absolute\npath of the file. Must be in BED or BAM format.")), ToolInput(tag="in_control_file", input_type=Boolean(optional=True), prefix="--control_file", doc=InputDocumentation(doc="[CONTROL_FILE [CONTROL_FILE ...]], -c [CONTROL_FILE [CONTROL_FILE ...]]\nName of the control library in BED or BAM format. This\ncan either be the relative or the absolute path of the\nfile. If you wish to run RECOGNICER without a control\nlibrary, simply do not enter the file.")), ToolInput(tag="in_species", input_type=Int(optional=True), prefix="--species", doc=InputDocumentation(doc="The species/genome used (ex: hg38)")), ToolInput(tag="in_redundancy_threshold", input_type=Int(optional=True), prefix="--redundancy_threshold", doc=InputDocumentation(doc="The number of copies of indentical reads allowed in a\nlibrary. Default value is 1")), ToolInput(tag="in_window_size", input_type=Int(optional=True), prefix="--window_size", doc=InputDocumentation(doc="Resolution of RECOGNICER. Default value is 200 (bp)")), ToolInput(tag="in_fragment_size", input_type=Int(optional=True), prefix="--fragment_size", doc=InputDocumentation(doc="The amount of shift from the beginning of a read to\nthe center of the DNA fragment represented by the\nread. Default value is 150 (bp).")), ToolInput(tag="in_effective_genome_fraction", input_type=Int(optional=True), prefix="--effective_genome_fraction", doc=InputDocumentation(doc="Effective genome as fraction of the genome size.\nDefault value is 0.74")), ToolInput(tag="in_false_discovery_rate", input_type=Float(optional=True), prefix="--false_discovery_rate", doc=InputDocumentation(doc="Remove all islands with an false_discovery_rate below\ncutoff. Default value is 0.01.")), ToolInput(tag="in_false_discovery_rate_df", input_type=String(optional=True), prefix="--false_discovery_rate_df", doc=InputDocumentation(doc="Cutoff for identification of significant changes been\nwild-type library and knockout library. Only provide a\nvalue when comparing two libraries.")), ToolInput(tag="in_output_directory", input_type=Directory(optional=True), prefix="--output_directory", doc=InputDocumentation(doc="Path of the directory in which results will be stored.\nDefault path is the current path")), ToolInput(tag="in_step_size", input_type=Int(optional=True), prefix="--step_size", doc=InputDocumentation(doc="Step Size: the number of windows in one graining unit.\nUsed for RECOGNICER algorithm")), ToolInput(tag="in_step_score", input_type=Int(optional=True), prefix="--step_score", doc=InputDocumentation(doc="Step Score: The minimum number of positive elements in\nthe graining unit to call the unit positive. Used for\nRECOGNICER algorithm")), ToolInput(tag="in_cpu", input_type=Int(optional=True), prefix="--cpu", doc=InputDocumentation(doc="CPU Core Count: The number of CPU cores RECOGNICER\nprogram will use when executing multi-processing\ntasks. Ideal core count is the species' number of\nchromosomes. Default value is the maximum number of\ncores avaiable in the system.")), ToolInput(tag="in_significant_reads", input_type=File(optional=True), prefix="--significant_reads", doc=InputDocumentation(doc="Output Significant Reads: Enter '--significant_reads'\nto have SICER produce a BED file of treatment reads\nfiltered by significant islands and WIG file of\nfiltered reads binned into windows")), ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="--verbose", doc=InputDocumentation(doc="increase console output verbosity")), ToolInput(tag="in_h", input_type=String(optional=True), prefix="-h", doc=InputDocumentation(doc="")), ToolInput(tag="in_use", input_type=String(), position=0, doc=InputDocumentation(doc="")), ToolInput(tag="in_more", input_type=String(), position=1, doc=InputDocumentation(doc="")), ToolInput(tag="in_information", input_type=String(), position=2, doc=InputDocumentation(doc=""))], outputs=[ToolOutput(tag="out_output_directory", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_output_directory", type_hint=File()), doc=OutputDocumentation(doc="Path of the directory in which results will be stored.\nDefault path is the current path")), ToolOutput(tag="out_significant_reads", output_type=File(optional=True), selector=InputSelector(input_to_select="in_significant_reads", type_hint=File()), doc=OutputDocumentation(doc="Output Significant Reads: Enter '--significant_reads'\nto have SICER produce a BED file of treatment reads\nfiltered by significant islands and WIG file of\nfiltered reads binned into windows"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Recognicer_Df_V0_1_0().translate("wdl", allow_empty_container=True)

