from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import String, Int, Boolean, Directory

Flash_V0_1_0 = CommandToolBuilder(tool="flash", base_command=["flash"], inputs=[ToolInput(tag="in_interleaved_input", input_type=String(optional=True), prefix="--interleaved-input", doc=InputDocumentation(doc=") or tab-delimited (see the --tab-delimited-input")), ToolInput(tag="in_phred_offset", input_type=Int(optional=True), prefix="--phred-offset", doc=InputDocumentation(doc="The smallest ASCII value of the characters used to\nrepresent quality values of bases in FASTQ files.\nIt should be set to either 33, which corresponds\nto the later Illumina platforms and Sanger\nplatforms, or 64, which corresponds to the\nearlier Illumina platforms.  Default: 33.")), ToolInput(tag="in_min_overlap", input_type=Int(optional=True), prefix="--min-overlap", doc=InputDocumentation(doc="The minimum required overlap length between two\nreads to provide a confident overlap.  Default:\n10bp.")), ToolInput(tag="in_max_overlap", input_type=Int(optional=True), prefix="--max-overlap", doc=InputDocumentation(doc="Maximum overlap length expected in approximately\n90% of read pairs.  It is by default set to 65bp,\nwhich works well for 100bp reads generated from a\n180bp library, assuming a normal distribution of\nfragment lengths.  Overlaps longer than the maximum\noverlap parameter are still considered as good\noverlaps, but the mismatch density (explained below)\nis calculated over the first max_overlap bases in\nthe overlapped region rather than the entire\noverlap.  Default: 65bp, or calculated from the\nspecified read length, fragment length, and fragment\nlength standard deviation.")), ToolInput(tag="in_max_mismatch_density", input_type=Int(optional=True), prefix="--max-mismatch-density", doc=InputDocumentation(doc="Maximum allowed ratio between the number of\nmismatched base pairs and the overlap length.\nTwo reads will not be combined with a given overlap\nif that overlap results in a mismatched base density\nhigher than this value.  Note: Any occurence of an\n'N' in either read is ignored and not counted\ntowards the mismatches or overlap length.  Our\nexperimental results suggest that higher values of\nthe maximum mismatch density yield larger\nnumbers of correctly merged read pairs but at\nthe expense of higher numbers of incorrectly\nmerged read pairs.  Default: 0.25.")), ToolInput(tag="in_allow_out_ies", input_type=Boolean(optional=True), prefix="--allow-outies", doc=InputDocumentation(doc="Also try combining read pairs in the 'outie'\norientation, e.g.\nRead 1: <-----------\nRead 2:       ------------>\nas opposed to only the 'innie' orientation, e.g.\nRead 1:       <------------\nRead 2: ----------->\nFLASH uses the same parameters when trying each\norientation.  If a read pair can be combined in\nboth 'innie' and 'outie' orientations, the\nbetter-fitting one will be chosen using the same\nscoring algorithm that FLASH normally uses.\nThis option also causes extra .innie and .outie\nhistogram files to be produced.")), ToolInput(tag="in_fragment_len_stddev", input_type=Int(optional=True), prefix="--fragment-len-stddev", doc=InputDocumentation(doc="Average read length, fragment length, and fragment\nstandard deviation.  These are convenience parameters\nonly, as they are only used for calculating the\nmaximum overlap (--max-overlap) parameter.\nThe maximum overlap is calculated as the overlap of\naverage-length reads from an average-size fragment\nplus 2.5 times the fragment length standard\ndeviation.  The default values are -r 100, -f 180,\nand -s 18, so this works out to a maximum overlap of\n65 bp.  If --max-overlap is specified, then the\nspecified value overrides the calculated value.\nIf you do not know the standard deviation of the\nfragment library, you can probably assume that the\nstandard deviation is 10% of the average fragment\nlength.")), ToolInput(tag="in_cap_mismatch_quals", input_type=Boolean(optional=True), prefix="--cap-mismatch-quals", doc=InputDocumentation(doc="Cap quality scores assigned at mismatch locations\nto 2.  This was the default behavior in FLASH v1.2.7\nand earlier.  Later versions will instead calculate\nsuch scores as max(|q1 - q2|, 2); that is, the\nabsolute value of the difference in quality scores,\nbut at least 2.  Essentially, the new behavior\nprevents a low quality base call that is likely a\nsequencing error from significantly bringing down\nthe quality of a high quality, likely correct base\ncall.")), ToolInput(tag="in_interleaved_output", input_type=Boolean(optional=True), prefix="--interleaved-output", doc=InputDocumentation(doc="Write the uncombined pairs in interleaved FASTQ")), ToolInput(tag="in_tab_delimited_input", input_type=Boolean(optional=True), prefix="--tab-delimited-input", doc=InputDocumentation(doc="Assume the input is in tab-delimited format\nrather than FASTQ, in the format described below in\n'--tab-delimited-output'.  In this mode you should\nprovide a single input file, each line of which must\ncontain either a read pair (5 fields) or a single\nread (3 fields).  FLASH will try to combine the read\npairs.  Single reads will be written to the output\nfile as-is if also using --tab-delimited-output;\notherwise they will be ignored.  Note that you may\nspecify '-' as the input file to read the\ntab-delimited data from standard input.")), ToolInput(tag="in_tab_delimited_output", input_type=Boolean(optional=True), prefix="--tab-delimited-output", doc=InputDocumentation(doc="Write output in tab-delimited format (not FASTQ).\nEach line will contain either a combined pair in the\nformat 'tag <tab> seq <tab> qual' or an uncombined\npair in the format 'tag <tab> seq_1 <tab> qual_1\n<tab> seq_2 <tab> qual_2'.")), ToolInput(tag="in_output_prefix", input_type=String(optional=True), prefix="--output-prefix", doc=InputDocumentation(doc="Prefix of output files.  Default: 'out'.")), ToolInput(tag="in_output_directory", input_type=Directory(optional=True), prefix="--output-directory", doc=InputDocumentation(doc="Path to directory for output files.  Default:\ncurrent working directory.")), ToolInput(tag="in_to_stdout", input_type=Boolean(optional=True), prefix="--to-stdout", doc=InputDocumentation(doc="Write the combined reads to standard output.  In\nthis mode, with FASTQ output (the default) the\nuncombined reads are discarded.  With tab-delimited\noutput, uncombined reads are included in the\ntab-delimited data written to standard output.\nIn both cases, histogram files are not written,\nand informational messages are sent to standard\nerror rather than to standard output.")), ToolInput(tag="in_compress", input_type=Boolean(optional=True), prefix="--compress", doc=InputDocumentation(doc="Compress the output files directly with zlib,\nusing the gzip container format.  Similar to\nspecifying --compress-prog=gzip and --suffix=gz,\nbut may be slightly faster.")), ToolInput(tag="in_compress_prog", input_type=Int(optional=True), prefix="--compress-prog", doc=InputDocumentation(doc="Pipe the output through the compression program\nPROG, which will be called as `PROG -c -',\nplus any arguments specified by --compress-prog-args.\nPROG must read uncompressed data from standard input\nand write compressed data to standard output when\ninvoked as noted above.\nExamples: gzip, bzip2, xz, pigz.")), ToolInput(tag="in_compress_prog_args", input_type=String(optional=True), prefix="--compress-prog-args", doc=InputDocumentation(doc="A string of additional arguments that will be passed\nto the compression program if one is specified with\n--compress-prog=PROG.  (The arguments '-c -' are\nstill passed in addition to explicitly specified\narguments.)")), ToolInput(tag="in_output_suffix", input_type=String(optional=True), prefix="--output-suffix", doc=InputDocumentation(doc="Use SUFFIX as the suffix of the output files\nafter '.fastq'.  A dot before the suffix is assumed,\nunless an empty suffix is provided.  Default:\nnothing; or 'gz' if -z is specified; or PROG if\n--compress-prog=PROG is specified.")), ToolInput(tag="in_threads", input_type=Int(optional=True), prefix="--threads", doc=InputDocumentation(doc="Set the number of worker threads.  This is in\naddition to the I/O threads.  Default: number of\nprocessors.  Note: if you need FLASH's output to\nappear deterministically or in the same order as\nthe original reads, you must specify -t 1\n(--threads=1).")), ToolInput(tag="in_quiet", input_type=Boolean(optional=True), prefix="--quiet", doc=InputDocumentation(doc="Do not print informational messages.")), ToolInput(tag="in_data_dot", input_type=String(), position=0, doc=InputDocumentation(doc="----------------------------------------------------------------------------\nMANDATORY INPUT")), ToolInput(tag="in_format_dot", input_type=String(), position=0, doc=InputDocumentation(doc="-I, --interleaved       Equivalent to specifying both --interleaved-input"))], outputs=[ToolOutput(tag="out_output_directory", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_output_directory", type_hint=File()), doc=OutputDocumentation(doc="Path to directory for output files.  Default:\ncurrent working directory."))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Flash_V0_1_0().translate("wdl", allow_empty_container=True)

