from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Array, String, Boolean, File, Int

Kleborate_V0_1_0 = CommandToolBuilder(tool="kleborate", base_command=["kleborate"], inputs=[ToolInput(tag="in_assemblies", input_type=Array(t=String(), optional=True), prefix="--assemblies", doc=InputDocumentation(doc="FASTA file(s) for assemblies")), ToolInput(tag="in_resistance", input_type=Boolean(optional=True), prefix="--resistance", doc=InputDocumentation(doc="Turn on resistance genes screening (default: no\nresistance gene screening)")), ToolInput(tag="in_kap_tive_k", input_type=Boolean(optional=True), prefix="--kaptive_k", doc=InputDocumentation(doc="Turn on Kaptive screening of K loci (default: do not\nrun Kaptive for K loci)")), ToolInput(tag="in_kap_tive_o", input_type=Boolean(optional=True), prefix="--kaptive_o", doc=InputDocumentation(doc="Turn on Kaptive screening of O loci (default: do not\nrun Kaptive for O loci)")), ToolInput(tag="in_kap_tive", input_type=Boolean(optional=True), prefix="--kaptive", doc=InputDocumentation(doc="Equivalent to --kaptive_k --kaptive_o")), ToolInput(tag="in_all", input_type=Boolean(optional=True), prefix="--all", doc=InputDocumentation(doc="Equivalent to --resistance --kaptive")), ToolInput(tag="in_outfile", input_type=File(optional=True), prefix="--outfile", doc=InputDocumentation(doc="File for detailed output (default:\nKleborate_results.txt)")), ToolInput(tag="in_kap_tive_k_outfile", input_type=File(optional=True), prefix="--kaptive_k_outfile", doc=InputDocumentation(doc="File for full Kaptive K locus output (default: do not\nsave Kaptive K locus results to separate file)")), ToolInput(tag="in_kap_tive_o_outfile", input_type=File(optional=True), prefix="--kaptive_o_outfile", doc=InputDocumentation(doc="File for full Kaptive O locus output (default: do not\nsave Kaptive O locus results to separate file)")), ToolInput(tag="in_min_identity", input_type=Int(optional=True), prefix="--min_identity", doc=InputDocumentation(doc="Minimum alignment identity for main results")), ToolInput(tag="in_min_coverage", input_type=Int(optional=True), prefix="--min_coverage", doc=InputDocumentation(doc="Minimum alignment coverage for main results")), ToolInput(tag="in_min_spurious_identity", input_type=Int(optional=True), prefix="--min_spurious_identity", doc=InputDocumentation(doc="Minimum alignment identity for spurious results")), ToolInput(tag="in_min_spurious_coverage", input_type=Int(optional=True), prefix="--min_spurious_coverage", doc=InputDocumentation(doc="Minimum alignment coverage for spurious results")), ToolInput(tag="in_force_index", input_type=Boolean(optional=True), prefix="--force_index", doc=InputDocumentation(doc="Rebuild the BLAST index at the start of execution"))], outputs=[ToolOutput(tag="out_outfile", output_type=File(optional=True), selector=InputSelector(input_to_select="in_outfile", type_hint=File()), doc=OutputDocumentation(doc="File for detailed output (default:\nKleborate_results.txt)")), ToolOutput(tag="out_kap_tive_k_outfile", output_type=File(optional=True), selector=InputSelector(input_to_select="in_kap_tive_k_outfile", type_hint=File()), doc=OutputDocumentation(doc="File for full Kaptive K locus output (default: do not\nsave Kaptive K locus results to separate file)")), ToolOutput(tag="out_kap_tive_o_outfile", output_type=File(optional=True), selector=InputSelector(input_to_select="in_kap_tive_o_outfile", type_hint=File()), doc=OutputDocumentation(doc="File for full Kaptive O locus output (default: do not\nsave Kaptive O locus results to separate file)"))], container="quay.io/biocontainers/kleborate:2.0.1--py_1", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Kleborate_V0_1_0().translate("wdl")

