from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, File, String

Kaks_Calculator_V0_1_0 = CommandToolBuilder(tool="KaKs_Calculator", base_command=["KaKs_Calculator"], inputs=[ToolInput(tag="in_axt_file_name", input_type=Boolean(optional=True), prefix="-i", doc=InputDocumentation(doc="Axt file name for calculating Ka & Ks.")), ToolInput(tag="in_output_file_name", input_type=File(optional=True), prefix="-o", doc=InputDocumentation(doc="Output file name for saving results.")), ToolInput(tag="in_genetic_code_table", input_type=Boolean(optional=True), prefix="-c", doc=InputDocumentation(doc="Genetic code table (Default = 1-Standard Code).\n1-Standard Code                         2-Vertebrate Mitochondrial Code\n3-Yeast Mitochondrial Code                      4-Mold Mitochondrial Code\n5-Invertebrate Mitochondrial Code                       6-Ciliate, Dasycladacean and Hexamita Code\n9-Echinoderm and Flatworm Mitochondrial Code                    10-Euplotid Nuclear Code\n11-Bacterial and Plant Plastid Code                     12-Alternative Yeast Nuclear Code\n13-Ascidian Mitochondrial Code                          14-Alternative Flatworm Mitochondrial Code\n15-Blepharisma Nuclear Code                     16-Chlorophycean Mitochondrial Code\n21-Trematode Mitochondrial Code                         22-Scenedesmus obliquus mitochondrial Code\n23-Thraustochytrium Mitochondrial Code\n(More information about the Genetic Codes: http://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi?mode=c)")), ToolInput(tag="in_methods_referencesdefault_ma", input_type=Boolean(optional=True), prefix="-m", doc=InputDocumentation(doc="Methods for estimating Ka and Ks and theirs references(Default = MA)")), ToolInput(tag="in_ng", input_type=String(), position=0, doc=InputDocumentation(doc="Nei, M. and Gojobori, T. (1986) Mol. Biol. Evol., 3, 418-426.")), ToolInput(tag="in_lwl", input_type=String(), position=1, doc=InputDocumentation(doc="Li, W.H., Wu, C.I. and Luo, C.C. (1985) Mol. Biol. Evol., 2, 150-174.")), ToolInput(tag="in_lpb", input_type=String(), position=2, doc=InputDocumentation(doc="Li, W.H. (1993) J. Mol. Evol., 36, 96-99.    Pamilo, P. and Bianchi, N.O. (1993) Mol. Biol. Evol., 10, 271-281.")), ToolInput(tag="in_ml_wl", input_type=String(), position=3, doc=InputDocumentation(doc="Tzeng, Y.H., Pan, R. and Li, W.H. (2004) Mol. Biol. Evol., 21, 2290-2298.")), ToolInput(tag="in_mlp_b", input_type=String(), position=4, doc=InputDocumentation(doc="Tzeng, Y.H., Pan, R. and Li, W.H. (2004) Mol. Biol. Evol., 21, 2290-2298.")), ToolInput(tag="in_gy", input_type=String(), position=5, doc=InputDocumentation(doc="Goldman, N. and Yang, Z. (1994) Mol. Biol. Evol., 11, 725-736.")), ToolInput(tag="in_yn", input_type=String(), position=6, doc=InputDocumentation(doc="Yang, Z. and Nielsen, R. (2000) Mol. Biol. Evol., 17, 32-43.")), ToolInput(tag="in_my_n", input_type=String(), position=7, doc=InputDocumentation(doc="Zhang, Z., Li, J. and Yu, J. (2006) BMC Evolutionary Biology, 6, 44.")), ToolInput(tag="in_ms", input_type=String(), position=8, doc=InputDocumentation(doc="Model Selection according to the AICc")), ToolInput(tag="in_ma", input_type=String(), position=9, doc=InputDocumentation(doc="Model Averaging on a set of candidate models")), ToolInput(tag="in_gng", input_type=String(), position=10, doc=InputDocumentation(doc="Wang, DP., Zhang, S., He, FH., Zhu, J.,Hu, SN. and Yu, J.(2009) Genomics, Proteomics and Bioinformatics. In press.")), ToolInput(tag="in_glw_l", input_type=String(), position=11, doc=InputDocumentation(doc="Wang, DP., Zhang, S., He, FH., Zhu, J.,Hu, SN. and Yu, J.(2009) Genomics, Proteomics and Bioinformatics. In press.")), ToolInput(tag="in_glp_b", input_type=String(), position=12, doc=InputDocumentation(doc="Wang, DP., Zhang, S., He, FH., Zhu, J.,Hu, SN. and Yu, J.(2009) Genomics, Proteomics and Bioinformatics. In press.")), ToolInput(tag="in_gml_wl", input_type=String(), position=13, doc=InputDocumentation(doc="Wang, DP., Zhang, S., He, FH., Zhu, J.,Hu, SN. and Yu, J.(2009) Genomics, Proteomics and Bioinformatics. In press.")), ToolInput(tag="in_gml_pb", input_type=String(), position=14, doc=InputDocumentation(doc="Wang, DP., Zhang, S., He, FH., Zhu, J.,Hu, SN. and Yu, J.(2009) Genomics, Proteomics and Bioinformatics. In press.")), ToolInput(tag="in_gyn", input_type=String(), position=15, doc=InputDocumentation(doc="Wang, DP., Zhang, S., He, FH., Zhu, J.,Hu, SN. and Yu, J.(2009) Genomics, Proteomics and Bioinformatics. In press.")), ToolInput(tag="in_gm_yn", input_type=String(), position=16, doc=InputDocumentation(doc="Wang, DP., Wan, HL., Zhang, S. and Yu, J. (2009) Biology Direct, 4:20 (16 June 2009)"))], outputs=[ToolOutput(tag="out_output_file_name", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output_file_name", type_hint=File()), doc=OutputDocumentation(doc="Output file name for saving results."))], container="quay.io/biocontainers/kakscalculator2:2.0.1--hc9558a2_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Kaks_Calculator_V0_1_0().translate("wdl")

