from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import String, File, Int, Directory, Float

Genome_Updater_Sh_V0_1_0 = CommandToolBuilder(tool="genome_updater.sh", base_command=["genome_updater.sh"], inputs=[ToolInput(tag="in_default_refseq", input_type=String(optional=True), prefix="-d", doc=InputDocumentation(doc="[genbank, refseq]\nDefault: refseq")), ToolInput(tag="in_category_default_all", input_type=String(optional=True), prefix="-c", doc=InputDocumentation(doc="Category [all, reference genome, representative genome, na]\nDefault: all")), ToolInput(tag="in_level_default_all", input_type=String(optional=True), prefix="-l", doc=InputDocumentation(doc="level [all, Complete Genome, Chromosome, Scaffold, Contig]\nDefault: all")), ToolInput(tag="in_formats_check_ftpftpncbinlmnihgovgenomesallreadmetxt", input_type=File(optional=True), prefix="-f", doc=InputDocumentation(doc="formats [genomic.fna.gz,assembly_report.txt, ...]\ncheck ftp://ftp.ncbi.nlm.nih.gov/genomes/all/README.txt for all file formats\nDefault: assembly_report.txt")), ToolInput(tag="in_top_references_download", input_type=Int(optional=True), prefix="-j", doc=InputDocumentation(doc="of top references for each species/taxids to download ['', species:INT, taxids:INT]. Example: 'species:3'. Selection is based on 1) RefSeq Category, 2) Assembly level, 3) Relation to type material and 4) Date (most recent first)\nDefault: ''")), ToolInput(tag="in_data_downloaded_updated", input_type=String(optional=True), prefix="-k", doc=InputDocumentation(doc=", no data is downloaded or updated - just checks for available sequences and changes")), ToolInput(tag="in_failed_downloads_data", input_type=String(optional=True), prefix="-i", doc=InputDocumentation(doc="failed downloads or any incomplete data from a previous run, keep current version")), ToolInput(tag="in_deletion_extra_found", input_type=Directory(optional=True), prefix="-x", doc=InputDocumentation(doc="the deletion of extra files if some are found in the repository folder")), ToolInput(tag="in_updated_assembly_accessions", input_type=String(optional=True), prefix="-u", doc=InputDocumentation(doc="of updated assembly accessions (Added/Removed, assembly accession, url)")), ToolInput(tag="in_updated_sequence_accessions", input_type=Int(optional=True), prefix="-r", doc=InputDocumentation(doc="of updated sequence accessions (Added/Removed, assembly accession, genbank accession, refseq accession, sequence length, taxid). Only available when file assembly_report.txt selected and successfully downloaded")), ToolInput(tag="in_list_urls_downloaded", input_type=String(optional=True), prefix="-p", doc=InputDocumentation(doc="list of URLs for downloaded and failed files")), ToolInput(tag="in_current_version_taxdumptargz", input_type=String(optional=True), prefix="-a", doc=InputDocumentation(doc="the current version of the Taxonomy database (taxdump.tar.gz)")), ToolInput(tag="in_output_directorydefault_tmpxxxxxxxxxx", input_type=Directory(optional=True), prefix="-o", doc=InputDocumentation(doc="output directory\nDefault: ./tmp.XXXXXXXXXX")), ToolInput(tag="in_labeldefault_timestamp_yyyymmddhhmmss", input_type=String(optional=True), prefix="-b", doc=InputDocumentation(doc="label\nDefault: current timestamp (YYYY-MM-DD_HH-MM-SS)")), ToolInput(tag="in_file_recover_data", input_type=File(optional=True), prefix="-e", doc=InputDocumentation(doc="'assembly_summary.txt' file to recover data from\nDefault: ''")), ToolInput(tag="in_default", input_type=Int(optional=True), prefix="-t", doc=InputDocumentation(doc="Default: 1")), ToolInput(tag="in_md_downloaded_files", input_type=Int(optional=True), prefix="-m", doc=InputDocumentation(doc="MD5 for downloaded files")), ToolInput(tag="in_output", input_type=String(optional=True), prefix="-s", doc=InputDocumentation(doc="output")), ToolInput(tag="in_output_download_progress", input_type=String(optional=True), prefix="-w", doc=InputDocumentation(doc="output with download progress (%) and download version at the end")), ToolInput(tag="in_exit_status_exit", input_type=Float(optional=True), prefix="-n", doc=InputDocumentation(doc="exit status. Exit Code = 1 if more than N files failed to download (integer for file number, float for percentage, 0 -> off)\nDefault: 0")), ToolInput(tag="in_print_debug_information", input_type=String(optional=True), prefix="-D", doc=InputDocumentation(doc="print debug information and exit")), ToolInput(tag="in_v_zero_dot_two_dot_four", input_type=Float(), position=0, doc=InputDocumentation(doc="-g Organism group (one or more comma-separated entries) [archaea, bacteria, fungi, human (also contained in vertebrate_mammalian), invertebrate, metagenomes (genbank), other (synthetic genomes - only genbank), plant, protozoa, vertebrate_mammalian, vertebrate_other, viral (only refseq)]. Example: archaea,bacteria"))], outputs=[ToolOutput(tag="out_output_directorydefault_tmpxxxxxxxxxx", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_output_directorydefault_tmpxxxxxxxxxx", type_hint=File()), doc=OutputDocumentation(doc="output directory\nDefault: ./tmp.XXXXXXXXXX"))], container="quay.io/biocontainers/genome_updater:0.2.4--0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Genome_Updater_Sh_V0_1_0().translate("wdl")

