from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Int, String, Boolean, Directory, Float

Mapdamage_V0_1_0 = CommandToolBuilder(tool="mapDamage", base_command=["mapDamage"], inputs=[ToolInput(tag="in_input", input_type=File(optional=True), prefix="--input", doc=InputDocumentation(doc="SAM/BAM file, must contain a valid header, use '-' for\nreading a BAM from stdin")), ToolInput(tag="in_reference", input_type=File(optional=True), prefix="--reference", doc=InputDocumentation(doc="Reference file in FASTA format")), ToolInput(tag="in_down_sample", input_type=Int(optional=True), prefix="--downsample", doc=InputDocumentation(doc="Downsample to a randomly selected fraction of the\nreads (if 0 < DOWNSAMPLE < 1), or a fixed number of\nrandomly selected reads (if DOWNSAMPLE >= 1). By\ndefault, no downsampling is performed.")), ToolInput(tag="in_down_sample_seed", input_type=String(optional=True), prefix="--downsample-seed", doc=InputDocumentation(doc="Seed value to use for downsampling. See documentation\nfor py module 'random' for default behavior.")), ToolInput(tag="in_merge_reference_sequences", input_type=Boolean(optional=True), prefix="--merge-reference-sequences", doc=InputDocumentation(doc="Ignore referece sequence names when tabulating reads\n(using '*' instead). Useful for alignments with a\nlarge number of reference sequnces, which may\notherwise result in excessive memory or disk usage due\nto the number of tables generated.")), ToolInput(tag="in_length", input_type=Int(optional=True), prefix="--length", doc=InputDocumentation(doc="read length, in nucleotides to consider [70]")), ToolInput(tag="in_around", input_type=Int(optional=True), prefix="--around", doc=InputDocumentation(doc="nucleotides to retrieve before/after reads [10]")), ToolInput(tag="in_min_base_qual", input_type=Int(optional=True), prefix="--min-basequal", doc=InputDocumentation(doc="minimun base quality Phred score considered, Phred-33\nassumed [0]")), ToolInput(tag="in_folder_name_store", input_type=Directory(optional=True), prefix="--folder", doc=InputDocumentation(doc="folder name to store results [results_FILENAME]")), ToolInput(tag="in_fast_a", input_type=Boolean(optional=True), prefix="--fasta", doc=InputDocumentation(doc="Write alignments in a FASTA file")), ToolInput(tag="in_plot_only", input_type=Boolean(optional=True), prefix="--plot-only", doc=InputDocumentation(doc="Run only plotting from a valid result folder")), ToolInput(tag="in_quiet", input_type=Boolean(optional=True), prefix="--quiet", doc=InputDocumentation(doc="Disable any output to stdout")), ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="--verbose", doc=InputDocumentation(doc="Display progression information during parsing")), ToolInput(tag="in_map_damage_modules", input_type=String(optional=True), prefix="--mapdamage-modules", doc=InputDocumentation(doc="Override the system wide installed mapDamage module")), ToolInput(tag="in_ymax", input_type=Float(optional=True), prefix="--ymax", doc=InputDocumentation(doc="graphical y-axis limit for nucleotide misincorporation\nfrequencies [0.3]")), ToolInput(tag="in_read_plot", input_type=Int(optional=True), prefix="--readplot", doc=InputDocumentation(doc="read length, in nucleotides, considered for plotting\nnucleotide misincorporations [25]")), ToolInput(tag="in_ref_plot", input_type=Int(optional=True), prefix="--refplot", doc=InputDocumentation(doc="the number of reference nucleotides to consider for\nploting base composition in the region located\nupstream and downstream of every read [10]")), ToolInput(tag="in_title", input_type=String(optional=True), prefix="--title", doc=InputDocumentation(doc="title used for plots []")), ToolInput(tag="in_rand", input_type=Int(optional=True), prefix="--rand", doc=InputDocumentation(doc="Number of random starting points for the likelihood")), ToolInput(tag="in_burn", input_type=Int(optional=True), prefix="--burn", doc=InputDocumentation(doc="Number of burnin iterations  [10000]")), ToolInput(tag="in_adjust", input_type=Int(optional=True), prefix="--adjust", doc=InputDocumentation(doc="Number of adjust proposal variance parameters")), ToolInput(tag="in_iter", input_type=Int(optional=True), prefix="--iter", doc=InputDocumentation(doc="Number of final MCMC iterations  [50000]")), ToolInput(tag="in_forward", input_type=Boolean(optional=True), prefix="--forward", doc=InputDocumentation(doc="Using only the 5' end of the seqs  [False]")), ToolInput(tag="in_reverse", input_type=Boolean(optional=True), prefix="--reverse", doc=InputDocumentation(doc="Using only the 3' end of the seqs  [False]")), ToolInput(tag="in_var_disp", input_type=Boolean(optional=True), prefix="--var-disp", doc=InputDocumentation(doc="Variable dispersion in the overhangs  [False]")), ToolInput(tag="in_jukes_cantor", input_type=Boolean(optional=True), prefix="--jukes-cantor", doc=InputDocumentation(doc="Use Jukes Cantor instead of HKY85  [False]")), ToolInput(tag="in_diff_hangs", input_type=Boolean(optional=True), prefix="--diff-hangs", doc=InputDocumentation(doc="The overhangs are different for 5' and 3'  [False]")), ToolInput(tag="in_fix_nicks", input_type=Boolean(optional=True), prefix="--fix-nicks", doc=InputDocumentation(doc="Fix the nick frequency vector (Only C.T from the 5'\nend and G.A from the 3' end)  [False]")), ToolInput(tag="in_use_raw_nick_freq", input_type=Boolean(optional=True), prefix="--use-raw-nick-freq", doc=InputDocumentation(doc="Use the raw nick frequency vector without smoothing\n[False]")), ToolInput(tag="in_single_stranded", input_type=Boolean(optional=True), prefix="--single-stranded", doc=InputDocumentation(doc="Single stranded protocol [False]")), ToolInput(tag="in_theme_bw", input_type=Boolean(optional=True), prefix="--theme-bw", doc=InputDocumentation(doc="Use black and white theme in post. pred. plot [False]")), ToolInput(tag="in_seq_length", input_type=Int(optional=True), prefix="--seq-length", doc=InputDocumentation(doc="How long sequence to use from each side [12]")), ToolInput(tag="in_stats_only", input_type=Boolean(optional=True), prefix="--stats-only", doc=InputDocumentation(doc="Run only statistical estimation from a valid result")), ToolInput(tag="in_check_r_packages", input_type=Boolean(optional=True), prefix="--check-R-packages", doc=InputDocumentation(doc="Check if the R modules are working")), ToolInput(tag="in_rescale", input_type=File(optional=True), prefix="--rescale", doc=InputDocumentation(doc="Rescale the quality scores in the BAM file using the\noutput from the statistical estimation")), ToolInput(tag="in_rescale_only", input_type=Boolean(optional=True), prefix="--rescale-only", doc=InputDocumentation(doc="Run only rescaling from a valid result folder")), ToolInput(tag="in_rescale_out", input_type=File(optional=True), prefix="--rescale-out", doc=InputDocumentation(doc="Write the rescaled BAM to this file")), ToolInput(tag="in_rescale_length_five_p", input_type=Int(optional=True), prefix="--rescale-length-5p", doc=InputDocumentation(doc="How many bases to rescale at the 5' termini; defaults\nto --seq-length.")), ToolInput(tag="in_rescale_length_three_p", input_type=Int(optional=True), prefix="--rescale-length-3p", doc=InputDocumentation(doc="How many bases to rescale at the 5' termini; defaults\nto --seq-length.")), ToolInput(tag="in_nostats_disabled_statistical", input_type=Directory(), position=0, doc=InputDocumentation(doc="--no-stats          Disabled statistical estimation, active by default"))], outputs=[ToolOutput(tag="out_rescale", output_type=File(optional=True), selector=InputSelector(input_to_select="in_rescale", type_hint=File()), doc=OutputDocumentation(doc="Rescale the quality scores in the BAM file using the\noutput from the statistical estimation"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Mapdamage_V0_1_0().translate("wdl", allow_empty_container=True)

