from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean

Efilter_V0_1_0 = CommandToolBuilder(tool="efilter", base_command=["efilter"], inputs=[ToolInput(tag="in_query", input_type=Boolean(optional=True), prefix="-query", doc=InputDocumentation(doc="Query string")), ToolInput(tag="in_sort", input_type=Boolean(optional=True), prefix="-sort", doc=InputDocumentation(doc="Result presentation order")), ToolInput(tag="in_days", input_type=Boolean(optional=True), prefix="-days", doc=InputDocumentation(doc="Number of days in the past")), ToolInput(tag="in_date_type", input_type=Boolean(optional=True), prefix="-datetype", doc=InputDocumentation(doc="Date field abbreviation")), ToolInput(tag="in_min_date", input_type=Boolean(optional=True), prefix="-mindate", doc=InputDocumentation(doc="Start of date range")), ToolInput(tag="in_maxdate", input_type=Boolean(optional=True), prefix="-maxdate", doc=InputDocumentation(doc="End of date range")), ToolInput(tag="in_field", input_type=Boolean(optional=True), prefix="-field", doc=InputDocumentation(doc="Query words individually in field")), ToolInput(tag="in_pairs", input_type=Boolean(optional=True), prefix="-pairs", doc=InputDocumentation(doc="Query overlapping word pairs")), ToolInput(tag="in_spell", input_type=Boolean(optional=True), prefix="-spell", doc=InputDocumentation(doc="Correct misspellings in query")), ToolInput(tag="in_pub", input_type=Boolean(optional=True), prefix="-pub", doc=InputDocumentation(doc="abstract, clinical, english, free, historical,\njournal, medline, preprint, published, review,\nstructured")), ToolInput(tag="in_journal", input_type=Boolean(optional=True), prefix="-journal", doc=InputDocumentation(doc="pnas, 'j bacteriol', ...")), ToolInput(tag="in_released", input_type=Boolean(optional=True), prefix="-released", doc=InputDocumentation(doc="last_week, last_month, last_year, prev_years")), ToolInput(tag="in_country", input_type=Boolean(optional=True), prefix="-country", doc=InputDocumentation(doc="usa:minnesota, united_kingdom, 'pacific ocean', ...")), ToolInput(tag="in_feature", input_type=Boolean(optional=True), prefix="-feature", doc=InputDocumentation(doc="gene, mrna, cds, mat_peptide, ...")), ToolInput(tag="in_location", input_type=Boolean(optional=True), prefix="-location", doc=InputDocumentation(doc="mitochondrion, chloroplast, plasmid, plastid")), ToolInput(tag="in_molecule", input_type=Boolean(optional=True), prefix="-molecule", doc=InputDocumentation(doc="genomic, mrna, trna, rrna, ncrna")), ToolInput(tag="in_organism", input_type=Boolean(optional=True), prefix="-organism", doc=InputDocumentation(doc="animals, archaea, bacteria, eukaryotes, fungi,\nhuman, insects, mammals, plants, prokaryotes,\nprotists, rodents, viruses")), ToolInput(tag="in_source", input_type=Boolean(optional=True), prefix="-source", doc=InputDocumentation(doc="genbank, insd, pdb, pir, refseq, swissprot, tpa")), ToolInput(tag="in_status", input_type=Boolean(optional=True), prefix="-status", doc=InputDocumentation(doc="alive")), ToolInput(tag="in_type", input_type=Boolean(optional=True), prefix="-type", doc=InputDocumentation(doc="coding, pseudo")), ToolInput(tag="in_class", input_type=Boolean(optional=True), prefix="-class", doc=InputDocumentation(doc="acceptor, donor, frameshift, indel, intron,\nmissense, nonsense, synonymous")), ToolInput(tag="in_kind", input_type=Boolean(optional=True), prefix="-kind", doc=InputDocumentation(doc="pathway")), ToolInput(tag="in_pathway", input_type=Boolean(optional=True), prefix="-pathway", doc=InputDocumentation(doc="reactome, wikipathways"))], outputs=[], container="quay.io/biocontainers/entrez-direct:13.9--pl526h375a9b1_1", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Efilter_V0_1_0().translate("wdl")

