from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import String, File, Directory, Int, Float

Hint_Tl_V0_1_0 = CommandToolBuilder(tool="hint_tl", base_command=["hint", "tl"], inputs=[ToolInput(tag="in_back_dir", input_type=String(optional=True), prefix="--backdir", doc=InputDocumentation(doc="[-c CUTOFF] [-o OUTDIR]")), ToolInput(tag="in_matrix_file", input_type=File(optional=True), prefix="--matrixfile", doc=InputDocumentation(doc="The matrix compressed file contains 1Mb and 100kb\nresolutions Hi-C contact matrix (.hic format), or 1Mb\nand 100kb resolution files seperate with ',', like\n/path/to/data_1Mb.cool,/path/to/data_100kb.cool or the\ndirectory that contain Hi-C interaction matrix in\nsparse or dense matrix format, interchromosomal\ninteraction matrices only. Absolute path is required")), ToolInput(tag="in_ref_dir", input_type=Directory(optional=True), prefix="--refdir", doc=InputDocumentation(doc="the reference directory that downloaded from dropbox\ndropbox. (https://www.dropbox.com/sh/2ufsyu4wvrboxxp/A\nABk5-_Fwy7jdM_t0vIsgYf4a?dl=0.)")), ToolInput(tag="in_enzyme", input_type=String(optional=True), prefix="--enzyme", doc=InputDocumentation(doc="Enzyme used in Hi-C experiment, DEFAULT: MboI")), ToolInput(tag="in_format", input_type=String(optional=True), prefix="--format", doc=InputDocumentation(doc="Format for the output contact matrix, DEFAULT: cooler")), ToolInput(tag="in_p_path", input_type=File(optional=True), prefix="--ppath", doc=InputDocumentation(doc="Path for pairix, use 'which pairix' to get the path")), ToolInput(tag="in_genome", input_type=String(optional=True), prefix="--genome", doc=InputDocumentation(doc="Specify your species, choose form hg38, hg19, and\nmm10. DEFAULT: hg19")), ToolInput(tag="in_chimeric", input_type=Int(optional=True), prefix="--chimeric", doc=InputDocumentation(doc="Chimeric read pairs with .pairsam format. If no\nchimeric reads provided, breakpoints in 100kb\nresolution will be output only")), ToolInput(tag="in_cut_off", input_type=Float(optional=True), prefix="--cutoff", doc=InputDocumentation(doc="Cutoff of the rank product for chromosomal pairs to\nselect candidate translocated chromosomal pairs,\ndefault = 0.05")), ToolInput(tag="in_outdir", input_type=File(optional=True), prefix="--outdir", doc=InputDocumentation(doc="Path to the output directory, where you want to store\nall the output files, if not set, the current\ndirectory will be used")), ToolInput(tag="in_name", input_type=String(optional=True), prefix="--name", doc=InputDocumentation(doc="Prefix for the result files. If not set, 'NA' will be\nused instead")), ToolInput(tag="in_threads", input_type=Int(optional=True), prefix="--threads", doc=InputDocumentation(doc="Number of threads for running HiNT-tl translocation\nbreakpoints detection part, DEFAULT: 16\n"))], outputs=[ToolOutput(tag="out_outdir", output_type=File(optional=True), selector=InputSelector(input_to_select="in_outdir", type_hint=File()), doc=OutputDocumentation(doc="Path to the output directory, where you want to store\nall the output files, if not set, the current\ndirectory will be used"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Hint_Tl_V0_1_0().translate("wdl", allow_empty_container=True)

