from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Boolean, Int, Float, Directory

Iva_V0_1_0 = CommandToolBuilder(tool="iva", base_command=["iva"], inputs=[ToolInput(tag="in_name_forward_reads", input_type=File(optional=True), prefix="-f", doc=InputDocumentation(doc="[.gz], --reads_fwd filename[.gz]\nName of forward reads fasta/q file. Must be used in\nconjunction with --reads_rev")), ToolInput(tag="in_name_reverse_reads", input_type=File(optional=True), prefix="-r", doc=InputDocumentation(doc="[.gz], --reads_rev filename[.gz]\nName of reverse reads fasta/q file. Must be used in\nconjunction with --reads_fwd")), ToolInput(tag="in_fr", input_type=File(optional=True), prefix="--fr", doc=InputDocumentation(doc="[.gz]    Name of interleaved fasta/q file")), ToolInput(tag="in_keep_files", input_type=Boolean(optional=True), prefix="--keep_files", doc=InputDocumentation(doc="Keep intermediate files (could be many!). Default is\nto delete all unnecessary files")), ToolInput(tag="in_contigs", input_type=File(optional=True), prefix="--contigs", doc=InputDocumentation(doc="[.gz]\nFasta file of contigs to be extended. Incompatible\nwith --reference")), ToolInput(tag="in_reference", input_type=File(optional=True), prefix="--reference", doc=InputDocumentation(doc="[.gz]\nEXPERIMENTAL! This option is EXPERIMENTAL, not\nrecommended, and has not been tested! Fasta file of\nreference genome, or parts thereof. IVA will try to\nassemble one contig per sequence in this file.\nIncompatible with --contigs")), ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="--verbose", doc=InputDocumentation(doc="Be verbose by printing messages to stdout. Use up to\nthree times for increasing verbosity.")), ToolInput(tag="in_s_malt_k", input_type=Int(optional=True), prefix="--smalt_k", doc=InputDocumentation(doc="kmer hash length in SMALT (the -k option in smalt\nindex) [19]")), ToolInput(tag="in_s_malt_s", input_type=Int(optional=True), prefix="--smalt_s", doc=InputDocumentation(doc="kmer hash step size in SMALT (the -s option in smalt\nindex) [11]")), ToolInput(tag="in_s_malt_id", input_type=Float(optional=True), prefix="--smalt_id", doc=InputDocumentation(doc="Minimum identity threshold for mapping to be reported\n(the -y option in smalt map) [0.5]")), ToolInput(tag="in_ctg_first_trim", input_type=Int(optional=True), prefix="--ctg_first_trim", doc=InputDocumentation(doc="Number of bases to trim off the end of every contig\nbefore extending for the first time [25]")), ToolInput(tag="in_ctg_iter_trim", input_type=Int(optional=True), prefix="--ctg_iter_trim", doc=InputDocumentation(doc="During iterative extension, number of bases to trim\noff the end of a contig when extension fails (then try\nextending again) [10]")), ToolInput(tag="in_ext_min_cov", input_type=Int(optional=True), prefix="--ext_min_cov", doc=InputDocumentation(doc="Minimum kmer depth needed to use that kmer to extend a\ncontig [10]")), ToolInput(tag="in_ext_min_ratio", input_type=Float(optional=True), prefix="--ext_min_ratio", doc=InputDocumentation(doc="Sets N, where kmer for extension must be at least N\ntimes more abundant than next most common kmer [4]")), ToolInput(tag="in_ext_max_bases", input_type=Int(optional=True), prefix="--ext_max_bases", doc=InputDocumentation(doc="Maximum number of bases to try to extend on each\niteration [100]")), ToolInput(tag="in_ext_min_clip", input_type=Int(optional=True), prefix="--ext_min_clip", doc=InputDocumentation(doc="Set minimum number of bases soft clipped off a read\nfor those bases to be used for extension [3]")), ToolInput(tag="in_max_contigs", input_type=Int(optional=True), prefix="--max_contigs", doc=InputDocumentation(doc="Maximum number of contigs allowed in the assembly. No\nmore seeds generated if the cutoff is reached [50]")), ToolInput(tag="in_make_new_seeds", input_type=Boolean(optional=True), prefix="--make_new_seeds", doc=InputDocumentation(doc="When no more contigs can be extended, generate a new\nseed. This is forced to be true when --contigs is not\nused")), ToolInput(tag="in_seed_start_length", input_type=Int(optional=True), prefix="--seed_start_length", doc=InputDocumentation(doc="When making a seed sequence, use the most common kmer\nof this length. Default is to use the minimum of\n(median read length, 95). Warning: it is not\nrecommended to set this higher than 95")), ToolInput(tag="in_seed_stop_length", input_type=Int(optional=True), prefix="--seed_stop_length", doc=InputDocumentation(doc="Stop extending seed using perfect matches from reads\nwhen this length is reached. Future extensions are\nthen made by treating the seed as a contig\n[0.9*max_insert]")), ToolInput(tag="in_seed_min_km_er_cov", input_type=Int(optional=True), prefix="--seed_min_kmer_cov", doc=InputDocumentation(doc="Minimum kmer coverage of initial seed [25]")), ToolInput(tag="in_seed_max_km_er_cov", input_type=Int(optional=True), prefix="--seed_max_kmer_cov", doc=InputDocumentation(doc="Maximum kmer coverage of initial seed [1000000]")), ToolInput(tag="in_seed_ext_max_bases", input_type=Int(optional=True), prefix="--seed_ext_max_bases", doc=InputDocumentation(doc="Maximum number of bases to try to extend on each\niteration [50]")), ToolInput(tag="in_seed_overlap_length", input_type=Int(optional=True), prefix="--seed_overlap_length", doc=InputDocumentation(doc="Number of overlapping bases needed between read and\nseed to use that read to extend [seed_start_length]")), ToolInput(tag="in_seed_ext_min_cov", input_type=Int(optional=True), prefix="--seed_ext_min_cov", doc=InputDocumentation(doc="Minimum kmer depth needed to use that kmer to extend a\ncontig [10]")), ToolInput(tag="in_seed_ext_min_ratio", input_type=Float(optional=True), prefix="--seed_ext_min_ratio", doc=InputDocumentation(doc="Sets N, where kmer for extension must be at least N\ntimes more abundant than next most common kmer [4]")), ToolInput(tag="in_trim_mo_matic", input_type=File(optional=True), prefix="--trimmomatic", doc=InputDocumentation(doc="Provide location of trimmomatic.jar file to enable\nread trimming. Required if --adapters used")), ToolInput(tag="in_trim_mo_qual", input_type=Int(optional=True), prefix="--trimmo_qual", doc=InputDocumentation(doc="Trimmomatic options used to quality trim reads\n[LEADING:10 TRAILING:10 SLIDINGWINDOW:4:20]")), ToolInput(tag="in_adapters", input_type=File(optional=True), prefix="--adapters", doc=InputDocumentation(doc="Fasta file of adapter sequences to be trimmed off\nreads. If used, must also use --trimmomatic. Default\nis file of adapters supplied with IVA")), ToolInput(tag="in_min_trimmed_length", input_type=Int(optional=True), prefix="--min_trimmed_length", doc=InputDocumentation(doc="Minimum length of read after trimming [50]")), ToolInput(tag="in_pcr_primers", input_type=File(optional=True), prefix="--pcr_primers", doc=InputDocumentation(doc="FASTA file of primers. The first perfect match found\nto a sequence in the primers file will be trimmed off\nthe start of each read. This is run after trimmomatic\n(if --trimmomatic used)")), ToolInput(tag="in_max_insert", input_type=Int(optional=True), prefix="--max_insert", doc=InputDocumentation(doc="Maximum insert size (includes read length). Reads with\ninferred insert size more than the maximum will not be\nused to extend contigs [800]")), ToolInput(tag="in_threads", input_type=Int(optional=True), prefix="--threads", doc=InputDocumentation(doc="Number of threads to use [1]")), ToolInput(tag="in_kmc_one_thread", input_type=Boolean(optional=True), prefix="--kmc_onethread", doc=InputDocumentation(doc="Force kmc to use one thread. By default the value of")), ToolInput(tag="in_strand_bias", input_type=Float(optional=True), prefix="--strand_bias", doc=InputDocumentation(doc="in [0,0.5]\nSet strand bias cutoff of mapped reads when trimming\ncontig ends, in the interval [0,0.5]. A value of x\nmeans that a base needs min(fwd_depth, rev_depth) /\ntotal_depth <= x. The only time this should be used is\nwith libraries with overlapping reads (ie fragment\nlength < 2*read length), and even then, it can make\nresults worse. If used, try a low value like 0.1 first\n[0]")), ToolInput(tag="in_test", input_type=Directory(optional=True), prefix="--test", doc=InputDocumentation(doc="Run using built in test data. All other options will\nbe ignored, except the mandatory output directory, and\n--trimmomatic and --threads can be also be used"))], outputs=[ToolOutput(tag="out_test", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_test", type_hint=File()), doc=OutputDocumentation(doc="Run using built in test data. All other options will\nbe ignored, except the mandatory output directory, and\n--trimmomatic and --threads can be also be used"))], container="quay.io/biocontainers/iva:1.0.11--py_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Iva_V0_1_0().translate("wdl")

