from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, File, Int

Gtftk_Select_By_Key_V0_1_0 = CommandToolBuilder(tool="gtftk_select_by_key", base_command=["gtftk", "select_by_key"], inputs=[ToolInput(tag="in_value", input_type=Boolean(optional=True), prefix="--value", doc=InputDocumentation(doc="A comma-separated list of values. (default: None)")), ToolInput(tag="in_file_with_values", input_type=Boolean(optional=True), prefix="--file-with-values", doc=InputDocumentation(doc="A file containing values as a single column. (default: None)")), ToolInput(tag="in_select_transcripts", input_type=Boolean(optional=True), prefix="--select-transcripts", doc=InputDocumentation(doc="A shortcuts for '-k feature -v transcript'. (default: False)")), ToolInput(tag="in_select_genes", input_type=Boolean(optional=True), prefix="--select-genes", doc=InputDocumentation(doc="A shortcuts for '-k feature -v gene'. (default: False)")), ToolInput(tag="in_select_exons", input_type=Boolean(optional=True), prefix="--select-exons", doc=InputDocumentation(doc="A shortcuts for '-k feature -v exon'. (default: False)")), ToolInput(tag="in_select_cds", input_type=Boolean(optional=True), prefix="--select-cds", doc=InputDocumentation(doc="A shortcuts for '-k feature -v CDS'. (default: False)")), ToolInput(tag="in_select_start_codon", input_type=Boolean(optional=True), prefix="--select-start-codon", doc=InputDocumentation(doc="A shortcuts for '-k feature -v start_codon'. (default: False)")), ToolInput(tag="in_input_file", input_type=Boolean(optional=True), prefix="--inputfile", doc=InputDocumentation(doc="Path to the GTF file. Default to STDIN (default: <stdin>)")), ToolInput(tag="in_output_file", input_type=File(optional=True), prefix="--outputfile", doc=InputDocumentation(doc="Output file. (default: <stdout>)")), ToolInput(tag="in_key", input_type=Boolean(optional=True), prefix="--key", doc=InputDocumentation(doc="The key name. (default: None)")), ToolInput(tag="in_col", input_type=Boolean(optional=True), prefix="--col", doc=InputDocumentation(doc="The column number (one-based) that contains the values in the file. File is tab-delimited. (default: 1)")), ToolInput(tag="in_invert_match", input_type=Boolean(optional=True), prefix="--invert-match", doc=InputDocumentation(doc="Not/invert match. Select lines whose selected key is not associated with the selected values. (default: False)")), ToolInput(tag="in_bed_format", input_type=Boolean(optional=True), prefix="--bed-format", doc=InputDocumentation(doc="Ask for bed format output. (default: False)")), ToolInput(tag="in_names", input_type=Boolean(optional=True), prefix="--names", doc=InputDocumentation(doc="If Bed output. The key(s) that should be used as name. (default: gene_id,transcript_id)")), ToolInput(tag="in_separator", input_type=Boolean(optional=True), prefix="--separator", doc=InputDocumentation(doc="If Bed output. The separator to be used for separating name elements (see -n). (default: |)")), ToolInput(tag="in_log", input_type=Boolean(optional=True), prefix="--log", doc=InputDocumentation(doc="Print some statistics about selected features. To be used in conjunction with -V 1/2. (default: False)")), ToolInput(tag="in_verbosity", input_type=Boolean(optional=True), prefix="--verbosity", doc=InputDocumentation(doc="Set output verbosity ([0-3]). (default: 0)")), ToolInput(tag="in_no_date", input_type=File(optional=True), prefix="--no-date", doc=InputDocumentation(doc="Do not add date to output file names. (default: False)")), ToolInput(tag="in_add_chr", input_type=Boolean(optional=True), prefix="--add-chr", doc=InputDocumentation(doc="Add 'chr' to chromosome names before printing output. (default: False)")), ToolInput(tag="in_tmp_dir", input_type=Boolean(optional=True), prefix="--tmp-dir", doc=InputDocumentation(doc="Keep all temporary files into this folder. (default: None)")), ToolInput(tag="in_keep_all", input_type=Boolean(optional=True), prefix="--keep-all", doc=InputDocumentation(doc="Try to keep all temporary files even if process does not terminate normally. (default: False)")), ToolInput(tag="in_logger_file", input_type=Boolean(optional=True), prefix="--logger-file", doc=InputDocumentation(doc="Stores the arguments passed to the command into a file. (default: None)")), ToolInput(tag="in_write_message_to_file", input_type=Boolean(optional=True), prefix="--write-message-to-file", doc=InputDocumentation(doc="Store all message into a file. (default: None)")), ToolInput(tag="in_optional_arguments", input_type=Int(), position=0, doc=InputDocumentation(doc="optional arguments:"))], outputs=[ToolOutput(tag="out_output_file", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output_file", type_hint=File()), doc=OutputDocumentation(doc="Output file. (default: <stdout>)")), ToolOutput(tag="out_no_date", output_type=File(optional=True), selector=InputSelector(input_to_select="in_no_date", type_hint=File()), doc=OutputDocumentation(doc="Do not add date to output file names. (default: False)"))], container="quay.io/biocontainers/pygtftk:1.2.7--py36h4edd4cd_1", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Gtftk_Select_By_Key_V0_1_0().translate("wdl")

