from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Int, Float, String

Sage_V0_1_0 = CommandToolBuilder(tool="SAGE", base_command=["SAGE"], inputs=[ToolInput(tag="in_known_hotspots", input_type=File(optional=True), prefix="-known_hotspots", doc=InputDocumentation(doc="Tab separated file of known hotspot")), ToolInput(tag="in_min_base_quality", input_type=Int(optional=True), prefix="-min_base_quality", doc=InputDocumentation(doc="Minimum quality for a base to be\nconsidered [13]")), ToolInput(tag="in_min_in_del_quality", input_type=Int(optional=True), prefix="-min_indel_quality", doc=InputDocumentation(doc="Low confidence filtering minimum\nindel quality [150]")), ToolInput(tag="in_min_in_del_vaf", input_type=Float(optional=True), prefix="-min_indel_vaf", doc=InputDocumentation(doc="Low confidence filtering minimum\nindel VAF [0.02]")), ToolInput(tag="in_min_mapping_quality", input_type=Int(optional=True), prefix="-min_mapping_quality", doc=InputDocumentation(doc="Minimum mapping quality for an\nalignment to be used [1]")), ToolInput(tag="in_min_snv_quality", input_type=Int(optional=True), prefix="-min_snv_quality", doc=InputDocumentation(doc="Low confidence filtering minimum\nSNV/MNV quality [100]")), ToolInput(tag="in_min_snv_vaf", input_type=Float(optional=True), prefix="-min_snv_vaf", doc=InputDocumentation(doc="Low confidence filtering minimum\nSNV/MNV VAF [0.005]")), ToolInput(tag="in_min_tumor_reads", input_type=Int(optional=True), prefix="-min_tumor_reads", doc=InputDocumentation(doc="Low confidence filtering minimum\ntumor reads [2]")), ToolInput(tag="in_out", input_type=File(optional=True), prefix="-out", doc=InputDocumentation(doc="Output VCF file to write. Gz")), ToolInput(tag="in_reference", input_type=String(optional=True), prefix="-reference", doc=InputDocumentation(doc="Name of reference sample")), ToolInput(tag="in_reference_bam", input_type=File(optional=True), prefix="-reference_bam", doc=InputDocumentation(doc="Path to reference bam file")), ToolInput(tag="in_tumor", input_type=String(optional=True), prefix="-tumor", doc=InputDocumentation(doc="Name of tumor sample")), ToolInput(tag="in_tumor_bam", input_type=File(optional=True), prefix="-tumor_bam", doc=InputDocumentation(doc="Path to tumor bam file")), ToolInput(tag="in_sage_hotspot_application", input_type=String(), position=0, doc=InputDocumentation(doc="-coding_regions <arg>                Coding regions bed file to search\nfor inframe indels")), ToolInput(tag="in_locations", input_type=String(), position=1, doc=InputDocumentation(doc="-max_het_binomial_likelihood <arg>   Up to 1 read of support for the ALT")), ToolInput(tag="in_supported_dot", input_type=String(), position=0, doc=InputDocumentation(doc="-ref_genome <arg>                    Path to the ref genome fasta file"))], outputs=[ToolOutput(tag="out_out", output_type=File(optional=True), selector=InputSelector(input_to_select="in_out", type_hint=File()), doc=OutputDocumentation(doc="Output VCF file to write. Gz"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Sage_V0_1_0().translate("wdl", allow_empty_container=True)

