from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Int, String, Directory

Snaptools_Snap_Pre_V0_1_0 = CommandToolBuilder(tool="snaptools_snap_pre", base_command=["snaptools", "snap-pre"], inputs=[ToolInput(tag="in_input_file", input_type=File(optional=True), prefix="--input-file", doc=InputDocumentation(doc="input bam, bed or bed.gz file. (default: None)")), ToolInput(tag="in_output_snap", input_type=File(optional=True), prefix="--output-snap", doc=InputDocumentation(doc="output snap file. (default: None)")), ToolInput(tag="in_genome_name", input_type=Int(optional=True), prefix="--genome-name", doc=InputDocumentation(doc="genome identifier (i.e. hg19, mm10). This tag does not\nchange anything unless merge or compare multiple snap\nfiles. (default: None)")), ToolInput(tag="in_genome_size", input_type=Int(optional=True), prefix="--genome-size", doc=InputDocumentation(doc="a txt file contains corresponding genome sizes. It\nmust be in the following format with the first column\nthe chromsome name and the second column as chromsome\nlength. This tag does not change anything unless merge\nor compare multiple snap files. (default: None)")), ToolInput(tag="in_barcode_file", input_type=File(optional=True), prefix="--barcode-file", doc=InputDocumentation(doc="a txt file contains pre-selected cell barcodes. If\n--barcode-file is given, snaptools will ignore any\nbarcodes not present in the --barcode-file. If it is\nNone, snaptools will automatically identify barcodes\nfrom bam file. The first column of --barcode-file must\nbe the selected barcodes and the other columns could\nbe any attributes of the barcode as desired\n(`ATGCTCTACTAC attr1 att2`). The attributes, however,\nwill not be kept in the snap file. This tag will be\nignored if the output snap file already exists.\n(default: None)")), ToolInput(tag="in_min_mapq", input_type=Int(optional=True), prefix="--min-mapq", doc=InputDocumentation(doc="min mappability score. Fargments with mappability\nscore less than --min-mapq will be filtered. (default:\n10)")), ToolInput(tag="in_min_cov", input_type=Int(optional=True), prefix="--min-cov", doc=InputDocumentation(doc="min number of fragments per barcode. barcodes of total\nfragments fewer than --min-cov will be considered when\ncreating the cell x bin count matrix. Note: because\nthe vast majority of barcodes contains very few reads,\nwe found by setting --min-cov, one can remove barcodes\nof low coverage without wasting time and storage.\nPlease note that this is not selection of good\nbarcodes for downstream clustering analysis, it is\nonly filterationof very low-quality barcodes.\n(default: 0)")), ToolInput(tag="in_max_num", input_type=Int(optional=True), prefix="--max-num", doc=InputDocumentation(doc="max number of barcodes to store. Barcodes are sorted\nbased on the coverage and only the top --max-num\nbarcodes will be stored. (default: 1000000)")), ToolInput(tag="in_keep_chrm", input_type=String(optional=True), prefix="--keep-chrm", doc=InputDocumentation(doc="a boolen tag indicates whether to keep fragments\nmapped to chrM. If set Fasle, fragments aligned to the\nmitochondrial sequence will be filtered. (default:\nTrue)")), ToolInput(tag="in_keep_single", input_type=String(optional=True), prefix="--keep-single", doc=InputDocumentation(doc="a boolen tag indicates whether to keep those reads\nwhose mates are not mapped or missing. If False,\nunpaired reads will be filtered. If True, unpaired\nreads will be simply treated as a fragment. Note: for\nsingle-end such as scTHS-seq, --keep-single must be\nTrue. (default: True)")), ToolInput(tag="in_keep_secondary", input_type=String(optional=True), prefix="--keep-secondary", doc=InputDocumentation(doc="a boolen tag indicates whether to keep secondary\nalignments. If False, secondary alignments will be\nfiltered. If True, a secondary alignments will be\ntreated as fragments just single-end. (default: False)")), ToolInput(tag="in_keep_discordant", input_type=String(optional=True), prefix="--keep-discordant", doc=InputDocumentation(doc="a boolen tag indicates whether to keep discordant read\npairs. (default: False)")), ToolInput(tag="in_tmp_folder", input_type=Directory(optional=True), prefix="--tmp-folder", doc=InputDocumentation(doc="a directory to store temporary files. If not given,\nsnaptools will automatically generate a temporary\nlocation to store temporary files. (default: None)")), ToolInput(tag="in_overwrite", input_type=String(optional=True), prefix="--overwrite", doc=InputDocumentation(doc="a boolen tag indicates whether to overwrite the matrix\nsession if it already exists. (default: False)")), ToolInput(tag="in_qc_file", input_type=File(optional=True), prefix="--qc-file", doc=InputDocumentation(doc="a boolen tag indicates whether to create a master qc\nfile. This .qc file contains basic quality control\nmetrics at the bulk level. Quality control is only\nestimated by selected barcodes only. (default: True)")), ToolInput(tag="in_verbose", input_type=String(optional=True), prefix="--verbose", doc=InputDocumentation(doc="a boolen tag indicates output the progress. (default:\nTrue)\n"))], outputs=[ToolOutput(tag="out_output_snap", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output_snap", type_hint=File()), doc=OutputDocumentation(doc="output snap file. (default: None)")), ToolOutput(tag="out_barcode_file", output_type=File(optional=True), selector=InputSelector(input_to_select="in_barcode_file", type_hint=File()), doc=OutputDocumentation(doc="a txt file contains pre-selected cell barcodes. If\n--barcode-file is given, snaptools will ignore any\nbarcodes not present in the --barcode-file. If it is\nNone, snaptools will automatically identify barcodes\nfrom bam file. The first column of --barcode-file must\nbe the selected barcodes and the other columns could\nbe any attributes of the barcode as desired\n(`ATGCTCTACTAC attr1 att2`). The attributes, however,\nwill not be kept in the snap file. This tag will be\nignored if the output snap file already exists.\n(default: None)"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Snaptools_Snap_Pre_V0_1_0().translate("wdl", allow_empty_container=True)

