from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Int, String, File, Boolean

Fastqtoca_V0_1_0 = CommandToolBuilder(tool="fastqToCA", base_command=["fastqToCA"], inputs=[ToolInput(tag="in_insert_size", input_type=Int(optional=True), prefix="-insertsize", doc=InputDocumentation(doc="d    Mates are on average i +- d bp apart.\nIf the word 'constant' follows the insert size, no changes will be\nmade to the insert size.")), ToolInput(tag="in_library_name", input_type=String(optional=True), prefix="-libraryname", doc=InputDocumentation(doc="The UID of the library these reads are added to.")), ToolInput(tag="in_technology", input_type=Int(optional=True), prefix="-technology", doc=InputDocumentation(doc="What instrument were these reads generated on ('illumina' is the default):\n'none'               -- don't set any features; use -feature to set them manually\n'sanger'             -- reads from dideoxy sequencers\n'454'                -- reads from 454 Life Sciences; FLX, Titanium, FLX+\n'illumina'           -- reads from Illumina; GAIIx, MiSeq, HiSeq; shorter than 160bp\n'illumina-long'      -- reads from Illumina; GAIIx, MiSeq, HiSeq; any length\n'moleculo'           -- reads from Illumina; Moleculo\n'pacbio-ccs'         -- reads from PacBio; Circular Consensus Sequence (CSS)\n'pacbio-corrected'   -- reads from PacBio; corrected reads from pacBioToCA\n'pacbio-raw'         -- reads from PacBio; uncorrected reads")), ToolInput(tag="in_type", input_type=File(optional=True), prefix="-type", doc=InputDocumentation(doc="What type of fastq ('sanger' is the default):\n'sanger'   -- QV's are PHRED, offset=33 '!', NCBI SRA data.\n'solexa'   -- QV's are Solexa, early Solexa data.\n'illumina' -- QV's are PHRED, offset=64 '@', Illumina reads from version 1.3 on.\nSee Cock, et al., 'The Sanger FASTQ file format for sequences with quality scores, and\nthe Solexa/Illumina FASTQ variants', doi:10.1093/nar/gkp1137")), ToolInput(tag="in_in_nie", input_type=Boolean(optional=True), prefix="-innie", doc=InputDocumentation(doc="The paired end reads are 5'-3' <-> 3'-5' (the usual case) (default)")), ToolInput(tag="in_out_tie", input_type=Boolean(optional=True), prefix="-outtie", doc=InputDocumentation(doc="The paired end reads are 3'-5' <-> 5'-3' (for Illumina Mate Pair reads)\nThis switch will reverse-complement every read, transforming outtie-oriented\nmates into innie-oriented mates.  This trick only works if all reads are the\nsame length.")), ToolInput(tag="in_reads", input_type=String(optional=True), prefix="-reads", doc=InputDocumentation(doc="Single ended reads, in fastq format.")), ToolInput(tag="in_nonrandom", input_type=Boolean(optional=True), prefix="-nonrandom", doc=InputDocumentation(doc="Mark the library as containing non-random reads.")), ToolInput(tag="in_feature", input_type=String(optional=True), prefix="-feature", doc=InputDocumentation(doc="V       Set feature F to value V.")), ToolInput(tag="in_stddev", input_type=String(optional=True), position=0, doc=InputDocumentation(doc=""))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Fastqtoca_V0_1_0().translate("wdl", allow_empty_container=True)

