from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, File, Int, Array, String

Pauvre_Browser_V0_1_0 = CommandToolBuilder(tool="pauvre_browser", base_command=["pauvre", "browser"], inputs=[ToolInput(tag="in_quiet", input_type=Boolean(optional=True), prefix="--quiet", doc=InputDocumentation(doc="Do not output warnings to stderr")), ToolInput(tag="in_chromosome_id", input_type=File(optional=True), prefix="--chromosomeid", doc=InputDocumentation(doc="The fasta sequence to observe. Use the header name of\nthe fasta file without the '>' character")), ToolInput(tag="in_dpi", input_type=Int(optional=True), prefix="--dpi", doc=InputDocumentation(doc="Change the dpi from the default 600 if you need it")), ToolInput(tag="in_no_timestamp", input_type=File(optional=True), prefix="--no_timestamp", doc=InputDocumentation(doc="Turn off time stamps in the filename output.")), ToolInput(tag="in_output_base_name", input_type=File(optional=True), prefix="--output-base-name", doc=InputDocumentation(doc="Specify a base name for the output file( s). The input\nfile base name is the default.")), ToolInput(tag="in_path", input_type=File(optional=True), prefix="--path", doc=InputDocumentation(doc="Set an explicit filepath for the output. Only do this\nif you have selected one output type.")), ToolInput(tag="in_plot_commands", input_type=Array(t=String(), optional=True), prefix="--plot_commands", doc=InputDocumentation(doc="Write strings here to select what to plot. The format\nfor each track is: <type>:<path>:<style>:<options> To\nplot the reference, the format is:\nref:<style>:<options> Surround each track string with\ndouble quotes and a space between subsequent strings.\n'bam:mybam.bam:depth' 'ref:colorful'")), ToolInput(tag="in_ratio", input_type=Array(t=String(), optional=True), prefix="--ratio", doc=InputDocumentation(doc="Enter the dimensions (arbitrary units) to plot the\nfigure. For example a figure that is seven times wider\nthan tall is: --ratio 7 1")), ToolInput(tag="in_reference", input_type=File(optional=True), prefix="--reference", doc=InputDocumentation(doc="The reference fasta file.")), ToolInput(tag="in_start", input_type=File(optional=True), prefix="--start", doc=InputDocumentation(doc="The start position to observe on the fasta file. Uses\n1-based indexing.")), ToolInput(tag="in_stop", input_type=File(optional=True), prefix="--stop", doc=InputDocumentation(doc="The stop position to observe on the fasta file. Uses\n1-based indexing.")), ToolInput(tag="in_transparent", input_type=Boolean(optional=True), prefix="--transparent", doc=InputDocumentation(doc="Specify this option if you DON'T want a transparent\nbackground. Default is on.\n")), ToolInput(tag="in_higher", input_type=String(), position=0, doc=InputDocumentation(doc="--fileform STRING [STRING ...]"))], outputs=[ToolOutput(tag="out_no_timestamp", output_type=File(optional=True), selector=InputSelector(input_to_select="in_no_timestamp", type_hint=File()), doc=OutputDocumentation(doc="Turn off time stamps in the filename output.")), ToolOutput(tag="out_path", output_type=File(optional=True), selector=InputSelector(input_to_select="in_path", type_hint=File()), doc=OutputDocumentation(doc="Set an explicit filepath for the output. Only do this\nif you have selected one output type."))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Pauvre_Browser_V0_1_0().translate("wdl", allow_empty_container=True)

