from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, File, String, Int, Directory

Rsat_Install_Organism_V0_1_0 = CommandToolBuilder(tool="rsat_install_organism", base_command=["rsat", "install-organism"], inputs=[ToolInput(tag="in_help", input_type=Boolean(optional=True), prefix="-help", doc=InputDocumentation(doc="(must be first argument) display options")), ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="-v", doc=InputDocumentation(doc="verbose")), ToolInput(tag="in_org", input_type=Boolean(optional=True), prefix="-org", doc=InputDocumentation(doc="organism name without spaces (e.g. Saccharomyces_cerevisiae)\nThe option -org can be used iteratively on the same\ncommand line to iterate the installation over multiple\norganisms.")), ToolInput(tag="in_org_file", input_type=File(optional=True), prefix="-org_file", doc=InputDocumentation(doc="Text file containing a list of organisms to install.\nThe first word of each row is taken as a query\norgnanism. Further information of the same row is\nignored.")), ToolInput(tag="in_all_organisms", input_type=Boolean(optional=True), prefix="-all_organisms", doc=InputDocumentation(doc="Install all the organisms found in the Refseq\ndirectory (see option -refseq).")), ToolInput(tag="in_tax_on", input_type=Boolean(optional=True), prefix="-taxon", doc=InputDocumentation(doc="taxon name (mutually exclusive with -org)\nThe installation will iterate over all organisms of\nthe selected taxon. Note that the command will only\napply to the organisms previously declared with the\ncommand install-organism -task config. The option\n-taxon is thus convenient for re-running installation\ntasks on previously installed organisms rather than\nfor installing new genomes downloaded e.g. from NCBI.\nThe option -taxon can be used iteratively on the same\ncommand line to iterate the installation over multiple\ntaxa.")), ToolInput(tag="in_group", input_type=String(optional=True), prefix="-group", doc=InputDocumentation(doc="group according to the NCBI/Refseq ftp/rsync\nsite (ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq).\nExamples: bacteria, fungi, protozoa, plant,\ninvertebrate, vertebrate_mammalian, vertebrate_other")), ToolInput(tag="in_species", input_type=Int(optional=True), prefix="-species", doc=InputDocumentation(doc="name, with space replaced by underscore\n(e.g. Saccharomyces_cerevisiae).\nNote that one species may correspond to several\norganisms if several strains are available for this\nspecies. In particular, for some bacteria there are\nseveral thousands of sequenced strains\n(e.g. Escherichia coli).\nThe option -max_strains can be used to restrict the\nnumber of installed strains per species.\nAn alternative is to specify a custom list of species\n+ strains with the option -org_table (this is the\nrecommended way to install multiple genomes from\nNCBI).")), ToolInput(tag="in_species_file", input_type=File(optional=True), prefix="-species_file", doc=InputDocumentation(doc="containing species names in the first")), ToolInput(tag="in_strain", input_type=String(optional=True), prefix="-strain", doc=InputDocumentation(doc="the download to a single strain or\nassembly. By default all strains of the selected\nspecies are downloaded, but this can be heavy for some\nbacterial species where thousands of strains have been\nsequenced (e.g. Escherichia coli).\nNote: the RSAT organism name is automatically built by\nconcatenating the NCBI species name and strain\nidentifier.")), ToolInput(tag="in_assembly_table", input_type=Boolean(optional=True), prefix="-assembly_table", doc=InputDocumentation(doc="Table extracted from the NCBI ftp site, indicating the\norganisms to install.\nExample: assembly table of fungi:\nftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/fungi/assembly_summary.txt\nThe table can be the full table for a given group, or\na custom subset, which can be selected on some\ninformative fields of the assembly table.\nWe recommend to select the subset of rows having\neither 'Complete genome' or 'Chromosome' in the 12th\ncolumn ('assembly_level'), and to add a filter on\n'refseq_category' in order to select 'representative\ngenome' or 'reference genome'.\nThe assembly table can also be filtered out manually\nin order to select a custom subset of the available\ngenomes.")), ToolInput(tag="in_refseq", input_type=Boolean(optional=True), prefix="-refseq", doc=InputDocumentation(doc="Local directory containing the mirror of the refseq\ngenomes found on the NCBI site:\nftp://ftp.ncbi.nih.gov/genomes/refseq\nNormally, the refseq directory is specified by\ndefining a global variable REFSEQ_DIR in the config\nfile. The option -refseq allows to overwrite this\nvalue.")), ToolInput(tag="in_skip", input_type=Boolean(optional=True), prefix="-skip", doc=InputDocumentation(doc="# Skip the first # organisms of the list.\nThis option is convenient to resume the installation\nof a list of organisms, when it has been interrupted.\nIt also allows to organize the installation by chunks\nof organisms.")), ToolInput(tag="in_skip_species", input_type=Boolean(optional=True), prefix="-skip_species", doc=InputDocumentation(doc="# skip the first # of the list of NCBI species.\nNote that one species can correspond to several\norganisms, if several strains have been sequenced.")), ToolInput(tag="in_last", input_type=Boolean(optional=True), prefix="-last", doc=InputDocumentation(doc="# Stop after the first # organisms of the list.\nThis option can be used to organize the installation\nby chunks of organisms.")), ToolInput(tag="in_last_species", input_type=Boolean(optional=True), prefix="-last_species", doc=InputDocumentation(doc="# Stop after the first # species of the NCBI")), ToolInput(tag="in_source", input_type=String(optional=True), prefix="-source", doc=InputDocumentation(doc="source")), ToolInput(tag="in_dir", input_type=File(optional=True), prefix="-dir", doc=InputDocumentation(doc="{install}\nAbsolute path of the installation directory.\nBEWARE : you should provide the absolute path of the\ninstallation directory, not the relative path.")), ToolInput(tag="in_batch", input_type=String(optional=True), prefix="-batch", doc=InputDocumentation(doc="some tasks (for example the calibration of oligos\nand dyads) in batch mode.  This options works on our\nlab cluster, but could be adapted for other\nconfigurations by adapting the parameters in\nRSAT_config.props.")), ToolInput(tag="in_queue", input_type=Boolean(optional=True), prefix="-queue", doc=InputDocumentation(doc="specify a queue to send the job with the option -batch.\nOn this instance of RSAT, default queue is rsat\nType 'qstat -f' to get a list of existing queues.")), ToolInput(tag="in_specify_alternative_file", input_type=Boolean(optional=True), prefix="-config", doc=InputDocumentation(doc="Specify an alternative organism configuration file for the\ngenome to be installed.\nBy default, the organism configuration file is\n$RSAT/public_html/data/genomes/supported_organisms.pl")), ToolInput(tag="in_backup_org_table", input_type=Boolean(optional=True), prefix="-backup_org_table", doc=InputDocumentation(doc="Store a backup copy of the organism table before\noverwriting it.")), ToolInput(tag="in_local", input_type=Boolean(optional=True), prefix="-local", doc=InputDocumentation(doc="Absolute path of a RSA local config file.\nBy default, the newly installed organism is added to\nthe main RSA config file is changed (provided the user\nhas write access to the RSA config file).\nIn addition to the organisms installed by the RSAT\nsystem administrator (found in\n/usr/local/share/rsat/public_html/data/supported_organisms.tab), users can\ninstall some organisms locally.\nFor this, the user must first define an environment\nvariable called RSA_LOCAL_CONFIG, and indicating the\nabsolute path of the local config file.\nE.g.\nexport RSA_LOCAL_CONFIG=/home/fred/RSA.local.config\nWhen install-organism is called with the option\n-local, the new organism is added to the file\nindicated by the environment variable RSA_LOCAL_CONFIG\nrather than the main RSA config file.")), ToolInput(tag="in_syn", input_type=Boolean(optional=True), prefix="-syn", doc=InputDocumentation(doc="synonym table\nA tab-delimited file containing two coloumns. The\nforst column contains a gene ID, the second a gene\nname.")), ToolInput(tag="in_up_from", input_type=Int(optional=True), prefix="-up_from", doc=InputDocumentation(doc="limit of the upstream regions (e.g. -800 for yeast)")), ToolInput(tag="in_up_to", input_type=Boolean(optional=True), prefix="-up_to", doc=InputDocumentation(doc="proximal limit of the upstream regions (e.g. -1)")), ToolInput(tag="in_genbank", input_type=Directory(optional=True), prefix="-genbank", doc=InputDocumentation(doc="directory (obsolete, see refseq)")), ToolInput(tag="in_pref_id", input_type=String(optional=True), prefix="-prefid", doc=InputDocumentation(doc="idname\npassed to parse-genbank.pl")), ToolInput(tag="in_date", input_type=String(optional=True), prefix="-date", doc=InputDocumentation(doc="Force the 'last_update' attribute to a given date.\nThis option is used by download-organism to ensure\nthat the local genome has the same installation date\nas the server, rather than using the date of download\nas update date.")), ToolInput(tag="in_ensembl", input_type=Boolean(optional=True), prefix="-ensembl", doc=InputDocumentation(doc="ENSEMBL directory. Directory containing the ENSEMBL\nflat files in Genbank format (ext .dat)\nExample:\nftp.ensembl.org/pub/current_worm/data/flatfiles/genbank")), ToolInput(tag="in_task", input_type=Boolean(optional=True), prefix="-task", doc=InputDocumentation(doc="specification of a single installation task")), ToolInput(tag="in_rm", input_type=Boolean(optional=True), prefix="-rm", doc=InputDocumentation(doc="calibrate oligo and dyad frequncies on repeat masked\nsequences, in addition to the non-masked sequences.")), ToolInput(tag="in_img_format", input_type=Boolean(optional=True), prefix="-img_format", doc=InputDocumentation(doc="image format for the graphs of sequence length distribution")), ToolInput(tag="in_install_organism", input_type=String(), position=0, doc=InputDocumentation(doc="AUTHOR")), ToolInput(tag="in_list_dot", input_type=String(), position=0, doc=InputDocumentation(doc="-organism Full name of the organism (e.g. 'Saccharomyces")), ToolInput(tag="in_e_dot_gdot", input_type=String(), position=0, doc=InputDocumentation(doc="install-organism -task dyads")), ToolInput(tag="in_available", input_type=String(), position=0, doc=InputDocumentation(doc="list available species on the NCBI FTP site.\nThis task cannot be combine with any\nother task.")), ToolInput(tag="in_download", input_type=String(), position=1, doc=InputDocumentation(doc="download genome from the NCBI FTP site\n(ftp://ftp.ncbi.nih.gov/genomes/refseq/)\nvia the rsync protocol.")), ToolInput(tag="in_list", input_type=String(), position=2, doc=InputDocumentation(doc="print the list of downloaded strains for the\nuser-specified species.")), ToolInput(tag="in_parse", input_type=String(), position=3, doc=InputDocumentation(doc="parse genome data (sequence, features) from \ngenbank-formatted flat files (gbff)\ndownloaded from NCBI. Parsing is ensured\nby the program parse-genbank.pl.")), ToolInput(tag="in_update_configuration_file", input_type=String(), position=4, doc=InputDocumentation(doc="update configuration file")), ToolInput(tag="in_start_stop", input_type=String(), position=5, doc=InputDocumentation(doc="calculate start and stop codon")), ToolInput(tag="in_frequencies", input_type=String(), position=6, doc=InputDocumentation(doc="allup        retrieve all upstream sequences")), ToolInput(tag="in_genome_segments", input_type=String(), position=7, doc=InputDocumentation(doc="retrieve sequences and limits of genome segments\n(intergenic, genic)")), ToolInput(tag="in_oligos", input_type=String(), position=8, doc=InputDocumentation(doc="calculate oligonucleotide frequencies\nThis task requires to specify, in\naddition, the type(s) of sequences for\nwhich oligo frequencies have to be\ncalculated (upstream_freq,\nintergenic_freq, genome_freq).")), ToolInput(tag="in_dyads", input_type=String(), position=9, doc=InputDocumentation(doc="calculate dyad frequencies\nThis task requires to specify, in\naddition, the type(s) of sequences for\nwhich dyad frequencies have to be\ncalculated (upstream_freq,\nintergenic_freq, genome_freq).")), ToolInput(tag="in_ncf", input_type=String(), position=10, doc=InputDocumentation(doc="calculate oligo and dyad frequencies in\nintergenic segments")), ToolInput(tag="in_upstream_freq", input_type=String(), position=11, doc=InputDocumentation(doc="calculate oligo and dyad frequencies for\nall upstream sequences")), ToolInput(tag="in_intergenic_freq", input_type=String(), position=12, doc=InputDocumentation(doc="calculate oligo and dyad frequencies for\nall intergenic sequences")), ToolInput(tag="in_genome_freq", input_type=String(), position=13, doc=InputDocumentation(doc="calculate oligo and dyad frequencies for\nthe whole genome sequence. This is not\nrecommended for higher organisms, where\nthe genome represents several Gigabases,\nand the computation of all oligo and dyad\nfrequencies might take ages.")), ToolInput(tag="in_clean", input_type=String(), position=14, doc=InputDocumentation(doc="remove unnecessary sequence files")), ToolInput(tag="in_species_dot", input_type=String(), position=0, doc=InputDocumentation(doc="install-organism -v 1 -group fungi -task download ")), ToolInput(tag="in_parallel_dot", input_type=String(), position=0, doc=InputDocumentation(doc="install-organism -v 1 ")), ToolInput(tag="in_download_organisms", input_type=String(), position=0, doc=InputDocumentation(doc="The program I<install-organism> performs all the formatting\nand calibration tasks for importing genomes from the reference\ndatabases (NCBI, EMBL) to RSAT.\nThe program I<download-organism> transfers the RSAT-formatted\ngenomes from a RSAT server.\nIf a genome is available on the RSAT server, it is recommended\nto use download-genomes in order to obtain it immediately in\nthe RSAT format, rather than install-genomes.\n"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Rsat_Install_Organism_V0_1_0().translate("wdl", allow_empty_container=True)

