class: CommandLineTool
id: ../../../../home/ubuntu/BiocondaCli/anvi_get_short_reads_from_bam.cwl
inputs:
- id: profile_db
  doc: Anvi'o profile database
  type: string
  inputBinding:
    prefix: --profile-db
- id: contigs_db
  doc: Anvi'o contigs database generated by 'anvi-gen- contigs'
  type: string
  inputBinding:
    prefix: --contigs-db
- id: collection_name
  doc: Collection name.
  type: string
  inputBinding:
    prefix: --collection-name
- id: bin_id
  doc: Bin name you are interested in.
  type: string
  inputBinding:
    prefix: --bin-id
- id: bin_ids_file
  doc: Text file for bins (each line should be a unique bin id).
  type: File
  inputBinding:
    prefix: --bin-ids-file
- id: output_file
  doc: File path to store results.
  type: File
  inputBinding:
    prefix: --output-file
- id: output_file_prefix
  doc: A prefix to be used while naming the output files (no file type extensions
    please; just a prefix).
  type: File
  inputBinding:
    prefix: --output-file-prefix
- id: gzip_output
  doc: When declared, output file(s) will be gzip compressed and the extension `.gz`
    will be added.
  type: boolean
  inputBinding:
    prefix: --gzip-output
- id: split_r_one_and_r_two
  doc: 'When declared, this program outputs 3 FASTA files for paired-end reads: one
    for R1, one for R2, and one for unpaired reads.'
  type: boolean
  inputBinding:
    prefix: --split-R1-and-R2
outputs: []
cwlVersion: v1.1
baseCommand:
- anvi-get-short-reads-from-bam
