from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Int, String, Boolean

Concoct_V0_1_0 = CommandToolBuilder(tool="concoct", base_command=["concoct"], inputs=[ToolInput(tag="in_coverage_file", input_type=File(optional=True), prefix="--coverage_file", doc=InputDocumentation(doc="specify the coverage file, containing a table where\neach row correspond to a contig, and each column\ncorrespond to a sample. The values are the average\ncoverage for this contig in that sample. All values\nare separated with tabs.")), ToolInput(tag="in_composition_file", input_type=File(optional=True), prefix="--composition_file", doc=InputDocumentation(doc="specify the composition file, containing sequences in\nfasta format. It is named the composition file since\nit is used to calculate the kmer composition (the\ngenomic signature) of each contig.")), ToolInput(tag="in_clusters", input_type=Int(optional=True), prefix="--clusters", doc=InputDocumentation(doc="specify maximal number of clusters for VGMM, default\n400.")), ToolInput(tag="in_km_er_length", input_type=Int(optional=True), prefix="--kmer_length", doc=InputDocumentation(doc="specify kmer length, default 4.")), ToolInput(tag="in_threads", input_type=Int(optional=True), prefix="--threads", doc=InputDocumentation(doc="Number of threads to use")), ToolInput(tag="in_length_threshold", input_type=Int(optional=True), prefix="--length_threshold", doc=InputDocumentation(doc="specify the sequence length threshold, contigs shorter\nthan this value will not be included. Defaults to\n1000.")), ToolInput(tag="in_read_length", input_type=Int(optional=True), prefix="--read_length", doc=InputDocumentation(doc="specify read length for coverage, default 100")), ToolInput(tag="in_total_percentage_pc_a", input_type=String(optional=True), prefix="--total_percentage_pca", doc=InputDocumentation(doc="The percentage of variance explained by the principal\ncomponents for the combined data.")), ToolInput(tag="in_basename", input_type=File(optional=True), prefix="--basename", doc=InputDocumentation(doc="Specify the basename for files or directory where\noutputwill be placed. Path to existing directory or\nbasenamewith a trailing '/' will be interpreted as a\ndirectory.If not provided, current directory will be\nused.")), ToolInput(tag="in_seed", input_type=Int(optional=True), prefix="--seed", doc=InputDocumentation(doc="Specify an integer to use as seed for clustering. 0\ngives a random seed, 1 is the default seed and any\nother positive integer can be used. Other values give\nArgumentTypeError.")), ToolInput(tag="in_iterations", input_type=Int(optional=True), prefix="--iterations", doc=InputDocumentation(doc="Specify maximum number of iterations for the VBGMM.\nDefault value is 500")), ToolInput(tag="in_no_cov_normalization", input_type=Boolean(optional=True), prefix="--no_cov_normalization", doc=InputDocumentation(doc="By default the coverage is normalized with regards to\nsamples, then normalized with regards of contigs and\nfinally log transformed. By setting this flag you skip\nthe normalization and only do log transorm of the\ncoverage.")), ToolInput(tag="in_no_total_coverage", input_type=Boolean(optional=True), prefix="--no_total_coverage", doc=InputDocumentation(doc="By default, the total coverage is added as a new\ncolumn in the coverage data matrix, independently of\ncoverage normalization but previous to log\ntransformation. Use this tag to escape this behaviour.")), ToolInput(tag="in_no_original_data", input_type=Boolean(optional=True), prefix="--no_original_data", doc=InputDocumentation(doc="By default the original data is saved to disk. For big\ndatasets, especially when a large k is used for\ncompositional data, this file can become very large.\nUse this tag if you don't want to save the original\ndata.")), ToolInput(tag="in_converge_out", input_type=Boolean(optional=True), prefix="--converge_out", doc=InputDocumentation(doc="Write convergence info to files.")), ToolInput(tag="in_debug", input_type=Boolean(optional=True), prefix="--debug", doc=InputDocumentation(doc="Debug parameters."))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Concoct_V0_1_0().translate("wdl", allow_empty_container=True)

