from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, Int

Ustacks_V0_1_0 = CommandToolBuilder(tool="ustacks", base_command=["ustacks"], inputs=[ToolInput(tag="in_name", input_type=Boolean(optional=True), prefix="--name", doc=InputDocumentation(doc=": a name for the sample (default: input file name minus the suffix).")), ToolInput(tag="in_keep_high_cov", input_type=Boolean(optional=True), prefix="--keep-high-cov", doc=InputDocumentation(doc=": disable the algorithm that removes highly-repetitive stacks and nearby errors.")), ToolInput(tag="in_high_cov_th_res", input_type=Boolean(optional=True), prefix="--high-cov-thres", doc=InputDocumentation(doc=": highly-repetitive stacks threshold, in standard deviation units (default: 3.0).")), ToolInput(tag="in_max_locus_stacks", input_type=Int(optional=True), prefix="--max-locus-stacks", doc=InputDocumentation(doc=": maximum number of stacks at a single de novo locus (default 3).")), ToolInput(tag="in_k_len", input_type=Int(optional=True), prefix="--k-len", doc=InputDocumentation(doc=": specify k-mer size for matching between alleles and loci (automatically calculated by default).")), ToolInput(tag="in_max_gaps", input_type=Boolean(optional=True), prefix="--max-gaps", doc=InputDocumentation(doc=": number of gaps allowed between stacks before merging (default: 2).")), ToolInput(tag="in_min_aln_len", input_type=Boolean(optional=True), prefix="--min-aln-len", doc=InputDocumentation(doc=": minimum length of aligned sequence in a gapped alignment (default: 0.80).")), ToolInput(tag="in_disable_gapped", input_type=Boolean(optional=True), prefix="--disable-gapped", doc=InputDocumentation(doc=": do not preform gapped alignments between stacks (default: gapped alignements enabled).")), ToolInput(tag="in_model_type", input_type=Boolean(optional=True), prefix="--model-type", doc=InputDocumentation(doc=": either 'snp' (default), 'bounded', or 'fixed'")), ToolInput(tag="in_alpha", input_type=Int(optional=True), prefix="--alpha", doc=InputDocumentation(doc=": chi square significance level required to call a heterozygote or homozygote, either 0.1, 0.05 (default), 0.01, or 0.001.")), ToolInput(tag="in_bound_low", input_type=Int(optional=True), prefix="--bound-low", doc=InputDocumentation(doc=": lower bound for epsilon, the error rate, between 0 and 1.0 (default 0).")), ToolInput(tag="in_bound_high", input_type=Int(optional=True), prefix="--bound-high", doc=InputDocumentation(doc=": upper bound for epsilon, the error rate, between 0 and 1.0 (default 1).")), ToolInput(tag="in_bc_err_freq", input_type=Int(optional=True), prefix="--bc-err-freq", doc=InputDocumentation(doc=": specify the barcode error frequency, between 0 and 1.0."))], outputs=[], container="quay.io/biocontainers/stacks:2.55--he513fc3_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Ustacks_V0_1_0().translate("wdl")

