from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, File, String

Soap_V0_1_0 = CommandToolBuilder(tool="soap", base_command=["soap"], inputs=[ToolInput(tag="in_str_query_file", input_type=Boolean(optional=True), prefix="-a", doc=InputDocumentation(doc="<str>   query a file, *.fq, *.fa")), ToolInput(tag="in_str_query_b", input_type=Boolean(optional=True), prefix="-b", doc=InputDocumentation(doc="<str>   query b file")), ToolInput(tag="in_str_sequences_table", input_type=Boolean(optional=True), prefix="-D", doc=InputDocumentation(doc="<str>   reference sequences indexing table, *.index format")), ToolInput(tag="in_str_output_alignment", input_type=File(optional=True), prefix="-o", doc=InputDocumentation(doc="<str>   output alignment file(txt)")), ToolInput(tag="in_int_match_mode", input_type=Boolean(optional=True), prefix="-M", doc=InputDocumentation(doc="<int>   match mode for each read or the seed part of read, which shouldn't contain more than 2 mismaches, [4]\n0: exact match only\n1: 1 mismatch match only\n2: 2 mismatch match only\n4: find the best hits")), ToolInput(tag="in_str_output_unmapped", input_type=File(optional=True), prefix="-u", doc=InputDocumentation(doc="<str>   output unmapped reads file")), ToolInput(tag="in_output_reads_reads", input_type=Boolean(optional=True), prefix="-t", doc=InputDocumentation(doc="output reads id instead reads name, [none]")), ToolInput(tag="in_int_align_initial", input_type=Boolean(optional=True), prefix="-l", doc=InputDocumentation(doc="<int>   align the initial n bps as a seed [256] means whole length of read")), ToolInput(tag="in_int_filter_lowquality", input_type=Boolean(optional=True), prefix="-n", doc=InputDocumentation(doc="<int>   filter low-quality reads containing >n Ns before alignment, [5]")), ToolInput(tag="in_var_9", input_type=Boolean(optional=True), prefix="-r", doc=InputDocumentation(doc="[0,1,2] how to report repeat hits, 0=none; 1=random one; 2=all, [1]")), ToolInput(tag="in_int_minimal_insert", input_type=Boolean(optional=True), prefix="-m", doc=InputDocumentation(doc="<int>   minimal insert size allowed, [400]")), ToolInput(tag="in_int_maximal_insert", input_type=Boolean(optional=True), prefix="-x", doc=InputDocumentation(doc="<int>   maximal insert size allowed, [600]")), ToolInput(tag="in_str_output_file", input_type=File(optional=True), prefix="-2", doc=InputDocumentation(doc="<str>   output file of unpaired alignment hits")), ToolInput(tag="in_int_maximum_number", input_type=Boolean(optional=True), prefix="-v", doc=InputDocumentation(doc="<int>   maximum number of mismatches allowed on a read. [5] bp")), ToolInput(tag="in_int_minimal_alignment", input_type=Boolean(optional=True), prefix="-s", doc=InputDocumentation(doc="<int>   minimal alignment length (for soft clip) [255] bp")), ToolInput(tag="in_int_one_continuous", input_type=Boolean(optional=True), prefix="-g", doc=InputDocumentation(doc="<int>   one continuous gap size allowed on a read. [0] bp")), ToolInput(tag="in_long_insert_size", input_type=Boolean(optional=True), prefix="-R", doc=InputDocumentation(doc="for long insert size of pair end reads RF. [none](means FR pair)")), ToolInput(tag="in_int_allow_exist", input_type=Boolean(optional=True), prefix="-e", doc=InputDocumentation(doc="<int>   will not allow gap exist inside n-bp edge of a read, default=5")), ToolInput(tag="in_int_number_use", input_type=Boolean(optional=True), prefix="-p", doc=InputDocumentation(doc="<int>   number of processors to use, [1]")), ToolInput(tag="in_soap_at_genomics_dot_org_dot_cn", input_type=String(), position=0, doc=InputDocumentation(doc="Usage:  soap [options]"))], outputs=[ToolOutput(tag="out_str_output_alignment", output_type=File(optional=True), selector=InputSelector(input_to_select="in_str_output_alignment", type_hint=File()), doc=OutputDocumentation(doc="<str>   output alignment file(txt)")), ToolOutput(tag="out_str_output_unmapped", output_type=File(optional=True), selector=InputSelector(input_to_select="in_str_output_unmapped", type_hint=File()), doc=OutputDocumentation(doc="<str>   output unmapped reads file")), ToolOutput(tag="out_str_output_file", output_type=File(optional=True), selector=InputSelector(input_to_select="in_str_output_file", type_hint=File()), doc=OutputDocumentation(doc="<str>   output file of unpaired alignment hits"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Soap_V0_1_0().translate("wdl", allow_empty_container=True)

