from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Int, Boolean, String

Meryl_Lookup_V0_1_0 = CommandToolBuilder(tool="meryl_lookup", base_command=["meryl-lookup"], inputs=[ToolInput(tag="in_sequence", input_type=Int(optional=True), prefix="-sequence", doc=InputDocumentation(doc="[<input2.fasta>] \")), ToolInput(tag="in_output", input_type=Boolean(optional=True), prefix="-output", doc=InputDocumentation(doc="<output1>      [<output2>]")), ToolInput(tag="in_min", input_type=Boolean(optional=True), prefix="-min", doc=InputDocumentation(doc="m    Ignore kmers with value below m")), ToolInput(tag="in_max", input_type=Boolean(optional=True), prefix="-max", doc=InputDocumentation(doc="m    Ignore kmers with value above m")), ToolInput(tag="in_threads", input_type=Int(optional=True), prefix="-threads", doc=InputDocumentation(doc="Number of threads to use when constructing lookup table.")), ToolInput(tag="in_memory", input_type=String(optional=True), prefix="-memory", doc=InputDocumentation(doc="Don't use more than m GB memory")), ToolInput(tag="in_existence", input_type=Boolean(optional=True), prefix="-existence", doc=InputDocumentation(doc="Report a tab-delimited line for each sequence showing the number of kmers\nin the sequence, in the database, and in both.\nMultiple input -mers may be supplied.  If no output is supplied, output is written\nto stdout.\noutput:  seqName <tab> mersInSeq <tab> mersInDB1 <tab> mersInSeq&DB1 [ <tab> mersInDB2 <tab> mersInSeq&DB2 ... ]\nseqName      - name of the sequence\nmersInSeq    - number of mers in the sequence\nmersInDB     - number of mers in the meryl database\nmersInSeq&DB - number of mers in the sequence that are\nalso in the database")), ToolInput(tag="in_dump", input_type=Boolean(optional=True), prefix="-dump", doc=InputDocumentation(doc="Report a tab-delimited line reporting each kmer in the input sequences, in\norder, annotated with the value of the kmer in the input database.  If the kmer\ndoes not exist in the database its value will be reported as zero.\nOnly one input may be supplied.  If no output is supplied, output is written\nto stdout.\noutput:  seqName <tab> seqId <tab> seqPos <tab> exists <tab> fwd-mer <tab> fwd-val <tab> rev-mer <tab> rev-val\nseqName    - name of the sequence this kmer is from\nseqId      - numeric version of the seqName (0-based)\nseqPos     - start position (0-based) of the kmer in the sequence\nexists     - 'T' if the kmer exists in the database, 'F' if it does not\nfwd-mer    - forward mer sequence\nfwd-val    - value of the forward mer in the database\nrev-mer    - reverse mer sequence\nrev-val    - value of the reverse mer in the database")), ToolInput(tag="in_include", input_type=Boolean(optional=True), prefix="-include", doc=InputDocumentation(doc="Extract sequences containing (-include) or not containing (-exclude) kmers in\nany input database.  Output sequences are written in the same format as the input\nsequences, with the number of kmers found added to the name.\nIf two input files are supplied, the corresponding sequences are treated as a pair,\nand two output files MUST be supplied.\noutput:  sequence given format (fasta or fastq) with the number of overlapping kmers appended\nif pairs of sequences are given, R1 will be stdout and R2 be named as <output.r2>\n<output.r2> will be automatically compressed if ends with .gz, .bz2, or xs\nseqName    - name of the sequence this kmer is from\nmersInBoth - number of mers in both sequence and in the database")), ToolInput(tag="in_speed_dot", input_type=String(), position=0, doc=InputDocumentation(doc="If the lookup table requires more memory than allowed, the program"))], outputs=[], container="quay.io/biocontainers/meryl:1.2--he1b5a44_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Meryl_Lookup_V0_1_0().translate("wdl")

