from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, String, Boolean

Jannovar_Hgvs_To_Vcf_V0_1_0 = CommandToolBuilder(tool="jannovar_hgvs_to_vcf", base_command=["jannovar", "hgvs-to-vcf"], inputs=[ToolInput(tag="in_reference_fast_a", input_type=File(optional=True), prefix="--reference-fasta", doc=InputDocumentation(doc="Path to reference FASTA file")), ToolInput(tag="in_database", input_type=File(optional=True), prefix="--database", doc=InputDocumentation(doc="Path to database .ser file")), ToolInput(tag="in_input_txt", input_type=File(optional=True), prefix="--input-txt", doc=InputDocumentation(doc="Input file  with  HGVS  transcript-level  changes,\nline-by-line")), ToolInput(tag="in_output_vcf", input_type=File(optional=True), prefix="--output-vcf", doc=InputDocumentation(doc="Output VCF file with chromosome-level changes")), ToolInput(tag="in_show_all", input_type=String(optional=True), prefix="--show-all", doc=InputDocumentation(doc="Show all effects")), ToolInput(tag="in_no_three_prime_shifting", input_type=Boolean(optional=True), prefix="--no-3-prime-shifting", doc=InputDocumentation(doc="Disable shifting towards 3' of transcript")), ToolInput(tag="in_three_letter_amino_acids", input_type=Boolean(optional=True), prefix="--3-letter-amino-acids", doc=InputDocumentation(doc="Enable usage of 3 letter amino acid codes")), ToolInput(tag="in_report_no_progress", input_type=Boolean(optional=True), prefix="--report-no-progress", doc=InputDocumentation(doc="Disable progress report, more quiet mode")), ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="--verbose", doc=InputDocumentation(doc="Enable verbose mode")), ToolInput(tag="in_very_verbose", input_type=Boolean(optional=True), prefix="--very-verbose", doc=InputDocumentation(doc="Enable very verbose mode")), ToolInput(tag="in_http_proxy", input_type=String(optional=True), prefix="--http-proxy", doc=InputDocumentation(doc="Set HTTP proxy to use, if any")), ToolInput(tag="in_https_proxy", input_type=String(optional=True), prefix="--https-proxy", doc=InputDocumentation(doc="Set HTTPS proxy to use, if any")), ToolInput(tag="in_ftp_proxy", input_type=String(optional=True), prefix="--ftp-proxy", doc=InputDocumentation(doc="Set FTP proxy to use, if any")), ToolInput(tag="in_jan_novar_cli", input_type=String(), position=0, doc=InputDocumentation(doc="")), ToolInput(tag="in_hgvs_to_vcf", input_type=String(), position=1, doc=InputDocumentation(doc=""))], outputs=[ToolOutput(tag="out_output_vcf", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output_vcf", type_hint=File()), doc=OutputDocumentation(doc="Output VCF file with chromosome-level changes"))], container="quay.io/biocontainers/jannovar-cli:0.35--0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Jannovar_Hgvs_To_Vcf_V0_1_0().translate("wdl")

