from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Int, Boolean, String, Float

Yass_V0_1_0 = CommandToolBuilder(tool="yass", base_command=["yass"], inputs=[ToolInput(tag="in_display_alignment_positions", input_type=File(optional=True), prefix="-d", doc=InputDocumentation(doc="0 : Display alignment positions (kept for compatibility)\n1 : Display alignment positions + alignments + stats (default)\n2 : Display blast-like tabular output\n3 : Display light tabular output (better for post-processing)\n4 : Display BED file output\n5 : Display PSL file output")), ToolInput(tag="in_process_forward_query", input_type=Int(optional=True), prefix="-r", doc=InputDocumentation(doc="0 : process forward (query) strand\n1 : process Reverse complement strand\n2 : process both forward and Reverse complement strands (default)")), ToolInput(tag="in_output_file", input_type=File(optional=True), prefix="-o", doc=InputDocumentation(doc="Output file")), ToolInput(tag="in_mask_regions_seed", input_type=Boolean(optional=True), prefix="-l", doc=InputDocumentation(doc="mask Lowercase regions (seed algorithm only)")), ToolInput(tag="in_sort_according_alignment", input_type=Int(optional=True), prefix="-s", doc=InputDocumentation(doc="Sort according to\n0 : alignment scores\n1 : entropy\n2 : mutual information (experimental)\n3 : both entropy and score\n4 : positions on the 1st file\n5 : positions on the 2nd file\n6 : alignment % id\n7 : 1st file sequence % id\n8 : 2nd file sequence % id\n10-18 : (0-8) + sort by first fasta chunks first\n20-28 : (0-8) + sort by second fasta chunks first\n30-38 : (0-8) + sort by both first/second chunks first\n40-48 : (10-18) + sort chunk results with (0-8) criterion (first fasta file)\n50-58 : (20-28) + sort chunk results with (0-8) criterion (second fasta file)\n60-68 : (30-38) + sort chunk results with (0-8) criterion (first fasta file)\n70-75 : (30-38) + sort chunk results with (0-8) criterion (second fasta file)\n80-85 : (30-38) + sort chunk results with (0-8) criterion (both fasta files)")), ToolInput(tag="in_display_current_version", input_type=Boolean(optional=True), prefix="-v", doc=InputDocumentation(doc="display the current Version")), ToolInput(tag="in_select_scoring_default", input_type=Int(optional=True), prefix="-M", doc=InputDocumentation(doc="select a scoring Matrix (default 3):\n[Match,Transversion,Transition],(Gopen,Gext)\n0 : [  1, -3, -2],( -8, -2)   1 : [  2, -3, -2],(-12, -4)\n2 : [  3, -3, -2],(-16, -4)   3 : [  5, -4, -3],(-16, -4)\n4 : [  5, -4, -2],(-16, -4)")), ToolInput(tag="in_reset_matchmismatchtransistionother_costs", input_type=Int(optional=True), prefix="-C", doc=InputDocumentation(doc="[,<int>[,<int>[,<int>]]]\nreset match/mismatch/transistion/other Costs (penalties)\nyou can also give the 16 values of matrix (ACGT order)")), ToolInput(tag="in_int_reset_gap", input_type=Int(optional=True), prefix="-G", doc=InputDocumentation(doc=",<int> reset Gap opening/extension penalties")), ToolInput(tag="in_real_reset_lambda", input_type=String(optional=True), prefix="-L", doc=InputDocumentation(doc=",<real> reset Lambda and K parameters of Gumbel law")), ToolInput(tag="in_xdrop_threshold_score", input_type=Int(optional=True), prefix="-X", doc=InputDocumentation(doc="Xdrop threshold score (default 25)")), ToolInput(tag="in_evalue_threshold_default", input_type=Int(optional=True), prefix="-E", doc=InputDocumentation(doc="E-value threshold (default 10)")), ToolInput(tag="in_low_complexity_filter", input_type=Float(optional=True), prefix="-e", doc=InputDocumentation(doc="low complexity filter :\nminimal allowed Entropy of trinucleotide distribution\nranging between 0 (no filter) and 6 (default 2.80)")), ToolInput(tag="in_limit_number_output", input_type=Int(optional=True), prefix="-O", doc=InputDocumentation(doc="limit number of Output alignments (default 1000000)")), ToolInput(tag="in_select_sequence_first", input_type=Int(optional=True), prefix="-S", doc=InputDocumentation(doc="Select sequence from the first multi-fasta file (default 0)\n* use 0 to select the full first multi-fasta file")), ToolInput(tag="in_forbid_aligning_regions", input_type=Int(optional=True), prefix="-T", doc=InputDocumentation(doc="forbid aligning too close regions (e.g. Tandem repeats)\nvalid for single sequence comparison only (default 16 bp)")), ToolInput(tag="in_seed_patterns_use", input_type=Int(optional=True), prefix="-p", doc=InputDocumentation(doc="seed Pattern(s)\n* use '#' for match\n* use '@' for match or transition\n* use '-' or '_' for joker\n* use ',' for seed separator (max: 32 seeds)\n- example with one seed :\nyass file.fas -p  '#@#--##--#-##@#'\n- example with two complementary seeds :\nyass file.fas -p '##-#-#@#-##@-##,##@#--@--##-#--###'\n(default  '###-#@-##@##,###--#-#--#-###')")), ToolInput(tag="in_seed_hit_consider", input_type=Int(optional=True), prefix="-c", doc=InputDocumentation(doc="seed hit Criterion : 1 or 2 seeds to consider a hit (default 2)")), ToolInput(tag="in_trim_overrepresented_seeds", input_type=Float(optional=True), prefix="-t", doc=InputDocumentation(doc="Trim out over-represented seeds codes\nranging between 0.0 (no trim) and +inf (default 0.001)")), ToolInput(tag="in_statistical_tolerance_alpha", input_type=Int(optional=True), prefix="-a", doc=InputDocumentation(doc="statistical tolerance Alpha (%) (default 5%)")), ToolInput(tag="in_indel_rate_default", input_type=Int(optional=True), prefix="-i", doc=InputDocumentation(doc="Indel rate (%)                  (default 8%)")), ToolInput(tag="in_mutation_rate_default", input_type=Int(optional=True), prefix="-m", doc=InputDocumentation(doc="Mutation rate (%)               (default 25%)")), ToolInput(tag="in_int_window_minmax", input_type=Int(optional=True), prefix="-W", doc=InputDocumentation(doc=",<int> Window <min,max> range for post-processing and grouping\nalignments (default <64,65536>)")), ToolInput(tag="in_window_size_coefficient", input_type=Int(optional=True), prefix="-w", doc=InputDocumentation(doc="Window size coefficient for post-processing and grouping\nalignments (default 16)\nNOTE : -w 0 disables post-processing\n"))], outputs=[ToolOutput(tag="out_display_alignment_positions", output_type=File(optional=True), selector=InputSelector(input_to_select="in_display_alignment_positions", type_hint=File()), doc=OutputDocumentation(doc="0 : Display alignment positions (kept for compatibility)\n1 : Display alignment positions + alignments + stats (default)\n2 : Display blast-like tabular output\n3 : Display light tabular output (better for post-processing)\n4 : Display BED file output\n5 : Display PSL file output")), ToolOutput(tag="out_output_file", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output_file", type_hint=File()), doc=OutputDocumentation(doc="Output file"))], container="quay.io/biocontainers/yass:1.14--h516909a_2", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Yass_V0_1_0().translate("wdl")

