from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Boolean, Int, Directory, String

Qualimap_Bamqc_V0_1_0 = CommandToolBuilder(tool="qualimap_bamqc", base_command=["qualimap", "bamqc"], inputs=[ToolInput(tag="in_bam", input_type=File(optional=True), prefix="-bam", doc=InputDocumentation(doc="Input mapping file in BAM format")), ToolInput(tag="in_paint_chromosome_limits", input_type=Boolean(optional=True), prefix="--paint-chromosome-limits", doc=InputDocumentation(doc="Paint chromosome limits inside charts")), ToolInput(tag="in_genome_gc_distr", input_type=Int(optional=True), prefix="--genome-gc-distr", doc=InputDocumentation(doc="Species to compare with genome GC\ndistribution. Possible values: HUMAN -\nhg19; MOUSE - mm9(default), mm10")), ToolInput(tag="in_feature_file", input_type=File(optional=True), prefix="--feature-file", doc=InputDocumentation(doc="Feature file with regions of interest in\nGFF/GTF or BED format")), ToolInput(tag="in_hm", input_type=Int(optional=True), prefix="-hm", doc=InputDocumentation(doc="Minimum size for a homopolymer to be\nconsidered in indel analysis (default is\n3)")), ToolInput(tag="in_collect_overlap_pairs", input_type=Boolean(optional=True), prefix="--collect-overlap-pairs", doc=InputDocumentation(doc="Activate this option to collect statistics\nof overlapping paired-end reads")), ToolInput(tag="in_nr", input_type=Int(optional=True), prefix="-nr", doc=InputDocumentation(doc="Number of reads analyzed in a chunk\n(default is 1000)")), ToolInput(tag="in_nt", input_type=Int(optional=True), prefix="-nt", doc=InputDocumentation(doc="Number of threads (default is 8)")), ToolInput(tag="in_nw", input_type=Int(optional=True), prefix="-nw", doc=InputDocumentation(doc="Number of windows (default is 400)")), ToolInput(tag="in_output_genome_coverage", input_type=File(optional=True), prefix="--output-genome-coverage", doc=InputDocumentation(doc="File to save per base non-zero coverage.\nWarning: large files are expected for\nlarge genomes")), ToolInput(tag="in_outside_stats", input_type=Boolean(optional=True), prefix="--outside-stats", doc=InputDocumentation(doc="Report information for the regions outside\nthose defined by feature-file  (ignored\nwhen -gff option is not set)")), ToolInput(tag="in_outdir", input_type=Directory(optional=True), prefix="-outdir", doc=InputDocumentation(doc="Output folder for HTML report and raw")), ToolInput(tag="in_out_format", input_type=String(optional=True), prefix="-outformat", doc=InputDocumentation(doc="Format of the output report (PDF, HTML or\nboth PDF:HTML, default is HTML).")), ToolInput(tag="in_sequencing_protocol", input_type=String(optional=True), prefix="--sequencing-protocol", doc=InputDocumentation(doc="Sequencing library protocol:\nstrand-specific-forward,\nstrand-specific-reverse or\nnon-strand-specific (default)")), ToolInput(tag="in_skip_duplicated", input_type=Boolean(optional=True), prefix="--skip-duplicated", doc=InputDocumentation(doc="Activate this option to skip duplicated\nalignments from the analysis. If the\nduplicates are not flagged in the BAM\nfile, then they will be detected by\nQualimap and can be selected for skipping.")), ToolInput(tag="in_skip_dup_mode", input_type=Int(optional=True), prefix="--skip-dup-mode", doc=InputDocumentation(doc="Specific type of duplicated alignments to\nskip (if this option is activated).\n0 : only flagged duplicates (default)\n1 : only estimated by Qualimap\n2 : both flagged and estimated\n")), ToolInput(tag="in_data_dot", input_type=String(), position=0, doc=InputDocumentation(doc="-outfile <arg>                       Output file for PDF report (default value"))], outputs=[ToolOutput(tag="out_outdir", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_outdir", type_hint=File()), doc=OutputDocumentation(doc="Output folder for HTML report and raw"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Qualimap_Bamqc_V0_1_0().translate("wdl", allow_empty_container=True)

