from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Int, Boolean, File, String

Make_Html2_Pl_V0_1_0 = CommandToolBuilder(tool="make_html2.pl", base_command=["make_html2.pl"], inputs=[ToolInput(tag="in_only_output_hairpins", input_type=Int(optional=True), prefix="-v", doc=InputDocumentation(doc="only output hairpins with score above 2")), ToolInput(tag="in_also_create_overview", input_type=Boolean(optional=True), prefix="-c", doc=InputDocumentation(doc="also create overview in excel format.")), ToolInput(tag="in_supply_file_known", input_type=File(optional=True), prefix="-k", doc=InputDocumentation(doc="supply file with known miRNAs")), ToolInput(tag="in_supply_survey_file", input_type=File(optional=True), prefix="-s", doc=InputDocumentation(doc="supply survey file if score cutoff is used to get information about how big is the confidence of resulting reads")), ToolInput(tag="in_report_complete_file", input_type=Boolean(optional=True), prefix="-e", doc=InputDocumentation(doc="report complete survey file")), ToolInput(tag="in_report_survey_current", input_type=Boolean(optional=True), prefix="-g", doc=InputDocumentation(doc="report survey for current score cutoff")), ToolInput(tag="in_rfam_file_check", input_type=File(optional=True), prefix="-r", doc=InputDocumentation(doc="Rfam file to check for already reported small RNA sequences")), ToolInput(tag="in_mirbasemrd_file_produced", input_type=File(optional=True), prefix="-q", doc=InputDocumentation(doc="miRBase.mrd file produced by quantifier module")), ToolInput(tag="in_signaturearf_file_mapped", input_type=File(optional=True), prefix="-x", doc=InputDocumentation(doc="signature.arf file with mapped reads to precursors")), ToolInput(tag="in_specify_organism_your", input_type=String(optional=True), prefix="-t", doc=InputDocumentation(doc="specify the organism from which your sequencing data was obtained")), ToolInput(tag="in_print_available_organisms", input_type=Boolean(optional=True), prefix="-u", doc=InputDocumentation(doc="print all available UCSC input organisms")), ToolInput(tag="in_timestamp_of_run", input_type=Boolean(optional=True), prefix="-y", doc=InputDocumentation(doc="timestamp of this run")), ToolInput(tag="in_sort_signature_sample", input_type=Boolean(optional=True), prefix="-o", doc=InputDocumentation(doc="sort signature by sample in pdf file, default is by beginning position")), ToolInput(tag="in_do_generate_pdfs", input_type=Boolean(optional=True), prefix="-d", doc=InputDocumentation(doc="do not generate pdfs")), ToolInput(tag="in_print_genomic_coordinates", input_type=Boolean(optional=True), prefix="-a", doc=InputDocumentation(doc="print genomic coordinates of mature sequence (still testing)")), ToolInput(tag="in_be_stringent_assigning", input_type=Boolean(optional=True), prefix="-l", doc=InputDocumentation(doc="be stringent when assigning miRNA-precursor connections like mmu-mir only is assigned to mmu-precursor"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Make_Html2_Pl_V0_1_0().translate("wdl", allow_empty_container=True)

