from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, File

Pplacer_V0_1_0 = CommandToolBuilder(tool="pplacer", base_command=["pplacer"], inputs=[ToolInput(tag="in_specify_path_reference", input_type=Boolean(optional=True), prefix="-c", doc=InputDocumentation(doc="Specify the path to the reference package.")), ToolInput(tag="in_specify_tree_filename", input_type=Boolean(optional=True), prefix="-t", doc=InputDocumentation(doc="Specify the reference tree filename.")), ToolInput(tag="in_specify_alignment_filename", input_type=Boolean(optional=True), prefix="-r", doc=InputDocumentation(doc="Specify the reference alignment filename.")), ToolInput(tag="in_supply_phyml_statstxt", input_type=Boolean(optional=True), prefix="-s", doc=InputDocumentation(doc="Supply a phyml stats.txt or a RAxML info file giving the model parameters.")), ToolInput(tag="in_specify_directory_containing", input_type=Boolean(optional=True), prefix="-d", doc=InputDocumentation(doc="Specify the directory containing the reference information.")), ToolInput(tag="in_calculate_posterior_probabilities", input_type=Boolean(optional=True), prefix="-p", doc=InputDocumentation(doc="Calculate posterior probabilities.")), ToolInput(tag="in_substitution_model_protein", input_type=Boolean(optional=True), prefix="-m", doc=InputDocumentation(doc="Substitution model. Protein: are LG, WAG, or JTT. Nucleotides: GTR.")), ToolInput(tag="in_model_freqs", input_type=Boolean(optional=True), prefix="--model-freqs", doc=InputDocumentation(doc="Use model frequencies instead of reference alignment frequencies.")), ToolInput(tag="in_gamma_cats", input_type=Boolean(optional=True), prefix="--gamma-cats", doc=InputDocumentation(doc="Number of categories for discrete gamma model.")), ToolInput(tag="in_gamma_alpha", input_type=Boolean(optional=True), prefix="--gamma-alpha", doc=InputDocumentation(doc="Specify the shape parameter for a discrete gamma model.")), ToolInput(tag="in_ml_tolerance", input_type=Boolean(optional=True), prefix="--ml-tolerance", doc=InputDocumentation(doc="1st stage branch len optimization tolerance (2nd stage to 1e-5). Default: 0.01.")), ToolInput(tag="in_pp_rel_err", input_type=Boolean(optional=True), prefix="--pp-rel-err", doc=InputDocumentation(doc="Relative error for the posterior probability calculation. Default is 0.01.")), ToolInput(tag="in_unif_prior", input_type=Boolean(optional=True), prefix="--unif-prior", doc=InputDocumentation(doc="Use a uniform prior rather than exponential.")), ToolInput(tag="in_inform_prior", input_type=Boolean(optional=True), prefix="--inform-prior", doc=InputDocumentation(doc="Use an informative exponential prior based on rooted distance to leaves.")), ToolInput(tag="in_prior_lower", input_type=Boolean(optional=True), prefix="--prior-lower", doc=InputDocumentation(doc="Lower bound for the informative prior mean. Default is 0.")), ToolInput(tag="in_start_pend", input_type=Boolean(optional=True), prefix="--start-pend", doc=InputDocumentation(doc="Starting pendant branch length. Default is 0.1.")), ToolInput(tag="in_max_pend", input_type=Boolean(optional=True), prefix="--max-pend", doc=InputDocumentation(doc="Set the maximum ML pendant branch length. Default is 2.")), ToolInput(tag="in_fig_cut_off", input_type=Boolean(optional=True), prefix="--fig-cutoff", doc=InputDocumentation(doc="The cutoff for determining figs. Default is 0; specify 0 to disable.")), ToolInput(tag="in_fig_eval_all", input_type=Boolean(optional=True), prefix="--fig-eval-all", doc=InputDocumentation(doc="Evaluate all likelihoods to ensure that the best location was selected.")), ToolInput(tag="in_fig_eval_discrepancy_tree", input_type=Boolean(optional=True), prefix="--fig-eval-discrepancy-tree", doc=InputDocumentation(doc="Write out a tree showing the discrepancies between the best complete and observed locations.")), ToolInput(tag="in_fig_tree", input_type=Boolean(optional=True), prefix="--fig-tree", doc=InputDocumentation(doc="Write out a tree showing the figs on the tree.")), ToolInput(tag="in_max_strikes", input_type=Boolean(optional=True), prefix="--max-strikes", doc=InputDocumentation(doc="Maximum number of strikes for baseball. 0 -> no ball playing. Default is 6.")), ToolInput(tag="in_strike_box", input_type=Boolean(optional=True), prefix="--strike-box", doc=InputDocumentation(doc="Set the size of the strike box in log likelihood units. Default is 3.")), ToolInput(tag="in_max_pitches", input_type=Boolean(optional=True), prefix="--max-pitches", doc=InputDocumentation(doc="Set the maximum number of pitches for baseball. Default is 40.")), ToolInput(tag="in_fantasy", input_type=Boolean(optional=True), prefix="--fantasy", doc=InputDocumentation(doc="Desired likelihood cutoff for fantasy baseball mode. 0 -> no fantasy.")), ToolInput(tag="in_fantasy_frac", input_type=Boolean(optional=True), prefix="--fantasy-frac", doc=InputDocumentation(doc="Fraction of fragments to use when running fantasy baseball. Default is 0.1.")), ToolInput(tag="in_write_masked", input_type=Boolean(optional=True), prefix="--write-masked", doc=InputDocumentation(doc="Write alignment masked to the region without gaps in the query.")), ToolInput(tag="in_verbosity", input_type=Boolean(optional=True), prefix="--verbosity", doc=InputDocumentation(doc="Set verbosity level. 0 is silent, and 2 is quite a lot. Default is 1.")), ToolInput(tag="in_out_dir", input_type=Boolean(optional=True), prefix="--out-dir", doc=InputDocumentation(doc="Specify the directory to write place files to.")), ToolInput(tag="in_specify_output_name", input_type=File(optional=True), prefix="-o", doc=InputDocumentation(doc="Specify the output file name")), ToolInput(tag="in_pretend", input_type=Boolean(optional=True), prefix="--pretend", doc=InputDocumentation(doc="Only check out the files then report. Do not run the analysis.")), ToolInput(tag="in_check_like", input_type=Boolean(optional=True), prefix="--check-like", doc=InputDocumentation(doc="Write out the likelihood of the reference tree, calculated two ways.")), ToolInput(tag="in_number_child_processes", input_type=Boolean(optional=True), prefix="-j", doc=InputDocumentation(doc="The number of child processes to spawn when doing placements. Default is 2.")), ToolInput(tag="in_timing", input_type=Boolean(optional=True), prefix="--timing", doc=InputDocumentation(doc="Display timing information after the pplacer run finishes.")), ToolInput(tag="in_no_pre_mask", input_type=Boolean(optional=True), prefix="--no-pre-mask", doc=InputDocumentation(doc="Don't pre-mask sequences before placement.")), ToolInput(tag="in_write_pre_masked", input_type=File(optional=True), prefix="--write-pre-masked", doc=InputDocumentation(doc="Write out the pre-masked sequences to the specified fasta file and exit.")), ToolInput(tag="in_map_mrca", input_type=File(optional=True), prefix="--map-mrca", doc=InputDocumentation(doc="Specify a file to write out MAP sequences for MRCAs and corresponding placements.")), ToolInput(tag="in_map_mrca_min", input_type=Boolean(optional=True), prefix="--map-mrca-min", doc=InputDocumentation(doc="Specify cutoff for inclusion in MAP sequence file. Default is 0.8.")), ToolInput(tag="in_map_identity", input_type=Boolean(optional=True), prefix="--map-identity", doc=InputDocumentation(doc="Add the percent identity of the query sequence to the nearest MAP sequence to each placement.")), ToolInput(tag="in_keep_at_most", input_type=Boolean(optional=True), prefix="--keep-at-most", doc=InputDocumentation(doc="The maximum number of placements we keep. Default is 7.")), ToolInput(tag="in_keep_factor", input_type=Boolean(optional=True), prefix="--keep-factor", doc=InputDocumentation(doc="Throw away anything that has ml_ratio below keep_factor times (best ml_ratio). Default is 0.01.")), ToolInput(tag="in_mrca_class", input_type=Boolean(optional=True), prefix="--mrca-class", doc=InputDocumentation(doc="Classify with MRCAs instead of a painted tree.")), ToolInput(tag="in_groups", input_type=Boolean(optional=True), prefix="--groups", doc=InputDocumentation(doc="Split query alignment into the specified number of groups.")), ToolInput(tag="in_always_refine", input_type=Boolean(optional=True), prefix="--always-refine", doc=InputDocumentation(doc="Always refine the model before placing.")), ToolInput(tag="in_help", input_type=Boolean(optional=True), prefix="--help", doc=InputDocumentation(doc="Display this list of options"))], outputs=[ToolOutput(tag="out_specify_output_name", output_type=File(optional=True), selector=InputSelector(input_to_select="in_specify_output_name", type_hint=File()), doc=OutputDocumentation(doc="Specify the output file name")), ToolOutput(tag="out_write_pre_masked", output_type=File(optional=True), selector=InputSelector(input_to_select="in_write_pre_masked", type_hint=File()), doc=OutputDocumentation(doc="Write out the pre-masked sequences to the specified fasta file and exit.")), ToolOutput(tag="out_map_mrca", output_type=File(optional=True), selector=InputSelector(input_to_select="in_map_mrca", type_hint=File()), doc=OutputDocumentation(doc="Specify a file to write out MAP sequences for MRCAs and corresponding placements."))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Pplacer_V0_1_0().translate("wdl", allow_empty_container=True)

