from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Int, Boolean, Array, String

Psite_V0_1_0 = CommandToolBuilder(tool="psite", base_command=["psite"], inputs=[ToolInput(tag="in_min_counts", input_type=Int(optional=True), prefix="--min_counts", doc=InputDocumentation(doc="Minimum counts required in normalization region to be\nincluded in metagene average (Default: 10)")), ToolInput(tag="in_normalize_over", input_type=Int(optional=True), prefix="--normalize_over", doc=InputDocumentation(doc="N  Portion of each window against which its individual\nraw count profile will be normalized. Specify two\nintegers, in nucleotide distance from landmark\n(negative for upstream, positive for downstream.\nSurround negative numbers with quotes.). (Default: 20\n50)")), ToolInput(tag="in_norm_region", input_type=Int(optional=True), prefix="--norm_region", doc=InputDocumentation(doc="N     Deprecated. Use ``--normalize_over`` instead.\nFormerly, Portion of each window against which its\nindividual raw count profile will be normalized.\nSpecify two integers, in nucleotide distance, from 5'\nend of window. (Default: 70 100)")), ToolInput(tag="in_require_upstream", input_type=Boolean(optional=True), prefix="--require_upstream", doc=InputDocumentation(doc="If supplied, the P-site offset is taken to be the\ndistance between the largest peak upstream of the\nstart codon and the start codon itself. Otherwise, the\nP-site offset is taken to be the distance between the\nlargest peak in the entire ROI and the start codon.\nIgnored if ``--constrain`` is used.")), ToolInput(tag="in_constrain", input_type=Int(optional=True), prefix="--constrain", doc=InputDocumentation(doc="X       Constrain P-site offset to be between specified\ndistance from start codon. Useful for noisy data.\n(Reasonable set: 10 15; default: not constrained)")), ToolInput(tag="in_aggregate", input_type=Boolean(optional=True), prefix="--aggregate", doc=InputDocumentation(doc="Estimate P-site from aggregate reads at each position,\ninstead of median normalized read density. Noisier,\nbut helpful for lower-count data or read lengths with\nfew counts. (Default: False)")), ToolInput(tag="in_keep", input_type=Boolean(optional=True), prefix="--keep", doc=InputDocumentation(doc="Save intermediate count files. Useful for additional\ncomputations (Default: False)")), ToolInput(tag="in_default", input_type=Int(optional=True), prefix="--default", doc=InputDocumentation(doc="Default 5' P-site offset for read lengths that are not\npresent or evaluated in the dataset. Unaffected by\n``--constrain`` (Default: 13)")), ToolInput(tag="in_quiet", input_type=Boolean(optional=True), prefix="--quiet", doc=InputDocumentation(doc="Suppress all warning messages. Cannot use with '-v'.")), ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="--verbose", doc=InputDocumentation(doc="Increase verbosity. With '-v', show every warning.\nWith '-vv', turn warnings into exceptions. Cannot use\nwith '-q'. (Default: show each type of warning once)")), ToolInput(tag="in_count_files", input_type=Array(t=String(), optional=True), prefix="--count_files", doc=InputDocumentation(doc="One or more count or alignment file(s) from a single\nsample or set of samples to be pooled.")), ToolInput(tag="in_count_file_format", input_type=String(optional=True), prefix="--countfile_format", doc=InputDocumentation(doc="Format of file containing alignments or counts\n(Default: BAM)")), ToolInput(tag="in_sum", input_type=String(optional=True), prefix="--sum", doc=InputDocumentation(doc="Sum used in normalization of counts and RPKM/RPNT\ncalculations (Default: total mapped reads/counts in\ndataset)")), ToolInput(tag="in_min_length", input_type=Int(optional=True), prefix="--min_length", doc=InputDocumentation(doc="Minimum read length required to be included (BAM &\nbowtie files only. Default: 25)")), ToolInput(tag="in_max_length", input_type=Int(optional=True), prefix="--max_length", doc=InputDocumentation(doc="Maximum read length permitted to be included (BAM &\nbowtie files only. Default: 100)")), ToolInput(tag="in_fig_format", input_type=String(optional=True), prefix="--figformat", doc=InputDocumentation(doc="File format for figure(s); Default: png)")), ToolInput(tag="in_fig_size", input_type=String(optional=True), prefix="--figsize", doc=InputDocumentation(doc="N         Figure width and height, in inches. (Default: use\nmatplotlibrc params)")), ToolInput(tag="in_title", input_type=String(optional=True), prefix="--title", doc=InputDocumentation(doc="Base title for plot(s).")), ToolInput(tag="in_cmap", input_type=Int(optional=True), prefix="--cmap", doc=InputDocumentation(doc="Matplotlib color map from which palette will be made\n(e.g. 'Blues','autumn','Set1'; default: use colors\nfrom ``--stylesheet`` if given, or color cycle in\nmatplotlibrc)")), ToolInput(tag="in_dpi", input_type=Int(optional=True), prefix="--dpi", doc=InputDocumentation(doc="Figure resolution (Default: 150)")), ToolInput(tag="in_stylesheet", input_type=Boolean(optional=True), prefix="--stylesheet", doc=InputDocumentation(doc="{Solarize_Light2,_classic_test_patch,bmh,classic,dark_background,fast,fivethirtyeight,ggplot,grayscale,seaborn,seaborn-bright,seaborn-colorblind,seaborn-dark,seaborn-dark-palette,seaborn-darkgrid,seaborn-deep,seaborn-muted,seaborn-notebook,seaborn-paper,seaborn-pastel,seaborn-poster,seaborn-talk,seaborn-ticks,seaborn-white,seaborn-whitegrid,tableau-colorblind10}\nUse this matplotlib stylesheet instead of matplotlibrc\nparams\n")), ToolInput(tag="in_out_base_p_offsets_dot_txt", input_type=String(), position=0, doc=InputDocumentation(doc="Tab-delimited text file with two columns. The first is read length,\nand the second the offset from the fiveprime end of that read length\nto the ribosomal P-site. This table can be supplied as the argument\nfor ``--offset`` when using ``--fiveprime_variable`` mapping in any\nof the other scripts in plastid.bin")), ToolInput(tag="in_out_base_meta_gene_profiles_dot_txt", input_type=String(), position=0, doc=InputDocumentation(doc="Metagene profiles, stratified by read length, before P-site offsets\nare applied.")), ToolInput(tag="in_out_base_k_raw_counts_dot_txt", input_type=String(), position=1, doc=InputDocumentation(doc="Saved if ``--keep`` is given on command line. Raw count vectors for each\nmetagene window specified in input ROI file, without P-site\nmapping rules applied, for reads of length `K`")), ToolInput(tag="in_out_base_k_norm_counts_dot_txt", input_type=String(), position=2, doc=InputDocumentation(doc="Saved if ``--keep`` is given on command line. Normalized count vectors\nfor each metagene window specified in input ROI file, without P-site\nmapping rules applied, for reads of length `K`")), ToolInput(tag="in_roi_file", input_type=String(), position=0, doc=InputDocumentation(doc="ROI file surrounding start codons, from ``metagene\ngenerate`` subprogram")), ToolInput(tag="in_out_base", input_type=String(), position=1, doc=InputDocumentation(doc="Basename for output files"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Psite_V0_1_0().translate("wdl", allow_empty_container=True)

