from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Array, String, File, Boolean, Int

Methplotlib_V0_1_0 = CommandToolBuilder(tool="methplotlib", base_command=["methplotlib"], inputs=[ToolInput(tag="in_methylation", input_type=Array(t=String(), optional=True), prefix="--methylation", doc=InputDocumentation(doc="methylation data in nanopolish, nanocompore or ont-\ncram format")), ToolInput(tag="in_names", input_type=Array(t=String(), optional=True), prefix="--names", doc=InputDocumentation(doc="names of datasets in --methylation")), ToolInput(tag="in_window", input_type=String(optional=True), prefix="--window", doc=InputDocumentation(doc="window (region) to which the visualisation has to be\nrestricted")), ToolInput(tag="in_gtf", input_type=File(optional=True), prefix="--gtf", doc=InputDocumentation(doc="add annotation based on a gtf file matching to your\nreference genome")), ToolInput(tag="in_bed", input_type=File(optional=True), prefix="--bed", doc=InputDocumentation(doc="add annotation based on a bed file matching to your\nreference genome")), ToolInput(tag="in_fast_a", input_type=String(optional=True), prefix="--fasta", doc=InputDocumentation(doc="required when --window is an entire chromosome, contig\nor transcript")), ToolInput(tag="in_simplify", input_type=Boolean(optional=True), prefix="--simplify", doc=InputDocumentation(doc="simplify annotation track to show genes rather than")), ToolInput(tag="in_smooth", input_type=String(optional=True), prefix="--smooth", doc=InputDocumentation(doc="When plotting frequencies points are averaged using a\nrolling window")), ToolInput(tag="in_dot_size", input_type=Int(optional=True), prefix="--dotsize", doc=InputDocumentation(doc="Control the size of dots in the per read plots")), ToolInput(tag="in_example", input_type=Boolean(optional=True), prefix="--example", doc=InputDocumentation(doc="Show example command and exit.")), ToolInput(tag="in_outfile", input_type=File(optional=True), prefix="--outfile", doc=InputDocumentation(doc="File to write results to. Default:\nmethylation_browser_{chr}_{start}_{end}.html. Use\n{region} as a shorthand for {chr}_{start}_{end} in the\nfilename. Missing paths will be created.")), ToolInput(tag="in_qc_file", input_type=File(optional=True), prefix="--qcfile", doc=InputDocumentation(doc="File to write the qc report to. Default: The path in\noutfile prefixed with qc_, default is qc_report_methyl\nation_browser_{chr}_{start}_{end}.html. Use {region}\nas a shorthand for {chr}_{start}_{end} in the\nfilename. Missing paths will be created.\n")), ToolInput(tag="in_v", input_type=Boolean(optional=True), prefix="-v", doc=InputDocumentation(doc="")), ToolInput(tag="in_transcripts", input_type=String(), position=0, doc=InputDocumentation(doc="--split               split, rather than overlay the methylation tracks"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Methplotlib_V0_1_0().translate("wdl", allow_empty_container=True)

