class: CommandLineTool
id: tadbit_segment.cwl
inputs:
- id: workdir
  doc: path to working directory (generated with the tool tadbit mapper)
  type: File
  inputBinding:
    prefix: --workdir
- id: tmp_db
  doc: if provided uses this directory to manipulate the database
  type: File
  inputBinding:
    prefix: --tmpdb
- id: no_sql
  doc: do not load/store data from/in sqlite database
  type: boolean
  inputBinding:
    prefix: --nosql
- id: all_bins
  doc: skip the filtering of bins for detection of TADs
  type: boolean
  inputBinding:
    prefix: --all_bins
- id: m_reads
  doc: path valid-pairs file (TADbit BAM format)
  type: File
  inputBinding:
    prefix: --mreads
- id: biases
  doc: path to file with precalculated biases by columns
  type: File
  inputBinding:
    prefix: --biases
- id: resolution
  doc: resolution at which to output matrices
  type: long
  inputBinding:
    prefix: --resolution
- id: norm_matrix
  doc: path to a matrix file with normalized read counts
  type: File
  inputBinding:
    prefix: --norm_matrix
- id: raw_matrix
  doc: path to a matrix file with raw read counts
  type: File
  inputBinding:
    prefix: --raw_matrix
- id: filter
  doc: "[[1, 2, 3, 4, 6, 7, 9, 10]] Use filters to define a set os valid pair of reads\
    \ e.g.: '--apply 1 2 3 4 8 9 10'. Where these numberscorrespond to: 1: self-circle,\
    \ 2: dangling-end, 3: error, 4: extra dangling-end, 5: too close from RES, 6:\
    \ too short, 7: too large, 8: over-represented, 9: duplicated, 10: random breaks,\
    \ 11: trans-chromosomic"
  type: long[]
  inputBinding:
    prefix: --filter
- id: nox
  doc: no display server (X screen)
  type: boolean
  inputBinding:
    prefix: --noX
- id: only_compartments
  doc: search A/B compartments using first eigen vector of the correlation matrix
  type: boolean
  inputBinding:
    prefix: --only_compartments
- id: only_tads
  doc: search TAD boundaries break-point detection algorithm
  type: boolean
  inputBinding:
    prefix: --only_tads
- id: verbose
  doc: print more messages
  type: boolean
  inputBinding:
    prefix: --verbose
- id: jobid
  doc: Use as input data generated by a job with a given jobid. Use tadbit describe
    to find out which.
  type: long
  inputBinding:
    prefix: --jobid
- id: chromosomes
  doc: Name of the chromosomes on which to search for TADs or compartments.
  type: string[]
  inputBinding:
    prefix: --chromosomes
- id: cpu
  doc: '[8] Maximum number of CPU cores available in the execution host. If higher
    than 1, tasks with multi-threading capabilities will enabled (if 0 all available)
    cores will be used'
  type: string
  inputBinding:
    prefix: --cpu
- id: force
  doc: overwrite previously run job
  type: boolean
  inputBinding:
    prefix: --force
- id: rich_in_a
  doc: path to a BED or bedGraph file with list of protein coding gene or other active
    epigenetic mark, to be used to label compartments instead of using the relative
    interaction count.
  type: File
  inputBinding:
    prefix: --rich_in_A
- id: fast_a
  doc: Path to fasta file with genome sequence, to compute GC content and use it to
    label compartments
  type: File
  inputBinding:
    prefix: --fasta
- id: save_corr
  doc: Save correlation matrix used to predict compartments.
  type: boolean
  inputBinding:
    prefix: --savecorr
- id: fix_corr_scale
  doc: Correlation matrix plot scaled between correlation 1 and -1 instead of maximum
    observed values.
  type: boolean
  inputBinding:
    prefix: --fix_corr_scale
- id: format
  doc: '[png] file format for figures'
  type: string
  inputBinding:
    prefix: --format
- id: n_evs
  doc: '[3] Number of eigenvectors to store. if "-1" all eigenvectors will be calculated'
  type: long
  inputBinding:
    prefix: --n_evs
- id: ev_index
  doc: "list of indexes of eigenvectors capturing compartments signal (one index per\
    \ chromosome, in the same order as chromosomes in fasta file). Example picking\
    \ the first eigenvector for all chromosomes but for chromosome 3: '--ev_index\
    \ 1 1 2 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1"
  type: long[]
  inputBinding:
    prefix: --ev_index
- id: max_tad_size
  doc: an integer defining the maximum size of TAD. Default defines it as the number
    of rows/columns
  type: long
  inputBinding:
    prefix: --max_tad_size
outputs: []
cwlVersion: v1.1
baseCommand:
- tadbit
- segment
