from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Array, String, File, Int, Boolean

Tombo_Plot_Per_Read_Roc_V0_1_0 = CommandToolBuilder(tool="tombo_plot_per_read_roc", base_command=["tombo", "plot", "per_read_roc"], inputs=[ToolInput(tag="in_per_read_statistics_filenames", input_type=Array(t=String(), optional=True), prefix="--per-read-statistics-filenames", doc=InputDocumentation(doc="Binary files containing per-read statistics from\nstatistical testing.")), ToolInput(tag="in_modified_locations", input_type=Array(t=String(), optional=True), prefix="--modified-locations", doc=InputDocumentation(doc="Modification description and bed format files\ncontaining single base locations of ground truth\nmodified sites. Bed files should contain 6 fields\nincluding strand. Format descriptions as\n'mod_name:locs.bed'. Example: 'CpG\nbisulfite':bisulfite_locs.bed")), ToolInput(tag="in_unmodified_locations", input_type=Array(t=String(), optional=True), prefix="--unmodified-locations", doc=InputDocumentation(doc="Bed format files containing single base locations of\nground truth unmodified sites. Bed files should\ncontain 6 fields including strand.")), ToolInput(tag="in_motif_descriptions", input_type=Array(t=String(), optional=True), prefix="--motif-descriptions", doc=InputDocumentation(doc="Ground truth, motif centered, modified base\ndescriptions for computing ROC and PR curves. Each\nstatistics file is associated with a set of motif\ndescriptions. Format descriptions as:\n'motif:mod_pos:name[::motif2:mod_pos2:name2...]'.\nmod_pos indicates the alternate-base within the motif\n(1-based index). Example: CCWGG:2:'dcm\n5mC'::GATC:2:'dam 6mA' would assess the performance of\na single Tombo statistics file for identification of\nE. coli dam and dcm methylation.")), ToolInput(tag="in_genome_fast_a", input_type=File(optional=True), prefix="--genome-fasta", doc=InputDocumentation(doc="FASTA file used to re-squiggle. For faster sequence\naccess.")), ToolInput(tag="in_statistics_per_block", input_type=Int(optional=True), prefix="--statistics-per-block", doc=InputDocumentation(doc="Number of randomly selected per-read, per-base\nstatistics to extract from each genomic block for\nplotting. Default: 100000")), ToolInput(tag="in_total_statistics_limit", input_type=Int(optional=True), prefix="--total-statistics-limit", doc=InputDocumentation(doc="Total per-read statistics to be extracted for\nplotting. Avoids memory overflow for large runs.\nDefault: 5000000")), ToolInput(tag="in_pdf_filename", input_type=File(optional=True), prefix="--pdf-filename", doc=InputDocumentation(doc="PDF filename to store plot(s). Default:\ntombo_results.per_reads_roc.pdf")), ToolInput(tag="in_quiet", input_type=Boolean(optional=True), prefix="--quiet", doc=InputDocumentation(doc="Don't print status information."))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Tombo_Plot_Per_Read_Roc_V0_1_0().translate("wdl", allow_empty_container=True)

