from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, String, Boolean, Int, Float

Ngmlr_V0_1_0 = CommandToolBuilder(tool="ngmlr", base_command=["ngmlr"], inputs=[ToolInput(tag="in_reference", input_type=File(optional=True), prefix="--reference", doc=InputDocumentation(doc="(required)  Path to the reference genome (FASTA/Q, can be gzipped)")), ToolInput(tag="in_query", input_type=File(optional=True), prefix="--query", doc=InputDocumentation(doc="Path to the read file (FASTA/Q) [/dev/stdin]")), ToolInput(tag="in_output", input_type=String(optional=True), prefix="--output", doc=InputDocumentation(doc="Adds RG:Z:<string> to all alignments in SAM/BAM [none]")), ToolInput(tag="in_skip_write", input_type=Boolean(optional=True), prefix="--skip-write", doc=InputDocumentation(doc="Don't write reference index to disk [false]")), ToolInput(tag="in_bam_fix", input_type=Boolean(optional=True), prefix="--bam-fix", doc=InputDocumentation(doc="Report reads with > 64k CIGAR operations as unmapped. Required to be compatibel to BAM format [false]")), ToolInput(tag="in_rg_id", input_type=String(optional=True), prefix="--rg-id", doc=InputDocumentation(doc="Adds RG:Z:<string> to all alignments in SAM/BAM [none]")), ToolInput(tag="in_rg_sm", input_type=String(optional=True), prefix="--rg-sm", doc=InputDocumentation(doc="RG header: Sample [none]")), ToolInput(tag="in_rg_lb", input_type=String(optional=True), prefix="--rg-lb", doc=InputDocumentation(doc="RG header: Library [none]")), ToolInput(tag="in_rg_pl", input_type=String(optional=True), prefix="--rg-pl", doc=InputDocumentation(doc="RG header: Platform [none]")), ToolInput(tag="in_rg_ds", input_type=String(optional=True), prefix="--rg-ds", doc=InputDocumentation(doc="RG header: Description [none]")), ToolInput(tag="in_rg_dt", input_type=String(optional=True), prefix="--rg-dt", doc=InputDocumentation(doc="RG header: Date (format: YYYY-MM-DD) [none]")), ToolInput(tag="in_rg_pu", input_type=String(optional=True), prefix="--rg-pu", doc=InputDocumentation(doc="RG header: Platform unit [none]")), ToolInput(tag="in_rg_pi", input_type=Int(optional=True), prefix="--rg-pi", doc=InputDocumentation(doc="RG header: Median insert size [none]")), ToolInput(tag="in_rg_pg", input_type=String(optional=True), prefix="--rg-pg", doc=InputDocumentation(doc="RG header: Programs [none]")), ToolInput(tag="in_rg_cn", input_type=String(optional=True), prefix="--rg-cn", doc=InputDocumentation(doc="RG header: sequencing center [none]")), ToolInput(tag="in_rg_fo", input_type=String(optional=True), prefix="--rg-fo", doc=InputDocumentation(doc="RG header: Flow order [none]")), ToolInput(tag="in_rg_ks", input_type=String(optional=True), prefix="--rg-ks", doc=InputDocumentation(doc="RG header: Key sequence [none]")), ToolInput(tag="in_threads", input_type=Int(optional=True), prefix="--threads", doc=InputDocumentation(doc="Number of threads [1]")), ToolInput(tag="in_pacbio_ont_presets", input_type=Boolean(optional=True), prefix="-x", doc=InputDocumentation(doc="<pacbio, ont>,  --presets <pacbio, ont>\nParameter presets for different sequencing technologies [pacbio]")), ToolInput(tag="in_min_identity", input_type=Int(optional=True), prefix="--min-identity", doc=InputDocumentation(doc="Alignments with an identity lower than this threshold will be discarded [0.65]")), ToolInput(tag="in_min_residues", input_type=Float(optional=True), prefix="--min-residues", doc=InputDocumentation(doc="Alignments containing less than <int> or (<float> * read length) residues will be discarded [0.25]")), ToolInput(tag="in_no_small_inv", input_type=Boolean(optional=True), prefix="--no-smallinv", doc=InputDocumentation(doc="Don't detect small inversions [false]")), ToolInput(tag="in_no_low_quality_split", input_type=Boolean(optional=True), prefix="--no-lowqualitysplit", doc=InputDocumentation(doc="Split alignments with poor quality [false]")), ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="--verbose", doc=InputDocumentation(doc="Debug output [false]")), ToolInput(tag="in_no_progress", input_type=Boolean(optional=True), prefix="--no-progress", doc=InputDocumentation(doc="Don't print progress info while mapping [false]")), ToolInput(tag="in_match", input_type=Float(optional=True), prefix="--match", doc=InputDocumentation(doc="Match score [2]")), ToolInput(tag="in_mismatch", input_type=Float(optional=True), prefix="--mismatch", doc=InputDocumentation(doc="Mismatch score [-5]")), ToolInput(tag="in_gap_open", input_type=Float(optional=True), prefix="--gap-open", doc=InputDocumentation(doc="Gap open score [-5]")), ToolInput(tag="in_gap_extend_max", input_type=Float(optional=True), prefix="--gap-extend-max", doc=InputDocumentation(doc="Gap open extend max [-5]")), ToolInput(tag="in_gap_extend_min", input_type=Float(optional=True), prefix="--gap-extend-min", doc=InputDocumentation(doc="Gap open extend min [-1]")), ToolInput(tag="in_gap_decay", input_type=Float(optional=True), prefix="--gap-decay", doc=InputDocumentation(doc="Gap extend decay [0.15]")), ToolInput(tag="in_km_er_length", input_type=Int(optional=True), prefix="--kmer-length", doc=InputDocumentation(doc="K-mer length in bases [13]")), ToolInput(tag="in_km_er_skip", input_type=Int(optional=True), prefix="--kmer-skip", doc=InputDocumentation(doc="Number of k-mers to skip when building the lookup table from the reference [2]")), ToolInput(tag="in_bin_size", input_type=Int(optional=True), prefix="--bin-size", doc=InputDocumentation(doc="Sets the size of the grid used during candidate search [4]")), ToolInput(tag="in_max_segments", input_type=Int(optional=True), prefix="--max-segments", doc=InputDocumentation(doc="Max number of segments allowed for a read per kb [1]")), ToolInput(tag="in_sub_read_length", input_type=Int(optional=True), prefix="--subread-length", doc=InputDocumentation(doc="Length of fragments reads are split into [256]")), ToolInput(tag="in_sub_read_corridor", input_type=Int(optional=True), prefix="--subread-corridor", doc=InputDocumentation(doc="Length of corridor sub-reads are aligned with [40]\n"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Ngmlr_V0_1_0().translate("wdl", allow_empty_container=True)

