from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, Directory, File, Int

Gtftk_Profile_V0_1_0 = CommandToolBuilder(tool="gtftk_profile", base_command=["gtftk", "profile"], inputs=[ToolInput(tag="in_input_file", input_type=Boolean(optional=True), prefix="--inputfile", doc=InputDocumentation(doc="A zip file containing a matrix as produced by mk_matrix. (default: None)")), ToolInput(tag="in_out_dir", input_type=Directory(optional=True), prefix="--out-dir", doc=InputDocumentation(doc="Output directory name. (default: draw_profile)")), ToolInput(tag="in_transcript_file", input_type=Boolean(optional=True), prefix="--transcript-file", doc=InputDocumentation(doc="A two columns file with the transcripts of interest and their classes. (default: None)")), ToolInput(tag="in_stat", input_type=Boolean(optional=True), prefix="--stat", doc=InputDocumentation(doc="The statistics to be computed. (default: mean)")), ToolInput(tag="in_confidence_interval", input_type=Boolean(optional=True), prefix="--confidence-interval", doc=InputDocumentation(doc="Add a confidence interval to estimate standard error of the mean. (default: False)")), ToolInput(tag="in_profile_colors", input_type=Boolean(optional=True), prefix="--profile-colors", doc=InputDocumentation(doc="Colors. (default: None)")), ToolInput(tag="in_color_order", input_type=Boolean(optional=True), prefix="--color-order", doc=InputDocumentation(doc="Factor ordering. comma-separated bwig labels or tx classes. (default: None)")), ToolInput(tag="in_group_by", input_type=Boolean(optional=True), prefix="--group-by", doc=InputDocumentation(doc="The variable used for grouping. (default: None)")), ToolInput(tag="in_facet_var", input_type=Boolean(optional=True), prefix="--facet-var", doc=InputDocumentation(doc="The variable to be used for splitting into facets. (default: None)")), ToolInput(tag="in_page_width", input_type=File(optional=True), prefix="--page-width", doc=InputDocumentation(doc="Output pdf file width (e.g. 7 inches). (default: None)")), ToolInput(tag="in_page_height", input_type=File(optional=True), prefix="--page-height", doc=InputDocumentation(doc="Output file height (e.g. 5 inches). (default: None)")), ToolInput(tag="in_page_format", input_type=File(optional=True), prefix="--page-format", doc=InputDocumentation(doc="Output file format. (default: pdf)")), ToolInput(tag="in_line_width", input_type=Boolean(optional=True), prefix="--line-width", doc=InputDocumentation(doc="Line width. (default: 1.25)")), ToolInput(tag="in_border_color", input_type=Boolean(optional=True), prefix="--border-color", doc=InputDocumentation(doc="Border color for the plot. (default: #777777)")), ToolInput(tag="in_x_lab", input_type=Boolean(optional=True), prefix="--x-lab", doc=InputDocumentation(doc="X axis label. (default: Selected genomic regions)")), ToolInput(tag="in_axis_text", input_type=Boolean(optional=True), prefix="--axis-text", doc=InputDocumentation(doc="Size of axis text. (default: 8)")), ToolInput(tag="in_strip_text", input_type=Boolean(optional=True), prefix="--strip-text", doc=InputDocumentation(doc="Size of strip text. (default: 8)")), ToolInput(tag="in_subset_b_wig", input_type=Boolean(optional=True), prefix="--subset-bwig", doc=InputDocumentation(doc="Use only a subset of the bigwigs for plotting (default: None)")), ToolInput(tag="in_facet_col", input_type=Boolean(optional=True), prefix="--facet-col", doc=InputDocumentation(doc="Number of facet columns. (default: 4)")), ToolInput(tag="in_show_group_number", input_type=Boolean(optional=True), prefix="--show-group-number", doc=InputDocumentation(doc="Show the number of element per group (groupe size). (default: False)")), ToolInput(tag="in_user_img_file", input_type=Boolean(optional=True), prefix="--user-img-file", doc=InputDocumentation(doc="Provide an alternative path for the image. (default: None)")), ToolInput(tag="in_upper_limit", input_type=Boolean(optional=True), prefix="--upper-limit", doc=InputDocumentation(doc="Upper limit based on quantile computed from unique values. (default: 0.95)")), ToolInput(tag="in_normalization_method", input_type=Boolean(optional=True), prefix="--normalization-method", doc=InputDocumentation(doc="The normalization method performed on a per bigwig basis. (default: none)")), ToolInput(tag="in_to_log", input_type=Boolean(optional=True), prefix="--to-log", doc=InputDocumentation(doc="Control whether the data should be log2-transform before plotting. (default: False)")), ToolInput(tag="in_title", input_type=Boolean(optional=True), prefix="--title", doc=InputDocumentation(doc="A title for the diagram. (default: )")), ToolInput(tag="in_dpi", input_type=Boolean(optional=True), prefix="--dpi", doc=InputDocumentation(doc="Dpi to use. (default: 300)")), ToolInput(tag="in_theme_plot_nine", input_type=Boolean(optional=True), prefix="--theme-plotnine", doc=InputDocumentation(doc="The theme for plotnine diagram. (default: bw)")), ToolInput(tag="in_palette", input_type=Boolean(optional=True), prefix="--palette", doc=InputDocumentation(doc="A color palette (see: https://tinyurl.com/ydacyfxx). (default: nipy_spectral)")), ToolInput(tag="in_list_b_wig", input_type=Boolean(optional=True), prefix="--list-bwig", doc=InputDocumentation(doc="List the bigwig files in the matrix file.. (default: False)")), ToolInput(tag="in_verbosity", input_type=Boolean(optional=True), prefix="--verbosity", doc=InputDocumentation(doc="Set output verbosity ([0-3]). (default: 0)")), ToolInput(tag="in_no_date", input_type=File(optional=True), prefix="--no-date", doc=InputDocumentation(doc="Do not add date to output file names. (default: False)")), ToolInput(tag="in_add_chr", input_type=Boolean(optional=True), prefix="--add-chr", doc=InputDocumentation(doc="Add 'chr' to chromosome names before printing output. (default: False)")), ToolInput(tag="in_tmp_dir", input_type=Boolean(optional=True), prefix="--tmp-dir", doc=InputDocumentation(doc="Keep all temporary files into this folder. (default: None)")), ToolInput(tag="in_keep_all", input_type=Boolean(optional=True), prefix="--keep-all", doc=InputDocumentation(doc="Try to keep all temporary files even if process does not terminate normally. (default: False)")), ToolInput(tag="in_logger_file", input_type=Boolean(optional=True), prefix="--logger-file", doc=InputDocumentation(doc="Stores the arguments passed to the command into a file. (default: None)")), ToolInput(tag="in_write_message_to_file", input_type=Boolean(optional=True), prefix="--write-message-to-file", doc=InputDocumentation(doc="Store all message into a file. (default: None)")), ToolInput(tag="in_arguments", input_type=Int(), position=0, doc=InputDocumentation(doc="Arguments:"))], outputs=[ToolOutput(tag="out_out_dir", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_out_dir", type_hint=File()), doc=OutputDocumentation(doc="Output directory name. (default: draw_profile)")), ToolOutput(tag="out_page_width", output_type=File(optional=True), selector=InputSelector(input_to_select="in_page_width", type_hint=File()), doc=OutputDocumentation(doc="Output pdf file width (e.g. 7 inches). (default: None)")), ToolOutput(tag="out_page_height", output_type=File(optional=True), selector=InputSelector(input_to_select="in_page_height", type_hint=File()), doc=OutputDocumentation(doc="Output file height (e.g. 5 inches). (default: None)")), ToolOutput(tag="out_page_format", output_type=File(optional=True), selector=InputSelector(input_to_select="in_page_format", type_hint=File()), doc=OutputDocumentation(doc="Output file format. (default: pdf)")), ToolOutput(tag="out_no_date", output_type=File(optional=True), selector=InputSelector(input_to_select="in_no_date", type_hint=File()), doc=OutputDocumentation(doc="Do not add date to output file names. (default: False)"))], container="quay.io/biocontainers/pygtftk:1.2.7--py36h4edd4cd_1", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Gtftk_Profile_V0_1_0().translate("wdl")

