class: CommandLineTool
id: EnrichmentAnalysis.cwl
inputs:
- id: in_ctrl
  doc: "Control density file generated by\nRiboDensityAtEachPosition.py.[required]."
  type: File?
  inputBinding:
    prefix: --ctrl
- id: in_treat
  doc: "Treat density file generated by\nRiboDensityAtEachPosition.py.[required]."
  type: File?
  inputBinding:
    prefix: --treat
- id: in_coordinate_file
  doc: "The file should contain the coordinate of start and\nstop codon. Generated\
    \ by\nOutputTranscriptInfo.py.[required]"
  type: File?
  inputBinding:
    prefix: --coordinateFile
- id: in_ot_put_prefix
  doc: Prefix of output files.[required]
  type: string?
  inputBinding:
    prefix: --otput_prefix
- id: in_unit_type
  doc: "Unit type we used for metagene analysis. Either 'nt'\nor 'codon'. default=codon"
  type: string?
  inputBinding:
    prefix: --unit_type
- id: in_filter_mode
  doc: "Mode for filtering transcripts. Either 'counts' or\n'RPKM'. default=counts."
  type: string?
  inputBinding:
    prefix: --filter_mode
- id: in_select_trans_list
  doc: "Selected transcript list used for metagene\nanalysis.This files requires the\
    \ first column must be\nthe transcript ID  with a column name."
  type: string?
  inputBinding:
    prefix: --select_trans_list
- id: in_minimum_cds_codon
  doc: "Minimum CDS codon (codon unit). CDS codons smaller\nthan \"minimum_cds_codon\"\
    \ will be skipped. default=150"
  type: long?
  inputBinding:
    prefix: --minimum_cds_codon
- id: in_minimum_cds_counts
  doc: "Minimum CDS counts. CDS counts smaller than\n\"min_cds_counts\" will be skipped.\
    \ default=128"
  type: long?
  inputBinding:
    prefix: --minimum_cds_counts
- id: in_minimum_norm_region_counts
  doc: "Minimum counts in normalization region . Counts in\nnorm region smaller than\
    \ \"minimum_norm_region_counts\"\nwill be skipped. default=64"
  type: long?
  inputBinding:
    prefix: --minimum_norm_region_counts
- id: in_upstream_codon
  doc: "Upstream codon corresponding to start codon (codon\nunit). While corresponding\
    \ to stop codon, it is the\ndownstream codon."
  type: string?
  inputBinding:
    prefix: --upstream_codon
- id: in_downstream_codon
  doc: "Downstream codon corresponding to start codon (codon\nunit). While corresponding\
    \ to stop codon, it is the\nupstream codon."
  type: string?
  inputBinding:
    prefix: --downstream_codon
- id: in_normalize_exclude_codon
  doc: "The region of the normalization for each genes should\nexclude the first \"\
    normalize_exclude_codon\".\ndefault=30"
  type: long?
  inputBinding:
    prefix: --normalize_exclude_codon
- id: in_ci
  doc: "The confidence intervals of mean on each position\nbased on t distribution.\
    \ default=0.95"
  type: double?
  inputBinding:
    prefix: --CI
- id: in_id_type
  doc: "define the id type users input. the default is\ntranscript id, if not, will\
    \ be transformed into\ntranscript id. default=transcript_id\n"
  type: string?
  inputBinding:
    prefix: --id-type
outputs:
- id: out_stdout
  doc: Standard output stream
  type: stdout
hints:
- class: DockerRequirement
  dockerPull: quay.io/biocontainers/ribominer:0.2.3.1--pyh3252c3a_0
cwlVersion: v1.1
baseCommand:
- EnrichmentAnalysis
