from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Int, String, File, Float, Boolean

Run_Panacota_Py_Prepare_V0_1_0 = CommandToolBuilder(tool="run_panacota.py_prepare", base_command=["run_panacota.py", "prepare"], inputs=[ToolInput(tag="in_species_taxid_download", input_type=Int(optional=True), prefix="-T", doc=InputDocumentation(doc="Species taxid to download, corresponding to the\n'species taxid' provided by the NCBI. This will\ndownload all sequences of this species and all its\nsub-species.A comma-separated list of species taxids\ncan also be provided. (Ex: -T 573 for Klebsiella\npneumoniae)")), ToolInput(tag="in_taxid_download_be", input_type=Int(optional=True), prefix="-t", doc=InputDocumentation(doc="Taxid to download. This can be the taxid of a sub-\nspecies, or of a specific strain. A taxid of a\nsubspecies will download all strains in this\nsubspecies EXCEPT the ones which have a specific\ntaxid.A comma-separated list of taxids can also be\nprovided.Ex: '-t 72407' will download all 'general' K.\npneumoniae subsp. pneumoniae strains, and '-t 1123862'\nwill download the strain K. pneumoniae subsp.\npneumoniae Kp13 (not included in -t 72407, as it is a\nstrain of the subspecies with a specific taxid).")), ToolInput(tag="in_species_download_corresponding", input_type=String(optional=True), prefix="-g", doc=InputDocumentation(doc="Species to download, corresponding to the 'organism\nname' provided by the NCBI. Give name between quotes\n(for example 'escherichia coli')")), ToolInput(tag="in_ncbi_section_download", input_type=String(optional=True), prefix="-s", doc=InputDocumentation(doc="NCBI section to download: all genbank, or only refseq\n(default)")), ToolInput(tag="in_assembly_level", input_type=String(optional=True), prefix="--assembly_level", doc=InputDocumentation(doc="Assembly levels of genomes to download (default: all).\nPossible levels are: 'all', 'complete', 'chromosome',\n'scaffold', 'contig'.You can also provide a comma-\nseparated list of assembly levels. For ex:\n'complete,chromosome'")), ToolInput(tag="in_give_path_directory", input_type=File(optional=True), prefix="-o", doc=InputDocumentation(doc="Give the path to the directory where you want to save\nthe downloaded database. In the given directory, it\nwill create a folder 'Database_init' containing all\nfasta files that will be sent to the control\nprocedure, as well as a folder 'refseq' with all\noriginal compressed files downloaded from refseq. By\ndefault, this output dir name is the ncbi_species name\nif given, or ncbi_species_taxid or ncbi_taxid\notherwise.")), ToolInput(tag="in_tmp", input_type=String(optional=True), prefix="--tmp", doc=InputDocumentation(doc="Specify where the temporary files (sequence split by\nrows of 'N', sequence with new contig names etc.) must\nbe saved. By default, it will be saved in your\nout_dir/tmp_files.")), ToolInput(tag="in_cut_n", input_type=Int(optional=True), prefix="--cutn", doc=InputDocumentation(doc="By default, each genome will be cut into new contigs\nwhen at least 5 'N' in a row are found in its\nsequence. If you don't want to cut genomes into new\ncontigs when there are rows of 'N', put 0 to this\noption. If you want to cut from a different number of\n'N' in a row, put this value to this option.")), ToolInput(tag="in_l_nine_zero", input_type=Int(optional=True), prefix="--l90", doc=InputDocumentation(doc="Maximum value of L90 allowed to keep a genome. Default\nis 100.")), ToolInput(tag="in_nb_cont", input_type=Int(optional=True), prefix="--nbcont", doc=InputDocumentation(doc="Maximum number of contigs allowed to keep a genome.\nDefault is 999.")), ToolInput(tag="in_min_dist", input_type=Float(optional=True), prefix="--min_dist", doc=InputDocumentation(doc="By default, genomes whose distance to the reference is\nnot between 1e-4 and 0.06 are discarded. You can\nspecify your own lower limit (instead of 1e-4) with\nthis option.")), ToolInput(tag="in_max_dist", input_type=Int(optional=True), prefix="--max_dist", doc=InputDocumentation(doc="By default, genomes whose distance to the reference is\nnot between 1e-4 and 0.06 are discarded. You can\nspecify your own lower limit (instead of 0.06) with\nthis option.")), ToolInput(tag="in_threads", input_type=Int(optional=True), prefix="--threads", doc=InputDocumentation(doc="Run 'N' downloads in parallel (default=1). Put 0 if\nyou want to use all cores of your computer.")), ToolInput(tag="in_no_refseq", input_type=Boolean(optional=True), prefix="--norefseq", doc=InputDocumentation(doc="If you already downloaded refseq genomes and do not\nwant to check them, add this option to directly go to\nthe next steps:quality control (L90, number of\ncontigs...) and mash filter. Don't forget to specify\nthe db_dir (-d option) where you already have your\ngenome sequences.")), ToolInput(tag="in_your_already_downloaded", input_type=File(optional=True), prefix="-d", doc=InputDocumentation(doc="If your already downloaded sequences are not in the\ndefault directory (outdir/Database_init), you can\nspecify here the path to those fasta files.")), ToolInput(tag="in_only_mash", input_type=Boolean(optional=True), prefix="--only-mash", doc=InputDocumentation(doc="Add this option if you already downloaded complete and\nrefseq genomes, and ran quality control (you have, a\nfile containing all genome names, as well as their\ngenome size, number of contigs and L90 values). It\nwill then get information on genomes quality from this\nfile, and run mash steps.")), ToolInput(tag="in_info", input_type=File(optional=True), prefix="--info", doc=InputDocumentation(doc="If you already ran the quality control, specify from\nwhich file PanACoTA can read this information, in\norder to proceed to the mash step. This file must\ncontain at least 4 columns, tab separated, with the\nfollowing headers: 'to_annotate', 'gsize', 'nb_conts',\n'L90'. Any other column will be ignored.")), ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="--verbose", doc=InputDocumentation(doc="Increase verbosity in stdout/stderr and log files.")), ToolInput(tag="in_quiet", input_type=Boolean(optional=True), prefix="--quiet", doc=InputDocumentation(doc="Do not display anything to stdout/stderr. log files\nwill still be created."))], outputs=[ToolOutput(tag="out_give_path_directory", output_type=File(optional=True), selector=InputSelector(input_to_select="in_give_path_directory", type_hint=File()), doc=OutputDocumentation(doc="Give the path to the directory where you want to save\nthe downloaded database. In the given directory, it\nwill create a folder 'Database_init' containing all\nfasta files that will be sent to the control\nprocedure, as well as a folder 'refseq' with all\noriginal compressed files downloaded from refseq. By\ndefault, this output dir name is the ncbi_species name\nif given, or ncbi_species_taxid or ncbi_taxid\notherwise."))], container="quay.io/biocontainers/panacota:1.2.0--py_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Run_Panacota_Py_Prepare_V0_1_0().translate("wdl")

