from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Int, Directory, Boolean

Simulate_Non_Silent_Ratio_V0_1_0 = CommandToolBuilder(tool="simulate_non_silent_ratio", base_command=["simulate_non_silent_ratio"], inputs=[ToolInput(tag="in_log_level", input_type=File(optional=True), prefix="--log-level", doc=InputDocumentation(doc="Write a log file (--log-level=DEBUG for debug mode,\n--log-level=INFO for info mode)")), ToolInput(tag="in_log", input_type=File(optional=True), prefix="--log", doc=InputDocumentation(doc="Path to log file. (accepts 'stdout')")), ToolInput(tag="in_input", input_type=File(optional=True), prefix="--input", doc=InputDocumentation(doc="gene FASTA file from extract_gene_seq.py script")), ToolInput(tag="in_mutations", input_type=File(optional=True), prefix="--mutations", doc=InputDocumentation(doc="DNA mutations file")), ToolInput(tag="in_bed", input_type=File(optional=True), prefix="--bed", doc=InputDocumentation(doc="BED file annotation of genes")), ToolInput(tag="in_processes", input_type=Int(optional=True), prefix="--processes", doc=InputDocumentation(doc="Number of processes to use. 0 indicates using a single\nprocess without using a multiprocessing pool (more\nmeans Faster, default: 0).")), ToolInput(tag="in_num_permutations", input_type=Int(optional=True), prefix="--num-permutations", doc=InputDocumentation(doc="Number of permutations for null model. p-value\nprecision increases with more permutations (Default:\n10000).")), ToolInput(tag="in_context", input_type=Int(optional=True), prefix="--context", doc=InputDocumentation(doc="Number of DNA bases to use as context. 0 indicates no\ncontext. 1 indicates only use the mutated base. 1.5\nindicates using the base context used in CHASM (http:/\n/wiki.chasmsoftware.org/index.php/CHASM_Overview). 2\nindicates using the mutated base and the upstream\nbase. 3 indicates using the mutated base and both the\nupstream and downstream bases. (Default: 1.5)")), ToolInput(tag="in_score_dir", input_type=Directory(optional=True), prefix="--score-dir", doc=InputDocumentation(doc="Directory containing score information in pickle files\n(Default: None).")), ToolInput(tag="in_by_sample", input_type=Boolean(optional=True), prefix="--by-sample", doc=InputDocumentation(doc="Report counts for observed mutations stratified by the\ntumor sample")), ToolInput(tag="in_use_unmapped", input_type=Boolean(optional=True), prefix="--use-unmapped", doc=InputDocumentation(doc="Use mutations that are not mapped to the the single\nreference transcript for a gene specified in the bed\nfile indicated by the -b option.")), ToolInput(tag="in_genome", input_type=File(optional=True), prefix="--genome", doc=InputDocumentation(doc="Path to the genome fasta file. Required if --use-\nunmapped flag is used. (Default: None)")), ToolInput(tag="in_observed_output", input_type=File(optional=True), prefix="--observed-output", doc=InputDocumentation(doc="Output text file of observed results (optional).")), ToolInput(tag="in_output", input_type=File(optional=True), prefix="--output", doc=InputDocumentation(doc="Output text file of simulation results\n"))], outputs=[ToolOutput(tag="out_observed_output", output_type=File(optional=True), selector=InputSelector(input_to_select="in_observed_output", type_hint=File()), doc=OutputDocumentation(doc="Output text file of observed results (optional).")), ToolOutput(tag="out_output", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output", type_hint=File()), doc=OutputDocumentation(doc="Output text file of simulation results\n"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Simulate_Non_Silent_Ratio_V0_1_0().translate("wdl", allow_empty_container=True)

