from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Int, String

Corrector_Ha_V0_1_0 = CommandToolBuilder(tool="Corrector_HA", base_command=["Corrector_HA"], inputs=[ToolInput(tag="in_set_kmer_size", input_type=Int(optional=True), prefix="-k", doc=InputDocumentation(doc="Set kmer size, default=17.\nNote: this value must be set to be same as that used in Kmer_freq program.")), ToolInput(tag="in_set_frequency_cutoff", input_type=Int(optional=True), prefix="-l", doc=InputDocumentation(doc="Set the low frequency cutoff, default=3.\nNote: the Kmers with frequency lower than(<=) this cutoff will be taken as\nsuspicious Kmers, while the others as credible Kmers.")), ToolInput(tag="in_set_minimum_length_continuous", input_type=Int(optional=True), prefix="-m", doc=InputDocumentation(doc="Set the minimum length of a continuous high-freq-kmer region, default=10.\nNote: use the Branch and bounding tree method, needs a continuous high\nfrequency region as the starting point. Here we set the minimum Kmer\nnumber in the continuous high-freq-kmer region.")), ToolInput(tag="in_set_maximum_change", input_type=Int(optional=True), prefix="-c", doc=InputDocumentation(doc="Set the maximum change allowed in one read, default=2.\nNote: set the allowed base number to be changed, when achieve this cutoff,\nthe program will turn to trim data instead of correction.")), ToolInput(tag="in_set_maximum_number", input_type=Int(optional=True), prefix="-n", doc=InputDocumentation(doc="Set the maximum node number allowed in the branch and bound tree, default=15000000.\nNote: this setting affects the maximum memory that will be used.")), ToolInput(tag="in_set_remain_data", input_type=Int(optional=True), prefix="-a", doc=InputDocumentation(doc="Set whether remain all the data(1) or not(0), default=0.\nNote: 1:remain all the data, 0:trim some low quality and suspicious data\nto get more accurate data; if set '1', no need to set -x and -r")), ToolInput(tag="in_set_ascii_shift", input_type=Int(optional=True), prefix="-Q", doc=InputDocumentation(doc="Set the ASCII shift of the quality value(Quality_shift), default=64.\nNote: usually 64 or 33 for Illumina data.")), ToolInput(tag="in_set_trim_region", input_type=Int(optional=True), prefix="-e", doc=InputDocumentation(doc="Set whether trim(1) suspicious region at the end of reads according to Q_value<=2 or not(0), default=0.\nNote: 0:no, trim low quality and suspicious data directly; 1:yes, just trim\nQ_value<=2 continues region.")), ToolInput(tag="in_set_trim_bases", input_type=Int(optional=True), prefix="-w", doc=InputDocumentation(doc="Set whether trim(1) error bases with Q_value<=2 instead of correct(0) it , default=0.\nNote: when base in Q_value<=2 continues region is detected error base,\n0:correct this base; 1:trim this end of reads.")), ToolInput(tag="in_set_quality_threshold", input_type=Int(optional=True), prefix="-q", doc=InputDocumentation(doc="Set Quality threshold of error bases, default=40.\nNote: bases with quality value higher than this threshold will not be\ncorrected when using BBT method.")), ToolInput(tag="in_set_trimmed_length", input_type=Int(optional=True), prefix="-x", doc=InputDocumentation(doc="Set the trimmed length at low-quality ends instead of correct them, default=KmerSize/2.\nNote: on the trimmed ends or real ends of reads, sequencing errors tend to be\nconcentrate, and is hard to correct all of them. In order to get higher accuracy\nin the final result, we further trimmed some length at these low-quality regions.")), ToolInput(tag="in_set_minimum_length_defaultnote", input_type=Int(optional=True), prefix="-r", doc=InputDocumentation(doc="Set the minimum length of trimmed read, default=50.\nNote:  This value set the minimum length of final reads that will be output.")), ToolInput(tag="in_set_thread_number", input_type=Int(optional=True), prefix="-t", doc=InputDocumentation(doc="Set the thread number, default=1.\nNote: the more thread number, the high speed, however, this should be less than\nthe number of CPU cores.")), ToolInput(tag="in_set_convert_read", input_type=Int(optional=True), prefix="-j", doc=InputDocumentation(doc="Set whether convert read1 and read2 corrected file into Pair-end file: 1, yes, 0, no; default=1.\nNote: program may discard some low quality reads in corrected file, it should be\nconvert into Pair-end file for using in some other porgram(for example: SOAPdenovo),\npair and single file as the result file when it set to '-j 1'. If you have only\nsingle-end reads, please do not use the function, that is, set option '-j 0'.\nThe default value is to deal with paired-end reads.")), ToolInput(tag="in_set_output_file", input_type=Int(optional=True), prefix="-o", doc=InputDocumentation(doc="Set output file format, default=1.\nNote: 0, fasta compressed file(*.fa.gz); 1, fastq compressed file(*.fq.gz);\n2, fasta text file(*.fa); 3, fastq text file(*.fq). If input read files are in fasta\nformat and ouput in fastq format, program will output all quality char=40+Quality_shift.")), ToolInput(tag="in_prefix_dot_freq_do_tgz", input_type=String(), position=0, doc=InputDocumentation(doc="")), ToolInput(tag="in_reads_file_list", input_type=String(), position=1, doc=InputDocumentation(doc=""))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Corrector_Ha_V0_1_0().translate("wdl", allow_empty_container=True)

