from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, String, Int, Directory

Hitea_V0_1_0 = CommandToolBuilder(tool="hitea", base_command=["hitea"], inputs=[ToolInput(tag="in_input_file_pairsam", input_type=File(optional=True), prefix="-i", doc=InputDocumentation(doc=":          Input file in pairsam format or unsorted-lossless bam format")), ToolInput(tag="in_restriction_endunuclease_used", input_type=String(optional=True), prefix="-e", doc=InputDocumentation(doc=":          Restriction endunuclease used for the assay (default: '', available:MboI,DpnII,HindIII,Arima,NcoI,NotI)")), ToolInput(tag="in_genome_build_used", input_type=Int(optional=True), prefix="-g", doc=InputDocumentation(doc=":          Genome build to be used (default:hg38, available: hg19)")), ToolInput(tag="in_fasta_format_file_teconsensus", input_type=File(optional=True), prefix="-n", doc=InputDocumentation(doc=":           fasta format file for TE-consensus sequences")), ToolInput(tag="in_fasta_format_file_repbase", input_type=File(optional=True), prefix="-b", doc=InputDocumentation(doc=":         fasta format file for Repbase subfamily sequences")), ToolInput(tag="in_fasta_format_file_polymorphic", input_type=File(optional=True), prefix="-p", doc=InputDocumentation(doc=":          fasta format file for Polymorphic sequences (header should be Family~name format")), ToolInput(tag="in_referencegenome_copies_tefamily", input_type=String(optional=True), prefix="-a", doc=InputDocumentation(doc=":            reference-genome copies for TE-family members")), ToolInput(tag="in_output_prefix_generating", input_type=String(optional=True), prefix="-o", doc=InputDocumentation(doc=":       Output prefix while generating report files (default: project)")), ToolInput(tag="in_working_directory_are", input_type=Directory(optional=True), prefix="-w", doc=InputDocumentation(doc=":          Working directory where the files are to be written")), ToolInput(tag="in_mapping_quality_threshold", input_type=Int(optional=True), prefix="-q", doc=InputDocumentation(doc=":     Mapping quality threshold for repeat anchored mate on the reference genome (default: 28)")), ToolInput(tag="in_minimum_clip_length", input_type=Int(optional=True), prefix="-s", doc=InputDocumentation(doc=":            Minimum clip length for detecting insertion (should be >=13bp) (default: 20)")), ToolInput(tag="in_remap_unmapped_reads", input_type=String(optional=True), prefix="-r", doc=InputDocumentation(doc=":           whether to remap unmapped clipped reads to the polymoprhic sequences (default:F)")), ToolInput(tag="in_file_is_experiment", input_type=File(optional=True), prefix="-x", doc=InputDocumentation(doc=":             whether the file is a WGS experiment (default:F)")), ToolInput(tag="in_display_help_message", input_type=String(optional=True), prefix="-h", doc=InputDocumentation(doc=":            Display help message"))], outputs=[], container="quay.io/biocontainers/hitea:0.1.5--0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Hitea_V0_1_0().translate("wdl")

