from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, String, Int, File

Sga_Graph_Diff_V0_1_0 = CommandToolBuilder(tool="sga_graph_diff", base_command=["sga", "graph-diff"], inputs=[ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="--verbose", doc=InputDocumentation(doc="display verbose output")), ToolInput(tag="in_prefix", input_type=String(optional=True), prefix="--prefix", doc=InputDocumentation(doc="prefix the output files with NAME")), ToolInput(tag="in_threads", input_type=Int(optional=True), prefix="--threads", doc=InputDocumentation(doc="use NUM computation threads")), ToolInput(tag="in_genome_size", input_type=Int(optional=True), prefix="--genome-size", doc=InputDocumentation(doc="(optional) set the size of the genome to be N bases\nthis is used to determine the number of bits to use in the bloom filter\nif unset, it will be calculated from the reference genome FASTA file")), ToolInput(tag="in_pre_cache_reference", input_type=String(optional=True), prefix="--precache-reference", doc=InputDocumentation(doc="precache the named chromosome of the reference genome\nIf STR is 'all' the entire reference will be cached")), ToolInput(tag="in_variant", input_type=File(optional=True), prefix="--variant", doc=InputDocumentation(doc="call variants present in the read set in FILE")), ToolInput(tag="in_base", input_type=File(optional=True), prefix="--base", doc=InputDocumentation(doc="use the read set in FILE as the base line for comparison\nif this option is not given, reference-based calls will be made")), ToolInput(tag="in_reference", input_type=File(optional=True), prefix="--reference", doc=InputDocumentation(doc="use the reference sequence in FILE")), ToolInput(tag="in_km_er", input_type=String(optional=True), prefix="--kmer", doc=InputDocumentation(doc="use K-mers to discover variants")), ToolInput(tag="in_min_discovery_count", input_type=Int(optional=True), prefix="--min-discovery-count", doc=InputDocumentation(doc="require a variant k-mer to be seen at least T times")), ToolInput(tag="in_algorithm", input_type=String(optional=True), prefix="--algorithm", doc=InputDocumentation(doc="select the assembly algorithm to use from: debruijn, string")), ToolInput(tag="in_min_overlap", input_type=Int(optional=True), prefix="--min-overlap", doc=InputDocumentation(doc="require at least N bp overlap when assembling using a string graph")), ToolInput(tag="in_min_dbg_count", input_type=Int(optional=True), prefix="--min-dbg-count", doc=InputDocumentation(doc="only use k-mers seen T times when assembling using a de Bruijn graph")), ToolInput(tag="in_ref", input_type=String(optional=True), prefix="--ref", doc=InputDocumentation(doc=""))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Sga_Graph_Diff_V0_1_0().translate("wdl", allow_empty_container=True)

