from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Boolean, Int, Float, String, Array

Samplot_Vcf_V0_1_0 = CommandToolBuilder(tool="samplot_vcf", base_command=["samplot", "vcf"], inputs=[ToolInput(tag="in_vcf", input_type=File(optional=True), prefix="--vcf", doc=InputDocumentation(doc="VCF file containing structural variants (default:\nNone)")), ToolInput(tag="in_out_dir", input_type=File(optional=True), prefix="--out-dir", doc=InputDocumentation(doc="path to write output PNGs (default: samplot-out)")), ToolInput(tag="in_ped", input_type=File(optional=True), prefix="--ped", doc=InputDocumentation(doc="path ped (or .fam) file (default: None)")), ToolInput(tag="in_dn_only", input_type=Boolean(optional=True), prefix="--dn_only", doc=InputDocumentation(doc="plots only putative de novo variants (PED file\nrequired) (default: False)")), ToolInput(tag="in_min_call_rate", input_type=Int(optional=True), prefix="--min_call_rate", doc=InputDocumentation(doc="only plot variants with at least this call-rate\n(default: 0.95)")), ToolInput(tag="in_filter", input_type=Float(optional=True), prefix="--filter", doc=InputDocumentation(doc="simple filter that samples must meet. Join multiple\nfilters with '&' and specify --filter multiple times\nfor 'or' e.g. DHFFC < 0.7 & SVTYPE = 'DEL' (default:\n[])")), ToolInput(tag="in_output_type", input_type=String(optional=True), prefix="--output_type", doc=InputDocumentation(doc="type of output figure (default: png)")), ToolInput(tag="in_max_he_ts", input_type=Int(optional=True), prefix="--max_hets", doc=InputDocumentation(doc="only plot variants with at most this many\nheterozygotes (default: None)")), ToolInput(tag="in_min_entries", input_type=Int(optional=True), prefix="--min_entries", doc=InputDocumentation(doc="try to include homref samples as controls to get this\nmany samples in plot (default: 6)")), ToolInput(tag="in_max_entries", input_type=Int(optional=True), prefix="--max_entries", doc=InputDocumentation(doc="only plot at most this many heterozygotes (default:\n10)")), ToolInput(tag="in_max_mb", input_type=Int(optional=True), prefix="--max_mb", doc=InputDocumentation(doc="skip variants longer than this many megabases\n(default: None)")), ToolInput(tag="in_min_bp", input_type=Int(optional=True), prefix="--min_bp", doc=InputDocumentation(doc="skip variants shorter than this many bases (default:\n20)")), ToolInput(tag="in_important_regions", input_type=File(optional=True), prefix="--important_regions", doc=InputDocumentation(doc="only report variants that overlap regions in this bed\nfile (default: None)")), ToolInput(tag="in_bams", input_type=Array(t=String(), optional=True), prefix="--bams", doc=InputDocumentation(doc="Space-delimited list of BAM/CRAM file names (default:\nNone)")), ToolInput(tag="in_sample_ids", input_type=Array(t=String(), optional=True), prefix="--sample_ids", doc=InputDocumentation(doc="Space-delimited list of sample IDs, must have same\norder as BAM/CRAM file names. BAM RG tag required if\nthis is omitted. (default: None)")), ToolInput(tag="in_command_file", input_type=File(optional=True), prefix="--command_file", doc=InputDocumentation(doc="store commands in this file. (default:\nsamplot_vcf_cmds.tmp)")), ToolInput(tag="in_format", input_type=String(optional=True), prefix="--format", doc=InputDocumentation(doc="comma separated list of FORMAT fields to include in\nsample plot title (default: AS,AP,DHFFC)")), ToolInput(tag="in_gff_three", input_type=Int(optional=True), prefix="--gff3", doc=InputDocumentation(doc="genomic regions (.gff with .tbi in same directory)\nused when building HTML table and table filters\n(default: None)")), ToolInput(tag="in_down_sample", input_type=Int(optional=True), prefix="--downsample", doc=InputDocumentation(doc="Number of normal reads/pairs to plot (default: 1)")), ToolInput(tag="in_manual_run", input_type=Boolean(optional=True), prefix="--manual_run", doc=InputDocumentation(doc="don't auto-run the samplot plot commands (command_file\nwill be deleted) (default: False)")), ToolInput(tag="in_debug", input_type=Boolean(optional=True), prefix="--debug", doc=InputDocumentation(doc="prints out the reason each skipped variant entry is\nskipped (default: False)\n"))], outputs=[ToolOutput(tag="out_out_dir", output_type=File(optional=True), selector=InputSelector(input_to_select="in_out_dir", type_hint=File()), doc=OutputDocumentation(doc="path to write output PNGs (default: samplot-out)"))], container="quay.io/biocontainers/samplot:1.1.3--pyh3252c3a_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Samplot_Vcf_V0_1_0().translate("wdl")

