version 1.0

task GetUTRSequences {
  input {
    File? input_files_fasta
    File? coordinate_file
    String? ot_put_prefix
  }
  command <<<
    GetUTRSequences \
      ~{if defined(input_files_fasta) then ("--input " +  '"' + input_files_fasta + '"') else ""} \
      ~{if defined(coordinate_file) then ("--coordinateFile " +  '"' + coordinate_file + '"') else ""} \
      ~{if defined(ot_put_prefix) then ("--otput_prefix " +  '"' + ot_put_prefix + '"') else ""}
  >>>
  runtime {
    docker: "quay.io/biocontainers/ribominer:0.2.3.1--pyh3252c3a_0"
  }
  parameter_meta {
    input_files_fasta: "Input file(s) in fasta format. All transcript\\nsequences are from the longest transcripts of protein\\ncoding genes which may be generated by\\nGetProteinCodingSequence.py"
    coordinate_file: "The file should contain the coordinate of start and\\nstop codon. Generated by RiboCode.[required]"
    ot_put_prefix: "Prefix of output files.[required]\\n"
  }
  output {
    File out_stdout = stdout()
  }
}