version 1.0

task MetageneAnalysis {
  input {
    File? bam_list_file
    File? input_files_bam
    File? coordinate_file
    String? ot_put_prefix
    Int? specific_reads_length
    Int? offset
    File? bam_file_legend
    String? unit_type
    String? filter_mode
    String? select_trans_list
    Int? minimum_cds_codon
    Int? minimum_cds_counts
    Int? minimum_norm_region_counts
    String? upstream_codon
    String? downstream_codon
    Int? normalize_exclude_codon
    Int? normalize_codon_density_num
    String? type
    String? id_type
    Float? ci
    Int? norm
    String? plot
  }
  command <<<
    MetageneAnalysis \
      ~{if defined(bam_list_file) then ("--bamListFile " +  '"' + bam_list_file + '"') else ""} \
      ~{if defined(input_files_bam) then ("--input " +  '"' + input_files_bam + '"') else ""} \
      ~{if defined(coordinate_file) then ("--coordinateFile " +  '"' + coordinate_file + '"') else ""} \
      ~{if defined(ot_put_prefix) then ("--otput_prefix " +  '"' + ot_put_prefix + '"') else ""} \
      ~{if defined(specific_reads_length) then ("--specific_reads_length " +  '"' + specific_reads_length + '"') else ""} \
      ~{if defined(offset) then ("--offset " +  '"' + offset + '"') else ""} \
      ~{if defined(bam_file_legend) then ("--bam_file_legend " +  '"' + bam_file_legend + '"') else ""} \
      ~{if defined(unit_type) then ("--unit_type " +  '"' + unit_type + '"') else ""} \
      ~{if defined(filter_mode) then ("--filter_mode " +  '"' + filter_mode + '"') else ""} \
      ~{if defined(select_trans_list) then ("--select_trans_list " +  '"' + select_trans_list + '"') else ""} \
      ~{if defined(minimum_cds_codon) then ("--minimum_cds_codon " +  '"' + minimum_cds_codon + '"') else ""} \
      ~{if defined(minimum_cds_counts) then ("--minimum_cds_counts " +  '"' + minimum_cds_counts + '"') else ""} \
      ~{if defined(minimum_norm_region_counts) then ("--minimum_norm_region_counts " +  '"' + minimum_norm_region_counts + '"') else ""} \
      ~{if defined(upstream_codon) then ("--upstream_codon " +  '"' + upstream_codon + '"') else ""} \
      ~{if defined(downstream_codon) then ("--downstream_codon " +  '"' + downstream_codon + '"') else ""} \
      ~{if defined(normalize_exclude_codon) then ("--normalize_exclude_codon " +  '"' + normalize_exclude_codon + '"') else ""} \
      ~{if defined(normalize_codon_density_num) then ("--normalize_codon_density_num " +  '"' + normalize_codon_density_num + '"') else ""} \
      ~{if defined(type) then ("--type " +  '"' + type + '"') else ""} \
      ~{if defined(id_type) then ("--id-type " +  '"' + id_type + '"') else ""} \
      ~{if defined(ci) then ("--CI " +  '"' + ci + '"') else ""} \
      ~{if defined(norm) then ("--norm " +  '"' + norm + '"') else ""} \
      ~{if defined(plot) then ("--plot " +  '"' + plot + '"') else ""}
  >>>
  runtime {
    docker: "quay.io/biocontainers/ribominer:0.2.3.1--pyh3252c3a_0"
  }
  parameter_meta {
    bam_list_file: "Bam file list, containing 4 columns.Namely\\nbamFiles,readLength, offSet, bamLegend. '-f' and '-i,\\n-r, -s, -t' parameters are mutually\\nexclusive.default=none."
    input_files_bam: "Input file(s) in bam format. All files should be split\\nby comma e.g. 1.bam,2.bam,3.bam[required]. '-i' and\\n'-f' are mutually exclusive. default=none"
    coordinate_file: "The file should contain the coordinate of start and\\nstop codon. Generated by\\nOutputTranscriptInfo.py.[required]"
    ot_put_prefix: "Prefix of output files.[required]"
    specific_reads_length: "Specific the lenght to do analysis, comma split. e.g.\\n'28,29,30'.If use all length set 'All'. Bam files diff\\nlength select split by '_' e.g. '28,29,30_ALL_27,28'\\n[required]. '-r' and '-f' are mutually exclusive."
    offset: "Specific the offset corresponding to read length,\\ncomma split. e.g. '12,13,13'. No offset set 0. Bam\\nfiles diff offset select split by '_' e.g.\\n'12,13,13_0_12,12' [required]. '-s' and '-f' are\\nmutually exclusive."
    bam_file_legend: "The legend of each bam files, comma split. e.g.\\n'condition1,condition2,condition3' [required]. '-t'\\nand '-f' are mutually exclusive."
    unit_type: "Unit type we used for metagene analysis. Either 'nt'\\nor 'codon'. default=codon"
    filter_mode: "Mode for filtering transcripts. Either 'counts' or\\n'RPKM'. default=counts."
    select_trans_list: "Selected transcript list used for metagene\\nanalysis.This files requires the first column must be\\nthe transcript ID  with a column name."
    minimum_cds_codon: "Minimum CDS codon (codon unit). CDS codons smaller\\nthan \\\"minimum_cds_codon\\\" will be skipped. default=150"
    minimum_cds_counts: "Minimum CDS counts. CDS counts smaller than\\n\\\"min_cds_counts\\\" will be skipped. default=128"
    minimum_norm_region_counts: "Minimum counts in normalization region . Counts in\\nnorm region smaller than \\\"minimum_norm_region_counts\\\"\\nwill be skipped. default=64"
    upstream_codon: "Upstream codon corresponding to start codon (codon\\nunit). While corresponding to stop codon, it is the\\ndownstream codon."
    downstream_codon: "Downstream codon corresponding to start codon (codon\\nunit). While corresponding to stop codon, it is the\\nupstream codon."
    normalize_exclude_codon: "The region of the normalization for each genes should\\nexclude the first \\\"normalize_exclude_codon\\\".\\ndefault=30"
    normalize_codon_density_num: "Output the first \\\"normalize_codon_density_num\\\" from\\nthe start codon for each qualified genes in each bam\\nfiles. Pay attention that this parameter should not\\nbigger than -l parameter. default=100"
    type: "Type of counts statistics.[CDS/cds or\\nUTR/utr].default=CDS"
    id_type: "define the id type users input. the default is\\ntranscript id, if not, will be transformed into\\ntranscript id. default=transcript_id"
    ci: "The confidence intervals of mean on each position\\nbased on t distribution. default=0.95"
    norm: "Use density beyound the first 30 (-e) codons for\\nnormalization or not. [yes/no]"
    plot: "Output a metagene plot or not. default=yes"
  }
  output {
    File out_stdout = stdout()
  }
}