from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, String, Int, Float

Metageneanalysis_V0_1_0 = CommandToolBuilder(tool="MetageneAnalysis", base_command=["MetageneAnalysis"], inputs=[ToolInput(tag="in_bam_list_file", input_type=File(optional=True), prefix="--bamListFile", doc=InputDocumentation(doc="Bam file list, containing 4 columns.Namely\nbamFiles,readLength, offSet, bamLegend. '-f' and '-i,\n-r, -s, -t' parameters are mutually\nexclusive.default=none.")), ToolInput(tag="in_input", input_type=File(optional=True), prefix="--input", doc=InputDocumentation(doc="Input file(s) in bam format. All files should be split\nby comma e.g. 1.bam,2.bam,3.bam[required]. '-i' and\n'-f' are mutually exclusive. default=none")), ToolInput(tag="in_coordinate_file", input_type=File(optional=True), prefix="--coordinateFile", doc=InputDocumentation(doc="The file should contain the coordinate of start and\nstop codon. Generated by\nOutputTranscriptInfo.py.[required]")), ToolInput(tag="in_ot_put_prefix", input_type=String(optional=True), prefix="--otput_prefix", doc=InputDocumentation(doc="Prefix of output files.[required]")), ToolInput(tag="in_specific_reads_length", input_type=Int(optional=True), prefix="--specific_reads_length", doc=InputDocumentation(doc="Specific the lenght to do analysis, comma split. e.g.\n'28,29,30'.If use all length set 'All'. Bam files diff\nlength select split by '_' e.g. '28,29,30_ALL_27,28'\n[required]. '-r' and '-f' are mutually exclusive.")), ToolInput(tag="in_offset", input_type=Int(optional=True), prefix="--offset", doc=InputDocumentation(doc="Specific the offset corresponding to read length,\ncomma split. e.g. '12,13,13'. No offset set 0. Bam\nfiles diff offset select split by '_' e.g.\n'12,13,13_0_12,12' [required]. '-s' and '-f' are\nmutually exclusive.")), ToolInput(tag="in_bam_file_legend", input_type=File(optional=True), prefix="--bam_file_legend", doc=InputDocumentation(doc="The legend of each bam files, comma split. e.g.\n'condition1,condition2,condition3' [required]. '-t'\nand '-f' are mutually exclusive.")), ToolInput(tag="in_unit_type", input_type=String(optional=True), prefix="--unit_type", doc=InputDocumentation(doc="Unit type we used for metagene analysis. Either 'nt'\nor 'codon'. default=codon")), ToolInput(tag="in_filter_mode", input_type=String(optional=True), prefix="--filter_mode", doc=InputDocumentation(doc="Mode for filtering transcripts. Either 'counts' or\n'RPKM'. default=counts.")), ToolInput(tag="in_select_trans_list", input_type=String(optional=True), prefix="--select_trans_list", doc=InputDocumentation(doc="Selected transcript list used for metagene\nanalysis.This files requires the first column must be\nthe transcript ID  with a column name.")), ToolInput(tag="in_minimum_cds_codon", input_type=Int(optional=True), prefix="--minimum_cds_codon", doc=InputDocumentation(doc="Minimum CDS codon (codon unit). CDS codons smaller\nthan 'minimum_cds_codon' will be skipped. default=150")), ToolInput(tag="in_minimum_cds_counts", input_type=Int(optional=True), prefix="--minimum_cds_counts", doc=InputDocumentation(doc="Minimum CDS counts. CDS counts smaller than\n'min_cds_counts' will be skipped. default=128")), ToolInput(tag="in_minimum_norm_region_counts", input_type=Int(optional=True), prefix="--minimum_norm_region_counts", doc=InputDocumentation(doc="Minimum counts in normalization region . Counts in\nnorm region smaller than 'minimum_norm_region_counts'\nwill be skipped. default=64")), ToolInput(tag="in_upstream_codon", input_type=String(optional=True), prefix="--upstream_codon", doc=InputDocumentation(doc="Upstream codon corresponding to start codon (codon\nunit). While corresponding to stop codon, it is the\ndownstream codon.")), ToolInput(tag="in_downstream_codon", input_type=String(optional=True), prefix="--downstream_codon", doc=InputDocumentation(doc="Downstream codon corresponding to start codon (codon\nunit). While corresponding to stop codon, it is the\nupstream codon.")), ToolInput(tag="in_normalize_exclude_codon", input_type=Int(optional=True), prefix="--normalize_exclude_codon", doc=InputDocumentation(doc="The region of the normalization for each genes should\nexclude the first 'normalize_exclude_codon'.\ndefault=30")), ToolInput(tag="in_normalize_codon_density_num", input_type=Int(optional=True), prefix="--normalize_codon_density_num", doc=InputDocumentation(doc="Output the first 'normalize_codon_density_num' from\nthe start codon for each qualified genes in each bam\nfiles. Pay attention that this parameter should not\nbigger than -l parameter. default=100")), ToolInput(tag="in_type", input_type=String(optional=True), prefix="--type", doc=InputDocumentation(doc="Type of counts statistics.[CDS/cds or\nUTR/utr].default=CDS")), ToolInput(tag="in_id_type", input_type=String(optional=True), prefix="--id-type", doc=InputDocumentation(doc="define the id type users input. the default is\ntranscript id, if not, will be transformed into\ntranscript id. default=transcript_id")), ToolInput(tag="in_ci", input_type=Float(optional=True), prefix="--CI", doc=InputDocumentation(doc="The confidence intervals of mean on each position\nbased on t distribution. default=0.95")), ToolInput(tag="in_norm", input_type=Int(optional=True), prefix="--norm", doc=InputDocumentation(doc="Use density beyound the first 30 (-e) codons for\nnormalization or not. [yes/no]")), ToolInput(tag="in_plot", input_type=String(optional=True), prefix="--plot", doc=InputDocumentation(doc="Output a metagene plot or not. default=yes"))], outputs=[], container="quay.io/biocontainers/ribominer:0.2.3.1--pyh3252c3a_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Metageneanalysis_V0_1_0().translate("wdl")

