version 1.0

task ReadsLengthOfSpecificRegions {
  input {
    File? input_file_bam
    String? left_position
    String? right_position
    String? prefix_output_files
    File? coordinate_file
    String? select_trans_list
    Int? type
    String? id_type
  }
  command <<<
    ReadsLengthOfSpecificRegions \
      ~{if defined(input_file_bam) then ("--input " +  '"' + input_file_bam + '"') else ""} \
      ~{if defined(left_position) then ("--left-position " +  '"' + left_position + '"') else ""} \
      ~{if defined(right_position) then ("--right-position " +  '"' + right_position + '"') else ""} \
      ~{if defined(prefix_output_files) then ("--output " +  '"' + prefix_output_files + '"') else ""} \
      ~{if defined(coordinate_file) then ("--coordinateFile " +  '"' + coordinate_file + '"') else ""} \
      ~{if defined(select_trans_list) then ("--select_trans_list " +  '"' + select_trans_list + '"') else ""} \
      ~{if defined(type) then ("--type " +  '"' + type + '"') else ""} \
      ~{if defined(id_type) then ("--id-type " +  '"' + id_type + '"') else ""}
  >>>
  runtime {
    docker: "quay.io/biocontainers/ribominer:0.2.3.1--pyh3252c3a_0"
  }
  parameter_meta {
    input_file_bam: "Input file in BAM format"
    left_position: "Left position of the sequence you are interested in."
    right_position: "Right position of the sequence you are interested in."
    prefix_output_files: "Prefix of output files.[required]"
    coordinate_file: "The file should contain the coordinate of start and\\nstop codon. Generated by\\nOutputTranscriptInfo.py.[required]"
    select_trans_list: "Selected transcript list used for metagene\\nanalysis.This files requires the first column must be\\nthe transcript ID  with a column name. default=none"
    type: "Type of regions.[CDS/5UTR/3UTR].default=CDS"
    id_type: "define the id type users input. the default is\\ntranscript id, if not, will be transformed into\\ntranscript id. default=transcript_id\\n"
  }
  output {
    File out_stdout = stdout()
  }
}