from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, File, String, Int

Fastprot_V0_1_0 = CommandToolBuilder(tool="fastprot", base_command=["fastprot"], inputs=[ToolInput(tag="in_detailed_help", input_type=Boolean(optional=True), prefix="--detailed-help", doc=InputDocumentation(doc="Print help, including all details and hidden\noptions, and exit")), ToolInput(tag="in_outfile", input_type=File(optional=True), prefix="--outfile", doc=InputDocumentation(doc="output filename. If not specified, output is\nwritten to stdout")), ToolInput(tag="in_input_format", input_type=String(optional=True), prefix="--input-format", doc=InputDocumentation(doc="input format. xml means the Fastphylo sequence\nXML format  (possible values='fasta',\n'phylip', 'xml' default=`fasta')")), ToolInput(tag="in_memory_efficient", input_type=Boolean(optional=True), prefix="--memory-efficient", doc=InputDocumentation(doc="memory efficient. Use less memory space and\nfast implementation. Only used with fasta and\nphylip format  (default=off)")), ToolInput(tag="in_output_format", input_type=String(optional=True), prefix="--output-format", doc=InputDocumentation(doc="output format. xml means the Fastphylo distance\nmatrix XML format  (possible\nvalues='phylip', 'xml', 'binary'\ndefault=`xml')")), ToolInput(tag="in_bootstraps", input_type=Int(optional=True), prefix="--bootstraps", doc=InputDocumentation(doc="Bootstrap num times and create matrix for each\n(default=`0')")), ToolInput(tag="in_no_incl_orig", input_type=Boolean(optional=True), prefix="--no-incl-orig", doc=InputDocumentation(doc="If the distance matrix from the original\nsequences should NOT be included - for\nbootstrapping  (default=off)")), ToolInput(tag="in_seed", input_type=Int(optional=True), prefix="--seed", doc=InputDocumentation(doc="Random seed. If not specified the current\ntimestamp will be used")), ToolInput(tag="in_distance_function", input_type=String(optional=True), prefix="--distance-function", doc=InputDocumentation(doc="Distance function  (possible values='ID',\n'JC', 'JCK', 'JCSS', 'WAG', 'JTT',\n'DAY', 'ARVE', 'MVR', 'LG'\ndefault=`WAG')")), ToolInput(tag="in_model_file", input_type=File(optional=True), prefix="--model-file", doc=InputDocumentation(doc="Read matrix and equilibrium distribution from\nfile, when used --distance-function is\ndisregarded")), ToolInput(tag="in_remove_indels", input_type=Boolean(optional=True), prefix="--remove-indels", doc=InputDocumentation(doc="Remove gap columns. A gap is denoted by '-'.\n(default=off)")), ToolInput(tag="in_maximum_likelihood", input_type=Boolean(optional=True), prefix="--maximum-likelihood", doc=InputDocumentation(doc="Compute a Maximum Likelihood estimate instead.\nCan not be used with --distance-function=ID,\nJC, JCK or JCSS or --sd  (default=off)")), ToolInput(tag="in_sd", input_type=Boolean(optional=True), prefix="--sd", doc=InputDocumentation(doc="Not yet implemented! Output a matrix with\nstandard deviations after the distance\nmatrix. Can not be used with\n--distance-function=ID, JC, JCK or JCSS or\n--maximum-likelihood  (default=off)")), ToolInput(tag="in_pfam", input_type=Boolean(optional=True), prefix="--pfam", doc=InputDocumentation(doc="use a normal distribution as distance prior,\nestimated from Pfam 7.2  (default=off)")), ToolInput(tag="in_speed", input_type=Int(optional=True), prefix="--speed", doc=InputDocumentation(doc="'Speed'. High speed results in low precision,\nonly affects ED calculations. Default is 5.\nValid range is [1,10].  (possible\nvalues='1', '2', '3', '4', '5',\n'6', '7', '8' default=`4')")), ToolInput(tag="in_print_relaxng_input", input_type=Boolean(optional=True), prefix="--print-relaxng-input", doc=InputDocumentation(doc="print the Relax NG schema for the XML input\nformat (Fastphylo protein sequence XML\nformat) and then exit  (default=off)")), ToolInput(tag="in_print_relaxng_output", input_type=Boolean(optional=True), prefix="--print-relaxng-output", doc=InputDocumentation(doc="print the Relax NG schema for the XML output\nformat (Fastphylo distance matrix XML format)\nand then exit.  (default=off)"))], outputs=[ToolOutput(tag="out_outfile", output_type=File(optional=True), selector=InputSelector(input_to_select="in_outfile", type_hint=File()), doc=OutputDocumentation(doc="output filename. If not specified, output is\nwritten to stdout"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Fastprot_V0_1_0().translate("wdl", allow_empty_container=True)

