from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, String, Int, Boolean, Float

Nucleoatac_Nuc_V0_1_0 = CommandToolBuilder(tool="nucleoatac_nuc", base_command=["nucleoatac", "nuc"], inputs=[ToolInput(tag="in_bed", input_type=File(optional=True), prefix="--bed", doc=InputDocumentation(doc="Regions for which to do stuff.")), ToolInput(tag="in_vm_at", input_type=File(optional=True), prefix="--vmat", doc=InputDocumentation(doc="VMat object")), ToolInput(tag="in_bam", input_type=File(optional=True), prefix="--bam", doc=InputDocumentation(doc="Accepts sorted BAM file")), ToolInput(tag="in_out", input_type=String(optional=True), prefix="--out", doc=InputDocumentation(doc="give output basename")), ToolInput(tag="in_fast_a", input_type=File(optional=True), prefix="--fasta", doc=InputDocumentation(doc="Indexed fasta file")), ToolInput(tag="in_pwm", input_type=Int(optional=True), prefix="--pwm", doc=InputDocumentation(doc="PWM descriptor file. Default is Human.PWM.txt included\nin package")), ToolInput(tag="in_sizes", input_type=File(optional=True), prefix="--sizes", doc=InputDocumentation(doc="File with fragment size distribution. Use if don't\nwant calculation of fragment size")), ToolInput(tag="in_occ_track", input_type=File(optional=True), prefix="--occ_track", doc=InputDocumentation(doc="bgzip compressed, tabix-indexed bedgraph file with\nocccupancy track. Otherwise occ not determined for nuc\npositions.")), ToolInput(tag="in_cores", input_type=Int(optional=True), prefix="--cores", doc=InputDocumentation(doc="Number of cores to use")), ToolInput(tag="in_write_all", input_type=Boolean(optional=True), prefix="--write_all", doc=InputDocumentation(doc="write all tracks")), ToolInput(tag="in_not_atac", input_type=Boolean(optional=True), prefix="--not_atac", doc=InputDocumentation(doc="data is not atac-seq")), ToolInput(tag="in_min_z", input_type=Float(optional=True), prefix="--min_z", doc=InputDocumentation(doc="Z-score threshold for nucleosome calls. Default is 3")), ToolInput(tag="in_min_lr", input_type=Float(optional=True), prefix="--min_lr", doc=InputDocumentation(doc="Log likelihood ratio threshold for nucleosome calls.\nDefault is 0")), ToolInput(tag="in_nuc_sep", input_type=Int(optional=True), prefix="--nuc_sep", doc=InputDocumentation(doc="Minimum separation between non-redundant nucleosomes.\nDefault is 120")), ToolInput(tag="in_redundant_sep", input_type=Int(optional=True), prefix="--redundant_sep", doc=InputDocumentation(doc="Minimum separation between redundant nucleosomes. Not\nrecommended to be below 15. Default is 25")), ToolInput(tag="in_sd", input_type=Int(optional=True), prefix="--sd", doc=InputDocumentation(doc="Standard deviation for smoothing. Affect the\nresolution at which nucleosomes can be positioned. Not\nrecommended to exceed 25 or to be smaller than 10.\nDefault is 10\n"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Nucleoatac_Nuc_V0_1_0().translate("wdl", allow_empty_container=True)

