from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Int, String, Float, Boolean, File

Mmseqs_Kmermatcher_V0_1_0 = CommandToolBuilder(tool="mmseqs_kmermatcher", base_command=["mmseqs", "kmermatcher"], inputs=[ToolInput(tag="in_alph_size", input_type=Int(optional=True), prefix="--alph-size", doc=InputDocumentation(doc="Alphabet size (range 2-21) [nucl:5,aa:13]")), ToolInput(tag="in_spaced_km_er_mode", input_type=Int(optional=True), prefix="--spaced-kmer-mode", doc=InputDocumentation(doc="0: use consecutive positions in k-mers; 1: use spaced k-mers [0]")), ToolInput(tag="in_spaced_km_er_pattern", input_type=String(optional=True), prefix="--spaced-kmer-pattern", doc=InputDocumentation(doc="User-specified spaced k-mer pattern []")), ToolInput(tag="in_mask", input_type=Int(optional=True), prefix="--mask", doc=InputDocumentation(doc="Mask sequences in k-mer stage: 0: w/o low complexity masking, 1: with low complexity masking [0]")), ToolInput(tag="in_mask_lower_case", input_type=Int(optional=True), prefix="--mask-lower-case", doc=InputDocumentation(doc="Lowercase letters will be excluded from k-mer search 0: include region, 1: exclude region [0]")), ToolInput(tag="in_kmer_length_set", input_type=Int(optional=True), prefix="-k", doc=InputDocumentation(doc="k-mer length (0: automatically set to optimum) [0]")), ToolInput(tag="in_split_memory_limit", input_type=Int(optional=True), prefix="--split-memory-limit", doc=InputDocumentation(doc="Set max memory per split. E.g. 800B, 5K, 10M, 1G. Default (0) to all available system memory [0]")), ToolInput(tag="in_min_seq_id", input_type=Float(optional=True), prefix="--min-seq-id", doc=InputDocumentation(doc="List matches above this sequence identity (for clustering) (range 0.0-1.0) [0.000]")), ToolInput(tag="in_cov_mode", input_type=Int(optional=True), prefix="--cov-mode", doc=InputDocumentation(doc="0: coverage of query and target\n1: coverage of target\n2: coverage of query\n3: target seq. length has to be at least x% of query length\n4: query seq. length has to be at least x% of target length\n5: short seq. needs to be at least x% of the other seq. length [0]")), ToolInput(tag="in_list_matches_fraction", input_type=Float(optional=True), prefix="-c", doc=InputDocumentation(doc="List matches above this fraction of aligned (covered) residues (see --cov-mode) [0.800]")), ToolInput(tag="in_km_er_per_seq", input_type=Int(optional=True), prefix="--kmer-per-seq", doc=InputDocumentation(doc="k-mers per sequence [0]")), ToolInput(tag="in_km_er_per_seq_scale", input_type=Int(optional=True), prefix="--kmer-per-seq-scale", doc=InputDocumentation(doc="Scale k-mer per sequence based on sequence length as kmer-per-seq val + scale x seqlen [nucl:0.200,aa:0.000]")), ToolInput(tag="in_adjust_km_er_len", input_type=Boolean(optional=True), prefix="--adjust-kmer-len", doc=InputDocumentation(doc="Adjust k-mer length based on specificity (only for nucleotides) [0]")), ToolInput(tag="in_hash_shift", input_type=Int(optional=True), prefix="--hash-shift", doc=InputDocumentation(doc="Shift k-mer hash initialization [67]")), ToolInput(tag="in_include_only_extendable", input_type=Boolean(optional=True), prefix="--include-only-extendable", doc=InputDocumentation(doc="Include only extendable [0]")), ToolInput(tag="in_ignore_multi_km_er", input_type=Boolean(optional=True), prefix="--ignore-multi-kmer", doc=InputDocumentation(doc="Skip k-mers occurring multiple times (>=2) [0]")), ToolInput(tag="in_sub_mat", input_type=File(optional=True), prefix="--sub-mat", doc=InputDocumentation(doc="Substitution matrix file [nucl:nucleotide.out,aa:blosum62.out]")), ToolInput(tag="in_max_seq_len", input_type=Int(optional=True), prefix="--max-seq-len", doc=InputDocumentation(doc="Maximum sequence length [65535]")), ToolInput(tag="in_threads", input_type=Int(optional=True), prefix="--threads", doc=InputDocumentation(doc="Number of CPU-cores used (all by default) [2]")), ToolInput(tag="in_compressed", input_type=Int(optional=True), prefix="--compressed", doc=InputDocumentation(doc="Write compressed output [0]")), ToolInput(tag="in_verbosity_level_errors", input_type=Int(optional=True), prefix="-v", doc=InputDocumentation(doc="Verbosity level: 0: quiet, 1: +errors, 2: +warnings, 3: +info [3]"))], outputs=[ToolOutput(tag="out_sub_mat", output_type=File(optional=True), selector=InputSelector(input_to_select="in_sub_mat", type_hint=File()), doc=OutputDocumentation(doc="Substitution matrix file [nucl:nucleotide.out,aa:blosum62.out]"))], container="quay.io/biocontainers/mmseqs2:13.45111--h2d02072_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Mmseqs_Kmermatcher_V0_1_0().translate("wdl")

