from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, File

Fa2Htgs_V0_1_0 = CommandToolBuilder(tool="fa2htgs", base_command=["fa2htgs"], inputs=[ToolInput(tag="in_filename_fasta_optionaldefault", input_type=Boolean(optional=True), prefix="-i", doc=InputDocumentation(doc="Filename for fasta input [File In]  Optional\ndefault = stdin")), ToolInput(tag="in_filename_seqsubmit_templatesub", input_type=Boolean(optional=True), prefix="-t", doc=InputDocumentation(doc="Filename for Seq-submit template [File In]\ndefault = template.sub")), ToolInput(tag="in_filename_optionaldefault_stdout", input_type=File(optional=True), prefix="-o", doc=InputDocumentation(doc="Filename for asn.1 output [File Out]  Optional\ndefault = stdout")), ToolInput(tag="in_log_errors_file", input_type=File(optional=True), prefix="-e", doc=InputDocumentation(doc="Log errors to file named: [File Out]  Optional")), ToolInput(tag="in_organism_name_optionaldefault", input_type=Boolean(optional=True), prefix="-n", doc=InputDocumentation(doc="Organism name? [String]  Optional\ndefault = Homo sapiens")), ToolInput(tag="in_sequence_name", input_type=Boolean(optional=True), prefix="-s", doc=InputDocumentation(doc="Sequence name? [String]")), ToolInput(tag="in_length_sequence_bp", input_type=Boolean(optional=True), prefix="-l", doc=InputDocumentation(doc="length of sequence in bp? [Integer]\ndefault = 0")), ToolInput(tag="in_genome_center_tag", input_type=Boolean(optional=True), prefix="-g", doc=InputDocumentation(doc="Genome Center tag? [String]")), ToolInput(tag="in_htgs_phase_default", input_type=Boolean(optional=True), prefix="-p", doc=InputDocumentation(doc="HTGS phase? [Integer]\ndefault = 1\nrange from 0 to 3")), ToolInput(tag="in_genbank_accession_update", input_type=Boolean(optional=True), prefix="-a", doc=InputDocumentation(doc="GenBank accession (if an update) [String]  Optional")), ToolInput(tag="in_remark_update_optional", input_type=Boolean(optional=True), prefix="-r", doc=InputDocumentation(doc="Remark for update? [String]  Optional")), ToolInput(tag="in_clone_name_optional", input_type=Boolean(optional=True), prefix="-c", doc=InputDocumentation(doc="Clone name? [String]  Optional")), ToolInput(tag="in_title_sequence_optional", input_type=Boolean(optional=True), prefix="-d", doc=InputDocumentation(doc="Title for sequence? [String]  Optional")), ToolInput(tag="in_take_comment_template", input_type=Boolean(optional=True), prefix="-m", doc=InputDocumentation(doc="Take comment from template ? [T/F]  Optional\ndefault = F")), ToolInput(tag="in_take_biosource_template", input_type=Boolean(optional=True), prefix="-u", doc=InputDocumentation(doc="Take biosource from template ? [T/F]  Optional\ndefault = F")), ToolInput(tag="in_secondary_accession_number", input_type=Boolean(optional=True), prefix="-x", doc=InputDocumentation(doc="Secondary accession number, separate by commas if multiple, s.t. U10000,L11000 [String]  Optional")), ToolInput(tag="in_clone_library_name", input_type=Boolean(optional=True), prefix="-C", doc=InputDocumentation(doc="Clone library name? [String]  Optional")), ToolInput(tag="in_map__optional", input_type=Boolean(optional=True), prefix="-M", doc=InputDocumentation(doc="Map? [String]  Optional")), ToolInput(tag="in_filename_comment_optional", input_type=Boolean(optional=True), prefix="-O", doc=InputDocumentation(doc="Filename for the comment: [File In]  Optional")), ToolInput(tag="in_filename_phrap_input_optional", input_type=Boolean(optional=True), prefix="-T", doc=InputDocumentation(doc="Filename for phrap input [File In]  Optional")), ToolInput(tag="in_contigs_use_separate", input_type=Boolean(optional=True), prefix="-P", doc=InputDocumentation(doc="Contigs to use, separate by commas if multiple [String]  Optional")), ToolInput(tag="in_filename_accession_optional", input_type=Boolean(optional=True), prefix="-A", doc=InputDocumentation(doc="Filename for accession list input [File In]  Optional")), ToolInput(tag="in_coordinates_resulting_sequence", input_type=Boolean(optional=True), prefix="-X", doc=InputDocumentation(doc="Coordinates are on the resulting sequence ? [T/F]  Optional\ndefault = F")), ToolInput(tag="in_sequence_optionaldefault_f", input_type=Boolean(optional=True), prefix="-D", doc=InputDocumentation(doc="HTGS_DRAFT sequence? [T/F]  Optional\ndefault = F")), ToolInput(tag="in_strain_name_optional", input_type=Boolean(optional=True), prefix="-S", doc=InputDocumentation(doc="Strain name? [String]  Optional")), ToolInput(tag="in_gap_length_default", input_type=Boolean(optional=True), prefix="-b", doc=InputDocumentation(doc="Gap length [Integer]\ndefault = 100\nrange from 0 to 1000000000")), ToolInput(tag="in_annotate_assemblyfragments_optionaldefault", input_type=Boolean(optional=True), prefix="-N", doc=InputDocumentation(doc="Annotate assembly_fragments [T/F]  Optional\ndefault = F")), ToolInput(tag="in_sp_clone_eg", input_type=Boolean(optional=True), prefix="-6", doc=InputDocumentation(doc="SP6 clone (e.g., Contig1,left) [String]  Optional")), ToolInput(tag="in_t_clone_eg", input_type=Boolean(optional=True), prefix="-7", doc=InputDocumentation(doc="T7 clone (e.g., Contig2,right) [String]  Optional")), ToolInput(tag="in_filename_phrap_contig_optional", input_type=Boolean(optional=True), prefix="-L", doc=InputDocumentation(doc="Filename for phrap contig order [File In]  Optional")), ToolInput(tag="in_htgsfulltop_keyword_optionaldefault", input_type=Boolean(optional=True), prefix="-f", doc=InputDocumentation(doc="htgs_fulltop keyword [T/F]  Optional\ndefault = F")), ToolInput(tag="in_htgsactivefin_keyword_optionaldefault", input_type=Boolean(optional=True), prefix="-v", doc=InputDocumentation(doc="htgs_activefin keyword [T/F]  Optional\ndefault = F")), ToolInput(tag="in_htgscancelled_keyword_optionaldefault", input_type=Boolean(optional=True), prefix="-q", doc=InputDocumentation(doc="htgs_cancelled keyword [T/F]  Optional\ndefault = F")), ToolInput(tag="in_filename_quality_optional", input_type=Boolean(optional=True), prefix="-Q", doc=InputDocumentation(doc="Filename for quality scores [File In]  Optional")), ToolInput(tag="in_whole_genome_shotgun", input_type=Boolean(optional=True), prefix="-w", doc=InputDocumentation(doc="Whole Genome Shotgun? [T/F]  Optional\ndefault = F")), ToolInput(tag="in_arbitrary_keyword_optional", input_type=Boolean(optional=True), prefix="-k", doc=InputDocumentation(doc="Arbitrary Keyword [String]  Optional"))], outputs=[ToolOutput(tag="out_filename_optionaldefault_stdout", output_type=File(optional=True), selector=InputSelector(input_to_select="in_filename_optionaldefault_stdout", type_hint=File()), doc=OutputDocumentation(doc="Filename for asn.1 output [File Out]  Optional\ndefault = stdout")), ToolOutput(tag="out_log_errors_file", output_type=File(optional=True), selector=InputSelector(input_to_select="in_log_errors_file", type_hint=File()), doc=OutputDocumentation(doc="Log errors to file named: [File Out]  Optional"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Fa2Htgs_V0_1_0().translate("wdl", allow_empty_container=True)

