from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, File, String, Int, Float, Directory

Kobas_Run_V0_1_0 = CommandToolBuilder(tool="kobas_run", base_command=["kobas-run"], inputs=[ToolInput(tag="in_list", input_type=Boolean(optional=True), prefix="--list", doc=InputDocumentation(doc="list available species, or list available databases\nfor specific species")), ToolInput(tag="in_in_file", input_type=File(optional=True), prefix="--infile", doc=InputDocumentation(doc="input data file")), ToolInput(tag="in_in_type", input_type=String(optional=True), prefix="--intype", doc=InputDocumentation(doc="input type (fasta:pro, fasta:nuc, blastout:xml,\nblastout:tab, id:refseqpro, id:uniprot, id:ensembl,\nid:ncbigene), default fasta:pro")), ToolInput(tag="in_species", input_type=Int(optional=True), prefix="--species", doc=InputDocumentation(doc="species abbreviation (for example: ko for KEGG\nOrthology, hsa for Homo sapiens, mmu for Mus musculus,\ndme for Drosophila melanogaster, ath for Arabidopsis\nthaliana, sce for Saccharomyces cerevisiae and eco for\nEscherichia coli K-12 MG1655)")), ToolInput(tag="in_evalue", input_type=Float(optional=True), prefix="--evalue", doc=InputDocumentation(doc="expect threshold for BLAST, default 1e-5")), ToolInput(tag="in_rank", input_type=Int(optional=True), prefix="--rank", doc=InputDocumentation(doc="rank cutoff for valid hits from BLAST result, default\n5")), ToolInput(tag="in_n_cpus", input_type=Int(optional=True), prefix="--nCPUs", doc=InputDocumentation(doc="number of CPUs to be used by BLAST, default 1")), ToolInput(tag="in_coverage", input_type=Int(optional=True), prefix="--coverage", doc=InputDocumentation(doc="subject coverage cutoff for BLAST, default 0")), ToolInput(tag="in_ortholog", input_type=String(optional=True), prefix="--ortholog", doc=InputDocumentation(doc="whether only use ortholog for cross-species annotation\nor not, default NO (If only use ortholog, give species\nabbr)")), ToolInput(tag="in_bg_file", input_type=File(optional=True), prefix="--bgfile", doc=InputDocumentation(doc="background file, the output of annotate (3 or 4-letter\nfile name is not allowed), or species abbreviation\n(for example: hsa for Homo sapiens, mmu for Mus\nmusculus, dme for Drosophila melanogaster, ath for\nArabidopsis thaliana, sce for Saccharomyces cerevisiae\nand eco for Escherichia coli K-12 MG1655), default\nsame species as annotate")), ToolInput(tag="in_db", input_type=Int(optional=True), prefix="--db", doc=InputDocumentation(doc="databases for selection, 1-letter abbreviation\nseparated by '/': K for KEGG PATHWAY, R for Reactome,\nB for BioCyc, p for PANTHER, o for OMIM, k for KEGG\nDISEASE, N for NHGRI GWAS Catalog and G for Gene\nOntology, S for Gene Ontology Slim, default\nK/R/B/p/o/k/N/G/S")), ToolInput(tag="in_method", input_type=String(optional=True), prefix="--method", doc=InputDocumentation(doc="choose statistical test method: b for binomial test, c\nfor chi-square test, h for hypergeometric test /\nFisher's exact test, and x for frequency list, default\nhypergeometric test / Fisher's exact test")), ToolInput(tag="in_fdr", input_type=String(optional=True), prefix="--fdr", doc=InputDocumentation(doc="choose false discovery rate (FDR) correction method:\nBH for Benjamini and Hochberg, BY for Benjamini and\nYekutieli, QVALUE, and None, default BH")), ToolInput(tag="in_outfile", input_type=File(optional=True), prefix="--outfile", doc=InputDocumentation(doc="output file for identification result, default stdout")), ToolInput(tag="in_cut_off", input_type=Int(optional=True), prefix="--cutoff", doc=InputDocumentation(doc="the gene number in a term is not less than the cutoff,\ndefault 5")), ToolInput(tag="in_kob_as_home", input_type=File(optional=True), prefix="--kobashome", doc=InputDocumentation(doc="Optional parameter. To set path to kobas_home, which\nis parent directory of sqlite3/ and seq_pep/ , default\nvalue is read from ~/.kobasrcwhere you set before\nrunning kobas. If you set this parameter, it means you\nset 'kobasdb' and 'blastdb' in this following\ndirectory. e.g. '-k /home/user/kobas/', means that you\nset kobasdb = /home/user/kobas/sqlite3/ and blastdb =\n/home/user/kobas/seq_pep/")), ToolInput(tag="in_blast_home", input_type=Directory(optional=True), prefix="--blasthome", doc=InputDocumentation(doc="Optional parameter. To set parent directory of blastx\nand blastp. If you set this parameter, it means you\nset 'blastx' and 'blastp' in this following directory.\nDefault value is read from ~/.kobasrc where you set\nbefore running kobas")), ToolInput(tag="in_blast_db", input_type=File(optional=True), prefix="--blastdb", doc=InputDocumentation(doc="Optional parameter. To set path to sep_pep/, default\nvalue is read from ~/.kobasrc where you set before\nrunning kobas")), ToolInput(tag="in_koba_sdb", input_type=File(optional=True), prefix="--kobasdb", doc=InputDocumentation(doc="Optional parameter. To set path to sqlite3/, default\nvalue is read from ~/.kobasrc where you set before\nrunning kobas, e.g. '-q /kobas_home/sqlite3/'")), ToolInput(tag="in_blastp", input_type=File(optional=True), prefix="--blastp", doc=InputDocumentation(doc="Optional parameter. To set path to blastp program,\ndefault value is read from ~/.kobasrc where you set\nbefore running kobas")), ToolInput(tag="in_blastx", input_type=File(optional=True), prefix="--blastx", doc=InputDocumentation(doc="Optional parameter. To set path to  blasx program,\ndefault value is read from ~/.kobasrc where you set\nbefore running kobas\n")), ToolInput(tag="in_run_kob_as_do_tpy", input_type=String(), position=0, doc=InputDocumentation(doc=""))], outputs=[ToolOutput(tag="out_bg_file", output_type=File(optional=True), selector=InputSelector(input_to_select="in_bg_file", type_hint=File()), doc=OutputDocumentation(doc="background file, the output of annotate (3 or 4-letter\nfile name is not allowed), or species abbreviation\n(for example: hsa for Homo sapiens, mmu for Mus\nmusculus, dme for Drosophila melanogaster, ath for\nArabidopsis thaliana, sce for Saccharomyces cerevisiae\nand eco for Escherichia coli K-12 MG1655), default\nsame species as annotate")), ToolOutput(tag="out_outfile", output_type=File(optional=True), selector=InputSelector(input_to_select="in_outfile", type_hint=File()), doc=OutputDocumentation(doc="output file for identification result, default stdout"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Kobas_Run_V0_1_0().translate("wdl", allow_empty_container=True)

