from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import String, Directory, Boolean, Int, Float, File

Truvari_V0_1_0 = CommandToolBuilder(tool="truvari", base_command=["truvari"], inputs=[ToolInput(tag="in_base", input_type=String(optional=True), prefix="--base", doc=InputDocumentation(doc="Baseline truth-set calls")), ToolInput(tag="in_comp", input_type=String(optional=True), prefix="--comp", doc=InputDocumentation(doc="Comparison set of calls")), ToolInput(tag="in_output", input_type=Directory(optional=True), prefix="--output", doc=InputDocumentation(doc="Output directory")), ToolInput(tag="in_reference", input_type=String(optional=True), prefix="--reference", doc=InputDocumentation(doc="Indexed fasta used to call variants")), ToolInput(tag="in_gia_b_report", input_type=Boolean(optional=True), prefix="--giabreport", doc=InputDocumentation(doc="Parse output TPs/FNs for GIAB annotations and create a")), ToolInput(tag="in_ref_dist", input_type=Int(optional=True), prefix="--refdist", doc=InputDocumentation(doc="Max reference location distance (500)")), ToolInput(tag="in_pct_sim", input_type=Int(optional=True), prefix="--pctsim", doc=InputDocumentation(doc="Min percent allele sequence similarity. Set to 0 to\nignore. (0.7)")), ToolInput(tag="in_pct_size", input_type=Int(optional=True), prefix="--pctsize", doc=InputDocumentation(doc="Min pct allele size similarity (minvarsize/maxvarsize)\n(0.7)")), ToolInput(tag="in_pc_to_vl", input_type=Float(optional=True), prefix="--pctovl", doc=InputDocumentation(doc="Minimum pct reciprocal overlap (0.0)")), ToolInput(tag="in_type_ignore", input_type=Boolean(optional=True), prefix="--typeignore", doc=InputDocumentation(doc="Variant types don't need to match to compare (False)")), ToolInput(tag="in_gt_comp", input_type=Boolean(optional=True), prefix="--gtcomp", doc=InputDocumentation(doc="Compare GenoTypes of matching calls")), ToolInput(tag="in_b_sample", input_type=String(optional=True), prefix="--bSample", doc=InputDocumentation(doc="Baseline calls sample to use (first)")), ToolInput(tag="in_c_sample", input_type=String(optional=True), prefix="--cSample", doc=InputDocumentation(doc="Comparison calls sample to use (first)")), ToolInput(tag="in_size_min", input_type=Int(optional=True), prefix="--sizemin", doc=InputDocumentation(doc="Minimum variant size to consider for comparison (50)")), ToolInput(tag="in_size_filt", input_type=Int(optional=True), prefix="--sizefilt", doc=InputDocumentation(doc="Minimum variant size to load into IntervalTree (30)")), ToolInput(tag="in_size_max", input_type=Int(optional=True), prefix="--sizemax", doc=InputDocumentation(doc="Maximum variant size to consider for comparison\n(50000)")), ToolInput(tag="in_pass_only", input_type=Boolean(optional=True), prefix="--passonly", doc=InputDocumentation(doc="Only consider calls with FILTER == PASS")), ToolInput(tag="in_no_ref", input_type=Boolean(optional=True), prefix="--no-ref", doc=InputDocumentation(doc="Don't include 0/0 or ./. GT calls (False)")), ToolInput(tag="in_include_bed", input_type=File(optional=True), prefix="--includebed", doc=InputDocumentation(doc="Bed file of regions in the genome to include only\ncalls overlapping")), ToolInput(tag="in_multi_match", input_type=Boolean(optional=True), prefix="--multimatch", doc=InputDocumentation(doc="Allow base calls to match multiple comparison calls\n(False)\n")), ToolInput(tag="in_report", input_type=String(), position=0, doc=InputDocumentation(doc="--debug               Verbose logging"))], outputs=[ToolOutput(tag="out_output", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_output", type_hint=File()), doc=OutputDocumentation(doc="Output directory"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Truvari_V0_1_0().translate("wdl", allow_empty_container=True)

