from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Int, String, Array, Boolean, Float

Tombo_Build_Model_Estimate_Alt_Reference_V0_1_0 = CommandToolBuilder(tool="tombo_build_model_estimate_alt_reference", base_command=["tombo", "build_model", "estimate_alt_reference"], inputs=[ToolInput(tag="in_tombo_model_alternative", input_type=File(optional=True), prefix="--alternate-model-filename", doc=InputDocumentation(doc="Tombo model for alternative likelihood ratio\nsignificance testing.")), ToolInput(tag="in_short_name_associate", input_type=Int(optional=True), prefix="--alternate-model-name", doc=InputDocumentation(doc="A short name to associate with this alternate model\n(e.g. 5mC, 6mA, etc.). This text will be included in\noutput filenames when this model is used for testing.")), ToolInput(tag="in_alternate_model_base", input_type=String(optional=True), prefix="--alternate-model-base", doc=InputDocumentation(doc="Non-standard base is an alternative to this base.")), ToolInput(tag="in_fast_five_based_irs", input_type=Array(t=Int(), optional=True), prefix="--fast5-basedirs", doc=InputDocumentation(doc="Directories containing fast5 files.")), ToolInput(tag="in_control_fast_five_based_irs", input_type=Array(t=Int(), optional=True), prefix="--control-fast5-basedirs", doc=InputDocumentation(doc="Set of directories containing fast5 files for control\nreads, containing only canonical nucleotides.")), ToolInput(tag="in_alternate_density_filename", input_type=File(optional=True), prefix="--alternate-density-filename", doc=InputDocumentation(doc="File containing k-mer level kernel density estimates\nfor the alternative sample saved using --save-density-\nbasename.")), ToolInput(tag="in_control_density_filename", input_type=File(optional=True), prefix="--control-density-filename", doc=InputDocumentation(doc="File containing k-mer level kernel density estimates\nfor the control sample saved using --save-density-\nbasename.")), ToolInput(tag="in_dna", input_type=Boolean(optional=True), prefix="--dna", doc=InputDocumentation(doc="Explicitly select canonical DNA model. Default:\nAutomatically determine from FAST5s")), ToolInput(tag="in_rna", input_type=Boolean(optional=True), prefix="--rna", doc=InputDocumentation(doc="Explicitly select canonical RNA model. Default:\nAutomatically determine from FAST5s")), ToolInput(tag="in_tombo_model_filename", input_type=File(optional=True), prefix="--tombo-model-filename", doc=InputDocumentation(doc="Tombo model filename. If no file is provided, the\ndefault DNA or RNA Tombo model will be used.")), ToolInput(tag="in_alt_fraction_percentile", input_type=Float(optional=True), prefix="--alt-fraction-percentile", doc=InputDocumentation(doc="When esitmating the alternative base incorporation\nrate, this percent of k-mers are assumed to have\nsignificantly shifted signal so the alternative\ndistribution minimally overlaps the standard base\ndistribution. Default: 5.000000")), ToolInput(tag="in_kernel_density_bandwidth", input_type=Float(optional=True), prefix="--kernel-density-bandwidth", doc=InputDocumentation(doc="Bandwidth applied when performing Gaussian kernal\ndensity esitmation on standard and alternative base\nsignal distributions. Default: 0.050000")), ToolInput(tag="in_minimum_km_er_observations", input_type=Int(optional=True), prefix="--minimum-kmer-observations", doc=InputDocumentation(doc="Number of each k-mer observations required in order to\nproduce a reference (genomic locations for standard\nreference and per-read for alternative reference).\nDefault: 1000")), ToolInput(tag="in_save_density_basename", input_type=String(optional=True), prefix="--save-density-basename", doc=InputDocumentation(doc="Basename to save alternative model estimation density\nestimation information. See scripts/debug_est_alt.R\nfor info use example. Default: Don't save.")), ToolInput(tag="in_processes", input_type=Int(optional=True), prefix="--processes", doc=InputDocumentation(doc="Number of processes. Default: 1")), ToolInput(tag="in_corrected_group", input_type=Int(optional=True), prefix="--corrected-group", doc=InputDocumentation(doc="FAST5 group created by resquiggle command. Default:\nRawGenomeCorrected_000")), ToolInput(tag="in_base_call_subgroups", input_type=Array(t=String(), optional=True), prefix="--basecall-subgroups", doc=InputDocumentation(doc="FAST5 subgroup(s) (under /Analyses/[--basecall-\ngroup]/) containing basecalls and created within\n[--corrected-group] containing re-squiggle results.\nDefault: ['BaseCalled_template']")), ToolInput(tag="in_quiet", input_type=Boolean(optional=True), prefix="--quiet", doc=InputDocumentation(doc="Don't print status information.")), ToolInput(tag="in__alternatemodelname", input_type=String(), position=0, doc=InputDocumentation(doc="--alternate-model-name")), ToolInput(tag="in__alternatemodelbase", input_type=String(), position=1, doc=InputDocumentation(doc="--alternate-model-base"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Tombo_Build_Model_Estimate_Alt_Reference_V0_1_0().translate("wdl", allow_empty_container=True)

