from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, String

Rnaseqc_V0_1_0 = CommandToolBuilder(tool="rnaseqc", base_command=["rnaseqc"], inputs=[ToolInput(tag="in_name_current_sample", input_type=Boolean(optional=True), prefix="-s", doc=InputDocumentation(doc="[sample], --sample=[sample]     The name of the current sample. Default:\nThe bam's filename")), ToolInput(tag="in_bed", input_type=Boolean(optional=True), prefix="--bed", doc=InputDocumentation(doc="=[BEDFILE]                   Optional input BED file containing\nnon-overlapping exons used for fragment\nsize calculations")), ToolInput(tag="in_fast_a", input_type=Boolean(optional=True), prefix="--fasta", doc=InputDocumentation(doc="=[fasta]                   Optional input FASTA/FASTQ file\ncontaining the reference sequence used\nfor parsing CRAM files")), ToolInput(tag="in_chimeric_distance", input_type=Boolean(optional=True), prefix="--chimeric-distance", doc=InputDocumentation(doc="=[DISTANCE]    Set the maximum accepted distance\nbetween read mates. Mates beyond this\ndistance will be counted as chimeric\npairs. Default: 2000000 [bp]")), ToolInput(tag="in_fragment_samples", input_type=Boolean(optional=True), prefix="--fragment-samples", doc=InputDocumentation(doc="=[SAMPLES]      Set the number of samples to take when\ncomputing fragment sizes. Requires the\n--bed argument. Default: 1000000")), ToolInput(tag="in_q", input_type=Boolean(optional=True), prefix="-q", doc=InputDocumentation(doc="[QUALITY],")), ToolInput(tag="in_mapping_quality", input_type=Boolean(optional=True), prefix="--mapping-quality", doc=InputDocumentation(doc="=[QUALITY]       Set the lower bound on read quality for\nexon coverage counting. Reads below this\nnumber are excluded from coverage\nmetrics. Default: 255")), ToolInput(tag="in_base_mismatch", input_type=Boolean(optional=True), prefix="--base-mismatch", doc=InputDocumentation(doc="=[MISMATCHES]      Set the maximum number of allowed\nmismatches between a read and the\nreference sequence. Reads with more than\nthis number of mismatches are excluded\nfrom coverage metrics. Default: 6")), ToolInput(tag="in_offset", input_type=Boolean(optional=True), prefix="--offset", doc=InputDocumentation(doc="=[OFFSET]                 Set the offset into the gene for the 3'\nand 5' windows in bias calculation. A\npositive value shifts the 3' and 5'\nwindows towards eachother, while a\nnegative value shifts them apart.\nDefault: 150 [bp]")), ToolInput(tag="in_window_size", input_type=Boolean(optional=True), prefix="--window-size", doc=InputDocumentation(doc="=[SIZE]              Set the size of the 3' and 5' windows in\nbias calculation. Default: 100 [bp]")), ToolInput(tag="in_gene_length", input_type=Boolean(optional=True), prefix="--gene-length", doc=InputDocumentation(doc="=[LENGTH]            Set the minimum size of a gene for bias\ncalculation. Genes below this size are\nignored in the calculation. Default: 600\n[bp]")), ToolInput(tag="in_legacy", input_type=Boolean(optional=True), prefix="--legacy", doc=InputDocumentation(doc="Use legacy counting rules. Gene and exon\ncounts match output of RNA-SeQC 1.1.9")), ToolInput(tag="in_stranded", input_type=Boolean(optional=True), prefix="--stranded", doc=InputDocumentation(doc="=[stranded]             Use strand-specific metrics. Only\nfeatures on the same strand of a read\nwill be considered. Allowed values are\n'RF', 'rf', 'FR', and 'fr'")), ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="--verbose", doc=InputDocumentation(doc="Give some feedback about what's going\non. Supply this argument twice for\nprogress updates while parsing the bam")), ToolInput(tag="in_filter_reads_specified", input_type=Boolean(optional=True), prefix="-t", doc=InputDocumentation(doc="[TAG...], --tag=[TAG...]        Filter out reads with the specified tag.")), ToolInput(tag="in_chimeric_tag", input_type=Boolean(optional=True), prefix="--chimeric-tag", doc=InputDocumentation(doc="=[TAG]              Reads maked with the specified tag will\nbe labeled as Chimeric. Defaults to 'mC'\nfor STAR")), ToolInput(tag="in_exclude_chimeric", input_type=Boolean(optional=True), prefix="--exclude-chimeric", doc=InputDocumentation(doc="Exclude chimeric reads from the read")), ToolInput(tag="in_rp_km", input_type=Boolean(optional=True), prefix="--rpkm", doc=InputDocumentation(doc="Output gene RPKM values instead of TPMs")), ToolInput(tag="in_coverage", input_type=Boolean(optional=True), prefix="--coverage", doc=InputDocumentation(doc="If this flag is provided, coverage\nstatistics for each transcript will be\nwritten to a table. Otherwise, only\nsummary coverage statistics are\ngenerated and added to the metrics table")), ToolInput(tag="in_coverage_mask", input_type=Boolean(optional=True), prefix="--coverage-mask", doc=InputDocumentation(doc="=[SIZE]            Sets how many bases at both ends of a\ntranscript are masked out when computing\nper-base exon coverage. Default: 500bp")), ToolInput(tag="in_detection_threshold", input_type=Boolean(optional=True), prefix="--detection-threshold", doc=InputDocumentation(doc="=[threshold] Number of counts on a gene to consider\nthe gene 'detected'. Additionally, genes\nbelow this limit are excluded from 3'\nbias computation. Default: 5 reads")), ToolInput(tag="in_gtf", input_type=String(), position=0, doc=InputDocumentation(doc="The input GTF file containing features\nto check the bam against")), ToolInput(tag="in_bam", input_type=String(), position=1, doc=InputDocumentation(doc="The input SAM/BAM file containing reads\nto process")), ToolInput(tag="in_output", input_type=String(), position=2, doc=InputDocumentation(doc="Output directory")), ToolInput(tag="in_counts", input_type=String(), position=3, doc=InputDocumentation(doc="-u, --unpaired                    Allow unpaired reads to be quantified."))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Rnaseqc_V0_1_0().translate("wdl", allow_empty_container=True)

