from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import String, File, Int, Boolean

Cansnper_V0_1_0 = CommandToolBuilder(tool="CanSNPer", base_command=["CanSNPer"], inputs=[ToolInput(tag="in_reference", input_type=String(optional=True), prefix="--reference", doc=InputDocumentation(doc="the name of the organism")), ToolInput(tag="in_query", input_type=File(optional=True), prefix="--query", doc=InputDocumentation(doc="fasta sequence file name that is to be analysed")), ToolInput(tag="in_db_path", input_type=File(optional=True), prefix="--db_path", doc=InputDocumentation(doc="path to CanSNPerDB.db")), ToolInput(tag="in_import_tree_file", input_type=File(optional=True), prefix="--import_tree_file", doc=InputDocumentation(doc="imports a tree structure into the database")), ToolInput(tag="in_import_snp_file", input_type=File(optional=True), prefix="--import_snp_file", doc=InputDocumentation(doc="imports a list of SNPs into the database")), ToolInput(tag="in_import_seq_file", input_type=File(optional=True), prefix="--import_seq_file", doc=InputDocumentation(doc="loads a sequence file into the database")), ToolInput(tag="in_strain_name", input_type=String(optional=True), prefix="--strain_name", doc=InputDocumentation(doc="the name of the strain")), ToolInput(tag="in_allow_differences", input_type=Int(optional=True), prefix="--allow_differences", doc=InputDocumentation(doc="allow a number of SNPs to be wrong, i.e.continue\nmoving down the tree even if none of the SNPs of the\nlower level are present [0]")), ToolInput(tag="in_tab_sep", input_type=Boolean(optional=True), prefix="--tab_sep", doc=InputDocumentation(doc="print the results in a simple tab separated format")), ToolInput(tag="in_draw_tree", input_type=Boolean(optional=True), prefix="--draw_tree", doc=InputDocumentation(doc="draw a pdf version of the tree, marking SNPs of the\nquery sequence")), ToolInput(tag="in_progressive_mauve", input_type=File(optional=True), prefix="--progressiveMauve", doc=InputDocumentation(doc="path to progressiveMauve binary file")), ToolInput(tag="in_list_snps", input_type=Boolean(optional=True), prefix="--list_snps", doc=InputDocumentation(doc="lists the SNPs of the given sequence")), ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="--verbose", doc=InputDocumentation(doc="prints some more information about the goings-ons of\nthe program while running")), ToolInput(tag="in_save_align", input_type=Boolean(optional=True), prefix="--save_align", doc=InputDocumentation(doc="saves the alignment file")), ToolInput(tag="in_num_threads", input_type=Int(optional=True), prefix="--num_threads", doc=InputDocumentation(doc="maximum number of threads CanSNPer is allowed to use,\nthe default [0] is no limit, CanSNPer will start one\nprocess per reference genome while aligning")), ToolInput(tag="in_delete_organism", input_type=Boolean(optional=True), prefix="-delete_organism", doc=InputDocumentation(doc="deletes all information in the database concerning an")), ToolInput(tag="in_tmp_path", input_type=File(optional=True), prefix="--tmp_path", doc=InputDocumentation(doc="where temporary files are stored")), ToolInput(tag="in_dev", input_type=Boolean(optional=True), prefix="--dev", doc=InputDocumentation(doc="dev mode")), ToolInput(tag="in_galaxy", input_type=Boolean(optional=True), prefix="--galaxy", doc=InputDocumentation(doc="argument used if Galaxy is running CanSNPer, do NOT\nuse.\n")), ToolInput(tag="in_organism", input_type=String(), position=0, doc=InputDocumentation(doc="-initialise_organism  initialise a new table for an organism"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Cansnper_V0_1_0().translate("wdl", allow_empty_container=True)

