from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, Directory, String

Cat_Bins_V0_1_0 = CommandToolBuilder(tool="CAT_bins", base_command=["CAT", "bins"], inputs=[ToolInput(tag="in_bin_folder", input_type=Boolean(optional=True), prefix="--bin_folder", doc=InputDocumentation(doc="Path to directory containing bins.")), ToolInput(tag="in_database_folder", input_type=Boolean(optional=True), prefix="--database_folder", doc=InputDocumentation(doc="Path to folder that contains database files.")), ToolInput(tag="in_taxonomy_folder", input_type=Boolean(optional=True), prefix="--taxonomy_folder", doc=InputDocumentation(doc="Path to folder that contains taxonomy files.")), ToolInput(tag="in_bin_suffix", input_type=Boolean(optional=True), prefix="--bin_suffix", doc=InputDocumentation(doc="Suffix of bins in bin folder (default: .fna).")), ToolInput(tag="in_range", input_type=Boolean(optional=True), prefix="--range", doc=InputDocumentation(doc="r parameter [0-49] (default: 5).")), ToolInput(tag="in_out_prefix", input_type=Boolean(optional=True), prefix="--out_prefix", doc=InputDocumentation(doc="Prefix for output files (default: out.BAT).")), ToolInput(tag="in_proteins_fast_a", input_type=Boolean(optional=True), prefix="--proteins_fasta", doc=InputDocumentation(doc="Path to concatenated predicted proteins fasta file\ngenerated during an earlier run of BAT. If supplied,\nBAT will skip the protein prediction step.")), ToolInput(tag="in_diamond_alignment", input_type=Boolean(optional=True), prefix="--diamond_alignment", doc=InputDocumentation(doc="Path to alignment table generated during an earlier\nrun of BAT. If supplied, BAT will skip the alignment\nstep and directly classify the bins. A concatenated\npredicted proteins fasta file should also be supplied\nwith argument [-p / --proteins].")), ToolInput(tag="in_path_to_prodigal", input_type=Boolean(optional=True), prefix="--path_to_prodigal", doc=InputDocumentation(doc="Path to Prodigal binaries. Supply if BAT can not find")), ToolInput(tag="in_force", input_type=Boolean(optional=True), prefix="--force", doc=InputDocumentation(doc="Force overwrite existing files.")), ToolInput(tag="in_quiet", input_type=Boolean(optional=True), prefix="--quiet", doc=InputDocumentation(doc="Suppress verbosity.")), ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="--verbose", doc=InputDocumentation(doc="Increase verbosity.")), ToolInput(tag="in_no_log", input_type=Boolean(optional=True), prefix="--no_log", doc=InputDocumentation(doc="Suppress log file.")), ToolInput(tag="in_i_know_what_im_doing", input_type=Boolean(optional=True), prefix="--I_know_what_Im_doing", doc=InputDocumentation(doc="Flag for experimental features.")), ToolInput(tag="in_nproc", input_type=Boolean(optional=True), prefix="--nproc", doc=InputDocumentation(doc="Number of cores to deploy by DIAMOND (default:\nmaximum).")), ToolInput(tag="in_sensitive", input_type=Boolean(optional=True), prefix="--sensitive", doc=InputDocumentation(doc="Run DIAMOND in sensitive mode (default: not enabled).")), ToolInput(tag="in_block_size", input_type=Boolean(optional=True), prefix="--block_size", doc=InputDocumentation(doc="DIAMOND block-size parameter (default: 2.0). Lower\nnumbers will decrease memory and temporary disk space\nusage.")), ToolInput(tag="in_index_chunks", input_type=Boolean(optional=True), prefix="--index_chunks", doc=InputDocumentation(doc="DIAMOND index-chunks parameter (default: 4). Set to 1\non high memory machines. The parameter has no effect\non temporary disk space usage.")), ToolInput(tag="in_tmpdir", input_type=Directory(optional=True), prefix="--tmpdir", doc=InputDocumentation(doc="Directory for temporary DIAMOND files (default:\ndirectory to which output files are written).")), ToolInput(tag="in_compress", input_type=Boolean(optional=True), prefix="--compress", doc=InputDocumentation(doc="Compress DIAMOND alignment file.")), ToolInput(tag="in_top", input_type=Boolean(optional=True), prefix="--top", doc=InputDocumentation(doc="DIAMOND top parameter [0-50] (default: 50). Governs\nhits within range of best hit that are written to the\nalignment file. This is not the [-r / --range]\nparameter! Can only be set with the\n[--I_know_what_Im_doing] flag, see README.md.\n")), ToolInput(tag="in_prodigal_dot", input_type=String(), position=0, doc=InputDocumentation(doc="--path_to_diamond     Path to DIAMOND binaries. Supply if BAT can not find")), ToolInput(tag="in_diamond_dot", input_type=String(), position=1, doc=InputDocumentation(doc="--no_stars            Suppress marking of suggestive classifications."))], outputs=[ToolOutput(tag="out_tmpdir", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_tmpdir", type_hint=File()), doc=OutputDocumentation(doc="Directory for temporary DIAMOND files (default:\ndirectory to which output files are written)."))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Cat_Bins_V0_1_0().translate("wdl", allow_empty_container=True)

