from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Int, Boolean, File, Float

Abra2_V0_1_0 = CommandToolBuilder(tool="abra2", base_command=["abra2"], inputs=[ToolInput(tag="in_am_q", input_type=Int(optional=True), prefix="--amq", doc=InputDocumentation(doc="Set mapq for alignments that map\nequally well to reference and an\nABRA generated contig.  default of\n-1 disables (default: -1)")), ToolInput(tag="in_ca", input_type=Boolean(optional=True), prefix="--ca", doc=InputDocumentation(doc="Contig anchor [M_bases_at_contig_edge,\nmax_mismatches_near_edge] (default:\n10,2)")), ToolInput(tag="in_cl", input_type=File(optional=True), prefix="--cl", doc=InputDocumentation(doc="Compression level of output bam file\n(s) (default: 5)")), ToolInput(tag="in_cons", input_type=Boolean(optional=True), prefix="--cons", doc=InputDocumentation(doc="Use positional consensus sequence when\naligning high quality soft clipping")), ToolInput(tag="in_contigs", input_type=Boolean(optional=True), prefix="--contigs", doc=InputDocumentation(doc="Optional file to which assembled\ncontigs are written")), ToolInput(tag="in_dist", input_type=Int(optional=True), prefix="--dist", doc=InputDocumentation(doc="Max read move distance (default: 1000)")), ToolInput(tag="in_gc", input_type=Boolean(optional=True), prefix="--gc", doc=InputDocumentation(doc="If specified, only reprocess regions\nthat contain at least one contig\ncontaining an indel or splice\n(experimental)")), ToolInput(tag="in_gkl", input_type=Boolean(optional=True), prefix="--gkl", doc=InputDocumentation(doc="If specified, use the GKL Intel\nDeflater.")), ToolInput(tag="in_gtf", input_type=Boolean(optional=True), prefix="--gtf", doc=InputDocumentation(doc="GTF file defining exons and transcripts")), ToolInput(tag="in_in", input_type=Boolean(optional=True), prefix="--in", doc=InputDocumentation(doc="Required list of input sam or bam file\n(s) separated by comma")), ToolInput(tag="in_in_vcf", input_type=Boolean(optional=True), prefix="--in-vcf", doc=InputDocumentation(doc="VCF containing known (or suspected)\nvariant sites.  Very large files\nshould be avoided.")), ToolInput(tag="in_index", input_type=Boolean(optional=True), prefix="--index", doc=InputDocumentation(doc="Enable BAM index generation when\noutputting sorted alignments (may\nrequire additonal memory)")), ToolInput(tag="in_junctions", input_type=Boolean(optional=True), prefix="--junctions", doc=InputDocumentation(doc="Splice junctions definition file")), ToolInput(tag="in_keep_tmp", input_type=Boolean(optional=True), prefix="--keep-tmp", doc=InputDocumentation(doc="Do not delete the temporary directory")), ToolInput(tag="in_km_er", input_type=Boolean(optional=True), prefix="--kmer", doc=InputDocumentation(doc="Optional assembly kmer size(delimit\nwith commas if multiple sizes\nspecified)")), ToolInput(tag="in_log", input_type=Boolean(optional=True), prefix="--log", doc=InputDocumentation(doc="Logging level (trace,debug,info,warn,\nerror) (default: info)")), ToolInput(tag="in_mac", input_type=Int(optional=True), prefix="--mac", doc=InputDocumentation(doc="Max assembled contigs (default: 64)")), ToolInput(tag="in_mad", input_type=Int(optional=True), prefix="--mad", doc=InputDocumentation(doc="Regions with average depth exceeding\nthis value will be downsampled\n(default: 1000)")), ToolInput(tag="in_mapq", input_type=Boolean(optional=True), prefix="--mapq", doc=InputDocumentation(doc="[Integer]                        Minimum mapping quality for a read to\nbe used in assembly and be eligible\nfor realignment (default: 20)")), ToolInput(tag="in_max_n", input_type=Boolean(optional=True), prefix="--maxn", doc=InputDocumentation(doc="[Integer]                        Maximum pre-pruned nodes in regional\nassembly (default: 150000)")), ToolInput(tag="in_mbq", input_type=Boolean(optional=True), prefix="--mbq", doc=InputDocumentation(doc="[Integer]                         Minimum base quality for inclusion in\nassembly.  This value is compared\nagainst the sum of base qualities\nper kmer position (default: 20)")), ToolInput(tag="in_mcl", input_type=Boolean(optional=True), prefix="--mcl", doc=InputDocumentation(doc="[Integer]                         Assembly minimum contig length\n(default: -1)")), ToolInput(tag="in_mcr", input_type=Int(optional=True), prefix="--mcr", doc=InputDocumentation(doc="Max number of cached reads per sample\nper thread (default: 1000000)")), ToolInput(tag="in_mer", input_type=Int(optional=True), prefix="--mer", doc=InputDocumentation(doc="Min edge pruning ratio.  Default value\nis appropriate for relatively\nsensitive somatic cases.  May be\nincreased for improved speed in\ngermline only cases. (default: 0.01)")), ToolInput(tag="in_mmr", input_type=Int(optional=True), prefix="--mmr", doc=InputDocumentation(doc="Max allowed mismatch rate when mapping\nreads back to contigs (default: 0.05)")), ToolInput(tag="in_mnf", input_type=Int(optional=True), prefix="--mnf", doc=InputDocumentation(doc="Assembly minimum node frequency\n(default: 1)")), ToolInput(tag="in_mrn", input_type=Float(optional=True), prefix="--mrn", doc=InputDocumentation(doc="Reads with noise score exceeding this\nvalue are not remapped.\nnumMismatches+(numIndels*2) <\nreadLength*mnr (default: 0.1)")), ToolInput(tag="in_mrr", input_type=Int(optional=True), prefix="--mrr", doc=InputDocumentation(doc="Regions containing more reads than\nthis value are not processed.  Use\n-1 to disable. (default: 1000000)")), ToolInput(tag="in_msr", input_type=Int(optional=True), prefix="--msr", doc=InputDocumentation(doc="Max reads to keep in memory per sample\nduring the sort phase.  When this\nvalue is exceeded, sort spills to\ndisk (default: 1000000)")), ToolInput(tag="in_no_edge_ci", input_type=Boolean(optional=True), prefix="--no-edge-ci", doc=InputDocumentation(doc="If specified, do not update alignments\nfor reads that have a complex indel\nat the read edge.  i.e. Do not allow\nalignments like: 90M10D10I")), ToolInput(tag="in_no_ndn", input_type=Boolean(optional=True), prefix="--no-ndn", doc=InputDocumentation(doc="If specified, do not allow adjacent N-\nD-N cigar elements")), ToolInput(tag="in_no_sort", input_type=Boolean(optional=True), prefix="--nosort", doc=InputDocumentation(doc="Do not attempt to sort final output")), ToolInput(tag="in_out", input_type=File(optional=True), prefix="--out", doc=InputDocumentation(doc="Required list of output sam or bam file\n(s) separated by comma")), ToolInput(tag="in_rcf", input_type=Float(optional=True), prefix="--rcf", doc=InputDocumentation(doc="Minimum read candidate fraction for\ntriggering assembly (default: 0.01)")), ToolInput(tag="in_ref", input_type=Boolean(optional=True), prefix="--ref", doc=InputDocumentation(doc="Genome reference location")), ToolInput(tag="in_sa", input_type=Boolean(optional=True), prefix="--sa", doc=InputDocumentation(doc="Skip assembly")), ToolInput(tag="in_sc", input_type=Boolean(optional=True), prefix="--sc", doc=InputDocumentation(doc="Soft clip contig args [max_contigs,\nmin_base_qual,frac_high_qual_bases,\nmin_soft_clip_len] (default:\n16,13,80,15)")), ToolInput(tag="in_sga", input_type=Boolean(optional=True), prefix="--sga", doc=InputDocumentation(doc="Scoring used for contig alignments\n(match, mismatch_penalty,\ngap_open_penalty,\ngap_extend_penalty) (default:\n8,32,48,1)")), ToolInput(tag="in_single", input_type=Boolean(optional=True), prefix="--single", doc=InputDocumentation(doc="Input is single end")), ToolInput(tag="in_skip", input_type=Boolean(optional=True), prefix="--skip", doc=InputDocumentation(doc="If no target specified, skip\nrealignment of chromosomes matching\nspecified regex.  Skipped reads are\noutput without modification.\nSpecify none to disable. (default:\nGL.*|hs37d5|chr.*random|chrUn.\n*|chrEBV|CMV|HBV|HCV.*|HIV.\n*|KSHV|HTLV.*|MCV|SV40|HPV.*)")), ToolInput(tag="in_sobs", input_type=Boolean(optional=True), prefix="--sobs", doc=InputDocumentation(doc="Do not use observed indels in original\nalignments to generate contigs")), ToolInput(tag="in_ssc", input_type=Boolean(optional=True), prefix="--ssc", doc=InputDocumentation(doc="Skip usage of soft clipped sequences\nas putative contigs")), ToolInput(tag="in_sua", input_type=Boolean(optional=True), prefix="--sua", doc=InputDocumentation(doc="Do not use unmapped reads anchored by\nmate to trigger assembly.  These\nreads are still eligible to\ncontribute to assembly")), ToolInput(tag="in_target_km_ers", input_type=Boolean(optional=True), prefix="--target-kmers", doc=InputDocumentation(doc="BED-like file containing target\nregions with per region kmer sizes\nin 4th column")), ToolInput(tag="in_targets", input_type=Boolean(optional=True), prefix="--targets", doc=InputDocumentation(doc="BED file containing target regions")), ToolInput(tag="in_threads", input_type=Int(optional=True), prefix="--threads", doc=InputDocumentation(doc="Number of threads (default: 4)")), ToolInput(tag="in_tmpdir", input_type=Boolean(optional=True), prefix="--tmpdir", doc=InputDocumentation(doc="Set the temp directory (overrides java.\nio.tmpdir)")), ToolInput(tag="in_u_jac", input_type=Boolean(optional=True), prefix="--ujac", doc=InputDocumentation(doc="If specified, use junction permuations\nas contigs (Experimental - may use\nexcessive memory and compute times)")), ToolInput(tag="in_und_up", input_type=Boolean(optional=True), prefix="--undup", doc=InputDocumentation(doc="Unset duplicate flag")), ToolInput(tag="in_ws", input_type=Boolean(optional=True), prefix="--ws", doc=InputDocumentation(doc="Processing window size and overlap\n(size,overlap) (default: 400,200)"))], outputs=[ToolOutput(tag="out_cl", output_type=File(optional=True), selector=InputSelector(input_to_select="in_cl", type_hint=File()), doc=OutputDocumentation(doc="Compression level of output bam file\n(s) (default: 5)")), ToolOutput(tag="out_out", output_type=File(optional=True), selector=InputSelector(input_to_select="in_out", type_hint=File()), doc=OutputDocumentation(doc="Required list of output sam or bam file\n(s) separated by comma"))], container="quay.io/biocontainers/abra2:2.23--hc9558a2_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Abra2_V0_1_0().translate("wdl")

