from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, String

Damidseq_Pipeline_V0_1_0 = CommandToolBuilder(tool="damidseq_pipeline", base_command=["damidseq_pipeline"], inputs=[ToolInput(tag="in_bam_files", input_type=Boolean(optional=True), prefix="--bamfiles", doc=InputDocumentation(doc="Only process BAM files")), ToolInput(tag="in_bed_tools_path", input_type=Boolean(optional=True), prefix="--bedtools_path", doc=InputDocumentation(doc="path to BEDTools executable (leave blank if in path)")), ToolInput(tag="in_bins", input_type=Boolean(optional=True), prefix="--bins", doc=InputDocumentation(doc="Width of bins to use for mapping reads\n[Current value: 75]")), ToolInput(tag="in_bowtie", input_type=Boolean(optional=True), prefix="--bowtie", doc=InputDocumentation(doc="Perform bowtie2 alignment [1/0]\n[Current value: 1]")), ToolInput(tag="in_bowtie_two_genome_dir", input_type=Boolean(optional=True), prefix="--bowtie2_genome_dir", doc=InputDocumentation(doc="Directory and basename for bowtie2 .bt2 indices")), ToolInput(tag="in_bowtie_two_path", input_type=Boolean(optional=True), prefix="--bowtie2_path", doc=InputDocumentation(doc="path to bowtie2 executable (leave blank if in path)")), ToolInput(tag="in_dam", input_type=Boolean(optional=True), prefix="--dam", doc=InputDocumentation(doc="Specify file to use as Dam control")), ToolInput(tag="in_extend_reads", input_type=Boolean(optional=True), prefix="--extend_reads", doc=InputDocumentation(doc="Perform read extension [1/0]\n[Current value: 1]")), ToolInput(tag="in_extension_method", input_type=Boolean(optional=True), prefix="--extension_method", doc=InputDocumentation(doc="Read extension method to use; options are:\nfull: Method used by version 1.3 and earlier.  Extends\nall reads to the value set with --len.\ngatc: Default for version 1.4 and above. Extends reads\nto --len or to the next GATC site, whichever is\nshorter.  Using this option increases peak resolution.\n[Current value: gatc]")), ToolInput(tag="in_full_data_files", input_type=Boolean(optional=True), prefix="--full_data_files", doc=InputDocumentation(doc="Output full binned ratio files (not only GATC array)")), ToolInput(tag="in_gatc_frag_file", input_type=Boolean(optional=True), prefix="--gatc_frag_file", doc=InputDocumentation(doc="GFF file containing all instances of the sequence GATC")), ToolInput(tag="in_just_align", input_type=Boolean(optional=True), prefix="--just_align", doc=InputDocumentation(doc="Just align the FASTQ files, generate BAM files, and exit")), ToolInput(tag="in_kde_plot", input_type=Boolean(optional=True), prefix="--kde_plot", doc=InputDocumentation(doc="create an Rplot of the kernel density fit for\nnormalisation (requires R)")), ToolInput(tag="in_keep_sam", input_type=Boolean(optional=True), prefix="--keep_sam", doc=InputDocumentation(doc="Do not delete .sam file")), ToolInput(tag="in_len", input_type=Boolean(optional=True), prefix="--len", doc=InputDocumentation(doc="Length to extend reads to\n[Current value: 300]")), ToolInput(tag="in_load_defaults", input_type=Boolean(optional=True), prefix="--load_defaults", doc=InputDocumentation(doc="Load this saved set of defaults\n(use 'list' to list current saved options)")), ToolInput(tag="in_max_norm_value", input_type=Boolean(optional=True), prefix="--max_norm_value", doc=InputDocumentation(doc="Maximum log2 value to limit normalisation search at\n(default = +5)\n[Current value: 5]")), ToolInput(tag="in_method_subtract", input_type=Boolean(optional=True), prefix="--method_subtract", doc=InputDocumentation(doc="Output values are (Dam_fusion - Dam) instead of\nlog2(Dam_fusion/Dam) (not recommended)")), ToolInput(tag="in_min_norm_value", input_type=Boolean(optional=True), prefix="--min_norm_value", doc=InputDocumentation(doc="Minimum log2 value to limit normalisation search at\n(default = -5)\n[Current value: -5]")), ToolInput(tag="in_no_file_filters", input_type=Boolean(optional=True), prefix="--no_file_filters", doc=InputDocumentation(doc="Do not trim filenames for output")), ToolInput(tag="in_norm_method", input_type=Boolean(optional=True), prefix="--norm_method", doc=InputDocumentation(doc="Normalisation method to use; options are:\nkde: use kernel density estimation of log2 GATC\nfragment ratio (default)\nrpm: use readcounts per million reads (not recommended\nfor most use cases)\n[Current value: kde]")), ToolInput(tag="in_norm_override", input_type=Boolean(optional=True), prefix="--norm_override", doc=InputDocumentation(doc="Normalise by this amount instead")), ToolInput(tag="in_norm_steps", input_type=Boolean(optional=True), prefix="--norm_steps", doc=InputDocumentation(doc="Number of points in normalisation routine (default = 300)\n[Current value: 300]")), ToolInput(tag="in_only_sam", input_type=Boolean(optional=True), prefix="--only_sam", doc=InputDocumentation(doc="Do not generate bam file")), ToolInput(tag="in_out_name", input_type=Boolean(optional=True), prefix="--out_name", doc=InputDocumentation(doc="Use this as the fusion-protein name when saving the final")), ToolInput(tag="in_ps_factor", input_type=Boolean(optional=True), prefix="--ps_factor", doc=InputDocumentation(doc="Value of c in c*(reads/bins) formula for calculating\npseudocounts (default = 10)\n[Current value: 10]")), ToolInput(tag="in_pseudo_counts", input_type=Boolean(optional=True), prefix="--pseudocounts", doc=InputDocumentation(doc="Add this value of psuedocounts instead (default: optimal\nnumber of pseudocounts determined algorithmically)")), ToolInput(tag="in_cutoff_average_q", input_type=Boolean(optional=True), prefix="--q", doc=InputDocumentation(doc="Cutoff average Q score for aligned reads\n[Current value: 30]")), ToolInput(tag="in_q_score_one_max", input_type=Boolean(optional=True), prefix="--qscore1max", doc=InputDocumentation(doc="max decile for normalising from Dam array\n[Current value: 1]")), ToolInput(tag="in_q_score_one_min", input_type=Boolean(optional=True), prefix="--qscore1min", doc=InputDocumentation(doc="min decile for normalising from Dam array\n[Current value: 0.4]")), ToolInput(tag="in_q_score_two_max", input_type=Boolean(optional=True), prefix="--qscore2max", doc=InputDocumentation(doc="max decile for normalising from fusion-protein array\n[Current value: 0.9]")), ToolInput(tag="in_reset_defaults", input_type=Boolean(optional=True), prefix="--reset_defaults", doc=InputDocumentation(doc="Delete user-defined parameters")), ToolInput(tag="in_sam_tools_path", input_type=Boolean(optional=True), prefix="--samtools_path", doc=InputDocumentation(doc="path to samtools executable (leave blank if in path)")), ToolInput(tag="in_save_defaults", input_type=Boolean(optional=True), prefix="--save_defaults", doc=InputDocumentation(doc="Save runtime parameters as default\n(provide a name to differentiate different genomes --\nthese can be loaded with 'load_defaults')")), ToolInput(tag="in_threads", input_type=Boolean(optional=True), prefix="--threads", doc=InputDocumentation(doc="threads for bowtie2 to use\n[Current value: 7]\n")), ToolInput(tag="in_ratio", input_type=String(), position=0, doc=InputDocumentation(doc="--output_format       Output tracks in this format [gff/bedgraph]"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Damidseq_Pipeline_V0_1_0().translate("wdl", allow_empty_container=True)

