from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import String, Directory, Float, Boolean, Int, File

Pathway_Pipeline_Py_V0_1_0 = CommandToolBuilder(tool="pathway_pipeline.py", base_command=["pathway_pipeline.py"], inputs=[ToolInput(tag="in_input", input_type=String(optional=True), prefix="--input", doc=InputDocumentation(doc="Input TSV table of gene family abundances (either the\nunstratified or stratified output of\nmetagenome_pipeline.py).")), ToolInput(tag="in_out_dir", input_type=Directory(optional=True), prefix="--out_dir", doc=InputDocumentation(doc="Output folder for pathway abundance output.")), ToolInput(tag="in_map", input_type=Float(optional=True), prefix="--map", doc=InputDocumentation(doc="Mapping of pathways to reactions. The default mapfile\nmaps MetaCyc reactions to prokaryotic MetaCyc pathways\n(default: /usr/local/lib/python3.6/site-packages/picru\nst2/default_files/pathway_mapfiles/metacyc_path2rxn_st\nruc_filt_pro.txt).")), ToolInput(tag="in_skip_min_path", input_type=Boolean(optional=True), prefix="--skip_minpath", doc=InputDocumentation(doc="Do not run MinPath to identify which pathways are\npresent as a first pass (on by default).")), ToolInput(tag="in_no_gap_fill", input_type=Boolean(optional=True), prefix="--no_gap_fill", doc=InputDocumentation(doc="Do not perform gap filling before predicting pathway\nabundances (Gap filling is on otherwise by default.")), ToolInput(tag="in_intermediate", input_type=Directory(optional=True), prefix="--intermediate", doc=InputDocumentation(doc="Output folder for intermediate files (will be deleted\notherwise).")), ToolInput(tag="in_processes", input_type=Int(optional=True), prefix="--processes", doc=InputDocumentation(doc="Number of processes to run in parallel (default: 1).")), ToolInput(tag="in_no_regroup", input_type=Boolean(optional=True), prefix="--no_regroup", doc=InputDocumentation(doc="Do not regroup input gene families to reactions as\nspecified in the regrouping mapfile.")), ToolInput(tag="in_coverage", input_type=Boolean(optional=True), prefix="--coverage", doc=InputDocumentation(doc="Calculate pathway coverages as well as abundances,\nwhich are experimental and only useful for advanced\nusers.")), ToolInput(tag="in_regroup_map", input_type=Float(optional=True), prefix="--regroup_map", doc=InputDocumentation(doc="Mapfile of ids to regroup gene families to before\nrunning MinPath. The default mapfile is for regrouping\nEC numbers to MetaCyc reactions (default:\n/usr/local/lib/python3.6/site-packages/picrust2/defaul\nt_files/pathway_mapfiles/ec_level4_to_metacyc_rxn.tsv)\n.")), ToolInput(tag="in_per_sequence_contrib", input_type=Boolean(optional=True), prefix="--per_sequence_contrib", doc=InputDocumentation(doc="Flag to specify that MinPath is run on the genes\ncontributed by each sequence (i.e. a predicted genome)\nindividually. Note this will greatly increase the\nruntime. The output will be the predicted pathway\nabundance contributed by each individual sequence.\nThis is in contrast to the default stratified output,\nwhich is the contribution to the community-wide\npathway abundances. Experimental pathway coverage\nstratified by contributing sequence will also be\noutput when --coverage is set. Options\n--per_sequence_contrib and --per_sequence_function\nneed to be set when this option is used (default:\nFalse).")), ToolInput(tag="in_per_sequence_abun", input_type=File(optional=True), prefix="--per_sequence_abun", doc=InputDocumentation(doc="Path to table of sequence abundances across samples\nnormalized by marker copy number (typically the\nnormalized sequence abundance table output at the\nmetagenome pipeline step). This input is required when\nthe --per_sequence_contrib option is set. (default:\nNone).")), ToolInput(tag="in_per_sequence_function", input_type=File(optional=True), prefix="--per_sequence_function", doc=InputDocumentation(doc="Path to table of function abundances per sequence,\nwhich was outputted at the hidden-state prediction\nstep. This input is required when the\n--per_sequence_contrib option is set. Note that this\nfile should be the same input table as used for the\nmetagenome pipeline step (default: None).")), ToolInput(tag="in_wide_table", input_type=Boolean(optional=True), prefix="--wide_table", doc=InputDocumentation(doc="Flag to specify that wide-format stratified table\nshould be output rather than metagenome contribution\ntable. This is the deprecated method of generating\nstratified tables since it is extremely memory\nintensive (default: False).")), ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="--verbose", doc=InputDocumentation(doc="If specified, print out wrapped commands and other\ndetails to screen."))], outputs=[ToolOutput(tag="out_out_dir", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_out_dir", type_hint=File()), doc=OutputDocumentation(doc="Output folder for pathway abundance output.")), ToolOutput(tag="out_intermediate", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_intermediate", type_hint=File()), doc=OutputDocumentation(doc="Output folder for intermediate files (will be deleted\notherwise)."))], container="quay.io/biocontainers/picrust2:2.4.1--py_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Pathway_Pipeline_Py_V0_1_0().translate("wdl")

