version 1.0

task AutoAugpl {
  input {
    File? genome
    File? training_set
    String? species
    String? hints
    String? est_ali
    Boolean? pasa
    File? cdna
    Boolean? pasa_polya_hints
    Boolean? use_existing
    Boolean? verbose
    Boolean? no_utr
    Boolean? working_dir
    Boolean? single_cpu
    Boolean? noninteractive
    String? cname
    Int? index
    Int? opt_rounds
    Int? max_intron_len
    Boolean? use_gmap_for_pasa
  }
  command <<<
    autoAug_pl \
      ~{if defined(genome) then ("--genome " +  '"' + genome + '"') else ""} \
      ~{if defined(training_set) then ("--trainingset " +  '"' + training_set + '"') else ""} \
      ~{if defined(species) then ("--species " +  '"' + species + '"') else ""} \
      ~{if defined(hints) then ("--hints " +  '"' + hints + '"') else ""} \
      ~{if defined(est_ali) then ("--estali " +  '"' + est_ali + '"') else ""} \
      ~{if (pasa) then "--pasa" else ""} \
      ~{if defined(cdna) then ("--cdna " +  '"' + cdna + '"') else ""} \
      ~{if (pasa_polya_hints) then "--pasapolyAhints" else ""} \
      ~{if (use_existing) then "--useexisting" else ""} \
      ~{if (verbose) then "--verbose" else ""} \
      ~{if (no_utr) then "--noutr" else ""} \
      ~{if (working_dir) then "--workingdir" else ""} \
      ~{if (single_cpu) then "--singleCPU" else ""} \
      ~{if (noninteractive) then "--noninteractive" else ""} \
      ~{if defined(cname) then ("--cname " +  '"' + cname + '"') else ""} \
      ~{if defined(index) then ("--index " +  '"' + index + '"') else ""} \
      ~{if defined(opt_rounds) then ("--optrounds " +  '"' + opt_rounds + '"') else ""} \
      ~{if defined(max_intron_len) then ("--maxIntronLen " +  '"' + max_intron_len + '"') else ""} \
      ~{if (use_gmap_for_pasa) then "--useGMAPforPASA" else ""}
  >>>
  runtime {
    docker: "None"
  }
  parameter_meta {
    genome: "fasta file with DNA sequences for training"
    training_set: "genesfile contains training genes in Genbank, GFF or protein FASTA format"
    species: "species name as used by AUGUSTUS"
    hints: "hints for gene predictions with AUGUSTUS"
    est_ali: "cDNA alignments in PSL format (as generated by BLAT and GMAP) are used to construct UTRs"
    pasa: "use PASA to construct a training set"
    cdna: "a fasta file with cDNA sequences (ESTs, mRNA)"
    pasa_polya_hints: "use PASA polyA hints as hints for the prediction"
    use_existing: "use and change the present config and parameter files if they exist for 'species'"
    verbose: "print more status info. Cumulative option, e.g. use -v -v -v to make this script very verbose"
    no_utr: "do not train and predict UTRs."
    working_dir: "=/path/to/wd/           In the working directory results and temporary files are stored.\\nDefault: current working directory"
    single_cpu: "run the complete program sequentially instead of parallel execution of jobs on a cluster"
    noninteractive: "bypass all manual interaction when using a SGE cluster"
    cname: "cluster name, only use it when \\\"noninteractive\\\" default:fe"
    index: "step index, default:0"
    opt_rounds: "optimization rounds - each meta parameter is optimized this often (default 1)"
    max_intron_len: "maximal length of an intron as used by PASA and BLAT, not by AUGUSTUS (default 100000)"
    use_gmap_for_pasa: "use GMAP instead of BLAT in the PASA run"
  }
  output {
    File out_stdout = stdout()
  }
}