from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Int, Boolean, Directory, File, Float, String

R_Scape_V0_1_0 = CommandToolBuilder(tool="R_scape", base_command=["R-scape"], inputs=[ToolInput(tag="in_eval_max_expected", input_type=Int(optional=True), prefix="-E", doc=InputDocumentation(doc=": Eval: max expected number of covNBPs allowed  [0.05]  (x>=0)")), ToolInput(tag="in_test_basepairs_other", input_type=Boolean(optional=True), prefix="-s", doc=InputDocumentation(doc=": two-set test: basepairs / all other pairs. Requires a given structure")), ToolInput(tag="in_structured", input_type=Boolean(optional=True), prefix="--structured", doc=InputDocumentation(doc=": This is a structural RNA of unknown structure")), ToolInput(tag="in_sample_contacts", input_type=Boolean(optional=True), prefix="--samplecontacts", doc=InputDocumentation(doc=": basepair-set sample size is all contacts (default for amino acids)")), ToolInput(tag="in_sample_bp", input_type=Boolean(optional=True), prefix="--samplebp", doc=InputDocumentation(doc=": basepair-set sample size is all 12-type basepairs (default for RNA/DNA)")), ToolInput(tag="in_sample_wc", input_type=Boolean(optional=True), prefix="--samplewc", doc=InputDocumentation(doc=": basepair-set sample size is WWc basepairs only")), ToolInput(tag="in_fold", input_type=Boolean(optional=True), prefix="--fold", doc=InputDocumentation(doc=": obtain the structure with maximum covariation")), ToolInput(tag="in_outdir", input_type=Directory(optional=True), prefix="--outdir", doc=InputDocumentation(doc=": specify a directory for all output files")), ToolInput(tag="in_out_name", input_type=File(optional=True), prefix="--outname", doc=InputDocumentation(doc=": specify a filename for all outputs")), ToolInput(tag="in_rtw_or_all", input_type=Boolean(optional=True), prefix="--r2rall", doc=InputDocumentation(doc=": make R2R plot all position in the alignment")), ToolInput(tag="in__be_verbose", input_type=Boolean(optional=True), prefix="-v", doc=InputDocumentation(doc=": be verbose")), ToolInput(tag="in_window", input_type=Int(optional=True), prefix="--window", doc=InputDocumentation(doc=": window size  (n>0)")), ToolInput(tag="in_slide", input_type=Int(optional=True), prefix="--slide", doc=InputDocumentation(doc=": window slide  [50]  (n>0)")), ToolInput(tag="in_one_msa", input_type=Boolean(optional=True), prefix="--onemsa", doc=InputDocumentation(doc=": if file has more than one msa, analyze only the first one")), ToolInput(tag="in_no_figures", input_type=Boolean(optional=True), prefix="--nofigures", doc=InputDocumentation(doc=": do not produce R2R and dotplot outputs")), ToolInput(tag="in_roc", input_type=Boolean(optional=True), prefix="--roc", doc=InputDocumentation(doc=": write .roc file")), ToolInput(tag="in_filter_xseqcons_residues", input_type=Float(optional=True), prefix="-F", doc=InputDocumentation(doc=": filter out seqs <x*seq_cons residues  (0<x<=1.0)")), ToolInput(tag="in_require_d_", input_type=Float(optional=True), prefix="-I", doc=InputDocumentation(doc=": require seqs to have < <x> id  [1.0]  (0<x<=1.0)")), ToolInput(tag="in_require_seqs_", input_type=Float(optional=True), prefix="-i", doc=InputDocumentation(doc=": require seqs to have >= <x> id  (0<=x<1.0)")), ToolInput(tag="in_t_start", input_type=Int(optional=True), prefix="--tstart", doc=InputDocumentation(doc=": min alignment position to analyze [1..alen]  (n>0)")), ToolInput(tag="in_tend", input_type=Int(optional=True), prefix="--tend", doc=InputDocumentation(doc=": max alignment position to analyze [1..alen]  (n>0)")), ToolInput(tag="in_consensus", input_type=Boolean(optional=True), prefix="--consensus", doc=InputDocumentation(doc=": analyze only consensus (seq_cons) positions")), ToolInput(tag="in_sub_msa", input_type=Int(optional=True), prefix="--submsa", doc=InputDocumentation(doc=": take n random sequences from the alignment, all if NULL  (n>0)")), ToolInput(tag="in_n_seq_min", input_type=Int(optional=True), prefix="--nseqmin", doc=InputDocumentation(doc=": minimum number of sequences in the alignment  (n>0)")), ToolInput(tag="in_gap_thresh", input_type=Float(optional=True), prefix="--gapthresh", doc=InputDocumentation(doc=": keep columns with < <x> fraction of gaps  [0.75]  (0<=x<=1)")), ToolInput(tag="in_mini_d", input_type=Float(optional=True), prefix="--minid", doc=InputDocumentation(doc=": minimum avgid of the given alignment  (0<x<=1.0)")), ToolInput(tag="in_max_id", input_type=Float(optional=True), prefix="--maxid", doc=InputDocumentation(doc=": maximum avgid of the given alignment  (0<x<=1.0)")), ToolInput(tag="in_tree_file", input_type=String(optional=True), prefix="--treefile", doc=InputDocumentation(doc=": provide external tree to use")), ToolInput(tag="in_v_shuffle", input_type=Boolean(optional=True), prefix="--vshuffle", doc=InputDocumentation(doc=": shuffle the residues in a column")), ToolInput(tag="in_c_shuffle", input_type=Boolean(optional=True), prefix="--cshuffle", doc=InputDocumentation(doc=": shuffle the columns of the alignment")), ToolInput(tag="in_cnt_maxd", input_type=Int(optional=True), prefix="--cntmaxD", doc=InputDocumentation(doc=": max distance for contact definition  [8.0]  (x>0)")), ToolInput(tag="in_pdb_file", input_type=File(optional=True), prefix="--pdbfile", doc=InputDocumentation(doc=": read pdb file from file <f>")), ToolInput(tag="in_cnt_mind", input_type=Int(optional=True), prefix="--cntmind", doc=InputDocumentation(doc=": min (j-i+1) for contact definition  [1]  (n>0)")), ToolInput(tag="in_only_pdb", input_type=Boolean(optional=True), prefix="--onlypdb", doc=InputDocumentation(doc=": use only structural info in pdbfile, ignore msa annotation if any")), ToolInput(tag="in_in_format", input_type=String(optional=True), prefix="--informat", doc=InputDocumentation(doc=": specify format")), ToolInput(tag="in_n_shuffle", input_type=Int(optional=True), prefix="--nshuffle", doc=InputDocumentation(doc=": number of shuffled alignments  (n>0)")), ToolInput(tag="in_chia", input_type=Boolean(optional=True), prefix="--CHIa", doc=InputDocumentation(doc=": CHI  ASC corrected statistic")), ToolInput(tag="in_chip", input_type=Boolean(optional=True), prefix="--CHIp", doc=InputDocumentation(doc=": CHI  APC corrected statistic")), ToolInput(tag="in_chi", input_type=Boolean(optional=True), prefix="--CHI", doc=InputDocumentation(doc=": CHI  statistic")), ToolInput(tag="in_gta", input_type=Boolean(optional=True), prefix="--GTa", doc=InputDocumentation(doc=": GT   ASC corrected statistic")), ToolInput(tag="in_gtp", input_type=Boolean(optional=True), prefix="--GTp", doc=InputDocumentation(doc=": GT   APC corrected statistic  [TRUE]")), ToolInput(tag="in_gt", input_type=Boolean(optional=True), prefix="--GT", doc=InputDocumentation(doc=": GT   statistic")), ToolInput(tag="in_mia", input_type=Boolean(optional=True), prefix="--MIa", doc=InputDocumentation(doc=": MI   ASC corrected statistic")), ToolInput(tag="in_mip", input_type=Boolean(optional=True), prefix="--MIp", doc=InputDocumentation(doc=": MI   APC corrected statistic")), ToolInput(tag="in_mi", input_type=Boolean(optional=True), prefix="--MI", doc=InputDocumentation(doc=": MI   statistic")), ToolInput(tag="in_mira", input_type=Boolean(optional=True), prefix="--MIra", doc=InputDocumentation(doc=": MIr  ASC corrected statistic")), ToolInput(tag="in_mir_p", input_type=Boolean(optional=True), prefix="--MIrp", doc=InputDocumentation(doc=": MIr  APC corrected statistic")), ToolInput(tag="in_mir", input_type=Boolean(optional=True), prefix="--MIr", doc=InputDocumentation(doc=": MIr  statistic")), ToolInput(tag="in_miga", input_type=Boolean(optional=True), prefix="--MIga", doc=InputDocumentation(doc=": MIg  ASC corrected statistic")), ToolInput(tag="in_mi_gp", input_type=Boolean(optional=True), prefix="--MIgp", doc=InputDocumentation(doc=": MIg  APC corrected statistic")), ToolInput(tag="in_mig", input_type=Boolean(optional=True), prefix="--MIg", doc=InputDocumentation(doc=": MIg  statistic")), ToolInput(tag="in_o_mesa", input_type=Boolean(optional=True), prefix="--OMESa", doc=InputDocumentation(doc=": OMES ASC corrected statistic")), ToolInput(tag="in_om_esp", input_type=Boolean(optional=True), prefix="--OMESp", doc=InputDocumentation(doc=": OMES APC corrected statistic")), ToolInput(tag="in_omes", input_type=Boolean(optional=True), prefix="--OMES", doc=InputDocumentation(doc=": OMES statistic")), ToolInput(tag="in_rafa", input_type=Boolean(optional=True), prefix="--RAFa", doc=InputDocumentation(doc=": RNAalifold ASC corrected statistic")), ToolInput(tag="in_r_afp", input_type=Boolean(optional=True), prefix="--RAFp", doc=InputDocumentation(doc=": RNAalifold APC corrected statistic")), ToolInput(tag="in_raf", input_type=Boolean(optional=True), prefix="--RAF", doc=InputDocumentation(doc=": RNAalifold statistic")), ToolInput(tag="in_raf_sa", input_type=Boolean(optional=True), prefix="--RAFSa", doc=InputDocumentation(doc=": RNAalifold-stacking ASC corrected statistic")), ToolInput(tag="in_raf_sp", input_type=Boolean(optional=True), prefix="--RAFSp", doc=InputDocumentation(doc=": RNAalifold-stacking APC corrected statistic")), ToolInput(tag="in_raf_s", input_type=Boolean(optional=True), prefix="--RAFS", doc=InputDocumentation(doc=": RNAalifold-stacking  statistic")), ToolInput(tag="in_ccfa", input_type=Boolean(optional=True), prefix="--CCFa", doc=InputDocumentation(doc=": Correlation Coefficient with Frobenius norm ASC corrected statistic")), ToolInput(tag="in_ccfp", input_type=Boolean(optional=True), prefix="--CCFp", doc=InputDocumentation(doc=": Correlation Coefficient with Frobenious norm  APC corrected statistic")), ToolInput(tag="in_ccf", input_type=Boolean(optional=True), prefix="--CCF", doc=InputDocumentation(doc=": Correlation Coefficient with Frobenious norm   statistic")), ToolInput(tag="in_ptf_p", input_type=Boolean(optional=True), prefix="--PTFp", doc=InputDocumentation(doc=": POTTS Frobenious ASC corrected statistic")), ToolInput(tag="in_p_tap", input_type=Boolean(optional=True), prefix="--PTAp", doc=InputDocumentation(doc=": POTTS Averages   ASC corrected statistic")), ToolInput(tag="in_pt_dp", input_type=Boolean(optional=True), prefix="--PTDp", doc=InputDocumentation(doc=": POTTS DI         ASC corrected statistic")), ToolInput(tag="in_cone_six", input_type=Boolean(optional=True), prefix="--C16", doc=InputDocumentation(doc=": use 16 covariation classes")), ToolInput(tag="in_c_two", input_type=Boolean(optional=True), prefix="--C2", doc=InputDocumentation(doc=": use 2 covariation classes")), ToolInput(tag="in_c_select", input_type=Boolean(optional=True), prefix="--CSELECT", doc=InputDocumentation(doc=": use C2 if nseq <= nseqthresh otherwise use C16  [TRUE]")), ToolInput(tag="in_naive", input_type=Boolean(optional=True), prefix="--naive", doc=InputDocumentation(doc=": sort results by cov score, no null model involved")), ToolInput(tag="in_null_phylo", input_type=Boolean(optional=True), prefix="--nullphylo", doc=InputDocumentation(doc=": nullphylo  statistics  [TRUE]")), ToolInput(tag="in_non_param", input_type=Boolean(optional=True), prefix="--nonparam", doc=InputDocumentation(doc=": non parameteric correlate  [TRUE]")), ToolInput(tag="in_potts", input_type=Boolean(optional=True), prefix="--potts", doc=InputDocumentation(doc=": potts couplings")), ToolInput(tag="in_dna", input_type=Boolean(optional=True), prefix="--dna", doc=InputDocumentation(doc=": use DNA alphabet")), ToolInput(tag="in_use_rna_alphabet", input_type=Boolean(optional=True), prefix="--rna", doc=InputDocumentation(doc=": use RNA alphabet")), ToolInput(tag="in_amino", input_type=Boolean(optional=True), prefix="--amino", doc=InputDocumentation(doc=": use protein alphabet")), ToolInput(tag="in_pt_muh", input_type=Float(optional=True), prefix="--ptmuh", doc=InputDocumentation(doc=": potts regularization parameters for training hi's  [0.01]  (x>=0)")), ToolInput(tag="in_pt_mue", input_type=Float(optional=True), prefix="--ptmue", doc=InputDocumentation(doc=": potts regularization parameters for training eij's  [0.20]  (x>=0)")), ToolInput(tag="in_plm", input_type=Boolean(optional=True), prefix="--PLM", doc=InputDocumentation(doc=": potts option for training")), ToolInput(tag="in_a_plm", input_type=Boolean(optional=True), prefix="--APLM", doc=InputDocumentation(doc=": potts option for training")), ToolInput(tag="in_out_potts", input_type=Boolean(optional=True), prefix="--outpotts", doc=InputDocumentation(doc=": write inferred potts parameters")), ToolInput(tag="in_gremlin", input_type=Boolean(optional=True), prefix="--gremlin", doc=InputDocumentation(doc=": reproduce gremlin")), ToolInput(tag="in_out_msa", input_type=Boolean(optional=True), prefix="--outmsa", doc=InputDocumentation(doc=": write actual msa used")), ToolInput(tag="in_out_tree", input_type=Boolean(optional=True), prefix="--outtree", doc=InputDocumentation(doc=": write phylogenetic tree used")), ToolInput(tag="in_out_null", input_type=Boolean(optional=True), prefix="--outnull", doc=InputDocumentation(doc=": write null alignments")), ToolInput(tag="in_all_branch", input_type=Boolean(optional=True), prefix="--allbranch", doc=InputDocumentation(doc=": output msa with all branges")), ToolInput(tag="in_v_output", input_type=Boolean(optional=True), prefix="--voutput", doc=InputDocumentation(doc=": verbose output")), ToolInput(tag="in_power", input_type=String(optional=True), prefix="--power", doc=InputDocumentation(doc=": calculate empirical power curve")), ToolInput(tag="in_double_subs", input_type=Boolean(optional=True), prefix="--doublesubs", doc=InputDocumentation(doc=": calculate power using double substitutions, default is single substitutions")), ToolInput(tag="in_minh_loop", input_type=Int(optional=True), prefix="--minhloop", doc=InputDocumentation(doc=": minimum hairpin loop length. If i-j is the closing pair: minhloop = j-1-1. Default is 0  (n>=0)")), ToolInput(tag="in_seed", input_type=Int(optional=True), prefix="--seed", doc=InputDocumentation(doc=": set RNG seed to <n>. Use 0 for a random seed.  [42]  (n>=0)")), ToolInput(tag="in_pm_ass", input_type=Float(optional=True), prefix="--pmass", doc=InputDocumentation(doc=": pmass for censored histogram of cov scores  [0.0005]  (0<x<=1)")), ToolInput(tag="in_options", input_type=Boolean(optional=True), prefix="-options", doc=InputDocumentation(doc="")), ToolInput(tag="in_var_93", input_type=String(), position=0, doc=InputDocumentation(doc="")), ToolInput(tag="in_structural", input_type=String(), position=1, doc=InputDocumentation(doc="")), ToolInput(tag="in_covariation", input_type=String(), position=2, doc=InputDocumentation(doc="")), ToolInput(tag="in_above", input_type=String(), position=3, doc=InputDocumentation(doc="")), ToolInput(tag="in_phylogenetic", input_type=String(), position=4, doc=InputDocumentation(doc="")), ToolInput(tag="in_expectation", input_type=String(), position=5, doc=InputDocumentation(doc="")), ToolInput(tag="in_msa_file", input_type=String(), position=6, doc=InputDocumentation(doc=""))], outputs=[ToolOutput(tag="out_outdir", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_outdir", type_hint=File()), doc=OutputDocumentation(doc=": specify a directory for all output files"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    R_Scape_V0_1_0().translate("wdl", allow_empty_container=True)

