from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Int, Boolean, String

Mash_Dist_V0_1_0 = CommandToolBuilder(tool="mash_dist", base_command=["mash", "dist"], inputs=[ToolInput(tag="in_parallelism_many_threads", input_type=Int(optional=True), prefix="-p", doc=InputDocumentation(doc="Parallelism. This many threads will be spawned for processing. [1]")), ToolInput(tag="in_list_input_lines", input_type=Boolean(optional=True), prefix="-l", doc=InputDocumentation(doc="List input. Lines in each <query> specify paths to sequence files,\none per line. The reference file is not affected.")), ToolInput(tag="in_table_output_report", input_type=Boolean(optional=True), prefix="-t", doc=InputDocumentation(doc="Table output (will not report p-values, but fields will be blank if\nthey do not meet the p-value threshold).")), ToolInput(tag="in_maximum_pvalue_report", input_type=Int(optional=True), prefix="-v", doc=InputDocumentation(doc="Maximum p-value to report. (0-1) [1.0]")), ToolInput(tag="in_maximum_distance_report", input_type=Int(optional=True), prefix="-d", doc=InputDocumentation(doc="Maximum distance to report. (0-1) [1.0]")), ToolInput(tag="in_show_comment_fields", input_type=Boolean(optional=True), prefix="-C", doc=InputDocumentation(doc="Show comment fields with reference/query names (denoted with ':').\n(0-1) [1.0]")), ToolInput(tag="in_kmer_size_hashes", input_type=Int(optional=True), prefix="-k", doc=InputDocumentation(doc="K-mer size. Hashes will be based on strings of this many\nnucleotides. Canonical nucleotides are used by default (see\nAlphabet options below). (1-32) [21]")), ToolInput(tag="in_sketch_size_sketch", input_type=Int(optional=True), prefix="-s", doc=InputDocumentation(doc="Sketch size. Each sketch will have at most this many non-redundant\nmin-hashes. [1000]")), ToolInput(tag="in_sketch_individual_sequences", input_type=Boolean(optional=True), prefix="-i", doc=InputDocumentation(doc="Sketch individual sequences, rather than whole files, e.g. for\nmulti-fastas of single-chromosome genomes or pair-wise gene\ncomparisons.")), ToolInput(tag="in_seed_provide_hash", input_type=Int(optional=True), prefix="-S", doc=InputDocumentation(doc="Seed to provide to the hash function. (0-4294967296) [42]")), ToolInput(tag="in_probability_threshold_size", input_type=Int(optional=True), prefix="-w", doc=InputDocumentation(doc="Probability threshold for warning about low k-mer size. (0-1)\n[0.01]")), ToolInput(tag="in_input_read_set", input_type=Boolean(optional=True), prefix="-r", doc=InputDocumentation(doc="Input is a read set. See Reads options below. Incompatible with -i.")), ToolInput(tag="in_use_bloom_filter", input_type=Int(optional=True), prefix="-b", doc=InputDocumentation(doc="Use a Bloom filter of this size (raw bytes or with K/M/G/T) to\nfilter out unique k-mers. This is useful if exact filtering with -m\nuses too much memory. However, some unique k-mers may pass\nerroneously, and copies cannot be counted beyond 2. Implies -r.")), ToolInput(tag="in_minimum_copies_kmer", input_type=Int(optional=True), prefix="-m", doc=InputDocumentation(doc="Minimum copies of each k-mer required to pass noise filter for\nreads. Implies -r. [1]")), ToolInput(tag="in_target_coverage_sketching", input_type=Int(optional=True), prefix="-c", doc=InputDocumentation(doc="Target coverage. Sketching will conclude if this coverage is\nreached before the end of the input file (estimated by average\nk-mer multiplicity). Implies -r.")), ToolInput(tag="in_genome_size_raw", input_type=Int(optional=True), prefix="-g", doc=InputDocumentation(doc="Genome size (raw bases or with K/M/G/T). If specified, will be used\nfor p-value calculation instead of an estimated size from k-mer\ncontent. Implies -r.")), ToolInput(tag="in_preserve_strand_default", input_type=Boolean(optional=True), prefix="-n", doc=InputDocumentation(doc="Preserve strand (by default, strand is ignored by using canonical\nDNA k-mers, which are alphabetical minima of forward-reverse\npairs). Implied if an alphabet is specified with -a or -z.")), ToolInput(tag="in_use_acid_az", input_type=Boolean(optional=True), prefix="-a", doc=InputDocumentation(doc="Use amino acid alphabet (A-Z, except BJOUXZ). Implies -n, -k 9.")), ToolInput(tag="in_alphabet_base_hashes", input_type=String(optional=True), prefix="-z", doc=InputDocumentation(doc="Alphabet to base hashes on (case ignored by default; see -Z).\nK-mers with other characters will be ignored. Implies -n.")), ToolInput(tag="in_preserve_case_ignored", input_type=Boolean(optional=True), prefix="-Z", doc=InputDocumentation(doc="Preserve case in k-mers and alphabet (case is ignored by default).\nSequence letters whose case is not in the current alphabet will be\nskipped when sketching.\n")), ToolInput(tag="in_option", input_type=String(), position=0, doc=InputDocumentation(doc="Description (range) [default]"))], outputs=[], container="quay.io/biocontainers/mash:2.3--ha61e061_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Mash_Dist_V0_1_0().translate("wdl")

