from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Int, Boolean, Float

Seqyclean_V0_1_0 = CommandToolBuilder(tool="seqyclean", base_command=["seqyclean"], inputs=[ToolInput(tag="in_turns_vector_trimming", input_type=File(optional=True), prefix="-v", doc=InputDocumentation(doc="- Turns on vector trimming, default=off. <filename> - is a path to a FASTA-file containing vector genomes.")), ToolInput(tag="in_turns_contaminants_screening", input_type=File(optional=True), prefix="-c", doc=InputDocumentation(doc="- Turns on contaminants screening, default=off, <filename> - is a path to a FASTA-file containing contaminant genomes.")), ToolInput(tag="in_common_size_kmer", input_type=Int(optional=True), prefix="-k", doc=InputDocumentation(doc="- Common size of k-mer, default=15")), ToolInput(tag="in_distance_consecutive_kmers", input_type=Boolean(optional=True), prefix="-d", doc=InputDocumentation(doc="- Distance between consecutive k-mers, default=1")), ToolInput(tag="in_kc", input_type=Int(optional=True), prefix="-kc", doc=InputDocumentation(doc="- Size of k-mer used in sampling contaminat genome, default=15")), ToolInput(tag="in_qual", input_type=Int(optional=True), prefix="-qual", doc=InputDocumentation(doc="<max_error_at_ends> - Turns on quality trimming, default=off. Error boundaries: max_average_error (default=0.01), max_error_at_ends (default=0.01)")), ToolInput(tag="in_bracket", input_type=Int(optional=True), prefix="-bracket", doc=InputDocumentation(doc="<max_avg_error> - Bracket window_size (default=0.794) and maximum_average_error (default=0.794) for quality trimming")), ToolInput(tag="in_window", input_type=Int(optional=True), prefix="-window", doc=InputDocumentation(doc="max_avg_error [window_size max_avg_error ...] - Parameters for window trimming. There are two windows with size of 50 and 10bp and max_avg_err of 0.794 by default.")), ToolInput(tag="in_ow", input_type=Boolean(optional=True), prefix="-ow", doc=InputDocumentation(doc="- Overwrite existing results, default=off")), ToolInput(tag="in_min_len", input_type=Int(optional=True), prefix="-minlen", doc=InputDocumentation(doc="- Minimum length of read to accept, default=50 bp.")), ToolInput(tag="in_poly_at", input_type=Boolean(optional=True), prefix="-polyat", doc=InputDocumentation(doc="[cdna] [cerr] [crng] - Turns on poly A/T trimming, default=off. Parameters: cdna (default=10) - maximum size of a poly tail, cerr (default=3) - maximum number of G/C nucleotides within a tail, cnrg (default=50) - range to look for a tail within a read.")), ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="-verbose", doc=InputDocumentation(doc="- Verbose output, default=off.")), ToolInput(tag="in_det_rep", input_type=Boolean(optional=True), prefix="-detrep", doc=InputDocumentation(doc="- Generate detailed report for each read, default=off.")), ToolInput(tag="in_dup", input_type=Boolean(optional=True), prefix="-dup", doc=InputDocumentation(doc="[-startdw 10][-sizedw 35] [-maxdup 3] - Turns on screening duplicated sequences, default=off. Here: -startdw (defalt=10) and -sizedw (default=25) are starting position and size of the window within a read, -maxdup (default=3) - maximum number of duplicated sequences allowed.")), ToolInput(tag="in_no_adapter_trim", input_type=Boolean(optional=True), prefix="-no_adapter_trim", doc=InputDocumentation(doc="- Turns off trimming of adapters, default=off.")), ToolInput(tag="in_number_yet_applicable", input_type=Int(optional=True), prefix="-t", doc=InputDocumentation(doc="- Number of threads (not yet applicable to Illumina mode), default=4.")), ToolInput(tag="in_fast_q", input_type=Boolean(optional=True), prefix="-fastq", doc=InputDocumentation(doc="- Output in FASTQ format, default=off.")), ToolInput(tag="in_fast_a_out", input_type=Boolean(optional=True), prefix="-fasta_out", doc=InputDocumentation(doc="- Output in FASTA format, default=off.")), ToolInput(tag="in_using_custom_filename", input_type=File(optional=True), prefix="-m", doc=InputDocumentation(doc="- Using custom barcodes, default=off. <filename> - a path to a FASTA-file with custom barcodes.")), ToolInput(tag="in_pairedend_mode_see", input_type=File(optional=True), prefix="-1", doc=InputDocumentation(doc="- Paired-end mode (see examples below)")), ToolInput(tag="in__singleend_mode", input_type=File(optional=True), prefix="-U", doc=InputDocumentation(doc="- Single-end mode")), ToolInput(tag="in_shuffle", input_type=Boolean(optional=True), prefix="-shuffle", doc=InputDocumentation(doc="- Store non-paired Illumina reads in shuffled file, default=off.")), ToolInput(tag="in_i_six_four", input_type=Boolean(optional=True), prefix="-i64", doc=InputDocumentation(doc="- Turns on 64-quality base, default = off.")), ToolInput(tag="in_adp", input_type=File(optional=True), prefix="-adp", doc=InputDocumentation(doc="- Turns on using custom adapters, default=off. <filename> - FASTA file with adapters")), ToolInput(tag="in_at", input_type=Float(optional=True), prefix="-at", doc=InputDocumentation(doc="- This option sets the similarity threshold for adapter trimming by overlap (only in paired-end mode). By default its value is set to 0.75.")), ToolInput(tag="in_overlap", input_type=Int(optional=True), prefix="-overlap", doc=InputDocumentation(doc="- This option turns on merging overlapping paired-end reads and <value> is the minimum overlap length. By default the minimum overlap length is 16 base pairs.")), ToolInput(tag="in_new_two_old", input_type=Boolean(optional=True), prefix="-new2old", doc=InputDocumentation(doc="- Switch to fix read IDs, default=off ( As is detailed in: http://contig.wordpress.com/2011/09/01/newbler-input-iii-a-quick-fix-for-the-new-illumina-fastq-header/#more-342 ).")), ToolInput(tag="in_gz", input_type=Boolean(optional=True), prefix="-gz", doc=InputDocumentation(doc="- compressed output (GZip format, .gz).")), ToolInput(tag="in_alen", input_type=Boolean(optional=True), prefix="-alen", doc=InputDocumentation(doc="- Maximum adapter length, default=30 bp.(only for paired-end mode)"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Seqyclean_V0_1_0().translate("wdl", allow_empty_container=True)

