from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Int, String, Boolean, File, Float

Cutesv_V0_1_0 = CommandToolBuilder(tool="cuteSV", base_command=["cuteSV"], inputs=[ToolInput(tag="in_threads", input_type=Int(optional=True), prefix="--threads", doc=InputDocumentation(doc="Number of threads to use.[16]")), ToolInput(tag="in_batches", input_type=Int(optional=True), prefix="--batches", doc=InputDocumentation(doc="Batch of genome segmentation interval.[10000000]")), ToolInput(tag="in_sample", input_type=String(optional=True), prefix="--sample", doc=InputDocumentation(doc="Sample name/id")), ToolInput(tag="in_retain_work_dir", input_type=Boolean(optional=True), prefix="--retain_work_dir", doc=InputDocumentation(doc="Enable to retain temporary folder and files.")), ToolInput(tag="in_report_read_id", input_type=Boolean(optional=True), prefix="--report_readid", doc=InputDocumentation(doc="Enable to report supporting read ids for each SV.")), ToolInput(tag="in_max_split_parts", input_type=Int(optional=True), prefix="--max_split_parts", doc=InputDocumentation(doc="Maximum number of split segments a read may be aligned\nbefore it is ignored.[7]")), ToolInput(tag="in_min_mapq", input_type=Int(optional=True), prefix="--min_mapq", doc=InputDocumentation(doc="Minimum mapping quality value of alignment to be taken\ninto account.[20]")), ToolInput(tag="in_min_read_len", input_type=Int(optional=True), prefix="--min_read_len", doc=InputDocumentation(doc="Ignores reads that only report alignments with not\nlonger than bp.[500]")), ToolInput(tag="in_merge_del_threshold", input_type=Int(optional=True), prefix="--merge_del_threshold", doc=InputDocumentation(doc="Maximum distance of deletion signals to be merged. In\nour paper, I used -md 500 to process HG002 real human\nsample data.[0]")), ToolInput(tag="in_merge_ins_threshold", input_type=Int(optional=True), prefix="--merge_ins_threshold", doc=InputDocumentation(doc="Maximum distance of insertion signals to be merged. In\nour paper, I used -mi 500 to process HG002 real human\nsample data.[100]")), ToolInput(tag="in_min_support", input_type=Int(optional=True), prefix="--min_support", doc=InputDocumentation(doc="Minimum number of reads that support a SV to be\nreported.[10]")), ToolInput(tag="in_min_size", input_type=Int(optional=True), prefix="--min_size", doc=InputDocumentation(doc="Minimum size of SV to be reported.[30]")), ToolInput(tag="in_max_size", input_type=Int(optional=True), prefix="--max_size", doc=InputDocumentation(doc="Maximum size of SV to be reported.[100000]")), ToolInput(tag="in_min_sig_length", input_type=Int(optional=True), prefix="--min_siglength", doc=InputDocumentation(doc="Minimum length of SV signal to be extracted.[10]")), ToolInput(tag="in_genotype", input_type=Boolean(optional=True), prefix="--genotype", doc=InputDocumentation(doc="Enable to generate genotypes.")), ToolInput(tag="in_gt_round", input_type=Int(optional=True), prefix="--gt_round", doc=InputDocumentation(doc="Maximum round of iteration for alignments searching if\nperform genotyping.[500]")), ToolInput(tag="in_ivcf", input_type=File(optional=True), prefix="-Ivcf", doc=InputDocumentation(doc="Optional given vcf file. Enable to perform force\ncalling. [NULL]")), ToolInput(tag="in_max_cluster_bias_ins", input_type=Int(optional=True), prefix="--max_cluster_bias_INS", doc=InputDocumentation(doc="Maximum distance to cluster read together for\ninsertion.[100]")), ToolInput(tag="in_diff_ratio_merging_ins", input_type=Float(optional=True), prefix="--diff_ratio_merging_INS", doc=InputDocumentation(doc="Do not merge breakpoints with basepair identity more\nthan [0.3] for insertion.")), ToolInput(tag="in_max_cluster_bias_del", input_type=Int(optional=True), prefix="--max_cluster_bias_DEL", doc=InputDocumentation(doc="Maximum distance to cluster read together for\ndeletion.[200]")), ToolInput(tag="in_diff_ratio_merging_del", input_type=Float(optional=True), prefix="--diff_ratio_merging_DEL", doc=InputDocumentation(doc="Do not merge breakpoints with basepair identity more\nthan [0.5] for deletion.")), ToolInput(tag="in_max_cluster_bias_in_v", input_type=Int(optional=True), prefix="--max_cluster_bias_INV", doc=InputDocumentation(doc="Maximum distance to cluster read together for\ninversion.[500]")), ToolInput(tag="in_max_cluster_bias_dup", input_type=Int(optional=True), prefix="--max_cluster_bias_DUP", doc=InputDocumentation(doc="Maximum distance to cluster read together for\nduplication.[500]")), ToolInput(tag="in_max_cluster_bias_tra", input_type=Int(optional=True), prefix="--max_cluster_bias_TRA", doc=InputDocumentation(doc="Maximum distance to cluster read together for\ntranslocation.[50]")), ToolInput(tag="in_diff_ratio_filtering_tra", input_type=Float(optional=True), prefix="--diff_ratio_filtering_TRA", doc=InputDocumentation(doc="Filter breakpoints with basepair identity less than\n[0.6] for translocation.\n")), ToolInput(tag="in_reference", input_type=String(), position=0, doc=InputDocumentation(doc="The reference genome in fasta format.")), ToolInput(tag="in_output", input_type=String(), position=1, doc=InputDocumentation(doc="Output VCF format file.")), ToolInput(tag="in_work_dir", input_type=String(), position=2, doc=InputDocumentation(doc="Work-directory for distributed jobs"))], outputs=[], container="quay.io/biocontainers/cutesv:1.0.10--py_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Cutesv_V0_1_0().translate("wdl")

