from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import String, Int

Wgdi_V0_1_0 = CommandToolBuilder(tool="wgdi", base_command=["wgdi"], inputs=[ToolInput(tag="in_show_homologous_dotplot", input_type=String(optional=True), prefix="-d", doc=InputDocumentation(doc="Show homologous gene dotplot")), ToolInput(tag="in_icl", input_type=String(optional=True), prefix="-icl", doc=InputDocumentation(doc="Improved version of ColinearScan")), ToolInput(tag="in_ks", input_type=Int(optional=True), prefix="-ks", doc=InputDocumentation(doc="Calculate Ka/Ks for homologous gene pairs by YN00")), ToolInput(tag="in_bk", input_type=String(optional=True), prefix="-bk", doc=InputDocumentation(doc="Show Ks of blocks in a dotplot")), ToolInput(tag="in_bi", input_type=String(optional=True), prefix="-bi", doc=InputDocumentation(doc="Collinearity and Ks speculate whole genome duplication")), ToolInput(tag="in_extract_eventrelated_genomic", input_type=String(optional=True), prefix="-c", doc=InputDocumentation(doc="Extract event-related genomic alignment")), ToolInput(tag="in_kp", input_type=String(optional=True), prefix="-kp", doc=InputDocumentation(doc="A simple way to get ks peaks")), ToolInput(tag="in_kf", input_type=String(optional=True), prefix="-kf", doc=InputDocumentation(doc="A simple way to draw ks distribution map")), ToolInput(tag="in_pf", input_type=String(optional=True), prefix="-pf", doc=InputDocumentation(doc="Gaussian fitting of ks distribution")), ToolInput(tag="in_pc", input_type=String(optional=True), prefix="-pc", doc=InputDocumentation(doc="Polyploid distinguish among subgenomes")), ToolInput(tag="in_show_eventrelated_alignment", input_type=String(optional=True), prefix="-a", doc=InputDocumentation(doc="Show event-related genomic alignment in a dotplot")), ToolInput(tag="in_at", input_type=String(optional=True), prefix="-at", doc=InputDocumentation(doc="Collinear genes construct phylogenetic trees")), ToolInput(tag="in_polyploidyindex_characterize_degree", input_type=String(optional=True), prefix="-p", doc=InputDocumentation(doc="Polyploidy-index characterize the degree of divergence\nbetween subgenomes of a polyploidy")), ToolInput(tag="in_show_subgenomes_genome", input_type=String(optional=True), prefix="-r", doc=InputDocumentation(doc="Show subgenomes in gene retention or genome")), ToolInput(tag="in_conf", input_type=String(optional=True), prefix="-conf", doc=InputDocumentation(doc="Display and modify the environment variable")), ToolInput(tag="in_fractionation", input_type=String(), position=0, doc=InputDocumentation(doc="-ci CIRCOS            A simple way to run circos"))], outputs=[], container="quay.io/biocontainers/wgdi:0.4.6--pyh7b7c402_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Wgdi_V0_1_0().translate("wdl")

