from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import String, File, Boolean, Int

Bis_Snp_V0_1_0 = CommandToolBuilder(tool="bis_snp", base_command=["bis-snp"], inputs=[ToolInput(tag="in_analysis_type", input_type=String(optional=True), prefix="--analysis_type", doc=InputDocumentation(doc="Type of analysis to run")), ToolInput(tag="in_input_file", input_type=File(optional=True), prefix="--input_file", doc=InputDocumentation(doc="Input file containing sequence\ndata (BAM or CRAM)")), ToolInput(tag="in_show_full_bam_list", input_type=Boolean(optional=True), prefix="--showFullBamList", doc=InputDocumentation(doc="Emit list of input BAM/CRAM\nfiles to log")), ToolInput(tag="in_read_buffer_size", input_type=Int(optional=True), prefix="--read_buffer_size", doc=InputDocumentation(doc="Number of reads per SAM file\nto buffer in memory")), ToolInput(tag="in_read_filter", input_type=String(optional=True), prefix="--read_filter", doc=InputDocumentation(doc="Filters to apply to reads\nbefore analysis")), ToolInput(tag="in_disable_read_filter", input_type=String(optional=True), prefix="--disable_read_filter", doc=InputDocumentation(doc="Read filters to disable")), ToolInput(tag="in_intervals", input_type=String(optional=True), prefix="--intervals", doc=InputDocumentation(doc="One or more genomic intervals\nover which to operate")), ToolInput(tag="in_exclude_intervals", input_type=String(optional=True), prefix="--excludeIntervals", doc=InputDocumentation(doc="One or more genomic intervals\nto exclude from processing")), ToolInput(tag="in_interval_set_rule", input_type=String(optional=True), prefix="--interval_set_rule", doc=InputDocumentation(doc="Set merging approach to use\nfor combining interval inputs\n(UNION|INTERSECTION)")), ToolInput(tag="in_interval_merging", input_type=String(optional=True), prefix="--interval_merging", doc=InputDocumentation(doc="Interval merging rule for\nabutting intervals (ALL|\nOVERLAPPING_ONLY)")), ToolInput(tag="in_interval_padding", input_type=String(optional=True), prefix="--interval_padding", doc=InputDocumentation(doc="Amount of padding (in bp) to\nadd to each interval")), ToolInput(tag="in_reference_sequence", input_type=File(optional=True), prefix="--reference_sequence", doc=InputDocumentation(doc="Reference sequence file")), ToolInput(tag="in_nondeterministic_random_seed", input_type=Boolean(optional=True), prefix="--nonDeterministicRandomSeed", doc=InputDocumentation(doc="Use a non-deterministic random")), ToolInput(tag="in_max_runtime_units", input_type=Int(optional=True), prefix="--maxRuntimeUnits", doc=InputDocumentation(doc="Unit of time used by\nmaxRuntime (NANOSECONDS|\nMICROSECONDS|MILLISECONDS|\nSECONDS|MINUTES|HOURS|DAYS)")), ToolInput(tag="in_downsampling_type", input_type=String(optional=True), prefix="--downsampling_type", doc=InputDocumentation(doc="Type of read downsampling to\nemploy at a given locus (NONE|\nALL_READS|BY_SAMPLE)")), ToolInput(tag="in_down_sample_to_fraction", input_type=String(optional=True), prefix="--downsample_to_fraction", doc=InputDocumentation(doc="Fraction of reads to\ndownsample to")), ToolInput(tag="in_down_sample_to_coverage", input_type=String(optional=True), prefix="--downsample_to_coverage", doc=InputDocumentation(doc="Target coverage threshold for\ndownsampling to coverage")), ToolInput(tag="in_baq", input_type=String(optional=True), prefix="--baq", doc=InputDocumentation(doc="Type of BAQ calculation to\napply in the engine (OFF|\nCALCULATE_AS_NECESSARY|\nRECALCULATE)")), ToolInput(tag="in_baq_gap_open_penalty", input_type=String(optional=True), prefix="--baqGapOpenPenalty", doc=InputDocumentation(doc="BAQ gap open penalty")), ToolInput(tag="in_refactor_ndn_cigar_string", input_type=Boolean(optional=True), prefix="--refactor_NDN_cigar_string", doc=InputDocumentation(doc="Reduce NDN elements in CIGAR")), ToolInput(tag="in_use_original_qualities", input_type=Boolean(optional=True), prefix="--useOriginalQualities", doc=InputDocumentation(doc="Use the base quality scores\nfrom the OQ tag")), ToolInput(tag="in_default_base_qualities", input_type=String(optional=True), prefix="--defaultBaseQualities", doc=InputDocumentation(doc="Assign a default base quality")), ToolInput(tag="in_performance_log", input_type=File(optional=True), prefix="--performanceLog", doc=InputDocumentation(doc="Write GATK runtime performance\nlog to this file")), ToolInput(tag="in_b_qsr", input_type=File(optional=True), prefix="--BQSR", doc=InputDocumentation(doc="Input covariates table file\nfor on-the-fly base quality\nscore recalibration")), ToolInput(tag="in_quantize_quals", input_type=Int(optional=True), prefix="--quantize_quals", doc=InputDocumentation(doc="Quantize quality scores to a\ngiven number of levels (with\n-BQSR)")), ToolInput(tag="in_static_quantized_quals", input_type=Int(optional=True), prefix="--static_quantized_quals", doc=InputDocumentation(doc="Use static quantized quality\nscores to a given number of\nlevels (with -BQSR)")), ToolInput(tag="in_disable_in_del_quals", input_type=Boolean(optional=True), prefix="--disable_indel_quals", doc=InputDocumentation(doc="Disable printing of base\ninsertion and deletion tags\n(with -BQSR)")), ToolInput(tag="in_emit_original_quals", input_type=Boolean(optional=True), prefix="--emit_original_quals", doc=InputDocumentation(doc="Emit the OQ tag with the\noriginal base qualities (with\n-BQSR)")), ToolInput(tag="in_preserve_q_scores_less_than", input_type=String(optional=True), prefix="--preserve_qscores_less_than", doc=InputDocumentation(doc="Don't recalibrate bases with\nquality scores less than this\nthreshold (with -BQSR)")), ToolInput(tag="in_global_q_score_prior", input_type=String(optional=True), prefix="--globalQScorePrior", doc=InputDocumentation(doc="Global Qscore Bayesian prior\nto use for BQSR")), ToolInput(tag="in_seconds_between_progress_updates", input_type=Boolean(optional=True), prefix="--secondsBetweenProgressUpdates", doc=InputDocumentation(doc="Time interval for process")), ToolInput(tag="in_validation_strictness", input_type=String(optional=True), prefix="--validation_strictness", doc=InputDocumentation(doc="How strict should we be with\nvalidation (STRICT|LENIENT|\nSILENT)")), ToolInput(tag="in_remove_program_records", input_type=Boolean(optional=True), prefix="--remove_program_records", doc=InputDocumentation(doc="Remove program records from\nthe SAM header")), ToolInput(tag="in_keep_program_records", input_type=Boolean(optional=True), prefix="--keep_program_records", doc=InputDocumentation(doc="Keep program records in the\nSAM header")), ToolInput(tag="in_sample_rename_mapping_file", input_type=File(optional=True), prefix="--sample_rename_mapping_file", doc=InputDocumentation(doc="Rename sample IDs on-the-fly\nat runtime using the provided\nmapping file")), ToolInput(tag="in_unsafe", input_type=String(optional=True), prefix="--unsafe", doc=InputDocumentation(doc="Enable unsafe operations:\nnothing will be checked at\nruntime (ALLOW_N_CIGAR_READS|\nALLOW_UNINDEXED_BAM|\nALLOW_UNSET_BAM_SORT_ORDER|\nNO_READ_ORDER_VERIFICATION|\nALLOW_SEQ_DICT_INCOMPATIBILITY|\nLENIENT_VCF_PROCESSING|ALL)")), ToolInput(tag="in_use_jdk_def_later", input_type=Boolean(optional=True), prefix="--use_jdk_deflater", doc=InputDocumentation(doc="Use the JDK Deflater instead\nof the IntelDeflater for\nwriting BAMs")), ToolInput(tag="in_use_jdk_inf_later", input_type=Boolean(optional=True), prefix="--use_jdk_inflater", doc=InputDocumentation(doc="Use the JDK Inflater instead\nof the IntelInflater for\nreading BAMs")), ToolInput(tag="in_sites_only", input_type=Boolean(optional=True), prefix="--sites_only", doc=InputDocumentation(doc="Output sites-only VCF")), ToolInput(tag="in_never_trim_vcf_format_field", input_type=Boolean(optional=True), prefix="--never_trim_vcf_format_field", doc=InputDocumentation(doc="Always output all the records\nin VCF FORMAT fields, even if\nsome are missing")), ToolInput(tag="in_bam_compression", input_type=Int(optional=True), prefix="--bam_compression", doc=InputDocumentation(doc="Compression level to use for\nwriting BAM files (0 - 9,\nhigher is more compressed)")), ToolInput(tag="in_simplify_bam", input_type=Boolean(optional=True), prefix="--simplifyBAM", doc=InputDocumentation(doc="Strip down read content and")), ToolInput(tag="in_num_threads", input_type=Int(optional=True), prefix="--num_threads", doc=InputDocumentation(doc="Number of data threads to\nallocate to this analysis")), ToolInput(tag="in_num_cpu_threads_per_data_thread", input_type=Int(optional=True), prefix="--num_cpu_threads_per_data_thread", doc=InputDocumentation(doc="Number of CPU threads to\nallocate per data thread")), ToolInput(tag="in_monitor_thread_efficiency", input_type=Boolean(optional=True), prefix="--monitorThreadEfficiency", doc=InputDocumentation(doc="Enable threading efficiency")), ToolInput(tag="in_pedigree_string", input_type=String(optional=True), prefix="--pedigreeString", doc=InputDocumentation(doc="Pedigree string for samples")), ToolInput(tag="in_pedigree_validation_type", input_type=String(optional=True), prefix="--pedigreeValidationType", doc=InputDocumentation(doc="Validation strictness for\npedigree (STRICT|SILENT)")), ToolInput(tag="in_variant_index_type", input_type=String(optional=True), prefix="--variant_index_type", doc=InputDocumentation(doc="Type of IndexCreator to use\nfor VCF/BCF indices\n(DYNAMIC_SEEK|DYNAMIC_SIZE|\nLINEAR|INTERVAL)")), ToolInput(tag="in_variant_index_parameter", input_type=String(optional=True), prefix="--variant_index_parameter", doc=InputDocumentation(doc="Parameter to pass to the\nVCF/BCF IndexCreator")), ToolInput(tag="in_reference_window_stop", input_type=String(optional=True), prefix="--reference_window_stop", doc=InputDocumentation(doc="Reference window stop")), ToolInput(tag="in_logging_level", input_type=String(optional=True), prefix="--logging_level", doc=InputDocumentation(doc="Set the minimum level of")), ToolInput(tag="in_max_runtime", input_type=Int(optional=True), prefix="-maxRuntime", doc=InputDocumentation(doc="")), ToolInput(tag="in_jar", input_type=Int(optional=True), prefix="-jar", doc=InputDocumentation(doc="")), ToolInput(tag="in_seed", input_type=String(), position=0, doc=InputDocumentation(doc="-maxRuntime,--maxRuntime <maxRuntime>                                                    Stop execution cleanly as soon ")), ToolInput(tag="in_string", input_type=String(), position=0, doc=InputDocumentation(doc="-fixMisencodedQuals,--fix_misencoded_quality_scores                                      Fix mis-encoded base quality ")), ToolInput(tag="in_scores", input_type=String(), position=1, doc=InputDocumentation(doc="-allowPotentiallyMisencodedQuals,--allow_potentially_misencoded_quality_scores           Ignore warnings about base ")), ToolInput(tag="in_locking", input_type=String(), position=0, doc=InputDocumentation(doc="-no_cmdline_in_header,--no_cmdline_in_header                                             Don't include the command line ")), ToolInput(tag="in__disablebamindexing_turn", input_type=String(), position=0, doc=InputDocumentation(doc="--disable_bam_indexing                                                                   Turn off on-the-fly creation ")), ToolInput(tag="in_files", input_type=String(), position=0, doc=InputDocumentation(doc="--generate_md5                                                                           Enable on-the-fly creation of ")), ToolInput(tag="in_monitoring", input_type=String(), position=0, doc=InputDocumentation(doc="-rgbl,--read_group_black_list <read_group_black_list>                                    Exclude read groups based on ")), ToolInput(tag="in_pedpedigree_pedigree_files", input_type=String(), position=1, doc=InputDocumentation(doc="-ped,--pedigree <pedigree>                                                               Pedigree files for samples")), ToolInput(tag="in_logging", input_type=String(), position=2, doc=InputDocumentation(doc="-log,--log_to_file <log_to_file>                                                         Set the logging location"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Bis_Snp_V0_1_0().translate("wdl", allow_empty_container=True)

