from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Directory, Boolean, File, String

Vamb_V0_1_0 = CommandToolBuilder(tool="vamb", base_command=["vamb"], inputs=[ToolInput(tag="in_output_directory_create", input_type=Directory(optional=True), prefix="--outdir", doc=InputDocumentation(doc="output directory to create")), ToolInput(tag="in_fast_a", input_type=Boolean(optional=True), prefix="--fasta", doc=InputDocumentation(doc="path to fasta file")), ToolInput(tag="in_t_nfs", input_type=Boolean(optional=True), prefix="--tnfs", doc=InputDocumentation(doc="path to .npz of TNF")), ToolInput(tag="in_names", input_type=Boolean(optional=True), prefix="--names", doc=InputDocumentation(doc="path to .npz of names of sequences")), ToolInput(tag="in_lengths", input_type=Boolean(optional=True), prefix="--lengths", doc=InputDocumentation(doc="path to .npz of seq lengths")), ToolInput(tag="in_bam_files", input_type=Boolean(optional=True), prefix="--bamfiles", doc=InputDocumentation(doc="[ ...]  paths to (multiple) BAM files")), ToolInput(tag="in_rp_km", input_type=Boolean(optional=True), prefix="--rpkm", doc=InputDocumentation(doc="path to .npz of RPKM")), ToolInput(tag="in_jgi", input_type=File(optional=True), prefix="--jgi", doc=InputDocumentation(doc="path to output of jgi_summarize_bam_contig_depths")), ToolInput(tag="in_ignore_contigs_shorter", input_type=Boolean(optional=True), prefix="-m", doc=InputDocumentation(doc="ignore contigs shorter than this [100]")), ToolInput(tag="in_ignore_reads_alignment", input_type=Boolean(optional=True), prefix="-s", doc=InputDocumentation(doc="ignore reads with alignment score below this [None]")), ToolInput(tag="in_ignore_reads_nucleotide", input_type=Boolean(optional=True), prefix="-z", doc=InputDocumentation(doc="ignore reads with nucleotide identity below this [None]")), ToolInput(tag="in_number_subprocesses_spawn", input_type=Boolean(optional=True), prefix="-p", doc=InputDocumentation(doc="number of subprocesses to spawn [min(8, nbamfiles)]")), ToolInput(tag="in_no_ref_check", input_type=Boolean(optional=True), prefix="--norefcheck", doc=InputDocumentation(doc="skip reference name hashing check [False]")), ToolInput(tag="in_min_fast_a", input_type=Boolean(optional=True), prefix="--minfasta", doc=InputDocumentation(doc="minimum bin size to output as fasta [None = no files]")), ToolInput(tag="in_n", input_type=Boolean(optional=True), prefix="-n", doc=InputDocumentation(doc="[ ...]          hidden neurons [Auto]")), ToolInput(tag="in_latent_neurons", input_type=Boolean(optional=True), prefix="-l", doc=InputDocumentation(doc="latent neurons [32]")), ToolInput(tag="in_alpha_weight_tnf", input_type=Boolean(optional=True), prefix="-a", doc=InputDocumentation(doc="alpha, weight of TNF versus depth loss [Auto]")), ToolInput(tag="in_beta_capacity_learn", input_type=Boolean(optional=True), prefix="-b", doc=InputDocumentation(doc="beta, capacity to learn [200.0]")), ToolInput(tag="in_dropout", input_type=Boolean(optional=True), prefix="-d", doc=InputDocumentation(doc="dropout [Auto]")), ToolInput(tag="in_cuda", input_type=Boolean(optional=True), prefix="--cuda", doc=InputDocumentation(doc="use GPU to train & cluster [False]")), ToolInput(tag="in_epochs", input_type=Boolean(optional=True), prefix="-e", doc=InputDocumentation(doc="epochs [500]")), ToolInput(tag="in_starting_batch_size", input_type=Boolean(optional=True), prefix="-t", doc=InputDocumentation(doc="starting batch size [64]")), ToolInput(tag="in_q", input_type=Boolean(optional=True), prefix="-q", doc=InputDocumentation(doc="[ [ ...]]        double batch size at epochs [25 75 150 300]")), ToolInput(tag="in_learning_rate", input_type=Boolean(optional=True), prefix="-r", doc=InputDocumentation(doc="learning rate [0.001]")), ToolInput(tag="in_size_count_successes", input_type=Boolean(optional=True), prefix="-w", doc=InputDocumentation(doc="size of window to count successes [200]")), ToolInput(tag="in_minimum_success_window", input_type=Boolean(optional=True), prefix="-u", doc=InputDocumentation(doc="minimum success in window [20]")), ToolInput(tag="in_minimum_cluster_size", input_type=Boolean(optional=True), prefix="-i", doc=InputDocumentation(doc="minimum cluster size [1]")), ToolInput(tag="in_stop_after_clusters", input_type=Boolean(optional=True), prefix="-c", doc=InputDocumentation(doc="stop after c clusters [None = infinite]")), ToolInput(tag="in_binsplit_separator", input_type=Boolean(optional=True), prefix="-o", doc=InputDocumentation(doc="binsplit separator [None = no split]")), ToolInput(tag="in_var_29", input_type=String(), position=0, doc=InputDocumentation(doc="")), ToolInput(tag="in_tnf_input", input_type=String(), position=1, doc=InputDocumentation(doc="")), ToolInput(tag="in_rp_km_input", input_type=String(), position=2, doc=InputDocumentation(doc=""))], outputs=[ToolOutput(tag="out_output_directory_create", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_output_directory_create", type_hint=File()), doc=OutputDocumentation(doc="output directory to create")), ToolOutput(tag="out_jgi", output_type=File(optional=True), selector=InputSelector(input_to_select="in_jgi", type_hint=File()), doc=OutputDocumentation(doc="path to output of jgi_summarize_bam_contig_depths"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Vamb_V0_1_0().translate("wdl", allow_empty_container=True)

