from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, Directory, File, String, Int

Gtftk_Ologram_V0_1_0 = CommandToolBuilder(tool="gtftk_ologram", base_command=["gtftk", "ologram"], inputs=[ToolInput(tag="in_more_bed", input_type=Boolean(optional=True), prefix="--more-bed", doc=InputDocumentation(doc="A list of bed files to be considered as additional genomic annotations. (default: None)")), ToolInput(tag="in_input_file", input_type=Boolean(optional=True), prefix="--inputfile", doc=InputDocumentation(doc="Path to the GTF file. Defaults to STDIN (default: <stdin>)")), ToolInput(tag="in_chrom_info", input_type=Boolean(optional=True), prefix="--chrom-info", doc=InputDocumentation(doc="Tabulated two-columns file. Chromosomes as column 1, sizes as column 2 (default: None)")), ToolInput(tag="in_peak_file", input_type=Boolean(optional=True), prefix="--peak-file", doc=InputDocumentation(doc="The file containing the peaks/regions to be annotated. (bed format). (default: None)")), ToolInput(tag="in_more_bed_labels", input_type=Boolean(optional=True), prefix="--more-bed-labels", doc=InputDocumentation(doc="A comma separated list of labels (see --more-bed) (default: None)")), ToolInput(tag="in_bed_excl", input_type=Boolean(optional=True), prefix="--bed-excl", doc=InputDocumentation(doc="Exclusion file. The chromosomes will be shortened by this much for the shuffles of peaks and features. (bed format). (default: None)")), ToolInput(tag="in_bed_incl", input_type=Boolean(optional=True), prefix="--bed-incl", doc=InputDocumentation(doc="Opposite of --bed-excl, will perform the same operation but keep only those regions. (default: None)")), ToolInput(tag="in_upstream", input_type=Boolean(optional=True), prefix="--upstream", doc=InputDocumentation(doc="Extend the TSS and TTS of in 5' by a given value. (default: 1000)")), ToolInput(tag="in_downstream", input_type=Boolean(optional=True), prefix="--downstream", doc=InputDocumentation(doc="Extend the TSS and TTS of in 3' by a given value. (default: 1000)")), ToolInput(tag="in_more_keys", input_type=Boolean(optional=True), prefix="--more-keys", doc=InputDocumentation(doc="A comma separated list of key used for labeling the genome. See Notes. (default: None)")), ToolInput(tag="in_no_basic_feature", input_type=Boolean(optional=True), prefix="--no-basic-feature", doc=InputDocumentation(doc="No statistics for basic features of GTF. Concentrates on --more-bed and --more-keys. (default: False)")), ToolInput(tag="in_nb_threads", input_type=Boolean(optional=True), prefix="--nb-threads", doc=InputDocumentation(doc="Number of threads for multiprocessing. (default: 1)")), ToolInput(tag="in_seed", input_type=Boolean(optional=True), prefix="--seed", doc=InputDocumentation(doc="Numpy random seed. (default: 42)")), ToolInput(tag="in_mini_batch_nb", input_type=Boolean(optional=True), prefix="--minibatch-nb", doc=InputDocumentation(doc="Number of minibatches of shuffles. (default: 10)")), ToolInput(tag="in_mini_batch_size", input_type=Boolean(optional=True), prefix="--minibatch-size", doc=InputDocumentation(doc="Size of each minibatch, in number of shuffles. (default: 20)")), ToolInput(tag="in_use_markov", input_type=Boolean(optional=True), prefix="--use-markov", doc=InputDocumentation(doc="Whether to use Markov model realisations (order 2) instead of independant shuffles. See notes. (default: False)")), ToolInput(tag="in_output_dir", input_type=Directory(optional=True), prefix="--outputdir", doc=InputDocumentation(doc="Output directory name. (default: ologram_output)")), ToolInput(tag="in_pdf_width", input_type=File(optional=True), prefix="--pdf-width", doc=InputDocumentation(doc="Output pdf file width (inches). (default: None)")), ToolInput(tag="in_pdf_height", input_type=File(optional=True), prefix="--pdf-height", doc=InputDocumentation(doc="Output pdf file height (inches). (default: None)")), ToolInput(tag="in_pdf_file_alt", input_type=Boolean(optional=True), prefix="--pdf-file-alt", doc=InputDocumentation(doc="Provide an alternative path for the main image. (default: None)")), ToolInput(tag="in_no_pdf", input_type=Boolean(optional=True), prefix="--no-pdf", doc=InputDocumentation(doc="Do not produce any image file. (default: False)")), ToolInput(tag="in_display_fit_quality", input_type=Boolean(optional=True), prefix="--display-fit-quality", doc=InputDocumentation(doc="Display the negative binomial fit quality on the diagrams. (default: False)")), ToolInput(tag="in_tsv_file_path", input_type=File(optional=True), prefix="--tsv-file-path", doc=InputDocumentation(doc="Provide an alternative path for text output file. (default: None)")), ToolInput(tag="in_sort_features", input_type=Boolean(optional=True), prefix="--sort-features", doc=InputDocumentation(doc="Whether to sort features in diagrams according to a computed statistic. (default: None)")), ToolInput(tag="in_no_gtf", input_type=Boolean(optional=True), prefix="--no-gtf", doc=InputDocumentation(doc="No GTF file is provided as input. (default: False)")), ToolInput(tag="in_force_chrom_gtf", input_type=Boolean(optional=True), prefix="--force-chrom-gtf", doc=InputDocumentation(doc="Discard silently, from GTF, genes outside chromosomes defined in --chrom-info. (default: False)")), ToolInput(tag="in_force_chrom_peak", input_type=Boolean(optional=True), prefix="--force-chrom-peak", doc=InputDocumentation(doc="Discard silently, from --peak-file, peaks outside chromosomes defined in --chrom-info. (default: False)")), ToolInput(tag="in_force_chrom_more_bed", input_type=Boolean(optional=True), prefix="--force-chrom-more-bed", doc=InputDocumentation(doc="Discard silently, from --more-bed files, regions outside chromosomes defined in --chrom-info. (default: False)")), ToolInput(tag="in_verbosity", input_type=Boolean(optional=True), prefix="--verbosity", doc=InputDocumentation(doc="Set output verbosity ([0-3]). (default: 0)")), ToolInput(tag="in_no_date", input_type=File(optional=True), prefix="--no-date", doc=InputDocumentation(doc="Do not add date to output file names. (default: False)")), ToolInput(tag="in_add_chr", input_type=Boolean(optional=True), prefix="--add-chr", doc=InputDocumentation(doc="Add 'chr' to chromosome names before printing output. (default: False)")), ToolInput(tag="in_tmp_dir", input_type=Boolean(optional=True), prefix="--tmp-dir", doc=InputDocumentation(doc="Keep all temporary files into this folder. (default: None)")), ToolInput(tag="in_keep_all", input_type=Boolean(optional=True), prefix="--keep-all", doc=InputDocumentation(doc="Try to keep all temporary files even if process does not terminate normally. (default: False)")), ToolInput(tag="in_logger_file", input_type=Boolean(optional=True), prefix="--logger-file", doc=InputDocumentation(doc="Stores the arguments passed to the command into a file. (default: None)")), ToolInput(tag="in_write_message_to_file", input_type=Boolean(optional=True), prefix="--write-message-to-file", doc=InputDocumentation(doc="Store all message into a file. (default: None)")), ToolInput(tag="in_the", input_type=String(), position=0, doc=InputDocumentation(doc="regions of the query (--peak-file) are located independently of the  reference (--inputfile or")), ToolInput(tag="in_regions_dot", input_type=String(), position=0, doc=InputDocumentation(doc="*  You can exclude regions from the shuffling. This is done by shuffling across a concatenated")), ToolInput(tag="in_arguments", input_type=Int(), position=0, doc=InputDocumentation(doc="Arguments:"))], outputs=[ToolOutput(tag="out_output_dir", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_output_dir", type_hint=File()), doc=OutputDocumentation(doc="Output directory name. (default: ologram_output)")), ToolOutput(tag="out_pdf_width", output_type=File(optional=True), selector=InputSelector(input_to_select="in_pdf_width", type_hint=File()), doc=OutputDocumentation(doc="Output pdf file width (inches). (default: None)")), ToolOutput(tag="out_pdf_height", output_type=File(optional=True), selector=InputSelector(input_to_select="in_pdf_height", type_hint=File()), doc=OutputDocumentation(doc="Output pdf file height (inches). (default: None)")), ToolOutput(tag="out_tsv_file_path", output_type=File(optional=True), selector=InputSelector(input_to_select="in_tsv_file_path", type_hint=File()), doc=OutputDocumentation(doc="Provide an alternative path for text output file. (default: None)")), ToolOutput(tag="out_no_date", output_type=File(optional=True), selector=InputSelector(input_to_select="in_no_date", type_hint=File()), doc=OutputDocumentation(doc="Do not add date to output file names. (default: False)"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Gtftk_Ologram_V0_1_0().translate("wdl", allow_empty_container=True)

