from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import String, File, Int, Boolean

Cnv_Facets_R_V0_1_0 = CommandToolBuilder(tool="cnv_facets.R", base_command=["cnv_facets.R"], inputs=[ToolInput(tag="in_out", input_type=String(optional=True), prefix="--out", doc=InputDocumentation(doc="Output prefix for the output files")), ToolInput(tag="in_snp_tumour", input_type=File(optional=True), prefix="--snp-tumour", doc=InputDocumentation(doc="BAM file for tumour sample")), ToolInput(tag="in_snp_normal", input_type=File(optional=True), prefix="--snp-normal", doc=InputDocumentation(doc="BAM file for normal sample")), ToolInput(tag="in_snp_vcf", input_type=File(optional=True), prefix="--snp-vcf", doc=InputDocumentation(doc="VCF file of SNPs where pileup is to be computed")), ToolInput(tag="in_snp_mapq", input_type=Int(optional=True), prefix="--snp-mapq", doc=InputDocumentation(doc="Sets the minimum threshold for mapping quality. Default 5")), ToolInput(tag="in_snp_baq", input_type=Int(optional=True), prefix="--snp-baq", doc=InputDocumentation(doc="Sets the minimum threshold for base quality. Default 10")), ToolInput(tag="in_snp_count_orphans", input_type=Boolean(optional=True), prefix="--snp-count-orphans", doc=InputDocumentation(doc="Do not discard anomalous read pairs")), ToolInput(tag="in_snp_nprocs", input_type=Int(optional=True), prefix="--snp-nprocs", doc=InputDocumentation(doc="Number of parallel processes to run to prepare the read pileup file.\nEach chromsome is assigned to a process. Default 1")), ToolInput(tag="in_pile_up", input_type=File(optional=True), prefix="--pileup", doc=InputDocumentation(doc="Pileup for matched normal (first sample) and tumour\n(second sample). Not needed if using BAM input. This file\nis the <prefix>.cvs.gz output of of a previous run of cnv_facet.R")), ToolInput(tag="in_depth", input_type=Int(optional=True), prefix="--depth", doc=InputDocumentation(doc="DEPTH, -d DEPTH DEPTH\nMinimum and maximum depth in normal sample for a position\nto be considered. Default 25 4000")), ToolInput(tag="in_targets", input_type=File(optional=True), prefix="--targets", doc=InputDocumentation(doc="BED file of target regions to scan. It may be the target regions\nfrom WEX or panel sequencing protocols. It is not required, even\nfor targeted sequencing, but it may improve the results")), ToolInput(tag="in_cval", input_type=Int(optional=True), prefix="--cval", doc=InputDocumentation(doc="CVAL, -cv CVAL CVAL\nCritical values for segmentation in pre-processing and\nprocessing. Larger values reduce segmentation. [25 150] is\nfacets default based on exome data. For whole genome consider\nincreasing to [25 400] and for targeted sequencing consider\nreducing them. Default 25 150")), ToolInput(tag="in_n_bhd_snp", input_type=Int(optional=True), prefix="--nbhd-snp", doc=InputDocumentation(doc="If an interval of size nbhd-snp contains more than one SNP,\nsample a random one.  This sampling reduces the SNP serial\ncorrelation. This value should be similar to the median insert\nsize of the libraries. If 'auto' and if using paired end BAM\ninput, use the estimated insert size from the normal bam file.\nOtherwise use 250. Default auto")), ToolInput(tag="in_annotation", input_type=File(optional=True), prefix="--annotation", doc=InputDocumentation(doc="Optional annotation file in BED format where the 4th column\ncontains the feature name (e.g. gene name). CNVs will be\nannotated with an additional INFO/TAG reporting all the\noveralapping features")), ToolInput(tag="in_g_build", input_type=String(optional=True), prefix="--gbuild", doc=InputDocumentation(doc="String indicating the reference genome build. Default hg38.")), ToolInput(tag="in_unmatched", input_type=Boolean(optional=True), prefix="--unmatched", doc=InputDocumentation(doc="Normal sample is unmatched. If set, heterozygote SNPs are\ncalled using tumor reads only and logOR calculations are different")), ToolInput(tag="in_no_cov_plot", input_type=Boolean(optional=True), prefix="--no-cov-plot", doc=InputDocumentation(doc="Do not produce coverage plots (they can be memory intensive).\nOther plots remain unaffected")), ToolInput(tag="in_rnd_seed", input_type=Int(optional=True), prefix="--rnd-seed", doc=InputDocumentation(doc="Seed for random number generator. Default: The name of the input file"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Cnv_Facets_R_V0_1_0().translate("wdl", allow_empty_container=True)

