from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Int, File, Boolean, String

Anvi_Script_Reformat_Fasta_V0_1_0 = CommandToolBuilder(tool="anvi_script_reformat_fasta", base_command=["anvi-script-reformat-fasta"], inputs=[ToolInput(tag="in_min_len", input_type=Int(optional=True), prefix="--min-len", doc=InputDocumentation(doc="Minimum length of contigs to keep (contigs shorter\nthan this value will not be included in the output\nfile). The default is 0, so nothing will be removed if\nyou do not declare a minimum size.")), ToolInput(tag="in_max_percentage_gaps", input_type=File(optional=True), prefix="--max-percentage-gaps", doc=InputDocumentation(doc="Maximum fraction of gaps in a sequence (any sequence\nwith more gaps will be removed from the output FASTA\nfile). The default is 100.000000.")), ToolInput(tag="in_max_gaps", input_type=Int(optional=True), prefix="--max-gaps", doc=InputDocumentation(doc="Maximum amount of gaps allowed per sequence in the\nalignment. Don't know which threshold to pick? Use\n--export-gap-counts-table to explore the gap counts\nper sequence distribution! (default: 1000000)")), ToolInput(tag="in_file__excludeids", input_type=File(optional=True), prefix="-i", doc=InputDocumentation(doc="FILE, --exclude-ids TXT FILE\nIDs to remove from the FASTA file. You cannot provide\nboth --keep-ids and --exclude-ids. (default: None)")), ToolInput(tag="in_export_gap_counts_table", input_type=Int(optional=True), prefix="--export-gap-counts-table", doc=InputDocumentation(doc="FILE\nExport a table with the number of gaps per sequence.\nPlease provide a prefix to name the file. (default:\nNone)")), ToolInput(tag="in_file__keepids", input_type=Int(optional=True), prefix="-I", doc=InputDocumentation(doc="FILE, --keep-ids TXT FILE\nIf provided, all IDs not in this file will be excluded\nfrom the reformatted FASTA file. Any additional\nfilters (such as --min-len) will still be applied to\nthe IDs in this file. You cannot provide both\n--exclude-ids and --keep-ids. (default: None)")), ToolInput(tag="in_file__default", input_type=File(optional=True), prefix="-o", doc=InputDocumentation(doc="FILE, --output-file FASTA FILE\nOutput file path. (default: None)")), ToolInput(tag="in_simplify_names", input_type=Boolean(optional=True), prefix="--simplify-names", doc=InputDocumentation(doc="Edit deflines to make sure they contigs have simple\nnames. (default: False)")), ToolInput(tag="in_prefix", input_type=String(optional=True), prefix="--prefix", doc=InputDocumentation(doc="Use this parameter if you would like to add a prefix\nto your contig names while simplifying them. The\nprefix must be a single word (you can use underscor\ncharacter, but nothing more!). (default: None)")), ToolInput(tag="in_seq_type", input_type=Int(optional=True), prefix="--seq-type", doc=InputDocumentation(doc="TYPE   Supply either 'NT' or 'AA' (if you want). If 'NT', any\ncharacters besides {A,C,T,G} will by replaced with\n'N'. If 'AA', any characters that are not 1-letter\namino acid characters will be replaced with 'X'. If\nyou don't supply anything, no charaters will be\nmodified. (default: None)")), ToolInput(tag="in_file__reportfile", input_type=File(optional=True), prefix="-r", doc=InputDocumentation(doc="FILE, --report-file REPORT FILE\nReport file path. When you run this program with\n`--simplify-names` flag, all changes to deflines will\nbe reported in this file in case you need to go back\nto this information later. It is not mandatory to\ndeclare one, but it is a very good idea to have it.\n(default: None)")), ToolInput(tag="in_fast_a", input_type=String(), position=0, doc=InputDocumentation(doc="üçï Can provide: ")), ToolInput(tag="in_contigs_fast_a", input_type=String(), position=1, doc=InputDocumentation(doc="üç∫ More on `anvi-script-reformat-fasta`:"))], outputs=[ToolOutput(tag="out_max_percentage_gaps", output_type=File(optional=True), selector=InputSelector(input_to_select="in_max_percentage_gaps", type_hint=File()), doc=OutputDocumentation(doc="Maximum fraction of gaps in a sequence (any sequence\nwith more gaps will be removed from the output FASTA\nfile). The default is 100.000000.")), ToolOutput(tag="out_file__default", output_type=File(optional=True), selector=InputSelector(input_to_select="in_file__default", type_hint=File()), doc=OutputDocumentation(doc="FILE, --output-file FASTA FILE\nOutput file path. (default: None)"))], container="quay.io/biocontainers/anvio-minimal:7--py_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Anvi_Script_Reformat_Fasta_V0_1_0().translate("wdl")

