from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, File

Sumaclust_V0_1_0 = CommandToolBuilder(tool="sumaclust", base_command=["sumaclust"], inputs=[ToolInput(tag="in_reference_sequence_shortest", input_type=Boolean(optional=True), prefix="-l", doc=InputDocumentation(doc=": Reference sequence length is the shortest.")), ToolInput(tag="in_reference_sequence_largest", input_type=Boolean(optional=True), prefix="-L", doc=InputDocumentation(doc=": Reference sequence length is the largest.")), ToolInput(tag="in_reference_sequence_alignment", input_type=Boolean(optional=True), prefix="-a", doc=InputDocumentation(doc=": Reference sequence length is the alignment length (default).")), ToolInput(tag="in_score_normalized_reference", input_type=Boolean(optional=True), prefix="-n", doc=InputDocumentation(doc=": Score is normalized by reference sequence length (default).")), ToolInput(tag="in_raw_score_normalized", input_type=Boolean(optional=True), prefix="-r", doc=InputDocumentation(doc=": Raw score, not normalized.")), ToolInput(tag="in_score_expressed_expressed", input_type=Boolean(optional=True), prefix="-d", doc=InputDocumentation(doc=": Score is expressed in distance (default : score is expressed in similarity).")), ToolInput(tag="in_score_threshold_normalized", input_type=Boolean(optional=True), prefix="-t", doc=InputDocumentation(doc="##.## : Score threshold for clustering. If the score is normalized and expressed in similarity (default),\nit is an identity, e.g. 0.95 for an identity of 95%. If the score is normalized\nand expressed in distance, it is (1.0 - identity), e.g. 0.05 for an identity of 95%.\nIf the score is not normalized and expressed in similarity, it is the length of the\nLongest Common Subsequence. If the score is not normalized and expressed in distance,\nit is (reference length - LCS length).\nOnly sequences with a similarity above ##.## with the center sequence of a cluster\nare assigned to that cluster. Default: 0.97.")), ToolInput(tag="in_exact_option_sequence", input_type=Boolean(optional=True), prefix="-e", doc=InputDocumentation(doc=": Exact option : A sequence is assigned to the cluster with the center sequence presenting the\nhighest similarity score > threshold, as opposed to the default 'fast' option where a sequence is\nassigned to the first cluster found with a center sequence presenting a score > threshold.")), ToolInput(tag="in_maximum_ratio_be", input_type=Boolean(optional=True), prefix="-R", doc=InputDocumentation(doc="##    : Maximum ratio between the counts of two sequences so that the less abundant one can be considered\nas a variant of the more abundant one. Default: 1.0.")), ToolInput(tag="in_multithreading_threads_using", input_type=Boolean(optional=True), prefix="-p", doc=InputDocumentation(doc="##    : Multithreading with ## threads using openMP.")), ToolInput(tag="in_sorting_by_be", input_type=Boolean(optional=True), prefix="-s", doc=InputDocumentation(doc="####  : Sorting by ####. Must be 'None' for no sorting, or a key in the fasta header of each sequence,\nexcept for the count that can be computed (default : sorting by count).")), ToolInput(tag="in_sorting_ascending_default", input_type=Boolean(optional=True), prefix="-o", doc=InputDocumentation(doc=": Sorting is in ascending order (default : descending).")), ToolInput(tag="in__ns_discarded", input_type=Boolean(optional=True), prefix="-g", doc=InputDocumentation(doc=": n's are replaced with a's (default: sequences with n's are discarded).")), ToolInput(tag="in_output_otu_table", input_type=File(optional=True), prefix="-B", doc=InputDocumentation(doc="###   : Output of the OTU table in BIOM format is activated, and written to file ###.")), ToolInput(tag="in_output_otu_map", input_type=File(optional=True), prefix="-O", doc=InputDocumentation(doc="###   : Output of the OTU map (observation map) is activated, and written to file ###.")), ToolInput(tag="in_output_fasta_format_written", input_type=File(optional=True), prefix="-F", doc=InputDocumentation(doc="###   : Output in FASTA format is written to file ### instead of standard output.")), ToolInput(tag="in_output_fasta_format_deactivated", input_type=Boolean(optional=True), prefix="-f", doc=InputDocumentation(doc=": Output in FASTA format is deactivated."))], outputs=[ToolOutput(tag="out_output_otu_table", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output_otu_table", type_hint=File()), doc=OutputDocumentation(doc="###   : Output of the OTU table in BIOM format is activated, and written to file ###.")), ToolOutput(tag="out_output_otu_map", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output_otu_map", type_hint=File()), doc=OutputDocumentation(doc="###   : Output of the OTU map (observation map) is activated, and written to file ###.")), ToolOutput(tag="out_output_fasta_format_written", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output_fasta_format_written", type_hint=File()), doc=OutputDocumentation(doc="###   : Output in FASTA format is written to file ### instead of standard output."))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Sumaclust_V0_1_0().translate("wdl", allow_empty_container=True)

