from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Int, File, Boolean, Directory, String

Minority_Freq_V0_1_0 = CommandToolBuilder(tool="minority_freq", base_command=["minority_freq"], inputs=[ToolInput(tag="in_start", input_type=Int(optional=True), prefix="--start", doc=InputDocumentation(doc="Starting position of the region of interest, 0-based\nindexing (default: None)")), ToolInput(tag="in_end", input_type=Int(optional=True), prefix="--end", doc=InputDocumentation(doc="Ending position of the region of interest, 0-based\nindexing. Note a half-open interval is used, i.e,\n[start:end) (default: None)")), ToolInput(tag="in_config", input_type=File(optional=True), prefix="--config", doc=InputDocumentation(doc="Report minority aminoacids - a .config file specifying\nreading frames expected (default: None)")), ToolInput(tag="in_minimum_read_depth", input_type=Int(optional=True), prefix="-c", doc=InputDocumentation(doc="Minimum read depth for reporting variants per locus\nand sample (default: 100)")), ToolInput(tag="in_name_patientsample_identifiers", input_type=Int(optional=True), prefix="-N", doc=InputDocumentation(doc=",name2,...    Patient/sample identifiers as comma separated strings\n(default: None)")), ToolInput(tag="in_number_threads_default", input_type=Int(optional=True), prefix="-t", doc=InputDocumentation(doc="Number of threads (default: 1)")), ToolInput(tag="in_freqs", input_type=Boolean(optional=True), prefix="--freqs", doc=InputDocumentation(doc="Indicates whether or not all frequencies should be\nstored (default: False)")), ToolInput(tag="in_indicates_coverage_be", input_type=File(optional=True), prefix="-d", doc=InputDocumentation(doc="Indicates whether coverage per locus should be written\nto output file (default: False)")), ToolInput(tag="in_output_directory_default", input_type=Directory(optional=True), prefix="-o", doc=InputDocumentation(doc="Output directory (default: /)")), ToolInput(tag="in_fasta_file_containing", input_type=File(optional=True), prefix="-r", doc=InputDocumentation(doc="Either a fasta file containing a reference sequence or\nthe reference name of the region/chromosome of\ninterest. The latter is expected if a region is\nspecified (default: None)\n")), ToolInput(tag="in_bam", input_type=String(), position=0, doc=InputDocumentation(doc="BAM file(s)"))], outputs=[ToolOutput(tag="out_indicates_coverage_be", output_type=File(optional=True), selector=InputSelector(input_to_select="in_indicates_coverage_be", type_hint=File()), doc=OutputDocumentation(doc="Indicates whether coverage per locus should be written\nto output file (default: False)")), ToolOutput(tag="out_output_directory_default", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_output_directory_default", type_hint=File()), doc=OutputDocumentation(doc="Output directory (default: /)"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Minority_Freq_V0_1_0().translate("wdl", allow_empty_container=True)

