from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import String, File, Int, Float, Boolean

Nanovar_V0_1_0 = CommandToolBuilder(tool="nanovar", base_command=["nanovar"], inputs=[ToolInput(tag="in_data_type", input_type=String(optional=True), prefix="--data_type", doc=InputDocumentation(doc="type of long-read data [ont]\nont - Oxford Nanopore Technologies\npacbio-clr - Pacific Biosciences CLR\npacbio-ccs - Pacific Biosciences CCS")), ToolInput(tag="in_filter_bed", input_type=File(optional=True), prefix="--filter_bed", doc=InputDocumentation(doc="BED file with genomic regions to be excluded [None]\n(e.g. telomeres and centromeres) Either specify name of in-built\nreference genome filter (i.e. hg38, hg19, mm10) or provide full\npath to own BED file.")), ToolInput(tag="in_min_cov", input_type=Int(optional=True), prefix="--mincov", doc=InputDocumentation(doc="minimum number of reads required to call a breakend [2]")), ToolInput(tag="in_min_len", input_type=Int(optional=True), prefix="--minlen", doc=InputDocumentation(doc="minimum length of SV to be detected [25]")), ToolInput(tag="in_split_pct", input_type=Float(optional=True), prefix="--splitpct", doc=InputDocumentation(doc="minimum percentage of unmapped bases within a long read\nto be considered as a split-read. 0.05<=p<=0.50 [0.05]")), ToolInput(tag="in_min_align", input_type=Int(optional=True), prefix="--minalign", doc=InputDocumentation(doc="minimum alignment length for single alignment reads [200]")), ToolInput(tag="in_buffer", input_type=Int(optional=True), prefix="--buffer", doc=InputDocumentation(doc="nucleotide length buffer for SV breakend clustering [50]")), ToolInput(tag="in_score", input_type=Float(optional=True), prefix="--score", doc=InputDocumentation(doc="score threshold for defining PASS/FAIL SVs in VCF [1.0]\nDefault score 1.0 was estimated from simulated analysis.")), ToolInput(tag="in_homo", input_type=Float(optional=True), prefix="--homo", doc=InputDocumentation(doc="lower limit of a breakend read ratio to classify a homozygous state [0.75]\n(i.e. Any breakend with homo<=ratio<=1.00 is classified as homozygous)")), ToolInput(tag="in_hetero", input_type=Float(optional=True), prefix="--hetero", doc=InputDocumentation(doc="lower limit of a breakend read ratio to classify a heterozygous state [0.35]\n(i.e. Any breakend with hetero<=ratio<homo is classified as heterozygous)")), ToolInput(tag="in_debug", input_type=Boolean(optional=True), prefix="--debug", doc=InputDocumentation(doc="run in debug mode")), ToolInput(tag="in_force", input_type=Boolean(optional=True), prefix="--force", doc=InputDocumentation(doc="run full pipeline (i.e. do not skip index generation)")), ToolInput(tag="in_quiet", input_type=Boolean(optional=True), prefix="--quiet", doc=InputDocumentation(doc="hide verbose")), ToolInput(tag="in_threads", input_type=Int(optional=True), prefix="--threads", doc=InputDocumentation(doc="number of available threads for use [1]")), ToolInput(tag="in_model", input_type=File(optional=True), prefix="--model", doc=InputDocumentation(doc="specify path to custom-built model")), ToolInput(tag="in_mm", input_type=File(optional=True), prefix="--mm", doc=InputDocumentation(doc="specify path to 'minimap2' executable")), ToolInput(tag="in_st", input_type=File(optional=True), prefix="--st", doc=InputDocumentation(doc="specify path to 'samtools' executable")), ToolInput(tag="in_mdb", input_type=File(optional=True), prefix="--mdb", doc=InputDocumentation(doc="specify path to 'makeblastdb' executable")), ToolInput(tag="in_wmk", input_type=File(optional=True), prefix="--wmk", doc=InputDocumentation(doc="specify path to 'windowmasker' executable")), ToolInput(tag="in_hsb", input_type=File(optional=True), prefix="--hsb", doc=InputDocumentation(doc="specify path to 'hs-blastn' executable"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Nanovar_V0_1_0().translate("wdl", allow_empty_container=True)

