from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, File

Maf2Maf_Pl_V0_1_0 = CommandToolBuilder(tool="maf2maf.pl", base_command=["maf2maf.pl"], inputs=[ToolInput(tag="in_input_maf", input_type=Boolean(optional=True), prefix="--input-maf", doc=InputDocumentation(doc="Path to input file in MAF format")), ToolInput(tag="in_output_maf", input_type=File(optional=True), prefix="--output-maf", doc=InputDocumentation(doc="Path to output MAF file [Default: STDOUT]")), ToolInput(tag="in_tmp_dir", input_type=Boolean(optional=True), prefix="--tmp-dir", doc=InputDocumentation(doc="Folder to retain intermediate VCFs/MAFs after runtime [Default: usually /tmp]")), ToolInput(tag="in_tum_depth_col", input_type=Boolean(optional=True), prefix="--tum-depth-col", doc=InputDocumentation(doc="Name of MAF column for read depth in tumor BAM [t_depth]")), ToolInput(tag="in_tum_rad_col", input_type=Boolean(optional=True), prefix="--tum-rad-col", doc=InputDocumentation(doc="Name of MAF column for reference allele depth in tumor BAM [t_ref_count]")), ToolInput(tag="in_tum_vad_col", input_type=Boolean(optional=True), prefix="--tum-vad-col", doc=InputDocumentation(doc="Name of MAF column for variant allele depth in tumor BAM [t_alt_count]")), ToolInput(tag="in_nrm_depth_col", input_type=Boolean(optional=True), prefix="--nrm-depth-col", doc=InputDocumentation(doc="Name of MAF column for read depth in normal BAM [n_depth]")), ToolInput(tag="in_nrm_rad_col", input_type=Boolean(optional=True), prefix="--nrm-rad-col", doc=InputDocumentation(doc="Name of MAF column for reference allele depth in normal BAM [n_ref_count]")), ToolInput(tag="in_nrm_vad_col", input_type=Boolean(optional=True), prefix="--nrm-vad-col", doc=InputDocumentation(doc="Name of MAF column for variant allele depth in normal BAM [n_alt_count]")), ToolInput(tag="in_retain_cols", input_type=Boolean(optional=True), prefix="--retain-cols", doc=InputDocumentation(doc="Comma-delimited list of columns to retain from the input MAF [Center,Verification_Status,Validation_Status,Mutation_Status,Sequencing_Phase,Sequence_Source,Validation_Method,Score,BAM_file,Sequencer,Tumor_Sample_UUID,Matched_Norm_Sample_UUID]")), ToolInput(tag="in_custom_enst", input_type=Boolean(optional=True), prefix="--custom-enst", doc=InputDocumentation(doc="List of custom ENST IDs that override canonical selection")), ToolInput(tag="in_vep_path", input_type=Boolean(optional=True), prefix="--vep-path", doc=InputDocumentation(doc="Folder containing the vep script [~/vep]")), ToolInput(tag="in_vep_data", input_type=Boolean(optional=True), prefix="--vep-data", doc=InputDocumentation(doc="VEP's base cache/plugin directory [~/.vep]")), ToolInput(tag="in_vep_forks", input_type=Boolean(optional=True), prefix="--vep-forks", doc=InputDocumentation(doc="Number of forked processes to use when running VEP [4]")), ToolInput(tag="in_buffer_size", input_type=Boolean(optional=True), prefix="--buffer-size", doc=InputDocumentation(doc="Number of variants VEP loads at a time; Reduce this for low memory systems [5000]")), ToolInput(tag="in_any_allele", input_type=Boolean(optional=True), prefix="--any-allele", doc=InputDocumentation(doc="When reporting co-located variants, allow mismatched variant alleles too")), ToolInput(tag="in_filter_vcf", input_type=Boolean(optional=True), prefix="--filter-vcf", doc=InputDocumentation(doc="A VCF for FILTER tag common_variant. Set to 0 to disable [~/.vep/ExAC_nonTCGA.r0.3.1.sites.vep.vcf.gz]")), ToolInput(tag="in_max_filter_ac", input_type=Boolean(optional=True), prefix="--max-filter-ac", doc=InputDocumentation(doc="Use tag common_variant if the filter-vcf reports a subpopulation AC higher than this [10]")), ToolInput(tag="in_species", input_type=Boolean(optional=True), prefix="--species", doc=InputDocumentation(doc="Ensembl-friendly name of species (e.g. mus_musculus for mouse) [homo_sapiens]")), ToolInput(tag="in_ncbi_build", input_type=Boolean(optional=True), prefix="--ncbi-build", doc=InputDocumentation(doc="NCBI reference assembly of variants in MAF (e.g. GRCm38 for mouse) [GRCh37]")), ToolInput(tag="in_cache_version", input_type=Boolean(optional=True), prefix="--cache-version", doc=InputDocumentation(doc="Version of offline cache to use with VEP (e.g. 75, 84, 91) [Default: Installed version]")), ToolInput(tag="in_ref_fast_a", input_type=Boolean(optional=True), prefix="--ref-fasta", doc=InputDocumentation(doc="Reference FASTA file [~/.vep/homo_sapiens/91_GRCh37/Homo_sapiens.GRCh37.75.dna.primary_assembly.fa.gz]")), ToolInput(tag="in_man", input_type=Boolean(optional=True), prefix="--man", doc=InputDocumentation(doc="Print the detailed manual"))], outputs=[ToolOutput(tag="out_output_maf", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output_maf", type_hint=File()), doc=OutputDocumentation(doc="Path to output MAF file [Default: STDOUT]"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Maf2Maf_Pl_V0_1_0().translate("wdl", allow_empty_container=True)

