#!/usr/bin/env cwl-runner

baseCommand:
- motifscan
class: CommandLineTool
cwlVersion: v1.0
id: motifscan
inputs:
- doc: 'Required input genomic regions file to scan motifs on. (default: None)'
  id: peak
  inputBinding:
    prefix: --peak
  type: string
- doc: 'Format of input genomic regions. See documents for more information. (default:
    bed5col)'
  id: format
  inputBinding:
    prefix: --format
  type: string
- doc: "Required pre-compiled motif file, which is generated by 'motifcompile'. (default:\
    \ None)"
  id: motif
  inputBinding:
    prefix: --motif
  type: string
- doc: "Required genome directory, which contains genome sequence and chromosome length.\
    \ You can run 'genomecompile' to generate the directory. (default: None)"
  id: genome
  inputBinding:
    prefix: --genome
  type: string
- doc: 'Gene annotation file, which is used to generate random controls when performing
    enrichment analysis. Sequence features on promoters are very different from those
    on whole genome. We will pick up random control regions with similar genomic locations
    for each input region instead of the whole genome. (default: None)'
  id: annotation
  inputBinding:
    prefix: --annotation
  type: string
- doc: 'Motif list file. If specified, the program will only perform MotifScan on
    given motifs in this list. (default: None)'
  id: motif_list
  inputBinding:
    prefix: --motif_list
  type: string
- doc: 'If set, MotifScan will neither generate random control regions nor perform
    enrichment analysis. (default: True)'
  id: no_enrichment
  inputBinding:
    prefix: --no_enrichment
  type: boolean
- doc: 'User specified control regions. (default: None)'
  id: control
  inputBinding:
    prefix: --control
  type: string
- doc: 'Format of control regions (same as -f for input regions). (default: bed5col)'
  id: c_format
  inputBinding:
    prefix: --cformat
  type: string
- doc: "Genomic location to apply MotifScan. 'all': perform MotifScan on the all input\
    \ regions. 'promoter': perform MotifScan on regions which located at promoter\
    \ regions. 'distal': perform MotifScan on regions which located at distal regions.\
    \ 'gene': perform MotifScan on regions which is the target(nearest one) of genes\
    \ within 10kbp distance cutoff. (default: all)"
  id: r
  inputBinding:
    prefix: -r
  type: string
- doc: "Upstream distance to TSS to define promoter regions. Valid when option -r\
    \ is set as 'promoter' or 'distal'. (default: 4000)"
  id: upstream
  inputBinding:
    prefix: --upstream
  type: string
- doc: "Upstream distance to TSS to define promoter regions. Valid when option -r\
    \ is set as 'promoter' or 'distal'. (default: 2000)"
  id: downstream
  inputBinding:
    prefix: --downstream
  type: string
- doc: 'The number of random control regions generated for each input region. For
    example, if you provide 1000 regions, 5000 random regions will be generated by
    default. (default: 5)'
  id: random_times
  inputBinding:
    prefix: --random_times
  type: string
- doc: 'The length of input regions to perform MotifScan around peak summit/midpoint.
    If set to 0, the whole peak region is used to perform MotifScan. (default: 1000)'
  id: peak_length
  inputBinding:
    prefix: --peak_length
  type: string
- doc: 'If set, MotifScan will report the sequence and position of each target. (default:
    False)'
  id: target_site
  inputBinding:
    prefix: --target_site
  type: boolean
- doc: "Output directory to the write the results. If unspecified, the results will\
    \ be written to 'motifscan_output_PREFIX_OF_INPUT_REGION' under working directory.\
    \ (default: None)"
  id: o
  inputBinding:
    prefix: -o
  type: string
