from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Directory, Boolean, Int, String, File

Kb_Ref_V0_1_0 = CommandToolBuilder(tool="kb_ref", base_command=["kb", "ref"], inputs=[ToolInput(tag="in_tmp", input_type=Directory(optional=True), prefix="--tmp", doc=InputDocumentation(doc="Override default temporary directory")), ToolInput(tag="in_keep_tmp", input_type=Boolean(optional=True), prefix="--keep-tmp", doc=InputDocumentation(doc="Do not delete the tmp directory")), ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="--verbose", doc=InputDocumentation(doc="Print debugging information")), ToolInput(tag="in_number_split_index", input_type=Int(optional=True), prefix="-n", doc=InputDocumentation(doc="Number of files to split the index into. If this\noption is specified, the FASTA that is normally used\nto create an index is split into `N` approximately-\nequal parts. Each of these FASTAs are indexed\nseparately. When using this option with `--workflow\nlamanno`, the intron FASTA is split into N-1\napproximately-equal parts and indexed, while the cDNA\nFASTA is not split and indexed.")), ToolInput(tag="in_download_prebuilt_index", input_type=String(optional=True), prefix="-d", doc=InputDocumentation(doc="Download a pre-built kallisto index (along with all\nnecessary files) instead of building it locally")), ToolInput(tag="in_use_option_override", input_type=Int(optional=True), prefix="-k", doc=InputDocumentation(doc="Use this option to override the k-mer length of the\nindex. Usually, the k-mer length automatically\ncalculated by `kb` provides the best results.")), ToolInput(tag="in_workflow", input_type=String(optional=True), prefix="--workflow", doc=InputDocumentation(doc="Type of workflow to prepare files for. Use `lamanno`\nfor RNA velocity based on La Manno et al. 2018 logic.\nUse `nucleus` for RNA velocity on single-nucleus RNA-\nseq reads. Use `kite` for feature barcoding. (default:\nstandard)")), ToolInput(tag="in_laman_no", input_type=Boolean(optional=True), prefix="--lamanno", doc=InputDocumentation(doc="Deprecated. Use `--workflow lamanno` instead.")), ToolInput(tag="in_overwrite", input_type=Boolean(optional=True), prefix="--overwrite", doc=InputDocumentation(doc="Overwrite existing kallisto index")), ToolInput(tag="in_path_kallisto_index", input_type=File(optional=True), prefix="-i", doc=InputDocumentation(doc="Path to the kallisto index to be constructed. If `-n`\nis also specified, this is the prefix for the n\nindices to construct.")), ToolInput(tag="in_path_transcripttogene_mapping", input_type=Int(optional=True), prefix="-g", doc=InputDocumentation(doc="Path to transcript-to-gene mapping to be generated")), ToolInput(tag="in_fone", input_type=Int(optional=True), prefix="-f1", doc=InputDocumentation(doc="[Optional with -d] Path to the cDNA FASTA (lamanno,\nnucleus) or mismatch FASTA (kite) to be generated")), ToolInput(tag="in_f_two", input_type=Int(optional=True), prefix="-f2", doc=InputDocumentation(doc="Path to the intron FASTA to be generated")), ToolInput(tag="in_cone", input_type=Int(optional=True), prefix="-c1", doc=InputDocumentation(doc="Path to generate cDNA transcripts-to-capture")), ToolInput(tag="in_c_two", input_type=Int(optional=True), prefix="-c2", doc=InputDocumentation(doc="Path to generate intron transcripts-to-capture")), ToolInput(tag="in_fast_a", input_type=String(), position=0, doc=InputDocumentation(doc="Genomic FASTA file(s), comma-delimited")), ToolInput(tag="in_gtf", input_type=String(), position=1, doc=InputDocumentation(doc="Reference GTF file(s), comma-delimited")), ToolInput(tag="in_feature", input_type=String(), position=2, doc=InputDocumentation(doc="[`kite` workflow only] Path to TSV containing barcodes\nand feature names."))], outputs=[], container="quay.io/biocontainers/kb-python:0.25.1--py_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Kb_Ref_V0_1_0().translate("wdl")

