from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean

Bacteria_Tradis_V0_1_0 = CommandToolBuilder(tool="bacteria_tradis", base_command=["bacteria_tradis"], inputs=[ToolInput(tag="in_text_file_listing", input_type=Boolean(optional=True), prefix="-f", doc=InputDocumentation(doc=": text file listing fastq files with tradis tags attached")), ToolInput(tag="in_tag_search_optional", input_type=Boolean(optional=True), prefix="-t", doc=InputDocumentation(doc=": tag to search for (optional.  If not set runs bwa in tagless mode with no filtering.)")), ToolInput(tag="in_reference_genome_fasta", input_type=Boolean(optional=True), prefix="-r", doc=InputDocumentation(doc=": reference genome in fasta format (.fa)")), ToolInput(tag="in_td", input_type=Boolean(optional=True), prefix="-td", doc=InputDocumentation(doc=": tag direction - 3 or 5 (optional. default = 3)")), ToolInput(tag="in_mm", input_type=Boolean(optional=True), prefix="-mm", doc=InputDocumentation(doc=": number of mismatches allowed when matching tag (optional. default = 0)")), ToolInput(tag="in_mapping_quality_score", input_type=Boolean(optional=True), prefix="-m", doc=InputDocumentation(doc=": mapping quality cutoff score (optional. default = 30)")), ToolInput(tag="in_custom_kmer_value_min", input_type=Boolean(optional=True), prefix="-k", doc=InputDocumentation(doc=": custom k-mer value (min seed length) (optional)")), ToolInput(tag="in_s_malt", input_type=Boolean(optional=True), prefix="--smalt", doc=InputDocumentation(doc=": use smalt rather than bwa as the mapper")), ToolInput(tag="in_s_malt_k", input_type=Boolean(optional=True), prefix="--smalt_k", doc=InputDocumentation(doc=": custom k-mer value for SMALT mapping (optional)")), ToolInput(tag="in_s_malt_s", input_type=Boolean(optional=True), prefix="--smalt_s", doc=InputDocumentation(doc=": custom step size for SMALT mapping (optional)")), ToolInput(tag="in_s_malt_y", input_type=Boolean(optional=True), prefix="--smalt_y", doc=InputDocumentation(doc=": custom y parameter for SMALT (optional. default = 0.96)")), ToolInput(tag="in_s_malt_r", input_type=Boolean(optional=True), prefix="--smalt_r", doc=InputDocumentation(doc=": custom r parameter for SMALT (optional. default = -1)")), ToolInput(tag="in_number_threads_use", input_type=Boolean(optional=True), prefix="-n", doc=InputDocumentation(doc=": number of threads to use for SMALT and samtools sort (optional. default = 1)")), ToolInput(tag="in_set_defaults_m", input_type=Boolean(optional=True), prefix="-e", doc=InputDocumentation(doc=": set defaults for essentiality experiment (smalt_r = 0, -m = 0)")), ToolInput(tag="in_verbose_debugging_output", input_type=Boolean(optional=True), prefix="-v", doc=InputDocumentation(doc=": verbose debugging output"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Bacteria_Tradis_V0_1_0().translate("wdl", allow_empty_container=True)

