class: CommandLineTool
id: novoindex.cwl
inputs:
- id: in_kmer_length_used
  doc: 99        is the k-mer length to be used for the index. Typically 14.
  type: boolean?
  inputBinding:
    prefix: -k
- id: in_step_size_index
  doc: 9         is the step size for the index. Typical values are from 1 to 3.
  type: boolean?
  inputBinding:
    prefix: -s
- id: in_sets_number_threads
  doc: 9         sets number of threads to use for indexing.
  type: boolean?
  inputBinding:
    prefix: -t
- id: in_sets_lower_case
  doc: sets lower case masking on. Lower case sequence will not be indexed.
  type: boolean?
  inputBinding:
    prefix: -m
- id: in_sets_bisulphite_indexing
  doc: sets bisulphite indexing and alignment mode for methylation experiments.
  type: boolean?
  inputBinding:
    prefix: -b
- id: in_add_m_tag
  doc: add an M5 tag to sequence headers.
  type: boolean?
  inputBinding:
    prefix: '-5'
- id: in_name_sets_name
  doc: "name      sets the an internal name for the reference sequence index. This\
    \ is\nused in report headers and as the AS: field in SAM SQ record.\nDefaults\
    \ to the indexfile name."
  type: boolean?
  inputBinding:
    prefix: -n
- id: in_novo_index
  doc: -k 99 -s 9 -m indexfile sequencefiles....
  type: string
  inputBinding:
    position: 0
- id: in_index_file
  doc: is the filename for the indexed reference sequence generated by novoindex.
  type: string
  inputBinding:
    position: 0
- id: in_sequence_files
  doc: a list of sequence files in fasta format to be included in the index.
  type: string
  inputBinding:
    position: 1
outputs:
- id: out_stdout
  doc: Standard output stream
  type: stdout
hints: []
cwlVersion: v1.1
baseCommand:
- novoindex
