from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, Int, String

Correct_Py_V0_1_0 = CommandToolBuilder(tool="correct.py", base_command=["correct.py"], inputs=[ToolInput(tag="in_prepend", input_type=Boolean(optional=True), prefix="--prepend", doc=InputDocumentation(doc="Prepend the corrected barcodes and orders to the\noriginal columns.")), ToolInput(tag="in_dist", input_type=Int(optional=True), prefix="--dist", doc=InputDocumentation(doc="NM edit distance threshold. Default: 1")), ToolInput(tag="in_mapq", input_type=Int(optional=True), prefix="--mapq", doc=InputDocumentation(doc="MAPQ threshold. Default: 20")), ToolInput(tag="in_pos", input_type=Int(optional=True), prefix="--pos", doc=InputDocumentation(doc="POS tolerance. Alignments will be ignored if abs(POS -\n1) is greater than this value. Set to greater than the\nbarcode length for no threshold. Default: 2")), ToolInput(tag="in_choose_by", input_type=String(optional=True), prefix="--choose-by", doc=InputDocumentation(doc="Choose the 'correct' barcode in a network of related\nbarcodes by either the count of how many times the\nbarcode was observed ('freq') or how connected the\nbarcode is to the others in the network ('connect').")), ToolInput(tag="in_no_check_ids", input_type=Boolean(optional=True), prefix="--no-check-ids", doc=InputDocumentation(doc="Don't check to make sure read pairs have identical\nids. By default, if this encounters a pair of reads in\nfamilies.tsv with ids that aren't identical (minus an\nending /1 or /2), it will throw an error.")), ToolInput(tag="in_limit", input_type=Int(optional=True), prefix="--limit", doc=InputDocumentation(doc="Limit the number of lines that will be read from each\ninput file, for testing purposes.")), ToolInput(tag="in_structures", input_type=Boolean(optional=True), prefix="--structures", doc=InputDocumentation(doc="Print a list of the unique isoforms")), ToolInput(tag="in_struct_human", input_type=Boolean(optional=True), prefix="--struct-human", doc=InputDocumentation(doc="[networks.png], --visualize [networks.png]\nProduce a visualization of the unique structures and\nwrite the image to this file. If you omit a filename,\nit will be displayed in a window.")), ToolInput(tag="in_viz_format", input_type=String(optional=True), prefix="--viz-format", doc=InputDocumentation(doc="Print log messages to this file instead of to stderr.\nWarning: Will overwrite the file.")), ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="--verbose", doc=InputDocumentation(doc="Print debug messages (very verbose).")), ToolInput(tag="in_phone_home", input_type=Boolean(optional=True), prefix="--phone-home", doc=InputDocumentation(doc="Report helpful usage data to the developer, to better\nunderstand the use cases and performance of the tool.\nThe only data which will be recorded is the name and\nversion of the tool, the size of the input data, the\ntime and memory taken to process it, and the IP\naddress of the machine running it. Also, if the script\nfails, it will report the name of the exception thrown\nand the line of code it occurred in. No parameters or\nfilenames are sent. All the reporting and recording\ncode is available at https://github.com/NickSto/ET.")), ToolInput(tag="in_galaxy", input_type=Boolean(optional=True), prefix="--galaxy", doc=InputDocumentation(doc="Tell the script it's running on Galaxy. Currently this\nonly affects data reported when phoning home.")), ToolInput(tag="in_test", input_type=Boolean(optional=True), prefix="--test", doc=InputDocumentation(doc="If reporting usage data, mark this as a test run.")), ToolInput(tag="in_families", input_type=String(), position=0, doc=InputDocumentation(doc="The sorted output of make-barcodes.awk. The important\npart is that it's a tab-delimited file with at least 2\ncolumns: the barcode sequence and order, and it must\nbe sorted in the same order as the 'reads' in the SAM\nfile.")), ToolInput(tag="in_reads", input_type=String(), position=1, doc=InputDocumentation(doc="The fasta/q file given to the aligner. Used to get\nbarcode sequences from read names.")), ToolInput(tag="in_sam", input_type=String(), position=2, doc=InputDocumentation(doc="Barcode alignment, in SAM format. Omit to read from\nstdin. The read names must be integers, representing\nthe (1-based) order they appear in the families file."))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Correct_Py_V0_1_0().translate("wdl", allow_empty_container=True)

