from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean

_Twofeat_V0_1_0 = CommandToolBuilder(tool="_twofeat", base_command=["_twofeat"], inputs=[ToolInput(tag="in_a_type", input_type=Boolean(optional=True), prefix="-atype", doc=InputDocumentation(doc="string     [*] By default every feature in the feature\ntable is allowed. You can set this to be any\nfeature type you wish to allow.\nSee http://www.ebi.ac.uk/embl/WebFeat/ for a\nlist of the EMBL feature types and see\nAppendix A of the Swissprot user manual in\nhttp://www.expasy.org/sprot/userman.html for\na list of the Swissprot feature types.\nThe type may be wildcarded by using '*'.\nIf you wish to allow more than one type,\nseparate their names with the character '|',\neg:\n*UTR | intron (Any string)")), ToolInput(tag="in_b_type", input_type=Boolean(optional=True), prefix="-btype", doc=InputDocumentation(doc="string     [*] By default every feature in the feature\ntable is allowed. You can set this to be any\nfeature type you wish to allow.\nSee http://www.ebi.ac.uk/embl/WebFeat/ for a\nlist of the EMBL feature types and see\nAppendix A of the Swissprot user manual in\nhttp://www.expasy.org/sprot/userman.html for\na list of the Swissprot feature types.\nThe type may be wildcarded by using '*'.\nIf you wish to allow more than one type,\nseparate their names with the character '|',\neg:\n*UTR | intron (Any string)")), ToolInput(tag="in_min_range", input_type=Boolean(optional=True), prefix="-minrange", doc=InputDocumentation(doc="integer    [0] If this is greater or equal to\n'maxrange', then no min or max range is\nspecified (Any integer value)")), ToolInput(tag="in_max_range", input_type=Boolean(optional=True), prefix="-maxrange", doc=InputDocumentation(doc="integer    [0] If this is less than or equal to\n'minrange', then no min or max range is\nspecified (Any integer value)")), ToolInput(tag="in_a_source", input_type=Boolean(optional=True), prefix="-asource", doc=InputDocumentation(doc="string     [*] By default any feature source in the\nfeature table is allowed. You can set this\nto match any feature source you wish to\nallow.\nThe source name is usually either the name\nof the program that detected the feature or\nit is the feature table (eg: EMBL) that the\nfeature came from.\nThe source may be wildcarded by using '*'.\nIf you wish to allow more than one source,\nseparate their names with the character '|',\neg:\ngene* | embl (Any string)")), ToolInput(tag="in_a_sense", input_type=Boolean(optional=True), prefix="-asense", doc=InputDocumentation(doc="menu       [0] By default any feature sense is allowed.\nYou can set this to match the required\nsense. (Values: 0 (Any sense); + (Forward\nsense); - (Reverse sense))")), ToolInput(tag="in_am_in_score", input_type=Boolean(optional=True), prefix="-aminscore", doc=InputDocumentation(doc="float      [0.0] If this is greater than or equal to\nthe maximum score, then any score is\nallowed. (Any numeric value)")), ToolInput(tag="in_a_max_score", input_type=Boolean(optional=True), prefix="-amaxscore", doc=InputDocumentation(doc="float      [0.0] If this is less than or equal to the\nmaximum score, then any score is permitted.\n(Any numeric value)")), ToolInput(tag="in_a_tag", input_type=Boolean(optional=True), prefix="-atag", doc=InputDocumentation(doc="string     [*] Tags are the types of extra values that\na feature may have. For example in the EMBL\nfeature table, a 'CDS' type of feature may\nhave the tags '/codon', '/codon_start',\n'/db_xref', '/EC_number', '/evidence',\n'/exception', '/function', '/gene',\n'/label', '/map', '/note', '/number',\n'/partial', '/product', '/protein_id',\n'/pseudo', '/standard_name', '/translation',\n'/transl_except', '/transl_table', or\n'/usedin'. Some of these tags also have\nvalues, for example '/gene' can have the\nvalue of the gene name.\nBy default any feature tag in the feature\ntable is allowed. You can set this to match\nany feature tag you wish to allow.\nThe tag may be wildcarded by using '*'.\nIf you wish to allow more than one tag,\nseparate their names with the character '|',\neg:\ngene | label (Any string)")), ToolInput(tag="in_a_value", input_type=Boolean(optional=True), prefix="-avalue", doc=InputDocumentation(doc="string     [*] Tag values are the values associated\nwith a feature tag. Tags are the types of\nextra values that a feature may have. For\nexample in the EMBL feature table, a 'CDS'\ntype of feature may have the tags '/codon',\n'/codon_start', '/db_xref', '/EC_number',\n'/evidence', '/exception', '/function',\n'/gene', '/label', '/map', '/note',\n'/number', '/partial', '/product',\n'/protein_id', '/pseudo', '/standard_name',\n'/translation', '/transl_except',\n'/transl_table', or '/usedin'. Only some of\nthese tags can have values, for example\n'/gene' can have the value of the gene name.\nBy default any feature tag value in the\nfeature table is allowed. You can set this\nto match any feature tag value you wish to\nallow.\nThe tag value may be wildcarded by using\n'*'.\nIf you wish to allow more than one tag\nvalue, separate their names with the\ncharacter '|', eg:\npax* | 10 (Any string)")), ToolInput(tag="in_b_source", input_type=Boolean(optional=True), prefix="-bsource", doc=InputDocumentation(doc="string     [*] By default any feature source in the\nfeature table is allowed. You can set this\nto match any feature source you wish to\nallow.\nThe source name is usually either the name\nof the program that detected the feature or\nit is the feature table (eg: EMBL) that the\nfeature came from.\nThe source may be wildcarded by using '*'.\nIf you wish to allow more than one source,\nseparate their names with the character '|',\neg:\ngene* | embl (Any string)")), ToolInput(tag="in_b_sense", input_type=Boolean(optional=True), prefix="-bsense", doc=InputDocumentation(doc="menu       [0] By default any feature sense is allowed.\nYou can set this to match the required\nsense. (Values: 0 (Any sense); + (Forward\nsense); - (Reverse sense))")), ToolInput(tag="in_bm_in_score", input_type=Boolean(optional=True), prefix="-bminscore", doc=InputDocumentation(doc="float      [0.0] If this is greater than or equal to\nthe maximum score, then any score is\nallowed. (Any numeric value)")), ToolInput(tag="in_bmax_score", input_type=Boolean(optional=True), prefix="-bmaxscore", doc=InputDocumentation(doc="float      [0.0] If this is less than or equal to the\nmaximum score, then any score is permitted.\n(Any numeric value)")), ToolInput(tag="in_b_tag", input_type=Boolean(optional=True), prefix="-btag", doc=InputDocumentation(doc="string     [*] Tags are the types of extra values that\na feature may have. For example in the EMBL\nfeature table, a 'CDS' type of feature may\nhave the tags '/codon', '/codon_start',\n'/db_xref', '/EC_number', '/evidence',\n'/exception', '/function', '/gene',\n'/label', '/map', '/note', '/number',\n'/partial', '/product', '/protein_id',\n'/pseudo', '/standard_name', '/translation',\n'/transl_except', '/transl_table', or\n'/usedin'. Some of these tags also have\nvalues, for example '/gene' can have the\nvalue of the gene name.\nBy default any feature tag in the feature\ntable is allowed. You can set this to match\nany feature tag you wish to allow.\nThe tag may be wildcarded by using '*'.\nIf you wish to allow more than one tag,\nseparate their names with the character '|',\neg:\ngene | label (Any string)")), ToolInput(tag="in_b_value", input_type=Boolean(optional=True), prefix="-bvalue", doc=InputDocumentation(doc="string     [*] Tag values are the values associated\nwith a feature tag. Tags are the types of\nextra values that a feature may have. For\nexample in the EMBL feature table, a 'CDS'\ntype of feature may have the tags '/codon',\n'/codon_start', '/db_xref', '/EC_number',\n'/evidence', '/exception', '/function',\n'/gene', '/label', '/map', '/note',\n'/number', '/partial', '/product',\n'/protein_id', '/pseudo', '/standard_name',\n'/translation', '/transl_except',\n'/transl_table', or '/usedin'. Only some of\nthese tags can have values, for example\n'/gene' can have the value of the gene name.\nBy default any feature tag value in the\nfeature table is allowed. You can set this\nto match any feature tag value you wish to\nallow.\nThe tag value may be wildcarded by using\n'*'.\nIf you wish to allow more than one tag\nvalue, separate their names with the\ncharacter '|', eg:\npax* | 10 (Any string)")), ToolInput(tag="in_overlap", input_type=Boolean(optional=True), prefix="-overlap", doc=InputDocumentation(doc="menu       [A] This allows you to specify the allowed\noverlaps of the features A and B.\nYou can allow any or no overlaps, specify\nthat they must or must not overlap, that one\nmust or must not be wholly enclosed within\nanother feature. (Values: A (Any); O\n(Overlap required); NO (No overlaps are\nallowed); NW (Overlap required but not\nwithin); AW (A must be all within B); BW (B\nmust be all within A))")), ToolInput(tag="in_range_type", input_type=Boolean(optional=True), prefix="-rangetype", doc=InputDocumentation(doc="menu       [N] This allows you to specify the positions\nfrom which the allowed minimum or maximum\ndistance between the features is measured\n(Values: N (From nearest ends); L (From left\nends); R (From right ends); F (From\nfurthest ends))")), ToolInput(tag="in_sense", input_type=Boolean(optional=True), prefix="-sense", doc=InputDocumentation(doc="menu       [A] This allows you to specify the required\nsense that the two features must be on. This\nis ignored (always 'Any') when looking at\nprotein sequence features. (Values: A (Any\nsense); S (Same sense); O (Opposite sense))")), ToolInput(tag="in_order", input_type=Boolean(optional=True), prefix="-order", doc=InputDocumentation(doc="menu       [A] This allows you to specify the required\norder of the two features. The order is\nmeasured from the start positions of the\nfeatures. This criterion is always applied\ndespite the specified overlap type required.\n(Values: A (Any); AB (Feature A then\nfeature B); BA (Feature B then feature A))")), ToolInput(tag="in_two_out", input_type=Boolean(optional=True), prefix="-twoout", doc=InputDocumentation(doc="toggle     [N] If you set this to be true, then the two\nfeatures themselves will be written out. If\nit is left as false, then a single feature\nwill be written out covering the two\nfeatures you found."))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    _Twofeat_V0_1_0().translate("wdl", allow_empty_container=True)

