from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, Int, File, String, Float

Haploview_V0_1_0 = CommandToolBuilder(tool="haploview", base_command=["haploview"], inputs=[ToolInput(tag="in_version", input_type=Boolean(optional=True), prefix="-version", doc=InputDocumentation(doc="Print out the version number")), ToolInput(tag="in_memory", input_type=Int(optional=True), prefix="-memory", doc=InputDocumentation(doc="allocates <memsize> megabytes of memory (default 512)")), ToolInput(tag="in_no_gui", input_type=Boolean(optional=True), prefix="-nogui", doc=InputDocumentation(doc="Command line output only")), ToolInput(tag="in_quiet", input_type=Boolean(optional=True), prefix="-quiet", doc=InputDocumentation(doc="Quiet mode- doesnt print any warnings or information to screen")), ToolInput(tag="in_log", input_type=File(optional=True), prefix="-log", doc=InputDocumentation(doc="Specify a logfile name (defaults to haploview.log if no name specified)")), ToolInput(tag="in_out", input_type=File(optional=True), prefix="-out", doc=InputDocumentation(doc="Specify a fileroot to be used for all output files")), ToolInput(tag="in_ped_file", input_type=File(optional=True), prefix="-pedfile", doc=InputDocumentation(doc="Specify an input file (or http:// location) in pedigree file format")), ToolInput(tag="in_hapmap", input_type=File(optional=True), prefix="-hapmap", doc=InputDocumentation(doc="Specify an input file (or http:// location) in HapMap format")), ToolInput(tag="in_phased_hmp_data", input_type=File(optional=True), prefix="-phasedhmpdata", doc=InputDocumentation(doc="Specify a HapMap PHASE data file (or http:// location)")), ToolInput(tag="in_phased_hmp_sample", input_type=File(optional=True), prefix="-phasedhmpsample", doc=InputDocumentation(doc="Specify a HapMap PHASE sample file (or http:// location)")), ToolInput(tag="in_phased_hmp_legend", input_type=File(optional=True), prefix="-phasedhmplegend", doc=InputDocumentation(doc="Specify a HapMap PHASE legend file (or http:// location)")), ToolInput(tag="in_gzip", input_type=Boolean(optional=True), prefix="-gzip", doc=InputDocumentation(doc="Indicates that phased input files use GZIP compression")), ToolInput(tag="in_hapmap_download", input_type=Boolean(optional=True), prefix="-hapmapDownload", doc=InputDocumentation(doc="Specify a phased HapMap download")), ToolInput(tag="in_haps", input_type=File(optional=True), prefix="-haps", doc=InputDocumentation(doc="Specify an input file (or http:// location) in .haps format")), ToolInput(tag="in_info", input_type=File(optional=True), prefix="-info", doc=InputDocumentation(doc="Specify a marker info file (or http:// location)")), ToolInput(tag="in_plink", input_type=File(optional=True), prefix="-plink", doc=InputDocumentation(doc="Specify a PLINK or other results file (or http:// location)")), ToolInput(tag="in_map", input_type=File(optional=True), prefix="-map", doc=InputDocumentation(doc="Specify a map file or binary map file (or http:// location)")), ToolInput(tag="in_non_snp", input_type=File(optional=True), prefix="-nonSNP", doc=InputDocumentation(doc="Specify that the accompanying PLINK file is non-SNP based output")), ToolInput(tag="in_select_cols", input_type=Boolean(optional=True), prefix="-selectCols", doc=InputDocumentation(doc="Activate the preloading column filter for PLINK loads")), ToolInput(tag="in_batch", input_type=File(optional=True), prefix="-batch", doc=InputDocumentation(doc="Batch mode. Each line in batch file should contain a genotype file\nfollowed by an optional info file, separated by a space.")), ToolInput(tag="in_blocks", input_type=File(optional=True), prefix="-blocks", doc=InputDocumentation(doc="Blocks file (or http:// location), one block per line, will force output for these blocks")), ToolInput(tag="in_track", input_type=File(optional=True), prefix="-track", doc=InputDocumentation(doc="Specify an input analysis track file (or http:// location)")), ToolInput(tag="in_exclude_markers", input_type=Int(optional=True), prefix="-excludeMarkers", doc=InputDocumentation(doc="Specify markers (in range 1-N where N is total number of markers) to be\nskipped for all analyses. Format: 1,2,5..12")), ToolInput(tag="in_skip_check", input_type=Boolean(optional=True), prefix="-skipcheck", doc=InputDocumentation(doc="Skips the various genotype file checks")), ToolInput(tag="in_chromosome", input_type=Boolean(optional=True), prefix="-chromosome", doc=InputDocumentation(doc="<1-22,X,Y>          Specifies the chromosome for this file or download")), ToolInput(tag="in_panel", input_type=Boolean(optional=True), prefix="-panel", doc=InputDocumentation(doc="<CEU,YRI,CHB+JPT>        Specifies the analysis panel for this HapMap download")), ToolInput(tag="in_startpos", input_type=Int(optional=True), prefix="-startpos", doc=InputDocumentation(doc="Specifies the start position in kb for this HapMap download")), ToolInput(tag="in_endpos", input_type=Int(optional=True), prefix="-endpos", doc=InputDocumentation(doc="Specifies the end position in kb for this HapMap download")), ToolInput(tag="in_release", input_type=Boolean(optional=True), prefix="-release", doc=InputDocumentation(doc="<16a,21,22>            Specifies the HapMap phase for this HapMap download (defaults to 21)")), ToolInput(tag="in_d_prime", input_type=Boolean(optional=True), prefix="-dprime", doc=InputDocumentation(doc="Outputs LD text to <fileroot>.LD")), ToolInput(tag="in_png", input_type=Boolean(optional=True), prefix="-png", doc=InputDocumentation(doc="Outputs LD display to <fileroot>.LD.PNG")), ToolInput(tag="in_compressed_png", input_type=Boolean(optional=True), prefix="-compressedpng", doc=InputDocumentation(doc="Outputs compressed LD display to <fileroot>.LD.PNG")), ToolInput(tag="in_svg", input_type=Boolean(optional=True), prefix="-svg", doc=InputDocumentation(doc="Outputs svg format LD display to <fileroot>.LD.SVG")), ToolInput(tag="in_info_track", input_type=Boolean(optional=True), prefix="-infoTrack", doc=InputDocumentation(doc="Downloads and displays HapMap info track on PNG image output")), ToolInput(tag="in_ld_color_scheme", input_type=String(optional=True), prefix="-ldcolorscheme", doc=InputDocumentation(doc="Specify an LD color scheme. <argument> should be one of:\nDEFAULT, RSQ, DPALT, GAB, GAM")), ToolInput(tag="in_ld_values", input_type=Boolean(optional=True), prefix="-ldvalues", doc=InputDocumentation(doc="<DPRIME,RSQ,NONE>     Specify what to print in LD image output. default is DPrime")), ToolInput(tag="in_check", input_type=Boolean(optional=True), prefix="-check", doc=InputDocumentation(doc="Outputs marker checks to <fileroot>.CHECK\nnote: -dprime  and -check default to no blocks output.\nUse -blockoutput to also output blocks")), ToolInput(tag="in_ind_check", input_type=Boolean(optional=True), prefix="-indcheck", doc=InputDocumentation(doc="Outputs genotype percent per individual to <fileroot>.INDCHECK")), ToolInput(tag="in_mendel", input_type=Boolean(optional=True), prefix="-mendel", doc=InputDocumentation(doc="Outputs Mendel error information to <fileroot>.MENDEL")), ToolInput(tag="in_male_he_ts", input_type=Boolean(optional=True), prefix="-malehets", doc=InputDocumentation(doc="Outputs male heterozygote information to <fileroot>.MALEHETS")), ToolInput(tag="in_block_output", input_type=Boolean(optional=True), prefix="-blockoutput", doc=InputDocumentation(doc="<GAB,GAM,SPI,ALL>  Output type. Gabriel, 4 gamete, spine output or all 3. default is Gabriel.")), ToolInput(tag="in_block_cut_high_ci", input_type=String(optional=True), prefix="-blockCutHighCI", doc=InputDocumentation(doc="Gabriel 'Strong LD' high confidence interval D' cutoff.")), ToolInput(tag="in_block_cut_low_ci", input_type=String(optional=True), prefix="-blockCutLowCI", doc=InputDocumentation(doc="Gabriel 'Strong LD' low confidence interval D' cutoff.")), ToolInput(tag="in_block_maf_thresh", input_type=String(optional=True), prefix="-blockMAFThresh", doc=InputDocumentation(doc="Gabriel MAF threshold.")), ToolInput(tag="in_block_rec_high_ci", input_type=String(optional=True), prefix="-blockRecHighCI", doc=InputDocumentation(doc="Gabriel recombination high confidence interval D' cutoff.")), ToolInput(tag="in_block_inform_frac", input_type=String(optional=True), prefix="-blockInformFrac", doc=InputDocumentation(doc="Gabriel fraction of informative markers required to be in LD.")), ToolInput(tag="in_block_four_gam_cut", input_type=Int(optional=True), prefix="-block4GamCut", doc=InputDocumentation(doc="4 Gamete block cutoff for frequency of 4th pairwise haplotype.")), ToolInput(tag="in_blocks_pine_dp", input_type=String(optional=True), prefix="-blockSpineDP", doc=InputDocumentation(doc="Solid Spine blocks D' cutoff for 'Strong LD")), ToolInput(tag="in_max_distance", input_type=Int(optional=True), prefix="-maxdistance", doc=InputDocumentation(doc="Maximum comparison distance in kilobases (integer). Default is 500")), ToolInput(tag="in_hap_thresh", input_type=String(optional=True), prefix="-hapthresh", doc=InputDocumentation(doc="Only output haps with at least this frequency")), ToolInput(tag="in_spacing", input_type=Int(optional=True), prefix="-spacing", doc=InputDocumentation(doc="Proportional spacing of markers in LD display. <threshold> is a value\nbetween 0 (no spacing) and 1 (max spacing). Default is 0")), ToolInput(tag="in_min_maf", input_type=Float(optional=True), prefix="-minMAF", doc=InputDocumentation(doc="Minimum minor allele frequency to include a marker. <threshold> is a value\nbetween 0 and 0.5. Default is .001")), ToolInput(tag="in_max_mendel", input_type=Int(optional=True), prefix="-maxMendel", doc=InputDocumentation(doc="Markers with more than <integer> Mendel errors will be excluded. Default is 1.")), ToolInput(tag="in_minge_no", input_type=Float(optional=True), prefix="-minGeno", doc=InputDocumentation(doc="Exclude markers with less than <threshold> valid data. <threshold> is a value\nbetween 0 and 1. Default is .75")), ToolInput(tag="in_hw_cut_off", input_type=Float(optional=True), prefix="-hwcutoff", doc=InputDocumentation(doc="Exclude markers with a HW p-value smaller than <threshold>. <threshold> is a value\nbetween 0 and 1. Default is .001")), ToolInput(tag="in_missing_cut_off", input_type=Float(optional=True), prefix="-missingCutoff", doc=InputDocumentation(doc="Exclude individuals with more than <threshold> fraction missing data.\n<threshold> is a value between 0 and 1. Default is .5")), ToolInput(tag="in_assoc_cc", input_type=Boolean(optional=True), prefix="-assocCC", doc=InputDocumentation(doc="Outputs case control association results to <fileroot>.ASSOC and <fileroot>.HAPASSOC")), ToolInput(tag="in_assoc_tdt", input_type=Boolean(optional=True), prefix="-assocTDT", doc=InputDocumentation(doc="Outputs trio association results to <fileroot>.ASSOC and <fileroot>.HAPASSOC")), ToolInput(tag="in_custom_assoc", input_type=File(optional=True), prefix="-customAssoc", doc=InputDocumentation(doc="Loads a set of custom tests for association.")), ToolInput(tag="in_perm_tests", input_type=Int(optional=True), prefix="-permtests", doc=InputDocumentation(doc="Performs <numtests> permutations on default association tests (or custom tests\nif a custom association file is specified) and writes to <fileroot>.PERMUT")), ToolInput(tag="in_pairwise_tagging", input_type=Boolean(optional=True), prefix="-pairwiseTagging", doc=InputDocumentation(doc="Generates pairwise tagging information in <fileroot>.TAGS and .TESTS")), ToolInput(tag="in_aggressive_tagging", input_type=Boolean(optional=True), prefix="-aggressiveTagging", doc=InputDocumentation(doc="As above but generates 2-marker haplotype tags unless specified otherwise by -aggressiveNumMarkers")), ToolInput(tag="in_aggressive_num_markers", input_type=Boolean(optional=True), prefix="-aggressiveNumMarkers", doc=InputDocumentation(doc="<2,3>     Specifies whether to use 2-marker haplotype tags or 2 and 3-marker haplotype tags.")), ToolInput(tag="in_max_num_tags", input_type=String(optional=True), prefix="-maxNumTags", doc=InputDocumentation(doc="Only selects <n> best tags.")), ToolInput(tag="in_dont_add_tags", input_type=Boolean(optional=True), prefix="-dontaddtags", doc=InputDocumentation(doc="Only uses forced in tags.")), ToolInput(tag="in_include_tags", input_type=String(optional=True), prefix="-includeTags", doc=InputDocumentation(doc="Forces in a comma separated list of marker names as tags.")), ToolInput(tag="in_include_tags_file", input_type=File(optional=True), prefix="-includeTagsFile", doc=InputDocumentation(doc="Forces in a file (or http:// location) of one marker name per line as tags.")), ToolInput(tag="in_exclude_tags", input_type=String(optional=True), prefix="-excludeTags", doc=InputDocumentation(doc="Excludes a comma separated list of marker names from being used as tags.")), ToolInput(tag="in_exclude_tags_file", input_type=File(optional=True), prefix="-excludeTagsFile", doc=InputDocumentation(doc="Excludes a file (or http:// location) of one marker name per line from being used as tags.")), ToolInput(tag="in_capture_alleles", input_type=File(optional=True), prefix="-captureAlleles", doc=InputDocumentation(doc="Capture only the alleles contained in a file (or http:// location) of one marker name per line.")), ToolInput(tag="in_design_scores", input_type=File(optional=True), prefix="-designScores", doc=InputDocumentation(doc="Specify design scores in a file (or http:// location) of one marker name and one score per line")), ToolInput(tag="in_min_design_scores", input_type=String(optional=True), prefix="-mindesignscores", doc=InputDocumentation(doc="Specify a minimum design score threshold.")), ToolInput(tag="in_min_tag_distance", input_type=String(optional=True), prefix="-mintagdistance", doc=InputDocumentation(doc="Specify a Minimum distance in bases between picked tags.")), ToolInput(tag="in_tag_lod_cut_off", input_type=String(optional=True), prefix="-taglodcutoff", doc=InputDocumentation(doc="Tagger LOD cutoff for creating multimarker tag haplotypes.")), ToolInput(tag="in_tag_rsq_cut_off", input_type=Int(optional=True), prefix="-tagrsqcutoff", doc=InputDocumentation(doc="Tagger r^2 cutoff."))], outputs=[ToolOutput(tag="out_non_snp", output_type=File(optional=True), selector=InputSelector(input_to_select="in_non_snp", type_hint=File()), doc=OutputDocumentation(doc="Specify that the accompanying PLINK file is non-SNP based output")), ToolOutput(tag="out_blocks", output_type=File(optional=True), selector=InputSelector(input_to_select="in_blocks", type_hint=File()), doc=OutputDocumentation(doc="Blocks file (or http:// location), one block per line, will force output for these blocks"))], container="quay.io/biocontainers/haploview:4.2--0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Haploview_V0_1_0().translate("wdl")

