from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Directory, Float, Int, Boolean, String

Refinem_Ssu_Erroneous_V0_1_0 = CommandToolBuilder(tool="refinem_ssu_erroneous", base_command=["refinem", "ssu_erroneous"], inputs=[ToolInput(tag="in_genome_ext", input_type=Directory(optional=True), prefix="--genome_ext", doc=InputDocumentation(doc="extension of genomes (other files in directory are\nignored) (default: fna)")), ToolInput(tag="in_evalue", input_type=Float(optional=True), prefix="--evalue", doc=InputDocumentation(doc="e-value threshold for identifying and classifying 16S\nrRNA genes (default: 1e-05)")), ToolInput(tag="in_concatenate", input_type=Int(optional=True), prefix="--concatenate", doc=InputDocumentation(doc="concatenate hits within the specified number of base\npairs (default: 200)")), ToolInput(tag="in_common_tax_on", input_type=Float(optional=True), prefix="--common_taxon", doc=InputDocumentation(doc="threshold for defining a taxon as common (default:\n10.0)")), ToolInput(tag="in_ssu_min_len", input_type=Int(optional=True), prefix="--ssu_min_len", doc=InputDocumentation(doc="minimum length of SSU 16S gene fragment to consider\nfor classification (default: 600)")), ToolInput(tag="in_ssu_domain", input_type=Float(optional=True), prefix="--ssu_domain", doc=InputDocumentation(doc="percent identity threshold for accepting domain\nclassification of SSU (default: 83.68)")), ToolInput(tag="in_ssu_phylum", input_type=Float(optional=True), prefix="--ssu_phylum", doc=InputDocumentation(doc="percent identity threshold for accepting phylum\nclassification of SSU (default: 86.35)")), ToolInput(tag="in_ssu_class", input_type=Float(optional=True), prefix="--ssu_class", doc=InputDocumentation(doc="percent identity threshold for accepting class\nclassification of SSU (default: 89.2)")), ToolInput(tag="in_ssu_order", input_type=Float(optional=True), prefix="--ssu_order", doc=InputDocumentation(doc="percent identity threshold for accepting order\nclassification of SSU (default: 92.25)")), ToolInput(tag="in_ssu_family", input_type=Float(optional=True), prefix="--ssu_family", doc=InputDocumentation(doc="percent identity threshold for accepting family\nclassification of SSU (default: 96.4)")), ToolInput(tag="in_ssu_genus", input_type=Float(optional=True), prefix="--ssu_genus", doc=InputDocumentation(doc="percent identity threshold for accepting genus\nclassification of SSU (default: 98.7)")), ToolInput(tag="in_cpus", input_type=Int(optional=True), prefix="--cpus", doc=InputDocumentation(doc="number of CPUs to use (default: 1)")), ToolInput(tag="in_silent", input_type=Boolean(optional=True), prefix="--silent", doc=InputDocumentation(doc="suppress output of logger")), ToolInput(tag="in_genome_nt_dir", input_type=String(), position=0, doc=InputDocumentation(doc="directory containing nucleotide scaffolds for each")), ToolInput(tag="in_genome", input_type=String(), position=1, doc=InputDocumentation(doc="taxon_profile_dir     directory with results of taxon_profile command")), ToolInput(tag="in_ssu_db", input_type=String(), position=2, doc=InputDocumentation(doc="BLAST database of 16S rRNA genes")), ToolInput(tag="in_ssu_taxonomy_file", input_type=String(), position=3, doc=InputDocumentation(doc="taxonomy file for genes in the 16S rRNA database")), ToolInput(tag="in_output_dir", input_type=String(), position=4, doc=InputDocumentation(doc="output directory"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Refinem_Ssu_Erroneous_V0_1_0().translate("wdl", allow_empty_container=True)

