from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Boolean, Int, String, Array

Magpurify_Phylo_Markers_V0_1_0 = CommandToolBuilder(tool="magpurify_phylo_markers", base_command=["magpurify", "phylo-markers"], inputs=[ToolInput(tag="in_db", input_type=File(optional=True), prefix="--db", doc=InputDocumentation(doc="Path to reference database. By default, the\nMAGPURIFYDB environmental variable is used (default:\nNone)")), ToolInput(tag="in_continue", input_type=Boolean(optional=True), prefix="--continue", doc=InputDocumentation(doc="Go straight to quality estimation and skip all\nprevious steps (default: False)")), ToolInput(tag="in_max_target_seqs", input_type=Int(optional=True), prefix="--max_target_seqs", doc=InputDocumentation(doc="Maximum number of targets reported in BLAST table\n(default: 1)")), ToolInput(tag="in_cut_off_type", input_type=String(optional=True), prefix="--cutoff_type", doc=InputDocumentation(doc="Use strict or sensitive %ID cutoff for taxonomically\nannotating genes (default: strict)")), ToolInput(tag="in_seq_type", input_type=String(optional=True), prefix="--seq_type", doc=InputDocumentation(doc="Choose to search genes versus DNA or protein database\n(default: protein)")), ToolInput(tag="in_hit_type", input_type=String(optional=True), prefix="--hit_type", doc=InputDocumentation(doc="Transfer taxonomy of all hits or top hit per gene\n(default: top_hit)")), ToolInput(tag="in_exclude_clades", input_type=Array(t=String(), optional=True), prefix="--exclude_clades", doc=InputDocumentation(doc="List of clades to exclude (ex: s__1300164.4) (default:\nNone)")), ToolInput(tag="in_bin_fract", input_type=Int(optional=True), prefix="--bin_fract", doc=InputDocumentation(doc="Min fraction of genes in bin that agree with consensus\ntaxonomy for bin annotation (default: 0.7)")), ToolInput(tag="in_contig_fract", input_type=Int(optional=True), prefix="--contig_fract", doc=InputDocumentation(doc="Min fraction of genes in that disagree with bin\ntaxonomy for filtering (default: 1.0)")), ToolInput(tag="in_allow_no_class", input_type=Boolean(optional=True), prefix="--allow_noclass", doc=InputDocumentation(doc="Allow a bin to be unclassfied and flag any classified\ncontigs (default: False)")), ToolInput(tag="in_threads", input_type=Int(optional=True), prefix="--threads", doc=InputDocumentation(doc="Number of CPUs to use (default: 1)")), ToolInput(tag="in_fna", input_type=String(), position=0, doc=InputDocumentation(doc="Path to input genome in FASTA format")), ToolInput(tag="in_out", input_type=String(), position=1, doc=InputDocumentation(doc="Output directory to store results and intermediate")), ToolInput(tag="in_files", input_type=String(), position=2, doc=InputDocumentation(doc="optional arguments:"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Magpurify_Phylo_Markers_V0_1_0().translate("wdl", allow_empty_container=True)

