from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Int, Float, String, Boolean, File

Igdiscover_Germlinefilter_V0_1_0 = CommandToolBuilder(tool="igdiscover_germlinefilter", base_command=["igdiscover", "germlinefilter"], inputs=[ToolInput(tag="in_cluster_size", input_type=Int(optional=True), prefix="--cluster-size", doc=InputDocumentation(doc="Consensus must represent at least N sequences.\nDefault: 0")), ToolInput(tag="in_cross_mapping_ratio", input_type=Float(optional=True), prefix="--cross-mapping-ratio", doc=InputDocumentation(doc="Ratio for detection of cross-mapping artifacts.\nDefault: 0.02")), ToolInput(tag="in_clo_no_type_ratio", input_type=Float(optional=True), prefix="--clonotype-ratio", doc=InputDocumentation(doc="Required ratio of 'clonotypes' counts between alleles.\nWorks only for genes named 'NAME*ALLELE'. Default: 0.1")), ToolInput(tag="in_exact_ratio", input_type=Float(optional=True), prefix="--exact-ratio", doc=InputDocumentation(doc="Required ratio of 'exact' counts between alleles.\nWorks only for genes named 'NAME*ALLELE'. Default: 0.1")), ToolInput(tag="in_cdr_three_shared_ratio", input_type=Int(optional=True), prefix="--cdr3-shared-ratio", doc=InputDocumentation(doc="Maximum allowed CDR3_shared_ratio. Default: 1.0")), ToolInput(tag="in_minimum_db_diff", input_type=Int(optional=True), prefix="--minimum-db-diff", doc=InputDocumentation(doc="Sequences must have at least N differences to the\ndatabase sequence. Default: 0")), ToolInput(tag="in_maximum_n", input_type=Int(optional=True), prefix="--maximum-N", doc=InputDocumentation(doc="Sequences must have at most COUNT 'N' bases. Default:\n0")), ToolInput(tag="in_unique_cdr_threes", input_type=Int(optional=True), prefix="--unique-cdr3s", doc=InputDocumentation(doc="Sequences must have at least N unique CDR3s within\nexact sequence matches. Default: 1")), ToolInput(tag="in_unique_j", input_type=Int(optional=True), prefix="--unique-J", doc=InputDocumentation(doc="Sequences must have at least N unique Js within exact\nsequence matches. Default: 0")), ToolInput(tag="in_unique_d_ratio", input_type=String(optional=True), prefix="--unique-D-ratio", doc=InputDocumentation(doc="Discard a sequence if another allele of this gene\nexists such that the ratio between their Ds_exact is\nless than RATIO")), ToolInput(tag="in_unique_d_threshold", input_type=String(optional=True), prefix="--unique-D-threshold", doc=InputDocumentation(doc="Apply the --unique-D-ratio filter only if the Ds_exact\nof the other allele is at least THRESHOLD")), ToolInput(tag="in_allow_stop", input_type=Boolean(optional=True), prefix="--allow-stop", doc=InputDocumentation(doc="Allow stop codons in sequences (uses the has_stop\ncolumn).Default: Do not allow stop codons.")), ToolInput(tag="in_whitelist", input_type=String(optional=True), prefix="--whitelist", doc=InputDocumentation(doc="Sequences that are never discarded or merged with\nothers, even if criteria for discarding them would\napply (except cross-mapping artifact removal, which is\nalways performed).")), ToolInput(tag="in_fast_a", input_type=File(optional=True), prefix="--fasta", doc=InputDocumentation(doc="Write new database in FASTA format to FILE")), ToolInput(tag="in_annotate", input_type=File(optional=True), prefix="--annotate", doc=InputDocumentation(doc="Write candidates.tab with filter annotations to FILE")), ToolInput(tag="in_candidates_dot_tab", input_type=String(), position=0, doc=InputDocumentation(doc="Tables (one or more) created by the 'discover' command"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Igdiscover_Germlinefilter_V0_1_0().translate("wdl", allow_empty_container=True)

