from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, File, Int, Float

Cfsan_Snp_Pipeline_Call_Consensus_V0_1_0 = CommandToolBuilder(tool="cfsan_snp_pipeline_call_consensus", base_command=["cfsan_snp_pipeline", "call_consensus"], inputs=[ToolInput(tag="in_force", input_type=Boolean(optional=True), prefix="--force", doc=InputDocumentation(doc="Force processing even when result file already exists\nand is newer than inputs. (default: False)")), ToolInput(tag="in_snp_list_file", input_type=File(optional=True), prefix="--snpListFile", doc=InputDocumentation(doc="Relative or absolute path to the SNP list file across\nall samples. (default: snplist.txt)")), ToolInput(tag="in_exclude_file", input_type=File(optional=True), prefix="--excludeFile", doc=InputDocumentation(doc="VCF file of positions to exclude. (default: None)")), ToolInput(tag="in_output", input_type=File(optional=True), prefix="--output", doc=InputDocumentation(doc="Output file. Relative or absolute path to the\nconsensus fasta file for this sample. (default:\nconsensus.fasta)")), ToolInput(tag="in_min_base_qual", input_type=Int(optional=True), prefix="--minBaseQual", doc=InputDocumentation(doc="Mimimum base quality score to count a read. All other\nsnp filters take effect after the low-quality reads\nare discarded. (default: 0)")), ToolInput(tag="in_min_cons_freq", input_type=Float(optional=True), prefix="--minConsFreq", doc=InputDocumentation(doc="Consensus frequency. Mimimum fraction of high-quality\nreads supporting the consensus to make a call.\n(default: 0.6)")), ToolInput(tag="in_min_cons_dpth", input_type=Int(optional=True), prefix="--minConsDpth", doc=InputDocumentation(doc="Consensus depth. Minimum number of high-quality reads\nsupporting the consensus to make a call. (default: 1)")), ToolInput(tag="in_min_cons_str_ddp_th", input_type=Int(optional=True), prefix="--minConsStrdDpth", doc=InputDocumentation(doc="Consensus strand depth. Minimum number of high-quality\nreads supporting the consensus which must be present\non both the forward and reverse strands to make a\ncall. (default: 0)")), ToolInput(tag="in_min_cons_st_rd_bias", input_type=Int(optional=True), prefix="--minConsStrdBias", doc=InputDocumentation(doc="Strand bias. Minimum fraction of the high-quality\nconsensus-supporting reads which must be present on\nboth the forward and reverse strands to make a call.\nThe numerator of this fraction is the number of high-\nquality consensus-supporting reads on one strand. The\ndenominator is the total number of high-quality\nconsensus-supporting reads on both strands combined.\n(default: 0)")), ToolInput(tag="in_vcf_filename", input_type=File(optional=True), prefix="--vcfFileName", doc=InputDocumentation(doc="VCF Output file name. If specified, a VCF file with\nthis file name will be created in the same directory\nas the consensus fasta file for this sample. (default:\nNone)")), ToolInput(tag="in_vcf_refname", input_type=File(optional=True), prefix="--vcfRefName", doc=InputDocumentation(doc="Name of the reference file. This is only used in the\ngenerated VCF file header. (default: Unknown\nreference)")), ToolInput(tag="in_vcf_all_pos", input_type=Boolean(optional=True), prefix="--vcfAllPos", doc=InputDocumentation(doc="Flag to cause VCF file generation at all positions,\nnot just the snp positions. This has no effect on the\nconsensus fasta file, it only affects the VCF file.\nThis capability is intended primarily as a diagnostic\ntool and enabling this flag will greatly increase\nexecution time. (default: False)")), ToolInput(tag="in_vcf_preserve_ref_case", input_type=Boolean(optional=True), prefix="--vcfPreserveRefCase", doc=InputDocumentation(doc="Flag to cause the VCF file generator to emit each\nreference base in uppercase/lowercase as it appears in\nthe reference sequence file. If not specified, the\nreference base is emitted in uppercase. (default:\nFalse)")), ToolInput(tag="in_vcf_failed_snp_gt", input_type=Boolean(optional=True), prefix="--vcfFailedSnpGt", doc=InputDocumentation(doc="{.,0,1}\nControls the VCF file GT data element when a snp fails\nfilters. Possible values: .) The GT element will be a\ndot, indicating unable to make a call. 0) The GT\nelement will be 0, indicating the reference base. 1)\nThe GT element will be the ALT index of the most\ncommonly occuring base, usually 1. (default: .)")), ToolInput(tag="in_verbose", input_type=Int(optional=True), prefix="--verbose", doc=InputDocumentation(doc="Verbose message level (0=no info, 5=lots) (default: 1)"))], outputs=[ToolOutput(tag="out_output", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output", type_hint=File()), doc=OutputDocumentation(doc="Output file. Relative or absolute path to the\nconsensus fasta file for this sample. (default:\nconsensus.fasta)")), ToolOutput(tag="out_vcf_filename", output_type=File(optional=True), selector=InputSelector(input_to_select="in_vcf_filename", type_hint=File()), doc=OutputDocumentation(doc="VCF Output file name. If specified, a VCF file with\nthis file name will be created in the same directory\nas the consensus fasta file for this sample. (default:\nNone)"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Cfsan_Snp_Pipeline_Call_Consensus_V0_1_0().translate("wdl", allow_empty_container=True)

