from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import String, File, Boolean, Float, Int, Directory

Checkm_Qa_V0_1_0 = CommandToolBuilder(tool="checkm_qa", base_command=["checkm", "qa"], inputs=[ToolInput(tag="in_out_format", input_type=String(optional=True), prefix="--out_format", doc=InputDocumentation(doc="desired output: (default: 1)\n1. summary of bin completeness and contamination\n2. extended summary of bin statistics (includes GC, genome size, ...)\n3. summary of bin quality for increasingly basal lineage-specific marker sets\n4. list of marker genes and their counts\n5. list of bin id, marker gene id, gene id\n6. list of marker genes present multiple times in a bin\n7. list of marker genes present multiple times on the same scaffold\n8. list indicating position of each marker gene within a bin\n9. FASTA file of marker genes identified in each bin")), ToolInput(tag="in_exclude_markers", input_type=File(optional=True), prefix="--exclude_markers", doc=InputDocumentation(doc="file specifying markers to exclude from marker sets")), ToolInput(tag="in_individual_markers", input_type=Boolean(optional=True), prefix="--individual_markers", doc=InputDocumentation(doc="treat marker as independent (i.e., ignore co-located set structure)")), ToolInput(tag="in_skip_adj_correction", input_type=Boolean(optional=True), prefix="--skip_adj_correction", doc=InputDocumentation(doc="do not exclude adjacent marker genes when estimating contamination")), ToolInput(tag="in_skip_pseudogene_correction", input_type=Boolean(optional=True), prefix="--skip_pseudogene_correction", doc=InputDocumentation(doc="skip identification and filtering of pseudogenes")), ToolInput(tag="in_aai_strain", input_type=Float(optional=True), prefix="--aai_strain", doc=InputDocumentation(doc="AAI threshold used to identify strain heterogeneity (default: 0.9)")), ToolInput(tag="in_alignment_file", input_type=File(optional=True), prefix="--alignment_file", doc=InputDocumentation(doc="produce file showing alignment of multi-copy genes and their AAI identity")), ToolInput(tag="in_ignore_thresholds", input_type=Boolean(optional=True), prefix="--ignore_thresholds", doc=InputDocumentation(doc="ignore model-specific score thresholds")), ToolInput(tag="in_e_value", input_type=Float(optional=True), prefix="--e_value", doc=InputDocumentation(doc="e-value cut off (default: 1e-10)")), ToolInput(tag="in_length", input_type=Int(optional=True), prefix="--length", doc=InputDocumentation(doc="percent overlap between target and query (default: 0.7)")), ToolInput(tag="in_coverage_file", input_type=File(optional=True), prefix="--coverage_file", doc=InputDocumentation(doc="file containing coverage of each sequence; coverage information added to table type 2 (see coverage command)")), ToolInput(tag="in_file", input_type=File(optional=True), prefix="--file", doc=InputDocumentation(doc="print results to file (default: stdout)")), ToolInput(tag="in_tab_table", input_type=Boolean(optional=True), prefix="--tab_table", doc=InputDocumentation(doc="print tab-separated values table")), ToolInput(tag="in_threads", input_type=Int(optional=True), prefix="--threads", doc=InputDocumentation(doc="number of threads (default: 1)")), ToolInput(tag="in_quiet", input_type=Boolean(optional=True), prefix="--quiet", doc=InputDocumentation(doc="suppress console output")), ToolInput(tag="in_tmpdir", input_type=Directory(optional=True), prefix="--tmpdir", doc=InputDocumentation(doc="specify an alternative directory for temporary files")), ToolInput(tag="in_marker_file", input_type=String(), position=0, doc=InputDocumentation(doc="marker file specified during analyze command")), ToolInput(tag="in_analyze_dir", input_type=String(), position=1, doc=InputDocumentation(doc="directory specified during analyze command"))], outputs=[], container="quay.io/biocontainers/checkm-genome:1.1.3--py_1", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Checkm_Qa_V0_1_0().translate("wdl")

