from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, Float, String, Int, File

Clipandmerge_V0_1_0 = CommandToolBuilder(tool="ClipAndMerge", base_command=["ClipAndMerge"], inputs=[ToolInput(tag="in_discard_bad_reads", input_type=Boolean(optional=True), prefix="-discardBadReads", doc=InputDocumentation(doc=": Discard reads after merging that do not fulfill the quality criteria. (default:\nfalse)")), ToolInput(tag="in_error_rate_merging", input_type=Float(optional=True), prefix="-e", doc=InputDocumentation(doc=": Error rate for merging forward and reverse reads. A value of 0.05 means that 5%\nmismatches are allowed in the overlap region. (default: 0.05)")), ToolInput(tag="in_forward_reads_agatcggaagagcacacgtctgaactccagtcac", input_type=String(optional=True), prefix="-f", doc=InputDocumentation(doc=": Forward reads adapter sequence. (default: AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC)")), ToolInput(tag="in_in_one", input_type=Int(optional=True), prefix="-in1", doc=InputDocumentation(doc="[]                       : Forward reads input file(s) in fastq(.gz) file format.")), ToolInput(tag="in_in_two", input_type=Int(optional=True), prefix="-in2", doc=InputDocumentation(doc="[]                       : Reverse reads input file(s) in fastq(.gz) file format.")), ToolInput(tag="in_discard_sequences_shorter", input_type=Int(optional=True), prefix="-l", doc=InputDocumentation(doc=": Discard sequences shorter than this number of nucleotides after adapter\nclipping. (default: 25)")), ToolInput(tag="in_last_base", input_type=Int(optional=True), prefix="-lastBase", doc=InputDocumentation(doc=": Reads are trimmed from the 3' end until given value is reached. Trimming is not\nperformed if read is already <= given value. If this option is given the\n'-trim3p' option is disregarded! Given value sould be 1-based! (default:\n2147483647)")), ToolInput(tag="in_log", input_type=File(optional=True), prefix="-log", doc=InputDocumentation(doc=": Write log messages to a file instead of the standard error stream.")), ToolInput(tag="in_require_alignment_length", input_type=Int(optional=True), prefix="-m", doc=InputDocumentation(doc=": Require a minimum adapter alignment length. If less nucleotides align with the\nadapter, the sequences are not clipped. (default: 8)")), ToolInput(tag="in_max_parallel_reads", input_type=Int(optional=True), prefix="-maxParallelReads", doc=InputDocumentation(doc=": Maximal number of reads, that can be processed in parallel. This number largely\ndepends on the processing system settings! Only change it if you know what you\nare doing! (default: 1000)")), ToolInput(tag="in_min_qual_bad_reads", input_type=Int(optional=True), prefix="-minQualBadReads", doc=InputDocumentation(doc=": Minimal base quality for keeping bad reads. If 0 is specified, then all reads\nare kept. (default: 0)")), ToolInput(tag="in_discard_sequences_is", input_type=Boolean(optional=True), prefix="-n", doc=InputDocumentation(doc=": Discard sequences with unknown (N) nucleotides. Default is to keep such\nsequences. (default: false)")), ToolInput(tag="in_no_clip_stats", input_type=Boolean(optional=True), prefix="-no_clip_stats", doc=InputDocumentation(doc=": Disable the display of clipping statistics. (default: false)")), ToolInput(tag="in_no_clipping", input_type=Boolean(optional=True), prefix="-no_clipping", doc=InputDocumentation(doc=": Skip adapter clipping. Only read merging is performed! (This is only recommended\nif every forward and reverse read has a corresponding partner in the other\nrespective fastq-file! Otherwise merging can not be performed correctly.\n(default: false)")), ToolInput(tag="in_no_merging", input_type=Boolean(optional=True), prefix="-no_merging", doc=InputDocumentation(doc=": Skip read merging for paired-end sequencing data! Only adapter clipping is\nperformed. This parameter is not needed for single-end data. (default: false)")), ToolInput(tag="in_no_qb_mm", input_type=Boolean(optional=True), prefix="-no_qbMM", doc=InputDocumentation(doc=": Do not perform quality based mismatch calculation for merging. Default is to\ntake quality scores into account. (default: false)")), ToolInput(tag="in_output_file_file", input_type=File(optional=True), prefix="-o", doc=InputDocumentation(doc=": Output file. If no file is provided, output will be written to System.out. If\nfile ends with '.gz', output will be gzipped.")), ToolInput(tag="in_minimal_number_nucleotides", input_type=Int(optional=True), prefix="-p", doc=InputDocumentation(doc=": Minimal number of nucleotides that have to overlap in order to merge the forward\nand reverse read. (default: 10)")), ToolInput(tag="in_minimum_base_quality", input_type=Int(optional=True), prefix="-q", doc=InputDocumentation(doc=": Minimum base quality for quality trimming. (default: 20)")), ToolInput(tag="in_qo", input_type=Int(optional=True), prefix="-qo", doc=InputDocumentation(doc=": Phred Score offset. (default: 33)")), ToolInput(tag="in_qt", input_type=Boolean(optional=True), prefix="-qt", doc=InputDocumentation(doc=": Enable quality trimming for non-merged reads. (default: true)")), ToolInput(tag="in_qual_freq_bad_reads", input_type=Float(optional=True), prefix="-qualFreqBadReads", doc=InputDocumentation(doc=": Percentage of reads that have to fulfill minimal base quality criterion.\n(default: 0.9)")), ToolInput(tag="in_reverse_reads_default", input_type=String(optional=True), prefix="-r", doc=InputDocumentation(doc=": Reverse reads adapter sequence. (default: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTA)")), ToolInput(tag="in_rm_no_partner", input_type=Boolean(optional=True), prefix="-rm_no_partner", doc=InputDocumentation(doc=": Remove reads with no pairing partner after adapter clipping. (default: false)")), ToolInput(tag="in_time_estimation", input_type=Boolean(optional=True), prefix="-timeEstimation", doc=InputDocumentation(doc=": Perform remaining time estimation. Note: this can take long for large gzipped\ninput files. (default: false)")), ToolInput(tag="in_trim_three_p", input_type=Int(optional=True), prefix="-trim3p", doc=InputDocumentation(doc=": Trim N nucleotides from the 3' end of each read. This step is performed after\nadapter clipping. Reverse reads are not reverse trancriped before trimming.\n(default: 0)")), ToolInput(tag="in_trim_five_p", input_type=Int(optional=True), prefix="-trim5p", doc=InputDocumentation(doc=": Trim N nucleotides from the 5' end of each read. This step is performed after\nadapter clipping. Reverse reads are not reverse transcriped before trimming.\n(default: 0)")), ToolInput(tag="in_reversefile_write_unmerged", input_type=File(optional=True), prefix="-u", doc=InputDocumentation(doc="REVERSE_FILE        : Write unmerged forward and reverse reads to extra files. Unmerged forward reads\nare written to the file 'FORWARD_FILE'. Unmerged reverse reads are written to\nthe file 'REVERSE_FILE', i.e. the regular output file then only contains merged\nreads!\nAttention: If the option '-rm_no_partner' is not selected the two given output\nfiles also contain forward/reverse reads with no pairing partner!\nIf filenames end with '.gz' gzipped output is produced!")), ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="-verbose", doc=InputDocumentation(doc=": Print additional processing information (default: false)"))], outputs=[ToolOutput(tag="out_output_file_file", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output_file_file", type_hint=File()), doc=OutputDocumentation(doc=": Output file. If no file is provided, output will be written to System.out. If\nfile ends with '.gz', output will be gzipped.")), ToolOutput(tag="out_reversefile_write_unmerged", output_type=File(optional=True), selector=InputSelector(input_to_select="in_reversefile_write_unmerged", type_hint=File()), doc=OutputDocumentation(doc="REVERSE_FILE        : Write unmerged forward and reverse reads to extra files. Unmerged forward reads\nare written to the file 'FORWARD_FILE'. Unmerged reverse reads are written to\nthe file 'REVERSE_FILE', i.e. the regular output file then only contains merged\nreads!\nAttention: If the option '-rm_no_partner' is not selected the two given output\nfiles also contain forward/reverse reads with no pairing partner!\nIf filenames end with '.gz' gzipped output is produced!"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Clipandmerge_V0_1_0().translate("wdl", allow_empty_container=True)

