version 1.0

task Gplassh {
  input {
    Boolean? mandatory_path_graph
    Boolean? mandatory_classifier_used
    Boolean? mandatory_mlplasmids_chosen
    Boolean? optional_output_name
    Boolean? optional_threshold_predict
    Boolean? optional_number_times
    Boolean? optional_gplas_filtering
    Boolean? optional_modularity_threshold
    Boolean? optional_path_corresponding
  }
  command <<<
    gplas_sh \
      ~{if (mandatory_path_graph) then "-i" else ""} \
      ~{if (mandatory_classifier_used) then "-c" else ""} \
      ~{if (mandatory_mlplasmids_chosen) then "-s" else ""} \
      ~{if (optional_output_name) then "-n" else ""} \
      ~{if (optional_threshold_predict) then "-t" else ""} \
      ~{if (optional_number_times) then "-x" else ""} \
      ~{if (optional_gplas_filtering) then "-f" else ""} \
      ~{if (optional_modularity_threshold) then "-q" else ""} \
      ~{if (optional_path_corresponding) then "-r" else ""}
  >>>
  runtime {
    docker: "quay.io/biocontainers/gplas:0.6.1--0"
  }
  parameter_meta {
    mandatory_path_graph: "Mandatory: Path to the graph file in *.gfa format used to extract nodes and links. Gfa file format"
    mandatory_classifier_used: "Mandatory: Classifier used to predict the contigs extracted from the input graph. String value: 'plasflow' or 'mlplasmids'"
    mandatory_mlplasmids_chosen: "Mandatory (if mlplasmids is chosen): Bacterial species from the graph file. If you have specified mlplasmids as classifier\\nyou need to provide one of the following three bacterial species:\\n'Enterococcus faecium','Klebsiella pneumoniae' or 'Escherichia coli'"
    optional_output_name: "Optional: Output name used in the files generated by gplas. Default: 'unnamed'"
    optional_threshold_predict: "Optional: Threshold to predict plasmid-derived sequences. Integer value ranging from 0 to 1.\\nDefault mlplasmids threshold: 0.5\\nDefault plasflow threshold: 0.7"
    optional_number_times: "Optional: Number of times gplas finds plasmid walks per each plasmid starting node. Integer value ranging from 1 to infinite.\\nDefault: 20"
    optional_gplas_filtering: "Optional: Gplas filtering threshold score to reject possible outcoming edges. Integer value ranging from 0 to 1.\\nDefault: 0.1"
    optional_modularity_threshold: "Optional: Modularity threshold to split components present in the plasmidome network. Integer value ranging from 0 to 1\\nDefault: 0.2"
    optional_path_corresponding: "Optional: Path to the complete reference genome corresponding to the graph given. For optimal results using this\\nbenchmarking flag, please name the reference genomes using the Unicycler scheme: e.g. '1 length=4123456' '2 length=10000' '3 length=2000'\\nfor your chromosome and plasmids. Fasta file format\\n"
  }
  output {
    File out_stdout = stdout()
  }
}