from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Int, Boolean, File, String, Float

Drep_Dereplicate_V0_1_0 = CommandToolBuilder(tool="dRep_dereplicate", base_command=["dRep", "dereplicate"], inputs=[ToolInput(tag="in_processors", input_type=Int(optional=True), prefix="--processors", doc=InputDocumentation(doc="threads (default: 6)")), ToolInput(tag="in_debug", input_type=Boolean(optional=True), prefix="--debug", doc=InputDocumentation(doc="make extra debugging output (default: False)")), ToolInput(tag="in_genomes_filter_ifbdb", input_type=Boolean(optional=True), prefix="-g", doc=InputDocumentation(doc="[GENOMES [GENOMES ...]], --genomes [GENOMES [GENOMES ...]]\ngenomes to filter in .fasta format. Not necessary if\nBdb or Wdb already exist. Can also input a text file\nwith paths to genomes, which results in fewer OS\nissues than wildcard expansion (default: None)")), ToolInput(tag="in_length", input_type=Int(optional=True), prefix="--length", doc=InputDocumentation(doc="Minimum genome length (default: 50000)")), ToolInput(tag="in_completeness", input_type=Int(optional=True), prefix="--completeness", doc=InputDocumentation(doc="Minumum genome completeness (default: 75)")), ToolInput(tag="in_contamination", input_type=Int(optional=True), prefix="--contamination", doc=InputDocumentation(doc="Maximum genome contamination (default: 25)")), ToolInput(tag="in_ignore_genome_quality", input_type=Boolean(optional=True), prefix="--ignoreGenomeQuality", doc=InputDocumentation(doc="Don't run checkM or do any quality filtering. NOT\nRECOMMENDED! This is useful for use with\nbacteriophages or eukaryotes or things where checkM\nscoring does not work. Will only choose genomes based\non length and N50 (default: False)")), ToolInput(tag="in_genome_info", input_type=File(optional=True), prefix="--genomeInfo", doc=InputDocumentation(doc="location of .csv file containing quality information\non the genomes. Must contain: ['genome'(basename of\n.fasta file of that genome), 'completeness'(0-100\nvalue for completeness of the genome),\n'contamination'(0-100 value of the contamination of\nthe genome)] (default: None)")), ToolInput(tag="in_check_m_method", input_type=String(optional=True), prefix="--checkM_method", doc=InputDocumentation(doc="Either lineage_wf (more accurate) or taxonomy_wf\n(faster) (default: lineage_wf)")), ToolInput(tag="in_set_recursion", input_type=Int(optional=True), prefix="--set_recursion", doc=InputDocumentation(doc="Increases the python recursion limit. NOT RECOMMENDED\nunless checkM is crashing due to recursion issues.\nRecommended to set to 2000 if needed, but setting this\ncould crash python (default: 0)")), ToolInput(tag="in_check_m_group_size", input_type=Int(optional=True), prefix="--checkm_group_size", doc=InputDocumentation(doc="The number of genomes passed to checkM at a time.\nIncreasing this increases RAM but makes checkM faster\n(default: 2000)")), ToolInput(tag="in_s_algorithm", input_type=String(optional=True), prefix="--S_algorithm", doc=InputDocumentation(doc="Algorithm for secondary clustering comaprisons:\nfastANI = Kmer-based approach; very fast\nANImf   = (DEFAULT) Align whole genomes with nucmer; filter alignment; compare aligned regions\nANIn    = Align whole genomes with nucmer; compare aligned regions\ngANI    = Identify and align ORFs; compare aligned ORFS\ngoANI   = Open source version of gANI; requires nsmimscan\n(default: ANImf)")), ToolInput(tag="in_mash_sketch", input_type=Int(optional=True), prefix="--MASH_sketch", doc=InputDocumentation(doc="MASH sketch size (default: 1000)")), ToolInput(tag="in_skip_mash", input_type=Boolean(optional=True), prefix="--SkipMash", doc=InputDocumentation(doc="Skip MASH clustering, just do secondary clustering on\nall genomes (default: False)")), ToolInput(tag="in_skip_secondary", input_type=Boolean(optional=True), prefix="--SkipSecondary", doc=InputDocumentation(doc="Skip secondary clustering, just perform MASH\nclustering (default: False)")), ToolInput(tag="in_n_preset", input_type=String(optional=True), prefix="--n_PRESET", doc=InputDocumentation(doc="Presets to pass to nucmer\ntight   = only align highly conserved regions\nnormal  = default ANIn parameters (default: normal)")), ToolInput(tag="in_p_ani", input_type=Float(optional=True), prefix="--P_ani", doc=InputDocumentation(doc="ANI threshold to form primary (MASH) clusters\n(default: 0.9)")), ToolInput(tag="in_s_ani", input_type=Float(optional=True), prefix="--S_ani", doc=InputDocumentation(doc="ANI threshold to form secondary clusters (default:\n0.99)")), ToolInput(tag="in_cov_thresh", input_type=Float(optional=True), prefix="--cov_thresh", doc=InputDocumentation(doc="Minmum level of overlap between genomes when doing\nsecondary comparisons (default: 0.1)")), ToolInput(tag="in_coverage_method", input_type=String(optional=True), prefix="--coverage_method", doc=InputDocumentation(doc="Method to calculate coverage of an alignment\n(for ANIn/ANImf only; gANI and fastANI can only do larger method)\ntotal   = 2*(aligned length) / (sum of total genome lengths)\nlarger  = max((aligned length / genome 1), (aligned_length / genome2))\n(default: larger)")), ToolInput(tag="in_cluster_alg", input_type=String(optional=True), prefix="--clusterAlg", doc=InputDocumentation(doc="Algorithm used to cluster genomes (passed to\nscipy.cluster.hierarchy.linkage (default: average)")), ToolInput(tag="in_multi_round_primary_clustering", input_type=Boolean(optional=True), prefix="--multiround_primary_clustering", doc=InputDocumentation(doc="Cluster each primary clunk separately and merge at the\nend with single linkage. Decreases RAM usage and\nincreases speed, and the cost of a minor loss in\nprecision and the inability to plot\nprimary_clustering_dendrograms. Especially helpful\nwhen clustering 5000+ genomes. Will be done with\nsingle linkage clustering (default: False)")), ToolInput(tag="in_primary_chunksize", input_type=Int(optional=True), prefix="--primary_chunksize", doc=InputDocumentation(doc="Impacts multiround_primary_clustering. If you have\nmore than this many genomes, process them in chunks of\nthis size. (default: 5000)")), ToolInput(tag="in_greedy_secondary_clustering", input_type=Boolean(optional=True), prefix="--greedy_secondary_clustering", doc=InputDocumentation(doc="Use a heuristic to avoid pair-wise comparisons when\ndoing secondary clustering. Will be done with single\nlinkage clustering. Only works for fastANI S_algorithm\noption at the moment (default: False)")), ToolInput(tag="in_run_tertiary_clustering", input_type=Boolean(optional=True), prefix="--run_tertiary_clustering", doc=InputDocumentation(doc="Run an additional round of clustering on the final\ngenome set. This is especially useful when greedy\nclustering is performed and/or to handle cases where\nsimilar genomes end up in different primary clusters.\nOnly works with dereplicate, not compare. (default:\nFalse)")), ToolInput(tag="in_completeness_weight", input_type=Int(optional=True), prefix="--completeness_weight", doc=InputDocumentation(doc="completeness weight (default: 1)")), ToolInput(tag="in_contamination_weight", input_type=Int(optional=True), prefix="--contamination_weight", doc=InputDocumentation(doc="contamination weight (default: 5)")), ToolInput(tag="in_strain_heterogeneity_weight", input_type=Int(optional=True), prefix="--strain_heterogeneity_weight", doc=InputDocumentation(doc="strain heterogeneity weight (default: 1)")), ToolInput(tag="in_n_five_zero_weight", input_type=Int(optional=True), prefix="--N50_weight", doc=InputDocumentation(doc="weight of log(genome N50) (default: 0.5)")), ToolInput(tag="in_size_weight", input_type=Int(optional=True), prefix="--size_weight", doc=InputDocumentation(doc="weight of log(genome size) (default: 0)")), ToolInput(tag="in_centrality_weight", input_type=Int(optional=True), prefix="--centrality_weight", doc=InputDocumentation(doc="Weight of (centrality - S_ani) (default: 1)")), ToolInput(tag="in_extra_weight_table", input_type=File(optional=True), prefix="--extra_weight_table", doc=InputDocumentation(doc="Path to a tab-separated file with two-columns, no\nheaders, listing genome and extra score to apply to\nthat genome (default: None)")), ToolInput(tag="in_warn_dist", input_type=Float(optional=True), prefix="--warn_dist", doc=InputDocumentation(doc="How far from the threshold to throw cluster warnings\n(default: 0.25)")), ToolInput(tag="in_warn_sim", input_type=Float(optional=True), prefix="--warn_sim", doc=InputDocumentation(doc="Similarity threshold for warnings between dereplicated\ngenomes (default: 0.98)")), ToolInput(tag="in_warn_aln", input_type=Float(optional=True), prefix="--warn_aln", doc=InputDocumentation(doc="Minimum aligned fraction for warnings between\ndereplicated genomes (ANIn) (default: 0.25)")), ToolInput(tag="in_genomes", input_type=String(optional=True), position=0, doc=InputDocumentation(doc=""))], outputs=[], container="quay.io/biocontainers/drep:3.2.0--py_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Drep_Dereplicate_V0_1_0().translate("wdl")

