from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, Int, File

Mapper_Pl_V0_1_0 = CommandToolBuilder(tool="mapper.pl", base_command=["mapper.pl"], inputs=[ToolInput(tag="in_input_file_seqtxt", input_type=Boolean(optional=True), prefix="-a", doc=InputDocumentation(doc="input file is seq.txt format")), ToolInput(tag="in_input_file_qseqtxt", input_type=Boolean(optional=True), prefix="-b", doc=InputDocumentation(doc="input file is qseq.txt format")), ToolInput(tag="in_input_file_fasta", input_type=Boolean(optional=True), prefix="-c", doc=InputDocumentation(doc="input file is fasta format")), ToolInput(tag="in_input_file_fastq", input_type=Boolean(optional=True), prefix="-e", doc=InputDocumentation(doc="input file is fastq format")), ToolInput(tag="in_input_file_config", input_type=Boolean(optional=True), prefix="-d", doc=InputDocumentation(doc="input file is a config file (see miRDeep2 documentation).\noptions -a, -b, -c or -e must be given with option -d.")), ToolInput(tag="in_threeletter_prefix_reads", input_type=Boolean(optional=True), prefix="-g", doc=InputDocumentation(doc="three-letter prefix for reads (by default 'seq')")), ToolInput(tag="in_convert_rna_map", input_type=Boolean(optional=True), prefix="-i", doc=InputDocumentation(doc="convert rna to dna alphabet (to map against genome)")), ToolInput(tag="in_remove_entries_have", input_type=Boolean(optional=True), prefix="-j", doc=InputDocumentation(doc="remove all entries that have a sequence that contains letters\nother than a,c,g,t,u,n,A,C,G,T,U,N")), ToolInput(tag="in_clip_adapter_sequence", input_type=Int(optional=True), prefix="-k", doc=InputDocumentation(doc="clip 3' adapter sequence")), ToolInput(tag="in_discard_reads_shorter", input_type=Int(optional=True), prefix="-l", doc=InputDocumentation(doc="discard reads shorter than int nts, default = 18")), ToolInput(tag="in_collapse_reads", input_type=Boolean(optional=True), prefix="-m", doc=InputDocumentation(doc="collapse reads")), ToolInput(tag="in_map_genome_indexed", input_type=File(optional=True), prefix="-p", doc=InputDocumentation(doc="map to genome (must be indexed by bowtie-build). The 'genome'\nstring must be the prefix of the bowtie index. For instance, if\nthe first indexed file is called 'h_sapiens_37_asm.1.ebwt' then\nthe prefix is 'h_sapiens_37_asm'.")), ToolInput(tag="in_map_takes_longer", input_type=Boolean(optional=True), prefix="-q", doc=InputDocumentation(doc="map with one mismatch in the seed (mapping takes longer)")), ToolInput(tag="in_read_allowed_map", input_type=Int(optional=True), prefix="-r", doc=InputDocumentation(doc="a read is allowed to map up to this number of positions in the genome\ndefault is 5")), ToolInput(tag="in_print_processed_reads", input_type=File(optional=True), prefix="-s", doc=InputDocumentation(doc="print processed reads to this file")), ToolInput(tag="in_print_read_mappings", input_type=File(optional=True), prefix="-t", doc=InputDocumentation(doc="print read mappings to this file")), ToolInput(tag="in_remove_directory_temporary", input_type=Boolean(optional=True), prefix="-u", doc=InputDocumentation(doc="do not remove directory with temporary files")), ToolInput(tag="in_outputs_progress_report", input_type=Boolean(optional=True), prefix="-v", doc=InputDocumentation(doc="outputs progress report")), ToolInput(tag="in_overwrite_existing_files", input_type=Boolean(optional=True), prefix="-n", doc=InputDocumentation(doc="overwrite existing files")), ToolInput(tag="in_number_threads_use", input_type=Boolean(optional=True), prefix="-o", doc=InputDocumentation(doc="number of threads to use for bowtie"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Mapper_Pl_V0_1_0().translate("wdl", allow_empty_container=True)

