from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import String, File, Float, Int, Boolean

Anvi_Run_Interacdome_V0_1_0 = CommandToolBuilder(tool="anvi_run_interacdome", base_command=["anvi-run-interacdome"], inputs=[ToolInput(tag="in_contigs_db", input_type=String(optional=True), prefix="--contigs-db", doc=InputDocumentation(doc="Anvi'o contigs database generated by 'anvi-gen-\ncontigs-database' (default: None)")), ToolInput(tag="in_interac_dome_data_dir", input_type=File(optional=True), prefix="--interacdome-data-dir", doc=InputDocumentation(doc="The path for the interacdome data to be stored. If you\nleave it as is without specifying anything, anvi'o\nwill set up everything in a pre-defined default\ndirectory. The advantage of using the default\ndirectory at the time of set up is that every user of\nanvi'o on a computer system will be using a single\ndata directory, but then you may need to run the setup\nprogram with superuser privileges. If you don't have\nsuperuser privileges, then you can use this parameter\nto tell anvi'o the location you wish to use to setup\nyour data. (default: None)")), ToolInput(tag="in_interac_dome_dataset", input_type=String(optional=True), prefix="--interacdome-dataset", doc=InputDocumentation(doc="Choose 'representable' to include Pfams that\ncorrespond to domain-ligand interactions that had\nnonredundant instances across three or more distinct\nPDB structures. InteracDomeauthors recommend using\nthis collection to learn more about domain binding\nproperties. Choose 'confident' to include Pfams that\ncorrespond to domain-ligand interactions that had\nnonredundant instances across three or more distinct\nPDB entries and achieved a cross-validated precision\nof at least 0.5. We recommend using this collection to\nannotate potential ligand-binding positions in protein\nsequences. The default is 'representable'.")), ToolInput(tag="in_min_binding_frequency", input_type=Float(optional=True), prefix="--min-binding-frequency", doc=InputDocumentation(doc="InteracDome has associated binding 'frequencies',\nwhich can be considered scores between 0 to 1 that\nquantify how likely a position is to be involved in\nbinding. Use this parameter to filter out low\nfrequencies. The default is 0.200000. Warning, your\ncontigs database size will grow massively if this is\nset to 0.0, but you're the boss.")), ToolInput(tag="in_min_hit_fraction", input_type=Float(optional=True), prefix="--min-hit-fraction", doc=InputDocumentation(doc="Any hits where the hit length--relative to the HMM\nprofile--divided by the total HMM profile length, is\nless than this value, it will be removed from the\nresults and will not contribute to binding\nfrequencies. The default is 0.5")), ToolInput(tag="in_information_content_cut_off", input_type=Float(optional=True), prefix="--information-content-cutoff", doc=InputDocumentation(doc="This parameter can be used to control for low-quality\ndomain hits. Each domain is composed of positions\n(match states) with varying degrees of conservancy,\nwhich can be quantified with information content (IC).\nHigh IC means highly conserved. For example, IC = 4\ncorresponds to 95% of the members of the Pfam sharing\nthe same amino acid at that position. By default,\nanvi'o demands that for an alignment of a user's gene\nwith a Pfam HMM, the gene sequence must match with the\nconsensus amino acid of each match state that has IC >\n4.000000. For context, it is common for a Pfam to not\neven have a position with an IC > 4, so these\nrepresent truly very conserved positions. You can\nmodify this with this parameter. For example, if you\nthink this is dumb, you can set this to 10000, and\nthen no domain hits will be removed for this reason.")), ToolInput(tag="in_num_threads", input_type=Int(optional=True), prefix="--num-threads", doc=InputDocumentation(doc="Maximum number of threads to use for multithreading\nwhenever possible. Very conservatively, the default is\n1. It is a good idea to not exceed the number of CPUs\n/ cores on your system. Plus, please be careful with\nthis option if you are running your commands on a SGE\n--if you are clusterizing your runs, and asking for\nmultiple threads to use, you may deplete your\nresources very fast. (default: 1)")), ToolInput(tag="in_just_do_it", input_type=Boolean(optional=True), prefix="--just-do-it", doc=InputDocumentation(doc="Don't bother me with questions or warnings, just do\nit. (default: False)")), ToolInput(tag="in_output_file_prefix", input_type=File(optional=True), prefix="--output-file-prefix", doc=InputDocumentation(doc="A prefix to be used while naming the output files (no\nfile type extensions please; just a prefix). (default:\nINTERACDOME)"))], outputs=[ToolOutput(tag="out_output_file_prefix", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output_file_prefix", type_hint=File()), doc=OutputDocumentation(doc="A prefix to be used while naming the output files (no\nfile type extensions please; just a prefix). (default:\nINTERACDOME)"))], container="quay.io/biocontainers/anvio-minimal:7--py_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Anvi_Run_Interacdome_V0_1_0().translate("wdl")

