from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Float, Boolean, Int, String, File

Dwgsim_V0_1_0 = CommandToolBuilder(tool="dwgsim", base_command=["dwgsim"], inputs=[ToolInput(tag="in_basecolorflow_error_first", input_type=Float(optional=True), prefix="-e", doc=InputDocumentation(doc="per base/color/flow error rate of the first read [from 0.020 to 0.020 by 0.000]")), ToolInput(tag="in_basecolorflow_error_second", input_type=Float(optional=True), prefix="-E", doc=InputDocumentation(doc="per base/color/flow error rate of the second read [from 0.020 to 0.020 by 0.000]")), ToolInput(tag="in_use_inner_distance", input_type=Boolean(optional=True), prefix="-i", doc=InputDocumentation(doc="use the inner distance instead of the outer distance for pairs [False]")), ToolInput(tag="in_outer_distance_two", input_type=Int(optional=True), prefix="-d", doc=InputDocumentation(doc="outer distance between the two ends for pairs [500]")), ToolInput(tag="in_standard_deviation_distance", input_type=Int(optional=True), prefix="-s", doc=InputDocumentation(doc="standard deviation of the distance for pairs [50.000]")), ToolInput(tag="in_number_read_pairs", input_type=Int(optional=True), prefix="-N", doc=InputDocumentation(doc="number of read pairs (-1 to disable) [-1]")), ToolInput(tag="in_mean_coverage_available", input_type=Float(optional=True), prefix="-C", doc=InputDocumentation(doc="mean coverage across available positions (-1 to disable) [100.00]")), ToolInput(tag="in_length_first_read", input_type=Int(optional=True), prefix="-1", doc=InputDocumentation(doc="length of the first read [70]")), ToolInput(tag="in_length_second_read", input_type=Int(optional=True), prefix="-2", doc=InputDocumentation(doc="length of the second read [70]")), ToolInput(tag="in_rate_of_mutations", input_type=Float(optional=True), prefix="-r", doc=InputDocumentation(doc="rate of mutations [0.0010]")), ToolInput(tag="in_frequency_simulate_mutations", input_type=Float(optional=True), prefix="-F", doc=InputDocumentation(doc="frequency of given mutation to simulate low fequency somatic mutations [0.5000]\nNB: freqeuncy F refers to the first strand of mutation, therefore mutations\non the second strand occour with a frequency of 1-F")), ToolInput(tag="in_fraction_mutations_are", input_type=Float(optional=True), prefix="-R", doc=InputDocumentation(doc="fraction of mutations that are indels [0.10]")), ToolInput(tag="in_probability_indel_extended", input_type=Float(optional=True), prefix="-X", doc=InputDocumentation(doc="probability an indel is extended [0.30]")), ToolInput(tag="in_minimum_length_indel", input_type=Int(optional=True), prefix="-I", doc=InputDocumentation(doc="the minimum length indel [1]")), ToolInput(tag="in_probability_random_dna", input_type=Float(optional=True), prefix="-y", doc=InputDocumentation(doc="probability of a random DNA read [0.05]")), ToolInput(tag="in_maximum_number_allowed", input_type=Int(optional=True), prefix="-n", doc=InputDocumentation(doc="maximum number of Ns allowed in a given read [0]")), ToolInput(tag="in_generate_reads_torrent", input_type=Int(optional=True), prefix="-c", doc=InputDocumentation(doc="generate reads for [0]:\n0: Illumina\n1: SOLiD\n2: Ion Torrent")), ToolInput(tag="in_generate_reads_default", input_type=Int(optional=True), prefix="-S", doc=InputDocumentation(doc="generate reads [0]:\n0: default (opposite strand for Illumina, same strand for SOLiD/Ion Torrent)\n1: same strand (mate pair)\n2: opposite strand (paired end)")), ToolInput(tag="in_flow_order_ion", input_type=String(optional=True), prefix="-f", doc=InputDocumentation(doc="the flow order for Ion Torrent data [(null)]")), ToolInput(tag="in_use_perbase_rate", input_type=Boolean(optional=True), prefix="-B", doc=InputDocumentation(doc="use a per-base error rate for Ion Torrent data [False]")), ToolInput(tag="in_haploid_mode", input_type=Boolean(optional=True), prefix="-H", doc=InputDocumentation(doc="haploid mode [False]")), ToolInput(tag="in_random_seed_uses", input_type=Int(optional=True), prefix="-z", doc=InputDocumentation(doc="random seed (-1 uses the current time) [-1]")), ToolInput(tag="in_generate_mutations_file", input_type=Boolean(optional=True), prefix="-M", doc=InputDocumentation(doc="generate a mutations file only [False]")), ToolInput(tag="in_mutations_txt_file", input_type=File(optional=True), prefix="-m", doc=InputDocumentation(doc="the mutations txt file to re-create [not using]")), ToolInput(tag="in_bedlike_file_set", input_type=File(optional=True), prefix="-b", doc=InputDocumentation(doc="the bed-like file set of candidate mutations [(null)]")), ToolInput(tag="in_vcf_file_set", input_type=File(optional=True), prefix="-v", doc=InputDocumentation(doc="the vcf file set of candidate mutations (use pl tag for strand) [(null)]")), ToolInput(tag="in_bed_regions_cover", input_type=File(optional=True), prefix="-x", doc=InputDocumentation(doc="the bed of regions to cover [not using]")), ToolInput(tag="in_read_prefix_prepend", input_type=String(optional=True), prefix="-P", doc=InputDocumentation(doc="a read prefix to prepend to each read name [not using]")), ToolInput(tag="in_fixed_base_quality", input_type=String(optional=True), prefix="-q", doc=InputDocumentation(doc="a fixed base quality to apply (single character) [not using]")), ToolInput(tag="in_standard_deviation_base", input_type=Float(optional=True), prefix="-Q", doc=InputDocumentation(doc="standard deviation of the base quality scores [2.00]")), ToolInput(tag="in_in_dot_ref_dot_fa", input_type=String(), position=0, doc=InputDocumentation(doc="")), ToolInput(tag="in_out_dot_prefix", input_type=String(), position=1, doc=InputDocumentation(doc=""))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Dwgsim_V0_1_0().translate("wdl", allow_empty_container=True)

