from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import String, Directory, Int, Boolean, File, Float

Ribo_Seed_V0_1_0 = CommandToolBuilder(tool="ribo_seed", base_command=["ribo", "seed"], inputs=[ToolInput(tag="in_reference_genbank", input_type=String(optional=True), prefix="--reference_genbank", doc=InputDocumentation(doc="genbank reference, used to estimate insert sizes, and\ncompare with QUAST")), ToolInput(tag="in_output", input_type=Directory(optional=True), prefix="--output", doc=InputDocumentation(doc="output directory; default: /")), ToolInput(tag="in_fast_q_one", input_type=Int(optional=True), prefix="--fastq1", doc=InputDocumentation(doc="forward fastq reads, can be compressed")), ToolInput(tag="in_fast_q_two", input_type=Int(optional=True), prefix="--fastq2", doc=InputDocumentation(doc="reverse fastq reads, can be compressed")), ToolInput(tag="in_fast_qs_one", input_type=Int(optional=True), prefix="--fastqS1", doc=InputDocumentation(doc="single fastq reads")), ToolInput(tag="in_flanking_length", input_type=Int(optional=True), prefix="--flanking_length", doc=InputDocumentation(doc="length of flanking regions, in bp; default: 1000")), ToolInput(tag="in_just_seed", input_type=Boolean(optional=True), prefix="--just_seed", doc=InputDocumentation(doc="Don't do an assembly, just generate the long read\n'seeds'; default: False")), ToolInput(tag="in_experiment_name", input_type=String(optional=True), prefix="--experiment_name", doc=InputDocumentation(doc="prefix for results files; default: riboSeed")), ToolInput(tag="in_mapper", input_type=String(optional=True), prefix="--mapper", doc=InputDocumentation(doc="available mappers: smalt and bwa; default: bwa")), ToolInput(tag="in_km_ers", input_type=Int(optional=True), prefix="--kmers", doc=InputDocumentation(doc="kmers used for final assembly, separated by commas\nsuch as21,33,55,77,99,127 . Can be set to 'auto',\nwhere SPAdes chooses. We ensure kmers are not too big\nor too close to read length; default:\n21,33,55,77,99,127")), ToolInput(tag="in_pre_km_ers", input_type=Int(optional=True), prefix="--pre_kmers", doc=InputDocumentation(doc="kmers used during seeding assemblies, separated bt\ncommas; default: 21,33,55,77,99")), ToolInput(tag="in_force_km_ers", input_type=Boolean(optional=True), prefix="--force_kmers", doc=InputDocumentation(doc="skip checking to see if kmerchoice is appropriate to\nread length. Sometimes kmers longer than reads can\nhelp in the final assembly, as the long reads\ngenerated by riboSeed contain kmers longer than the\nread length")), ToolInput(tag="in_score_min", input_type=Int(optional=True), prefix="--score_min", doc=InputDocumentation(doc="If using smalt, this sets the '-m' param; default with\nsmalt is inferred from read length. If using BWA,\nreads mapping with ASscore lower than this will be\nrejected; default with BWA is half of read length")), ToolInput(tag="in_min_assembly_len", input_type=Int(optional=True), prefix="--min_assembly_len", doc=InputDocumentation(doc="if initial SPAdes assembly largest contig is not at\nleast as long as --min_assembly_len, reject. Set this\nto the length of the seed sequence; if it is not\nachieved, seeding across regions will likely fail;\ndefault: 6000")), ToolInput(tag="in_include_shorts", input_type=Boolean(optional=True), prefix="--include_shorts", doc=InputDocumentation(doc="if assembled contig is smaller than")), ToolInput(tag="in_subtract", input_type=Boolean(optional=True), prefix="--subtract", doc=InputDocumentation(doc="if --subtract reads already used in previousround of\nsubassembly will not be included in subsequent rounds.\nThis can lead to problems with multiple mapping and\ninflated coverage.")), ToolInput(tag="in_linear", input_type=Boolean(optional=True), prefix="--linear", doc=InputDocumentation(doc="if genome is known to not be circular and a region of\ninterest (including flanking bits) extends past\nchromosome end, this extends the seqence past\nchromosome origin forward by --padding; default: False")), ToolInput(tag="in_min_flank_depth", input_type=Int(optional=True), prefix="--min_flank_depth", doc=InputDocumentation(doc="a subassembly will not be performed if this minimum\ndepth is not achieved on both the 3' and5' end of the\npseudocontig. default: 0")), ToolInput(tag="in_sub_assembler", input_type=String(optional=True), prefix="--subassembler", doc=InputDocumentation(doc="assembler to use for subassembly scheme. SPAdes is\nused by default, but Skesa is a new addition that\nseems to work for subassembly and is faster")), ToolInput(tag="in_ref_as_contig", input_type=String(optional=True), prefix="--ref_as_contig", doc=InputDocumentation(doc="ignore: reference will not be used in subassembly.\ntrusted: SPAdes will use the seed sequences as a\n--trusted-contig; untrusted: SPAdes will treat as\n--untrusted-contig. infer: if mapping percentage over\n80%, 'trusted'; else 'untrusted'. See SPAdes docs for\ndetails. default: infer")), ToolInput(tag="in_additional_libs", input_type=String(optional=True), prefix="--additional_libs", doc=InputDocumentation(doc="include this string (usually additional library, but\ncould be other SPAdes args) these libraries in final\nassembly in addition to the reads supplied as -F and\n-R. They must be supplied according to SPAdes arg\nnaming scheme. Use at own risk.default: None")), ToolInput(tag="in_clean_temps", input_type=Boolean(optional=True), prefix="--clean_temps", doc=InputDocumentation(doc="if --clean_temps, mapping files will be removed once\nthey are no no longer needed during the mapping\niterations to save space; default: False")), ToolInput(tag="in_enable_spades_error_core_ction", input_type=Boolean(optional=True), prefix="--enable-spades-error-corection", doc=InputDocumentation(doc="Default behaviour should be to skip read error\ncorrection: http://cab.spbu.ru/benchmarking-tools-for-\nde-novo-microbial-assembly/ . This re-enables\nitdefault: False")), ToolInput(tag="in_skip_control", input_type=Boolean(optional=True), prefix="--skip_control", doc=InputDocumentation(doc="if --skip_control, no de novo assembly will be done;\ndefault: False")), ToolInput(tag="in_iterations", input_type=Int(optional=True), prefix="--iterations", doc=InputDocumentation(doc="if iterations>1, multiple seedings will occur after\nsubassembly of seed regions; if setting --target_len,\nseedings will continue until --iterations are\ncompleted or --target_len is matched or exceeded;\ndefault: 3")), ToolInput(tag="in_verbosity", input_type=File(optional=True), prefix="--verbosity", doc=InputDocumentation(doc="Logger writes debug to file in output dir; this sets\nverbosity level sent to stderr. 1 = debug(), 2 =\ninfo(), 3 = warning(), 4 = error() and 5 = critical();\ndefault: 2")), ToolInput(tag="in_target_len", input_type=Float(optional=True), prefix="--target_len", doc=InputDocumentation(doc="if set, iterations will continue until contigs reach\nthis length, or max iterations (set by --iterations)\nhave been completed. Set as fraction of original seed\nlength by giving a decimal between 0 and 5, or set as\nan absolute number of base pairs by giving an integer\ngreater than 50. Not used by default")), ToolInput(tag="in_serialize", input_type=Boolean(optional=True), prefix="--serialize", doc=InputDocumentation(doc="if --serialize, runs seeding and assembly without\nmultiprocessing. This is recommended for machines with\nless than 8GB RAM: False")), ToolInput(tag="in_consensus", input_type=Boolean(optional=True), prefix="--consensus", doc=InputDocumentation(doc="if --initial_consensus, generate a mpileup-based\nconsesnsus instead of doing a proper spades\nsubassembly")), ToolInput(tag="in_s_malt_scoring", input_type=Int(optional=True), prefix="--smalt_scoring", doc=InputDocumentation(doc="if mapping with SMALT, submit custom smalt scoring via\nsmalt -S scorespec option; default:\nmatch=1,subst=-4,gapopen=-4,gapext=-3")), ToolInput(tag="in_mapper_args", input_type=Int(optional=True), prefix="--mapper_args", doc=InputDocumentation(doc="submit custom parameters to mapper. And by mapper, I\nmean bwa, cause we dont support this option for SMALT,\nsorry. This requires knowledge of your chosen mapper's\noptional arguments. Proceed with caution! default: -L\n0,0 -U 0 -a")), ToolInput(tag="in_spades_exe", input_type=File(optional=True), prefix="--spades_exe", doc=InputDocumentation(doc="Path to SPAdes executable; default: spades.py")), ToolInput(tag="in_sam_tools_exe", input_type=File(optional=True), prefix="--samtools_exe", doc=InputDocumentation(doc="Path to samtools executable; default: samtools")), ToolInput(tag="in_skes_a_exe", input_type=File(optional=True), prefix="--skesa_exe", doc=InputDocumentation(doc="Path to skesa executable; default: skesa")), ToolInput(tag="in_s_malt_exe", input_type=File(optional=True), prefix="--smalt_exe", doc=InputDocumentation(doc="Path to smalt executable; default: smalt")), ToolInput(tag="in_bwa_exe", input_type=File(optional=True), prefix="--bwa_exe", doc=InputDocumentation(doc="Path to BWA executable; default: bwa")), ToolInput(tag="in_quast_exe", input_type=File(optional=True), prefix="--quast_exe", doc=InputDocumentation(doc="Path to quast executable; default: quast")), ToolInput(tag="in_bcf_tools_exe", input_type=File(optional=True), prefix="--bcftools_exe", doc=InputDocumentation(doc="Path to bcftools executable; default: bcftools")), ToolInput(tag="in_cores", input_type=String(optional=True), prefix="--cores", doc=InputDocumentation(doc="cores to be used; default: None")), ToolInput(tag="in_threads", input_type=String(optional=True), prefix="--threads", doc=InputDocumentation(doc="if your cores are hyperthreaded, set number threads to\nthe number of threads per processer.If unsure, see\n'cat /proc/cpuinfo' under 'cpu cores', or 'lscpu'\nunder 'Thread(s) per core'.: 1")), ToolInput(tag="in_memory", input_type=Int(optional=True), prefix="--memory", doc=InputDocumentation(doc="system memory available; default: 8")), ToolInput(tag="in_inferred", input_type=String(), position=0, doc=InputDocumentation(doc="--damn_the_torpedos   Ignore certain errors, full speed ahead!"))], outputs=[ToolOutput(tag="out_output", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_output", type_hint=File()), doc=OutputDocumentation(doc="output directory; default: /")), ToolOutput(tag="out_verbosity", output_type=File(optional=True), selector=InputSelector(input_to_select="in_verbosity", type_hint=File()), doc=OutputDocumentation(doc="Logger writes debug to file in output dir; this sets\nverbosity level sent to stderr. 1 = debug(), 2 =\ninfo(), 3 = warning(), 4 = error() and 5 = critical();\ndefault: 2"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Ribo_Seed_V0_1_0().translate("wdl", allow_empty_container=True)

