class: CommandLineTool
id: ChorusNGSfilter.cwl
inputs:
- id: jellyfish
  doc: The path where Jellyfish software installed
  type: string
  inputBinding:
    prefix: --jellyfish
- id: genome
  doc: Fasta format genome file, should include all sequences from genome
  type: string
  inputBinding:
    prefix: --genome
- id: input
  doc: 'Fastq format input files contain reads from whole genome shotgun sequencing,
    files can be gzipped. Multiple files separate with ",". For example: 1.fq.gz,2.fq.gz'
  type: string
  inputBinding:
    prefix: --input
- id: gzipped
  doc: 'Input fastq file is gzipped(gz) or uncompressed(text). (Default: gz)'
  type: string
  inputBinding:
    prefix: --gzipped
- id: threads
  doc: 'Number of threads or CPUs to use. (Default: 1)'
  type: string
  inputBinding:
    prefix: --threads
- id: km_er
  doc: 'Length of k-mer used for counting k-mers in input fastq files. (Default: 17)'
  type: string
  inputBinding:
    prefix: --kmer
- id: probe
  doc: The bed format probe file generated by Chorus
  type: string
  inputBinding:
    prefix: --probe
- id: output
  doc: 'Output bed format probe file with k-mer score. (Default: output.bed)'
  type: string
  inputBinding:
    prefix: --output
- id: j
  doc: /opt/software/jellyfish/bin/jellyfish -p probe.bed -o output.bed
  type: boolean
  inputBinding:
    prefix: -j
outputs: []
cwlVersion: v1.1
baseCommand:
- ChorusNGSfilter
