version 1.0

task RunMidaspySpecies {
  input {
    Int? fastafastq_file_containing_st_mate
    Int? fastafastq_file_containing_nd_mate
    Int? number_reads_use
    Int? number_threads_use
    File? path_reference_default
    Boolean? remove_temp
    Int? word_size
    Float? map_id
    Float? aln_cov
    Int? read_length
    String outdir
  }
  command <<<
    run_midas_py species \
      ~{outdir} \
      ~{if defined(fastafastq_file_containing_st_mate) then ("-1 " +  '"' + fastafastq_file_containing_st_mate + '"') else ""} \
      ~{if defined(fastafastq_file_containing_nd_mate) then ("-2 " +  '"' + fastafastq_file_containing_nd_mate + '"') else ""} \
      ~{if defined(number_reads_use) then ("-n " +  '"' + number_reads_use + '"') else ""} \
      ~{if defined(number_threads_use) then ("-t " +  '"' + number_threads_use + '"') else ""} \
      ~{if defined(path_reference_default) then ("-d " +  '"' + path_reference_default + '"') else ""} \
      ~{if (remove_temp) then "--remove_temp" else ""} \
      ~{if defined(word_size) then ("--word_size " +  '"' + word_size + '"') else ""} \
      ~{if defined(map_id) then ("--mapid " +  '"' + map_id + '"') else ""} \
      ~{if defined(aln_cov) then ("--aln_cov " +  '"' + aln_cov + '"') else ""} \
      ~{if defined(read_length) then ("--read_length " +  '"' + read_length + '"') else ""}
  >>>
  parameter_meta {
    fastafastq_file_containing_st_mate: "FASTA/FASTQ file containing 1st mate if using paired-end reads.\\nOtherwise FASTA/FASTQ containing unpaired reads.\\nCan be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2)"
    fastafastq_file_containing_nd_mate: "FASTA/FASTQ file containing 2nd mate if using paired-end reads.\\nCan be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2)"
    number_reads_use: "Number of reads to use from input file(s) (use all)"
    number_threads_use: "Number of threads to use for database search (1)"
    path_reference_default: "Path to reference database\\nBy default, the MIDAS_DB environmental variable is used"
    remove_temp: "Remove intermediate files generated by MIDAS (False).\\nUseful to reduce disk space of MIDAS output"
    word_size: "Word size for BLAST search (28)\\nUse word sizes > 16 for greatest efficiency."
    map_id: "Discard reads with alignment identity < MAPID\\nBy default gene-specific species-level cutoffs are used\\nValues between 0-100 accepted"
    aln_cov: "Discard reads with alignment coverage < ALN_COV (0.75)\\nValues between 0-1 accepted"
    read_length: "Trim reads to READ_LENGTH and discard reads with length < READ_LENGTH\\nBy default, reads are not trimmed or filtered"
    outdir: "Path to directory to store results.\\nDirectory name should correspond to sample identifier"
  }
  output {
    File out_stdout = stdout()
  }
}