from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Array, Int, File, String, Float, Boolean

Tombo_Test_Significance_V0_1_0 = CommandToolBuilder(tool="tombo_test_significance", base_command=["tombo", "test_significance"], inputs=[ToolInput(tag="in_fast_five_based_irs", input_type=Array(t=Int(), optional=True), prefix="--fast5-basedirs", doc=InputDocumentation(doc="Directories containing fast5 files.")), ToolInput(tag="in_statistics_file_basename", input_type=File(optional=True), prefix="--statistics-file-basename", doc=InputDocumentation(doc="File base name to save base by base statistics from\ntesting. Filenames will be [--statistics-file-basename\n].[--alternate-bases]?.tombo.stats")), ToolInput(tag="in_control_fast_five_based_irs", input_type=Array(t=Int(), optional=True), prefix="--control-fast5-basedirs", doc=InputDocumentation(doc="Control set of directories containing fast5 files.\nThese reads should contain only standard nucleotides.")), ToolInput(tag="in_tombo_model_filename", input_type=File(optional=True), prefix="--tombo-model-filename", doc=InputDocumentation(doc="Tombo model for event-less resquiggle and significance\ntesting. If no model is provided the default DNA or\nRNA tombo model will be used.")), ToolInput(tag="in_alternate_model_filenames", input_type=Array(t=String(), optional=True), prefix="--alternate-model-filenames", doc=InputDocumentation(doc="Tombo models for alternative likelihood ratio\nsignificance testing.")), ToolInput(tag="in_alternate_bases", input_type=String(optional=True), prefix="--alternate-bases", doc=InputDocumentation(doc="[{5mC} ...]\nDefault non-standard base model for testing.")), ToolInput(tag="in_fishers_method_context", input_type=Int(optional=True), prefix="--fishers-method-context", doc=InputDocumentation(doc="Number of context bases up and downstream over which\nto compute Fisher's method combined p-values. Note:\nNot applicable for alternative model likelihood ratio\ntests. Default: 1.")), ToolInput(tag="in_minimum_test_reads", input_type=Int(optional=True), prefix="--minimum-test-reads", doc=InputDocumentation(doc="Number of reads required at a position to perform\nsignificance testing or contribute to model\nestimation. Default: 5")), ToolInput(tag="in_single_read_threshold", input_type=Float(optional=True), prefix="--single-read-threshold", doc=InputDocumentation(doc="P-value or log likelihood ratio threshold when\ncomputing fraction of significant reads at each\ngenomic position. Default: pvalue:0.01; likelihood\nratio:2")), ToolInput(tag="in_multiprocess_region_size", input_type=Int(optional=True), prefix="--multiprocess-region-size", doc=InputDocumentation(doc="Size of regions over which to multiprocesses statistic\ncomputation. For very deep samples a smaller value is\nrecommmended in order to control memory consumption.\nDefault: 10000")), ToolInput(tag="in_processes", input_type=Int(optional=True), prefix="--processes", doc=InputDocumentation(doc="Number of processes. Default: 1")), ToolInput(tag="in_corrected_group", input_type=Int(optional=True), prefix="--corrected-group", doc=InputDocumentation(doc="FAST5 group created by resquiggle command. Default:\nRawGenomeCorrected_000")), ToolInput(tag="in_base_call_subgroups", input_type=Array(t=String(), optional=True), prefix="--basecall-subgroups", doc=InputDocumentation(doc="FAST5 subgroup(s) (under Analyses/[corrected-group])\ncontaining basecalls. Default: ['BaseCalled_template']")), ToolInput(tag="in_quiet", input_type=Boolean(optional=True), prefix="--quiet", doc=InputDocumentation(doc="Don't print status information."))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Tombo_Test_Significance_V0_1_0().translate("wdl", allow_empty_container=True)

