from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, String, Int, Float, Boolean

Htsbox_Pileup_V0_1_0 = CommandToolBuilder(tool="htsbox_pileup", base_command=["htsbox", "pileup"], inputs=[ToolInput(tag="in_reference_genome", input_type=File(optional=True), prefix="-f", doc=InputDocumentation(doc="reference genome [null]")), ToolInput(tag="in_region", input_type=String(optional=True), prefix="-r", doc=InputDocumentation(doc="region [null]")), ToolInput(tag="in_bed_position_list", input_type=File(optional=True), prefix="-b", doc=InputDocumentation(doc="BED or position list file to include [null]")), ToolInput(tag="in_minimum_base_quality", input_type=Int(optional=True), prefix="-Q", doc=InputDocumentation(doc="minimum base quality [0]")), ToolInput(tag="in_minimum_mapping_quality", input_type=Int(optional=True), prefix="-q", doc=InputDocumentation(doc="minimum mapping quality [0]")), ToolInput(tag="in_minimum_query_length", input_type=Int(optional=True), prefix="-l", doc=InputDocumentation(doc="minimum query length [0]")), ToolInput(tag="in_minimum_supplementary_alignment", input_type=Int(optional=True), prefix="-S", doc=InputDocumentation(doc="minimum supplementary alignment length [0]")), ToolInput(tag="in_ignore_queries_perbase", input_type=Float(optional=True), prefix="-V", doc=InputDocumentation(doc="ignore queries with per-base divergence >FLOAT [1]")), ToolInput(tag="in_ignore_bases_intbp", input_type=Int(optional=True), prefix="-T", doc=InputDocumentation(doc="ignore bases within INT-bp from either end of a read [0]")), ToolInput(tag="in_base_quality_depth", input_type=Boolean(optional=True), prefix="-d", doc=InputDocumentation(doc="base quality as depth")), ToolInput(tag="in_drop_alleles_depthint", input_type=Int(optional=True), prefix="-s", doc=InputDocumentation(doc="drop alleles with depth<INT [1]")), ToolInput(tag="in_use_mark_bases", input_type=Boolean(optional=True), prefix="-e", doc=InputDocumentation(doc="use '*' to mark deleted bases")), ToolInput(tag="in_show_variants_only", input_type=Boolean(optional=True), prefix="-v", doc=InputDocumentation(doc="show variants only")), ToolInput(tag="in_output_vcf_format", input_type=Boolean(optional=True), prefix="-c", doc=InputDocumentation(doc="output in the VCF format (force -v)")), ToolInput(tag="in_show_count_allele", input_type=Boolean(optional=True), prefix="-C", doc=InputDocumentation(doc="show count of each allele on both strands")), ToolInput(tag="in_output_consensus_fasta", input_type=Boolean(optional=True), prefix="-F", doc=InputDocumentation(doc="output the consensus in FASTA")), ToolInput(tag="in_majorityallele_fasta_majfa", input_type=Boolean(optional=True), prefix="-M", doc=InputDocumentation(doc="majority-allele FASTA (majfa; force -F)")), ToolInput(tag="in_randomallele_fasta_randfa", input_type=Boolean(optional=True), prefix="-R", doc=InputDocumentation(doc="random-allele FASTA (randfa; force -F)")), ToolInput(tag="in_character_bases_identical", input_type=String(optional=True), prefix="-x", doc=InputDocumentation(doc="character for bases identical to the reference [ref]")), ToolInput(tag="in_soft_mask_avgsumfloatsqrtavgsum", input_type=Float(optional=True), prefix="-D", doc=InputDocumentation(doc="soft mask if sumQ > avgSum+FLOAT*sqrt(avgSum) (force -F) [3.00]")), ToolInput(tag="in_unitig_calling_mode", input_type=Boolean(optional=True), prefix="-u", doc=InputDocumentation(doc="unitig calling mode (-d -V.01 -S300 -q20 -Q3 -s5)")), ToolInput(tag="in_pile_up", input_type=String(), position=0, doc=InputDocumentation(doc="")), ToolInput(tag="in_in_one_dot_bam", input_type=Int(), position=1, doc=InputDocumentation(doc=""))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Htsbox_Pileup_V0_1_0().translate("wdl", allow_empty_container=True)

