from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import String, Boolean, Int, File

Samtools_Depth_V0_1_0 = CommandToolBuilder(tool="samtools_depth", base_command=["samtools", "depth"], inputs=[ToolInput(tag="in_list_positions_regions", input_type=String(optional=True), prefix="-b", doc=InputDocumentation(doc="list of positions or regions")), ToolInput(tag="in_use_customized_files", input_type=Boolean(optional=True), prefix="-X", doc=InputDocumentation(doc="use customized index files")), ToolInput(tag="in_list_input_bam", input_type=String(optional=True), prefix="-f", doc=InputDocumentation(doc="list of input BAM filenames, one per line [null]")), ToolInput(tag="in_print_file_header", input_type=Boolean(optional=True), prefix="-H", doc=InputDocumentation(doc="print a file header")), ToolInput(tag="in_read_threshold_ignore", input_type=Int(optional=True), prefix="-l", doc=InputDocumentation(doc="read length threshold (ignore reads shorter than <int>) [0]")), ToolInput(tag="in_maximum_coverage_depth", input_type=Int(optional=True), prefix="-d", doc=InputDocumentation(doc="maximum coverage depth [8000]. If 0, depth is set to the maximum\ninteger value, effectively removing any depth limit.")), ToolInput(tag="in_where_write_output", input_type=File(optional=True), prefix="-o", doc=InputDocumentation(doc="where to write output to [stdout]")), ToolInput(tag="in_base_quality_threshold", input_type=Int(optional=True), prefix="-q", doc=InputDocumentation(doc="base quality threshold [0]")), ToolInput(tag="in_mapping_quality_threshold", input_type=Int(optional=True), prefix="-Q", doc=InputDocumentation(doc="mapping quality threshold [0]")), ToolInput(tag="in_chrfromto__region", input_type=Boolean(optional=True), prefix="-r", doc=InputDocumentation(doc="<chr:from-to>    region")), ToolInput(tag="in_remove_specified_flags", input_type=String(optional=True), prefix="-g", doc=InputDocumentation(doc="remove the specified flags from the set used to filter out reads")), ToolInput(tag="in_add_specified_flags", input_type=Int(optional=True), prefix="-G", doc=InputDocumentation(doc="add the specified flags to the set used to filter out reads\nThe default set is UNMAP,SECONDARY,QCFAIL,DUP or 0x704")), ToolInput(tag="in_include_reads_deletions", input_type=Boolean(optional=True), prefix="-J", doc=InputDocumentation(doc="include reads with deletions in depth computation")), ToolInput(tag="in_overlapping_section_basesof", input_type=Boolean(optional=True), prefix="-s", doc=InputDocumentation(doc="for the overlapping section of a read pair, count only the bases\nof a single read. This option requires raising the base quality\nthreshold to 1.")), ToolInput(tag="in_input_fmt_option", input_type=File(optional=True), prefix="--input-fmt-option", doc=InputDocumentation(doc="[=VAL]\nSpecify a single input file format option in the form\nof OPTION or OPTION=VALUE")), ToolInput(tag="in_reference", input_type=File(optional=True), prefix="--reference", doc=InputDocumentation(doc="Reference sequence FASTA FILE [null]")), ToolInput(tag="in_verbosity", input_type=Int(optional=True), prefix="--verbosity", doc=InputDocumentation(doc="Set level of verbosity")), ToolInput(tag="in_in_one_dot_bam", input_type=Int(), position=0, doc=InputDocumentation(doc=""))], outputs=[], container="quay.io/biocontainers/samtools:1.11--h6270b1f_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Samtools_Depth_V0_1_0().translate("wdl")

