from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import String, File, Int, Boolean

Bcftools_Consensus_V0_1_0 = CommandToolBuilder(tool="bcftools_consensus", base_command=["bcftools", "consensus"], inputs=[ToolInput(tag="in_sample", input_type=String(optional=True), prefix="--sample", doc=InputDocumentation(doc="apply variants of the given sample")), ToolInput(tag="in_chain", input_type=File(optional=True), prefix="--chain", doc=InputDocumentation(doc="write a chain file for liftover")), ToolInput(tag="in_exclude", input_type=String(optional=True), prefix="--exclude", doc=InputDocumentation(doc="exclude sites for which the expression is true (see man page for details)")), ToolInput(tag="in_fast_a_ref", input_type=File(optional=True), prefix="--fasta-ref", doc=InputDocumentation(doc="reference sequence in fasta format")), ToolInput(tag="in_haplotype", input_type=Int(optional=True), prefix="--haplotype", doc=InputDocumentation(doc="choose which allele to use from the FORMAT/GT field, note\nthe codes are case-insensitive:\n1: first allele from GT, regardless of phasing\n2: second allele from GT, regardless of phasing\nR: REF allele in het genotypes\nA: ALT allele\nLR,LA: longer allele and REF/ALT if equal length\nSR,SA: shorter allele and REF/ALT if equal length\n1pIu,2pIu: first/second allele for phased and IUPAC code for unphased GTs")), ToolInput(tag="in_include", input_type=String(optional=True), prefix="--include", doc=InputDocumentation(doc="select sites for which the expression is true (see man page for details)")), ToolInput(tag="in_iupac_codes", input_type=Boolean(optional=True), prefix="--iupac-codes", doc=InputDocumentation(doc="output variants in the form of IUPAC ambiguity codes")), ToolInput(tag="in_mask", input_type=File(optional=True), prefix="--mask", doc=InputDocumentation(doc="replace regions with N")), ToolInput(tag="in_missing", input_type=String(optional=True), prefix="--missing", doc=InputDocumentation(doc="output <char> instead of skipping the missing genotypes")), ToolInput(tag="in_output", input_type=File(optional=True), prefix="--output", doc=InputDocumentation(doc="write output to a file [standard output]")), ToolInput(tag="in_prefix", input_type=String(optional=True), prefix="--prefix", doc=InputDocumentation(doc="prefix to add to output sequence names")), ToolInput(tag="in_file_dot_vcf_do_tgz", input_type=File(), position=0, doc=InputDocumentation(doc=""))], outputs=[ToolOutput(tag="out_output", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output", type_hint=File()), doc=OutputDocumentation(doc="write output to a file [standard output]"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Bcftools_Consensus_V0_1_0().translate("wdl", allow_empty_container=True)

