from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import String, File, Boolean, Int

Anvi_Gen_Gene_Level_Stats_Databases_V0_1_0 = CommandToolBuilder(tool="anvi_gen_gene_level_stats_databases", base_command=["anvi-gen-gene-level-stats-databases"], inputs=[ToolInput(tag="in_contigs_db", input_type=String(optional=True), prefix="--contigs-db", doc=InputDocumentation(doc="Anvi'o contigs database generated by 'anvi-gen-\ncontigs'")), ToolInput(tag="in_profile_db", input_type=String(optional=True), prefix="--profile-db", doc=InputDocumentation(doc="Anvi'o profile database")), ToolInput(tag="in_collection_name", input_type=String(optional=True), prefix="--collection-name", doc=InputDocumentation(doc="Collection name.")), ToolInput(tag="in_bin_id", input_type=String(optional=True), prefix="--bin-id", doc=InputDocumentation(doc="Bin name you are interested in.")), ToolInput(tag="in_bin_ids_file", input_type=File(optional=True), prefix="--bin-ids-file", doc=InputDocumentation(doc="Text file for bins (each line should be a unique bin\nid).")), ToolInput(tag="in_zeros_are_outliers", input_type=Boolean(optional=True), prefix="--zeros-are-outliers", doc=InputDocumentation(doc="If you want all zero coverage positions to be treated\nlike outliers then use this flag. The reason to treat\nzero coverage as outliers is because when mapping\nreads to a reference we could get many zero positions\ndue to accessory genes. These positions then skew the\naverage values that we compute.")), ToolInput(tag="in_outliers_threshold", input_type=Int(optional=True), prefix="--outliers-threshold", doc=InputDocumentation(doc="Threshold to use for the outlier detection. The\ndefault value is '1.5'. Absolute deviation around the\nmedian is used. To read more about the method please\nrefer to: 'How to Detect and Handle Outliers' by Boris\nIglewicz and David Hoaglin\n(doi:10.1016/j.jesp.2013.03.013).")), ToolInput(tag="in_just_do_it", input_type=Boolean(optional=True), prefix="--just-do-it", doc=InputDocumentation(doc="Don't bother me with questions or warnings, just do")), ToolInput(tag="in_in_seq_stats", input_type=Boolean(optional=True), prefix="--inseq-stats", doc=InputDocumentation(doc="Provide if working with INSeq/Tn-Seq genomic data.\nWith this, all gene level coverage stats will be\ncalculated using INSeq/Tn-Seq statistical methods.\n")), ToolInput(tag="in_it_dot", input_type=String(), position=0, doc=InputDocumentation(doc="INSEQ DATA:"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Anvi_Gen_Gene_Level_Stats_Databases_V0_1_0().translate("wdl", allow_empty_container=True)

