from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Int, File, Float, Boolean, Array, String

Tombo_Resquiggle_V0_1_0 = CommandToolBuilder(tool="tombo_resquiggle", base_command=["tombo", "resquiggle"], inputs=[ToolInput(tag="in_minimap_two_executable", input_type=Int(optional=True), prefix="--minimap2-executable", doc=InputDocumentation(doc="Path to minimap2 executable.")), ToolInput(tag="in_minimap_two_index", input_type=Int(optional=True), prefix="--minimap2-index", doc=InputDocumentation(doc="Path to minimap2 index (with map-ont preset) file\ncorresponding to the [genome_fasta] provided.")), ToolInput(tag="in_bwa_mem_executable", input_type=File(optional=True), prefix="--bwa-mem-executable", doc=InputDocumentation(doc="Path to bwa-mem executable.")), ToolInput(tag="in_graph_map_executable", input_type=File(optional=True), prefix="--graphmap-executable", doc=InputDocumentation(doc="Path to graphmap executable.")), ToolInput(tag="in_alignment_batch_size", input_type=Int(optional=True), prefix="--alignment-batch-size", doc=InputDocumentation(doc="Number of reads included in each alignment call. Note:\nA new system mapping call is made for each batch\n(including loading of the genome), so it is advised to\nuse larger values for larger genomes. Default: 1000")), ToolInput(tag="in_tombo_model_filename", input_type=File(optional=True), prefix="--tombo-model-filename", doc=InputDocumentation(doc="Tombo model for event-less resquiggle and significance\ntesting. If no model is provided the default DNA or\nRNA tombo model will be used.")), ToolInput(tag="in_match_expected_value", input_type=Float(optional=True), prefix="--match-expected-value", doc=InputDocumentation(doc="Expected value when a matched event to genomic\nsequence is encountered. Default: 0.500000")), ToolInput(tag="in_skip_penalty", input_type=Float(optional=True), prefix="--skip-penalty", doc=InputDocumentation(doc="Penalty applied to skipped genomic bases in event to\nsequence assignment. Default: 1.000000")), ToolInput(tag="in_bandwidth", input_type=Int(optional=True), prefix="--bandwidth", doc=InputDocumentation(doc="Bandwidth of events for dynamic sequence to event\nmapping. Default: 501")), ToolInput(tag="in_skip_index", input_type=Boolean(optional=True), prefix="--skip-index", doc=InputDocumentation(doc="Skip creation of tombo index. This drastically slows\ndownstream tombo commands. Default stores tombo index\nnamed '.[--fast5-basedir].[--corrected-\ngroup].tombo.index' to be loaded automatically for\ndownstream commands.")), ToolInput(tag="in_failed_reads_filename", input_type=File(optional=True), prefix="--failed-reads-filename", doc=InputDocumentation(doc="Output failed read filenames with assoicated error.\nDefault: Do not store failed reads.")), ToolInput(tag="in_include_event_stdev", input_type=Boolean(optional=True), prefix="--include-event-stdev", doc=InputDocumentation(doc="Include corrected event standard deviation in output\nFAST5 data.")), ToolInput(tag="in_obs_per_base_filter", input_type=Array(t=String(), optional=True), prefix="--obs-per-base-filter", doc=InputDocumentation(doc="Filter reads baseed on observations per base\npercentile thresholds. Format thresholds as\n'percentile:thresh [pctl2:thresh2 ...]'. For example\nto filter reads with 99th pctl > 200 obs/base or max >\n5k obs/base use '99:200 100:5000'.")), ToolInput(tag="in_outlier_threshold", input_type=Int(optional=True), prefix="--outlier-threshold", doc=InputDocumentation(doc="Windosrize the signal at this number of scale values.\nNegative value disables outlier clipping. Default:\n5.000000")), ToolInput(tag="in_processes", input_type=Int(optional=True), prefix="--processes", doc=InputDocumentation(doc="Number of processes. Default: 2")), ToolInput(tag="in_align_processes", input_type=Int(optional=True), prefix="--align-processes", doc=InputDocumentation(doc="Number of processes to use for parsing and aligning\noriginal basecalls. Each process will independently\nload the genome into memory, so use caution with\nlarger genomes (e.g. human). Default: 1")), ToolInput(tag="in_align_threads_per_process", input_type=Int(optional=True), prefix="--align-threads-per-process", doc=InputDocumentation(doc="Number of threads to use for aligner system call.\nDefault: [--processes] / (2 * [--align-processes)]")), ToolInput(tag="in_re_squiggle_processes", input_type=Int(optional=True), prefix="--resquiggle-processes", doc=InputDocumentation(doc="Number of processes to use for resquiggle algorithm.\nDefault: [--processes] / 2")), ToolInput(tag="in_corrected_group", input_type=Int(optional=True), prefix="--corrected-group", doc=InputDocumentation(doc="FAST5 group created by resquiggle command. Default:\nRawGenomeCorrected_000")), ToolInput(tag="in_base_call_group", input_type=Int(optional=True), prefix="--basecall-group", doc=InputDocumentation(doc="FAST5 group obtain original basecalls (under Analyses\ngroup). Default: Basecall_1D_000")), ToolInput(tag="in_base_call_subgroups", input_type=Array(t=String(), optional=True), prefix="--basecall-subgroups", doc=InputDocumentation(doc="FAST5 subgroup(s) (under Analyses/[corrected-group])\ncontaining basecalls. Default: ['BaseCalled_template']")), ToolInput(tag="in_overwrite", input_type=Boolean(optional=True), prefix="--overwrite", doc=InputDocumentation(doc="Overwrite previous corrected group in FAST5 files.\nNote: only effects --corrected-group or --new-\ncorrected-group.")), ToolInput(tag="in_quiet", input_type=Boolean(optional=True), prefix="--quiet", doc=InputDocumentation(doc="Don't print status information.")), ToolInput(tag="in_fast_five_basedir", input_type=Int(), position=0, doc=InputDocumentation(doc="Directory containing fast5 files. All files ending in\n'fast5' found recursively within this base directory\nwill be processed.")), ToolInput(tag="in_genome_fast_a", input_type=String(), position=1, doc=InputDocumentation(doc="Path to fasta file for mapping."))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Tombo_Resquiggle_V0_1_0().translate("wdl", allow_empty_container=True)

