from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, String, Int, Boolean, Directory, Float

Basenji_Data2_Py_V0_1_0 = CommandToolBuilder(tool="basenji_data2.py", base_command=["basenji_data2.py"], inputs=[ToolInput(tag="in_alignment_net_file", input_type=File(optional=True), prefix="-a", doc=InputDocumentation(doc="Alignment .net file")), ToolInput(tag="in_set_blacklist_nucleotides", input_type=String(optional=True), prefix="-b", doc=InputDocumentation(doc="Set blacklist nucleotides to a baseline value.")), ToolInput(tag="in_break", input_type=Int(optional=True), prefix="--break", doc=InputDocumentation(doc="Break in half contigs above length [Default: none]")), ToolInput(tag="in_crop", input_type=Int(optional=True), prefix="--crop", doc=InputDocumentation(doc="Crop bp off each end [Default: 0]")), ToolInput(tag="in_downsample_the_segments", input_type=String(optional=True), prefix="-d", doc=InputDocumentation(doc="Down-sample the segments")), ToolInput(tag="in_commaseparated_list_files", input_type=String(optional=True), prefix="-g", doc=InputDocumentation(doc="Comma-separated list of assembly gaps BED files\n[Default: none]")), ToolInput(tag="in_interpolate_nans", input_type=Boolean(optional=True), prefix="-i", doc=InputDocumentation(doc="Interpolate NaNs [Default: False]")), ToolInput(tag="in_sequence_length", input_type=Int(optional=True), prefix="-l", doc=InputDocumentation(doc="Sequence length [Default: 131072]")), ToolInput(tag="in_local", input_type=Boolean(optional=True), prefix="--local", doc=InputDocumentation(doc="Run jobs locally as opposed to on SLURM [Default:\nFalse]")), ToolInput(tag="in_alignment_net_fill", input_type=Int(optional=True), prefix="-n", doc=InputDocumentation(doc="Alignment net fill size minimum [Default: 100000]")), ToolInput(tag="in_output_directory", input_type=Directory(optional=True), prefix="-o", doc=InputDocumentation(doc="Output directory [Default: data_out]")), ToolInput(tag="in_number_parallel_processes", input_type=Int(optional=True), prefix="-p", doc=InputDocumentation(doc="Number parallel processes [Default: none]")), ToolInput(tag="in_sequences_tfrecord_file", input_type=File(optional=True), prefix="-r", doc=InputDocumentation(doc="Sequences per TFRecord file [Default: 256]")), ToolInput(tag="in_restart", input_type=Boolean(optional=True), prefix="--restart", doc=InputDocumentation(doc="Skip already read HDF5 coverage values. [Default:\nFalse]")), ToolInput(tag="in_seed", input_type=Int(optional=True), prefix="--seed", doc=InputDocumentation(doc="Random seed [Default: 44]")), ToolInput(tag="in_snap", input_type=String(optional=True), prefix="--snap", doc=InputDocumentation(doc="Snap sequences to multiple of the given value\n[Default: none]")), ToolInput(tag="in_stride_train", input_type=String(optional=True), prefix="--stride_train", doc=InputDocumentation(doc="Stride to advance train sequences [Default:\nseq_length]")), ToolInput(tag="in_stride_test", input_type=Float(optional=True), prefix="--stride_test", doc=InputDocumentation(doc="Stride to advance valid and test sequences [Default:\n1.0]")), ToolInput(tag="in_soft", input_type=Boolean(optional=True), prefix="--soft", doc=InputDocumentation(doc="Soft clip values, applying sqrt to the execess above\nthe threshold [Default: False]")), ToolInput(tag="in_proportion_data_testing", input_type=Float(optional=True), prefix="-t", doc=InputDocumentation(doc="Proportion of the data for testing [Default: 0.1]")), ToolInput(tag="in_commaseparated_genome_segments", input_type=String(optional=True), prefix="-u", doc=InputDocumentation(doc="Comma-separated genome unmappable segments to set to")), ToolInput(tag="in_u_map_clip", input_type=String(optional=True), prefix="--umap_clip", doc=InputDocumentation(doc="Clip unmappable regions to this percentile in the\nsequences' distribution of values")), ToolInput(tag="in_sum_pool_width", input_type=Int(optional=True), prefix="-w", doc=InputDocumentation(doc="Sum pool width [Default: 128]")), ToolInput(tag="in_proportion_data_validation", input_type=Float(optional=True), prefix="-v", doc=InputDocumentation(doc="Proportion of the data for validation [Default: 0.1]")), ToolInput(tag="in_na", input_type=String(), position=0, doc=InputDocumentation(doc="--umap_t=UMAP_T       Remove sequences with more than this unmappable bin %"))], outputs=[ToolOutput(tag="out_output_directory", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_output_directory", type_hint=File()), doc=OutputDocumentation(doc="Output directory [Default: data_out]"))], container="quay.io/biocontainers/basenji:0.5--py_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Basenji_Data2_Py_V0_1_0().translate("wdl")

