from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Directory, Boolean, Int, File, String, Float

Bactopia_Datasets_Py_V0_1_0 = CommandToolBuilder(tool="bactopia_datasets.py", base_command=["bactopia-datasets.py"], inputs=[ToolInput(tag="in_outdir", input_type=Directory(optional=True), prefix="--outdir", doc=InputDocumentation(doc="Directory to write output. (Default ./datasets)")), ToolInput(tag="in_skip_ariba", input_type=Boolean(optional=True), prefix="--skip_ariba", doc=InputDocumentation(doc="Skip setup of Ariba datasets")), ToolInput(tag="in_ariba", input_type=Int(optional=True), prefix="--ariba", doc=InputDocumentation(doc="Comma separated list of Ariba datasets to download and\nsetup. Available datasets include: argannot, card,\nncbi, megares, plasmidfinder, resfinder,\nsrst2_argannot, vfdb_core, vfdb_full, virulencefinder\n(Default: 'vfdb_core,card') Use --available_datasets\nto see the full list.")), ToolInput(tag="in_download_available_schemas", input_type=File(optional=True), prefix="--species", doc=InputDocumentation(doc="Download available MLST schemas and completed genomes\nfor a given species or a list of species in a text\nfile.")), ToolInput(tag="in_skip_mlst", input_type=Boolean(optional=True), prefix="--skip_mlst", doc=InputDocumentation(doc="Skip setup of MLST schemas for each species")), ToolInput(tag="in_skip_pro_kk_a", input_type=Boolean(optional=True), prefix="--skip_prokka", doc=InputDocumentation(doc="Skip creation of a Prokka formatted fasta for each")), ToolInput(tag="in_assembly_level", input_type=String(optional=True), prefix="--assembly_level", doc=InputDocumentation(doc="Assembly levels of genomes to download (Default:\ncomplete).")), ToolInput(tag="in_limit", input_type=Int(optional=True), prefix="--limit", doc=InputDocumentation(doc="If available completed genomes exceeds a given limit,\na random subsample will be taken. (Default 1000)")), ToolInput(tag="in_accessions", input_type=String(optional=True), prefix="--accessions", doc=InputDocumentation(doc="A list of RefSeq accessions to download.")), ToolInput(tag="in_identity", input_type=Float(optional=True), prefix="--identity", doc=InputDocumentation(doc="CD-HIT (-c) sequence identity threshold. (Default:\n0.9)")), ToolInput(tag="in_overlap", input_type=Float(optional=True), prefix="--overlap", doc=InputDocumentation(doc="CD-HIT (-s) length difference cutoff. (Default: 0.8)")), ToolInput(tag="in_max_memory", input_type=Int(optional=True), prefix="--max_memory", doc=InputDocumentation(doc="CD-HIT (-M) memory limit (in MB). (Default: unlimited")), ToolInput(tag="in_fast_cluster", input_type=Boolean(optional=True), prefix="--fast_cluster", doc=InputDocumentation(doc="Use CD-HIT's (-g 0) fast clustering algorithm, instead\nof the accurate but slow algorithm.")), ToolInput(tag="in_skip_min_mer", input_type=Boolean(optional=True), prefix="--skip_minmer", doc=InputDocumentation(doc="Skip download of pre-computed minmer datasets (mash,\nsourmash)")), ToolInput(tag="in_skip_pls_db", input_type=Boolean(optional=True), prefix="--skip_plsdb", doc=InputDocumentation(doc="Skip download of pre-computed PLSDB datbases (blast,\nmash)")), ToolInput(tag="in_skip_amr", input_type=Boolean(optional=True), prefix="--skip_amr", doc=InputDocumentation(doc="Skip download of antimicrobial resistance databases\n(e.g. AMRFinder+)")), ToolInput(tag="in_prodigal_tf", input_type=File(optional=True), prefix="--prodigal_tf", doc=InputDocumentation(doc="A pre-built Prodigal training file to add to the\nspecies annotation folder. Requires a single species\n(--species) and will replace existing training files.")), ToolInput(tag="in_reference", input_type=File(optional=True), prefix="--reference", doc=InputDocumentation(doc="A reference genome (FASTA/GenBank (preferred)) file or\ndirectory to be added to the optional folder for\nvariant calling. Requires a single species\n(--species).")), ToolInput(tag="in_mapping", input_type=File(optional=True), prefix="--mapping", doc=InputDocumentation(doc="A reference sequence (FASTA) file or directory to be\nadded to the optional folder for mapping. Requires a\nsingle species (--species).")), ToolInput(tag="in_genes", input_type=File(optional=True), prefix="--genes", doc=InputDocumentation(doc="A gene sequence (FASTA) file or directory to be added\nto the optional folder for BLAST. Requires a single\nspecies (--species).")), ToolInput(tag="in_proteins", input_type=File(optional=True), prefix="--proteins", doc=InputDocumentation(doc="A protein sequence (FASTA) file or directory to be\nadded to the optional folder for BLAST. Requires a\nsingle species (--species).")), ToolInput(tag="in_primers", input_type=File(optional=True), prefix="--primers", doc=InputDocumentation(doc="A primer sequence (FASTA) file or directory to be\nadded to the optional folder for BLAST. Requires a\nsingle species (--species).")), ToolInput(tag="in_force_optional", input_type=Boolean(optional=True), prefix="--force_optional", doc=InputDocumentation(doc="Overwrite any existing files in the optional folders")), ToolInput(tag="in_cpus", input_type=Int(optional=True), prefix="--cpus", doc=InputDocumentation(doc="Number of cpus to use. (Default: 1)")), ToolInput(tag="in_clear_cache", input_type=Boolean(optional=True), prefix="--clear_cache", doc=InputDocumentation(doc="Remove any existing cache.")), ToolInput(tag="in_force", input_type=Boolean(optional=True), prefix="--force", doc=InputDocumentation(doc="Forcibly overwrite existing datasets.")), ToolInput(tag="in_force_ariba", input_type=Boolean(optional=True), prefix="--force_ariba", doc=InputDocumentation(doc="Forcibly overwrite existing Ariba datasets.")), ToolInput(tag="in_force_mlst", input_type=Boolean(optional=True), prefix="--force_mlst", doc=InputDocumentation(doc="Forcibly overwrite existing MLST datasets.")), ToolInput(tag="in_force_pro_kk_a", input_type=Boolean(optional=True), prefix="--force_prokka", doc=InputDocumentation(doc="Forcibly overwrite existing Prokka datasets.")), ToolInput(tag="in_force_min_mer", input_type=Boolean(optional=True), prefix="--force_minmer", doc=InputDocumentation(doc="Forcibly overwrite existing minmer datasets.")), ToolInput(tag="in_force_pls_db", input_type=Boolean(optional=True), prefix="--force_plsdb", doc=InputDocumentation(doc="Forcibly overwrite existing PLSDB datasets.")), ToolInput(tag="in_force_amr", input_type=Boolean(optional=True), prefix="--force_amr", doc=InputDocumentation(doc="Forcibly overwrite existing antimicrobial resistance")), ToolInput(tag="in_available_datasets", input_type=Boolean(optional=True), prefix="--available_datasets", doc=InputDocumentation(doc="List Ariba reference datasets and MLST schemas\navailable for setup.")), ToolInput(tag="in_depends", input_type=Boolean(optional=True), prefix="--depends", doc=InputDocumentation(doc="Verify dependencies are installed.")), ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="--verbose", doc=InputDocumentation(doc="Print debug related text.")), ToolInput(tag="in_silent", input_type=Boolean(optional=True), prefix="--silent", doc=InputDocumentation(doc="Only critical errors will be printed.")), ToolInput(tag="in_ariba_dot", input_type=String(), position=1, doc=InputDocumentation(doc="optional arguments:")), ToolInput(tag="in__includegenus_fasta", input_type=String(), position=0, doc=InputDocumentation(doc="--include_genus       Include all genus members in the Prokka proteins FASTA")), ToolInput(tag="in_datasets_dot", input_type=String(), position=0, doc=InputDocumentation(doc="--keep_files          Keep all downloaded and intermediate files."))], outputs=[ToolOutput(tag="out_outdir", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_outdir", type_hint=File()), doc=OutputDocumentation(doc="Directory to write output. (Default ./datasets)"))], container="quay.io/biocontainers/bactopia:1.6.1--0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Bactopia_Datasets_Py_V0_1_0().translate("wdl")

