from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Int, Boolean, File, String

Agat_Sp_Fix_Longest_Orf_Pl_V0_1_0 = CommandToolBuilder(tool="agat_sp_fix_longest_ORF.pl", base_command=["agat_sp_fix_longest_ORF.pl"], inputs=[ToolInput(tag="in_model", input_type=Int(optional=True), prefix="--model", doc=InputDocumentation(doc="Kind of ORF Model you want to fix. By default all are used. To\nselect specific models writte e.g --model 1,4\nModel1 = The original ORF is part of the new ORF; the new ORF is\nlonger\nModel2 = The original ORF and the new one are different; the new\none is longer, they do not overlap each other.\nModel3 = The original ORF and the new one are different; the new\none is longer, they overlap each other.\nModel4 = The new ORF is shorter due to the presence of stop\ncodon in the original ORF.\nModel5 = The new ORF is shorter but the original ORF has not\npremature stop codon. The shorter predicted ORF can be due to\nthe fact that the original ORF does not start by a start codon,\nwhile we force here the prediction to have a start codon. A ORF\nwihtout start can be the fact of an incomplete or fragmented\nORF: annotation tool didn't predict the start because: * the\nstart region is NNNN * the start region is XXXX * correct\nnucleotides but prediction tool did not annotate this part (e.g\nincomplete evidence in evidence-based prediction)\nModel6 = The ORF is same size but not correct frame (+1 or +2 bp\ngives a frame shift).")), ToolInput(tag="in_gff", input_type=Boolean(optional=True), prefix="--gff", doc=InputDocumentation(doc="Input GTF/GFF file.")), ToolInput(tag="in_fast_a", input_type=File(optional=True), prefix="--fasta", doc=InputDocumentation(doc="Imput fasta file.")), ToolInput(tag="in_codon", input_type=Int(optional=True), prefix="--codon", doc=InputDocumentation(doc="Codon table to use. [default 1]")), ToolInput(tag="in_split", input_type=Int(optional=True), prefix="--split", doc=InputDocumentation(doc="This option is usefull for Model2. Indeed when the prediction is\nnon overlapping the original cds, it is possible to split the\ngene into two different genes. By default we don't split it. We\nkeep the longest. If you want to split it type: -s")), ToolInput(tag="in_outfile", input_type=File(optional=True), prefix="--outfile", doc=InputDocumentation(doc="Output GFF file. If no output file is specified, the output will\nbe written to STDOUT.")), ToolInput(tag="in_verbose_mode_default", input_type=Boolean(optional=True), prefix="-v", doc=InputDocumentation(doc="verbose mode. Default off. -v 1 minimum verbosity, -v 3 maximum")), ToolInput(tag="in_agat_sp_fix_longest_or_fdot_pl", input_type=String(), position=0, doc=InputDocumentation(doc="Description:")), ToolInput(tag="in_verbosity", input_type=String(), position=0, doc=InputDocumentation(doc="-h or --help"))], outputs=[ToolOutput(tag="out_outfile", output_type=File(optional=True), selector=InputSelector(input_to_select="in_outfile", type_hint=File()), doc=OutputDocumentation(doc="Output GFF file. If no output file is specified, the output will\nbe written to STDOUT."))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Agat_Sp_Fix_Longest_Orf_Pl_V0_1_0().translate("wdl", allow_empty_container=True)

