from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, String, Int, Directory, Float, Boolean

Vcontact2_V0_1_0 = CommandToolBuilder(tool="vcontact2", base_command=["vcontact2"], inputs=[ToolInput(tag="in_raw_proteins", input_type=File(optional=True), prefix="--raw-proteins", doc=InputDocumentation(doc="FASTA-formatted proteins file. If provided alongside\n--proteins-fn, vConTACT will start prior to PC\ngeneration. (default: None)")), ToolInput(tag="in_rel_mode", input_type=String(optional=True), prefix="--rel-mode", doc=InputDocumentation(doc="Method to use to create the protein-protein similarity\nedge file. This is what the PC clustering is applied\nagainst. (default: Diamond)")), ToolInput(tag="in_blast_fp", input_type=File(optional=True), prefix="--blast-fp", doc=InputDocumentation(doc="Blast results file in CSV or TSV format. Used for\ngenerating the protein clusters. If provided alongside\n--proteins-fn, vConTACT will start from the PC-\ngeneration step. If raw proteins are provided, THIS\nWILL BE SKIPPED. Reference DBs CANNOT BE USED IF THIS\nOPTION IS ENABLED!! (default: None)")), ToolInput(tag="in_proteins_fp", input_type=File(optional=True), prefix="--proteins-fp", doc=InputDocumentation(doc="A file linking the protein name (as in the blast file)\nand the genome names (csv or tsv). If provided\nalongside --blast-fp, vConTACT will start from the PC-\ngeneration step. If provided alongside --raw-proteins,\nvConTACT will start from creating the all-verses-all\nprotein comparison and then generate protein clusters.\n(default: None)")), ToolInput(tag="in_db", input_type=String(optional=True), prefix="--db", doc=InputDocumentation(doc="Select a reference database to de novo cluster\nproteins against. If 'None' selected, be aware that\nthere will be no references included in the analysis.\n(default: ProkaryoticViralRefSeq85-ICTV)")), ToolInput(tag="in_pcs_mode", input_type=String(optional=True), prefix="--pcs-mode", doc=InputDocumentation(doc="Whether to use ClusterONE or MCL for Protein Cluster\n(PC) generation. (default: MCL)")), ToolInput(tag="in_vcs_mode", input_type=String(optional=True), prefix="--vcs-mode", doc=InputDocumentation(doc="Whether to use ClusterONE or MCL for Viral Cluster\n(VC) generation. (default: ClusterONE)")), ToolInput(tag="in_cone_bin", input_type=Int(optional=True), prefix="--c1-bin", doc=InputDocumentation(doc="Location for clusterONE java file. This MUST SPECIFY\nthe full path to 'cluster_one-1.0.jar'. (default:\nNone)")), ToolInput(tag="in_blastp_bin", input_type=File(optional=True), prefix="--blastp-bin", doc=InputDocumentation(doc="Location for BLASTP executable. Path only used if\nvConTACT cant find in $PATH. (default: None)")), ToolInput(tag="in_diamond_bin", input_type=File(optional=True), prefix="--diamond-bin", doc=InputDocumentation(doc="Location for DIAMOND executable. Path only used if\nvConTACT cant find in $PATH. (default: None)")), ToolInput(tag="in_output_dir", input_type=Directory(optional=True), prefix="--output-dir", doc=InputDocumentation(doc="Output directory (default: vContact_Output)")), ToolInput(tag="in_threads", input_type=Int(optional=True), prefix="--threads", doc=InputDocumentation(doc="Number of CPUs to use. If nothing is provided,\nvConTACT will attempt to identify the number of\navailable CPUs. (default: 1)")), ToolInput(tag="in_blast_evalue", input_type=Float(optional=True), prefix="--blast-evalue", doc=InputDocumentation(doc="E-value used by BLASTP when creating the protein-\nprotein similarity network. (--rel-mode BLASTP only)\n(default: 0.0001)")), ToolInput(tag="in_max_overlap", input_type=Int(optional=True), prefix="--max-overlap", doc=InputDocumentation(doc="Specifies the maximum allowed overlap between two\nclusters. (ClusterONE only) (default: 0.8)")), ToolInput(tag="in_penalty", input_type=Float(optional=True), prefix="--penalty", doc=InputDocumentation(doc="Sets a penalty value for the inclusion of each node.\nIt can be used to model the possibility of uncharted\nconnections for each node, so nodes with only a single\nweak connection to a cluster will not be added to the\ncluster as the penalty value will outweigh the\nbenefits of adding the node. (ClusterONE only)\n(default: 2.0)")), ToolInput(tag="in_haircut", input_type=Float(optional=True), prefix="--haircut", doc=InputDocumentation(doc="Apply a haircut transformation as a post-processing\nstep on the detected clusters. A haircut\ntransformation removes dangling nodes from a cluster.\n(ClusterONE only) (default: 0.1)")), ToolInput(tag="in_pc_inflation", input_type=Float(optional=True), prefix="--pc-inflation", doc=InputDocumentation(doc="Inflation value to define proteins clusters with MCL.\n(default: 2.0) (MCL only) (default: 2.0)")), ToolInput(tag="in_vc_inflation", input_type=Float(optional=True), prefix="--vc-inflation", doc=InputDocumentation(doc="Inflation parameter to define contig clusters with\nMCL. (MCL only) (default: 2.0)")), ToolInput(tag="in_min_density", input_type=Int(optional=True), prefix="--min-density", doc=InputDocumentation(doc="Sets the minimum density of predicted complexes.\n(ClusterONE only) (default: 0.3)")), ToolInput(tag="in_min_size", input_type=Int(optional=True), prefix="--min-size", doc=InputDocumentation(doc="Minimum size for the Viral Cluster. (default: 2)")), ToolInput(tag="in_vc_overlap", input_type=Float(optional=True), prefix="--vc-overlap", doc=InputDocumentation(doc="Specifies the maximum allowed overlap between two\nclusters. (ClusterONE only) (default: 0.9)")), ToolInput(tag="in_vc_penalty", input_type=Float(optional=True), prefix="--vc-penalty", doc=InputDocumentation(doc="Sets a penalty value for the inclusion of each node.\nIt can be used to model the possibility of uncharted\nconnections for each node, so nodes with only a single\nweak connection to a cluster will not be added to the\ncluster as the penalty value will outweigh the\nbenefits of adding the node. (ClusterONE only)\n(default: 2.0)")), ToolInput(tag="in_vc_haircut", input_type=Float(optional=True), prefix="--vc-haircut", doc=InputDocumentation(doc="Apply a haircut transformation as a post-processing\nstep on the detected clusters. A haircut\ntransformation removes dangling nodes from a cluster.\n(ClusterONE only) (default: 0.55)")), ToolInput(tag="in_merge_method", input_type=String(optional=True), prefix="--merge-method", doc=InputDocumentation(doc="Specifies the method to be used to merge highly\noverlapping complexes. (ClusterONE only) (default:\nsingle)")), ToolInput(tag="in_similarity", input_type=String(optional=True), prefix="--similarity", doc=InputDocumentation(doc="Specifies the similarity function to be used in the\nmerging step. (ClusterONE only) (default: match)")), ToolInput(tag="in_seed_method", input_type=String(optional=True), prefix="--seed-method", doc=InputDocumentation(doc="Specifies the seed generation method to use.\n(ClusterONE only) (default: nodes)")), ToolInput(tag="in_optimize", input_type=Boolean(optional=True), prefix="--optimize", doc=InputDocumentation(doc="Optimize hierarchical distances during second-pass of\nthe viral clusters (default: False)")), ToolInput(tag="in_sig", input_type=Float(optional=True), prefix="--sig", doc=InputDocumentation(doc="Significance threshold in the contig similarity\nnetwork. (default: 1.0)")), ToolInput(tag="in_max_sig", input_type=Int(optional=True), prefix="--max-sig", doc=InputDocumentation(doc="Significance threshold in the contig similarity\nnetwork. (default: 300)")), ToolInput(tag="in_permissive", input_type=Boolean(optional=True), prefix="--permissive", doc=InputDocumentation(doc="Use permissive affiliation for associating VCs with\nreference sequences. (default: False)")), ToolInput(tag="in_mod_inflation", input_type=Float(optional=True), prefix="--mod-inflation", doc=InputDocumentation(doc="Inflation parameter to define protein modules with\nMCL. (default: 5.0)")), ToolInput(tag="in_mod_sig", input_type=Float(optional=True), prefix="--mod-sig", doc=InputDocumentation(doc="Significance threshold in the protein cluster\nsimilarity network. (default: 1.0)")), ToolInput(tag="in_mod_shared_min", input_type=Int(optional=True), prefix="--mod-shared-min", doc=InputDocumentation(doc="Minimal number (inclusive) of contigs a PC must appear\ninto to be taken into account in the modules\ncomputing. (default: 3)")), ToolInput(tag="in_link_sig", input_type=Float(optional=True), prefix="--link-sig", doc=InputDocumentation(doc="Significitaivity threshold to link a cluster and a\nmodule (default: 1.0)")), ToolInput(tag="in_link_prop", input_type=Float(optional=True), prefix="--link-prop", doc=InputDocumentation(doc="Proportion of a module's PC a contig must have to be\nconsidered as displaying this module. (default: 0.5)")), ToolInput(tag="in_export_backend_suported", input_type=Boolean(optional=True), prefix="-e", doc=InputDocumentation(doc="[EXPORTS [EXPORTS ...]], --exports [EXPORTS [EXPORTS ...]]\nExport backend. Suported values are 'csv', 'krona' and\n'cytoscape' (default: ['csv'])")), ToolInput(tag="in_cluster_filter", input_type=Boolean(optional=True), prefix="--cluster-filter", doc=InputDocumentation(doc="[CLUSTER_FILTER [CLUSTER_FILTER ...]]\nId of the clusters to export (Cytoscape export only).\n(default: [0])")), ToolInput(tag="in_criterion", input_type=String(optional=True), prefix="--criterion", doc=InputDocumentation(doc="Pooling criterion for cluster export (Cytoscape export\nonly). (default: predicted_family)")), ToolInput(tag="in_contigs", input_type=File(optional=True), prefix="--contigs", doc=InputDocumentation(doc="Contig info file (tsv or csv) (default: None)")), ToolInput(tag="in_pcs", input_type=File(optional=True), prefix="--pcs", doc=InputDocumentation(doc="Protein clusters info file (tsv or csv) (default:\nNone)")), ToolInput(tag="in_pc_profiles", input_type=String(optional=True), prefix="--pc-profiles", doc=InputDocumentation(doc="Protein cluster profiles of the contigs (tsv or csv)\n(default: None)")), ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="--verbose", doc=InputDocumentation(doc="Verbosity level : -v warning, -vv info, -vvv debug,\n(default info) (default: -2)")), ToolInput(tag="in_force_overwrite", input_type=Boolean(optional=True), prefix="--force-overwrite", doc=InputDocumentation(doc="Overwrite existing files (default: False)\n"))], outputs=[ToolOutput(tag="out_output_dir", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_output_dir", type_hint=File()), doc=OutputDocumentation(doc="Output directory (default: vContact_Output)"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Vcontact2_V0_1_0().translate("wdl", allow_empty_container=True)

