class: CommandLineTool
id: gplas.sh.cwl
inputs:
- id: in_mandatory_path_graph
  doc: 'Mandatory: Path to the graph file in *.gfa format used to extract nodes and
    links. Gfa file format'
  type: boolean?
  inputBinding:
    prefix: -i
- id: in_mandatory_classifier_used
  doc: "Mandatory: Classifier used to predict the contigs extracted from the input\
    \ graph. String value: 'plasflow' or 'mlplasmids'"
  type: boolean?
  inputBinding:
    prefix: -c
- id: in_mandatory_mlplasmids_chosen
  doc: "Mandatory (if mlplasmids is chosen): Bacterial species from the graph file.\
    \ If you have specified mlplasmids as classifier\nyou need to provide one of the\
    \ following three bacterial species:\n'Enterococcus faecium','Klebsiella pneumoniae'\
    \ or 'Escherichia coli'"
  type: boolean?
  inputBinding:
    prefix: -s
- id: in_optional_output_name
  doc: "Optional: Output name used in the files generated by gplas. Default: 'unnamed'"
  type: boolean?
  inputBinding:
    prefix: -n
- id: in_optional_threshold_predict
  doc: "Optional: Threshold to predict plasmid-derived sequences. Integer value ranging\
    \ from 0 to 1.\nDefault mlplasmids threshold: 0.5\nDefault plasflow threshold:\
    \ 0.7"
  type: boolean?
  inputBinding:
    prefix: -t
- id: in_optional_number_times
  doc: "Optional: Number of times gplas finds plasmid walks per each plasmid starting\
    \ node. Integer value ranging from 1 to infinite.\nDefault: 20"
  type: boolean?
  inputBinding:
    prefix: -x
- id: in_optional_gplas_filtering
  doc: "Optional: Gplas filtering threshold score to reject possible outcoming edges.\
    \ Integer value ranging from 0 to 1.\nDefault: 0.1"
  type: boolean?
  inputBinding:
    prefix: -f
- id: in_optional_modularity_threshold
  doc: "Optional: Modularity threshold to split components present in the plasmidome\
    \ network. Integer value ranging from 0 to 1\nDefault: 0.2"
  type: boolean?
  inputBinding:
    prefix: -q
- id: in_optional_path_corresponding
  doc: "Optional: Path to the complete reference genome corresponding to the graph\
    \ given. For optimal results using this\nbenchmarking flag, please name the reference\
    \ genomes using the Unicycler scheme: e.g. '1 length=4123456' '2 length=10000'\
    \ '3 length=2000'\nfor your chromosome and plasmids. Fasta file format\n"
  type: boolean?
  inputBinding:
    prefix: -r
outputs:
- id: out_stdout
  doc: Standard output stream
  type: stdout
hints:
- class: DockerRequirement
  dockerPull: quay.io/biocontainers/gplas:0.6.1--0
cwlVersion: v1.1
baseCommand:
- gplas.sh
