from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Int, Boolean

Chipassoc_V0_1_0 = CommandToolBuilder(tool="ChIPAssoc", base_command=["ChIPAssoc"], inputs=[ToolInput(tag="in_bed", input_type=File(optional=True), prefix="--bed", doc=InputDocumentation(doc="BED file of genomic coordinates (e.g. ChIP-Seq peaks).\nThe center of each peak will be used to compute the\ndistance from a gene.")), ToolInput(tag="in_db", input_type=File(optional=True), prefix="--db", doc=InputDocumentation(doc="Gene annotation table. This can be a sqlite3 local db\nfile, BED file or genome version of UCSC. The BED file\nmust have an extension of '.bed'")), ToolInput(tag="in_g_set", input_type=File(optional=True), prefix="--gset", doc=InputDocumentation(doc="Gene set to see the association with the genomic\ncoordinates given through -b. Multiple gene sets can\nbe given by repeatedly using this option (e.g. -g\ngeneset1.txt -g geneset2.txt -g geneset3.txt). Genes\nmust be given in a TXT file each line of which has a\nrefseq accession ID or an official gene symbol (i.e. a\nsingle-column TXT file).")), ToolInput(tag="in_min", input_type=Int(optional=True), prefix="--min", doc=InputDocumentation(doc="The lower limit of the distance to consider in\nassociation analysis (in kb). Note that the lower and\nupper limits must be reasonably apart to obtain a\nmeaningful result. By default 0kb.")), ToolInput(tag="in_max", input_type=Int(optional=True), prefix="--max", doc=InputDocumentation(doc="The upper limit of the distance to consider in\nassociation analysis (in kb). Note that the lower and\nupper limits must be reasonably apart to obtain a\nmeaningful result. By default 200kb.")), ToolInput(tag="in_name", input_type=File(optional=True), prefix="--name", doc=InputDocumentation(doc="Experiment name. This will be used to name the output\nfile. If an experiment name is not given, input BED\nfile name will be used instead.")), ToolInput(tag="in_lab", input_type=Int(optional=True), prefix="--lab", doc=InputDocumentation(doc="Label for each gene set. Likewise, multiple gene set\nlabels can be given by repeatedly using this option\n(e.g. -l label1 -l label2 -l label3). If labels are\nnot given, 'gene set' will be used by default.")), ToolInput(tag="in_rbg", input_type=File(optional=True), prefix="--rbg", doc=InputDocumentation(doc="Background gene set. If a set of genes is given using\nthis option, the gene set will be used as background\nor null set when running KS test. Otherwise, all\nrefseq genes will be used as background. Background\ngenes must be given in a TXT file each line of which\nhas a refseq accession ID or an offical gene symbol\n(i.e. a single-column TXT file).")), ToolInput(tag="in_g_name_two", input_type=Boolean(optional=True), prefix="--gname2", doc=InputDocumentation(doc="If this switch is on, gene or transcript IDs in files\ngiven through -g will be considered as official gene\nsymbols.\n"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Chipassoc_V0_1_0().translate("wdl", allow_empty_container=True)

