class: CommandLineTool
id: NanoPlot.cwl
inputs:
- id: in_threads
  doc: Set the allowed number of threads to be used by the script
  type: long
  inputBinding:
    prefix: --threads
- id: in_verbose
  doc: Write log messages also to terminal.
  type: boolean
  inputBinding:
    prefix: --verbose
- id: in_store
  doc: Store the extracted data in a pickle file for future plotting.
  type: boolean
  inputBinding:
    prefix: --store
- id: in_raw
  doc: Store the extracted data in tab separated file.
  type: boolean
  inputBinding:
    prefix: --raw
- id: in_huge
  doc: Input data is one very large file.
  type: boolean
  inputBinding:
    prefix: --huge
- id: in_outdir
  doc: Specify directory in which output has to be created.
  type: Directory
  inputBinding:
    prefix: --outdir
- id: in_prefix
  doc: Specify an optional prefix to be used for the output files.
  type: string
  inputBinding:
    prefix: --prefix
- id: in_maxlength
  doc: Hide reads longer than length specified.
  type: long
  inputBinding:
    prefix: --maxlength
- id: in_minlength
  doc: Hide reads shorter than length specified.
  type: long
  inputBinding:
    prefix: --minlength
- id: in_drop_outliers
  doc: Drop outlier reads with extreme long length.
  type: boolean
  inputBinding:
    prefix: --drop_outliers
- id: in_down_sample
  doc: Reduce dataset to N reads by random sampling.
  type: string
  inputBinding:
    prefix: --downsample
- id: in_log_length
  doc: Additionally show logarithmic scaling of lengths in plots.
  type: boolean
  inputBinding:
    prefix: --loglength
- id: in_percent_qual
  doc: Use qualities as theoretical percent identities.
  type: boolean
  inputBinding:
    prefix: --percentqual
- id: in_a_length
  doc: Use aligned read lengths rather than sequenced length (bam mode)
  type: boolean
  inputBinding:
    prefix: --alength
- id: in_min_qual
  doc: Drop reads with an average quality lower than specified.
  type: string
  inputBinding:
    prefix: --minqual
- id: in_runtime_until
  doc: Only take the N first hours of a run
  type: string
  inputBinding:
    prefix: --runtime_until
- id: in_read_type
  doc: "Which read type to extract information about from summary. Options are 1D,\
    \ 2D,\n1D2"
  type: string
  inputBinding:
    prefix: --readtype
- id: in_barcoded
  doc: Use if you want to split the summary file by barcode
  type: boolean
  inputBinding:
    prefix: --barcoded
- id: in_no_supplementary
  doc: Use if you want to remove supplementary alignments
  type: boolean
  inputBinding:
    prefix: --no_supplementary
- id: in_color
  doc: Specify a valid matplotlib color for the plots
  type: string
  inputBinding:
    prefix: --color
- id: in_colormap
  doc: Specify a valid matplotlib colormap for the heatmap
  type: string
  inputBinding:
    prefix: --colormap
- id: in_format
  doc: Specify the output format of the plots.
  type: string
  inputBinding:
    prefix: --format
- id: in_plots
  doc: "[{kde,hex,dot,pauvre} [{kde,hex,dot,pauvre} ...]]\nSpecify which bivariate\
    \ plots have to be made."
  type: boolean
  inputBinding:
    prefix: --plots
- id: in_list_colors
  doc: List the colors which are available for plotting and exit.
  type: boolean
  inputBinding:
    prefix: --listcolors
- id: in_list_colormaps
  doc: List the colors which are available for plotting and exit.
  type: boolean
  inputBinding:
    prefix: --listcolormaps
- id: in_no_n_five_zero
  doc: Hide the N50 mark in the read length histogram
  type: boolean
  inputBinding:
    prefix: --no-N50
- id: in_n_five_zero
  doc: Show the N50 mark in the read length histogram
  type: boolean
  inputBinding:
    prefix: --N50
- id: in_title
  doc: Add a title to all plots, requires quoting if using spaces
  type: string
  inputBinding:
    prefix: --title
- id: in_font_scale
  doc: Scale the font of the plots by a factor
  type: string
  inputBinding:
    prefix: --font_scale
- id: in_dpi
  doc: Set the dpi for saving images
  type: string
  inputBinding:
    prefix: --dpi
- id: in_hide_stats
  doc: Not adding Pearson R stats in some bivariate plots
  type: boolean
  inputBinding:
    prefix: --hide_stats
- id: in_fast_q
  doc: Data is in one or more default fastq file(s).
  type: File[]
  inputBinding:
    prefix: --fastq
- id: in_fast_a
  doc: Data is in one or more fasta file(s).
  type: File[]
  inputBinding:
    prefix: --fasta
- id: in_fast_q_rich
  doc: "Data is in one or more fastq file(s) generated by albacore, MinKNOW or guppy\n\
    with additional information concerning channel and time."
  type: File[]
  inputBinding:
    prefix: --fastq_rich
- id: in_fast_q_minimal
  doc: "Data is in one or more fastq file(s) generated by albacore, MinKNOW or guppy\n\
    with additional information concerning channel and time. Is extracted swiftly\n\
    without elaborate checks."
  type: File[]
  inputBinding:
    prefix: --fastq_minimal
- id: in_summary
  doc: Data is in one or more summary file(s) generated by albacore or guppy.
  type: File[]
  inputBinding:
    prefix: --summary
- id: in_bam
  doc: Data is in one or more sorted bam file(s).
  type: File[]
  inputBinding:
    prefix: --bam
- id: in_ub_am
  doc: Data is in one or more unmapped bam file(s).
  type: File[]
  inputBinding:
    prefix: --ubam
- id: in_cram
  doc: Data is in one or more sorted cram file(s).
  type: File[]
  inputBinding:
    prefix: --cram
- id: in_pickle
  doc: Data is a pickle file stored earlier.
  type: File
  inputBinding:
    prefix: --pickle
- id: in_v
  doc: ''
  type: boolean
  inputBinding:
    prefix: -v
outputs:
- id: out_stdout
  doc: Standard output stream
  type: stdout
- id: out_outdir
  doc: Specify directory in which output has to be created.
  type: Directory
  outputBinding:
    glob: $(inputs.in_outdir)
cwlVersion: v1.1
baseCommand:
- NanoPlot
