from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Directory, File, String, Int

Chisel_V0_1_0 = CommandToolBuilder(tool="chisel", base_command=["chisel"], inputs=[ToolInput(tag="in_run_dir", input_type=Directory(optional=True), prefix="--rundir", doc=InputDocumentation(doc="Running directory (default: current directory)")), ToolInput(tag="in_tumor", input_type=File(optional=True), prefix="--tumor", doc=InputDocumentation(doc="Barcoded single-cell BAM file")), ToolInput(tag="in_normal", input_type=File(optional=True), prefix="--normal", doc=InputDocumentation(doc="Matched-normal BAM file")), ToolInput(tag="in_reference", input_type=String(optional=True), prefix="--reference", doc=InputDocumentation(doc="Reference genome")), ToolInput(tag="in_list_phased", input_type=File(optional=True), prefix="--listphased", doc=InputDocumentation(doc="Phased SNPs file (lines of heterozygous germline SNPs\nmust contain either 0|1 or 1|0)")), ToolInput(tag="in_size", input_type=Int(optional=True), prefix="--size", doc=InputDocumentation(doc="Bin size, with or without 'kb' or 'Mb'")), ToolInput(tag="in_block_size", input_type=Int(optional=True), prefix="--blocksize", doc=InputDocumentation(doc="Size of the haplotype blocks (default: 50kb, use 0 to\ndisable)")), ToolInput(tag="in_chromosomes", input_type=Int(optional=True), prefix="--chromosomes", doc=InputDocumentation(doc="Space-separeted list of chromosomes between apices\n(default: 'chr1 chr2 chr3 chr4 chr5 chr6 chr7 chr8\nchr9 chr10 chr11 chr12 chr13 chr14 chr15 chr16 chr17\nchr18 chr19 chr20 chr21 chr22')")), ToolInput(tag="in_min_reads", input_type=Int(optional=True), prefix="--minreads", doc=InputDocumentation(doc="Minimum number total reads to select cells (default:\n500000)")), ToolInput(tag="in_max_ploidy", input_type=Int(optional=True), prefix="--maxploidy", doc=InputDocumentation(doc="Maximum total copy number to consider for balanced\ncluster (default: 4, corresponding to a WGD)")), ToolInput(tag="in_upper_k", input_type=Int(optional=True), prefix="--upperk", doc=InputDocumentation(doc="Maximum number of bin clusters (default: 100, use 0 to\nconsider maximum number of clusters)")), ToolInput(tag="in_bcf_tools", input_type=File(optional=True), prefix="--bcftools", doc=InputDocumentation(doc="Path to the directory to 'bcftools' executable,\nrequired in default mode (default: bcftools is\ndirectly called as it is in user $PATH)")), ToolInput(tag="in_sam_tools", input_type=File(optional=True), prefix="--samtools", doc=InputDocumentation(doc="Path to the directory to 'samtools' executable,\nrequired in default mode (default: samtools is\ndirectly called as it is in user $PATH)")), ToolInput(tag="in_seed", input_type=String(optional=True), prefix="--seed", doc=InputDocumentation(doc="Random seed for replication (default: None)")), ToolInput(tag="in_jobs", input_type=Int(optional=True), prefix="--jobs", doc=InputDocumentation(doc="Number of parallele jobs to use (default: equal to\nnumber of available processors)\n"))], outputs=[], container="quay.io/biocontainers/chisel:0.0.5--py_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Chisel_V0_1_0().translate("wdl")

