from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, Int, File, String, Float

Run_Discosnprad_Sh_V0_1_0 = CommandToolBuilder(tool="run_discoSnpRad.sh", base_command=["run_discoSnpRad.sh"], inputs=[ToolInput(tag="in_f_of", input_type=Boolean(optional=True), prefix="--fof", doc=InputDocumentation(doc="<file name of a file of file(s)>\nThe input read files indicated in a file of file(s)\nExample: -r bank.fof with bank.fof containing the two lines\ndata_sample/reads_sequence1.fasta\ndata_sample/reads_sequence2.fasta.gz")), ToolInput(tag="in_k_size", input_type=Int(optional=True), prefix="--k_size", doc=InputDocumentation(doc="<int value>\nSet the length of used kmers. Must fit the compiled value.\nDefault=31")), ToolInput(tag="in_min_coverage", input_type=Int(optional=True), prefix="--min_coverage", doc=InputDocumentation(doc="<int value>\nSet the minimal coverage per read set: Used by kissnp2 (don't use kmers with lower coverage) and kissreads (read coherency threshold).\nThis coverage can be automatically detected per read set (in this case use 'auto' or specified per read set, see the documentation.)\nDefault=3")), ToolInput(tag="in_high_precision", input_type=Boolean(optional=True), prefix="--high_precision", doc=InputDocumentation(doc="lower recall / higher precision mode. With this parameter no symmetrical crossroads may be traversed during bubble detection (by default up to 5 symmetrical crossroads may be traversed during bubble detection).")), ToolInput(tag="in_graph", input_type=File(optional=True), prefix="--graph", doc=InputDocumentation(doc="Reuse a previously created graph (.h5 file)")), ToolInput(tag="in_prefix", input_type=String(optional=True), prefix="--prefix", doc=InputDocumentation(doc="All out files will start with this prefix. Default='discoRes'")), ToolInput(tag="in_max_coverage", input_type=Int(optional=True), prefix="--max_coverage", doc=InputDocumentation(doc="Set the maximal coverage for each read set: Used by kissnp2 (don't use kmers with higher coverage).\nDefault=2^31-1")), ToolInput(tag="in_no_low_complexity", input_type=Boolean(optional=True), prefix="--no_low_complexity", doc=InputDocumentation(doc="Remove low complexity bubbles")), ToolInput(tag="in_deletion_max_size", input_type=Int(optional=True), prefix="--deletion_max_size", doc=InputDocumentation(doc="discoSnpRad will search for deletions of size from 0 to D included. Default=0 (no deletion)")), ToolInput(tag="in_src", input_type=Boolean(optional=True), prefix="--src", doc=InputDocumentation(doc="[src_path]\nperforms clustering of variants with short_read_connector\nsrc_path: **absolute** path to short_read_connector directory, containing the 'short_read_connector.sh' file.\n-Note1: short read connector must be compiled.\n-Note2: if no value is given, it assumes short_read_connector.sh is in the PATH env variable.\n-Note3: with this option, discoSnpRad outputs a vcf file containing the variants clustered by locus")), ToolInput(tag="in_max_size_cluster", input_type=Int(optional=True), prefix="--max_size_cluster", doc=InputDocumentation(doc="Discards cluster containing more than this number of variants. (Default 150)\nRequires the -S or --src option")), ToolInput(tag="in_max_missing", input_type=Float(optional=True), prefix="--max_missing", doc=InputDocumentation(doc="Remove variants with more than max_missing % missing values. (Default 0.95, removes variants detected in 5% and less populations)")), ToolInput(tag="in_min_rank", input_type=Float(optional=True), prefix="--min_rank", doc=InputDocumentation(doc="Remove variants whose rank is smaller than this threshold. (Default 0.4)")), ToolInput(tag="in_max_snp_per_bubble", input_type=Int(optional=True), prefix="--max_snp_per_bubble", doc=InputDocumentation(doc="discoSnpRad will search up to P SNPs in a unique bubble. Default=5")), ToolInput(tag="in_max_substitutions", input_type=Int(optional=True), prefix="--max_substitutions", doc=InputDocumentation(doc="Set the number of authorized substitutions used while mapping reads on found SNPs (kissreads). Default=10")), ToolInput(tag="in_max_threads", input_type=Int(optional=True), prefix="--max_threads", doc=InputDocumentation(doc="Max number of used threads. 0 means all threads\ndefault 0")), ToolInput(tag="in_wraith_mode_only", input_type=Boolean(optional=True), prefix="-w", doc=InputDocumentation(doc="Wraith mode: only show all discoSnpRad commands without running them")), ToolInput(tag="in_verbose_avoids_output", input_type=Int(optional=True), prefix="-v", doc=InputDocumentation(doc="Verbose 0 (avoids progress output) or 1 (enables progress output) -- default=1."))], outputs=[], container="quay.io/biocontainers/discosnp:2.5.4--h8b12597_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Run_Discosnprad_Sh_V0_1_0().translate("wdl")

