from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Int, String, Boolean, File, Float

Sage_V0_1_0 = CommandToolBuilder(tool="SAGE", base_command=["SAGE"], inputs=[ToolInput(tag="in_base_qual_fixed_penalty", input_type=Int(optional=True), prefix="-base_qual_fixed_penalty", doc=InputDocumentation(doc="Fixed penalty to\napply to base\nquality [12]")), ToolInput(tag="in_b_qr_enabled", input_type=String(optional=True), prefix="-bqr_enabled", doc=InputDocumentation(doc="BQR (Base Quality\nRecalibration)\nenabled [true]")), ToolInput(tag="in_b_qr_max_alt_count", input_type=Int(optional=True), prefix="-bqr_max_alt_count", doc=InputDocumentation(doc="BQR maximum alt\ncount to be an\nerror [3]")), ToolInput(tag="in_b_qr_min_map_qual", input_type=Int(optional=True), prefix="-bqr_min_map_qual", doc=InputDocumentation(doc="BQR min base\nquality remap\nqual [10]")), ToolInput(tag="in_b_qr_plot", input_type=String(optional=True), prefix="-bqr_plot", doc=InputDocumentation(doc="BQR plots [true]")), ToolInput(tag="in_b_qr_sample_size", input_type=Int(optional=True), prefix="-bqr_sample_size", doc=InputDocumentation(doc="BQR sampling size\nper autosome\n[2000000]")), ToolInput(tag="in_chr", input_type=String(optional=True), prefix="-chr", doc=InputDocumentation(doc="Run for single")), ToolInput(tag="in_hard_filter_enabled", input_type=Boolean(optional=True), prefix="-hard_filter_enabled", doc=InputDocumentation(doc="All filters are\nhard [false]")), ToolInput(tag="in_hard_min_tumor_qual", input_type=Int(optional=True), prefix="-hard_min_tumor_qual", doc=InputDocumentation(doc="Hard minimum\ntumor quality\n[30]")), ToolInput(tag="in_hard_min_tumor_raw_alt_support", input_type=Int(optional=True), prefix="-hard_min_tumor_raw_alt_support", doc=InputDocumentation(doc="Hard minimum\ntumor raw alt\nsupport [2]")), ToolInput(tag="in_hard_min_tumor_raw_base_quality", input_type=Int(optional=True), prefix="-hard_min_tumor_raw_base_quality", doc=InputDocumentation(doc="Hard minimum\ntumor raw base\nquality [0]")), ToolInput(tag="in_high_confidence_bed", input_type=File(optional=True), prefix="-high_confidence_bed", doc=InputDocumentation(doc="High confidence\nregions bed file")), ToolInput(tag="in_high_confidence_max_germline_rel_raw_base_qual", input_type=String(optional=True), prefix="-high_confidence_max_germline_rel_raw_base_qual", doc=InputDocumentation(doc="Maximum")), ToolInput(tag="in_high_confidence_max_germline_vaf", input_type=String(optional=True), prefix="-high_confidence_max_germline_vaf", doc=InputDocumentation(doc="Maximum")), ToolInput(tag="in_high_confidence_min_germline_depth", input_type=String(optional=True), prefix="-high_confidence_min_germline_depth", doc=InputDocumentation(doc="Minimum")), ToolInput(tag="in_high_confidence_min_germline_depth_allo_some", input_type=String(optional=True), prefix="-high_confidence_min_germline_depth_allosome", doc=InputDocumentation(doc="Minimum")), ToolInput(tag="in_high_confidence_min_tumor_qual", input_type=String(optional=True), prefix="-high_confidence_min_tumor_qual", doc=InputDocumentation(doc="Minimum")), ToolInput(tag="in_high_confidence_min_tumor_vaf", input_type=String(optional=True), prefix="-high_confidence_min_tumor_vaf", doc=InputDocumentation(doc="Minimum")), ToolInput(tag="in_highly_polymorphic_genes", input_type=Int(optional=True), prefix="-highly_polymorphic_genes", doc=InputDocumentation(doc="Genes to exclude\nevent distance\npenalty\n[HLA-A,HLA-B,HLA-\nC,HLA-DQA1,HLA-DQ\nB1,HLA-DRB1]")), ToolInput(tag="in_hotspot_max_germline_rel_raw_base_qual", input_type=Float(optional=True), prefix="-hotspot_max_germline_rel_raw_base_qual", doc=InputDocumentation(doc="Maximum hotspot\ngermline relative\nquality [0.5]")), ToolInput(tag="in_hotspot_max_germline_vaf", input_type=Float(optional=True), prefix="-hotspot_max_germline_vaf", doc=InputDocumentation(doc="Maximum hotspot\ngermline VAF\n[0.1]")), ToolInput(tag="in_hotspot_min_germline_depth", input_type=Int(optional=True), prefix="-hotspot_min_germline_depth", doc=InputDocumentation(doc="Minimum hotspot\ngermline depth\n[0]")), ToolInput(tag="in_hotspot_min_germline_depth_allo_some", input_type=Int(optional=True), prefix="-hotspot_min_germline_depth_allosome", doc=InputDocumentation(doc="Minimum hotspot\ngermline depth\n[0]")), ToolInput(tag="in_hotspot_min_tumor_qual", input_type=Int(optional=True), prefix="-hotspot_min_tumor_qual", doc=InputDocumentation(doc="Minimum hotspot\ntumor quality\n[70]")), ToolInput(tag="in_hotspot_min_tumor_vaf", input_type=Float(optional=True), prefix="-hotspot_min_tumor_vaf", doc=InputDocumentation(doc="Minimum hotspot\ntumor VAF [0.005]")), ToolInput(tag="in_hotspots", input_type=String(optional=True), prefix="-hotspots", doc=InputDocumentation(doc="Hotspots")), ToolInput(tag="in_jitter_min_repeat_count", input_type=Int(optional=True), prefix="-jitter_min_repeat_count", doc=InputDocumentation(doc="Minimum repeat\ncount before\napplying jitter\npenalty [3]")), ToolInput(tag="in_jitter_penalty", input_type=Float(optional=True), prefix="-jitter_penalty", doc=InputDocumentation(doc="Penalty to apply\nto qual score\nwhen read context\nmatches with\njitter [0.25]")), ToolInput(tag="in_low_confidence_max_germline_rel_raw_base_qual", input_type=String(optional=True), prefix="-low_confidence_max_germline_rel_raw_base_qual", doc=InputDocumentation(doc="Maximum")), ToolInput(tag="in_low_confidence_max_germline_vaf", input_type=String(optional=True), prefix="-low_confidence_max_germline_vaf", doc=InputDocumentation(doc="Maximum")), ToolInput(tag="in_low_confidence_min_germline_depth", input_type=String(optional=True), prefix="-low_confidence_min_germline_depth", doc=InputDocumentation(doc="Minimum")), ToolInput(tag="in_low_confidence_min_germline_depth_allo_some", input_type=String(optional=True), prefix="-low_confidence_min_germline_depth_allosome", doc=InputDocumentation(doc="Minimum")), ToolInput(tag="in_low_confidence_min_tumor_qual", input_type=String(optional=True), prefix="-low_confidence_min_tumor_qual", doc=InputDocumentation(doc="Minimum")), ToolInput(tag="in_low_confidence_min_tumor_vaf", input_type=String(optional=True), prefix="-low_confidence_min_tumor_vaf", doc=InputDocumentation(doc="Minimum")), ToolInput(tag="in_map_qual_fixed_penalty", input_type=Int(optional=True), prefix="-map_qual_fixed_penalty", doc=InputDocumentation(doc="Fixed penalty to\napply to map\nquality [15]")), ToolInput(tag="in_map_qual_improper_pair_penalty", input_type=Int(optional=True), prefix="-map_qual_improper_pair_penalty", doc=InputDocumentation(doc="Penalty to apply\nto map qual when\nSAM record does\nnot have the\nProperPair flag\n[15]")), ToolInput(tag="in_map_qual_read_events_penalty", input_type=Int(optional=True), prefix="-map_qual_read_events_penalty", doc=InputDocumentation(doc="Penalty to apply\nto map qual for\nadditional\ndistance from ref\n[8]")), ToolInput(tag="in_max_read_depth", input_type=Int(optional=True), prefix="-max_read_depth", doc=InputDocumentation(doc="Max depth to look\nfor evidence\n[1000]")), ToolInput(tag="in_max_read_depth_panel", input_type=Int(optional=True), prefix="-max_read_depth_panel", doc=InputDocumentation(doc="Max depth to look\nfor evidence in\npanel [100000]")), ToolInput(tag="in_max_realignment_depth", input_type=Int(optional=True), prefix="-max_realignment_depth", doc=InputDocumentation(doc="Max depth to\ncheck for\nrealignment\n[1000]")), ToolInput(tag="in_min_map_quality", input_type=Int(optional=True), prefix="-min_map_quality", doc=InputDocumentation(doc="Min map quality\nto apply to\nnon-hotspot\nvariants [10]")), ToolInput(tag="in_mnv_enabled", input_type=String(optional=True), prefix="-mnv_enabled", doc=InputDocumentation(doc="Enable MNVs\n[true]")), ToolInput(tag="in_mnv_filter_enabled", input_type=Boolean(optional=True), prefix="-mnv_filter_enabled", doc=InputDocumentation(doc="Enable")), ToolInput(tag="in_out", input_type=File(optional=True), prefix="-out", doc=InputDocumentation(doc="Path to output")), ToolInput(tag="in_panel_max_germline_vaf", input_type=Float(optional=True), prefix="-panel_max_germline_vaf", doc=InputDocumentation(doc="Maximum panel\ngermline VAF\n[0.04]")), ToolInput(tag="in_panel_min_germline_depth", input_type=Int(optional=True), prefix="-panel_min_germline_depth", doc=InputDocumentation(doc="Minimum panel\ngermline depth\n[0]")), ToolInput(tag="in_panel_min_germline_depth_allo_some", input_type=Int(optional=True), prefix="-panel_min_germline_depth_allosome", doc=InputDocumentation(doc="Minimum panel\ngermline depth\n[0]")), ToolInput(tag="in_panel_min_tumor_qual", input_type=Int(optional=True), prefix="-panel_min_tumor_qual", doc=InputDocumentation(doc="Minimum panel\ntumor quality\n[100]")), ToolInput(tag="in_panel_min_tumor_vaf", input_type=Float(optional=True), prefix="-panel_min_tumor_vaf", doc=InputDocumentation(doc="Minimum panel\ntumor VAF [0.015]")), ToolInput(tag="in_panel_only", input_type=Boolean(optional=True), prefix="-panel_only", doc=InputDocumentation(doc="Only examine\npanel for\nvariants")), ToolInput(tag="in_read_context_flank_size", input_type=Int(optional=True), prefix="-read_context_flank_size", doc=InputDocumentation(doc="Size of read\ncontext flank\n[10]")), ToolInput(tag="in_read_edge_fixed_penalty", input_type=Int(optional=True), prefix="-read_edge_fixed_penalty", doc=InputDocumentation(doc="Fixed penalty to\napply to distance\nfrom read edge\n[0]")), ToolInput(tag="in_ref_genome", input_type=File(optional=True), prefix="-ref_genome", doc=InputDocumentation(doc="Path to indexed\nref genome fasta\nfile")), ToolInput(tag="in_reference", input_type=String(optional=True), prefix="-reference", doc=InputDocumentation(doc="Name of reference")), ToolInput(tag="in_slice_size", input_type=Int(optional=True), prefix="-slice_size", doc=InputDocumentation(doc="Slice size\n[100000]")), ToolInput(tag="in_soft_filter_enabled", input_type=Boolean(optional=True), prefix="-soft_filter_enabled", doc=InputDocumentation(doc="Enable soft\nfilters [true]")), ToolInput(tag="in_threads", input_type=Int(optional=True), prefix="-threads", doc=InputDocumentation(doc="Number of threads\n[2]")), ToolInput(tag="in_tumor", input_type=String(optional=True), prefix="-tumor", doc=InputDocumentation(doc="Name of tumor")), ToolInput(tag="in_sage_application", input_type=String(), position=0, doc=InputDocumentation(doc="-assembly <arg>                                         Assembly, must be\none of [hg19,\nhg38]")), ToolInput(tag="in_chromosome", input_type=String(), position=1, doc=InputDocumentation(doc="-coverage_bed <arg>                                     Coverage is")), ToolInput(tag="in_optionally", input_type=String(), position=0, doc=InputDocumentation(doc="supplied bed")), ToolInput(tag="in_max_germline_alt", input_type=String(), position=1, doc=InputDocumentation(doc="support mnv")), ToolInput(tag="in_vcf", input_type=String(), position=0, doc=InputDocumentation(doc="-panel_bed <arg>                                        Panel regions bed")), ToolInput(tag="in_panelmaxgermlinerelrawbasequal_arg_maximum", input_type=File(), position=1, doc=InputDocumentation(doc="-panel_max_germline_rel_raw_base_qual <arg>             Maximum panel")), ToolInput(tag="in_validationstringency_arg_sam", input_type=File(), position=1, doc=InputDocumentation(doc="-validation_stringency <arg>                            SAM validation"))], outputs=[ToolOutput(tag="out_out", output_type=File(optional=True), selector=InputSelector(input_to_select="in_out", type_hint=File()), doc=OutputDocumentation(doc="Path to output"))], container="quay.io/biocontainers/hmftools-sage:2.7--0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Sage_V0_1_0().translate("wdl")

