from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, Directory

Mapsplice_Py_V0_1_0 = CommandToolBuilder(tool="mapsplice.py", base_command=["mapsplice.py"], inputs=[ToolInput(tag="in_chromosome_dir", input_type=Boolean(optional=True), prefix="--chromosome-dir", doc=InputDocumentation(doc="<string>     reference sequence directory")), ToolInput(tag="in_string_path_prefix", input_type=Boolean(optional=True), prefix="-x", doc=InputDocumentation(doc="<string>     path and prefix of bowtie index")), ToolInput(tag="in_string_end_", input_type=Boolean(optional=True), prefix="-1", doc=InputDocumentation(doc="/                            <string>     end 1 reads / single end reads")), ToolInput(tag="in_string_end_reads", input_type=Boolean(optional=True), prefix="-2", doc=InputDocumentation(doc="/                            <string>     end 2 reads")), ToolInput(tag="in_threads", input_type=Boolean(optional=True), prefix="--threads", doc=InputDocumentation(doc="<int>        number of threads, default: 1")), ToolInput(tag="in_output", input_type=Directory(optional=True), prefix="--output", doc=InputDocumentation(doc="<string>     output directory, default: ./mapsplice_out")), ToolInput(tag="in_qual_scale", input_type=Boolean(optional=True), prefix="--qual-scale", doc=InputDocumentation(doc="<string>     quality scale, phred64(default) or phred33 or solexa64")), ToolInput(tag="in_bam", input_type=Boolean(optional=True), prefix="--bam", doc=InputDocumentation(doc="output alignment in BAM format, default: SAM format.")), ToolInput(tag="in_keep_tmp", input_type=Boolean(optional=True), prefix="--keep-tmp", doc=InputDocumentation(doc="keep intermediate files, default: off")), ToolInput(tag="in_se_glen", input_type=Boolean(optional=True), prefix="--seglen", doc=InputDocumentation(doc="<int>        segment length, default: 25")), ToolInput(tag="in_min_map_len", input_type=Boolean(optional=True), prefix="--min-map-len", doc=InputDocumentation(doc="<int>        minimum alignment length, default: 50")), ToolInput(tag="in_maxhits_int_alignments", input_type=Boolean(optional=True), prefix="-k", doc=InputDocumentation(doc="/max-hits                    <int>        maximum alignments per read, default: 4")), ToolInput(tag="in_min_intron", input_type=Boolean(optional=True), prefix="--min-intron", doc=InputDocumentation(doc="<int>        minimum intron length, default: 50")), ToolInput(tag="in_max_intron", input_type=Boolean(optional=True), prefix="--max-intron", doc=InputDocumentation(doc="<int>        maximum intron length, default: 300,000")), ToolInput(tag="in_non_canonical_double_anchor", input_type=Boolean(optional=True), prefix="--non-canonical-double-anchor", doc=InputDocumentation(doc="also search for non-canonical junctions in double anchor case, default: off(same parameter as --non-canonical in versions previous to MapSplice 2.2.0)")), ToolInput(tag="in_non_canonical_single_anchor", input_type=Boolean(optional=True), prefix="--non-canonical-single-anchor", doc=InputDocumentation(doc="also search for non-canonical junctions in single anchor case, default: off")), ToolInput(tag="in_filtering", input_type=Boolean(optional=True), prefix="--filtering", doc=InputDocumentation(doc="<int>        The stringency level of filtering splice junctions in the range of [1, 2].\nDefault is 2.\n1: Less stringent filtering, with higher sensitivity of splice junction detection.\n2: Standard filtering.")), ToolInput(tag="in_splice_mis", input_type=Boolean(optional=True), prefix="--splice-mis", doc=InputDocumentation(doc="<int>        Maximum number of mismatches that are allowed in the first/last segment\ncrossing a splice junction in the range of [0, 2]. Default is 1.\n(Maximum number of mismatches that are allowed in the middle segment\ncrossing a splice junction is always fixed at 2.)")), ToolInput(tag="in_max_append_mis", input_type=Boolean(optional=True), prefix="--max-append-mis", doc=InputDocumentation(doc="<int>        Maximum number of mismatches allowed to append a high error exonic segment\nnext to an adjacent low error segment. Default is 3.")), ToolInput(tag="in_ins", input_type=Boolean(optional=True), prefix="--ins", doc=InputDocumentation(doc="<int>        maximum insertion length, default: 6, must in range [0, 10]")), ToolInput(tag="in_del", input_type=Boolean(optional=True), prefix="--del", doc=InputDocumentation(doc="<int>        maximum deletion length, default: 6")), ToolInput(tag="in_fusion_non_canonical", input_type=Boolean(optional=True), prefix="--fusion-non-canonical", doc=InputDocumentation(doc="also search for fusion junction, default: off")), ToolInput(tag="in_min_fusion_distance", input_type=Boolean(optional=True), prefix="--min-fusion-distance", doc=InputDocumentation(doc="<int>        Minimum distance between two gapped segments to be considered as fusion candidate.\ndefault is 10,000. Please set this to lower value(e.g 200) to be more sensitive to\ncircular RNA detection.")), ToolInput(tag="in_gene_gtf", input_type=Boolean(optional=True), prefix="--gene-gtf", doc=InputDocumentation(doc="<string>     Gene annotation file in GTF format, used to annotate fusion junctions. Can be downloaded\nfrom ENSEMBL ftp site. (e.g, for human hg19: Homo_sapiens.GRCh37.66.gtf.gz). Required\nfor the detection of Circular RNA."))], outputs=[ToolOutput(tag="out_output", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_output", type_hint=File()), doc=OutputDocumentation(doc="<string>     output directory, default: ./mapsplice_out"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Mapsplice_Py_V0_1_0().translate("wdl", allow_empty_container=True)

