from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import String, File, Directory, Int, Boolean, Float

Get_Organelle_From_Assembly_Py_V0_1_0 = CommandToolBuilder(tool="get_organelle_from_assembly.py", base_command=["get_organelle_from_assembly.py"], inputs=[ToolInput(tag="in_flag_fungusmtfungus_rna", input_type=String(optional=True), prefix="-F", doc=InputDocumentation(doc="This flag should be followed with embplant_pt\n(embryophyta plant plastome), other_pt (non-\nembryophyta plant plastome), embplant_mt (plant\nmitochondrion), embplant_nr (plant nuclear ribosomal\nRNA), animal_mt (animal mitochondrion), fungus_mt\n(fungus mitochondrion), fungus_nr (fungus nuclear\nribosomal RNA), or embplant_mt,other_pt,fungus_mt (the\ncombination of any of above organelle genomes split by\ncomma(s), which might be computationally more\nintensive than separate runs), or anonym (uncertain\norganelle genome type, with customized gene database\n('--genes'), which is suggested only when the above\ndatabase is genetically distant from your sample) or\nraw (disentangle the raw graph directly without\ntagging).")), ToolInput(tag="in_input_assembly_graph", input_type=File(optional=True), prefix="-g", doc=InputDocumentation(doc="Input assembly graph (fastg/gfa) file. The format will\nbe recognized by the file name suffix.")), ToolInput(tag="in_output_directory_overwriting", input_type=Directory(optional=True), prefix="-o", doc=InputDocumentation(doc="Output directory. Overwriting files if directory")), ToolInput(tag="in_max_depth", input_type=Int(optional=True), prefix="--max-depth", doc=InputDocumentation(doc="Input a float or integer number. filter graph file by\na maximum depth. Default: inf.")), ToolInput(tag="in_config_dir", input_type=File(optional=True), prefix="--config-dir", doc=InputDocumentation(doc="The directory where the configuration file and default\ndatabases were placed. The default value also can be\nchanged by adding 'export GETORG_PATH=your_favor' to\nthe shell script (e.g. ~/.bash_profile or ~/.bashrc)\nDefault: /root/.GetOrganelle")), ToolInput(tag="in_genes", input_type=Int(optional=True), prefix="--genes", doc=InputDocumentation(doc="Followed with a customized database (a fasta file or\nthe base name of a blast database) containing or made\nof ONE set of protein coding genes and ribosomal RNAs\nextracted from ONE reference genome that you want to\nassemble. Should be a list of databases split by\ncomma(s) on a multi-organelle mode, with the same list\nlength to organelle_type (followed by '-F'). This is\noptional for any organelle mentioned in '-F' but\nrequired for 'anonym'. By default, certain database(s)\nin /root/.GetOrganelle/LabelDatabase would be used\ncontingent on the organelle types chosen (-F). The\ndefault value no longer holds when '--genes' or '--ex-\ngenes' is used.")), ToolInput(tag="in_ex_genes", input_type=Int(optional=True), prefix="--ex-genes", doc=InputDocumentation(doc="This is optional and Not suggested, since non-target\ncontigs could contribute information for better\ndownstream coverage-based clustering. Followed with a\ncustomized database (a fasta file or the base name of\na blast database) containing or made of protein coding\ngenes and ribosomal RNAs extracted from reference\ngenome(s) that you want to exclude. Could be a list of\ndatabases split by comma(s) but NOT required to have\nthe same list length to organelle_type (followed by\n'-F'). The default value no longer holds when '--\ngenes' or '--ex-genes' is used.")), ToolInput(tag="in_no_slim", input_type=Boolean(optional=True), prefix="--no-slim", doc=InputDocumentation(doc="Disable slimming process and directly disentangle the\noriginal assembly graph. Default: False")), ToolInput(tag="in_slim_options", input_type=String(optional=True), prefix="--slim-options", doc=InputDocumentation(doc="Other options for calling slim_graph.py")), ToolInput(tag="in_max_slim_extending_len", input_type=Int(optional=True), prefix="--max-slim-extending-len", doc=InputDocumentation(doc="This is used to limit the extending length, below\nwhich a 'non-hit contig' is allowed to be distant from\na 'hit contig' to be kept. See more under\nslim_graph.py:--max-slim-extending-len. Default: 15000\n(-F embplant_pt), 50000 (-F\nembplant_mt/fungus_mt/other_pt), 12500 (-F\nembplant_nr/fungus_nr/animal_mt),\nmaximum_of_type1_type2 (-F type1,type2), inf (-F\nanonym)")), ToolInput(tag="in_spades_out_dir", input_type=File(optional=True), prefix="--spades-out-dir", doc=InputDocumentation(doc="Input spades output directory with 'scaffolds.fasta'\nand 'scaffolds.paths', which are used for scaffolding\ndisconnected contigs with GAPs. Default: disabled")), ToolInput(tag="in_depth_factor", input_type=Float(optional=True), prefix="--depth-factor", doc=InputDocumentation(doc="Depth factor for differentiate genome type of contigs.\nThe genome type of contigs are determined by blast.\nDefault: 10.0")), ToolInput(tag="in_type_f", input_type=Float(optional=True), prefix="--type-f", doc=InputDocumentation(doc="Type factor for identifying contig type tag when\nmultiple tags exist in one contig. Default:3.0")), ToolInput(tag="in_contamination_depth", input_type=Float(optional=True), prefix="--contamination-depth", doc=InputDocumentation(doc="Depth factor for confirming contamination in parallel\ncontigs. Default: 3.0")), ToolInput(tag="in_contamination_similarity", input_type=Float(optional=True), prefix="--contamination-similarity", doc=InputDocumentation(doc="Similarity threshold for confirming contaminating\ncontigs. Default: 0.9")), ToolInput(tag="in_no_degenerate", input_type=Boolean(optional=True), prefix="--no-degenerate", doc=InputDocumentation(doc="Disable making consensus from parallel contig based on\nnucleotide degenerate table.")), ToolInput(tag="in_degenerate_depth", input_type=Float(optional=True), prefix="--degenerate-depth", doc=InputDocumentation(doc="Depth factor for confirming parallel contigs. Default:\n1.5")), ToolInput(tag="in_degenerate_similarity", input_type=Float(optional=True), prefix="--degenerate-similarity", doc=InputDocumentation(doc="Similarity threshold for confirming parallel contigs.\nDefault: 0.98")), ToolInput(tag="in_disentangle_time_limit", input_type=File(optional=True), prefix="--disentangle-time-limit", doc=InputDocumentation(doc="Time limit (second) for each try of disentangling a\ngraph file as a circular genome. Disentangling a graph\nas contigs is not limited. Default: 3600")), ToolInput(tag="in_expected_max_size", input_type=Int(optional=True), prefix="--expected-max-size", doc=InputDocumentation(doc="Expected maximum target genome size(s) for\ndisentangling. Should be a list of INTEGER numbers\nsplit by comma(s) on a multi-organelle mode, with the\nsame list length to organelle_type (followed by '-F').\nDefault: 250000 (-F embplant_pt/fungus_mt), 25000 (-F\nembplant_nr/fungus_nr/animal_mt), 1000000 (-F\nembplant_mt/other_pt), 1000000,1000000,250000 (-F\nother_pt,embplant_mt,fungus_mt)")), ToolInput(tag="in_expected_min_size", input_type=Int(optional=True), prefix="--expected-min-size", doc=InputDocumentation(doc="Expected minimum target genome size(s) for\ndisentangling. Should be a list of INTEGER numbers\nsplit by comma(s) on a multi-organelle mode, with the\nsame list length to organelle_type (followed by '-F').\nDefault: 10000 for all.")), ToolInput(tag="in_reverse_lsc", input_type=Boolean(optional=True), prefix="--reverse-lsc", doc=InputDocumentation(doc="For '-F embplant_pt' with complete circular result, by\ndefault, the direction of the starting contig (usually\nthe LSC contig) is determined as the direction with\nless ORFs. Choose this option to reverse the direction\nof the starting contig when result is circular.\nActually, both directions are biologically equivalent\nto each other. The reordering of the direction is only\nfor easier downstream analysis.")), ToolInput(tag="in_max_paths_num", input_type=Int(optional=True), prefix="--max-paths-num", doc=InputDocumentation(doc="Repeats would dramatically increase the number of\npotential isomers (paths). This option was used to\nexport a certain amount of paths out of all possible\npaths per assembly graph. Default: 1000")), ToolInput(tag="in_keep_all_polymorphic", input_type=Boolean(optional=True), prefix="--keep-all-polymorphic", doc=InputDocumentation(doc="By default, this script would pick the contig with\nhighest coverage among all parallel (polymorphic)\ncontigs when degenerating was not applicable. Choose\nthis flag to export all combinations.")), ToolInput(tag="in_min_sigma", input_type=Int(optional=True), prefix="--min-sigma", doc=InputDocumentation(doc="Minimum deviation factor for excluding non-target\ncontigs. Default:0.1")), ToolInput(tag="in_max_multiplicity", input_type=Int(optional=True), prefix="--max-multiplicity", doc=InputDocumentation(doc="Maximum multiplicity of contigs for disentangling\ngenome paths. Should be 1~12. Default:8")), ToolInput(tag="in_maximum_threads_use", input_type=String(optional=True), prefix="-t", doc=InputDocumentation(doc="Maximum threads to use.")), ToolInput(tag="in_prefix", input_type=String(optional=True), prefix="--prefix", doc=InputDocumentation(doc="Add extra prefix to resulting files under the output")), ToolInput(tag="in_which_bandage", input_type=File(optional=True), prefix="--which-bandage", doc=InputDocumentation(doc="Assign the path to bandage binary file if not added to\nthe path. Default: try $PATH")), ToolInput(tag="in_keep_temp", input_type=Boolean(optional=True), prefix="--keep-temp", doc=InputDocumentation(doc="Choose to keep the running temp/index files.")), ToolInput(tag="in_continue", input_type=Boolean(optional=True), prefix="--continue", doc=InputDocumentation(doc="Several check points based on files produced, rather\nthan on the log file, so keep in mind that this script\nwill not detect the difference between this input\nparameters and the previous ones.")), ToolInput(tag="in_random_seed", input_type=Int(optional=True), prefix="--random-seed", doc=InputDocumentation(doc="Default: 12345")), ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="--verbose", doc=InputDocumentation(doc="Verbose output. Choose to enable verbose running")), ToolInput(tag="in_min_depth", input_type=Int(optional=True), prefix="--min-depth", doc=InputDocumentation(doc="")), ToolInput(tag="in_exists_dot", input_type=String(), position=0, doc=InputDocumentation(doc="--min-depth=MIN_DEPTH")), ToolInput(tag="in_log_handler_dot", input_type=String(), position=0, doc=InputDocumentation(doc="-h                    print brief introduction for frequently-used options."))], outputs=[ToolOutput(tag="out_output_directory_overwriting", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_output_directory_overwriting", type_hint=File()), doc=OutputDocumentation(doc="Output directory. Overwriting files if directory"))], container="quay.io/biocontainers/getorganelle:1.7.3.4--pyh3252c3a_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Get_Organelle_From_Assembly_Py_V0_1_0().translate("wdl")

