from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Directory, String, Int, Boolean

Ribo_Run_V0_1_0 = CommandToolBuilder(tool="ribo_run", base_command=["ribo", "run"], inputs=[ToolInput(tag="in_reference_fast_a", input_type=File(optional=True), prefix="--reference_fasta", doc=InputDocumentation(doc="path to a (multi)fasta or a directory containing one\nor more chromosomal sequences in fasta format.\nRequired, unless using a config file")), ToolInput(tag="in_config", input_type=File(optional=True), prefix="--config", doc=InputDocumentation(doc="config file; if none given, create one; default: /")), ToolInput(tag="in_outputdir__defaulttriboseedpipelineresults", input_type=Directory(optional=True), prefix="-o", doc=InputDocumentation(doc="/output/dir/, --output /output/dir/\noutput directory; default:\n/2020-09-10T0043_riboSeed_pipeline_results/")), ToolInput(tag="in_experiment_name", input_type=String(optional=True), prefix="--experiment_name", doc=InputDocumentation(doc="prefix for results files; default: inferred")), ToolInput(tag="in_kingdom", input_type=String(optional=True), prefix="--Kingdom", doc=InputDocumentation(doc="whether to look for eukaryotic, archaeal, or bacterial\nrDNA; default: bac")), ToolInput(tag="in_ss__specificfeatures", input_type=Int(optional=True), prefix="-S", doc=InputDocumentation(doc=":23S:5S, --specific_features 16S:23S:5S\ncolon:separated -- specific features; default:\n16S:23S:5S")), ToolInput(tag="in_clusters", input_type=Int(optional=True), prefix="--clusters", doc=InputDocumentation(doc="number of rDNA clusters;if submitting multiple\nrecords, must be a colon:separated list whose length\nmatches number of genbank records. Default is inferred\nfrom specific feature with fewest hits")), ToolInput(tag="in_cluster_file", input_type=File(optional=True), prefix="--cluster_file", doc=InputDocumentation(doc="clustered_loci file output from riboSelect;this is\ncreated by default from run_riboSeed, but if you don't\nagree with the operon structure predicted by\nriboSelect, you can use your alternate clustered_loci\nfile. default: None")), ToolInput(tag="in_fast_q_one", input_type=Int(optional=True), prefix="--fastq1", doc=InputDocumentation(doc="path to forward fastq file, can be compressed")), ToolInput(tag="in_fast_q_two", input_type=Int(optional=True), prefix="--fastq2", doc=InputDocumentation(doc="path to reverse fastq file, can be compressed")), ToolInput(tag="in_fast_q_single_one", input_type=Int(optional=True), prefix="--fastq_single1", doc=InputDocumentation(doc="path to single fastq file")), ToolInput(tag="in_score_min", input_type=Int(optional=True), prefix="--score_min", doc=InputDocumentation(doc="If using smalt, this sets the '-m' param; default with\nsmalt is inferred from read length. If using BWA,\nreads mapping with ASscore lower than this will be\nrejected; default with BWA is half of read length")), ToolInput(tag="in_ref_as_contig", input_type=String(optional=True), prefix="--ref_as_contig", doc=InputDocumentation(doc="ignore: reference will not be used in subassembly.\ntrusted: SPAdes will use the seed sequences as a\n--trusted-contig; untrusted: SPAdes will treat as\n--untrusted-contig. infer: if mapping percentage over\n80%, 'trusted'; else 'untrusted'. See SPAdes docs for\ndetails. default: infer")), ToolInput(tag="in_linear", input_type=Boolean(optional=True), prefix="--linear", doc=InputDocumentation(doc="if genome is known to not be circular and a region of\ninterest (including flanking bits) extends past\nchromosome end, this extends the seqence past\nchromosome origin forward by --padding; default: False")), ToolInput(tag="in_sub_assembler", input_type=String(optional=True), prefix="--subassembler", doc=InputDocumentation(doc="assembler to use for subassembly scheme. SPAdes is\nused by default, but Skesa is a new addition that\nseems to work for subassembly and is faster")), ToolInput(tag="in_just_seed", input_type=Boolean(optional=True), prefix="--just_seed", doc=InputDocumentation(doc="Don't do an assembly, just generate the long read\n'seeds'; default: False")), ToolInput(tag="in_flanking_length", input_type=Int(optional=True), prefix="--flanking_length", doc=InputDocumentation(doc="length of flanking regions, in bp; default: 1000")), ToolInput(tag="in_kmers_kmers_used", input_type=Int(optional=True), prefix="-k", doc=InputDocumentation(doc=",33,55,77,99,127, --kmers 21,33,55,77,99,127\nkmers used for final assembly, separated by commas\nsuch as21,33,55,77,99,127. Can be set to 'auto', where\nSPAdes chooses. We ensure kmers are not too big or too\nclose to read length; default: 21,33,55,77,99,127")), ToolInput(tag="in_force_km_ers", input_type=Boolean(optional=True), prefix="--force_kmers", doc=InputDocumentation(doc="skip checking to see if kmerchoice is appropriate to\nread length. Sometimes kmers longer than reads can\nhelp in the final assembly, as the long reads\ngenerated by riboSeed contain kmers longer than the\nread length")), ToolInput(tag="in__prekmers_kmers", input_type=Int(optional=True), prefix="-p", doc=InputDocumentation(doc=",33,55,77,99, --pre_kmers 21,33,55,77,99\nkmers used during seeding assemblies, separated bt\ncommas; default: 21,33,55,77,99")), ToolInput(tag="in_min_flank_depth", input_type=Int(optional=True), prefix="--min_flank_depth", doc=InputDocumentation(doc="a subassembly won't be performed if this minimum depth\nis not achieved on both the 3' and5' end of the\npseudocontig. default: 0")), ToolInput(tag="in_clean_temps", input_type=Boolean(optional=True), prefix="--clean_temps", doc=InputDocumentation(doc="if --clean_temps, mapping files will be removed once\nthey are no no longer needed during the mapping\niterations to save space; default: False")), ToolInput(tag="in_iterations", input_type=Int(optional=True), prefix="--iterations", doc=InputDocumentation(doc="if iterations>1, multiple seedings will occur after\nsubassembly of seed regions; if setting --target_len,\nseedings will continue until --iterations are\ncompleted or --target_len is matched or exceeded;\ndefault: 3")), ToolInput(tag="in_skip_control", input_type=Boolean(optional=True), prefix="--skip_control", doc=InputDocumentation(doc="if --skip_control, no de novo assembly will be done;\ndefault: False")), ToolInput(tag="in_verbosity", input_type=File(optional=True), prefix="--verbosity", doc=InputDocumentation(doc="Logger writes debug to file in output dir; this sets\nverbosity level sent to stderr. 1 = debug(), 2 =\ninfo(), 3 = warning(), 4 = error() and 5 = critical();\ndefault: 2")), ToolInput(tag="in_cores", input_type=Int(optional=True), prefix="--cores", doc=InputDocumentation(doc="cores used; default: None")), ToolInput(tag="in_memory", input_type=Int(optional=True), prefix="--memory", doc=InputDocumentation(doc="cores for multiprocessing; default: 8")), ToolInput(tag="in_damn_the_torpedos", input_type=Boolean(optional=True), prefix="--damn_the_torpedos", doc=InputDocumentation(doc="Ignore certain errors, full speed ahead!")), ToolInput(tag="in_stages", input_type=String(optional=True), prefix="--stages", doc=InputDocumentation(doc="[{sketch,spec,snag,score,stack,none} ...]\nWhich assessment stages you wish to run: sketch, spec,\nsnag, score, stack. Any combination thereof")), ToolInput(tag="in_threads", input_type=String(optional=True), prefix="--threads", doc=InputDocumentation(doc="if your cores are hyperthreaded, set number threads to\nthe number of threads per processer.If unsure, see\n'cat /proc/cpuinfo' under 'cpu cores', or 'lscpu'\nunder 'Thread(s) per core'.: 1")), ToolInput(tag="in_additional_libs", input_type=String(optional=True), prefix="--additional_libs", doc=InputDocumentation(doc="include these libraries in final assembly in addition\nto the reads supplied as -F and -R. They must be\nsupplied according to SPAdes arg naming scheme. Use at\nown risk.default: None")), ToolInput(tag="in_serialize", input_type=Boolean(optional=True), prefix="--serialize", doc=InputDocumentation(doc="if --serialize, runs seeding and assembly without\nmultiprocessing. We recommend this for machines with\nless than 8GB RAM: False"))], outputs=[ToolOutput(tag="out_outputdir__defaulttriboseedpipelineresults", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_outputdir__defaulttriboseedpipelineresults", type_hint=File()), doc=OutputDocumentation(doc="/output/dir/, --output /output/dir/\noutput directory; default:\n/2020-09-10T0043_riboSeed_pipeline_results/")), ToolOutput(tag="out_cluster_file", output_type=File(optional=True), selector=InputSelector(input_to_select="in_cluster_file", type_hint=File()), doc=OutputDocumentation(doc="clustered_loci file output from riboSelect;this is\ncreated by default from run_riboSeed, but if you don't\nagree with the operon structure predicted by\nriboSelect, you can use your alternate clustered_loci\nfile. default: None")), ToolOutput(tag="out_verbosity", output_type=File(optional=True), selector=InputSelector(input_to_select="in_verbosity", type_hint=File()), doc=OutputDocumentation(doc="Logger writes debug to file in output dir; this sets\nverbosity level sent to stderr. 1 = debug(), 2 =\ninfo(), 3 = warning(), 4 = error() and 5 = critical();\ndefault: 2"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Ribo_Run_V0_1_0().translate("wdl", allow_empty_container=True)

