from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, Directory, String

Hlaprofiler_Pl_Build_V0_1_0 = CommandToolBuilder(tool="HLAProfiler.pl_build", base_command=["HLAProfiler.pl", "build"], inputs=[ToolInput(tag="in_transcripts", input_type=Boolean(optional=True), prefix="-transcripts", doc=InputDocumentation(doc="|t          location of fasta file containing transcripts. Currently only GENCODE transcripts are supported.(required)")), ToolInput(tag="in_transcript_gtf", input_type=Boolean(optional=True), prefix="-transcript_gtf", doc=InputDocumentation(doc="|g       location of gtf file containing transcripts corresponding to the -transcripts option. Currently only GENCODE transcripts are supported.(required)")), ToolInput(tag="in_exclusion_bed", input_type=Boolean(optional=True), prefix="-exclusion_bed", doc=InputDocumentation(doc="|e        location of bed file containing the coordinated any regions to be excluded from the distractome. i.e. HLA region.(required)")), ToolInput(tag="in_reference", input_type=Boolean(optional=True), prefix="-reference", doc=InputDocumentation(doc="|r            location of fasta file containing HLA reference. IPD-IMGT/HLA reference recommended.(required)")), ToolInput(tag="in_cwd", input_type=Boolean(optional=True), prefix="-cwd", doc=InputDocumentation(doc="File containing the names of common and well-documented alleles. This file can be blank but must be specified.(required)")), ToolInput(tag="in_output_dir", input_type=Directory(optional=True), prefix="-output_dir", doc=InputDocumentation(doc="|o           location of output directory(default:'.')")), ToolInput(tag="in_database_name", input_type=Boolean(optional=True), prefix="-database_name", doc=InputDocumentation(doc="|db       name of the HLA database to be created(default:hla)")), ToolInput(tag="in_kraken_path", input_type=Boolean(optional=True), prefix="-kraken_path", doc=InputDocumentation(doc="|kp base directory of kraken installation. (default:base directory of path returned by `which kraken`)")), ToolInput(tag="in_k_mer", input_type=Boolean(optional=True), prefix="-k_mer", doc=InputDocumentation(doc="|k                size of the k-mer used to create database.(default:31)")), ToolInput(tag="in_minimizer", input_type=Boolean(optional=True), prefix="-minimizer", doc=InputDocumentation(doc="|mi           size of the k-mer minimizer used to crate database.(default:13)")), ToolInput(tag="in_num_reads", input_type=Boolean(optional=True), prefix="-num_reads", doc=InputDocumentation(doc="|nr           number of reads to simulated per reference allele for k-mer profile creations.(default:500000)")), ToolInput(tag="in_read_length", input_type=Boolean(optional=True), prefix="-read_length", doc=InputDocumentation(doc="|rl         length of reads simulated for k-mer profile. Same as the length of the k-mers in the profile.(default:50)")), ToolInput(tag="in_max_insert", input_type=Boolean(optional=True), prefix="-max_insert", doc=InputDocumentation(doc="|m           maximum size of insert (default:1000)")), ToolInput(tag="in_scale", input_type=Boolean(optional=True), prefix="-scale", doc=InputDocumentation(doc="|sc               scale of pareto distribution to determine insert size (default:80)")), ToolInput(tag="in_shape", input_type=Boolean(optional=True), prefix="-shape", doc=InputDocumentation(doc="|sh               shape of pareto distribution to determine insert size (default:0.7)")), ToolInput(tag="in_seed", input_type=Boolean(optional=True), prefix="-seed", doc=InputDocumentation(doc="|sd                seed of random number generator for simulation (default:1234)")), ToolInput(tag="in_filter_reads", input_type=Boolean(optional=True), prefix="-filter_reads", doc=InputDocumentation(doc="|f         toggle whether or not to filter reads using in the HLA database when building the k-mer profile.It is STRONGLY recommended to use the default for this setting. Possibile values 0 or 1. (default:1)")), ToolInput(tag="in_intermediate_files", input_type=Boolean(optional=True), prefix="-intermediate_files", doc=InputDocumentation(doc="|if  toggles flag to keep intermediate files (default:off)")), ToolInput(tag="in_threads", input_type=Boolean(optional=True), prefix="-threads", doc=InputDocumentation(doc="|c      number of threads to uses for processing.(default:1)")), ToolInput(tag="in_options", input_type=String(), position=0, doc=InputDocumentation(doc=""))], outputs=[ToolOutput(tag="out_output_dir", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_output_dir", type_hint=File()), doc=OutputDocumentation(doc="|o           location of output directory(default:'.')"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Hlaprofiler_Pl_Build_V0_1_0().translate("wdl", allow_empty_container=True)

