from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, File, Int

Gtftk_Overlapping_V0_1_0 = CommandToolBuilder(tool="gtftk_overlapping", base_command=["gtftk", "overlapping"], inputs=[ToolInput(tag="in_annotate_gtf", input_type=Boolean(optional=True), prefix="--annotate-gtf", doc=InputDocumentation(doc="All lines of the original GTF will be printed. (default: False)")), ToolInput(tag="in_input_file", input_type=Boolean(optional=True), prefix="--inputfile", doc=InputDocumentation(doc="Path to the GTF file. Default to STDIN (default: <stdin>)")), ToolInput(tag="in_output_file", input_type=File(optional=True), prefix="--outputfile", doc=InputDocumentation(doc="Output file. (default: <stdout>)")), ToolInput(tag="in_chrom_info", input_type=Boolean(optional=True), prefix="--chrom-info", doc=InputDocumentation(doc="Chromosome information. A tabulated two-columns file with chromosomes as column 1 and sizes as column 2 (default: None)")), ToolInput(tag="in_upstream", input_type=Boolean(optional=True), prefix="--upstream", doc=InputDocumentation(doc="Extend the region in 5'. Used to define the region around the TSS/TTS. (default: 1500)")), ToolInput(tag="in_downstream", input_type=Boolean(optional=True), prefix="--downstream", doc=InputDocumentation(doc="Extend the region in 3'. Used to define the region around the TSS/TTS. (default: 1500)")), ToolInput(tag="in_feature_type", input_type=Boolean(optional=True), prefix="--feature-type", doc=InputDocumentation(doc="The feature of interest. (default: transcript)")), ToolInput(tag="in_same_stranded_ness", input_type=Boolean(optional=True), prefix="--same-strandedness", doc=InputDocumentation(doc="Require same strandedness (default: False)")), ToolInput(tag="in_diff_stranded_ness", input_type=Boolean(optional=True), prefix="--diff-strandedness", doc=InputDocumentation(doc="Require different strandedness (default: False)")), ToolInput(tag="in_invert_match", input_type=Boolean(optional=True), prefix="--invert-match", doc=InputDocumentation(doc="Not/Invert match. (default: False)")), ToolInput(tag="in_key_name", input_type=Boolean(optional=True), prefix="--key-name", doc=InputDocumentation(doc="The name of the key. (default: None)")), ToolInput(tag="in_bool", input_type=Boolean(optional=True), prefix="--bool", doc=InputDocumentation(doc="When --annotate-gtf is used use 0 or 1 as key values (instead of overlapping transcripts id). (default: False)")), ToolInput(tag="in_annotate_all", input_type=Boolean(optional=True), prefix="--annotate-all", doc=InputDocumentation(doc="When --annotate-gtf annotate all transcripts (default value would be '0'). (default: False)")), ToolInput(tag="in_verbosity", input_type=Boolean(optional=True), prefix="--verbosity", doc=InputDocumentation(doc="Set output verbosity ([0-3]). (default: 0)")), ToolInput(tag="in_no_date", input_type=File(optional=True), prefix="--no-date", doc=InputDocumentation(doc="Do not add date to output file names. (default: False)")), ToolInput(tag="in_add_chr", input_type=Boolean(optional=True), prefix="--add-chr", doc=InputDocumentation(doc="Add 'chr' to chromosome names before printing output. (default: False)")), ToolInput(tag="in_tmp_dir", input_type=Boolean(optional=True), prefix="--tmp-dir", doc=InputDocumentation(doc="Keep all temporary files into this folder. (default: None)")), ToolInput(tag="in_keep_all", input_type=Boolean(optional=True), prefix="--keep-all", doc=InputDocumentation(doc="Try to keep all temporary files even if process does not terminate normally. (default: False)")), ToolInput(tag="in_logger_file", input_type=Boolean(optional=True), prefix="--logger-file", doc=InputDocumentation(doc="Stores the arguments passed to the command into a file. (default: None)")), ToolInput(tag="in_write_message_to_file", input_type=Boolean(optional=True), prefix="--write-message-to-file", doc=InputDocumentation(doc="Store all message into a file. (default: None)")), ToolInput(tag="in_arguments", input_type=Int(), position=0, doc=InputDocumentation(doc="Arguments:"))], outputs=[ToolOutput(tag="out_output_file", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output_file", type_hint=File()), doc=OutputDocumentation(doc="Output file. (default: <stdout>)")), ToolOutput(tag="out_no_date", output_type=File(optional=True), selector=InputSelector(input_to_select="in_no_date", type_hint=File()), doc=OutputDocumentation(doc="Do not add date to output file names. (default: False)"))], container="quay.io/biocontainers/pygtftk:1.2.7--py36h4edd4cd_1", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Gtftk_Overlapping_V0_1_0().translate("wdl")

