from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import String, File, Int, Boolean

Anvi_Get_Sequences_For_Hmm_Hits_V0_1_0 = CommandToolBuilder(tool="anvi_get_sequences_for_hmm_hits", base_command=["anvi-get-sequences-for-hmm-hits"], inputs=[ToolInput(tag="in_contigs_db", input_type=String(optional=True), prefix="--contigs-db", doc=InputDocumentation(doc="Anvi'o contigs database generated by 'anvi-gen-\ncontigs'")), ToolInput(tag="in_profile_db", input_type=String(optional=True), prefix="--profile-db", doc=InputDocumentation(doc="Anvi'o profile database")), ToolInput(tag="in_collection_name", input_type=String(optional=True), prefix="--collection-name", doc=InputDocumentation(doc="Collection name.")), ToolInput(tag="in_bin_id", input_type=String(optional=True), prefix="--bin-id", doc=InputDocumentation(doc="Bin name you are interested in.")), ToolInput(tag="in_bin_ids_file", input_type=File(optional=True), prefix="--bin-ids-file", doc=InputDocumentation(doc="Text file for bins (each line should be a unique bin\nid).")), ToolInput(tag="in_external_genomes", input_type=File(optional=True), prefix="--external-genomes", doc=InputDocumentation(doc="A two-column TAB-delimited flat text file that lists\nanvi'o contigs databases. The first item in the header\nline should read 'name', and the second should read\n'contigs_db_path'. Each line in the file should\ndescribe a single entry, where the first column is the\nname of the genome (or MAG), and the second column is\nthe anvi'o contigs database generated for this genome.")), ToolInput(tag="in_internal_genomes", input_type=File(optional=True), prefix="--internal-genomes", doc=InputDocumentation(doc="A five-column TAB-delimited flat text file. The header\nline must contain these columns: 'name', 'bin_id',\n'collection_id', 'profile_db_path', 'contigs_db_path'.\nEach line should list a single entry, where 'name' can\nbe any name to describe the anvi'o bin identified as\n'bin_id' that is stored in a collection.")), ToolInput(tag="in_hmm_sources", input_type=Int(optional=True), prefix="--hmm-sources", doc=InputDocumentation(doc="NAME\nGet sequences for a specific list of HMM sources. You\ncan list one or more sources by separating them from\neach other with a comma character (i.e., '--hmm-\nsources source_1,source_2,source_3'). If you would\nlike to see a list of available sources in the contigs\ndatabase, run this program with '--list-hmm-sources'\nflag.")), ToolInput(tag="in_gene_names", input_type=Int(optional=True), prefix="--gene-names", doc=InputDocumentation(doc="HIT NAME\nGet sequences only for a specific gene name. Each name\nshould be separated from each other by a comma\ncharacter. For instance, if you want to get back only\nRecA and Ribosomal_L27, you can type '--gene-names\nRecA,Ribosomal_L27', and you will get any and every\nhit that matches these names in any source. If you\nwould like to see a list of available gene names, you\ncan use '--list-available-gene-names' flag.")), ToolInput(tag="in_list_hmm_sources", input_type=Boolean(optional=True), prefix="--list-hmm-sources", doc=InputDocumentation(doc="List available HMM sources in the contigs database and\nquit.")), ToolInput(tag="in_list_available_gene_names", input_type=Boolean(optional=True), prefix="--list-available-gene-names", doc=InputDocumentation(doc="List available gene names in HMM sources selection and\nquit.")), ToolInput(tag="in_output_file", input_type=File(optional=True), prefix="--output-file", doc=InputDocumentation(doc="File path to store results.")), ToolInput(tag="in_no_wrap", input_type=File(optional=True), prefix="--no-wrap", doc=InputDocumentation(doc="Do not be wrap sequences nicely in the output file.")), ToolInput(tag="in_get_aa_sequences", input_type=Boolean(optional=True), prefix="--get-aa-sequences", doc=InputDocumentation(doc="Store amino acid sequences instead.")), ToolInput(tag="in_concatenate_genes", input_type=Boolean(optional=True), prefix="--concatenate-genes", doc=InputDocumentation(doc="Concatenate output genes in the same order to create a\nmulti-gene alignment output that is suitable for\nphylogenomic analyses.")), ToolInput(tag="in_partition_file", input_type=File(optional=True), prefix="--partition-file", doc=InputDocumentation(doc="Some commonly used software for phylogenetic analyses\n(e.g., IQ-TREE, RAxML, etc) allow users to\nspecify/test different substitution models for each\ngene of a concatenated multiple sequence alignments.\nFor this, they use a special file format called a\n'partition file', which indicates the site for each\ngene in the alignment. You can use this parameter to\ndeclare an output path for anvi'o to report a NEXUS\nformat partition file in addition to your FASTA output\n(requested by Massimiliano Molari in #1333).")), ToolInput(tag="in_max_num_genes_missing_from_bin", input_type=Int(optional=True), prefix="--max-num-genes-missing-from-bin", doc=InputDocumentation(doc="This filter removes bins (or genomes) from your\nanalysis. If you have a list of gene names, you can\nuse this parameter to omit any bin (or external\ngenome) that is missing more than a number of genes\nyou desire. For instance, if you have 100 genome bins,\nand you are interested in working with 5 ribosomal\nproteins, you can use '--max-num-genes-missing-from-\nbin 4' to remove the bins that are missing more than 4\nof those 5 genes. This is especially useful for\nphylogenomic analyses. Parameter 0 will remove any bin\nthat is missing any of the genes.")), ToolInput(tag="in_min_num_bins_gene_occurs", input_type=Int(optional=True), prefix="--min-num-bins-gene-occurs", doc=InputDocumentation(doc="This filter removes genes from your analysis. Let's\nassume you have 100 bins to get sequences for HMM\nhits. If you want to work only with genes among all\nthe hits that occur in at least X number of bins, and\ndiscard the rest of them, you can use this flag. If\nyou say '--min-num-bins-gene-occurs 90', each gene in\nthe analysis will be required at least to appear in 90\ngenomes. If a gene occurs in less than that number of\ngenomes, it simply will not be reported. This is\nespecially useful for phylogenomic analyses, where you\nmay want to only focus on genes that are prevalent\nacross the set of genomes you wish to analyze.")), ToolInput(tag="in_align_with", input_type=String(optional=True), prefix="--align-with", doc=InputDocumentation(doc="The multiple sequence alignment program to use when\nmultiple sequence alignment is necessary. To see all\navailable options, use the flag `--list-aligners`.")), ToolInput(tag="in_separator", input_type=String(optional=True), prefix="--separator", doc=InputDocumentation(doc="A word that will be used to sepaate concatenated gene\nsequences from each other (IF you are using this\nprogram with `--concatenate-genes` flag). The default\nis 'XXX' for amino acid sequences, and 'NNN' for DNA\nsequences")), ToolInput(tag="in_return_best_hit", input_type=Boolean(optional=True), prefix="--return-best-hit", doc=InputDocumentation(doc="A bin may contain more than one hit for a gene name in\na given HMM source. For instance, there may be\nmultiple RecA hits in a genome bin from Campbell et\nal.. Using this flag, will go through all of the gene\nnames that appear multiple times, and remove all but\nthe one with the lowest e-value. Good for whenever you\nreally need to get only a single copy of single-copy\ncore genes from a genome bin.")), ToolInput(tag="in_just_do_it", input_type=Boolean(optional=True), prefix="--just-do-it", doc=InputDocumentation(doc="Don't bother me with questions or warnings, just do\nit.\n"))], outputs=[ToolOutput(tag="out_output_file", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output_file", type_hint=File()), doc=OutputDocumentation(doc="File path to store results.")), ToolOutput(tag="out_no_wrap", output_type=File(optional=True), selector=InputSelector(input_to_select="in_no_wrap", type_hint=File()), doc=OutputDocumentation(doc="Do not be wrap sequences nicely in the output file.")), ToolOutput(tag="out_partition_file", output_type=File(optional=True), selector=InputSelector(input_to_select="in_partition_file", type_hint=File()), doc=OutputDocumentation(doc="Some commonly used software for phylogenetic analyses\n(e.g., IQ-TREE, RAxML, etc) allow users to\nspecify/test different substitution models for each\ngene of a concatenated multiple sequence alignments.\nFor this, they use a special file format called a\n'partition file', which indicates the site for each\ngene in the alignment. You can use this parameter to\ndeclare an output path for anvi'o to report a NEXUS\nformat partition file in addition to your FASTA output\n(requested by Massimiliano Molari in #1333)."))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Anvi_Get_Sequences_For_Hmm_Hits_V0_1_0().translate("wdl", allow_empty_container=True)

