class: CommandLineTool
id: anvi_split.cwl
inputs:
- id: in_contigs_db
  doc: "Anvi'o contigs database generated by 'anvi-gen-\ncontigs'"
  type: string
  inputBinding:
    prefix: --contigs-db
- id: in_genomes_storage
  doc: Anvi'o genomes storage file
  type: File
  inputBinding:
    prefix: --genomes-storage
- id: in_skip_variability_tables
  doc: "Processing variability tables in profile database\nmight take a very long\
    \ time. With this flag you will\nbe asking anvi'o to skip them."
  type: boolean
  inputBinding:
    prefix: --skip-variability-tables
- id: in_compress_auxiliary_data
  doc: "When declared, the auxiliary data file in the\nresulting output will be compressed.\
    \ This saves space,\nbut it takes long. Also, if you are planning to\ncompress\
    \ the entire later using GZIP, it is even\nuseless to do. But you are the boss!"
  type: File
  inputBinding:
    prefix: --compress-auxiliary-data
- id: in_bin_id
  doc: Bin name you are interested in.
  type: string
  inputBinding:
    prefix: --bin-id
- id: in_output_dir
  doc: Directory path for output files
  type: File
  inputBinding:
    prefix: --output-dir
- id: in_list_collections
  doc: Show available collections and exit.
  type: boolean
  inputBinding:
    prefix: --list-collections
- id: in_skip_hierarchical_clustering
  doc: "If you are not planning to use the interactive\ninterface (or if you have\
    \ other means to add a tree of\ncontigs in the database) you may skip the step\
    \ where\nhierarchical clustering of your items are preformed\nbased on default\
    \ clustering recipes matching to your\ndatabase type."
  type: boolean
  inputBinding:
    prefix: --skip-hierarchical-clustering
- id: in_enforce_hierarchical_clustering
  doc: "If you have more than 25,000 splits in your merged\nprofile, anvi-merge will\
    \ automatically skip the\nhierarchical clustering of splits (by setting --skip-\n\
    hierarchical-clustering flag on). This is due to the\nfact that computational\
    \ time required for hierarchical\nclustering increases exponentially with the\
    \ number of\nitems being clustered. Based on our experience we\ndecided that 25,000\
    \ splits is about the maximum we\nshould try. However, this is not a theoretical\
    \ limit,\nand you can overwrite this heuristic by using this\nflag, which would\
    \ tell anvi'o to attempt to cluster\nsplits regardless."
  type: boolean
  inputBinding:
    prefix: --enforce-hierarchical-clustering
- id: in_distance
  doc: "The distance metric for the hierarchical clustering.\nIf you do not use this\
    \ flag, the default distance\nmetric will be used for each clustering configuration\n\
    which is \"euclidean\"."
  type: string
  inputBinding:
    prefix: --distance
- id: in_linkage
  doc: "The same story with the `--distance`, except, the\nsystem default for this\
    \ one is ward.\n"
  type: string
  inputBinding:
    prefix: --linkage
- id: in_p
  doc: ''
  type: string
  inputBinding:
    prefix: -p
- id: in_here_dot
  doc: "-p PAN_OR_PROFILE_DB, --pan-or-profile-db PAN_OR_PROFILE_DB\nAnvi'o pan or\
    \ profile database (and even genes\ndatabase in appropriate contexts)."
  type: string
  inputBinding:
    position: 0
- id: in_separately_dot
  doc: "-C COLLECTION_NAME, --collection-name COLLECTION_NAME\nCollection name."
  type: string
  inputBinding:
    position: 0
outputs:
- id: out_stdout
  doc: Standard output stream
  type: stdout
- id: out_compress_auxiliary_data
  doc: "When declared, the auxiliary data file in the\nresulting output will be compressed.\
    \ This saves space,\nbut it takes long. Also, if you are planning to\ncompress\
    \ the entire later using GZIP, it is even\nuseless to do. But you are the boss!"
  type: File
  outputBinding:
    glob: $(inputs.in_compress_auxiliary_data)
- id: out_output_dir
  doc: Directory path for output files
  type: File
  outputBinding:
    glob: $(inputs.in_output_dir)
cwlVersion: v1.1
baseCommand:
- anvi-split
