version 1.0

task Anvigetshortreadsmappingtoagene {
  input {
    String? contigs_db
    String? s_sorted_indexed
    String? gene_caller_id
    File? genes_of_interest
    Int? leeway
    File? output_file_prefix
  }
  command <<<
    anvi_get_short_reads_mapping_to_a_gene \
      ~{if defined(contigs_db) then ("--contigs-db " +  '"' + contigs_db + '"') else ""} \
      ~{if defined(s_sorted_indexed) then ("-i " +  '"' + s_sorted_indexed + '"') else ""} \
      ~{if defined(gene_caller_id) then ("--gene-caller-id " +  '"' + gene_caller_id + '"') else ""} \
      ~{if defined(genes_of_interest) then ("--genes-of-interest " +  '"' + genes_of_interest + '"') else ""} \
      ~{if defined(leeway) then ("--leeway " +  '"' + leeway + '"') else ""} \
      ~{if defined(output_file_prefix) then ("--output-file-prefix " +  '"' + output_file_prefix + '"') else ""}
  >>>
  parameter_meta {
    contigs_db: "Anvi'o contigs database generated by 'anvi-gen-\\ncontigs'"
    s_sorted_indexed: "(S) [INPUT_BAM(S) ...], --input-files INPUT_BAM(S) [INPUT_BAM(S) ...]\\nSorted and indexed BAM files to analyze. It is\\nessential that all BAM files must be the result of\\nmappings against the same contigs."
    gene_caller_id: "A single gene id."
    genes_of_interest: "A file with anvi'o gene caller IDs. There should be\\nonly one column in the file, and each line should\\ncorrespond to a unique gene caller id (without a\\ncolumn header)."
    leeway: "The minimum number of nucleotides for a given short\\nread mapping into the gene context for it to be\\nreported. You must consider the length of your short\\nreads, as well as the length of the gene you are\\ntargeting. The default is 100 nts."
    output_file_prefix: "A prefix to be used while naming the output files (no\\nfile type extensions please; just a prefix).\\n"
  }
  output {
    File out_stdout = stdout()
    File out_output_file_prefix = "${in_output_file_prefix}"
  }
}