from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import String, File, Boolean

Anvi_Summarize_V0_1_0 = CommandToolBuilder(tool="anvi_summarize", base_command=["anvi-summarize"], inputs=[ToolInput(tag="in_pan_or_profile_db", input_type=String(optional=True), prefix="--pan-or-profile-db", doc=InputDocumentation(doc="Anvi'o pan or profile database (and even genes\ndatabase in appropriate contexts).")), ToolInput(tag="in_contigs_db", input_type=String(optional=True), prefix="--contigs-db", doc=InputDocumentation(doc="Anvi'o contigs database generated by 'anvi-gen-\ncontigs'")), ToolInput(tag="in_genomes_storage", input_type=File(optional=True), prefix="--genomes-storage", doc=InputDocumentation(doc="Anvi'o genomes storage file")), ToolInput(tag="in_in_it_gene_coverages", input_type=Boolean(optional=True), prefix="--init-gene-coverages", doc=InputDocumentation(doc="Initialize gene coverage and detection data. This is a\nvery computationally expensive step, but it is\nnecessary when you need gene level coverage data. The\nreason this is very computationally expensive is\nbecause anvi'o computes gene coverages by going back\nto actual coverage values of each gene to average\nthem, instead of using contig average coverage values,\nfor extreme accuracy.")), ToolInput(tag="in_reformat_contig_names", input_type=Boolean(optional=True), prefix="--reformat-contig-names", doc=InputDocumentation(doc="Reformat contig names while generating the summary\noutput so they look fancy. With this flag, anvi'o will\nreplace the original names of contigs to those that\ninclude the bin name as a prefix in resulting summary\noutput files per bin. Use this flag carefully as it\nmay influence your downstream analyses due to the fact\nthat your original contig names in your input FASTA\nfile for the contigs database will not be in the\nsummary output. Although, anvi'o will report a\nconversion map per bin so you can recover the original\ncontig name if you have to.")), ToolInput(tag="in_report_aa_seqs_for_gene_calls", input_type=File(optional=True), prefix="--report-aa-seqs-for-gene-calls", doc=InputDocumentation(doc="You can use this flag if you would like amino acid AND\ndna sequences for your gene calls in the genes output\nfile. By default, only dna sequences are reported.")), ToolInput(tag="in_report_dna_sequences", input_type=Boolean(optional=True), prefix="--report-DNA-sequences", doc=InputDocumentation(doc="By default, this program reports amino acid sequences.\nUse this flag to report DNA sequences instead. Also\nnote, since gene clusters are aligned via amino acid\nsequences, using this flag removes alignment\ninformation manifesting in the form of gap characters\n(`-` characters) that would be present if amino acid\nsequences were reported. Read the warnings during\nruntime for more detailed information.")), ToolInput(tag="in_collection_name", input_type=String(optional=True), prefix="--collection-name", doc=InputDocumentation(doc="Collection name.")), ToolInput(tag="in_output_dir", input_type=File(optional=True), prefix="--output-dir", doc=InputDocumentation(doc="Directory path for output files")), ToolInput(tag="in_list_collections", input_type=Boolean(optional=True), prefix="--list-collections", doc=InputDocumentation(doc="Show available collections and exit.")), ToolInput(tag="in_taxonomic_level", input_type=String(optional=True), prefix="--taxonomic-level", doc=InputDocumentation(doc="The taxonomic level to use whenever relevant and/or\navailable. The default taxonomic level is t_genus, but\nif you choose something specific, anvi'o will focus on\nthat whenever possible.")), ToolInput(tag="in_cog_data_dir", input_type=File(optional=True), prefix="--cog-data-dir", doc=InputDocumentation(doc="The directory path for your COG setup. Anvi'o will try\nto use the default path if you do not specify\nanything.")), ToolInput(tag="in_quick_summary", input_type=Boolean(optional=True), prefix="--quick-summary", doc=InputDocumentation(doc="When declared the summary output will be generated as\nquickly as possible, with minimum amount of essential\ninformation about bins.")), ToolInput(tag="in_just_do_it", input_type=Boolean(optional=True), prefix="--just-do-it", doc=InputDocumentation(doc="Don't bother me with questions or warnings, just do\nit.\n"))], outputs=[ToolOutput(tag="out_report_aa_seqs_for_gene_calls", output_type=File(optional=True), selector=InputSelector(input_to_select="in_report_aa_seqs_for_gene_calls", type_hint=File()), doc=OutputDocumentation(doc="You can use this flag if you would like amino acid AND\ndna sequences for your gene calls in the genes output\nfile. By default, only dna sequences are reported.")), ToolOutput(tag="out_output_dir", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output_dir", type_hint=File()), doc=OutputDocumentation(doc="Directory path for output files"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Anvi_Summarize_V0_1_0().translate("wdl", allow_empty_container=True)

