version 1.0

task Anvisplit {
  input {
    String? contigs_db
    File? genomes_storage
    Boolean? skip_variability_tables
    File? compress_auxiliary_data
    String? bin_id
    File? output_dir
    Boolean? list_collections
    Boolean? skip_hierarchical_clustering
    Boolean? enforce_hierarchical_clustering
    String? distance
    String? linkage
    String? p
    String here_dot
    String separately_dot
  }
  command <<<
    anvi_split \
      ~{here_dot} \
      ~{separately_dot} \
      ~{if defined(contigs_db) then ("--contigs-db " +  '"' + contigs_db + '"') else ""} \
      ~{if defined(genomes_storage) then ("--genomes-storage " +  '"' + genomes_storage + '"') else ""} \
      ~{if (skip_variability_tables) then "--skip-variability-tables" else ""} \
      ~{if (compress_auxiliary_data) then "--compress-auxiliary-data" else ""} \
      ~{if defined(bin_id) then ("--bin-id " +  '"' + bin_id + '"') else ""} \
      ~{if defined(output_dir) then ("--output-dir " +  '"' + output_dir + '"') else ""} \
      ~{if (list_collections) then "--list-collections" else ""} \
      ~{if (skip_hierarchical_clustering) then "--skip-hierarchical-clustering" else ""} \
      ~{if (enforce_hierarchical_clustering) then "--enforce-hierarchical-clustering" else ""} \
      ~{if defined(distance) then ("--distance " +  '"' + distance + '"') else ""} \
      ~{if defined(linkage) then ("--linkage " +  '"' + linkage + '"') else ""} \
      ~{if defined(p) then ("-p " +  '"' + p + '"') else ""}
  >>>
  parameter_meta {
    contigs_db: "Anvi'o contigs database generated by 'anvi-gen-\\ncontigs'"
    genomes_storage: "Anvi'o genomes storage file"
    skip_variability_tables: "Processing variability tables in profile database\\nmight take a very long time. With this flag you will\\nbe asking anvi'o to skip them."
    compress_auxiliary_data: "When declared, the auxiliary data file in the\\nresulting output will be compressed. This saves space,\\nbut it takes long. Also, if you are planning to\\ncompress the entire later using GZIP, it is even\\nuseless to do. But you are the boss!"
    bin_id: "Bin name you are interested in."
    output_dir: "Directory path for output files"
    list_collections: "Show available collections and exit."
    skip_hierarchical_clustering: "If you are not planning to use the interactive\\ninterface (or if you have other means to add a tree of\\ncontigs in the database) you may skip the step where\\nhierarchical clustering of your items are preformed\\nbased on default clustering recipes matching to your\\ndatabase type."
    enforce_hierarchical_clustering: "If you have more than 25,000 splits in your merged\\nprofile, anvi-merge will automatically skip the\\nhierarchical clustering of splits (by setting --skip-\\nhierarchical-clustering flag on). This is due to the\\nfact that computational time required for hierarchical\\nclustering increases exponentially with the number of\\nitems being clustered. Based on our experience we\\ndecided that 25,000 splits is about the maximum we\\nshould try. However, this is not a theoretical limit,\\nand you can overwrite this heuristic by using this\\nflag, which would tell anvi'o to attempt to cluster\\nsplits regardless."
    distance: "The distance metric for the hierarchical clustering.\\nIf you do not use this flag, the default distance\\nmetric will be used for each clustering configuration\\nwhich is \\\"euclidean\\\"."
    linkage: "The same story with the `--distance`, except, the\\nsystem default for this one is ward.\\n"
    p: ""
    here_dot: "-p PAN_OR_PROFILE_DB, --pan-or-profile-db PAN_OR_PROFILE_DB\\nAnvi'o pan or profile database (and even genes\\ndatabase in appropriate contexts)."
    separately_dot: "-C COLLECTION_NAME, --collection-name COLLECTION_NAME\\nCollection name."
  }
  output {
    File out_stdout = stdout()
    File out_compress_auxiliary_data = "${in_compress_auxiliary_data}"
    File out_output_dir = "${in_output_dir}"
  }
}