from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import String, File, Boolean, Float, Int

Anvi_Refine_V0_1_0 = CommandToolBuilder(tool="anvi_refine", base_command=["anvi-refine"], inputs=[ToolInput(tag="in_profile_db", input_type=String(optional=True), prefix="--profile-db", doc=InputDocumentation(doc="Anvi'o profile database")), ToolInput(tag="in_contigs_db", input_type=String(optional=True), prefix="--contigs-db", doc=InputDocumentation(doc="Anvi'o contigs database generated by 'anvi-gen-\ncontigs'")), ToolInput(tag="in_collection_name", input_type=String(optional=True), prefix="--collection-name", doc=InputDocumentation(doc="Collection name.")), ToolInput(tag="in_bin_id", input_type=String(optional=True), prefix="--bin-id", doc=InputDocumentation(doc="Bin name you are interested in.")), ToolInput(tag="in_bin_ids_file", input_type=File(optional=True), prefix="--bin-ids-file", doc=InputDocumentation(doc="Text file for bins (each line should be a unique bin\nid).")), ToolInput(tag="in_find_from_split_name", input_type=String(optional=True), prefix="--find-from-split-name", doc=InputDocumentation(doc="If you don't know the bin name you want to work with\nbut if you know the split name it contains you can use\nthis parameter to tell anvi'o the split name, and so\nit can find the bin for you automatically. This is\nsomething extremely difficult for anvi'o to do, but it\ndoes it anyway because you.")), ToolInput(tag="in_tree", input_type=String(optional=True), prefix="--tree", doc=InputDocumentation(doc="NEWICK formatted tree structure")), ToolInput(tag="in_skip_hierarchical_clustering", input_type=Boolean(optional=True), prefix="--skip-hierarchical-clustering", doc=InputDocumentation(doc="Skip hierarchical clustering for the splits in the\nrefined bin, if you skip clustering you need to\nprovide your own newick formatted tree using --tree\nparameter.")), ToolInput(tag="in_load_full_state", input_type=Boolean(optional=True), prefix="--load-full-state", doc=InputDocumentation(doc="Often the minimum and maximum values defined for the\nan entire profile database that contains all contigs\ndo not scale well when you wish to work with a single\nbin in the refine mode. For this reason, the default\nbehavior of anvi-refine is to ignore min/max values\nset in the default state. This flag is your way of\ntelling anvi'o to not do that, and load the state\nstored in the profile database as is.")), ToolInput(tag="in_additional_view", input_type=File(optional=True), prefix="--additional-view", doc=InputDocumentation(doc="A TAB-delimited file for an additional view to be used\nin the interface. This file should contain all split\nnames, and values for each of them in all samples.\nEach column in this file must correspond to a sample\nname. Content of this file will be called 'user_view',\nwhich will be available as a new item in the 'views'\ncombo box in the interface")), ToolInput(tag="in_additional_layers", input_type=File(optional=True), prefix="--additional-layers", doc=InputDocumentation(doc="A TAB-delimited file for additional layers for splits.\nThe first column of this file must be split names, and\nthe remaining columns should be unique attributes. The\nfile does not need to contain all split names, or\nvalues for each split in every column. Anvi'o will try\nto deal with missing data nicely. Each column in this\nfile will be visualized as a new layer in the tree.")), ToolInput(tag="in_taxonomic_level", input_type=String(optional=True), prefix="--taxonomic-level", doc=InputDocumentation(doc="The taxonomic level to use whenever relevant and/or\navailable. The default taxonomic level is t_genus, but\nif you choose something specific, anvi'o will focus on\nthat whenever possible.")), ToolInput(tag="in_hide_outlier_sn_vs", input_type=Boolean(optional=True), prefix="--hide-outlier-SNVs", doc=InputDocumentation(doc="During profiling, anvi'o marks positions of single-\nnucleotide variations (SNVs) that originate from\nplaces in contigs where coverage values are a bit\n'sketchy'. If you would like to avoid SNVs in those\npositions of splits in applicable projects you can use\nthis flag, and the interface would hide SNVs that are\nmarked as 'outlier' (although it is clearly the best\nto see everything, no one will judge you if you end up\nusing this flag) (plus, there may or may not be some\nhistorical data on this here:\nhttps://github.com/meren/anvio/issues/309).")), ToolInput(tag="in_title", input_type=File(optional=True), prefix="--title", doc=InputDocumentation(doc="Title for the interface. If you are working with a\nRUNINFO dict, the title will be determined based on\ninformation stored in that file. Regardless, you can\noverride that value using this parameter.")), ToolInput(tag="in_export_svg", input_type=File(optional=True), prefix="--export-svg", doc=InputDocumentation(doc="The SVG output file path.")), ToolInput(tag="in_dry_run", input_type=Boolean(optional=True), prefix="--dry-run", doc=InputDocumentation(doc="Don't do anything real. Test everything, and stop\nright before wherever the developer said 'well, this\nis enough testing', and decided to print out results.")), ToolInput(tag="in_skip_in_it_functions", input_type=Boolean(optional=True), prefix="--skip-init-functions", doc=InputDocumentation(doc="When declared, function calls for genes will not be\ninitialized (therefore will be missing from all\nrelevant interfaces or output files). The use of this\nflag may reduce the memory fingerprint and processing\ntime for large datasets.")), ToolInput(tag="in_skip_auto_ordering", input_type=Boolean(optional=True), prefix="--skip-auto-ordering", doc=InputDocumentation(doc="When declared, the attempt to include automatically\ngenerated orders of items based on additional data is\nskipped. In case those buggers cause issues with your\ndata, and you still want to see your stuff and deal\nwith the other issue maybe later.")), ToolInput(tag="in_ip_address", input_type=Float(optional=True), prefix="--ip-address", doc=InputDocumentation(doc="IP address for the HTTP server. The default ip address\n(0.0.0.0) should work just fine for most.")), ToolInput(tag="in_port_number", input_type=Int(optional=True), prefix="--port-number", doc=InputDocumentation(doc="Port number to use for anvi'o services. If nothing is\ndeclared, anvi'o will try to find a suitable port\nnumber, starting from the default port number, 8080.")), ToolInput(tag="in_browser_path", input_type=File(optional=True), prefix="--browser-path", doc=InputDocumentation(doc="By default, anvi'o will use your default browser to\nlaunch the interactive interface. If you would like to\nuse something else than your system default, you can\nprovide a full path for an alternative browser using\nthis parameter, and hope for the best. For instance we\nare using this parameter to call Google's experimental\nbrowser, Canary, which performs better with demanding\nvisualizations.")), ToolInput(tag="in_read_only", input_type=Boolean(optional=True), prefix="--read-only", doc=InputDocumentation(doc="When the interactive interface is started with this\nflag, all 'database write' operations will be\ndisabled.")), ToolInput(tag="in_server_only", input_type=Boolean(optional=True), prefix="--server-only", doc=InputDocumentation(doc="The default behavior is to start the local server, and\nfire up a browser that connects to the server. If you\nhave other plans, and want to start the server without\ncalling the browser, this is the flag you need.")), ToolInput(tag="in_password_protected", input_type=Boolean(optional=True), prefix="--password-protected", doc=InputDocumentation(doc="If this flag is set, command line tool will ask you to\nenter a password and interactive interface will be\nonly accessible after entering same password. This\noption is recommended for shared machines like\nclusters or shared networks where computers are not\nisolated.\n")), ToolInput(tag="in_interface_dot", input_type=String(), position=0, doc=InputDocumentation(doc="--split-hmm-layers    When declared, this flag tells the interface to split\nevery gene found in HMM searches that were performed\nagainst non-singlecopy gene HMM profiles into their\nown layer. Please see the documentation for details."))], outputs=[ToolOutput(tag="out_export_svg", output_type=File(optional=True), selector=InputSelector(input_to_select="in_export_svg", type_hint=File()), doc=OutputDocumentation(doc="The SVG output file path."))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Anvi_Refine_V0_1_0().translate("wdl", allow_empty_container=True)

