from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Array, File, String, Int, Directory

Macs2_Predictd_V0_1_0 = CommandToolBuilder(tool="macs2_predictd", base_command=["macs2", "predictd"], inputs=[ToolInput(tag="in_i_file", input_type=Array(t=File(), optional=True), prefix="--ifile", doc=InputDocumentation(doc="ChIP-seq alignment file. If multiple files are given\nas '-t A B C', then they will all be read and\ncombined. Note that pair-end data is not supposed to\nwork with this command. REQUIRED.")), ToolInput(tag="in_format", input_type=String(optional=True), prefix="--format", doc=InputDocumentation(doc="Format of tag file, 'AUTO', 'BED' or 'ELAND' or\n'ELANDMULTI' or 'ELANDEXPORT' or 'SAM' or 'BAM' or\n'BOWTIE' or 'BAMPE' or 'BEDPE'. The default AUTO\noption will let MACS decide which format the file is.\nPlease check the definition in README file if you\nchoose ELAND/ELANDMULTI/ELANDEXPORT/SAM/BAM/BOWTIE.\nDEFAULT: 'AUTO'")), ToolInput(tag="in_g_size", input_type=Int(optional=True), prefix="--gsize", doc=InputDocumentation(doc="Effective genome size. It can be 1.0e+9 or 1000000000,\nor shortcuts:'hs' for human (2.7e9), 'mm' for mouse\n(1.87e9), 'ce' for C. elegans (9e7) and 'dm' for\nfruitfly (1.2e8), Default:hs")), ToolInput(tag="in_t_size", input_type=Int(optional=True), prefix="--tsize", doc=InputDocumentation(doc="Tag size. This will override the auto detected tag\nsize. DEFAULT: Not set")), ToolInput(tag="in_bw", input_type=Int(optional=True), prefix="--bw", doc=InputDocumentation(doc="Band width for picking regions to compute fragment\nsize. This value is only used while building the\nshifting model. DEFAULT: 300")), ToolInput(tag="in_d_min", input_type=Int(optional=True), prefix="--d-min", doc=InputDocumentation(doc="Minimum fragment size in basepair. Any predicted\nfragment size less than this will be excluded.\nDEFAULT: 20")), ToolInput(tag="in_mfold__mfold", input_type=Int(optional=True), prefix="-m", doc=InputDocumentation(doc="MFOLD, --mfold MFOLD MFOLD\nSelect the regions within MFOLD range of high-\nconfidence enrichment ratio against background to\nbuild model. Fold-enrichment in regions must be lower\nthan upper limit, and higher than the lower limit. Use\nas '-m 10 30'. DEFAULT:5 50")), ToolInput(tag="in_outdir", input_type=Directory(optional=True), prefix="--outdir", doc=InputDocumentation(doc="If specified all output files will be written to that\ndirectory. Default: the current working directory")), ToolInput(tag="in_r_file", input_type=File(optional=True), prefix="--rfile", doc=InputDocumentation(doc="PREFIX of filename of R script for drawing\nX-correlation figure. DEFAULT:'predictd' and R file\nwill be predicted_model.R")), ToolInput(tag="in_buffer_size", input_type=Int(optional=True), prefix="--buffer-size", doc=InputDocumentation(doc="Buffer size for incrementally increasing internal\narray size to store reads alignment information. In\nmost cases, you don't have to change this parameter.\nHowever, if there are large number of\nchromosomes/contigs/scaffolds in your alignment, it's\nrecommended to specify a smaller buffer size in order\nto decrease memory usage (but it will take longer time\nto read alignment files). Minimum memory requested for\nreading an alignment file is about # of CHROMOSOME *\nBUFFER_SIZE * 8 Bytes. DEFAULT: 100000")), ToolInput(tag="in_verbose", input_type=Int(optional=True), prefix="--verbose", doc=InputDocumentation(doc="Set verbose level of runtime message. 0: only show\ncritical message, 1: show additional warning message,\n2: show process information, 3: show debug messages.\nDEFAULT:2\n"))], outputs=[ToolOutput(tag="out_outdir", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_outdir", type_hint=File()), doc=OutputDocumentation(doc="If specified all output files will be written to that\ndirectory. Default: the current working directory"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Macs2_Predictd_V0_1_0().translate("wdl", allow_empty_container=True)

