from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, Directory, File

Smeg_Growth_Est_V0_1_0 = CommandToolBuilder(tool="smeg_growth_est", base_command=["smeg", "growth_est"], inputs=[ToolInput(tag="in_reads_directory_reads", input_type=Boolean(optional=True), prefix="-r", doc=InputDocumentation(doc="Reads directory (single-end reads)")), ToolInput(tag="in_sample_filename_extension", input_type=Boolean(optional=True), prefix="-x", doc=InputDocumentation(doc="Sample filename extension (fq, fastq, fastq.gz) [default fastq]")), ToolInput(tag="in_output_directory", input_type=Directory(optional=True), prefix="-o", doc=InputDocumentation(doc="Output directory")), ToolInput(tag="in_species_database_directory", input_type=Boolean(optional=True), prefix="-s", doc=InputDocumentation(doc="Species database directory")), ToolInput(tag="in_int_smeg_method", input_type=Boolean(optional=True), prefix="-m", doc=InputDocumentation(doc="INT    SMEG method (0 = de novo-based method, 1 = reference-based method) [default = 0]")), ToolInput(tag="in_float_coverage_cutoff", input_type=Boolean(optional=True), prefix="-c", doc=InputDocumentation(doc="FLOAT  Coverage cutoff (>= 0.5) [default 0.5]")), ToolInput(tag="in_int_minimum_number", input_type=Boolean(optional=True), prefix="-u", doc=InputDocumentation(doc="INT    Minimum number of SNPs to estimate growth rate [default = 100]")), ToolInput(tag="in_path_file_listing", input_type=Boolean(optional=True), prefix="-l", doc=InputDocumentation(doc="Path to file listing a subset of reads for analysis\n[default = analyze all samples in Reads directory]")), ToolInput(tag="in_float_cluster_detection", input_type=Boolean(optional=True), prefix="-d", doc=InputDocumentation(doc="FLOAT  Cluster detection threshold (range 0.1 - 1) [default = 0.2]")), ToolInput(tag="in_float_samplespecific_snp", input_type=Boolean(optional=True), prefix="-t", doc=InputDocumentation(doc="FLOAT  Sample-specific SNP assignment threshold (range 0.1 - 1) [default = 0.6]")), ToolInput(tag="in_file_listing_reference", input_type=Boolean(optional=True), prefix="-g", doc=InputDocumentation(doc="File listing reference genomes for growth rate estimation")), ToolInput(tag="in_file_listing_path", input_type=Boolean(optional=True), prefix="-a", doc=InputDocumentation(doc="File listing FULL PATH to DESMAN-resolved strains in fasta format (core-genes)")), ToolInput(tag="in_int_max_number", input_type=Boolean(optional=True), prefix="-n", doc=InputDocumentation(doc="INT    Max number of mismatch [default = use default bowtie2 threshold]")), ToolInput(tag="in_merge_output_tables", input_type=File(optional=True), prefix="-e", doc=InputDocumentation(doc="merge output tables into a single matrix file and generate heatmap")), ToolInput(tag="in_int_number_threads", input_type=Boolean(optional=True), prefix="-p", doc=InputDocumentation(doc="INT    Number of threads [default 4]"))], outputs=[ToolOutput(tag="out_output_directory", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_output_directory", type_hint=File()), doc=OutputDocumentation(doc="Output directory")), ToolOutput(tag="out_merge_output_tables", output_type=File(optional=True), selector=InputSelector(input_to_select="in_merge_output_tables", type_hint=File()), doc=OutputDocumentation(doc="merge output tables into a single matrix file and generate heatmap"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Smeg_Growth_Est_V0_1_0().translate("wdl", allow_empty_container=True)

