from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, File

Setupgenomicdata_Py_V0_1_0 = CommandToolBuilder(tool="setupGenomicData.py", base_command=["setupGenomicData.py"], inputs=[ToolInput(tag="in_all", input_type=Boolean(optional=True), prefix="--all", doc=InputDocumentation(doc="Fetch all data sets.")), ToolInput(tag="in_h_gone_nine", input_type=Boolean(optional=True), prefix="--hg19", doc=InputDocumentation(doc="Fetch human genome files.")), ToolInput(tag="in_hg_three_eight", input_type=Boolean(optional=True), prefix="--hg38", doc=InputDocumentation(doc="Fetch human genome files.")), ToolInput(tag="in_mm_nine", input_type=Boolean(optional=True), prefix="--mm9", doc=InputDocumentation(doc="Fetch mouse files.")), ToolInput(tag="in_mm_one_zero", input_type=Boolean(optional=True), prefix="--mm10", doc=InputDocumentation(doc="Fetch mouse files.")), ToolInput(tag="in_zv_nine", input_type=Boolean(optional=True), prefix="--zv9", doc=InputDocumentation(doc="Fetch zebrafish files.")), ToolInput(tag="in_zv_one_zero", input_type=Boolean(optional=True), prefix="--zv10", doc=InputDocumentation(doc="Fetch zebrafish files.")), ToolInput(tag="in_h_gone_nine_genome_path", input_type=File(optional=True), prefix="--hg19-genome-path", doc=InputDocumentation(doc="Path to an already existing hg19 genome (all\nchromosomes in the same file).")), ToolInput(tag="in_hg_three_eight_genome_path", input_type=File(optional=True), prefix="--hg38-genome-path", doc=InputDocumentation(doc="Path to an already existing hg38 genome (all\nchromosomes in the same file).")), ToolInput(tag="in_mm_nine_genome_path", input_type=File(optional=True), prefix="--mm9-genome-path", doc=InputDocumentation(doc="Path to an already existing mm9 genome (all\nchromosomes in the same file).")), ToolInput(tag="in_mm_one_zero_genome_path", input_type=File(optional=True), prefix="--mm10-genome-path", doc=InputDocumentation(doc="Path to an already existing mm9 genome (all\nchromosomes in the same file).")), ToolInput(tag="in_zv_nine_genome_path", input_type=File(optional=True), prefix="--zv9-genome-path", doc=InputDocumentation(doc="Path to an already existing zv9 genome (all\nchromosomes in the same file).")), ToolInput(tag="in_zv_one_zero_genome_path", input_type=File(optional=True), prefix="--zv10-genome-path", doc=InputDocumentation(doc="Path to an already existing zv10 genome (all\nchromosomes in the same file).")), ToolInput(tag="in_h_gone_nine_gtf_path", input_type=File(optional=True), prefix="--hg19-gtf-path", doc=InputDocumentation(doc="Path to an already existing hg19 GTF file.")), ToolInput(tag="in_hg_three_eight_gtf_path", input_type=File(optional=True), prefix="--hg38-gtf-path", doc=InputDocumentation(doc="Path to an already existing hg38 GTF file.")), ToolInput(tag="in_mm_nine_gtf_path", input_type=File(optional=True), prefix="--mm9-gtf-path", doc=InputDocumentation(doc="Path to an already existing mm9 GTF file.")), ToolInput(tag="in_mm_one_zero_gtf_path", input_type=File(optional=True), prefix="--mm10-gtf-path", doc=InputDocumentation(doc="Path to an already existing mm10 GTF file.")), ToolInput(tag="in_zv_nine_gtf_path", input_type=File(optional=True), prefix="--zv9-gtf-path", doc=InputDocumentation(doc="Path to an already existing zv9 GTF file.")), ToolInput(tag="in_zv_one_zero_gtf_path", input_type=File(optional=True), prefix="--zv10-gtf-path", doc=InputDocumentation(doc="Path to an already existing zv10 GTF file.")), ToolInput(tag="in_hg_three_eight_rm", input_type=Boolean(optional=True), prefix="--hg38-rm", doc=InputDocumentation(doc="Fetch repeat masker files for hg38 from Broad"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Setupgenomicdata_Py_V0_1_0().translate("wdl", allow_empty_container=True)

