from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, File

Maf2Vcf_Pl_V0_1_0 = CommandToolBuilder(tool="maf2vcf.pl", base_command=["maf2vcf.pl"], inputs=[ToolInput(tag="in_input_maf", input_type=Boolean(optional=True), prefix="--input-maf", doc=InputDocumentation(doc="Path to input file in MAF format")), ToolInput(tag="in_output_dir", input_type=File(optional=True), prefix="--output-dir", doc=InputDocumentation(doc="Path to output directory where VCFs will be stored, one per TN-pair")), ToolInput(tag="in_output_vcf", input_type=Boolean(optional=True), prefix="--output-vcf", doc=InputDocumentation(doc="Path to output multi-sample VCF containing all TN-pairs [<output-dir>/<input-maf-name>.vcf]")), ToolInput(tag="in_ref_fast_a", input_type=Boolean(optional=True), prefix="--ref-fasta", doc=InputDocumentation(doc="Path to reference Fasta file [~/.vep/homo_sapiens/102_GRCh37/Homo_sapiens.GRCh37.dna.toplevel.fa.gz]")), ToolInput(tag="in_per_tn_v_cfs", input_type=Boolean(optional=True), prefix="--per-tn-vcfs", doc=InputDocumentation(doc="Specify this to generate VCFs per-TN pair, in addition to the multi-sample VCF")), ToolInput(tag="in_tum_depth_col", input_type=Boolean(optional=True), prefix="--tum-depth-col", doc=InputDocumentation(doc="Name of MAF column for read depth in tumor BAM [t_depth]")), ToolInput(tag="in_tum_rad_col", input_type=Boolean(optional=True), prefix="--tum-rad-col", doc=InputDocumentation(doc="Name of MAF column for reference allele depth in tumor BAM [t_ref_count]")), ToolInput(tag="in_tum_vad_col", input_type=Boolean(optional=True), prefix="--tum-vad-col", doc=InputDocumentation(doc="Name of MAF column for variant allele depth in tumor BAM [t_alt_count]")), ToolInput(tag="in_nrm_depth_col", input_type=Boolean(optional=True), prefix="--nrm-depth-col", doc=InputDocumentation(doc="Name of MAF column for read depth in normal BAM [n_depth]")), ToolInput(tag="in_nrm_rad_col", input_type=Boolean(optional=True), prefix="--nrm-rad-col", doc=InputDocumentation(doc="Name of MAF column for reference allele depth in normal BAM [n_ref_count]")), ToolInput(tag="in_nrm_vad_col", input_type=Boolean(optional=True), prefix="--nrm-vad-col", doc=InputDocumentation(doc="Name of MAF column for variant allele depth in normal BAM [n_alt_count]")), ToolInput(tag="in_man", input_type=Boolean(optional=True), prefix="--man", doc=InputDocumentation(doc="Print the detailed manual"))], outputs=[ToolOutput(tag="out_output_dir", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output_dir", type_hint=File()), doc=OutputDocumentation(doc="Path to output directory where VCFs will be stored, one per TN-pair"))], container="quay.io/biocontainers/vcf2maf:1.6.20--hdfd78af_1", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Maf2Vcf_Pl_V0_1_0().translate("wdl")

