from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import String, File, Int, Boolean, Float

Ribotish_Predict_V0_1_0 = CommandToolBuilder(tool="ribotish_predict", base_command=["ribotish", "predict"], inputs=[ToolInput(tag="in_tis_enriched_riboseq", input_type=String(optional=True), prefix="-t", doc=InputDocumentation(doc="TIS enriched riboseq bam files, comma seperated")), ToolInput(tag="in_ordinary_riboseq_bam", input_type=String(optional=True), prefix="-b", doc=InputDocumentation(doc="Ordinary riboseq bam files, comma seperated")), ToolInput(tag="in_gene_annotation_file_orf", input_type=File(optional=True), prefix="-g", doc=InputDocumentation(doc="Gene annotation file for ORF prediction")), ToolInput(tag="in_genome_fasta_file", input_type=File(optional=True), prefix="-f", doc=InputDocumentation(doc="Genome fasta file")), ToolInput(tag="in_output_result_file", input_type=File(optional=True), prefix="-o", doc=InputDocumentation(doc="Output result file")), ToolInput(tag="in_only_test_input", input_type=Int(optional=True), prefix="-i", doc=InputDocumentation(doc="Only test input candidate ORFs, format: transID start\nstop (0 based, half open)")), ToolInput(tag="in_gene_format", input_type=File(optional=True), prefix="--geneformat", doc=InputDocumentation(doc="Gene annotation file format (gtf, bed, gpd, gff,\ndefault: auto)")), ToolInput(tag="in_tis_para", input_type=String(optional=True), prefix="--tispara", doc=InputDocumentation(doc="Input offset parameter files for -t bam files")), ToolInput(tag="in_ribo_para", input_type=String(optional=True), prefix="--ribopara", doc=InputDocumentation(doc="Input offset parameter files for -b bam files")), ToolInput(tag="in_n_parts", input_type=Int(optional=True), prefix="--nparts", doc=InputDocumentation(doc="Group transcript according to TIS reads density\nquantile (default: 10)")), ToolInput(tag="in_gene_file_file", input_type=File(optional=True), prefix="-a", doc=InputDocumentation(doc="Gene file for known protein coding gene annotation and\nTIS background estimation instead of -g gene file")), ToolInput(tag="in_output_tis_background", input_type=File(optional=True), prefix="-e", doc=InputDocumentation(doc="Output TIS background estimation result (default:\ntisBackground.txt)")), ToolInput(tag="in_input_background_estimation", input_type=File(optional=True), prefix="-s", doc=InputDocumentation(doc="Input background estimation result file instead of\ninstant estimation")), ToolInput(tag="in_trans_profile", input_type=String(optional=True), prefix="--transprofile", doc=InputDocumentation(doc="Output RPF P-site profile for each transcript")), ToolInput(tag="in_in_profile", input_type=String(optional=True), prefix="--inprofile", doc=InputDocumentation(doc="Input RPF P-site profile for each transcript, instead\nof reading bam reads, save time for re-running")), ToolInput(tag="in_chr_map", input_type=File(optional=True), prefix="--chrmap", doc=InputDocumentation(doc="Input chromosome id mapping table file if annotation\nchr ids are not same as chr ids in bam/fasta files")), ToolInput(tag="in_all_result", input_type=String(optional=True), prefix="--allresult", doc=InputDocumentation(doc="All result output without FDR q-value threshold\n(default: output + '_all.txt', 'off' to turn off)")), ToolInput(tag="in_alt", input_type=Boolean(optional=True), prefix="--alt", doc=InputDocumentation(doc="Use alternative start codons (all codons with 1 base\ndifferent from ATG)")), ToolInput(tag="in_alt_codons", input_type=String(optional=True), prefix="--altcodons", doc=InputDocumentation(doc="Use provided alternative start codons, comma\nseperated, eg. CTG,GTG,ACG")), ToolInput(tag="in_tis_two_ribo", input_type=Boolean(optional=True), prefix="--tis2ribo", doc=InputDocumentation(doc="Add TIS bam counts to ribo, if specified or -b not")), ToolInput(tag="in_harr_width", input_type=Int(optional=True), prefix="--harrwidth", doc=InputDocumentation(doc="Flanking region for harr data, in codons (default: 15)")), ToolInput(tag="in_enrich_test", input_type=Boolean(optional=True), prefix="--enrichtest", doc=InputDocumentation(doc="Use enrich test instead of frame test")), ToolInput(tag="in_min_aalen", input_type=Int(optional=True), prefix="--minaalen", doc=InputDocumentation(doc="Min amino acid length of candidate ORF (default: 6)")), ToolInput(tag="in_gene_filter", input_type=String(optional=True), prefix="--genefilter", doc=InputDocumentation(doc="Only process given genes")), ToolInput(tag="in_tp_th", input_type=Float(optional=True), prefix="--tpth", doc=InputDocumentation(doc="TIS p value threshold (default: 0.05)")), ToolInput(tag="in_fp_th", input_type=Float(optional=True), prefix="--fpth", doc=InputDocumentation(doc="Frame p value threshold (default: 0.05)")), ToolInput(tag="in_min_pth", input_type=Int(optional=True), prefix="--minpth", doc=InputDocumentation(doc="At least one of TIS or frame p value should be lower\nthan this threshold (default: 0.05)")), ToolInput(tag="in_fsp_th", input_type=String(optional=True), prefix="--fspth", doc=InputDocumentation(doc="Fisher's p value threshold")), ToolInput(tag="in_fsq_th", input_type=String(optional=True), prefix="--fsqth", doc=InputDocumentation(doc="Fisher's FDR q value threshold")), ToolInput(tag="in_frame_local_best", input_type=Boolean(optional=True), prefix="--framelocalbest", doc=InputDocumentation(doc="Only report local best frame test results")), ToolInput(tag="in_frame_best", input_type=Boolean(optional=True), prefix="--framebest", doc=InputDocumentation(doc="Only report best frame test results")), ToolInput(tag="in_longest", input_type=Boolean(optional=True), prefix="--longest", doc=InputDocumentation(doc="Only report longest possible ORF results")), ToolInput(tag="in_seq", input_type=Boolean(optional=True), prefix="--seq", doc=InputDocumentation(doc="Report ORF sequences")), ToolInput(tag="in_aa_seq", input_type=Boolean(optional=True), prefix="--aaseq", doc=InputDocumentation(doc="Report amino acid sequences")), ToolInput(tag="in_blocks", input_type=Boolean(optional=True), prefix="--blocks", doc=InputDocumentation(doc="Report all exon block positions for predicted ORFs")), ToolInput(tag="in_max_nh", input_type=Int(optional=True), prefix="--maxNH", doc=InputDocumentation(doc="Max NH value allowed for bam alignments (default: 5)")), ToolInput(tag="in_min_mapq", input_type=Int(optional=True), prefix="--minMapQ", doc=InputDocumentation(doc="Min MapQ value required for bam alignments (default:\n1)")), ToolInput(tag="in_secondary", input_type=Boolean(optional=True), prefix="--secondary", doc=InputDocumentation(doc="Use bam secondary alignments")), ToolInput(tag="in_paired", input_type=Boolean(optional=True), prefix="--paired", doc=InputDocumentation(doc="Reads are paired end")), ToolInput(tag="in_no_compatible", input_type=Boolean(optional=True), prefix="--nocompatible", doc=InputDocumentation(doc="Do not require reads compatible with transcript splice")), ToolInput(tag="in_number_of_processes", input_type=Int(optional=True), prefix="-p", doc=InputDocumentation(doc="Number of processes")), ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="--verbose", doc=InputDocumentation(doc="Increase output verbosity")), ToolInput(tag="in_provided", input_type=String(), position=0, doc=InputDocumentation(doc="--harr                The data is treated with harringtonine (instead of")), ToolInput(tag="in_junctions", input_type=String(), position=0, doc=InputDocumentation(doc="--compatiblemis COMPATIBLEMIS"))], outputs=[ToolOutput(tag="out_output_result_file", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output_result_file", type_hint=File()), doc=OutputDocumentation(doc="Output result file"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Ribotish_Predict_V0_1_0().translate("wdl", allow_empty_container=True)

