from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, File, Int, String, Float

Varlociraptor_Preprocess_Variants_V0_1_0 = CommandToolBuilder(tool="varlociraptor_preprocess_variants", base_command=["varlociraptor", "preprocess", "variants"], inputs=[ToolInput(tag="in_omit_insert_size", input_type=Boolean(optional=True), prefix="--omit-insert-size", doc=InputDocumentation(doc="Do not consider insert size when calculating support for a variant. Use this flag when\nprocessing amplicon data, where indels do not impact the observed insert size")), ToolInput(tag="in_alignment_properties", input_type=File(optional=True), prefix="--alignment-properties", doc=InputDocumentation(doc="Alignment properties JSON file for sample. If not provided, properties will be estimated from the given BAM\nfile.")), ToolInput(tag="in_bam", input_type=File(optional=True), prefix="--bam", doc=InputDocumentation(doc="BAM file with aligned reads from a single sample.")), ToolInput(tag="in_candidates", input_type=File(optional=True), prefix="--candidates", doc=InputDocumentation(doc="VCF/BCF file to process (if omitted, read from STDIN).")), ToolInput(tag="in_max_depth", input_type=Int(optional=True), prefix="--max-depth", doc=InputDocumentation(doc="Maximum number of observations to use for calling. If locus is exceeding this number, downsampling is\nperformed. [default: 200]")), ToolInput(tag="in_output", input_type=File(optional=True), prefix="--output", doc=InputDocumentation(doc="BCF file that shall contain the results (if omitted, write to STDOUT).")), ToolInput(tag="in_stranded_ness", input_type=String(optional=True), prefix="--strandedness", doc=InputDocumentation(doc="Strandedness of sequencing protocol in case of paired-end (opposite strand as usual or same strand as with\nmate-pair sequencing.) [default: opposite]  [possible values: opposite, same]")), ToolInput(tag="in_in_del_window", input_type=Int(optional=True), prefix="--indel-window", doc=InputDocumentation(doc="Number of bases to consider left and right of breakpoint when calculating read support. Currently\nimplemented maximum value is 64. [default: 64]")), ToolInput(tag="in_spurious_del_rate", input_type=Float(optional=True), prefix="--spurious-del-rate", doc=InputDocumentation(doc="Rate of spuriosly deleted bases by the sequencer (Illumina: 5.1e-6, see Schirmer et al. BMC Bioinformatics\n2016). [default: 5.1e-6]")), ToolInput(tag="in_spurious_del_ext_rate", input_type=Float(optional=True), prefix="--spurious-delext-rate", doc=InputDocumentation(doc="Extension rate of spurious deletions by the sequencer (Illumina: 0.0, see Schirmer et al. BMC Bioinformatics\n2016) [default: 0.0]")), ToolInput(tag="in_spurious_ins_rate", input_type=Float(optional=True), prefix="--spurious-ins-rate", doc=InputDocumentation(doc="Rate of spuriously inserted bases by the sequencer (Illumina: 2.8e-6, see Schirmer et al. BMC Bioinformatics\n2016). [default: 2.8e-6]")), ToolInput(tag="in_spurious_in_sext_rate", input_type=Float(optional=True), prefix="--spurious-insext-rate", doc=InputDocumentation(doc="Extension rate of spurious insertions by the sequencer (Illumina: 0.0, see Schirmer et al. BMC\nBioinformatics 2016) [default: 0.0]")), ToolInput(tag="in_reference_dot_fast_a", input_type=String(), position=0, doc=InputDocumentation(doc=""))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Varlociraptor_Preprocess_Variants_V0_1_0().translate("wdl", allow_empty_container=True)

