class: CommandLineTool
id: anvi_estimate_metabolism.cwl
inputs:
- id: in_contigs_db
  doc: "Anvi'o contigs database generated by 'anvi-gen-\ncontigs-database' (default:\
    \ None)"
  type: string?
  inputBinding:
    prefix: --contigs-db
- id: in_meta_genome_mode
  doc: "Treat a given contigs database as a metagenome rather\nthan treating it as\
    \ a single genome. (default: False)"
  type: boolean?
  inputBinding:
    prefix: --metagenome-mode
- id: in_kegg_data_dir
  doc: "The directory path for your KEGG setup, which will\ninclude things like KOfam\
    \ profiles and KEGG MODULE\ndata. Anvi'o will try to use the default path if you\n\
    do not specify anything. (default: None)"
  type: File?
  inputBinding:
    prefix: --kegg-data-dir
- id: in_collection_name
  doc: 'Collection name. (default: None)'
  type: string?
  inputBinding:
    prefix: --collection-name
- id: in_bin_id
  doc: 'Bin name you are interested in. (default: None)'
  type: string?
  inputBinding:
    prefix: --bin-id
- id: in_bin_ids_file
  doc: "Text file for bins (each line should be a unique bin\nid). (default: None)"
  type: File?
  inputBinding:
    prefix: --bin-ids-file
- id: in_external_genomes
  doc: "A two-column TAB-delimited flat text file that lists\nanvi'o contigs databases.\
    \ The first item in the header\nline should read 'name', and the second should\
    \ read\n'contigs_db_path'. Each line in the file should\ndescribe a single entry,\
    \ where the first column is the\nname of the genome (or MAG), and the second column\
    \ is\nthe anvi'o contigs database generated for this genome.\n(default: None)"
  type: File?
  inputBinding:
    prefix: --external-genomes
- id: in_internal_genomes
  doc: "A five-column TAB-delimited flat text file. The header\nline must contain\
    \ these columns: 'name', 'bin_id',\n'collection_id', 'profile_db_path', 'contigs_db_path'.\n\
    Each line should list a single entry, where 'name' can\nbe any name to describe\
    \ the anvi'o bin identified as\n'bin_id' that is stored in a collection. (default:\n\
    None)"
  type: File?
  inputBinding:
    prefix: --internal-genomes
- id: in_meta_genomes
  doc: "A two-column TAB-delimited flat text file. The header\nline must contain these\
    \ columns: 'name',\n'contigs_db_path', and 'profile_db_path'. Each line\nshould\
    \ list a single entry, where 'name' can be any\nname to describe the metagenome\
    \ stored in the anvi'o\ncontigs database. In this context, the anvi'o profiles\n\
    associated with contigs database must be SINGLE\nPROFILES, as in generated by\
    \ the program `anvi-\nprofile` and not `anvi-merge`. (default: None)"
  type: File?
  inputBinding:
    prefix: --metagenomes
- id: in_module_completion_threshold
  doc: "This threshold defines the point at which we consider\na KEGG module to be\
    \ 'complete' or 'present' in a given\ngenome or bin. It is the fraction of steps\
    \ that must\nbe complete in in order for the entire module to be\nmarked complete.\
    \ The default is 0.75."
  type: long?
  inputBinding:
    prefix: --module-completion-threshold
- id: in_output_file_prefix
  doc: "A prefix to be used while naming the output files (no\nfile type extensions\
    \ please; just a prefix). (default:\nNone)"
  type: File?
  inputBinding:
    prefix: --output-file-prefix
- id: in_kegg_output_modes
  doc: "Use this flag to indicate what information you want in\nthe kegg metabolism\
    \ output files, by providing a\ncomma-separated list of output modes (each 'mode'\
    \ you\nprovide will result in a different output file, all\nwith the same prefix).\
    \ The default output modes are\n'kofam_hits' and 'complete_modules'. To see a\
    \ list of\navailable output modes, run this script with the flag\n--list-available-modes.\
    \ (default: None)"
  type: File?
  inputBinding:
    prefix: --kegg-output-modes
- id: in_list_available_modes
  doc: "Use this flag to see the available output modes and\ntheir descriptions. (default:\
    \ False)"
  type: boolean?
  inputBinding:
    prefix: --list-available-modes
- id: in_custom_output_headers
  doc: "For use with the 'custom' output mode. Provide a\ncomma-separated list of\
    \ headers to include in the\noutput matrix. To see a list of available headers,\
    \ run\nthis script with the flag --list-available-output-\nheaders. (default:\
    \ None)"
  type: string?
  inputBinding:
    prefix: --custom-output-headers
- id: in_list_available_output_headers
  doc: "Use this flag to see the available output headers.\n(default: False)"
  type: boolean?
  inputBinding:
    prefix: --list-available-output-headers
- id: in_matrix_format
  doc: "If you want to generate the output in several sparse\nmatrices instead of\
    \ one file, use this flag. In each\nmatrix, contigs DBs will be arranged in columns\
    \ and\nKEGG modules in rows. This output option is especially\nappropriate for\
    \ input option #3. (default: False)"
  type: boolean?
  inputBinding:
    prefix: --matrix-format
- id: in_include_metadata
  doc: "When asking for --matrix-format, you can use this flag\nto make sure the output\
    \ matrix files include columns\nwith metadata for each KEGG Module or KO (like\
    \ the\nmodule name and category for example) before the\nsample columns. (default:\
    \ False)"
  type: boolean?
  inputBinding:
    prefix: --include-metadata
- id: in_include_zeros
  doc: "If you use this flag, long-format output files will\ninclude modules with\
    \ 0 percent completeness score.\n(default: False)"
  type: boolean?
  inputBinding:
    prefix: --include-zeros
- id: in_only_complete
  doc: "Choose this flag if you want only modules over the\nmodule completeness threshold\
    \ to be included in any\noutput files. (default: False)"
  type: boolean?
  inputBinding:
    prefix: --only-complete
- id: in_get_raw_data_as_json
  doc: "If you want the raw metabolism estimation data\ndictionary in JSON-format,\
    \ provide a filename prefix\nto this argument.The program will then output a file\n\
    with the .json extension containing this data.\n(default: None)"
  type: File?
  inputBinding:
    prefix: --get-raw-data-as-json
- id: in_store_json_without_estimation
  doc: "This flag is used to control what is stored in the\nJSON-formatted metabolism\
    \ data dictionary. When this\nflag is provided alongside the --get-raw-data-as-json\n\
    flag, the JSON file will be created without running\nmetabolism estimation, and\
    \ that file will consequently\ninclude only information about KOfam hits and gene\n\
    calls. The idea is that you can then modify this file\nas you like and re-run\
    \ this program using the flag\n--estimate-from-json. (default: False)"
  type: boolean?
  inputBinding:
    prefix: --store-json-without-estimation
- id: in_estimate_from_json
  doc: "If you have a JSON file containing KOfam hits and gene\ncall information from\
    \ your contigs database (such as a\nfile produced using the --get-raw-data-as-json\
    \ flag),\nyou can provide that file to this flag and KEGG\nmetabolism estimates\
    \ will be computed from the\ninformation within instead of from a contigs database.\n\
    (default: None)"
  type: File?
  inputBinding:
    prefix: --estimate-from-json
- id: in_respectively_dot
  doc: "-p PROFILE_DB, --profile-db PROFILE_DB\nAnvi'o profile database (default:\
    \ None)"
  type: string
  inputBinding:
    position: 0
- id: in_kegg_metabolism
  doc: 'üç∫ More on `anvi-estimate-metabolism`:'
  type: string
  inputBinding:
    position: 0
outputs:
- id: out_stdout
  doc: Standard output stream
  type: stdout
- id: out_output_file_prefix
  doc: "A prefix to be used while naming the output files (no\nfile type extensions\
    \ please; just a prefix). (default:\nNone)"
  type: File?
  outputBinding:
    glob: $(inputs.in_output_file_prefix)
- id: out_kegg_output_modes
  doc: "Use this flag to indicate what information you want in\nthe kegg metabolism\
    \ output files, by providing a\ncomma-separated list of output modes (each 'mode'\
    \ you\nprovide will result in a different output file, all\nwith the same prefix).\
    \ The default output modes are\n'kofam_hits' and 'complete_modules'. To see a\
    \ list of\navailable output modes, run this script with the flag\n--list-available-modes.\
    \ (default: None)"
  type: File?
  outputBinding:
    glob: $(inputs.in_kegg_output_modes)
- id: out_get_raw_data_as_json
  doc: "If you want the raw metabolism estimation data\ndictionary in JSON-format,\
    \ provide a filename prefix\nto this argument.The program will then output a file\n\
    with the .json extension containing this data.\n(default: None)"
  type: File?
  outputBinding:
    glob: $(inputs.in_get_raw_data_as_json)
hints:
- class: DockerRequirement
  dockerPull: quay.io/biocontainers/anvio-minimal:7--py_0
cwlVersion: v1.1
baseCommand:
- anvi-estimate-metabolism
