from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Int, String, Directory, Boolean, Float

Evaluate_Assembly_Using_Mapping_Py_V0_1_0 = CommandToolBuilder(tool="evaluate_assembly_using_mapping.py", base_command=["evaluate_assembly_using_mapping.py"], inputs=[ToolInput(tag="in_input_assembly_fasta", input_type=File(optional=True), prefix="-f", doc=InputDocumentation(doc="input assembly fasta file.")), ToolInput(tag="in_input_files_unpaired", input_type=Int(optional=True), prefix="-1", doc=InputDocumentation(doc="Input file(s) with unpaired (single-end) reads to be\nadded to the pool. files could be comma-separated\nlists such as 'seq1,seq2'.")), ToolInput(tag="in_max_lib_len", input_type=Int(optional=True), prefix="--max-lib-len", doc=InputDocumentation(doc="Corresponding to '-X' option in Bowtie2. Default:\n1200.")), ToolInput(tag="in_yesnoauto_input_fasta", input_type=String(optional=True), prefix="-c", doc=InputDocumentation(doc="(yes/no/auto) input fasta is circular. If auto was\nchosen, the input fasta would be treated as circular\nwhen the sequence name ends with '(circular)'.\nDefault: auto")), ToolInput(tag="in_output_folder", input_type=Directory(optional=True), prefix="-o", doc=InputDocumentation(doc="output folder.")), ToolInput(tag="in_threads", input_type=String(optional=True), prefix="-t", doc=InputDocumentation(doc="threads.")), ToolInput(tag="in_continue", input_type=Int(optional=True), prefix="--continue", doc=InputDocumentation(doc="Seed for random number generator. Default: 12345")), ToolInput(tag="in_draw", input_type=Boolean(optional=True), prefix="--draw", doc=InputDocumentation(doc="Draw density plot using matplotlib, which should be")), ToolInput(tag="in_plot_title", input_type=File(optional=True), prefix="--plot-title", doc=InputDocumentation(doc="Default: `the file name of the input fasta`")), ToolInput(tag="in_plot_subtitle", input_type=Int(optional=True), prefix="--plot-subtitle", doc=InputDocumentation(doc="A 4-space indicates a line break. Default: None")), ToolInput(tag="in_plot_transparent", input_type=Boolean(optional=True), prefix="--plot-transparent", doc=InputDocumentation(doc="Default: False")), ToolInput(tag="in_plot_x_density", input_type=Float(optional=True), prefix="--plot-x-density", doc=InputDocumentation(doc="Default: 12000.0")), ToolInput(tag="in_plot_x_gap_dots", input_type=Int(optional=True), prefix="--plot-x-gap-dots", doc=InputDocumentation(doc="Number of sites added in-between isolated contigs.\nDefault: 3000")), ToolInput(tag="in_plot_figure_height", input_type=Float(optional=True), prefix="--plot-figure-height", doc=InputDocumentation(doc="Default: 5.0")), ToolInput(tag="in_plot_y_lim", input_type=String(optional=True), prefix="--plot-y-lim", doc=InputDocumentation(doc="Y axis value limit.")), ToolInput(tag="in_plot_font", input_type=String(optional=True), prefix="--plot-font", doc=InputDocumentation(doc="For plot of unicode characters in some environments.\nUse 'Times New Roman','Arial' etc. Default: none.")), ToolInput(tag="in_disable_customized_error_rate", input_type=Int(optional=True), prefix="--disable-customized-error-rate", doc=InputDocumentation(doc="Assign the path to Bowtie2 binary files if not added\nto the path. Default: try\nGetOrganelleDep/linux/bowtie2 first, then $PATH")), ToolInput(tag="in_bowtie_two_mode", input_type=Int(optional=True), prefix="--bowtie2-mode", doc=InputDocumentation(doc="Default: --sensitive")), ToolInput(tag="in_bowtie_two_options", input_type=Int(optional=True), prefix="--bowtie2-options", doc=InputDocumentation(doc="Default: --no-discordant --dovetail")), ToolInput(tag="in_stat_mode", input_type=Int(optional=True), prefix="--stat-mode", doc=InputDocumentation(doc="Statistical mode for counting multiple hits of a\nsingle read: best/all. The all mode is meaningful only\nwhen '-k <INT>' was included in '--bowtie2-options'.\nDefault: best")), ToolInput(tag="in_debug", input_type=Boolean(optional=True), prefix="--debug", doc=InputDocumentation(doc="Turn on debug mode.")), ToolInput(tag="in_installed_dot", input_type=String(), position=0, doc=InputDocumentation(doc="--plot-format=PLOT_FORMAT"))], outputs=[ToolOutput(tag="out_output_folder", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_output_folder", type_hint=File()), doc=OutputDocumentation(doc="output folder."))], container="quay.io/biocontainers/getorganelle:1.7.3.4--pyh3252c3a_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Evaluate_Assembly_Using_Mapping_Py_V0_1_0().translate("wdl")

