version 1.0

task IsONclust {
  input {
    File? fast_q
    File? fl_nc
    File? ccs
    Int? number_cores_allocated
    Int? debugging_prints_status
    Int? filters_reads_default
    Boolean? ont
    Boolean? iso_seq
    File? use_old_sorted_file
    Boolean? consensus
    Int? abundance_ratio
    Float? rc_identity_threshold
    Boolean? medaka
    Int? kmer_size_default
    Int? window_size_default
    Int? min_shared
    Float? mapped_threshold
    Float? aligned_threshold
    String? batch_type
    Int? min_fraction
    Int? min_prob_no_hits
    Directory? out_folder
  }
  command <<<
    isONclust \
      ~{if defined(fast_q) then ("--fastq " +  '"' + fast_q + '"') else ""} \
      ~{if defined(fl_nc) then ("--flnc " +  '"' + fl_nc + '"') else ""} \
      ~{if defined(ccs) then ("--ccs " +  '"' + ccs + '"') else ""} \
      ~{if defined(number_cores_allocated) then ("--t " +  '"' + number_cores_allocated + '"') else ""} \
      ~{if defined(debugging_prints_status) then ("--d " +  '"' + debugging_prints_status + '"') else ""} \
      ~{if defined(filters_reads_default) then ("--q " +  '"' + filters_reads_default + '"') else ""} \
      ~{if (ont) then "--ont" else ""} \
      ~{if (iso_seq) then "--isoseq" else ""} \
      ~{if (use_old_sorted_file) then "--use_old_sorted_file" else ""} \
      ~{if (consensus) then "--consensus" else ""} \
      ~{if defined(abundance_ratio) then ("--abundance_ratio " +  '"' + abundance_ratio + '"') else ""} \
      ~{if defined(rc_identity_threshold) then ("--rc_identity_threshold " +  '"' + rc_identity_threshold + '"') else ""} \
      ~{if (medaka) then "--medaka" else ""} \
      ~{if defined(kmer_size_default) then ("--k " +  '"' + kmer_size_default + '"') else ""} \
      ~{if defined(window_size_default) then ("--w " +  '"' + window_size_default + '"') else ""} \
      ~{if defined(min_shared) then ("--min_shared " +  '"' + min_shared + '"') else ""} \
      ~{if defined(mapped_threshold) then ("--mapped_threshold " +  '"' + mapped_threshold + '"') else ""} \
      ~{if defined(aligned_threshold) then ("--aligned_threshold " +  '"' + aligned_threshold + '"') else ""} \
      ~{if defined(batch_type) then ("--batch_type " +  '"' + batch_type + '"') else ""} \
      ~{if defined(min_fraction) then ("--min_fraction " +  '"' + min_fraction + '"') else ""} \
      ~{if defined(min_prob_no_hits) then ("--min_prob_no_hits " +  '"' + min_prob_no_hits + '"') else ""} \
      ~{if defined(out_folder) then ("--outfolder " +  '"' + out_folder + '"') else ""}
  >>>
  parameter_meta {
    fast_q: "Path to consensus fastq file(s) (default: False)"
    fl_nc: "The flnc reads generated by the isoseq3 algorithm (BAM\\nfile) (default: False)"
    ccs: "Path to consensus BAM file(s) (default: False)"
    number_cores_allocated: "Number of cores allocated for clustering (default: 8)"
    debugging_prints_status: "For debugging, prints status of clustering and\\nminimizer database every p reads processed. (default:\\n10000)"
    filters_reads_default: "Filters reads with average phred quality value under\\nthis number (default = 7.0). (default: 7.0)"
    ont: "Clustering of ONT transcript reads. (default: False)"
    iso_seq: "Clustering of PacBio Iso-Seq reads. (default: False)"
    use_old_sorted_file: "Using already existing sorted file if present in\\nspecified output directory. (default: False)"
    consensus: "After clustering, (1) run spoa on all clusters, (2)\\ndetect reverse complements, (3) run medaka. (default:\\nFalse)"
    abundance_ratio: "Threshold for --consensus algorithm. Consider only\\nclusters larger than a fraction of number of total\\nreads (default 0.1) (default: 0.1)"
    rc_identity_threshold: "Threshold for --consensus algorithm. Define a reverse\\ncomplement if identity is over this threshold (default\\n0.9) (default: 0.9)"
    medaka: "Run final medaka polishing algorithm. (default: False)"
    kmer_size_default: "Kmer size (default: 15)"
    window_size_default: "Window size (default: 50)"
    min_shared: "Minmum number of minimizers shared between read and\\ncluster (default: 5)"
    mapped_threshold: "Minmum mapped fraction of read to be included in\\ncluster. The density of minimizers to classify a\\nregion as mapped depends on quality of the read.\\n(default: 0.7)"
    aligned_threshold: "Minmum aligned fraction of read to be included in\\ncluster. Aligned identity depends on the quality of\\nthe read. (default: 0.4)"
    batch_type: "In parrallel mode, how to split the reads into chunks\\n\\\"total_nt\\\", \\\"nr_reads\\\", or \\\"weighted\\\" (default:\\ntotal_nt) (default: total_nt)"
    min_fraction: "Minmum fraction of minimizers shared compared to best\\nhit, in order to continue mapping. (default: 0.8)"
    min_prob_no_hits: "Minimum probability for i consecutive minimizers to be\\ndifferent between read and representative and still\\nconsidered as mapped region, under assumption that\\nthey come from the same transcript (depends on read\\nquality). (default: 0.1)"
    out_folder: "A fasta file with transcripts that are shared between\\nsamples and have perfect illumina support. (default:\\nNone)\\n"
  }
  output {
    File out_stdout = stdout()
    File out_use_old_sorted_file = "${in_use_old_sorted_file}"
    Directory out_out_folder = "${in_out_folder}"
  }
}