from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Int, Boolean, File, Float, String

Kraken2_V0_1_0 = CommandToolBuilder(tool="kraken2", base_command=["kraken2"], inputs=[ToolInput(tag="in_db", input_type=Int(optional=True), prefix="--db", doc=InputDocumentation(doc="Name for Kraken 2 DB\n(default: none)")), ToolInput(tag="in_threads", input_type=Int(optional=True), prefix="--threads", doc=InputDocumentation(doc="Number of threads (default: 1)")), ToolInput(tag="in_quick", input_type=Boolean(optional=True), prefix="--quick", doc=InputDocumentation(doc="Quick operation (use first hit or hits)")), ToolInput(tag="in_unclassified_out", input_type=File(optional=True), prefix="--unclassified-out", doc=InputDocumentation(doc="Print unclassified sequences to filename")), ToolInput(tag="in_classified_out", input_type=File(optional=True), prefix="--classified-out", doc=InputDocumentation(doc="Print classified sequences to filename")), ToolInput(tag="in_output", input_type=File(optional=True), prefix="--output", doc=InputDocumentation(doc="Print output to filename (default: stdout); '-' will\nsuppress normal output")), ToolInput(tag="in_confidence", input_type=Float(optional=True), prefix="--confidence", doc=InputDocumentation(doc="Confidence score threshold (default: 0.0); must be\nin [0, 1].")), ToolInput(tag="in_minimum_base_quality", input_type=Int(optional=True), prefix="--minimum-base-quality", doc=InputDocumentation(doc="Minimum base quality used in classification (def: 0,\nonly effective with FASTQ input).")), ToolInput(tag="in_report", input_type=File(optional=True), prefix="--report", doc=InputDocumentation(doc="Print a report with aggregrate counts/clade to file")), ToolInput(tag="in_use_mpa_style", input_type=Boolean(optional=True), prefix="--use-mpa-style", doc=InputDocumentation(doc="With --report, format report output like Kraken 1's")), ToolInput(tag="in_report_minimizer_data", input_type=String(optional=True), prefix="--report-minimizer-data", doc=InputDocumentation(doc=", report minimizer and distinct minimizer\ncount information in addition to normal Kraken report")), ToolInput(tag="in_memory_mapping", input_type=Boolean(optional=True), prefix="--memory-mapping", doc=InputDocumentation(doc="Avoids loading database into RAM")), ToolInput(tag="in_paired", input_type=Boolean(optional=True), prefix="--paired", doc=InputDocumentation(doc="The filenames provided have paired-end reads")), ToolInput(tag="in_use_names", input_type=Boolean(optional=True), prefix="--use-names", doc=InputDocumentation(doc="Print scientific names instead of just taxids")), ToolInput(tag="in_gzip_compressed", input_type=Boolean(optional=True), prefix="--gzip-compressed", doc=InputDocumentation(doc="Input files are compressed with gzip")), ToolInput(tag="in_bzip_two_compressed", input_type=Boolean(optional=True), prefix="--bzip2-compressed", doc=InputDocumentation(doc="Input files are compressed with bzip2")), ToolInput(tag="in_minimum_hit_groups", input_type=Int(optional=True), prefix="--minimum-hit-groups", doc=InputDocumentation(doc="Minimum number of hit groups (overlapping k-mers\nsharing the same minimizer) needed to make a call\n(default: 2)")), ToolInput(tag="in_kraken_mpa_report", input_type=String(), position=0, doc=InputDocumentation(doc="--report-zero-counts    With --report, report counts for ALL taxa, even if"))], outputs=[ToolOutput(tag="out_output", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output", type_hint=File()), doc=OutputDocumentation(doc="Print output to filename (default: stdout); '-' will\nsuppress normal output"))], container="quay.io/biocontainers/kraken2:2.1.1--pl526hc9558a2_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Kraken2_V0_1_0().translate("wdl")

