from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, File

Gtftk_Get_Tx_Seq_V0_1_0 = CommandToolBuilder(tool="gtftk_get_tx_seq", base_command=["gtftk", "get_tx_seq"], inputs=[ToolInput(tag="in_input_file", input_type=Boolean(optional=True), prefix="--inputfile", doc=InputDocumentation(doc="Path to the GTF file. Default to STDIN (default: <stdin>)")), ToolInput(tag="in_output_file", input_type=File(optional=True), prefix="--outputfile", doc=InputDocumentation(doc="Output FASTA file. (default: <stdout>)")), ToolInput(tag="in_genome", input_type=Boolean(optional=True), prefix="--genome", doc=InputDocumentation(doc="The genome in fasta format. Accept path with wildcards (e.g. *.fa). (default: None)")), ToolInput(tag="in_with_introns", input_type=Boolean(optional=True), prefix="--with-introns", doc=InputDocumentation(doc="Set to true to include intronic regions. (default: False)")), ToolInput(tag="in_separator", input_type=Boolean(optional=True), prefix="--separator", doc=InputDocumentation(doc="To separate info in header. (default: |)")), ToolInput(tag="in_label", input_type=Boolean(optional=True), prefix="--label", doc=InputDocumentation(doc="A set of key for the header. (default: feature,transcript_id,gene_id,seqid,start,end)")), ToolInput(tag="in_sleuth_format", input_type=Boolean(optional=True), prefix="--sleuth-format", doc=InputDocumentation(doc="Produce output in sleuth format (still experimental). (default: False)")), ToolInput(tag="in_delete_version", input_type=Boolean(optional=True), prefix="--delete-version", doc=InputDocumentation(doc="In case of --sleuth-format, delete gene_id or transcript_id version number (e.g '.2' in ENSG56765.2). (default: False)")), ToolInput(tag="in_assembly", input_type=Boolean(optional=True), prefix="--assembly", doc=InputDocumentation(doc="In case of --sleuth-format, an assembly version. (default: GRCm38)")), ToolInput(tag="in_del_chr", input_type=Boolean(optional=True), prefix="--del-chr", doc=InputDocumentation(doc="When using --sleuth-format delete 'chr' in sequence id. (default: False)")), ToolInput(tag="in_no_rev_comp", input_type=Boolean(optional=True), prefix="--no-rev-comp", doc=InputDocumentation(doc="Don't reverse complement sequence corresponding to gene on minus strand. (default: False)")), ToolInput(tag="in_explicit", input_type=Boolean(optional=True), prefix="--explicit", doc=InputDocumentation(doc="Write explicitly the name of the keys in the header. (default: False)")), ToolInput(tag="in_verbosity", input_type=Boolean(optional=True), prefix="--verbosity", doc=InputDocumentation(doc="Set output verbosity ([0-3]). (default: 0)")), ToolInput(tag="in_no_date", input_type=File(optional=True), prefix="--no-date", doc=InputDocumentation(doc="Do not add date to output file names. (default: False)")), ToolInput(tag="in_add_chr", input_type=Boolean(optional=True), prefix="--add-chr", doc=InputDocumentation(doc="Add 'chr' to chromosome names before printing output. (default: False)")), ToolInput(tag="in_tmp_dir", input_type=Boolean(optional=True), prefix="--tmp-dir", doc=InputDocumentation(doc="Keep all temporary files into this folder. (default: None)")), ToolInput(tag="in_keep_all", input_type=Boolean(optional=True), prefix="--keep-all", doc=InputDocumentation(doc="Try to keep all temporary files even if process does not terminate normally. (default: False)")), ToolInput(tag="in_logger_file", input_type=Boolean(optional=True), prefix="--logger-file", doc=InputDocumentation(doc="Stores the arguments passed to the command into a file. (default: None)")), ToolInput(tag="in_write_message_to_file", input_type=Boolean(optional=True), prefix="--write-message-to-file", doc=InputDocumentation(doc="Store all message into a file. (default: None)"))], outputs=[ToolOutput(tag="out_output_file", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output_file", type_hint=File()), doc=OutputDocumentation(doc="Output FASTA file. (default: <stdout>)")), ToolOutput(tag="out_no_date", output_type=File(optional=True), selector=InputSelector(input_to_select="in_no_date", type_hint=File()), doc=OutputDocumentation(doc="Do not add date to output file names. (default: False)"))], container="quay.io/biocontainers/pygtftk:1.2.7--py36h4edd4cd_1", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Gtftk_Get_Tx_Seq_V0_1_0().translate("wdl")

