from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Directory, Boolean, String

Brass_Pl_V0_1_0 = CommandToolBuilder(tool="brass.pl", base_command=["brass.pl"], inputs=[ToolInput(tag="in_outdir", input_type=Directory(optional=True), prefix="-outdir", doc=InputDocumentation(doc="Folder to output result to.")), ToolInput(tag="in_tumour", input_type=Boolean(optional=True), prefix="-tumour", doc=InputDocumentation(doc="Tumour BAM file")), ToolInput(tag="in_normal", input_type=Boolean(optional=True), prefix="-normal", doc=InputDocumentation(doc="Normal BAM file")), ToolInput(tag="in_depth", input_type=Boolean(optional=True), prefix="-depth", doc=InputDocumentation(doc="Regions of excessive sequencing depth to be ignored")), ToolInput(tag="in_genome", input_type=Boolean(optional=True), prefix="-genome", doc=InputDocumentation(doc="Genome fasta file")), ToolInput(tag="in_species", input_type=Boolean(optional=True), prefix="-species", doc=InputDocumentation(doc="Species name")), ToolInput(tag="in_assembly", input_type=Boolean(optional=True), prefix="-assembly", doc=InputDocumentation(doc="Assembly name")), ToolInput(tag="in_protocol", input_type=Boolean(optional=True), prefix="-protocol", doc=InputDocumentation(doc="Sequencing protocol (WGS|WXS|RNA)")), ToolInput(tag="in_g_cache", input_type=Boolean(optional=True), prefix="-g_cache", doc=InputDocumentation(doc="Genome cache file.")), ToolInput(tag="in_viral", input_type=Boolean(optional=True), prefix="-viral", doc=InputDocumentation(doc="Virus sequences from NCBI")), ToolInput(tag="in_microbe", input_type=Boolean(optional=True), prefix="-microbe", doc=InputDocumentation(doc="Microbe sequences file prefix from NCBI, please exclude '.N.fa.2bit'")), ToolInput(tag="in_gc_bins", input_type=Boolean(optional=True), prefix="-gcbins", doc=InputDocumentation(doc="5 column bed coord file, col 4 number of non-N bases, col 5 GC fraction.")), ToolInput(tag="in_cent_tel", input_type=Boolean(optional=True), prefix="-centtel", doc=InputDocumentation(doc="TSV file of usable regions of the chromosome\nExample in perl/share/Rdefault/")), ToolInput(tag="in_min_cn", input_type=Boolean(optional=True), prefix="-mincn", doc=InputDocumentation(doc="Minimum CN change for copynumber_flag [0.3].")), ToolInput(tag="in_repeats", input_type=Boolean(optional=True), prefix="-repeats", doc=InputDocumentation(doc="Repeat file, see 'make-repeat-file' (legacy)")), ToolInput(tag="in_samp_stat", input_type=Boolean(optional=True), prefix="-sampstat", doc=InputDocumentation(doc="ASCAT sample statistics file or file containing\nNormalContamination 0.XXXXX [0.25]\nPloidy X.XXX [2.0]")), ToolInput(tag="in_platform", input_type=Boolean(optional=True), prefix="-platform", doc=InputDocumentation(doc="Sequencing platform (when not defined in BAM header)")), ToolInput(tag="in_tum_name", input_type=Boolean(optional=True), prefix="-tum_name", doc=InputDocumentation(doc="Tumour sample name (when not defined in BAM header)")), ToolInput(tag="in_norm_name", input_type=Boolean(optional=True), prefix="-norm_name", doc=InputDocumentation(doc="Normal sample name (when not defined in BAM header)")), ToolInput(tag="in_filter", input_type=Boolean(optional=True), prefix="-filter", doc=InputDocumentation(doc="bgzip tabix-ed normal panel groups file")), ToolInput(tag="in_no_clean", input_type=Boolean(optional=True), prefix="-noclean", doc=InputDocumentation(doc="Don't tidyup the processing areas.")), ToolInput(tag="in_cpus", input_type=Boolean(optional=True), prefix="-cpus", doc=InputDocumentation(doc="Number of cores to use. [1]\n- recommend max 2 during 'input' process.")), ToolInput(tag="in_process", input_type=Boolean(optional=True), prefix="-process", doc=InputDocumentation(doc="Only process this step then exit, optionally set -index")), ToolInput(tag="in_index", input_type=Boolean(optional=True), prefix="-index", doc=InputDocumentation(doc="Optionally restrict '-p' to single job")), ToolInput(tag="in_limit", input_type=Boolean(optional=True), prefix="-limit", doc=InputDocumentation(doc="Define with -p and -i, see below")), ToolInput(tag="in_man", input_type=Boolean(optional=True), prefix="-man", doc=InputDocumentation(doc="Full documentation.")), ToolInput(tag="in_version", input_type=Boolean(optional=True), prefix="-version", doc=InputDocumentation(doc="Version")), ToolInput(tag="in_optional", input_type=String(), position=0, doc=InputDocumentation(doc="-mingroup  -j   Minimum reads to call group [2]."))], outputs=[ToolOutput(tag="out_outdir", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_outdir", type_hint=File()), doc=OutputDocumentation(doc="Folder to output result to."))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Brass_Pl_V0_1_0().translate("wdl", allow_empty_container=True)

