from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Int, File, Directory, Boolean, Float, String

Slamdunk_All_V0_1_0 = CommandToolBuilder(tool="slamdunk_all", base_command=["slamdunk", "all"], inputs=[ToolInput(tag="in_o", input_type=Int(optional=True), prefix="-o", doc=InputDocumentation(doc="[-5 TRIM5] [-a MAXPOLYA] [-n TOPN]")), ToolInput(tag="in_reference", input_type=File(optional=True), prefix="--reference", doc=InputDocumentation(doc="Reference fasta file")), ToolInput(tag="in_bed", input_type=File(optional=True), prefix="--bed", doc=InputDocumentation(doc="BED file with 3'UTR coordinates")), ToolInput(tag="in_filter_bed", input_type=File(optional=True), prefix="--filterbed", doc=InputDocumentation(doc="BED file with 3'UTR coordinates to filter multimappers\n(activates -m)")), ToolInput(tag="in_vcf", input_type=File(optional=True), prefix="--vcf", doc=InputDocumentation(doc="Skip SNP step and provide custom variant file.")), ToolInput(tag="in_output_dir", input_type=Directory(optional=True), prefix="--outputDir", doc=InputDocumentation(doc="Output directory for slamdunk run.")), ToolInput(tag="in_trim_five_p", input_type=Int(optional=True), prefix="--trim-5p", doc=InputDocumentation(doc="Number of bp removed from 5' end of all reads\n(default: 12)")), ToolInput(tag="in_max_polya", input_type=Int(optional=True), prefix="--max-polya", doc=InputDocumentation(doc="Max number of As at the 3' end of a read (default: 4)")), ToolInput(tag="in_top_n", input_type=Int(optional=True), prefix="--topn", doc=InputDocumentation(doc="Max. number of alignments to report per read (default:\n1)")), ToolInput(tag="in_threads", input_type=Int(optional=True), prefix="--threads", doc=InputDocumentation(doc="Thread number (default: 1)")), ToolInput(tag="in_quant_seq", input_type=Boolean(optional=True), prefix="--quantseq", doc=InputDocumentation(doc="Run plain Quantseq alignment without SLAM-seq scoring")), ToolInput(tag="in_end_to_end", input_type=Boolean(optional=True), prefix="--endtoend", doc=InputDocumentation(doc="Use a end to end alignment algorithm for mapping.")), ToolInput(tag="in_multimap", input_type=Boolean(optional=True), prefix="--multimap", doc=InputDocumentation(doc="Use reference to resolve multimappers (requires -n >\n1).")), ToolInput(tag="in_min_mq", input_type=Int(optional=True), prefix="--min-mq", doc=InputDocumentation(doc="Minimum mapping quality (default: 2)")), ToolInput(tag="in_min_identity", input_type=Int(optional=True), prefix="--min-identity", doc=InputDocumentation(doc="Minimum alignment identity (default: 0.95)")), ToolInput(tag="in_max_nm", input_type=Int(optional=True), prefix="--max-nm", doc=InputDocumentation(doc="Maximum NM for alignments (default: -1)")), ToolInput(tag="in_min_coverage", input_type=Int(optional=True), prefix="--min-coverage", doc=InputDocumentation(doc="Minimimum coverage to call variant (default: 10)")), ToolInput(tag="in_var_fraction", input_type=Float(optional=True), prefix="--var-fraction", doc=InputDocumentation(doc="Minimimum variant fraction to call variant (default:\n0.8)")), ToolInput(tag="in_conversion_threshold", input_type=Int(optional=True), prefix="--conversion-threshold", doc=InputDocumentation(doc="Number of T>C conversions required to count read as\nT>C read (default: 1)")), ToolInput(tag="in_max_read_length", input_type=Int(optional=True), prefix="--max-read-length", doc=InputDocumentation(doc="Max read length in BAM file")), ToolInput(tag="in_min_base_qual", input_type=Int(optional=True), prefix="--min-base-qual", doc=InputDocumentation(doc="Min base quality for T -> C conversions (default: 27)")), ToolInput(tag="in_sample_name", input_type=String(optional=True), prefix="--sampleName", doc=InputDocumentation(doc="Use this sample name for all supplied samples")), ToolInput(tag="in_sample_type", input_type=String(optional=True), prefix="--sampleType", doc=InputDocumentation(doc="Use this sample type for all supplied samples")), ToolInput(tag="in_sample_time", input_type=String(optional=True), prefix="--sampleTime", doc=InputDocumentation(doc="Use this sample time for all supplied samples")), ToolInput(tag="in_sample_index", input_type=Int(optional=True), prefix="--sample-index", doc=InputDocumentation(doc="Run analysis only for sample <i>. Use for distributing\nslamdunk analysis on a cluster (index is 1-based).")), ToolInput(tag="in_skip_sam", input_type=Boolean(optional=True), prefix="--skip-sam", doc=InputDocumentation(doc="Output BAM while mapping. Slower but, uses less hard\ndisk.\n")), ToolInput(tag="in_files", input_type=String(), position=0, doc=InputDocumentation(doc="Single csv/tsv file (recommended) containing all\nsample files and sample info or a list of all sample\nBAM/FASTA(gz)/FASTQ(gz) files"))], outputs=[ToolOutput(tag="out_output_dir", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_output_dir", type_hint=File()), doc=OutputDocumentation(doc="Output directory for slamdunk run."))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Slamdunk_All_V0_1_0().translate("wdl", allow_empty_container=True)

