from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, Int, String, File

Bowtie2_Align_S_V0_1_0 = CommandToolBuilder(tool="bowtie2_align_s", base_command=["bowtie2-align-s"], inputs=[ToolInput(tag="in_query_input_files_fastq", input_type=Boolean(optional=True), prefix="-q", doc=InputDocumentation(doc="query input files are FASTQ .fq/.fastq (default)")), ToolInput(tag="in_tab_five", input_type=Boolean(optional=True), prefix="--tab5", doc=InputDocumentation(doc="query input files are TAB5 .tab5")), ToolInput(tag="in_tab_six", input_type=Boolean(optional=True), prefix="--tab6", doc=InputDocumentation(doc="query input files are TAB6 .tab6")), ToolInput(tag="in_q_seq", input_type=Boolean(optional=True), prefix="--qseq", doc=InputDocumentation(doc="query input files are in Illumina's qseq format")), ToolInput(tag="in_query_input_files_multifasta", input_type=Boolean(optional=True), prefix="-f", doc=InputDocumentation(doc="query input files are (multi-)FASTA .fa/.mfa")), ToolInput(tag="in_query_input_files_raw", input_type=Boolean(optional=True), prefix="-r", doc=InputDocumentation(doc="query input files are raw one-sequence-per-line")), ToolInput(tag="in_intiint_query_input", input_type=Int(optional=True), prefix="-F", doc=InputDocumentation(doc=":<int>,i:<int> query input files are continuous FASTA where reads\nare substrings (k-mers) extracted from a FASTA file <s>\nand aligned at offsets 1, 1+i, 1+2i ... end of reference")), ToolInput(tag="in_m_m_r", input_type=Boolean(optional=True), prefix="-c", doc=InputDocumentation(doc="<m1>, <m2>, <r> are sequences themselves, not files")), ToolInput(tag="in_skip", input_type=Int(optional=True), prefix="--skip", doc=InputDocumentation(doc="skip the first <int> reads/pairs in the input (none)")), ToolInput(tag="in_up_to", input_type=Int(optional=True), prefix="--upto", doc=InputDocumentation(doc="stop after first <int> reads/pairs (no limit)")), ToolInput(tag="in_trim_five", input_type=Int(optional=True), prefix="--trim5", doc=InputDocumentation(doc="trim <int> bases from 5'/left end of reads (0)")), ToolInput(tag="in_trim_three", input_type=Int(optional=True), prefix="--trim3", doc=InputDocumentation(doc="trim <int> bases from 3'/right end of reads (0)")), ToolInput(tag="in_trim_to", input_type=Boolean(optional=True), prefix="--trim-to", doc=InputDocumentation(doc="[3:|5:]<int> trim reads exceeding <int> bases from either 3' or 5' end\nIf the read end is not specified then it defaults to 3 (0)")), ToolInput(tag="in_phred_three_three", input_type=Boolean(optional=True), prefix="--phred33", doc=InputDocumentation(doc="qualities are Phred+33 (default)")), ToolInput(tag="in_phred_six_four", input_type=Boolean(optional=True), prefix="--phred64", doc=InputDocumentation(doc="qualities are Phred+64")), ToolInput(tag="in_int_quals", input_type=Boolean(optional=True), prefix="--int-quals", doc=InputDocumentation(doc="qualities encoded as space-delimited integers")), ToolInput(tag="in_sensitive", input_type=Int(optional=True), prefix="--sensitive", doc=InputDocumentation(doc=",1,1.15 (default)")), ToolInput(tag="in_n_ceil", input_type=Int(optional=True), prefix="--n-ceil", doc=InputDocumentation(doc="func for max # non-A/C/G/Ts permitted in aln (L,0,0.15)")), ToolInput(tag="in_d_pad", input_type=Int(optional=True), prefix="--dpad", doc=InputDocumentation(doc="include <int> extra ref chars on sides of DP table (15)")), ToolInput(tag="in_g_bar", input_type=Int(optional=True), prefix="--gbar", doc=InputDocumentation(doc="disallow gaps within <int> nucs of read extremes (4)")), ToolInput(tag="in_ignore_quals", input_type=Boolean(optional=True), prefix="--ignore-quals", doc=InputDocumentation(doc="treat all quality values as 30 on Phred scale (off)")), ToolInput(tag="in_no_fw", input_type=Boolean(optional=True), prefix="--nofw", doc=InputDocumentation(doc="do not align forward (original) version of read (off)")), ToolInput(tag="in_norc", input_type=Boolean(optional=True), prefix="--norc", doc=InputDocumentation(doc="do not align reverse-complement version of read (off)")), ToolInput(tag="in_no_one_mm_up_front", input_type=Boolean(optional=True), prefix="--no-1mm-upfront", doc=InputDocumentation(doc="do not allow 1 mismatch alignments before attempting to\nscan for the optimal seeded alignments")), ToolInput(tag="in_end_to_end", input_type=Boolean(optional=True), prefix="--end-to-end", doc=InputDocumentation(doc="entire read must align; no clipping (on)")), ToolInput(tag="in_ma", input_type=Int(optional=True), prefix="--ma", doc=InputDocumentation(doc="match bonus (0 for --end-to-end, 2 for --local)")), ToolInput(tag="in_mp", input_type=Int(optional=True), prefix="--mp", doc=InputDocumentation(doc="max penalty for mismatch; lower qual = lower penalty (6)")), ToolInput(tag="in_np", input_type=Int(optional=True), prefix="--np", doc=InputDocumentation(doc="penalty for non-A/C/G/Ts in read/ref (1)")), ToolInput(tag="in_rdg", input_type=Int(optional=True), prefix="--rdg", doc=InputDocumentation(doc=",<int>  read gap open, extend penalties (5,3)")), ToolInput(tag="in_rfg", input_type=Int(optional=True), prefix="--rfg", doc=InputDocumentation(doc=",<int>  reference gap open, extend penalties (5,3)")), ToolInput(tag="in_score_min", input_type=Int(optional=True), prefix="--score-min", doc=InputDocumentation(doc="min acceptable alignment score w/r/t read length\n(G,20,8 for local, L,-0.6,-0.6 for end-to-end)")), ToolInput(tag="in_mini_ns", input_type=Int(optional=True), prefix="--minins", doc=InputDocumentation(doc="minimum fragment length (0)")), ToolInput(tag="in_max_ins", input_type=Int(optional=True), prefix="--maxins", doc=InputDocumentation(doc="maximum fragment length (500)")), ToolInput(tag="in_fr", input_type=String(optional=True), prefix="--fr", doc=InputDocumentation(doc="align fw/rev, rev/fw, fw/fw (--fr)")), ToolInput(tag="in_no_mixed", input_type=Boolean(optional=True), prefix="--no-mixed", doc=InputDocumentation(doc="suppress unpaired alignments for paired reads")), ToolInput(tag="in_no_discordant", input_type=Boolean(optional=True), prefix="--no-discordant", doc=InputDocumentation(doc="suppress discordant alignments for paired reads")), ToolInput(tag="in_dovetail", input_type=Boolean(optional=True), prefix="--dovetail", doc=InputDocumentation(doc="concordant when mates extend past each other")), ToolInput(tag="in_no_contain", input_type=Boolean(optional=True), prefix="--no-contain", doc=InputDocumentation(doc="not concordant when one mate alignment contains other")), ToolInput(tag="in_no_overlap", input_type=Boolean(optional=True), prefix="--no-overlap", doc=InputDocumentation(doc="not concordant when mates overlap at all")), ToolInput(tag="in_align_paired_reads", input_type=Int(optional=True), prefix="--align-paired-reads", doc=InputDocumentation(doc="will, by default, attempt to align unpaired BAM reads.\nUse this option to align paired-end reads instead.")), ToolInput(tag="in_preserve_tags", input_type=Boolean(optional=True), prefix="--preserve-tags", doc=InputDocumentation(doc="Preserve tags from the original BAM record by\nappending them to the end of the corresponding SAM output.")), ToolInput(tag="in_time", input_type=Boolean(optional=True), prefix="--time", doc=InputDocumentation(doc="print wall-clock time taken by search phases")), ToolInput(tag="in_quiet", input_type=Boolean(optional=True), prefix="--quiet", doc=InputDocumentation(doc="print nothing to stderr except serious errors")), ToolInput(tag="in_met_file", input_type=File(optional=True), prefix="--met-file", doc=InputDocumentation(doc="send metrics to file at <path> (off)")), ToolInput(tag="in_met_stderr", input_type=Boolean(optional=True), prefix="--met-stderr", doc=InputDocumentation(doc="send metrics to stderr (off)")), ToolInput(tag="in_met", input_type=Int(optional=True), prefix="--met", doc=InputDocumentation(doc="report internal counters & metrics every <int> secs (1)")), ToolInput(tag="in_no_unal", input_type=Boolean(optional=True), prefix="--no-unal", doc=InputDocumentation(doc="suppress SAM records for unaligned reads")), ToolInput(tag="in_no_head", input_type=Boolean(optional=True), prefix="--no-head", doc=InputDocumentation(doc="suppress header lines, i.e. lines starting with @")), ToolInput(tag="in_no_sq", input_type=Boolean(optional=True), prefix="--no-sq", doc=InputDocumentation(doc="suppress @SQ header lines")), ToolInput(tag="in_rg_id", input_type=String(optional=True), prefix="--rg-id", doc=InputDocumentation(doc="set read group id, reflected in @RG line and RG:Z: opt field")), ToolInput(tag="in_rg", input_type=String(optional=True), prefix="--rg", doc=InputDocumentation(doc="add <text> ('lab:value') to @RG line of SAM header.\nNote: @RG line only printed when --rg-id is set.")), ToolInput(tag="in_omit_sec_seq", input_type=Boolean(optional=True), prefix="--omit-sec-seq", doc=InputDocumentation(doc="put '*' in SEQ and QUAL fields for secondary alignments.")), ToolInput(tag="in_sam_no_qname_trunc", input_type=String(optional=True), prefix="--sam-no-qname-trunc", doc=InputDocumentation(doc="standard behavior of truncating readname at first whitespace\nat the expense of generating non-standard SAM.")), ToolInput(tag="in_xeq", input_type=Boolean(optional=True), prefix="--xeq", doc=InputDocumentation(doc="Use '='/'X', instead of 'M,' to specify matches/mismatches in SAM record.")), ToolInput(tag="in_soft_clipped_unmapped_tlen", input_type=String(optional=True), prefix="--soft-clipped-unmapped-tlen", doc=InputDocumentation(doc="soft-clipped bases when reporting TLEN")), ToolInput(tag="in_threads", input_type=Int(optional=True), prefix="--threads", doc=InputDocumentation(doc="number of alignment threads to launch (1)")), ToolInput(tag="in_reorder", input_type=Boolean(optional=True), prefix="--reorder", doc=InputDocumentation(doc="force SAM output order to match order of input reads")), ToolInput(tag="in_mm", input_type=Boolean(optional=True), prefix="--mm", doc=InputDocumentation(doc="use memory-mapped I/O for index; many 'bowtie's can share")), ToolInput(tag="in_qc_filter", input_type=Boolean(optional=True), prefix="--qc-filter", doc=InputDocumentation(doc="filter out reads that are bad according to QSEQ filter")), ToolInput(tag="in_seed", input_type=Int(optional=True), prefix="--seed", doc=InputDocumentation(doc="seed for random number generator (0)")), ToolInput(tag="in_non_deterministic", input_type=String(optional=True), prefix="--non-deterministic", doc=InputDocumentation(doc="rand. gen. arbitrarily instead of using read attributes")), ToolInput(tag="in_note", input_type=String(), position=0, doc=InputDocumentation(doc="")), ToolInput(tag="in_var_62", input_type=String(), position=0, doc=InputDocumentation(doc="")), ToolInput(tag="in_bowtie_two_align", input_type=Int(), position=0, doc=InputDocumentation(doc="")), ToolInput(tag="in_bt_two_idx", input_type=Int(), position=0, doc=InputDocumentation(doc="")), ToolInput(tag="in_i", input_type=String(), position=0, doc=InputDocumentation(doc="")), ToolInput(tag="in_m_one", input_type=Int(), position=0, doc=InputDocumentation(doc="")), ToolInput(tag="in_m_two", input_type=Int(), position=0, doc=InputDocumentation(doc="")), ToolInput(tag="in_var_68", input_type=String(), position=0, doc=InputDocumentation(doc="")), ToolInput(tag="in_var_69", input_type=String(), position=0, doc=InputDocumentation(doc="")), ToolInput(tag="in_specified", input_type=String(), position=0, doc=InputDocumentation(doc="")), ToolInput(tag="in_file", input_type=File(), position=1, doc=InputDocumentation(doc="")), ToolInput(tag="in_files", input_type=String(), position=1, doc=InputDocumentation(doc="")), ToolInput(tag="in_index", input_type=String(), position=1, doc=InputDocumentation(doc="")), ToolInput(tag="in_many", input_type=String(), position=1, doc=InputDocumentation(doc="")), ToolInput(tag="in_are", input_type=String(), position=2, doc=InputDocumentation(doc="")), ToolInput(tag="in_filename", input_type=File(), position=2, doc=InputDocumentation(doc="")), ToolInput(tag="in_for", input_type=String(), position=2, doc=InputDocumentation(doc="")), ToolInput(tag="in_times_dot", input_type=String(), position=2, doc=InputDocumentation(doc="")), ToolInput(tag="in_with", input_type=String(), position=2, doc=InputDocumentation(doc="")), ToolInput(tag="in_e_dot_gdot", input_type=String(), position=3, doc=InputDocumentation(doc="")), ToolInput(tag="in_var_81", input_type=String(), position=3, doc=InputDocumentation(doc="")), ToolInput(tag="in_interleaved", input_type=String(), position=3, doc=InputDocumentation(doc="")), ToolInput(tag="in_prefix", input_type=String(), position=3, doc=InputDocumentation(doc="")), ToolInput(tag="in_unaligned", input_type=String(), position=3, doc=InputDocumentation(doc="")), ToolInput(tag="in_unpaired", input_type=String(), position=3, doc=InputDocumentation(doc="")), ToolInput(tag="in_var_86", input_type=String(), position=4, doc=InputDocumentation(doc="")), ToolInput(tag="in_output", input_type=String(), position=4, doc=InputDocumentation(doc="")), ToolInput(tag="in_paired_end", input_type=String(), position=4, doc=InputDocumentation(doc="")), ToolInput(tag="in_fast_q", input_type=String(), position=5, doc=InputDocumentation(doc="")), ToolInput(tag="in_sorted", input_type=String(), position=5, doc=InputDocumentation(doc="")), ToolInput(tag="in_by", input_type=String(), position=6, doc=InputDocumentation(doc="")), ToolInput(tag="in_read", input_type=String(), position=7, doc=InputDocumentation(doc=""))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Bowtie2_Align_S_V0_1_0().translate("wdl", allow_empty_container=True)

