from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, Directory, String, File, Float, Int

Mirge2_0_Predict_V0_1_0 = CommandToolBuilder(tool="miRge2.0_predict", base_command=["miRge2.0", "predict"], inputs=[ToolInput(tag="in_two_options_file", input_type=Boolean(optional=True), prefix="-s", doc=InputDocumentation(doc="[sample <required> [sample <required> ...]]\ntwo options: 1. A file where each row represents one\nsample name; 2. *.fastq *.fastq ... Or *.fastq.gz\n*.fastq.gz ...")), ToolInput(tag="in_directory_outputs_default", input_type=Directory(optional=True), prefix="-o", doc=InputDocumentation(doc="the directory of the outputs (default: current\ndirectory)")), ToolInput(tag="in_mirna_database_default", input_type=String(optional=True), prefix="-d", doc=InputDocumentation(doc="the miRNA database (default: miRBase. miRGeneDB is\noptional)")), ToolInput(tag="in_pb", input_type=File(optional=True), prefix="-pb", doc=InputDocumentation(doc="the path to the system's bowtie binary")), ToolInput(tag="in_lib", input_type=File(optional=True), prefix="-lib", doc=InputDocumentation(doc="the path to the miRge libraries")), ToolInput(tag="in_sp", input_type=String(optional=True), prefix="-sp", doc=InputDocumentation(doc="the species can be human, mouse, fruitfly, nematode,\nrat and zebrafish (novel miRNA detection is confined\nin human and mouse)")), ToolInput(tag="in_ps", input_type=File(optional=True), prefix="-ps", doc=InputDocumentation(doc="the path to the system's samtools binary")), ToolInput(tag="in_pr", input_type=File(optional=True), prefix="-pr", doc=InputDocumentation(doc="the path to the system's rnafold binary")), ToolInput(tag="in_ex", input_type=Float(optional=True), prefix="-ex", doc=InputDocumentation(doc="the threshold of the proportion of canonical reads for\nthe miRNAs to determine whether keeping them or not\nwhen counting. Users can set it between 0 and 0.5.\n(default: 0.1)")), ToolInput(tag="in_ad", input_type=String(optional=True), prefix="-ad", doc=InputDocumentation(doc="the adapter need to be removed which could be\nillumina, ion or a defined sequence (default: none)")), ToolInput(tag="in_phred_six_four", input_type=Boolean(optional=True), prefix="-phred64", doc=InputDocumentation(doc="phred64 format(default: 64)")), ToolInput(tag="in_spike_in", input_type=Boolean(optional=True), prefix="-spikeIn", doc=InputDocumentation(doc="switch to annotate spike-ins if the bowtie index files\nare loacted at the path of bowtie's index files\n(default: off)")), ToolInput(tag="in_tcf", input_type=Boolean(optional=True), prefix="-tcf", doc=InputDocumentation(doc="switch to write trimmed and collapsed fasta file\n(default: off)")), ToolInput(tag="in_di", input_type=Boolean(optional=True), prefix="-di", doc=InputDocumentation(doc="switch to calculate of isomirs entropy (default: off)")), ToolInput(tag="in_cpu", input_type=Int(optional=True), prefix="-cpu", doc=InputDocumentation(doc="the number of processors to use for trimming, qc, and\nalignment (default: 1)")), ToolInput(tag="in_a_i", input_type=Boolean(optional=True), prefix="-ai", doc=InputDocumentation(doc="switch to calculate of A to I editing (default: off)")), ToolInput(tag="in_gff", input_type=Boolean(optional=True), prefix="-gff", doc=InputDocumentation(doc="switch to output results in gff format (default: off)")), ToolInput(tag="in_trf", input_type=Boolean(optional=True), prefix="-trf", doc=InputDocumentation(doc="switch to analyze tRNA fragment (default: off)")), ToolInput(tag="in_ws", input_type=File(optional=True), prefix="-ws", doc=InputDocumentation(doc="the file containing the overall samples to analysis\nfor novel miRNA prediction. No header, just a list of\n*.fastq file names in a column. Names of files can be\nto your choosing (e.g. filestochecknovel.txt)")), ToolInput(tag="in_min_l", input_type=Int(optional=True), prefix="-minl", doc=InputDocumentation(doc="the minimum length of the reatined reads for novel\nmiRNA detection (default: 16)")), ToolInput(tag="in_max_l", input_type=Int(optional=True), prefix="-maxl", doc=InputDocumentation(doc="the maximum length of the reatined reads for novel\nmiRNA detection (default: 25)")), ToolInput(tag="in_cc", input_type=Int(optional=True), prefix="-cc", doc=InputDocumentation(doc="the maximum read count of the reatined reads for novel\nmiRNA detection (default: 2)")), ToolInput(tag="in_ml", input_type=Int(optional=True), prefix="-ml", doc=InputDocumentation(doc="the maximum number of mapping loci for the retained\nreads for novel miRNA detection (default: 3)")), ToolInput(tag="in_sl", input_type=Int(optional=True), prefix="-sl", doc=InputDocumentation(doc="the seed length when invoking Bowtie for novel miRNA\ndetection (default: 25)")), ToolInput(tag="in_olc", input_type=Int(optional=True), prefix="-olc", doc=InputDocumentation(doc="the length of overlapped seqence when joining reads\ninto longer sequences based on the coordinate on the\ngenome for novel miRNA detection (default: 14)")), ToolInput(tag="in_clc", input_type=Int(optional=True), prefix="-clc", doc=InputDocumentation(doc="the maximum length of the clustered sequences for\nnovel miRNA detection (default: 30)")), ToolInput(tag="in_args", input_type=String(optional=True), position=0, doc=InputDocumentation(doc=""))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Mirge2_0_Predict_V0_1_0().translate("wdl", allow_empty_container=True)

