from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, Int, String, File, Float

Vardict_Java_V0_1_0 = CommandToolBuilder(tool="vardict_java", base_command=["vardict-java"], inputs=[ToolInput(tag="in_indicate_move_indels", input_type=Boolean(optional=True), prefix="-3", doc=InputDocumentation(doc="Indicate to move indels to 3-prime if alternative alignment can be achieved.")), ToolInput(tag="in_number_std_pair", input_type=Int(optional=True), prefix="-A", doc=InputDocumentation(doc="The number of STD. A pair will be considered for DEL if INSERT > INSERT_SIZE + INSERT_STD_AMT *")), ToolInput(tag="in_amplicon", input_type=Boolean(optional=True), prefix="--amplicon", doc=InputDocumentation(doc="<int:float>           Indicate it's amplicon based calling.  Reads that don't map to the amplicon will be skipped.  A read pair\nis considered belonging  to the amplicon if the edges are less than int bp to the amplicon, and overlap\nfraction is at least float.  Default: 10:0.95")), ToolInput(tag="in_adaptor", input_type=String(optional=True), prefix="-adaptor", doc=InputDocumentation(doc="Filter adaptor sequences so that they are not used in realignment. Multiple adaptors can be supplied by\nsetting them with comma, like: --adaptor ACGTTGCTC,ACGGGGTCTC,ACGCGGCTAG .")), ToolInput(tag="in_minimum_reads_determine", input_type=Int(optional=True), prefix="-B", doc=InputDocumentation(doc="The minimum # of reads to determine strand bias, default 2")), ToolInput(tag="in_indexed_bam_file", input_type=File(optional=True), prefix="-b", doc=InputDocumentation(doc="The indexed BAM file")), ToolInput(tag="in_indicate_chromosome_names", input_type=Boolean(optional=True), prefix="-C", doc=InputDocumentation(doc="Indicate the chromosome names are just numbers, such as 1, 2, not chr1, chr2 (deprecated)")), ToolInput(tag="in_the_column_chromosome", input_type=Int(optional=True), prefix="-c", doc=InputDocumentation(doc="The column for chromosome")), ToolInput(tag="in_chimeric", input_type=Boolean(optional=True), prefix="-chimeric", doc=InputDocumentation(doc="Indicate to turn off chimeric reads filtering.")), ToolInput(tag="in_debug", input_type=Boolean(optional=True), prefix="--debug", doc=InputDocumentation(doc="Debug mode.  Will print some error messages and append full genotype at the end.")), ToolInput(tag="in_delimiter_split_tab", input_type=String(optional=True), prefix="-d", doc=InputDocumentation(doc="The delimiter for split region_info, default to tab '\t'")), ToolInput(tag="in_del_dup_var", input_type=Boolean(optional=True), prefix="-deldupvar", doc=InputDocumentation(doc="Turn on deleting of duplicate variants. Variants in this mode are considered and outputted only if start\nposition of variant is inside the region interest.")), ToolInput(tag="in_default_printer", input_type=String(optional=True), prefix="--default-printer", doc=InputDocumentation(doc="The printer type used for different outputs. Default: OUT (i.e. System.out).")), ToolInput(tag="in_column_region_end", input_type=Int(optional=True), prefix="-E", doc=InputDocumentation(doc="The column for region end, e.g. gene end")), ToolInput(tag="in_column_segment_ends", input_type=Int(optional=True), prefix="-e", doc=InputDocumentation(doc="The column for segment ends in the region, e.g. exon ends")), ToolInput(tag="in_hexical_filter_reads", input_type=Int(optional=True), prefix="-F", doc=InputDocumentation(doc="The hexical to filter reads using samtools. Default: 0x504 (filter 2nd alignments, unmapped reads and\nduplicates).  Use -F 0 to turn it off.")), ToolInput(tag="in_threshold_allele_frequency", input_type=Float(optional=True), prefix="-f", doc=InputDocumentation(doc="The threshold for allele frequency, default: 0.01 or 1%")), ToolInput(tag="in_fisher", input_type=Boolean(optional=True), prefix="-fisher", doc=InputDocumentation(doc="Experimental feature: Changes R script (teststrandbias.R and testsomatic.) to Java implementation of\nFisher exact test.")), ToolInput(tag="in_reference_fasta_indexed", input_type=File(optional=True), prefix="-G", doc=InputDocumentation(doc="The reference fasta. Should be indexed (.fai).\nDefaults to: /ngs/reference_data/genomes/Hsapiens/hg19/seq/hg19.fa.  Also short commands can be used to\nset path to:\nhg19 - /ngs/reference_data/genomes/Hsapiens/hg19/seq/hg19.fa\nhg38 - /ngs/reference_data/genomes/Hsapiens/hg38/seq/hg38.fa\nmm10 - /ngs/reference_data/genomes/Mmusculus/mm10/seq/mm10.fa")), ToolInput(tag="in_column_gene_name", input_type=Int(optional=True), prefix="-g", doc=InputDocumentation(doc="The column for gene name, or segment annotation")), ToolInput(tag="in__print_help", input_type=Boolean(optional=True), prefix="-H", doc=InputDocumentation(doc=",--?                              Print this help page")), ToolInput(tag="in_indel_size_bp", input_type=Int(optional=True), prefix="-I", doc=InputDocumentation(doc="The indel size.  Default: 50bp")), ToolInput(tag="in_splice", input_type=Boolean(optional=True), prefix="--splice", doc=InputDocumentation(doc="Output splicing read counts")), ToolInput(tag="in_crisp_r", input_type=Int(optional=True), prefix="--crispr", doc=InputDocumentation(doc="The genomic position that CRISPR/Cas9 suppose to cut, typically 3bp from the PAM NGG site and within the")), ToolInput(tag="in_crispr_mode_minimum", input_type=String(optional=True), prefix="-j", doc=InputDocumentation(doc="In CRISPR mode, the minimum amount in bp that a read needs to overlap with cutting site.  If a read does\nnot meet the criteria,\nit will not be used for variant calling, since it is likely just a partially amplified PCR.  Default: not\nset, or no filtering")), ToolInput(tag="in_include_ns_total", input_type=Boolean(optional=True), prefix="-K", doc=InputDocumentation(doc="Include Ns in the total depth calculation")), ToolInput(tag="in_indicate_perform_default", input_type=Int(optional=True), prefix="-k", doc=InputDocumentation(doc="Indicate whether to perform local realignment.  Default: 1.  Set to 0 to disable it.  For Ion or PacBio,\n0 is recommended.")), ToolInput(tag="in_minimum_structural_variant", input_type=Int(optional=True), prefix="-L", doc=InputDocumentation(doc="The minimum structural variant length to be presented using <DEL> <DUP> <INV> <INS>, etc. Default: 1000.\nAny indel, complex variants less than this will be spelled out with exact nucleotides.")), ToolInput(tag="in_minimum_matches_read", input_type=Int(optional=True), prefix="-M", doc=InputDocumentation(doc="The minimum matches for a read to be considered. If, after soft-clipping, the matched bp is less than\nINT, then the read is discarded. It's meant for PCR based targeted sequencing where there's no insert and\nthe matching is only the primers.\nDefault: 0, or no filtering")), ToolInput(tag="in_set_reads_more", input_type=Int(optional=True), prefix="-m", doc=InputDocumentation(doc="If set, reads with mismatches more than INT will be filtered and ignored.  Gaps are not counted as")), ToolInput(tag="in_m_freq", input_type=Float(optional=True), prefix="-mfreq", doc=InputDocumentation(doc="The variant frequency threshold to determine variant as good in case of monomer MSI. Default: 0.25")), ToolInput(tag="in_sample_name_used", input_type=String(optional=True), prefix="-N", doc=InputDocumentation(doc="The sample name to be used directly.  Will overwrite -n option")), ToolInput(tag="in_regular_expression_extract", input_type=String(optional=True), prefix="-n", doc=InputDocumentation(doc="The regular expression to extract sample name from BAM filenames.  Default to: /([^\/\._]+?)_[^\/]*.bam/")), ToolInput(tag="in_nm_freq", input_type=Float(optional=True), prefix="-nmfreq", doc=InputDocumentation(doc="The variant frequency threshold to determine variant as good in case of non-monomer MSI. Default: 0.1")), ToolInput(tag="in_reads_have_mean", input_type=String(optional=True), prefix="-O", doc=InputDocumentation(doc="The reads should have at least mean MapQ to be considered a valid variant.  Default: no filtering")), ToolInput(tag="in_qratio_goodqualityreadsbadqualityreads_quality", input_type=Float(optional=True), prefix="-o", doc=InputDocumentation(doc="The Qratio of (good_quality_reads)/(bad_quality_reads+0.5).  The quality is defined by -q option.\nDefault: 1.5")), ToolInput(tag="in_read_position_filter", input_type=Int(optional=True), prefix="-P", doc=InputDocumentation(doc="The read position filter.  If the mean variants position is less that specified, it's considered false")), ToolInput(tag="in_pileup_regardless_frequency", input_type=Boolean(optional=True), prefix="-p", doc=InputDocumentation(doc="Do pileup regardless of the frequency")), ToolInput(tag="in_set_reads_less", input_type=Int(optional=True), prefix="-Q", doc=InputDocumentation(doc="If set, reads with mapping quality less than INT will be filtered and ignored")), ToolInput(tag="in_phred_score_base", input_type=Int(optional=True), prefix="-q", doc=InputDocumentation(doc="The phred score for a base to be considered a good call.  Default: 25 (for Illumina)\nFor PGM, set it to ~15, as PGM tends to under estimate base quality.")), ToolInput(tag="in_region_interest_format", input_type=String(optional=True), prefix="-R", doc=InputDocumentation(doc="The region of interest.  In the format of chr:start-end.  If end is omitted, then a single position.  No\nBED is needed.")), ToolInput(tag="in_minimum_variant_reads", input_type=Int(optional=True), prefix="-r", doc=InputDocumentation(doc="The minimum # of variant reads, default 2")), ToolInput(tag="in_column_region_start", input_type=Int(optional=True), prefix="-S", doc=InputDocumentation(doc="The column for region start, e.g. gene start")), ToolInput(tag="in_column_segment_starts", input_type=Int(optional=True), prefix="-s", doc=InputDocumentation(doc="The column for segment starts in the region, e.g. exon starts")), ToolInput(tag="in_trim", input_type=Int(optional=True), prefix="--trim", doc=InputDocumentation(doc="Trim bases after [INT] bases in the reads")), ToolInput(tag="in_ded_up", input_type=Boolean(optional=True), prefix="--dedup", doc=InputDocumentation(doc="Indicate to remove duplicated reads.  Only one pair with same start positions will be kept")), ToolInput(tag="in_th", input_type=Int(optional=True), prefix="-th", doc=InputDocumentation(doc="Threads count.")), ToolInput(tag="in_no_sv", input_type=Boolean(optional=True), prefix="--nosv", doc=InputDocumentation(doc="Turn off structural variant calling.")), ToolInput(tag="in_indicate_mode_usingforward", input_type=Boolean(optional=True), prefix="-u", doc=InputDocumentation(doc="Indicate unique mode, which when mate pairs overlap, the overlapping part will be counted only once using\nforward read only.")), ToolInput(tag="in_un", input_type=Boolean(optional=True), prefix="-UN", doc=InputDocumentation(doc="Indicate unique mode, which when mate pairs overlap, the overlapping part will be counted only once using\nfirst read only.")), ToolInput(tag="in_lowest_frequency_normal", input_type=Float(optional=True), prefix="-V", doc=InputDocumentation(doc="The lowest frequency in the normal sample allowed for a putative somatic mutation.  Defaults to 0.05")), ToolInput(tag="in_vcf_format_output", input_type=Boolean(optional=True), prefix="-v", doc=InputDocumentation(doc="VCF format output")), ToolInput(tag="in_vs", input_type=String(optional=True), prefix="-VS", doc=InputDocumentation(doc="How strict to be when reading a SAM or BAM.")), ToolInput(tag="in_insert_size", input_type=Int(optional=True), prefix="--insert-size", doc=InputDocumentation(doc="The insert size.  Used for SV calling.  Default: 300")), ToolInput(tag="in_extension_bp_look", input_type=Int(optional=True), prefix="-X", doc=InputDocumentation(doc="Extension of bp to look for mismatches after insersion or deletion.  Default to 2 bp, or only calls when\nthey're within 2 bp.")), ToolInput(tag="in_number_nucleotide_extend", input_type=Int(optional=True), prefix="-x", doc=InputDocumentation(doc="The number of nucleotide to extend for each segment, default: 0")), ToolInput(tag="in_ref_extension", input_type=Int(optional=True), prefix="--ref-extension", doc=InputDocumentation(doc="Extension of bp of reference to build lookup table. Default to 1200 bp. Increase the number will slowdown\nthe program. The main purpose is to call large indels with 1000 bp that can be missed by discordant mate\npairs.")), ToolInput(tag="in_down_sample", input_type=Float(optional=True), prefix="--downsample", doc=InputDocumentation(doc="For downsampling fraction.  e.g. 0.7 means roughly 70% downsampling.  Default: No downsampling.  Use with")), ToolInput(tag="in_indicate_coordinates_zerobased", input_type=Int(optional=True), prefix="-z", doc=InputDocumentation(doc="Indicate whether coordinates are zero-based, as IGV uses.  Default: 1 for BED file or amplicon BED file.\nUse 0 to turn it off. When using the -R option, it's set to 0")), ToolInput(tag="in_insert_std_dot", input_type=String(), position=0, doc=InputDocumentation(doc="Default: 4")), ToolInput(tag="in_guide_dot", input_type=String(), position=1, doc=InputDocumentation(doc="For")), ToolInput(tag="in_mismatches_dot", input_type=String(), position=0, doc=InputDocumentation(doc="Valid only for bowtie2/TopHat or BWA aln followed by sampe.  BWA mem is calculated as NM - Indels.")), ToolInput(tag="in_positive_dot", input_type=String(), position=0, doc=InputDocumentation(doc="Default: 5")), ToolInput(tag="in_strict", input_type=String(), position=1, doc=InputDocumentation(doc="- throw an exception if something looks wrong.")), ToolInput(tag="in_emit_warnings_keep", input_type=String(), position=2, doc=InputDocumentation(doc="- Emit warnings but keep going if possible.")), ToolInput(tag="in_silent", input_type=String(), position=3, doc=InputDocumentation(doc="- Like LENIENT, only don't emit warning messages.")), ToolInput(tag="in_winsertstd_int_insert", input_type=String(), position=0, doc=InputDocumentation(doc="-W,--insert-std <INT>               The insert size STD.  Used for SV calling.  Default: 100")), ToolInput(tag="in_caution_dot", input_type=String(), position=1, doc=InputDocumentation(doc="The"))], outputs=[], container="quay.io/biocontainers/vardict-java:1.8.2--0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Vardict_Java_V0_1_0().translate("wdl")

