class: CommandLineTool
id: ../../../../home/ubuntu/BiocondaCli/motif_fit.cwl
inputs:
- id: input_file
  doc: 'Path to a FASTA file. (default: None)'
  type: string
  inputBinding:
    prefix: --input-file
- id: training_size
  doc: 'Size of the random sequence sample to use for fitting the discriminative model.
    If None then all instances are used. (default: None)'
  type: string
  inputBinding:
    prefix: --training-size
- id: model_filename
  doc: 'File name for serialized model object. (default: motif.model)'
  type: string
  inputBinding:
    prefix: --model-filename
- id: output_dir
  doc: 'Path to output directory. (default: out)'
  type: string
  inputBinding:
    prefix: --output-dir
- id: min_subarray_size
  doc: 'Minimal size in number of nucleotides of the motives to search. (default:
    7)'
  type: long
  inputBinding:
    prefix: --min-subarray-size
- id: max_subarray_size
  doc: 'Maximal size in number of nucleotides of the motives to search. (default:
    10)'
  type: long
  inputBinding:
    prefix: --max-subarray-size
- id: min_motif_count
  doc: 'Minimal number of occurrences for a motif sequence to be accepted. (default:
    1)'
  type: long
  inputBinding:
    prefix: --min-motif-count
- id: min_cluster_size
  doc: 'Minimal number of motif sequences in a cluster to accept the cluster. (default:
    1)'
  type: long
  inputBinding:
    prefix: --min-cluster-size
- id: random_state
  doc: 'Random seed. (default: 1)'
  type: string
  inputBinding:
    prefix: --random-state
- id: log_full_state
  doc: 'If set, log all the internal parameters values and motif database of the motif
    finder. Warning: it can generate large logging files. (default: False)'
  type: boolean
  inputBinding:
    prefix: --log-full-state
- id: negative_input_file
  doc: 'Path to a FASTA file. (default: None)'
  type: string
  inputBinding:
    prefix: --negative-input-file
- id: negative_ratio
  doc: 'Factor multiplying the training-size to obtain the number of negative instances
    generated by random permutation. (default: 2)'
  type: string
  inputBinding:
    prefix: --negative-ratio
- id: shuffle_order
  doc: 'When shuffling sequences to create negative instances, consider k-mers of
    specified order. (default: 2)'
  type: string
  inputBinding:
    prefix: --shuffle-order
- id: nbits
  doc: 'Number of bits used to express the graph kernel features. A value of 20 corresponds
    to 2**20=1 million possible features. (default: 20)'
  type: string
  inputBinding:
    prefix: --nbits
- id: complexity
  doc: 'Size of the generalization of k-mers for graphs. (default: 4)'
  type: string
  inputBinding:
    prefix: --complexity
- id: n_iter_search
  doc: 'Number of randomly generated hyper parameter configurations tried during the
    discriminative model optimization. A value of 1 implies using the estimator default
    values. (default: 1)'
  type: string
  inputBinding:
    prefix: --n-iter-search
- id: n_jobs
  doc: 'Number of cores to use in multiprocessing. (default: 2)'
  type: string
  inputBinding:
    prefix: --n-jobs
- id: n_blocks
  doc: 'Number of blocks in which to divide the input for the multiprocessing elaboration.
    (default: 8)'
  type: string
  inputBinding:
    prefix: --n-blocks
- id: block_size
  doc: 'Number of instances per block for the multiprocessing elaboration. (default:
    None)'
  type: string
  inputBinding:
    prefix: -block-size
- id: pre_processor_n_jobs
  doc: 'Number of cores to use in multiprocessing. (default: 4)'
  type: string
  inputBinding:
    prefix: --pre-processor-n-jobs
- id: pre_processor_n_blocks
  doc: 'Number of blocks in which to divide the input for the multiprocessing elaboration.
    (default: 10)'
  type: string
  inputBinding:
    prefix: --pre-processor-n-blocks
- id: pre_processor_block_size
  doc: 'Number of instances per block for the multiprocessing elaboration. (default:
    None)'
  type: string
  inputBinding:
    prefix: --pre-processor-block-size
outputs: []
cwlVersion: v1.1
baseCommand:
- motif
- fit
