class: CommandLineTool
id: PlotTransCoverage.cwl
inputs:
- id: in_input
  doc: "Coverage file generated by\nCoverageOfEachTrans.py.[required]"
  type: File?
  inputBinding:
    prefix: --input
- id: in_ot_put_prefix
  doc: Prefix of output files.[required]
  type: string?
  inputBinding:
    prefix: --otput_prefix
- id: in_coordinate_file
  doc: "The file should contain the coordinate of start and\nstop codon. Generated\
    \ by\nOutputTranscriptInfo.py.[required]"
  type: File?
  inputBinding:
    prefix: --coordinateFile
- id: in_target_gene
  doc: "Target genes for plot. If '-m single-gene', '-t' is a\ngene name or gene id\
    \ or transcript id; elif '-m gene-\nlist', '-t' is a file containing some gene\
    \ names or\ngene ids or transcript ids with a column name."
  type: File?
  inputBinding:
    prefix: --target-gene
- id: in_id_type
  doc: "define the id type users input. the default is\ntranscript id, if not, will\
    \ be transformed into\ntranscript id. default=transcript_id"
  type: string?
  inputBinding:
    prefix: --id-type
- id: in_color
  doc: Color for plot. default=lightskyblue
  type: string?
  inputBinding:
    prefix: --color
- id: in_ymax
  doc: The max of ylim. default=none
  type: long?
  inputBinding:
    prefix: --ymax
- id: in_ymin
  doc: The min of ylim. default=none
  type: long?
  inputBinding:
    prefix: --ymin
- id: in_type
  doc: "Type for input. Either 'single-gene' or 'gene-list'.\ndefault=single-gene."
  type: string?
  inputBinding:
    prefix: --type
- id: in_mode
  doc: "Mode of calculation. Either coverage or density of a\ntranscript or transcripts.\
    \ default=coverage\n"
  type: string?
  inputBinding:
    prefix: --mode
outputs:
- id: out_stdout
  doc: Standard output stream
  type: stdout
hints:
- class: DockerRequirement
  dockerPull: quay.io/biocontainers/ribominer:0.2.3.1--pyh3252c3a_0
cwlVersion: v1.1
baseCommand:
- PlotTransCoverage
