class: CommandLineTool
id: ../../../mimodd_map.cwl
inputs:
- id: in_ofile
  doc: "redirect the binned variant counts to this file\n(default: stdout)."
  type: File
  inputBinding:
    prefix: --ofile
- id: in_quiet
  doc: suppress warning messages about plotting problems.
  type: boolean
  inputBinding:
    prefix: --quiet
- id: in_bin_sizes
  doc: "list of bin sizes to be used for histogram plots and\nlinkage reports (default:\
    \ 1Mb and 500Kb)"
  type: long[]
  inputBinding:
    prefix: --bin-sizes
- id: in_mapping_sample
  doc: "name of the sample (as appearing in the input vcf\nfile) for which variants\
    \ should be mapped (cannot be\nused in SVD mode)"
  type: File
  inputBinding:
    prefix: --mapping-sample
- id: in_related_parent
  doc: "name of the sample to provide related parent strain\n(mutagenesis strain)\
    \ variants for the analysis in\nVariant Allele Frequency (VAF) mode."
  type: string
  inputBinding:
    prefix: --related-parent
- id: in_unrelated_parent
  doc: "name of the sample to provide unrelated parent strain\n(mapping strain) variants\
    \ for the analysis in Variant\nAllele Frequency (VAF) mode."
  type: string
  inputBinding:
    prefix: --unrelated-parent
- id: in_infer_missing
  doc: "if variant data for either the related or the\nunrelated parent strain is\
    \ not provided, the tool can\ntry to infer the alleles present in that parent\
    \ from\nthe allele spectrum found in the mapping sample. Use\nwith caution on\
    \ carefully filtered variant lists only!"
  type: boolean
  inputBinding:
    prefix: --infer-missing
- id: in_contrast_sample
  doc: "name of the sample (as appearing in the input vcf\nfile) that provides the\
    \ contrast for the mapping\nsample"
  type: File
  inputBinding:
    prefix: --contrast-sample
- id: in_text_file
  doc: "generate text-based output for every variant position\nand save it to the\
    \ specified file. This file can be\nused as input during later runs of the tool,\
    \ which\nwill speed up replotting."
  type: File
  inputBinding:
    prefix: --text-file
- id: in_seq_info
  doc: "if an input file does not specify required contig\ninformation (chromosome\
    \ names and sizes), extract this\ninformation from SEQINFO_FILE. This file can\
    \ be a\nreference genome file in fasta format or a CloudMap-\nstyle sequence dictionary\
    \ file. This option is never\nnecessary with MiModD-generated input, but can be\n\
    useful with input files generated by third-party tools\nthat do not record contig\
    \ information in their output."
  type: File
  inputBinding:
    prefix: --seqinfo
- id: in_cloud_map
  doc: "generate valid input for the original CloudMap Mapping\ntools and save it\
    \ to the text output file specified by\nthe \"-t\" option. This option can only\
    \ be used in \"SVD\"\nor \"VAF\" mode, which have an equivalent in CloudMap."
  type: boolean
  inputBinding:
    prefix: --cloudmap
- id: in_plot_file
  doc: "generate graphical output and store it in the given\nfile (default: no graphical\
    \ output)"
  type: File
  inputBinding:
    prefix: --plot-file
- id: in_fit_width
  doc: do not autoscale x-axes to size of largest contig
  type: boolean
  inputBinding:
    prefix: --fit-width
- id: in_no_scatter
  doc: "do not produce scatter plots of observed segregation\nrates; just plot histograms"
  type: boolean
  inputBinding:
    prefix: --no-scatter
- id: in_loess_span
  doc: "span parameter for the Loess regression line through\nthe linkage data (default:\
    \ 0.1, specify 0 to skip\ncalculation)"
  type: double
  inputBinding:
    prefix: --loess-span
- id: in_ylim_scatter
  doc: 'upper limit for scatter plot y-axis (default: 1)'
  type: double
  inputBinding:
    prefix: --ylim-scatter
- id: in_points_colors
  doc: color(s) for scatter plot data points
  type: string[]
  inputBinding:
    prefix: --points-colors
- id: in_loess_colors
  doc: "color(s) for regression line(s) through scatter plot\ndata"
  type: string[]
  inputBinding:
    prefix: --loess-colors
- id: in_no_hist
  doc: "do not produce linkage histogram plots; only generate\nscatter plots"
  type: boolean
  inputBinding:
    prefix: --no-hist
- id: in_no_kde
  doc: do not add kernel density estimate lines to histogram
  type: boolean
  inputBinding:
    prefix: --no-kde
- id: in_hist_colors
  doc: "list of colors to be used for plotting histogram bars\nof different width\
    \ (default: darkgrey and red)\n"
  type: string[]
  inputBinding:
    prefix: --hist-colors
- id: in_analysis_mode
  doc: "specify \"SVD\" for Simple Variant Density analysis,\n\"VAF\" for Variant\
    \ Allele Frequency analysis or \"VAC\"\nfor Variant Allele Contrast analysis."
  type: string
  inputBinding:
    position: 0
- id: in_input_file
  doc: "valid input files are VCF files or per-variant report\nfiles (as generated\
    \ by this tool with the \"-t\" option\nor by the CloudMap Hawaiian Variant Density\
    \ Mapping\ntool)."
  type: string
  inputBinding:
    position: 1
- id: in_plots
  doc: '--ylim-hist FLOAT     upper limit for histogram plot y-axis (default: auto)'
  type: string
  inputBinding:
    position: 0
outputs:
- id: out_stdout
  doc: Standard output stream
  type: stdout
- id: out_plot_file
  doc: "generate graphical output and store it in the given\nfile (default: no graphical\
    \ output)"
  type: File
  outputBinding:
    glob: $(inputs.in_plot_file)
cwlVersion: v1.1
baseCommand:
- mimodd
- map
