class: CommandLineTool
id: mimodd_map.cwl
inputs:
- id: analysis_mode
  doc: specify "SVD" for Simple Variant Density analysis, "VAF" for Variant Allele
    Frequency analysis or "VAC" for Variant Allele Contrast analysis.
  type: string
  inputBinding:
    position: 0
- id: input_file
  doc: valid input files are VCF files or per-variant report files (as generated by
    this tool with the "-t" option or by the CloudMap Hawaiian Variant Density Mapping
    tool).
  type: string
  inputBinding:
    position: 1
- id: ofile
  doc: 'redirect the binned variant counts to this file (default: stdout).'
  type: string
  inputBinding:
    prefix: --ofile
- id: quiet
  doc: suppress warning messages about plotting problems.
  type: boolean
  inputBinding:
    prefix: --quiet
- id: bin_sizes
  doc: 'list of bin sizes to be used for histogram plots and linkage reports (default:
    1Mb and 500Kb)'
  type: long[]
  inputBinding:
    prefix: --bin-sizes
- id: mapping_sample
  doc: name of the sample (as appearing in the input vcf file) for which variants
    should be mapped (cannot be used in SVD mode)
  type: string
  inputBinding:
    prefix: --mapping-sample
- id: related_parent
  doc: name of the sample to provide related parent strain (mutagenesis strain) variants
    for the analysis in Variant Allele Frequency (VAF) mode.
  type: string
  inputBinding:
    prefix: --related-parent
- id: unrelated_parent
  doc: name of the sample to provide unrelated parent strain (mapping strain) variants
    for the analysis in Variant Allele Frequency (VAF) mode.
  type: string
  inputBinding:
    prefix: --unrelated-parent
- id: infer_missing
  doc: if variant data for either the related or the unrelated parent strain is not
    provided, the tool can try to infer the alleles present in that parent from the
    allele spectrum found in the mapping sample. Use with caution on carefully filtered
    variant lists only!
  type: boolean
  inputBinding:
    prefix: --infer-missing
- id: contrast_sample
  doc: name of the sample (as appearing in the input vcf file) that provides the contrast
    for the mapping sample
  type: string
  inputBinding:
    prefix: --contrast-sample
- id: text_file
  doc: generate text-based output for every variant position and save it to the specified
    file. This file can be used as input during later runs of the tool, which will
    speed up replotting.
  type: string
  inputBinding:
    prefix: --text-file
- id: seq_info
  doc: if an input file does not specify required contig information (chromosome names
    and sizes), extract this information from SEQINFO_FILE. This file can be a reference
    genome file in fasta format or a CloudMap- style sequence dictionary file. This
    option is never necessary with MiModD-generated input, but can be useful with
    input files generated by third-party tools that do not record contig information
    in their output.
  type: string
  inputBinding:
    prefix: --seqinfo
- id: cloud_map
  doc: generate valid input for the original CloudMap Mapping tools and save it to
    the text output file specified by the "-t" option. This option can only be used
    in "SVD" or "VAF" mode, which have an equivalent in CloudMap.
  type: boolean
  inputBinding:
    prefix: --cloudmap
- id: plot_file
  doc: 'generate graphical output and store it in the given file (default: no graphical
    output)'
  type: File
  inputBinding:
    prefix: --plot-file
- id: fit_width
  doc: do not autoscale x-axes to size of largest contig
  type: boolean
  inputBinding:
    prefix: --fit-width
- id: no_scatter
  doc: do not produce scatter plots of observed segregation rates; just plot histograms
  type: boolean
  inputBinding:
    prefix: --no-scatter
- id: loess_span
  doc: 'span parameter for the Loess regression line through the linkage data (default:
    0.1, specify 0 to skip calculation)'
  type: double
  inputBinding:
    prefix: --loess-span
- id: ylim_scatter
  doc: 'upper limit for scatter plot y-axis (default: 1)'
  type: double
  inputBinding:
    prefix: --ylim-scatter
- id: points_colors
  doc: color(s) for scatter plot data points
  type: string[]
  inputBinding:
    prefix: --points-colors
- id: loess_colors
  doc: color(s) for regression line(s) through scatter plot data
  type: string[]
  inputBinding:
    prefix: --loess-colors
- id: no_hist
  doc: do not produce linkage histogram plots; only generate scatter plots
  type: boolean
  inputBinding:
    prefix: --no-hist
- id: no_kde
  doc: do not add kernel density estimate lines to histogram plots
  type: boolean
  inputBinding:
    prefix: --no-kde
- id: ylim_hist
  doc: 'upper limit for histogram plot y-axis (default: auto)'
  type: double
  inputBinding:
    prefix: --ylim-hist
- id: hist_colors
  doc: 'list of colors to be used for plotting histogram bars of different width (default:
    darkgrey and red)'
  type: string[]
  inputBinding:
    prefix: --hist-colors
outputs: []
cwlVersion: v1.1
baseCommand:
- mimodd
- map
