from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Int, File, Boolean, String, Float

Seqkit_Bam_V0_1_0 = CommandToolBuilder(tool="seqkit_bam", base_command=["seqkit", "bam"], inputs=[ToolInput(tag="in_bins", input_type=Int(optional=True), prefix="--bins", doc=InputDocumentation(doc="number of histogram bins (default -1)")), ToolInput(tag="in_bundle", input_type=Int(optional=True), prefix="--bundle", doc=InputDocumentation(doc="partition BAM file into loci (-1) or bundles with this minimum size")), ToolInput(tag="in_count", input_type=File(optional=True), prefix="--count", doc=InputDocumentation(doc="count reads per reference and save to this file")), ToolInput(tag="in_delay", input_type=Int(optional=True), prefix="--delay", doc=InputDocumentation(doc="sleep this many seconds after plotting (default 1)")), ToolInput(tag="in_dump", input_type=Boolean(optional=True), prefix="--dump", doc=InputDocumentation(doc="print histogram data to stderr instead of plotting")), ToolInput(tag="in_exclude_ids", input_type=File(optional=True), prefix="--exclude-ids", doc=InputDocumentation(doc="exclude records with IDs contained in this file")), ToolInput(tag="in_exec_after", input_type=String(optional=True), prefix="--exec-after", doc=InputDocumentation(doc="execute command after reporting")), ToolInput(tag="in_exec_before", input_type=String(optional=True), prefix="--exec-before", doc=InputDocumentation(doc="execute command before reporting")), ToolInput(tag="in_field", input_type=String(optional=True), prefix="--field", doc=InputDocumentation(doc="target fields")), ToolInput(tag="in_grep_ids", input_type=File(optional=True), prefix="--grep-ids", doc=InputDocumentation(doc="only keep records with IDs contained in this file")), ToolInput(tag="in_idx_count", input_type=Boolean(optional=True), prefix="--idx-count", doc=InputDocumentation(doc="fast read per reference counting based on the BAM index")), ToolInput(tag="in_idx_stat", input_type=Boolean(optional=True), prefix="--idx-stat", doc=InputDocumentation(doc="fast statistics based on the BAM index")), ToolInput(tag="in_img", input_type=File(optional=True), prefix="--img", doc=InputDocumentation(doc="save histogram to this PDF/image file")), ToolInput(tag="in_list_fields", input_type=Boolean(optional=True), prefix="--list-fields", doc=InputDocumentation(doc="list all available BAM record features")), ToolInput(tag="in_log", input_type=Boolean(optional=True), prefix="--log", doc=InputDocumentation(doc="log10(x+1) transform numeric values")), ToolInput(tag="in_map_qual", input_type=Int(optional=True), prefix="--map-qual", doc=InputDocumentation(doc="minimum mapping quality")), ToolInput(tag="in_pass", input_type=Boolean(optional=True), prefix="--pass", doc=InputDocumentation(doc="passthrough mode (forward filtered BAM to output)")), ToolInput(tag="in_pretty", input_type=Boolean(optional=True), prefix="--pretty", doc=InputDocumentation(doc="pretty print certain TSV outputs")), ToolInput(tag="in_prim_only", input_type=Boolean(optional=True), prefix="--prim-only", doc=InputDocumentation(doc="filter out non-primary alignment records")), ToolInput(tag="in_print_freq", input_type=Int(optional=True), prefix="--print-freq", doc=InputDocumentation(doc="print/report after this many records (-1 for print after EOF) (default -1)")), ToolInput(tag="in_quiet_mode", input_type=Boolean(optional=True), prefix="--quiet-mode", doc=InputDocumentation(doc="supress all plotting to stderr")), ToolInput(tag="in_range_max", input_type=Float(optional=True), prefix="--range-max", doc=InputDocumentation(doc="discard record with field (-f) value greater than this flag (default NaN)")), ToolInput(tag="in_range_min", input_type=Float(optional=True), prefix="--range-min", doc=InputDocumentation(doc="discard record with field (-f) value less than this flag (default NaN)")), ToolInput(tag="in_reset", input_type=Boolean(optional=True), prefix="--reset", doc=InputDocumentation(doc="reset histogram after every report")), ToolInput(tag="in_silent_mode", input_type=Boolean(optional=True), prefix="--silent-mode", doc=InputDocumentation(doc="supress TSV output to stderr")), ToolInput(tag="in_stat", input_type=Boolean(optional=True), prefix="--stat", doc=InputDocumentation(doc="print BAM satistics of the input files")), ToolInput(tag="in_tool", input_type=String(optional=True), prefix="--tool", doc=InputDocumentation(doc="invoke toolbox in YAML format (see documentation)")), ToolInput(tag="in_top_bam", input_type=File(optional=True), prefix="--top-bam", doc=InputDocumentation(doc="save the top -? records to this bam file")), ToolInput(tag="in_alphabet_guess_seq_length", input_type=Int(optional=True), prefix="--alphabet-guess-seq-length", doc=InputDocumentation(doc="length of sequence prefix of the first FASTA record based on which seqkit guesses the sequence type (0 for whole seq) (default 10000)")), ToolInput(tag="in_id_ncbi", input_type=Boolean(optional=True), prefix="--id-ncbi", doc=InputDocumentation(doc="FASTA head is NCBI-style, e.g. >gi|110645304|ref|NC_002516.2| Pseud...")), ToolInput(tag="in_id_regexp", input_type=String(optional=True), prefix="--id-regexp", doc=InputDocumentation(doc="regular expression for parsing ID (default '^(\\S+)\\s?')")), ToolInput(tag="in_in_file_list", input_type=File(optional=True), prefix="--infile-list", doc=InputDocumentation(doc="file of input files list (one file per line), if given, they are appended to files from cli arguments")), ToolInput(tag="in_line_width", input_type=Int(optional=True), prefix="--line-width", doc=InputDocumentation(doc="line width when outputing FASTA format (0 for no wrap) (default 60)")), ToolInput(tag="in_out_file", input_type=File(optional=True), prefix="--out-file", doc=InputDocumentation(doc="out file ('-' for stdout, suffix .gz for gzipped out) (default '-')")), ToolInput(tag="in_quiet", input_type=Boolean(optional=True), prefix="--quiet", doc=InputDocumentation(doc="be quiet and do not show extra information")), ToolInput(tag="in_seq_type", input_type=String(optional=True), prefix="--seq-type", doc=InputDocumentation(doc="sequence type (dna|rna|protein|unlimit|auto) (for auto, it automatically detect by the first sequence) (default 'auto')")), ToolInput(tag="in_threads", input_type=Int(optional=True), prefix="--threads", doc=InputDocumentation(doc="number of CPUs. (default value: 1 for single-CPU PC, 2 for others. can also set with environment variable SEQKIT_THREADS) (default 2)"))], outputs=[ToolOutput(tag="out_out_file", output_type=File(optional=True), selector=InputSelector(input_to_select="in_out_file", type_hint=File()), doc=OutputDocumentation(doc="out file ('-' for stdout, suffix .gz for gzipped out) (default '-')"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Seqkit_Bam_V0_1_0().translate("wdl", allow_empty_container=True)

