version 1.0

task AnviSummarize {
  input {
    String? pan_or_profile_db
    String? contigs_db
    String? genomes_storage
    Boolean? in_it_gene_coverages
    Boolean? reformat_contig_names
    Boolean? report_aa_seqs_for_gene_calls
    Boolean? report_dna_sequences
    String? collection_name
    String? output_dir
    Boolean? list_collections
    String? taxonomic_level
    String? cog_data_dir
    Boolean? quick_summary
    Boolean? just_do_it
  }
  command <<<
    anvi-summarize \
      ~{if defined(pan_or_profile_db) then ("--pan-or-profile-db " +  '"' + pan_or_profile_db + '"') else ""} \
      ~{if defined(contigs_db) then ("--contigs-db " +  '"' + contigs_db + '"') else ""} \
      ~{if defined(genomes_storage) then ("--genomes-storage " +  '"' + genomes_storage + '"') else ""} \
      ~{true="--init-gene-coverages" false="" in_it_gene_coverages} \
      ~{true="--reformat-contig-names" false="" reformat_contig_names} \
      ~{true="--report-aa-seqs-for-gene-calls" false="" report_aa_seqs_for_gene_calls} \
      ~{true="--report-DNA-sequences" false="" report_dna_sequences} \
      ~{if defined(collection_name) then ("--collection-name " +  '"' + collection_name + '"') else ""} \
      ~{if defined(output_dir) then ("--output-dir " +  '"' + output_dir + '"') else ""} \
      ~{true="--list-collections" false="" list_collections} \
      ~{if defined(taxonomic_level) then ("--taxonomic-level " +  '"' + taxonomic_level + '"') else ""} \
      ~{if defined(cog_data_dir) then ("--cog-data-dir " +  '"' + cog_data_dir + '"') else ""} \
      ~{true="--quick-summary" false="" quick_summary} \
      ~{true="--just-do-it" false="" just_do_it}
  >>>
  parameter_meta {
    pan_or_profile_db: "Anvi'o pan or profile database (and even genes database in appropriate contexts)."
    contigs_db: "Anvi'o contigs database generated by 'anvi-gen- contigs'"
    genomes_storage: "Anvi'o genomes storage file"
    in_it_gene_coverages: "Initialize gene coverage and detection data. This is a very computationally expensive step, but it is necessary when you need gene level coverage data. The reason this is very computationally expensive is because anvi'o computes gene coverages by going back to actual coverage values of each gene to average them, instead of using contig average coverage values, for extreme accuracy."
    reformat_contig_names: "Reformat contig names while generating the summary output so they look fancy. With this flag, anvi'o will replace the original names of contigs to those that include the bin name as a prefix in resulting summary output files per bin. Use this flag carefully as it may influence your downstream analyses due to the fact that your original contig names in your input FASTA file for the contigs database will not be in the summary output. Although, anvi'o will report a conversion map per bin so you can recover the original contig name if you have to."
    report_aa_seqs_for_gene_calls: "You can use this flag if you would like amino acid AND dna sequences for your gene calls in the genes output file. By default, only dna sequences are reported."
    report_dna_sequences: "By default, this program reports amino acid sequences. Use this flag to report DNA sequences instead. Also note, since gene clusters are aligned via amino acid sequences, using this flag removes alignment information manifesting in the form of gap characters (`-` characters) that would be present if amino acid sequences were reported. Read the warnings during runtime for more detailed information."
    collection_name: "Collection name."
    output_dir: "Directory path for output files"
    list_collections: "Show available collections and exit."
    taxonomic_level: "The taxonomic level to use whenever relevant and/or available. The default taxonomic level is t_genus, but if you choose something specific, anvi'o will focus on that whenever possible."
    cog_data_dir: "The directory path for your COG setup. Anvi'o will try to use the default path if you do not specify anything."
    quick_summary: "When declared the summary output will be generated as quickly as possible, with minimum amount of essential information about bins."
    just_do_it: "Don't bother me with questions or warnings, just do it."
  }
}