from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, File

Ngmerge_V0_1_0 = CommandToolBuilder(tool="NGmerge", base_command=["NGmerge"], inputs=[ToolInput(tag="in_file_file_forward", input_type=Boolean(optional=True), prefix="-1", doc=InputDocumentation(doc="<file>       Input FASTQ file with reads from forward direction")), ToolInput(tag="in_file_file_reverse", input_type=Boolean(optional=True), prefix="-2", doc=InputDocumentation(doc="<file>       Input FASTQ file with reads from reverse direction")), ToolInput(tag="in_file_output_fastq", input_type=File(optional=True), prefix="-o", doc=InputDocumentation(doc="<file>       Output FASTQ file(s):\n- in 'stitch' mode (def.), the file of merged reads\n- in 'adapter-removal' mode (-a), the output files\nwill be <file>_1.fastq and <file>_2.fastq")), ToolInput(tag="in_int_overlap_reads", input_type=Boolean(optional=True), prefix="-m", doc=InputDocumentation(doc="<int>        Minimum overlap of the paired-end reads (def. 20)")), ToolInput(tag="in_float_mismatches_allow", input_type=Boolean(optional=True), prefix="-p", doc=InputDocumentation(doc="<float>      Mismatches to allow in the overlapped region\n(a fraction of the overlap length; def. 0.10)")), ToolInput(tag="in_use_mode_also", input_type=Boolean(optional=True), prefix="-a", doc=InputDocumentation(doc="Use 'adapter-removal' mode (also sets -d option)")), ToolInput(tag="in_option_check_overhangs", input_type=Boolean(optional=True), prefix="-d", doc=InputDocumentation(doc="Option to check for dovetailing (with 3' overhangs)")), ToolInput(tag="in_int_minimum_overlap", input_type=Boolean(optional=True), prefix="-e", doc=InputDocumentation(doc="<int>        Minimum overlap of dovetailed alignments (def. 50)")), ToolInput(tag="in_option_produce_read", input_type=Boolean(optional=True), prefix="-s", doc=InputDocumentation(doc="Option to produce shortest stitched read")), ToolInput(tag="in_file_log_file_stitching", input_type=Boolean(optional=True), prefix="-l", doc=InputDocumentation(doc="<file>       Log file for stitching results of each read pair")), ToolInput(tag="in_file_fastq_files", input_type=File(optional=True), prefix="-f", doc=InputDocumentation(doc="<file>       FASTQ files for reads that failed stitching\n(output as <file>_1.fastq and <file>_2.fastq)")), ToolInput(tag="in_file_log_file_reads", input_type=Boolean(optional=True), prefix="-c", doc=InputDocumentation(doc="<file>       Log file for dovetailed reads (adapter sequences)")), ToolInput(tag="in_file_log_file_formatted", input_type=Boolean(optional=True), prefix="-j", doc=InputDocumentation(doc="<file>       Log file for formatted alignments of merged reads")), ToolInput(tag="in_option_gzip_z", input_type=Boolean(optional=True), prefix="-z", doc=InputDocumentation(doc="Option to gzip (-z) or not (-y) FASTQ output(s)")), ToolInput(tag="in_option_produce_outputs", input_type=Boolean(optional=True), prefix="-i", doc=InputDocumentation(doc="Option to produce interleaved FASTQ output(s)")), ToolInput(tag="in_file_use_given", input_type=Boolean(optional=True), prefix="-w", doc=InputDocumentation(doc="<file>       Use given error profile for merged qual scores")), ToolInput(tag="in_use_method_merged", input_type=Boolean(optional=True), prefix="-g", doc=InputDocumentation(doc="Use 'fastq-join' method for merged qual scores")), ToolInput(tag="in_int_offset_def", input_type=Boolean(optional=True), prefix="-q", doc=InputDocumentation(doc="<int>        FASTQ quality offset (def. 33)")), ToolInput(tag="in_int_score_based", input_type=Boolean(optional=True), prefix="-u", doc=InputDocumentation(doc="<int>        Maximum input quality score (0-based; def. 40)")), ToolInput(tag="in_int_number_use", input_type=Boolean(optional=True), prefix="-n", doc=InputDocumentation(doc="<int>        Number of threads to use (def. 1)")), ToolInput(tag="in_option_print_status", input_type=Boolean(optional=True), prefix="-v", doc=InputDocumentation(doc="Option to print status updates/counts to stderr"))], outputs=[ToolOutput(tag="out_file_output_fastq", output_type=File(optional=True), selector=InputSelector(input_to_select="in_file_output_fastq", type_hint=File()), doc=OutputDocumentation(doc="<file>       Output FASTQ file(s):\n- in 'stitch' mode (def.), the file of merged reads\n- in 'adapter-removal' mode (-a), the output files\nwill be <file>_1.fastq and <file>_2.fastq")), ToolOutput(tag="out_file_fastq_files", output_type=File(optional=True), selector=InputSelector(input_to_select="in_file_fastq_files", type_hint=File()), doc=OutputDocumentation(doc="<file>       FASTQ files for reads that failed stitching\n(output as <file>_1.fastq and <file>_2.fastq)"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Ngmerge_V0_1_0().translate("wdl", allow_empty_container=True)

