from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, Int, String, File

Centrifuge_Class_V0_1_0 = CommandToolBuilder(tool="centrifuge_class", base_command=["centrifuge-class"], inputs=[ToolInput(tag="in_query_input_files_fastq", input_type=Boolean(optional=True), prefix="-q", doc=InputDocumentation(doc="query input files are FASTQ .fq/.fastq (default)")), ToolInput(tag="in_q_seq", input_type=Boolean(optional=True), prefix="--qseq", doc=InputDocumentation(doc="query input files are in Illumina's qseq format")), ToolInput(tag="in_query_input_files_multifasta", input_type=Boolean(optional=True), prefix="-f", doc=InputDocumentation(doc="query input files are (multi-)FASTA .fa/.mfa")), ToolInput(tag="in_query_input_files_raw", input_type=Boolean(optional=True), prefix="-r", doc=InputDocumentation(doc="query input files are raw one-sequence-per-line")), ToolInput(tag="in_m_m_r", input_type=Boolean(optional=True), prefix="-c", doc=InputDocumentation(doc="<m1>, <m2>, <r> are sequences themselves, not files")), ToolInput(tag="in_skip", input_type=Int(optional=True), prefix="--skip", doc=InputDocumentation(doc="skip the first <int> reads/pairs in the input (none)")), ToolInput(tag="in_up_to", input_type=Int(optional=True), prefix="--upto", doc=InputDocumentation(doc="stop after first <int> reads/pairs (no limit)")), ToolInput(tag="in_trim_five", input_type=Int(optional=True), prefix="--trim5", doc=InputDocumentation(doc="trim <int> bases from 5'/left end of reads (0)")), ToolInput(tag="in_trim_three", input_type=Int(optional=True), prefix="--trim3", doc=InputDocumentation(doc="trim <int> bases from 3'/right end of reads (0)")), ToolInput(tag="in_phred_three_three", input_type=Boolean(optional=True), prefix="--phred33", doc=InputDocumentation(doc="qualities are Phred+33 (default)")), ToolInput(tag="in_phred_six_four", input_type=Boolean(optional=True), prefix="--phred64", doc=InputDocumentation(doc="qualities are Phred+64")), ToolInput(tag="in_int_quals", input_type=Boolean(optional=True), prefix="--int-quals", doc=InputDocumentation(doc="qualities encoded as space-delimited integers")), ToolInput(tag="in_ignore_quals", input_type=Boolean(optional=True), prefix="--ignore-quals", doc=InputDocumentation(doc="treat all quality values as 30 on Phred scale (off)")), ToolInput(tag="in_no_fw", input_type=Boolean(optional=True), prefix="--nofw", doc=InputDocumentation(doc="do not align forward (original) version of read (off)")), ToolInput(tag="in_norc", input_type=Boolean(optional=True), prefix="--norc", doc=InputDocumentation(doc="do not align reverse-complement version of read (off)")), ToolInput(tag="in_min_hit_len", input_type=Int(optional=True), prefix="--min-hitlen", doc=InputDocumentation(doc="minimum length of partial hits (default 22, must be greater than 15)")), ToolInput(tag="in_report_upto_assignments", input_type=Int(optional=True), prefix="-k", doc=InputDocumentation(doc="report upto <int> distinct, primary assignments for each read or pair")), ToolInput(tag="in_host_tax_ids", input_type=String(optional=True), prefix="--host-taxids", doc=InputDocumentation(doc="comma-separated list of taxonomic IDs that will be preferred in classification")), ToolInput(tag="in_exclude_tax_ids", input_type=String(optional=True), prefix="--exclude-taxids", doc=InputDocumentation(doc="comma-separated list of taxonomic IDs that will be excluded in classification")), ToolInput(tag="in_out_fmt", input_type=String(optional=True), prefix="--out-fmt", doc=InputDocumentation(doc="define output format, either 'tab' or 'sam' (tab)")), ToolInput(tag="in_tab_fmt_cols", input_type=Int(optional=True), prefix="--tab-fmt-cols", doc=InputDocumentation(doc="columns in tabular format, comma separated\ndefault: readID,seqID,taxID,score,2ndBestScore,hitLength,queryLength,numMatches")), ToolInput(tag="in_time", input_type=Boolean(optional=True), prefix="--time", doc=InputDocumentation(doc="print wall-clock time taken by search phases")), ToolInput(tag="in_quiet", input_type=Boolean(optional=True), prefix="--quiet", doc=InputDocumentation(doc="print nothing to stderr except serious errors")), ToolInput(tag="in_met_file", input_type=File(optional=True), prefix="--met-file", doc=InputDocumentation(doc="send metrics to file at <path> (off)")), ToolInput(tag="in_met_stderr", input_type=Boolean(optional=True), prefix="--met-stderr", doc=InputDocumentation(doc="send metrics to stderr (off)")), ToolInput(tag="in_met", input_type=Int(optional=True), prefix="--met", doc=InputDocumentation(doc="report internal counters & metrics every <int> secs (1)")), ToolInput(tag="in_threads", input_type=Int(optional=True), prefix="--threads", doc=InputDocumentation(doc="number of alignment threads to launch (1)")), ToolInput(tag="in_mm", input_type=Boolean(optional=True), prefix="--mm", doc=InputDocumentation(doc="use memory-mapped I/O for index; many instances can share")), ToolInput(tag="in_qc_filter", input_type=Boolean(optional=True), prefix="--qc-filter", doc=InputDocumentation(doc="filter out reads that are bad according to QSEQ filter")), ToolInput(tag="in_seed", input_type=Int(optional=True), prefix="--seed", doc=InputDocumentation(doc="seed for random number generator (0)")), ToolInput(tag="in_non_deterministic", input_type=String(optional=True), prefix="--non-deterministic", doc=InputDocumentation(doc="rand. gen. arbitrarily instead of using read attributes")), ToolInput(tag="in_cf_idx", input_type=String(), position=0, doc=InputDocumentation(doc="")), ToolInput(tag="in_m_one", input_type=Int(), position=0, doc=InputDocumentation(doc="")), ToolInput(tag="in_m_two", input_type=Int(), position=0, doc=InputDocumentation(doc="")), ToolInput(tag="in_var_34", input_type=String(), position=0, doc=InputDocumentation(doc="")), ToolInput(tag="in_report", input_type=String(), position=0, doc=InputDocumentation(doc="")), ToolInput(tag="in_specified", input_type=String(), position=0, doc=InputDocumentation(doc="")), ToolInput(tag="in_file", input_type=File(), position=1, doc=InputDocumentation(doc="")), ToolInput(tag="in_files", input_type=String(), position=1, doc=InputDocumentation(doc="")), ToolInput(tag="in_index", input_type=String(), position=1, doc=InputDocumentation(doc="")), ToolInput(tag="in_many", input_type=String(), position=1, doc=InputDocumentation(doc="")), ToolInput(tag="in_filename", input_type=File(), position=2, doc=InputDocumentation(doc="")), ToolInput(tag="in_for", input_type=String(), position=2, doc=InputDocumentation(doc="")), ToolInput(tag="in_times_dot", input_type=String(), position=2, doc=InputDocumentation(doc="")), ToolInput(tag="in_with", input_type=String(), position=2, doc=InputDocumentation(doc="")), ToolInput(tag="in_e_dot_gdot", input_type=String(), position=3, doc=InputDocumentation(doc="")), ToolInput(tag="in_classification", input_type=String(), position=3, doc=InputDocumentation(doc="")), ToolInput(tag="in_prefix", input_type=String(), position=3, doc=InputDocumentation(doc="")), ToolInput(tag="in_tabular", input_type=String(), position=3, doc=InputDocumentation(doc="")), ToolInput(tag="in_unpaired", input_type=String(), position=3, doc=InputDocumentation(doc="")), ToolInput(tag="in_output", input_type=String(), position=4, doc=InputDocumentation(doc=""))], outputs=[], container="quay.io/biocontainers/centrifuge-core:1.0.4_beta--he513fc3_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Centrifuge_Class_V0_1_0().translate("wdl")

