from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, Int, String, File

Repeatmasker_V0_1_0 = CommandToolBuilder(tool="RepeatMasker", base_command=["RepeatMasker"], inputs=[ToolInput(tag="in_pa", input_type=Boolean(optional=True), prefix="-pa", doc=InputDocumentation(doc="(rallel) [number]\nThe number of processors to use in parallel (only works for batch\nfiles or sequences over 50 kb)")), ToolInput(tag="in_slow_search_more", input_type=Boolean(optional=True), prefix="-s", doc=InputDocumentation(doc="Slow search; 0-5% more sensitive, 2-3 times slower than default")), ToolInput(tag="in_quick_search_sensitive", input_type=Boolean(optional=True), prefix="-q", doc=InputDocumentation(doc="Quick search; 5-10% less sensitive, 2-5 times faster than default")), ToolInput(tag="in_qq", input_type=Int(optional=True), prefix="-qq", doc=InputDocumentation(doc="job; about 10% less sensitive, 4->10 times faster than default\n(quick searches are fine under most circumstances) repeat options")), ToolInput(tag="in_no_low", input_type=Boolean(optional=True), prefix="-nolow", doc=InputDocumentation(doc="Does not mask low_complexity DNA or simple repeats")), ToolInput(tag="in_no_int", input_type=Boolean(optional=True), prefix="-noint", doc=InputDocumentation(doc="Only masks low complex/simple repeats (no interspersed repeats)")), ToolInput(tag="in_no_rna", input_type=Boolean(optional=True), prefix="-norna", doc=InputDocumentation(doc="Does not mask small RNA (pseudo) genes")), ToolInput(tag="in_alu", input_type=Boolean(optional=True), prefix="-alu", doc=InputDocumentation(doc="Only masks Alus (and 7SLRNA, SVA and LTR5)(only for primate DNA)")), ToolInput(tag="in_div", input_type=Boolean(optional=True), prefix="-div", doc=InputDocumentation(doc="[number]\nMasks only those repeats < x percent diverged from consensus seq")), ToolInput(tag="in_lib", input_type=Boolean(optional=True), prefix="-lib", doc=InputDocumentation(doc="[filename]\nAllows use of a custom library (e.g. from another species)")), ToolInput(tag="in_cut_off", input_type=Boolean(optional=True), prefix="-cutoff", doc=InputDocumentation(doc="[number]\nSets cutoff score for masking repeats when using -lib (default 225)")), ToolInput(tag="in_species", input_type=String(optional=True), prefix="-species", doc=InputDocumentation(doc="Specify the species or clade of the input sequence. The species name\nmust be a valid NCBI Taxonomy Database species name and be contained\nin the RepeatMasker repeat database. Some examples are:\n-species human\n-species mouse\n-species rattus\n-species 'ciona savignyi'\n-species arabidopsis\nOther commonly used species:\nmammal, carnivore, rodentia, rat, cow, pig, cat, dog, chicken, fugu,\ndanio, 'ciona intestinalis' drosophila, anopheles, elegans,\ndiatoaea, artiodactyl, arabidopsis, rice, wheat, and maize")), ToolInput(tag="in_is_only", input_type=Boolean(optional=True), prefix="-is_only", doc=InputDocumentation(doc="Only clips E coli insertion elements out of fasta and .qual files")), ToolInput(tag="in_is_clip", input_type=Boolean(optional=True), prefix="-is_clip", doc=InputDocumentation(doc="Clips IS elements before analysis (default: IS only reported)")), ToolInput(tag="in_no_is", input_type=Boolean(optional=True), prefix="-no_is", doc=InputDocumentation(doc="Skips bacterial insertion element check")), ToolInput(tag="in_gc", input_type=Boolean(optional=True), prefix="-gc", doc=InputDocumentation(doc="[number]\nUse matrices calculated for 'number' percentage background GC level")), ToolInput(tag="in_gc_calc", input_type=Boolean(optional=True), prefix="-gccalc", doc=InputDocumentation(doc="RepeatMasker calculates the GC content even for batch files/small\nseqs")), ToolInput(tag="in_frag", input_type=Boolean(optional=True), prefix="-frag", doc=InputDocumentation(doc="[number]\nMaximum sequence length masked without fragmenting (default 60000,\n300000 for DeCypher)")), ToolInput(tag="in_no_cut", input_type=Boolean(optional=True), prefix="-nocut", doc=InputDocumentation(doc="Skips the steps in which repeats are excised")), ToolInput(tag="in_noisy", input_type=Boolean(optional=True), prefix="-noisy", doc=InputDocumentation(doc="Prints search engine progress report to screen (defaults to .stderr\nfile)")), ToolInput(tag="in_no_post", input_type=Boolean(optional=True), prefix="-nopost", doc=InputDocumentation(doc="Do not postprocess the results of the run ( i.e. call ProcessRepeats\n). NOTE: This options should only be used when ProcessRepeats will\nbe run manually on the results.")), ToolInput(tag="in_dir", input_type=File(optional=True), prefix="-dir", doc=InputDocumentation(doc="[directory name]\nWrites output to this directory (default is query file directory,\n'-dir .' will write to current directory).")), ToolInput(tag="in_lignmentswrites_alignments_file", input_type=File(optional=True), prefix="-a", doc=InputDocumentation(doc="(lignments)\nWrites alignments in .align output file")), ToolInput(tag="in_in_v", input_type=Boolean(optional=True), prefix="-inv", doc=InputDocumentation(doc="Alignments are presented in the orientation of the repeat (with\noption -a)")), ToolInput(tag="in_l_cam_big", input_type=Boolean(optional=True), prefix="-lcambig", doc=InputDocumentation(doc="Outputs ambiguous DNA transposon fragments using a lower case name.\nAll other repeats are listed in upper case. Ambiguous fragments\nmatch multiple repeat elements and can only be called based on\nflanking repeat information.")), ToolInput(tag="in_small", input_type=Boolean(optional=True), prefix="-small", doc=InputDocumentation(doc="Returns complete .masked sequence in lower case")), ToolInput(tag="in_xsmall", input_type=Boolean(optional=True), prefix="-xsmall", doc=InputDocumentation(doc="Returns repetitive regions in lowercase (rest capitals) rather than\nmasked")), ToolInput(tag="in_returns_repetitive_regions_masked", input_type=Boolean(optional=True), prefix="-x", doc=InputDocumentation(doc="Returns repetitive regions masked with Xs rather than Ns")), ToolInput(tag="in_poly", input_type=Boolean(optional=True), prefix="-poly", doc=InputDocumentation(doc="Reports simple repeats that may be polymorphic (in file.poly)")), ToolInput(tag="in_source", input_type=Boolean(optional=True), prefix="-source", doc=InputDocumentation(doc="Includes for each annotation the HSP 'evidence'. Currently this\noption is only available with the '-html' output format listed\nbelow.")), ToolInput(tag="in_html", input_type=File(optional=True), prefix="-html", doc=InputDocumentation(doc="Creates an additional output file in xhtml format.")), ToolInput(tag="in_ace", input_type=File(optional=True), prefix="-ace", doc=InputDocumentation(doc="Creates an additional output file in ACeDB format")), ToolInput(tag="in_gff", input_type=Boolean(optional=True), prefix="-gff", doc=InputDocumentation(doc="Creates an additional Gene Feature Finding format output")), ToolInput(tag="in_creates_additional_file", input_type=Boolean(optional=True), prefix="-u", doc=InputDocumentation(doc="Creates an additional annotation file not processed by")), ToolInput(tag="in_no_id", input_type=Boolean(optional=True), prefix="-no_id", doc=InputDocumentation(doc="Leaves out final column with unique ID for each element (was\ndefault)")), ToolInput(tag="in_xclncalculates_repeat_densities", input_type=Boolean(optional=True), prefix="-e", doc=InputDocumentation(doc="(xcln)\nCalculates repeat densities (in .tbl) excluding runs of >=20 N/Xs in\nthe query")), ToolInput(tag="in_sequence_dot", input_type=String(), position=0, doc=InputDocumentation(doc="-e(ngine) [crossmatch|wublast|abblast|ncbi|hmmer|decypher]\nUse an alternate search engine to the default.")), ToolInput(tag="in_process_repeats", input_type=String(), position=0, doc=InputDocumentation(doc="-xm Creates an additional output file in cross_match format (for"))], outputs=[ToolOutput(tag="out_dir", output_type=File(optional=True), selector=InputSelector(input_to_select="in_dir", type_hint=File()), doc=OutputDocumentation(doc="[directory name]\nWrites output to this directory (default is query file directory,\n'-dir .' will write to current directory).")), ToolOutput(tag="out_lignmentswrites_alignments_file", output_type=File(optional=True), selector=InputSelector(input_to_select="in_lignmentswrites_alignments_file", type_hint=File()), doc=OutputDocumentation(doc="(lignments)\nWrites alignments in .align output file")), ToolOutput(tag="out_html", output_type=File(optional=True), selector=InputSelector(input_to_select="in_html", type_hint=File()), doc=OutputDocumentation(doc="Creates an additional output file in xhtml format.")), ToolOutput(tag="out_ace", output_type=File(optional=True), selector=InputSelector(input_to_select="in_ace", type_hint=File()), doc=OutputDocumentation(doc="Creates an additional output file in ACeDB format"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Repeatmasker_V0_1_0().translate("wdl", allow_empty_container=True)

