from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, File

Diamond_V0_1_0 = CommandToolBuilder(tool="diamond", base_command=["diamond"], inputs=[ToolInput(tag="in_threads", input_type=Boolean(optional=True), prefix="--threads", doc=InputDocumentation(doc="(-p)           number of CPU threads")), ToolInput(tag="in_db", input_type=Boolean(optional=True), prefix="--db", doc=InputDocumentation(doc="(-d)                database file")), ToolInput(tag="in_out", input_type=File(optional=True), prefix="--out", doc=InputDocumentation(doc="(-o)               output file")), ToolInput(tag="in_out_fmt", input_type=Boolean(optional=True), prefix="--outfmt", doc=InputDocumentation(doc="(-f)            output format\n0   = BLAST pairwise\n5   = BLAST XML\n6   = BLAST tabular\n100 = DIAMOND alignment archive (DAA)\n101 = SAM\nValue 6 may be followed by a space-separated list of these keywords:\nqseqid means Query Seq - id\nqlen means Query sequence length\nsseqid means Subject Seq - id\nsallseqid means All subject Seq - id(s), separated by a ';'\nslen means Subject sequence length\nqstart means Start of alignment in query\nqend means End of alignment in query\nsstart means Start of alignment in subject\nsend means End of alignment in subject\nqseq means Aligned part of query sequence\nfull_qseq means Query sequence\nsseq means Aligned part of subject sequence\nfull_sseq means Subject sequence\nevalue means Expect value\nbitscore means Bit score\nscore means Raw score\nlength means Alignment length\npident means Percentage of identical matches\nnident means Number of identical matches\nmismatch means Number of mismatches\npositive means Number of positive - scoring matches\ngapopen means Number of gap openings\ngaps means Total number of gaps\nppos means Percentage of positive - scoring matches\nqframe means Query frame\nbtop means Blast traceback operations(BTOP)\ncigar means CIGAR string\nstaxids means unique Subject Taxonomy ID(s), separated by a ';' (in numerical order)\nsscinames means unique Subject Scientific Name(s), separated by a ';'\nsskingdoms means unique Subject Super Kingdom(s), separated by a ';'\nskingdoms means unique Subject Kingdom(s), separated by a ';'\nsphylums means unique Subject Phylum(s), separated by a ';'\nstitle means Subject Title\nsalltitles means All Subject Title(s), separated by a '<>'\nqcovhsp means Query Coverage Per HSP\nscovhsp means Subject Coverage Per HSP\nqtitle means Query title\nqqual means Query quality values for the aligned part of the query\nfull_qqual means Query quality values\nqstrand means Query strand\nDefault: qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore")), ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="--verbose", doc=InputDocumentation(doc="(-v)           verbose console output")), ToolInput(tag="in_log", input_type=Boolean(optional=True), prefix="--log", doc=InputDocumentation(doc="enable debug log")), ToolInput(tag="in_quiet", input_type=Boolean(optional=True), prefix="--quiet", doc=InputDocumentation(doc="disable console output")), ToolInput(tag="in_header", input_type=Boolean(optional=True), prefix="--header", doc=InputDocumentation(doc="Write header lines to blast tabular format.")), ToolInput(tag="in_in", input_type=Boolean(optional=True), prefix="--in", doc=InputDocumentation(doc="input reference file in FASTA format")), ToolInput(tag="in_taxon_map", input_type=Boolean(optional=True), prefix="--taxonmap", doc=InputDocumentation(doc="protein accession to taxid mapping file")), ToolInput(tag="in_tax_on_nodes", input_type=Boolean(optional=True), prefix="--taxonnodes", doc=InputDocumentation(doc="taxonomy nodes.dmp from NCBI")), ToolInput(tag="in_tax_on_names", input_type=Boolean(optional=True), prefix="--taxonnames", doc=InputDocumentation(doc="taxonomy names.dmp from NCBI")), ToolInput(tag="in_query", input_type=Boolean(optional=True), prefix="--query", doc=InputDocumentation(doc="(-q)             input query file")), ToolInput(tag="in_strand", input_type=Boolean(optional=True), prefix="--strand", doc=InputDocumentation(doc="query strands to search (both/minus/plus)")), ToolInput(tag="in_un", input_type=Boolean(optional=True), prefix="--un", doc=InputDocumentation(doc="file for unaligned queries")), ToolInput(tag="in_al", input_type=Boolean(optional=True), prefix="--al", doc=InputDocumentation(doc="file or aligned queries")), ToolInput(tag="in_un_fmt", input_type=Boolean(optional=True), prefix="--unfmt", doc=InputDocumentation(doc="format of unaligned query file (fasta/fastq)")), ToolInput(tag="in_al_fmt", input_type=Boolean(optional=True), prefix="--alfmt", doc=InputDocumentation(doc="format of aligned query file (fasta/fastq)")), ToolInput(tag="in_unal", input_type=Boolean(optional=True), prefix="--unal", doc=InputDocumentation(doc="report unaligned queries (0=no, 1=yes)")), ToolInput(tag="in_max_target_seqs", input_type=Boolean(optional=True), prefix="--max-target-seqs", doc=InputDocumentation(doc="(-k)   maximum number of target sequences to report alignments for (default=25)")), ToolInput(tag="in_top", input_type=Boolean(optional=True), prefix="--top", doc=InputDocumentation(doc="report alignments within this percentage range of top alignment score (overrides --max-target-seqs)")), ToolInput(tag="in_max_hsps", input_type=Boolean(optional=True), prefix="--max-hsps", doc=InputDocumentation(doc="maximum number of HSPs per subject sequence to report for each query (default=unlimited)")), ToolInput(tag="in_range_culling", input_type=Boolean(optional=True), prefix="--range-culling", doc=InputDocumentation(doc="restrict hit culling to overlapping query ranges")), ToolInput(tag="in_compress", input_type=Boolean(optional=True), prefix="--compress", doc=InputDocumentation(doc="compression for output files (0=none, 1=gzip)")), ToolInput(tag="in_evalue", input_type=Boolean(optional=True), prefix="--evalue", doc=InputDocumentation(doc="(-e)            maximum e-value to report alignments (default=0.001)")), ToolInput(tag="in_min_score", input_type=Boolean(optional=True), prefix="--min-score", doc=InputDocumentation(doc="minimum bit score to report alignments (overrides e-value setting)")), ToolInput(tag="in_id", input_type=Boolean(optional=True), prefix="--id", doc=InputDocumentation(doc="minimum identity% to report an alignment")), ToolInput(tag="in_query_cover", input_type=Boolean(optional=True), prefix="--query-cover", doc=InputDocumentation(doc="minimum query cover% to report an alignment")), ToolInput(tag="in_subject_cover", input_type=Boolean(optional=True), prefix="--subject-cover", doc=InputDocumentation(doc="minimum subject cover% to report an alignment")), ToolInput(tag="in_sensitive", input_type=Boolean(optional=True), prefix="--sensitive", doc=InputDocumentation(doc="enable sensitive mode (default: fast)")), ToolInput(tag="in_more_sensitive", input_type=Boolean(optional=True), prefix="--more-sensitive", doc=InputDocumentation(doc="enable more sensitive mode (default: fast)")), ToolInput(tag="in_block_size", input_type=Boolean(optional=True), prefix="--block-size", doc=InputDocumentation(doc="(-b)        sequence block size in billions of letters (default=2.0)")), ToolInput(tag="in_index_chunks", input_type=Boolean(optional=True), prefix="--index-chunks", doc=InputDocumentation(doc="(-c)      number of chunks for index processing (default=4)")), ToolInput(tag="in_tmpdir", input_type=Boolean(optional=True), prefix="--tmpdir", doc=InputDocumentation(doc="(-t)            directory for temporary files")), ToolInput(tag="in_gap_open", input_type=Boolean(optional=True), prefix="--gapopen", doc=InputDocumentation(doc="gap open penalty")), ToolInput(tag="in_gap_extend", input_type=Boolean(optional=True), prefix="--gapextend", doc=InputDocumentation(doc="gap extension penalty")), ToolInput(tag="in_frameshift", input_type=Boolean(optional=True), prefix="--frameshift", doc=InputDocumentation(doc="(-F)        frame shift penalty (default=disabled)")), ToolInput(tag="in_long_reads", input_type=Boolean(optional=True), prefix="--long-reads", doc=InputDocumentation(doc="short for --range-culling --top 10 -F 15")), ToolInput(tag="in_matrix", input_type=Boolean(optional=True), prefix="--matrix", doc=InputDocumentation(doc="score matrix for protein alignment (default=BLOSUM62)")), ToolInput(tag="in_custom_matrix", input_type=Boolean(optional=True), prefix="--custom-matrix", doc=InputDocumentation(doc="file containing custom scoring matrix")), ToolInput(tag="in_lambda", input_type=Boolean(optional=True), prefix="--lambda", doc=InputDocumentation(doc="lambda parameter for custom matrix")), ToolInput(tag="in_k_parameter_custom", input_type=Boolean(optional=True), prefix="--K", doc=InputDocumentation(doc="K parameter for custom matrix")), ToolInput(tag="in_comp_based_stats", input_type=Boolean(optional=True), prefix="--comp-based-stats", doc=InputDocumentation(doc="enable composition based statistics (0/1=default)")), ToolInput(tag="in_masking", input_type=Boolean(optional=True), prefix="--masking", doc=InputDocumentation(doc="enable masking of low complexity regions (0/1=default)")), ToolInput(tag="in_query_gen_code", input_type=Boolean(optional=True), prefix="--query-gencode", doc=InputDocumentation(doc="genetic code to use to translate query (see user manual)")), ToolInput(tag="in_s_all_titles", input_type=Boolean(optional=True), prefix="--salltitles", doc=InputDocumentation(doc="include full subject titles in DAA file")), ToolInput(tag="in_s_all_seq_id", input_type=Boolean(optional=True), prefix="--sallseqid", doc=InputDocumentation(doc="include all subject ids in DAA file")), ToolInput(tag="in_no_self_hits", input_type=Boolean(optional=True), prefix="--no-self-hits", doc=InputDocumentation(doc="suppress reporting of identical self hits")), ToolInput(tag="in_taxon_list", input_type=Boolean(optional=True), prefix="--taxonlist", doc=InputDocumentation(doc="restrict search to list of taxon ids (comma-separated)")), ToolInput(tag="in_tax_on_exclude", input_type=Boolean(optional=True), prefix="--taxon-exclude", doc=InputDocumentation(doc="exclude list of taxon ids (comma-separated)")), ToolInput(tag="in_algo", input_type=Boolean(optional=True), prefix="--algo", doc=InputDocumentation(doc="Seed search algorithm (0=double-indexed/1=query-indexed)")), ToolInput(tag="in_bin", input_type=Boolean(optional=True), prefix="--bin", doc=InputDocumentation(doc="number of query bins for seed search")), ToolInput(tag="in_min_orf", input_type=Boolean(optional=True), prefix="--min-orf", doc=InputDocumentation(doc="(-l)           ignore translated sequences without an open reading frame of at least this length")), ToolInput(tag="in_freq_sd", input_type=Boolean(optional=True), prefix="--freq-sd", doc=InputDocumentation(doc="number of standard deviations for ignoring frequent seeds")), ToolInput(tag="in_id_two", input_type=Boolean(optional=True), prefix="--id2", doc=InputDocumentation(doc="minimum number of identities for stage 1 hit")), ToolInput(tag="in_x_drop", input_type=Boolean(optional=True), prefix="--xdrop", doc=InputDocumentation(doc="(-x)             xdrop for ungapped alignment")), ToolInput(tag="in_band", input_type=Boolean(optional=True), prefix="--band", doc=InputDocumentation(doc="band for dynamic programming computation")), ToolInput(tag="in_shapes", input_type=Boolean(optional=True), prefix="--shapes", doc=InputDocumentation(doc="(-s)            number of seed shapes (0 = all available)")), ToolInput(tag="in_shape_mask", input_type=Boolean(optional=True), prefix="--shape-mask", doc=InputDocumentation(doc="seed shapes")), ToolInput(tag="in_rank_ratio", input_type=Boolean(optional=True), prefix="--rank-ratio", doc=InputDocumentation(doc="include subjects within this ratio of last hit")), ToolInput(tag="in_culling_overlap", input_type=Boolean(optional=True), prefix="--culling-overlap", doc=InputDocumentation(doc="minimum range overlap with higher scoring hit to delete a hit (default=50%)")), ToolInput(tag="in_tax_on_k", input_type=Boolean(optional=True), prefix="--taxon-k", doc=InputDocumentation(doc="maximum number of targets to report per species")), ToolInput(tag="in_range_cover", input_type=Boolean(optional=True), prefix="--range-cover", doc=InputDocumentation(doc="percentage of query range to be covered for range culling (default=50%)")), ToolInput(tag="in_db_size", input_type=Boolean(optional=True), prefix="--dbsize", doc=InputDocumentation(doc="effective database size (in letters)")), ToolInput(tag="in_no_auto_append", input_type=Boolean(optional=True), prefix="--no-auto-append", doc=InputDocumentation(doc="disable auto appending of DAA and DMND file extensions")), ToolInput(tag="in_xml_b_lord_format", input_type=Boolean(optional=True), prefix="--xml-blord-format", doc=InputDocumentation(doc="Use gnl|BL_ORD_ID| style format in XML output")), ToolInput(tag="in_stop_match_score", input_type=Boolean(optional=True), prefix="--stop-match-score", doc=InputDocumentation(doc="Set the match score of stop codons against each other.")), ToolInput(tag="in_tant_an_min_mask_prob", input_type=Boolean(optional=True), prefix="--tantan-minMaskProb", doc=InputDocumentation(doc="minimum repeat probability for masking (default=0.9)")), ToolInput(tag="in_file_buffer_size", input_type=Boolean(optional=True), prefix="--file-buffer-size", doc=InputDocumentation(doc="file buffer size in bytes (default=67108864)")), ToolInput(tag="in_da_a", input_type=Boolean(optional=True), prefix="--daa", doc=InputDocumentation(doc="(-a)               DIAMOND alignment archive (DAA) file")), ToolInput(tag="in_forward_only", input_type=Boolean(optional=True), prefix="--forwardonly", doc=InputDocumentation(doc="only show alignments of forward strand")), ToolInput(tag="in_seq", input_type=Boolean(optional=True), prefix="--seq", doc=InputDocumentation(doc="Sequence numbers to display."))], outputs=[ToolOutput(tag="out_out", output_type=File(optional=True), selector=InputSelector(input_to_select="in_out", type_hint=File()), doc=OutputDocumentation(doc="(-o)               output file"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Diamond_V0_1_0().translate("wdl", allow_empty_container=True)

