from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import String, Int, File, Boolean, Directory, Float

Telescope_Assign_V0_1_0 = CommandToolBuilder(tool="telescope_assign", base_command=["telescope", "assign"], inputs=[ToolInput(tag="in_attribute", input_type=String(optional=True), prefix="--attribute", doc=InputDocumentation(doc="GTF attribute that defines a transposable element\nlocus. GTF features that share the same value for\n--attribute will be considered as part of the same\nlocus. (default: locus)")), ToolInput(tag="in_no_feature_key", input_type=String(optional=True), prefix="--no_feature_key", doc=InputDocumentation(doc="Used internally to represent alignments. Must be\ndifferent from all other feature names. (default:\n__no_feature)")), ToolInput(tag="in_n_cpu", input_type=Int(optional=True), prefix="--ncpu", doc=InputDocumentation(doc="Number of cores to use. (Multiple cores not supported\nyet). (default: 1)")), ToolInput(tag="in_tempdir", input_type=File(optional=True), prefix="--tempdir", doc=InputDocumentation(doc="Path to temporary directory. Temporary files will be\nstored here. Default uses python tempfile package to\ncreate the temporary directory. (default: None)")), ToolInput(tag="in_quiet", input_type=Boolean(optional=True), prefix="--quiet", doc=InputDocumentation(doc="Silence (most) output. (default: False)")), ToolInput(tag="in_debug", input_type=Boolean(optional=True), prefix="--debug", doc=InputDocumentation(doc="Print debug messages. (default: False)")), ToolInput(tag="in_log_file", input_type=File(optional=True), prefix="--logfile", doc=InputDocumentation(doc="Log output to this file. (default: None)")), ToolInput(tag="in_outdir", input_type=Directory(optional=True), prefix="--outdir", doc=InputDocumentation(doc="Output directory. (default: .)")), ToolInput(tag="in_exp_tag", input_type=String(optional=True), prefix="--exp_tag", doc=InputDocumentation(doc="Experiment tag (default: telescope)")), ToolInput(tag="in_updated_sam", input_type=Boolean(optional=True), prefix="--updated_sam", doc=InputDocumentation(doc="Generate an updated alignment file. (default: False)")), ToolInput(tag="in_reassign_mode", input_type=String(optional=True), prefix="--reassign_mode", doc=InputDocumentation(doc="Reassignment mode. After EM is complete, each fragment\nis reassigned according to the expected value of its\nmembership weights. The reassignment method is the\nmethod for resolving the 'best' reassignment for\nfragments that have multiple possible reassignments.\nAvailable modes are: 'exclude' - fragments with\nmultiple best assignments are excluded from the final\ncounts; 'choose' - the best assignment is randomly\nchosen from among the set of best assignments;\n'average' - the fragment is divided evenly among the\nbest assignments; 'conf' - only assignments that\nexceed a certain threshold (see --conf_prob) are\naccepted; 'unique' - only uniquely aligned reads are\nincluded. NOTE: Results using all assignment modes are\nincluded in the Telescope report by default. This\nargument determines what mode will be used for the\n'final counts' column. (default: exclude)")), ToolInput(tag="in_conf_prob", input_type=Float(optional=True), prefix="--conf_prob", doc=InputDocumentation(doc="Minimum probability for high confidence assignment.\n(default: 0.9)")), ToolInput(tag="in_overlap_mode", input_type=String(optional=True), prefix="--overlap_mode", doc=InputDocumentation(doc="Overlap mode. The method used to determine whether a\nfragment overlaps feature. (default: threshold)")), ToolInput(tag="in_overlap_threshold", input_type=Float(optional=True), prefix="--overlap_threshold", doc=InputDocumentation(doc="Fraction of fragment that must be contained within a\nfeature to be assigned to that locus. Ignored if\n--overlap_method is not 'threshold'. (default: 0.2)")), ToolInput(tag="in_annotation_class", input_type=String(optional=True), prefix="--annotation_class", doc=InputDocumentation(doc="Annotation class to use for finding overlaps. Both\nhtseq and intervaltree appear to yield identical\nresults. Performance differences are TBD. (default:\nintervaltree)")), ToolInput(tag="in_pi_prior", input_type=Int(optional=True), prefix="--pi_prior", doc=InputDocumentation(doc="Prior on π. Equivalent to adding n unique reads.\n(default: 0)")), ToolInput(tag="in_theta_prior", input_type=Int(optional=True), prefix="--theta_prior", doc=InputDocumentation(doc="Prior on θ. Equivalent to adding n non-unique reads.\nNOTE: It is recommended to set this prior to a large\nvalue. This increases the penalty for non-unique reads\nand improves accuracy. (default: 200000)")), ToolInput(tag="in_em_epsilon", input_type=Float(optional=True), prefix="--em_epsilon", doc=InputDocumentation(doc="EM Algorithm Epsilon cutoff (default: 1e-7)")), ToolInput(tag="in_max_iter", input_type=Int(optional=True), prefix="--max_iter", doc=InputDocumentation(doc="EM Algorithm maximum iterations (default: 100)")), ToolInput(tag="in_use_likelihood", input_type=Boolean(optional=True), prefix="--use_likelihood", doc=InputDocumentation(doc="Use difference in log-likelihood as convergence\ncriteria. (default: False)")), ToolInput(tag="in_skip_em", input_type=Boolean(optional=True), prefix="--skip_em", doc=InputDocumentation(doc="Exits after loading alignment and saving checkpoint\nfile. (default: False)\n")), ToolInput(tag="in_sam_file", input_type=String(), position=0, doc=InputDocumentation(doc="Path to alignment file. Alignment file can be in SAM\nor BAM format. File must be collated so that all\nalignments for a read pair appear sequentially in the\nfile.")), ToolInput(tag="in_gtf_file", input_type=String(), position=1, doc=InputDocumentation(doc="Path to annotation file (GTF format)"))], outputs=[ToolOutput(tag="out_log_file", output_type=File(optional=True), selector=InputSelector(input_to_select="in_log_file", type_hint=File()), doc=OutputDocumentation(doc="Log output to this file. (default: None)")), ToolOutput(tag="out_outdir", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_outdir", type_hint=File()), doc=OutputDocumentation(doc="Output directory. (default: .)"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Telescope_Assign_V0_1_0().translate("wdl", allow_empty_container=True)

