from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, String

Pgdspider2_Cli_V0_1_0 = CommandToolBuilder(tool="PGDSpider2_cli", base_command=["PGDSpider2-cli"], inputs=[ToolInput(tag="in_input_file", input_type=File(optional=True), prefix="-inputfile", doc=InputDocumentation(doc="The input file for the conversion (mandatory).")), ToolInput(tag="in_input_format", input_type=File(optional=True), prefix="-inputformat", doc=InputDocumentation(doc="Format of the input file. Mandatory if not defined\nin the answer (SPID) file.\nPossible input formats:\nPGD, ARLEQUIN, BAM, BAPS, BATWING, BCF, CONVERT, EIGENSOFT, FASTA,\nFASTQ, FSTAT, GDA, GENELAND, GENEPOP, GENETIX, HGDP, HGDP_CEPH,\nIMMANC, IM, IMA2, MAF, MEGA, MIGRATE, MSA, NEWHYBRIDS, NEXUS,\nONESAMP, PED, PHYLIP, SAM, STRUCTURE, VCF, XMFA")), ToolInput(tag="in_output_file", input_type=File(optional=True), prefix="-outputfile", doc=InputDocumentation(doc="The output file for the conversion (mandatory).")), ToolInput(tag="in_output_format", input_type=File(optional=True), prefix="-outputformat", doc=InputDocumentation(doc="Format of the output file. Mandatory if not defined\nin the answer (SPID) file.\nPossible output formats:\nPGD, ARLEQUIN, BAM, BAMOVA, BAPS, BATWING, BAYENV, BCF, EIGENSOFT,\nFASTA, FASTQ, FDIST2, FSTAT, GDA, GENELAND, GENEPOP,\nGENETIX,GESTE_BAYE_SCAN, IMMANC, IM, IMA2, KML, MEGA, MIGRATE, MSA,\nMSVAR, NEWHYBRIDS, NEXUS, ONESAMP, PED, PHYLIP, SAM, STRUCTURAMA,\nSTRUCTURE, VCF, XMFA")), ToolInput(tag="in_spid", input_type=File(optional=True), prefix="-spid", doc=InputDocumentation(doc="SPID file containing preanswered conversion questions\n(mandatory).")), ToolInput(tag="in_pgd_spider_cli", input_type=String(), position=0, doc=InputDocumentation(doc="")), ToolInput(tag="in_options_dot_dot_dot", input_type=String(), position=1, doc=InputDocumentation(doc=""))], outputs=[ToolOutput(tag="out_output_file", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output_file", type_hint=File()), doc=OutputDocumentation(doc="The output file for the conversion (mandatory).")), ToolOutput(tag="out_output_format", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output_format", type_hint=File()), doc=OutputDocumentation(doc="Format of the output file. Mandatory if not defined\nin the answer (SPID) file.\nPossible output formats:\nPGD, ARLEQUIN, BAM, BAMOVA, BAPS, BATWING, BAYENV, BCF, EIGENSOFT,\nFASTA, FASTQ, FDIST2, FSTAT, GDA, GENELAND, GENEPOP,\nGENETIX,GESTE_BAYE_SCAN, IMMANC, IM, IMA2, KML, MEGA, MIGRATE, MSA,\nMSVAR, NEWHYBRIDS, NEXUS, ONESAMP, PED, PHYLIP, SAM, STRUCTURAMA,\nSTRUCTURE, VCF, XMFA"))], container="quay.io/biocontainers/pgdspider:2.1.1.5--0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Pgdspider2_Cli_V0_1_0().translate("wdl")

