from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean

Sortgrcd_V0_1_0 = CommandToolBuilder(tool="sortgrcd", base_command=["sortgrcd"], inputs=[ToolInput(tag="in_cn", input_type=Boolean(optional=True), prefix="-CN", doc=InputDocumentation(doc=":    Minimum % of coverage (0-100)")), ToolInput(tag="in_en", input_type=Boolean(optional=True), prefix="-EN", doc=InputDocumentation(doc=":    Report only the best (N=1) or all (N=2) results per gene locus (1)")), ToolInput(tag="in_fn", input_type=Boolean(optional=True), prefix="-FN", doc=InputDocumentation(doc=":    Filter Level (0 -> 3: no -> stringent)")), ToolInput(tag="in_gn", input_type=Boolean(optional=True), prefix="-GN", doc=InputDocumentation(doc=":    Genetic code (0: universal)")), ToolInput(tag="in_hn", input_type=Boolean(optional=True), prefix="-HN", doc=InputDocumentation(doc=":    Minimum spaln score")), ToolInput(tag="in_jn", input_type=Boolean(optional=True), prefix="-JN", doc=InputDocumentation(doc=":    Minimum ORF length (300)")), ToolInput(tag="in_maximum_total_number_missmatches", input_type=Boolean(optional=True), prefix="-MN", doc=InputDocumentation(doc=":    Maximum total number of missmatches")), ToolInput(tag="in_maximum_total_number_noncanonical", input_type=Boolean(optional=True), prefix="-NN", doc=InputDocumentation(doc=":    Maximum total number of non-canonical boundaries")), ToolInput(tag="in_on", input_type=Boolean(optional=True), prefix="-ON", doc=InputDocumentation(doc=":    Output format. 0:Gff3, 3:BED, 4:Native, 5:Intron,\n6:cDNA, 7:translate, 8:CDS, 15:unique intron")), ToolInput(tag="in_pn", input_type=Boolean(optional=True), prefix="-PN", doc=InputDocumentation(doc=":    Minimum Overall % identity (0-100)")), ToolInput(tag="in_maximum_total_number_unpaired", input_type=Boolean(optional=True), prefix="-UN", doc=InputDocumentation(doc=":    Maximum total number of unpaired bases in gaps")), ToolInput(tag="in_ln", input_type=Boolean(optional=True), prefix="-lN", doc=InputDocumentation(doc=":    Number of residues per line for -O6 or -O7 (60)")), ToolInput(tag="in_maximum_allowed_missmatches", input_type=Boolean(optional=True), prefix="-mN", doc=InputDocumentation(doc=":    Maximum allowed missmatches at both exon boundaries")), ToolInput(tag="in_allow_noncanonical_boundary", input_type=Boolean(optional=True), prefix="-nN", doc=InputDocumentation(doc=":    allow non-canonical boundary? [0: no; 1: AT-AN; 2: 1bp mismatch; 3: any]")), ToolInput(tag="in_maximum_allowed_bases", input_type=Boolean(optional=True), prefix="-uN", doc=InputDocumentation(doc=":    Maximum allowed unpaired bases in gaps at both exon boundaries")), ToolInput(tag="in_gs", input_type=Boolean(optional=True), prefix="-gS", doc=InputDocumentation(doc=":    Specify the .grp file name")), ToolInput(tag="in_sa", input_type=Boolean(optional=True), prefix="-Sa", doc=InputDocumentation(doc=":    sort chromosomes in the alphabetical order of identifier (default)")), ToolInput(tag="in_sb", input_type=Boolean(optional=True), prefix="-Sb", doc=InputDocumentation(doc=":    sort chromosomes in the order of abundance mapped on them")), ToolInput(tag="in_sc", input_type=Boolean(optional=True), prefix="-Sc", doc=InputDocumentation(doc=":    sort chromosomes in the order of apparence in the genome db")), ToolInput(tag="in_sr", input_type=Boolean(optional=True), prefix="-Sr", doc=InputDocumentation(doc=":    sort records mapped on minus strand in the reverse order of genomic positions"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Sortgrcd_V0_1_0().translate("wdl", allow_empty_container=True)

