from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Int, File, Boolean, String, Directory

Mimseq_V0_1_0 = CommandToolBuilder(tool="mimseq", base_command=["mimseq"], inputs=[ToolInput(tag="in_species", input_type=Int(optional=True), prefix="--species", doc=InputDocumentation(doc="Species being analyzed for which to load pre-packaged\ndata files (prioritized over -t, -o and -m). Options\nare: Hsap, Hsap38, Mmus, Scer, Spom, Dmel, Drer, Ecol")), ToolInput(tag="in_trnas_trnas_genomic", input_type=File(optional=True), prefix="-t", doc=InputDocumentation(doc="tRNAs, --trnas genomic tRNAs\nGenomic tRNA fasta file, e.g. from gtRNAdb or\ntRNAscan-SE. Already avalable in data folder for a few\nmodel organisms.")), ToolInput(tag="in_file__trnaout", input_type=File(optional=True), prefix="-o", doc=InputDocumentation(doc="out file, --trnaout tRNA out file\ntRNA.out file generated by tRNAscan-SE (also may be\navailable on gtRNAdb). Contains information about tRNA\nfeatures, including introns.")), ToolInput(tag="in_trnas__mitotrnas", input_type=File(optional=True), prefix="-m", doc=InputDocumentation(doc="tRNAs, --mito-trnas mitochondrial tRNAs\nMitochondrial tRNA fasta file. Should be downloaded\nfrom mitotRNAdb for species of interest. Already\navailable in data folder for a few model organisms.")), ToolInput(tag="in_pre_trnas", input_type=Boolean(optional=True), prefix="--pretRNAs", doc=InputDocumentation(doc="Input reference sequences are pretRNAs. Enabling this\noption will disable the removal of intron sequences\nand addition of 3'-CCA to generate mature tRNA\nsequences. Useful for mapping and discovering pretRNA\nsequence reads.")), ToolInput(tag="in_no_cluster", input_type=Boolean(optional=True), prefix="--no-cluster", doc=InputDocumentation(doc="Disable usearch sequence clustering of tRNAs by\nisodecoder which drastically reduces the rate of\nmulti-mapping reads. Default is enabled.")), ToolInput(tag="in_cluster_id", input_type=Boolean(optional=True), prefix="--cluster-id", doc=InputDocumentation(doc="[clustering identity threshold]\nIdentity cutoff for usearch clustering between 0 and\n1. Default is 0.97.")), ToolInput(tag="in_de_conv_cov_ratio", input_type=Boolean(optional=True), prefix="--deconv-cov-ratio", doc=InputDocumentation(doc="[deconvolution coverage threshold]\nThreshold for ratio between coverage at 3' end and\nmismatch used for deconvolution. Coverage reductions\ngreater than the threshold will result in non-\ndeconvoluted sequences. Default is 0.5 (i.e. less than\n50% reduction required for deconvolution).")), ToolInput(tag="in_threads", input_type=Int(optional=True), prefix="--threads", doc=InputDocumentation(doc="number\nSet processor threads to use during read alignment and\nread counting.")), ToolInput(tag="in_post_trans_mod_off", input_type=Boolean(optional=True), prefix="--posttrans-mod-off", doc=InputDocumentation(doc="Disable post-transcriptional modification of tRNAs,\ni.e. addition of 3'-CCA and 5'-G (His) to mature\nsequences. Disable for certain prokaryotes (e.g. E.\ncoli) where this is genomically encoded. Leave enabled\n(default) for all eukaryotes.")), ToolInput(tag="in_control_condition", input_type=String(optional=True), prefix="--control-condition", doc=InputDocumentation(doc="condition\nName of control/wild-type condition as per user\ndefined group specified in sample data input. This\nmust exactly match the group name specified in sample\ndata. This is used for differential expression\nanalysis so that results are always in the form\nmutant/treatment vs WT/control. REQUIRED")), ToolInput(tag="in_no_cca_analysis", input_type=Boolean(optional=True), prefix="--no-cca-analysis", doc=InputDocumentation(doc="Disable analysis of 3'-CCA ends. When enabled, this\ncalculates proportions of CC vs CCA ending reads per\ncluster and performs DESeq2 analysis. Useful for\ncomparing functional to non-functional mature tRNAs.\nDefault is enabled.")), ToolInput(tag="in_double_cca", input_type=Boolean(optional=True), prefix="--double-cca", doc=InputDocumentation(doc="Enable analysis of 3'-CCACCA tagging for tRNA\ndegradation pathway. Note that this will alter the\noutput of the CCA analysis pipeline.")), ToolInput(tag="in_local_modo_mics", input_type=Boolean(optional=True), prefix="--local-modomics", doc=InputDocumentation(doc="Disable retrieval of Modomics data from online.\nInstead use older locally stored data. Warning - this\nleads to usage of older Modomics data!")), ToolInput(tag="in_max_mismatches", input_type=Int(optional=True), prefix="--max-mismatches", doc=InputDocumentation(doc="mismatches\nMaximum mismatches allowed. If specified between 0.0\nand 1.0, then treated as a fraction of read length.\nOtherwise, treated as integer number of mismatches.\nDefault is an automatic ultrafast value calculated by\nGSNAP; see GSNAP help for more info.")), ToolInput(tag="in_remap_mismatches", input_type=Int(optional=True), prefix="--remap-mismatches", doc=InputDocumentation(doc="mismatches for remap\nMaximum number of mismatches allowed during remapping\nof all reads. Treated similarly to --max-mismatches.\nThis is important to control misalignment of reads to\nsimilar clusters/tRNAs Note that the SNP index will be\nupdated with new SNPs from the first round of\nalignment and so this should be relatively small to\nprohibit misalignment.")), ToolInput(tag="in_no_snp_tolerance", input_type=Boolean(optional=True), prefix="--no-snp-tolerance", doc=InputDocumentation(doc="Disable GSNAP SNP-tolerant read alignment, where known\nmodifications from Modomics are mapped as SNPs.\nDefault is enabled.")), ToolInput(tag="in_name__experiment", input_type=String(optional=True), prefix="-n", doc=InputDocumentation(doc="name, --name experiment name\nName of experiment. Note, output files and indices\nwill have this as a prefix. REQUIRED")), ToolInput(tag="in_out_dir", input_type=Directory(optional=True), prefix="--out-dir", doc=InputDocumentation(doc="directory\nOutput directory. Default is current directory. Cannot\nbe an existing directory.")), ToolInput(tag="in_keep_temp", input_type=Boolean(optional=True), prefix="--keep-temp", doc=InputDocumentation(doc="Keeps multi-mapping and unmapped bam files from GSNAP\nalignments. Default is false.")), ToolInput(tag="in_min_cov", input_type=Int(optional=True), prefix="--min-cov", doc=InputDocumentation(doc="coverage per cluster\nMinimum coverage per cluster required to include this\ncluster in coverage plots, modification analysis, and\n3'-CCA analysis. Can be a fraction of total mapped\nreads between 0 and 1, or an integer of absolute\ncoverage. Any cluster not meeting the threshold in 1\nor more sample will be excluded. Note that all\nclusters are included for differential expression\nanalysis with DESeq2. Default = 0.0005 (0.05% mapped\nreads).")), ToolInput(tag="in_max_multi", input_type=Int(optional=True), prefix="--max-multi", doc=InputDocumentation(doc="coverage multithreading\nMaximum number of bam files to run bedtools coverage\non simultaneously. Increasing this number reduces\nprocessing time by increasing number of files\nprocessed simultaneously. However, depending on the\nsize of the bam files to process and available memory,\ntoo many files processed at once can cause termination\nof mim-tRNAseq due to insufficient memory. If mim-\ntRNAseq fails during coverage calculation, lower this\nnumber. Increase at your own discretion. Default is 3.")), ToolInput(tag="in_remap", input_type=Boolean(optional=True), prefix="--remap", doc=InputDocumentation(doc="Enable detection of unannotated (potential)\nmodifications from misincorporation data. These are\ndefined as having a total misincorporation rate higher\nthan the threshold set with --misinc-thresh. These\nmodifications are then appended to already known ones,\nand read alignment is reperformed. Very useful for\npoorly annotated species in Modomics. Due to\nrealignment and misincorporation parsing, enabling\nthis option slows the analysis down considerably.")), ToolInput(tag="in_mis_inc_thresh", input_type=Boolean(optional=True), prefix="--misinc-thresh", doc=InputDocumentation(doc="[threshold for unannotated mods]\nThreshold of total misincorporation rate at a position\nin a cluster used to call unannotated modifications.\nValue between 0 and 1, default is 0.1 (10%\nmisincorporation).\n")), ToolInput(tag="in_sample_data", input_type=String(), position=0, doc=InputDocumentation(doc="Sample data sheet in text format, tab-separated.\nColumn 1: full path to fastq (or fastq.gz). Column 2:\ncondition/group."))], outputs=[ToolOutput(tag="out_file__trnaout", output_type=File(optional=True), selector=InputSelector(input_to_select="in_file__trnaout", type_hint=File()), doc=OutputDocumentation(doc="out file, --trnaout tRNA out file\ntRNA.out file generated by tRNAscan-SE (also may be\navailable on gtRNAdb). Contains information about tRNA\nfeatures, including introns.")), ToolOutput(tag="out_out_dir", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_out_dir", type_hint=File()), doc=OutputDocumentation(doc="directory\nOutput directory. Default is current directory. Cannot\nbe an existing directory."))], container="quay.io/biocontainers/mimseq:0.3.3.2--pyhdfd78af_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Mimseq_V0_1_0().translate("wdl")

