from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, String

Blast2Rma_V0_1_0 = CommandToolBuilder(tool="blast2rma", base_command=["blast2rma"], inputs=[ToolInput(tag="in_format", input_type=Boolean(optional=True), prefix="--format", doc=InputDocumentation(doc="[string]                Input file format. Mandatory option. Legal values: Unknown, DAA, BlastText, BlastXML, BlastTab, LastMAF, RapSearch2Aln, IlluminaReporter, RDPAssignmentDetails, RDPStandalone, Mothur, SAM, References_as_FastA")), ToolInput(tag="in_blast_mode", input_type=Boolean(optional=True), prefix="--blastMode", doc=InputDocumentation(doc="[string]            Blast mode. Default value: Unknown. Legal values: Unknown, BlastN, BlastP, BlastX, Classifier")), ToolInput(tag="in_reads_files_fasta", input_type=Boolean(optional=True), prefix="--reads", doc=InputDocumentation(doc="[string(s)]              Reads file(s) (fasta or fastq, gzipped ok).")), ToolInput(tag="in_metadata_file", input_type=Boolean(optional=True), prefix="--metaDataFile", doc=InputDocumentation(doc="[string(s)]     Files containing metadata to be included in RMA6 files.")), ToolInput(tag="in_use_compression", input_type=Boolean(optional=True), prefix="--useCompression", doc=InputDocumentation(doc="Compress reads and matches in RMA file (smaller files, longer to generate. Default value: true.")), ToolInput(tag="in_paired_suffix_length", input_type=Boolean(optional=True), prefix="--pairedSuffixLength", doc=InputDocumentation(doc="[number]   Length of name suffix used to distinguish between name of read and its mate. Default value: 0.")), ToolInput(tag="in_paired_reads_in_one_file", input_type=Boolean(optional=True), prefix="--pairedReadsInOneFile", doc=InputDocumentation(doc="Are paired reads in one file (usually they are in two). Default value: false.")), ToolInput(tag="in_max_matches_per_read", input_type=Boolean(optional=True), prefix="--maxMatchesPerRead", doc=InputDocumentation(doc="[number]     Max matches per read. Default value: 100.")), ToolInput(tag="in_classify", input_type=Boolean(optional=True), prefix="--classify", doc=InputDocumentation(doc="Run classification algorithm. Default value: true.")), ToolInput(tag="in_min_score", input_type=Boolean(optional=True), prefix="--minScore", doc=InputDocumentation(doc="[number]             Min score. Default value: 50.0.")), ToolInput(tag="in_max_expected", input_type=Boolean(optional=True), prefix="--maxExpected", doc=InputDocumentation(doc="[number]          Max expected. Default value: 0.01.")), ToolInput(tag="in_min_percent_identity", input_type=Boolean(optional=True), prefix="--minPercentIdentity", doc=InputDocumentation(doc="[number]   Min percent identity. Default value: 0.0.")), ToolInput(tag="in_top_percent", input_type=Boolean(optional=True), prefix="--topPercent", doc=InputDocumentation(doc="[number]          Top percent. Default value: 10.0.")), ToolInput(tag="in_min_support_percent", input_type=Boolean(optional=True), prefix="--minSupportPercent", doc=InputDocumentation(doc="[number]   Min support as percent of assigned reads (0==off). Default value: 0.05.")), ToolInput(tag="in_min_support", input_type=Boolean(optional=True), prefix="--minSupport", doc=InputDocumentation(doc="[number]          Min support (0==off). Default value: 0.")), ToolInput(tag="in_min_percent_read_cover", input_type=Boolean(optional=True), prefix="--minPercentReadCover", doc=InputDocumentation(doc="[number]   Min percent of read length to be covered by alignments. Default value: 0.0.")), ToolInput(tag="in_min_percent_reference_cover", input_type=Boolean(optional=True), prefix="--minPercentReferenceCover", doc=InputDocumentation(doc="[number]   Min percent of reference length to be covered by alignments. Default value: 0.0.")), ToolInput(tag="in_lca_algorithm", input_type=Boolean(optional=True), prefix="--lcaAlgorithm", doc=InputDocumentation(doc="[string]        Set the LCA algorithm to use for taxonomic assignment. Default value: naive. Legal values: naive, weighted, longReads")), ToolInput(tag="in_lca_coverage_percent", input_type=Boolean(optional=True), prefix="--lcaCoveragePercent", doc=InputDocumentation(doc="[number]   Set the percent for the LCA to cover. Default value: 100.0.")), ToolInput(tag="in_read_assignment_mode", input_type=Boolean(optional=True), prefix="--readAssignmentMode", doc=InputDocumentation(doc="[string]   Set the read assignment mode. Default value: readCount. Legal values: readCount, readLength, alignedBases, readMagnitude")), ToolInput(tag="in_con_file", input_type=Boolean(optional=True), prefix="--conFile", doc=InputDocumentation(doc="[string]              File of contaminant taxa (one Id or name per line).")), ToolInput(tag="in_parse_tax_on_names", input_type=Boolean(optional=True), prefix="--parseTaxonNames", doc=InputDocumentation(doc="Parse taxon names. Default value: true.")), ToolInput(tag="in_gi_two_tax_a", input_type=Boolean(optional=True), prefix="--gi2taxa", doc=InputDocumentation(doc="[string]             GI-to-Taxonomy mapping file.")), ToolInput(tag="in_acc_two_tax_a", input_type=Boolean(optional=True), prefix="--acc2taxa", doc=InputDocumentation(doc="[string]            Accessopm-to-Taxonomy mapping file.")), ToolInput(tag="in_syn_two_tax_a", input_type=Boolean(optional=True), prefix="--syn2taxa", doc=InputDocumentation(doc="[string]            Synonyms-to-Taxonomy mapping file.")), ToolInput(tag="in_gi_two_eggnog", input_type=Boolean(optional=True), prefix="--gi2eggnog", doc=InputDocumentation(doc="[string]      GI-to-EGGNOG mapping file.")), ToolInput(tag="in_acc_two_eggnog", input_type=Boolean(optional=True), prefix="--acc2eggnog", doc=InputDocumentation(doc="[string]     Accession-to-EGGNOG mapping file.")), ToolInput(tag="in_syn_two_eggnog", input_type=Boolean(optional=True), prefix="--syn2eggnog", doc=InputDocumentation(doc="[string]     Synonyms-to-EGGNOG mapping file.")), ToolInput(tag="in_tags_four_eggnog", input_type=Boolean(optional=True), prefix="--tags4eggnog", doc=InputDocumentation(doc="[string]    Tags for EGGNOG id parsing (must set to activate id parsing).")), ToolInput(tag="in_gi_two_interpro_two_go", input_type=Boolean(optional=True), prefix="--gi2interpro2go", doc=InputDocumentation(doc="[string]   GI-to-INTERPRO2GO mapping file.")), ToolInput(tag="in_acc_two_interpro_two_go", input_type=Boolean(optional=True), prefix="--acc2interpro2go", doc=InputDocumentation(doc="[string]   Accession-to-INTERPRO2GO mapping file.")), ToolInput(tag="in_syn_two_interpro_two_go", input_type=Boolean(optional=True), prefix="--syn2interpro2go", doc=InputDocumentation(doc="[string]   Synonyms-to-INTERPRO2GO mapping file.")), ToolInput(tag="in_tags_four_interpro_two_go", input_type=Boolean(optional=True), prefix="--tags4interpro2go", doc=InputDocumentation(doc="[string]   Tags for INTERPRO2GO id parsing (must set to activate id parsing).")), ToolInput(tag="in_gi_two_kegg", input_type=Boolean(optional=True), prefix="--gi2kegg", doc=InputDocumentation(doc="[string]          GI-to-KEGG mapping file.")), ToolInput(tag="in_acc_two_kegg", input_type=Boolean(optional=True), prefix="--acc2kegg", doc=InputDocumentation(doc="[string]         Accession-to-KEGG mapping file.")), ToolInput(tag="in_syn_two_kegg", input_type=Boolean(optional=True), prefix="--syn2kegg", doc=InputDocumentation(doc="[string]         Synonyms-to-KEGG mapping file.")), ToolInput(tag="in_tags_four_kegg", input_type=Boolean(optional=True), prefix="--tags4kegg", doc=InputDocumentation(doc="[string]        Tags for KEGG id parsing (must set to activate id parsing).")), ToolInput(tag="in_gi_two_seed", input_type=Boolean(optional=True), prefix="--gi2seed", doc=InputDocumentation(doc="[string]          GI-to-SEED mapping file.")), ToolInput(tag="in_acc_two_seed", input_type=Boolean(optional=True), prefix="--acc2seed", doc=InputDocumentation(doc="[string]         Accession-to-SEED mapping file.")), ToolInput(tag="in_syn_two_seed", input_type=Boolean(optional=True), prefix="--syn2seed", doc=InputDocumentation(doc="[string]         Synonyms-to-SEED mapping file.")), ToolInput(tag="in_tags_four_seed", input_type=Boolean(optional=True), prefix="--tags4seed", doc=InputDocumentation(doc="[string]        Tags for SEED id parsing (must set to activate id parsing).")), ToolInput(tag="in_first_word_is_accession", input_type=Boolean(optional=True), prefix="--firstWordIsAccession", doc=InputDocumentation(doc="First word in reference header is accession number (set to 'true' for NCBI-nr downloaded Sep 2016 or later). Default value: true.")), ToolInput(tag="in_accession_tags", input_type=Boolean(optional=True), prefix="--accessionTags", doc=InputDocumentation(doc="[string(s)]   List of accession tags. Default value(s): gb| ref|.")), ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="--verbose", doc=InputDocumentation(doc="Echo commandline options and be verbose. Default value: false.")), ToolInput(tag="in_input", input_type=String(), position=0, doc=InputDocumentation(doc="-i, --in [string(s)]                 Input BLAST file[s] (gzipped ok). Mandatory option.")), ToolInput(tag="in_output", input_type=String(), position=1, doc=InputDocumentation(doc="-o, --out [string(s)]                Output file(s), one for each input file, or a directory. Mandatory option.")), ToolInput(tag="in__paired_paired", input_type=String(), position=2, doc=InputDocumentation(doc="-p, --paired                         Reads are paired. Default value: false.")), ToolInput(tag="in_parameters", input_type=String(), position=3, doc=InputDocumentation(doc="-lg, --longReads                     Parse and analyse as long reads. Default value: false."))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Blast2Rma_V0_1_0().translate("wdl", allow_empty_container=True)

