from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, String

Pbaa_Cluster_V0_1_0 = CommandToolBuilder(tool="pbaa_cluster", base_command=["pbaa", "cluster"], inputs=[ToolInput(tag="in_filter", input_type=Boolean(optional=True), prefix="--filter", doc=InputDocumentation(doc="INT    Variants with coverage lower than filter\nwill be ignored. [3]")), ToolInput(tag="in_trim_ends", input_type=Boolean(optional=True), prefix="--trim-ends", doc=InputDocumentation(doc="INT    Number of bases to trim from both sides of\nreads during graph construction and variant\ndetection. [5]")), ToolInput(tag="in_pile_size", input_type=Boolean(optional=True), prefix="--pile-size", doc=InputDocumentation(doc="INT    The number of best alignments to keep for\neach read during error correction. [30]")), ToolInput(tag="in_min_var_frequency", input_type=Boolean(optional=True), prefix="--min-var-frequency", doc=InputDocumentation(doc="FLOAT  Minimum coverage frequency within a pile.\n[0.3]")), ToolInput(tag="in_max_alignments_per_read", input_type=Boolean(optional=True), prefix="--max-alignments-per-read", doc=InputDocumentation(doc="INT    The number of random alignments, for each\nread, within a guide grouping [1000]")), ToolInput(tag="in_max_reads_per_guide", input_type=Boolean(optional=True), prefix="--max-reads-per-guide", doc=InputDocumentation(doc="INT    The number randomly selected reads to use\nwithin a guide grouping. [500]")), ToolInput(tag="in_iterations", input_type=Boolean(optional=True), prefix="--iterations", doc=InputDocumentation(doc="INT    Number of iterations to run k-means. [9]")), ToolInput(tag="in_seed", input_type=Boolean(optional=True), prefix="--seed", doc=InputDocumentation(doc="INT    Randomization seed. [1984]")), ToolInput(tag="in_max_consensus_reads", input_type=Boolean(optional=True), prefix="--max-consensus-reads", doc=InputDocumentation(doc="INT    Maximum number of reads to use per cluster\nconsensus. [100]")), ToolInput(tag="in_max_amplicon_size", input_type=Boolean(optional=True), prefix="--max-amplicon-size", doc=InputDocumentation(doc="INT    Upper read length cutoff, longer reads will\nbe skipped. [15000]")), ToolInput(tag="in_min_read_qv", input_type=Boolean(optional=True), prefix="--min-read-qv", doc=InputDocumentation(doc="FLOAT  Low read QV cutoff. [20]")), ToolInput(tag="in_off_target_groups", input_type=Boolean(optional=True), prefix="--off-target-groups", doc=InputDocumentation(doc="STR    Group names to exclude, i.e. these loci are\noff-target (not amplified).")), ToolInput(tag="in_min_cluster_frequency", input_type=Boolean(optional=True), prefix="--min-cluster-frequency", doc=InputDocumentation(doc="FLOAT  Low frequency cluster cutoff. [0.1]")), ToolInput(tag="in_min_cluster_read_count", input_type=Boolean(optional=True), prefix="--min-cluster-read-count", doc=InputDocumentation(doc="INT    Low read count cluster cutoff. [5]")), ToolInput(tag="in_max_u_chime_score", input_type=Boolean(optional=True), prefix="--max-uchime-score", doc=InputDocumentation(doc="FLOAT  High UCHIME score cutoff. [1]")), ToolInput(tag="in_num_threads", input_type=Boolean(optional=True), prefix="--num-threads", doc=InputDocumentation(doc="INT    Number of threads to use, 0 means\nautodetection. [0]")), ToolInput(tag="in_log_level", input_type=Boolean(optional=True), prefix="--log-level", doc=InputDocumentation(doc="STR    Set log level. Valid choices: (TRACE,\nDEBUG, INFO, WARN, FATAL). [WARN]")), ToolInput(tag="in_log_file", input_type=Boolean(optional=True), prefix="--log-file", doc=InputDocumentation(doc="FILE   Log to a file, instead of stderr.")), ToolInput(tag="in_files_dot", input_type=String(), position=0, doc=InputDocumentation(doc="prefix                     STR    Output prefix for run."))], outputs=[], container="quay.io/biocontainers/pbaa:1.0.0--hdfd78af_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Pbaa_Cluster_V0_1_0().translate("wdl")

