version 1.0

task AnviImportTaxonomyForGenes {
  input {
    String? contigs_db
    String? parser
    File? s_input_files
    Boolean? just_do_it
  }
  command <<<
    anvi-import-taxonomy-for-genes \
      ~{if defined(contigs_db) then ("--contigs-db " +  '"' + contigs_db + '"') else ""} \
      ~{if defined(parser) then ("--parser " +  '"' + parser + '"') else ""} \
      ~{if defined(s_input_files) then ("-i " +  '"' + s_input_files + '"') else ""} \
      ~{true="--just-do-it" false="" just_do_it}
  >>>
  parameter_meta {
    contigs_db: "Anvi'o contigs database generated by 'anvi-gen- contigs'"
    parser: "Parser to make sense of the input files. There are 3 parsers readily available: ['default_matrix', 'centrifuge', 'kaiju']. It is OK if you do not select a parser, but in that case there will be no additional contigs available except the identification of single- copy genes in your contigs for later use. Using a parser will not prevent the analysis of single-copy genes, but make anvio more powerful to help you make sense of your results. Please see the documentation, or get in touch with the developers if you have any questions regarding parsers."
    s_input_files: "(S) [FILE(S) ...], --input-files FILE(S) [FILE(S) ...] Input file(s) for selected parser. Each parser (except \"blank\") requires input files to process that you generate before running anvio. Please see the documentation for details."
    just_do_it: "Don't bother me with questions or warnings, just do it."
  }
}