class: CommandLineTool
id: ../../../../home/ubuntu/BiocondaCli/ORFcount.cwl
inputs:
- id: stranded
  doc: 'whether the data is strand-specific, reverse means reversed strand interpretation.
    (default: yes)'
  type: string
  inputBinding:
    prefix: --stranded
- id: mina_qual
  doc: skip all reads with alignment quality lower than the given minimum value (default:10)
  type: long
  inputBinding:
    prefix: --minaqual
- id: count_mode
  doc: 'mode to handle reads overlapping more than one ORF (choices: union,intersection-strict;default:
    intersection-strict)'
  type: string
  inputBinding:
    prefix: --count_mode
- id: gtf
  doc: ORF gtf file generated by RiboCode, default:final_result
  type: string
  inputBinding:
    prefix: --gtf
- id: rpf_mapping_file
  doc: ribo-seq BAM/SAM file aligned to the genome, multiple files should be separated
    with ","
  type: string
  inputBinding:
    prefix: --rpf_mapping_file
- id: first_exclude_codons
  doc: excluding the reads aligned to the first few codons of the ORF, default:15
  type: string
  inputBinding:
    prefix: --first_exclude_codons
- id: last_exclude_codons
  doc: excluding the reads aligned to the last few codons of the ORF, default:5
  type: string
  inputBinding:
    prefix: --last_exclude_codons
- id: exclude_min_orf
  doc: the minimal length(nt) of ORF for excluding the reads aligned to first and
    last few codons, default:100
  type: string
  inputBinding:
    prefix: --exclude_min_ORF
- id: min_read
  doc: minimal read length for the counting of RPF,default:26
  type: long
  inputBinding:
    prefix: --min_read
- id: max_read
  doc: maximal read length for the counting of RPF,default:34
  type: long
  inputBinding:
    prefix: --max_read
- id: output
  doc: write out all ORF counts into a file, default is to write to standard output
  type: string
  inputBinding:
    prefix: --output
outputs: []
cwlVersion: v1.1
baseCommand:
- ORFcount
