class: CommandLineTool
id: tAI.cwl
inputs:
- id: in_input
  doc: "Input file(s) in fasta format. All files should be\nsplit by comma e.g. 1.fasta,2.fasta,3.fasta[required]\n\
    . Note: input sequence must be cds sequences which\ncould be generated by GetProteinCodingSequence.py."
  type: File?
  inputBinding:
    prefix: --input
- id: in_trna_copy_numbers
  doc: tRNA copy numbers file.
  type: File?
  inputBinding:
    prefix: --tRNA_copy_numbers
- id: in_ot_put_prefix
  doc: Prefix of output files.[required]
  type: string?
  inputBinding:
    prefix: --otput_prefix
- id: in_trans_file_legend
  doc: "The legend of each fasta files, comma split. e.g.\n'condition1,condition2,condition3'\
    \ [required]"
  type: File?
  inputBinding:
    prefix: --trans_file_legend
- id: in_upstream_codon
  doc: "Upstream codon corresponding to start codon (codon\nunit). While corresponding\
    \ to stop codon, it is the\ndownstream codon.default=0"
  type: long?
  inputBinding:
    prefix: --upstream_codon
- id: in_downstream_codon
  doc: "Downstream codon corresponding to start codon (codon\nunit). While corresponding\
    \ to stop codon, it is the\nupstream codon.default=500"
  type: long?
  inputBinding:
    prefix: --downstream_codon
- id: in_table
  doc: "Which genetic code to use. default is the standard\ncode.table = 1, is the\
    \ standard table, ref: https://ww\nw.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi\n"
  type: long?
  inputBinding:
    prefix: --table
outputs:
- id: out_stdout
  doc: Standard output stream
  type: stdout
hints:
- class: DockerRequirement
  dockerPull: quay.io/biocontainers/ribominer:0.2.3.1--pyh3252c3a_0
cwlVersion: v1.1
baseCommand:
- tAI
