from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Int, File, Boolean, String, Directory, Float

Protgraph_V0_1_0 = CommandToolBuilder(tool="protgraph", base_command=["protgraph"], inputs=[ToolInput(tag="in_num_of_entries", input_type=Int(optional=True), prefix="--num_of_entries", doc=InputDocumentation(doc="Number of entries across all files (summed). if given,\nit will an estimation of the runtime")), ToolInput(tag="in_exclude_accessions", input_type=File(optional=True), prefix="--exclude_accessions", doc=InputDocumentation(doc="A csv file only containing accessions in the first row\nwhich should be excluded for processsing. Setting this\nvalue may reduce the reading performance and therefore\nthe throughput performance overall.")), ToolInput(tag="in_num_of_processes", input_type=Int(optional=True), prefix="--num_of_processes", doc=InputDocumentation(doc="The number of processes used to process entries. Each\nprocess can process an entry individually. The default\nvalue is 'cores - 1', since one process is reserved\nfor reading entries")), ToolInput(tag="in_verify_graph", input_type=Boolean(optional=True), prefix="--verify_graph", doc=InputDocumentation(doc="Set this flag to perform a check whether the graph was\ngenerated correctly. Here we explicitly check for\nparallel edges, for DAG and other properties.")), ToolInput(tag="in_skip_isoforms", input_type=Boolean(optional=True), prefix="--skip_isoforms", doc=InputDocumentation(doc="Set this flag to exclude isoforms 'VAR_SEQ' (and\npossible modification on them like variations, etc...)\nfrom the FeatureTable")), ToolInput(tag="in_skip_variants", input_type=Boolean(optional=True), prefix="--skip_variants", doc=InputDocumentation(doc="Set this flag to exclude 'VARIANT' from the")), ToolInput(tag="in_skip_signal", input_type=Boolean(optional=True), prefix="--skip_signal", doc=InputDocumentation(doc="Set this flag to exclude skipping the signal peptide\n('SIGNAL' in FeatureTable) of specific proteins")), ToolInput(tag="in_digestion", input_type=String(optional=True), prefix="--digestion", doc=InputDocumentation(doc="Set the digestion method. The full digestion cleaves\nat every edge, which can be useful for retrieving all\npossible peptides with arbitrary cutting points. The\ndigestion method skip skips the digestion completely.\nDefault: Trypsin")), ToolInput(tag="in_no_merge", input_type=Boolean(optional=True), prefix="--no_merge", doc=InputDocumentation(doc="Set this flag to skip the merging process for chains\nof nodes and edges into a single node. Setting this\noption could drastically increase the size of the\ngraph, especially its depth.")), ToolInput(tag="in_annotate_mono_weights", input_type=Boolean(optional=True), prefix="--annotate_mono_weights", doc=InputDocumentation(doc="Set this to annotate nodes and edges with the\nmonoisotopic weights. (Values are taken from the mass\ndictionary)")), ToolInput(tag="in_annotate_a_vrg_weights", input_type=Boolean(optional=True), prefix="--annotate_avrg_weights", doc=InputDocumentation(doc="Set this to annotate nodes and edges with the average\nweights. (Values are taken from the mass dictionary)")), ToolInput(tag="in_annotate_mono_weight_to_end", input_type=Boolean(optional=True), prefix="--annotate_mono_weight_to_end", doc=InputDocumentation(doc="Set this to annotate nodes and edges with the\nmonoisotopic end weights. This weight informs about\nhow much weight is at least left to get to the end\nNode. NOTE: Applying this, also sets the monoisotopic\nweights")), ToolInput(tag="in_annotate_a_vrg_weight_to_end", input_type=Boolean(optional=True), prefix="--annotate_avrg_weight_to_end", doc=InputDocumentation(doc="Set this to annotate nodes and edges with the average\nend weights. This weight informs about how much weight\nis at least left to get to the end Node. NOTE:\nApplying this, also sets the average weights")), ToolInput(tag="in_mass_dict_type", input_type=String(optional=True), prefix="--mass_dict_type", doc=InputDocumentation(doc="Set the type of the mass dictionary for amino acid.\nDefault is set to int")), ToolInput(tag="in_mass_dict_factor", input_type=Int(optional=True), prefix="--mass_dict_factor", doc=InputDocumentation(doc="Set the factor for the masses inside the\nmass_dictionary. The default is set to 1 000 000 000,\nso that each mass can be converted into integers.")), ToolInput(tag="in_calc_num_possibilities", input_type=Boolean(optional=True), prefix="--calc_num_possibilities", doc=InputDocumentation(doc="If this is set, the number of all possible (non\nrepeating) paths from the start to the end node will\nbe calculated. This uses a dynamic programming\napproach to calculate this in an efficient manner.")), ToolInput(tag="in_calc_num_possibilities_mis_cleavages", input_type=Boolean(optional=True), prefix="--calc_num_possibilities_miscleavages", doc=InputDocumentation(doc="If this is set, the number of all possible (non\nrepeating) paths from the start to the end node will\nbe calculated. This returns a list, sorted by the\nnumber of miscleavages (beginning at 0). Example:\nReturns: [1, 3, 5, 2] -> 1 path with 0 miscleavages, 3\npaths with 1 miscleavage, 5 paths with 2 miscleavages,\netc ... This uses a dynamic programming approach to\ncalculate this in an efficient manner. NOTE: This may\nget memory heavy, depending on the proteins\n(especially on Titin)")), ToolInput(tag="in_calc_num_possibilities_hops", input_type=Boolean(optional=True), prefix="--calc_num_possibilities_hops", doc=InputDocumentation(doc="If this is set, the number of all possible (non\nrepeating) paths from the start to the end node will\nbe calculated. This returns a list, sorted by the\nnumber of hops (beginning at 0). Example: Returns: [0,\n3, 5, 2] -> 0 paths with 0 hops, 3 paths with 1 hop, 5\npaths with 2 hops, etc ... This uses a dynamic\nprogramming approach to calculate this in an efficient\nmanner. NOTE: This mis even more memory heavy then\nbinning on miscleavages. Of course it depends on the\nproteins (especially on Titin) NOTE: The dedicated\nstart and end node is not counted here. If you\ntraverse a graph, expect +2 more nodes in a path!")), ToolInput(tag="in_output_csv", input_type=File(optional=True), prefix="--output_csv", doc=InputDocumentation(doc="Set the output file, which will contain information\nabout the ProteinGaph (in csv) NOTE: It will write to\n'protein_graph_statistics.csv' and overwrite if such a\nfile exists. Default is set to the current working\ndirectory")), ToolInput(tag="in_export_output_folder", input_type=Directory(optional=True), prefix="--export_output_folder", doc=InputDocumentation(doc="Set the output folder to specify the dirctory of\nexported graphs (dot, graphml, gml) NOTE: It will\noverwrite exisiting files. Default is set the current\nworking directory")), ToolInput(tag="in_export_in_directories", input_type=Boolean(optional=True), prefix="--export_in_directories", doc=InputDocumentation(doc="Set this flag to export files in directories (coded by\naccession) instead of directly by the accession name.\nThis could be useful if millions of proteins are added\ninto this tool, since a folder with millions of\nentries can be problematic in some cases.")), ToolInput(tag="in_export_dot", input_type=Boolean(optional=True), prefix="--export_dot", doc=InputDocumentation(doc="Set this flag to export a dot file for each protein")), ToolInput(tag="in_export_graph_ml", input_type=Boolean(optional=True), prefix="--export_graphml", doc=InputDocumentation(doc="Set this flag to export a GraphML file for each\nprotein. This is the recommended export method.")), ToolInput(tag="in_export_gml", input_type=Boolean(optional=True), prefix="--export_gml", doc=InputDocumentation(doc="Set this flag to export a GML file for each protein")), ToolInput(tag="in_export_pickle", input_type=Boolean(optional=True), prefix="--export_pickle", doc=InputDocumentation(doc="Set this flag to export a Pickle file for each protein")), ToolInput(tag="in_export_red_is_graph", input_type=Boolean(optional=True), prefix="--export_redisgraph", doc=InputDocumentation(doc="Set this flag to export to a redis-server having the\nmodule RedisGraph loaded.")), ToolInput(tag="in_red_is_graph_host", input_type=String(optional=True), prefix="--redisgraph_host", doc=InputDocumentation(doc="Set the host name for the redis-server having the\nmodule RedisGraph. Default: localhost")), ToolInput(tag="in_red_is_graph_port", input_type=Int(optional=True), prefix="--redisgraph_port", doc=InputDocumentation(doc="Set the port for the redis-server having the module\nRedisGraph. Default: 6379")), ToolInput(tag="in_red_is_graph_graph", input_type=String(optional=True), prefix="--redisgraph_graph", doc=InputDocumentation(doc="Set the graph name on the redis-server having the\nmodule RedisGraph. Default 'proteins'")), ToolInput(tag="in_export_postgres", input_type=Boolean(optional=True), prefix="--export_postgres", doc=InputDocumentation(doc="Set this flag to export to a postgresql server.NOTE:\nThis will try to create the tables 'nodes' and 'edges'\non a specified database. Make sure the database in\nwhich the data should be saved also exists.")), ToolInput(tag="in_postgres_host", input_type=Float(optional=True), prefix="--postgres_host", doc=InputDocumentation(doc="Set the host name for the postgresql server. Default:\n127.0.0.1")), ToolInput(tag="in_postgres_port", input_type=Int(optional=True), prefix="--postgres_port", doc=InputDocumentation(doc="Set the port for the postgresql server. Default: 5433")), ToolInput(tag="in_postgres_user", input_type=String(optional=True), prefix="--postgres_user", doc=InputDocumentation(doc="Set the username for the postgresql server. Default:\npostgres")), ToolInput(tag="in_postgres_password", input_type=String(optional=True), prefix="--postgres_password", doc=InputDocumentation(doc="Set the password for the postgresql server. Default:\ndeveloper")), ToolInput(tag="in_postgres_database", input_type=String(optional=True), prefix="--postgres_database", doc=InputDocumentation(doc="Set the database which will be used for the postgresql\nserver. Default: proteins")), ToolInput(tag="in_export_mysql", input_type=Boolean(optional=True), prefix="--export_mysql", doc=InputDocumentation(doc="Set this flag to export to a MySQL server.NOTE: This\nwill try to create the tables 'nodes' and 'edges' on a\nspecified database. Make sure the database in which\nthe data should be saved also exists.")), ToolInput(tag="in_mysql_host", input_type=Float(optional=True), prefix="--mysql_host", doc=InputDocumentation(doc="Set the host name for the MySQL server. Default:\n127.0.0.1")), ToolInput(tag="in_mysql_port", input_type=Int(optional=True), prefix="--mysql_port", doc=InputDocumentation(doc="Set the port for the MySQL server. Default: 3306")), ToolInput(tag="in_mysql_user", input_type=String(optional=True), prefix="--mysql_user", doc=InputDocumentation(doc="Set the username for the MySQL server. Default: root")), ToolInput(tag="in_mysql_password", input_type=String(optional=True), prefix="--mysql_password", doc=InputDocumentation(doc="Set the password for the MySQL server. Default:\n<empty>")), ToolInput(tag="in_mysql_database", input_type=String(optional=True), prefix="--mysql_database", doc=InputDocumentation(doc="Set the database which will be used for the MySQL\nserver. Default: proteins")), ToolInput(tag="in_export_peptide_postgres", input_type=Boolean(optional=True), prefix="--export_peptide_postgres", doc=InputDocumentation(doc="Set this flag to export peptides (specifically paths)\nto a postgresql server.NOTE: This will try to create\nthe tables 'accessions' and 'peptides' on a specified\ndatabase. Make sure the database in which the data\nshould be saved also exists. If problems occur, try to\ndelete the generated tables and retry again.")), ToolInput(tag="in_pep_postgres_host", input_type=Float(optional=True), prefix="--pep_postgres_host", doc=InputDocumentation(doc="Set the host name for the postgresql server. Default:\n127.0.0.1")), ToolInput(tag="in_pep_postgres_port", input_type=Int(optional=True), prefix="--pep_postgres_port", doc=InputDocumentation(doc="Set the port for the postgresql server. Default: 5433")), ToolInput(tag="in_pep_postgres_user", input_type=String(optional=True), prefix="--pep_postgres_user", doc=InputDocumentation(doc="Set the username for the postgresql server. Default:\npostgres")), ToolInput(tag="in_pep_postgres_password", input_type=String(optional=True), prefix="--pep_postgres_password", doc=InputDocumentation(doc="Set the password for the postgresql server. Default:\ndeveloper")), ToolInput(tag="in_pep_postgres_database", input_type=String(optional=True), prefix="--pep_postgres_database", doc=InputDocumentation(doc="Set the database which will be used for the postgresql\nserver. Default: proteins")), ToolInput(tag="in_pep_postgres_hops", input_type=Int(optional=True), prefix="--pep_postgres_hops", doc=InputDocumentation(doc="Set the number of hops (max length of path) which\nshould be used to get paths from a graph. NOTE: the\nlarger the number the more memory may be needed. This\ndepends on the protein which currently is processed.\nDefault is set to 'None', so all lengths are\nconsidered.")), ToolInput(tag="in_pep_postgres_mis_cleavages", input_type=Int(optional=True), prefix="--pep_postgres_miscleavages", doc=InputDocumentation(doc="Set this number to filter the generated paths by their\nmiscleavages.The protein graphs do contain infomration\nabout 'infinite' miscleavages and therefor also return\nthose paths/peptides. If setting (default) to '-1',\nall results are considered. However you can limit the\nnumber of miscleavages, if needed.")), ToolInput(tag="in_pep_postgres_skip_x", input_type=Boolean(optional=True), prefix="--pep_postgres_skip_x", doc=InputDocumentation(doc="Set this flag to skip to skip all entries, which\ncontain an X")), ToolInput(tag="in_pep_postgres_no_duplicates", input_type=Boolean(optional=True), prefix="--pep_postgres_no_duplicates", doc=InputDocumentation(doc="Set this flag to not insert duplicates into the\ndatabase. NOTE: Setting this value decreases the\nperformance drastically")), ToolInput(tag="in_pep_postgres_use_i_graph", input_type=Boolean(optional=True), prefix="--pep_postgres_use_igraph", doc=InputDocumentation(doc="Set this flag to use igraph instead of netx. NOTE: If\nsetting this flag, the peptide generation will be\nconsiderably faster but also consumes much more\nmemory. Also, the igraph implementation DOES NOT go\nover each single edge, so some (repeating results) may\nnever be discovered when using this flag.")), ToolInput(tag="in_pep_postgres_min_pep_length", input_type=Int(optional=True), prefix="--pep_postgres_min_pep_length", doc=InputDocumentation(doc="Set the minimum peptide length to filter out smaller\nexisting path/peptides. Here, the actual number of\naminoacid for a peptide is referenced. Default: 0")), ToolInput(tag="in_pep_postgres_batch_size", input_type=Int(optional=True), prefix="--pep_postgres_batch_size", doc=InputDocumentation(doc="Set the batch size. This defines how many peptides are\ninserted at once. Default: 25000")), ToolInput(tag="in_export_peptide_mysql", input_type=Boolean(optional=True), prefix="--export_peptide_mysql", doc=InputDocumentation(doc="Set this flag to export peptides (specifically paths)\nto a MySQL server.NOTE: This will try to create the\ntables 'accessions' and 'peptides' on a specified\ndatabase. Make sure the database in which the data\nshould be saved also exists. If problems occur, try to\ndelete the generated tables and retry again.")), ToolInput(tag="in_pep_mysql_host", input_type=Float(optional=True), prefix="--pep_mysql_host", doc=InputDocumentation(doc="Set the host name for the mysql server. Default:\n127.0.0.1")), ToolInput(tag="in_pep_mysql_port", input_type=Int(optional=True), prefix="--pep_mysql_port", doc=InputDocumentation(doc="Set the port for the mysql server. Default: 3306")), ToolInput(tag="in_pep_mysql_user", input_type=String(optional=True), prefix="--pep_mysql_user", doc=InputDocumentation(doc="Set the username for the mysql server. Default: root")), ToolInput(tag="in_pep_mysql_password", input_type=String(optional=True), prefix="--pep_mysql_password", doc=InputDocumentation(doc="Set the password for the mysql server. Default: ''")), ToolInput(tag="in_pep_mysql_database", input_type=String(optional=True), prefix="--pep_mysql_database", doc=InputDocumentation(doc="Set the database which will be used for the mysql\nserver. Default: proteins")), ToolInput(tag="in_pep_mysql_hops", input_type=Int(optional=True), prefix="--pep_mysql_hops", doc=InputDocumentation(doc="Set the number of hops (max length of path) which\nshould be used to get paths from a graph. NOTE: the\nlarger the number the more memory may be needed. This\ndepends on the protein which currently is processed.\nDefault is set to 'None', so all lengths are\nconsidered.")), ToolInput(tag="in_pep_mysql_mis_cleavages", input_type=Int(optional=True), prefix="--pep_mysql_miscleavages", doc=InputDocumentation(doc="Set this number to filter the generated paths by their\nmiscleavages.The protein graphs do contain infomration\nabout 'infinite' miscleavages and therefor also return\nthose paths/peptides. If setting (default) to '-1',\nall results are considered. However you can limit the\nnumber of miscleavages, if needed.")), ToolInput(tag="in_pep_mysql_skip_x", input_type=Boolean(optional=True), prefix="--pep_mysql_skip_x", doc=InputDocumentation(doc="Set this flag to skip to skip all entries, which\ncontain an X")), ToolInput(tag="in_pep_mysql_no_duplicates", input_type=Boolean(optional=True), prefix="--pep_mysql_no_duplicates", doc=InputDocumentation(doc="Set this flag to not insert duplicates into the\ndatabase. NOTE: Setting this value decreases the\nperformance drastically")), ToolInput(tag="in_pep_mysql_use_i_graph", input_type=Boolean(optional=True), prefix="--pep_mysql_use_igraph", doc=InputDocumentation(doc="Set this flag to use igraph instead of netx. NOTE: If\nsetting this flag, the peptide generation will be\nconsiderably faster but also consumes much more\nmemory. Also, the igraph implementation DOES NOT go\nover each single edge, so some (repeating results) may\nnever be discovered when using this flag.")), ToolInput(tag="in_pep_mysql_min_pep_length", input_type=Int(optional=True), prefix="--pep_mysql_min_pep_length", doc=InputDocumentation(doc="Set the minimum peptide length to filter out smaller\nexisting path/peptides. Here, the actual number of\naminoacid for a peptide is referenced. Default: 0")), ToolInput(tag="in_pep_mysql_batch_size", input_type=Int(optional=True), prefix="--pep_mysql_batch_size", doc=InputDocumentation(doc="Set the batch size. This defines how many peptides are\ninserted at once. Default: 25000")), ToolInput(tag="in_export_peptide_fast_a", input_type=Boolean(optional=True), prefix="--export_peptide_fasta", doc=InputDocumentation(doc="Set this flag to export peptides into a single fasta\nfile.")), ToolInput(tag="in_pep_fast_a_out", input_type=File(optional=True), prefix="--pep_fasta_out", doc=InputDocumentation(doc="Set the output file for the peptide fasta export.\nDefault: '${pwd}/peptides.fasta'. NOTE: This will\noverwrite existing files.")), ToolInput(tag="in_pep_fast_a_hops", input_type=Int(optional=True), prefix="--pep_fasta_hops", doc=InputDocumentation(doc="Set the number of hops (max length of path) which\nshould be used to get paths from a graph. NOTE: the\nlarger the number the more memory may be needed. This\ndepends on the protein which currently is processed.\nDefault is set to 'None', so all lengths are\nconsidered.")), ToolInput(tag="in_pep_fast_a_mis_cleavages", input_type=Int(optional=True), prefix="--pep_fasta_miscleavages", doc=InputDocumentation(doc="Set this number to filter the generated paths by their\nmiscleavages.The protein graphs do contain infomration\nabout 'infinite' miscleavages and therefor also return\nthose paths/peptides. If setting (default) to '-1',\nall results are considered. However you can limit the\nnumber of miscleavages, if needed.")), ToolInput(tag="in_pep_fast_a_skip_x", input_type=Boolean(optional=True), prefix="--pep_fasta_skip_x", doc=InputDocumentation(doc="Set this flag to skip to skip all entries, which\ncontain an X")), ToolInput(tag="in_pep_fast_a_use_i_graph", input_type=Boolean(optional=True), prefix="--pep_fasta_use_igraph", doc=InputDocumentation(doc="Set this flag to use igraph instead of netx. NOTE: If\nsetting this flag, the peptide generation will be\nconsiderably faster but also consumes much more\nmemory. Also, the igraph implementation DOES NOT go\nover each single edge, so some (repeating results) may\nnever be discovered when using this flag.")), ToolInput(tag="in_pep_fast_a_min_pep_length", input_type=Int(optional=True), prefix="--pep_fasta_min_pep_length", doc=InputDocumentation(doc="Set the minimum peptide length to filter out smaller\nexisting path/peptides. Here, the actual number of\naminoacid for a peptide is referenced. Default: 0")), ToolInput(tag="in_pep_fast_a_batch_size", input_type=Int(optional=True), prefix="--pep_fasta_batch_size", doc=InputDocumentation(doc="Set the batch size. This defines how many peptides are\nprocessed and written at once. Default: 25000")), ToolInput(tag="in_export_gremlin", input_type=Boolean(optional=True), prefix="--export_gremlin", doc=InputDocumentation(doc="Set this flag to export the graphs via gremlin to a\ngremlin server.NOTE: The export is very slow, since it\nexecutes each node as a single query (tested on\nJanusGraph and Apache Gremlin Server). This exporter\nis not well implemented and may not work. This is due\nto difficulties implementing such an exporter in a\nglobal manner. To reduce the number of errors: Try to\nhave a stable connection to the gremlin-server and\nalso allocate enough resource for it, so that it can\nprocess the queries quick enough.")), ToolInput(tag="in_gremlin_url", input_type=Int(optional=True), prefix="--gremlin_url", doc=InputDocumentation(doc="Set the url to the gremlin URL (no authentication).\nDefault: 'ws://localhost:8182/gremlin'")), ToolInput(tag="in_gremlin_traversal_source", input_type=String(optional=True), prefix="--gremlin_traversal_source", doc=InputDocumentation(doc="Set the traversal source for remote. Default 'g'\n")), ToolInput(tag="in_files", input_type=String(), position=0, doc=InputDocumentation(doc="Files containing the Swissprot/EMBL-Entries (either in\n.dat or .txt)")), ToolInput(tag="in_feature_table", input_type=String(), position=0, doc=InputDocumentation(doc="--skip_init_met, -sm  Set this flag to exclude the skipping of the initiator"))], outputs=[ToolOutput(tag="out_output_csv", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output_csv", type_hint=File()), doc=OutputDocumentation(doc="Set the output file, which will contain information\nabout the ProteinGaph (in csv) NOTE: It will write to\n'protein_graph_statistics.csv' and overwrite if such a\nfile exists. Default is set to the current working\ndirectory")), ToolOutput(tag="out_export_output_folder", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_export_output_folder", type_hint=File()), doc=OutputDocumentation(doc="Set the output folder to specify the dirctory of\nexported graphs (dot, graphml, gml) NOTE: It will\noverwrite exisiting files. Default is set the current\nworking directory")), ToolOutput(tag="out_pep_fast_a_out", output_type=File(optional=True), selector=InputSelector(input_to_select="in_pep_fast_a_out", type_hint=File()), doc=OutputDocumentation(doc="Set the output file for the peptide fasta export.\nDefault: '${pwd}/peptides.fasta'. NOTE: This will\noverwrite existing files."))], container="quay.io/biocontainers/protgraph:0.0.2b3--pyhdfd78af_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Protgraph_V0_1_0().translate("wdl")

