from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import String, File, Boolean, Float, Int

Rnaphylo_V0_1_0 = CommandToolBuilder(tool="RNAPhylo", base_command=["RNAPhylo"], inputs=[ToolInput(tag="in__phylogeny_tree", input_type=String(optional=True), prefix="-t", doc=InputDocumentation(doc=": The phylogeny tree")), ToolInput(tag="in_phylogenetic_model_single", input_type=File(optional=True), prefix="-s", doc=InputDocumentation(doc=": The phylogenetic model for single bases")), ToolInput(tag="in_phylogenetic_model_paired", input_type=File(optional=True), prefix="-p", doc=InputDocumentation(doc=": The phylogenetic model for paired bases")), ToolInput(tag="in_scfg_grammar_file", input_type=File(optional=True), prefix="-g", doc=InputDocumentation(doc=": The SCFG grammar file.")), ToolInput(tag="in_float_lower_bound", input_type=Boolean(optional=True), prefix="--l", doc=InputDocumentation(doc="<float> : The lower bound for paired emission probability to be interesting")), ToolInput(tag="in_partition", input_type=Boolean(optional=True), prefix="--partition", doc=InputDocumentation(doc=": Use partition function")), ToolInput(tag="in_in_format", input_type=String(optional=True), prefix="--informat", doc=InputDocumentation(doc=": The format of input alignment")), ToolInput(tag="in_degen_is_absent", input_type=Boolean(optional=True), prefix="--degen-is-absent", doc=InputDocumentation(doc=": model degenerate nucleotides as absent")), ToolInput(tag="in_fragmentary", input_type=Boolean(optional=True), prefix="--fragmentary", doc=InputDocumentation(doc=": model fragmentary (or equivalently truncated) sequences (implies --degen-is-absent)")), ToolInput(tag="in_only_ss_cons", input_type=Boolean(optional=True), prefix="--only-ss-cons", doc=InputDocumentation(doc=": only allow emission of base pairs in SS_cons (to test this aspect of --fragmentary)")), ToolInput(tag="in_gap_thresh", input_type=Float(optional=True), prefix="--gap-thresh", doc=InputDocumentation(doc=": gap threshold for removing columns.  --gap-thresh 1 means retain all columns.  Default: 0.7")), ToolInput(tag="in_partition_close_to_fuzzy_limit", input_type=Int(optional=True), prefix="--partition-close-to-fuzzy-limit", doc=InputDocumentation(doc=": 'fuzzy' nucleotide positions are those that are on the edge of                                          a truncation, or have a suspicious number of degenerate nucleotides (see --suspicious-degen-nucs).                                      When a nucleotide is within <int> nucleotides of a fuzzy position, the partition function cannot                                        be calculated on it, as if the nucleotide were absent.  This flag is meaningless unless --partition is                                          also used.  A value of -1 disables this feature.  Default: -1.  Recommend 3 if you use it.")), ToolInput(tag="in_suspicious_degen_n_ucs", input_type=Int(optional=True), prefix="--suspicious-degen-nucs", doc=InputDocumentation(doc=": when there are <int> or more consecutive degenerate nucleotides                                         they are considered suspicious (see --partition-close-to-fuzzy-limit)                                   Default: 2.")), ToolInput(tag="in_ignore_all_gap", input_type=Boolean(optional=True), prefix="--ignore-all-gap", doc=InputDocumentation(doc=": don't complain about alignments containing columns with no nucleotides"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Rnaphylo_V0_1_0().translate("wdl", allow_empty_container=True)

