from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Boolean, Int, String

Agat_Sp_Manage_Ids_Pl_V0_1_0 = CommandToolBuilder(tool="agat_sp_manage_IDs.pl", base_command=["agat_sp_manage_IDs.pl"], inputs=[ToolInput(tag="in_gff", input_type=File(optional=True), prefix="--gff", doc=InputDocumentation(doc="Input GTF/GFF file.")), ToolInput(tag="in_gap", input_type=Boolean(optional=True), prefix="--gap", doc=InputDocumentation(doc="Integer - Increment the next gene (level1 feature) suffix with\nthis value. Defauft 0.")), ToolInput(tag="in_ensembl", input_type=Boolean(optional=True), prefix="--ensembl", doc=InputDocumentation(doc="Boolean - For an ID Ensembl like (e.g PREFIXG00000000022). The\nID is built as follow: $opt_prefix.$letterCode.0*.Number where\nthe number of 0 is adapted in order to have 11 digits.")), ToolInput(tag="in_prefix", input_type=Boolean(optional=True), prefix="--prefix", doc=InputDocumentation(doc="String - Add a specific prefix to the ID. By defaut if will be\nthe feature type (3rd column).")), ToolInput(tag="in_type_dependent", input_type=Boolean(optional=True), prefix="--type_dependent", doc=InputDocumentation(doc="Boolean - Activate type_dependent numbering. The number is\ndepedendent of the feature type. i.e instead of: NbV1Ch01\nAUGUSTUS gene 97932 99714 0.06 - . ID=gene1 NbV1Ch01 AUGUSTUS\nmRNA 97932 99714 0.06 - . ID=mRNA2 NbV1Ch01 AUGUSTUS exon 97932\n98571 . - . ID=exon3 NbV1Ch01 AUGUSTUS exon 98679 98844 . - .\nID=exon4 You will get: NbV1Ch01 AUGUSTUS gene 97932 99714 0.06 -\n. ID=gene1 NbV1Ch01 AUGUSTUS mRNA 97932 99714 0.06 - . ID=mRNA1\nNbV1Ch01 AUGUSTUS exon 97932 98571 . - . ID=exon1 NbV1Ch01\nAUGUSTUS exon 98679 98844 . - . ID=exon2")), ToolInput(tag="in_collective", input_type=Boolean(optional=True), prefix="--collective", doc=InputDocumentation(doc="Boolean - In the case of discontinuous features (i.e. a single\nfeature that exists over multiple genomic locations like CDS,\nUTR) we set a uniq ID by default. If you wish to set the a\ncollective ID for those feature, please activate this option.")), ToolInput(tag="in_tair", input_type=Boolean(optional=True), prefix="--tair", doc=InputDocumentation(doc="Boolean - Tair like Output:\nNbV1Ch01 TAIR10 gene 5928 8737 . - . ID=AT1G01020 NbV1Ch01\nTAIR10 mRNA 5928 8737 . - . ID=AT1G01020.1 NbV1Ch01 TAIR10 exon\n5928 8737 . - . ID=AT1G01020.1-exon1")), ToolInput(tag="in_nb", input_type=Boolean(optional=True), prefix="--nb", doc=InputDocumentation(doc="Integer - Start numbering to this value. Default 1.")), ToolInput(tag="in_primary_tag_option", input_type=Int(optional=True), prefix="-p", doc=InputDocumentation(doc="primary tag option, case insensitive, list. Allow to specied the\nfeature types that will be handled. You can specified a specific\nfeature by given its primary tag name (column 3) as: cds, Gene,\nMrNa You can specify directly all the feature of a particular\nlevel: level2=mRNA,ncRNA,tRNA,etc level3=CDS,exon,UTR,etc By\ndefault all feature are taken into account. fill the option by\nthe value 'all' will have the same behaviour.")), ToolInput(tag="in_outfile", input_type=File(optional=True), prefix="--outfile", doc=InputDocumentation(doc="String - Output GFF file. If no output file is specified, the\noutput will be written to STDOUT.")), ToolInput(tag="in_agat_sp_manage_ids_do_tpl", input_type=String(), position=0, doc=InputDocumentation(doc="Description:"))], outputs=[ToolOutput(tag="out_outfile", output_type=File(optional=True), selector=InputSelector(input_to_select="in_outfile", type_hint=File()), doc=OutputDocumentation(doc="String - Output GFF file. If no output file is specified, the\noutput will be written to STDOUT."))], container="quay.io/biocontainers/agat:0.6.0--pl526r35_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Agat_Sp_Manage_Ids_Pl_V0_1_0().translate("wdl")

