from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import String, Int, File, Boolean, Directory, Float

Akita_Data_Py_V0_1_0 = CommandToolBuilder(tool="akita_data.py", base_command=["akita_data.py"], inputs=[ToolInput(tag="in_set_blacklist_nucleotides", input_type=String(optional=True), prefix="-b", doc=InputDocumentation(doc="Set blacklist nucleotides to a baseline value.")), ToolInput(tag="in_break", input_type=Int(optional=True), prefix="--break", doc=InputDocumentation(doc="Break in half contigs above length [Default: 8388608]")), ToolInput(tag="in_crop", input_type=Int(optional=True), prefix="--crop", doc=InputDocumentation(doc="Crop bp off each end [Default: 0]")), ToolInput(tag="in_positions_diagonal_ignore", input_type=Int(optional=True), prefix="-d", doc=InputDocumentation(doc="Positions on the diagonal to ignore [Default: 2]")), ToolInput(tag="in_generate_cross_split", input_type=String(optional=True), prefix="-f", doc=InputDocumentation(doc="Generate cross fold split [Default: none]")), ToolInput(tag="in_genome_assembly_gaps", input_type=File(optional=True), prefix="-g", doc=InputDocumentation(doc="Genome assembly gaps BED [Default: none]")), ToolInput(tag="in_gaussian_kernel_stddev", input_type=Int(optional=True), prefix="-k", doc=InputDocumentation(doc="Gaussian kernel stddev to smooth values [Default: 0]")), ToolInput(tag="in_sequence_length", input_type=Int(optional=True), prefix="-l", doc=InputDocumentation(doc="Sequence length [Default: 131072]")), ToolInput(tag="in_limit", input_type=File(optional=True), prefix="--limit", doc=InputDocumentation(doc="Limit to segments that overlap regions in a BED file")), ToolInput(tag="in_local", input_type=Boolean(optional=True), prefix="--local", doc=InputDocumentation(doc="Run jobs locally as opposed to on SLURM [Default:\nFalse]")), ToolInput(tag="in_output_directory", input_type=Directory(optional=True), prefix="-o", doc=InputDocumentation(doc="Output directory [Default: data_out]")), ToolInput(tag="in_number_parallel_processes", input_type=Int(optional=True), prefix="-p", doc=InputDocumentation(doc="Number parallel processes [Default: none]")), ToolInput(tag="in_sequences_tfrecord_file", input_type=File(optional=True), prefix="-r", doc=InputDocumentation(doc="Sequences per TFRecord file [Default: 128]")), ToolInput(tag="in_restart", input_type=Boolean(optional=True), prefix="--restart", doc=InputDocumentation(doc="Continue progress from midpoint. [Default: False]")), ToolInput(tag="in_sample", input_type=String(optional=True), prefix="--sample", doc=InputDocumentation(doc="Down-sample the segments")), ToolInput(tag="in_seed", input_type=Int(optional=True), prefix="--seed", doc=InputDocumentation(doc="Random seed [Default: 44]")), ToolInput(tag="in_stride_train", input_type=String(optional=True), prefix="--stride_train", doc=InputDocumentation(doc="Stride to advance train sequences [Default:\nseq_length]")), ToolInput(tag="in_stride_test", input_type=String(optional=True), prefix="--stride_test", doc=InputDocumentation(doc="Stride to advance valid and test sequences [Default:\nseq_length]")), ToolInput(tag="in_split_test", input_type=Boolean(optional=True), prefix="--split_test", doc=InputDocumentation(doc="Exit after split. [Default: False]")), ToolInput(tag="in_proportion_data_testing", input_type=Float(optional=True), prefix="-t", doc=InputDocumentation(doc="Proportion of the data for testing [Default: 0.05]")), ToolInput(tag="in_unmappable_regions_bed", input_type=String(optional=True), prefix="-u", doc=InputDocumentation(doc="Unmappable regions in BED format")), ToolInput(tag="in_u_map_midpoints", input_type=Int(optional=True), prefix="--umap_midpoints", doc=InputDocumentation(doc="Regions with midpoints to exclude in BED format. Used\nfor 4C/HiC.")), ToolInput(tag="in_u_map_t", input_type=Float(optional=True), prefix="--umap_t", doc=InputDocumentation(doc="Remove sequences with more than this unmappable bin %\n[Default: 0.3]")), ToolInput(tag="in_u_map_set", input_type=String(optional=True), prefix="--umap_set", doc=InputDocumentation(doc="Set unmappable regions to this percentile in the\nsequences' distribution of values")), ToolInput(tag="in_sum_pool_width", input_type=Int(optional=True), prefix="-w", doc=InputDocumentation(doc="Sum pool width [Default: 128]")), ToolInput(tag="in_proportion_data_validation", input_type=Float(optional=True), prefix="-v", doc=InputDocumentation(doc="Proportion of the data for validation [Default: 0.05]")), ToolInput(tag="in_snap", input_type=String(optional=True), prefix="--snap", doc=InputDocumentation(doc="snap stride to multiple for binned targets in bp, if\nnot None seq_length must be a multiple of snap")), ToolInput(tag="in_as_obs_exp", input_type=Boolean(optional=True), prefix="--as_obsexp", doc=InputDocumentation(doc="save targets as obsexp profiles")), ToolInput(tag="in_global_obs_exp", input_type=Boolean(optional=True), prefix="--global_obsexp", doc=InputDocumentation(doc="use pre-calculated by-chromosome obs/exp")), ToolInput(tag="in_no_log", input_type=Boolean(optional=True), prefix="--no_log", doc=InputDocumentation(doc="do not take log for obs/exp")), ToolInput(tag="in_fast_a_file", input_type=String(), position=0, doc=InputDocumentation(doc="")), ToolInput(tag="in_targets_file", input_type=String(), position=1, doc=InputDocumentation(doc=""))], outputs=[ToolOutput(tag="out_output_directory", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_output_directory", type_hint=File()), doc=OutputDocumentation(doc="Output directory [Default: data_out]"))], container="quay.io/biocontainers/basenji:0.5--py_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Akita_Data_Py_V0_1_0().translate("wdl")

