version 1.0

task CoverageOfEachTrans {
  input {
    File? bam_list_file
    File? input_files_bam
    File? coordinate_file
    String? ot_put_prefix
    Int? specific_reads_length
    Int? offset
    File? bam_file_legend
    String? select_trans_list
    String? id_type
    String? mode
  }
  command <<<
    CoverageOfEachTrans \
      ~{if defined(bam_list_file) then ("--bamListFile " +  '"' + bam_list_file + '"') else ""} \
      ~{if defined(input_files_bam) then ("--input " +  '"' + input_files_bam + '"') else ""} \
      ~{if defined(coordinate_file) then ("--coordinateFile " +  '"' + coordinate_file + '"') else ""} \
      ~{if defined(ot_put_prefix) then ("--otput_prefix " +  '"' + ot_put_prefix + '"') else ""} \
      ~{if defined(specific_reads_length) then ("--specific_reads_length " +  '"' + specific_reads_length + '"') else ""} \
      ~{if defined(offset) then ("--offset " +  '"' + offset + '"') else ""} \
      ~{if defined(bam_file_legend) then ("--bam_file_legend " +  '"' + bam_file_legend + '"') else ""} \
      ~{if defined(select_trans_list) then ("--select_trans_list " +  '"' + select_trans_list + '"') else ""} \
      ~{if defined(id_type) then ("--id-type " +  '"' + id_type + '"') else ""} \
      ~{if defined(mode) then ("--mode " +  '"' + mode + '"') else ""}
  >>>
  runtime {
    docker: "quay.io/biocontainers/ribominer:0.2.3.1--pyh3252c3a_0"
  }
  parameter_meta {
    bam_list_file: "Bam file list, containing 4 columns.Namely\\nbamFiles,readLength, offSet, bamLegend. '-f' and '-i,\\n-r, -s, -t' parameters are mutually\\nexclusive.default=none."
    input_files_bam: "Input file(s) in bam format. All files should be split\\nby comma e.g. 1.bam,2.bam,3.bam[required]. '-i' and\\n'-f' are mutually exclusive. default=none"
    coordinate_file: "The file should contain the coordinate of start and\\nstop codon. Generated by\\nOutputTranscriptInfo.py.[required]"
    ot_put_prefix: "Prefix of output files.[required],default=none"
    specific_reads_length: "Specific the lenght to do analysis, comma split. e.g.\\n'28,29,30'.If use all length set 'All'. Bam files diff\\nlength select split by '_' e.g. '28,29,30_ALL_27,28'\\n[required]. '-r' and '-f' are mutually exclusive."
    offset: "Specific the offset corresponding to read length,\\ncomma split. e.g. '12,13,13'. No offset set 0. Bam\\nfiles diff offset select split by '_' e.g.\\n'12,13,13_0_12,12' [required]. '-s' and '-f' are\\nmutually exclusive."
    bam_file_legend: "The legend of each bam files, comma split. e.g.\\n'condition1,condition2,condition3' [required]. '-t'\\nand '-f' are mutually exclusive."
    select_trans_list: "Selected transcript list used for metagene\\nanalysis.This files requires the first column must be\\nthe transcript ID  with a column name."
    id_type: "define the id type users input. the default is\\ntranscript id, if not, will be transformed into\\ntranscript id. default=transcript_id"
    mode: "Mode of calculation. Either coverage or density of a\\ntranscript or transcripts. default=coverage\\n"
  }
  output {
    File out_stdout = stdout()
  }
}