from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Int, Boolean, String

Rnalien_V0_1_0 = CommandToolBuilder(tool="RNAlien", base_command=["RNAlien"], inputs=[ToolInput(tag="in_input_fast_a_file_path", input_type=File(optional=True), prefix="--inputfastafilepath", doc=InputDocumentation(doc="Path to input fasta file")), ToolInput(tag="in_input_alignment_file_path", input_type=File(optional=True), prefix="--inputalignmentfilepath", doc=InputDocumentation(doc="Path to input alignment file")), ToolInput(tag="in_output_path", input_type=File(optional=True), prefix="--outputpath", doc=InputDocumentation(doc="Path to output directory. Default:\ncurrent working directory")), ToolInput(tag="in_input_taxid", input_type=Int(optional=True), prefix="--inputtaxid", doc=InputDocumentation(doc="NCBI taxonomy ID number of input RNA")), ToolInput(tag="in_input_evalue_cut_off", input_type=Int(optional=True), prefix="--inputevaluecutoff", doc=InputDocumentation(doc="Evalue cutoff for cmsearch filtering.\nDefault: 0.001")), ToolInput(tag="in_input_blast_database", input_type=File(optional=True), prefix="--inputblastdatabase", doc=InputDocumentation(doc="Specify name of blast database to use,\nin offline mode the filepath to the\nblast database (/home/user/nt_v5).\nDefault: nt")), ToolInput(tag="in_length_filter", input_type=Boolean(optional=True), prefix="--lengthfilter", doc=InputDocumentation(doc="Filter blast hits per genomic length.\nDefault: True")), ToolInput(tag="in_coverage_filter", input_type=Boolean(optional=True), prefix="--coveragefilter", doc=InputDocumentation(doc="Filter blast hits by coverage of at\nleast 80%. Default: True")), ToolInput(tag="in_single_hit_per_tax", input_type=Boolean(optional=True), prefix="--singlehitpertax", doc=InputDocumentation(doc="Only the best blast hit per taxonomic\nentry is considered. Default: False")), ToolInput(tag="in_blast_soft_masking", input_type=Boolean(optional=True), prefix="--blastsoftmasking", doc=InputDocumentation(doc="Toggles blast query softmasking,\nmeaning masking of non-conserved regions\non the query. Default: False")), ToolInput(tag="in_input_query_selection_method", input_type=String(optional=True), prefix="--inputqueryselectionmethod", doc=InputDocumentation(doc="Method for selection of queries\n(filtering,clustering). Default:\nfiltering")), ToolInput(tag="in_input_query_number", input_type=Int(optional=True), prefix="--inputquerynumber", doc=InputDocumentation(doc="Number of queries used for candidate\nsearch. Default: 5")), ToolInput(tag="in_threads", input_type=Int(optional=True), prefix="--threads", doc=InputDocumentation(doc="Number of available cpu slots/cores.\nDefault: 1")), ToolInput(tag="in_taxonomy_restriction", input_type=String(optional=True), prefix="--taxonomyrestriction", doc=InputDocumentation(doc="Restrict search space to taxonomic")), ToolInput(tag="in_session_identifica_tor", input_type=String(optional=True), prefix="--sessionidentificator", doc=InputDocumentation(doc="Optional session id that is used\ninstead of automatically generated one.")), ToolInput(tag="in_perform_evaluation", input_type=Boolean(optional=True), prefix="--performevaluation", doc=InputDocumentation(doc="Perform evaluation step. Default: True")), ToolInput(tag="in_check_setup", input_type=Boolean(optional=True), prefix="--checksetup", doc=InputDocumentation(doc="Just prints installed tool versions\nand performs connection check. Default:\nFalse")), ToolInput(tag="in_taxonomy_dump_path", input_type=File(optional=True), prefix="--taxonomydumppath", doc=InputDocumentation(doc="Path to NCBI taxonomy dump directory.")), ToolInput(tag="in_offline_mode", input_type=Boolean(optional=True), prefix="--offlinemode", doc=InputDocumentation(doc="Uses locally installed blast and\ndatabases. Default: False")), ToolInput(tag="in_numeric_version", input_type=Boolean(optional=True), prefix="--numeric-version", doc=InputDocumentation(doc="Print just the version number")), ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="--verbose", doc=InputDocumentation(doc="Loud verbosity")), ToolInput(tag="in_quiet", input_type=Boolean(optional=True), prefix="--quiet", doc=InputDocumentation(doc="Quiet verbosity")), ToolInput(tag="in_organism", input_type=String(), position=0, doc=InputDocumentation(doc="-z --inputnscicutoff=NUM             Only candidate sequences with a")), ToolInput(tag="in_kingdom", input_type=String(), position=0, doc=InputDocumentation(doc="(bacteria,archea,eukaryia,cellularorganisms,viruses)."))], outputs=[ToolOutput(tag="out_output_path", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output_path", type_hint=File()), doc=OutputDocumentation(doc="Path to output directory. Default:\ncurrent working directory"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Rnalien_V0_1_0().translate("wdl", allow_empty_container=True)

