from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, File

Blastall_V0_1_0 = CommandToolBuilder(tool="blastall", base_command=["blastall"], inputs=[ToolInput(tag="in_program_name", input_type=Boolean(optional=True), prefix="-p", doc=InputDocumentation(doc="Program Name [String]")), ToolInput(tag="in_database_default_", input_type=Boolean(optional=True), prefix="-d", doc=InputDocumentation(doc="Database [String]\ndefault = nr")), ToolInput(tag="in_query_file_default", input_type=Boolean(optional=True), prefix="-i", doc=InputDocumentation(doc="Query File [File In]\ndefault = stdin")), ToolInput(tag="in_expectation_value_e", input_type=Boolean(optional=True), prefix="-e", doc=InputDocumentation(doc="Expectation value (E) [Real]\ndefault = 10.0")), ToolInput(tag="in_alignment_view_options", input_type=Boolean(optional=True), prefix="-m", doc=InputDocumentation(doc="alignment view options:")), ToolInput(tag="in_blast_report_file", input_type=File(optional=True), prefix="-o", doc=InputDocumentation(doc="BLAST report Output File [File Out]  Optional\ndefault = stdout")), ToolInput(tag="in_filter_query_sequence", input_type=Boolean(optional=True), prefix="-F", doc=InputDocumentation(doc="Filter query sequence (DUST with blastn, SEG with others) [String]\ndefault = T")), ToolInput(tag="in_cost_open_gap", input_type=Boolean(optional=True), prefix="-G", doc=InputDocumentation(doc="Cost to open a gap (-1 invokes default behavior) [Integer]\ndefault = -1")), ToolInput(tag="in_cost_extend_gap", input_type=Boolean(optional=True), prefix="-E", doc=InputDocumentation(doc="Cost to extend a gap (-1 invokes default behavior) [Integer]\ndefault = -1")), ToolInput(tag="in_x_dropoff_value_gapped", input_type=Boolean(optional=True), prefix="-X", doc=InputDocumentation(doc="X dropoff value for gapped alignment (in bits) (zero invokes default behavior)\nblastn 30, megablast 20, tblastx 0, all others 15 [Integer]\ndefault = 0")), ToolInput(tag="in_show_gis_deflines", input_type=Boolean(optional=True), prefix="-I", doc=InputDocumentation(doc="Show GI's in deflines [T/F]\ndefault = F")), ToolInput(tag="in_penalty_nucleotide_mismatch", input_type=Boolean(optional=True), prefix="-q", doc=InputDocumentation(doc="Penalty for a nucleotide mismatch (blastn only) [Integer]\ndefault = -3")), ToolInput(tag="in_reward_nucleotide_match", input_type=Boolean(optional=True), prefix="-r", doc=InputDocumentation(doc="Reward for a nucleotide match (blastn only) [Integer]\ndefault = 1")), ToolInput(tag="in_number_database_sequences", input_type=Boolean(optional=True), prefix="-v", doc=InputDocumentation(doc="Number of database sequences to show one-line descriptions for (V) [Integer]\ndefault = 500")), ToolInput(tag="in_number_database_sequence", input_type=Boolean(optional=True), prefix="-b", doc=InputDocumentation(doc="Number of database sequence to show alignments for (B) [Integer]\ndefault = 250")), ToolInput(tag="in_threshold_extending_hits", input_type=Boolean(optional=True), prefix="-f", doc=InputDocumentation(doc="Threshold for extending hits, default if zero\nblastp 11, blastn 0, blastx 12, tblastn 13\ntblastx 13, megablast 0 [Real]\ndefault = 0")), ToolInput(tag="in_perform_gapped_alignment", input_type=Boolean(optional=True), prefix="-g", doc=InputDocumentation(doc="Perform gapped alignment (not available with tblastx) [T/F]\ndefault = T")), ToolInput(tag="in_query_genetic_code", input_type=Boolean(optional=True), prefix="-Q", doc=InputDocumentation(doc="Query Genetic code to use [Integer]\ndefault = 1")), ToolInput(tag="in_db_genetic_code", input_type=Boolean(optional=True), prefix="-D", doc=InputDocumentation(doc="DB Genetic code (for tblast[nx] only) [Integer]\ndefault = 1")), ToolInput(tag="in_number_use_default", input_type=Boolean(optional=True), prefix="-a", doc=InputDocumentation(doc="Number of processors to use [Integer]\ndefault = 1")), ToolInput(tag="in_seqalign_file_optional", input_type=File(optional=True), prefix="-O", doc=InputDocumentation(doc="SeqAlign file [File Out]  Optional")), ToolInput(tag="in_believe_defline_default", input_type=Boolean(optional=True), prefix="-J", doc=InputDocumentation(doc="Believe the query defline [T/F]\ndefault = F")), ToolInput(tag="in_matrix_default_blosum", input_type=Boolean(optional=True), prefix="-M", doc=InputDocumentation(doc="Matrix [String]\ndefault = BLOSUM62")), ToolInput(tag="in_word_size_default", input_type=Boolean(optional=True), prefix="-W", doc=InputDocumentation(doc="Word size, default if zero (blastn 11, megablast 28, all others 3) [Integer]\ndefault = 0")), ToolInput(tag="in_effective_length_database", input_type=Boolean(optional=True), prefix="-z", doc=InputDocumentation(doc="Effective length of the database (use zero for the real size) [Real]\ndefault = 0")), ToolInput(tag="in_number_best_keep", input_type=Boolean(optional=True), prefix="-K", doc=InputDocumentation(doc="Number of best hits from a region to keep. Off by default.")), ToolInput(tag="in_multiple_hit_single", input_type=Boolean(optional=True), prefix="-P", doc=InputDocumentation(doc="0 for multiple hit, 1 for single hit (does not apply to blastn) [Integer]\ndefault = 0")), ToolInput(tag="in_effective_length_search", input_type=Boolean(optional=True), prefix="-Y", doc=InputDocumentation(doc="Effective length of the search space (use zero for the real size) [Real]\ndefault = 0")), ToolInput(tag="in_query_strands_search", input_type=Boolean(optional=True), prefix="-S", doc=InputDocumentation(doc="Query strands to search against database (for blast[nx], and tblastx)\n3 is both, 1 is top, 2 is bottom [Integer]\ndefault = 3")), ToolInput(tag="in_produce_html_output", input_type=Boolean(optional=True), prefix="-T", doc=InputDocumentation(doc="Produce HTML output [T/F]\ndefault = F")), ToolInput(tag="in_restrict_search_list", input_type=Boolean(optional=True), prefix="-l", doc=InputDocumentation(doc="Restrict search of database to list of GI's [String]  Optional")), ToolInput(tag="in_use_case_filtering", input_type=Boolean(optional=True), prefix="-U", doc=InputDocumentation(doc="Use lower case filtering of FASTA sequence [T/F]  Optional")), ToolInput(tag="in_x_dropoff_value_ungapped", input_type=Boolean(optional=True), prefix="-y", doc=InputDocumentation(doc="X dropoff value for ungapped extensions in bits (0.0 invokes default behavior)\nblastn 20, megablast 10, all others 7 [Real]\ndefault = 0.0")), ToolInput(tag="in_x_dropoff_value_final", input_type=Boolean(optional=True), prefix="-Z", doc=InputDocumentation(doc="X dropoff value for final gapped alignment in bits (0.0 invokes default behavior)\nblastn/megablast 100, tblastx 0, all others 25 [Integer]\ndefault = 0")), ToolInput(tag="in_psitblastn_checkpoint_file", input_type=Boolean(optional=True), prefix="-R", doc=InputDocumentation(doc="PSI-TBLASTN checkpoint file [File In]  Optional")), ToolInput(tag="in_megablast_search_default", input_type=Boolean(optional=True), prefix="-n", doc=InputDocumentation(doc="MegaBlast search [T/F]\ndefault = F")), ToolInput(tag="in_location_query_optional", input_type=Boolean(optional=True), prefix="-L", doc=InputDocumentation(doc="Location on query sequence [String]  Optional")), ToolInput(tag="in_multiple_hits_window", input_type=Boolean(optional=True), prefix="-A", doc=InputDocumentation(doc="Multiple Hits window size, default if zero (blastn/megablast 0, all others 40 [Integer]\ndefault = 0")), ToolInput(tag="in_frame_shift_oof", input_type=Boolean(optional=True), prefix="-w", doc=InputDocumentation(doc="Frame shift penalty (OOF algorithm for blastx) [Integer]\ndefault = 0")), ToolInput(tag="in_length_invokes_behavior", input_type=Boolean(optional=True), prefix="-t", doc=InputDocumentation(doc="Length of the largest intron allowed in a translated nucleotide sequence when linking multiple distinct alignments. (0 invokes default behavior; a negative value disables linking.) [Integer]\ndefault = 0")), ToolInput(tag="in_number_concatenated_tblastn", input_type=Boolean(optional=True), prefix="-B", doc=InputDocumentation(doc="Number of concatenated queries, for blastn and tblastn [Integer]  Optional\ndefault = 0")), ToolInput(tag="in_force_use_legacy", input_type=Boolean(optional=True), prefix="-V", doc=InputDocumentation(doc="Force use of the legacy BLAST engine [T/F]  Optional\ndefault = F")), ToolInput(tag="in_use_compositionbased_adjustments", input_type=Boolean(optional=True), prefix="-C", doc=InputDocumentation(doc="Use composition-based score adjustments for blastp or tblastn:\nAs first character:\nD or d: default (equivalent to T)\n0 or F or f: no composition-based statistics\n2 or T or t: Composition-based score adjustments as in Bioinformatics 21:902-911,\n1: Composition-based statistics as in NAR 29:2994-3005, 2001\n2005, conditioned on sequence properties\n3: Composition-based score adjustment as in Bioinformatics 21:902-911,\n2005, unconditionally\nFor programs other than tblastn, must either be absent or be D, F or 0.\nAs second character, if first character is equivalent to 1, 2, or 3:\nU or u: unified p-value combining alignment p-value and compositional p-value in round 1 only")), ToolInput(tag="in_compute_smithwaterman_alignments", input_type=Boolean(optional=True), prefix="-s", doc=InputDocumentation(doc="Compute locally optimal Smith-Waterman alignments (This option is only\navailable for gapped tblastn.) [T/F]\ndefault = F\n"))], outputs=[ToolOutput(tag="out_blast_report_file", output_type=File(optional=True), selector=InputSelector(input_to_select="in_blast_report_file", type_hint=File()), doc=OutputDocumentation(doc="BLAST report Output File [File Out]  Optional\ndefault = stdout")), ToolOutput(tag="out_seqalign_file_optional", output_type=File(optional=True), selector=InputSelector(input_to_select="in_seqalign_file_optional", type_hint=File()), doc=OutputDocumentation(doc="SeqAlign file [File Out]  Optional"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Blastall_V0_1_0().translate("wdl", allow_empty_container=True)

