from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Int, Boolean, String, Array

Vtools_Associate_V0_1_0 = CommandToolBuilder(tool="vtools_associate", base_command=["vtools", "associate"], inputs=[ToolInput(tag="in_jobs", input_type=Int(optional=True), prefix="--jobs", doc=InputDocumentation(doc="Number of processes to carry out association tests.")), ToolInput(tag="in_mpi", input_type=Boolean(optional=True), prefix="-mpi", doc=InputDocumentation(doc="Submit vtools association job to cluster, please check\nbash script.")), ToolInput(tag="in_verbosity", input_type=String(optional=True), prefix="--verbosity", doc=InputDocumentation(doc="Output error and warning (0), info (1), debug (2) and\ntrace (3) information to standard output (default to\n1).")), ToolInput(tag="in_covariates", input_type=Boolean(optional=True), prefix="--covariates", doc=InputDocumentation(doc="[COVARIATES [COVARIATES ...]]\nOptional phenotypes that will be passed to statistical\ntests as covariates. Values of these phenotypes should\nbe integer or float.")), ToolInput(tag="in_var_info", input_type=Boolean(optional=True), prefix="--var_info", doc=InputDocumentation(doc="[VAR_INFO [VAR_INFO ...]], --var-info [VAR_INFO [VAR_INFO ...]]\nOptional variant information fields (e.g. minor allele\nfrequency from 1000 genomes project) that will be\npassed to statistical tests. The fields could be any\nannotation fields of with integer or float values,\nincluding those from used annotation databases (use\n'vtools show fields' to see a list of usable fields).")), ToolInput(tag="in_geno_info", input_type=Boolean(optional=True), prefix="--geno_info", doc=InputDocumentation(doc="[GENO_INFO [GENO_INFO ...]], --geno-info [GENO_INFO [GENO_INFO ...]]\nOptional genotype fields (e.g. quality score of\ngenotype calls, cf. 'vtools show genotypes') that will\nbe passed to statistical tests. Note that the fields\nshould exist for all samples that are tested.")), ToolInput(tag="in_geno_name", input_type=String(optional=True), prefix="--geno_name", doc=InputDocumentation(doc="Field name of genotype, default to 'GT'. If another\nfield name is specified, for example if imputation\nscores are available as 'DS' (dosage), then the given\nfield 'DS' will be used as genotype data for\nassociation analysis.")), ToolInput(tag="in_methods", input_type=Array(t=String(), optional=True), prefix="--methods", doc=InputDocumentation(doc="Method of one or more association tests. Parameters\nfor each method should be specified together as a\nquoted long argument (e.g. --methods 'm --alternative\n2' 'm1 --permute 1000'), although the common method\nparameters can be specified separately, as long as\nthey do not conflict with command arguments. (e.g.\n--methods m1 m2 -p 1000 is equivalent to --methods 'm1\n-p 1000' 'm2 -p 1000'.). You can use command 'vtools\nshow tests' for a list of association tests, and\n'vtools show test TST' for details about a test.\nCustomized association tests can be specified as\nmod_name.test_name where mod_name should be a Python\nmodule (system wide or in the current directory), and\ntest_name should be a subclass of NullTest.")), ToolInput(tag="in_group_variants_separated", input_type=Boolean(optional=True), prefix="-g", doc=InputDocumentation(doc="[GROUP_BY [GROUP_BY ...]], --group_by [GROUP_BY [GROUP_BY ...]], --group-by [GROUP_BY [GROUP_BY ...]]\nGroup variants by fields. If specified, variants will\nbe separated into groups and are tested one by one.")), ToolInput(tag="in_limiting_variants_samples", input_type=Boolean(optional=True), prefix="-s", doc=InputDocumentation(doc="[COND [COND ...]], --samples [COND [COND ...]]\nLimiting variants from samples that match conditions\nthat use columns shown in command 'vtools show sample'\n(e.g. 'aff=1', 'filename like 'MG%''). Each line of\nthe sample table (vtools show samples) is considered\nas samples. If genotype of a physical sample is\nscattered into multiple samples (e.g. imported\nchromosome by chromosome), they should be merged using\ncommand vtools admin.")), ToolInput(tag="in_genotypes", input_type=Boolean(optional=True), prefix="--genotypes", doc=InputDocumentation(doc="[COND [COND ...]]\nLimiting genotypes to those matching conditions that\nuse columns shown in command 'vtools show genotypes'\n(e.g. 'GQ>15'). Genotypes failing such conditions will\nbe regarded as missing genotypes.")), ToolInput(tag="in_discard_samples", input_type=Boolean(optional=True), prefix="--discard_samples", doc=InputDocumentation(doc="[EXPR [EXPR ...]], --discard-samples [EXPR [EXPR ...]]\nDiscard samples that match specified conditions within\neach test group (defined by parameter --group_by).\nCurrently only expressions in the form of '%(NA)>p' is\nprovidedted to remove samples that have more 100*p\npercent of missing values.")), ToolInput(tag="in_discard_variants", input_type=Boolean(optional=True), prefix="--discard_variants", doc=InputDocumentation(doc="[EXPR [EXPR ...]], --discard-variants [EXPR [EXPR ...]]\nDiscard variant sites based on specified conditions\nwithin each test group. Currently only expressions in\nthe form of '%(NA)>p' is provided to remove variant\nsites that have more than 100*p percent of missing\ngenotypes. Note that this filter will be applied after\n'--discard_samples' is applied, if the latter also is\nspecified.")), ToolInput(tag="in_to_db", input_type=String(optional=True), prefix="--to_db", doc=InputDocumentation(doc="Name of a database to which results from association\ntests will be written. Groups with existing results in\nthe database will be ignored unless parameter --force\nis used.")), ToolInput(tag="in_delimiter", input_type=String(optional=True), prefix="--delimiter", doc=InputDocumentation(doc="Delimiter use to separate columns of output. The\ndefault output uses multiple spaces to align columns\nof output. Use '-d,' for csv output, or -d'\t' for\ntab-delimited output.")), ToolInput(tag="in_force", input_type=Boolean(optional=True), prefix="--force", doc=InputDocumentation(doc="Analyze all groups including those that have recorded\nresults in the result database.\n")), ToolInput(tag="in_variants", input_type=String(), position=0, doc=InputDocumentation(doc="Table of variants to be tested.")), ToolInput(tag="in_phenotypes", input_type=String(), position=1, doc=InputDocumentation(doc="A list of phenotypes that will be passed to the\nassociation statistics calculator. Currently only a\nsingle phenotype is allowed."))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Vtools_Associate_V0_1_0().translate("wdl", allow_empty_container=True)

