version 1.0

task RiboCode {
  input {
    Directory? an_not_dir
    File? config_file
    String? longest_orf
    Float? pval_cut_off
    String? start_codon
    String? alt_start_codons
    String? stop_codon
    Int? transl_table
    Int? min_aa_length
    File? output_gtf
    File? output_bed
    File? output_name
  }
  command <<<
    RiboCode \
      ~{if defined(an_not_dir) then ("--annot_dir " +  '"' + an_not_dir + '"') else ""} \
      ~{if defined(config_file) then ("--config_file " +  '"' + config_file + '"') else ""} \
      ~{if defined(longest_orf) then ("--longest-orf " +  '"' + longest_orf + '"') else ""} \
      ~{if defined(pval_cut_off) then ("--pval-cutoff " +  '"' + pval_cut_off + '"') else ""} \
      ~{if defined(start_codon) then ("--start_codon " +  '"' + start_codon + '"') else ""} \
      ~{if defined(alt_start_codons) then ("--alt_start_codons " +  '"' + alt_start_codons + '"') else ""} \
      ~{if defined(stop_codon) then ("--stop_codon " +  '"' + stop_codon + '"') else ""} \
      ~{if defined(transl_table) then ("--transl_table " +  '"' + transl_table + '"') else ""} \
      ~{if defined(min_aa_length) then ("--min-AA-length " +  '"' + min_aa_length + '"') else ""} \
      ~{if (output_gtf) then "--output-gtf" else ""} \
      ~{if (output_bed) then "--output-bed" else ""} \
      ~{if defined(output_name) then ("--output-name " +  '"' + output_name + '"') else ""}
  >>>
  runtime {
    docker: "None"
  }
  parameter_meta {
    an_not_dir: "transcripts annotation directory, generated by\\nprepare_transcripts."
    config_file: "list bam file and P-sites information in this file,\\nplease refer to the example file in data folder."
    longest_orf: "Default: yes, the region from most distal AUG to stop\\nwas defined as an ORF. If set to no , the position of\\nstart codon will be automatically determined by\\nprogram."
    pval_cut_off: "P-value cutoff for ORF filtering, default 0.05"
    start_codon: "The canonical start codon. default: ATG"
    alt_start_codons: "The alternative start codon, such as CTG,GTG, default:\\nNone. Multiple codons should be separated by comma."
    stop_codon: "Stop codon, default: TAA,TAG,TGA"
    transl_table: "ORF translation table(Default: 1). Assign the correct\\ngenetic code based on your organism, [please refer: ht\\ntps://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi\\n]"
    min_aa_length: "The minimal length of predicted peptides,default 20"
    output_gtf: "output the gtf file of predicted ORFs"
    output_bed: "output the bed file of predicted ORFs"
    output_name: "output file name, default: final_result"
  }
  output {
    File out_stdout = stdout()
    File out_output_gtf = "${in_output_gtf}"
    File out_output_bed = "${in_output_bed}"
    File out_output_name = "${in_output_name}"
  }
}