from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Directory, Boolean, File, String, Int, Float

Coronaspades_Py_V0_1_0 = CommandToolBuilder(tool="coronaspades.py", base_command=["coronaspades.py"], inputs=[ToolInput(tag="in_directory_store_files", input_type=Directory(optional=True), prefix="-o", doc=InputDocumentation(doc="directory to store all the resulting files (required)")), ToolInput(tag="in_ion_torrent", input_type=Boolean(optional=True), prefix="--iontorrent", doc=InputDocumentation(doc="this flag is required for IonTorrent data")), ToolInput(tag="in_test", input_type=Boolean(optional=True), prefix="--test", doc=InputDocumentation(doc="runs SPAdes on toy dataset")), ToolInput(tag="in_file_interlaced_pairedend", input_type=File(optional=True), prefix="--12", doc=InputDocumentation(doc="file with interlaced forward and reverse paired-end reads")), ToolInput(tag="in_file_forward_pairedend", input_type=File(optional=True), prefix="-1", doc=InputDocumentation(doc="file with forward paired-end reads")), ToolInput(tag="in_file_reverse_pairedend", input_type=File(optional=True), prefix="-2", doc=InputDocumentation(doc="file with reverse paired-end reads")), ToolInput(tag="in_file_unpaired_reads", input_type=File(optional=True), prefix="-s", doc=InputDocumentation(doc="file with unpaired reads")), ToolInput(tag="in_merged", input_type=File(optional=True), prefix="--merged", doc=InputDocumentation(doc="file with merged forward and reverse paired-end reads")), ToolInput(tag="in_pe_one_two", input_type=Boolean(optional=True), prefix="--pe-12", doc=InputDocumentation(doc="<#> <filename>      file with interlaced reads for paired-end library number <#>.\nOlder deprecated syntax is -pe<#>-12 <filename>")), ToolInput(tag="in_pe_one", input_type=Boolean(optional=True), prefix="--pe-1", doc=InputDocumentation(doc="<#> <filename>       file with forward reads for paired-end library number <#>.\nOlder deprecated syntax is -pe<#>-1 <filename>")), ToolInput(tag="in_pe_two", input_type=Boolean(optional=True), prefix="--pe-2", doc=InputDocumentation(doc="<#> <filename>       file with reverse reads for paired-end library number <#>.\nOlder deprecated syntax is -pe<#>-2 <filename>")), ToolInput(tag="in_pe_s", input_type=Boolean(optional=True), prefix="--pe-s", doc=InputDocumentation(doc="<#> <filename>       file with unpaired reads for paired-end library number <#>.\nOlder deprecated syntax is -pe<#>-s <filename>")), ToolInput(tag="in_pe_m", input_type=Boolean(optional=True), prefix="--pe-m", doc=InputDocumentation(doc="<#> <filename>       file with merged reads for paired-end library number <#>.\nOlder deprecated syntax is -pe<#>-m <filename>")), ToolInput(tag="in_pe_or", input_type=Boolean(optional=True), prefix="--pe-or", doc=InputDocumentation(doc="<#> <or>            orientation of reads for paired-end library number <#>\n(<or> = fr, rf, ff).\nOlder deprecated syntax is -pe<#>-<or>")), ToolInput(tag="in_mp_one_two", input_type=Boolean(optional=True), prefix="--mp-12", doc=InputDocumentation(doc="<#> <filename>      file with interlaced reads for mate-pair library number <#>.\nOlder deprecated syntax is -mp<#>-12 <filename>")), ToolInput(tag="in_mp_one", input_type=Boolean(optional=True), prefix="--mp-1", doc=InputDocumentation(doc="<#> <filename>       file with forward reads for mate-pair library number <#>.\nOlder deprecated syntax is -mp<#>-1 <filename>")), ToolInput(tag="in_mp_two", input_type=Boolean(optional=True), prefix="--mp-2", doc=InputDocumentation(doc="<#> <filename>       file with reverse reads for mate-pair library number <#>.\nOlder deprecated syntax is -mp<#>-2 <filename>")), ToolInput(tag="in_mp_s", input_type=Boolean(optional=True), prefix="--mp-s", doc=InputDocumentation(doc="<#> <filename>       file with unpaired reads for mate-pair library number <#>.\nOlder deprecated syntax is -mp<#>-s <filename>")), ToolInput(tag="in_mp_or", input_type=Boolean(optional=True), prefix="--mp-or", doc=InputDocumentation(doc="<#> <or>            orientation of reads for mate-pair library number <#>\n(<or> = fr, rf, ff).\nOlder deprecated syntax is -mp<#>-<or>")), ToolInput(tag="in_hq_mp_one_two", input_type=Boolean(optional=True), prefix="--hqmp-12", doc=InputDocumentation(doc="<#> <filename>    file with interlaced reads for high-quality mate-pair library number <#>.\nOlder deprecated syntax is -hqmp<#>-12 <filename>")), ToolInput(tag="in_hq_mp_one", input_type=Boolean(optional=True), prefix="--hqmp-1", doc=InputDocumentation(doc="<#> <filename>     file with forward reads for high-quality mate-pair library number <#>.\nOlder deprecated syntax is -hqmp<#>-1 <filename>")), ToolInput(tag="in_hq_mp_two", input_type=Boolean(optional=True), prefix="--hqmp-2", doc=InputDocumentation(doc="<#> <filename>     file with reverse reads for high-quality mate-pair library number <#>.\nOlder deprecated syntax is -hqmp<#>-2 <filename>")), ToolInput(tag="in_hq_mp_s", input_type=Boolean(optional=True), prefix="--hqmp-s", doc=InputDocumentation(doc="<#> <filename>     file with unpaired reads for high-quality mate-pair library number <#>.\nOlder deprecated syntax is -hqmp<#>-s <filename>")), ToolInput(tag="in_hq_mp_or", input_type=Boolean(optional=True), prefix="--hqmp-or", doc=InputDocumentation(doc="<#> <or>          orientation of reads for high-quality mate-pair library number <#>\n(<or> = fr, rf, ff).\nOlder deprecated syntax is -hqmp<#>-<or>")), ToolInput(tag="in_sanger", input_type=File(optional=True), prefix="--sanger", doc=InputDocumentation(doc="file with Sanger reads")), ToolInput(tag="in_pac_bio", input_type=File(optional=True), prefix="--pacbio", doc=InputDocumentation(doc="file with PacBio reads")), ToolInput(tag="in_nano_pore", input_type=File(optional=True), prefix="--nanopore", doc=InputDocumentation(doc="file with Nanopore reads")), ToolInput(tag="in_trusted_contigs", input_type=File(optional=True), prefix="--trusted-contigs", doc=InputDocumentation(doc="file with trusted contigs")), ToolInput(tag="in_untrusted_contigs", input_type=File(optional=True), prefix="--untrusted-contigs", doc=InputDocumentation(doc="file with untrusted contigs")), ToolInput(tag="in_only_error_correction", input_type=Boolean(optional=True), prefix="--only-error-correction", doc=InputDocumentation(doc="runs only read error correction (without assembling)")), ToolInput(tag="in_only_assembler", input_type=Boolean(optional=True), prefix="--only-assembler", doc=InputDocumentation(doc="runs only assembling (without read error correction)")), ToolInput(tag="in_careful", input_type=Boolean(optional=True), prefix="--careful", doc=InputDocumentation(doc="tries to reduce number of mismatches and short indels")), ToolInput(tag="in_checkpoints", input_type=String(optional=True), prefix="--checkpoints", doc=InputDocumentation(doc="save intermediate check-points ('last', 'all')")), ToolInput(tag="in_continue", input_type=Boolean(optional=True), prefix="--continue", doc=InputDocumentation(doc="continue run from the last available check-point (only -o should be specified)")), ToolInput(tag="in_restart_from", input_type=Int(optional=True), prefix="--restart-from", doc=InputDocumentation(doc="restart run with updated options and from the specified check-point\n('ec', 'as', 'k<int>', 'mc', 'last')")), ToolInput(tag="in_disable_gzip_output", input_type=Boolean(optional=True), prefix="--disable-gzip-output", doc=InputDocumentation(doc="forces error correction not to compress the corrected reads")), ToolInput(tag="in_disable_rr", input_type=Boolean(optional=True), prefix="--disable-rr", doc=InputDocumentation(doc="disables repeat resolution stage of assembling")), ToolInput(tag="in_dataset", input_type=File(optional=True), prefix="--dataset", doc=InputDocumentation(doc="file with dataset description in YAML format")), ToolInput(tag="in_threads", input_type=Int(optional=True), prefix="--threads", doc=InputDocumentation(doc="number of threads. [default: 16]")), ToolInput(tag="in_memory", input_type=Int(optional=True), prefix="--memory", doc=InputDocumentation(doc="RAM limit for SPAdes in Gb (terminates if exceeded). [default: 250]")), ToolInput(tag="in_tmp_dir", input_type=Directory(optional=True), prefix="--tmp-dir", doc=InputDocumentation(doc="directory for temporary files. [default: <output_dir>/tmp]")), ToolInput(tag="in_list_kmer_sizes", input_type=Int(optional=True), prefix="-k", doc=InputDocumentation(doc="[<int> ...]        list of k-mer sizes (must be odd and less than 128)\n[default: 'auto']")), ToolInput(tag="in_cov_cut_off", input_type=Float(optional=True), prefix="--cov-cutoff", doc=InputDocumentation(doc="coverage cutoff value (a positive float number, or 'auto', or 'off')\n[default: 'off']")), ToolInput(tag="in_phred_offset", input_type=Int(optional=True), prefix="--phred-offset", doc=InputDocumentation(doc="PHRED quality offset in the input reads (33 or 64),\n[default: auto-detect]")), ToolInput(tag="in_custom_hmms", input_type=Directory(optional=True), prefix="--custom-hmms", doc=InputDocumentation(doc="directory with custom hmms that replace default ones,\n[default: None]\n")), ToolInput(tag="in_spades_do_tpy", input_type=String(), position=0, doc=InputDocumentation(doc=""))], outputs=[], container="quay.io/biocontainers/spades:3.15.2--h633aebb_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Coronaspades_Py_V0_1_0().translate("wdl")

