from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean

Hiddendomains_V0_1_0 = CommandToolBuilder(tool="hiddenDomains", base_command=["hiddenDomains"], inputs=[ToolInput(tag="in_width_bin_default", input_type=Boolean(optional=True), prefix="-b", doc=InputDocumentation(doc="BIN_WIDTH\nThe width of the bin. Default is 1000bp.")), ToolInput(tag="in_input_files_isare", input_type=Boolean(optional=True), prefix="-B", doc=InputDocumentation(doc="The input file(s) is(are) in BED format (the default is BAM). NOTE: All\nread files have to have to same format (either BED or BAM). Use\nbinReads.pl as a stand alone program if you have a more complicated set up.")), ToolInput(tag="in_controlreadsa_bed_file", input_type=Boolean(optional=True), prefix="-c", doc=InputDocumentation(doc="ControlReads\nA BED or BAM file that contains aligned read reads. Use the -B option to\nspeicfy BED format. BAM format is the default.")), ToolInput(tag="in_controlreadsbinsif_you_already", input_type=Boolean(optional=True), prefix="-C", doc=InputDocumentation(doc="ControlReadsBins\nIf you have already binned your control reads, you can pass them in\nwith this option and they will be used directly.")), ToolInput(tag="in_chrominfotxtif_you_get", input_type=Boolean(optional=True), prefix="-g", doc=InputDocumentation(doc="ChromInfo.txt\nIf you get an out of bounds error when uploading a bed file to the UCSC\ngenome browser, you can use this option to trim the bounds to the size of\nthe chromosomes.\nChromInfo.txt should be a tab delimited file with the chromosome names\nin the first column and their sizes in the second column.  Here is an\nexample ChromInfo.txt file for a genome with 3 chromosomes:\nchr1    197195432\nchr2    181748087\nchr3    159599783")), ToolInput(tag="in_outputprefixhiddendomains_generates_files", input_type=Boolean(optional=True), prefix="-o", doc=InputDocumentation(doc="OutputPrefix\nhiddenDomains generates four or five files with names that start with\nOutputPrefix.\n1) '_domains.txt': A text file with all of the enriched domains and\nposterior probabilities.\n2) '_vis.bed': A BED file for visualization, which contains one line per\nsignificantly enriched bin - this allows for color coding based on the\nposterior probability.\n3) '_analysis.bed': The second BED file is for analysis, and this merges\nall consecutive bins with posterior probabiliites greater than\nMIN_POSTERIOR (as specified with the -p option) or the default value,\n0 - which merges all consecutive significat bins.\n4) '_treatment_bins.txt': A file with the read counts per bin.\n5) '_control_bins.txt': A file with the read counts per bin.")), ToolInput(tag="in_minposteriortoss_parts_minposterior", input_type=Boolean(optional=True), prefix="-p", doc=InputDocumentation(doc="MIN_POSTERIOR\nToss out parts of domains that have posterior values that are less\nthan MIN_POSTERIOR. You can set this to any value you want, but for\nreference, domainsToBed bins according the following scale:\n>= 0.9\n0.9 > posterior >= 0.8\n0.8 > posterior >= 0.7\n0.7 > posterior >= min posterior for significance\nThe default value is 0; everything is merged by default.")), ToolInput(tag="in_minmapqthe_minimum_mapq", input_type=Boolean(optional=True), prefix="-q", doc=InputDocumentation(doc="MIN_MAPQ\nThe minimum MAPQ score. Default is 30.")), ToolInput(tag="in_treatmentreadsa_bed_bam", input_type=Boolean(optional=True), prefix="-t", doc=InputDocumentation(doc="TreatmentReads\nA BED or BAM file that contains aligned read reads. Use the -B option to\nspeicfy BED format. BAM format is the default.\n")), ToolInput(tag="in_more_options", input_type=Boolean(optional=True), prefix="-MORE_OPTIONS", doc=InputDocumentation(doc="")), ToolInput(tag="in_options", input_type=Boolean(optional=True), prefix="-OPTIONS", doc=InputDocumentation(doc=""))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Hiddendomains_V0_1_0().translate("wdl", allow_empty_container=True)

