from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean

Malt_Build_V0_1_0 = CommandToolBuilder(tool="malt_build", base_command=["malt-build"], inputs=[ToolInput(tag="in_input", input_type=Boolean(optional=True), prefix="--input", doc=InputDocumentation(doc="[string(s)]              Input reference files in FastA format (or specify a single directory). Mandatory option.")), ToolInput(tag="in_sequence_type", input_type=Boolean(optional=True), prefix="--sequenceType", doc=InputDocumentation(doc="[string]          Sequence type. Mandatory option. Legal values: DNA, Protein")), ToolInput(tag="in_input_gff", input_type=Boolean(optional=True), prefix="--inputGFF", doc=InputDocumentation(doc="[string(s)]        Files that provide CDS annotations of DNA input files in GFF format (or specify a single directory).")), ToolInput(tag="in_index", input_type=Boolean(optional=True), prefix="--index", doc=InputDocumentation(doc="[string]                 Name of index directory. Mandatory option.")), ToolInput(tag="in_threads", input_type=Boolean(optional=True), prefix="--threads", doc=InputDocumentation(doc="[number]               Number of worker threads. Default value: 8.")), ToolInput(tag="in_step", input_type=Boolean(optional=True), prefix="--step", doc=InputDocumentation(doc="[number]                 Step size used to advance seed; a value greater than 1 will reduce index size, but also sensitivity. Default value: 1.")), ToolInput(tag="in_shapes", input_type=Boolean(optional=True), prefix="--shapes", doc=InputDocumentation(doc="[string(s)]            Seed shape(s). Default value(s): default.")), ToolInput(tag="in_max_hits_per_seed", input_type=Boolean(optional=True), prefix="--maxHitsPerSeed", doc=InputDocumentation(doc="[number]       Maximum number of hits per seed. Default value: 1000.")), ToolInput(tag="in_protein_reduct", input_type=Boolean(optional=True), prefix="--proteinReduct", doc=InputDocumentation(doc="[string]        Name or definition of protein alphabet reduction (BLOSUM50_10,BLOSUM50_11,BLOSUM50_15,BLOSUM50_4,BLOSUM50_8,DIAMOND_11,GBMR4,HSDM17,MALT_10,SDM12,UNREDUCED). Default value: DIAMOND_11.")), ToolInput(tag="in_gi_two_eggnog", input_type=Boolean(optional=True), prefix="--gi2eggnog", doc=InputDocumentation(doc="[string]      GI-to-EGGNOG mapping file (deprecated).")), ToolInput(tag="in_acc_two_eggnog", input_type=Boolean(optional=True), prefix="--acc2eggnog", doc=InputDocumentation(doc="[string]     Accession-to-EGGNOG mapping file.")), ToolInput(tag="in_syn_two_eggnog", input_type=Boolean(optional=True), prefix="--syn2eggnog", doc=InputDocumentation(doc="[string]     Synonyms-to-EGGNOG mapping file.")), ToolInput(tag="in_gi_two_interpro_two_go", input_type=Boolean(optional=True), prefix="--gi2interpro2go", doc=InputDocumentation(doc="[string]   GI-to-INTERPRO2GO mapping file (deprecated).")), ToolInput(tag="in_acc_two_interpro_two_go", input_type=Boolean(optional=True), prefix="--acc2interpro2go", doc=InputDocumentation(doc="[string]   Accession-to-INTERPRO2GO mapping file.")), ToolInput(tag="in_syn_two_interpro_two_go", input_type=Boolean(optional=True), prefix="--syn2interpro2go", doc=InputDocumentation(doc="[string]   Synonyms-to-INTERPRO2GO mapping file.")), ToolInput(tag="in_gi_two_kegg", input_type=Boolean(optional=True), prefix="--gi2kegg", doc=InputDocumentation(doc="[string]          GI-to-KEGG mapping file (deprecated).")), ToolInput(tag="in_acc_two_kegg", input_type=Boolean(optional=True), prefix="--acc2kegg", doc=InputDocumentation(doc="[string]         Accession-to-KEGG mapping file.")), ToolInput(tag="in_syn_two_kegg", input_type=Boolean(optional=True), prefix="--syn2kegg", doc=InputDocumentation(doc="[string]         Synonyms-to-KEGG mapping file.")), ToolInput(tag="in_gi_two_seed", input_type=Boolean(optional=True), prefix="--gi2seed", doc=InputDocumentation(doc="[string]          GI-to-SEED mapping file (deprecated).")), ToolInput(tag="in_acc_two_seed", input_type=Boolean(optional=True), prefix="--acc2seed", doc=InputDocumentation(doc="[string]         Accession-to-SEED mapping file.")), ToolInput(tag="in_syn_two_seed", input_type=Boolean(optional=True), prefix="--syn2seed", doc=InputDocumentation(doc="[string]         Synonyms-to-SEED mapping file.")), ToolInput(tag="in_gi_two_taxonomy", input_type=Boolean(optional=True), prefix="--gi2taxonomy", doc=InputDocumentation(doc="[string]   GI-to-Taxonomy mapping file (deprecated).")), ToolInput(tag="in_acc_two_taxonomy", input_type=Boolean(optional=True), prefix="--acc2taxonomy", doc=InputDocumentation(doc="[string]   Accession-to-Taxonomy mapping file.")), ToolInput(tag="in_syn_two_taxonomy", input_type=Boolean(optional=True), prefix="--syn2taxonomy", doc=InputDocumentation(doc="[string]   Synonyms-to-Taxonomy mapping file.")), ToolInput(tag="in_parse_tax_on_names", input_type=Boolean(optional=True), prefix="--parseTaxonNames", doc=InputDocumentation(doc="Parse taxon names. Default value: true.")), ToolInput(tag="in_no_fun", input_type=Boolean(optional=True), prefix="--noFun", doc=InputDocumentation(doc="Turn off functional classifications for provided mapping files (set this when using GFF files for DNA references). Default value: false.")), ToolInput(tag="in_first_word_is_accession", input_type=Boolean(optional=True), prefix="--firstWordIsAccession", doc=InputDocumentation(doc="First word in reference header is accession number. Default value: true.")), ToolInput(tag="in_accession_tags", input_type=Boolean(optional=True), prefix="--accessionTags", doc=InputDocumentation(doc="[string(s)]   List of accession tags. Default value(s): gb| ref|.")), ToolInput(tag="in_first_word_only", input_type=Boolean(optional=True), prefix="--firstWordOnly", doc=InputDocumentation(doc="Save only first word of reference header. Default value: false.")), ToolInput(tag="in_random", input_type=Boolean(optional=True), prefix="--random", doc=InputDocumentation(doc="[number]              Random number generator seed. Default value: 666.")), ToolInput(tag="in_hash_scale_factor", input_type=Boolean(optional=True), prefix="--hashScaleFactor", doc=InputDocumentation(doc="[number]     Hash table scale factor. Default value: 0.9.")), ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="--verbose", doc=InputDocumentation(doc="Echo commandline options and be verbose. Default value: false."))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Malt_Build_V0_1_0().translate("wdl", allow_empty_container=True)

