from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Array, String, Boolean, Int, File

Nanoraw_Plot_Kmer_V0_1_0 = CommandToolBuilder(tool="nanoraw_plot_kmer", base_command=["nanoraw", "plot_kmer"], inputs=[ToolInput(tag="in_base_call_subgroups", input_type=Array(t=String(), optional=True), prefix="--basecall-subgroups", doc=InputDocumentation(doc="FAST5 subgroup (under Analyses/[corrected-group])\nwhere individual template and/or complement reads are\nstored. Default: ['BaseCalled_template']")), ToolInput(tag="in_two_d", input_type=Boolean(optional=True), prefix="--2d", doc=InputDocumentation(doc="Input contains 2D reads. Equivalent to `--basecall-\nsubgroups BaseCalled_template BaseCalled_complement`")), ToolInput(tag="in_fast_five_based_irs", input_type=Array(t=Int(), optional=True), prefix="--fast5-basedirs", doc=InputDocumentation(doc="Directories containing fast5 files.")), ToolInput(tag="in_upstream_bases", input_type=String(optional=True), prefix="--upstream-bases", doc=InputDocumentation(doc="Upstream bases in k-mer. Should be one of {0,1,2,3}.\nDefault: 1")), ToolInput(tag="in_downstream_bases", input_type=String(optional=True), prefix="--downstream-bases", doc=InputDocumentation(doc="Downstream bases in k-mer. Should be one of {0,1,2,3}.\nDefault: 2")), ToolInput(tag="in_read_mean", input_type=Boolean(optional=True), prefix="--read-mean", doc=InputDocumentation(doc="Plot kmer event means across reads as opposed to each")), ToolInput(tag="in_corrected_group", input_type=Int(optional=True), prefix="--corrected-group", doc=InputDocumentation(doc="FAST5 group to access/plot created by\ngenome_resquiggle script. Default:\nRawGenomeCorrected_000")), ToolInput(tag="in_pdf_filename", input_type=File(optional=True), prefix="--pdf-filename", doc=InputDocumentation(doc="PDF filename to store plot(s). Default:\nNanopore_kmer_distribution.pdf")), ToolInput(tag="in_num_reads", input_type=Int(optional=True), prefix="--num-reads", doc=InputDocumentation(doc="Number of reads to plot (one region per read).\nDefault: 500")), ToolInput(tag="in_r_data_filename", input_type=File(optional=True), prefix="--r-data-filename", doc=InputDocumentation(doc="Filename to save R data structure. Defualt: Don't save")), ToolInput(tag="in_dont_plot", input_type=Boolean(optional=True), prefix="--dont-plot", doc=InputDocumentation(doc="Don't actually plot the result. Useful when you only\nwant the R data file.")), ToolInput(tag="in_quiet", input_type=Boolean(optional=True), prefix="--quiet", doc=InputDocumentation(doc="Don't print status information.")), ToolInput(tag="in_event_dot", input_type=String(), position=0, doc=InputDocumentation(doc="--num-kmer-threshold NUM_KMER_THRESHOLD"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Nanoraw_Plot_Kmer_V0_1_0().translate("wdl", allow_empty_container=True)

