from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, File, String, Directory

Comparemotifs_Pl_V0_1_0 = CommandToolBuilder(tool="compareMotifs.pl", base_command=["compareMotifs.pl"], inputs=[ToolInput(tag="in_reduce_thresh", input_type=Boolean(optional=True), prefix="-reduceThresh", doc=InputDocumentation(doc="<#> (Similarity Threshold used to remove similar motifs, default: 0.6)")), ToolInput(tag="in_match_thresh", input_type=Boolean(optional=True), prefix="-matchThresh", doc=InputDocumentation(doc="<#|T#> (Similarity Threshold to report alignments with known motifs.  Use T#\n[i.e. T10] to match the top # known motifs regardless of similarity, default: T10)")), ToolInput(tag="in_known", input_type=File(optional=True), prefix="-known", doc=InputDocumentation(doc="(Library of known motifs to compare to input motifs, default:\n'/usr/local/share/homer-4.11-2/.//data/knownTFs//all/all.motifs'")), ToolInput(tag="in_p_value", input_type=Boolean(optional=True), prefix="-pvalue", doc=InputDocumentation(doc="<#> (Remove Motifs with a p-value higher than #, default: keep all)")), ToolInput(tag="in_remove_motifs_less_default", input_type=Boolean(optional=True), prefix="-F", doc=InputDocumentation(doc="<#> (Remove Motifs with fold enrichment [target%/bg%] less than <#>, default: keep all)")), ToolInput(tag="in_info", input_type=Boolean(optional=True), prefix="-info", doc=InputDocumentation(doc="<#> (Remove Motifs with information content less than #, default: not used)")), ToolInput(tag="in_remove_motifs_background_default", input_type=Boolean(optional=True), prefix="-b", doc=InputDocumentation(doc="<#> (Remove Motifs with background percentage less than #, default: not used)")), ToolInput(tag="in_mint", input_type=Boolean(optional=True), prefix="-minT", doc=InputDocumentation(doc="<#> (Remove Motifs with less than # number of target instances, default: not used)")), ToolInput(tag="in_site_reduce", input_type=Boolean(optional=True), prefix="-siteReduce", doc=InputDocumentation(doc="(If homer2 known with -siteReduce was used, motifs will be reduced by\ntheir calculated co-occurence.")), ToolInput(tag="in_stat", input_type=String(optional=True), prefix="-stat", doc=InputDocumentation(doc="(Stat used to compute matrix similarity.\ndefault: correlation)")), ToolInput(tag="in_bits", input_type=Boolean(optional=True), prefix="-bits", doc=InputDocumentation(doc="(scale logos to bit content, default present nucleotide percentage)")), ToolInput(tag="in_rna", input_type=Boolean(optional=True), prefix="-rna", doc=InputDocumentation(doc="(output RNA motifs, use RNA motif/miRNA database for comparison)")), ToolInput(tag="in_norev_opp", input_type=Boolean(optional=True), prefix="-norevopp", doc=InputDocumentation(doc="(do not check for matches on reverse strand)")), ToolInput(tag="in_reduced_motifs", input_type=File(optional=True), prefix="-reducedMotifs", doc=InputDocumentation(doc="(will remove redundant motifs, output unique ones, and then quit)")), ToolInput(tag="in_matrix", input_type=File(optional=True), prefix="-matrix", doc=InputDocumentation(doc="(will compute all pairwise similarity scores to matrix, then quit)")), ToolInput(tag="in_no_facts", input_type=Boolean(optional=True), prefix="-nofacts", doc=InputDocumentation(doc="(leave out the humor)")), ToolInput(tag="in_dbview", input_type=Boolean(optional=True), prefix="-dbview", doc=InputDocumentation(doc="(internal option )")), ToolInput(tag="in_db_table", input_type=File(optional=True), prefix="-dbTable", doc=InputDocumentation(doc="(information to use in dbview, internal option)")), ToolInput(tag="in_cpu", input_type=Boolean(optional=True), prefix="-cpu", doc=InputDocumentation(doc="<#> (number of CPUs to use)")), ToolInput(tag="in_basic", input_type=Boolean(optional=True), prefix="-basic", doc=InputDocumentation(doc="(don't compare to known motifs or print similar motifs)")), ToolInput(tag="in_seq_logo", input_type=Boolean(optional=True), prefix="-seqlogo", doc=InputDocumentation(doc="(revert to using seqlogo/weblogo/ghostscript, by default uses SVG now)")), ToolInput(tag="in_motifs_file", input_type=File(), position=0, doc=InputDocumentation(doc="")), ToolInput(tag="in_output_directory", input_type=Directory(), position=1, doc=InputDocumentation(doc=""))], outputs=[ToolOutput(tag="out_reduced_motifs", output_type=File(optional=True), selector=InputSelector(input_to_select="in_reduced_motifs", type_hint=File()), doc=OutputDocumentation(doc="(will remove redundant motifs, output unique ones, and then quit)")), ToolOutput(tag="out_matrix", output_type=File(optional=True), selector=InputSelector(input_to_select="in_matrix", type_hint=File()), doc=OutputDocumentation(doc="(will compute all pairwise similarity scores to matrix, then quit)")), ToolOutput(tag="out_output_directory", output_type=Directory(), selector=InputSelector(input_to_select="in_output_directory", type_hint=File()), doc=OutputDocumentation(doc=""))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Comparemotifs_Pl_V0_1_0().translate("wdl", allow_empty_container=True)

