from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, Int, Float, File, Directory, String

Checkm_Lineage_Wf_V0_1_0 = CommandToolBuilder(tool="checkm_lineage_wf", base_command=["checkm", "lineage_wf"], inputs=[ToolInput(tag="in_reduced_tree", input_type=Boolean(optional=True), prefix="--reduced_tree", doc=InputDocumentation(doc="use reduced tree (requires <16GB of memory) for determining lineage of each bin")), ToolInput(tag="in_ali", input_type=Boolean(optional=True), prefix="--ali", doc=InputDocumentation(doc="generate HMMER alignment file for each bin")), ToolInput(tag="in_nt", input_type=Boolean(optional=True), prefix="--nt", doc=InputDocumentation(doc="generate nucleotide gene sequences for each bin")), ToolInput(tag="in_genes", input_type=Boolean(optional=True), prefix="--genes", doc=InputDocumentation(doc="bins contain genes as amino acids instead of nucleotide contigs")), ToolInput(tag="in_unique", input_type=Int(optional=True), prefix="--unique", doc=InputDocumentation(doc="minimum number of unique phylogenetic markers required to use lineage-specific marker set (default: 10)")), ToolInput(tag="in_multi", input_type=Int(optional=True), prefix="--multi", doc=InputDocumentation(doc="maximum number of multi-copy phylogenetic markers before defaulting to domain-level marker set (default: 10)")), ToolInput(tag="in_force_domain", input_type=Boolean(optional=True), prefix="--force_domain", doc=InputDocumentation(doc="use domain-level sets for all bins")), ToolInput(tag="in_no_refinement", input_type=Boolean(optional=True), prefix="--no_refinement", doc=InputDocumentation(doc="do not perform lineage-specific marker set refinement")), ToolInput(tag="in_individual_markers", input_type=Boolean(optional=True), prefix="--individual_markers", doc=InputDocumentation(doc="treat marker as independent (i.e., ignore co-located set structure)")), ToolInput(tag="in_skip_adj_correction", input_type=Boolean(optional=True), prefix="--skip_adj_correction", doc=InputDocumentation(doc="do not exclude adjacent marker genes when estimating contamination")), ToolInput(tag="in_skip_pseudogene_correction", input_type=Boolean(optional=True), prefix="--skip_pseudogene_correction", doc=InputDocumentation(doc="skip identification and filtering of pseudogenes")), ToolInput(tag="in_aai_strain", input_type=Float(optional=True), prefix="--aai_strain", doc=InputDocumentation(doc="AAI threshold used to identify strain heterogeneity (default: 0.9)")), ToolInput(tag="in_alignment_file", input_type=File(optional=True), prefix="--alignment_file", doc=InputDocumentation(doc="produce file showing alignment of multi-copy genes and their AAI identity")), ToolInput(tag="in_ignore_thresholds", input_type=Boolean(optional=True), prefix="--ignore_thresholds", doc=InputDocumentation(doc="ignore model-specific score thresholds")), ToolInput(tag="in_e_value", input_type=Float(optional=True), prefix="--e_value", doc=InputDocumentation(doc="e-value cut off (default: 1e-10)")), ToolInput(tag="in_length", input_type=Int(optional=True), prefix="--length", doc=InputDocumentation(doc="percent overlap between target and query (default: 0.7)")), ToolInput(tag="in_file", input_type=File(optional=True), prefix="--file", doc=InputDocumentation(doc="print results to file (default: stdout)")), ToolInput(tag="in_tab_table", input_type=Boolean(optional=True), prefix="--tab_table", doc=InputDocumentation(doc="print tab-separated values table")), ToolInput(tag="in_extension", input_type=Directory(optional=True), prefix="--extension", doc=InputDocumentation(doc="extension of bins (other files in directory are ignored) (default: fna)")), ToolInput(tag="in_threads", input_type=Int(optional=True), prefix="--threads", doc=InputDocumentation(doc="number of threads (default: 1)")), ToolInput(tag="in_p_placer_threads", input_type=Int(optional=True), prefix="--pplacer_threads", doc=InputDocumentation(doc="number of threads used by pplacer (memory usage increases linearly with additional threads) (default: 1)")), ToolInput(tag="in_quiet", input_type=Boolean(optional=True), prefix="--quiet", doc=InputDocumentation(doc="suppress console output")), ToolInput(tag="in_tmpdir", input_type=Directory(optional=True), prefix="--tmpdir", doc=InputDocumentation(doc="specify an alternative directory for temporary files")), ToolInput(tag="in_bin_dir", input_type=String(), position=0, doc=InputDocumentation(doc="directory containing bins (fasta format)")), ToolInput(tag="in_output_dir", input_type=String(), position=1, doc=InputDocumentation(doc="directory to write output files"))], outputs=[], container="quay.io/biocontainers/checkm-genome:1.1.3--py_1", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Checkm_Lineage_Wf_V0_1_0().translate("wdl")

