from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import String, Int, Float, Boolean, File

Slim_Graph_Py_V0_1_0 = CommandToolBuilder(tool="slim_graph.py", base_command=["slim_graph.py"], inputs=[ToolInput(tag="in_followed_fungusmt_embryophyta", input_type=String(optional=True), prefix="-F", doc=InputDocumentation(doc="followed with mode embplant_pt, other_pt, embplant_mt,\nembplant_nr, animal_mt, fungus_mt, fungus_nr (which\nmeans embryophyta plastome, non-embryophyta plastome,\nplant mitochondrion, plant nuclear ribosomal RNA,\nanimal mitochondrion, fungus mitochondrion, fungus\nnuclear ribosomal RNA separately), or a combination of\nabove split by comma(s) (corresponds to certain\narguments as following listed).\n------------------------------------------------------\nembplant_pt      ' --include-priority\n/root/.GetOrganelle/LabelDatabase/embplant_pt.fasta'\n------------------------------------------------------\nother_pt         ' --include-priority\n/root/.GetOrganelle/LabelDatabase/other_pt.fasta'\n------------------------------------------------------\nembplant_mt      ' --include-priority\n/root/.GetOrganelle/LabelDatabase/embplant_mt.fasta'\n------------------------------------------------------\nembplant_nr      ' --include-priority\n/root/.GetOrganelle/LabelDatabase/embplant_nr.fasta'\n------------------------------------------------------\nanimal_mt        ' --include-priority\n/root/.GetOrganelle/LabelDatabase/animal_mt.fasta'\n------------------------------------------------------\nfungus_mt        ' --include-priority\n/root/.GetOrganelle/LabelDatabase/fungus_mt.fasta'\n------------------------------------------------------\nfungus_nr        ' --include-priority\n/root/.GetOrganelle/LabelDatabase/fungus_nr.fasta'\n------------------------------------------------------\nother_pt,embplant_mt,fungus_mt   ' --include-priority\n/root/.GetOrganelle/LabelDatabase/other_pt.fasta,/root\n/.GetOrganelle/LabelDatabase/embplant_mt.fasta,/root/.\nGetOrganelle/LabelDatabase/fungus_mt.fasta'\n------------------------------------------------------\nFor easy usage and compatibility of old versions,\nfollowing redirection would be automatically fulfilled\nwithout warning:   plant_cp->embplant_pt;\nplant_pt->embplant_pt;  plant_mt->embplant_mt;\nplant_nr->embplant_nr")), ToolInput(tag="in_followed_animalmt_fungusmtfungusnr", input_type=String(optional=True), prefix="-E", doc=InputDocumentation(doc="followed with mode embplant_pt, other_pt, embplant_mt,\nembplant_nr, animal_mt, fungus_mt,fungus_nr (which\nmeans embryophyta plastome, non-embryophyta plastome,\nplant mitochondrion, plant nuclear ribosomal RNA,\nanimal mitochondrion, fungus mitochondrion, fungus\nnuclear ribosomal RNA separately), or a combination of\nabove split by comma(s) (be similar to -F and\ncorresponds to certain arguments as following listed).\n------------------------------------------------------\nembplant_pt      ' --exclude\n/root/.GetOrganelle/LabelDatabase/embplant_pt.fasta'\n------------------------------------------------------\nembplant_mt      ' --exclude\n/root/.GetOrganelle/LabelDatabase/embplant_mt.fasta'\n------------------------------------------------------\nembplant_nr      ' --exclude\n/root/.GetOrganelle/LabelDatabase/embplant_nr.fasta'\n------------------------------------------------------\nanimal_mt        ' --exclude\n/root/.GetOrganelle/LabelDatabase/animal_mt.fasta'\n------------------------------------------------------\nfungus_mt        ' --exclude\n/root/.GetOrganelle/LabelDatabase/fungus_mt.fasta'\n------------------------------------------------------\nfungus_nr        ' --exclude\n/root/.GetOrganelle/LabelDatabase/fungus_nr.fasta'\n------------------------------------------------------\nembplant_mt,embplant_nr          ' --exclude /root/.Ge\ntOrganelle/LabelDatabase/embplant_mt.fasta,/root/.GetO\nrganelle/LabelDatabase/embplant_nr.fasta'\n------------------------------------------------------\nFor easy usage and compatibility of old versions,\nfollowing redirection would be automatically fulfilled\nwithout warning:   plant_cp->embplant_pt;\nplant_pt->embplant_pt;  plant_mt->embplant_mt;\nplant_nr->embplant_nr")), ToolInput(tag="in_no_hits", input_type=String(optional=True), prefix="--no-hits", doc=InputDocumentation(doc="Provide treatment for non-hitting contigs.\n------------------------------------------------------\nex_no_con        keep those connect with hitting-\ninclude contigs. (Default)\n------------------------------------------------------\nex_no_hit        exclude all.\n------------------------------------------------------\nkeep_all         keep all\n------------------------------------------------------")), ToolInput(tag="in_max_slim_extending_len", input_type=Int(optional=True), prefix="--max-slim-extending-len", doc=InputDocumentation(doc="This is used to limit the extending length, below\nwhich a 'non-hit contig' is allowed to be distant from\na 'hit contig' to be kept. This distance is measured\nby the shortest distance connecting those two contigs,\nweighted by the depth of the 'hit contig'. This is\nused only when '--no-hits ex_no_con' was chosen.\nShould be a single INTEGER number or inf (meaning\ninfinite). It is supposed to be half of the maximum\nexpected genome size to be safe, but could be much\nsmaller if the LabelDatabse is closely related.\nDefault: 15000 (-F embplant_pt), 50000 (-F\nembplant_mt/fungus_mt/other_pt), 12500 (-F\nembplant_nr/fungus_nr/animal_mt),\nmaximum_of_type1_type2 (-F type1,type2), inf (cases\nwithout using -F)")), ToolInput(tag="in_significant", input_type=Float(optional=True), prefix="--significant", doc=InputDocumentation(doc="Within a contig, if the query-score of hitting A is\nmore than given times (Default: 3.0) of the query-\nscore of hitting B, this contig would be treated as\nonly A related, rather than both.")), ToolInput(tag="in_depth_cut_off", input_type=Float(optional=True), prefix="--depth-cutoff", doc=InputDocumentation(doc="After detection for target coverage, those beyond\ncertain times (depth cutoff) of the detected coverage\nwould be excluded. Default: 10000.0")), ToolInput(tag="in_min_depth", input_type=Int(optional=True), prefix="--min-depth", doc=InputDocumentation(doc="Input a float or integer number. Filter fastg file by\na minimum depth. Default: 0.0.")), ToolInput(tag="in_max_depth", input_type=Int(optional=True), prefix="--max-depth", doc=InputDocumentation(doc="Input a float or integer number. filter fastg file by\na maximum depth. Default: inf.")), ToolInput(tag="in_merge", input_type=Boolean(optional=True), prefix="--merge", doc=InputDocumentation(doc="Merge all possible contigs.")), ToolInput(tag="in_include", input_type=String(optional=True), prefix="--include", doc=InputDocumentation(doc="followed by Blastn database(s)")), ToolInput(tag="in_include_priority", input_type=String(optional=True), prefix="--include-priority", doc=InputDocumentation(doc="followed by Blastn database(s).")), ToolInput(tag="in_exclude", input_type=String(optional=True), prefix="--exclude", doc=InputDocumentation(doc="followed by Blastn database(s).")), ToolInput(tag="in_exclude_priority", input_type=String(optional=True), prefix="--exclude-priority", doc=InputDocumentation(doc="followed by Blastn database(s)")), ToolInput(tag="in_no_hits_labeled_tab", input_type=Boolean(optional=True), prefix="--no-hits-labeled-tab", doc=InputDocumentation(doc="Choose to disable producing tab file")), ToolInput(tag="in_keep_temp", input_type=Boolean(optional=True), prefix="--keep-temp", doc=InputDocumentation(doc="Choose to disable deleting temp files produced by\nblast and this script")), ToolInput(tag="in_out_dir", input_type=File(optional=True), prefix="--out-dir", doc=InputDocumentation(doc="By default the output would be along with the input\nfastg file. But you could assign a new directory with\nthis option.")), ToolInput(tag="in_evalue", input_type=Float(optional=True), prefix="--evalue", doc=InputDocumentation(doc="blastn evalue threshold. Default: 1e-25")), ToolInput(tag="in_prefix", input_type=String(optional=True), prefix="--prefix", doc=InputDocumentation(doc="Add prefix to the output basename. Conflict with '--\nout-base'.")), ToolInput(tag="in_out_base", input_type=File(optional=True), prefix="--out-base", doc=InputDocumentation(doc="By default the output basename would be modified based\non the input fastg file. But you could assign a new\nbasename with this option. Conflict with '--prefix'.\nConflict with multiple input files!")), ToolInput(tag="in_log", input_type=Boolean(optional=True), prefix="--log", doc=InputDocumentation(doc="Generate log file.")), ToolInput(tag="in_wrapper", input_type=Boolean(optional=True), prefix="--wrapper", doc=InputDocumentation(doc="Wrapper mode logging when called by get_organelle*.py.\nDefault: False")), ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="--verbose", doc=InputDocumentation(doc="For debug usage.")), ToolInput(tag="in_continue", input_type=Boolean(optional=True), prefix="--continue", doc=InputDocumentation(doc="Specified for calling from get_organelle_from_reads.py")), ToolInput(tag="in_no_overwrite", input_type=Boolean(optional=True), prefix="--no-overwrite", doc=InputDocumentation(doc="Overwrite existing output result.")), ToolInput(tag="in_which_blast", input_type=File(optional=True), prefix="--which-blast", doc=InputDocumentation(doc="Assign the path to BLAST binary files if not added to\nthe path. Default: try '/GetOrganelleDep/linux/ncbi-\nblast' first, then $PATH")), ToolInput(tag="in_config_dir", input_type=File(optional=True), prefix="--config-dir", doc=InputDocumentation(doc="The directory where the default databases were placed.\nThe default value also can be changed by adding\n'export GETORG_PATH=your_favor' to the shell script\n(e.g. ~/.bash_profile or ~/.bashrc) Default:\n/root/.GetOrganelle")), ToolInput(tag="in_threads", input_type=String(optional=True), prefix="--threads", doc=InputDocumentation(doc="Threads for blastn.\n")), ToolInput(tag="in_your_fast_g_files", input_type=String(), position=0, doc=InputDocumentation(doc=""))], outputs=[], container="quay.io/biocontainers/getorganelle:1.7.3.5--pyh3252c3a_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Slim_Graph_Py_V0_1_0().translate("wdl")

