from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, Directory, File, String, Int

Run_Snp_Pipeline_Sh_V0_1_0 = CommandToolBuilder(tool="run_snp_pipeline.sh", base_command=["run_snp_pipeline.sh"], inputs=[ToolInput(tag="in_force", input_type=Boolean(optional=True), prefix="--force", doc=InputDocumentation(doc="Force processing even when result files already exist\nand are newer than inputs (default: False)")), ToolInput(tag="in_mirror", input_type=Directory(optional=True), prefix="--mirror", doc=InputDocumentation(doc="Create a mirror copy of the reference directory and\nall the sample directories.  Use this option to avoid\npolluting the reference directory and sample\ndirectories with the intermediate files generated by\nthe snp pipeline.  A 'reference' subdirectory and a\n'samples' subdirectory are created under the output\ndirectory (see the -o option).  One directory per\nsample is created under the 'samples' directory.\nThree suboptions allow a choice of how the reference\nand samples are mirrored.\n-m soft : creates soft links to the fasta and fastq\nfiles instead of copying\n-m hard : creates hard links to the fasta and fastq\nfiles instead of copying\n-m copy : copies the fasta and fastq files\n(default: None)")), ToolInput(tag="in_conf", input_type=File(optional=True), prefix="--conf", doc=InputDocumentation(doc="Relative or absolute path to a configuration file for\noverriding defaults and defining extra parameters for\nthe tools and scripts within the pipeline.\nNote: A default parameter configuration file named\n'snppipeline.conf' is used whenever the pipeline\nis run without the -c option.  The configuration\nfile used for each run is copied into the log\ndirectory, capturing the parameters used during\nthe run. (default: None)")), ToolInput(tag="in_slurmtorque__queuemgr", input_type=String(optional=True), prefix="-Q", doc=InputDocumentation(doc="|slurm|torque, --queue_mgr grid|slurm|torque\nJob queue manager for remote parallel job execution in\nan HPC environment. Currently 'grid', 'slurm', and\n'torque' are supported. If not specified, the pipeline\nwill execute locally. (default: None)")), ToolInput(tag="in_out_dir", input_type=Directory(optional=True), prefix="--out_dir", doc=InputDocumentation(doc="Output directory for the result files. Additional\nsubdirectories are automatically created under the\noutput directory for logs files and the mirrored\nsamples and reference files (see the -m option). The\noutput directory will be created if it does not\nalready exist. If not specified, the output files are\nwritten to the current working directory. If you re-\nrun the pipeline on previously processed samples, and\nspecify a different output directory, the pipeline\nwill not rebuild everything unless you either force a\nrebuild (see the -f option) or you request mirrored\ninputs (see the -m option). (default: .)")), ToolInput(tag="in_samples_dir", input_type=File(optional=True), prefix="--samples_dir", doc=InputDocumentation(doc="Relative or absolute path to the parent directory of\nall the sample directories.  The -s option should be\nused when all the sample directories are in\nsubdirectories immediately below a parent directory.\nNote: You must specify either the -s or -S option, but\nnot both.\nNote: The specified directory should contain only a\ncollection of sample directories, nothing else.\nNote: Unless you request mirrored inputs, see the\n-m option, additional files will be written to\neach of the sample directories during the\nexecution of the SNP Pipeline (default: None)")), ToolInput(tag="in_samples_file", input_type=File(optional=True), prefix="--samples_file", doc=InputDocumentation(doc="Relative or absolute path to a file listing all of the\nsample directories.  The -S option should be used when\nthe samples are not under a common parent directory.\nNote: If you are not mirroring the samples (see the\n-m option), you can improve parallel processing\nperformance by sorting the the list of\ndirectories descending by size, largest first.\nThe -m option automatically generates a sorted\ndirectory list.\nNote: You must specify either the -s or -S option, but\nnot both.\nNote: Unless you request mirrored inputs, see the\n-m option, additional files will be written to\neach of the sample directories during the\nexecution of the SNP Pipeline (default: None)")), ToolInput(tag="in_purge", input_type=Directory(optional=True), prefix="--purge", doc=InputDocumentation(doc="Purge the intermediate output files (the entire\n'samples' directory) when the pipeline completes\nsuccessfully. (default: False)")), ToolInput(tag="in_verbose", input_type=Int(optional=True), prefix="--verbose", doc=InputDocumentation(doc="Verbose message level (0=no info, 5=lots) (default: 1)")), ToolInput(tag="in_cfsan_snp_pipeline", input_type=String(), position=0, doc=InputDocumentation(doc="")), ToolInput(tag="in_run", input_type=String(), position=1, doc=InputDocumentation(doc=""))], outputs=[ToolOutput(tag="out_mirror", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_mirror", type_hint=File()), doc=OutputDocumentation(doc="Create a mirror copy of the reference directory and\nall the sample directories.  Use this option to avoid\npolluting the reference directory and sample\ndirectories with the intermediate files generated by\nthe snp pipeline.  A 'reference' subdirectory and a\n'samples' subdirectory are created under the output\ndirectory (see the -o option).  One directory per\nsample is created under the 'samples' directory.\nThree suboptions allow a choice of how the reference\nand samples are mirrored.\n-m soft : creates soft links to the fasta and fastq\nfiles instead of copying\n-m hard : creates hard links to the fasta and fastq\nfiles instead of copying\n-m copy : copies the fasta and fastq files\n(default: None)")), ToolOutput(tag="out_purge", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_purge", type_hint=File()), doc=OutputDocumentation(doc="Purge the intermediate output files (the entire\n'samples' directory) when the pipeline completes\nsuccessfully. (default: False)"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Run_Snp_Pipeline_Sh_V0_1_0().translate("wdl", allow_empty_container=True)

