from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Int, Boolean, String

Variantbreak_V0_1_0 = CommandToolBuilder(tool="variantbreak", base_command=["variantbreak"], inputs=[ToolInput(tag="in_annotation_file_dir", input_type=File(optional=True), prefix="--annotation_file_dir", doc=InputDocumentation(doc="path to single annotation file or directory containing annotation\nfiles of GTF/GFF or BED formats. Formats: .gtf/.gff/.gff3/.bed")), ToolInput(tag="in_filter_file_dir", input_type=File(optional=True), prefix="--filter_file_dir", doc=InputDocumentation(doc="path to single filter file or directory containing filter files of\nBED format. Format: .bed")), ToolInput(tag="in_del_annotate_size", input_type=Int(optional=True), prefix="--del_annotate_size", doc=InputDocumentation(doc="Deletions with sizes lower or equal to this value will have the\nentire deleted region annotated. Any genes that intersect with\nthe deleted region will be included as annotation. On the contrary,\nif deletion size is greater than this value, only the deletion\nbreakends will be annotated, omitting any annotation of the middle\ndeleted region. In other words, only genes intersecting with the\ndeletion breakends will be included as annotation. This is done to\nreduce false annotations due to false large deletions. Note that\nthe value to be set is an absolute deletion size, do not use minus\n'-'. Use value '-1' to disable this threshold and annotate all\ndeleted regions despite of size. [20000]")), ToolInput(tag="in_merge_buffer", input_type=Int(optional=True), prefix="--merge_buffer", doc=InputDocumentation(doc="nucleotide length buffer for SV breakend clustering [400]")), ToolInput(tag="in_promoter_size", input_type=Int(optional=True), prefix="--promoter_size", doc=InputDocumentation(doc="length in base-pairs upstream of TSS to define promoter region\n[1000]")), ToolInput(tag="in_max_annotation", input_type=Int(optional=True), prefix="--max_annotation", doc=InputDocumentation(doc="maximum number of annotation entries to be recorded in the\ndataframe for each SV [3]")), ToolInput(tag="in_cluster_sample", input_type=Boolean(optional=True), prefix="--cluster_sample", doc=InputDocumentation(doc="performs hierarchical clustering on samples")), ToolInput(tag="in_auto_filter", input_type=Boolean(optional=True), prefix="--auto_filter", doc=InputDocumentation(doc="automatically removes variants that intersected\nwith all filter BED files")), ToolInput(tag="in_sep", input_type=File(optional=True), prefix="--sep", doc=InputDocumentation(doc="single character field delimiter for output dataframe CSV file\n(e.g. '\t' for tab or ',' for comma) [,]")), ToolInput(tag="in_filename", input_type=File(optional=True), prefix="--filename", doc=InputDocumentation(doc="File name prefix of output files [output]")), ToolInput(tag="in_quiet", input_type=Boolean(optional=True), prefix="--quiet", doc=InputDocumentation(doc="hide verbose")), ToolInput(tag="in_exist_dot", input_type=String(), position=0, doc=InputDocumentation(doc="optional arguments:"))], outputs=[ToolOutput(tag="out_sep", output_type=File(optional=True), selector=InputSelector(input_to_select="in_sep", type_hint=File()), doc=OutputDocumentation(doc="single character field delimiter for output dataframe CSV file\n(e.g. '\t' for tab or ',' for comma) [,]")), ToolOutput(tag="out_filename", output_type=File(optional=True), selector=InputSelector(input_to_select="in_filename", type_hint=File()), doc=OutputDocumentation(doc="File name prefix of output files [output]"))], container="quay.io/biocontainers/variantbreak:1.0.4--py_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Variantbreak_V0_1_0().translate("wdl")

