from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, String, Boolean, Int, Directory

Snpsplit_Genome_Preparation_V0_1_0 = CommandToolBuilder(tool="SNPsplit_genome_preparation", base_command=["SNPsplit_genome_preparation"], inputs=[ToolInput(tag="in_vcf_file", input_type=File(optional=True), prefix="--vcf_file", doc=InputDocumentation(doc="Mandatory file specifying SNP information for mouse strains from the Mouse Genomes Project\n(http://www.sanger.ac.uk/science/data/mouse-genomes-project). The file used and approved is\ncalled 'mgp.v5.merged.snps_all.dbSNP142.vcf.gz'. Please note that future versions\nof this file or entirely different VCF files might not work out-of-the-box but may require some\ntweaking. SNP calls are read from the VCF files, and high confidence SNPs are written into\na folder in the current working directory called SNPs_<strain_name>/chr<chromosome>.txt,\nin the following format:\nSNP-ID     Chromosome  Position    Strand   Ref/SNP\nexample:   33941939        9       68878541       1       T/G")), ToolInput(tag="in_strain", input_type=String(optional=True), prefix="--strain", doc=InputDocumentation(doc="The strain you would like to use as SNP (ALT) genome. Mandatory. For an overview of strain names\njust run SNPsplit_genome_preparation selecting '--list_strains'.")), ToolInput(tag="in_list_strains", input_type=Boolean(optional=True), prefix="--list_strains", doc=InputDocumentation(doc="Displays a list of strain names present in the VCF file for use with '--strain <strain_name>'.")), ToolInput(tag="in_dual_hybrid", input_type=Boolean(optional=True), prefix="--dual_hybrid", doc=InputDocumentation(doc="Optional. The resulting genome will no longer relate to the original reference specified with '--reference_genome'.\nInstead the new Reference (Ref) is defined by '--strain <strain_name>' and the new SNP genome\nis defined by '--strain2 <strain_name>'. '--dual_hybrid' automatically sets '--full_sequence'.\nThis will invoke a multi-step process:\n1) Read/filter SNPs for first strain (specified with '--strain <strain_name>')\n2) Write full SNP incorporated (and optionally N-masked) genome sequence for first strain\n3) Read/filter SNPs for second strain (specified with '--strain2 <strain_name>')\n4) Write full SNP incorporated (and optionally N-masked) genome sequence for second strain\n5) Generate new Ref/Alt SNP annotations for Strain1/Strain2\n6) Set first strain as new reference genome and construct full SNP incorporated (and optionally\nN-masked) genome sequences for Strain1/Strain2")), ToolInput(tag="in_strain_two", input_type=Int(optional=True), prefix="--strain2", doc=InputDocumentation(doc="Optional for constructing dual hybrid genomes (see '--dual_hybrid' for more information). For an\noverview of strain names just run SNPsplit_genome_preparation selecting '--list_strains'.")), ToolInput(tag="in_reference_genome", input_type=Boolean(optional=True), prefix="--reference_genome", doc=InputDocumentation(doc="The path to the reference genome, typically the strain 'Black6' (C57BL/6J), e.g.\n'--reference_genome /bi/scratch/Genomes/Mouse/GRCm38/'. Expects one or more FastA files in this folder\n(file extension: .fa or .fasta).")), ToolInput(tag="in_skip_filtering", input_type=Boolean(optional=True), prefix="--skip_filtering", doc=InputDocumentation(doc="This option skips reading and filtering the VCF file. This assumes that a folder named\n'SNPs_<Strain_Name>' exists in the working directory, and that text files with SNP information\nare contained therein in the following format:\nSNP-ID     Chromosome  Position    Strand   Ref/SNP\nexample:   33941939        9       68878541       1       T/G")), ToolInput(tag="in_n_masking", input_type=Directory(optional=True), prefix="--nmasking", doc=InputDocumentation(doc="Write out a genome version for the strain specified where Ref bases are replaced with 'N'. In the\nRef/SNP example T/G the N-masked genome would now carry an N instead of the T. The N-masked genome\nis written to a folder called  '<strain_name>_N-masked/'. Default: ON.")), ToolInput(tag="in_full_sequence", input_type=Directory(optional=True), prefix="--full_sequence", doc=InputDocumentation(doc="Write out a genome version for the strain specified where Ref bases are replaced with the SNP base.\nIn the Ref/SNP example T/G the full sequence genome would now carry a G instead of the T. The full\nsequence genome is written out to folder called '<strain_name>_full_sequence/'. May be set in\naddition to '--nmasking'. Default: OFF.")), ToolInput(tag="in_no_n_masking", input_type=Boolean(optional=True), prefix="--no_nmasking", doc=InputDocumentation(doc="Disable N-masking if it is not desirable. Will automatically set '--full_sequence' instead.")), ToolInput(tag="in_genome_build", input_type=Boolean(optional=True), prefix="--genome_build", doc=InputDocumentation(doc="[name]         Name of the mouse genome build, e.g. GRCm38. Will be incorporated into some of the output files.\nDefaults to 'GRCm38'."))], outputs=[ToolOutput(tag="out_vcf_file", output_type=File(optional=True), selector=InputSelector(input_to_select="in_vcf_file", type_hint=File()), doc=OutputDocumentation(doc="Mandatory file specifying SNP information for mouse strains from the Mouse Genomes Project\n(http://www.sanger.ac.uk/science/data/mouse-genomes-project). The file used and approved is\ncalled 'mgp.v5.merged.snps_all.dbSNP142.vcf.gz'. Please note that future versions\nof this file or entirely different VCF files might not work out-of-the-box but may require some\ntweaking. SNP calls are read from the VCF files, and high confidence SNPs are written into\na folder in the current working directory called SNPs_<strain_name>/chr<chromosome>.txt,\nin the following format:\nSNP-ID     Chromosome  Position    Strand   Ref/SNP\nexample:   33941939        9       68878541       1       T/G")), ToolOutput(tag="out_n_masking", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_n_masking", type_hint=File()), doc=OutputDocumentation(doc="Write out a genome version for the strain specified where Ref bases are replaced with 'N'. In the\nRef/SNP example T/G the N-masked genome would now carry an N instead of the T. The N-masked genome\nis written to a folder called  '<strain_name>_N-masked/'. Default: ON.")), ToolOutput(tag="out_full_sequence", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_full_sequence", type_hint=File()), doc=OutputDocumentation(doc="Write out a genome version for the strain specified where Ref bases are replaced with the SNP base.\nIn the Ref/SNP example T/G the full sequence genome would now carry a G instead of the T. The full\nsequence genome is written out to folder called '<strain_name>_full_sequence/'. May be set in\naddition to '--nmasking'. Default: OFF."))], container="quay.io/biocontainers/snpsplit:0.4.0--0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Snpsplit_Genome_Preparation_V0_1_0().translate("wdl")

