from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import String, Boolean, Float, File, Int

Rtg_Cgsim_V0_1_0 = CommandToolBuilder(tool="rtg_cgsim", base_command=["rtg", "cgsim"], inputs=[ToolInput(tag="in_input", input_type=String(optional=True), prefix="--input", doc=InputDocumentation(doc="SDF containing input genome")), ToolInput(tag="in_output", input_type=String(optional=True), prefix="--output", doc=InputDocumentation(doc="name for reads output SDF")), ToolInput(tag="in_abundance", input_type=Boolean(optional=True), prefix="--abundance", doc=InputDocumentation(doc="if set, the user-supplied distribution\nrepresents desired abundance")), ToolInput(tag="in_allow_unknowns", input_type=Boolean(optional=True), prefix="--allow-unknowns", doc=InputDocumentation(doc="allow reads to be drawn from template\nfragments containing unknown nucleotides")), ToolInput(tag="in_coverage", input_type=Float(optional=True), prefix="--coverage", doc=InputDocumentation(doc="coverage, must be positive")), ToolInput(tag="in_distribution", input_type=File(optional=True), prefix="--distribution", doc=InputDocumentation(doc="file containing probability distribution for\nsequence selection")), ToolInput(tag="in_dna_fraction", input_type=Boolean(optional=True), prefix="--dna-fraction", doc=InputDocumentation(doc="if set, the user-supplied distribution\nrepresents desired DNA fraction")), ToolInput(tag="in_max_fragment_size", input_type=Int(optional=True), prefix="--max-fragment-size", doc=InputDocumentation(doc="maximum fragment size (Default is 500)")), ToolInput(tag="in_min_fragment_size", input_type=Int(optional=True), prefix="--min-fragment-size", doc=InputDocumentation(doc="minimum fragment size (Default is 350)")), ToolInput(tag="in_n_rate", input_type=Float(optional=True), prefix="--n-rate", doc=InputDocumentation(doc="rate that the machine will generate new\nunknowns in the read (Default is 0.0)")), ToolInput(tag="in_num_reads", input_type=Int(optional=True), prefix="--num-reads", doc=InputDocumentation(doc="number of reads to be generated")), ToolInput(tag="in_taxonomy_distribution", input_type=File(optional=True), prefix="--taxonomy-distribution", doc=InputDocumentation(doc="file containing probability distribution for\nsequence selection expressed by taxonomy id")), ToolInput(tag="in_cg_read_version", input_type=Int(optional=True), prefix="--cg-read-version", doc=InputDocumentation(doc="select Complete Genomics read structure\nversion, 1 (35 bp) or 2 (29 bp)")), ToolInput(tag="in_comment", input_type=String(optional=True), prefix="--comment", doc=InputDocumentation(doc="comment to include in the generated SDF")), ToolInput(tag="in_no_names", input_type=Boolean(optional=True), prefix="--no-names", doc=InputDocumentation(doc="do not create read names in the output SDF")), ToolInput(tag="in_no_qualities", input_type=Boolean(optional=True), prefix="--no-qualities", doc=InputDocumentation(doc="do not create read qualities in the output")), ToolInput(tag="in_sam_rg", input_type=File(optional=True), prefix="--sam-rg", doc=InputDocumentation(doc="|FILE         file containing a single valid read group\nSAM header line or a string in the form\n'@RG\tID:READGROUP1\tSM:BACT_SAMPLE\tPL:ILLUMINA'")), ToolInput(tag="in_seed", input_type=Int(optional=True), prefix="--seed", doc=InputDocumentation(doc="seed for random number generator")), ToolInput(tag="in_sdf", input_type=String(), position=0, doc=InputDocumentation(doc="-q, --qual-range=STRING          set the range of base quality values"))], outputs=[], container="quay.io/biocontainers/rtg-tools:3.12--0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Rtg_Cgsim_V0_1_0().translate("wdl")

