#!/usr/bin/env cwl-runner

baseCommand:
- mamotif
class: CommandLineTool
cwlVersion: v1.0
id: mamotif
inputs:
- doc: 'Path of peaks file of sample 1. BED and MACS format are currently supported.Please
    refer to documents for details. (default: None)'
  id: p1
  inputBinding:
    prefix: --p1
  type: string
- doc: 'Path of peaks file of sample 2. (default: None)'
  id: p2
  inputBinding:
    prefix: --p2
  type: string
- doc: 'Path of reads file of sample 1. BED format are currently supported. (default:
    None)'
  id: r1
  inputBinding:
    prefix: --r1
  type: string
- doc: 'Path of reads file of sample 2. (default: None)'
  id: r2
  inputBinding:
    prefix: --r2
  type: string
- doc: "Reads shiftsize of sample 1. This value is used to shift reads towards 3'\
    \ directionto account for the actual binding site. Set as half of the fragment\
    \ length. (default: 100)"
  id: s1
  inputBinding:
    prefix: --s1
  type: string
- doc: 'Reads shiftsize of sample 2. (default: 100)'
  id: s2
  inputBinding:
    prefix: --s2
  type: string
- doc: "Required pre-compiled motif file, which is generated by 'motifcompile'. (default:\
    \ None)"
  id: motif
  inputBinding:
    prefix: --motif
  type: string
- doc: "Required genome directory, which contains genome sequence and chromosome length.\
    \ You can run 'genomecompile' to generate the directory. (default: None)"
  id: genome
  inputBinding:
    prefix: --genome
  type: string
- doc: 'Gene annotation file, which is used to generate random controls when performing
    enrichment analysis. Sequence features on promoters are very different from those
    on whole genome. We will pick up random control regions with similar genomic locations
    for each input region instead of the whole genome. (default: None)'
  id: annotation
  inputBinding:
    prefix: --annotation
  type: string
- doc: 'Half width of the window size when calculating read densities. Each window
    with length of 2*width is centered at peak summit or midpoint. This value should
    match the typical length of peaks, thus we recommend 1000 for sharp histone marks
    like H3K4me3 and H3K9/27ac, or 500 for transcription factors or DNase- Seq. (default:
    1000)'
  id: w
  inputBinding:
    prefix: -w
  type: string
- doc: 'Summit to summit distance cutoff for common peaks. Default=width/2. Only peaks
    with summit distance less than than this value are considered as real common peaks
    of two samples. (default: None)'
  id: d
  inputBinding:
    prefix: -d
  type: string
- doc: 'Times of permutation to test the enrichment of peak overlap between two samples.
    (default: 5)'
  id: n
  inputBinding:
    prefix: -n
  type: string
- doc: 'P-value cutoff to define biased (sample 1/2-specific) peaks. (default: 0.01)'
  id: p
  inputBinding:
    prefix: -p
  type: string
- doc: 'M-value cutoff to distinguish biased peaks from unbiased peaks. Peaks with
    M-value>=M_cutoff and P-value<=P_cutoff as defined as sample1-biased(specific)
    peaks, while peaks with M-value<=-1*M_cutoff and P-value<=P_cutoff are defined
    as sample2-biased peaks. (default: 1.0)'
  id: m_cut_off
  inputBinding:
    prefix: --m_cutoff
  type: string
- doc: 'Motif list file. If specified, the program will only perform MotifScan on
    given motifs in this list. (default: None)'
  id: motif_list
  inputBinding:
    prefix: --motif_list
  type: string
- doc: 'If set, MotifScan will neither generate random control regions nor perform
    enrichment analysis. (default: True)'
  id: no_enrichment
  inputBinding:
    prefix: --no_enrichment
  type: boolean
- doc: "Genomic location to apply MotifScan. 'all': perform MotifScan on the all input\
    \ regions. 'promoter': perform MotifScan on regions which located at promoter\
    \ regions. 'distal': perform MotifScan on regions which located at distal regions.\
    \ 'gene': perform MotifScan on regions which is the target(nearest one) of genes\
    \ within 10kbp distance cutoff. (default: all)"
  id: r
  inputBinding:
    prefix: -r
  type: string
- doc: "Upstream distance to TSS to define promoter regions. Valid when option -r\
    \ is set as 'promoter' or 'distal'. (default: 4000)"
  id: upstream
  inputBinding:
    prefix: --upstream
  type: string
- doc: "Upstream distance to TSS to define promoter regions. Valid when option -r\
    \ is set as 'promoter' or 'distal'. (default: 2000)"
  id: downstream
  inputBinding:
    prefix: --downstream
  type: string
- doc: 'The number of random control regions generated for each input region. For
    example, if you provide 1000 regions, 5000 random regions will be generated by
    default. (default: 5)'
  id: random_times
  inputBinding:
    prefix: --random_times
  type: string
- doc: 'The length of input regions to perform MotifScan around peak summit/midpoint.
    If set to 0, the whole peak region is used to perform MotifScan. (default: 1000)'
  id: peak_length
  inputBinding:
    prefix: --peak_length
  type: string
- doc: 'Using negative test of this pk (default: False)'
  id: negative
  inputBinding:
    prefix: --negative
  type: boolean
- doc: 'correction type of pvalues, no correction or benjamin or bonferroni,default=benjamin
    (default: benjamin)'
  id: correction
  inputBinding:
    prefix: --correction
  type: string
- doc: 'By default, MAnorm will output the results of unique and merged common peaks
    of two samples. With this option on, MAnorm will output two extra files containing
    the results of the original(unmerged) peaks in two samples. With this option on,
    MAmotif will also output the exact target sites of all motifs. (default: False)'
  id: s
  inputBinding:
    prefix: -s
  type: boolean
- doc: 'Comparison name, this is used as the folder name and prefix of output files.
    (default: None)'
  id: o
  inputBinding:
    prefix: -o
  type: string
