class: CommandLineTool
id: anvi_estimate_scg_taxonomy.cwl
inputs:
- id: in_contigs_db
  doc: "Anvi'o contigs database generated by 'anvi-gen-\ncontigs'"
  type: string?
  inputBinding:
    prefix: --contigs-db
- id: in_meta_genome_mode
  doc: "Treat a given contigs database as a metagenome rather\nthan treating it as\
    \ a single genome."
  type: boolean?
  inputBinding:
    prefix: --metagenome-mode
- id: in_profile_db
  doc: Anvi'o profile database
  type: string?
  inputBinding:
    prefix: --profile-db
- id: in_collection_name
  doc: Collection name.
  type: string?
  inputBinding:
    prefix: --collection-name
- id: in_meta_genomes
  doc: "A two-column TAB-delimited flat text file. The header\nline must contain these\
    \ columns: 'name',\n'contigs_db_path', and 'profile_db_path'. Each line\nshould\
    \ list a single entry, where 'name' can be any\nname to describe the metagenome\
    \ stored in the anvi'o\ncontigs database. In this context, the anvi'o profiles\n\
    associated with contigs database must be SINGLE\nPROFILES, as in generated by\
    \ the program `anvi-\nprofile` and not `anvi-merge`."
  type: File?
  inputBinding:
    prefix: --metagenomes
- id: in_output_file
  doc: File path to store results.
  type: File?
  inputBinding:
    prefix: --output-file
- id: in_output_file_prefix
  doc: "A prefix to be used while naming the output files (no\nfile type extensions\
    \ please; just a prefix)."
  type: File?
  inputBinding:
    prefix: --output-file-prefix
- id: in_taxonomic_level
  doc: "The taxonomic level to use whenever relevant and/or\navailable. The default\
    \ taxonomic level is None, but if\nyou choose something specific, anvi'o will\
    \ focus on\nthat whenever possible."
  type: string?
  inputBinding:
    prefix: --taxonomic-level
- id: in_matrix_format
  doc: "If you want the reports to look like sparse matrices\nwhenever possible, declare\
    \ this flag. Matrices are\nespecially good to use when you are working with\n\
    internal/external genomes since they can show you\nquickly the distribution of\
    \ each taxon across all\nmetagenomes in programs like EXCEL. WELL TRY IT AND\n\
    SEE."
  type: boolean?
  inputBinding:
    prefix: --matrix-format
- id: in_raw_output
  doc: "Just store the raw output without any processing of\nthe primary data structure."
  type: boolean?
  inputBinding:
    prefix: --raw-output
- id: in_num_threads
  doc: "Maximum number of threads to use for multithreading\nwhenever possible. Very\
    \ conservatively, the default is\n1. It is a good idea to not exceed the number\
    \ of CPUs\n/ cores on your system. Plus, please be careful with\nthis option if\
    \ you are running your commands on a SGE\n--if you are clusterizing your runs,\
    \ and asking for\nmultiple threads to use, you may deplete your\nresources very\
    \ fast."
  type: long?
  inputBinding:
    prefix: --num-threads
- id: in_scg_name_for_meta_genome_mode
  doc: "When running in metagenome mode, anvi'o automatically\nchooses the most frequent\
    \ single-copy core gene to\nestimate the taxonomic composition within a contigs\n\
    database. If you have a different preference you can\nuse this parameter to communicate\
    \ that."
  type: string?
  inputBinding:
    prefix: --scg-name-for-metagenome-mode
- id: in_report_scg_frequencies
  doc: "Report SCG frequencies in a TAB-delimited file and\nquit. This is a great\
    \ way to decide which SCG name to\nuse in metagenome mode (we often wish to use\
    \ the most\nfrequent SCG to increase the detection of taxa)."
  type: File?
  inputBinding:
    prefix: --report-scg-frequencies
- id: in_just_do_it
  doc: Don't bother me with questions or warnings, just do
  type: boolean?
  inputBinding:
    prefix: --just-do-it
- id: in_simplify_taxonomy_information
  doc: "The taxonomy output may include a large number of\nnames that contain clade-specific\
    \ code for not-yet-\ncharacterized taxa. With this flag you can simplify\ntaxon\
    \ names. This will influence all output files and\ndisplays as the use of this\
    \ flag will on-the-fly trim\ntaxonomic levels with clade-specific code names."
  type: boolean?
  inputBinding:
    prefix: --simplify-taxonomy-information
- id: in_compute_scg_coverages
  doc: "When this flag is declared, anvi'o will go back to the\nprofile database to\
    \ learn coverage statistics of\nsingle-copy core genes for which we have taxonomy\n\
    information."
  type: boolean?
  inputBinding:
    prefix: --compute-scg-coverages
- id: in_update_profile_db_with_taxonomy
  doc: "When anvi'o knows all both taxonomic affiliations and\ncoverages across samples\
    \ for single-copy core genes,\nit can, in theory add this information to the profile\n\
    database. With this flag you can instruct anvi'o to do\nthat and find information\
    \ on taxonomy in the `layers`\ntab of your interactive interface."
  type: boolean?
  inputBinding:
    prefix: --update-profile-db-with-taxonomy
- id: in_taxonomy_database
  doc: "Path to the directory that contains the BLAST\ndatabases for single-copy core\
    \ genes. You will almost\nnever need to use this parameter unless you are trying\n\
    something very fancy. But when you do, you can tell\nanvi'o where to look for\
    \ database files through this\nparameter.\n"
  type: File?
  inputBinding:
    prefix: --taxonomy-database
- id: in_it_dot
  doc: 'ADVANCED:'
  type: string
  inputBinding:
    position: 0
outputs:
- id: out_stdout
  doc: Standard output stream
  type: stdout
- id: out_output_file
  doc: File path to store results.
  type: File?
  outputBinding:
    glob: $(inputs.in_output_file)
- id: out_output_file_prefix
  doc: "A prefix to be used while naming the output files (no\nfile type extensions\
    \ please; just a prefix)."
  type: File?
  outputBinding:
    glob: $(inputs.in_output_file_prefix)
hints: []
cwlVersion: v1.1
baseCommand:
- anvi-estimate-scg-taxonomy
