class: CommandLineTool
id: malt_run.cwl
inputs:
- id: in_mode
  doc: '[string]                  Program mode. Mandatory option. Legal values: Unknown,
    BlastN, BlastP, BlastX, Classifier'
  type: boolean?
  inputBinding:
    prefix: --mode
- id: in_alignment_type
  doc: '[string]        Type of alignment to be performed. Default value: Local. Legal
    values: Local, SemiGlobal'
  type: boolean?
  inputBinding:
    prefix: --alignmentType
- id: in_in_file
  doc: '[string(s)]             Input file(s) containing queries in FastA or FastQ
    format (gzip or zip ok). Mandatory option.'
  type: boolean?
  inputBinding:
    prefix: --inFile
- id: in_index
  doc: '[string]                 Index directory as generated by MaltBuild. Mandatory
    option.'
  type: boolean?
  inputBinding:
    prefix: --index
- id: in_output
  doc: '[string(s)]             Output RMA file(s) or directory.'
  type: File?
  inputBinding:
    prefix: --output
- id: in_include_unaligned
  doc: 'Include unaligned queries in RMA output file. Default value: false.'
  type: File?
  inputBinding:
    prefix: --includeUnaligned
- id: in_alignments
  doc: '[string(s)]         Output alignment file(s) or directory or STDOUT.'
  type: File?
  inputBinding:
    prefix: --alignments
- id: in_format
  doc: '[string]                Alignment output format. Default value: SAM. Legal
    values: SAM, Tab, Text'
  type: boolean?
  inputBinding:
    prefix: --format
- id: in_gzip_alignments
  doc: 'Compress alignments using gzip. Default value: true.'
  type: boolean?
  inputBinding:
    prefix: --gzipAlignments
- id: in_sam_soft_clip
  doc: 'Use soft clipping in SAM files (BlastN mode only). Default value: false.'
  type: boolean?
  inputBinding:
    prefix: --samSoftClip
- id: in_sparse_sam
  doc: 'Produce sparse SAM format (smaller, faster, but only suitable for MEGAN).
    Default value: false.'
  type: boolean?
  inputBinding:
    prefix: --sparseSAM
- id: in_out_aligned
  doc: '[string(s)]        Aligned reads output file(s) or directory or STDOUT.'
  type: File?
  inputBinding:
    prefix: --outAligned
- id: in_gzip_aligned
  doc: 'Compress aligned reads output using gzip. Default value: true.'
  type: boolean?
  inputBinding:
    prefix: --gzipAligned
- id: in_out_unaligned
  doc: '[string(s)]      Unaligned reads output file(s) or directory or STDOUT.'
  type: File?
  inputBinding:
    prefix: --outUnaligned
- id: in_gzip_unaligned
  doc: 'Compress unaligned reads output using gzip. Default value: true.'
  type: boolean?
  inputBinding:
    prefix: --gzipUnaligned
- id: in_num_threads
  doc: '[number]            Number of worker threads. Default value: 8.'
  type: boolean?
  inputBinding:
    prefix: --numThreads
- id: in_memory_mode
  doc: '[string]          Memory mode. Default value: load. Legal values: load, page,
    map'
  type: boolean?
  inputBinding:
    prefix: --memoryMode
- id: in_max_tables
  doc: '[number]            Set the maximum number of seed tables to use (0=all).
    Default value: 0.'
  type: boolean?
  inputBinding:
    prefix: --maxTables
- id: in_replicate_query_cache
  doc: 'Cache results for replicated queries. Default value: false.'
  type: boolean?
  inputBinding:
    prefix: --replicateQueryCache
- id: in_min_bit_score
  doc: '[number]           Minimum bit score. Default value: 50.0.'
  type: boolean?
  inputBinding:
    prefix: --minBitScore
- id: in_max_expected
  doc: '[number]           Maximum expected score. Default value: 1.0.'
  type: boolean?
  inputBinding:
    prefix: --maxExpected
- id: in_min_percent_identity
  doc: '[number]   Minimum percent identity. Default value: 0.0.'
  type: boolean?
  inputBinding:
    prefix: --minPercentIdentity
- id: in_max_alignments_per_query
  doc: '[number]   Maximum number of alignments per query. Default value: 25.'
  type: boolean?
  inputBinding:
    prefix: --maxAlignmentsPerQuery
- id: in_max_alignments_per_ref
  doc: '[number]   Maximum number of (non-overlapping) alignments per reference. Default
    value: 1.'
  type: boolean?
  inputBinding:
    prefix: --maxAlignmentsPerRef
- id: in_match_score
  doc: '[number]           Match score. Default value: 2.'
  type: boolean?
  inputBinding:
    prefix: --matchScore
- id: in_mismatch_score
  doc: '[number]        Mismatch score. Default value: -3.'
  type: boolean?
  inputBinding:
    prefix: --mismatchScore
- id: in_set_lambda
  doc: '[number]            Parameter Lambda for BLASTN statistics. Default value:
    0.625.'
  type: boolean?
  inputBinding:
    prefix: --setLambda
- id: in_set_k
  doc: '[number]                  Parameter K for BLASTN statistics. Default value:
    0.41.'
  type: boolean?
  inputBinding:
    prefix: --setK
- id: in_submatrix
  doc: '[string]           Protein substitution matrix to use. Default value: BLOSUM62.
    Legal values: BLOSUM45, BLOSUM50, BLOSUM62, BLOSUM80, BLOSUM90'
  type: boolean?
  inputBinding:
    prefix: --subMatrix
- id: in_forward_only
  doc: 'Align query forward strand only. Default value: false.'
  type: boolean?
  inputBinding:
    prefix: --forwardOnly
- id: in_reverse_only
  doc: 'Align query reverse strand only. Default value: false.'
  type: boolean?
  inputBinding:
    prefix: --reverseOnly
- id: in_top_percent
  doc: '[number]          Top percent value for LCA algorithm. Default value: 10.0.'
  type: boolean?
  inputBinding:
    prefix: --topPercent
- id: in_min_support_percent
  doc: '[number]   Min support value for LCA algorithm as a percent of assigned reads
    (0==off). Default value: 0.001.'
  type: boolean?
  inputBinding:
    prefix: --minSupportPercent
- id: in_min_support
  doc: '[number]          Min support value for LCA algorithm (overrides --minSupportPercent).
    Default value: 0.'
  type: boolean?
  inputBinding:
    prefix: --minSupport
- id: in_min_percent_identity_lca
  doc: '[number]   Min percent identity used by LCA algorithm. Default value: 0.0.'
  type: boolean?
  inputBinding:
    prefix: --minPercentIdentityLCA
- id: in_use_min_percent_identity_filter_lca
  doc: 'Use percent identity assignment filter. Default value: false.'
  type: boolean?
  inputBinding:
    prefix: --useMinPercentIdentityFilterLCA
- id: in_weighted_lca
  doc: 'Use the weighted LCA for taxonomic assignment. Default value: false.'
  type: boolean?
  inputBinding:
    prefix: --weightedLCA
- id: in_lca_coverage_percent
  doc: '[number]   Set the percent for the LCA to cover. Default value: 100.0.'
  type: boolean?
  inputBinding:
    prefix: --lcaCoveragePercent
- id: in_magnitudes
  doc: 'Reads have magnitudes (to be used in taxonomic or functional analysis). Default
    value: false.'
  type: boolean?
  inputBinding:
    prefix: --magnitudes
- id: in_con_file
  doc: '[string]              File of contaminant taxa (one Id or name per line).'
  type: boolean?
  inputBinding:
    prefix: --conFile
- id: in_max_seeds_per_frame
  doc: '[number]    Maximum number of seed matches per offset per read frame. Default
    value: 100.'
  type: boolean?
  inputBinding:
    prefix: --maxSeedsPerFrame
- id: in_max_seeds_per_ref
  doc: '[number]      Maximum number of seed matches per read and reference. Default
    value: 20.'
  type: boolean?
  inputBinding:
    prefix: --maxSeedsPerRef
- id: in_seed_shift
  doc: '[number]            Seed shift. Default value: 1.'
  type: boolean?
  inputBinding:
    prefix: --seedShift
- id: in_gap_open
  doc: '[number]              Gap open penalty. Default value: 11.'
  type: boolean?
  inputBinding:
    prefix: --gapOpen
- id: in_gap_extend
  doc: '[number]            Gap extension penalty. Default value: 1.'
  type: boolean?
  inputBinding:
    prefix: --gapExtend
- id: in_band
  doc: '[number]                 Band width/2 for banded alignment. Default value:
    4.'
  type: boolean?
  inputBinding:
    prefix: --band
- id: in_replicate_query_cache_bits
  doc: '[number]   Bits used for caching replicate queries (size is then 2^bits).
    Default value: 20.'
  type: boolean?
  inputBinding:
    prefix: --replicateQueryCacheBits
- id: in_x_part
  doc: 'Show part of the table in human readable form for debugging. Default value:
    false.'
  type: boolean?
  inputBinding:
    prefix: --xPart
- id: in_verbose
  doc: 'Echo commandline options and be verbose. Default value: false.'
  type: boolean?
  inputBinding:
    prefix: --verbose
outputs:
- id: out_stdout
  doc: Standard output stream
  type: stdout
- id: out_output
  doc: '[string(s)]             Output RMA file(s) or directory.'
  type: File?
  outputBinding:
    glob: $(inputs.in_output)
- id: out_include_unaligned
  doc: 'Include unaligned queries in RMA output file. Default value: false.'
  type: File?
  outputBinding:
    glob: $(inputs.in_include_unaligned)
- id: out_alignments
  doc: '[string(s)]         Output alignment file(s) or directory or STDOUT.'
  type: File?
  outputBinding:
    glob: $(inputs.in_alignments)
- id: out_out_aligned
  doc: '[string(s)]        Aligned reads output file(s) or directory or STDOUT.'
  type: File?
  outputBinding:
    glob: $(inputs.in_out_aligned)
- id: out_out_unaligned
  doc: '[string(s)]      Unaligned reads output file(s) or directory or STDOUT.'
  type: File?
  outputBinding:
    glob: $(inputs.in_out_unaligned)
hints: []
cwlVersion: v1.1
baseCommand:
- malt-run
