from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Boolean, Int

Phigaro_V0_1_0 = CommandToolBuilder(tool="phigaro", base_command=["phigaro"], inputs=[ToolInput(tag="in_fast_a_file", input_type=File(optional=True), prefix="--fasta-file", doc=InputDocumentation(doc="Assembly scaffolds/contigs or full genomes, required")), ToolInput(tag="in_config", input_type=File(optional=True), prefix="--config", doc=InputDocumentation(doc="Path to the config file, not required. The deafult is\n/root/.phigaro/config.yml")), ToolInput(tag="in_print_vogs", input_type=Boolean(optional=True), prefix="--print-vogs", doc=InputDocumentation(doc="Print phage vogs for each region")), ToolInput(tag="in_extension", input_type=File(optional=True), prefix="--extension", doc=InputDocumentation(doc="Type of the output: html, tsv, gff, bed or stdout.\nDefault is html. You can specify several file formats\nwith a space as a separator. Example: -e tsv html\nstdout.")), ToolInput(tag="in_output", input_type=File(optional=True), prefix="--output", doc=InputDocumentation(doc="Output filename for html and txt outputs. Required by\ndefault, but not required for stdout only output.")), ToolInput(tag="in_not_open", input_type=File(optional=True), prefix="--not-open", doc=InputDocumentation(doc="Do not open html file automatically, if html output\ntype is specified.")), ToolInput(tag="in_threads", input_type=Int(optional=True), prefix="--threads", doc=InputDocumentation(doc="Num of threads (default is num of CPUs=8)")), ToolInput(tag="in_no_cleanup", input_type=Boolean(optional=True), prefix="--no-cleanup", doc=InputDocumentation(doc="Do not delete any temporary files that was generated\nby Phigaro (HMMER & Prodigal outputs and some others).")), ToolInput(tag="in_substitute_output", input_type=File(optional=True), prefix="--substitute-output", doc=InputDocumentation(doc="If you have precomputed prodigal and/or hmmer data you\ncan provide paths to the files in the following\nformat: program:address/to/the/file. In place of\nprogram you should write hmmer or prodigal. If you\nneed to provide both files you should pass them\nseparetely as two parametres.")), ToolInput(tag="in_save_fast_a", input_type=Boolean(optional=True), prefix="--save-fasta", doc=InputDocumentation(doc="Save all phage fasta sequences in a fasta file.")), ToolInput(tag="in_delete_shorts", input_type=Boolean(optional=True), prefix="--delete-shorts", doc=InputDocumentation(doc="Exclude sequences with length < 20000 automatically.")), ToolInput(tag="in_mode", input_type=Int(optional=True), prefix="--mode", doc=InputDocumentation(doc="You can launch Phigaro at one of 3 modes: basic, abs,\nwithout_gc. Default is basic. Read more about modes at\nhttps://github.com/bobeobibo/phigaro/\n")), ToolInput(tag="in_v", input_type=Boolean(optional=True), prefix="-V", doc=InputDocumentation(doc=""))], outputs=[ToolOutput(tag="out_extension", output_type=File(optional=True), selector=InputSelector(input_to_select="in_extension", type_hint=File()), doc=OutputDocumentation(doc="Type of the output: html, tsv, gff, bed or stdout.\nDefault is html. You can specify several file formats\nwith a space as a separator. Example: -e tsv html\nstdout.")), ToolOutput(tag="out_output", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output", type_hint=File()), doc=OutputDocumentation(doc="Output filename for html and txt outputs. Required by\ndefault, but not required for stdout only output.")), ToolOutput(tag="out_not_open", output_type=File(optional=True), selector=InputSelector(input_to_select="in_not_open", type_hint=File()), doc=OutputDocumentation(doc="Do not open html file automatically, if html output\ntype is specified."))], container="quay.io/biocontainers/phigaro:2.3.0--pyh7b7c402_1", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Phigaro_V0_1_0().translate("wdl")

