version 1.0

task Kissreads2 {
  input {
    Boolean? phasing
    Boolean? arg_option_homogeneous
    Boolean? predictions
    Boolean? reads
    Boolean? unco
    Boolean? co
    Boolean? hamming
    Boolean? coverage_file
    Boolean? arg_size_k
    Boolean? index_stride
    Boolean? size_seeds
    Boolean? output_fast_a
    Boolean? genotype
    Boolean? nb_cores
    Boolean? verbose
    Boolean? version
  }
  command <<<
    kissreads2 \
      ~{true="-phasing" false="" phasing} \
      ~{true="-x" false="" arg_option_homogeneous} \
      ~{true="-predictions" false="" predictions} \
      ~{true="-reads" false="" reads} \
      ~{true="-unco" false="" unco} \
      ~{true="-co" false="" co} \
      ~{true="-hamming" false="" hamming} \
      ~{true="-coverage_file" false="" coverage_file} \
      ~{true="-k" false="" arg_size_k} \
      ~{true="-index_stride" false="" index_stride} \
      ~{true="-size_seeds" false="" size_seeds} \
      ~{true="-output_fasta" false="" output_fast_a} \
      ~{true="-genotype" false="" genotype} \
      ~{true="-nb-cores" false="" nb_cores} \
      ~{true="-verbose" false="" verbose} \
      ~{true="-version" false="" version}
  >>>
  parameter_meta {
    phasing: "(0 arg) :    [Experimental] Phase alleles mapped by the same (pair of) read(s)"
    arg_option_homogeneous: "(0 arg) :    radseq option, homogeneous read depth along the prediction "
    predictions: "(1 arg) :    Input predictions"
    reads: "(1 arg) :    Input reads"
    unco: "(1 arg) :    Output uncoherent file name  [default '/dev/null']"
    co: "(1 arg) :    Output coherent file name"
    hamming: "(1 arg) :    Maximal hamming distance authorized while maping  [default '1']"
    coverage_file: "(1 arg) :    File (.h5) generated by kissnp2, containing the coverage threshold per read set  [default '_removemeplease']"
    arg_size_k: "(1 arg) :    Size of k, used as minial overlap and kmer spanning read coherence  [default '31']"
    index_stride: "(1 arg) :    Index Stride  [default '2']"
    size_seeds: "(1 arg) :    Size of the used seeds (distinct from the size of k)  [default '25']"
    output_fast_a: "(0 arg) :    Output standart Fasta. By default the output is formatted especially for the discoSnp++ pipeline"
    genotype: "(0 arg) :    Compute genotypes"
    nb_cores: "(1 arg) :    number of cores  [default '0']"
    verbose: "(1 arg) :    verbosity level  [default '1']"
    version: "(0 arg) :    version"
  }
}