from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, String

Pbmm2_Align_V0_1_0 = CommandToolBuilder(tool="pbmm2_align", base_command=["pbmm2", "align"], inputs=[ToolInput(tag="in_chunk_size", input_type=Boolean(optional=True), prefix="--chunk-size", doc=InputDocumentation(doc="INT    Process N records per chunk. [100]")), ToolInput(tag="in_sort", input_type=Boolean(optional=True), prefix="--sort", doc=InputDocumentation(doc="Generate sorted BAM file.")), ToolInput(tag="in_sort_memory", input_type=Boolean(optional=True), prefix="--sort-memory", doc=InputDocumentation(doc="STR    Memory per thread for sorting. [768M]")), ToolInput(tag="in_sort_threads", input_type=Boolean(optional=True), prefix="--sort-threads", doc=InputDocumentation(doc="INT    Number of threads used for sorting; 0 means 25% of -j, maximum 8. [0]")), ToolInput(tag="in_preset", input_type=Boolean(optional=True), prefix="--preset", doc=InputDocumentation(doc="STR    Set alignment mode. See below for preset parameter details. Valid choices:\n(SUBREAD, CCS, HIFI, ISOSEQ, UNROLLED). [SUBREAD]")), ToolInput(tag="in_int_kmer_size", input_type=Boolean(optional=True), prefix="-k", doc=InputDocumentation(doc="INT    k-mer size (no larger than 28). [-1]")), ToolInput(tag="in_int_minimizer_window", input_type=Boolean(optional=True), prefix="-w", doc=InputDocumentation(doc="INT    Minimizer window size. [-1]")), ToolInput(tag="in_no_km_er_compression", input_type=Boolean(optional=True), prefix="--no-kmer-compression", doc=InputDocumentation(doc="Disable homopolymer-compressed k-mer (compression is active for SUBREAD & UNROLLED\npresets).")), ToolInput(tag="in_int_matching_score", input_type=Boolean(optional=True), prefix="-A", doc=InputDocumentation(doc="INT    Matching score. [-1]")), ToolInput(tag="in_int_mismatch_penalty", input_type=Boolean(optional=True), prefix="-B", doc=InputDocumentation(doc="INT    Mismatch penalty. [-1]")), ToolInput(tag="in_int_zdrop_score", input_type=Boolean(optional=True), prefix="-z", doc=InputDocumentation(doc="INT    Z-drop score. [-1]")), ToolInput(tag="in_int_zdrop_inversion", input_type=Boolean(optional=True), prefix="-Z", doc=InputDocumentation(doc="INT    Z-drop inversion score. [-1]")), ToolInput(tag="in_int_bandwidth_used", input_type=Boolean(optional=True), prefix="-r", doc=InputDocumentation(doc="INT    Bandwidth used in chaining and DP-based alignment. [-1]")), ToolInput(tag="in_int_stop_enlongation", input_type=Boolean(optional=True), prefix="-g", doc=InputDocumentation(doc="INT    Stop chain enlongation if there are no minimizers in N bp. [-1]")), ToolInput(tag="in_gap_open_one", input_type=Boolean(optional=True), prefix="--gap-open-1", doc=InputDocumentation(doc="INT    Gap open penalty 1. [-1]")), ToolInput(tag="in_gap_open_two", input_type=Boolean(optional=True), prefix="--gap-open-2", doc=InputDocumentation(doc="INT    Gap open penalty 2. [-1]")), ToolInput(tag="in_gap_extend_one", input_type=Boolean(optional=True), prefix="--gap-extend-1", doc=InputDocumentation(doc="INT    Gap extension penalty 1. [-1]")), ToolInput(tag="in_gap_extend_two", input_type=Boolean(optional=True), prefix="--gap-extend-2", doc=InputDocumentation(doc="INT    Gap extension penalty 2. [-1]")), ToolInput(tag="in_lj_min_ratio", input_type=Boolean(optional=True), prefix="--lj-min-ratio", doc=InputDocumentation(doc="FLOAT  Long join flank ratio. [-1]")), ToolInput(tag="in_int_max_intron", input_type=Boolean(optional=True), prefix="-G", doc=InputDocumentation(doc="INT    Max intron length (changes -r). [-1]")), ToolInput(tag="in_int_cost_noncanonical", input_type=Boolean(optional=True), prefix="-C", doc=InputDocumentation(doc="INT    Cost for a non-canonical GT-AG splicing (effective in ISOSEQ preset). [-1]")), ToolInput(tag="in_no_splice_flank", input_type=Boolean(optional=True), prefix="--no-splice-flank", doc=InputDocumentation(doc="Do not prefer splice flanks GT-AG (effective in ISOSEQ preset).")), ToolInput(tag="in_sample", input_type=Boolean(optional=True), prefix="--sample", doc=InputDocumentation(doc="STR    Sample name for all read groups. Defaults, in order of precedence: SM field in\ninput read group, biosample name, well sample name, 'UnnamedSample'.")), ToolInput(tag="in_rg", input_type=Boolean(optional=True), prefix="--rg", doc=InputDocumentation(doc="STR    Read group header line such as '@RG\tID:xyz\tSM:abc'. Only for FASTA/Q inputs.")), ToolInput(tag="in_min_concordance_perc", input_type=Boolean(optional=True), prefix="--min-concordance-perc", doc=InputDocumentation(doc="FLOAT  Minimum alignment concordance in percent. [70]")), ToolInput(tag="in_min_id_perc", input_type=Boolean(optional=True), prefix="--min-id-perc", doc=InputDocumentation(doc="FLOAT  Minimum sequence identity in percent. [0]")), ToolInput(tag="in_min_gap_comp_id_perc", input_type=Boolean(optional=True), prefix="--min-gap-comp-id-perc", doc=InputDocumentation(doc="FLOAT  Minimum gap compressed sequence identity in percent. [0]")), ToolInput(tag="in_min_length", input_type=Boolean(optional=True), prefix="--min-length", doc=InputDocumentation(doc="INT    Minimum mapped read length in basepairs. [50]")), ToolInput(tag="in_best_n", input_type=Boolean(optional=True), prefix="--best-n", doc=InputDocumentation(doc="INT    Output at maximum N alignments for each read, 0 means no maximum. [0]")), ToolInput(tag="in_strip", input_type=Boolean(optional=True), prefix="--strip", doc=InputDocumentation(doc="Remove all kinetic and extra QV tags. Output cannot be polished.")), ToolInput(tag="in_split_by_sample", input_type=Boolean(optional=True), prefix="--split-by-sample", doc=InputDocumentation(doc="One output BAM per sample.")), ToolInput(tag="in_unmapped", input_type=Boolean(optional=True), prefix="--unmapped", doc=InputDocumentation(doc="Include unmapped records in output.")), ToolInput(tag="in_bam_index", input_type=Boolean(optional=True), prefix="--bam-index", doc=InputDocumentation(doc="STR    Generate index for sorted BAM output. Valid choices: (NONE, BAI, CSI). [BAI]")), ToolInput(tag="in_median_filter", input_type=Boolean(optional=True), prefix="--median-filter", doc=InputDocumentation(doc="Pick one read per ZMW of median length.")), ToolInput(tag="in_z_mw", input_type=Boolean(optional=True), prefix="--zmw", doc=InputDocumentation(doc="Process ZMW Reads, subreadset.xml input required (activates UNROLLED preset).")), ToolInput(tag="in_hq_region", input_type=Boolean(optional=True), prefix="--hqregion", doc=InputDocumentation(doc="Process HQ region of each ZMW, subreadset.xml input required (activates UNROLLED\npreset).")), ToolInput(tag="in_collapse_homopolymers", input_type=Boolean(optional=True), prefix="--collapse-homopolymers", doc=InputDocumentation(doc="Collapse homopolymers in reads and reference.")), ToolInput(tag="in_num_threads", input_type=Boolean(optional=True), prefix="--num-threads", doc=InputDocumentation(doc="INT    Number of threads to use, 0 means autodetection. [0]")), ToolInput(tag="in_log_level", input_type=Boolean(optional=True), prefix="--log-level", doc=InputDocumentation(doc="STR    Set log level. Valid choices: (TRACE, DEBUG, INFO, WARN, FATAL). [WARN]")), ToolInput(tag="in_log_file", input_type=Boolean(optional=True), prefix="--log-file", doc=InputDocumentation(doc="FILE   Log to a file, instead of stderr.")), ToolInput(tag="in_out_dot_aligned_dot_bam_vertical_line_xml", input_type=String(), position=0, doc=InputDocumentation(doc="")), ToolInput(tag="in_ref_dot_f_a_vertical_line_xml_vertical_line_mmi", input_type=String(), position=0, doc=InputDocumentation(doc="")), ToolInput(tag="in_str", input_type=String(), position=1, doc=InputDocumentation(doc="")), ToolInput(tag="in_in_dot_bam_vertical_line_xml_vertical_line_f_a_vertical_line_fq_vertical_line_gz_vertical_line_fof_n", input_type=String(), position=1, doc=InputDocumentation(doc="")), ToolInput(tag="in_input", input_type=String(), position=2, doc=InputDocumentation(doc="")), ToolInput(tag="in_output", input_type=String(), position=2, doc=InputDocumentation(doc="")), ToolInput(tag="in_reference", input_type=String(), position=2, doc=InputDocumentation(doc="")), ToolInput(tag="in_bam", input_type=String(), position=3, doc=InputDocumentation(doc="")), ToolInput(tag="in_fast_a", input_type=String(), position=3, doc=InputDocumentation(doc="")), ToolInput(tag="in_or", input_type=String(), position=4, doc=InputDocumentation(doc="")), ToolInput(tag="in_dataset", input_type=String(), position=5, doc=InputDocumentation(doc=""))], outputs=[], container="quay.io/biocontainers/pbmm2:1.4.0--h56fc30b_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Pbmm2_Align_V0_1_0().translate("wdl")

