from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, Float, Int, String, File

Varcall_V0_1_0 = CommandToolBuilder(tool="varcall", base_command=["varcall"], inputs=[ToolInput(tag="in_calculate_statistics", input_type=Boolean(optional=True), prefix="-s", doc=InputDocumentation(doc="Calculate statistics")), ToolInput(tag="in_version_calculate_variants", input_type=Boolean(optional=True), prefix="-v", doc=InputDocumentation(doc="|version  Calculate variants bases on supplied parameters (see -S)")), ToolInput(tag="in_reference_fasta_required", input_type=Boolean(optional=True), prefix="-f", doc=InputDocumentation(doc="Reference fasta (required if using bams, ignored otherwise)")), ToolInput(tag="in_min_locii_depth", input_type=Boolean(optional=True), prefix="-m", doc=InputDocumentation(doc="Min locii depth (1)")), ToolInput(tag="in_min_allele_depth", input_type=Boolean(optional=True), prefix="-a", doc=InputDocumentation(doc="Min allele depth (2)")), ToolInput(tag="in_min_allele_pct", input_type=Boolean(optional=True), prefix="-p", doc=InputDocumentation(doc="Min allele pct by quality (0)")), ToolInput(tag="in_min_qual", input_type=Boolean(optional=True), prefix="-q", doc=InputDocumentation(doc="Min qual (3)")), ToolInput(tag="in_min_mapping_quality", input_type=Boolean(optional=True), prefix="-Q", doc=InputDocumentation(doc="Min mapping quality (0)")), ToolInput(tag="in_min_pct_balance", input_type=Boolean(optional=True), prefix="-b", doc=InputDocumentation(doc="Min pct balance (strand/total) (0)")), ToolInput(tag="in_max_duplicate_read", input_type=Float(optional=True), prefix="-D", doc=InputDocumentation(doc="Max duplicate read fraction (depth/length per position) (1)")), ToolInput(tag="in_minimum_diversity_cv", input_type=Float(optional=True), prefix="-d", doc=InputDocumentation(doc="Minimum diversity (CV from optimal depth) (0.25)")), ToolInput(tag="in_minimum_agreement_cv", input_type=Float(optional=True), prefix="-G", doc=InputDocumentation(doc="Minimum agreement (Weighted CV of positional variation) (0.25)")), ToolInput(tag="in_zero_filters_set", input_type=Boolean(optional=True), prefix="-0", doc=InputDocumentation(doc="Zero out all filters, set e-value filter to 1, report everything")), ToolInput(tag="in_running_bam_turn", input_type=Boolean(optional=True), prefix="-B", doc=InputDocumentation(doc="If running from a BAM, turn off BAQ correction (false)")), ToolInput(tag="in_homopolymer_repeat_indel", input_type=Boolean(optional=True), prefix="-R", doc=InputDocumentation(doc="Homopolymer repeat indel filtering (8)")), ToolInput(tag="in_alpha_filter_use", input_type=Float(optional=True), prefix="-e", doc=InputDocumentation(doc="Alpha filter to use, requires -l or -S (.05)")), ToolInput(tag="in_global_minimum_error", input_type=Float(optional=True), prefix="-g", doc=InputDocumentation(doc="Global minimum error rate (default: assume phred is ok)")), ToolInput(tag="in_number_locii_mil", input_type=Int(optional=True), prefix="-l", doc=InputDocumentation(doc="Number of locii in total pileup used for bonferroni (1 mil)")), ToolInput(tag="in_pos_output_pos", input_type=String(optional=True), prefix="-x", doc=InputDocumentation(doc=":POS  Output this pos only, then quit")), ToolInput(tag="in_read_statistics_params", input_type=File(optional=True), prefix="-S", doc=InputDocumentation(doc="Read in statistics and params from a previous run with -s (do this!)")), ToolInput(tag="in_calculate_intarget_stats", input_type=File(optional=True), prefix="-A", doc=InputDocumentation(doc="Calculate in-target stats using the annotation file (requires -o)")), ToolInput(tag="in_output_prefix_works", input_type=String(optional=True), prefix="-o", doc=InputDocumentation(doc="Output prefix (works with -s or -v)")), ToolInput(tag="in_list_output_vcf", input_type=File(optional=True), prefix="-F", doc=InputDocumentation(doc="List of file types to output (var, varsum, eav, vcf)")), ToolInput(tag="in_pcr_an_not", input_type=Boolean(optional=True), prefix="--pcr-annot", doc=InputDocumentation(doc="BED      Only include reads adhering to the expected amplicons")), ToolInput(tag="in_stranded", input_type=Boolean(optional=True), prefix="--stranded", doc=InputDocumentation(doc="TYPE     Can be FR (the default), FF, FR.  Used with pcr-annot")), ToolInput(tag="in_diversity", input_type=Boolean(optional=True), prefix="--diversity", doc=InputDocumentation(doc="|d FLOAT    Alias for -d")), ToolInput(tag="in_agreement", input_type=Boolean(optional=True), prefix="--agreement", doc=InputDocumentation(doc="|G FLOAT    Alias for -G")), ToolInput(tag="in_no_indels", input_type=Boolean(optional=True), prefix="--no-indels", doc=InputDocumentation(doc="Ignore all indels")), ToolInput(tag="in_prefix_dot_var", input_type=String(), position=0, doc=InputDocumentation(doc="Variant calls in tab delimited 'varcall' format")), ToolInput(tag="in_prefix_do_tea_v", input_type=String(), position=1, doc=InputDocumentation(doc="Variant calls in tab delimited 'ea-var' format")), ToolInput(tag="in_prefix_dot_cse", input_type=String(), position=2, doc=InputDocumentation(doc="Variant calls in tab delimited 'varprowl' format")), ToolInput(tag="in_prefix_dot_vcf", input_type=String(), position=3, doc=InputDocumentation(doc="Variant calls, in vcf format")), ToolInput(tag="in_prefix_dot_var_sum", input_type=String(), position=4, doc=InputDocumentation(doc="Summary of variant calls")), ToolInput(tag="in_prefix_dot_tgt_dot_var", input_type=String(), position=5, doc=InputDocumentation(doc="On-target version of .var")), ToolInput(tag="in_prefix_dot_tgt_dot_cse", input_type=String(), position=6, doc=InputDocumentation(doc="On-target version of .cse")), ToolInput(tag="in_prefix_dot_tgt_dot_var_sum", input_type=String(), position=7, doc=InputDocumentation(doc="On-target version of .varsum")), ToolInput(tag="in_prefix_dot_stats", input_type=String(), position=0, doc=InputDocumentation(doc="Stats output")), ToolInput(tag="in_prefix_dot_noise", input_type=String(), position=1, doc=InputDocumentation(doc="Non-reference, non-homozygous allele summary")), ToolInput(tag="in_prefix_do_tx_noise", input_type=String(), position=2, doc=InputDocumentation(doc="Like noise, but with context-specific rates"))], outputs=[ToolOutput(tag="out_list_output_vcf", output_type=File(optional=True), selector=InputSelector(input_to_select="in_list_output_vcf", type_hint=File()), doc=OutputDocumentation(doc="List of file types to output (var, varsum, eav, vcf)"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Varcall_V0_1_0().translate("wdl", allow_empty_container=True)

