from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, Int, String

Ataqv_V0_1_0 = CommandToolBuilder(tool="ataqv", base_command=["ataqv"], inputs=[ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="--verbose", doc=InputDocumentation(doc=": show more details and progress updates.")), ToolInput(tag="in_threads", input_type=Int(optional=True), prefix="--threads", doc=InputDocumentation(doc=": the maximum number of threads to use (right now, only for calculating TSS enrichment).")), ToolInput(tag="in_peak_file", input_type=Boolean(optional=True), prefix="--peak-file", doc=InputDocumentation(doc="'file name'\nA BED file of peaks called for alignments in the BAM file. Specify 'auto' to use the\nBAM file name with '.peaks' appended, or if the BAM file contains read groups, to\nassume each read group has a peak file whose name is the read group ID with '.peaks'\nappended. If you specify a single filename instead of 'auto' with read groups, the\nsame peaks will be used for all reads -- be sure this is what you want.")), ToolInput(tag="in_tss_file", input_type=Boolean(optional=True), prefix="--tss-file", doc=InputDocumentation(doc="'file name'\nA BED file of transcription start sites for the experiment organism. If supplied,\na TSS enrichment score will be calculated according to the ENCODE data standards.\nThis calculation requires that the BAM file of alignments be indexed.")), ToolInput(tag="in_tss_extension", input_type=Boolean(optional=True), prefix="--tss-extension", doc=InputDocumentation(doc="'size'\nIf a TSS enrichment score is requested, it will be calculated for a region of\n'size' bases to either side of transcription start sites. The default is 1000bp.")), ToolInput(tag="in_excluded_region_file", input_type=Boolean(optional=True), prefix="--excluded-region-file", doc=InputDocumentation(doc="'file name'\nA BED file containing excluded regions. Peaks or TSS overlapping these will be ignored.\nMay be given multiple times.")), ToolInput(tag="in_metrics_file", input_type=Boolean(optional=True), prefix="--metrics-file", doc=InputDocumentation(doc="'file name'\nThe JSON file to which metrics will be written. The default filename will be based on\nthe BAM file, with the suffix '.ataqv.json'.")), ToolInput(tag="in_log_problematic_reads", input_type=Boolean(optional=True), prefix="--log-problematic-reads", doc=InputDocumentation(doc="If given, problematic reads will be logged to a file per read group, with names\nderived from the read group IDs, with '.problems' appended. If no read groups\nare found, the reads will be written to one file named after the BAM file.")), ToolInput(tag="in_name", input_type=Boolean(optional=True), prefix="--name", doc=InputDocumentation(doc="'name'\nA label to be used for the metrics when there are no read groups. If there are read\ngroups, each will have its metrics named using its ID field. With no read groups and\nno --name given, your metrics will be named after the alignment file.")), ToolInput(tag="in_ignore_read_groups", input_type=Boolean(optional=True), prefix="--ignore-read-groups", doc=InputDocumentation(doc="Even if read groups are present in the BAM file, ignore them and combine metrics\nfor all reads under a single sample and library named with the --name option. This\nalso implies that a single peak file will be used for all reads; see the --peak option.")), ToolInput(tag="in_description", input_type=Boolean(optional=True), prefix="--description", doc=InputDocumentation(doc="'description'\nA short description of the experiment.")), ToolInput(tag="in_url", input_type=Boolean(optional=True), prefix="--url", doc=InputDocumentation(doc="'URL'\nA URL for more detail on the experiment (perhaps using a DOI).")), ToolInput(tag="in_library_description", input_type=Boolean(optional=True), prefix="--library-description", doc=InputDocumentation(doc="'description'\nUse this description for all libraries in the BAM file, instead of using the DS\nfield from each read group.")), ToolInput(tag="in_autosomal_reference_file", input_type=Boolean(optional=True), prefix="--autosomal-reference-file", doc=InputDocumentation(doc="'file name'\nA file containing autosomal reference names, one per line. The names must match\nthe reference names in the alignment file exactly, or the metrics based on counts\nof autosomal alignments will be wrong.")), ToolInput(tag="in_mitochondrial_reference_name", input_type=Boolean(optional=True), prefix="--mitochondrial-reference-name", doc=InputDocumentation(doc="'name'\nIf the reference name for mitochondrial DNA in your alignment file is not 'chrM',.\nuse this option to supply the correct name. Again, if this name is wrong, all the\nmeasurements involving mitochondrial alignments will be wrong.\n")), ToolInput(tag="in_organism", input_type=String(), position=0, doc=InputDocumentation(doc="Autosomal References\n-------  ------------------\nfly  2R 2L 3R 3L 4\nhuman  1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22\nmouse  1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19\nrat  1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20\nworm  I II III IV V\nyeast  I II III IV V VI VII VIII IX X XI XII XIII XIV XV XVI"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Ataqv_V0_1_0().translate("wdl", allow_empty_container=True)

