from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, String, Boolean, Float, Int

Motus_Profile_V0_1_0 = CommandToolBuilder(tool="motus_profile", base_command=["motus", "profile"], inputs=[ToolInput(tag="in_input_files_reads_forward_formatted", input_type=File(), prefix="-f", doc=InputDocumentation(doc="input file(s) for reads in forward orientation, fastq formatted")), ToolInput(tag="in_input_files_reads_reverse_formatted", input_type=File(), prefix="-r", doc=InputDocumentation(doc="input file(s) for reads in reverse orientation, fastq formatted")), ToolInput(tag="in_input_files_reads_mate_formatted", input_type=File(), prefix="-s", doc=InputDocumentation(doc="input file(s) for reads without mate, fastq formatted")), ToolInput(tag="in_sample_name", input_type=String(optional=True), prefix="-n", doc=InputDocumentation(doc="sample name")), ToolInput(tag="in_provide_sam_bam", input_type=File(), prefix="-i", doc=InputDocumentation(doc="provide sam or bam input file(s)")), ToolInput(tag="in_provide_mgc_reads", input_type=File(optional=True), prefix="-m", doc=InputDocumentation(doc="provide a mgc reads count file")), ToolInput(tag="in_output_file_name", input_type=File(optional=True), prefix="-o", doc=InputDocumentation(doc="output file name [stdout]")), ToolInput(tag="in_save_result_step", input_type=File(optional=True), prefix="-I", doc=InputDocumentation(doc="save the result of bwa in bam format (intermediate step) [None]")), ToolInput(tag="in_save_mgc_reads", input_type=File(optional=True), prefix="-M", doc=InputDocumentation(doc="save the mgc reads count (intermediate step) [None]")), ToolInput(tag="in_profile_only_species", input_type=Boolean(optional=True), prefix="-e", doc=InputDocumentation(doc="profile only reference species (ref_mOTUs)")), ToolInput(tag="in_print_result_counts", input_type=Boolean(optional=True), prefix="-c", doc=InputDocumentation(doc="print result as counts instead of relative abundances")), ToolInput(tag="in_print_ncbi_i", input_type=Boolean(optional=True), prefix="-p", doc=InputDocumentation(doc="print NCBI id")), ToolInput(tag="in_print_full_name", input_type=Boolean(optional=True), prefix="-u", doc=InputDocumentation(doc="print the full name of the species")), ToolInput(tag="in_print_rank_taxonomy", input_type=Boolean(optional=True), prefix="-q", doc=InputDocumentation(doc="print the full rank taxonomy")), ToolInput(tag="in_print_result_biom", input_type=Boolean(optional=True), prefix="-B", doc=InputDocumentation(doc="print result in BIOM format")), ToolInput(tag="in_print_result_cami", input_type=Float(optional=True), prefix="-C", doc=InputDocumentation(doc="print result in CAMI format (BioBoxes format 0.9.1)\nValues: [precision, recall, parenthesis]")), ToolInput(tag="in_taxonomic_level_values", input_type=String(optional=True), prefix="-k", doc=InputDocumentation(doc="taxonomic level [mOTU]\nValues: [kingdom, phylum, class, order, family, genus, mOTU]")), ToolInput(tag="in_number_marker_genes", input_type=Int(optional=True), prefix="-g", doc=InputDocumentation(doc="number of marker genes cutoff: 1=higher recall, 10=higher precision [3]")), ToolInput(tag="in_min_length_alignment", input_type=Int(optional=True), prefix="-l", doc=InputDocumentation(doc="min. length of alignment for the reads (number of nucleotides) [75]")), ToolInput(tag="in_number_of_threads", input_type=Int(optional=True), prefix="-t", doc=InputDocumentation(doc="number of threads [1]")), ToolInput(tag="in_verbose_level_error", input_type=Int(optional=True), prefix="-v", doc=InputDocumentation(doc="verbose level: 1=error, 2=warning, 3=message, 4+=debugging [3]")), ToolInput(tag="in_type_read_counts", input_type=String(optional=True), prefix="-y", doc=InputDocumentation(doc="type of read counts [insert.scaled_counts]\nValues: [base.coverage, insert.raw_counts, insert.scaled_counts]\n"))], outputs=[ToolOutput(tag="out_output_file_name", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output_file_name", type_hint=File()), doc=OutputDocumentation(doc="output file name [stdout]"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Motus_Profile_V0_1_0().translate("wdl", allow_empty_container=True)

