from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, Int, String, File

Pmdtools_V0_1_0 = CommandToolBuilder(tool="pmdtools", base_command=["pmdtools"], inputs=[ToolInput(tag="in_adjusts_s", input_type=Boolean(optional=True), prefix="--adjustss", doc=InputDocumentation(doc="strand orientation aware penalization of base\nqualities for single stranded libraries (use with\n--CpG for udg-treated ss libs)")), ToolInput(tag="in_number", input_type=Int(optional=True), prefix="--number", doc=InputDocumentation(doc="stop after these many reads have been processed")), ToolInput(tag="in_chromosome", input_type=String(optional=True), prefix="--chromosome", doc=InputDocumentation(doc="only process data from this chromosome")), ToolInput(tag="in_require_mapq", input_type=String(optional=True), prefix="--requiremapq", doc=InputDocumentation(doc="only process sequences with mapping quality at least\nthis great")), ToolInput(tag="in_read_length", input_type=Int(optional=True), prefix="--readlength", doc=InputDocumentation(doc="only process sequences with this read length")), ToolInput(tag="in_maxlength", input_type=Int(optional=True), prefix="--maxlength", doc=InputDocumentation(doc="only process sequences with max this read length")), ToolInput(tag="in_minlength", input_type=Int(optional=True), prefix="--minlength", doc=InputDocumentation(doc="only process sequences with min this read length")), ToolInput(tag="in_max_gc", input_type=Int(optional=True), prefix="--maxGC", doc=InputDocumentation(doc="only process sequences with max this GC content of the\naligning reference sequence")), ToolInput(tag="in_ming_c", input_type=Int(optional=True), prefix="--minGC", doc=InputDocumentation(doc="only process sequences with min this GC content of the\naligning reference sequence")), ToolInput(tag="in_require_base_q", input_type=String(optional=True), prefix="--requirebaseq", doc=InputDocumentation(doc="only process bases with base quality at least this\ngreat")), ToolInput(tag="in_deamination", input_type=Boolean(optional=True), prefix="--deamination", doc=InputDocumentation(doc="output base frequencies in the read at positions where\nthere are C or G in the reference")), ToolInput(tag="in_udg_plus", input_type=Boolean(optional=True), prefix="--UDGplus", doc=InputDocumentation(doc="only use Cs and Gs in CpG context")), ToolInput(tag="in_ss", input_type=Boolean(optional=True), prefix="--ss", doc=InputDocumentation(doc="single stranded")), ToolInput(tag="in_pm_dsp_rim", input_type=Boolean(optional=True), prefix="--PMDSprim", doc=InputDocumentation(doc="PMDSprim")), ToolInput(tag="in_udg_minus", input_type=Boolean(optional=True), prefix="--UDGminus", doc=InputDocumentation(doc="use all bases (placeholder)")), ToolInput(tag="in_ecoli_cpg", input_type=Boolean(optional=True), prefix="--EcoliCpG", doc=InputDocumentation(doc="Ecoli CpG (first 5' position and two 3' positions and\nCpG context)")), ToolInput(tag="in_ecoli", input_type=Boolean(optional=True), prefix="--Ecoli", doc=InputDocumentation(doc="Ecoli (first 5' position and two 3' positions only)")), ToolInput(tag="in_masks_s", input_type=Boolean(optional=True), prefix="--maskss", doc=InputDocumentation(doc="mask to lower case bases where ref is C (taking strand\ninto account)")), ToolInput(tag="in_leipzig_simple", input_type=Boolean(optional=True), prefix="--Leipzigsimple", doc=InputDocumentation(doc="Leipzigsimple")), ToolInput(tag="in_no_cpg", input_type=Boolean(optional=True), prefix="--noCpG", doc=InputDocumentation(doc="dont use Cs and Gs in CpG context")), ToolInput(tag="in_first", input_type=Boolean(optional=True), prefix="--first", doc=InputDocumentation(doc="outputs the deamination rate at the first position\nonly, but with a standard error")), ToolInput(tag="in_range", input_type=Int(optional=True), prefix="--range", doc=InputDocumentation(doc="output deamination patterns for this many positions\nfrom the sequence terminus (default=30)")), ToolInput(tag="in_polymorphism_ancient", input_type=String(optional=True), prefix="--polymorphism_ancient", doc=InputDocumentation(doc="True biological polymorphism between the ancient\nindividual and the reference sequence")), ToolInput(tag="in_polymorphism_contamination", input_type=String(optional=True), prefix="--polymorphism_contamination", doc=InputDocumentation(doc="True biological polymorphism between the contaminants\nand the reference sequence")), ToolInput(tag="in_pm_dp_param", input_type=String(optional=True), prefix="--PMDpparam", doc=InputDocumentation(doc="parameter p in geometric probability distribution of\nPMD")), ToolInput(tag="in_pmd_constant", input_type=String(optional=True), prefix="--PMDconstant", doc=InputDocumentation(doc="constant C in geometric probability distribution of\nPMD")), ToolInput(tag="in_no_clips", input_type=Boolean(optional=True), prefix="--noclips", doc=InputDocumentation(doc="no clips")), ToolInput(tag="in_no_indels", input_type=Boolean(optional=True), prefix="--noindels", doc=InputDocumentation(doc="no indels")), ToolInput(tag="in_only_clips", input_type=Boolean(optional=True), prefix="--onlyclips", doc=InputDocumentation(doc="only clips")), ToolInput(tag="in_only_deletions", input_type=Boolean(optional=True), prefix="--onlydeletions", doc=InputDocumentation(doc="only deletions")), ToolInput(tag="in_only_insertions", input_type=Boolean(optional=True), prefix="--onlyinsertions", doc=InputDocumentation(doc="only insertions")), ToolInput(tag="in_no_deletions", input_type=Boolean(optional=True), prefix="--nodeletions", doc=InputDocumentation(doc="no deletions")), ToolInput(tag="in_no_insertions", input_type=Boolean(optional=True), prefix="--noinsertions", doc=InputDocumentation(doc="no insertions")), ToolInput(tag="in_not_reverse", input_type=Boolean(optional=True), prefix="--notreverse", doc=InputDocumentation(doc="no reverse complement alignments")), ToolInput(tag="in_print_ds", input_type=Boolean(optional=True), prefix="--printDS", doc=InputDocumentation(doc="print PMD scores")), ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="--verbose", doc=InputDocumentation(doc="verbose")), ToolInput(tag="in_print_alignments", input_type=Boolean(optional=True), prefix="--printalignments", doc=InputDocumentation(doc="print human readable alignments")), ToolInput(tag="in_mask_terminal_deamination_s", input_type=String(optional=True), prefix="--maskterminaldeaminations", doc=InputDocumentation(doc="mask terminal deaminations")), ToolInput(tag="in_mask_terminal_bases", input_type=String(optional=True), prefix="--maskterminalbases", doc=InputDocumentation(doc="mask terminal bases")), ToolInput(tag="in_threshold", input_type=String(optional=True), prefix="--threshold", doc=InputDocumentation(doc="only output sequences with PMD score above this\nthreshold")), ToolInput(tag="in_upper_threshold", input_type=String(optional=True), prefix="--upperthreshold", doc=InputDocumentation(doc="only output sequences with PMD score below this\nthreshold")), ToolInput(tag="in_perc_identity", input_type=String(optional=True), prefix="--perc_identity", doc=InputDocumentation(doc="only output sequences with percent identity above this\nthreshold")), ToolInput(tag="in_adjust_base_q", input_type=Boolean(optional=True), prefix="--adjustbaseq", doc=InputDocumentation(doc="apply PMD-aware adjustment of base quality scores\nspecific to C>T and G>A mismatches to the reference")), ToolInput(tag="in_adjust_base_q_all", input_type=Boolean(optional=True), prefix="--adjustbaseq_all", doc=InputDocumentation(doc="apply PMD-aware adjustment of base quality scores\nregardless of observed bases")), ToolInput(tag="in_dry", input_type=Boolean(optional=True), prefix="--dry", doc=InputDocumentation(doc="print SAM input without any filters")), ToolInput(tag="in_sam_tools_path", input_type=File(optional=True), prefix="--samtoolspath", doc=InputDocumentation(doc="full path to samtools")), ToolInput(tag="in_base_composition", input_type=Boolean(optional=True), prefix="--basecomposition", doc=InputDocumentation(doc="basecomposition")), ToolInput(tag="in_estimate", input_type=Boolean(optional=True), prefix="--estimate", doc=InputDocumentation(doc="two-terminus estimate of contamination")), ToolInput(tag="in_estimate_base", input_type=String(optional=True), prefix="--estimatebase", doc=InputDocumentation(doc="position of base used fortwo-terminus estimate of\ncontamination")), ToolInput(tag="in_platypus", input_type=Boolean(optional=True), prefix="--platypus", doc=InputDocumentation(doc="output big list of base frequencies for platypus")), ToolInput(tag="in_writes_am_field", input_type=Boolean(optional=True), prefix="--writesamfield", doc=InputDocumentation(doc="add 'DS:Z:<PMDS>' field to SAM output, will overwrite\nif already present")), ToolInput(tag="in_basic", input_type=Int(optional=True), prefix="--basic", doc=InputDocumentation(doc="only output reads with a C>T mismatch this many\nbasepairs from the 5' end")), ToolInput(tag="in_terminal", input_type=Boolean(optional=True), prefix="--terminal", doc=InputDocumentation(doc="only output reads with a C>T mismatch in either 3' or\n5' end")), ToolInput(tag="in_stats", input_type=Boolean(optional=True), prefix="--stats", doc=InputDocumentation(doc="output summarizing statistics to stderr")), ToolInput(tag="in_pm_dsp_rim_threshold", input_type=String(), position=0, doc=InputDocumentation(doc="--UDGhalf             only use Cs and Gs in CpG context, the first and last")), ToolInput(tag="in_custom_terminus", input_type=String(), position=0, doc=InputDocumentation(doc="--flagss              flag SNPs (taking strand into account)")), ToolInput(tag="in_refseq", input_type=String(), position=1, doc=InputDocumentation(doc="--header              output the SAM header"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Pmdtools_V0_1_0().translate("wdl", allow_empty_container=True)

