from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, File, String

Allelecounter_Pl_V0_1_0 = CommandToolBuilder(tool="alleleCounter.pl", base_command=["alleleCounter.pl"], inputs=[ToolInput(tag="in_bam", input_type=Boolean(optional=True), prefix="-bam", doc=InputDocumentation(doc="BAM/CRAM file (expects co-located index)\n- if CRAM see '-ref'")), ToolInput(tag="in_output", input_type=File(optional=True), prefix="-output", doc=InputDocumentation(doc="Output file [STDOUT]")), ToolInput(tag="in_loci", input_type=File(optional=True), prefix="-loci", doc=InputDocumentation(doc="Alternate loci file (just needs chr pos)\n- output is different, counts for each residue")), ToolInput(tag="in_ref", input_type=Boolean(optional=True), prefix="-ref", doc=InputDocumentation(doc="genome.fa, required for CRAM (with colocated .fai)")), ToolInput(tag="in_min_qual", input_type=Boolean(optional=True), prefix="-minqual", doc=InputDocumentation(doc="Minimum base quality to include (integer) [30]")), ToolInput(tag="in_map_qual", input_type=Boolean(optional=True), prefix="-mapqual", doc=InputDocumentation(doc="Minimum mapping quality of read (integer) [35]")), ToolInput(tag="in_gender", input_type=Boolean(optional=True), prefix="-gender", doc=InputDocumentation(doc="flag, presence indicates loci file to be treated as gender SNPs.\n- cannot be used with 's'")), ToolInput(tag="in_snp_six", input_type=File(optional=True), prefix="-snp6", doc=InputDocumentation(doc="flag, presence indicates loci file is SNP6 format.\n- cannot be used with 'g'\n- changes output format")), ToolInput(tag="in_version", input_type=Boolean(optional=True), prefix="-version", doc=InputDocumentation(doc="Version number")), ToolInput(tag="in_allele_counts_do_tpl", input_type=String(), position=0, doc=InputDocumentation(doc="Required:"))], outputs=[ToolOutput(tag="out_output", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output", type_hint=File()), doc=OutputDocumentation(doc="Output file [STDOUT]")), ToolOutput(tag="out_loci", output_type=File(optional=True), selector=InputSelector(input_to_select="in_loci", type_hint=File()), doc=OutputDocumentation(doc="Alternate loci file (just needs chr pos)\n- output is different, counts for each residue")), ToolOutput(tag="out_snp_six", output_type=File(optional=True), selector=InputSelector(input_to_select="in_snp_six", type_hint=File()), doc=OutputDocumentation(doc="flag, presence indicates loci file is SNP6 format.\n- cannot be used with 'g'\n- changes output format"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Allelecounter_Pl_V0_1_0().translate("wdl", allow_empty_container=True)

