from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, Int, String, File

Cutot_V0_1_0 = CommandToolBuilder(tool="cutot", base_command=["cutot"], inputs=[ToolInput(tag="in_no_menu", input_type=Boolean(optional=True), prefix="-nomenu", doc=InputDocumentation(doc="Prevent the menu interface being displayed")), ToolInput(tag="in_nowarn", input_type=Boolean(optional=True), prefix="-nowarn", doc=InputDocumentation(doc="Prevent warnings about sequences being displayed")), ToolInput(tag="in_silent", input_type=Boolean(optional=True), prefix="-silent", doc=InputDocumentation(doc="Overwrite files silently")), ToolInput(tag="in_totals", input_type=Boolean(optional=True), prefix="-totals", doc=InputDocumentation(doc="Concatenate all genes in inputfile")), ToolInput(tag="in_machine", input_type=Boolean(optional=True), prefix="-machine", doc=InputDocumentation(doc="Machine readable output")), ToolInput(tag="in_human", input_type=Boolean(optional=True), prefix="-human", doc=InputDocumentation(doc="Human readable output")), ToolInput(tag="in_code", input_type=Int(optional=True), prefix="-code", doc=InputDocumentation(doc="Genetic code as defined under menu 3 option 5")), ToolInput(tag="in_f_type", input_type=Int(optional=True), prefix="-f_type", doc=InputDocumentation(doc="Fop/CBI codons as defined by menu 3 option 6")), ToolInput(tag="in_c_type", input_type=Int(optional=True), prefix="-c_type", doc=InputDocumentation(doc="Cai fitness values as defined by menu 3 option 7")), ToolInput(tag="in_char_column_separator", input_type=Boolean(optional=True), prefix="-t", doc=InputDocumentation(doc="(char)      Column separator to be used in output files (comma,tab,space)")), ToolInput(tag="in_cai", input_type=Boolean(optional=True), prefix="-cai", doc=InputDocumentation(doc="calculate Codon Adaptation Index (CAI)")), ToolInput(tag="in_fop", input_type=Boolean(optional=True), prefix="-fop", doc=InputDocumentation(doc="calculate Frequency of OPtimal codons index (FOP)")), ToolInput(tag="in_cbi", input_type=Boolean(optional=True), prefix="-cbi", doc=InputDocumentation(doc="calculate Codon Bias Index (CBI)")), ToolInput(tag="in_enc", input_type=Boolean(optional=True), prefix="-enc", doc=InputDocumentation(doc="Effective Number of Codons (ENc)")), ToolInput(tag="in_gc", input_type=Boolean(optional=True), prefix="-gc", doc=InputDocumentation(doc="G+C content of gene (all 3 codon positions)")), ToolInput(tag="in_gcs_three", input_type=Boolean(optional=True), prefix="-gcs3", doc=InputDocumentation(doc="GC of synonymous codons 3rd positions")), ToolInput(tag="in_sil_base", input_type=Boolean(optional=True), prefix="-sil_base", doc=InputDocumentation(doc="Base composition at synonymous third codon positions")), ToolInput(tag="in_l_sym", input_type=Boolean(optional=True), prefix="-L_sym", doc=InputDocumentation(doc="Number of synonymous codons")), ToolInput(tag="in_l_aa", input_type=Boolean(optional=True), prefix="-L_aa", doc=InputDocumentation(doc="Total number of synonymous and non-synonymous codons")), ToolInput(tag="in_all_indices", input_type=Boolean(optional=True), prefix="-all_indices", doc=InputDocumentation(doc="All the above indices")), ToolInput(tag="in_aro", input_type=Boolean(optional=True), prefix="-aro", doc=InputDocumentation(doc="Calculate aromaticity of protein")), ToolInput(tag="in_hyd", input_type=Boolean(optional=True), prefix="-hyd", doc=InputDocumentation(doc="Calculate hydropathicity of protein")), ToolInput(tag="in_cai_file", input_type=String(optional=True), prefix="-cai_file", doc=InputDocumentation(doc="User input file of CAI values")), ToolInput(tag="in_cbi_file", input_type=String(optional=True), prefix="-cbi_file", doc=InputDocumentation(doc="User input file of CBI values")), ToolInput(tag="in_fop_file", input_type=String(optional=True), prefix="-fop_file", doc=InputDocumentation(doc="User input file of Fop values")), ToolInput(tag="in_coa_cu", input_type=Boolean(optional=True), prefix="-coa_cu", doc=InputDocumentation(doc="COA of codon usage frequencies")), ToolInput(tag="in_coa_rs_cu", input_type=Boolean(optional=True), prefix="-coa_rscu", doc=InputDocumentation(doc="COA of Relative Synonymous Codon Usage")), ToolInput(tag="in_coa_aa", input_type=Boolean(optional=True), prefix="-coa_aa", doc=InputDocumentation(doc="COA of amino acid usage frequencies")), ToolInput(tag="in_coa_expert", input_type=Boolean(optional=True), prefix="-coa_expert", doc=InputDocumentation(doc="Generate detailed(expert) statistics on COA")), ToolInput(tag="in_coa_axes", input_type=Int(optional=True), prefix="-coa_axes", doc=InputDocumentation(doc="Select number of axis to record")), ToolInput(tag="in_coa_num", input_type=Int(optional=True), prefix="-coa_num", doc=InputDocumentation(doc="Select number of genes to use to identify optimal codons\nvalues can be whole numbers or a percentage (5 or 10%)")), ToolInput(tag="in_aau", input_type=Boolean(optional=True), prefix="-aau", doc=InputDocumentation(doc="Amino Acid Usage (AAU)")), ToolInput(tag="in_ra_au", input_type=Boolean(optional=True), prefix="-raau", doc=InputDocumentation(doc="Relative Amino Acid Usage (RAAU)")), ToolInput(tag="in_cu", input_type=Boolean(optional=True), prefix="-cu", doc=InputDocumentation(doc="Codon Usage (CU) (default)")), ToolInput(tag="in_cut_ab", input_type=Boolean(optional=True), prefix="-cutab", doc=InputDocumentation(doc="Tabulation of codon usage")), ToolInput(tag="in_cut_ot", input_type=Boolean(optional=True), prefix="-cutot", doc=InputDocumentation(doc="Tabulation of dataset's codon usage")), ToolInput(tag="in_rs_cu", input_type=Boolean(optional=True), prefix="-rscu", doc=InputDocumentation(doc="Relative Synonymous Codon Usage (RSCU)")), ToolInput(tag="in_fast_a", input_type=Boolean(optional=True), prefix="-fasta", doc=InputDocumentation(doc="fasta format")), ToolInput(tag="in_tidy", input_type=Boolean(optional=True), prefix="-tidy", doc=InputDocumentation(doc="fasta format")), ToolInput(tag="in_reader", input_type=Boolean(optional=True), prefix="-reader", doc=InputDocumentation(doc="Reader format (codons are separated by spaces)")), ToolInput(tag="in_transl", input_type=Boolean(optional=True), prefix="-transl", doc=InputDocumentation(doc="Conceptual translation of DNA to amino acid")), ToolInput(tag="in_base", input_type=Boolean(optional=True), prefix="-base", doc=InputDocumentation(doc="Detailed report of codon G+C composition")), ToolInput(tag="in_din_uc", input_type=Boolean(optional=True), prefix="-dinuc", doc=InputDocumentation(doc="Dinucleotide usage of the three codon pos.")), ToolInput(tag="in_no_blk", input_type=File(optional=True), prefix="-noblk", doc=InputDocumentation(doc="No bulk output to be written to file"))], outputs=[ToolOutput(tag="out_no_blk", output_type=File(optional=True), selector=InputSelector(input_to_select="in_no_blk", type_hint=File()), doc=OutputDocumentation(doc="No bulk output to be written to file"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Cutot_V0_1_0().translate("wdl", allow_empty_container=True)

