version 1.0

task Anvigetsequencesforgenecalls {
  input {
    String? contigs_db
    String? gene_caller_ids
    Int? flank_length
    String? delimiter
    Boolean? report_extended_def_lines
    File? wrap
    File? export_gff_three
    Boolean? get_aa_sequences
    File? external_gene_calls
    File? genomes_storage
    File? genome_names
    File? output_file
  }
  command <<<
    anvi_get_sequences_for_gene_calls \
      ~{if defined(contigs_db) then ("--contigs-db " +  '"' + contigs_db + '"') else ""} \
      ~{if defined(gene_caller_ids) then ("--gene-caller-ids " +  '"' + gene_caller_ids + '"') else ""} \
      ~{if defined(flank_length) then ("--flank-length " +  '"' + flank_length + '"') else ""} \
      ~{if defined(delimiter) then ("--delimiter " +  '"' + delimiter + '"') else ""} \
      ~{if (report_extended_def_lines) then "--report-extended-deflines" else ""} \
      ~{if defined(wrap) then ("--wrap " +  '"' + wrap + '"') else ""} \
      ~{if (export_gff_three) then "--export-gff3" else ""} \
      ~{if (get_aa_sequences) then "--get-aa-sequences" else ""} \
      ~{if defined(external_gene_calls) then ("--external-gene-calls " +  '"' + external_gene_calls + '"') else ""} \
      ~{if defined(genomes_storage) then ("--genomes-storage " +  '"' + genomes_storage + '"') else ""} \
      ~{if defined(genome_names) then ("--genome-names " +  '"' + genome_names + '"') else ""} \
      ~{if defined(output_file) then ("--output-file " +  '"' + output_file + '"') else ""}
  >>>
  runtime {
    docker: "quay.io/biocontainers/anvio-minimal:7--py_0"
  }
  parameter_meta {
    contigs_db: "Anvi'o contigs database generated by 'anvi-gen-\\ncontigs-database' (default: None)"
    gene_caller_ids: "Gene caller ids. Multiple of them can be declared\\nseparated by a delimiter (the default is a comma). In\\nanvi-gen-variability-profile, if you declare nothing\\nyou will get all genes matching your other filtering\\ncriteria. In other programs, you may get everything,\\nnothing, or an error. It really depends on the\\nsituation. Fortunately, mistakes are cheap, so it's\\nworth a try. (default: None)"
    flank_length: "Extend sequences for gene calls with additional\\nnucleotides from both ends. If the seqeunce for a\\ntarget gene is between nucleotide positions START and\\nSTOP, using a flank lenght of M will give you a\\nsequence that starts at START - M and ends at STOP +\\nM. (default: 0)"
    delimiter: "The delimiter to parse multiple input terms. The\\ndefault is ','."
    report_extended_def_lines: "When declared, the deflines in the resulting FASTA\\nfile will contain more information. (default: False)"
    wrap: "When to wrap sequences when storing them in a FASTA\\nfile. The default is '120'. A value of '0' would be\\nequivalent to 'do not wrap'."
    export_gff_three: "If this is true, the output file will be in GFF3\\nformat. (default: False)"
    get_aa_sequences: "Store amino acid sequences instead. (default: False)"
    external_gene_calls: "An optional external gene calls file path that\\nprecisely describes the set of gene sequences\\nexported. Using this file you can create an anvi'o\\ncontigs database from the resulting genes FASTA file\\nwithout having to do a gene calling from scratch.\\n(default: None)"
    genomes_storage: "Anvi'o genomes storage file (default: None)"
    genome_names: "Genome names to 'focus'. You can use this parameter to\\nlimit the genomes included in your analysis. You can\\nprovide these names as a comma-separated list of\\nnames, or you can put them in a file, where you have a\\nsingle genome name in each line, and provide the file\\npath. (default: None)"
    output_file: "File path to store results. (default: None)"
  }
  output {
    File out_stdout = stdout()
    File out_export_gff_three = "${in_export_gff_three}"
    File out_output_file = "${in_output_file}"
  }
}