from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, String, File, Int

Mixcr_Exportalignments_V0_1_0 = CommandToolBuilder(tool="mixcr_exportAlignments", base_command=["mixcr", "exportAlignments"], inputs=[ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="--verbose", doc=InputDocumentation(doc="Verbose warning messages.")), ToolInput(tag="in_aa_feature", input_type=String(optional=True), prefix="-aaFeature", doc=InputDocumentation(doc="Export amino acid sequence of specified gene feature")), ToolInput(tag="in_aa_feature_imputed", input_type=String(optional=True), prefix="-aaFeatureImputed", doc=InputDocumentation(doc="Export amino acid sequence of specified gene feature using letters from\ngermline (marked lowercase) for uncovered regions")), ToolInput(tag="in_aa_mutations", input_type=String(optional=True), prefix="-aaMutations", doc=InputDocumentation(doc="Extract amino acid mutations for specific gene feature")), ToolInput(tag="in_aa_mutations_relative", input_type=String(optional=True), prefix="-aaMutationsRelative", doc=InputDocumentation(doc="PARAM\nExtract amino acid mutations for specific gene feature relative to another\nfeature.")), ToolInput(tag="in_a_vrg_feature_quality", input_type=String(optional=True), prefix="-avrgFeatureQuality", doc=InputDocumentation(doc="Export average quality of specified gene feature")), ToolInput(tag="in_chains", input_type=String(optional=True), prefix="--chains", doc=InputDocumentation(doc="Limit export to specific chain (e.g. TRA or IGH) (fractions will be\nrecalculated)")), ToolInput(tag="in_c_alignment", input_type=Boolean(optional=True), prefix="-cAlignment", doc=InputDocumentation(doc="Export best C alignment")), ToolInput(tag="in_c_alignments", input_type=Boolean(optional=True), prefix="-cAlignments", doc=InputDocumentation(doc="Export all C alignments")), ToolInput(tag="in_c_best_identity_percent", input_type=Boolean(optional=True), prefix="-cBestIdentityPercent", doc=InputDocumentation(doc="C best alignment identity percent")), ToolInput(tag="in_c_families", input_type=Boolean(optional=True), prefix="-cFamilies", doc=InputDocumentation(doc="Export all C gene family anmes (e.g. TRBV12 for TRBV12-3*00)")), ToolInput(tag="in_c_family", input_type=Boolean(optional=True), prefix="-cFamily", doc=InputDocumentation(doc="Export best C hit family name (e.g. TRBV12 for TRBV12-3*00)")), ToolInput(tag="in_c_gene", input_type=Boolean(optional=True), prefix="-cGene", doc=InputDocumentation(doc="Export best C hit gene name (e.g. TRBV12-3 for TRBV12-3*00)")), ToolInput(tag="in_c_genes", input_type=Boolean(optional=True), prefix="-cGenes", doc=InputDocumentation(doc="Export all C gene names (e.g. TRBV12-3 for TRBV12-3*00)")), ToolInput(tag="in_chit", input_type=Boolean(optional=True), prefix="-cHit", doc=InputDocumentation(doc="Export best C hit")), ToolInput(tag="in_chits", input_type=Boolean(optional=True), prefix="-cHits", doc=InputDocumentation(doc="Export all C hits")), ToolInput(tag="in_chit_score", input_type=Boolean(optional=True), prefix="-cHitScore", doc=InputDocumentation(doc="Export score for best C hit")), ToolInput(tag="in_chits_with_score", input_type=Boolean(optional=True), prefix="-cHitsWithScore", doc=InputDocumentation(doc="Export all C hits with score")), ToolInput(tag="in_c_identity_percents", input_type=Boolean(optional=True), prefix="-cIdentityPercents", doc=InputDocumentation(doc="C alignment identity percents")), ToolInput(tag="in_cloneid", input_type=Boolean(optional=True), prefix="-cloneId", doc=InputDocumentation(doc="To which clone alignment was attached (make sure using .clna file as input\nfor exportAlignments)")), ToolInput(tag="in_cloneid_with_mapping_type", input_type=Boolean(optional=True), prefix="-cloneIdWithMappingType", doc=InputDocumentation(doc="To which clone alignment was attached with additional info on mapping type\n(make sure using .clna file as input for exportAlignments)")), ToolInput(tag="in_d_alignment", input_type=Boolean(optional=True), prefix="-dAlignment", doc=InputDocumentation(doc="Export best D alignment")), ToolInput(tag="in_d_alignments", input_type=Boolean(optional=True), prefix="-dAlignments", doc=InputDocumentation(doc="Export all D alignments")), ToolInput(tag="in_d_best_identity_percent", input_type=Boolean(optional=True), prefix="-dBestIdentityPercent", doc=InputDocumentation(doc="D best alignment identity percent")), ToolInput(tag="in_default_anchor_points", input_type=Boolean(optional=True), prefix="-defaultAnchorPoints", doc=InputDocumentation(doc="Outputs a list of default reference points (like CDR2Begin, FR4End, etc.\nsee documentation for the full list and formatting)")), ToolInput(tag="in_descr_r_one", input_type=Boolean(optional=True), prefix="-descrR1", doc=InputDocumentation(doc="Export description line from initial .fasta or .fastq file (deprecated)")), ToolInput(tag="in_descr_r_two", input_type=Boolean(optional=True), prefix="-descrR2", doc=InputDocumentation(doc="Export description line from initial .fasta or .fastq file (deprecated)")), ToolInput(tag="in_descr_sr_one", input_type=Boolean(optional=True), prefix="-descrsR1", doc=InputDocumentation(doc="Export description lines from initial .fasta or .fastq file for R1 reads\n(only available if -OsaveOriginalReads=true was used in align command)")), ToolInput(tag="in_descr_sr_two", input_type=Boolean(optional=True), prefix="-descrsR2", doc=InputDocumentation(doc="Export description lines from initial .fastq file for R2 reads (only\navailable if -OsaveOriginalReads=true was used in align command)")), ToolInput(tag="in_d_families", input_type=Boolean(optional=True), prefix="-dFamilies", doc=InputDocumentation(doc="Export all D gene family anmes (e.g. TRBV12 for TRBV12-3*00)")), ToolInput(tag="in_d_family", input_type=Boolean(optional=True), prefix="-dFamily", doc=InputDocumentation(doc="Export best D hit family name (e.g. TRBV12 for TRBV12-3*00)")), ToolInput(tag="in_d_gene", input_type=Boolean(optional=True), prefix="-dGene", doc=InputDocumentation(doc="Export best D hit gene name (e.g. TRBV12-3 for TRBV12-3*00)")), ToolInput(tag="in_d_genes", input_type=Boolean(optional=True), prefix="-dGenes", doc=InputDocumentation(doc="Export all D gene names (e.g. TRBV12-3 for TRBV12-3*00)")), ToolInput(tag="in_d_hit", input_type=Boolean(optional=True), prefix="-dHit", doc=InputDocumentation(doc="Export best D hit")), ToolInput(tag="in_d_hits", input_type=Boolean(optional=True), prefix="-dHits", doc=InputDocumentation(doc="Export all D hits")), ToolInput(tag="in_dh_its_core", input_type=Boolean(optional=True), prefix="-dHitScore", doc=InputDocumentation(doc="Export score for best D hit")), ToolInput(tag="in_d_hits_with_score", input_type=Boolean(optional=True), prefix="-dHitsWithScore", doc=InputDocumentation(doc="Export all D hits with score")), ToolInput(tag="in_d_identity_percents", input_type=Boolean(optional=True), prefix="-dIdentityPercents", doc=InputDocumentation(doc="D alignment identity percents")), ToolInput(tag="in_force_overwrite", input_type=File(optional=True), prefix="--force-overwrite", doc=InputDocumentation(doc="Force overwrite of output file(s).")), ToolInput(tag="in_j_alignment", input_type=Boolean(optional=True), prefix="-jAlignment", doc=InputDocumentation(doc="Export best J alignment")), ToolInput(tag="in_j_alignments", input_type=Boolean(optional=True), prefix="-jAlignments", doc=InputDocumentation(doc="Export all J alignments")), ToolInput(tag="in_j_best_identity_percent", input_type=Boolean(optional=True), prefix="-jBestIdentityPercent", doc=InputDocumentation(doc="J best alignment identity percent")), ToolInput(tag="in_j_families", input_type=Boolean(optional=True), prefix="-jFamilies", doc=InputDocumentation(doc="Export all J gene family anmes (e.g. TRBV12 for TRBV12-3*00)")), ToolInput(tag="in_j_family", input_type=Boolean(optional=True), prefix="-jFamily", doc=InputDocumentation(doc="Export best J hit family name (e.g. TRBV12 for TRBV12-3*00)")), ToolInput(tag="in_j_gene", input_type=Boolean(optional=True), prefix="-jGene", doc=InputDocumentation(doc="Export best J hit gene name (e.g. TRBV12-3 for TRBV12-3*00)")), ToolInput(tag="in_j_genes", input_type=Boolean(optional=True), prefix="-jGenes", doc=InputDocumentation(doc="Export all J gene names (e.g. TRBV12-3 for TRBV12-3*00)")), ToolInput(tag="in_jh_it", input_type=Boolean(optional=True), prefix="-jHit", doc=InputDocumentation(doc="Export best J hit")), ToolInput(tag="in_j_hits", input_type=Boolean(optional=True), prefix="-jHits", doc=InputDocumentation(doc="Export all J hits")), ToolInput(tag="in_jh_its_core", input_type=Boolean(optional=True), prefix="-jHitScore", doc=InputDocumentation(doc="Export score for best J hit")), ToolInput(tag="in_j_hits_with_score", input_type=Boolean(optional=True), prefix="-jHitsWithScore", doc=InputDocumentation(doc="Export all J hits with score")), ToolInput(tag="in_j_identity_percents", input_type=Boolean(optional=True), prefix="-jIdentityPercents", doc=InputDocumentation(doc="J alignment identity percents")), ToolInput(tag="in_length_of", input_type=Int(optional=True), prefix="-lengthOf", doc=InputDocumentation(doc="Export length of specified gene feature.")), ToolInput(tag="in_min_feature_quality", input_type=String(optional=True), prefix="-minFeatureQuality", doc=InputDocumentation(doc="Export minimal quality of specified gene feature")), ToolInput(tag="in_mutations_detailed", input_type=String(optional=True), prefix="-mutationsDetailed", doc=InputDocumentation(doc="Detailed list of nucleotide and corresponding amino acid mutations. Format\n<nt_mutation>:<aa_mutation_individual>:<aa_mutation_cumulative>, where\n<aa_mutation_individual> is an expected amino acid mutation given no\nother mutations have occurred, and <aa_mutation_cumulative> amino acid\nmutation is the observed amino acid mutation combining effect from all\nother. WARNING: format may change in following versions.")), ToolInput(tag="in_mutations_detailed_relative", input_type=String(optional=True), prefix="-mutationsDetailedRelative", doc=InputDocumentation(doc="PARAM\nDetailed list of nucleotide and corresponding amino acid mutations written,\npositions relative to specified gene feature. Format <nt_mutation>:\n<aa_mutation_individual>:<aa_mutation_cumulative>, where\n<aa_mutation_individual> is an expected amino acid mutation given no\nother mutations have occurred, and <aa_mutation_cumulative> amino acid\nmutation is the observed amino acid mutation combining effect from all\nother. WARNING: format may change in following versions.")), ToolInput(tag="in_limit", input_type=String(optional=True), prefix="--limit", doc=InputDocumentation(doc="Output only first N records")), ToolInput(tag="in_n_feature", input_type=String(optional=True), prefix="-nFeature", doc=InputDocumentation(doc="Export nucleotide sequence of specified gene feature")), ToolInput(tag="in_n_feature_imputed", input_type=String(optional=True), prefix="-nFeatureImputed", doc=InputDocumentation(doc="Export nucleotide sequence of specified gene feature using letters from\ngermline (marked lowercase) for uncovered regions")), ToolInput(tag="in_n_mutations", input_type=String(optional=True), prefix="-nMutations", doc=InputDocumentation(doc="Extract nucleotide mutations for specific gene feature; relative to\ngermline sequence.")), ToolInput(tag="in_n_mutations_relative", input_type=String(optional=True), prefix="-nMutationsRelative", doc=InputDocumentation(doc="PARAM\nExtract nucleotide mutations for specific gene feature relative to another\nfeature.")), ToolInput(tag="in_no_warnings", input_type=Boolean(optional=True), prefix="--no-warnings", doc=InputDocumentation(doc="Suppress all warning messages.")), ToolInput(tag="in_preset", input_type=Int(optional=True), prefix="--preset", doc=InputDocumentation(doc="Specify preset of export fields (possible values: 'full', 'min'; 'full' by\ndefault)")), ToolInput(tag="in_preset_file", input_type=File(optional=True), prefix="--preset-file", doc=InputDocumentation(doc="Specify preset file of export fields")), ToolInput(tag="in_position_in_reference_of", input_type=String(optional=True), prefix="-positionInReferenceOf", doc=InputDocumentation(doc="Export position of specified reference point inside referencesequences\n(clonal sequence / read sequence).")), ToolInput(tag="in_position_of", input_type=String(optional=True), prefix="-positionOf", doc=InputDocumentation(doc="Export position of specified reference point inside targetsequences (clonal\nsequence / read sequence).")), ToolInput(tag="in_q_feature", input_type=String(optional=True), prefix="-qFeature", doc=InputDocumentation(doc="Export quality string of specified gene feature")), ToolInput(tag="in_read_history", input_type=Boolean(optional=True), prefix="-readHistory", doc=InputDocumentation(doc="Export read history")), ToolInput(tag="in_read_id", input_type=Boolean(optional=True), prefix="-readId", doc=InputDocumentation(doc="Export id of read corresponding to alignment (deprecated)")), ToolInput(tag="in_read_ids", input_type=Boolean(optional=True), prefix="-readIds", doc=InputDocumentation(doc="Export id(s) of read(s) corresponding to alignment")), ToolInput(tag="in_target_qualities", input_type=Boolean(optional=True), prefix="-targetQualities", doc=InputDocumentation(doc="Export aligned sequence (target) qualities, separated with comma")), ToolInput(tag="in_targets", input_type=Boolean(optional=True), prefix="-targets", doc=InputDocumentation(doc="Export number of targets")), ToolInput(tag="in_target_sequences", input_type=Boolean(optional=True), prefix="-targetSequences", doc=InputDocumentation(doc="Export aligned sequences (targets), separated with comma")), ToolInput(tag="in_top_chains", input_type=Boolean(optional=True), prefix="-topChains", doc=InputDocumentation(doc="Top chains")), ToolInput(tag="in_with_spaces", input_type=Boolean(optional=True), prefix="--with-spaces", doc=InputDocumentation(doc="Output column headers with spaces.")), ToolInput(tag="in_v_alignment", input_type=Boolean(optional=True), prefix="-vAlignment", doc=InputDocumentation(doc="Export best V alignment")), ToolInput(tag="in_v_alignments", input_type=Boolean(optional=True), prefix="-vAlignments", doc=InputDocumentation(doc="Export all V alignments")), ToolInput(tag="in_v_best_identity_percent", input_type=Boolean(optional=True), prefix="-vBestIdentityPercent", doc=InputDocumentation(doc="V best alignment identity percent")), ToolInput(tag="in_v_families", input_type=Boolean(optional=True), prefix="-vFamilies", doc=InputDocumentation(doc="Export all V gene family anmes (e.g. TRBV12 for TRBV12-3*00)")), ToolInput(tag="in_v_family", input_type=Boolean(optional=True), prefix="-vFamily", doc=InputDocumentation(doc="Export best V hit family name (e.g. TRBV12 for TRBV12-3*00)")), ToolInput(tag="in_v_gene", input_type=Boolean(optional=True), prefix="-vGene", doc=InputDocumentation(doc="Export best V hit gene name (e.g. TRBV12-3 for TRBV12-3*00)")), ToolInput(tag="in_v_genes", input_type=Boolean(optional=True), prefix="-vGenes", doc=InputDocumentation(doc="Export all V gene names (e.g. TRBV12-3 for TRBV12-3*00)")), ToolInput(tag="in_v_hit", input_type=Boolean(optional=True), prefix="-vHit", doc=InputDocumentation(doc="Export best V hit")), ToolInput(tag="in_v_hits", input_type=Boolean(optional=True), prefix="-vHits", doc=InputDocumentation(doc="Export all V hits")), ToolInput(tag="in_vh_its_core", input_type=Boolean(optional=True), prefix="-vHitScore", doc=InputDocumentation(doc="Export score for best V hit")), ToolInput(tag="in_v_hits_with_score", input_type=Boolean(optional=True), prefix="-vHitsWithScore", doc=InputDocumentation(doc="Export all V hits with score")), ToolInput(tag="in_v_identity_percents", input_type=Boolean(optional=True), prefix="-vIdentityPercents", doc=InputDocumentation(doc="V alignment identity percents")), ToolInput(tag="in_fv", input_type=Boolean(optional=True), prefix="-fv", doc=InputDocumentation(doc="")), ToolInput(tag="in_scr", input_type=String(), position=0, doc=InputDocumentation(doc="")), ToolInput(tag="in_export_alignments", input_type=String(), position=1, doc=InputDocumentation(doc=""))], outputs=[ToolOutput(tag="out_force_overwrite", output_type=File(optional=True), selector=InputSelector(input_to_select="in_force_overwrite", type_hint=File()), doc=OutputDocumentation(doc="Force overwrite of output file(s)."))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Mixcr_Exportalignments_V0_1_0().translate("wdl", allow_empty_container=True)

