version 1.0

task RnaQUAST.py {
  input {
    Array[String] reference
    Array[String] gtf
    String? gene_db
    Array[String] transcripts
    Array[String] alignment
    String? reads_alignment
    String? left_reads
    String? right_reads
    String? single_reads
    String? gmap_index
    String? output_dir
    Boolean? test
    Boolean? debug
    String? threads
    Array[String] labels
    Boolean? strand_specific
    Int? min_alignment
    Boolean? no_plots
    Boolean? blat
    Boolean? gene_mark
    Boolean? run_quality_assessment
    String? lower_threshold
    String? upper_threshold
    Boolean? prokaryote
    Boolean? disable_infer_genes
    Boolean? disable_infer_transcripts
    String? busco_lineage
    String python
  }
  command <<<
    rnaQUAST.py \
      ~{python} \
      ~{if defined(reference) then ("--reference " +  '"' + reference + '"') else ""} \
      ~{if defined(gtf) then ("--gtf " +  '"' + gtf + '"') else ""} \
      ~{if defined(gene_db) then ("--gene_db " +  '"' + gene_db + '"') else ""} \
      ~{if defined(transcripts) then ("--transcripts " +  '"' + transcripts + '"') else ""} \
      ~{if defined(alignment) then ("--alignment " +  '"' + alignment + '"') else ""} \
      ~{if defined(reads_alignment) then ("--reads_alignment " +  '"' + reads_alignment + '"') else ""} \
      ~{if defined(left_reads) then ("--left_reads " +  '"' + left_reads + '"') else ""} \
      ~{if defined(right_reads) then ("--right_reads " +  '"' + right_reads + '"') else ""} \
      ~{if defined(single_reads) then ("--single_reads " +  '"' + single_reads + '"') else ""} \
      ~{if defined(gmap_index) then ("--gmap_index " +  '"' + gmap_index + '"') else ""} \
      ~{if defined(output_dir) then ("--output_dir " +  '"' + output_dir + '"') else ""} \
      ~{true="--test" false="" test} \
      ~{true="--debug" false="" debug} \
      ~{if defined(threads) then ("--threads " +  '"' + threads + '"') else ""} \
      ~{if defined(labels) then ("--labels " +  '"' + labels + '"') else ""} \
      ~{true="--strand_specific" false="" strand_specific} \
      ~{if defined(min_alignment) then ("--min_alignment " +  '"' + min_alignment + '"') else ""} \
      ~{true="--no_plots" false="" no_plots} \
      ~{true="--blat" false="" blat} \
      ~{true="--gene_mark" false="" gene_mark} \
      ~{true="--meta" false="" run_quality_assessment} \
      ~{if defined(lower_threshold) then ("--lower_threshold " +  '"' + lower_threshold + '"') else ""} \
      ~{if defined(upper_threshold) then ("--upper_threshold " +  '"' + upper_threshold + '"') else ""} \
      ~{true="--prokaryote" false="" prokaryote} \
      ~{true="--disable_infer_genes" false="" disable_infer_genes} \
      ~{true="--disable_infer_transcripts" false="" disable_infer_transcripts} \
      ~{if defined(busco_lineage) then ("--busco_lineage " +  '"' + busco_lineage + '"') else ""}
  >>>
  parameter_meta {
    reference: "Single file (or several files for meta RNA) with reference genome in FASTA format or *.txt file with one-per-line list of FASTA files with reference sequences"
    gtf: "File with gene coordinates (or several files or *.txt file with one-per-line list of GTF / GFF files for meta RNA). We recommend to use files downloaded from GENCODE or Ensembl [GTF/GFF]"
    gene_db: "Path to the gene database generated by gffutils to be used"
    transcripts: "File(s) with transcripts [FASTA]"
    alignment: "File(s) with transcript alignments to the reference genome [PSL]"
    reads_alignment: "File with read alignments to the reference genome [SAM]"
    left_reads: "File with forward paired-end reads [FASTQ or gzip- compressed]"
    right_reads: "File with reverse paired-end reads [FASTQ or gzip- compressed]"
    single_reads: "File with unpaired reads [FASTQ or gzip-compressed]"
    gmap_index: "Folder containing GMAP index for the reference genome"
    output_dir: "Directory to store all results [default: rnaQUAST_results/results_<datetime>]"
    test: "Run rnaQUAST on the test data from the test_data folder, output directory is rnaOUAST_test_output"
    debug: "Report detailed information, typically used only for detecting problems."
    threads: "Maximum number of threads, default: min(number of CPUs / 2, 16)"
    labels: "Name(s) of assemblies that will be used in the reports"
    strand_specific: "Set if transcripts were assembled using strand- specific RNA-Seq data"
    min_alignment: "Minimal alignment length, default: 50"
    no_plots: "Do not draw plots (to speed up computation)"
    blat: "Run with BLAT alignment tool (http://hgwdev.cse.ucsc.edu/~kent/exe/) instead of GMAP"
    gene_mark: "Run with GeneMarkS-T tool (http://topaz.gatech.edu/GeneMark/)"
    run_quality_assessment: "Run QUALITY ASSESSMENT FOR METATRANSCRIPTOME ASSEMBLIES"
    lower_threshold: "Lower threshold for x-assembled/covered/matched metrics, default: 0.5"
    upper_threshold: "Upper threshold for x-assembled/covered/matched metrics, default: 0.95"
    prokaryote: "Use this option if the genome is prokaryotic"
    disable_infer_genes: "Use this option if your GTF file already contains genes records"
    disable_infer_transcripts: "Use this option if your GTF already contains transcripts records"
    busco_lineage: "Run with BUSCO tool (http://busco.ezlab.org/). Path to the BUSCO lineage data to be used (Eukaryota, Metazoa, Arthropoda, Vertebrata or Fungi)"
    python: ""
  }
}