from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean

Tssar_V0_1_0 = CommandToolBuilder(tool="TSSAR", base_command=["TSSAR"], inputs=[ToolInput(tag="in_libp", input_type=Boolean(optional=True), prefix="--libP", doc=InputDocumentation(doc="*libraryP.sam* AND --libM *libraryM.sam*\nInput library (P .. Plus; M .. Minus) in SAM format. The plus\nlibrary is the one with enriched TSS (for dRNA-seq this means that\nthe plus library is the treated library, while the minus library is\nthe untreated library)")), ToolInput(tag="in_fast_a", input_type=Boolean(optional=True), prefix="--fasta", doc=InputDocumentation(doc="*genome.fa* OR --g_size *INT*\nEither the location of reference genome sequence in fasta file\nformat OR the genome size in *INT*. The fasta file is only used to\nparse the genome size so just one of the two must be specified.")), ToolInput(tag="in_min_peak", input_type=Boolean(optional=True), prefix="--minPeak", doc=InputDocumentation(doc="*INT*\nMinimal Peak size in *INT*. Only positions where read start count in\nthe (+)library is greater or equal then *INT* are evaluated to be a\nTSS. Positions with less reads are seen as backgroound noise and not\nconsidered. Default is *3*.")), ToolInput(tag="in_pval", input_type=Boolean(optional=True), prefix="--pval", doc=InputDocumentation(doc="*FLOAT*\nMaximal P-value for each position to be annotated as a TSS. Default\nis *1e-04*.")), ToolInput(tag="in_winsize", input_type=Boolean(optional=True), prefix="--winSize", doc=InputDocumentation(doc="*INT*\nSize of the window which slides over the genome and defines the\nstatistical properties of the local model. Default is *1,000*.")), ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="--verbose", doc=InputDocumentation(doc="If set, some progress reports are printed to STDERR during\ncomputation.")), ToolInput(tag="in_pro_rata", input_type=Boolean(optional=True), prefix="--prorata", doc=InputDocumentation(doc="If set, the information from the SAM file how many times a read was\nmapped to the genome is used, if present. If the read maps *n* times\nto the genome, each position is counted only *1/n* times. Usefull in\ncombination with e.g. segemehl mapper, which can report suboptimal\nmapping positions and/or reports all location where a read maps\noptimally. Default is off.")), ToolInput(tag="in_score", input_type=Boolean(optional=True), prefix="--score", doc=InputDocumentation(doc="*p|d*\nIf score mode is *p* the p-value is used as score in the TSS BED\nfile. If score mode is *d* the peak difference is used as score in\nthe TSS BED file. Default is *d*. Also used for clustering, which\nadvices to use 'd', since the p-value often becomes zero for\nconsecutive positions, thus disabling a proper merging of\nconsecutive positions to the best one.")), ToolInput(tag="in_no_cluster", input_type=Boolean(optional=True), prefix="--nocluster", doc=InputDocumentation(doc="If --nocluster is set all positions annotated as TSS are reported.\nIf --cluster is set consecutive TSS positions are clustered and only\nthe 'best' position is reported. 'Best' position depends on the\nsetting of --score (see above). Either the position with the lowest\np-Value or the position with the highest peak difference between\nplus and minus library is reported. Default is --cluster. The option\n--range defines the maximal distance for two significant positions\nto be called 'consecutive'.")), ToolInput(tag="in_range", input_type=Boolean(optional=True), prefix="--range", doc=InputDocumentation(doc="*INT*\nThe maximal distance for two significant positions to be be\nclustered together if option --cluster is set. Default is *3* nt. If\n--cluster is set to --nocluster, --range is ignored.")), ToolInput(tag="in_no_clean", input_type=Boolean(optional=True), prefix="--noclean", doc=InputDocumentation(doc="If --clean is set, all temporary files which are created during the\ncomputation are deleted afterwards. With --noclean they are stored.\nMainly for debugging purpose. Default setting is --clean.")), ToolInput(tag="in_tmpdir", input_type=Boolean(optional=True), prefix="--tmpdir", doc=InputDocumentation(doc="*DIR*\nSpecifies where the temporary files should be stored. Default is\n*/tmp*.")), ToolInput(tag="in_man", input_type=Boolean(optional=True), prefix="--man", doc=InputDocumentation(doc="Print a long version of the man-page."))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Tssar_V0_1_0().translate("wdl", allow_empty_container=True)

