from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, String, File, Int, Directory

Xtea_V0_1_0 = CommandToolBuilder(tool="xtea", base_command=["xtea"], inputs=[ToolInput(tag="in_decompress", input_type=Boolean(optional=True), prefix="--decompress", doc=InputDocumentation(doc="Decompress the rep lib and reference file")), ToolInput(tag="in_mosaic", input_type=Boolean(optional=True), prefix="--mosaic", doc=InputDocumentation(doc="Calling mosaic events from high coverage data")), ToolInput(tag="in_case_control", input_type=Boolean(optional=True), prefix="--case_control", doc=InputDocumentation(doc="Run in case control mode")), ToolInput(tag="in_user", input_type=Boolean(optional=True), prefix="--user", doc=InputDocumentation(doc="Use user specific parameters instead of automatically\ncalculated ones")), ToolInput(tag="in_force", input_type=Boolean(optional=True), prefix="--force", doc=InputDocumentation(doc="Force to start from the very beginning")), ToolInput(tag="in_tumor", input_type=Boolean(optional=True), prefix="--tumor", doc=InputDocumentation(doc="Working on tumor samples")), ToolInput(tag="in_purity", input_type=String(optional=True), prefix="--purity", doc=InputDocumentation(doc="Tumor purity")), ToolInput(tag="in_lsf", input_type=Boolean(optional=True), prefix="--lsf", doc=InputDocumentation(doc="Indiates submit to LSF system")), ToolInput(tag="in_slur_m", input_type=Boolean(optional=True), prefix="--slurm", doc=InputDocumentation(doc="Indiates submit to slurm system")), ToolInput(tag="in_resume", input_type=File(optional=True), prefix="--resume", doc=InputDocumentation(doc="Resume the running, which will skip the step if output\nfile already exists!")), ToolInput(tag="in_id", input_type=File(optional=True), prefix="--id", doc=InputDocumentation(doc="sample id list file")), ToolInput(tag="in_par", input_type=File(optional=True), prefix="--par", doc=InputDocumentation(doc="parameter file")), ToolInput(tag="in_lib", input_type=File(optional=True), prefix="--lib", doc=InputDocumentation(doc="TE lib config file")), ToolInput(tag="in_bam", input_type=File(optional=True), prefix="--bam", doc=InputDocumentation(doc="Input bam file")), ToolInput(tag="in_x_one_zero", input_type=File(optional=True), prefix="--x10", doc=InputDocumentation(doc="Input 10X bam file")), ToolInput(tag="in_cores", input_type=Int(optional=True), prefix="--cores", doc=InputDocumentation(doc="number of cores")), ToolInput(tag="in_memory", input_type=String(optional=True), prefix="--memory", doc=InputDocumentation(doc="Memory limit in GB")), ToolInput(tag="in_partition", input_type=String(optional=True), prefix="--partition", doc=InputDocumentation(doc="Which queue to run the job")), ToolInput(tag="in_time", input_type=String(optional=True), prefix="--time", doc=InputDocumentation(doc="Time limit")), ToolInput(tag="in_path", input_type=Directory(optional=True), prefix="--path", doc=InputDocumentation(doc="Working folder")), ToolInput(tag="in_ref", input_type=String(optional=True), prefix="--ref", doc=InputDocumentation(doc="reference genome")), ToolInput(tag="in_gene", input_type=File(optional=True), prefix="--gene", doc=InputDocumentation(doc="Gene annotation file")), ToolInput(tag="in_x_tea", input_type=Directory(optional=True), prefix="--xtea", doc=InputDocumentation(doc="xTEA folder")), ToolInput(tag="in_flag", input_type=Int(optional=True), prefix="--flag", doc=InputDocumentation(doc="Flag indicates which step to run (1-clip, 2-disc,\n4-barcode, 8-xfilter, 16-filter, 32-asm)")), ToolInput(tag="in_rep_type", input_type=Int(optional=True), prefix="--reptype", doc=InputDocumentation(doc="Type of repeats working on: 1-L1, 2-Alu, 4-SVA,\n8-HERV, 16-Mitochondrial")), ToolInput(tag="in_fl_klen", input_type=File(optional=True), prefix="--flklen", doc=InputDocumentation(doc="flank region file")), ToolInput(tag="in_n_clip", input_type=String(optional=True), prefix="--nclip", doc=InputDocumentation(doc="cutoff of minimum # of clipped reads")), ToolInput(tag="in_cr", input_type=String(optional=True), prefix="--cr", doc=InputDocumentation(doc="cutoff of minimum # of clipped reads whose mates map\nin repetitive regions")), ToolInput(tag="in_nd", input_type=String(optional=True), prefix="--nd", doc=InputDocumentation(doc="cutoff of minimum # of discordant pair")), ToolInput(tag="in_nf_clip", input_type=String(optional=True), prefix="--nfclip", doc=InputDocumentation(doc="cutoff of minimum # of clipped reads in filtering step")), ToolInput(tag="in_nf_disc", input_type=String(optional=True), prefix="--nfdisc", doc=InputDocumentation(doc="cutoff of minimum # of discordant pair of each sample\nin filtering step")), ToolInput(tag="in_teilen", input_type=Int(optional=True), prefix="--teilen", doc=InputDocumentation(doc="minimum length of the insertion for future analysis")), ToolInput(tag="in_output", input_type=File(optional=True), prefix="--output", doc=InputDocumentation(doc="The output file")), ToolInput(tag="in_blacklist", input_type=File(optional=True), prefix="--blacklist", doc=InputDocumentation(doc="Reference panel database for filtering, or a blacklist\nregion\n"))], outputs=[ToolOutput(tag="out_resume", output_type=File(optional=True), selector=InputSelector(input_to_select="in_resume", type_hint=File()), doc=OutputDocumentation(doc="Resume the running, which will skip the step if output\nfile already exists!")), ToolOutput(tag="out_output", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output", type_hint=File()), doc=OutputDocumentation(doc="The output file"))], container="quay.io/biocontainers/xtea:0.1.6--0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Xtea_V0_1_0().translate("wdl")

