class: CommandLineTool
id: LiBis.cwl
inputs:
- id: file
  doc: Enter a number, 0 means using parameter to set up, 1 means using text file
    to set up
  type: File
  inputBinding:
    prefix: --file
- id: setting_file
  doc: Setting txt file name. Ignore if -f is 0.
  type: string
  inputBinding:
    prefix: --settingfile
- id: n
  doc: '[NAME [NAME ...]], --name [NAME [NAME ...]] Required. Fastq file name.'
  type: boolean
  inputBinding:
    prefix: -n
- id: clip
  doc: 'Clip mode. 0 means close. 1 means open. default: 0'
  type: string
  inputBinding:
    prefix: --clip
- id: l
  doc: '[LABEL [LABEL ...]], --label [LABEL [LABEL ...]] Labels for samples'
  type: boolean
  inputBinding:
    prefix: -l
- id: genome
  doc: Genome the reference belong to.(Use for plotting) hg18/hg19/mm10/mm9 and so
    on. Plotting script will not avaliable if leave it blank. default=hg38.
  type: string
  inputBinding:
    prefix: --genome
- id: window
  doc: Window length for clipping mode, default=30
  type: string
  inputBinding:
    prefix: --window
- id: step
  doc: Step size for clipping mode.
  type: string
  inputBinding:
    prefix: --step
- id: process
  doc: Process using for one pipeline. Normally bsmap will cost 8 cpu number. So total
    will be 8p.
  type: string
  inputBinding:
    prefix: --process
- id: ref
  doc: Required. Reference genome file name.
  type: string
  inputBinding:
    prefix: --ref
- id: quality_control
  doc: Do quality control
  type: boolean
  inputBinding:
    prefix: --QualityControl
- id: trim
  doc: Do trimming.
  type: boolean
  inputBinding:
    prefix: --trim
- id: binsize
  doc: Plot setting. Set the bin size for averaging methylation ratio among samples,
    default=1000k
  type: string
  inputBinding:
    prefix: --binsize
- id: filter
  doc: Minimal length for recombined reads
  type: string
  inputBinding:
    prefix: --filter
- id: bam
  doc: "Processed bam file for the first step. If bam files are offered here, the\
    \ first step of bsmap will be skipped. BAM files can only be generated by BSMAP.\
    \ Different files should be seperated by ','. If there's no bam file for part\
    \ of the samples, leave the space blank. For example: a.bam,b.bam,,,,f.bam"
  type: string
  inputBinding:
    prefix: --bam
- id: m_call
  doc: Run mcall for mapped bams or not
  type: boolean
  inputBinding:
    prefix: --mcall
- id: plot
  doc: Generate the final report or not
  type: boolean
  inputBinding:
    prefix: --plot
- id: no_check
  doc: Skip the checking step for result folders. Using this parameter may rewrite
    the previous results.
  type: boolean
  inputBinding:
    prefix: --nocheck
- id: moab_s_module
  doc: Run LiBis as MOABS module. Please only apply this when integrating LiBis to
    MOABS.
  type: boolean
  inputBinding:
    prefix: --MOABSmodule
- id: multiple_mapping
  doc: Apply multiple mapping after first around bsmap
  type: boolean
  inputBinding:
    prefix: --multiplemapping
- id: pair_end_filter
  doc: Apply pairend filter for all LiBis rescued reads. Only keep reads have uniquely
    mapped pair by BSMAP or LiBis.
  type: boolean
  inputBinding:
    prefix: --pairendfilter
- id: bam_u
  doc: Given processed bam file contains unmapped reads. Only use with -bam is open.
  type: boolean
  inputBinding:
    prefix: --bamu
- id: gzip
  doc: Temporary files are in gz format.
  type: boolean
  inputBinding:
    prefix: --gzip
- id: full_mode
  doc: Keep all temp files.
  type: boolean
  inputBinding:
    prefix: --fullmode
outputs: []
cwlVersion: v1.1
baseCommand:
- LiBis
