from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, File, String

Real_Tbl2Asn_V0_1_0 = CommandToolBuilder(tool="real_tbl2asn", base_command=["real-tbl2asn"], inputs=[ToolInput(tag="in_path_files_optional", input_type=Boolean(optional=True), prefix="-p", doc=InputDocumentation(doc="Path to Files [String]  Optional")), ToolInput(tag="in_path_results_optional", input_type=Boolean(optional=True), prefix="-r", doc=InputDocumentation(doc="Path for Results [String]  Optional")), ToolInput(tag="in_single_input_file", input_type=Boolean(optional=True), prefix="-i", doc=InputDocumentation(doc="Single Input File [File In]  Optional")), ToolInput(tag="in_single_output_file", input_type=File(optional=True), prefix="-o", doc=InputDocumentation(doc="Single Output File [File Out]  Optional")), ToolInput(tag="in_suffix__optionaldefault", input_type=Boolean(optional=True), prefix="-x", doc=InputDocumentation(doc="Suffix [String]  Optional\ndefault = .fsa")), ToolInput(tag="in_recurse_optionaldefault_f", input_type=Boolean(optional=True), prefix="-E", doc=InputDocumentation(doc="Recurse [T/F]  Optional\ndefault = F")), ToolInput(tag="in_template_file_optional", input_type=Boolean(optional=True), prefix="-t", doc=InputDocumentation(doc="Template File [File In]  Optional")), ToolInput(tag="in_file_typea_anyru", input_type=Boolean(optional=True), prefix="-a", doc=InputDocumentation(doc="File Type\na Any\nr20u Runs of 20+ Ns are gaps, 100 Ns are unknown length\nr20k Runs of 20+ Ns are gaps, 100 Ns are known length\nr10u Runs of 10+ Ns are gaps, 100 Ns are unknown length\nr10k Runs of 10+ Ns are gaps, 100 Ns are known length\ns FASTA Set (s Batch, s1 Pop, s2 Phy, s3 Mut, s4 Eco, s9 Small-genome)\nd FASTA Delta, di FASTA Delta with Implicit Gaps\nl FASTA+Gap Alignment (l Batch, l1 Pop, l2 Phy, l3 Mut, l4 Eco, l9 Small-genome)\nz FASTA with Gap Lines\ne PHRAP/ACE\nb ASN.1 for -M flag")), ToolInput(tag="in_read_fastas_optionaldefault", input_type=Boolean(optional=True), prefix="-s", doc=InputDocumentation(doc="Read FASTAs as Set [T/F]  Optional\ndefault = F")), ToolInput(tag="in_genomic_product_set", input_type=Boolean(optional=True), prefix="-g", doc=InputDocumentation(doc="Genomic Product Set [T/F]  Optional\ndefault = F")), ToolInput(tag="in_delayed_genomic_product", input_type=Boolean(optional=True), prefix="-J", doc=InputDocumentation(doc="Delayed Genomic Product Set [T/F]  Optional\ndefault = F")), ToolInput(tag="in_feature_id_linkso", input_type=Boolean(optional=True), prefix="-F", doc=InputDocumentation(doc="Feature ID Links\no By Overlap\np By Product\nl By Label and Location\ns Suppress links forced by -M")), ToolInput(tag="in_accession__optional", input_type=Boolean(optional=True), prefix="-A", doc=InputDocumentation(doc="Accession [String]  Optional")), ToolInput(tag="in_genome_center_tag", input_type=Boolean(optional=True), prefix="-C", doc=InputDocumentation(doc="Genome Center Tag [String]  Optional")), ToolInput(tag="in_organism_name_optional", input_type=Boolean(optional=True), prefix="-n", doc=InputDocumentation(doc="Organism Name [String]  Optional")), ToolInput(tag="in_source_qualifiers_optional", input_type=Boolean(optional=True), prefix="-j", doc=InputDocumentation(doc="Source Qualifiers [String]  Optional")), ToolInput(tag="in_comment__optional", input_type=Boolean(optional=True), prefix="-y", doc=InputDocumentation(doc="Comment [String]  Optional")), ToolInput(tag="in_comment_file_optional", input_type=Boolean(optional=True), prefix="-Y", doc=InputDocumentation(doc="Comment File [File In]  Optional")), ToolInput(tag="in_descriptors_file_optional", input_type=Boolean(optional=True), prefix="-D", doc=InputDocumentation(doc="Descriptors File [File In]  Optional")), ToolInput(tag="in_single_table_file", input_type=Boolean(optional=True), prefix="-f", doc=InputDocumentation(doc="Single Table File [File In]  Optional")), ToolInput(tag="in_cds_flags_combine", input_type=Boolean(optional=True), prefix="-k", doc=InputDocumentation(doc="CDS Flags (combine any of the following letters)\nc Annotate Longest ORF\nr Allow Runon ORFs\nm Allow Alternative Starts\nk Set Conflict on Mismatch")), ToolInput(tag="in_verification_combine_following", input_type=Boolean(optional=True), prefix="-V", doc=InputDocumentation(doc="Verification (combine any of the following letters)\nv Validate with Normal Stringency\nr Validate without Country Check\nc BarCode Validation\nb Generate GenBank Flatfile\ng Generate Gene Report\nt Validate with TSA Check")), ToolInput(tag="in_validate_use_v", input_type=Boolean(optional=True), prefix="-v", doc=InputDocumentation(doc="Validate (obsolete: use -V v) [T/F]  Optional\ndefault = F")), ToolInput(tag="in_generate_genbank_file", input_type=Boolean(optional=True), prefix="-b", doc=InputDocumentation(doc="Generate GenBank File (obsolete: use -V b) [T/F]  Optional\ndefault = F")), ToolInput(tag="in_seq_id_file", input_type=Boolean(optional=True), prefix="-q", doc=InputDocumentation(doc="Seq ID from File Name [T/F]  Optional\ndefault = F")), ToolInput(tag="in_genprodset_nucprotset_optionaldefault", input_type=Boolean(optional=True), prefix="-u", doc=InputDocumentation(doc="GenProdSet to NucProtSet [T/F]  Optional\ndefault = F")), ToolInput(tag="in_alignment_gap_flags", input_type=Boolean(optional=True), prefix="-G", doc=InputDocumentation(doc="Alignment Gap Flags (comma separated fields, e.g., p,-,-,-,?,. )\nn Nucleotide or p Protein,\nBegin, Middle, End Gap Characters,\nMissing Characters, Match Characters")), ToolInput(tag="in_remote_sequence_record", input_type=Boolean(optional=True), prefix="-R", doc=InputDocumentation(doc="Remote Sequence Record Fetching from ID [T/F]  Optional\ndefault = F")), ToolInput(tag="in_smart_feature_annotation", input_type=Boolean(optional=True), prefix="-S", doc=InputDocumentation(doc="Smart Feature Annotation [T/F]  Optional\ndefault = F")), ToolInput(tag="in_mrna_title_policys", input_type=Boolean(optional=True), prefix="-Q", doc=InputDocumentation(doc="mRNA Title Policy\ns Special mRNA Titles\nr RefSeq mRNA Titles")), ToolInput(tag="in_remove_unnecessary_gene", input_type=Boolean(optional=True), prefix="-U", doc=InputDocumentation(doc="Remove Unnecessary Gene Xref [T/F]  Optional\ndefault = F")), ToolInput(tag="in_force_local_proteinidtranscriptid", input_type=Boolean(optional=True), prefix="-L", doc=InputDocumentation(doc="Force Local protein_id/transcript_id [T/F]  Optional\ndefault = F")), ToolInput(tag="in_remote_taxonomy_lookup", input_type=Boolean(optional=True), prefix="-T", doc=InputDocumentation(doc="Remote Taxonomy Lookup [T/F]  Optional\ndefault = F")), ToolInput(tag="in_remote_publication_lookup", input_type=Boolean(optional=True), prefix="-P", doc=InputDocumentation(doc="Remote Publication Lookup [T/F]  Optional\ndefault = F")), ToolInput(tag="in_log_progress_optionaldefault", input_type=Boolean(optional=True), prefix="-W", doc=InputDocumentation(doc="Log Progress [T/F]  Optional\ndefault = F")), ToolInput(tag="in_save_bioseqset_optionaldefault", input_type=Boolean(optional=True), prefix="-K", doc=InputDocumentation(doc="Save Bioseq-set [T/F]  Optional\ndefault = F")), ToolInput(tag="in_hold_publishy_hold", input_type=Boolean(optional=True), prefix="-H", doc=InputDocumentation(doc="Hold Until Publish\ny Hold for One Year\nmm/dd/yyyy")), ToolInput(tag="in_discrepancy_report_output", input_type=File(optional=True), prefix="-Z", doc=InputDocumentation(doc="Discrepancy Report Output File [File Out]  Optional")), ToolInput(tag="in_cleanup_combine_following", input_type=Boolean(optional=True), prefix="-c", doc=InputDocumentation(doc="Cleanup (combine any of the following letters)\nd Correct Collection Dates (assume month first)\nD Correct Collection Dates (assume day first)\nb Append note to coding regions that overlap other coding regions with similar product names and do not contain 'ABC'\nx Extend partial ends of features by one or two nucleotides to abut gaps or sequence ends\np Add exception to non-extendable partials\ns Add exception to short introns\nf Fix product names")), ToolInput(tag="in_cleanup_log_file", input_type=File(optional=True), prefix="-z", doc=InputDocumentation(doc="Cleanup Log File [File Out]  Optional")), ToolInput(tag="in_extra_flags_combine", input_type=Boolean(optional=True), prefix="-X", doc=InputDocumentation(doc="Extra Flags (combine any of the following letters)\nA Automatic definition line generator\nC Apply comments in .cmt files to all sequences\nE Treat like eukaryota in the Discrepancy Report")), ToolInput(tag="in_project_version_number", input_type=Boolean(optional=True), prefix="-N", doc=InputDocumentation(doc="Project Version Number [Integer]  Optional\ndefault = 0")), ToolInput(tag="in_single_structured_comment", input_type=Boolean(optional=True), prefix="-w", doc=InputDocumentation(doc="Single Structured Comment File (overrides the use of -X C) [File In]  Optional")), ToolInput(tag="in_master_genome_flagsn", input_type=Boolean(optional=True), prefix="-M", doc=InputDocumentation(doc="Master Genome Flags\nn Normal\nb Big Sequence\np Power Option\nt TSA")), ToolInput(tag="in_add_type_used", input_type=Boolean(optional=True), prefix="-l", doc=InputDocumentation(doc="Add type of evidence used to assert linkage across assembly gaps (only for TSA records). Must be one of the following:")), ToolInput(tag="in_lineage_use_discrepancy", input_type=Boolean(optional=True), prefix="-m", doc=InputDocumentation(doc="Lineage to use for Discrepancy Report tests")), ToolInput(tag="in_paired_ends", input_type=String(), position=0, doc=InputDocumentation(doc="align-genus")), ToolInput(tag="in_align_x_genus", input_type=String(), position=1, doc=InputDocumentation(doc="align-trnscpt")), ToolInput(tag="in_within_clone", input_type=String(), position=2, doc=InputDocumentation(doc="clone-contig")), ToolInput(tag="in_map", input_type=String(), position=3, doc=InputDocumentation(doc="strobe"))], outputs=[ToolOutput(tag="out_single_output_file", output_type=File(optional=True), selector=InputSelector(input_to_select="in_single_output_file", type_hint=File()), doc=OutputDocumentation(doc="Single Output File [File Out]  Optional")), ToolOutput(tag="out_discrepancy_report_output", output_type=File(optional=True), selector=InputSelector(input_to_select="in_discrepancy_report_output", type_hint=File()), doc=OutputDocumentation(doc="Discrepancy Report Output File [File Out]  Optional")), ToolOutput(tag="out_cleanup_log_file", output_type=File(optional=True), selector=InputSelector(input_to_select="in_cleanup_log_file", type_hint=File()), doc=OutputDocumentation(doc="Cleanup Log File [File Out]  Optional"))], container="quay.io/biocontainers/tbl2asn-forever:25.7.2f--h516909a_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Real_Tbl2Asn_V0_1_0().translate("wdl")

