from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Int, Boolean, String

Rsem_Prepare_Reference_V0_1_0 = CommandToolBuilder(tool="rsem_prepare_reference", base_command=["rsem-prepare-reference"], inputs=[ToolInput(tag="in_gtf", input_type=File(optional=True), prefix="--gtf", doc=InputDocumentation(doc="If this option is on, RSEM assumes that 'reference_fasta_file(s)'\ncontains the sequence of a genome, and will extract transcript\nreference sequences using the gene annotations specified in <file>,\nwhich should be in GTF format.\nIf this and '--gff3' options are off, RSEM will assume\n'reference_fasta_file(s)' contains the reference transcripts. In\nthis case, RSEM assumes that name of each sequence in the\nMulti-FASTA files is its transcript_id.\n(Default: off)")), ToolInput(tag="in_gff_three", input_type=File(optional=True), prefix="--gff3", doc=InputDocumentation(doc="The annotation file is in GFF3 format instead of GTF format. RSEM\nwill first convert it to GTF format with the file name\n'reference_name.gtf'. Please make sure that 'reference_name.gtf'\ndoes not exist. (Default: off)")), ToolInput(tag="in_gff_three_rna_patterns", input_type=Int(optional=True), prefix="--gff3-RNA-patterns", doc=InputDocumentation(doc="<pattern> is a comma-separated list of transcript categories, e.g.\n'mRNA,rRNA'. Only transcripts that match the <pattern> will be\nextracted. (Default: 'mRNA')")), ToolInput(tag="in_gff_three_genes_as_transcripts", input_type=Boolean(optional=True), prefix="--gff3-genes-as-transcripts", doc=InputDocumentation(doc="This option is designed for untypical organisms, such as viruses,\nwhose GFF3 files only contain genes. RSEM will assume each gene as a\nunique transcript when it converts the GFF3 file into GTF format.")), ToolInput(tag="in_trusted_sources", input_type=String(optional=True), prefix="--trusted-sources", doc=InputDocumentation(doc="<sources> is a comma-separated list of trusted sources, e.g.\n'ENSEMBL,HAVANA'. Only transcripts coming from these sources will be\nextracted. If this option is off, all sources are accepted.\n(Default: off)")), ToolInput(tag="in_transcript_to_gene_map", input_type=File(optional=True), prefix="--transcript-to-gene-map", doc=InputDocumentation(doc="Use information from <file> to map from transcript (isoform) ids to\ngene ids. Each line of <file> should be of the form:\ngene_id transcript_id\nwith the two fields separated by a tab character.\nIf you are using a GTF file for the 'UCSC Genes' gene set from the\nUCSC Genome Browser, then the 'knownIsoforms.txt' file (obtained\nfrom the 'Downloads' section of the UCSC Genome Browser site) is of\nthis format.\nIf this option is off, then the mapping of isoforms to genes depends\non whether the '--gtf' option is specified. If '--gtf' is specified,\nthen RSEM uses the 'gene_id' and 'transcript_id' attributes in the\nGTF file. Otherwise, RSEM assumes that each sequence in the\nreference sequence files is a separate gene.\n(Default: off)")), ToolInput(tag="in_allele_to_gene_map", input_type=File(optional=True), prefix="--allele-to-gene-map", doc=InputDocumentation(doc="Use information from <file> to provide gene_id and transcript_id\ninformation for each allele-specific transcript. Each line of <file>\nshould be of the form:\ngene_id transcript_id allele_id\nwith the fields separated by a tab character.\nThis option is designed for quantifying allele-specific expression.\nIt is only valid if '--gtf' option is not specified. allele_id\nshould be the sequence names presented in the Multi-FASTA-formatted\nfiles.\n(Default: off)")), ToolInput(tag="in_polya", input_type=Boolean(optional=True), prefix="--polyA", doc=InputDocumentation(doc="Add poly(A) tails to the end of all reference isoforms. The length\nof poly(A) tail added is specified by '--polyA-length' option. STAR\naligner users may not want to use this option. (Default: do not add\npoly(A) tail to any of the isoforms)")), ToolInput(tag="in_polya_length", input_type=Int(optional=True), prefix="--polyA-length", doc=InputDocumentation(doc="The length of the poly(A) tails to be added. (Default: 125)")), ToolInput(tag="in_no_polya_subset", input_type=File(optional=True), prefix="--no-polyA-subset", doc=InputDocumentation(doc="Only meaningful if '--polyA' is specified. Do not add poly(A) tails\nto those transcripts listed in <file>. <file> is a file containing a\nlist of transcript_ids. (Default: off)")), ToolInput(tag="in_bowtie", input_type=Boolean(optional=True), prefix="--bowtie", doc=InputDocumentation(doc="Build Bowtie indices. (Default: off)")), ToolInput(tag="in_bowtie_path", input_type=File(optional=True), prefix="--bowtie-path", doc=InputDocumentation(doc="The path to the Bowtie executables. (Default: the path to Bowtie\nexecutables is assumed to be in the user's PATH environment\nvariable)")), ToolInput(tag="in_bowtie_two", input_type=Boolean(optional=True), prefix="--bowtie2", doc=InputDocumentation(doc="Build Bowtie 2 indices. (Default: off)")), ToolInput(tag="in_bowtie_two_path", input_type=File(optional=True), prefix="--bowtie2-path", doc=InputDocumentation(doc="The path to the Bowtie 2 executables. (Default: the path to Bowtie 2\nexecutables is assumed to be in the user's PATH environment\nvariable)")), ToolInput(tag="in_star", input_type=Boolean(optional=True), prefix="--star", doc=InputDocumentation(doc="Build STAR indices. (Default: off)")), ToolInput(tag="in_star_path", input_type=File(optional=True), prefix="--star-path", doc=InputDocumentation(doc="The path to STAR's executable. (Default: the path to STAR executable\nis assumed to be in user's PATH environment variable)")), ToolInput(tag="in_star_sj_db_overhang", input_type=Int(optional=True), prefix="--star-sjdboverhang", doc=InputDocumentation(doc="Length of the genomic sequence around annotated junction. It is only\nused for STAR to build splice junctions database and not needed for\nBowtie or Bowtie2. It will be passed as the --sjdbOverhang option to\nSTAR. According to STAR's manual, its ideal value is\nmax(ReadLength)-1, e.g. for 2x101 paired-end reads, the ideal value\nis 101-1=100. In most cases, the default value of 100 will work as\nwell as the ideal value. (Default: 100)")), ToolInput(tag="in_his_at_two_hca", input_type=Boolean(optional=True), prefix="--hisat2-hca", doc=InputDocumentation(doc="Build HISAT2 indices on the transcriptome according to Human Cell\nAtlas (HCA) SMART-Seq2 pipeline. (Default: off)")), ToolInput(tag="in_his_at_two_path", input_type=File(optional=True), prefix="--hisat2-path", doc=InputDocumentation(doc="The path to the HISAT2 executables. (Default: the path to HISAT2\nexecutables is assumed to be in the user's PATH environment\nvariable)")), ToolInput(tag="in_num_threads", input_type=Int(optional=True), prefix="--num-threads", doc=InputDocumentation(doc="Number of threads to use for building STAR's genome indices.\n(Default: 1)")), ToolInput(tag="in_quiet", input_type=Boolean(optional=True), prefix="--quiet", doc=InputDocumentation(doc="Suppress the output of logging information. (Default: off)")), ToolInput(tag="in_prep_pr_sem", input_type=Boolean(optional=True), prefix="--prep-pRSEM", doc=InputDocumentation(doc="A Boolean indicating whether to prepare reference files for pRSEM,\nincluding building Bowtie indices for a genome and selecting\ntraining set isoforms. The index files will be used for aligning\nChIP-seq reads in prior-enhanced RSEM and the training set isoforms\nwill be used for learning prior. A path to Bowtie executables and a\nmappability file in bigWig format are required when this option is\non. Currently, Bowtie2 is not supported for prior-enhanced RSEM.\n(Default: off)")), ToolInput(tag="in_mapp_ability_bigwig_file", input_type=File(optional=True), prefix="--mappability-bigwig-file", doc=InputDocumentation(doc="Full path to a whole-genome mappability file in bigWig format. This\nfile is required for running prior-enhanced RSEM. It is used for\nselecting a training set of isoforms for prior-learning. This file\ncan be either downloaded from UCSC Genome Browser or generated by\nGEM (Derrien et al., 2012, PLoS One). (Default: '')"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Rsem_Prepare_Reference_V0_1_0().translate("wdl", allow_empty_container=True)

