from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, File, String

Biscuit_Pileup_V0_1_0 = CommandToolBuilder(tool="biscuit_pileup", base_command=["biscuit", "pileup"], inputs=[ToolInput(tag="in_region_optional_specified", input_type=Boolean(optional=True), prefix="-g", doc=InputDocumentation(doc="region (optional, if not specified the whole bam will be processed).")), ToolInput(tag="in_number_of_threads", input_type=Boolean(optional=True), prefix="-q", doc=InputDocumentation(doc="number of threads [3].")), ToolInput(tag="in_nomeseq_mode", input_type=Boolean(optional=True), prefix="-N", doc=InputDocumentation(doc="NOMe-seq mode [off]")), ToolInput(tag="in_somatic_mode_also", input_type=Boolean(optional=True), prefix="-S", doc=InputDocumentation(doc="somatic mode, must also provide -T (tumor BAM) and -I (normal BAM) arguments [off]")), ToolInput(tag="in_somatic_mode_tumor", input_type=Boolean(optional=True), prefix="-T", doc=InputDocumentation(doc="somatic mode, tumor BAM")), ToolInput(tag="in_somatic_mode_normal", input_type=Boolean(optional=True), prefix="-I", doc=InputDocumentation(doc="somatic mode, normal BAM")), ToolInput(tag="in_pileup_output_file", input_type=File(optional=True), prefix="-o", doc=InputDocumentation(doc="pileup output file [stdout]")), ToolInput(tag="in_pileup_statistics_output", input_type=Boolean(optional=True), prefix="-w", doc=InputDocumentation(doc="pileup statistics output prefix [same as output]")), ToolInput(tag="in_verbose_print_info", input_type=Boolean(optional=True), prefix="-v", doc=InputDocumentation(doc="verbose (<5 print additional info for diagnosis, >5 debug).")), ToolInput(tag="in_min_base_quality", input_type=Boolean(optional=True), prefix="-b", doc=InputDocumentation(doc="min base quality [20].")), ToolInput(tag="in_minimum_mapping_quality", input_type=Boolean(optional=True), prefix="-m", doc=InputDocumentation(doc="minimum mapping quality [40].")), ToolInput(tag="in_minimum_alignment_score", input_type=Boolean(optional=True), prefix="-a", doc=InputDocumentation(doc="minimum alignment score (from AS-tag) [40].")), ToolInput(tag="in_max_cytosine_retention", input_type=Boolean(optional=True), prefix="-t", doc=InputDocumentation(doc="max cytosine retention in a read [999999].")), ToolInput(tag="in_minimum_read_length", input_type=Boolean(optional=True), prefix="-l", doc=InputDocumentation(doc="minimum read length [10].")), ToolInput(tag="in_minimum_distance_end", input_type=Boolean(optional=True), prefix="-e", doc=InputDocumentation(doc="minimum distance to end of a read [3].")), ToolInput(tag="in_redistribution_ambiguous_genotyping", input_type=Boolean(optional=True), prefix="-r", doc=InputDocumentation(doc="NO redistribution of ambiguous (Y/R) calls in SNP genotyping.")), ToolInput(tag="in_filtering_secondary_mapping", input_type=Boolean(optional=True), prefix="-c", doc=InputDocumentation(doc="NO filtering secondary mapping.")), ToolInput(tag="in_double_counting_cytosine", input_type=Boolean(optional=True), prefix="-d", doc=InputDocumentation(doc="NO double counting cytosine in overlapping mate reads.")), ToolInput(tag="in_no_filtering_duplicate", input_type=Boolean(optional=True), prefix="-u", doc=InputDocumentation(doc="NO filtering of duplicate.")), ToolInput(tag="in_filtering_improper_pair", input_type=Boolean(optional=True), prefix="-p", doc=InputDocumentation(doc="NO filtering of improper pair.")), ToolInput(tag="in_maximum_nm_tag", input_type=Boolean(optional=True), prefix="-n", doc=InputDocumentation(doc="maximum NM tag [999999].")), ToolInput(tag="in_error_rate_", input_type=Boolean(optional=True), prefix="-E", doc=InputDocumentation(doc="error rate [0.001].")), ToolInput(tag="in_mutation_rate_", input_type=Boolean(optional=True), prefix="-M", doc=InputDocumentation(doc="mutation rate [0.001].")), ToolInput(tag="in_somatic_mutation_rate", input_type=Boolean(optional=True), prefix="-x", doc=InputDocumentation(doc="somatic mutation rate [0.001].")), ToolInput(tag="in_contamination_rate_", input_type=Boolean(optional=True), prefix="-C", doc=InputDocumentation(doc="contamination rate [0.010].")), ToolInput(tag="in_prior_probability_heterozygous", input_type=Boolean(optional=True), prefix="-P", doc=InputDocumentation(doc="prior probability for heterozygous variant [0.333].")), ToolInput(tag="in_prior_probability_homozygous", input_type=Boolean(optional=True), prefix="-Q", doc=InputDocumentation(doc="prior probability for homozygous variant [0.333].")), ToolInput(tag="in_pile_up", input_type=String(), position=0, doc=InputDocumentation(doc="")), ToolInput(tag="in_ref_dot_fa", input_type=String(), position=1, doc=InputDocumentation(doc="")), ToolInput(tag="in_in_dot_bam_dot_dot", input_type=String(), position=2, doc=InputDocumentation(doc=""))], outputs=[ToolOutput(tag="out_pileup_output_file", output_type=File(optional=True), selector=InputSelector(input_to_select="in_pileup_output_file", type_hint=File()), doc=OutputDocumentation(doc="pileup output file [stdout]"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Biscuit_Pileup_V0_1_0().translate("wdl", allow_empty_container=True)

