class: CommandLineTool
id: ../../../../home/ubuntu/BiocondaCli/rsem_control_fdr.cwl
inputs:
- id: hard_threshold
  doc: 'Use hard threshold method to control FDR. If this option is set, only those
    genes/transcripts with their PPDE >= 1 - fdr_rate are called as DE. (Default:
    on)'
  type: boolean
  inputBinding:
    prefix: --hard-threshold
- id: soft_threshold
  doc: "Use soft threshold method to control FDR. If this option is set, this program\
    \ will try to report as many genes/transcripts as possible, as long as their average\
    \ PPDE >= 1 - fdr_rate. This option is equivalent to use EBSeq's 'crit_fun' for\
    \ FDR control. (Default: off)"
  type: boolean
  inputBinding:
    prefix: --soft-threshold
- id: h_slash_help
  doc: Show help information.
  type: boolean
  inputBinding:
    prefix: -h/--help
- id: input_file
  doc: This should be the main result file generated by 'rsem-run-ebseq', which contains
    all genes/transcripts and their associated statistics.
  type: string
  inputBinding:
    position: 0
- id: fdr_rate
  doc: The desire false discovery rate (FDR).
  type: string
  inputBinding:
    position: 1
- id: output_file
  doc: This file is a subset of the 'input_file'. It only contains the genes/transcripts
    called as differentially expressed (DE). When more than 2 conditions exist, DE
    is defined as not all conditions are equally expressed. Because statistical significance
    does not necessarily mean biological significance, users should also refer to
    the fold changes to decide which genes/transcripts are biologically significant.
    When more than two conditions exist, this file will not contain fold change information
    and users need to calculate it from 'input_file.condmeans' by themselves.
  type: string
  inputBinding:
    position: 2
outputs: []
cwlVersion: v1.1
baseCommand:
- rsem-control-fdr
