from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import String, Directory, Int, Boolean, File

Sonicparanoid_Extract_V0_1_0 = CommandToolBuilder(tool="sonicparanoid_extract", base_command=["sonicparanoid-extract"], inputs=[ToolInput(tag="in_input_table", input_type=String(optional=True), prefix="--input-table", doc=InputDocumentation(doc="Table with output groups generated using\nSonicParanoid. NOTE: the headers must not be modified\nor removed.")), ToolInput(tag="in_output_directory", input_type=Directory(optional=True), prefix="--output-directory", doc=InputDocumentation(doc="The directory in which the results will be stored.")), ToolInput(tag="in_fast_a_directory", input_type=Directory(optional=True), prefix="--fasta-directory", doc=InputDocumentation(doc="Directory containing the proteome files in FASTA\nformat.")), ToolInput(tag="in_min_sp", input_type=Int(optional=True), prefix="--min-sp", doc=InputDocumentation(doc="Exctract clusters with genes from at least --min-sp\nspecies. (Default=2)")), ToolInput(tag="in_max_sp", input_type=Int(optional=True), prefix="--max-sp", doc=InputDocumentation(doc="Exctract clusters with genes from at max --max-sp\nspecies. (Default=2)")), ToolInput(tag="in_ids_list", input_type=Int(optional=True), prefix="--ids-list", doc=InputDocumentation(doc="Extract a list of clusters by their ids. NOTE: the ids\nshould be separated by a comma (e.g., --ids-list\n20,23,40,22); this parameter bypasses the --min-sp and\nmax-sp.")), ToolInput(tag="in_min_conf", input_type=Int(optional=True), prefix="--min-conf", doc=InputDocumentation(doc="Keep only orthologs with a confidence higher than\n--min-conf. (Default=0.05)")), ToolInput(tag="in_fast_a", input_type=Boolean(optional=True), prefix="--fasta", doc=InputDocumentation(doc="Generate a FASTA for with the proteins in each")), ToolInput(tag="in_an_not_cols", input_type=Int(optional=True), prefix="--annot-cols", doc=InputDocumentation(doc="Column ids containing annotations for the new HDR.\nNOTE: the column positions should be separated by a\ncomma (e.g., --annot-cols 1,2,4,5); this parameter\nbypasses.")), ToolInput(tag="in_an_not_file", input_type=File(optional=True), prefix="--annot-file", doc=InputDocumentation(doc="Table with tab-serataed fields with annotations.")), ToolInput(tag="in_gene_col_id", input_type=Int(optional=True), prefix="--gene-col-id", doc=InputDocumentation(doc="Column number of the annotation file containing the\nprotein id.")), ToolInput(tag="in_debug", input_type=Boolean(optional=True), prefix="--debug", doc=InputDocumentation(doc="Output debug information.")), ToolInput(tag="in_cluster_dot", input_type=String(), position=0, doc=InputDocumentation(doc="-mf, --multiple-fasta"))], outputs=[ToolOutput(tag="out_output_directory", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_output_directory", type_hint=File()), doc=OutputDocumentation(doc="The directory in which the results will be stored."))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Sonicparanoid_Extract_V0_1_0().translate("wdl", allow_empty_container=True)

