from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Boolean, String, Int, Directory, Float

Rnabloom_V0_1_0 = CommandToolBuilder(tool="rnabloom", base_command=["rnabloom"], inputs=[ToolInput(tag="in_left", input_type=File(optional=True), prefix="--left", doc=InputDocumentation(doc="left reads file(s)")), ToolInput(tag="in_right", input_type=File(optional=True), prefix="--right", doc=InputDocumentation(doc="right reads file(s)")), ToolInput(tag="in_pool", input_type=File(optional=True), prefix="--pool", doc=InputDocumentation(doc="list of read files for pooled assembly")), ToolInput(tag="in_long", input_type=File(optional=True), prefix="-long", doc=InputDocumentation(doc="long reads file(s)\n(Requires `minimap2` and `racon` in\nPATH. Presets `-k 17 -c 3 -indel 10 -e\n3 -p 0.8 -overlap 200 -tip 100` unless\neach option is defined otherwise.)")), ToolInput(tag="in_ref", input_type=File(optional=True), prefix="-ref", doc=InputDocumentation(doc="reference transcripts file(s) for\nguiding the assembly process")), ToolInput(tag="in_rev_comp_left", input_type=Boolean(optional=True), prefix="--revcomp-left", doc=InputDocumentation(doc="reverse-complement left reads [false]")), ToolInput(tag="in_rev_comp_right", input_type=Boolean(optional=True), prefix="--revcomp-right", doc=InputDocumentation(doc="reverse-complement right reads [false]")), ToolInput(tag="in_rev_comp_long", input_type=Boolean(optional=True), prefix="--revcomp-long", doc=InputDocumentation(doc="reverse-complement long reads [false]")), ToolInput(tag="in_stranded", input_type=Boolean(optional=True), prefix="--stranded", doc=InputDocumentation(doc="reads are strand specific [false]")), ToolInput(tag="in_name", input_type=String(optional=True), prefix="--name", doc=InputDocumentation(doc="assembly name [rnabloom]")), ToolInput(tag="in_prefix", input_type=String(optional=True), prefix="-prefix", doc=InputDocumentation(doc="name prefix in FASTA header for\nassembled transcripts")), ToolInput(tag="in_uracil", input_type=Boolean(optional=True), prefix="--uracil", doc=InputDocumentation(doc="output uracils (U) in place of thymines\n(T) in assembled transcripts [false]")), ToolInput(tag="in_threads", input_type=Int(optional=True), prefix="--threads", doc=InputDocumentation(doc="number of threads to run [2]")), ToolInput(tag="in_outdir", input_type=Directory(optional=True), prefix="--outdir", doc=InputDocumentation(doc="output directory [//rnabloom_assembly]")), ToolInput(tag="in_force", input_type=Boolean(optional=True), prefix="--force", doc=InputDocumentation(doc="force overwrite existing files [false]")), ToolInput(tag="in_km_er", input_type=Int(optional=True), prefix="--kmer", doc=InputDocumentation(doc="k-mer size [25]")), ToolInput(tag="in_stage", input_type=Int(optional=True), prefix="-stage", doc=InputDocumentation(doc="assembly termination stage\nshort reads: [3]\n1. construct graph\n2. assemble fragments\n3. assemble transcripts\nlong reads: [3]\n1. construct graph\n2. correct reads\n3. assemble transcripts")), ToolInput(tag="in_qual_dbg", input_type=Int(optional=True), prefix="--qual-dbg", doc=InputDocumentation(doc="minimum base quality in reads for\nconstructing DBG [3]")), ToolInput(tag="in_qual_frag", input_type=Int(optional=True), prefix="--qual-frag", doc=InputDocumentation(doc="minimum base quality in reads for\nfragment reconstruction [3]")), ToolInput(tag="in_min_cov", input_type=Int(optional=True), prefix="--mincov", doc=InputDocumentation(doc="minimum k-mer coverage [1]")), ToolInput(tag="in_hash", input_type=Int(optional=True), prefix="-hash", doc=InputDocumentation(doc="number of hash functions for all Bloom\nfilters [2]")), ToolInput(tag="in_sbf_hash", input_type=Int(optional=True), prefix="--sbf-hash", doc=InputDocumentation(doc="number of hash functions for screening\nBloom filter [2]")), ToolInput(tag="in_dbg_bf_hash", input_type=Int(optional=True), prefix="--dbgbf-hash", doc=InputDocumentation(doc="number of hash functions for de Bruijn\ngraph Bloom filter [2]")), ToolInput(tag="in_cbf_hash", input_type=Int(optional=True), prefix="--cbf-hash", doc=InputDocumentation(doc="number of hash functions for k-mer\ncounting Bloom filter [2]")), ToolInput(tag="in_pkb_f_hash", input_type=Int(optional=True), prefix="--pkbf-hash", doc=InputDocumentation(doc="number of hash functions for paired\nk-mers Bloom filter [2]")), ToolInput(tag="in_num_km_ers", input_type=Int(optional=True), prefix="--num-kmers", doc=InputDocumentation(doc="expected number of unique k-mers in\ninput reads")), ToolInput(tag="in_nt_card", input_type=Boolean(optional=True), prefix="-ntcard", doc=InputDocumentation(doc="count unique k-mers in input reads with\nntCard [false]\n(Requires `ntcard` in PATH. If this\noption is used along with `-long`, the\nvalue for `-c` is set automatically\nbased on the ntCard histogram, unless\n`-c` is defined otherwise)")), ToolInput(tag="in_memory", input_type=Float(optional=True), prefix="--memory", doc=InputDocumentation(doc="total amount of memory (GB) for all\nBloom filters [auto]")), ToolInput(tag="in_sbf_mem", input_type=Float(optional=True), prefix="--sbf-mem", doc=InputDocumentation(doc="amount of memory (GB) for screening\nBloom filter [auto]")), ToolInput(tag="in_dbg_bf_mem", input_type=Float(optional=True), prefix="--dbgbf-mem", doc=InputDocumentation(doc="amount of memory (GB) for de Bruijn\ngraph Bloom filter [auto]")), ToolInput(tag="in_cbf_mem", input_type=Float(optional=True), prefix="--cbf-mem", doc=InputDocumentation(doc="amount of memory (GB) for k-mer\ncounting Bloom filter [auto]")), ToolInput(tag="in_pkb_f_mem", input_type=Float(optional=True), prefix="--pkbf-mem", doc=InputDocumentation(doc="amount of memory (GB) for paired kmers\nBloom filter [auto]")), ToolInput(tag="in_fpr", input_type=Float(optional=True), prefix="--fpr", doc=InputDocumentation(doc="maximum allowable false-positive rate\nof Bloom filters [0.01]")), ToolInput(tag="in_save_bf", input_type=Boolean(optional=True), prefix="-savebf", doc=InputDocumentation(doc="save graph (Bloom filters) from stage 1\nto disk [false]")), ToolInput(tag="in_tip_length", input_type=Int(optional=True), prefix="-tiplength", doc=InputDocumentation(doc="maximum number of bases in a tip [5]")), ToolInput(tag="in_look_ahead", input_type=Int(optional=True), prefix="-lookahead", doc=InputDocumentation(doc="number of k-mers to look ahead during\ngraph traversal [3]")), ToolInput(tag="in_sample", input_type=Int(optional=True), prefix="-sample", doc=InputDocumentation(doc="sample size for estimating\nread/fragment lengths [1000]")), ToolInput(tag="in_err_corr_it_r", input_type=Int(optional=True), prefix="--errcorritr", doc=InputDocumentation(doc="number of iterations of\nerror-correction in reads [1]")), ToolInput(tag="in_max_cov_grad", input_type=Float(optional=True), prefix="--maxcovgrad", doc=InputDocumentation(doc="maximum k-mer coverage gradient for\nerror correction [0.50]")), ToolInput(tag="in_in_del", input_type=Int(optional=True), prefix="-indel", doc=InputDocumentation(doc="maximum size of indels to be collapsed\n[1]")), ToolInput(tag="in_percent", input_type=Float(optional=True), prefix="--percent", doc=InputDocumentation(doc="minimum percent identity of sequences\nto be collapsed [0.90]")), ToolInput(tag="in_length", input_type=Int(optional=True), prefix="-length", doc=InputDocumentation(doc="minimum transcript length in output\nassembly [200]")), ToolInput(tag="in_norr", input_type=Boolean(optional=True), prefix="-norr", doc=InputDocumentation(doc="skip redundancy reduction for assembled\ntranscripts [false]\n(will not create 'transcripts.nr.fa')")), ToolInput(tag="in_merge_pool", input_type=Boolean(optional=True), prefix="-mergepool", doc=InputDocumentation(doc="merge pooled assemblies [false]\n(Requires `-pool`; overrides `-norr`)")), ToolInput(tag="in_overlap", input_type=Int(optional=True), prefix="-overlap", doc=InputDocumentation(doc="minimum number of overlapping bases\nbetween reads [10]")), ToolInput(tag="in_bound", input_type=Int(optional=True), prefix="-bound", doc=InputDocumentation(doc="maximum distance between read mates\n[500]")), ToolInput(tag="in_extend", input_type=Boolean(optional=True), prefix="-extend", doc=InputDocumentation(doc="extend fragments outward during\nfragment reconstruction [false]")), ToolInput(tag="in_no_fc", input_type=Boolean(optional=True), prefix="-nofc", doc=InputDocumentation(doc="turn off assembly consistency with\nfragment paired k-mers [false]")), ToolInput(tag="in_sensitive", input_type=Boolean(optional=True), prefix="-sensitive", doc=InputDocumentation(doc="assemble transcripts in sensitive mode\n[false]")), ToolInput(tag="in_artifact", input_type=Boolean(optional=True), prefix="-artifact", doc=InputDocumentation(doc="keep potential sequencing artifacts\n[false]")), ToolInput(tag="in_chimera", input_type=Boolean(optional=True), prefix="-chimera", doc=InputDocumentation(doc="keep potential chimeras [false]")), ToolInput(tag="in_stratum", input_type=Float(optional=True), prefix="-stratum", doc=InputDocumentation(doc="fragments lower than the specified\nstratum are extended only if they are\nbranch-free in the graph [e0]")), ToolInput(tag="in_pair", input_type=Int(optional=True), prefix="-pair", doc=InputDocumentation(doc="minimum number of consecutive k-mer\npairs for assembling transcripts [10]")), ToolInput(tag="in_polya", input_type=Int(optional=True), prefix="--polya", doc=InputDocumentation(doc="prioritize assembly of transcripts with\npoly-A tails of the minimum length\nspecified [0]")), ToolInput(tag="in_mm_opt", input_type=Int(optional=True), prefix="-mmopt", doc=InputDocumentation(doc="options for minimap2 [-r 150]\n(`-x` and `-t` are already in use)")), ToolInput(tag="in_l_rop", input_type=Float(optional=True), prefix="-lrop", doc=InputDocumentation(doc="minimum proportion of matching bases\nwithin long-read overlaps [0.4]")), ToolInput(tag="in_lrrd", input_type=Int(optional=True), prefix="-lrrd", doc=InputDocumentation(doc="min read depth required for long-read\nassembly [2]")), ToolInput(tag="in_lrp_b", input_type=Boolean(optional=True), prefix="-lrpb", doc=InputDocumentation(doc="use PacBio preset for minimap2 [false]")), ToolInput(tag="in_debug", input_type=Boolean(optional=True), prefix="-debug", doc=InputDocumentation(doc="print debugging information [false]")), ToolInput(tag="in_jar", input_type=String(optional=True), prefix="-jar", doc=InputDocumentation(doc="")), ToolInput(tag="in_java", input_type=String(), position=0, doc=InputDocumentation(doc=""))], outputs=[ToolOutput(tag="out_outdir", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_outdir", type_hint=File()), doc=OutputDocumentation(doc="output directory [//rnabloom_assembly]"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Rnabloom_V0_1_0().translate("wdl", allow_empty_container=True)

