from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Directory, Int, String, Boolean

Dram_V_Py_Annotate_V0_1_0 = CommandToolBuilder(tool="DRAM_v.py_annotate", base_command=["DRAM-v.py", "annotate"], inputs=[ToolInput(tag="in_input_fast_a", input_type=File(optional=True), prefix="--input_fasta", doc=InputDocumentation(doc="fasta file, output from (default: None)")), ToolInput(tag="in_vir_sorter_affi_contigs", input_type=File(optional=True), prefix="--virsorter_affi_contigs", doc=InputDocumentation(doc="VirSorter VIRSorter_affi-contigs.tab output file\n(default: None)")), ToolInput(tag="in_output_dir", input_type=Directory(optional=True), prefix="--output_dir", doc=InputDocumentation(doc="output directory (default: None)")), ToolInput(tag="in_min_contig_size", input_type=Int(optional=True), prefix="--min_contig_size", doc=InputDocumentation(doc="minimum contig size to be used for gene prediction\n(default: 2500)")), ToolInput(tag="in_prodigal_mode", input_type=String(optional=True), prefix="--prodigal_mode", doc=InputDocumentation(doc="Mode of prodigal to use for gene calling. NOTE: normal\nor single mode require genomes which are high quality\nwith low contamination and long contigs (average\nlength >3 Kbp). (default: meta)")), ToolInput(tag="in_trans_table", input_type=String(optional=True), prefix="--trans_table", doc=InputDocumentation(doc="Translation table for prodigal to use for gene calling\n(default: 11)")), ToolInput(tag="in_bit_score_threshold", input_type=Int(optional=True), prefix="--bit_score_threshold", doc=InputDocumentation(doc="minimum bitScore of search to retain hits (default:\n60)")), ToolInput(tag="in_rbh_bit_score_threshold", input_type=Int(optional=True), prefix="--rbh_bit_score_threshold", doc=InputDocumentation(doc="minimum bitScore of reverse best hits to retain hits\n(default: 350)")), ToolInput(tag="in_custom_db_name", input_type=String(optional=True), prefix="--custom_db_name", doc=InputDocumentation(doc="Names of custom databases, can be usedmultiple times.\n(default: None)")), ToolInput(tag="in_custom_fast_a_loc", input_type=String(optional=True), prefix="--custom_fasta_loc", doc=InputDocumentation(doc="Location of fastas to annotated against, can be used\nmultiple times butmust match nubmer of\ncustom_db_name's (default: None)")), ToolInput(tag="in_use_uniref", input_type=Boolean(optional=True), prefix="--use_uniref", doc=InputDocumentation(doc="Annotate these fastas against UniRef, drastically\ndecreases run time and memory requirements (default:\nFalse)")), ToolInput(tag="in_low_mem_mode", input_type=Boolean(optional=True), prefix="--low_mem_mode", doc=InputDocumentation(doc="Skip annotating with uniref and use kofam instead of\nKEGG genes even if provided. Drastically decreases\nmemory usage (default: False)")), ToolInput(tag="in_skip_trnascan", input_type=Int(optional=True), prefix="--skip_trnascan", doc=InputDocumentation(doc="number of processors to use (default: 10)"))], outputs=[ToolOutput(tag="out_input_fast_a", output_type=File(optional=True), selector=InputSelector(input_to_select="in_input_fast_a", type_hint=File()), doc=OutputDocumentation(doc="fasta file, output from (default: None)")), ToolOutput(tag="out_vir_sorter_affi_contigs", output_type=File(optional=True), selector=InputSelector(input_to_select="in_vir_sorter_affi_contigs", type_hint=File()), doc=OutputDocumentation(doc="VirSorter VIRSorter_affi-contigs.tab output file\n(default: None)")), ToolOutput(tag="out_output_dir", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_output_dir", type_hint=File()), doc=OutputDocumentation(doc="output directory (default: None)"))], container="quay.io/biocontainers/dram:1.1.1--py_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Dram_V_Py_Annotate_V0_1_0().translate("wdl")

