from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, File, Int, Directory, String

Qualimap_Multi_Bamqc_V0_1_0 = CommandToolBuilder(tool="qualimap_multi_bamqc", base_command=["qualimap", "multi-bamqc"], inputs=[ToolInput(tag="in_paint_chromosome_limits", input_type=Boolean(optional=True), prefix="--paint-chromosome-limits", doc=InputDocumentation(doc="Only for -r mode. Paint chromosome limits\ninside charts")), ToolInput(tag="in_data", input_type=File(optional=True), prefix="--data", doc=InputDocumentation(doc="File describing the input data. Format of the\nfile is a 2- or 3-column tab-delimited table.\nColumn 1: sample name\nColumn 2: either path to the BAM QC result or\npath to BAM file (-r mode)\nColumn 3: group name (activates sample group\nmarking)")), ToolInput(tag="in_feature_file", input_type=File(optional=True), prefix="--feature-file", doc=InputDocumentation(doc="Only for -r mode. Feature file with regions of\ninterest in GFF/GTF or BED format")), ToolInput(tag="in_hm", input_type=Int(optional=True), prefix="-hm", doc=InputDocumentation(doc="Only for -r mode. Minimum size for a\nhomopolymer to be considered in indel analysis\n(default is 3)")), ToolInput(tag="in_nr", input_type=Int(optional=True), prefix="-nr", doc=InputDocumentation(doc="Only for -r mode. Number of reads analyzed in\na chunk (default is 1000)")), ToolInput(tag="in_nw", input_type=Int(optional=True), prefix="-nw", doc=InputDocumentation(doc="Only for -r mode. Number of windows (default\nis 400)")), ToolInput(tag="in_outdir", input_type=Directory(optional=True), prefix="-outdir", doc=InputDocumentation(doc="Output folder for HTML report and raw data.")), ToolInput(tag="in_outfile", input_type=File(optional=True), prefix="-outfile", doc=InputDocumentation(doc="Output file for PDF report (default value is\nreport.pdf).")), ToolInput(tag="in_out_format", input_type=String(optional=True), prefix="-outformat", doc=InputDocumentation(doc="Format of the output report (PDF, HTML or both\nPDF:HTML, default is HTML).")), ToolInput(tag="in_sequencing_protocol", input_type=String(optional=True), prefix="--sequencing-protocol", doc=InputDocumentation(doc="Only for -r mode. Sequencing library protocol:\nstrand-specific-forward,\nstrand-specific-reverse or non-strand-specific\n(default)")), ToolInput(tag="in_run_bam_qc", input_type=Boolean(optional=True), prefix="--run-bamqc", doc=InputDocumentation(doc="Raw BAM files are provided as input. If this\noption is activated BAM QC process first will\nbe run for each sample, then multi-sample\nanalysis will be performed.\n"))], outputs=[ToolOutput(tag="out_outdir", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_outdir", type_hint=File()), doc=OutputDocumentation(doc="Output folder for HTML report and raw data.")), ToolOutput(tag="out_outfile", output_type=File(optional=True), selector=InputSelector(input_to_select="in_outfile", type_hint=File()), doc=OutputDocumentation(doc="Output file for PDF report (default value is\nreport.pdf)."))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Qualimap_Multi_Bamqc_V0_1_0().translate("wdl", allow_empty_container=True)

