from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Int, Boolean, Float

Hhsearch_V0_1_0 = CommandToolBuilder(tool="hhsearch", base_command=["hhsearch"], inputs=[ToolInput(tag="in_inputquery_multiple_sequence", input_type=File(optional=True), prefix="-i", doc=InputDocumentation(doc="input/query multiple sequence alignment (a2m, a3m, FASTA) or HMM")), ToolInput(tag="in_database_name_eg", input_type=Int(optional=True), prefix="-d", doc=InputDocumentation(doc="database name (e.g. uniprot20_29Feb2012)\nMultiple databases may be specified with '-d <db1> -d <db2> ...'")), ToolInput(tag="in_evalue_cutoff_inclusion", input_type=Boolean(optional=True), prefix="-e", doc=InputDocumentation(doc="[0,1]   E-value cutoff for inclusion in result alignment (def=0.001)")), ToolInput(tag="in_no_tags", input_type=Boolean(optional=True), prefix="-notags", doc=InputDocumentation(doc="do NOT / do neutralize His-, C-myc-, FLAG-tags, and trypsin\nrecognition sequence to background distribution (def=-notags)")), ToolInput(tag="in_write_results_file", input_type=File(optional=True), prefix="-o", doc=InputDocumentation(doc="write results in standard format to file (default=<infile.hhr>)")), ToolInput(tag="in_o_a_three_m", input_type=File(optional=True), prefix="-oa3m", doc=InputDocumentation(doc="write result MSA with significant matches in a3m format")), ToolInput(tag="in_blast_tab", input_type=Int(optional=True), prefix="-blasttab", doc=InputDocumentation(doc="write result in tabular BLAST format (compatible to -m 8 or -outfmt 6 output)\n1      2      3           4         5        6      8    9      10   11   12\n'query target #match/tLen #mismatch #gapOpen qstart qend tstart tend eval score'")), ToolInput(tag="in_opsi", input_type=File(optional=True), prefix="-opsi", doc=InputDocumentation(doc="write result MSA of significant matches in PSI-BLAST format")), ToolInput(tag="in_oh_hm", input_type=File(optional=True), prefix="-ohhm", doc=InputDocumentation(doc="write HHM file for result MSA of significant matches")), ToolInput(tag="in_add_cons", input_type=Boolean(optional=True), prefix="-add_cons", doc=InputDocumentation(doc="generate consensus sequence as master sequence of query MSA (default=don't)")), ToolInput(tag="in_hide_cons", input_type=Boolean(optional=True), prefix="-hide_cons", doc=InputDocumentation(doc="don't show consensus sequence in alignments (default=show)")), ToolInput(tag="in_hide_pred", input_type=Boolean(optional=True), prefix="-hide_pred", doc=InputDocumentation(doc="don't show predicted 2ndary structure in alignments (default=show)")), ToolInput(tag="in_hide_dssp", input_type=Boolean(optional=True), prefix="-hide_dssp", doc=InputDocumentation(doc="don't show DSSP 2ndary structure in alignments (default=show)")), ToolInput(tag="in_show_ss_conf", input_type=Boolean(optional=True), prefix="-show_ssconf", doc=InputDocumentation(doc="show confidences for predicted 2ndary structure in alignments")), ToolInput(tag="in_of_as", input_type=File(optional=True), prefix="-Ofas", doc=InputDocumentation(doc="write pairwise alignments in FASTA xor A2M (-Oa2m) xor A3M (-Oa3m) format")), ToolInput(tag="in_seq", input_type=Int(optional=True), prefix="-seq", doc=InputDocumentation(doc="max. number of query/template sequences displayed (default=1)")), ToolInput(tag="in_ali_w", input_type=Int(optional=True), prefix="-aliw", doc=InputDocumentation(doc="number of columns per line in alignment list (default=80)")), ToolInput(tag="in_minimum_probability_default", input_type=Boolean(optional=True), prefix="-p", doc=InputDocumentation(doc="[0,100]     minimum probability in summary and alignment list (default=20)")), ToolInput(tag="in_inf_maximum_evalue", input_type=Boolean(optional=True), prefix="-E", doc=InputDocumentation(doc="[0,inf[     maximum E-value in summary and alignment list (default=1E+06)")), ToolInput(tag="in_maximum_number_lines", input_type=Int(optional=True), prefix="-Z", doc=InputDocumentation(doc="maximum number of lines in summary hit list (default=500)")), ToolInput(tag="in_minimum_number_lines", input_type=Int(optional=True), prefix="-z", doc=InputDocumentation(doc="minimum number of lines in summary hit list (default=10)")), ToolInput(tag="in_maximum_number_alignments", input_type=Int(optional=True), prefix="-B", doc=InputDocumentation(doc="maximum number of alignments in alignment list (default=500)")), ToolInput(tag="in_minimum_number_alignments", input_type=Int(optional=True), prefix="-b", doc=InputDocumentation(doc="minimum number of alignments in alignment list (default=10)")), ToolInput(tag="in_all", input_type=Boolean(optional=True), prefix="-all", doc=InputDocumentation(doc="show all sequences in result MSA; do not filter result MSA")), ToolInput(tag="in_id", input_type=Boolean(optional=True), prefix="-id", doc=InputDocumentation(doc="[0,100]  maximum pairwise sequence identity (def=90)")), ToolInput(tag="in_diff", input_type=Boolean(optional=True), prefix="-diff", doc=InputDocumentation(doc="[0,inf[  filter MSAs by selecting most diverse set of sequences, keeping\nat least this many seqs in each MSA block of length 50\nZero and non-numerical values turn off the filtering. (def=100)")), ToolInput(tag="in_cov", input_type=Boolean(optional=True), prefix="-cov", doc=InputDocumentation(doc="[0,100]  minimum coverage with master sequence (%) (def=0)")), ToolInput(tag="in_qid", input_type=Boolean(optional=True), prefix="-qid", doc=InputDocumentation(doc="[0,100]  minimum sequence identity with master sequence (%) (def=0)")), ToolInput(tag="in_qsc", input_type=Boolean(optional=True), prefix="-qsc", doc=InputDocumentation(doc="[0,100]  minimum score per column with master sequence (default=-20.0)")), ToolInput(tag="in_neff", input_type=Boolean(optional=True), prefix="-neff", doc=InputDocumentation(doc="[1,inf]  target diversity of multiple sequence alignment (default=off)")), ToolInput(tag="in_mark", input_type=Boolean(optional=True), prefix="-mark", doc=InputDocumentation(doc="do not filter out sequences marked by '>@'in their name line")), ToolInput(tag="in_no_realign", input_type=Boolean(optional=True), prefix="-norealign", doc=InputDocumentation(doc="do NOT realign displayed hits with MAC algorithm (def=realign)")), ToolInput(tag="in_ovlp", input_type=Int(optional=True), prefix="-ovlp", doc=InputDocumentation(doc="banded alignment: forbid <ovlp> largest diagonals |i-j| of DP matrix (def=0)")), ToolInput(tag="in_mact", input_type=Boolean(optional=True), prefix="-mact", doc=InputDocumentation(doc="[0,1[         posterior prob threshold for MAC realignment controlling greedi-\nness at alignment ends: 0:global >0.1:local (default=0.35)")), ToolInput(tag="in_glob", input_type=Boolean(optional=True), prefix="-glob", doc=InputDocumentation(doc="use global/local alignment mode for searching/ranking (def=local)")), ToolInput(tag="in_realign", input_type=Boolean(optional=True), prefix="-realign", doc=InputDocumentation(doc="realign displayed hits with max. accuracy (MAC) algorithm")), ToolInput(tag="in_excl", input_type=Int(optional=True), prefix="-excl", doc=InputDocumentation(doc="exclude query positions from the alignment, e.g. '1-33,97-168'")), ToolInput(tag="in_realign_max", input_type=Int(optional=True), prefix="-realign_max", doc=InputDocumentation(doc="realign max. <int> hits (default=500)")), ToolInput(tag="in_alt", input_type=Int(optional=True), prefix="-alt", doc=InputDocumentation(doc="show up to this many alternative alignments with raw score > smin(def=4)")), ToolInput(tag="in_sm_in", input_type=Float(optional=True), prefix="-smin", doc=InputDocumentation(doc="minimum raw score for alternative alignments (def=20.0)")), ToolInput(tag="in_shift", input_type=Boolean(optional=True), prefix="-shift", doc=InputDocumentation(doc="[-1,1]       profile-profile score offset (def=-0.03)")), ToolInput(tag="in_corr", input_type=Boolean(optional=True), prefix="-corr", doc=InputDocumentation(doc="[0,1]         weight of term for pair correlations (def=0.10)")), ToolInput(tag="in_sc", input_type=Boolean(optional=True), prefix="-sc", doc=InputDocumentation(doc="<int>         amino acid score         (tja: template HMM at column j) (def=1)\n0       = log2 Sum(tja*qia/pa)   (pa: aa background frequencies)\n1       = log2 Sum(tja*qia/pqa)  (pqa = 1/2*(pa+ta) )\n2       = log2 Sum(tja*qia/ta)   (ta: av. aa freqs in template)\n3       = log2 Sum(tja*qia/qa)   (qa: av. aa freqs in query)\n5       local amino acid composition correction")), ToolInput(tag="in_ssm", input_type=Boolean(optional=True), prefix="-ssm", doc=InputDocumentation(doc="{0,..,4}    0:   no ss scoring\n1,2: ss scoring after or during alignment  [default=2]\n3,4: ss scoring after or during alignment, predicted vs. predicted")), ToolInput(tag="in_ssw", input_type=Boolean(optional=True), prefix="-ssw", doc=InputDocumentation(doc="[0,1]          weight of ss score  (def=0.11)")), ToolInput(tag="in_ssa", input_type=Boolean(optional=True), prefix="-ssa", doc=InputDocumentation(doc="[0,1]          SS substitution matrix = (1-ssa)*I + ssa*full-SS-substition-matrix [def=1.00)")), ToolInput(tag="in_wg", input_type=Boolean(optional=True), prefix="-wg", doc=InputDocumentation(doc="use global sequence weighting for realignment!")), ToolInput(tag="in_gap_b", input_type=Boolean(optional=True), prefix="-gapb", doc=InputDocumentation(doc="[0,inf[  Transition pseudocount admixture (def=1.00)")), ToolInput(tag="in_gap_d", input_type=Boolean(optional=True), prefix="-gapd", doc=InputDocumentation(doc="[0,inf[  Transition pseudocount admixture for open gap (default=0.15)")), ToolInput(tag="in_gape", input_type=Boolean(optional=True), prefix="-gape", doc=InputDocumentation(doc="[0,1.5]  Transition pseudocount admixture for extend gap (def=1.00)")), ToolInput(tag="in_gap_f", input_type=Boolean(optional=True), prefix="-gapf", doc=InputDocumentation(doc="]0,inf]  factor to increase/reduce gap open penalty for deletes (def=0.60)")), ToolInput(tag="in_gap_g", input_type=Boolean(optional=True), prefix="-gapg", doc=InputDocumentation(doc="]0,inf]  factor to increase/reduce gap open penalty for inserts (def=0.60)")), ToolInput(tag="in_gap_h", input_type=Boolean(optional=True), prefix="-gaph", doc=InputDocumentation(doc="]0,inf]  factor to increase/reduce gap extend penalty for deletes(def=0.60)")), ToolInput(tag="in_gap_i", input_type=Boolean(optional=True), prefix="-gapi", doc=InputDocumentation(doc="]0,inf]  factor to increase/reduce gap extend penalty for inserts(def=0.60)")), ToolInput(tag="in_eg_q", input_type=Boolean(optional=True), prefix="-egq", doc=InputDocumentation(doc="[0,inf[  penalty (bits) for end gaps aligned to query residues (def=0.00)")), ToolInput(tag="in_egt", input_type=Boolean(optional=True), prefix="-egt", doc=InputDocumentation(doc="[0,inf[  penalty (bits) for end gaps aligned to template residues (def=0.00)")), ToolInput(tag="in_pc_hhm_con_txt_mode", input_type=Boolean(optional=True), prefix="-pc_hhm_contxt_mode", doc=InputDocumentation(doc="{0,..,3}   position dependence of pc admixture 'tau' (pc mode, default=2)\n0: no pseudo counts:    tau = 0\n1: constant             tau = a\n2: diversity-dependent: tau = a/(1+((Neff[i]-1)/b)^c)\n3: CSBlast admixture:   tau = a(1+b)/(Neff[i]+b)\n(Neff[i]: number of effective seqs in local MSA around column i)")), ToolInput(tag="in_pc_hhm_con_txt_a", input_type=Boolean(optional=True), prefix="-pc_hhm_contxt_a", doc=InputDocumentation(doc="[0,1]        overall pseudocount admixture (def=0.9)")), ToolInput(tag="in_pc_hhm_con_txt_b", input_type=Boolean(optional=True), prefix="-pc_hhm_contxt_b", doc=InputDocumentation(doc="[1,inf[      Neff threshold value for mode 2 (def=4.0)")), ToolInput(tag="in_pc_hhm_con_txt_c", input_type=Boolean(optional=True), prefix="-pc_hhm_contxt_c", doc=InputDocumentation(doc="[0,3]        extinction exponent c for mode 2 (def=1.0)")), ToolInput(tag="in_pc_hhm_no_con_txt_mode", input_type=Boolean(optional=True), prefix="-pc_hhm_nocontxt_mode", doc=InputDocumentation(doc="{0,..,3}   position dependence of pc admixture 'tau' (pc mode, default=2)\n0: no pseudo counts:    tau = 0\n1: constant             tau = a\n2: diversity-dependent: tau = a/(1+((Neff[i]-1)/b)^c)\n(Neff[i]: number of effective seqs in local MSA around column i)")), ToolInput(tag="in_pc_hhm_no_con_txt_a", input_type=Boolean(optional=True), prefix="-pc_hhm_nocontxt_a", doc=InputDocumentation(doc="[0,1]        overall pseudocount admixture (def=1.0)")), ToolInput(tag="in_pc_hhm_no_con_txt_b", input_type=Boolean(optional=True), prefix="-pc_hhm_nocontxt_b", doc=InputDocumentation(doc="[1,inf[      Neff threshold value for mode 2 (def=1.5)")), ToolInput(tag="in_pc_hhm_no_con_txt_c", input_type=Boolean(optional=True), prefix="-pc_hhm_nocontxt_c", doc=InputDocumentation(doc="[0,3]        extinction exponent c for mode 2 (def=1.0)")), ToolInput(tag="in_no_con_txt", input_type=Boolean(optional=True), prefix="-nocontxt", doc=InputDocumentation(doc="use substitution-matrix instead of context-specific pseudocounts")), ToolInput(tag="in_con_txt", input_type=File(optional=True), prefix="-contxt", doc=InputDocumentation(doc="context file for computing context-specific pseudocounts (default=)")), ToolInput(tag="in_csw", input_type=Boolean(optional=True), prefix="-csw", doc=InputDocumentation(doc="[0,inf]  weight of central position in cs pseudocount mode (def=1.6)")), ToolInput(tag="in_csb", input_type=Boolean(optional=True), prefix="-csb", doc=InputDocumentation(doc="[0,1]    weight decay parameter for positions in cs pc mode (def=0.9)")), ToolInput(tag="in_verbose_mode_screen", input_type=Int(optional=True), prefix="-v", doc=InputDocumentation(doc="verbose mode: 0:no screen output  1:only warnings  2: verbose (def=2)")), ToolInput(tag="in_cpu", input_type=Int(optional=True), prefix="-cpu", doc=InputDocumentation(doc="number of CPUs to use (for shared memory SMPs) (default=2)")), ToolInput(tag="in_scores", input_type=File(optional=True), prefix="-scores", doc=InputDocumentation(doc="write scores for all pairwise comparisons to file")), ToolInput(tag="in_a_tab", input_type=Boolean(optional=True), prefix="-atab", doc=InputDocumentation(doc="<file> write all alignments in tabular layout to file")), ToolInput(tag="in_max_seq", input_type=Int(optional=True), prefix="-maxseq", doc=InputDocumentation(doc="max number of input rows (def=65535)")), ToolInput(tag="in_max_res", input_type=Int(optional=True), prefix="-maxres", doc=InputDocumentation(doc="max number of HMM columns (def=20001)")), ToolInput(tag="in_maxmem", input_type=Boolean(optional=True), prefix="-maxmem", doc=InputDocumentation(doc="[1,inf[ limit memory for realignment (in GB) (def=3.0)"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Hhsearch_V0_1_0().translate("wdl", allow_empty_container=True)

