from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import String, File, Directory, Boolean, Float, Int

Chewiesnake_V0_1_0 = CommandToolBuilder(tool="chewiesnake", base_command=["chewiesnake"], inputs=[ToolInput(tag="in_prodigal", input_type=String(optional=True), prefix="--prodigal", doc=InputDocumentation(doc="[--bsr_threshold BSR_THRESHOLD]")), ToolInput(tag="in_sample_list", input_type=File(optional=True), prefix="--sample_list", doc=InputDocumentation(doc="List of samples to analyze, as a two column tsv file\nwith columns sample and assembly. Can be generated\nwith provided script create_sampleSheet,sh")), ToolInput(tag="in_working_directory", input_type=Directory(optional=True), prefix="--working_directory", doc=InputDocumentation(doc="Working directory where results are saved")), ToolInput(tag="in_cond_a_prefix", input_type=File(optional=True), prefix="--condaprefix", doc=InputDocumentation(doc="Path of default conda environment, enables recycling\nbuilt environments, default is in folder conda_env in\nrepository directory.")), ToolInput(tag="in_reads", input_type=Boolean(optional=True), prefix="--reads", doc=InputDocumentation(doc="Input data is reads. Assemble (using shovill) prior to\nallele calling (default is contigs)")), ToolInput(tag="in_scheme", input_type=File(optional=True), prefix="--scheme", doc=InputDocumentation(doc="Path to directory of the cgmlst scheme")), ToolInput(tag="in_bsr_threshold", input_type=Float(optional=True), prefix="--bsr_threshold", doc=InputDocumentation(doc="blast scoring ratio threshold to use , default = 0.6")), ToolInput(tag="in_size_threshold", input_type=Int(optional=True), prefix="--size_threshold", doc=InputDocumentation(doc="size threshold, default at 0.2 means alleles with size\nvariation of +-20 percent will be tagged as ASM/ALM ,\ndefault = 0.2")), ToolInput(tag="in_distance_method", input_type=Int(optional=True), prefix="--distance_method", doc=InputDocumentation(doc="Grapetree distance method; default = 3")), ToolInput(tag="in_clustering_method", input_type=Int(optional=True), prefix="--clustering_method", doc=InputDocumentation(doc="The agglomeration method to be used for hierarchical\nclustering. This should be (an unambiguous\nabbreviation of) one of 'ward.D', 'ward.D2', 'single',\n'complete', 'average' (= UPGMA), 'mcquitty' (= WPGMA),\n'median' (= WPGMC) or 'centroid' (= UPGMC); default =\nsingle")), ToolInput(tag="in_distance_threshold", input_type=Int(optional=True), prefix="--distance_threshold", doc=InputDocumentation(doc="A single distance threshold for the extraction of sub-\ntrees; default = 10")), ToolInput(tag="in_address_range", input_type=Int(optional=True), prefix="--address_range", doc=InputDocumentation(doc="A comma separated set of cutoff values for defining\nthe clustering address (Default:\n1,5,10,20,50,100,200,1000)")), ToolInput(tag="in_report", input_type=Boolean(optional=True), prefix="--report", doc=InputDocumentation(doc="Create html report")), ToolInput(tag="in_comparison", input_type=Boolean(optional=True), prefix="--comparison", doc=InputDocumentation(doc="Compare these results to pre-computed allele database")), ToolInput(tag="in_comparison_db", input_type=File(optional=True), prefix="--comparison_db", doc=InputDocumentation(doc="Path to pre-computed allele database for comparison")), ToolInput(tag="in_joining_threshold", input_type=Int(optional=True), prefix="--joining_threshold", doc=InputDocumentation(doc="A distance threshold for joining data with\ncomparsion_db; default = 30")), ToolInput(tag="in_remove_frame_shifts", input_type=Boolean(optional=True), prefix="--remove_frameshifts", doc=InputDocumentation(doc="remove frameshift alleles by deviating allele length")), ToolInput(tag="in_allele_length_threshold", input_type=Int(optional=True), prefix="--allele_length_threshold", doc=InputDocumentation(doc="Maximum tolerated allele length deviance compared to\nmedian allele length of locus; choose integer for\n'absolute frameshift mode and float (0..1) for\n'relative' frameshift mode ; default=0.1")), ToolInput(tag="in_frameshift_mode", input_type=Int(optional=True), prefix="--frameshift_mode", doc=InputDocumentation(doc="Whether to consider absolute or relative differences\nof allele length for frameshifts removal. Choose from\n'absolute' and 'relative', default='relative'")), ToolInput(tag="in_min_trimmed_length", input_type=Int(optional=True), prefix="--min_trimmed_length", doc=InputDocumentation(doc="Minimum length of a read to keep, default = 15")), ToolInput(tag="in_assembler", input_type=String(optional=True), prefix="--assembler", doc=InputDocumentation(doc="Assembler to use in shovill, choose from megahit\nvelvet skesa spades (default: spades)")), ToolInput(tag="in_shov_ill_output_options", input_type=String(optional=True), prefix="--shovill_output_options", doc=InputDocumentation(doc="Extra output options for shovill (default: '')")), ToolInput(tag="in_shov_ill_extra_opts", input_type=String(optional=True), prefix="--shovill_extraopts", doc=InputDocumentation(doc="Extra options for shovill (default: '')")), ToolInput(tag="in_shov_ill_modules", input_type=String(optional=True), prefix="--shovill_modules", doc=InputDocumentation(doc="Module options for shovill, choose from --noreadcorr\n--trim --nostitch --nocorr --noreadcorr (default: '--\nnoreadcorr')")), ToolInput(tag="in_shov_ill_depth", input_type=Int(optional=True), prefix="--shovill_depth", doc=InputDocumentation(doc="Sub-sample --R1/--R2 to this depth. Disable with\n--depth 0 (default: 100)")), ToolInput(tag="in_shov_ill_tmpdir", input_type=Directory(optional=True), prefix="--shovill_tmpdir", doc=InputDocumentation(doc="Fast temporary directory (default: '')")), ToolInput(tag="in_use_cond_a", input_type=Boolean(optional=True), prefix="--use_conda", doc=InputDocumentation(doc="Utilize '--useconda' option, i.e. all software\ndependencies are available in a single env")), ToolInput(tag="in_cond_a_front_end", input_type=Boolean(optional=True), prefix="--conda_frontend", doc=InputDocumentation(doc="Do not mamba but conda as frontend to create\nindividual module' software environments")), ToolInput(tag="in_threads_sample", input_type=Int(optional=True), prefix="--threads_sample", doc=InputDocumentation(doc="Number of Threads to use per sample, default = 10")), ToolInput(tag="in_threads", input_type=Int(optional=True), prefix="--threads", doc=InputDocumentation(doc="Number of Threads to use. Note that samples can only\nbe processed sequentially due to the required database\naccess. However the allele calling can be executed in\nparallel for the different loci, default = 10")), ToolInput(tag="in_dry_run", input_type=Boolean(optional=True), prefix="--dryrun", doc=InputDocumentation(doc="Snakemake dryrun. Only calculate graph without\nexecuting anything")), ToolInput(tag="in_force_all", input_type=Boolean(optional=True), prefix="--forceall", doc=InputDocumentation(doc="Snakemake force. Force recalculation of all steps")), ToolInput(tag="in_unlock", input_type=Boolean(optional=True), prefix="--unlock", doc=InputDocumentation(doc="unlock snakemake")), ToolInput(tag="in_logdir", input_type=File(optional=True), prefix="--logdir", doc=InputDocumentation(doc="Path to directory whete .snakemake output is to be\nsaved\n"))], outputs=[ToolOutput(tag="out_logdir", output_type=File(optional=True), selector=InputSelector(input_to_select="in_logdir", type_hint=File()), doc=OutputDocumentation(doc="Path to directory whete .snakemake output is to be\nsaved\n"))], container="quay.io/biocontainers/chewiesnake:3.0.0--1", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Chewiesnake_V0_1_0().translate("wdl")

