from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, File, Int

Feelnc_Filter_Pl_V0_1_0 = CommandToolBuilder(tool="FEELnc_filter.pl", base_command=["FEELnc_filter.pl"], inputs=[ToolInput(tag="in_man", input_type=Boolean(optional=True), prefix="--man", doc=InputDocumentation(doc="Open man page")), ToolInput(tag="in_verbosity", input_type=Boolean(optional=True), prefix="--verbosity", doc=InputDocumentation(doc="Level of verbosity 0, 1 and 2 [default 1]")), ToolInput(tag="in_in_file", input_type=File(optional=True), prefix="--infile", doc=InputDocumentation(doc="Specify the GTF file to be filtered (such as a cufflinks transcripts/merged .GTF file)")), ToolInput(tag="in_mrna_file", input_type=File(optional=True), prefix="--mRNAfile", doc=InputDocumentation(doc="Specify the annotation GTF file to be filtered on based on sense exon overlap (file of protein coding annotation)")), ToolInput(tag="in_size", input_type=Int(optional=True), prefix="--size", doc=InputDocumentation(doc="Keep transcript with a minimal size (default 200)")), ToolInput(tag="in_biotype", input_type=Boolean(optional=True), prefix="--biotype", doc=InputDocumentation(doc="Only consider transcript(s) from the reference annotation having this(these) biotype(s) (e.g : -b transcript_biotype=protein_coding,pseudogene) [default undef i.e all transcripts]")), ToolInput(tag="in_linc_only", input_type=Boolean(optional=True), prefix="--linconly", doc=InputDocumentation(doc="Keep only long intergenic/interveaning ncRNAs [default FALSE]")), ToolInput(tag="in_mono_ex", input_type=Boolean(optional=True), prefix="--monoex", doc=InputDocumentation(doc="=-1|0|1             Keep monoexonic transcript(s): mode to be selected from : -1 keep monoexonic antisense (for RNASeq stranded protocol), 1 keep all monoexonic, 0 remove all monoexonic   [default 0]")), ToolInput(tag="in_biex", input_type=Int(optional=True), prefix="--biex", doc=InputDocumentation(doc="Discard biexonic transcripts having one exon size lower to this value (default 25)")), ToolInput(tag="in_min_frac_over", input_type=Int(optional=True), prefix="--minfrac_over", doc=InputDocumentation(doc="Minimal fraction out of the candidate lncRNA size to be considered for overlap [default 0 i.e 1nt]")), ToolInput(tag="in_proc", input_type=Int(optional=True), prefix="--proc", doc=InputDocumentation(doc="Number of thread for computing overlap [default 4]")), ToolInput(tag="in_out_log", input_type=File(optional=True), prefix="--outlog", doc=InputDocumentation(doc="Specify the log file of output which [default infile.log]"))], outputs=[ToolOutput(tag="out_out_log", output_type=File(optional=True), selector=InputSelector(input_to_select="in_out_log", type_hint=File()), doc=OutputDocumentation(doc="Specify the log file of output which [default infile.log]"))], container="quay.io/biocontainers/feelnc:0.2--pl526_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Feelnc_Filter_Pl_V0_1_0().translate("wdl")

