version 1.0

task Metaplots {
  input {
    Directory? an_not_dir
    File? rpf_mapping_file
    File? input_file
    String? stranded
    Int? minimum_length
    Int? maximum_length
    Int? p_value_one_cut_off
    Int? p_value_two_cut_off
    Int? frame_zero_percent
    File? out_name
  }
  command <<<
    metaplots \
      ~{if defined(an_not_dir) then ("--annot_dir " +  '"' + an_not_dir + '"') else ""} \
      ~{if defined(rpf_mapping_file) then ("--rpf_mapping_file " +  '"' + rpf_mapping_file + '"') else ""} \
      ~{if defined(input_file) then ("--input_file " +  '"' + input_file + '"') else ""} \
      ~{if defined(stranded) then ("--stranded " +  '"' + stranded + '"') else ""} \
      ~{if defined(minimum_length) then ("--minimum-length " +  '"' + minimum_length + '"') else ""} \
      ~{if defined(maximum_length) then ("--maximum-length " +  '"' + maximum_length + '"') else ""} \
      ~{if defined(p_value_one_cut_off) then ("--pvalue1_cutoff " +  '"' + p_value_one_cut_off + '"') else ""} \
      ~{if defined(p_value_two_cut_off) then ("--pvalue2_cutoff " +  '"' + p_value_two_cut_off + '"') else ""} \
      ~{if defined(frame_zero_percent) then ("--frame0_percent " +  '"' + frame_zero_percent + '"') else ""} \
      ~{if defined(out_name) then ("--outname " +  '"' + out_name + '"') else ""}
  >>>
  runtime {
    docker: "None"
  }
  parameter_meta {
    an_not_dir: "transcripts annotation directory, generated by\\nprepare_transcripts."
    rpf_mapping_file: "ribo-seq BAM/SAM file aligned to the transcriptome."
    input_file: "the file list the ribo-seq BAM/SAM files aligned to\\nthe transcriptome."
    stranded: "whether the data is strand-specific, reverse means\\nreversed strand interpretation.(default: yes)"
    minimum_length: "minimum length of read to output, default 24"
    maximum_length: "maximum length of read to output, default 36"
    p_value_one_cut_off: "pvalue cutoff of frame0 > frame2 for automatically\\npredicting P-site location, default 0.001"
    p_value_two_cut_off: "pvalue cutoff of frame0 > frame2 for automatically\\npredicting P-site location, default 0.001"
    frame_zero_percent: "proportion threshold of the number of reads in frame0,\\ndefined by f0/(f0+f1+f2), default 0.65"
    out_name: "name of output pdf file(default: metaplots)"
  }
  output {
    File out_stdout = stdout()
    File out_out_name = "${in_out_name}"
  }
}