from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Array, String, File, Directory, Boolean, Int, Float

Rnaquast_Py_V0_1_0 = CommandToolBuilder(tool="rnaQUAST.py", base_command=["rnaQUAST.py"], inputs=[ToolInput(tag="in_reference", input_type=Array(t=String(), optional=True), prefix="--reference", doc=InputDocumentation(doc="Single file (or several files for meta RNA) with\nreference genome in FASTA format or *.txt file with\none-per-line list of FASTA files with reference\nsequences")), ToolInput(tag="in_gtf", input_type=Array(t=String(), optional=True), prefix="--gtf", doc=InputDocumentation(doc="File with gene coordinates (or several files or *.txt\nfile with one-per-line list of GTF / GFF files for\nmeta RNA). We recommend to use files downloaded from\nGENCODE or Ensembl [GTF/GFF]")), ToolInput(tag="in_gene_db", input_type=File(optional=True), prefix="--gene_db", doc=InputDocumentation(doc="Path to the gene database generated by gffutils to be")), ToolInput(tag="in_alignment", input_type=Array(t=String(), optional=True), prefix="--alignment", doc=InputDocumentation(doc="File(s) with transcript alignments to the reference\ngenome [PSL]")), ToolInput(tag="in_reads_alignment", input_type=File(optional=True), prefix="--reads_alignment", doc=InputDocumentation(doc="File with read alignments to the reference genome\n[SAM]")), ToolInput(tag="in_left_reads", input_type=File(optional=True), prefix="--left_reads", doc=InputDocumentation(doc="File with forward paired-end reads [FASTQ or gzip-\ncompressed]")), ToolInput(tag="in_right_reads", input_type=File(optional=True), prefix="--right_reads", doc=InputDocumentation(doc="File with reverse paired-end reads [FASTQ or gzip-\ncompressed]")), ToolInput(tag="in_single_reads", input_type=File(optional=True), prefix="--single_reads", doc=InputDocumentation(doc="File with unpaired reads [FASTQ or gzip-compressed]")), ToolInput(tag="in_gmap_index", input_type=Directory(optional=True), prefix="--gmap_index", doc=InputDocumentation(doc="Folder containing GMAP index for the reference genome")), ToolInput(tag="in_output_dir", input_type=Directory(optional=True), prefix="--output_dir", doc=InputDocumentation(doc="Directory to store all results [default:\nrnaQUAST_results/results_<datetime>]")), ToolInput(tag="in_test", input_type=Directory(optional=True), prefix="--test", doc=InputDocumentation(doc="Run rnaQUAST on the test data from the test_data\nfolder, output directory is rnaOUAST_test_output")), ToolInput(tag="in_debug", input_type=Boolean(optional=True), prefix="--debug", doc=InputDocumentation(doc="Report detailed information, typically used only for\ndetecting problems.")), ToolInput(tag="in_threads", input_type=Int(optional=True), prefix="--threads", doc=InputDocumentation(doc="Maximum number of threads, default: min(number of CPUs\n/ 2, 16)")), ToolInput(tag="in_labels", input_type=Array(t=String(), optional=True), prefix="--labels", doc=InputDocumentation(doc="Name(s) of assemblies that will be used in the reports")), ToolInput(tag="in_strand_specific", input_type=Boolean(optional=True), prefix="--strand_specific", doc=InputDocumentation(doc="Set if transcripts were assembled using strand-\nspecific RNA-Seq data")), ToolInput(tag="in_min_alignment", input_type=Int(optional=True), prefix="--min_alignment", doc=InputDocumentation(doc="Minimal alignment length, default: 50")), ToolInput(tag="in_no_plots", input_type=Boolean(optional=True), prefix="--no_plots", doc=InputDocumentation(doc="Do not draw plots (to speed up computation)")), ToolInput(tag="in_blat", input_type=Boolean(optional=True), prefix="--blat", doc=InputDocumentation(doc="Run with BLAT alignment tool\n(http://hgwdev.cse.ucsc.edu/~kent/exe/) instead of\nGMAP")), ToolInput(tag="in_gene_mark", input_type=Boolean(optional=True), prefix="--gene_mark", doc=InputDocumentation(doc="Run with GeneMarkS-T tool\n(http://topaz.gatech.edu/GeneMark/)")), ToolInput(tag="in_meta", input_type=Boolean(optional=True), prefix="--meta", doc=InputDocumentation(doc="Run QUALITY ASSESSMENT FOR METATRANSCRIPTOME")), ToolInput(tag="in_upper_threshold", input_type=Float(optional=True), prefix="--upper_threshold", doc=InputDocumentation(doc="Upper threshold for x-assembled/covered/matched\nmetrics, default: 0.95")), ToolInput(tag="in_prokaryote", input_type=Boolean(optional=True), prefix="--prokaryote", doc=InputDocumentation(doc="Use this option if the genome is prokaryotic")), ToolInput(tag="in_disable_infer_genes", input_type=Boolean(optional=True), prefix="--disable_infer_genes", doc=InputDocumentation(doc="Use this option if your GTF file already contains\ngenes records")), ToolInput(tag="in_disable_infer_transcripts", input_type=Boolean(optional=True), prefix="--disable_infer_transcripts", doc=InputDocumentation(doc="Use this option if your GTF already contains\ntranscripts records")), ToolInput(tag="in_busco", input_type=File(optional=True), prefix="--busco", doc=InputDocumentation(doc="Run with BUSCO tool (http://busco.ezlab.org/).\nAutomated lineage selection [auto-lineage], BUSCO\ndataset name or a path to the lineage data to be used")), ToolInput(tag="in_used", input_type=String(), position=0, doc=InputDocumentation(doc="-c TRANSCRIPTS [TRANSCRIPTS ...], --transcripts TRANSCRIPTS [TRANSCRIPTS ...]")), ToolInput(tag="in_assemblies", input_type=String(), position=0, doc=InputDocumentation(doc="--lower_threshold LOWER_THRESHOLD"))], outputs=[ToolOutput(tag="out_test", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_test", type_hint=File()), doc=OutputDocumentation(doc="Run rnaQUAST on the test data from the test_data\nfolder, output directory is rnaOUAST_test_output"))], container="quay.io/biocontainers/rnaquast:2.2.0--0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Rnaquast_Py_V0_1_0().translate("wdl")

