from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Int, Boolean, File, Float

Mmseqs_Linsearch_V0_1_0 = CommandToolBuilder(tool="mmseqs_linsearch", base_command=["mmseqs", "linsearch"], inputs=[ToolInput(tag="in_comp_bias_corr", input_type=Int(optional=True), prefix="--comp-bias-corr", doc=InputDocumentation(doc="Correct for locally biased amino acid composition (range 0-1) [1]")), ToolInput(tag="in_add_self_matches", input_type=Boolean(optional=True), prefix="--add-self-matches", doc=InputDocumentation(doc="Artificially add entries of queries with themselves (for clustering) [0]")), ToolInput(tag="in_seed_sub_mat", input_type=File(optional=True), prefix="--seed-sub-mat", doc=InputDocumentation(doc="Substitution matrix file for k-mer generation [nucl:nucleotide.out,aa:blosum62.out]")), ToolInput(tag="in_mask", input_type=Int(optional=True), prefix="--mask", doc=InputDocumentation(doc="Mask sequences in k-mer stage: 0: w/o low complexity masking, 1: with low complexity masking [0]")), ToolInput(tag="in_mask_lower_case", input_type=Int(optional=True), prefix="--mask-lower-case", doc=InputDocumentation(doc="Lowercase letters will be excluded from k-mer search 0: include region, 1: exclude region [0]")), ToolInput(tag="in_split_memory_limit", input_type=Int(optional=True), prefix="--split-memory-limit", doc=InputDocumentation(doc="Set max memory per split. E.g. 800B, 5K, 10M, 1G. Default (0) to all available system memory [0]")), ToolInput(tag="in_add_string_convert", input_type=Boolean(optional=True), prefix="-a", doc=InputDocumentation(doc="Add backtrace string (convert to alignments with mmseqs convertalis module) [0]")), ToolInput(tag="in_alignment_mode", input_type=Int(optional=True), prefix="--alignment-mode", doc=InputDocumentation(doc="How to compute the alignment:\n0: automatic\n1: only score and end_pos\n2: also start_pos and cov\n3: also seq.id\n4: only ungapped alignment [2]")), ToolInput(tag="in_alignment_output_mode", input_type=Int(optional=True), prefix="--alignment-output-mode", doc=InputDocumentation(doc="How to compute the alignment:\n0: automatic\n1: only score and end_pos\n2: also start_pos and cov\n3: also seq.id\n4: only ungapped alignment\n5: score only (output) cluster format [0]")), ToolInput(tag="in_wrapped_scoring", input_type=Boolean(optional=True), prefix="--wrapped-scoring", doc=InputDocumentation(doc="Double the (nucleotide) query sequence during the scoring process to allow wrapped diagonal scoring around end and start [0]")), ToolInput(tag="in_list_matches_evalue", input_type=Float(optional=True), prefix="-e", doc=InputDocumentation(doc="List matches below this E-value (range 0.0-inf) [1.000E-03]")), ToolInput(tag="in_min_seq_id", input_type=Float(optional=True), prefix="--min-seq-id", doc=InputDocumentation(doc="List matches above this sequence identity (for clustering) (range 0.0-1.0) [0.000]")), ToolInput(tag="in_min_aln_len", input_type=Int(optional=True), prefix="--min-aln-len", doc=InputDocumentation(doc="Minimum alignment length (range 0-INT_MAX) [0]")), ToolInput(tag="in_seq_id_mode", input_type=Int(optional=True), prefix="--seq-id-mode", doc=InputDocumentation(doc="0: alignment length 1: shorter, 2: longer sequence [0]")), ToolInput(tag="in_alt_ali", input_type=Int(optional=True), prefix="--alt-ali", doc=InputDocumentation(doc="Show up to this many alternative alignments [0]")), ToolInput(tag="in_list_matches_fraction", input_type=Float(optional=True), prefix="-c", doc=InputDocumentation(doc="List matches above this fraction of aligned (covered) residues (see --cov-mode) [0.000]")), ToolInput(tag="in_cov_mode", input_type=Int(optional=True), prefix="--cov-mode", doc=InputDocumentation(doc="0: coverage of query and target\n1: coverage of target\n2: coverage of query\n3: target seq. length has to be at least x% of query length\n4: query seq. length has to be at least x% of target length\n5: short seq. needs to be at least x% of the other seq. length [0]")), ToolInput(tag="in_max_rejected", input_type=Int(optional=True), prefix="--max-rejected", doc=InputDocumentation(doc="Maximum rejected alignments before alignment calculation for a query is stopped [2147483647]")), ToolInput(tag="in_max_accept", input_type=Int(optional=True), prefix="--max-accept", doc=InputDocumentation(doc="Maximum accepted alignments before alignment calculation for a query is stopped [2147483647]")), ToolInput(tag="in_score_bias", input_type=Float(optional=True), prefix="--score-bias", doc=InputDocumentation(doc="Score bias when computing SW alignment (in bits) [0.000]")), ToolInput(tag="in_realign", input_type=Boolean(optional=True), prefix="--realign", doc=InputDocumentation(doc="Compute more conservative, shorter alignments (scores and E-values not changed) [0]")), ToolInput(tag="in_realign_score_bias", input_type=Float(optional=True), prefix="--realign-score-bias", doc=InputDocumentation(doc="Additional bias when computing realignment [-0.200]")), ToolInput(tag="in_realign_max_seqs", input_type=Int(optional=True), prefix="--realign-max-seqs", doc=InputDocumentation(doc="Maximum number of results to return in realignment [2147483647]")), ToolInput(tag="in_gap_open", input_type=Int(optional=True), prefix="--gap-open", doc=InputDocumentation(doc="Gap open cost [nucl:5,aa:11]")), ToolInput(tag="in_gap_extend", input_type=Int(optional=True), prefix="--gap-extend", doc=InputDocumentation(doc="Gap extension cost [nucl:2,aa:1]")), ToolInput(tag="in_z_drop", input_type=Int(optional=True), prefix="--zdrop", doc=InputDocumentation(doc="Maximal allowed difference between score values before alignment is truncated  (nucleotide alignment only) [40]")), ToolInput(tag="in_km_er_per_seq", input_type=Int(optional=True), prefix="--kmer-per-seq", doc=InputDocumentation(doc="k-mers per sequence [21]")), ToolInput(tag="in_km_er_per_seq_scale", input_type=Int(optional=True), prefix="--kmer-per-seq-scale", doc=InputDocumentation(doc="Scale k-mer per sequence based on sequence length as kmer-per-seq val + scale x seqlen [nucl:0.200,aa:0.000]")), ToolInput(tag="in_pick_n_sim_km_er", input_type=Int(optional=True), prefix="--pick-n-sim-kmer", doc=InputDocumentation(doc="Add N similar k-mers to search [1]")), ToolInput(tag="in_result_direction", input_type=Int(optional=True), prefix="--result-direction", doc=InputDocumentation(doc="result is 0: query, 1: target centric [1]")), ToolInput(tag="in_pc_a", input_type=Float(optional=True), prefix="--pca", doc=InputDocumentation(doc="Pseudo count admixture strength [1.000]")), ToolInput(tag="in_pcb", input_type=Float(optional=True), prefix="--pcb", doc=InputDocumentation(doc="Pseudo counts: Neff at half of maximum admixture (range 0.0-inf) [1.500]")), ToolInput(tag="in_min_length", input_type=Int(optional=True), prefix="--min-length", doc=InputDocumentation(doc="Minimum codon number in open reading frames [30]")), ToolInput(tag="in_max_length", input_type=Int(optional=True), prefix="--max-length", doc=InputDocumentation(doc="Maximum codon number in open reading frames [32734]")), ToolInput(tag="in_max_gaps", input_type=Int(optional=True), prefix="--max-gaps", doc=InputDocumentation(doc="Maximum number of codons with gaps or unknown residues before an open reading frame is rejected [2147483647]")), ToolInput(tag="in_contig_start_mode", input_type=Int(optional=True), prefix="--contig-start-mode", doc=InputDocumentation(doc="Contig start can be 0: incomplete, 1: complete, 2: both [2]")), ToolInput(tag="in_contig_end_mode", input_type=Int(optional=True), prefix="--contig-end-mode", doc=InputDocumentation(doc="Contig end can be 0: incomplete, 1: complete, 2: both [2]")), ToolInput(tag="in_orf_start_mode", input_type=Int(optional=True), prefix="--orf-start-mode", doc=InputDocumentation(doc="Orf fragment can be 0: from start to stop, 1: from any to stop, 2: from last encountered start to stop (no start in the middle) [1]")), ToolInput(tag="in_forward_frames", input_type=Int(optional=True), prefix="--forward-frames", doc=InputDocumentation(doc="Comma-separated list of frames on the forward strand to be extracted [1,2,3]")), ToolInput(tag="in_reverse_frames", input_type=Int(optional=True), prefix="--reverse-frames", doc=InputDocumentation(doc="Comma-separated list of frames on the reverse strand to be extracted [1,2,3]")), ToolInput(tag="in_translation_table", input_type=Int(optional=True), prefix="--translation-table", doc=InputDocumentation(doc="1) CANONICAL, 2) VERT_MITOCHONDRIAL, 3) YEAST_MITOCHONDRIAL, 4) MOLD_MITOCHONDRIAL, 5) INVERT_MITOCHONDRIAL, 6) CILIATE\n9) FLATWORM_MITOCHONDRIAL, 10) EUPLOTID, 11) PROKARYOTE, 12) ALT_YEAST, 13) ASCIDIAN_MITOCHONDRIAL, 14) ALT_FLATWORM_MITOCHONDRIAL\n15) BLEPHARISMA, 16) CHLOROPHYCEAN_MITOCHONDRIAL, 21) TREMATODE_MITOCHONDRIAL, 22) SCENEDESMUS_MITOCHONDRIAL\n23) THRAUSTOCHYTRIUM_MITOCHONDRIAL, 24) PTEROBRANCHIA_MITOCHONDRIAL, 25) GRACILIBACTERIA, 26) PACHYSOLEN, 27) KARYORELICT, 28) CONDYLOSTOMA\n29) MESODINIUM, 30) PERTRICH, 31) BLASTOCRITHIDIA [1]")), ToolInput(tag="in_translate", input_type=Int(optional=True), prefix="--translate", doc=InputDocumentation(doc="Translate ORF to amino acid [0]")), ToolInput(tag="in_use_all_table_starts", input_type=Boolean(optional=True), prefix="--use-all-table-starts", doc=InputDocumentation(doc="Use all alternatives for a start codon in the genetic table, if false - only ATG (AUG) [0]")), ToolInput(tag="in_id_offset", input_type=Int(optional=True), prefix="--id-offset", doc=InputDocumentation(doc="Numeric ids in index file are offset by this value [0]")), ToolInput(tag="in_add_orf_stop", input_type=Boolean(optional=True), prefix="--add-orf-stop", doc=InputDocumentation(doc="Add stop codon '*' at complete start and end [0]")), ToolInput(tag="in_search_type", input_type=Int(optional=True), prefix="--search-type", doc=InputDocumentation(doc="Search type 0: auto 1: amino acid, 2: translated, 3: nucleotide, 4: translated nucleotide alignment [0]")), ToolInput(tag="in_sub_mat", input_type=File(optional=True), prefix="--sub-mat", doc=InputDocumentation(doc="Substitution matrix file [nucl:nucleotide.out,aa:blosum62.out]")), ToolInput(tag="in_max_seq_len", input_type=Int(optional=True), prefix="--max-seq-len", doc=InputDocumentation(doc="Maximum sequence length [65535]")), ToolInput(tag="in_db_load_mode", input_type=Int(optional=True), prefix="--db-load-mode", doc=InputDocumentation(doc="Database preload mode 0: auto, 1: fread, 2: mmap, 3: mmap+touch [0]")), ToolInput(tag="in_threads", input_type=Int(optional=True), prefix="--threads", doc=InputDocumentation(doc="Number of CPU-cores used (all by default) [2]")), ToolInput(tag="in_compressed", input_type=Int(optional=True), prefix="--compressed", doc=InputDocumentation(doc="Write compressed output [0]")), ToolInput(tag="in_verbosity_level_errors", input_type=Int(optional=True), prefix="-v", doc=InputDocumentation(doc="Verbosity level: 0: quiet, 1: +errors, 2: +warnings, 3: +info [3]")), ToolInput(tag="in_mpi_runner", input_type=Int(optional=True), prefix="--mpi-runner", doc=InputDocumentation(doc="Use MPI on compute cluster with this MPI command (e.g. 'mpirun -np 42') []")), ToolInput(tag="in_force_reuse", input_type=Boolean(optional=True), prefix="--force-reuse", doc=InputDocumentation(doc="Reuse tmp filse in tmp/latest folder ignoring parameters and version changes [0]")), ToolInput(tag="in_remove_tmp_files", input_type=Boolean(optional=True), prefix="--remove-tmp-files", doc=InputDocumentation(doc="Delete temporary files [0]")), ToolInput(tag="in_create_lookup", input_type=Int(optional=True), prefix="--create-lookup", doc=InputDocumentation(doc="Create database lookup file (can be very large) [0]")), ToolInput(tag="in_chain_alignments", input_type=Int(optional=True), prefix="--chain-alignments", doc=InputDocumentation(doc="Chain overlapping alignments [0]")), ToolInput(tag="in_merge_query", input_type=Int(optional=True), prefix="--merge-query", doc=InputDocumentation(doc="Combine ORFs/split sequences to a single entry [1]"))], outputs=[ToolOutput(tag="out_seed_sub_mat", output_type=File(optional=True), selector=InputSelector(input_to_select="in_seed_sub_mat", type_hint=File()), doc=OutputDocumentation(doc="Substitution matrix file for k-mer generation [nucl:nucleotide.out,aa:blosum62.out]")), ToolOutput(tag="out_sub_mat", output_type=File(optional=True), selector=InputSelector(input_to_select="in_sub_mat", type_hint=File()), doc=OutputDocumentation(doc="Substitution matrix file [nucl:nucleotide.out,aa:blosum62.out]"))], container="quay.io/biocontainers/mmseqs2:13.45111--h2d02072_0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Mmseqs_Linsearch_V0_1_0().translate("wdl")

