from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Int, Boolean, Float, String

Sequnwinder_V0_1_0 = CommandToolBuilder(tool="sequnwinder", base_command=["sequnwinder"], inputs=[ToolInput(tag="in_out", input_type=File(optional=True), prefix="--out", doc=InputDocumentation(doc=": Ouput file prefix. All output will be put into a directory with the prefix name")), ToolInput(tag="in_threads", input_type=Int(optional=True), prefix="--threads", doc=InputDocumentation(doc=": Use n threads to train SeqUnwinder model. Default is 5 threads")), ToolInput(tag="in_debug", input_type=Boolean(optional=True), prefix="--debug", doc=InputDocumentation(doc=": Flag to run in debug mode; prints extra output")), ToolInput(tag="in_meme_path", input_type=File(optional=True), prefix="--memepath", doc=InputDocumentation(doc=": path to the meme bin dir (default: meme is in $PATH)")), ToolInput(tag="in_geninfo", input_type=File(optional=True), prefix="--geninfo", doc=InputDocumentation(doc="This file should list the lengths of all chromosomes on separate lines using the format chrName<tab>chrLength")), ToolInput(tag="in_genre_gs", input_type=Boolean(optional=True), prefix="--genregs", doc=InputDocumentation(doc="<List of TF binding sites with annotations; eg: chr1:151736000  Shared;Proximal>")), ToolInput(tag="in_win", input_type=Int(optional=True), prefix="--win", doc=InputDocumentation(doc=": Size of the genomic regions in bp. Default = 150.")), ToolInput(tag="in_maker_and_regs", input_type=Boolean(optional=True), prefix="--makerandregs", doc=InputDocumentation(doc=": Flag to make random genomic regions as an extra outgroup class in classification (Only applicable when genome is provide.)")), ToolInput(tag="in_mink", input_type=Int(optional=True), prefix="--mink", doc=InputDocumentation(doc=": Minimum length of k-mer (default = 4)")), ToolInput(tag="in_max_k", input_type=Int(optional=True), prefix="--maxk", doc=InputDocumentation(doc=": Maximum length of k-mer (default = 5)")), ToolInput(tag="in_regularization_constant_default", input_type=Int(optional=True), prefix="--r", doc=InputDocumentation(doc=": Regularization constant (default = 10)")), ToolInput(tag="in_number_folds_cross", input_type=Int(optional=True), prefix="--x", doc=InputDocumentation(doc=": Number of folds for cross validation, default = 3.")), ToolInput(tag="in_merge_low", input_type=Boolean(optional=True), prefix="--mergelow", doc=InputDocumentation(doc=": Flag to merge subclasses with less than 200 sites with other relevant classes. By default, all subclasses with less that 200 sites are removed.")), ToolInput(tag="in_mins_can_len", input_type=Int(optional=True), prefix="--minscanlen", doc=InputDocumentation(doc=": Minimum length of the window to scan K-mer models. Default=8.")), ToolInput(tag="in_max_scan_len", input_type=Int(optional=True), prefix="--maxscanlen", doc=InputDocumentation(doc=": Maximum length of the window to scan K-mer models. Default=14.")), ToolInput(tag="in_hills_thresh", input_type=Float(optional=True), prefix="--hillsthresh", doc=InputDocumentation(doc=": Scoring threshold to identify hills. Default=0.1.")), ToolInput(tag="in_meme_min_w", input_type=Int(optional=True), prefix="--mememinw", doc=InputDocumentation(doc=": minw arg for MEME. Default=6.")), ToolInput(tag="in_meme_max_w", input_type=Int(optional=True), prefix="--mememaxw", doc=InputDocumentation(doc=": maxw arg for MEME. Default=13. This value should always be less than 'maxscanlen'.")), ToolInput(tag="in_me_men_motifs", input_type=Int(optional=True), prefix="--memenmotifs", doc=InputDocumentation(doc=": Number of motifs MEME should find in each condition (default=3)")), ToolInput(tag="in_meme_args", input_type=String(optional=True), prefix="--memeargs", doc=InputDocumentation(doc=": Additional args for MEME (default:  -dna -mod zoops -revcomp -nostatus)")), ToolInput(tag="in_meme_search_win", input_type=Int(optional=True), prefix="--memesearchwin", doc=InputDocumentation(doc=": Window around hills to search for discriminative motifs. Default=16. (Only applicable when run with 'genregs').")), ToolInput(tag="in_motif_min_roc", input_type=Int(optional=True), prefix="--motifminROC", doc=InputDocumentation(doc=": minimum class-specific ROC required to report motif. Default=0.7.        --a <int>: Maximum number of allowed ADMM iterations. Default=400.")), ToolInput(tag="in_and", input_type=String(), position=0, doc=InputDocumentation(doc="--seq <path>: A directory containing fasta format files corresponding to every named chromosome is required")), ToolInput(tag="in_or", input_type=String(), position=0, doc=InputDocumentation(doc="--genseqs <DNA sequences around at TF binding sites; eg: ATGC...TGC     Shared;Proximal>"))], outputs=[ToolOutput(tag="out_out", output_type=File(optional=True), selector=InputSelector(input_to_select="in_out", type_hint=File()), doc=OutputDocumentation(doc=": Ouput file prefix. All output will be put into a directory with the prefix name"))], container="quay.io/biocontainers/sequnwinder:0.1.4--0", version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Sequnwinder_V0_1_0().translate("wdl")

