from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Directory, Int, String

Jass_Preprocessing_V0_1_0 = CommandToolBuilder(tool="jass_preprocessing", base_command=["jass_preprocessing"], inputs=[ToolInput(tag="in_g_was_info", input_type=File(optional=True), prefix="--gwas-info", doc=InputDocumentation(doc="Path to the file describing the format of the\nindividual GWASs files with correct header")), ToolInput(tag="in_ref_path", input_type=File(optional=True), prefix="--ref-path", doc=InputDocumentation(doc="reference panel location (used to determine which snp\nto impute)")), ToolInput(tag="in_input_folder", input_type=Directory(optional=True), prefix="--input-folder", doc=InputDocumentation(doc="Path to the folder containing the Raw GWASs summary\nstatistic files, must end by '/'")), ToolInput(tag="in_diagnostic_folder", input_type=Directory(optional=True), prefix="--diagnostic-folder", doc=InputDocumentation(doc="Path to the reporting information on the PreProcessing\nsuch as the SNPs sample size distribution")), ToolInput(tag="in_output_folder", input_type=Directory(optional=True), prefix="--output-folder", doc=InputDocumentation(doc="Location of main ouput folder for preprocessed GWAS\nfiles (splitted by chromosome)")), ToolInput(tag="in_output_folder_one_file", input_type=File(optional=True), prefix="--output-folder-1-file", doc=InputDocumentation(doc="optional location to store the preprocessing in one\ntabular file with one chromosome columns (useful to\ncompute LDSC correlation for instance)")), ToolInput(tag="in_percent_sample_size", input_type=Int(optional=True), prefix="--percent-sample-size", doc=InputDocumentation(doc="the proportion (between 0 and 1) of the 90th\npercentile of the sample size used to filter the SNPs")), ToolInput(tag="in_minimum_maf", input_type=String(optional=True), prefix="--minimum-MAF", doc=InputDocumentation(doc="Filter the reference panel by minimum allele frequency")), ToolInput(tag="in_mask_mhc", input_type=String(optional=True), prefix="--mask-MHC", doc=InputDocumentation(doc="Whether the MHC region should be masked or not.\ndefault is False")), ToolInput(tag="in_additional_masked_region", input_type=Int(optional=True), prefix="--additional-masked-region", doc=InputDocumentation(doc="List of dictionary containing coordinate of region to\nmask. For example :[{'chr':6, 'start':50000000, 'end':\n70000000}, {'chr':6, 'start':100000000, 'end':\n120000000}]\n"))], outputs=[ToolOutput(tag="out_output_folder", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_output_folder", type_hint=File()), doc=OutputDocumentation(doc="Location of main ouput folder for preprocessed GWAS\nfiles (splitted by chromosome)")), ToolOutput(tag="out_output_folder_one_file", output_type=File(optional=True), selector=InputSelector(input_to_select="in_output_folder_one_file", type_hint=File()), doc=OutputDocumentation(doc="optional location to store the preprocessing in one\ntabular file with one chromosome columns (useful to\ncompute LDSC correlation for instance)"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Jass_Preprocessing_V0_1_0().translate("wdl", allow_empty_container=True)

