from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Boolean, Directory, File

Sv2_V0_1_0 = CommandToolBuilder(tool="sv2", base_command=["sv2"], inputs=[ToolInput(tag="in_bam", input_type=Boolean(optional=True), prefix="-bam", doc=InputDocumentation(doc="...     bam file(s)")), ToolInput(tag="in_vcf", input_type=Boolean(optional=True), prefix="-vcf", doc=InputDocumentation(doc="...     vcf files(s) of SVs")), ToolInput(tag="in_bed", input_type=Boolean(optional=True), prefix="-bed", doc=InputDocumentation(doc="...     bed files(s) of SVs")), ToolInput(tag="in_snv", input_type=Boolean(optional=True), prefix="-snv", doc=InputDocumentation(doc="...     snv vcf files(s), must be bgzipped and tabixed")), ToolInput(tag="in_ped", input_type=Boolean(optional=True), prefix="-ped", doc=InputDocumentation(doc="...     ped files(s)")), ToolInput(tag="in_genome", input_type=Boolean(optional=True), prefix="-genome", doc=InputDocumentation(doc="reference genome build <hg19, hg38, mm10> [default: hg19]")), ToolInput(tag="in_pcr_free", input_type=Boolean(optional=True), prefix="-pcrfree", doc=InputDocumentation(doc="GC content normalization for pcr free sequences")), ToolInput(tag="in_bwa_mem_flagged", input_type=Boolean(optional=True), prefix="-M", doc=InputDocumentation(doc="bwa mem -M compatibility, split-reads flagged as secondary instead of supplementary")), ToolInput(tag="in_merge", input_type=Boolean(optional=True), prefix="-merge", doc=InputDocumentation(doc="merge overlapping SV breakpoints after genotyping")), ToolInput(tag="in_min_ovr", input_type=Boolean(optional=True), prefix="-min-ovr", doc=InputDocumentation(doc="minimum reciprocal overlap for merging [default: 0.8]")), ToolInput(tag="in_no_an_no", input_type=Boolean(optional=True), prefix="-no-anno", doc=InputDocumentation(doc="genotype without annotating variants")), ToolInput(tag="in_pre", input_type=Directory(optional=True), prefix="-pre", doc=InputDocumentation(doc="preprocessing output directory, skips preprocessing")), ToolInput(tag="in_feats", input_type=Directory(optional=True), prefix="-feats", doc=InputDocumentation(doc="feature extraction output directory, skips feature extraction")), ToolInput(tag="in_load_clf", input_type=Boolean(optional=True), prefix="-load-clf", doc=InputDocumentation(doc="add custom classifiers (-load-clf <clf.json>)")), ToolInput(tag="in_clf", input_type=Boolean(optional=True), prefix="-clf", doc=InputDocumentation(doc="define classifier for genotyping [default:default]")), ToolInput(tag="in_download", input_type=Boolean(optional=True), prefix="-download", doc=InputDocumentation(doc="download required data files")), ToolInput(tag="in_h_gone_nine", input_type=Boolean(optional=True), prefix="-hg19", doc=InputDocumentation(doc="hg19 fasta file")), ToolInput(tag="in_hg_three_eight", input_type=Boolean(optional=True), prefix="-hg38", doc=InputDocumentation(doc="hg38 fasta file")), ToolInput(tag="in_mm_one_zero", input_type=Boolean(optional=True), prefix="-mm10", doc=InputDocumentation(doc="mm10 fasta file")), ToolInput(tag="in_ini", input_type=Boolean(optional=True), prefix="-ini", doc=InputDocumentation(doc="configuration INI file [default: $SV2_INSTALL_PATH/config/sv2.ini]")), ToolInput(tag="in_log", input_type=Boolean(optional=True), prefix="-log", doc=InputDocumentation(doc="log file for standard error messages [default: STDOUT]")), ToolInput(tag="in_tmp_dir", input_type=Boolean(optional=True), prefix="-tmp-dir", doc=InputDocumentation(doc="directory for temporary files [default: working directory]")), ToolInput(tag="in_seed", input_type=Boolean(optional=True), prefix="-seed", doc=InputDocumentation(doc="random seed for preprocessing shuffling [default: 42]")), ToolInput(tag="in_out", input_type=Boolean(optional=True), prefix="-out", doc=InputDocumentation(doc="output prefix [default: sv2_genotypes]")), ToolInput(tag="in_o_dir", input_type=File(optional=True), prefix="-odir", doc=InputDocumentation(doc="output path, location for sv2 output directories [default: working directory]"))], outputs=[ToolOutput(tag="out_pre", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_pre", type_hint=File()), doc=OutputDocumentation(doc="preprocessing output directory, skips preprocessing")), ToolOutput(tag="out_feats", output_type=Directory(optional=True), selector=InputSelector(input_to_select="in_feats", type_hint=File()), doc=OutputDocumentation(doc="feature extraction output directory, skips feature extraction")), ToolOutput(tag="out_o_dir", output_type=File(optional=True), selector=InputSelector(input_to_select="in_o_dir", type_hint=File()), doc=OutputDocumentation(doc="output path, location for sv2 output directories [default: working directory]"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Sv2_V0_1_0().translate("wdl", allow_empty_container=True)

