from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import Int, Float, String, File, Boolean

Oncogemini_Bottleneck_V0_1_0 = CommandToolBuilder(tool="oncogemini_bottleneck", base_command=["oncogemini", "bottleneck"], inputs=[ToolInput(tag="in_mind_p", input_type=Int(optional=True), prefix="--minDP", doc=InputDocumentation(doc="Minimum depth required in all samples default is 0)")), ToolInput(tag="in_ming_q", input_type=Int(optional=True), prefix="--minGQ", doc=InputDocumentation(doc="Minimum genotype quality required in all samples (default\nis 0)")), ToolInput(tag="in_max_norm", input_type=Float(optional=True), prefix="--maxNorm", doc=InputDocumentation(doc="Specify a maximum normal sample AF to allow (default is 0)")), ToolInput(tag="in_min_slope", input_type=Float(optional=True), prefix="--minSlope", doc=InputDocumentation(doc="Minimum slope required for the AFs across samples (default\nis 0.05)")), ToolInput(tag="in_min_r", input_type=Float(optional=True), prefix="--minR", doc=InputDocumentation(doc="Minimum r correlation coefficient required for AFs\n(default is 0.5)")), ToolInput(tag="in_samples", input_type=String(optional=True), prefix="--samples", doc=InputDocumentation(doc="Rather than including all samples, a string of comma-\nseparated specified samples to use (default is 'All')")), ToolInput(tag="in_min_end", input_type=Float(optional=True), prefix="--minEnd", doc=InputDocumentation(doc="Minimum AF required of the sample representing the final\ntimepoint (default is 0)")), ToolInput(tag="in_end_diff", input_type=Float(optional=True), prefix="--endDiff", doc=InputDocumentation(doc="Minimum required AF difference between the samples\nrepresenting the first and final timepoints (default is 0)")), ToolInput(tag="in_patient", input_type=File(optional=True), prefix="--patient", doc=InputDocumentation(doc="Specify a patient to filter (should correspond to a\npatient_id in ped file)")), ToolInput(tag="in_columns", input_type=String(optional=True), prefix="--columns", doc=InputDocumentation(doc="A list of columns that you would like returned (default is\n'*', which returns every column)")), ToolInput(tag="in_filter", input_type=String(optional=True), prefix="--filter", doc=InputDocumentation(doc="Restrictions to apply to variants (SQL syntax)")), ToolInput(tag="in_purity", input_type=Boolean(optional=True), prefix="--purity", doc=InputDocumentation(doc="Using purity estimates in ped file, make corrections to AF\nto be used")), ToolInput(tag="in_somatic_only", input_type=Boolean(optional=True), prefix="--somatic_only", doc=InputDocumentation(doc="Only include variants that have been marked as somatic via\nthe set_somatic command")), ToolInput(tag="in_cancers", input_type=String(optional=True), prefix="--cancers", doc=InputDocumentation(doc="Restrict results to variants/genes associated with\nspecific cancer types by entering a comma-separated string\nof cancer type abbreviations (see documents for\nabbreviations) REQUIRES that db include\ncivic_gene_abbrevations and/or cgi_gene_abbreviations\n"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Oncogemini_Bottleneck_V0_1_0().translate("wdl", allow_empty_container=True)

