from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, String, Boolean, Int

Autoaug_Pl_V0_1_0 = CommandToolBuilder(tool="autoAug.pl", base_command=["autoAug.pl"], inputs=[ToolInput(tag="in_genome", input_type=File(optional=True), prefix="--genome", doc=InputDocumentation(doc="fasta file with DNA sequences for training")), ToolInput(tag="in_training_set", input_type=File(optional=True), prefix="--trainingset", doc=InputDocumentation(doc="genesfile contains training genes in Genbank, GFF or protein FASTA format")), ToolInput(tag="in_species", input_type=String(optional=True), prefix="--species", doc=InputDocumentation(doc="species name as used by AUGUSTUS")), ToolInput(tag="in_hints", input_type=String(optional=True), prefix="--hints", doc=InputDocumentation(doc="hints for gene predictions with AUGUSTUS")), ToolInput(tag="in_est_ali", input_type=String(optional=True), prefix="--estali", doc=InputDocumentation(doc="cDNA alignments in PSL format (as generated by BLAT and GMAP) are used to construct UTRs")), ToolInput(tag="in_pasa", input_type=Boolean(optional=True), prefix="--pasa", doc=InputDocumentation(doc="use PASA to construct a training set")), ToolInput(tag="in_cdna", input_type=File(optional=True), prefix="--cdna", doc=InputDocumentation(doc="a fasta file with cDNA sequences (ESTs, mRNA)")), ToolInput(tag="in_pasa_polya_hints", input_type=Boolean(optional=True), prefix="--pasapolyAhints", doc=InputDocumentation(doc="use PASA polyA hints as hints for the prediction")), ToolInput(tag="in_use_existing", input_type=Boolean(optional=True), prefix="--useexisting", doc=InputDocumentation(doc="use and change the present config and parameter files if they exist for 'species'")), ToolInput(tag="in_verbose", input_type=Boolean(optional=True), prefix="--verbose", doc=InputDocumentation(doc="print more status info. Cumulative option, e.g. use -v -v -v to make this script very verbose")), ToolInput(tag="in_no_utr", input_type=Boolean(optional=True), prefix="--noutr", doc=InputDocumentation(doc="do not train and predict UTRs.")), ToolInput(tag="in_working_dir", input_type=Boolean(optional=True), prefix="--workingdir", doc=InputDocumentation(doc="=/path/to/wd/           In the working directory results and temporary files are stored.\nDefault: current working directory")), ToolInput(tag="in_single_cpu", input_type=Boolean(optional=True), prefix="--singleCPU", doc=InputDocumentation(doc="run the complete program sequentially instead of parallel execution of jobs on a cluster")), ToolInput(tag="in_noninteractive", input_type=Boolean(optional=True), prefix="--noninteractive", doc=InputDocumentation(doc="bypass all manual interaction when using a SGE cluster")), ToolInput(tag="in_cname", input_type=String(optional=True), prefix="--cname", doc=InputDocumentation(doc="cluster name, only use it when 'noninteractive' default:fe")), ToolInput(tag="in_index", input_type=Int(optional=True), prefix="--index", doc=InputDocumentation(doc="step index, default:0")), ToolInput(tag="in_opt_rounds", input_type=Int(optional=True), prefix="--optrounds", doc=InputDocumentation(doc="optimization rounds - each meta parameter is optimized this often (default 1)")), ToolInput(tag="in_max_intron_len", input_type=Int(optional=True), prefix="--maxIntronLen", doc=InputDocumentation(doc="maximal length of an intron as used by PASA and BLAT, not by AUGUSTUS (default 100000)")), ToolInput(tag="in_use_gmap_for_pasa", input_type=Boolean(optional=True), prefix="--useGMAPforPASA", doc=InputDocumentation(doc="use GMAP instead of BLAT in the PASA run"))], outputs=[], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Autoaug_Pl_V0_1_0().translate("wdl", allow_empty_container=True)

