version 1.0

task Anvidisplaymetabolism {
  input {
    String? contigs_db
    Boolean? meta_genome_mode
    File? kegg_data_dir
    String? collection_name
    String? bin_id
    File? bin_ids_file
    Int? module_completion_threshold
    Float? ip_address
    Int? port_number
    File? browser_path
    Boolean? server_only
    Boolean? password_protected
    String respectively_dot
    String interactive
  }
  command <<<
    anvi_display_metabolism \
      ~{respectively_dot} \
      ~{interactive} \
      ~{if defined(contigs_db) then ("--contigs-db " +  '"' + contigs_db + '"') else ""} \
      ~{if (meta_genome_mode) then "--metagenome-mode" else ""} \
      ~{if defined(kegg_data_dir) then ("--kegg-data-dir " +  '"' + kegg_data_dir + '"') else ""} \
      ~{if defined(collection_name) then ("--collection-name " +  '"' + collection_name + '"') else ""} \
      ~{if defined(bin_id) then ("--bin-id " +  '"' + bin_id + '"') else ""} \
      ~{if defined(bin_ids_file) then ("--bin-ids-file " +  '"' + bin_ids_file + '"') else ""} \
      ~{if defined(module_completion_threshold) then ("--module-completion-threshold " +  '"' + module_completion_threshold + '"') else ""} \
      ~{if defined(ip_address) then ("--ip-address " +  '"' + ip_address + '"') else ""} \
      ~{if defined(port_number) then ("--port-number " +  '"' + port_number + '"') else ""} \
      ~{if defined(browser_path) then ("--browser-path " +  '"' + browser_path + '"') else ""} \
      ~{if (server_only) then "--server-only" else ""} \
      ~{if (password_protected) then "--password-protected" else ""}
  >>>
  runtime {
    docker: "quay.io/biocontainers/anvio-minimal:7--py_0"
  }
  parameter_meta {
    contigs_db: "Anvi'o contigs database generated by 'anvi-gen-\\ncontigs-database' (default: None)"
    meta_genome_mode: "Treat a given contigs database as a metagenome rather\\nthan treating it as a single genome. (default: False)"
    kegg_data_dir: "The directory path for your KEGG setup, which will\\ninclude things like KOfam profiles and KEGG MODULE\\ndata. Anvi'o will try to use the default path if you\\ndo not specify anything. (default: None)"
    collection_name: "Collection name. (default: None)"
    bin_id: "Bin name you are interested in. (default: None)"
    bin_ids_file: "Text file for bins (each line should be a unique bin\\nid). (default: None)"
    module_completion_threshold: "This threshold defines the point at which we consider\\na KEGG module to be 'complete' or 'present' in a given\\ngenome or bin. It is the fraction of steps that must\\nbe complete in in order for the entire module to be\\nmarked complete. The default is 0.75."
    ip_address: "IP address for the HTTP server. The default ip address\\n(0.0.0.0) should work just fine for most."
    port_number: "Port number to use for anvi'o services. If nothing is\\ndeclared, anvi'o will try to find a suitable port\\nnumber, starting from the default port number, 8080.\\n(default: None)"
    browser_path: "By default, anvi'o will use your default browser to\\nlaunch the interactive interface. If you would like to\\nuse something else than your system default, you can\\nprovide a full path for an alternative browser using\\nthis parameter, and hope for the best. For instance we\\nare using this parameter to call Google's experimental\\nbrowser, Canary, which performs better with demanding\\nvisualizations. (default: None)"
    server_only: "The default behavior is to start the local server, and\\nfire up a browser that connects to the server. If you\\nhave other plans, and want to start the server without\\ncalling the browser, this is the flag you need.\\n(default: False)"
    password_protected: "If this flag is set, command line tool will ask you to\\nenter a password and interactive interface will be\\nonly accessible after entering same password. This\\noption is recommended for shared machines like\\nclusters or shared networks where computers are not\\nisolated. (default: False)"
    respectively_dot: "-p PROFILE_DB, --profile-db PROFILE_DB\\nAnvi'o profile database (default: None)"
    interactive: "üç∫ More on `anvi-display-metabolism`:"
  }
  output {
    File out_stdout = stdout()
  }
}