from datetime import datetime
from typing import List, Optional, Dict, Any

from janis_core import *
from janis_core.types.common_data_types import File, Int, String

Ispcr_V0_1_0 = CommandToolBuilder(tool="isPcr", base_command=["isPcr"], inputs=[ToolInput(tag="in_ooc", input_type=File(optional=True), prefix="-ooc", doc=InputDocumentation(doc="Use overused tile file N.ooc.  N should correspond to\nthe tileSize")), ToolInput(tag="in_tile_size", input_type=Int(optional=True), prefix="-tileSize", doc=InputDocumentation(doc="the size of match that triggers an alignment.\nDefault is 11 .")), ToolInput(tag="in_step_size", input_type=Int(optional=True), prefix="-stepSize", doc=InputDocumentation(doc="spacing between tiles. Default is 5.")), ToolInput(tag="in_maxsize", input_type=Int(optional=True), prefix="-maxSize", doc=InputDocumentation(doc="- Maximum size of PCR product (default 4000)")), ToolInput(tag="in_minsize", input_type=Int(optional=True), prefix="-minSize", doc=InputDocumentation(doc="- Minimum size of PCR product (default 0)")), ToolInput(tag="in_min_perfect", input_type=Int(optional=True), prefix="-minPerfect", doc=InputDocumentation(doc="- Minimum size of perfect match at 3' end of primer (default 15)")), ToolInput(tag="in_min_good", input_type=Int(optional=True), prefix="-minGood", doc=InputDocumentation(doc="- Minimum size where there must be 2 matches for each mismatch (default 15)")), ToolInput(tag="in_mask", input_type=File(optional=True), prefix="-mask", doc=InputDocumentation(doc="Mask out repeats.  Alignments won't be started in masked region\nbut may extend through it in nucleotide searches.  Masked areas\nare ignored entirely in protein or translated searches. Types are\nlower - mask out lower cased sequence\nupper - mask out upper cased sequence\nout   - mask according to database.out RepeatMasker .out file\nfile.out - mask database according to RepeatMasker file.out")), ToolInput(tag="in_make_ooc", input_type=File(optional=True), prefix="-makeOoc", doc=InputDocumentation(doc="Make overused tile file. Database needs to be complete genome.")), ToolInput(tag="in_rep_match", input_type=Int(optional=True), prefix="-repMatch", doc=InputDocumentation(doc="sets the number of repetitions of a tile allowed before\nit is marked as overused.  Typically this is 256 for tileSize\n12, 1024 for tile size 11, 4096 for tile size 10.\nDefault is 1024.  Only comes into play with makeOoc")), ToolInput(tag="in_flip_reverse", input_type=String(optional=True), prefix="-flipReverse", doc=InputDocumentation(doc="complement reverse (second) primer before using")), ToolInput(tag="in_out", input_type=String(optional=True), prefix="-out", doc=InputDocumentation(doc="- Output format.  Either\nfa - fasta with position, primers in header (default)\nbed - tab delimited format. Fields: chrom/start/end/name/score/strand\npsl - blat format.\n")), ToolInput(tag="in_database", input_type=String(), position=0, doc=InputDocumentation(doc="")), ToolInput(tag="in_query", input_type=String(), position=1, doc=InputDocumentation(doc=""))], outputs=[ToolOutput(tag="out_mask", output_type=File(optional=True), selector=InputSelector(input_to_select="in_mask", type_hint=File()), doc=OutputDocumentation(doc="Mask out repeats.  Alignments won't be started in masked region\nbut may extend through it in nucleotide searches.  Masked areas\nare ignored entirely in protein or translated searches. Types are\nlower - mask out lower cased sequence\nupper - mask out upper cased sequence\nout   - mask according to database.out RepeatMasker .out file\nfile.out - mask database according to RepeatMasker file.out"))], container=None, version="v0.1.0")


if __name__ == "__main__":
    # or "cwl"
    Ispcr_V0_1_0().translate("wdl", allow_empty_container=True)

